{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import operator\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "def wrap_ttest(df, label_column, comparison_columns=None, alpha=.05, return_all=False, correction_method='bonferroni', mincount=3, pval_return_corrected=True):\n",
    "    try:\n",
    "        '''Verify precondition that label column exists and has exactly 2 unique values'''\n",
    "        label_values = df[label_column].unique()\n",
    "        if len(label_values) != 2:\n",
    "            print(\"Incorrectly Formatted Dataframe! Label column must have exactly 2 unique values.\")\n",
    "            return None\n",
    "        \n",
    "        '''Partition dataframe into two sets, one for each of the two unique values from the label column'''\n",
    "        partition1 = df.loc[df[label_column] == label_values[0]]\n",
    "        partition2 = df.loc[df[label_column] == label_values[1]]\n",
    "\n",
    "        '''If no comparison columns specified, use all columns except the specified labed column'''\n",
    "        if not comparison_columns:\n",
    "            comparison_columns = list(df.columns)\n",
    "            comparison_columns.remove(label_column)\n",
    "\n",
    "        '''Determine the number of real valued columns on which we will do t-tests'''\n",
    "        number_of_comparisons = len(comparison_columns)\n",
    "\n",
    "        '''Store comparisons and p-values in two arrays'''\n",
    "        comparisons = []\n",
    "        pvals = []\n",
    "        \n",
    "        '''Loop through each comparison column, perform the t-test, and record the p-val'''\n",
    "        for column in comparison_columns:\n",
    "            if len(partition1[column].dropna(axis=0)) <= mincount:\n",
    "                continue\n",
    "            elif len(partition2[column].dropna(axis=0)) <= mincount:\n",
    "                continue\n",
    "            else:\n",
    "                stat, pval = scipy.stats.ttest_ind(partition1[column].dropna(axis=0), partition2[column].dropna(axis=0))\n",
    "                comparisons.append(column)\n",
    "                pvals.append(pval)\n",
    "            \n",
    "        '''Correct for multiple testing to determine if each comparison meets the new cutoff'''\n",
    "        results = statsmodels.stats.multitest.multipletests(pvals=pvals, alpha=alpha, method=correction_method)\n",
    "        reject = results[0]\n",
    "\n",
    "        '''Format results in a pandas dataframe'''\n",
    "        results_df = pd.DataFrame(columns=['Comparison','P_Value'])\n",
    "\n",
    "        '''If return all, add all comparisons and p-values to dataframe'''\n",
    "        if return_all:\n",
    "            results_df['Comparison'] = comparisons\n",
    "            results_df['P_Value'] = pvals\n",
    "\n",
    "            '''Else only add significant comparisons'''\n",
    "        else:\n",
    "            for i in range(0, len(reject)):\n",
    "                if reject[i]:\n",
    "                    if pval_return_corrected:\n",
    "                        results_df = results_df.append({'Comparison':comparisons[i],'P_Value':results[1][i]}, ignore_index=True)\n",
    "                    else:\n",
    "                        results_df = results_df.append({'Comparison':comparisons[i],'P_Value':pvals[i]}, ignore_index=True)\n",
    "\n",
    "\n",
    "        '''Sort dataframe by ascending p-value'''\n",
    "        results_df = results_df.sort_values(by='P_Value', ascending=True)\n",
    "        results_df = results_df.reset_index(drop=True)\n",
    "\n",
    "        '''If results df is not empty, return it, else return None'''\n",
    "        if len(results_df) > 0:\n",
    "            return results_df\n",
    "        else:\n",
    "            print(\"Empty\")\n",
    "            return results[1]\n",
    "\n",
    "\n",
    "    except:\n",
    "        print(\"Incorrectly Formatted Dataframe!\")\n",
    "        return results[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Your version of cptac (0.8.0) is out-of-date. Latest is 0.8.1. Please run 'pip install --upgrade cptac' to update it. (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 10)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load cancer data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "en = cptac.Endometrial()\n",
    "br = cptac.Brca()\n",
    "cl = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PIK3CA'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter out hotspot mutations\n",
    "Hotspots are:\n",
    "E542K\n",
    "E545K\n",
    "H1047R\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the mutation type, and proteomics for PIK3CA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 48 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    }
   ],
   "source": [
    "proteomics = en.get_proteomics(tissue_type=\"tumor\")\n",
    "mut_type = en.get_genotype_all_vars(\"PIK3CA\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join mutation type and proteomics together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "joined = mut_type.join(proteomics)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02290</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.6840</td>\n",
       "      <td>0.9840</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.36300</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.7370</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.07570</td>\n",
       "      <td>-0.4730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.01050</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.32000</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>0.2430</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.07140</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.1560</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.03010</td>\n",
       "      <td>-0.4670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>Amplification</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.374</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.5370</td>\n",
       "      <td>0.3110</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.22000</td>\n",
       "      <td>-0.5620</td>\n",
       "      <td>0.9370</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>0.0513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E726K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-0.7120</td>\n",
       "      <td>0.0462</td>\n",
       "      <td>-0.0471</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>1.3500</td>\n",
       "      <td>1.2100</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.06990</td>\n",
       "      <td>-0.4010</td>\n",
       "      <td>0.55700</td>\n",
       "      <td>0.8270</td>\n",
       "      <td>0.3480</td>\n",
       "      <td>-0.187</td>\n",
       "      <td>-0.10700</td>\n",
       "      <td>-0.830</td>\n",
       "      <td>0.06200</td>\n",
       "      <td>-0.5280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>Amplification</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.280</td>\n",
       "      <td>-0.7360</td>\n",
       "      <td>-0.5520</td>\n",
       "      <td>0.1140</td>\n",
       "      <td>0.2730</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.8500</td>\n",
       "      <td>...</td>\n",
       "      <td>0.13100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00364</td>\n",
       "      <td>0.7630</td>\n",
       "      <td>0.0784</td>\n",
       "      <td>-0.203</td>\n",
       "      <td>0.01320</td>\n",
       "      <td>-1.200</td>\n",
       "      <td>-0.35700</td>\n",
       "      <td>0.0299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.290</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>3.1700</td>\n",
       "      <td>-0.9070</td>\n",
       "      <td>0.0317</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17900</td>\n",
       "      <td>-0.5210</td>\n",
       "      <td>0.05230</td>\n",
       "      <td>0.5150</td>\n",
       "      <td>0.5850</td>\n",
       "      <td>0.138</td>\n",
       "      <td>-0.19000</td>\n",
       "      <td>-0.966</td>\n",
       "      <td>-0.00627</td>\n",
       "      <td>-0.2490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.266</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.4700</td>\n",
       "      <td>0.3980</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.14000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.01220</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>0.5530</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.06420</td>\n",
       "      <td>-0.437</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.4980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.K111E</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>0.692</td>\n",
       "      <td>0.5890</td>\n",
       "      <td>0.5870</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0835</td>\n",
       "      <td>0.1720</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.02250</td>\n",
       "      <td>0.2980</td>\n",
       "      <td>-0.12400</td>\n",
       "      <td>0.1780</td>\n",
       "      <td>0.1470</td>\n",
       "      <td>-0.114</td>\n",
       "      <td>0.13700</td>\n",
       "      <td>1.240</td>\n",
       "      <td>0.00682</td>\n",
       "      <td>-0.0233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 11002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation       Location  Mutation_Status   A1BG     A2M  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00006   Missense_Mutation        p.E545K  Single_mutation -1.180 -0.8630   \n",
       "C3L-00008      Wildtype_Tumor    No_mutation   Wildtype_Tumor -0.685 -1.0700   \n",
       "C3L-00032   Missense_Mutation        p.E545K  Single_mutation -0.528 -1.3200   \n",
       "C3L-00090      Wildtype_Tumor    No_mutation   Wildtype_Tumor -1.670 -1.1900   \n",
       "C3L-00098       Amplification  Amplification  Single_mutation -0.374 -0.0206   \n",
       "...                       ...            ...              ...    ...     ...   \n",
       "C3N-01520   Missense_Mutation        p.E726K  Single_mutation -1.070 -0.7120   \n",
       "C3N-01521       Amplification  Amplification  Single_mutation -1.280 -0.7360   \n",
       "C3N-01537      Wildtype_Tumor    No_mutation   Wildtype_Tumor -0.290 -0.3200   \n",
       "C3N-01802      Wildtype_Tumor    No_mutation   Wildtype_Tumor  0.266  1.3900   \n",
       "C3N-01825   Missense_Mutation        p.K111E  Single_mutation  0.692  0.5890   \n",
       "\n",
       "Name         A2ML1  A4GALT    AAAS    AACS   AADAT  ...   ZSWIM8  ZSWIM9  \\\n",
       "Patient_ID                                          ...                    \n",
       "C3L-00006  -0.8020  0.2220  0.2560  0.6650  1.2800  ... -0.08770     NaN   \n",
       "C3L-00008  -0.6840  0.9840  0.1350  0.3340  1.3000  ... -0.03560     NaN   \n",
       "C3L-00032   0.4350     NaN -0.2400  1.0400 -0.0213  ...  0.00112 -0.1450   \n",
       "C3L-00090  -0.4430  0.2430 -0.0993  0.7570  0.7400  ...  0.07250 -0.0552   \n",
       "C3L-00098  -0.5370  0.3110  0.3750  0.0131 -1.1000  ... -0.17600     NaN   \n",
       "...            ...     ...     ...     ...     ...  ...      ...     ...   \n",
       "C3N-01520   0.0462 -0.0471  0.2270  1.3500  1.2100  ... -0.06990 -0.4010   \n",
       "C3N-01521  -0.5520  0.1140  0.2730  1.0700  0.8500  ...  0.13100     NaN   \n",
       "C3N-01537   3.1700 -0.9070  0.0317 -0.0425     NaN  ... -0.17900 -0.5210   \n",
       "C3N-01802  -0.0655  0.4700  0.3980 -0.1340  0.4610  ... -0.14000     NaN   \n",
       "C3N-01825   0.5870     NaN -0.0835  0.1720 -0.2150  ... -0.02250  0.2980   \n",
       "\n",
       "Name           ZW10  ZWILCH   ZWINT   ZXDC   ZYG11B    ZYX    ZZEF1    ZZZ3  \n",
       "Patient_ID                                                                   \n",
       "C3L-00006   0.02290  0.1090     NaN -0.332 -0.43300 -1.020 -0.12300 -0.0859  \n",
       "C3L-00008   0.36300  1.0700  0.7370 -0.564 -0.00461 -1.130 -0.07570 -0.4730  \n",
       "C3L-00032   0.01050 -0.1160     NaN  0.151 -0.07400 -0.540  0.32000 -0.4190  \n",
       "C3L-00090  -0.07140  0.0933  0.1560 -0.398 -0.07520 -0.797 -0.03010 -0.4670  \n",
       "C3L-00098  -1.22000 -0.5620  0.9370 -0.646  0.20700 -1.850 -0.17600  0.0513  \n",
       "...             ...     ...     ...    ...      ...    ...      ...     ...  \n",
       "C3N-01520   0.55700  0.8270  0.3480 -0.187 -0.10700 -0.830  0.06200 -0.5280  \n",
       "C3N-01521  -0.00364  0.7630  0.0784 -0.203  0.01320 -1.200 -0.35700  0.0299  \n",
       "C3N-01537   0.05230  0.5150  0.5850  0.138 -0.19000 -0.966 -0.00627 -0.2490  \n",
       "C3N-01802  -0.01220  0.2500  0.5530  0.387  0.06420 -0.437  0.10400 -0.4980  \n",
       "C3N-01825  -0.12400  0.1780  0.1470 -0.114  0.13700  1.240  0.00682 -0.0233  \n",
       "\n",
       "[95 rows x 11002 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "joined"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Filter for missense and wildtyoe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "missense = joined.loc[joined.Mutation == \"Missense_Mutation\"]\n",
    "\n",
    "endo_both = missense.append(wildtype)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Drop unncecesary columns\n",
    "The resulting dataframe is what we will be working with from this point forward. It contains the mutations type and the proteomics for each gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.4120</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.4330</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.4190</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.0740</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.32000</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00139</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.467</td>\n",
       "      <td>0.3700</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4340</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>-0.1750</td>\n",
       "      <td>0.1810</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.67500</td>\n",
       "      <td>0.2390</td>\n",
       "      <td>0.1400</td>\n",
       "      <td>1.070</td>\n",
       "      <td>0.6070</td>\n",
       "      <td>0.486</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>-0.632</td>\n",
       "      <td>-0.20300</td>\n",
       "      <td>-0.0685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00143</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.120</td>\n",
       "      <td>-1.3100</td>\n",
       "      <td>0.9120</td>\n",
       "      <td>0.418</td>\n",
       "      <td>-0.0768</td>\n",
       "      <td>0.8460</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3110</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00212</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-1.270</td>\n",
       "      <td>-1.2700</td>\n",
       "      <td>-0.222</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>-0.620</td>\n",
       "      <td>0.36300</td>\n",
       "      <td>-0.4630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00156</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.232</td>\n",
       "      <td>-0.0910</td>\n",
       "      <td>0.6180</td>\n",
       "      <td>1.630</td>\n",
       "      <td>-0.2500</td>\n",
       "      <td>1.3300</td>\n",
       "      <td>-1.7200</td>\n",
       "      <td>-0.8890</td>\n",
       "      <td>0.6330</td>\n",
       "      <td>...</td>\n",
       "      <td>0.06520</td>\n",
       "      <td>-0.2360</td>\n",
       "      <td>0.5010</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>-0.428</td>\n",
       "      <td>-0.2120</td>\n",
       "      <td>-1.340</td>\n",
       "      <td>0.31700</td>\n",
       "      <td>-0.9430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01211</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.636</td>\n",
       "      <td>-0.3710</td>\n",
       "      <td>0.9730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6370</td>\n",
       "      <td>-0.1660</td>\n",
       "      <td>-0.1220</td>\n",
       "      <td>0.2320</td>\n",
       "      <td>0.3790</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.41600</td>\n",
       "      <td>-0.7510</td>\n",
       "      <td>0.1790</td>\n",
       "      <td>0.658</td>\n",
       "      <td>0.4700</td>\n",
       "      <td>0.270</td>\n",
       "      <td>-0.1480</td>\n",
       "      <td>-0.139</td>\n",
       "      <td>-0.02990</td>\n",
       "      <td>0.1310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01217</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.743</td>\n",
       "      <td>-1.3700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1930</td>\n",
       "      <td>0.6940</td>\n",
       "      <td>0.3380</td>\n",
       "      <td>0.0989</td>\n",
       "      <td>-0.0137</td>\n",
       "      <td>...</td>\n",
       "      <td>0.18100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1960</td>\n",
       "      <td>0.274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.501</td>\n",
       "      <td>-0.0468</td>\n",
       "      <td>0.653</td>\n",
       "      <td>-0.04640</td>\n",
       "      <td>-0.1600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01219</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.295</td>\n",
       "      <td>-0.0589</td>\n",
       "      <td>0.4900</td>\n",
       "      <td>-0.109</td>\n",
       "      <td>-0.0495</td>\n",
       "      <td>0.0374</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>-0.3210</td>\n",
       "      <td>0.1830</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.29000</td>\n",
       "      <td>0.0457</td>\n",
       "      <td>0.2460</td>\n",
       "      <td>0.589</td>\n",
       "      <td>0.3930</td>\n",
       "      <td>-0.199</td>\n",
       "      <td>-0.1590</td>\n",
       "      <td>-1.090</td>\n",
       "      <td>0.06270</td>\n",
       "      <td>-0.3790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.290</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>3.1700</td>\n",
       "      <td>-0.907</td>\n",
       "      <td>0.0317</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2730</td>\n",
       "      <td>-0.0747</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17900</td>\n",
       "      <td>-0.5210</td>\n",
       "      <td>0.0523</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.5850</td>\n",
       "      <td>0.138</td>\n",
       "      <td>-0.1900</td>\n",
       "      <td>-0.966</td>\n",
       "      <td>-0.00627</td>\n",
       "      <td>-0.2490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.266</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.470</td>\n",
       "      <td>0.3980</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.14000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0122</td>\n",
       "      <td>0.250</td>\n",
       "      <td>0.5530</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.0642</td>\n",
       "      <td>-0.437</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.4980</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation   A1BG     A2M   A2ML1  A4GALT    AAAS    AACS  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00006   Missense_Mutation -1.180 -0.8630 -0.8020   0.222  0.2560  0.6650   \n",
       "C3L-00032   Missense_Mutation -0.528 -1.3200  0.4350     NaN -0.2400  1.0400   \n",
       "C3L-00139   Missense_Mutation -0.467  0.3700 -0.3390     NaN  0.4340  0.0358   \n",
       "C3L-00143   Missense_Mutation -1.120 -1.3100  0.9120   0.418 -0.0768  0.8460   \n",
       "C3L-00156   Missense_Mutation -0.232 -0.0910  0.6180   1.630 -0.2500  1.3300   \n",
       "...                       ...    ...     ...     ...     ...     ...     ...   \n",
       "C3N-01211      Wildtype_Tumor -0.636 -0.3710  0.9730     NaN  0.6370 -0.1660   \n",
       "C3N-01217      Wildtype_Tumor -0.743 -1.3700     NaN     NaN -0.1930  0.6940   \n",
       "C3N-01219      Wildtype_Tumor -0.295 -0.0589  0.4900  -0.109 -0.0495  0.0374   \n",
       "C3N-01537      Wildtype_Tumor -0.290 -0.3200  3.1700  -0.907  0.0317 -0.0425   \n",
       "C3N-01802      Wildtype_Tumor  0.266  1.3900 -0.0655   0.470  0.3980 -0.1340   \n",
       "\n",
       "Name         AADAT   AAED1   AAGAB  ...   ZSWIM8  ZSWIM9    ZW10  ZWILCH  \\\n",
       "Patient_ID                          ...                                    \n",
       "C3L-00006   1.2800 -0.3390  0.4120  ... -0.08770     NaN  0.0229   0.109   \n",
       "C3L-00032  -0.0213 -0.0479  0.4190  ...  0.00112 -0.1450  0.0105  -0.116   \n",
       "C3L-00139  -0.1750  0.1810  0.1160  ... -0.67500  0.2390  0.1400   1.070   \n",
       "C3L-00143  -0.1210     NaN -0.3110  ... -0.00212     NaN -1.1900  -1.270   \n",
       "C3L-00156  -1.7200 -0.8890  0.6330  ...  0.06520 -0.2360  0.5010   0.352   \n",
       "...            ...     ...     ...  ...      ...     ...     ...     ...   \n",
       "C3N-01211  -0.1220  0.2320  0.3790  ... -0.41600 -0.7510  0.1790   0.658   \n",
       "C3N-01217   0.3380  0.0989 -0.0137  ...  0.18100     NaN  0.1960   0.274   \n",
       "C3N-01219   0.1020 -0.3210  0.1830  ... -0.29000  0.0457  0.2460   0.589   \n",
       "C3N-01537      NaN -0.2730 -0.0747  ... -0.17900 -0.5210  0.0523   0.515   \n",
       "C3N-01802   0.4610  1.0400  0.3630  ... -0.14000     NaN -0.0122   0.250   \n",
       "\n",
       "Name         ZWINT   ZXDC  ZYG11B    ZYX    ZZEF1    ZZZ3  \n",
       "Patient_ID                                                 \n",
       "C3L-00006      NaN -0.332 -0.4330 -1.020 -0.12300 -0.0859  \n",
       "C3L-00032      NaN  0.151 -0.0740 -0.540  0.32000 -0.4190  \n",
       "C3L-00139   0.6070  0.486  0.1690 -0.632 -0.20300 -0.0685  \n",
       "C3L-00143  -1.2700 -0.222 -0.3200 -0.620  0.36300 -0.4630  \n",
       "C3L-00156   0.0839 -0.428 -0.2120 -1.340  0.31700 -0.9430  \n",
       "...            ...    ...     ...    ...      ...     ...  \n",
       "C3N-01211   0.4700  0.270 -0.1480 -0.139 -0.02990  0.1310  \n",
       "C3N-01217      NaN -0.501 -0.0468  0.653 -0.04640 -0.1600  \n",
       "C3N-01219   0.3930 -0.199 -0.1590 -1.090  0.06270 -0.3790  \n",
       "C3N-01537   0.5850  0.138 -0.1900 -0.966 -0.00627 -0.2490  \n",
       "C3N-01802   0.5530  0.387  0.0642 -0.437  0.10400 -0.4980  \n",
       "\n",
       "[84 rows x 11000 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_prot_and_mutations = endo_both.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "endo_prot_and_mutations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run T-test and difference of Median for each cancer type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the median proteomic expression among all samples, for each gene.\n",
    "This is done for missense mutations and wildtype separately\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "missense = endo_prot_and_mutations[endo_prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = endo_prot_and_mutations[endo_prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #get all wildtype mutations\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the medain difference between missense and wildtype for each gene.\n",
    "This is done by (for each gene) subtracting the median of the missense mutations from the median of the wildtype.\n",
    "This means that if the difference is positive the proteomics of the wildtype have higher expression than the missense mutation. If the difference is negative then the missense mutation has a higher expression than the wildtype."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.07100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.10300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.47135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>0.22545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>0.07410</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median\n",
       "A1BG                 0.07100\n",
       "A2M                  0.10300\n",
       "A2ML1                0.47135\n",
       "A4GALT               0.22545\n",
       "AAAS                 0.07410"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_d = {}\n",
    "\n",
    "for gene in endo_prot_and_mutations:\n",
    "    if gene == \"Mutation\": continue\n",
    "    dif_endo =  missense_med[gene] - wt_med[gene]\n",
    "    endo_d[gene] = dif_endo\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(endo_d, orient='index', columns=['Difference_In_Median'])\n",
    "median_diff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Do a t-test for every gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.9999312004874984"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = list(endo_prot_and_mutations.columns[1:])\n",
    "n = wrap_ttest(endo_prot_and_mutations, 'Mutation', correction_method=\"fdr_bh\", mincount=15)\n",
    "min(n)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join difference in median and t-test p value into the same dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Step 4 join median and p value together\n",
    "# endo_df = median_diff.join(t_test)\n",
    "# endo_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BRCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "no_duplicates = proteomics[proteomics.columns[~fil]]\n",
    "# no_duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 83 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    }
   ],
   "source": [
    "#DEALING WITH DUPLICATES\n",
    "#flatten the multiindex\n",
    "duplicates = cptac.utils.reduce_multiindex(duplicates, flatten=True)\n",
    "\n",
    "# We need to figure which of databaseID we want to use, for each protein\n",
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "duplicate_gene_names = list(set(duplicates.columns.get_level_values('Name'))) #get a list of genes names that are duplicated\n",
    "duplicates = cptac.utils.reduce_multiindex(duplicates, flatten=True)\n",
    "duplicates\n",
    "\n",
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "\n",
    "mut_type = br.get_genotype_all_vars(\"PIK3CA\")\n",
    "joined = mut_type.join(duplicates)\n",
    "\n",
    "#select samples containing hotspot mutations\n",
    "br_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "                    joined.Location.str.contains('E545K') |\n",
    "                    joined.Location.str.contains('H1047R')]\n",
    "\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "missense = joined.loc[joined.Mutation == \"Missense_Mutation\"]\n",
    "brca_both = missense.append(wildtype)\n",
    "\n",
    "#drop unncessary columns\n",
    "test_prot_and_mutations = brca_both.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "\n",
    "#step 3 do a t test for every gene and report the p-value\n",
    "cols = list(test_prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(test_prot_and_mutations, 'Mutation', cols, correction_method= \"fdr_bh\", return_all=True)\n",
    "\n",
    "# t_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for each duplicated gene name select the one with the min p val and append[ to no_duplicat df\n",
    "selected_dup = []\n",
    "for name in duplicate_gene_names:\n",
    "    gene_duplicate = t_test[t_test.Comparison.str.contains(name)]\n",
    "    #select the one with the min p value for t test\n",
    "    min_row = gene_duplicate.loc[gene_duplicate.P_Value == gene_duplicate.P_Value.min()]\n",
    "    name = min_row['Comparison'].values\n",
    "    selected_dup.append(name)\n",
    "\n",
    "selected= []\n",
    "\n",
    "\n",
    "no_dup = cptac.utils.reduce_multiindex(no_duplicates, flatten=True)\n",
    "no_dup.columns\n",
    "for name in no_dup.columns:\n",
    "    selected.append(name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 83 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSCAN31</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CPT001846</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.3964</td>\n",
       "      <td>1.3302</td>\n",
       "      <td>-5.0948</td>\n",
       "      <td>0.7674</td>\n",
       "      <td>-1.6845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1022</td>\n",
       "      <td>-0.5814</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.7592</td>\n",
       "      <td>0.4711</td>\n",
       "      <td>0.6018</td>\n",
       "      <td>0.2062</td>\n",
       "      <td>-0.2137</td>\n",
       "      <td>-2.1219</td>\n",
       "      <td>0.0860</td>\n",
       "      <td>2.5814</td>\n",
       "      <td>-0.2852</td>\n",
       "      <td>-0.1074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X03BR010</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.5386</td>\n",
       "      <td>0.0663</td>\n",
       "      <td>-3.7769</td>\n",
       "      <td>-0.0033</td>\n",
       "      <td>1.0579</td>\n",
       "      <td>0.8396</td>\n",
       "      <td>-0.5769</td>\n",
       "      <td>2.5209</td>\n",
       "      <td>0.3357</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1708</td>\n",
       "      <td>-0.0591</td>\n",
       "      <td>-1.7682</td>\n",
       "      <td>-1.3061</td>\n",
       "      <td>0.1523</td>\n",
       "      <td>-0.6977</td>\n",
       "      <td>-1.3316</td>\n",
       "      <td>0.8303</td>\n",
       "      <td>-0.3795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X05BR004</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.7901</td>\n",
       "      <td>-0.3931</td>\n",
       "      <td>-3.9995</td>\n",
       "      <td>0.1256</td>\n",
       "      <td>-1.8863</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3140</td>\n",
       "      <td>-0.0507</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.3219</td>\n",
       "      <td>-0.5268</td>\n",
       "      <td>-0.0487</td>\n",
       "      <td>-0.4863</td>\n",
       "      <td>-1.5257</td>\n",
       "      <td>0.5247</td>\n",
       "      <td>-0.7801</td>\n",
       "      <td>-0.6180</td>\n",
       "      <td>-0.6423</td>\n",
       "      <td>0.1904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X05BR009</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>3.6316</td>\n",
       "      <td>3.5662</td>\n",
       "      <td>-2.4971</td>\n",
       "      <td>-0.9388</td>\n",
       "      <td>-2.3314</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3822</td>\n",
       "      <td>-0.1004</td>\n",
       "      <td>0.0427</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.1106</td>\n",
       "      <td>-0.8816</td>\n",
       "      <td>-0.3294</td>\n",
       "      <td>-0.7691</td>\n",
       "      <td>-1.5278</td>\n",
       "      <td>0.1102</td>\n",
       "      <td>-0.5196</td>\n",
       "      <td>1.3474</td>\n",
       "      <td>-1.0431</td>\n",
       "      <td>0.1388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X05BR043</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.0922</td>\n",
       "      <td>0.8715</td>\n",
       "      <td>-2.5619</td>\n",
       "      <td>-0.0461</td>\n",
       "      <td>-3.3529</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1559</td>\n",
       "      <td>-0.0154</td>\n",
       "      <td>-0.0768</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2157</td>\n",
       "      <td>-0.0073</td>\n",
       "      <td>-0.1810</td>\n",
       "      <td>1.1883</td>\n",
       "      <td>1.3825</td>\n",
       "      <td>0.6651</td>\n",
       "      <td>-0.3629</td>\n",
       "      <td>1.2292</td>\n",
       "      <td>-0.7839</td>\n",
       "      <td>-0.2382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X15BR003</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>2.4710</td>\n",
       "      <td>1.0447</td>\n",
       "      <td>1.6000</td>\n",
       "      <td>-0.2389</td>\n",
       "      <td>-0.2619</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.2958</td>\n",
       "      <td>0.1298</td>\n",
       "      <td>0.3695</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3656</td>\n",
       "      <td>-1.0569</td>\n",
       "      <td>0.8718</td>\n",
       "      <td>-0.4324</td>\n",
       "      <td>-0.7942</td>\n",
       "      <td>-2.4072</td>\n",
       "      <td>1.2313</td>\n",
       "      <td>0.6114</td>\n",
       "      <td>-0.6675</td>\n",
       "      <td>-0.8818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X18BR004</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.4652</td>\n",
       "      <td>-0.2157</td>\n",
       "      <td>-3.6000</td>\n",
       "      <td>-0.3496</td>\n",
       "      <td>-0.9725</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2205</td>\n",
       "      <td>-0.7924</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.0780</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.3147</td>\n",
       "      <td>0.0305</td>\n",
       "      <td>-0.2289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2189</td>\n",
       "      <td>-0.3281</td>\n",
       "      <td>0.4254</td>\n",
       "      <td>-0.1496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X18BR006</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.0999</td>\n",
       "      <td>0.7852</td>\n",
       "      <td>-8.1266</td>\n",
       "      <td>-0.5701</td>\n",
       "      <td>-2.1021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5732</td>\n",
       "      <td>0.4928</td>\n",
       "      <td>-0.2838</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3722</td>\n",
       "      <td>0.0816</td>\n",
       "      <td>0.0086</td>\n",
       "      <td>0.3131</td>\n",
       "      <td>-0.2260</td>\n",
       "      <td>-0.6128</td>\n",
       "      <td>0.7913</td>\n",
       "      <td>0.8553</td>\n",
       "      <td>-0.0371</td>\n",
       "      <td>-0.3874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X18BR017</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.8934</td>\n",
       "      <td>1.6886</td>\n",
       "      <td>-6.0217</td>\n",
       "      <td>-0.0481</td>\n",
       "      <td>0.1299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.5536</td>\n",
       "      <td>0.5748</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2404</td>\n",
       "      <td>0.6150</td>\n",
       "      <td>0.2906</td>\n",
       "      <td>0.2303</td>\n",
       "      <td>-0.3553</td>\n",
       "      <td>0.4313</td>\n",
       "      <td>1.0197</td>\n",
       "      <td>1.0025</td>\n",
       "      <td>1.8005</td>\n",
       "      <td>-0.5533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X20BR001</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>1.0545</td>\n",
       "      <td>1.0394</td>\n",
       "      <td>-3.5713</td>\n",
       "      <td>-0.7841</td>\n",
       "      <td>-1.1205</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2457</td>\n",
       "      <td>-0.5970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4385</td>\n",
       "      <td>0.1229</td>\n",
       "      <td>0.1229</td>\n",
       "      <td>-1.2017</td>\n",
       "      <td>-1.4247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7539</td>\n",
       "      <td>-0.2229</td>\n",
       "      <td>0.1569</td>\n",
       "      <td>0.2495</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86 rows × 9451 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation    A1BG     A2M   A2ML1    AAAS    AACS   AADAT  \\\n",
       "Patient_ID                                                                      \n",
       "CPT001846   Missense_Mutation  1.3964  1.3302 -5.0948  0.7674 -1.6845     NaN   \n",
       "X03BR010    Missense_Mutation  1.5386  0.0663 -3.7769 -0.0033  1.0579  0.8396   \n",
       "X05BR004    Missense_Mutation  0.7901 -0.3931 -3.9995  0.1256 -1.8863     NaN   \n",
       "X05BR009    Missense_Mutation  3.6316  3.5662 -2.4971 -0.9388 -2.3314     NaN   \n",
       "X05BR043    Missense_Mutation  1.0922  0.8715 -2.5619 -0.0461 -3.3529     NaN   \n",
       "...                       ...     ...     ...     ...     ...     ...     ...   \n",
       "X15BR003       Wildtype_Tumor  2.4710  1.0447  1.6000 -0.2389 -0.2619     NaN   \n",
       "X18BR004       Wildtype_Tumor -0.4652 -0.2157 -3.6000 -0.3496 -0.9725     NaN   \n",
       "X18BR006       Wildtype_Tumor  0.0999  0.7852 -8.1266 -0.5701 -2.1021     NaN   \n",
       "X18BR017       Wildtype_Tumor  0.8934  1.6886 -6.0217 -0.0481  0.1299     NaN   \n",
       "X20BR001       Wildtype_Tumor  1.0545  1.0394 -3.5713 -0.7841 -1.1205     NaN   \n",
       "\n",
       "Name         AAED1   AAGAB    AAK1  ...  ZSCAN31  ZSWIM8    ZW10  ZWILCH  \\\n",
       "Patient_ID                          ...                                    \n",
       "CPT001846   2.1022 -0.5814  0.2916  ...  -0.7592  0.4711  0.6018  0.2062   \n",
       "X03BR010   -0.5769  2.5209  0.3357  ...      NaN  0.1708 -0.0591 -1.7682   \n",
       "X05BR004       NaN  0.3140 -0.0507  ...  -2.3219 -0.5268 -0.0487 -0.4863   \n",
       "X05BR009    0.3822 -0.1004  0.0427  ...  -0.1106 -0.8816 -0.3294 -0.7691   \n",
       "X05BR043   -1.1559 -0.0154 -0.0768  ...  -0.2157 -0.0073 -0.1810  1.1883   \n",
       "...            ...     ...     ...  ...      ...     ...     ...     ...   \n",
       "X15BR003    1.2958  0.1298  0.3695  ...  -0.3656 -1.0569  0.8718 -0.4324   \n",
       "X18BR004       NaN  0.2205 -0.7924  ...  -2.0780  0.9906  0.3147  0.0305   \n",
       "X18BR006   -0.5732  0.4928 -0.2838  ...  -0.3722  0.0816  0.0086  0.3131   \n",
       "X18BR017       NaN  1.5536  0.5748  ...  -0.2404  0.6150  0.2906  0.2303   \n",
       "X20BR001       NaN  0.2457 -0.5970  ...   0.4385  0.1229  0.1229 -1.2017   \n",
       "\n",
       "Name         ZWINT    ZXDC  ZYG11B     ZYX   ZZEF1    ZZZ3  \n",
       "Patient_ID                                                  \n",
       "CPT001846  -0.2137 -2.1219  0.0860  2.5814 -0.2852 -0.1074  \n",
       "X03BR010   -1.3061  0.1523 -0.6977 -1.3316  0.8303 -0.3795  \n",
       "X05BR004   -1.5257  0.5247 -0.7801 -0.6180 -0.6423  0.1904  \n",
       "X05BR009   -1.5278  0.1102 -0.5196  1.3474 -1.0431  0.1388  \n",
       "X05BR043    1.3825  0.6651 -0.3629  1.2292 -0.7839 -0.2382  \n",
       "...            ...     ...     ...     ...     ...     ...  \n",
       "X15BR003   -0.7942 -2.4072  1.2313  0.6114 -0.6675 -0.8818  \n",
       "X18BR004   -0.2289     NaN  0.2189 -0.3281  0.4254 -0.1496  \n",
       "X18BR006   -0.2260 -0.6128  0.7913  0.8553 -0.0371 -0.3874  \n",
       "X18BR017   -0.3553  0.4313  1.0197  1.0025  1.8005 -0.5533  \n",
       "X20BR001   -1.4247     NaN -0.7539 -0.2229  0.1569  0.2495  \n",
       "\n",
       "[86 rows x 9451 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#use the list of selected genes to filter dataframe\n",
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "proteomics = cptac.utils.reduce_multiindex(proteomics, flatten=True)\n",
    "selected_prot = proteomics[proteomics.columns.intersection(selected)]\n",
    "\n",
    "#get mutation_tyoe\n",
    "mut_type = br.get_genotype_all_vars('PIK3CA')\n",
    "\n",
    "#join proteomics and mutation type\n",
    "joined = mut_type.join(selected_prot)\n",
    "\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "missense = joined.loc[joined.Mutation == \"Missense_Mutation\"]\n",
    "brca_both = missense.append(wildtype)\n",
    "\n",
    "\n",
    "#drop unncessary columns\n",
    "brca_prot_and_mutations = brca_both.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "#drop databaseID identifier\n",
    "split = brca_prot_and_mutations.columns.str.split('_', 1).str[0]\n",
    "brca_prot_and_mutations.columns=split\n",
    "\n",
    "#remove duplicate columns\n",
    "brca_prot_and_mutations = brca_prot_and_mutations.loc[:,~brca_prot_and_mutations.columns.duplicated()]\n",
    "brca_prot_and_mutations\n",
    "\n",
    "# prot_and_mutations.loc[prot_and_mutations[\"Mutation\"]== \"Missense_Mutation\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BCL2L13</td>\n",
       "      <td>0.251581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HSPE1-MOB4</td>\n",
       "      <td>0.251581</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Comparison   P_Value\n",
       "0     BCL2L13  0.251581\n",
       "1  HSPE1-MOB4  0.251581"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now run ttest\n",
    "n = wrap_ttest(brca_prot_and_mutations, 'Mutation', correction_method= \"fdr_bh\", pval_return_corrected=True, alpha=.3)\n",
    "n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>AAMDC</th>\n",
       "      <th>AAMP</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNHIT6</th>\n",
       "      <th>ZNRD1</th>\n",
       "      <th>ZNRF2</th>\n",
       "      <th>ZPR1</th>\n",
       "      <th>ZRANB2</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01CO022</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.906</td>\n",
       "      <td>-0.778</td>\n",
       "      <td>1.4500</td>\n",
       "      <td>0.359</td>\n",
       "      <td>0.2190</td>\n",
       "      <td>0.7710</td>\n",
       "      <td>-0.0471</td>\n",
       "      <td>-0.9160</td>\n",
       "      <td>0.4770</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1940</td>\n",
       "      <td>0.1740</td>\n",
       "      <td>0.2990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6200</td>\n",
       "      <td>-0.7180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO003</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.692</td>\n",
       "      <td>-1.220</td>\n",
       "      <td>-0.4380</td>\n",
       "      <td>0.828</td>\n",
       "      <td>-0.0137</td>\n",
       "      <td>0.2960</td>\n",
       "      <td>-0.0302</td>\n",
       "      <td>-1.2100</td>\n",
       "      <td>0.1550</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1400</td>\n",
       "      <td>0.1040</td>\n",
       "      <td>-0.1330</td>\n",
       "      <td>-0.1020</td>\n",
       "      <td>-1.0800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0430</td>\n",
       "      <td>-0.2430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.190</td>\n",
       "      <td>-1.580</td>\n",
       "      <td>-0.6300</td>\n",
       "      <td>0.474</td>\n",
       "      <td>0.4050</td>\n",
       "      <td>0.2790</td>\n",
       "      <td>0.0172</td>\n",
       "      <td>-0.7120</td>\n",
       "      <td>-0.0105</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5520</td>\n",
       "      <td>-0.0722</td>\n",
       "      <td>0.3080</td>\n",
       "      <td>0.1620</td>\n",
       "      <td>-0.2670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7660</td>\n",
       "      <td>-0.1950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO028</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.340</td>\n",
       "      <td>0.343</td>\n",
       "      <td>-1.6600</td>\n",
       "      <td>0.504</td>\n",
       "      <td>0.1620</td>\n",
       "      <td>0.3190</td>\n",
       "      <td>-0.2980</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>-0.1050</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0595</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>0.1450</td>\n",
       "      <td>0.1970</td>\n",
       "      <td>-0.3990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7540</td>\n",
       "      <td>-0.4230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.220</td>\n",
       "      <td>-0.575</td>\n",
       "      <td>-0.5960</td>\n",
       "      <td>0.299</td>\n",
       "      <td>-0.0254</td>\n",
       "      <td>0.7640</td>\n",
       "      <td>0.1040</td>\n",
       "      <td>-0.6940</td>\n",
       "      <td>0.3560</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2850</td>\n",
       "      <td>-0.1480</td>\n",
       "      <td>0.0751</td>\n",
       "      <td>0.0503</td>\n",
       "      <td>0.2300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5900</td>\n",
       "      <td>-0.0644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20CO001</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.629</td>\n",
       "      <td>-0.068</td>\n",
       "      <td>0.0734</td>\n",
       "      <td>-0.654</td>\n",
       "      <td>0.2130</td>\n",
       "      <td>0.1550</td>\n",
       "      <td>0.5650</td>\n",
       "      <td>-0.0406</td>\n",
       "      <td>-0.1540</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4150</td>\n",
       "      <td>-0.0765</td>\n",
       "      <td>-0.0385</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4260</td>\n",
       "      <td>0.5190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20CO003</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.123</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-0.4520</td>\n",
       "      <td>0.266</td>\n",
       "      <td>-0.2280</td>\n",
       "      <td>0.0484</td>\n",
       "      <td>0.1710</td>\n",
       "      <td>0.7850</td>\n",
       "      <td>0.3000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0415</td>\n",
       "      <td>0.0830</td>\n",
       "      <td>-0.0374</td>\n",
       "      <td>-0.1320</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6810</td>\n",
       "      <td>-0.2970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO004</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.280</td>\n",
       "      <td>0.518</td>\n",
       "      <td>-0.8720</td>\n",
       "      <td>0.258</td>\n",
       "      <td>0.4240</td>\n",
       "      <td>0.2980</td>\n",
       "      <td>-0.1500</td>\n",
       "      <td>-0.6580</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4490</td>\n",
       "      <td>-0.0616</td>\n",
       "      <td>0.3850</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>-0.0437</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.8840</td>\n",
       "      <td>0.0088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO006</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.515</td>\n",
       "      <td>-1.210</td>\n",
       "      <td>-0.0283</td>\n",
       "      <td>0.178</td>\n",
       "      <td>-0.3030</td>\n",
       "      <td>0.1500</td>\n",
       "      <td>0.3990</td>\n",
       "      <td>-1.0000</td>\n",
       "      <td>0.2060</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5610</td>\n",
       "      <td>0.4740</td>\n",
       "      <td>0.0836</td>\n",
       "      <td>-0.2510</td>\n",
       "      <td>0.1760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0584</td>\n",
       "      <td>-0.3010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24CO005</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>85 rows × 8068 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation   A1BG   A1CF     A2M   AAAS    AACS   AAGAB  \\\n",
       "Patient_ID                                                                   \n",
       "01CO022     Missense_Mutation  0.906 -0.778  1.4500  0.359  0.2190  0.7710   \n",
       "05CO003     Missense_Mutation -0.692 -1.220 -0.4380  0.828 -0.0137  0.2960   \n",
       "05CO006     Missense_Mutation -1.190 -1.580 -0.6300  0.474  0.4050  0.2790   \n",
       "05CO028     Missense_Mutation -1.340  0.343 -1.6600  0.504  0.1620  0.3190   \n",
       "05CO032     Missense_Mutation -1.220 -0.575 -0.5960  0.299 -0.0254  0.7640   \n",
       "...                       ...    ...    ...     ...    ...     ...     ...   \n",
       "20CO001        Wildtype_Tumor  0.629 -0.068  0.0734 -0.654  0.2130  0.1550   \n",
       "20CO003        Wildtype_Tumor -0.123 -1.070 -0.4520  0.266 -0.2280  0.0484   \n",
       "22CO004        Wildtype_Tumor -1.280  0.518 -0.8720  0.258  0.4240  0.2980   \n",
       "22CO006        Wildtype_Tumor  0.515 -1.210 -0.0283  0.178 -0.3030  0.1500   \n",
       "24CO005        Wildtype_Tumor    NaN    NaN     NaN    NaN     NaN     NaN   \n",
       "\n",
       "Name          AAK1   AAMDC    AAMP  ...  ZNHIT6  ZNRD1   ZNRF2    ZPR1  \\\n",
       "Patient_ID                          ...                                  \n",
       "01CO022    -0.0471 -0.9160  0.4770  ...     NaN    NaN     NaN  0.1940   \n",
       "05CO003    -0.0302 -1.2100  0.1550  ...     NaN    NaN -1.1400  0.1040   \n",
       "05CO006     0.0172 -0.7120 -0.0105  ...     NaN    NaN -0.5520 -0.0722   \n",
       "05CO028    -0.2980 -0.2150 -0.1050  ...     NaN    NaN -0.0595  0.1690   \n",
       "05CO032     0.1040 -0.6940  0.3560  ...  0.0105    NaN -0.2850 -0.1480   \n",
       "...            ...     ...     ...  ...     ...    ...     ...     ...   \n",
       "20CO001     0.5650 -0.0406 -0.1540  ...     NaN    NaN -0.1340  0.4150   \n",
       "20CO003     0.1710  0.7850  0.3000  ...     NaN    NaN     NaN -0.0415   \n",
       "22CO004    -0.1500 -0.6580  0.0066  ...     NaN    NaN  0.4490 -0.0616   \n",
       "22CO006     0.3990 -1.0000  0.2060  ...     NaN    NaN -0.5610  0.4740   \n",
       "24CO005        NaN     NaN     NaN  ...     NaN    NaN     NaN     NaN   \n",
       "\n",
       "Name        ZRANB2    ZW10  ZWILCH  ZWINT     ZYX   ZZEF1  \n",
       "Patient_ID                                                 \n",
       "01CO022     0.1740  0.2990     NaN    NaN -0.6200 -0.7180  \n",
       "05CO003    -0.1330 -0.1020 -1.0800    NaN -0.0430 -0.2430  \n",
       "05CO006     0.3080  0.1620 -0.2670    NaN -0.7660 -0.1950  \n",
       "05CO028     0.1450  0.1970 -0.3990    NaN -0.7540 -0.4230  \n",
       "05CO032     0.0751  0.0503  0.2300    NaN -0.5900 -0.0644  \n",
       "...            ...     ...     ...    ...     ...     ...  \n",
       "20CO001    -0.0765 -0.0385     NaN    NaN  0.4260  0.5190  \n",
       "20CO003     0.0830 -0.0374 -0.1320    NaN  0.6810 -0.2970  \n",
       "22CO004     0.3850  0.1160 -0.0437    NaN -0.8840  0.0088  \n",
       "22CO006     0.0836 -0.2510  0.1760    NaN  0.0584 -0.3010  \n",
       "24CO005        NaN     NaN     NaN    NaN     NaN     NaN  \n",
       "\n",
       "[85 rows x 8068 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "proteomics = cl.get_proteomics(tissue_type=\"tumor\")\n",
    "\n",
    "prot_list = list(proteomics.columns)\n",
    "\n",
    "mut_type = cl.get_genotype_all_vars(\"PIK3CA\")\n",
    "\n",
    "\n",
    "joined = mut_type.join(proteomics)\n",
    "\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "missense = joined.loc[joined.Mutation == 'nonsynonymous SNV']\n",
    "colon_both = missense.append(wildtype)\n",
    "\n",
    "#drop unncessary columns\n",
    "col_prot_and_mutations = colon_both.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "col_prot_and_mutations[\"Mutation\"] = col_prot_and_mutations['Mutation'].replace(['nonsynonymous SNV'], 'Missense_Mutation')\n",
    "col_prot_and_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'prot_and_mutations' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-21-fc3f0449f6b8>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m#get the difference in medians\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mmissense\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprot_and_mutations\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mprot_and_mutations\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMutation\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"Missense_Mutation\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;31m#get all missense_mutations\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mwt\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprot_and_mutations\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mprot_and_mutations\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMutation\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"Wildtype_Tumor\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;31m#getn all wildtype\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mmissense_med\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmissense\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmedian\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m#get the medain of the missense. This will give the median expression among all samples, for each gene\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mwt_med\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmedian\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m#get the median of the wildtype\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'prot_and_mutations' is not defined"
     ]
    }
   ],
   "source": [
    "#get the difference in medians\n",
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "colon_d = {}\n",
    "\n",
    "for prot in proteomics:\n",
    "    dif_colon = missense_med[prot] - wt_med[prot]\n",
    "    colon_d[prot] = dif_colon\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(colon_d, orient='index', columns=['Difference_In_Median'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#do a t test for every gene and report the p-value\n",
    "cols = list(col_prot_and_mutations.columns[1:])\n",
    "t_test = wrap_ttest(col_prot_and_mutations, 'Mutation', cols, correction_method= \"fdr_bh\")\n",
    "# t_test = t_test.set_index('Comparison')\n",
    "t_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# t_test.head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#join median and p value together\n",
    "colon_df = median_diff.join(t_test)\n",
    "colon_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine Endo, Brca, and Colon into one table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine brca and endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "brca_df=brca_df.rename_axis(None)\n",
    "brca_endo = brca_df.join(endo_df, lsuffix='_Brca', rsuffix='_Endo')\n",
    "brca_endo.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_table = brca_endo.join(colon_df, rsuffix='_colon')\n",
    "final_table.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Only rows that contain a significant pvalue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "significant = final_table.loc[(final_table['P_Value_Brca'] <= .05) |\n",
    "                (final_table['P_Value_Endo'] <= .05) |\n",
    "                (final_table['P_Value'] <= .05)]\n",
    "significant.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "significant.to_csv(\"~/WhenMutationsDontMatter/PIK3CA/csv_files/proteomics.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
