{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter out hotspot mutations\n",
    "Hotspot:\n",
    "E542K\n",
    "E535K\n",
    "H1047R\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "en = cptac.Endometrial()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PIK3CA'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on method get_genotype_all_vars in module cptac.dataset:\n",
      "\n",
      "get_genotype_all_vars(mutations_genes, mutations_filter=None, show_location=True, mutation_hotspot=None) method of cptac.endometrial.Endometrial instance\n",
      "    Return a dataframe that has the mutation type and wheather or not it is a multiple mutation\n",
      "    Parameters:\n",
      "    mutation_genes (str, or list or array-like of str): The gene(s) to get mutation data for.\n",
      "    mutations_filter (list, optional):  List of mutations to prioritize when filtering out multiple mutations, in order of priority.\n",
      "    show_location (bool, optional): Whether to include the Location column from the mutation dataframe. Defaults to True.\n",
      "    mutation_hotspot (optional): a list of hotspots\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(en.get_genotype_all_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 48 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 382)\n"
     ]
    }
   ],
   "source": [
    "mut = en.get_genotype_all_vars(gene, mutation_hotspot=['E542K', 'E535K', 'H1047R'])\n",
    "prot = en.get_proteomics()\n",
    "protfilter = prot.index.str.endswith('.N')\n",
    "prot = prot[~protfilter]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02290</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.6840</td>\n",
       "      <td>0.9840</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.36300</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.7370</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.07570</td>\n",
       "      <td>-0.4730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.01050</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.32000</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>0.2430</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.07140</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.1560</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.03010</td>\n",
       "      <td>-0.4670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>Amplification</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.374</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.5370</td>\n",
       "      <td>0.3110</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.22000</td>\n",
       "      <td>-0.5620</td>\n",
       "      <td>0.9370</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>0.0513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E726K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-0.7120</td>\n",
       "      <td>0.0462</td>\n",
       "      <td>-0.0471</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>1.3500</td>\n",
       "      <td>1.2100</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.06990</td>\n",
       "      <td>-0.4010</td>\n",
       "      <td>0.55700</td>\n",
       "      <td>0.8270</td>\n",
       "      <td>0.3480</td>\n",
       "      <td>-0.187</td>\n",
       "      <td>-0.10700</td>\n",
       "      <td>-0.830</td>\n",
       "      <td>0.06200</td>\n",
       "      <td>-0.5280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>Amplification</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.280</td>\n",
       "      <td>-0.7360</td>\n",
       "      <td>-0.5520</td>\n",
       "      <td>0.1140</td>\n",
       "      <td>0.2730</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.8500</td>\n",
       "      <td>...</td>\n",
       "      <td>0.13100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00364</td>\n",
       "      <td>0.7630</td>\n",
       "      <td>0.0784</td>\n",
       "      <td>-0.203</td>\n",
       "      <td>0.01320</td>\n",
       "      <td>-1.200</td>\n",
       "      <td>-0.35700</td>\n",
       "      <td>0.0299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.290</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>3.1700</td>\n",
       "      <td>-0.9070</td>\n",
       "      <td>0.0317</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17900</td>\n",
       "      <td>-0.5210</td>\n",
       "      <td>0.05230</td>\n",
       "      <td>0.5150</td>\n",
       "      <td>0.5850</td>\n",
       "      <td>0.138</td>\n",
       "      <td>-0.19000</td>\n",
       "      <td>-0.966</td>\n",
       "      <td>-0.00627</td>\n",
       "      <td>-0.2490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.266</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.4700</td>\n",
       "      <td>0.3980</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.14000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.01220</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>0.5530</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.06420</td>\n",
       "      <td>-0.437</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.4980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.K111E</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>0.692</td>\n",
       "      <td>0.5890</td>\n",
       "      <td>0.5870</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0835</td>\n",
       "      <td>0.1720</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.02250</td>\n",
       "      <td>0.2980</td>\n",
       "      <td>-0.12400</td>\n",
       "      <td>0.1780</td>\n",
       "      <td>0.1470</td>\n",
       "      <td>-0.114</td>\n",
       "      <td>0.13700</td>\n",
       "      <td>1.240</td>\n",
       "      <td>0.00682</td>\n",
       "      <td>-0.0233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 11002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation       Location  Mutation_Status   A1BG     A2M  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00006   Missense_Mutation        p.E545K  Single_mutation -1.180 -0.8630   \n",
       "C3L-00008      Wildtype_Tumor    No_mutation   Wildtype_Tumor -0.685 -1.0700   \n",
       "C3L-00032   Missense_Mutation        p.E545K  Single_mutation -0.528 -1.3200   \n",
       "C3L-00090      Wildtype_Tumor    No_mutation   Wildtype_Tumor -1.670 -1.1900   \n",
       "C3L-00098       Amplification  Amplification  Single_mutation -0.374 -0.0206   \n",
       "...                       ...            ...              ...    ...     ...   \n",
       "C3N-01520   Missense_Mutation        p.E726K  Single_mutation -1.070 -0.7120   \n",
       "C3N-01521       Amplification  Amplification  Single_mutation -1.280 -0.7360   \n",
       "C3N-01537      Wildtype_Tumor    No_mutation   Wildtype_Tumor -0.290 -0.3200   \n",
       "C3N-01802      Wildtype_Tumor    No_mutation   Wildtype_Tumor  0.266  1.3900   \n",
       "C3N-01825   Missense_Mutation        p.K111E  Single_mutation  0.692  0.5890   \n",
       "\n",
       "Name         A2ML1  A4GALT    AAAS    AACS   AADAT  ...   ZSWIM8  ZSWIM9  \\\n",
       "Patient_ID                                          ...                    \n",
       "C3L-00006  -0.8020  0.2220  0.2560  0.6650  1.2800  ... -0.08770     NaN   \n",
       "C3L-00008  -0.6840  0.9840  0.1350  0.3340  1.3000  ... -0.03560     NaN   \n",
       "C3L-00032   0.4350     NaN -0.2400  1.0400 -0.0213  ...  0.00112 -0.1450   \n",
       "C3L-00090  -0.4430  0.2430 -0.0993  0.7570  0.7400  ...  0.07250 -0.0552   \n",
       "C3L-00098  -0.5370  0.3110  0.3750  0.0131 -1.1000  ... -0.17600     NaN   \n",
       "...            ...     ...     ...     ...     ...  ...      ...     ...   \n",
       "C3N-01520   0.0462 -0.0471  0.2270  1.3500  1.2100  ... -0.06990 -0.4010   \n",
       "C3N-01521  -0.5520  0.1140  0.2730  1.0700  0.8500  ...  0.13100     NaN   \n",
       "C3N-01537   3.1700 -0.9070  0.0317 -0.0425     NaN  ... -0.17900 -0.5210   \n",
       "C3N-01802  -0.0655  0.4700  0.3980 -0.1340  0.4610  ... -0.14000     NaN   \n",
       "C3N-01825   0.5870     NaN -0.0835  0.1720 -0.2150  ... -0.02250  0.2980   \n",
       "\n",
       "Name           ZW10  ZWILCH   ZWINT   ZXDC   ZYG11B    ZYX    ZZEF1    ZZZ3  \n",
       "Patient_ID                                                                   \n",
       "C3L-00006   0.02290  0.1090     NaN -0.332 -0.43300 -1.020 -0.12300 -0.0859  \n",
       "C3L-00008   0.36300  1.0700  0.7370 -0.564 -0.00461 -1.130 -0.07570 -0.4730  \n",
       "C3L-00032   0.01050 -0.1160     NaN  0.151 -0.07400 -0.540  0.32000 -0.4190  \n",
       "C3L-00090  -0.07140  0.0933  0.1560 -0.398 -0.07520 -0.797 -0.03010 -0.4670  \n",
       "C3L-00098  -1.22000 -0.5620  0.9370 -0.646  0.20700 -1.850 -0.17600  0.0513  \n",
       "...             ...     ...     ...    ...      ...    ...      ...     ...  \n",
       "C3N-01520   0.55700  0.8270  0.3480 -0.187 -0.10700 -0.830  0.06200 -0.5280  \n",
       "C3N-01521  -0.00364  0.7630  0.0784 -0.203  0.01320 -1.200 -0.35700  0.0299  \n",
       "C3N-01537   0.05230  0.5150  0.5850  0.138 -0.19000 -0.966 -0.00627 -0.2490  \n",
       "C3N-01802  -0.01220  0.2500  0.5530  0.387  0.06420 -0.437  0.10400 -0.4980  \n",
       "C3N-01825  -0.12400  0.1780  0.1470 -0.114  0.13700  1.240  0.00682 -0.0233  \n",
       "\n",
       "[95 rows x 11002 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "joined = mut.join(prot)\n",
    "joined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00362</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.9240</td>\n",
       "      <td>-0.445</td>\n",
       "      <td>1.57</td>\n",
       "      <td>-0.678</td>\n",
       "      <td>0.173</td>\n",
       "      <td>0.436</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0496</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1080</td>\n",
       "      <td>-0.153</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.109</td>\n",
       "      <td>-0.3360</td>\n",
       "      <td>-0.8220</td>\n",
       "      <td>-0.0338</td>\n",
       "      <td>0.12100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00324</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.4490</td>\n",
       "      <td>-0.440</td>\n",
       "      <td>1.15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.645</td>\n",
       "      <td>0.316</td>\n",
       "      <td>0.0809</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0449</td>\n",
       "      <td>0.0452</td>\n",
       "      <td>-0.1520</td>\n",
       "      <td>-0.197</td>\n",
       "      <td>0.295</td>\n",
       "      <td>-0.188</td>\n",
       "      <td>-0.2650</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>-0.1130</td>\n",
       "      <td>-0.30200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00383</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>0.0729</td>\n",
       "      <td>0.567</td>\n",
       "      <td>1.71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.278</td>\n",
       "      <td>0.646</td>\n",
       "      <td>0.3780</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2790</td>\n",
       "      <td>-0.3780</td>\n",
       "      <td>0.0125</td>\n",
       "      <td>-0.305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.149</td>\n",
       "      <td>-0.0526</td>\n",
       "      <td>-0.0311</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.00801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01212</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>-0.142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.211</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0.4920</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0406</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0924</td>\n",
       "      <td>0.121</td>\n",
       "      <td>0.433</td>\n",
       "      <td>-1.050</td>\n",
       "      <td>-0.3160</td>\n",
       "      <td>-0.4000</td>\n",
       "      <td>0.1230</td>\n",
       "      <td>-0.39100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 11002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation  Location  Mutation_Status    A1BG    A2M  \\\n",
       "Patient_ID                                                                \n",
       "C3L-00362   Missense_Mutation  p.H1047R  Single_mutation -0.9240 -0.445   \n",
       "C3N-00324   Missense_Mutation  p.H1047R  Single_mutation -0.4490 -0.440   \n",
       "C3N-00383   Missense_Mutation  p.H1047R  Single_mutation  0.0729  0.567   \n",
       "C3N-01212   Missense_Mutation  p.H1047R  Single_mutation  0.0308 -0.142   \n",
       "\n",
       "Name        A2ML1  A4GALT   AAAS   AACS   AADAT  ...  ZSWIM8  ZSWIM9    ZW10  \\\n",
       "Patient_ID                                       ...                           \n",
       "C3L-00362    1.57  -0.678  0.173  0.436  0.2270  ... -0.0496     NaN  0.1080   \n",
       "C3N-00324    1.15     NaN -0.645  0.316  0.0809  ...  0.0449  0.0452 -0.1520   \n",
       "C3N-00383    1.71     NaN -0.278  0.646  0.3780  ...  0.2790 -0.3780  0.0125   \n",
       "C3N-01212     NaN     NaN  0.211  0.927  0.4920  ...  0.0406     NaN  0.0924   \n",
       "\n",
       "Name        ZWILCH  ZWINT   ZXDC  ZYG11B     ZYX   ZZEF1     ZZZ3  \n",
       "Patient_ID                                                         \n",
       "C3L-00362   -0.153    NaN  0.109 -0.3360 -0.8220 -0.0338  0.12100  \n",
       "C3N-00324   -0.197  0.295 -0.188 -0.2650  0.2500 -0.1130 -0.30200  \n",
       "C3N-00383   -0.305    NaN  0.149 -0.0526 -0.0311  0.4320  0.00801  \n",
       "C3N-01212    0.121  0.433 -1.050 -0.3160 -0.4000  0.1230 -0.39100  \n",
       "\n",
       "[4 rows x 11002 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "joined[joined.Location.str.contains('H1047R')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00362</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.9240</td>\n",
       "      <td>-0.445</td>\n",
       "      <td>1.57</td>\n",
       "      <td>-0.678</td>\n",
       "      <td>0.173</td>\n",
       "      <td>0.436</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0496</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1080</td>\n",
       "      <td>-0.153</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.109</td>\n",
       "      <td>-0.3360</td>\n",
       "      <td>-0.8220</td>\n",
       "      <td>-0.0338</td>\n",
       "      <td>0.12100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00324</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.4490</td>\n",
       "      <td>-0.440</td>\n",
       "      <td>1.15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.645</td>\n",
       "      <td>0.316</td>\n",
       "      <td>0.0809</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0449</td>\n",
       "      <td>0.0452</td>\n",
       "      <td>-0.1520</td>\n",
       "      <td>-0.197</td>\n",
       "      <td>0.295</td>\n",
       "      <td>-0.188</td>\n",
       "      <td>-0.2650</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>-0.1130</td>\n",
       "      <td>-0.30200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00383</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>0.0729</td>\n",
       "      <td>0.567</td>\n",
       "      <td>1.71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.278</td>\n",
       "      <td>0.646</td>\n",
       "      <td>0.3780</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2790</td>\n",
       "      <td>-0.3780</td>\n",
       "      <td>0.0125</td>\n",
       "      <td>-0.305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.149</td>\n",
       "      <td>-0.0526</td>\n",
       "      <td>-0.0311</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.00801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01212</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>-0.142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.211</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0.4920</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0406</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0924</td>\n",
       "      <td>0.121</td>\n",
       "      <td>0.433</td>\n",
       "      <td>-1.050</td>\n",
       "      <td>-0.3160</td>\n",
       "      <td>-0.4000</td>\n",
       "      <td>0.1230</td>\n",
       "      <td>-0.39100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 11002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation  Location  Mutation_Status    A1BG    A2M  \\\n",
       "Patient_ID                                                                \n",
       "C3L-00362   Missense_Mutation  p.H1047R  Single_mutation -0.9240 -0.445   \n",
       "C3N-00324   Missense_Mutation  p.H1047R  Single_mutation -0.4490 -0.440   \n",
       "C3N-00383   Missense_Mutation  p.H1047R  Single_mutation  0.0729  0.567   \n",
       "C3N-01212   Missense_Mutation  p.H1047R  Single_mutation  0.0308 -0.142   \n",
       "\n",
       "Name        A2ML1  A4GALT   AAAS   AACS   AADAT  ...  ZSWIM8  ZSWIM9    ZW10  \\\n",
       "Patient_ID                                       ...                           \n",
       "C3L-00362    1.57  -0.678  0.173  0.436  0.2270  ... -0.0496     NaN  0.1080   \n",
       "C3N-00324    1.15     NaN -0.645  0.316  0.0809  ...  0.0449  0.0452 -0.1520   \n",
       "C3N-00383    1.71     NaN -0.278  0.646  0.3780  ...  0.2790 -0.3780  0.0125   \n",
       "C3N-01212     NaN     NaN  0.211  0.927  0.4920  ...  0.0406     NaN  0.0924   \n",
       "\n",
       "Name        ZWILCH  ZWINT   ZXDC  ZYG11B     ZYX   ZZEF1     ZZZ3  \n",
       "Patient_ID                                                         \n",
       "C3L-00362   -0.153    NaN  0.109 -0.3360 -0.8220 -0.0338  0.12100  \n",
       "C3N-00324   -0.197  0.295 -0.188 -0.2650  0.2500 -0.1130 -0.30200  \n",
       "C3N-00383   -0.305    NaN  0.149 -0.0526 -0.0311  0.4320  0.00801  \n",
       "C3N-01212    0.121  0.433 -1.050 -0.3160 -0.4000  0.1230 -0.39100  \n",
       "\n",
       "[4 rows x 11002 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "joined[joined.Location.str.contains('H1047R')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endo get difference in median and t-test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>-1.1800</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.256000</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.412</td>\n",
       "      <td>-0.66400</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.33200</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.1230</td>\n",
       "      <td>-0.085900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>-0.6850</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.684</td>\n",
       "      <td>0.984</td>\n",
       "      <td>0.135000</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>0.1390</td>\n",
       "      <td>1.330</td>\n",
       "      <td>-0.36700</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.737</td>\n",
       "      <td>-0.56400</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.0757</td>\n",
       "      <td>-0.473000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>-0.5280</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.240000</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.419</td>\n",
       "      <td>-0.50000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.15100</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.3200</td>\n",
       "      <td>-0.419000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>-1.6700</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.443</td>\n",
       "      <td>0.243</td>\n",
       "      <td>-0.099300</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>-0.9290</td>\n",
       "      <td>0.229</td>\n",
       "      <td>-0.22300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.0714</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.156</td>\n",
       "      <td>-0.39800</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.0301</td>\n",
       "      <td>-0.467000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>-0.3740</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.537</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.375000</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.565</td>\n",
       "      <td>-0.10100</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.2200</td>\n",
       "      <td>-0.5620</td>\n",
       "      <td>0.937</td>\n",
       "      <td>-0.64600</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.1760</td>\n",
       "      <td>0.051300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NX5.N</th>\n",
       "      <td>-0.0014</td>\n",
       "      <td>0.6160</td>\n",
       "      <td>-0.641</td>\n",
       "      <td>-0.204</td>\n",
       "      <td>0.027600</td>\n",
       "      <td>-1.4100</td>\n",
       "      <td>-0.4050</td>\n",
       "      <td>-0.5610</td>\n",
       "      <td>-0.373</td>\n",
       "      <td>0.12400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.21500</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>-0.1900</td>\n",
       "      <td>-1.3600</td>\n",
       "      <td>-1.100</td>\n",
       "      <td>-0.19700</td>\n",
       "      <td>-0.30700</td>\n",
       "      <td>0.682</td>\n",
       "      <td>0.0984</td>\n",
       "      <td>-0.020200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NX6.N</th>\n",
       "      <td>0.2980</td>\n",
       "      <td>0.3950</td>\n",
       "      <td>-1.240</td>\n",
       "      <td>-0.178</td>\n",
       "      <td>-0.180000</td>\n",
       "      <td>-0.5910</td>\n",
       "      <td>-0.5980</td>\n",
       "      <td>0.4020</td>\n",
       "      <td>-0.384</td>\n",
       "      <td>0.15200</td>\n",
       "      <td>...</td>\n",
       "      <td>0.05470</td>\n",
       "      <td>0.1980</td>\n",
       "      <td>-0.0139</td>\n",
       "      <td>-0.8080</td>\n",
       "      <td>-0.324</td>\n",
       "      <td>0.16300</td>\n",
       "      <td>-0.14500</td>\n",
       "      <td>0.855</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>-0.000114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NX7.N</th>\n",
       "      <td>0.1110</td>\n",
       "      <td>0.2100</td>\n",
       "      <td>-0.751</td>\n",
       "      <td>-0.335</td>\n",
       "      <td>0.000865</td>\n",
       "      <td>-1.3700</td>\n",
       "      <td>-0.2730</td>\n",
       "      <td>0.1060</td>\n",
       "      <td>-0.717</td>\n",
       "      <td>0.00865</td>\n",
       "      <td>...</td>\n",
       "      <td>0.17600</td>\n",
       "      <td>0.3960</td>\n",
       "      <td>-0.3640</td>\n",
       "      <td>-1.6400</td>\n",
       "      <td>-1.480</td>\n",
       "      <td>0.45300</td>\n",
       "      <td>-0.16800</td>\n",
       "      <td>0.411</td>\n",
       "      <td>-0.1760</td>\n",
       "      <td>0.218000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NX8.N</th>\n",
       "      <td>0.9280</td>\n",
       "      <td>0.9520</td>\n",
       "      <td>-1.380</td>\n",
       "      <td>-0.345</td>\n",
       "      <td>-0.002170</td>\n",
       "      <td>-0.6570</td>\n",
       "      <td>-0.3630</td>\n",
       "      <td>0.0807</td>\n",
       "      <td>-0.200</td>\n",
       "      <td>0.03500</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.13500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0926</td>\n",
       "      <td>-0.1550</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.00177</td>\n",
       "      <td>0.05320</td>\n",
       "      <td>0.850</td>\n",
       "      <td>0.0444</td>\n",
       "      <td>0.001660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NX9.N</th>\n",
       "      <td>0.8990</td>\n",
       "      <td>0.1360</td>\n",
       "      <td>-1.630</td>\n",
       "      <td>-0.468</td>\n",
       "      <td>-0.293000</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>-0.7120</td>\n",
       "      <td>-0.2610</td>\n",
       "      <td>-0.466</td>\n",
       "      <td>0.23100</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00943</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3490</td>\n",
       "      <td>-0.5490</td>\n",
       "      <td>-0.468</td>\n",
       "      <td>-0.15400</td>\n",
       "      <td>0.10800</td>\n",
       "      <td>1.730</td>\n",
       "      <td>0.1670</td>\n",
       "      <td>-0.059000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows × 10999 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name          A1BG     A2M  A2ML1  A4GALT      AAAS    AACS   AADAT   AAED1  \\\n",
       "Patient_ID                                                                    \n",
       "C3L-00006  -1.1800 -0.8630 -0.802   0.222  0.256000  0.6650  1.2800 -0.3390   \n",
       "C3L-00008  -0.6850 -1.0700 -0.684   0.984  0.135000  0.3340  1.3000  0.1390   \n",
       "C3L-00032  -0.5280 -1.3200  0.435     NaN -0.240000  1.0400 -0.0213 -0.0479   \n",
       "C3L-00090  -1.6700 -1.1900 -0.443   0.243 -0.099300  0.7570  0.7400 -0.9290   \n",
       "C3L-00098  -0.3740 -0.0206 -0.537   0.311  0.375000  0.0131 -1.1000     NaN   \n",
       "...            ...     ...    ...     ...       ...     ...     ...     ...   \n",
       "NX5.N      -0.0014  0.6160 -0.641  -0.204  0.027600 -1.4100 -0.4050 -0.5610   \n",
       "NX6.N       0.2980  0.3950 -1.240  -0.178 -0.180000 -0.5910 -0.5980  0.4020   \n",
       "NX7.N       0.1110  0.2100 -0.751  -0.335  0.000865 -1.3700 -0.2730  0.1060   \n",
       "NX8.N       0.9280  0.9520 -1.380  -0.345 -0.002170 -0.6570 -0.3630  0.0807   \n",
       "NX9.N       0.8990  0.1360 -1.630  -0.468 -0.293000 -1.1000 -0.7120 -0.2610   \n",
       "\n",
       "Name        AAGAB     AAK1  ...   ZSWIM8  ZSWIM9    ZW10  ZWILCH  ZWINT  \\\n",
       "Patient_ID                  ...                                           \n",
       "C3L-00006   0.412 -0.66400  ... -0.08770     NaN  0.0229  0.1090    NaN   \n",
       "C3L-00008   1.330 -0.36700  ... -0.03560     NaN  0.3630  1.0700  0.737   \n",
       "C3L-00032   0.419 -0.50000  ...  0.00112 -0.1450  0.0105 -0.1160    NaN   \n",
       "C3L-00090   0.229 -0.22300  ...  0.07250 -0.0552 -0.0714  0.0933  0.156   \n",
       "C3L-00098   0.565 -0.10100  ... -0.17600     NaN -1.2200 -0.5620  0.937   \n",
       "...           ...      ...  ...      ...     ...     ...     ...    ...   \n",
       "NX5.N      -0.373  0.12400  ...  0.21500  0.1070 -0.1900 -1.3600 -1.100   \n",
       "NX6.N      -0.384  0.15200  ...  0.05470  0.1980 -0.0139 -0.8080 -0.324   \n",
       "NX7.N      -0.717  0.00865  ...  0.17600  0.3960 -0.3640 -1.6400 -1.480   \n",
       "NX8.N      -0.200  0.03500  ... -0.13500     NaN  0.0926 -0.1550  0.151   \n",
       "NX9.N      -0.466  0.23100  ... -0.00943     NaN -0.3490 -0.5490 -0.468   \n",
       "\n",
       "Name           ZXDC   ZYG11B    ZYX   ZZEF1      ZZZ3  \n",
       "Patient_ID                                             \n",
       "C3L-00006  -0.33200 -0.43300 -1.020 -0.1230 -0.085900  \n",
       "C3L-00008  -0.56400 -0.00461 -1.130 -0.0757 -0.473000  \n",
       "C3L-00032   0.15100 -0.07400 -0.540  0.3200 -0.419000  \n",
       "C3L-00090  -0.39800 -0.07520 -0.797 -0.0301 -0.467000  \n",
       "C3L-00098  -0.64600  0.20700 -1.850 -0.1760  0.051300  \n",
       "...             ...      ...    ...     ...       ...  \n",
       "NX5.N      -0.19700 -0.30700  0.682  0.0984 -0.020200  \n",
       "NX6.N       0.16300 -0.14500  0.855  0.1160 -0.000114  \n",
       "NX7.N       0.45300 -0.16800  0.411 -0.1760  0.218000  \n",
       "NX8.N      -0.00177  0.05320  0.850  0.0444  0.001660  \n",
       "NX9.N      -0.15400  0.10800  1.730  0.1670 -0.059000  \n",
       "\n",
       "[144 rows x 10999 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "en.get_proteomics()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 48 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 382)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:153: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:257: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.07100</td>\n",
       "      <td>0.720461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.10300</td>\n",
       "      <td>0.681688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>-0.47135</td>\n",
       "      <td>0.136212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>-0.22545</td>\n",
       "      <td>0.492356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.07410</td>\n",
       "      <td>0.295868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>-0.08700</td>\n",
       "      <td>0.808610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>0.03240</td>\n",
       "      <td>0.755896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>0.13800</td>\n",
       "      <td>0.251409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0.00210</td>\n",
       "      <td>0.398421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.01570</td>\n",
       "      <td>0.978939</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10999 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median   P_Value\n",
       "A1BG                -0.07100  0.720461\n",
       "A2M                 -0.10300  0.681688\n",
       "A2ML1               -0.47135  0.136212\n",
       "A4GALT              -0.22545  0.492356\n",
       "AAAS                -0.07410  0.295868\n",
       "...                      ...       ...\n",
       "ZXDC                -0.08700  0.808610\n",
       "ZYG11B               0.03240  0.755896\n",
       "ZYX                  0.13800  0.251409\n",
       "ZZEF1                0.00210  0.398421\n",
       "ZZZ3                 0.01570  0.978939\n",
       "\n",
       "[10999 rows x 2 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "proteomics = en.get_proteomics()\n",
    "protfilter = proteomics.index.str.endswith('.N') #filter out normal samples\n",
    "proteomics = proteomics[~protfilter]\n",
    "\n",
    "\n",
    "prot_list = list(proteomics.columns)\n",
    "\n",
    "mut_type = en.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "prot_and_mutations = mut_type.join(proteomics)\n",
    "\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Missense_Mutation']\n",
    "get = prot_and_mutations['Mutation'].isin(compare)\n",
    "missense_wt = prot_and_mutations[get]\n",
    "missense_wt['Mutation'].value_counts()\n",
    "\n",
    "# Step 2 get the difference in medians\n",
    "missense = missense_wt[missense_wt.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = missense_wt[missense_wt.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "endo_d = {}\n",
    "\n",
    "for prot in proteomics:\n",
    "    dif_endo = wt_med[prot] - missense_med[prot]\n",
    "    endo_d[prot] = dif_endo\n",
    "#     print(wt_med[prot], '-', missense_med[prot], '=',dif_endo)\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(endo_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#step 3 do a t test for every gene and report the p-value\n",
    "cols = list(missense_wt.columns[1:])\n",
    "t_test = u.wrap_ttest(missense_wt, 'Mutation', cols, return_all = True)\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "# Step 4 join median and p value together\n",
    "endo_df = median_diff.join(t_test)\n",
    "endo_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Brca get difference in median and t-test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "br = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#separate proteomics into genes with duplicate database ID and genes with only one databaseID\n",
    "proteomics = br.get_proteomics()\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "no_duplicates = proteomics[proteomics.columns[~fil]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 83 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 382)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG_NP_570602.2</th>\n",
       "      <td>-0.64290</td>\n",
       "      <td>0.605521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M_NP_000005.2</th>\n",
       "      <td>0.26040</td>\n",
       "      <td>0.917772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1_NP_653271.2|NP_001269353.1</th>\n",
       "      <td>0.74900</td>\n",
       "      <td>0.070253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS_NP_056480.1|NP_001166937.1</th>\n",
       "      <td>-0.16450</td>\n",
       "      <td>0.093375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS_NP_076417.2|NP_001306769.1|NP_001306768.1</th>\n",
       "      <td>0.07180</td>\n",
       "      <td>0.899917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC_NP_079388.3|NP_001035743.1</th>\n",
       "      <td>-0.08945</td>\n",
       "      <td>0.583564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B_NP_078922.1</th>\n",
       "      <td>0.13290</td>\n",
       "      <td>0.124112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX_NP_001010972.1</th>\n",
       "      <td>-0.47940</td>\n",
       "      <td>0.426235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1_NP_055928.3</th>\n",
       "      <td>0.02700</td>\n",
       "      <td>0.537838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3_NP_056349.1|NP_001295166.1</th>\n",
       "      <td>0.05820</td>\n",
       "      <td>0.804589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9452 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Difference_In_Median   P_Value\n",
       "A1BG_NP_570602.2                                            -0.64290  0.605521\n",
       "A2M_NP_000005.2                                              0.26040  0.917772\n",
       "A2ML1_NP_653271.2|NP_001269353.1                             0.74900  0.070253\n",
       "AAAS_NP_056480.1|NP_001166937.1                             -0.16450  0.093375\n",
       "AACS_NP_076417.2|NP_001306769.1|NP_001306768.1               0.07180  0.899917\n",
       "...                                                              ...       ...\n",
       "ZXDC_NP_079388.3|NP_001035743.1                             -0.08945  0.583564\n",
       "ZYG11B_NP_078922.1                                           0.13290  0.124112\n",
       "ZYX_NP_001010972.1                                          -0.47940  0.426235\n",
       "ZZEF1_NP_055928.3                                            0.02700  0.537838\n",
       "ZZZ3_NP_056349.1|NP_001295166.1                              0.05820  0.804589\n",
       "\n",
       "[9452 rows x 2 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#FOR NO DUPLICATES\n",
    "#get the pval and min for no duplicates\n",
    "no_duplicates = br.reduce_multiindex(no_duplicates, flatten=True)\n",
    "\n",
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "mut_type = br.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "prot_and_mutations = mut_type.join(no_duplicates)\n",
    "\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Missense_Mutation']\n",
    "get = prot_and_mutations['Mutation'].isin(compare)\n",
    "missense_wt = prot_and_mutations[get]\n",
    "missense_wt['Mutation'].value_counts()\n",
    "\n",
    "# Step 2 get the difference in medians\n",
    "missense = missense_wt[missense_wt.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = missense_wt[missense_wt.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "no_dup_d = {}\n",
    "\n",
    "for prot in no_duplicates:\n",
    "    dif_brca = wt_med[prot] - missense_med[prot]\n",
    "    no_dup_d[prot] = dif_brca\n",
    "#     print(wt_med[prot], '-', missense_med[prot], '=',dif_endo)\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(no_dup_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#step 3 do a t test for every gene and report the p-value\n",
    "cols = list(missense_wt.columns[1:])\n",
    "t_test = u.wrap_ttest(missense_wt, 'Mutation', cols, return_all = True)\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "# Step 4 join median and p value together\n",
    "no_dup_df = median_diff.join(t_test)\n",
    "no_dup_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 83 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 382)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABCB7_NP_001258628.1|NP_001258626.1</th>\n",
       "      <td>0.28370</td>\n",
       "      <td>0.253549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCB7_NP_004290.2|NP_001258625.1|NP_001258627.1</th>\n",
       "      <td>0.18180</td>\n",
       "      <td>0.641102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABLIM3_NP_001287947.1|NP_001287944.1</th>\n",
       "      <td>-0.33190</td>\n",
       "      <td>0.852467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABLIM3_NP_001287956.1</th>\n",
       "      <td>-0.37175</td>\n",
       "      <td>0.135988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACIN1_NP_001158286.1</th>\n",
       "      <td>0.10270</td>\n",
       "      <td>0.800310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YBX3_NP_003642.3</th>\n",
       "      <td>-0.03920</td>\n",
       "      <td>0.559371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZAK_NP_057737.2</th>\n",
       "      <td>-0.09030</td>\n",
       "      <td>0.780759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZAK_NP_598407.1</th>\n",
       "      <td>-0.21270</td>\n",
       "      <td>0.989303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF185_NP_001171579.1|NP_001171584.1</th>\n",
       "      <td>-0.19150</td>\n",
       "      <td>0.171569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF185_NP_001171580.1|NP_001171578.1|NP_001171581.1</th>\n",
       "      <td>-0.50745</td>\n",
       "      <td>0.687588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>655 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Difference_In_Median  \\\n",
       "ABCB7_NP_001258628.1|NP_001258626.1                              0.28370   \n",
       "ABCB7_NP_004290.2|NP_001258625.1|NP_001258627.1                  0.18180   \n",
       "ABLIM3_NP_001287947.1|NP_001287944.1                            -0.33190   \n",
       "ABLIM3_NP_001287956.1                                           -0.37175   \n",
       "ACIN1_NP_001158286.1                                             0.10270   \n",
       "...                                                                  ...   \n",
       "YBX3_NP_003642.3                                                -0.03920   \n",
       "ZAK_NP_057737.2                                                 -0.09030   \n",
       "ZAK_NP_598407.1                                                 -0.21270   \n",
       "ZNF185_NP_001171579.1|NP_001171584.1                            -0.19150   \n",
       "ZNF185_NP_001171580.1|NP_001171578.1|NP_0011715...              -0.50745   \n",
       "\n",
       "                                                     P_Value  \n",
       "ABCB7_NP_001258628.1|NP_001258626.1                 0.253549  \n",
       "ABCB7_NP_004290.2|NP_001258625.1|NP_001258627.1     0.641102  \n",
       "ABLIM3_NP_001287947.1|NP_001287944.1                0.852467  \n",
       "ABLIM3_NP_001287956.1                               0.135988  \n",
       "ACIN1_NP_001158286.1                                0.800310  \n",
       "...                                                      ...  \n",
       "YBX3_NP_003642.3                                    0.559371  \n",
       "ZAK_NP_057737.2                                     0.780759  \n",
       "ZAK_NP_598407.1                                     0.989303  \n",
       "ZNF185_NP_001171579.1|NP_001171584.1                0.171569  \n",
       "ZNF185_NP_001171580.1|NP_001171578.1|NP_0011715...  0.687588  \n",
       "\n",
       "[655 rows x 2 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#DEALING WITH DUPLICATES\n",
    "#get the pval and min for duplicates\n",
    "\n",
    "duplicates = br.reduce_multiindex(duplicates, flatten=True)\n",
    "\n",
    "# We need to figure which of databaseID we want to use, for each protein\n",
    "proteomics = br.get_proteomics()\n",
    "protfilter = proteomics.index.str.endswith('.N') #filter out normal samples\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "duplicate_gene_names = list(set(duplicates.columns.get_level_values('Name'))) #get a list of genes names that are duplicated\n",
    "duplicates = br.reduce_multiindex(duplicates, flatten=True)\n",
    "\n",
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "\n",
    "mut_type = br.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "prot_and_mutations = mut_type.join(duplicates)\n",
    "\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Missense_Mutation']\n",
    "get = prot_and_mutations['Mutation'].isin(compare)\n",
    "missense_wt = prot_and_mutations[get]\n",
    "missense_wt['Mutation'].value_counts()\n",
    "\n",
    "# Step 2 get the difference in medians\n",
    "missense = missense_wt[missense_wt.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = missense_wt[missense_wt.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "duplicates_d = {}\n",
    "\n",
    "for prot in duplicates:\n",
    "    dif_brca = wt_med[prot] - missense_med[prot]\n",
    "    duplicates_d[prot] = dif_brca\n",
    "#     print(wt_med[prot], '-', missense_med[prot], '=',dif_endo)\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(duplicates_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#step 3 do a t test for every gene and report the p-value\n",
    "cols = list(missense_wt.columns[1:])\n",
    "t_test = u.wrap_ttest(missense_wt, 'Mutation', cols, return_all = True)\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "# Step 4 join median and p value together\n",
    "dup_df = median_diff.join(t_test)\n",
    "dup_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG_NP_570602.2</th>\n",
       "      <td>-0.64290</td>\n",
       "      <td>0.605521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1_NP_653271.2|NP_001269353.1</th>\n",
       "      <td>0.74900</td>\n",
       "      <td>0.070253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M_NP_000005.2</th>\n",
       "      <td>0.26040</td>\n",
       "      <td>0.917772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS_NP_056480.1|NP_001166937.1</th>\n",
       "      <td>-0.16450</td>\n",
       "      <td>0.093375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS_NP_076417.2|NP_001306769.1|NP_001306768.1</th>\n",
       "      <td>0.07180</td>\n",
       "      <td>0.899917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC_NP_079388.3|NP_001035743.1</th>\n",
       "      <td>-0.08945</td>\n",
       "      <td>0.583564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B_NP_078922.1</th>\n",
       "      <td>0.13290</td>\n",
       "      <td>0.124112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX_NP_001010972.1</th>\n",
       "      <td>-0.47940</td>\n",
       "      <td>0.426235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1_NP_055928.3</th>\n",
       "      <td>0.02700</td>\n",
       "      <td>0.537838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3_NP_056349.1|NP_001295166.1</th>\n",
       "      <td>0.05820</td>\n",
       "      <td>0.804589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9771 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Difference_In_Median   P_Value\n",
       "A1BG_NP_570602.2                                            -0.64290  0.605521\n",
       "A2ML1_NP_653271.2|NP_001269353.1                             0.74900  0.070253\n",
       "A2M_NP_000005.2                                              0.26040  0.917772\n",
       "AAAS_NP_056480.1|NP_001166937.1                             -0.16450  0.093375\n",
       "AACS_NP_076417.2|NP_001306769.1|NP_001306768.1               0.07180  0.899917\n",
       "...                                                              ...       ...\n",
       "ZXDC_NP_079388.3|NP_001035743.1                             -0.08945  0.583564\n",
       "ZYG11B_NP_078922.1                                           0.13290  0.124112\n",
       "ZYX_NP_001010972.1                                          -0.47940  0.426235\n",
       "ZZEF1_NP_055928.3                                            0.02700  0.537838\n",
       "ZZZ3_NP_056349.1|NP_001295166.1                              0.05820  0.804589\n",
       "\n",
       "[9771 rows x 2 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#for each duplicated gene name select the one with the min p val and append to no_duplicat df\n",
    "for name in duplicate_gene_names:\n",
    "    gene_duplicate = dup_df[dup_df.index.str.contains(name)]\n",
    "    #select the one with the min p value for t test\n",
    "    min_row = gene_duplicate[gene_duplicate.P_Value == gene_duplicate.P_Value.min()]\n",
    "    #append the selected version onto no_duplicate dataframe\n",
    "    no_dup_df=no_dup_df.append(min_row)\n",
    "no_dup_df=no_dup_df.sort_index()\n",
    "no_dup_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'A2ML1'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#reformat the table by getting rid of databaseIDs so we can join it to Endo and BRca\n",
    "text='A2ML1_NP_653271.2|NP_001269353.1'\n",
    "sep = '_'\n",
    "text.split(sep)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "split = no_dup_df.index.str.split(sep, 1).str[0]\n",
    "no_dup_df['new_index']=split\n",
    "brca_df = no_dup_df.set_index(\"new_index\")\n",
    "# split[\"A\"].str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>new_index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.64290</td>\n",
       "      <td>0.605521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.74900</td>\n",
       "      <td>0.070253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.26040</td>\n",
       "      <td>0.917772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.16450</td>\n",
       "      <td>0.093375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0.07180</td>\n",
       "      <td>0.899917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>-0.08945</td>\n",
       "      <td>0.583564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>0.13290</td>\n",
       "      <td>0.124112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>-0.47940</td>\n",
       "      <td>0.426235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0.02700</td>\n",
       "      <td>0.537838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.05820</td>\n",
       "      <td>0.804589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9771 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Difference_In_Median   P_Value\n",
       "new_index                                \n",
       "A1BG                   -0.64290  0.605521\n",
       "A2ML1                   0.74900  0.070253\n",
       "A2M                     0.26040  0.917772\n",
       "AAAS                   -0.16450  0.093375\n",
       "AACS                    0.07180  0.899917\n",
       "...                         ...       ...\n",
       "ZXDC                   -0.08945  0.583564\n",
       "ZYG11B                  0.13290  0.124112\n",
       "ZYX                    -0.47940  0.426235\n",
       "ZZEF1                   0.02700  0.537838\n",
       "ZZZ3                    0.05820  0.804589\n",
       "\n",
       "[9771 rows x 2 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_df "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon get difference in median and t-test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "cl = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 382)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 382)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:153: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:257: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.15300</td>\n",
       "      <td>0.873969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.33610</td>\n",
       "      <td>0.021796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.07100</td>\n",
       "      <td>0.971735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.04400</td>\n",
       "      <td>0.557351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>-0.03700</td>\n",
       "      <td>0.452855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZW10</th>\n",
       "      <td>0.00230</td>\n",
       "      <td>0.756621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZWILCH</th>\n",
       "      <td>0.18430</td>\n",
       "      <td>0.233675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZWINT</th>\n",
       "      <td>-0.32775</td>\n",
       "      <td>0.546515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>-0.13700</td>\n",
       "      <td>0.921572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>-0.01600</td>\n",
       "      <td>0.391801</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8067 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median   P_Value\n",
       "A1BG                 0.15300  0.873969\n",
       "A1CF                 0.33610  0.021796\n",
       "A2M                  0.07100  0.971735\n",
       "AAAS                -0.04400  0.557351\n",
       "AACS                -0.03700  0.452855\n",
       "...                      ...       ...\n",
       "ZW10                 0.00230  0.756621\n",
       "ZWILCH               0.18430  0.233675\n",
       "ZWINT               -0.32775  0.546515\n",
       "ZYX                 -0.13700  0.921572\n",
       "ZZEF1               -0.01600  0.391801\n",
       "\n",
       "[8067 rows x 2 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "proteomics = cl.get_proteomics()\n",
    "protfilter = proteomics.index.str.endswith('.N')\n",
    "proteomics = proteomics[~protfilter]\n",
    "\n",
    "prot_list = list(proteomics.columns)\n",
    "\n",
    "mut_type = cl.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "prot_and_mutations = mut_type.join(proteomics)\n",
    "\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','nonsynonymous SNV']\n",
    "get = prot_and_mutations['Mutation'].isin(compare)\n",
    "missense_wt = prot_and_mutations[get]\n",
    "missense_wt['Mutation'].value_counts()\n",
    "\n",
    "# Step 2 get the difference in medians\n",
    "missense = missense_wt[missense_wt.Mutation == \"nonsynonymous SNV\"] #get all missense_mutations\n",
    "wt = missense_wt[missense_wt.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "colon_d = {}\n",
    "\n",
    "for prot in proteomics:\n",
    "    dif_colon = wt_med[prot] - missense_med[prot]\n",
    "    colon_d[prot] = dif_colon\n",
    "#     print(wt_med[prot], '-', missense_med[prot], '=',dif_endo)\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(colon_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#step 3 do a t test for every gene and report the p-value\n",
    "cols = list(missense_wt.columns[1:])\n",
    "t_test = u.wrap_ttest(missense_wt, 'Mutation', cols, return_all = True)\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "# Step 4 join median and p value together\n",
    "colon_df = median_diff.join(t_test)\n",
    "colon_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine Endo, Brca, and Colon into one table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "brca_df=brca_df.rename_axis(None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median_Brca</th>\n",
       "      <th>P_Value_Brca</th>\n",
       "      <th>Difference_In_Median_Endo</th>\n",
       "      <th>P_Value_Endo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.64290</td>\n",
       "      <td>0.605521</td>\n",
       "      <td>-0.07100</td>\n",
       "      <td>0.720461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.26040</td>\n",
       "      <td>0.917772</td>\n",
       "      <td>-0.10300</td>\n",
       "      <td>0.681688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.74900</td>\n",
       "      <td>0.070253</td>\n",
       "      <td>-0.47135</td>\n",
       "      <td>0.136212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.16450</td>\n",
       "      <td>0.093375</td>\n",
       "      <td>-0.07410</td>\n",
       "      <td>0.295868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0.07180</td>\n",
       "      <td>0.899917</td>\n",
       "      <td>0.03900</td>\n",
       "      <td>0.734326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>-0.08945</td>\n",
       "      <td>0.583564</td>\n",
       "      <td>-0.08700</td>\n",
       "      <td>0.808610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>0.13290</td>\n",
       "      <td>0.124112</td>\n",
       "      <td>0.03240</td>\n",
       "      <td>0.755896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>-0.47940</td>\n",
       "      <td>0.426235</td>\n",
       "      <td>0.13800</td>\n",
       "      <td>0.251409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0.02700</td>\n",
       "      <td>0.537838</td>\n",
       "      <td>0.00210</td>\n",
       "      <td>0.398421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.05820</td>\n",
       "      <td>0.804589</td>\n",
       "      <td>0.01570</td>\n",
       "      <td>0.978939</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9771 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median_Brca  P_Value_Brca  Difference_In_Median_Endo  \\\n",
       "A1BG                     -0.64290      0.605521                   -0.07100   \n",
       "A2M                       0.26040      0.917772                   -0.10300   \n",
       "A2ML1                     0.74900      0.070253                   -0.47135   \n",
       "AAAS                     -0.16450      0.093375                   -0.07410   \n",
       "AACS                      0.07180      0.899917                    0.03900   \n",
       "...                           ...           ...                        ...   \n",
       "ZXDC                     -0.08945      0.583564                   -0.08700   \n",
       "ZYG11B                    0.13290      0.124112                    0.03240   \n",
       "ZYX                      -0.47940      0.426235                    0.13800   \n",
       "ZZEF1                     0.02700      0.537838                    0.00210   \n",
       "ZZZ3                      0.05820      0.804589                    0.01570   \n",
       "\n",
       "        P_Value_Endo  \n",
       "A1BG        0.720461  \n",
       "A2M         0.681688  \n",
       "A2ML1       0.136212  \n",
       "AAAS        0.295868  \n",
       "AACS        0.734326  \n",
       "...              ...  \n",
       "ZXDC        0.808610  \n",
       "ZYG11B      0.755896  \n",
       "ZYX         0.251409  \n",
       "ZZEF1       0.398421  \n",
       "ZZZ3        0.978939  \n",
       "\n",
       "[9771 rows x 4 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_endo = brca_df.join(endo_df, lsuffix='_Brca', rsuffix='_Endo')\n",
    "brca_endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median_Brca</th>\n",
       "      <th>P_Value_Brca</th>\n",
       "      <th>Difference_In_Median_Endo</th>\n",
       "      <th>P_Value_Endo</th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.64290</td>\n",
       "      <td>0.605521</td>\n",
       "      <td>-0.07100</td>\n",
       "      <td>0.720461</td>\n",
       "      <td>0.153</td>\n",
       "      <td>0.873969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.26040</td>\n",
       "      <td>0.917772</td>\n",
       "      <td>-0.10300</td>\n",
       "      <td>0.681688</td>\n",
       "      <td>0.071</td>\n",
       "      <td>0.971735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.74900</td>\n",
       "      <td>0.070253</td>\n",
       "      <td>-0.47135</td>\n",
       "      <td>0.136212</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.16450</td>\n",
       "      <td>0.093375</td>\n",
       "      <td>-0.07410</td>\n",
       "      <td>0.295868</td>\n",
       "      <td>-0.044</td>\n",
       "      <td>0.557351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0.07180</td>\n",
       "      <td>0.899917</td>\n",
       "      <td>0.03900</td>\n",
       "      <td>0.734326</td>\n",
       "      <td>-0.037</td>\n",
       "      <td>0.452855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>-0.08945</td>\n",
       "      <td>0.583564</td>\n",
       "      <td>-0.08700</td>\n",
       "      <td>0.808610</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>0.13290</td>\n",
       "      <td>0.124112</td>\n",
       "      <td>0.03240</td>\n",
       "      <td>0.755896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>-0.47940</td>\n",
       "      <td>0.426235</td>\n",
       "      <td>0.13800</td>\n",
       "      <td>0.251409</td>\n",
       "      <td>-0.137</td>\n",
       "      <td>0.921572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0.02700</td>\n",
       "      <td>0.537838</td>\n",
       "      <td>0.00210</td>\n",
       "      <td>0.398421</td>\n",
       "      <td>-0.016</td>\n",
       "      <td>0.391801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.05820</td>\n",
       "      <td>0.804589</td>\n",
       "      <td>0.01570</td>\n",
       "      <td>0.978939</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9771 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median_Brca  P_Value_Brca  Difference_In_Median_Endo  \\\n",
       "A1BG                     -0.64290      0.605521                   -0.07100   \n",
       "A2M                       0.26040      0.917772                   -0.10300   \n",
       "A2ML1                     0.74900      0.070253                   -0.47135   \n",
       "AAAS                     -0.16450      0.093375                   -0.07410   \n",
       "AACS                      0.07180      0.899917                    0.03900   \n",
       "...                           ...           ...                        ...   \n",
       "ZXDC                     -0.08945      0.583564                   -0.08700   \n",
       "ZYG11B                    0.13290      0.124112                    0.03240   \n",
       "ZYX                      -0.47940      0.426235                    0.13800   \n",
       "ZZEF1                     0.02700      0.537838                    0.00210   \n",
       "ZZZ3                      0.05820      0.804589                    0.01570   \n",
       "\n",
       "        P_Value_Endo  Difference_In_Median   P_Value  \n",
       "A1BG        0.720461                 0.153  0.873969  \n",
       "A2M         0.681688                 0.071  0.971735  \n",
       "A2ML1       0.136212                   NaN       NaN  \n",
       "AAAS        0.295868                -0.044  0.557351  \n",
       "AACS        0.734326                -0.037  0.452855  \n",
       "...              ...                   ...       ...  \n",
       "ZXDC        0.808610                   NaN       NaN  \n",
       "ZYG11B      0.755896                   NaN       NaN  \n",
       "ZYX         0.251409                -0.137  0.921572  \n",
       "ZZEF1       0.398421                -0.016  0.391801  \n",
       "ZZZ3        0.978939                   NaN       NaN  \n",
       "\n",
       "[9771 rows x 6 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_table = brca_endo.join(colon_df, rsuffix='_colon')\n",
    "final_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# final_table.to_csv(\"proteomics.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# endo_and_colon = pd.concat([brca_df, endo_and_colon], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# final_table = pd.concat([endo_and_colon, brca_df], axis=1, sort=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# final_table"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
