{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PTEN loss Survival Analysis for HNSCC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import cptac\n",
    "import cptac.utils as ut\n",
    "import re\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import scipy\n",
    "import lifelines\n",
    "from lifelines import KaplanMeierFitter\n",
    "from lifelines import CoxPHFitter\n",
    "from lifelines.statistics import proportional_hazard_test\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import warnings\n",
    "#warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Params\n",
    "cancer_object: Object. The loaded cancer data set.\n",
    "all_prot: List. trans proteins used to get proteomics data. \n",
    "gene_in: String. Gene used to get mutation status for the Mutation col.\n",
    "utils: utils package from cptac.  \n",
    "\n",
    "Returns a dataframe with trans proteomics and mutation status of a specific gene.\n",
    "'''\n",
    "\n",
    "def all_prot_format_df(cancer_object, all_prot, gene_in = 'PTEN', utils = ut):\n",
    "    mut_type = cancer_object.get_genotype_all_vars(gene_in)\n",
    "    \n",
    "    if cancer_object.get_cancer_type() not in ('luad', 'endometrial'):\n",
    "        # Keep only tumor samples from proteomics\n",
    "        prot_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "            mutations_genes = [gene_in], omics_df_name = 'proteomics', omics_genes = all_prot, tissue_type = 'tumor')\n",
    "        # Reduce a multiindex \n",
    "        if isinstance(prot_and_mutations.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "            prot_and_mutations = utils.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "            prot_and_mutations = rename_duplicate_cols(prot_and_mutations)\n",
    "        \n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        mut_type = mut_type[['Mutation']] \n",
    "        prot_df = prot_and_mutations.iloc[:,:-4] # Keep only proteomics\n",
    "        merged = prot_df.join(mut_type, how = 'outer')\n",
    "        \n",
    "        # Keep only Wildtype and deletion\n",
    "        compare = ['Wildtype_Tumor','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "    \n",
    "    \n",
    "    # Luad has no somatic mutations for PTEN which changes some things\n",
    "    elif cancer_object.get_cancer_type() in ('luad'): \n",
    "        # get_genotype_all_vars add cnv data under the column PTEN\n",
    "        mut_type = mut_type.drop(columns= gene_in)\n",
    "        # different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "        omics = cancer_object.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics',genes1= gene_in, \n",
    "            genes2= all_prot, tissue_type = 'tumor')\n",
    "        omics = utils.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "        omics = rename_duplicate_cols(omics)\n",
    "        omics = omics.drop(columns='PTEN_CNV')\n",
    "       \n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        merged = omics.join(mut_type, how = 'outer') \n",
    "        # Keep only No_Mutation (wildtype) and deletion\n",
    "        compare = ['No_Mutation','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "        mut_wt['Mutation'] = np.where(\n",
    "            mut_wt['Mutation'] == 'No_Mutation', 'Wildtype_Tumor', 'Deletion')\n",
    "        \n",
    "    elif cancer_object.get_cancer_type() in ('endometrial'):\n",
    "        # merge cnv with genotype all mut type\n",
    "        prot_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "            mutations_genes = [gene_in], omics_df_name = 'proteomics', omics_genes = all_prot, tissue_type = 'tumor') # drop Normal samples\n",
    "\n",
    "        prot_df = prot_and_mutations.iloc[:,:-4] #drop Mutation and Location cols\n",
    "        mut_type = mut_type[['Mutation']] # Get Mutation col that includes CNV\n",
    "        merged = prot_df.join(mut_type, how = 'outer') # merge \n",
    "\n",
    "        # Create Truncation category and keep truncation and wt\n",
    "        compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "        mut_wt['Mutation'] = np.where(\n",
    "                        mut_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "    \n",
    "    mut_wt = mut_wt.rename(columns={col: re.sub(r'_proteomics', '', col) for col in mut_wt.columns.tolist()})\n",
    "    return mut_wt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "''' Adds '_i' to duplicate col names where the int i increases by one with each duplicate \n",
    "Returns a df with unique column names. '''\n",
    "\n",
    "def rename_duplicate_cols(df):\n",
    "    cols = pd.Series(df.columns[:])\n",
    "\n",
    "    for dup in cols[cols.duplicated()].unique(): \n",
    "        cols[cols[cols == dup].index.values.tolist()] = [dup + '_' + str(i) if i != 0 else dup for i in range(sum(cols == dup))]\n",
    "\n",
    "    # rename the columns with the cols list.\n",
    "    df.columns=cols\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_survival_cols(cancer_object):\n",
    "    cancer = cancer_object.get_cancer_type()\n",
    "    #print(cancer)\n",
    "    \n",
    "    # get unique survival col names\n",
    "    if cancer == 'gbm':\n",
    "        cols = ['vital_status', 'path_diag_to_last_contact_days', 'path_diag_to_death_days']\n",
    "    if cancer == 'hnscc':\n",
    "        cols = ['VITAL_STATUS', 'PATHO_DIAG_TO_LAST_CONT_DAYS', 'PATHO_DIAG_TO_DEATH_DAYS']\n",
    "    if cancer == 'luad':\n",
    "        cols = ['Vital Status', 'Path Diag to Last Contact(Day)', 'Path Diag to Death(days)']\n",
    "    if cancer == 'ovarian':\n",
    "        cols = ['Vital_Status', 'Days_Between_Collection_And_Last_Contact', 'Days_Between_Collection_And_Death']\n",
    "    if cancer == 'endometrial':\n",
    "        cols = ['Vital Status', 'Path Diag to Last Contact(Day)', 'Path Diag to Death(days)']\n",
    "    \n",
    "    # lscc no follow up or clin matching survival cols\n",
    "\n",
    "    if cancer == 'brca':\n",
    "        cols = ['Vital Status (at time of last contact)'] \n",
    "        print('Use csv')\n",
    "    if cancer == 'colon':\n",
    "        cols = ['Vital Status (at time of last contact)'] \n",
    "        print('Use csv')\n",
    "\n",
    "    return cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_most_recent_last_contact_df(survival_in):\n",
    "    d = {}\n",
    "    df = survival_in[['days_to_last_contact']]\n",
    "    index_list = list(df.index.unique())\n",
    "    for i in index_list:\n",
    "        one_index_df = df.loc[[i]]\n",
    "        # Get the latest number when duplicate records\n",
    "        most_recent = one_index_df['days_to_last_contact'].max() # get most recent (highest num)\n",
    "        d[i] = most_recent\n",
    "        \n",
    "    last_contact_df = pd.DataFrame.from_dict(d, orient='index', columns=['days_to_last_contact'])\n",
    "    last_contact_df.index.name = 'Patient_ID'\n",
    "    \n",
    "    return last_contact_df   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 1: Import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The HNSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    }
   ],
   "source": [
    "cancer = 'HNSCC'\n",
    "ca = cptac.Hnscc()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "followup\n",
      " Living      128\n",
      "Deceased     12\n",
      "Name: vital_status, dtype: int64 \n",
      "\n",
      "clin\n",
      " Living      87\n",
      "Deceased    14\n",
      "Name: follow_up_vital_status, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "cols = get_survival_cols(ca)\n",
    "\n",
    "f = ca.get_followup(tissue_type = 'tumor')\n",
    "df2 = f[cols]\n",
    "df3 = df2.rename(columns = {cols[0]: 'vital_status', cols[1]: 'days_to_last_contact', cols[2]: 'days_to_death'})\n",
    "print('followup\\n', df3.vital_status.value_counts(), '\\n')\n",
    "\n",
    "clin = ca.get_clinical(tissue_type = 'tumor')\n",
    "print('clin\\n', clin.follow_up_vital_status.value_counts())\n",
    "cd = clin[['follow_up_days', 'follow_up_vital_status']]\n",
    "cdf = cd.rename(columns = {'follow_up_days': 'days_to_last_contact_or_death', 'follow_up_vital_status': 'vital_status'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "num death events in followup: 12\n",
      "num death events in clin: 14\n"
     ]
    }
   ],
   "source": [
    "fd = df3.loc[df3['vital_status'] == 'Deceased']\n",
    "followup_list = list(fd.index)\n",
    "print('num death events in followup:', len(followup_list))\n",
    "\n",
    "cd = cdf.loc[cdf['vital_status'] == 'Deceased']\n",
    "clin_list = list(cd.index)\n",
    "print('num death events in clin:', len(clin_list))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find samples in clin and not in followup (with death events)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['C3N-00828', 'C3N-03009']"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find samples in clin and not in followup (with death events)\n",
    "only_in_clin = [e for e in clin_list if e not in followup_list]\n",
    "len(only_in_clin)\n",
    "only_in_clin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "clin\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>follow_up_days</th>\n",
       "      <th>follow_up_vital_status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3N-00828</th>\n",
       "      <td>951.0</td>\n",
       "      <td>Deceased</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        follow_up_days follow_up_vital_status\n",
       "Patient_ID                                       \n",
       "C3N-00828            951.0               Deceased"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = 'C3N-00828'\n",
    "#a = 'C3N-03009'\n",
    "print('clin')\n",
    "cd.loc[[a]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "followup\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>VITAL_STATUS</th>\n",
       "      <th>PATHO_DIAG_TO_LAST_CONT_DAYS</th>\n",
       "      <th>PATHO_DIAG_TO_DEATH_DAYS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3N-00828</th>\n",
       "      <td>Living</td>\n",
       "      <td>393</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00828</th>\n",
       "      <td>Living</td>\n",
       "      <td>635</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00828</th>\n",
       "      <td>Living</td>\n",
       "      <td>826</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       VITAL_STATUS  PATHO_DIAG_TO_LAST_CONT_DAYS  \\\n",
       "Patient_ID                                              \n",
       "C3N-00828        Living                           393   \n",
       "C3N-00828        Living                           635   \n",
       "C3N-00828        Living                           826   \n",
       "\n",
       "Name        PATHO_DIAG_TO_DEATH_DAYS  \n",
       "Patient_ID                            \n",
       "C3N-00828                        NaN  \n",
       "C3N-00828                        NaN  \n",
       "C3N-00828                        NaN  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('followup')\n",
    "df2.loc[a]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check the 2 samples with negative days"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "followup\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>VITAL_STATUS</th>\n",
       "      <th>PATHO_DIAG_TO_LAST_CONT_DAYS</th>\n",
       "      <th>PATHO_DIAG_TO_DEATH_DAYS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00995</th>\n",
       "      <td>Deceased</td>\n",
       "      <td>-10</td>\n",
       "      <td>-9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00299</th>\n",
       "      <td>Deceased</td>\n",
       "      <td>-15</td>\n",
       "      <td>294.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       VITAL_STATUS  PATHO_DIAG_TO_LAST_CONT_DAYS  \\\n",
       "Patient_ID                                              \n",
       "C3L-00995      Deceased                           -10   \n",
       "C3N-00299      Deceased                           -15   \n",
       "\n",
       "Name        PATHO_DIAG_TO_DEATH_DAYS  \n",
       "Patient_ID                            \n",
       "C3L-00995                       -9.0  \n",
       "C3N-00299                      294.0  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check samples with neg days to death\n",
    "a = ['C3L-00995', 'C3N-00299']\n",
    "print('followup')\n",
    "df2.loc[a]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "clin\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>follow_up_days</th>\n",
       "      <th>follow_up_vital_status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00995</th>\n",
       "      <td>-10.0</td>\n",
       "      <td>Deceased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00299</th>\n",
       "      <td>-15.0</td>\n",
       "      <td>Deceased</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        follow_up_days follow_up_vital_status\n",
       "Patient_ID                                       \n",
       "C3L-00995            -10.0               Deceased\n",
       "C3N-00299            -15.0               Deceased"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('clin')\n",
    "cd.loc[a]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepare Data for Multivariate Kaplan Meier Plots and Cox's Proportional Hazard Test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Living      87\n",
      "Deceased    14\n",
      "Name: follow_up_vital_status, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "clin = ca.get_clinical(tissue_type = 'tumor')\n",
    "print(clin.follow_up_vital_status.value_counts())\n",
    "df = clin[['follow_up_days', 'follow_up_vital_status']]\n",
    "df = df.rename(columns = {'follow_up_days': 'days_to_last_contact_or_death', 'follow_up_vital_status': 'vital_status'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_asarray.py:83: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray\n",
      "  return array(a, dtype, copy=False, order=order)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 107 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 17)\n"
     ]
    }
   ],
   "source": [
    "del_wt = all_prot_format_df(ca, ['PTEN']) # get Mutation col (PTEN del/wt)\n",
    "mut_survival = df.join(del_wt, on = 'Patient_ID', how = 'left') # combine keeping only turmors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "99"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop row with NaN in vital status : (99 rows)\n",
    "mut_survival = mut_survival[mut_survival['vital_status'].notna()]\n",
    "mut_survival = mut_survival[mut_survival['days_to_last_contact_or_death'].notna()]\n",
    "mut_survival = mut_survival[mut_survival['Mutation'].notna()]\n",
    "len(mut_survival)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    85\n",
       "True     14\n",
       "Name: vital_status, dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Make the vital_status column boolean\n",
    "mut_survival['vital_status'] = mut_survival['vital_status'].replace('Living', False) # deceased = True\n",
    "mut_survival['vital_status'] = mut_survival['vital_status'].astype('bool') # careful: nan vals will change to bool\n",
    "df_genes = mut_survival\n",
    "\n",
    "mut_survival.vital_status.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# map numeric values to correspond with mutation status\n",
    "mutation_map = {\"Wildtype_Tumor\" : 0, \"Deletion\" : 1}\n",
    "\n",
    "df_genes['PTEN'] = df_genes['Mutation'].map(mutation_map)\n",
    "\n",
    "#Then we will drop missing values, as missing values \n",
    "#will throw an error in the functions we will use in lifelines.\n",
    "df_clean = df_genes.dropna(axis=0, how='any').copy()\n",
    "df_clean = df_clean.drop(columns= 'Mutation') # drop col with str values (info mapped to PTEN as int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    85\n",
      "True     14\n",
      "Name: vital_status, dtype: int64\n",
      "0    82\n",
      "1    17\n",
      "Name: PTEN, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(df_clean.vital_status.value_counts())\n",
    "print(df_clean.PTEN.value_counts())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Verify that your columns are the correct data types.  The event of interest, in this case Vital_Status needs to contain boolean values, and all other columns in the table must be of a numeric type (either int64 or float64)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "days_to_last_contact_or_death : float64\n",
      "vital_status : bool\n",
      "PTEN : int64\n"
     ]
    }
   ],
   "source": [
    "for col in df_clean.columns:\n",
    "    print(col, \":\", df_clean[col].dtype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check mirroring"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "wt = df_clean.loc[df_clean['PTEN'] == 0]\n",
    "ch = {'test': [1200, False, 1], 't2': [800, True, 1], 't3':[200, True, 1], 't4': [300, True, 1], 't5': [600, True, 1]}\n",
    "cols = ['days_to_last_contact_or_death', 'vital_status', 'PTEN']\n",
    "test = pd.DataFrame.from_dict(ch, orient = 'index', columns = cols)\n",
    "tdf = wt.append(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>days_to_last_contact_or_death</th>\n",
       "      <th>vital_status</th>\n",
       "      <th>PTEN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>test</th>\n",
       "      <td>1200</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t2</th>\n",
       "      <td>800</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t3</th>\n",
       "      <td>200</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t4</th>\n",
       "      <td>300</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t5</th>\n",
       "      <td>600</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>test2</th>\n",
       "      <td>1200</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d2</th>\n",
       "      <td>800</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d3</th>\n",
       "      <td>200</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d4</th>\n",
       "      <td>300</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       days_to_last_contact_or_death  vital_status  PTEN\n",
       "test                            1200         False     1\n",
       "t2                               800          True     1\n",
       "t3                               200          True     1\n",
       "t4                               300          True     1\n",
       "t5                               600          True     1\n",
       "test2                           1200          True     0\n",
       "d2                               800          True     0\n",
       "d3                               200          True     0\n",
       "d4                               300          True     0"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ch = {'test2': [1200, True, 0], 'd2': [800, True, 0], 'd3':[200, True, 0], 'd4': [300, True, 0]}\n",
    "cols = ['days_to_last_contact_or_death', 'vital_status', 'PTEN']\n",
    "test2 = pd.DataFrame.from_dict(ch, orient = 'index', columns = cols)\n",
    "tdf = test.append(test2)\n",
    "tdf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 5: Plot Multivariate Survival Risk"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "0 is wt. 1 is mutant."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x21f974a8048>"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAEICAYAAABPgw/pAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXhU1f3H8feXXQ2LsgkEBBFFBAQMKio14oZWxT7Vn9SVtkqxRdta16otdStqf/5al8qDrVW0Cq0rtajUhaIWlCCIrBJAIAHZZDFAkMD398e9oUOckJmQyZ2ZfF7Pk4e595577vdMwnfOPffOuebuiIhI5qsXdQAiIlIzlNBFRLKEErqISJZQQhcRyRJK6CIiWUIJXUQkSyihS8YzszFmdmcN1POUmd1TzX07mVmJmdWvotwwM3u/ehGmDzObYmZXRx2H7E0JPc2Z2edmdkaFdXslhbDMGjM7KGbd1WY2JWZ5iJnNNrMtZrbezN42s84x2480s7+H2zab2Rwzu6E8QZlZIzMbZWaLzWxreMwnY+uoEOMpZvafsK4vzewDM+tfQ2/LXtx9hLvfnYq6y4Xv+a4waW8xs0/M7LyYGFa4e46770phDKPMbGcYw6bw/R1gZr8M15WYWWlMnCVmNi/c18PfW0nMz80x9bqZXRxzrAbhus6pao/UPCX07NEA+Gm8DWZ2BDAO+AXQHOgC/BHYHW7vCnwIrAR6uXtz4GIgD2gaVvMCcAFwaVjHscBM4PQ4x2sGvAY8AhwCdAB+A+xItlEWSJe/02nungO0IHj/xptZi1qOYUIYQ2vgfeAl4Lfhh0kOMKI8zvDnmJh9j41Zn+PuD8Rs+xK4q6ozDElv6fIfRfbfg8CNlSSYPsAyd3/bA1+5+4vuviLc/hvgP+5+g7uvBnD3Re5+qbtvCs8QzgSGuPsMdy9z983u/pi7/znO8Y4M63je3Xe5+3Z3n+zuc2BPj/DZ8sJm1jnsDTYIl6eY2b1m9gGwDfilmRXEHsDMfm5mE8PXe4ZKzGxBbM857GmuN7N+4fLfzeyL8MxhqpnFJryEuPtu4BngIKBbJW0YZmZLzewrM1tmZpfFq8vMHjSz982seZIx7ASeBg4FWibbhjjeAL4GLk92RzOrZ2Z3mNlyM1trZuPK22NmTczsWTPbEJ5VzDCztuG2hN4jSZwSevYoAKYAN8bZ9jHQ3cz+z8xOM7OcCtvPIOiBV+YM4CN3X5lgLJ8Bu8zsaTM7x8wOTnC/WFcAwwnOEB4BjjKzbjHbLwWei7Pf88D3YpbPBta7+8fh8usESbgNwfvy12QDC3ux3wd2AsvjbD8IeBg4x92bAicBsyuUqWdmTwC9gbPcfbMF4/CbzKxTAjE0BoYBRe6+Ptk2xOHAncCvzaxhkvsOC39OAw4HcoBHw21XEZzRdST44BkBbE/kPZLkKaFnhlfC/+ibzGwTwel+PL8CrjOz1rEr3X0pkE8w9PE3YH3Yqy1P7C2B1fs4flXb9+LuW4BTCJLEE8A6M5tY3jNL0FPuPq/8bAB4lTBRh4m9OzAxzn7PAReY2YHh8l6J392fDM9QdgCjgGOT6B2fGL7/pcDvgMvdfW0lZXcDPc3sAHdf7e7zYrY1JPjgOQQ43923hbGtcPcWMWdO8fxPGMNK4DjgwgRjB/g49u/IzM6O3ejuE4F1QLIXOy8DHnL3pe5eAtwGDA3PVnYS/P0cEZ6tzQz/PmDf75FUgxJ6Zrgw/I/ewt1bAD+OV8jd5xKMXd8aZ9t0d/8fd28NDAS+Bdwebt4AtNvH8avaHi+WBe4+zN1zgZ5Ae+D3SVRR8WzgOf7b874UeKU8EVY4biGwADg/TOoXhPtiZvXNbLSZLTGzLcDn4W6tEoxpevj+H0zwYTIwXiF33wpcQtAbXW1m/zSz7jFFjgCGAL9x968TPHa5v4V/B23cfZC7z0xi336xf0fu/macMncQ/F00SaLe9ux9prKc4JpOW4KhqTcJrjesMrMHzKxhAu+RVIMSevb5NXANQW88LnefQXAxrWe46i3gu/uo8y3geDPLrU5A7r4QeCrmeFuBA2OKHBpvtwrLk4FWZtaHILHHG24pVz7sMgSYHyZ5CD4IhhAMITUHOofrLZF27Aks6IX+GLjCzPpWUuZNdz+T4INwIcGZSrkFBEM2r5vZUckcO9Xc/V9AIZV0GiqxCjgsZrkTUAascfed7v4bd+9BMKxyHnBleKx9vUdSDUroWSZMXhOA68vXWXAL4TVm1iZc7k7Qc50eFvk1cFJ4ge7QsMwR4cWsFu7+FvAv4GUzOy680NjUzEaY2Q8qxmBm3c3sF+UfAGbWkSDBlh9vNvCtcMy4OcEpelXtKiMY53+QYKjiX/soPh44C7iWvRN/U4I7bTYQfKDcV9Vx9xHPBuBPBMNcezGztmZ2QThOvAMoAXZV2P954JfAWxbcZZRObgduTqL888DPzaxLOIx3H8HdOGXhNZte4XWHLQRDMLsSeY8keUro2ekugjswym0iSOCfmlkJwR0NLwMPALj7EmAAQY91npltBl4kuND6VVjHRcAkgg+LzcBcgtsa34pz/K+AE4APzWwrQSKfS3DbZHkvcAIwh+DWx9cSbNdzBL3rv4cJPq7wTp1pBD3CCTGbxhEMBxQD8/nvB0x1/R4418x6V1hfj6CtqwhuBzyVOD1ed3+a4Hf1TniXTPmXk6q8KFpNn9je96HHHQJz9w+Aj5Ko90mCoZWpwDKCawzXhdsOJfgg3kJwZvJv4FkSfI8kOaYHXIiIZAf10EVEsoQSuohIllBCFxHJEkroIiJZokFUB27VqpV37tw5qsOLiGSkmTNnrg+/IPgNkSX0zp07U1BQUHVBERHZw8y+MX9QOQ25iIhkCSV0EZEsoYQuIpIlIhtDF5G6Y+fOnRQVFVFaWhp1KBmjSZMm5Obm0rBh4tPTK6GLSMoVFRXRtGlTOnfujFlSk1vWSe7Ohg0bKCoqokuXLgnvpyEXEUm50tJSWrZsqWSeIDOjZcuWSZ/RKKGLSK1QMk9Odd6vjBtymXfDvVzxl4c5t11Xvt2uK2W7d/PT2W9xfvsjGHzo4ZTuKuMXn7zDdzocyRltO1NS9jW3zJnCxbndyW/TiU1fl3L73Kl8r1MPTmmVy4Yd2/nVvPe44rCenNiyPWtKt3LX/A8Y1rkX/Q9pR/H2r7hvwTSu7nIsfQ9uS4eh59HpmkuifhtERL5BPfQkzPt4Fv/5c9LPFBaRNFC/fn369OlDz549ufjiiykuLqZPnz706dOHQw89lA4dOuxZ/vrrr/eUL/8ZPXo0APn5+eTl5e2pt6CggPz8/ITjmDlzJr169eKII47g+uuvpyanMM+4HvoxD93Oxw/dvte6jyuUqbh8ZoXlcyosn1dhueJTdy8K/+13cFtYtI6hiQQqImnlgAMOYPbs2QBcdtllTJgwYc/yqFGjyMnJ4cYbb4xbvqK1a9fy+uuvc845FbNJ1a699lrGjh3LiSeeyLnnnssbb7xRrXriSaiHbmaDzWyRmRWa2TceQGxmzc3sH2b2iZnNM7Pv10h0aWZE176M6Br3EZIikkEGDhxIYWFh1QUrcdNNN3HPPfckvd/q1avZsmULAwYMwMy48soreeWVV6odR0VV9tDDZwE+RtDRLQJmmNlEd58fU+wnBA/jPd/MWgOLzOyv1XiieVrr1TzufDgikoQ/PFHI4qUlNVpnt8Nz+Ok1RyRUtqysjNdff53Bgwfvs9z27dvp06fPnuXbbruNSy4Jrp8NGDCAl19+mXfffZemTZvuKbNo0aI9ZSqaMmUKxcXF5Ob+91nrubm5FBcXJxR3IhIZcjkeKHT3pQBmNp7waeoxZRxoasFl2RyCZwRW+szHTLWkZBMQPHxTRDJLbIIeOHAgP/zhD/dZfl9DLgB33HEH99xzD/fff/+edUcdddQ+94k3Xl6Td/8kktA7ACtjlosIHgAc61FgIsEDX5sCl7j77ooVmdlwYDhAp06peg5u6jz0WfDc3MsjjkMkkyXak65pVSXoZA0aNIg777yT6dP/+6zxqnroubm5FBUV7VlXVFRE+/btayymRBJ6vI+Pih8zZwOzgUFAV+BfZvaeu2/Zayf3scBYgLy8vIx7OvVPjugXdQgikkZuv/12RowYweGHHw5U3UNv0aIFTZs2Zfr06ZxwwgmMGzeO6667rsbiSSShFwEdY5ZzCXrisb4PjPbgfKLQzJYB3YGPaiTKNNGjWauoQxCRWlJxDH3w4MF7bl0sd+6559K6dXLX1h5//HGGDRvG9u3bOeecc2rsDhdILKHPALqZWRegGBgKXFqhzArgdOA9M2sLHAUsrbEo08RnX30JaAxdJBOVlFR+IXbUqFHfWLdr1664ZadMmbLX8syZM5OKIy8vj7lz5ya1T6KqTOjuXmZmI4E3gfrAk+4+z8xGhNvHAHcDT5nZpwRDNLe4+/pUBJyKK+SJenvOhzRoUI+rIjm6iMi+JfTFInefBEyqsG5MzOtVwFk1G1r6+W7OMTRupC/Xikh6yrhvikZ1hRxg3PhDIju2iEhVMi6hR2npjo1RhyAiUikl9CRM+DK4kDEq2jBEROJSQk/CFS2PjToEEZFK6QpfEnIbNSO3UbOowxCRakiX6XNvv/12OnbsSE5OTk03UT30ZCwu3RB1CCJSTekyfe7555/PyJEj6datWzVasW/qoSfhhY3zeWHj/KoLikhai2r6XIATTzyRdu3aVfvY+6IeehKGtepTdSER2ad5N9zLlk8W1midzY7tzjEVHnxTmSinz23RokVCMVaXEnoS2jVsWnUhEUlL6TB9bqopoSdhYWlKZjMQqVMS7UnXtHSYPlc99DTy8sYFANwXcRwikh6SnT431XRRNAk/bNWPH7bSnOgidUH5EE35z623fuNxytWaPvfmm28mNzeXbdu2kZubG3emx+qyeI9Eqg15eXleUFAQybGra1yX7wBw5bKXI45EJLMsWLCAo48+OuowMk68983MZrp7XrzyGnJJwrzta6MOQUSkUkroSZi4aREA91dRTkQkCkroSRje+rioQxARqZQSehJaNjgw6hBERCqlhJ6EOdvWRB2CiEillNCT8M/NnwHwu4jjEBGJR/ehJ+Ha1v25tnX/qMMQkWr4/PPP6dmzZ0rqnjJlCueddx4AEydO3DPVbm1TDz0JLRo0iToEEUlzF1xwARdccEEkx06oh25mg81skZkVmtk3vi5lZjeZ2ezwZ66Z7TKzrHui8qxtq5m1bXXUYYhINZWVlXHVVVfRu3dvLrroIrZt28Zdd91F//796dmzJ8OHD6f8y5YPP/wwPXr0oHfv3gwdOhSArVu38oMf/ID+/fvTt29fXn311W8c46mnnmLkyJEADBs2jOuvv56TTjqJww8/nBdeeGFPuQcffJD+/fvTu3dvfv3rX9dI+6pM6GZWH3gMOAfoAXzPzHrElnH3B929j7v3AW4D/u3uX9ZIhGnkjc2FvLG5+nMoi0ggPz+fp556CoCdO3eSn5/Ps88+C8C2bdvIz89nwoQJAGzevJn8/HxeeuklANavX09+fj7/+Mc/APjiiy8SPu6iRYsYPnw4c+bMoVmzZvzxj39k5MiRzJgxg7lz57J9+3Zee+01AEaPHs2sWbOYM2cOY8aMAeDee+9l0KBBzJgxg3fffZebbrqJrVu37vOYq1ev5v333+e1117bM33A5MmTWbx4MR999BGzZ89m5syZTJ06NeF2VCaRHvrxQKG7L3X3r4HxwJB9lP8e8Px+R5aGRrY5npFtjo86DBGppo4dO3LyyScDcPnll/P+++/z7rvvcsIJJ9CrVy/eeecd5s2bB0Dv3r257LLLePbZZ2nQIBidnjx5MqNHj6ZPnz7k5+dTWlrKihUr9nnMCy+8kHr16tGjRw/WrFmzp57JkyfTt29f+vXrx8KFC1m8ePF+ty+RMfQOwMqY5SLghHgFzexAYDAwspLtw4HhAJ06dUoq0HTQtH7jqEMQyQpTpkzZ87phw4Z7LR944IF7LTdv3nyv5VatWu21fOihhyZ8XDP7xvKPf/xjCgoK6NixI6NGjaK0tBSAf/7zn0ydOpWJEydy9913M2/ePNydF198kaOOOmqvesoTdTyNG/83b5QP57g7t912Gz/60Y8Sjj0RifTQLc66ymb0Oh/4oLLhFncf6+557p6X7Axl6aBg6yoKtq6KOgwRqaYVK1Ywbdo0AJ5//nlOOeUUIPiQKCkp2TPGvXv3blauXMlpp53GAw88wKZNmygpKeHss8/mkUce2ZOYZ82aVa04zj77bJ588klKSkoAKC4uZu3a/Z8rKpEeehHQMWY5F6gsqw0lS4dbAP61ZUnUIYjIfjj66KN5+umn+dGPfkS3bt249tpr2bhxI7169aJz58707x/clrxr1y4uv/xyNm/ejLvz85//nBYtWnDnnXfys5/9jN69e+PudO7cec+YezLOOussFixYwIABAwDIycnh2WefpU2bNvvVviqnzzWzBsBnwOlAMTADuNTd51Uo1xxYBnR0931fJSAzp88dc1hwn+mI5cn/AkXqMk2fWz01Pn2uu5eZ2UjgTaA+8KS7zzOzEeH2MWHR7wCTE0nmmerAeg2jDkFEpFIJfbHI3ScBkyqsG1Nh+SngqZoKLB19uLUIgCsjjkNEJB59UzQJ72xZFnUIIhnL3b9xl4lUrjpPk1NCT8INbQdEHYJIRmrSpAkbNmygZcuWSuoJcHc2bNhAkybJTTeihJ6ExvX0dolUR25uLkVFRaxbty7qUDJGkyZNyM3NTWofZagkfFASfL9KY+giyWnYsCFdunSJOoysp4SehKlffR51CCIilVJCT8JNh54cdQgiIpVSQk9CA9PzQEQkfSmhJ+G9r5YDGkMXkfSkhJ6E90v2PU2miEiUlNCTcFu7gVGHICJSKQ0Ki4hkCfXQkzAlvG1RY+giko6U0JPw4dbiqEMQEamUEnoSbtF96CKSxpTQk/TBqgIeb9mBh/oMAmDCyoXM3LiaB3qfBsBzK+Yzd/M67ut1KgDjPp/L4pKN3N0zuKD6l2VzWL5tC6OOCR599cTST1hTupU7epwEwONLZrF55w5u7X4iAI8snsmO3bu48ajg4dS//yx4KMjPjgzmt//doo9oXK8+13U7DoDRC6fTvGFjru3aF4B75v+Htk0O4prDjwVg1Lz3OezAZny/S28A7pz7Ht1yDubKzj0B+OWn/6Zn89Zc2qkHADfPeZfjDm7HJR27A3DD7Hc4uVUHvpsbPFPx+llvcXqbwxjSoRsAP/l4Mue268q323WlbPdufjr7Lc5vfwSDDz2c0l1l/OKTd/hOhyM5o21nSsq+5pY5U7g4tzv5bTqx6etSbp87le916sEprXLpMPQ8Ol1zSY397kSynS6KJmH10SdTmnNI1GHUCZ/PmsOcp/8edRgiGaXKR9ClSiY+gm7kbbMBePS3fSKOJPv1O7gtAB9vrPxp6iJ10X49gk4kClcc1jPqEEQyjhK6pKUTW7aPOgSRjKOELmlpTWnWPmtcJGUSuihqZoPNbJGZFZrZrZWUyTez2WY2z8z+XbNhSl1z1/wPuGv+B1GHIZJRquyhm1l94DHgTKAImGFmE919fkyZFsAfgcHuvsLM2qQqYKkbhnXuFXUIIhknkSGX44FCd18KYGbjgSHA/JgylwIvufsKAHdfW9OBSt3S/5B2UYcgknESGXLpAKyMWS4K18U6EjjYzKaY2Uwz03Qnsl+Kt39F8favog5DJKMk0kO3OOsq3rzeADgOOB04AJhmZtPd/bO9KjIbDgwH6NSpU/LRSp1x34JpAFwUcRwimSSRhF4EdIxZzgVWxSmz3t23AlvNbCpwLLBXQnf3scBYCL5YVN2gJftd3eXYqEMQyTiJDLnMALqZWRczawQMBSZWKPMqMNDMGpjZgcAJwIKaDVXqkr4Ht6Vv+G1REUlMlT10dy8zs5HAm0B94El3n2dmI8LtY9x9gZm9AcwBdgN/cve5qQxcstvyrZsBGBBxHCKZJKEvFrn7JGBShXVjKiw/CDxYc6FJXfbAog+B4HRQRBKjb4pKWhoRTv8rIolTQpe01Kt566hDEMk4SuiSlpaUbAI0hi6SDCX0JBUuK9kzL7qkztufTKNBg3pcHnUgIhlECT0JZ56qKWpqy4U5PWjcSA/UEkmGEnoShgxuz5DBmqe7Nowb3zLqEEQyjhK6pKXlOzZFHYJIxlFCl7T03JefAnBnxHGIZBIldElLlx6i+dBFkqWELmnpsMYtog5BJOMooUtaWrpjY9QhiGQcJXRJSxO+DOZ2GxVtGCIZRQld0tIVLTUfukiylNAlLeU2ahZ1CCIZRwld0tLi0g1RhyCScZTQJS29sHE+AHdHHIdIJlFCl7Q0rFWfqEMQyThK6JKW2jVsyoov5tLv4Lb88ugBdDigKTO+XM1Tn3/Kr3qcTNsmBzF9wyqeWT6Xu44ZSMvGB/D++iKeXzGfe3t+ixaNmjBl7Qr+XrSQ+3vnk9OgEW+t+ZyXiz/jf48dRJP6DXjji6X8Y1Uhf+hzBg3q1eOfq5cwafUSHut3FgCvFi/m7bXLebjvGQC8WLSID9YX81CfQQBMWLmQmRtX80Dv0wB4bsV85m5ex329TgVg3OdzWVyykbt7DgTgL8vmsHzbFkYdcwoATyz9hDWlW7mjx0kAPL5kFj0GDuDWZ56ovTdasooSuqSl1UefzLbS9VC6JepQas27KwvZ/M6OqMOQDGbuHsmB8/LyvKCgIJJjS/orn3P+0d/WnaGXaadfAcCAt5+JOBJJZ2Y2093z4m3ThNMiIlkioYRuZoPNbJGZFZrZrXG255vZZjObHf78quZDFclujyyeySOLZ0YdhmSwKsfQzaw+8BhwJlAEzDCzie4+v0LR99z9vBTEKFIn7Ni9K+oQJMMlclH0eKDQ3ZcCmNl4YAhQMaGLyH648ajjow5BMlwiQy4dgJUxy0XhuooGmNknZva6mR0TryIzG25mBWZWsG7dumqEKyIilUmkh25x1lW8NeZj4DB3LzGzc4FXgG7f2Ml9LDAWgrtckoxVJKv9/rPgrq8BEcchmSuRHnoR0DFmORdYFVvA3be4e0n4ehLQ0Mxa1ViUIiJSpUR66DOAbmbWBSgGhgKXxhYws0OBNe7uZnY8wQeFZlcSScLPjox7a7FIwqpM6O5eZmYjgTeB+sCT7j7PzEaE28cAFwHXmlkZsB0Y6lF9Y0lEpI5K6Kv/4TDKpArrxsS8fhR4tGZDE6lbfrfoIwBejDgOyVyay0UkTTSuVz/qECTDKaGLpInruh0XdQiS4TSXi4hIllAPXSRNjF44HYBXI45DMpcSukiaaN6wcdQhSIZTQhdJE9d27Rt1CJLhNIYuIpIl1EOXtFW4rGTPk4vqgk8+mkrjRvV4K+pAJGMpoUtaOvPUNlGHUOuaeWMa7tJJs1SfErqkpSGD2zNkcPuow6hV48bHnXVaJGHqDoiIZAn10EXSxJh1wXzoV0Ych2QuJXSRNNGuYU7UIUiGU0IXSRNDWnSPOgTJcBpDFxHJEuqhi6SJP66dAWgMXapPCV0kTXRq1DzqECTDKaGLpInzWhwZdQiS4TSGLiKSJdRDF0kTj6z9ENAYulSfErpImuja+JCoQ5AMl9CQi5kNNrNFZlZoZrfuo1x/M9tlZhfVXIgidcO5zbtxbvNuUYchGazKhG5m9YHHgHOAHsD3zKxHJeXuB96s6SBFRKRqiQy5HA8UuvtSADMbDwwB5lcodx3wItC/RiMUqSP+b03wTFGNoUt1JZLQOwArY5aLgBNiC5hZB+A7wCD2kdDNbDgwHKBTp07JxiqS1Y45oHXUIUiGSyShW5x1XmH598At7r7LLF7xcCf3scBYgLy8vIp1iNRpZzXrGnUIkuESSehFQMeY5VxgVYUyecD4MJm3As41szJ3f6VGohQRkSolktBnAN3MrAtQDAwFLo0t4O5dyl+b2VPAa0rmIsn53Zr/ABpDl+qrMqG7e5mZjSS4e6U+8KS7zzOzEeH2MSmOUaRO6HvAoVGHIBkuoS8WufskYFKFdXETubsP2/+wROqe05sdHnUIkuE0l4uISJbQV/9F0sT9X3wAaAxdqk8JXSRNnHBQh6hDkAynhC6SJvKbdo46BMlwGkMXEckS6qGLpInfrn4P0Bi6VJ8SukiaOCVH8xvJ/lFCF0kTA5seFnUIkuGU0EXSRJnvjjoEyXBK6CJp4sHwPvQfRByHZC4ldJE08S3dtij7SQldJE2cnNOx6kIi+6CELpImduwuizoEyXBK6CJp4qE10wC4JuI4JHMpoYukiUHNunDAlnX869RLuGXOFC7O7U5+m05s+rqU2+dO5XudenBKq1w27NjOr+a9xxWH9eTElu1ZU7qVu+Z/wLDOveh/SDuKt3/FfQumcXWXY+l7cFuWb93MA4s+ZETXvvRq3polJZt46LOP+MkR/ejRrBWfffUlf1hcwE+75XFk00OYv2U9jxV+zA1HHk/XnBZ8unkdY5bM4uajTuCwg5oza+Ma/rTsE3559AA6HNCUGV+u5qnPP+VXPU6mbZODmL5hFc8sn8tdxwykZeMDeH99Ec+vmM+9Pb9Fi0ZNmLJ2BX8vWsj9vfPJadCIt9Z8zsvFn/G/xw6iSf0GvPHFUv6xqpA/9DmDBvXq8c/VS5i0egmP9TsLgFeLF/P22uU83PcMAF4sWsQH64t5qM8gACasXMjMjat5oPdpADy3Yj5zN6/jvl6nAjDu87ksLtnI3T0HAvCXZXNYvm0Lo445BYAnln7CmtKt3NHjJAAeXzKLzTt3cGv3EwF4ZPFMduzexY1HHQ/A7z8rAOBnR+YB8LtFH9G4Xn2u63YcAKMXTqd5w8Zc27UvAPfM/w8DzxjErc88UeN/Q0roImmiU94ltFvwQdRhSIrNWVtM2/c/TEnd5h7Ns5rz8vK8oKAgkmOLpKORt80G4NHf9ok4EkmlaadfAcCAt5+p1v5mNtPd8+Jt0+RcIiJZQkMuIiK1aNS894HgIc01TQldRKQWHXZgs5TVrYQuIlKLvt+ld8rqTmgM3cwGm9kiMys0s1vjbB9iZnPMbLaZFZjZKTUfqoiI7EuVPUl0ZoYAAAgXSURBVHQzqw88BpwJFAEzzGyiu8+PKfY2MNHd3cx6A38DuqciYBGRTHbn3OBBJm+loO5EhlyOBwrdfSmAmY0HhgB7Erq7l8SUPwiI5l5IEZE01y3n4JTVnUhC7wCsjFkuAk6oWMjMvgP8FmgDfDteRWY2HBgO0KmTns4iInXPlZ17pqzuRMbQLc66b/TA3f1ld+8OXAjcHa8idx/r7nnunte6devkIhURkX1KpIdeBMTO65kLrKqssLtPNbOuZtbK3dfvb4AiItnkl5/+G4B3U1B3Igl9BtDNzLoAxcBQ4NLYAmZ2BLAkvCjaD2gEbKjpYEVEMl3P5qkbnagyobt7mZmNJPhiU33gSXefZ2Yjwu1jgO8CV5rZTmA7cIlHNUmMiEgau7RTj5TVndAXi9x9EjCpwroxMa/vB+6v2dBERCQZ+qaoiEgtunlOMHr+XgrqVkIXEalFxx3cLmV1K6GLiNSiSzqm7kv0mg9dRCRLqIcukkYKl5XseXKRZKep0ybTsEE9ZqagbiV0kTRx5qltog5BasExDdvSqEG8L+DvPyV0kTQxZHB7hgxuH3UYkmLjxndNWd0aQxcRyRLqoYuI1KL7v/gAgCtTULcSuohILTrhoA4pq1sJXUSkFuU37ZyyujWGLiKSJdRDFxGpRb9dHcziojF0EZEMd0pO6h6/qYQuIlKLBjY9LGV1K6GLiNSiMt+dsrqV0EVEatGD4X3oP0hB3UroIiK16FspvG1RCV1EpBadnNMxZXUroYuI1KIdu8tSVrcSuohILXpozTQArklB3QkldDMbDPwBqA/8yd1HV9h+GXBLuFgCXOvun9RkoCIi2WDksZenrO4qE7qZ1QceA84EioAZZjbR3efHFFsGnOruG83sHGAscEIqAhYRyWQLBg1LWd2JzOVyPFDo7kvd/WtgPDAktoC7/8fdN4aL04Hcmg1TRESqkkhC7wCsjFkuCtdV5ofA6/E2mNlwMysws4J169YlHqWIiFQpkYQe7+F3Hreg2WkECf2WeNvdfay757l7XuvWrROPUkREqpTIRdEiIPbGyVxgVcVCZtYb+BNwjrtvqJnwREQkUYn00GcA3cysi5k1AoYCE2MLmFkn4CXgCnf/rObDFBGRqlTZQ3f3MjMbCbxJcNvik+4+z8xGhNvHAL8CWgJ/NDOAMnfPS13YIiJSUUL3obv7JGBShXVjYl5fDVxds6GJiEgy9Ag6EZEsoYQuIpIllNBFRLKEErqISJZQQhcRyRJK6CIiWUIJXUQkSyihi4hkCT2xSESkFnU7PCdldSuhi4jUop9ec0TK6taQi4hIllBCFxHJEkroIiJZQgldRCRLKKGLiGQJJXQRkSyhhC4ikiWU0EVEsoS5ezQHNlsHLK/m7q2A9TUYTiZQm+sGtblu2J82H+bureNtiCyh7w8zK6hrD6FWm+sGtbluSFWbNeQiIpIllNBFRLJEpib0sVEHEAG1uW5Qm+uGlLQ5I8fQRUTkmzK1hy4iIhUooYuIZIm0TOhm1tHM3jWzBWY2z8x+Gq4/xMz+ZWaLw38PjtnnNjMrNLNFZnZ2dNFXn5nVN7NZZvZauJzt7W1hZi+Y2cLwdz2gDrT55+Hf9Fwze97MmmRbm83sSTNba2ZzY9Yl3UYzO87MPg23PWxmVtttSVQlbX4w/NueY2Yvm1mLmG2pabO7p90P0A7oF75uCnwG9AAeAG4N198K3B++7gF8AjQGugBLgPpRt6Ma7b4BeA54LVzO9vY+DVwdvm4EtMjmNgMdgGXAAeHy34Bh2dZm4FtAP2BuzLqk2wh8BAwADHgdOCfqtiXZ5rOABuHr+2ujzWnZQ3f31e7+cfj6K2ABwX+GIQRJgPDfC8PXQ4Dx7r7D3ZcBhcDxtRv1/jGzXODbwJ9iVmdze5sR/Cf4M4C7f+3um8jiNocaAAeYWQPgQGAVWdZmd58KfFlhdVJtNLN2QDN3n+ZBphsXs0/aiddmd5/s7mXh4nQgN3ydsjanZUKPZWadgb7Ah0Bbd18NQdIH2oTFOgArY3YrCtdlkt8DNwO7Y9Zlc3sPB9YBfwmHmf5kZgeRxW1292Lgd8AKYDWw2d0nk8VtjpFsGzuEryuuz1Q/IOhxQwrbnNYJ3cxygBeBn7n7ln0VjbMuY+7HNLPzgLXuPjPRXeKsy5j2hhoQnKI+7u59ga0Ep+KVyfg2h+PGQwhOs9sDB5nZ5fvaJc66jGpzAiprY9a03cxuB8qAv5avilOsRtqctgndzBoSJPO/uvtL4eo14WkJ4b9rw/VFQMeY3XMJTmUzxcnABWb2OTAeGGRmz5K97YWgDUXu/mG4/AJBgs/mNp8BLHP3de6+E3gJOInsbnO5ZNtYxH+HKGLXZxQzuwo4D7gsHEaBFLY5LRN6eGX3z8ACd38oZtNE4Krw9VXAqzHrh5pZYzPrAnQjuLiQEdz9NnfPdffOwFDgHXe/nCxtL4C7fwGsNLOjwlWnA/PJ4jYTDLWcaGYHhn/jpxNcH8rmNpdLqo3hsMxXZnZi+F5dGbNPRjCzwcAtwAXuvi1mU+raHPXV4UquGJ9CcKoxB5gd/pwLtATeBhaH/x4Ss8/tBFeLF5HGV8MTaHs+/73LJavbC/QBCsLf8yvAwXWgzb8BFgJzgWcI7nTIqjYDzxNcI9hJ0Ov8YXXaCOSF79MS4FHCb7an408lbS4kGCsvz2FjUt1mffVfRCRLpOWQi4iIJE8JXUQkSyihi4hkCSV0EZEsoYQuIpIllNBFRLKEErqISJb4f4wLLqK3dPAUAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_title = cancer+\" Survival Risk: PTEN loss\"\n",
    "\n",
    "cph = CoxPHFitter()\n",
    "cph.fit(tdf, duration_col = \"days_to_last_contact_or_death\", \n",
    "        event_col = \"vital_status\")\n",
    "cph.plot_partial_effects_on_outcome('PTEN', [0,1], cmap='coolwarm', title=plot_title)    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Proportional hazard assumption looks okay.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cph.check_assumptions(tdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "covariate\n",
       "PTEN   -0.188063\n",
       "Name: coef, dtype: float64"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cph.params_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "covariate\n",
       "PTEN    1.688784\n",
       "Name: exp(coef), dtype: float64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cph.hazard_ratios_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>model</th>\n",
       "      <td>lifelines.CoxPHFitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>duration col</th>\n",
       "      <td>'days_to_last_contact_or_death'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event col</th>\n",
       "      <td>'vital_status'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>baseline estimation</th>\n",
       "      <td>breslow</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>number of observations</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>number of events observed</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>partial log-likelihood</th>\n",
       "      <td>-6.466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>time fit was run</th>\n",
       "      <td>2020-11-13 22:35:39 UTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>model</th>\n",
       "      <td>untransformed variables</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div><table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th style=\"min-width: 12px;\"></th>\n",
       "      <th style=\"min-width: 12px;\">coef</th>\n",
       "      <th style=\"min-width: 12px;\">exp(coef)</th>\n",
       "      <th style=\"min-width: 12px;\">se(coef)</th>\n",
       "      <th style=\"min-width: 12px;\">coef lower 95%</th>\n",
       "      <th style=\"min-width: 12px;\">coef upper 95%</th>\n",
       "      <th style=\"min-width: 12px;\">exp(coef) lower 95%</th>\n",
       "      <th style=\"min-width: 12px;\">exp(coef) upper 95%</th>\n",
       "      <th style=\"min-width: 12px;\">z</th>\n",
       "      <th style=\"min-width: 12px;\">p</th>\n",
       "      <th style=\"min-width: 12px;\">-log2(p)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>0.524</td>\n",
       "      <td>1.689</td>\n",
       "      <td>1.151</td>\n",
       "      <td>-1.732</td>\n",
       "      <td>2.780</td>\n",
       "      <td>0.177</td>\n",
       "      <td>16.125</td>\n",
       "      <td>0.455</td>\n",
       "      <td>0.649</td>\n",
       "      <td>0.624</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table><br><div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Concordance</th>\n",
       "      <td>0.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Partial AIC</th>\n",
       "      <td>14.932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>log-likelihood ratio test</th>\n",
       "      <td>0.226 on 1 df</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>-log2(p) of ll-ratio test</th>\n",
       "      <td>0.657</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/latex": [
       "\\begin{tabular}{lrrrrrrrrrr}\n",
       "\\toprule\n",
       "{} &  coef &  exp(coef) &  se(coef) &  coef lower 95\\% &  coef upper 95\\% &  exp(coef) lower 95\\% &  exp(coef) upper 95\\% &     z &     p &  -log2(p) \\\\\n",
       "covariate &       &            &           &                 &                 &                      &                      &       &       &           \\\\\n",
       "\\midrule\n",
       "PTEN      & 0.524 &      1.689 &     1.151 &          -1.732 &           2.780 &                0.177 &               16.125 & 0.455 & 0.649 &     0.624 \\\\\n",
       "\\bottomrule\n",
       "\\end{tabular}\n"
      ],
      "text/plain": [
       "<lifelines.CoxPHFitter: fitted with 6 total observations, 1 right-censored observations>\n",
       "             duration col = 'days_to_last_contact_or_death'\n",
       "                event col = 'vital_status'\n",
       "      baseline estimation = breslow\n",
       "   number of observations = 6\n",
       "number of events observed = 5\n",
       "   partial log-likelihood = -6.466\n",
       "         time fit was run = 2020-11-13 22:35:39 UTC\n",
       "                    model = untransformed variables\n",
       "\n",
       "---\n",
       "            coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%\n",
       "covariate                                                                                                         \n",
       "PTEN       0.524      1.689      1.151           -1.732            2.780                0.177               16.125\n",
       "\n",
       "              z     p   -log2(p)\n",
       "covariate                       \n",
       "PTEN      0.455 0.649      0.624\n",
       "---\n",
       "Concordance = 0.600\n",
       "Partial AIC = 14.932\n",
       "log-likelihood ratio test = 0.226 on 1 df\n",
       "-log2(p) of ll-ratio test = 0.657"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cph.print_summary(model=\"untransformed variables\", decimals=3)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 6: Cox's Proportional Hazard Test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With the proportional_hazard_test function, we can now perform Cox's Proportional Hazard Test on the data to determine how each attribute contributes to our cohort's overall survival. This is shown by the hazard ratio in the column labeled -log2(p) below. In general, a hazard ratio of 1 suggests that an attribute has no effect on overall survival. A ratio less than 1 suggests that an attribute contributes to lower survival risk. A ratio greater than 1 suggests that an attribute contributes to higher survival risk."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = proportional_hazard_test(cph, df_clean, time_transform='rank')\n",
    "results.print_summary(decimals=3, model=\"untransformed variables\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below, we show the confidence interval for the hazard ratio. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.title(cancer)\n",
    "cph.plot()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# log(hazards ratio) = coef = -0.192"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Conclusions: Because the p-value was greater than 0.05, there is insufficient evidence to suggest the PTEN loss effects survival risk."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
