{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make Figure 6: Mitotic Cell Cycle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebooks takes all the trans genes that are significant in multiple cancers and runs a GSEA using Reactome. It then takes a subset of genes from the Mitotic Cell Cycle pathway and creates a heatmap. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cptac version: 0.8.4\n"
     ]
    }
   ],
   "source": [
    "print('cptac version:', cptac.version())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 1: Run GSEA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First read in sig_pval_heatmap.csv into a df. This csv file contains only genes with a significant p-value in at least one cancer. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_df = pd.read_csv('../Make_Tables/csv/mult_sig_pval_heatmap.csv') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_list = list(sig_df.Proteomics.unique()) # list of genes with a sig pval in mult cancers\n",
    "prot_enr = gp.enrichr(gene_list = prot_list, description='Tumor_partition', gene_sets='Reactome_2016', outdir='/Enrichr')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Term</th>\n",
       "      <th>Overlap</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Adjusted P-value</th>\n",
       "      <th>Old P-value</th>\n",
       "      <th>Old Adjusted P-value</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>Combined Score</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Gene_set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gene Expression Homo sapiens R-HSA-74160</td>\n",
       "      <td>76/1631</td>\n",
       "      <td>7.790079e-17</td>\n",
       "      <td>1.191882e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.807059</td>\n",
       "      <td>104.116894</td>\n",
       "      <td>DDX47;WDR3;WDR4;HNRNPU;NAT10;ZC3H8;ADAR;PPP2R2...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Cell Cycle, Mitotic Homo sapiens R-HSA-69278</td>\n",
       "      <td>37/462</td>\n",
       "      <td>2.324896e-15</td>\n",
       "      <td>1.778545e-12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.824493</td>\n",
       "      <td>162.561772</td>\n",
       "      <td>TOP2A;AHCTF1;PCNA;MCM7;DCTN1;BUB1B;PPP2R2A;ORC...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Processing of Capped Intron-Containing Pre-mRN...</td>\n",
       "      <td>24/193</td>\n",
       "      <td>1.716598e-14</td>\n",
       "      <td>8.754651e-12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7.491104</td>\n",
       "      <td>237.436894</td>\n",
       "      <td>RANBP2;DDX5;NUP210;CPSF1;NUP155;NCBP1;NUP133;D...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Cell Cycle Homo sapiens R-HSA-1640170</td>\n",
       "      <td>38/566</td>\n",
       "      <td>2.579814e-13</td>\n",
       "      <td>9.867789e-11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.044446</td>\n",
       "      <td>117.231872</td>\n",
       "      <td>TOP2A;AHCTF1;PCNA;MCM7;DCTN1;BUB1B;PPP2R2A;ORC...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rRNA modification in the nucleus Homo sapiens ...</td>\n",
       "      <td>14/58</td>\n",
       "      <td>4.819842e-13</td>\n",
       "      <td>1.474872e-10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14.540922</td>\n",
       "      <td>412.393134</td>\n",
       "      <td>UTP6;DDX47;IMP3;WDR3;HEATR1;NAT10;WDR75;IMP4;P...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Term  Overlap       P-value  \\\n",
       "0           Gene Expression Homo sapiens R-HSA-74160  76/1631  7.790079e-17   \n",
       "1       Cell Cycle, Mitotic Homo sapiens R-HSA-69278   37/462  2.324896e-15   \n",
       "2  Processing of Capped Intron-Containing Pre-mRN...   24/193  1.716598e-14   \n",
       "3              Cell Cycle Homo sapiens R-HSA-1640170   38/566  2.579814e-13   \n",
       "4  rRNA modification in the nucleus Homo sapiens ...    14/58  4.819842e-13   \n",
       "\n",
       "   Adjusted P-value  Old P-value  Old Adjusted P-value  Odds Ratio  \\\n",
       "0      1.191882e-13            0                     0    2.807059   \n",
       "1      1.778545e-12            0                     0    4.824493   \n",
       "2      8.754651e-12            0                     0    7.491104   \n",
       "3      9.867789e-11            0                     0    4.044446   \n",
       "4      1.474872e-10            0                     0   14.540922   \n",
       "\n",
       "   Combined Score                                              Genes  \\\n",
       "0      104.116894  DDX47;WDR3;WDR4;HNRNPU;NAT10;ZC3H8;ADAR;PPP2R2...   \n",
       "1      162.561772  TOP2A;AHCTF1;PCNA;MCM7;DCTN1;BUB1B;PPP2R2A;ORC...   \n",
       "2      237.436894  RANBP2;DDX5;NUP210;CPSF1;NUP155;NCBP1;NUP133;D...   \n",
       "3      117.231872  TOP2A;AHCTF1;PCNA;MCM7;DCTN1;BUB1B;PPP2R2A;ORC...   \n",
       "4      412.393134  UTP6;DDX47;IMP3;WDR3;HEATR1;NAT10;WDR75;IMP4;P...   \n",
       "\n",
       "        Gene_set  \n",
       "0  Reactome_2016  \n",
       "1  Reactome_2016  \n",
       "2  Reactome_2016  \n",
       "3  Reactome_2016  \n",
       "4  Reactome_2016  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_enr.res2d.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 2: Get the list of significant genes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cell Cycle, Mitotic Homo sapiens R-HSA-69278\n",
      "total genes: 37\n"
     ]
    }
   ],
   "source": [
    "index = 1\n",
    "trans = prot_enr.res2d.Genes[index]\n",
    "genes = trans.split(';')\n",
    "print(prot_enr.res2d.Term[index])\n",
    "print('total genes:',len(genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Catagorize genes\n",
    "narrowed_cell_cycle = ['DCTN1', 'ACTR1A', 'MAD2L1', 'MCM4', 'PCNA', 'MCM5', 'MCM2','PCM1',\n",
    "              'MCM7', 'TPR', 'MCM3', 'MCM6', 'NUP153', 'RFC3', 'CDK11B', 'XPO1','PRKCB', 'RFC2', 'NUF2', \n",
    "              'RAB8A', 'RFC4', 'RANBP2', 'CENPF', 'GINS4', 'NDC80', 'OPTN', 'PPP2R2A', 'PPP2R2D',\n",
    "              'TPX2', 'GINS2', 'BUB1B', 'TOP2A', 'TOPBP1', 'TP53', 'NUP210', 'MSH2', 'MSH6']\n",
    "\n",
    "dna_synth = ['MCM4', 'PCNA', 'MCM5', 'MCM2','PCM1', 'MCM7', 'MCM3', 'RFC3', \n",
    "             'RFC2', 'RFC4', 'GINS4', 'GINS2', 'TOP2A']\n",
    "\n",
    "dynactin = ['DCTN1','ACTR1A']\n",
    "spindle = ['BUB1B', 'CENPF', 'MAD2L1', 'NDC80', 'NUF2', 'PCM1', 'TPX2','DCTN1','ACTR1A']\n",
    "\n",
    "checkpoints = ['BUB1B','PPP2R2A', 'PPP2R2D', 'TOPBP1','MAD2L1']\n",
    "\n",
    "nuclear_pore_transport = ['TPR','NUP153','NUP210', 'XPO1','RANBP2']\n",
    "other_transport = ['RAB8A']\n",
    "other = ['PRKCB','OPTN','CDK11B']\n",
    "\n",
    "path_list = {'DNA_Synth':dna_synth, 'Spindle':spindle, 'Dynactin': dynactin, 'Checkpoints':checkpoints,\n",
    "            'Nuclear_Pore_Transport':nuclear_pore_transport}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DCTN1</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>-0.382988</td>\n",
       "      <td>Gbm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTR1A</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>-0.491126</td>\n",
       "      <td>Gbm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MCM4</td>\n",
       "      <td>0.001303</td>\n",
       "      <td>1.236484</td>\n",
       "      <td>Gbm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MAD2L1</td>\n",
       "      <td>0.001303</td>\n",
       "      <td>0.546914</td>\n",
       "      <td>Gbm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PCNA</td>\n",
       "      <td>0.001545</td>\n",
       "      <td>0.631486</td>\n",
       "      <td>Gbm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2037</th>\n",
       "      <td>BUB1B</td>\n",
       "      <td>0.026111</td>\n",
       "      <td>-0.559000</td>\n",
       "      <td>Endo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2051</th>\n",
       "      <td>NUF2</td>\n",
       "      <td>0.034789</td>\n",
       "      <td>-0.612500</td>\n",
       "      <td>Endo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2056</th>\n",
       "      <td>CENPF</td>\n",
       "      <td>0.037588</td>\n",
       "      <td>-0.439000</td>\n",
       "      <td>Endo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2065</th>\n",
       "      <td>TPR</td>\n",
       "      <td>0.042634</td>\n",
       "      <td>-0.184300</td>\n",
       "      <td>Endo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2075</th>\n",
       "      <td>PPP2R2A</td>\n",
       "      <td>0.048016</td>\n",
       "      <td>0.372600</td>\n",
       "      <td>Endo</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>76 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Proteomics   P_Value   Medians Cancer\n",
       "3         DCTN1  0.000015 -0.382988    Gbm\n",
       "4        ACTR1A  0.000038 -0.491126    Gbm\n",
       "20         MCM4  0.001303  1.236484    Gbm\n",
       "23       MAD2L1  0.001303  0.546914    Gbm\n",
       "26         PCNA  0.001545  0.631486    Gbm\n",
       "...         ...       ...       ...    ...\n",
       "2037      BUB1B  0.026111 -0.559000   Endo\n",
       "2051       NUF2  0.034789 -0.612500   Endo\n",
       "2056      CENPF  0.037588 -0.439000   Endo\n",
       "2065        TPR  0.042634 -0.184300   Endo\n",
       "2075    PPP2R2A  0.048016  0.372600   Endo\n",
       "\n",
       "[76 rows x 4 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bool_df = sig_df.Proteomics.isin(narrowed_cell_cycle)\n",
    "df = sig_df[bool_df]\n",
    "plot_df = df.loc[df['P_Value'] <= 0.05] # Only plot sig genes\n",
    "plot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "ordered_prot_list = ['MSH2', 'MSH6', 'GINS2', 'GINS4', 'MCM2', 'MCM3', 'MCM4', 'MCM5', 'MCM6', 'MCM7',\n",
    "                     'RFC2', 'RFC3', 'RFC4', 'RFC5',  'PCNA', 'PCM1', 'TOP2A','PPP2R2A', 'PPP2R2D', \n",
    "                     'TOPBP1', 'TP53', 'MAD2L1', 'BUB1B', 'CENPF', 'NDC80', 'NUF2', 'TPX2','DCTN1','ACTR1A']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "# Reorder index to reorder the protein order in heatmap\n",
    "plot_df[\"Index\"] = plot_df[\"Proteomics\"] + \"_\" + plot_df[\"Cancer\"]\n",
    "plot_df = plot_df.set_index(\"Index\")\n",
    "\n",
    "ordered_list = []\n",
    "cancer_list = ['Brca', 'Colon', 'Endo', 'Gbm', 'Hnscc', 'Lscc', 'Luad', 'Ov']\n",
    "for gene in ordered_prot_list:\n",
    "    for cancer in cancer_list:\n",
    "        ordered_list.append(gene+'_'+cancer)\n",
    "\n",
    "ordered_df = plot_df.reindex(ordered_list)\n",
    "ordered_df = ordered_df.loc[ordered_df['P_Value'] < 0.05] # Keep sig "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['EC', 'GBM', 'HNSCC', 'LUAD'], dtype=object)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename cancers with abbreviations                               \n",
    "ordered_df = ordered_df.replace({'Hnscc': 'HNSCC', 'Luad': 'LUAD', 'Lscc': 'LSCC', 'Ovarian': 'OV', 'Colon': 'CO',\n",
    "                    'Kidney': 'ccRCC', 'Breast': 'BR', 'Gbm': 'GBM', 'Endo': 'EC'})\n",
    "ordered_df['Cancer'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min p-value: 1.5008898009485894e-05\n"
     ]
    }
   ],
   "source": [
    "print('min p-value:', ordered_df.P_Value.min())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"1467\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"1467\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"1467\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '1467' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"1467\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"1467\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"1467\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '1467' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"1467\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"b999f95f-dee9-4bd6-a57f-839aded8d277\" data-root-id=\"1468\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"6cc172cb-0c0b-4a39-8be4-29fed653f629\":{\"roots\":{\"references\":[{\"attributes\":{\"children\":[{\"id\":\"1419\"},{\"id\":\"1445\"}]},\"id\":\"1468\",\"type\":\"Row\"},{\"attributes\":{\"below\":[{\"id\":\"1452\"}],\"center\":[{\"id\":\"1454\"},{\"id\":\"1457\"}],\"left\":[{\"id\":\"1455\"}],\"plot_height\":200,\"plot_width\":140,\"renderers\":[{\"id\":\"1465\"}],\"title\":{\"id\":\"1490\"},\"toolbar\":{\"id\":\"1459\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"1443\"},\"x_scale\":{\"id\":\"1448\"},\"y_range\":{\"id\":\"1444\"},\"y_scale\":{\"id\":\"1450\"}},\"id\":\"1445\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"factors\":[\"LUAD\",\"HNSCC\",\"GBM\",\"EC\"]},\"id\":\"1418\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"1430\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"data\":{\"Medians\":{\"__ndarray__\":\"AAAAAAAA+D8AAAAAAAD4PwAAAAAAAPg/\",\"dtype\":\"float64\",\"shape\":[3]},\"P_Value\":{\"__ndarray__\":\"mpmZmZmZqT/8qfHSTWJQP/Fo44i1+OQ+\",\"dtype\":\"float64\",\"shape\":[3]},\"index\":[0,1,2],\"size\":{\"__ndarray__\":\"reqW3HH5IUC4/yvzJ7k0QMT/+Z/2REFA\",\"dtype\":\"float64\",\"shape\":[3]},\"size2\":{\"__ndarray__\":\"keNze0L3B0Cg/4+ZiqEbQFuqoiqeBidA\",\"dtype\":\"float64\",\"shape\":[3]},\"x_axis\":[\"\",\"\",\"\"],\"y_axis\":[\"5.0e-02\",\"1.0e-03\",\"1.0e-05\"]},\"selected\":{\"id\":\"1503\"},\"selection_policy\":{\"id\":\"1504\"}},\"id\":\"1461\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"axis\":{\"id\":\"1429\"},\"dimension\":1,\"ticker\":null},\"id\":\"1431\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"1492\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"1448\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"text\":\"\"},\"id\":\"1490\",\"type\":\"Title\"},{\"attributes\":{\"high\":1.5,\"low\":1.5,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"1442\",\"type\":\"LinearColorMapper\"},{\"attributes\":{},\"id\":\"1450\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"callback\":null},\"id\":\"1432\",\"type\":\"HoverTool\"},{\"attributes\":{\"axis_label\":\"FDR p-value\",\"formatter\":{\"id\":\"1497\"},\"major_label_orientation\":1.5707963267948966,\"major_label_text_font_size\":\"11pt\",\"ticker\":{\"id\":\"1453\"}},\"id\":\"1452\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"1432\"}]},\"id\":\"1433\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"1495\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{},\"id\":\"1453\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"1497\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"axis\":{\"id\":\"1452\"},\"ticker\":null},\"id\":\"1454\",\"type\":\"Grid\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1416\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Proteomics\"},\"y\":{\"field\":\"Cancer\"}},\"id\":\"1438\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"1499\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"formatter\":{\"id\":\"1499\"},\"major_label_text_font_size\":\"11pt\",\"ticker\":{\"id\":\"1456\"}},\"id\":\"1455\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"data_source\":{\"id\":\"1435\"},\"glyph\":{\"id\":\"1437\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1438\"},\"selection_glyph\":null,\"view\":{\"id\":\"1440\"}},\"id\":\"1439\",\"type\":\"GlyphRenderer\"},{\"attributes\":{},\"id\":\"1501\",\"type\":\"Selection\"},{\"attributes\":{},\"id\":\"1456\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"1491\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"1502\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"axis\":{\"id\":\"1455\"},\"dimension\":1,\"ticker\":null},\"id\":\"1457\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"1503\",\"type\":\"Selection\"},{\"attributes\":{\"callback\":null},\"id\":\"1458\",\"type\":\"HoverTool\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1416\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Proteomics\"},\"y\":{\"field\":\"Cancer\"}},\"id\":\"1437\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"1504\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"source\":{\"id\":\"1435\"}},\"id\":\"1440\",\"type\":\"CDSView\"},{\"attributes\":{\"below\":[{\"id\":\"1426\"}],\"center\":[{\"id\":\"1428\"},{\"id\":\"1431\"}],\"left\":[{\"id\":\"1429\"}],\"plot_height\":350,\"plot_width\":800,\"renderers\":[{\"id\":\"1439\"}],\"right\":[{\"id\":\"1441\"}],\"title\":{\"id\":\"1488\"},\"toolbar\":{\"id\":\"1433\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"1417\"},\"x_scale\":{\"id\":\"1422\"},\"y_range\":{\"id\":\"1418\"},\"y_scale\":{\"id\":\"1424\"}},\"id\":\"1419\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"1458\"}]},\"id\":\"1459\",\"type\":\"Toolbar\"},{\"attributes\":{\"high\":1.2364838744723563,\"low\":-1.2364838744723563,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"1416\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"data\":{\"Cancer\":[\"EC\",\"GBM\",\"EC\",\"GBM\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"HNSCC\",\"GBM\",\"LUAD\"],\"Index\":[\"MSH2_Endo\",\"MSH2_Gbm\",\"MSH6_Endo\",\"MSH6_Gbm\",\"GINS2_Gbm\",\"GINS2_Hnscc\",\"GINS4_Gbm\",\"GINS4_Hnscc\",\"MCM2_Gbm\",\"MCM2_Hnscc\",\"MCM3_Gbm\",\"MCM3_Hnscc\",\"MCM4_Gbm\",\"MCM4_Hnscc\",\"MCM5_Gbm\",\"MCM5_Hnscc\",\"MCM7_Gbm\",\"MCM7_Hnscc\",\"RFC2_Endo\",\"RFC2_Gbm\",\"RFC2_Hnscc\",\"RFC3_Gbm\",\"RFC3_Hnscc\",\"RFC4_Gbm\",\"RFC4_Hnscc\",\"PCNA_Gbm\",\"PCNA_Hnscc\",\"PCM1_Endo\",\"PCM1_Gbm\",\"TOP2A_Endo\",\"TOP2A_Gbm\",\"TOP2A_Hnscc\",\"PPP2R2A_Endo\",\"PPP2R2A_Gbm\",\"PPP2R2D_Gbm\",\"PPP2R2D_Hnscc\",\"TOPBP1_Endo\",\"TOPBP1_Gbm\",\"MAD2L1_Gbm\",\"MAD2L1_Hnscc\",\"BUB1B_Endo\",\"BUB1B_Gbm\",\"CENPF_Endo\",\"CENPF_Gbm\",\"NDC80_Endo\",\"NDC80_Gbm\",\"NUF2_Endo\",\"NUF2_Gbm\",\"NUF2_Hnscc\",\"TPX2_Endo\",\"TPX2_Gbm\",\"TPX2_Hnscc\",\"DCTN1_Endo\",\"DCTN1_Gbm\",\"DCTN1_Hnscc\",\"ACTR1A_Gbm\",\"ACTR1A_Luad\"],\"Medians\":{\"__ndarray__\":\"ObTIdr6f2r+eLRrKvcnaP8l2vp8aL+G/Go0cy+WE3T/qQJUJyTXhP8B/Fmg3POA/B/4/YCq/5z/AldB+mELePz1d4gR+EO4/QM6JLucj3j+RRTmDGabtP79/RZbuJN8/Az6uUKPI8z9ADyzSvNfgP6QLQhZo2+s/QCCFX1R13j8cdU1XUpXwPwAtQ/CqfOI/BCuHFtnO17+e6+yHTzXYPz/L5omTxdc/pgTEWRtl3D8Aytlg1hPQP2+YrGHgAtc/wPUdJ2GQ3D+vNu3+IjXkP8FgojJn7do/4XoUrkfh2j9FiiRFLq3WPyhcj8L1KOS/aL6lY0Y66T9A3j9ZcebjP5tVn6ut2Nc/td6aBLx4zr/trNymp8bev4J6l2+k482/vAUSFD/G4783FhJ7go7fP2KP+qdRgOE/gEiSf0Wn2D8W2c73U+Phvx0wEqiIjeY/f2q8dJMY3L/Yd8ZdqbnhP9v5fmq8dOO/alxbYSSH5D+amZmZmZnjv2ni5a+G/+8/QFMSnjuB5T+amZmZmZnpv9EAVasFfOQ/wAr8ZrBm3z8dOGdEaW/QPzGtsgfegti//xp4L7n/vr+lU3Fmm27fv3e+nxov3eG/\",\"dtype\":\"float64\",\"shape\":[57]},\"P_Value\":{\"__ndarray__\":\"4aOQJAJxlT/E06xyLSqcP3kiDRJfP5Q/sYhiHP4Boz/HCqE/DjykP98YYdV/2o4/waxfpsmHkj9QyZQcoK1vPyus/eH1MmI/Ym7gs6kzoD8mCMD4z4lrPyoqy5pgips/N1dWxjNZVT9qnsP04q+fP/f+cnaRK2I/kpvwoSGNoz85R8RyUAplP+fBC3JWsp8/JnjW4HYqmj8U/Pu067mDPwAnC+/VNpk/exSErH8vcz/eAbNND46RP1v6BFMBS4g/vPjHaP0QmD+3qNUBoVFZPzYlC7w+M50/x2S2kSXniD/U5dNnQ+pkP9sXRhG1tJM/6D/+QwAGpT92hxPg/nCBPwdjZDKPlag/lfb9djW+mD9du25oJD6eP+0BmO2B4Jg/q+RBl+pYVD+sLNIcheeJPzdXVsYzWVU/LER59DUAoj/w9FPUvbyaP+1MCaLJfqQ/ILOwXcE+oz8+2FO7kJKRP3dcptZXBY8/1uS9FBbdkj90LmEz18+hP7qLtx9uB4Q/WiPK37x0mD/iWnaQWuKVP8d09g61OaA/GVlZ9ZJzpD8VRLxyv4qQP2CGNTzXee8+rAW8L29spD95jDvoUCwEP6afSs7T33w/\",\"dtype\":\"float64\",\"shape\":[57]},\"Proteomics\":[\"MSH2\",\"MSH2\",\"MSH6\",\"MSH6\",\"GINS2\",\"GINS2\",\"GINS4\",\"GINS4\",\"MCM2\",\"MCM2\",\"MCM3\",\"MCM3\",\"MCM4\",\"MCM4\",\"MCM5\",\"MCM5\",\"MCM7\",\"MCM7\",\"RFC2\",\"RFC2\",\"RFC2\",\"RFC3\",\"RFC3\",\"RFC4\",\"RFC4\",\"PCNA\",\"PCNA\",\"PCM1\",\"PCM1\",\"TOP2A\",\"TOP2A\",\"TOP2A\",\"PPP2R2A\",\"PPP2R2A\",\"PPP2R2D\",\"PPP2R2D\",\"TOPBP1\",\"TOPBP1\",\"MAD2L1\",\"MAD2L1\",\"BUB1B\",\"BUB1B\",\"CENPF\",\"CENPF\",\"NDC80\",\"NDC80\",\"NUF2\",\"NUF2\",\"NUF2\",\"TPX2\",\"TPX2\",\"TPX2\",\"DCTN1\",\"DCTN1\",\"DCTN1\",\"ACTR1A\",\"ACTR1A\"],\"size\":{\"__ndarray__\":\"mW4Sm2YyJ0CgudSHdo8lQKytObVmiidAiZ7Ee8fCI0C2IIzCtWIjQA4Cp/cVLClA5LtWbIUSKEChYlQ+daowQMM75g8hVDJAKvQNMB64JEACm69TBBYxQEfbGnXjsSVADKL3+uLtM0BeH2Mud9okQIx5ZUNZVTJAdCYhknmXI0AegKNWt+QxQIyUB2IA2iRArazMO4kAJkAChA+UI9srQBggdWNxOSZAybAiekAXMECqW0ZSlmUoQLSOP0xJmypA7DWgJgKBJkChkxBp42ozQL7dzB8CWCVAE7SsqDJ1KkDa3Mi8TOkxQNyKrCwNtCdA9c/CbvcnI0CKgT4qOpgsQGbE7UiiNyJARnsQRWtWJkBoFmyCHyIlQE9vitAfTiZAD60B7MUSNEAV8Z1cozgqQAyi9/ri7TNA1RxopmIWJEBeaIXTWt8lQOR9a4MMTyNAbP2qtrevI0DCSYNLDGQoQEadFNDGIylAqmZ/xyX3J0B9SUTclyYkQEQNFRi+wytAADkP4FZoJkALgciUARMnQM2/9ZLhtSRAMKPkzVVSI0AFl8M04MAoQP2XeHwJqUBAF8LDZW5UI0BUxv79D38+QJwiGuOXui1A\",\"dtype\":\"float64\",\"shape\":[57]},\"size2\":{\"__ndarray__\":\"d5MYJN7tDkArohu1SL8MQJDnTJwzYw9ADH5b+glZCkBI1rpY8tgJQF8BGqUOyBBAQ33k8lgMEEAsLhtTnDgWQFn6MhUscBhA4poSQCigC0ACJOrEBcgWQLR5zkYv7QxAZi1K+YOSGkAof9k99M0LQBCi3FnMcRhA8N3WwkwfCkB9VS8en9sXQGUbCthVzQtA55C7TwxWDUABWAq4F5ISQMoqnITsoQ1ADEEu+FV0FUDH5y6MuUMQQCNfKojbvBFA5keAM1gBDkCBb2vh2eMZQP58ZipYdQxAYs0dG3ejEUB4JmGmu+EXQCW5kDu8mg9ARxVZPp+KCUBcVtTGJhATQDNbkrYtSghAXqTABo/IDUDgHZBYfy0MQGmUuMB/vQ1AFDwCkLLDGkAO9hOTF3sRQGYtSvmDkhpAHNGKiIPICkAoi1xvzikNQNqn5Fm7vglA5lE5np8/CkAsMQLdskIQQNlouIqEwhBA4ohUXzL0D0CnDFvQH94KQIOzuLp+ghJAVaFp1XPgDUBkAbYbAsQOQBFV8sMsnQtA6y6GEh3DCUCuZIJ4lYAQQKfKoFC3NiZAdK2vh+jFCUDjLv9TtVQkQL3BZpe60RNA\",\"dtype\":\"float64\",\"shape\":[57]}},\"selected\":{\"id\":\"1501\"},\"selection_policy\":{\"id\":\"1502\"}},\"id\":\"1435\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"1422\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"color_mapper\":{\"id\":\"1416\"},\"formatter\":{\"id\":\"1491\"},\"location\":[0,0],\"ticker\":{\"id\":\"1492\"}},\"id\":\"1441\",\"type\":\"ColorBar\"},{\"attributes\":{\"text\":\"\"},\"id\":\"1488\",\"type\":\"Title\"},{\"attributes\":{},\"id\":\"1424\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1442\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"1463\",\"type\":\"Scatter\"},{\"attributes\":{\"source\":{\"id\":\"1461\"}},\"id\":\"1466\",\"type\":\"CDSView\"},{\"attributes\":{\"formatter\":{\"id\":\"1493\"},\"major_label_orientation\":1.5707963267948966,\"major_label_text_font_size\":\"12pt\",\"ticker\":{\"id\":\"1427\"}},\"id\":\"1426\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"factors\":[\"MSH2\",\"MSH6\",\"GINS2\",\"GINS4\",\"MCM2\",\"MCM3\",\"MCM4\",\"MCM5\",\"MCM7\",\"RFC2\",\"RFC3\",\"RFC4\",\"PCNA\",\"PCM1\",\"TOP2A\",\"PPP2R2A\",\"PPP2R2D\",\"TOPBP1\",\"MAD2L1\",\"BUB1B\",\"CENPF\",\"NDC80\",\"NUF2\",\"TPX2\",\"DCTN1\",\"ACTR1A\"]},\"id\":\"1417\",\"type\":\"FactorRange\"},{\"attributes\":{\"data_source\":{\"id\":\"1461\"},\"glyph\":{\"id\":\"1463\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1464\"},\"selection_glyph\":null,\"view\":{\"id\":\"1466\"}},\"id\":\"1465\",\"type\":\"GlyphRenderer\"},{\"attributes\":{},\"id\":\"1427\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"factors\":[\"\"]},\"id\":\"1443\",\"type\":\"FactorRange\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1442\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"1464\",\"type\":\"Scatter\"},{\"attributes\":{\"axis\":{\"id\":\"1426\"},\"ticker\":null},\"id\":\"1428\",\"type\":\"Grid\"},{\"attributes\":{\"factors\":[\"5.0e-02\",\"1.0e-03\",\"1.0e-05\"]},\"id\":\"1444\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"1493\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"formatter\":{\"id\":\"1495\"},\"major_label_text_font_size\":\"12pt\",\"ticker\":{\"id\":\"1430\"}},\"id\":\"1429\",\"type\":\"CategoricalAxis\"}],\"root_ids\":[\"1468\"]},\"title\":\"Bokeh Application\",\"version\":\"2.0.1\"}};\n",
       "  var render_items = [{\"docid\":\"6cc172cb-0c0b-4a39-8be4-29fed653f629\",\"root_ids\":[\"1468\"],\"roots\":{\"1468\":\"b999f95f-dee9-4bd6-a57f-839aded8d277\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "1468"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "test = p.plotCircleHeatMap(ordered_df, circle_var='P_Value', color_var='Medians', x_axis='Proteomics', y_axis='Cancer',\n",
    "                    plot_height=350, plot_width=800, x_axis_lab='', font_size=12, legend_max=0.05, \n",
    "                    legend_med = 0.001, legend_min=0.00001, save_png='Fig_6_Cell_Cycle.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "\"Invalid object to show. The object to passed to show must be one of:\n\n* a LayoutDOM (e.g. a Plot or Widget or Layout)\n* a Bokeh Application\n* a callable suitable to an application FunctionHandler\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-17-e35fe7577b40>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mbokeh\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mplotting\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mfigure\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mshow\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[0mshow\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtest\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\bokeh\\io\\showing.py\u001b[0m in \u001b[0;36mshow\u001b[1;34m(obj, browser, new, notebook_handle, notebook_url, **kw)\u001b[0m\n\u001b[0;32m    128\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    129\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mLayoutDOM\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mis_application\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mcallable\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 130\u001b[1;33m         \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0m_BAD_SHOW_MSG\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    131\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    132\u001b[0m     \u001b[1;31m# TODO (bev) check callable signature more thoroughly\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: \"Invalid object to show. The object to passed to show must be one of:\n\n* a LayoutDOM (e.g. a Plot or Widget or Layout)\n* a Bokeh Application\n* a callable suitable to an application FunctionHandler\n"
     ]
    }
   ],
   "source": [
    "from bokeh.plotting import figure, show\n",
    "show(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
