{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make Figure 6: Mitotic Cell Cycle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebooks takes all the trans genes that are significant in multiple cancers and runs a GSEA using Reactome. It then takes a subset of genes from the Mitotic Cell Cycle pathway and creates a heatmap. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cptac version: 0.9.6\n"
     ]
    }
   ],
   "source": [
    "print('cptac version:', cptac.version())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 1: Run GSEA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First read in sig_pval_heatmap.csv into a df. This csv file contains only genes with a significant p-value in at least one cancer. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_df = pd.read_csv('../Make_Tables/csv/all_heatmap.csv')\n",
    "mult_sig_df = pd.read_csv('../Make_Tables/csv/mult_sig_pval_heatmap.csv') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_list = list(mult_sig_df.Proteomics.unique()) # list of genes with a sig pval in mult cancers\n",
    "prot_enr = gp.enrichr(gene_list = prot_list, description='Tumor_partition', gene_sets='Reactome_2016')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_set</th>\n",
       "      <th>Term</th>\n",
       "      <th>Overlap</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Adjusted P-value</th>\n",
       "      <th>Old P-value</th>\n",
       "      <th>Old Adjusted P-value</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>Combined Score</th>\n",
       "      <th>Genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Reactome_2016</td>\n",
       "      <td>Gene Expression Homo sapiens R-HSA-74160</td>\n",
       "      <td>78/1631</td>\n",
       "      <td>1.086596e-16</td>\n",
       "      <td>7.932147e-14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.366774</td>\n",
       "      <td>123.756929</td>\n",
       "      <td>DDX47;WDR3;WDR4;HNRNPU;NAT10;ZC3H8;ADAR;PPP2R2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Reactome_2016</td>\n",
       "      <td>Cell Cycle, Mitotic Homo sapiens R-HSA-69278</td>\n",
       "      <td>39/462</td>\n",
       "      <td>3.268842e-16</td>\n",
       "      <td>1.193127e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.737497</td>\n",
       "      <td>204.581499</td>\n",
       "      <td>TOP2A;AHCTF1;PCNA;MCM7;DCTN1;NCAPG;BUB1B;PPP2R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Reactome_2016</td>\n",
       "      <td>Processing of Capped Intron-Containing Pre-mRN...</td>\n",
       "      <td>25/193</td>\n",
       "      <td>5.415573e-15</td>\n",
       "      <td>1.317789e-12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8.976485</td>\n",
       "      <td>294.873035</td>\n",
       "      <td>DDX5;DHX9;HNRNPU;USP39;ELAVL1;EFTUD2;NUP85;POL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Reactome_2016</td>\n",
       "      <td>Cell Cycle Homo sapiens R-HSA-1640170</td>\n",
       "      <td>40/566</td>\n",
       "      <td>5.170850e-14</td>\n",
       "      <td>9.436801e-12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.722236</td>\n",
       "      <td>144.468093</td>\n",
       "      <td>TOP2A;AHCTF1;PCNA;MCM7;DCTN1;NCAPG;BUB1B;PPP2R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Reactome_2016</td>\n",
       "      <td>rRNA modification in the nucleus Homo sapiens ...</td>\n",
       "      <td>14/58</td>\n",
       "      <td>9.127898e-13</td>\n",
       "      <td>1.332673e-10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>18.679369</td>\n",
       "      <td>517.834512</td>\n",
       "      <td>UTP6;DDX47;IMP3;WDR3;HEATR1;NAT10;WDR75;IMP4;P...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene_set                                               Term  Overlap  \\\n",
       "0  Reactome_2016           Gene Expression Homo sapiens R-HSA-74160  78/1631   \n",
       "1  Reactome_2016       Cell Cycle, Mitotic Homo sapiens R-HSA-69278   39/462   \n",
       "2  Reactome_2016  Processing of Capped Intron-Containing Pre-mRN...   25/193   \n",
       "3  Reactome_2016              Cell Cycle Homo sapiens R-HSA-1640170   40/566   \n",
       "4  Reactome_2016  rRNA modification in the nucleus Homo sapiens ...    14/58   \n",
       "\n",
       "        P-value  Adjusted P-value  Old P-value  Old Adjusted P-value  \\\n",
       "0  1.086596e-16      7.932147e-14            0                     0   \n",
       "1  3.268842e-16      1.193127e-13            0                     0   \n",
       "2  5.415573e-15      1.317789e-12            0                     0   \n",
       "3  5.170850e-14      9.436801e-12            0                     0   \n",
       "4  9.127898e-13      1.332673e-10            0                     0   \n",
       "\n",
       "   Odds Ratio  Combined Score  \\\n",
       "0    3.366774      123.756929   \n",
       "1    5.737497      204.581499   \n",
       "2    8.976485      294.873035   \n",
       "3    4.722236      144.468093   \n",
       "4   18.679369      517.834512   \n",
       "\n",
       "                                               Genes  \n",
       "0  DDX47;WDR3;WDR4;HNRNPU;NAT10;ZC3H8;ADAR;PPP2R2...  \n",
       "1  TOP2A;AHCTF1;PCNA;MCM7;DCTN1;NCAPG;BUB1B;PPP2R...  \n",
       "2  DDX5;DHX9;HNRNPU;USP39;ELAVL1;EFTUD2;NUP85;POL...  \n",
       "3  TOP2A;AHCTF1;PCNA;MCM7;DCTN1;NCAPG;BUB1B;PPP2R...  \n",
       "4  UTP6;DDX47;IMP3;WDR3;HEATR1;NAT10;WDR75;IMP4;P...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_enr.res2d.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 2: Get the list of significant genes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cell Cycle, Mitotic Homo sapiens R-HSA-69278\n",
      "Adjusted p-value: 1.1931272982172198e-13\n",
      "total genes: 39\n"
     ]
    }
   ],
   "source": [
    "index = 1\n",
    "trans = prot_enr.res2d.Genes[index]\n",
    "genes = trans.split(';')\n",
    "print(prot_enr.res2d.Term[index])\n",
    "print('Adjusted p-value:', prot_enr.res2d['Adjusted P-value'][index])\n",
    "print('total genes:',len(genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Catagorize genes\n",
    "narrowed_cell_cycle = ['DCTN1', 'ACTR1A', 'MAD2L1', 'MCM4', 'PCNA', 'MCM5', 'MCM2','PCM1',\n",
    "              'MCM7', 'TPR', 'MCM3', 'MCM6', 'NUP153', 'RFC3', 'CDK11B', 'XPO1','PRKCB', 'RFC2', 'NUF2', \n",
    "              'RAB8A', 'RFC4', 'RANBP2', 'CENPF', 'GINS4', 'NDC80', 'OPTN', 'PPP2R2A', 'PPP2R2D',\n",
    "              'TPX2', 'GINS2', 'BUB1B', 'TOP2A', 'TOPBP1', 'NUP210', 'MSH2', 'MSH6', 'CDK11B', '']\n",
    "\n",
    "dna_synth = ['MCM4', 'PCNA', 'MCM5', 'MCM2','PCM1', 'MCM7', 'MCM3', 'RFC3', \n",
    "             'RFC2', 'RFC4', 'GINS4', 'GINS2', 'TOP2A']\n",
    "\n",
    "dynactin = ['DCTN1','ACTR1A']\n",
    "spindle = ['BUB1B', 'CENPF', 'MAD2L1', 'NDC80', 'NUF2', 'PCM1', 'TPX2','DCTN1','ACTR1A']\n",
    "\n",
    "checkpoints = ['BUB1B','PPP2R2A', 'PPP2R2D', 'TOPBP1','MAD2L1']\n",
    "\n",
    "nuclear_pore_transport = ['TPR','NUP153','NUP210', 'XPO1','RANBP2']\n",
    "other_transport = ['RAB8A']\n",
    "other = ['PRKCB','OPTN','CDK11B']\n",
    "\n",
    "path_list = {'DNA_Synth':dna_synth, 'Spindle':spindle, 'Dynactin': dynactin, 'Checkpoints':checkpoints,\n",
    "            'Nuclear_Pore_Transport':nuclear_pore_transport}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DCTN1</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>-0.382988</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ACTR1A</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>-0.491126</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>MCM6</td>\n",
       "      <td>0.001303</td>\n",
       "      <td>1.085407</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>MCM4</td>\n",
       "      <td>0.001303</td>\n",
       "      <td>1.236484</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>MAD2L1</td>\n",
       "      <td>0.001303</td>\n",
       "      <td>0.546914</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66220</th>\n",
       "      <td>BUB1B</td>\n",
       "      <td>0.026111</td>\n",
       "      <td>-0.559000</td>\n",
       "      <td>EC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66269</th>\n",
       "      <td>NUF2</td>\n",
       "      <td>0.034789</td>\n",
       "      <td>-0.612500</td>\n",
       "      <td>EC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66277</th>\n",
       "      <td>CENPF</td>\n",
       "      <td>0.037588</td>\n",
       "      <td>-0.439000</td>\n",
       "      <td>EC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66315</th>\n",
       "      <td>TPR</td>\n",
       "      <td>0.042634</td>\n",
       "      <td>-0.184300</td>\n",
       "      <td>EC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66351</th>\n",
       "      <td>PPP2R2A</td>\n",
       "      <td>0.048016</td>\n",
       "      <td>0.372600</td>\n",
       "      <td>EC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>81 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics   P_Value   Medians Cancer\n",
       "6          DCTN1  0.000015 -0.382988    GBM\n",
       "9         ACTR1A  0.000038 -0.491126    GBM\n",
       "106         MCM6  0.001303  1.085407    GBM\n",
       "107         MCM4  0.001303  1.236484    GBM\n",
       "114       MAD2L1  0.001303  0.546914    GBM\n",
       "...          ...       ...       ...    ...\n",
       "66220      BUB1B  0.026111 -0.559000     EC\n",
       "66269       NUF2  0.034789 -0.612500     EC\n",
       "66277      CENPF  0.037588 -0.439000     EC\n",
       "66315        TPR  0.042634 -0.184300     EC\n",
       "66351    PPP2R2A  0.048016  0.372600     EC\n",
       "\n",
       "[81 rows x 4 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bool_df = all_df.Proteomics.isin(narrowed_cell_cycle)\n",
    "df = all_df[bool_df]\n",
    "plot_df = df.loc[df['P_Value'] <= 0.05] # Only plot sig genes\n",
    "plot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "ordered_prot_list = ['MSH2', 'MSH6', 'GINS2', 'GINS4', 'MCM2', 'MCM3', 'MCM4', 'MCM5', 'MCM6', 'MCM7',\n",
    "                     'RFC2', 'RFC3', 'RFC4', 'RFC5',  'PCNA', 'TOP2A', 'CDK11B', 'PPP2R2A', 'PPP2R2D', \n",
    "                     'TOPBP1', 'MAD2L1', 'BUB1B', 'CENPF', 'NDC80', 'NUF2', 'TPX2', 'PCM1', 'DCTN1','ACTR1A']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-10-a1aee8cd8482>:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  plot_df[\"Index\"] = plot_df[\"Proteomics\"] + \"_\" + plot_df[\"Cancer\"]\n"
     ]
    }
   ],
   "source": [
    "# Reorder index to reorder the protein order in heatmap\n",
    "plot_df[\"Index\"] = plot_df[\"Proteomics\"] + \"_\" + plot_df[\"Cancer\"]\n",
    "plot_df = plot_df.set_index(\"Index\")\n",
    "\n",
    "ordered_list = []\n",
    "cancer_list = ['BR', 'CO', 'EC', 'GBM', 'HNSCC', 'LSCC', 'LUAD', 'OV']\n",
    "for gene in ordered_prot_list:\n",
    "    for cancer in cancer_list:\n",
    "        ordered_list.append(gene+'_'+cancer)\n",
    "\n",
    "ordered_df = plot_df.reindex(ordered_list)\n",
    "ordered_df = ordered_df.loc[ordered_df['P_Value'] < 0.05] # Keep sig "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "min p-value: 1.5008898009485896e-05\n"
     ]
    }
   ],
   "source": [
    "print('min p-value:', ordered_df.P_Value.min())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"1053\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"1053\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.3.3.min.js\": \"dM3QQsP+wXdHg42wTqW85BjZQdLNNIXqlPw/BgKoExPmTG7ZLML4EGqLMfqHT6ON\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.3.3.min.js\": \"8x57I4YuIfu8XyZfFo0XVr2WAT8EK4rh/uDe3wF7YuW2FNUSNEpJbsPaB1nJ2fz2\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.3.3.min.js\": \"3QTqdz9LyAm2i0sG5XTePsHec3UHWwVsrOL68SYRoAXsafvfAyqtQ+h440+qIBhS\"};\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.3.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.3.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.3.3.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"1053\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"1053\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.3.3.min.js\": \"dM3QQsP+wXdHg42wTqW85BjZQdLNNIXqlPw/BgKoExPmTG7ZLML4EGqLMfqHT6ON\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.3.3.min.js\": \"8x57I4YuIfu8XyZfFo0XVr2WAT8EK4rh/uDe3wF7YuW2FNUSNEpJbsPaB1nJ2fz2\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.3.3.min.js\": \"3QTqdz9LyAm2i0sG5XTePsHec3UHWwVsrOL68SYRoAXsafvfAyqtQ+h440+qIBhS\"};\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.3.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.3.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.3.3.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"1053\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"d742b56b-5363-4a98-a654-a5b21176020e\" data-root-id=\"1054\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"80eeb155-ab88-40ba-8c42-80d26248b0bb\":{\"defs\":[],\"roots\":{\"references\":[{\"attributes\":{\"children\":[{\"id\":\"1005\"},{\"id\":\"1031\"}]},\"id\":\"1054\",\"type\":\"Row\"},{\"attributes\":{\"axis\":{\"id\":\"1038\"},\"ticker\":null},\"id\":\"1040\",\"type\":\"Grid\"},{\"attributes\":{\"data\":{\"Cancer\":[\"EC\",\"GBM\",\"LSCC\",\"EC\",\"GBM\",\"LSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"LSCC\",\"GBM\",\"HNSCC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"HNSCC\",\"EC\",\"GBM\",\"EC\",\"GBM\",\"HNSCC\",\"GBM\",\"LUAD\"],\"Index\":[\"MSH2_EC\",\"MSH2_GBM\",\"MSH2_LSCC\",\"MSH6_EC\",\"MSH6_GBM\",\"MSH6_LSCC\",\"GINS2_GBM\",\"GINS2_HNSCC\",\"GINS4_GBM\",\"GINS4_HNSCC\",\"MCM2_GBM\",\"MCM2_HNSCC\",\"MCM3_GBM\",\"MCM3_HNSCC\",\"MCM4_GBM\",\"MCM4_HNSCC\",\"MCM5_GBM\",\"MCM5_HNSCC\",\"MCM6_GBM\",\"MCM7_GBM\",\"MCM7_HNSCC\",\"RFC2_EC\",\"RFC2_GBM\",\"RFC2_HNSCC\",\"RFC3_GBM\",\"RFC3_HNSCC\",\"RFC3_LSCC\",\"RFC4_GBM\",\"RFC4_HNSCC\",\"PCNA_GBM\",\"PCNA_HNSCC\",\"TOP2A_EC\",\"TOP2A_GBM\",\"TOP2A_HNSCC\",\"CDK11B_EC\",\"CDK11B_GBM\",\"PPP2R2A_EC\",\"PPP2R2A_GBM\",\"PPP2R2D_GBM\",\"PPP2R2D_HNSCC\",\"TOPBP1_EC\",\"TOPBP1_GBM\",\"MAD2L1_GBM\",\"MAD2L1_HNSCC\",\"BUB1B_EC\",\"BUB1B_GBM\",\"CENPF_EC\",\"CENPF_GBM\",\"NDC80_EC\",\"NDC80_GBM\",\"NUF2_EC\",\"NUF2_GBM\",\"NUF2_HNSCC\",\"TPX2_EC\",\"TPX2_GBM\",\"TPX2_HNSCC\",\"PCM1_EC\",\"PCM1_GBM\",\"DCTN1_EC\",\"DCTN1_GBM\",\"DCTN1_HNSCC\",\"ACTR1A_GBM\",\"ACTR1A_LUAD\"],\"Medians\":{\"__ndarray__\":\"ObTIdr6f2r+eLRrKvcnaP/JjzF1LyPE/yHa+nxov4b8ajRzL5YTdP7n8h/Tb1/Y/6kCVCck14T/AfxZoNzzgPwb+P2Aqv+c//5XQfphC3j87XeIEfhDuP3/OiS7nI94/kUU5gxmm7T+/f0WW7iTfPwM+rlCjyPM/YA8s0rzX4D+kC0IWaNvrP0AghV9Udd4/SJxVDNRd8T8bdU1XUpXwPwAtQ/CqfOI/AiuHFtnO17+g6+yHTzXYP7/L5omTxdc/pgTEWRtl3D8/ydlg1hPQP1TBqKROQOI/bpisYeAC1z8/9h0nYZDcP7A27f4iNeQ/gGCiMmft2j8pXI/C9Sjkv2a+pWNGOuk/YN4/WXHm4z9Qwkzbv7LUv9vYSSMTd9g/m1Wfq63Y1z+y3poEvHjOv+us3Kanxt6//XmXb6Tjzb+8BRIUP8bjvzcWEnuCjt8/Yo/6p1GA4T+ASJJ/RafYPxfZzvdT4+G/HTASqIiN5j9/arx0kxjcv9h3xl2pueE/2/l+arx0479qXFthJIfkP5mZmZmZmeO/aeLlr4b/7z8gUxKeO4HlP5qZmZmZmem/0ABVqwV85D//CvxmsGbfP+F6FK5H4do/R4okRS6t1j8dOGdEaW/QPzGtsgfegti//Rt4L7n/vr+jU3Fmm27fv3e+nxov3eG/\",\"dtype\":\"float64\",\"order\":\"little\",\"shape\":[63]},\"P_Value\":{\"__ndarray__\":\"zaOQJAJxlT+u06xyLSqcP5SNLgXSeqY/ZyINEl8/lD+kiGIc/gGjP4gj1mUhqaY/vQqhPw48pD+5HGHVf9qOP6isX6bJh5I/XsmUHKCtbz9gq/3h9TJiP1lu4LOpM6A/bQfA+M+Jaz/GKcuaYIqbP8xWVsYzWVU/D57D9OKvnz99/nJ2kStiP6Wb8KEhjaM/zFZWxjNZVT9SRsRyUAplP6/BC3JWsp8/GHjW4HYqmj8M/Pu067mDP+crC+/VNpk/DBSErH8vcz+mAbNND46RP6Z+hkHKUaM/YfoEUwFLiD+r+Mdo/RCYP/ym1QGhUVk/+yQLvD4znT/IF0YRtbSTP+c//kMABqU/x4UT4P5wgT+ij84+PW6WPy+NjeZ9E3g/A2NkMo+VqD+E9v12Nb6YP0a7bmgkPp4/GAKY7YHgmD/u4kGX6lhUP6ks0hyF54k/zFZWxjNZVT8sRHn0NQCiP+z0U9S9vJo/6EwJosl+pD8Vs7BdwT6jPzXYU7uQkpE/Q1ym1lcFjz+/5L0UFt2SP3AuYTPXz6E/pYu3H24HhD8NHsrfvHSYP8padpBa4pU/tnT2DrU5oD+kW1n1knOkP6tktpEl54g/SOXTZ0PqZD8KRLxyv4qQP2GGNTzXee8+pwW8L29spD9ijDvoUCwEP1efSs7T33w/\",\"dtype\":\"float64\",\"order\":\"little\",\"shape\":[63]},\"Proteomics\":[\"MSH2\",\"MSH2\",\"MSH2\",\"MSH6\",\"MSH6\",\"MSH6\",\"GINS2\",\"GINS2\",\"GINS4\",\"GINS4\",\"MCM2\",\"MCM2\",\"MCM3\",\"MCM3\",\"MCM4\",\"MCM4\",\"MCM5\",\"MCM5\",\"MCM6\",\"MCM7\",\"MCM7\",\"RFC2\",\"RFC2\",\"RFC2\",\"RFC3\",\"RFC3\",\"RFC3\",\"RFC4\",\"RFC4\",\"PCNA\",\"PCNA\",\"TOP2A\",\"TOP2A\",\"TOP2A\",\"CDK11B\",\"CDK11B\",\"PPP2R2A\",\"PPP2R2A\",\"PPP2R2D\",\"PPP2R2D\",\"TOPBP1\",\"TOPBP1\",\"MAD2L1\",\"MAD2L1\",\"BUB1B\",\"BUB1B\",\"CENPF\",\"CENPF\",\"NDC80\",\"NDC80\",\"NUF2\",\"NUF2\",\"NUF2\",\"TPX2\",\"TPX2\",\"TPX2\",\"PCM1\",\"PCM1\",\"DCTN1\",\"DCTN1\",\"DCTN1\",\"ACTR1A\",\"ACTR1A\"],\"size\":{\"__ndarray__\":\"n24Sm2YyJ0CmudSHdo8lQB6IC8YWwSJAsa05tWaKJ0COnsR7x8IjQKtzdCjHtCJAuSCMwrViI0BOAaf3FSwpQO67VmyFEihAoGJUPnWqMEDkO+YPIVQyQCz0DTAeuCRAFpuvUwQWMUBd2xp147ElQByi9/ri7TNAbx9jLnfaJECgeWVDWVUyQG4mIZJ5lyNAHKL3+uLtM0A/gKNWt+QxQJeUB2IA2iRAsKzMO4kAJkADhA+UI9srQO0edWNxOSZA2rAiekAXMEC+W0ZSlmUoQHV+KG/LqSNAs44/TEmbKkDwNaAmAoEmQNWTEGnjajNAyt3MHwJYJUDiiqwsDbQnQPXPwm73JyNAHoI+KjqYLEAcHjnsH+0mQEQ3I6C70S5AZ8TtSKI3IkBLexBFa1YmQGwWbIIfIiVARG+K0B9OJkBRrQHsxRI0QBbxnVyjOCpAHKL3+uLtM0DVHGimYhYkQF9ohdNa3yVA5X1rgwxPI0Bw/aq2t68jQMVJg0sMZChAUJ0U0MYjKUCxZn/HJfcnQH5JRNyXJiRASg0VGL7DK0BNOg/gVmgmQBKByJQBEydA07/1kuG1JEByouTNVVIjQBm0rKgydSpA7tzIvEzpMUAKl8M04MAoQP2XeHwJqUBAGMLDZW5UI0BZxv79D38+QKwiGuOXui1A\",\"dtype\":\"float64\",\"order\":\"little\",\"shape\":[63]},\"size2\":{\"__ndarray__\":\"f5MYJN7tDkAyohu1SL8MQNIKurJzAQlAl+dMnDNjD0ASflv6CVkKQI/vRYsJ8QhATNa6WPLYCUDfABqlDsgQQEl95PJYDBBAKy4bU5w4FkCG+jIVLHAYQOaaEkAooAtAHSTqxAXIFkDRec5GL+0MQHotSvmDkhpAP3/ZPfTNC0ArotxZzHEYQOjd1sJMHwpAei1K+YOSGkCpVS8en9sXQHQbCthVzQtA65C7TwxWDUACWAq4F5ISQDwpnITsoQ1AI0Eu+FV0FUDU5y6MuUMQQEdTi+m5NwpAIl8qiNu8EUDrR4AzWAEOQMdva+HZ4xlADn1mKlh1DEAtuZA7vJoPQEcVWT6figlAv1bUxiYQE0B7faHlf5EOQNgkwmrSixRANFuSti1KCEBkpMAGj8gNQOYdkFh/LQxAW5S4wH+9DUBsPAKQssMaQA/2E5MXexFAei1K+YOSGkAc0YqIg8gKQCmLXG/OKQ1A3KfkWbu+CUDqUTmenz8KQC4xAt2yQhBA4Gi4ioTCEEDsiFRfMvQPQKgMW9Af3gpAh7O4un6CEkARo2nVc+ANQG0BthsCxA5AGVXywyydC0DtLYYSHcMJQGbNHRt3oxFAkyZhprvhF0CxZIJ4lYAQQKfKoFC3NiZAdq2vh+jFCUDmLv9TtVQkQMjBZpe60RNA\",\"dtype\":\"float64\",\"order\":\"little\",\"shape\":[63]}},\"selected\":{\"id\":\"1073\"},\"selection_policy\":{\"id\":\"1072\"}},\"id\":\"1021\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"formatter\":{\"id\":\"1070\"},\"major_label_policy\":{\"id\":\"1071\"},\"major_label_text_font_size\":\"11pt\",\"ticker\":{\"id\":\"1042\"}},\"id\":\"1041\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"callback\":null},\"id\":\"1018\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"1057\",\"type\":\"Title\"},{\"attributes\":{},\"id\":\"1016\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1002\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\"},\"x\":{\"field\":\"Proteomics\"},\"y\":{\"field\":\"Cancer\"}},\"id\":\"1024\",\"type\":\"Scatter\"},{\"attributes\":{\"data_source\":{\"id\":\"1047\"},\"glyph\":{\"id\":\"1049\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1050\"},\"view\":{\"id\":\"1052\"}},\"id\":\"1051\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"factors\":[\"LUAD\",\"LSCC\",\"HNSCC\",\"GBM\",\"EC\"]},\"id\":\"1004\",\"type\":\"FactorRange\"},{\"attributes\":{\"factors\":[\"5.0e-02\",\"1.0e-03\",\"1.0e-05\"]},\"id\":\"1030\",\"type\":\"FactorRange\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1028\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"1050\",\"type\":\"Scatter\"},{\"attributes\":{\"below\":[{\"id\":\"1012\"}],\"center\":[{\"id\":\"1014\"},{\"id\":\"1017\"}],\"height\":350,\"left\":[{\"id\":\"1015\"}],\"renderers\":[{\"id\":\"1025\"}],\"right\":[{\"id\":\"1027\"}],\"title\":{\"id\":\"1055\"},\"toolbar\":{\"id\":\"1019\"},\"toolbar_location\":null,\"width\":800,\"x_range\":{\"id\":\"1003\"},\"x_scale\":{\"id\":\"1008\"},\"y_range\":{\"id\":\"1004\"},\"y_scale\":{\"id\":\"1010\"}},\"id\":\"1005\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"axis_label\":\"FDR p-value\",\"formatter\":{\"id\":\"1067\"},\"major_label_orientation\":1.5707963267948966,\"major_label_policy\":{\"id\":\"1068\"},\"major_label_text_font_size\":\"11pt\",\"ticker\":{\"id\":\"1039\"}},\"id\":\"1038\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"1055\",\"type\":\"Title\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1028\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"1049\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"1061\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{},\"id\":\"1062\",\"type\":\"AllLabels\"},{\"attributes\":{\"below\":[{\"id\":\"1038\"}],\"center\":[{\"id\":\"1040\"},{\"id\":\"1043\"}],\"height\":200,\"left\":[{\"id\":\"1041\"}],\"renderers\":[{\"id\":\"1051\"}],\"title\":{\"id\":\"1057\"},\"toolbar\":{\"id\":\"1045\"},\"toolbar_location\":null,\"width\":140,\"x_range\":{\"id\":\"1029\"},\"x_scale\":{\"id\":\"1034\"},\"y_range\":{\"id\":\"1030\"},\"y_scale\":{\"id\":\"1036\"}},\"id\":\"1031\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{},\"id\":\"1067\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1002\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\"},\"x\":{\"field\":\"Proteomics\"},\"y\":{\"field\":\"Cancer\"}},\"id\":\"1023\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"1039\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"1068\",\"type\":\"AllLabels\"},{\"attributes\":{},\"id\":\"1036\",\"type\":\"CategoricalScale\"},{\"attributes\":{},\"id\":\"1013\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"data\":{\"Medians\":{\"__ndarray__\":\"AAAAAAAA+D8AAAAAAAD4PwAAAAAAAPg/\",\"dtype\":\"float64\",\"order\":\"little\",\"shape\":[3]},\"P_Value\":{\"__ndarray__\":\"mpmZmZmZqT/8qfHSTWJQP/Fo44i1+OQ+\",\"dtype\":\"float64\",\"order\":\"little\",\"shape\":[3]},\"index\":[0,1,2],\"size\":{\"__ndarray__\":\"reqW3HH5IUC4/yvzJ7k0QMT/+Z/2REFA\",\"dtype\":\"float64\",\"order\":\"little\",\"shape\":[3]},\"size2\":{\"__ndarray__\":\"keNze0L3B0Cg/4+ZiqEbQFuqoiqeBidA\",\"dtype\":\"float64\",\"order\":\"little\",\"shape\":[3]},\"x_axis\":[\"\",\"\",\"\"],\"y_axis\":[\"5.0e-02\",\"1.0e-03\",\"1.0e-05\"]},\"selected\":{\"id\":\"1075\"},\"selection_policy\":{\"id\":\"1074\"}},\"id\":\"1047\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"1059\",\"type\":\"NoOverlap\"},{\"attributes\":{},\"id\":\"1074\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"axis\":{\"id\":\"1015\"},\"dimension\":1,\"ticker\":null},\"id\":\"1017\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"1075\",\"type\":\"Selection\"},{\"attributes\":{\"formatter\":{\"id\":\"1064\"},\"major_label_policy\":{\"id\":\"1065\"},\"major_label_text_font_size\":\"12pt\",\"ticker\":{\"id\":\"1016\"}},\"id\":\"1015\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"factors\":[\"MSH2\",\"MSH6\",\"GINS2\",\"GINS4\",\"MCM2\",\"MCM3\",\"MCM4\",\"MCM5\",\"MCM6\",\"MCM7\",\"RFC2\",\"RFC3\",\"RFC4\",\"PCNA\",\"TOP2A\",\"CDK11B\",\"PPP2R2A\",\"PPP2R2D\",\"TOPBP1\",\"MAD2L1\",\"BUB1B\",\"CENPF\",\"NDC80\",\"NUF2\",\"TPX2\",\"PCM1\",\"DCTN1\",\"ACTR1A\"]},\"id\":\"1003\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"1010\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"active_multi\":null,\"tools\":[{\"id\":\"1044\"}]},\"id\":\"1045\",\"type\":\"Toolbar\"},{\"attributes\":{\"source\":{\"id\":\"1021\"}},\"id\":\"1026\",\"type\":\"CDSView\"},{\"attributes\":{\"axis\":{\"id\":\"1041\"},\"dimension\":1,\"ticker\":null},\"id\":\"1043\",\"type\":\"Grid\"},{\"attributes\":{\"source\":{\"id\":\"1047\"}},\"id\":\"1052\",\"type\":\"CDSView\"},{\"attributes\":{},\"id\":\"1072\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"1073\",\"type\":\"Selection\"},{\"attributes\":{\"high\":1.4277,\"low\":-1.4277,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"1002\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"color_mapper\":{\"id\":\"1002\"},\"location\":[0,0],\"major_label_policy\":{\"id\":\"1059\"}},\"id\":\"1027\",\"type\":\"ColorBar\"},{\"attributes\":{\"callback\":null},\"id\":\"1044\",\"type\":\"HoverTool\"},{\"attributes\":{\"active_multi\":null,\"tools\":[{\"id\":\"1018\"}]},\"id\":\"1019\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"1042\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"1070\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{},\"id\":\"1071\",\"type\":\"AllLabels\"},{\"attributes\":{\"high\":1.5,\"low\":1.5,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"1028\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"factors\":[\"\"]},\"id\":\"1029\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"1008\",\"type\":\"CategoricalScale\"},{\"attributes\":{},\"id\":\"1064\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"formatter\":{\"id\":\"1061\"},\"major_label_orientation\":1.5707963267948966,\"major_label_policy\":{\"id\":\"1062\"},\"major_label_text_font_size\":\"12pt\",\"ticker\":{\"id\":\"1013\"}},\"id\":\"1012\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"1065\",\"type\":\"AllLabels\"},{\"attributes\":{\"data_source\":{\"id\":\"1021\"},\"glyph\":{\"id\":\"1023\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1024\"},\"view\":{\"id\":\"1026\"}},\"id\":\"1025\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"axis\":{\"id\":\"1012\"},\"ticker\":null},\"id\":\"1014\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"1034\",\"type\":\"CategoricalScale\"}],\"root_ids\":[\"1054\"]},\"title\":\"Bokeh Application\",\"version\":\"2.3.3\"}};\n",
       "  var render_items = [{\"docid\":\"80eeb155-ab88-40ba-8c42-80d26248b0bb\",\"root_ids\":[\"1054\"],\"roots\":{\"1054\":\"d742b56b-5363-4a98-a654-a5b21176020e\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "1054"
      }
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div style=\"display: table;\"><div style=\"display: table-row;\"><div style=\"display: table-cell;\"><b title=\"bokeh.plotting.figure.Figure\">Figure</b>(</div><div style=\"display: table-cell;\">id&nbsp;=&nbsp;'1005', <span id=\"1415\" style=\"cursor: pointer;\">&hellip;)</span></div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">above&nbsp;=&nbsp;[],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">align&nbsp;=&nbsp;'start',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">aspect_ratio&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">aspect_scale&nbsp;=&nbsp;1,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">background&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">background_fill_alpha&nbsp;=&nbsp;1.0,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">background_fill_color&nbsp;=&nbsp;'#ffffff',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">below&nbsp;=&nbsp;[CategoricalAxis(id='1012', ...)],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">border_fill_alpha&nbsp;=&nbsp;1.0,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">border_fill_color&nbsp;=&nbsp;'#ffffff',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">center&nbsp;=&nbsp;[Grid(id='1014', ...), Grid(id='1017', ...)],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">css_classes&nbsp;=&nbsp;[],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">disabled&nbsp;=&nbsp;False,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">extra_x_ranges&nbsp;=&nbsp;{},</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">extra_y_ranges&nbsp;=&nbsp;{},</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">frame_height&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">frame_width&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">height&nbsp;=&nbsp;350,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">height_policy&nbsp;=&nbsp;'auto',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">hidpi&nbsp;=&nbsp;True,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">inner_height&nbsp;=&nbsp;0,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">inner_width&nbsp;=&nbsp;0,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">js_event_callbacks&nbsp;=&nbsp;{},</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">js_property_callbacks&nbsp;=&nbsp;{},</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">left&nbsp;=&nbsp;[CategoricalAxis(id='1015', ...)],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">lod_factor&nbsp;=&nbsp;10,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">lod_interval&nbsp;=&nbsp;300,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">lod_threshold&nbsp;=&nbsp;2000,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">lod_timeout&nbsp;=&nbsp;500,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">margin&nbsp;=&nbsp;(0, 0, 0, 0),</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">match_aspect&nbsp;=&nbsp;False,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">max_height&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">max_width&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">min_border&nbsp;=&nbsp;5,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">min_border_bottom&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">min_border_left&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">min_border_right&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">min_border_top&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">min_height&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">min_width&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">name&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outer_height&nbsp;=&nbsp;0,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outer_width&nbsp;=&nbsp;0,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outline_line_alpha&nbsp;=&nbsp;1.0,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outline_line_cap&nbsp;=&nbsp;'butt',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outline_line_color&nbsp;=&nbsp;'#e5e5e5',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outline_line_dash&nbsp;=&nbsp;[],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outline_line_dash_offset&nbsp;=&nbsp;0,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outline_line_join&nbsp;=&nbsp;'bevel',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">outline_line_width&nbsp;=&nbsp;1,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">output_backend&nbsp;=&nbsp;'canvas',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">renderers&nbsp;=&nbsp;[GlyphRenderer(id='1025', ...)],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">reset_policy&nbsp;=&nbsp;'standard',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">right&nbsp;=&nbsp;[ColorBar(id='1027', ...)],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">sizing_mode&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">subscribed_events&nbsp;=&nbsp;[],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">syncable&nbsp;=&nbsp;True,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">tags&nbsp;=&nbsp;[],</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">title&nbsp;=&nbsp;Title(id='1055', ...),</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">title_location&nbsp;=&nbsp;'above',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">toolbar&nbsp;=&nbsp;Toolbar(id='1019', ...),</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">toolbar_location&nbsp;=&nbsp;None,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">toolbar_sticky&nbsp;=&nbsp;True,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">visible&nbsp;=&nbsp;True,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">width&nbsp;=&nbsp;800,</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">width_policy&nbsp;=&nbsp;'auto',</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">x_range&nbsp;=&nbsp;FactorRange(id='1003', ...),</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">x_scale&nbsp;=&nbsp;CategoricalScale(id='1008', ...),</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">y_range&nbsp;=&nbsp;FactorRange(id='1004', ...),</div></div><div class=\"1414\" style=\"display: none;\"><div style=\"display: table-cell;\"></div><div style=\"display: table-cell;\">y_scale&nbsp;=&nbsp;CategoricalScale(id='1010', ...))</div></div></div>\n",
       "<script>\n",
       "(function() {\n",
       "  var expanded = false;\n",
       "  var ellipsis = document.getElementById(\"1415\");\n",
       "  ellipsis.addEventListener(\"click\", function() {\n",
       "    var rows = document.getElementsByClassName(\"1414\");\n",
       "    for (var i = 0; i < rows.length; i++) {\n",
       "      var el = rows[i];\n",
       "      el.style.display = expanded ? \"none\" : \"table-row\";\n",
       "    }\n",
       "    ellipsis.innerHTML = expanded ? \"&hellip;)\" : \"&lsaquo;&lsaquo;&lsaquo;\";\n",
       "    expanded = !expanded;\n",
       "  });\n",
       "})();\n",
       "</script>\n"
      ],
      "text/plain": [
       "Figure(id='1005', ...)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p.plotCircleHeatMap(ordered_df, circle_var='P_Value', color_var='Medians', x_axis='Proteomics', y_axis='Cancer',\n",
    "                    plot_height=350, plot_width=800, x_axis_lab='', font_size=12, legend_max=0.05, \n",
    "                    legend_med = 0.001, legend_min=0.00001, save_png='Fig_6_Cell_Cycle.png')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The manuscript mentions the average of the FDR p-values for MCM complex proteins in EC. Here is where we calculate the average for these p-values."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Average of FDR p-values for MCM proteins in EC: 0.11949892063850753\n"
     ]
    }
   ],
   "source": [
    "# average FDR p-val of MCM2-7 in EC\n",
    "mcm = ['MCM2','MCM3','MCM4','MCM5','MCM6','MCM7']\n",
    "bool_df = all_df.Proteomics.isin(mcm)\n",
    "df = all_df[bool_df]\n",
    "df = df.loc[df['Cancer'] == 'EC'] \n",
    "print('Average of FDR p-values for MCM proteins in EC:', df.P_Value.mean())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The manuscript also mentions the strong FDR p-values for DCTN1 and ACTR1A in GBM as shown below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FDR p-values of DCTN1 nd ACTR1A in GBM:\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DCTN1</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>-0.382988</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ACTR1A</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>-0.491126</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Proteomics   P_Value   Medians Cancer\n",
       "6      DCTN1  0.000015 -0.382988    GBM\n",
       "9     ACTR1A  0.000038 -0.491126    GBM"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# P-values for DCTN1 and ACTR1A in GBM\n",
    "axon_health = ['DCTN1','ACTR1A']\n",
    "bool_df = all_df.Proteomics.isin(axon_health)\n",
    "df = all_df[bool_df]\n",
    "df = df.loc[df['Cancer'] == 'GBM'] \n",
    "print('FDR p-values of DCTN1 nd ACTR1A in GBM:\\n')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
