{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find cell cycle genes without enough data to do a t-test in each cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import warnings\n",
    "#warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Returns a dataframe with proteomics and mutation type\n",
    "\n",
    "# all_prot: list of trans genes\n",
    "\n",
    "def all_prot_format_df(cancer_object, all_prot, gene_in = 'PTEN', utils = u):\n",
    "    mut_type = cancer_object.get_genotype_all_vars(gene_in)\n",
    "    \n",
    "    if cancer_object.get_cancer_type() not in ('luad', 'endometrial'):\n",
    "        # Keep only tumor samples from proteomics\n",
    "        prot_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "            mutations_genes = [gene_in], omics_df_name = 'proteomics', omics_genes = all_prot, tissue_type = 'tumor')\n",
    "        # Reduce a multiindex \n",
    "        if isinstance(prot_and_mutations.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "            prot_and_mutations = utils.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "            prot_and_mutations = rename_duplicate_cols(prot_and_mutations)\n",
    "        \n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        mut_type = mut_type[['Mutation']] \n",
    "        prot_df = prot_and_mutations.iloc[:,:-4] # Keep only proteomics\n",
    "        merged = prot_df.join(mut_type)\n",
    "        \n",
    "        # Keep only Wildtype and deletion\n",
    "        compare = ['Wildtype_Tumor','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "    \n",
    "    \n",
    "    # Luad has no somatic mutations for PTEN which changes some things\n",
    "    elif cancer_object.get_cancer_type() in ('luad'): \n",
    "        # get_genotype_all_vars add cnv data under the column PTEN\n",
    "        mut_type = mut_type.drop(columns= gene_in)\n",
    "        # different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "        omics = cancer_object.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics',genes1= gene_in, \n",
    "            genes2= all_prot, tissue_type = 'tumor')\n",
    "        omics = utils.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "        omics = rename_duplicate_cols(omics)\n",
    "        omics = omics.drop(columns='PTEN_CNV')\n",
    "       \n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        merged = omics.join(mut_type) \n",
    "        # Keep only No_Mutation (wildtype) and deletion\n",
    "        compare = ['No_Mutation','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "        mut_wt['Mutation'] = np.where(\n",
    "            mut_wt['Mutation'] == 'No_Mutation', 'Wildtype_Tumor', 'Deletion')\n",
    "        \n",
    "    elif cancer_object.get_cancer_type() in ('endometrial'):\n",
    "\n",
    "        # merge cnv with genotype all mut type\n",
    "        prot_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "            mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = all_prot, tissue_type = 'tumor') # drop Normal samples\n",
    "\n",
    "        prot_df = prot_and_mutations.iloc[:,:-4] #drop Mutation and Location cols\n",
    "        mut_type = mut_type[['Mutation']] # Get Mutation col that includes CNV\n",
    "        merged = prot_df.join(mut_type) # merge \n",
    "\n",
    "        # Create Truncation category and keep truncation and wt\n",
    "        compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "        mut_wt['Mutation'] = np.where(\n",
    "                        mut_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "    \n",
    "    mut_wt = mut_wt.rename(columns={col: re.sub(r'_proteomics', '', col) for col in mut_wt.columns.tolist()})\n",
    "    return mut_wt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Adds '_i' to duplicate col names where i is a number (increases with each duplicate)\n",
    "\n",
    "def rename_duplicate_cols(df):\n",
    "    cols = pd.Series(df.columns[:])\n",
    "\n",
    "    for dup in cols[cols.duplicated()].unique(): \n",
    "        cols[cols[cols == dup].index.values.tolist()] = [dup + '_' + str(i) if i != 0 else dup for i in range(sum(cols == dup))]\n",
    "\n",
    "    # rename the columns with the cols list.\n",
    "    df.columns=cols\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that endometrial index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that luad index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The HNSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that ovarian index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                            \r"
     ]
    }
   ],
   "source": [
    "g = cptac.Gbm()\n",
    "en = cptac.Endometrial()\n",
    "h = cptac.Hnscc()\n",
    "l = cptac.Luad()\n",
    "ls = cptac.Lscc()\n",
    "o = cptac.Ovarian()\n",
    "col = cptac.Colon()\n",
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "df: DataFrame containing omics data and a binary column.\n",
    "mincount: Int of the minimum num of actual values needed to pass cutoff and not be listed in not_enough_data list. \n",
    "omics_cols: Names of columns to check if there is enough data. \n",
    "\n",
    "Returns: List of genes with not enough data (num of non-NaN values are less than mincount).\n",
    "'''\n",
    "\n",
    "def find_few_data_genes(df, mincount, omics_cols = None, binary_labels = ('Deletion','Wildtype_Tumor'), \n",
    "    binary_col = 'Mutation'):\n",
    "    \n",
    "    # Separate into binary groups\n",
    "    label_1 = binary_labels[0]\n",
    "    label_2 = binary_labels[1]\n",
    "    partition1 = df.loc[df[binary_col] == label_1]\n",
    "    partition2 = df.loc[df[binary_col] == label_2]\n",
    "    \n",
    "    # Get list of columns\n",
    "    if omics_cols is None:\n",
    "        omics_cols = list(df.columns)\n",
    "        omics_cols.remove(binary_col)\n",
    "    else:\n",
    "        omics_cols = set(omics_cols)\n",
    "        if binary_col in (omics_cols):\n",
    "            omics_cols.remove(binary_col)\n",
    "        \n",
    "    # Append genes with less data than mincount (for mut or wt) to list\n",
    "    not_enough_data = []\n",
    "    for c in omics_cols:\n",
    "        #print(len(partition1[c].dropna(axis='rows')))\n",
    "        #print(len(partition2[c].dropna(axis='rows')))\n",
    "        if len(partition1[c].dropna(axis='rows')) <= mincount: #non-nan vals less than min\n",
    "            not_enough_data.append(c)\n",
    "            continue\n",
    "        elif len(partition2[c].dropna(axis='rows')) <= mincount:\n",
    "            not_enough_data.append(c)\n",
    "            continue\n",
    "        \n",
    "    \n",
    "    print('genes with not enough data: ', len(not_enough_data), '/', len(omics_cols))        \n",
    "    return not_enough_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_obj = {'Endo':en, 'Colon':col, 'Luad':l, 'Lscc': ls, 'Gbm': g, 'Ov': o, 'Brca': b, 'Hnscc': h}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_list = ['DCTN1', 'ACTR1A', 'MAD2L1', 'MCM4', 'PCNA', 'MCM5', 'MCM2','PCM1',\n",
    "              'MCM7', 'TPR', 'MCM3', 'MCM6', 'NUP153', 'RFC3', 'CDK11B', 'XPO1','PRKCB', 'RFC2', 'NUF2', \n",
    "              'RAB8A', 'RFC4', 'RANBP2', 'CENPF', 'GINS4', 'NDC80', 'OPTN', 'PPP2R2A', 'PPP2R2D',\n",
    "              'TPX2', 'GINS2', 'BUB1B', 'TOP2A', 'TOPBP1', 'TP53', 'NUP210', 'MSH2', 'MSH6']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Endo\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_asarray.py:83: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray\n",
      "  return array(a, dtype, copy=False, order=order)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 55)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:66: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: BUB1B, CENPF, GINS2, NUF2, TOPBP1 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 101 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes with not enough data:  0 / 37\n",
      "\n",
      "\n",
      "Colon\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: XPO1 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 34)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01890 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 34)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 35)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:46: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 96 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes with not enough data:  7 / 37\n",
      "\n",
      "\n",
      "Luad\n",
      "genes with not enough data:  1 / 37\n",
      "\n",
      "\n",
      "Lscc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 96 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 14)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes with not enough data:  0 / 37\n",
      "\n",
      "\n",
      "Gbm\n",
      "genes with not enough data:  0 / 37\n",
      "\n",
      "\n",
      "Ov\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 79 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 14)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes with not enough data:  0 / 37\n",
      "\n",
      "\n",
      "Brca\n",
      "genes with not enough data:  1 / 37\n",
      "\n",
      "\n",
      "Hnscc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: DCTN1 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes with not enough data:  0 / 37\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 107 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    }
   ],
   "source": [
    "mincount = 5\n",
    "gene = 'PTEN'\n",
    "\n",
    "few_data_prot = {}\n",
    "for c in cancer_obj:\n",
    "    print(c)\n",
    "    #prot = cancer_obj[c].get_proteomics()\n",
    "    #if isinstance(prot.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "    #    prot = u.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "    #prot_list = list(prot.columns)\n",
    "    mut_wt = all_prot_format_df(cancer_obj[c], prot_list)\n",
    "    if cancer_obj[c].get_cancer_type() in ('endometrial'):\n",
    "        labels = ('Truncation', 'Wildtype_Tumor')\n",
    "    else:\n",
    "        labels = ('Deletion', 'Wildtype_Tumor')\n",
    "    nd_list = find_few_data_genes(mut_wt, mincount, omics_cols = prot_list, binary_labels = labels)\n",
    "    few_data_prot[c] = nd_list\n",
    "    print('\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Endo': [],\n",
       " 'Colon': ['NUF2', 'GINS4', 'CENPF', 'NDC80', 'BUB1B', 'TOPBP1', 'GINS2'],\n",
       " 'Luad': ['XPO1'],\n",
       " 'Lscc': [],\n",
       " 'Gbm': [],\n",
       " 'Ov': [],\n",
       " 'Brca': ['DCTN1'],\n",
       " 'Hnscc': []}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "few_data_prot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "col = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Endo\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 52)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:63: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Colon\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: BUB1B, CENPF, GINS2, NUF2, TOPBP1 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 101 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Luad\n",
      "Lscc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: XPO1 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 34)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01890 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 34)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 35)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:46: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 96 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 96 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 14)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gbm\n",
      "Ov\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 79 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 14)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: DCTN1 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Hnscc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 107 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    }
   ],
   "source": [
    "cancer_obj = {'Endo':en, 'Colon':col, 'Luad':l, 'Lscc': ls, 'Gbm': g, 'Ov': o, 'Brca': b, 'Hnscc': h}\n",
    "missing = {}\n",
    "for c in cancer_obj:\n",
    "    print(c)\n",
    "    mut_wt = all_prot_format_df(cancer_obj[c], prot_list)\n",
    "    null_columns = mut_wt.columns[mut_wt.isnull().all()]\n",
    "    missing[c] = null_columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Endo': Index([], dtype='object', name='Name'),\n",
       " 'Colon': Index(['NUF2', 'CENPF', 'GINS2', 'BUB1B', 'TOPBP1'], dtype='object', name='Name'),\n",
       " 'Luad': Index(['XPO1'], dtype='object', name='Name'),\n",
       " 'Lscc': Index([], dtype='object', name='Name'),\n",
       " 'Gbm': Index([], dtype='object', name='Name'),\n",
       " 'Ov': Index([], dtype='object', name='Name'),\n",
       " 'Brca': Index(['DCTN1'], dtype='object', name='Name'),\n",
       " 'Hnscc': Index([], dtype='object', name='Name')}"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "BUB1B, CENPF, GINS2, NUF2, TOPBP1 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_asarray.py:83: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray\n",
      "  return array(a, dtype, copy=False, order=order)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: BUB1B, CENPF, GINS2, NUF2, TOPBP1 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 101 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>DCTN1</th>\n",
       "      <th>ACTR1A</th>\n",
       "      <th>MAD2L1</th>\n",
       "      <th>MCM4</th>\n",
       "      <th>PCNA</th>\n",
       "      <th>MCM5</th>\n",
       "      <th>MCM2</th>\n",
       "      <th>PCM1</th>\n",
       "      <th>MCM7</th>\n",
       "      <th>TPR</th>\n",
       "      <th>...</th>\n",
       "      <th>TPX2</th>\n",
       "      <th>GINS2</th>\n",
       "      <th>BUB1B</th>\n",
       "      <th>TOP2A</th>\n",
       "      <th>TOPBP1</th>\n",
       "      <th>TP53</th>\n",
       "      <th>NUP210</th>\n",
       "      <th>MSH2</th>\n",
       "      <th>MSH6</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01CO001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO005</th>\n",
       "      <td>-0.2120</td>\n",
       "      <td>0.0468</td>\n",
       "      <td>-0.0627</td>\n",
       "      <td>0.1760</td>\n",
       "      <td>0.0313</td>\n",
       "      <td>0.0517</td>\n",
       "      <td>0.0363</td>\n",
       "      <td>-0.2020</td>\n",
       "      <td>0.258</td>\n",
       "      <td>0.1740</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.277</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.489</td>\n",
       "      <td>0.5620</td>\n",
       "      <td>0.3090</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO006</th>\n",
       "      <td>-0.1250</td>\n",
       "      <td>0.0500</td>\n",
       "      <td>-0.1620</td>\n",
       "      <td>0.2420</td>\n",
       "      <td>-0.0178</td>\n",
       "      <td>0.1790</td>\n",
       "      <td>0.2330</td>\n",
       "      <td>0.0144</td>\n",
       "      <td>0.215</td>\n",
       "      <td>0.0299</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6340</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.796</td>\n",
       "      <td>-0.3520</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>0.0907</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO008</th>\n",
       "      <td>-0.2510</td>\n",
       "      <td>-0.3780</td>\n",
       "      <td>-0.0747</td>\n",
       "      <td>-0.4790</td>\n",
       "      <td>-0.3960</td>\n",
       "      <td>-0.3320</td>\n",
       "      <td>-0.4490</td>\n",
       "      <td>0.1920</td>\n",
       "      <td>-0.368</td>\n",
       "      <td>0.3660</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6550</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2520</td>\n",
       "      <td>0.1140</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO013</th>\n",
       "      <td>-0.4470</td>\n",
       "      <td>-0.4460</td>\n",
       "      <td>1.0600</td>\n",
       "      <td>0.7490</td>\n",
       "      <td>1.0100</td>\n",
       "      <td>0.9190</td>\n",
       "      <td>0.7990</td>\n",
       "      <td>-0.4050</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0.4820</td>\n",
       "      <td>...</td>\n",
       "      <td>0.468</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.390</td>\n",
       "      <td>0.0852</td>\n",
       "      <td>0.6560</td>\n",
       "      <td>0.6460</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21CO007</th>\n",
       "      <td>-0.2500</td>\n",
       "      <td>-0.4660</td>\n",
       "      <td>0.0842</td>\n",
       "      <td>0.3090</td>\n",
       "      <td>0.7900</td>\n",
       "      <td>0.4900</td>\n",
       "      <td>0.6050</td>\n",
       "      <td>-0.5170</td>\n",
       "      <td>0.565</td>\n",
       "      <td>0.2520</td>\n",
       "      <td>...</td>\n",
       "      <td>0.642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0887</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1130</td>\n",
       "      <td>0.3840</td>\n",
       "      <td>0.3030</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO004</th>\n",
       "      <td>-0.2610</td>\n",
       "      <td>-0.5280</td>\n",
       "      <td>0.1390</td>\n",
       "      <td>-0.1260</td>\n",
       "      <td>0.4260</td>\n",
       "      <td>-0.1300</td>\n",
       "      <td>-0.1700</td>\n",
       "      <td>-0.5210</td>\n",
       "      <td>-0.456</td>\n",
       "      <td>0.1550</td>\n",
       "      <td>...</td>\n",
       "      <td>0.438</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.850</td>\n",
       "      <td>0.0758</td>\n",
       "      <td>0.0689</td>\n",
       "      <td>0.0213</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO006</th>\n",
       "      <td>-0.0767</td>\n",
       "      <td>0.1290</td>\n",
       "      <td>0.1260</td>\n",
       "      <td>-0.1990</td>\n",
       "      <td>0.1890</td>\n",
       "      <td>-0.0601</td>\n",
       "      <td>-0.1810</td>\n",
       "      <td>0.0689</td>\n",
       "      <td>-0.172</td>\n",
       "      <td>0.0756</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7110</td>\n",
       "      <td>-0.2720</td>\n",
       "      <td>-0.1750</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24CO005</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27CO004</th>\n",
       "      <td>-0.2750</td>\n",
       "      <td>0.4810</td>\n",
       "      <td>0.3500</td>\n",
       "      <td>0.0931</td>\n",
       "      <td>-0.5320</td>\n",
       "      <td>0.0663</td>\n",
       "      <td>0.0366</td>\n",
       "      <td>-0.3360</td>\n",
       "      <td>0.102</td>\n",
       "      <td>0.1940</td>\n",
       "      <td>...</td>\n",
       "      <td>0.044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.848</td>\n",
       "      <td>0.5650</td>\n",
       "      <td>-0.0547</td>\n",
       "      <td>0.0879</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>101 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name         DCTN1  ACTR1A  MAD2L1    MCM4    PCNA    MCM5    MCM2    PCM1  \\\n",
       "Patient_ID                                                                   \n",
       "01CO001        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   \n",
       "01CO005    -0.2120  0.0468 -0.0627  0.1760  0.0313  0.0517  0.0363 -0.2020   \n",
       "01CO006    -0.1250  0.0500 -0.1620  0.2420 -0.0178  0.1790  0.2330  0.0144   \n",
       "01CO008    -0.2510 -0.3780 -0.0747 -0.4790 -0.3960 -0.3320 -0.4490  0.1920   \n",
       "01CO013    -0.4470 -0.4460  1.0600  0.7490  1.0100  0.9190  0.7990 -0.4050   \n",
       "...            ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "21CO007    -0.2500 -0.4660  0.0842  0.3090  0.7900  0.4900  0.6050 -0.5170   \n",
       "22CO004    -0.2610 -0.5280  0.1390 -0.1260  0.4260 -0.1300 -0.1700 -0.5210   \n",
       "22CO006    -0.0767  0.1290  0.1260 -0.1990  0.1890 -0.0601 -0.1810  0.0689   \n",
       "24CO005        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   \n",
       "27CO004    -0.2750  0.4810  0.3500  0.0931 -0.5320  0.0663  0.0366 -0.3360   \n",
       "\n",
       "Name         MCM7     TPR  ...   TPX2  GINS2  BUB1B   TOP2A  TOPBP1   TP53  \\\n",
       "Patient_ID                 ...                                               \n",
       "01CO001       NaN     NaN  ...    NaN    NaN    NaN     NaN     NaN    NaN   \n",
       "01CO005     0.258  0.1740  ... -0.277    NaN    NaN -0.0488     NaN -0.489   \n",
       "01CO006     0.215  0.0299  ...    NaN    NaN    NaN -0.6340     NaN -0.796   \n",
       "01CO008    -0.368  0.3660  ...    NaN    NaN    NaN -0.6550     NaN    NaN   \n",
       "01CO013     0.927  0.4820  ...  0.468    NaN    NaN  0.5180     NaN -1.390   \n",
       "...           ...     ...  ...    ...    ...    ...     ...     ...    ...   \n",
       "21CO007     0.565  0.2520  ...  0.642    NaN    NaN  0.0887     NaN    NaN   \n",
       "22CO004    -0.456  0.1550  ...  0.438    NaN    NaN  0.1710     NaN  0.850   \n",
       "22CO006    -0.172  0.0756  ...    NaN    NaN    NaN  0.1450     NaN    NaN   \n",
       "24CO005       NaN     NaN  ...    NaN    NaN    NaN     NaN     NaN    NaN   \n",
       "27CO004     0.102  0.1940  ...  0.044    NaN    NaN  0.4400     NaN -0.848   \n",
       "\n",
       "Name        NUP210    MSH2    MSH6        Mutation  \n",
       "Patient_ID                                          \n",
       "01CO001        NaN     NaN     NaN  Wildtype_Tumor  \n",
       "01CO005     0.5620  0.3090  0.4610        Deletion  \n",
       "01CO006    -0.3520  0.2270  0.0907  Wildtype_Tumor  \n",
       "01CO008     0.2520  0.1140  0.0143  Wildtype_Tumor  \n",
       "01CO013     0.0852  0.6560  0.6460  Wildtype_Tumor  \n",
       "...            ...     ...     ...             ...  \n",
       "21CO007    -0.1130  0.3840  0.3030  Wildtype_Tumor  \n",
       "22CO004     0.0758  0.0689  0.0213  Wildtype_Tumor  \n",
       "22CO006     0.7110 -0.2720 -0.1750  Wildtype_Tumor  \n",
       "24CO005        NaN     NaN     NaN  Wildtype_Tumor  \n",
       "27CO004     0.5650 -0.0547  0.0879        Deletion  \n",
       "\n",
       "[101 rows x 38 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_wt = all_prot_format_df(col, prot_list)\n",
    "mut_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>DCTN1</th>\n",
       "      <th>ACTR1A</th>\n",
       "      <th>MAD2L1</th>\n",
       "      <th>MCM4</th>\n",
       "      <th>PCNA</th>\n",
       "      <th>MCM5</th>\n",
       "      <th>MCM2</th>\n",
       "      <th>PCM1</th>\n",
       "      <th>MCM7</th>\n",
       "      <th>TPR</th>\n",
       "      <th>...</th>\n",
       "      <th>TPX2</th>\n",
       "      <th>GINS2</th>\n",
       "      <th>BUB1B</th>\n",
       "      <th>TOP2A</th>\n",
       "      <th>TOPBP1</th>\n",
       "      <th>TP53</th>\n",
       "      <th>NUP210</th>\n",
       "      <th>MSH2</th>\n",
       "      <th>MSH6</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>-0.2660</td>\n",
       "      <td>-0.01850</td>\n",
       "      <td>0.639</td>\n",
       "      <td>0.759</td>\n",
       "      <td>0.800</td>\n",
       "      <td>0.6890</td>\n",
       "      <td>0.6150</td>\n",
       "      <td>0.267</td>\n",
       "      <td>0.8020</td>\n",
       "      <td>-0.0715</td>\n",
       "      <td>...</td>\n",
       "      <td>0.925</td>\n",
       "      <td>1.68000</td>\n",
       "      <td>1.420000</td>\n",
       "      <td>1.380</td>\n",
       "      <td>0.3830</td>\n",
       "      <td>0.2950</td>\n",
       "      <td>0.6940</td>\n",
       "      <td>-0.7670</td>\n",
       "      <td>-1.950</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>0.0384</td>\n",
       "      <td>0.21200</td>\n",
       "      <td>0.309</td>\n",
       "      <td>-0.440</td>\n",
       "      <td>-0.285</td>\n",
       "      <td>-0.2580</td>\n",
       "      <td>-0.0715</td>\n",
       "      <td>0.727</td>\n",
       "      <td>-0.2180</td>\n",
       "      <td>-0.2350</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.786</td>\n",
       "      <td>-0.06140</td>\n",
       "      <td>0.021800</td>\n",
       "      <td>0.168</td>\n",
       "      <td>-0.1510</td>\n",
       "      <td>-0.8710</td>\n",
       "      <td>-0.0541</td>\n",
       "      <td>-0.1170</td>\n",
       "      <td>-0.144</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>-0.4350</td>\n",
       "      <td>-0.47900</td>\n",
       "      <td>0.294</td>\n",
       "      <td>1.660</td>\n",
       "      <td>1.220</td>\n",
       "      <td>1.5900</td>\n",
       "      <td>1.4100</td>\n",
       "      <td>-0.308</td>\n",
       "      <td>1.7000</td>\n",
       "      <td>0.1700</td>\n",
       "      <td>...</td>\n",
       "      <td>2.090</td>\n",
       "      <td>0.30900</td>\n",
       "      <td>0.936000</td>\n",
       "      <td>3.280</td>\n",
       "      <td>0.8440</td>\n",
       "      <td>3.0100</td>\n",
       "      <td>-0.5780</td>\n",
       "      <td>0.2110</td>\n",
       "      <td>0.407</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00137</th>\n",
       "      <td>-0.0387</td>\n",
       "      <td>0.00114</td>\n",
       "      <td>0.837</td>\n",
       "      <td>0.913</td>\n",
       "      <td>0.469</td>\n",
       "      <td>1.0600</td>\n",
       "      <td>0.8550</td>\n",
       "      <td>0.248</td>\n",
       "      <td>1.2100</td>\n",
       "      <td>0.0681</td>\n",
       "      <td>...</td>\n",
       "      <td>0.422</td>\n",
       "      <td>1.16000</td>\n",
       "      <td>0.940000</td>\n",
       "      <td>0.629</td>\n",
       "      <td>0.4180</td>\n",
       "      <td>0.4410</td>\n",
       "      <td>1.1600</td>\n",
       "      <td>0.6680</td>\n",
       "      <td>0.450</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00139</th>\n",
       "      <td>-0.2770</td>\n",
       "      <td>-0.50400</td>\n",
       "      <td>0.520</td>\n",
       "      <td>1.560</td>\n",
       "      <td>1.570</td>\n",
       "      <td>1.6800</td>\n",
       "      <td>1.5100</td>\n",
       "      <td>0.386</td>\n",
       "      <td>1.3500</td>\n",
       "      <td>0.2580</td>\n",
       "      <td>...</td>\n",
       "      <td>2.130</td>\n",
       "      <td>-0.01940</td>\n",
       "      <td>1.330000</td>\n",
       "      <td>3.120</td>\n",
       "      <td>0.9720</td>\n",
       "      <td>-1.2200</td>\n",
       "      <td>1.3600</td>\n",
       "      <td>0.6790</td>\n",
       "      <td>0.828</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>-0.0934</td>\n",
       "      <td>0.10300</td>\n",
       "      <td>0.589</td>\n",
       "      <td>0.728</td>\n",
       "      <td>0.663</td>\n",
       "      <td>0.6760</td>\n",
       "      <td>0.4810</td>\n",
       "      <td>0.640</td>\n",
       "      <td>0.4520</td>\n",
       "      <td>-0.1380</td>\n",
       "      <td>...</td>\n",
       "      <td>0.265</td>\n",
       "      <td>0.79300</td>\n",
       "      <td>0.195000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.0932</td>\n",
       "      <td>-0.2570</td>\n",
       "      <td>0.5130</td>\n",
       "      <td>-0.1980</td>\n",
       "      <td>0.510</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>-0.3890</td>\n",
       "      <td>-0.97900</td>\n",
       "      <td>0.191</td>\n",
       "      <td>0.300</td>\n",
       "      <td>0.822</td>\n",
       "      <td>0.0119</td>\n",
       "      <td>0.2310</td>\n",
       "      <td>-0.458</td>\n",
       "      <td>-0.0943</td>\n",
       "      <td>0.3050</td>\n",
       "      <td>...</td>\n",
       "      <td>0.620</td>\n",
       "      <td>0.67500</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.384</td>\n",
       "      <td>0.4400</td>\n",
       "      <td>1.2000</td>\n",
       "      <td>0.9370</td>\n",
       "      <td>0.1570</td>\n",
       "      <td>0.563</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>-0.3070</td>\n",
       "      <td>-0.77300</td>\n",
       "      <td>0.648</td>\n",
       "      <td>0.836</td>\n",
       "      <td>-0.069</td>\n",
       "      <td>0.4240</td>\n",
       "      <td>0.7840</td>\n",
       "      <td>-0.254</td>\n",
       "      <td>0.7000</td>\n",
       "      <td>-0.0173</td>\n",
       "      <td>...</td>\n",
       "      <td>1.230</td>\n",
       "      <td>-0.00635</td>\n",
       "      <td>1.150000</td>\n",
       "      <td>0.964</td>\n",
       "      <td>0.5770</td>\n",
       "      <td>-0.0333</td>\n",
       "      <td>0.7190</td>\n",
       "      <td>-0.0961</td>\n",
       "      <td>0.176</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>-0.4780</td>\n",
       "      <td>0.14000</td>\n",
       "      <td>-0.755</td>\n",
       "      <td>-0.119</td>\n",
       "      <td>0.368</td>\n",
       "      <td>-0.1780</td>\n",
       "      <td>0.1380</td>\n",
       "      <td>-0.713</td>\n",
       "      <td>0.4100</td>\n",
       "      <td>0.0605</td>\n",
       "      <td>...</td>\n",
       "      <td>1.040</td>\n",
       "      <td>0.19700</td>\n",
       "      <td>0.627000</td>\n",
       "      <td>0.668</td>\n",
       "      <td>0.7550</td>\n",
       "      <td>0.2920</td>\n",
       "      <td>1.3100</td>\n",
       "      <td>-0.2970</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>0.2370</td>\n",
       "      <td>-0.10800</td>\n",
       "      <td>0.467</td>\n",
       "      <td>0.603</td>\n",
       "      <td>0.158</td>\n",
       "      <td>0.6970</td>\n",
       "      <td>0.7970</td>\n",
       "      <td>-0.638</td>\n",
       "      <td>0.8660</td>\n",
       "      <td>-0.0725</td>\n",
       "      <td>...</td>\n",
       "      <td>0.646</td>\n",
       "      <td>0.08970</td>\n",
       "      <td>0.862000</td>\n",
       "      <td>1.290</td>\n",
       "      <td>0.4660</td>\n",
       "      <td>1.5100</td>\n",
       "      <td>-0.3090</td>\n",
       "      <td>0.1580</td>\n",
       "      <td>0.245</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name         DCTN1   ACTR1A  MAD2L1   MCM4   PCNA    MCM5    MCM2   PCM1  \\\n",
       "Patient_ID                                                                 \n",
       "C3L-00006  -0.2660 -0.01850   0.639  0.759  0.800  0.6890  0.6150  0.267   \n",
       "C3L-00032   0.0384  0.21200   0.309 -0.440 -0.285 -0.2580 -0.0715  0.727   \n",
       "C3L-00098  -0.4350 -0.47900   0.294  1.660  1.220  1.5900  1.4100 -0.308   \n",
       "C3L-00137  -0.0387  0.00114   0.837  0.913  0.469  1.0600  0.8550  0.248   \n",
       "C3L-00139  -0.2770 -0.50400   0.520  1.560  1.570  1.6800  1.5100  0.386   \n",
       "...            ...      ...     ...    ...    ...     ...     ...    ...   \n",
       "C3N-01520  -0.0934  0.10300   0.589  0.728  0.663  0.6760  0.4810  0.640   \n",
       "C3N-01521  -0.3890 -0.97900   0.191  0.300  0.822  0.0119  0.2310 -0.458   \n",
       "C3N-01537  -0.3070 -0.77300   0.648  0.836 -0.069  0.4240  0.7840 -0.254   \n",
       "C3N-01802  -0.4780  0.14000  -0.755 -0.119  0.368 -0.1780  0.1380 -0.713   \n",
       "C3N-01825   0.2370 -0.10800   0.467  0.603  0.158  0.6970  0.7970 -0.638   \n",
       "\n",
       "Name          MCM7     TPR  ...   TPX2    GINS2     BUB1B  TOP2A  TOPBP1  \\\n",
       "Patient_ID                  ...                                            \n",
       "C3L-00006   0.8020 -0.0715  ...  0.925  1.68000  1.420000  1.380  0.3830   \n",
       "C3L-00032  -0.2180 -0.2350  ... -0.786 -0.06140  0.021800  0.168 -0.1510   \n",
       "C3L-00098   1.7000  0.1700  ...  2.090  0.30900  0.936000  3.280  0.8440   \n",
       "C3L-00137   1.2100  0.0681  ...  0.422  1.16000  0.940000  0.629  0.4180   \n",
       "C3L-00139   1.3500  0.2580  ...  2.130 -0.01940  1.330000  3.120  0.9720   \n",
       "...            ...     ...  ...    ...      ...       ...    ...     ...   \n",
       "C3N-01520   0.4520 -0.1380  ...  0.265  0.79300  0.195000  1.000  0.0932   \n",
       "C3N-01521  -0.0943  0.3050  ...  0.620  0.67500  0.000046  0.384  0.4400   \n",
       "C3N-01537   0.7000 -0.0173  ...  1.230 -0.00635  1.150000  0.964  0.5770   \n",
       "C3N-01802   0.4100  0.0605  ...  1.040  0.19700  0.627000  0.668  0.7550   \n",
       "C3N-01825   0.8660 -0.0725  ...  0.646  0.08970  0.862000  1.290  0.4660   \n",
       "\n",
       "Name          TP53  NUP210    MSH2   MSH6        Mutation  \n",
       "Patient_ID                                                 \n",
       "C3L-00006   0.2950  0.6940 -0.7670 -1.950      Truncation  \n",
       "C3L-00032  -0.8710 -0.0541 -0.1170 -0.144      Truncation  \n",
       "C3L-00098   3.0100 -0.5780  0.2110  0.407  Wildtype_Tumor  \n",
       "C3L-00137   0.4410  1.1600  0.6680  0.450      Truncation  \n",
       "C3L-00139  -1.2200  1.3600  0.6790  0.828  Wildtype_Tumor  \n",
       "...            ...     ...     ...    ...             ...  \n",
       "C3N-01520  -0.2570  0.5130 -0.1980  0.510      Truncation  \n",
       "C3N-01521   1.2000  0.9370  0.1570  0.563  Wildtype_Tumor  \n",
       "C3N-01537  -0.0333  0.7190 -0.0961  0.176  Wildtype_Tumor  \n",
       "C3N-01802   0.2920  1.3100 -0.2970 -0.332  Wildtype_Tumor  \n",
       "C3N-01825   1.5100 -0.3090  0.1580  0.245  Wildtype_Tumor  \n",
       "\n",
       "[68 rows x 38 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([], dtype='object', name='Name')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "null_columns = mut_wt.columns[mut_wt.isnull().all()]\n",
    "mut_wt[mut_wt.isnull().any(axis=1)][null_columns].head()\n",
    "null_columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# checks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 101 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    }
   ],
   "source": [
    "# Get df\n",
    "cancer_obj = col\n",
    "prot = cancer_obj.get_proteomics()\n",
    "if cancer_obj == l:\n",
    "    prot = u.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "prot_list = list(prot.columns)\n",
    "#prot_list = list(u.get_interacting_proteins_biogrid('KRAS', num_results=100))\n",
    "#prot_list = ['KRAS']\n",
    "\n",
    "mut_wt = all_prot_format_df(cancer_obj, prot_list)\n",
    "\n",
    "mut_wt_cols = list(mut_wt.columns[:-1])\n",
    "#print('Last column name:', mut_wt_cols[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "find_few_data_genes() got an unexpected keyword argument 'gene'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-14-746d22040b6d>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;31m# Test\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mmincount\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[0mnd_genes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mfind_few_data_genes\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmut_wt\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmincount\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mprot_list\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mgene\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m'PTEN'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnd_genes\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m: find_few_data_genes() got an unexpected keyword argument 'gene'"
     ]
    }
   ],
   "source": [
    "# Test\n",
    "mincount = 0\n",
    "nd_genes = find_few_data_genes(mut_wt, mincount, prot_list, gene = 'PTEN')\n",
    "print(len(nd_genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nd_genes[5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'CCDC140' # gene in not_enough_data list (< mincount)\n",
    "\n",
    "prot = cancer_obj.get_proteomics()\n",
    "if cancer_obj == l:\n",
    "    prot = u.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "test_prot_list = list(prot.columns)\n",
    "test_mut_wt = all_prot_format_df(cancer_obj, test_prot_list)\n",
    "\n",
    "gene_df = test_mut_wt[[gene,'Mutation']]\n",
    "mut_df = gene_df.loc[gene_df['Mutation'] == 'Missense']\n",
    "wt_df = gene_df.loc[gene_df['Mutation'] == 'Wildtype_Tumor']\n",
    "\n",
    "print('Num of missense with proteomics data:', len(mut_df[gene].dropna()), '(<=', mincount, '?)')\n",
    "print('Num of wt with proteomics data:', len(wt_df[gene].dropna()), '(<=', mincount, '?)')\n",
    "mut_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
