{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Phosphoproteomics GSEA and Heatmap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook looks at the significant phosphosites in at least one cancer. Pancancer heatmaps are created with circle size showing significance and color showing differences in median."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sig in mult cancers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sig in mult cancers: 12\n"
     ]
    }
   ],
   "source": [
    "mult_df = pd.read_csv('Make_Tables\\csv\\mult_sig_heatmap_phospho.csv')\n",
    "print('sig in mult cancers:', len(mult_df.Phospho.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phospho</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DNMBP_S1436</td>\n",
       "      <td>0.000425</td>\n",
       "      <td>1.63160</td>\n",
       "      <td>Luad</td>\n",
       "      <td>DNMBP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOS1_S1161</td>\n",
       "      <td>0.000547</td>\n",
       "      <td>1.30250</td>\n",
       "      <td>Luad</td>\n",
       "      <td>SOS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PARP4_S1186</td>\n",
       "      <td>0.009200</td>\n",
       "      <td>1.54200</td>\n",
       "      <td>Luad</td>\n",
       "      <td>PARP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIAA1217_S1568</td>\n",
       "      <td>0.022649</td>\n",
       "      <td>1.69130</td>\n",
       "      <td>Luad</td>\n",
       "      <td>KIAA1217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PDAP1_S170</td>\n",
       "      <td>0.025903</td>\n",
       "      <td>-1.01535</td>\n",
       "      <td>Luad</td>\n",
       "      <td>PDAP1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Phospho   P_Value  Medians Cancer      Gene\n",
       "0     DNMBP_S1436  0.000425  1.63160   Luad     DNMBP\n",
       "1      SOS1_S1161  0.000547  1.30250   Luad      SOS1\n",
       "2     PARP4_S1186  0.009200  1.54200   Luad     PARP4\n",
       "3  KIAA1217_S1568  0.022649  1.69130   Luad  KIAA1217\n",
       "4      PDAP1_S170  0.025903 -1.01535   Luad     PDAP1"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get list of just proteins\n",
    "mult_df['Gene'] = mult_df['Phospho']\n",
    "mult_df['Gene'] = mult_df['Gene'].replace(r'(_[\\w][\\d]*[\\w]?[\\d]*[\\w]?[\\d]*)*$', '', regex=True)\n",
    "prot_list = list(mult_df.Gene.unique()) \n",
    "mult_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2020-08-10 13:49:21,573 Warning: No enrich terms using library Reactome_2016 when cutoff = 0.05\n"
     ]
    }
   ],
   "source": [
    "prot_enr = gp.enrichr(gene_list = prot_list, description='Tumor_partition', gene_sets='Reactome_2016', \n",
    "                       outdir='/Enrichr')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Term</th>\n",
       "      <th>Overlap</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Adjusted P-value</th>\n",
       "      <th>Old P-value</th>\n",
       "      <th>Old Adjusted P-value</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>Combined Score</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Gene_set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Interleukin-1 processing Homo sapiens R-HSA-44...</td>\n",
       "      <td>1/7</td>\n",
       "      <td>0.004193</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>238.095238</td>\n",
       "      <td>1303.413145</td>\n",
       "      <td>NFKB2</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IkBA variant leads to EDA-ID Homo sapiens R-HS...</td>\n",
       "      <td>1/7</td>\n",
       "      <td>0.004193</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>238.095238</td>\n",
       "      <td>1303.413145</td>\n",
       "      <td>NFKB2</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IKBKG deficiency causes anhidrotic ectodermal ...</td>\n",
       "      <td>1/8</td>\n",
       "      <td>0.004791</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>208.333333</td>\n",
       "      <td>1112.724603</td>\n",
       "      <td>NFKB2</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Type I hemidesmosome assembly Homo sapiens R-H...</td>\n",
       "      <td>1/9</td>\n",
       "      <td>0.005388</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>185.185185</td>\n",
       "      <td>967.327670</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EGFR Transactivation by Gastrin Homo sapiens R...</td>\n",
       "      <td>1/9</td>\n",
       "      <td>0.005388</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>185.185185</td>\n",
       "      <td>967.327670</td>\n",
       "      <td>SOS1</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Term Overlap   P-value  \\\n",
       "0  Interleukin-1 processing Homo sapiens R-HSA-44...     1/7  0.004193   \n",
       "1  IkBA variant leads to EDA-ID Homo sapiens R-HS...     1/7  0.004193   \n",
       "2  IKBKG deficiency causes anhidrotic ectodermal ...     1/8  0.004791   \n",
       "3  Type I hemidesmosome assembly Homo sapiens R-H...     1/9  0.005388   \n",
       "4  EGFR Transactivation by Gastrin Homo sapiens R...     1/9  0.005388   \n",
       "\n",
       "   Adjusted P-value  Old P-value  Old Adjusted P-value  Odds Ratio  \\\n",
       "0               1.0            0                     0  238.095238   \n",
       "1               1.0            0                     0  238.095238   \n",
       "2               1.0            0                     0  208.333333   \n",
       "3               1.0            0                     0  185.185185   \n",
       "4               1.0            0                     0  185.185185   \n",
       "\n",
       "   Combined Score  Genes       Gene_set  \n",
       "0     1303.413145  NFKB2  Reactome_2016  \n",
       "1     1303.413145  NFKB2  Reactome_2016  \n",
       "2     1112.724603  NFKB2  Reactome_2016  \n",
       "3      967.327670   PLEC  Reactome_2016  \n",
       "4      967.327670   SOS1  Reactome_2016  "
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_enr.res2d.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sig in one cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sig in one cancer: 889\n"
     ]
    }
   ],
   "source": [
    "sig_df = pd.read_csv('Make_Tables\\csv\\sig_pval_heatmap_phospho.csv')\n",
    "print('sig in one cancer:', len(sig_df.Phospho.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phospho</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EEF1B2_S8</td>\n",
       "      <td>0.000107</td>\n",
       "      <td>2.1124</td>\n",
       "      <td>Luad</td>\n",
       "      <td>EEF1B2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DNMBP_S1436</td>\n",
       "      <td>0.000425</td>\n",
       "      <td>1.6316</td>\n",
       "      <td>Luad</td>\n",
       "      <td>DNMBP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CTBP2_S905</td>\n",
       "      <td>0.000450</td>\n",
       "      <td>1.2298</td>\n",
       "      <td>Luad</td>\n",
       "      <td>CTBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SOS1_S1161</td>\n",
       "      <td>0.000547</td>\n",
       "      <td>1.3025</td>\n",
       "      <td>Luad</td>\n",
       "      <td>SOS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>XPO5_S416</td>\n",
       "      <td>0.000726</td>\n",
       "      <td>2.6570</td>\n",
       "      <td>Luad</td>\n",
       "      <td>XPO5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Phospho   P_Value  Medians Cancer    Gene\n",
       "0    EEF1B2_S8  0.000107   2.1124   Luad  EEF1B2\n",
       "1  DNMBP_S1436  0.000425   1.6316   Luad   DNMBP\n",
       "2   CTBP2_S905  0.000450   1.2298   Luad   CTBP2\n",
       "3   SOS1_S1161  0.000547   1.3025   Luad    SOS1\n",
       "4    XPO5_S416  0.000726   2.6570   Luad    XPO5"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get list of just proteins\n",
    "sig_df['Gene'] = sig_df['Phospho']\n",
    "sig_df['Gene'] = sig_df['Gene'].replace(r'(_[\\w][\\d]*[\\w]?[\\d]*[\\w]?[\\d]*)*$', '', regex=True)\n",
    "prot_list = list(sig_df.Gene.unique()) \n",
    "sig_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_enr = gp.enrichr(gene_list = one, description='Tumor_partition', gene_sets='Reactome_2016', \n",
    "                       outdir='/Enrichr')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Term</th>\n",
       "      <th>Overlap</th>\n",
       "      <th>Adjusted P-value</th>\n",
       "      <th>Genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Signaling by Rho GTPases Homo sapiens R-HSA-19...</td>\n",
       "      <td>26/367</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>ARHGAP5;FGD1;ARHGAP35;AKAP13;XPO1;ABL1;SRGAP2;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rho GTPase cycle Homo sapiens R-HSA-194840</td>\n",
       "      <td>14/122</td>\n",
       "      <td>0.000113</td>\n",
       "      <td>ARHGEF12;ARHGEF18;ARHGAP5;ARHGAP35;FGD1;VAV2;B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Interactions of Rev with host cellular protein...</td>\n",
       "      <td>8/35</td>\n",
       "      <td>0.000172</td>\n",
       "      <td>RANBP2;NUP214;NPM1;XPO1;TPR;NUP35;NUP153;NUP98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SUMO E3 ligases SUMOylate target proteins Homo...</td>\n",
       "      <td>12/96</td>\n",
       "      <td>0.000173</td>\n",
       "      <td>NUP214;RANBP2;TOP2B;MDC1;TPR;NUP35;NUP153;NUP9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SUMOylation Homo sapiens R-HSA-2990846</td>\n",
       "      <td>12/101</td>\n",
       "      <td>0.000242</td>\n",
       "      <td>NUP214;RANBP2;MDC1;TOP2B;TPR;NUP35;NUP153;NUP9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Cell Cycle Homo sapiens R-HSA-1640170</td>\n",
       "      <td>30/566</td>\n",
       "      <td>0.000316</td>\n",
       "      <td>MDC1;GOLGA2;PCM1;XPO1;TPR;RBBP8;E2F4;TP53BP1;B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NEP/NS2 Interacts with the Cellular Export Mac...</td>\n",
       "      <td>7/30</td>\n",
       "      <td>0.000351</td>\n",
       "      <td>RANBP2;NUP214;XPO1;TPR;NUP35;NUP153;NUP98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Export of Viral Ribonucleoproteins from Nucleu...</td>\n",
       "      <td>7/31</td>\n",
       "      <td>0.000390</td>\n",
       "      <td>RANBP2;NUP214;XPO1;TPR;NUP35;NUP153;NUP98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Cell Cycle, Mitotic Homo sapiens R-HSA-69278</td>\n",
       "      <td>26/462</td>\n",
       "      <td>0.000371</td>\n",
       "      <td>GOLGA2;PCM1;XPO1;TPR;E2F4;BTRC;SKP2;CLASP1;CDK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Signaling by Robo receptor Homo sapiens R-HSA-...</td>\n",
       "      <td>7/32</td>\n",
       "      <td>0.000393</td>\n",
       "      <td>ENAH;ABL1;NCK2;SRGAP2;SOS1;CLASP1;CLASP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Nuclear import of Rev protein Homo sapiens R-H...</td>\n",
       "      <td>7/32</td>\n",
       "      <td>0.000357</td>\n",
       "      <td>RANBP2;NUP214;NPM1;TPR;NUP35;NUP153;NUP98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NRAGE signals death through JNK Homo sapiens R...</td>\n",
       "      <td>8/45</td>\n",
       "      <td>0.000331</td>\n",
       "      <td>AKAP13;ARHGEF12;BAD;ARHGEF18;ARHGEF2;SOS1;FGD1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Transport of the SLBP independent Mature mRNA ...</td>\n",
       "      <td>7/33</td>\n",
       "      <td>0.000377</td>\n",
       "      <td>RANBP2;NUP214;NCBP1;TPR;NUP35;NUP153;NUP98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Rev-mediated nuclear export of HIV RNA Homo sa...</td>\n",
       "      <td>7/33</td>\n",
       "      <td>0.000350</td>\n",
       "      <td>RANBP2;NUP214;XPO1;TPR;NUP35;NUP153;NUP98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Transport of the SLBP Dependant Mature mRNA Ho...</td>\n",
       "      <td>7/34</td>\n",
       "      <td>0.000405</td>\n",
       "      <td>RANBP2;NUP214;NCBP1;TPR;NUP35;NUP153;NUP98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Term Overlap  \\\n",
       "0   Signaling by Rho GTPases Homo sapiens R-HSA-19...  26/367   \n",
       "1          Rho GTPase cycle Homo sapiens R-HSA-194840  14/122   \n",
       "2   Interactions of Rev with host cellular protein...    8/35   \n",
       "3   SUMO E3 ligases SUMOylate target proteins Homo...   12/96   \n",
       "4              SUMOylation Homo sapiens R-HSA-2990846  12/101   \n",
       "5               Cell Cycle Homo sapiens R-HSA-1640170  30/566   \n",
       "6   NEP/NS2 Interacts with the Cellular Export Mac...    7/30   \n",
       "7   Export of Viral Ribonucleoproteins from Nucleu...    7/31   \n",
       "8        Cell Cycle, Mitotic Homo sapiens R-HSA-69278  26/462   \n",
       "9   Signaling by Robo receptor Homo sapiens R-HSA-...    7/32   \n",
       "10  Nuclear import of Rev protein Homo sapiens R-H...    7/32   \n",
       "11  NRAGE signals death through JNK Homo sapiens R...    8/45   \n",
       "12  Transport of the SLBP independent Mature mRNA ...    7/33   \n",
       "13  Rev-mediated nuclear export of HIV RNA Homo sa...    7/33   \n",
       "14  Transport of the SLBP Dependant Mature mRNA Ho...    7/34   \n",
       "\n",
       "    Adjusted P-value                                              Genes  \n",
       "0           0.000039  ARHGAP5;FGD1;ARHGAP35;AKAP13;XPO1;ABL1;SRGAP2;...  \n",
       "1           0.000113  ARHGEF12;ARHGEF18;ARHGAP5;ARHGAP35;FGD1;VAV2;B...  \n",
       "2           0.000172     RANBP2;NUP214;NPM1;XPO1;TPR;NUP35;NUP153;NUP98  \n",
       "3           0.000173  NUP214;RANBP2;TOP2B;MDC1;TPR;NUP35;NUP153;NUP9...  \n",
       "4           0.000242  NUP214;RANBP2;MDC1;TOP2B;TPR;NUP35;NUP153;NUP9...  \n",
       "5           0.000316  MDC1;GOLGA2;PCM1;XPO1;TPR;RBBP8;E2F4;TP53BP1;B...  \n",
       "6           0.000351          RANBP2;NUP214;XPO1;TPR;NUP35;NUP153;NUP98  \n",
       "7           0.000390          RANBP2;NUP214;XPO1;TPR;NUP35;NUP153;NUP98  \n",
       "8           0.000371  GOLGA2;PCM1;XPO1;TPR;E2F4;BTRC;SKP2;CLASP1;CDK...  \n",
       "9           0.000393           ENAH;ABL1;NCK2;SRGAP2;SOS1;CLASP1;CLASP2  \n",
       "10          0.000357          RANBP2;NUP214;NPM1;TPR;NUP35;NUP153;NUP98  \n",
       "11          0.000331  AKAP13;ARHGEF12;BAD;ARHGEF18;ARHGEF2;SOS1;FGD1...  \n",
       "12          0.000377         RANBP2;NUP214;NCBP1;TPR;NUP35;NUP153;NUP98  \n",
       "13          0.000350          RANBP2;NUP214;XPO1;TPR;NUP35;NUP153;NUP98  \n",
       "14          0.000405         RANBP2;NUP214;NCBP1;TPR;NUP35;NUP153;NUP98  "
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_enr.res2d[['Term', 'Overlap', 'Adjusted P-value', 'Genes']].head(15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      Signaling by Rho GTPases Homo sapiens R-HSA-19...\n",
       "1             Rho GTPase cycle Homo sapiens R-HSA-194840\n",
       "2      Interactions of Rev with host cellular protein...\n",
       "3      SUMO E3 ligases SUMOylate target proteins Homo...\n",
       "4                 SUMOylation Homo sapiens R-HSA-2990846\n",
       "                             ...                        \n",
       "741    Extracellular matrix organization Homo sapiens...\n",
       "742    Metabolism of lipids and lipoproteins Homo sap...\n",
       "743    Metabolism of amino acids and derivatives Homo...\n",
       "744    GPCR downstream signaling Homo sapiens R-HSA-3...\n",
       "745                Metabolism Homo sapiens R-HSA-1430728\n",
       "Name: Term, Length: 746, dtype: object"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_enr.res2d['Term']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make Heatmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUMO E3 ligases SUMOylate target proteins Homo sapiens R-HSA-3108232\n",
      "Num genes: 17\n"
     ]
    }
   ],
   "source": [
    "i = 4\n",
    "genes = prot_enr.res2d.Genes[i]\n",
    "g_list = genes.split(';')\n",
    "print(prot_enr.res2d.Term[i])\n",
    "print('Num genes:', len(g_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "# only one cancer\n",
    "\n",
    "c = 'Endo' #'Luad'\n",
    "sig_df = sig_df.loc[sig_df['P_Value'] < 0.05]\n",
    "one_df = sig_df.loc[sig_df['Cancer'] == c]\n",
    "len(one_df)\n",
    "one = list(one_df.Gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [],
   "source": [
    "get = sig_df['Gene'].isin(g_list)\n",
    "plot_df = sig_df[get]\n",
    "plot_df = plot_df.loc[sig_df['P_Value'] < 0.05]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"5158\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"5158\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"5158\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '5158' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"5158\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"5158\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"5158\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '5158' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"5158\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"d2d9e664-eed6-47eb-8df9-dffc444e7b13\" data-root-id=\"5159\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"c40239a3-6f4f-425c-9a47-ff521026bdef\":{\"roots\":{\"references\":[{\"attributes\":{\"children\":[{\"id\":\"5085\"},{\"id\":\"5136\"}]},\"id\":\"5159\",\"type\":\"Row\"},{\"attributes\":{},\"id\":\"5093\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"5411\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"5141\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"5149\"}]},\"id\":\"5150\",\"type\":\"Toolbar\"},{\"attributes\":{\"high\":1.9881,\"low\":-1.9881,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"5082\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"axis\":{\"id\":\"5092\"},\"ticker\":null},\"id\":\"5094\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"5400\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"axis_label\":\"FDR_P-Values\",\"formatter\":{\"id\":\"5404\"},\"major_label_orientation\":1.5707963267948966,\"ticker\":{\"id\":\"5144\"}},\"id\":\"5143\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"factors\":[\"5.0e-02\",\"1.0e-03\",\"6.4e-05\"]},\"id\":\"5135\",\"type\":\"FactorRange\"},{\"attributes\":{\"formatter\":{\"id\":\"5400\"},\"major_label_text_font_size\":\"12pt\",\"ticker\":{\"id\":\"5096\"}},\"id\":\"5095\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"5402\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"5144\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"factors\":[\"NUP54_S289\",\"RANBP2_S1869\",\"RANBP2_T2504\",\"NUP50_S208\",\"POM121C_S130S131\",\"RPA1_S38\",\"NUP214_S964\",\"XRCC4_S328\",\"NUP50_S221\",\"TPR_S1662\",\"NUP98_T1053\",\"NUP153_T388\",\"MDC1_T1321\",\"TPR_T653\",\"TPR_T650\",\"PML_S512\",\"PML_S480\",\"TOP2B_S1550\",\"NUP98_T653\",\"NUP35_T48\",\"NUP98_S1047\",\"TOP2B_S1576\",\"NUP153_S547\",\"TPR_S648\",\"NUP153_S553\",\"HNRNPC_S233\",\"RANBP2_S1613\",\"TP53BP1_T1614\",\"NUP214_S610\",\"HDAC7_S546\",\"NUP35_T280\",\"NUP153_T413\"]},\"id\":\"5083\",\"type\":\"FactorRange\"},{\"attributes\":{\"text\":\"\"},\"id\":\"5397\",\"type\":\"Title\"},{\"attributes\":{\"data\":{\"Medians\":{\"__ndarray__\":\"AAAAAAAA+D8AAAAAAAD4PwAAAAAAAPg/\",\"dtype\":\"float64\",\"shape\":[3]},\"P_Value\":{\"__ndarray__\":\"mpmZmZmZqT/8qfHSTWJQP9klw/mRyxA/\",\"dtype\":\"float64\",\"shape\":[3]},\"index\":[0,1,2],\"size\":{\"__ndarray__\":\"reqW3HH5IUC4/yvzJ7k0QL9vUZ939zxA\",\"dtype\":\"float64\",\"shape\":[3]},\"size2\":{\"__ndarray__\":\"keNze0L3B0Cg/4+ZiqEbQCr14BSlTyNA\",\"dtype\":\"float64\",\"shape\":[3]},\"x_axis\":[\"\",\"\",\"\"],\"y_axis\":[\"5.0e-02\",\"1.0e-03\",\"6.4e-05\"]},\"selected\":{\"id\":\"5410\"},\"selection_policy\":{\"id\":\"5411\"}},\"id\":\"5152\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"5096\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"5403\",\"type\":\"BasicTicker\"},{\"attributes\":{\"text\":\"\"},\"id\":\"5395\",\"type\":\"Title\"},{\"attributes\":{\"color_mapper\":{\"id\":\"5082\"},\"formatter\":{\"id\":\"5402\"},\"location\":[0,0],\"ticker\":{\"id\":\"5403\"}},\"id\":\"5107\",\"type\":\"ColorBar\"},{\"attributes\":{\"axis\":{\"id\":\"5143\"},\"ticker\":null},\"id\":\"5145\",\"type\":\"Grid\"},{\"attributes\":{\"factors\":[\"\"]},\"id\":\"5134\",\"type\":\"FactorRange\"},{\"attributes\":{\"high\":1.5,\"low\":1.5,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"5133\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"factors\":[\"Luad\",\"Endo\"]},\"id\":\"5084\",\"type\":\"FactorRange\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"5082\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Phospho\"},\"y\":{\"field\":\"Cancer\"}},\"id\":\"5104\",\"type\":\"Scatter\"},{\"attributes\":{\"axis\":{\"id\":\"5095\"},\"dimension\":1,\"ticker\":null},\"id\":\"5097\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"5404\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"data_source\":{\"id\":\"5101\"},\"glyph\":{\"id\":\"5103\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"5104\"},\"selection_glyph\":null,\"view\":{\"id\":\"5106\"}},\"id\":\"5105\",\"type\":\"GlyphRenderer\"},{\"attributes\":{},\"id\":\"5398\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"formatter\":{\"id\":\"5406\"},\"ticker\":{\"id\":\"5147\"}},\"id\":\"5146\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"below\":[{\"id\":\"5092\"}],\"center\":[{\"id\":\"5094\"},{\"id\":\"5097\"}],\"left\":[{\"id\":\"5095\"}],\"plot_height\":400,\"plot_width\":1500,\"renderers\":[{\"id\":\"5105\"}],\"right\":[{\"id\":\"5107\"}],\"title\":{\"id\":\"5395\"},\"toolbar\":{\"id\":\"5099\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"5083\"},\"x_scale\":{\"id\":\"5088\"},\"y_range\":{\"id\":\"5084\"},\"y_scale\":{\"id\":\"5090\"}},\"id\":\"5085\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"source\":{\"id\":\"5101\"}},\"id\":\"5106\",\"type\":\"CDSView\"},{\"attributes\":{\"source\":{\"id\":\"5152\"}},\"id\":\"5157\",\"type\":\"CDSView\"},{\"attributes\":{},\"id\":\"5406\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{},\"id\":\"5147\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"callback\":null},\"id\":\"5098\",\"type\":\"HoverTool\"},{\"attributes\":{\"data_source\":{\"id\":\"5152\"},\"glyph\":{\"id\":\"5154\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"5155\"},\"selection_glyph\":null,\"view\":{\"id\":\"5157\"}},\"id\":\"5156\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"data\":{\"Cancer\":[\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\",\"Endo\"],\"Gene\":[\"NUP54\",\"RANBP2\",\"RANBP2\",\"NUP50\",\"POM121C\",\"RPA1\",\"NUP214\",\"XRCC4\",\"NUP50\",\"TPR\",\"NUP98\",\"NUP153\",\"MDC1\",\"TPR\",\"TPR\",\"PML\",\"PML\",\"TOP2B\",\"NUP98\",\"NUP35\",\"NUP98\",\"TOP2B\",\"NUP153\",\"TPR\",\"NUP153\",\"HNRNPC\",\"RANBP2\",\"TP53BP1\",\"NUP214\",\"HDAC7\",\"NUP35\",\"NUP153\"],\"Medians\":{\"__ndarray__\":\"Io51cRuN8D/c14FzRpTkPwisHFpku/A/7zhFR3L57z924JwRpb3jv3Gsi9toAOa/AAAAAACA/T/zH9JvXwfnP3PXEvJBz/8/okW28/1U8T8Y7IZti7LWPw4tsp3vp+k/ih9j7lpC5T+U9gZfmEziP+xRuB6F69s/Rbbz/dR45j9WDi2yne/nP5/Nqs/VVtQ/8tJNYhBY1j/qBDQRNjzjP86qz9VW7Nc/qMZLN4lB5D/RkVz+Q/reP+xRuB6F6+k/rPpcbcX+0T+vlGWIY13Ev13cRgN4C9A/6Nms+lxt4T/njCjtDb7fPxIUP8bcteQ/FmpN845T4j+sHFpkO9/jPw==\",\"dtype\":\"float64\",\"shape\":[32]},\"P_Value\":{\"__ndarray__\":\"7wLBwPNsfD9dyGA5jjyDP/iYq72uMJc/+diWm6/smD+yD/tDAHadPybMZS3K86E/JsxlLcrzoT+509KmzSmoP+Ub6PwwSqg/2SXD+ZHLED8IxLNXIuFCP6fzplBduEo/5yBVlDgrZT/vWhTjy4NuP0X6UtNYuG8/0KTVOL0xgD+GkokclUmAP5zbV2/9dYo/HrRYfn9Wkj95ULYOGimUPxB/TSwm5Jo/Ki89hcOQmz+95A9uOf+bP6qFz2hcS54/XsiJwG3ZoD9FuWmTXiChP3auliIQUqE/cAXynh/loj/B2ewSNJekP/zmOuVyfaU/pkbA0pTupz/UYcFpkYioPw==\",\"dtype\":\"float64\",\"shape\":[32]},\"Phospho\":[\"NUP54_S289\",\"RANBP2_S1869\",\"RANBP2_T2504\",\"NUP50_S208\",\"POM121C_S130S131\",\"RPA1_S38\",\"NUP214_S964\",\"XRCC4_S328\",\"NUP50_S221\",\"TPR_S1662\",\"NUP98_T1053\",\"NUP153_T388\",\"MDC1_T1321\",\"TPR_T653\",\"TPR_T650\",\"PML_S512\",\"PML_S480\",\"TOP2B_S1550\",\"NUP98_T653\",\"NUP35_T48\",\"NUP98_S1047\",\"TOP2B_S1576\",\"NUP153_S547\",\"TPR_S648\",\"NUP153_S553\",\"HNRNPC_S233\",\"RANBP2_S1613\",\"TP53BP1_T1614\",\"NUP214_S610\",\"HDAC7_S546\",\"NUP35_T280\",\"NUP153_T413\"],\"index\":[19,44,109,121,152,202,210,284,298,633,639,643,685,702,708,742,743,796,860,888,957,961,962,996,1028,1034,1037,1072,1097,1118,1177,1184],\"size\":{\"__ndarray__\":\"dtuxxqbSLUDSJAFywAEsQCSSRFf5uSZAorLmljBLJkAmVEQvWkolQCvaWfWHGiRAK9pZ9YcaJEDONKYWKVIiQFYFNqIjSiJAv29Rn3f3PEACc1KPnWA2QOw4LJXaVTVA8wCO2QngMUCaE96rMccwQHTnB3xxqTBAwS7Auy0KLUDuRCW3XgEtQNu0XBH9FypAsGgvdJAiKEDIK8q9A5EnQP0oYG+J1iVAGtFOb3+wJUBR8bcKpJglQGKox7aAHyVACkA8zex7JEAOj/li3mIkQHmtQqSOUSRAdjuDUOvLI0CIkgjw6kcjQCEKWNs/BiNAyo140e9gIkAkMrTPzjoiQA==\",\"dtype\":\"float64\",\"shape\":[32]},\"size2\":{\"__ndarray__\":\"T5J2hMThE0CMGFah1asSQIVtW3RMTQ5ALe6IHpa5DUAzcLDpImMMQI/Nd5wKzgpAj813nArOCkAS8TIejG0IQHNcnS3aYghAKvXgFKVPI0ACRMNpJ9YdQDqh5cZ4chxARAFod2LVF0AjGijl7F4WQJo0tfpBNxZA1skqfR5cE0BJg8MkP1YTQJJ46GBTZRFAIJsf+AoXEEC2jw39BGwPQKc2gJQMHg1AzRZplFTrDEBs7J9jhcsMQIPgtPMAKgxADgD7ZuZPC0C9vkzZfS4LQPeRA9toFwtA8/muwDllCkALbgtAObUJQCy4ynmqXQlAYxL2wT+BCEDaQvAUaU4IQA==\",\"dtype\":\"float64\",\"shape\":[32]}},\"selected\":{\"id\":\"5408\"},\"selection_policy\":{\"id\":\"5409\"}},\"id\":\"5101\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"5408\",\"type\":\"Selection\"},{\"attributes\":{},\"id\":\"5088\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"5133\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"5155\",\"type\":\"Scatter\"},{\"attributes\":{\"axis\":{\"id\":\"5146\"},\"dimension\":1,\"ticker\":null},\"id\":\"5148\",\"type\":\"Grid\"},{\"attributes\":{\"below\":[{\"id\":\"5143\"}],\"center\":[{\"id\":\"5145\"},{\"id\":\"5148\"}],\"left\":[{\"id\":\"5146\"}],\"plot_height\":200,\"plot_width\":120,\"renderers\":[{\"id\":\"5156\"}],\"title\":{\"id\":\"5397\"},\"toolbar\":{\"id\":\"5150\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"5134\"},\"x_scale\":{\"id\":\"5139\"},\"y_range\":{\"id\":\"5135\"},\"y_scale\":{\"id\":\"5141\"}},\"id\":\"5136\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"5098\"}]},\"id\":\"5099\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"5409\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"5090\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"5133\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"5154\",\"type\":\"Scatter\"},{\"attributes\":{\"callback\":null},\"id\":\"5149\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"5410\",\"type\":\"Selection\"},{\"attributes\":{\"formatter\":{\"id\":\"5398\"},\"major_label_orientation\":1.5707963267948966,\"major_label_text_font_size\":\"12pt\",\"ticker\":{\"id\":\"5093\"}},\"id\":\"5092\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"5139\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"5082\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Phospho\"},\"y\":{\"field\":\"Cancer\"}},\"id\":\"5103\",\"type\":\"Scatter\"}],\"root_ids\":[\"5159\"]},\"title\":\"Bokeh Application\",\"version\":\"2.0.1\"}};\n",
       "  var render_items = [{\"docid\":\"c40239a3-6f4f-425c-9a47-ff521026bdef\",\"root_ids\":[\"5159\"],\"roots\":{\"5159\":\"d2d9e664-eed6-47eb-8df9-dffc444e7b13\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "5159"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "p.plotCircleHeatMap(plot_df, circle_var = 'P_Value', color_var='Medians', x_axis= 'Phospho', y_axis = 'Cancer',\n",
    "                    plot_height=400, plot_width= 1500, legend_min = .01, legend_max = 0.05) #,save_png = '.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
