{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Interacting Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "\n",
    "import sys\n",
    "sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\WhenMutationsDontMatter\\\\')\n",
    "import plot_utils as pu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that endometrial index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that ovarian index is up-to-date...    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LUAD dataset is under publication embargo until July 01, 2020. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 4)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                            \r"
     ]
    }
   ],
   "source": [
    "gbm = cptac.Gbm()\n",
    "endo = cptac.Endometrial()\n",
    "h = cptac.Hnscc()\n",
    "l = cptac.Luad()\n",
    "o = cptac.Ovarian()\n",
    "col = cptac.Colon()\n",
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11141"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = gbm.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "len(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG_proteomics</th>\n",
       "      <th>A2M_proteomics</th>\n",
       "      <th>AAAS_proteomics</th>\n",
       "      <th>AACS_proteomics</th>\n",
       "      <th>AADAT_proteomics</th>\n",
       "      <th>AAED1_proteomics</th>\n",
       "      <th>AAGAB_proteomics</th>\n",
       "      <th>AAK1_proteomics</th>\n",
       "      <th>AAMDC_proteomics</th>\n",
       "      <th>AAMP_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>0.077630</td>\n",
       "      <td>0.487228</td>\n",
       "      <td>-0.254208</td>\n",
       "      <td>-0.144373</td>\n",
       "      <td>0.551881</td>\n",
       "      <td>-0.025276</td>\n",
       "      <td>-0.467451</td>\n",
       "      <td>-0.089511</td>\n",
       "      <td>-0.078806</td>\n",
       "      <td>0.329603</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047437</td>\n",
       "      <td>-0.105908</td>\n",
       "      <td>-0.347076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.459635</td>\n",
       "      <td>0.079452</td>\n",
       "      <td>-0.784983</td>\n",
       "      <td>-0.488441</td>\n",
       "      <td>0.167990</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>-0.145975</td>\n",
       "      <td>0.798796</td>\n",
       "      <td>0.184242</td>\n",
       "      <td>-0.470603</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.390211</td>\n",
       "      <td>0.245466</td>\n",
       "      <td>-0.609998</td>\n",
       "      <td>0.118625</td>\n",
       "      <td>-0.086927</td>\n",
       "      <td>...</td>\n",
       "      <td>0.161975</td>\n",
       "      <td>-0.213093</td>\n",
       "      <td>0.235571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.107421</td>\n",
       "      <td>0.048724</td>\n",
       "      <td>0.138403</td>\n",
       "      <td>-0.290141</td>\n",
       "      <td>0.405037</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>0.821991</td>\n",
       "      <td>1.096470</td>\n",
       "      <td>-0.094421</td>\n",
       "      <td>-0.106304</td>\n",
       "      <td>0.084578</td>\n",
       "      <td>0.176402</td>\n",
       "      <td>-0.248151</td>\n",
       "      <td>0.014061</td>\n",
       "      <td>-0.699773</td>\n",
       "      <td>-0.638462</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.065534</td>\n",
       "      <td>-0.306717</td>\n",
       "      <td>0.879991</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.883564</td>\n",
       "      <td>-0.172222</td>\n",
       "      <td>0.011876</td>\n",
       "      <td>-0.131889</td>\n",
       "      <td>-0.503581</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>-0.064567</td>\n",
       "      <td>0.129385</td>\n",
       "      <td>0.047751</td>\n",
       "      <td>-0.118187</td>\n",
       "      <td>0.237434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.303847</td>\n",
       "      <td>0.322163</td>\n",
       "      <td>-0.555479</td>\n",
       "      <td>-0.363414</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.254535</td>\n",
       "      <td>0.463653</td>\n",
       "      <td>0.580230</td>\n",
       "      <td>0.503044</td>\n",
       "      <td>-0.604986</td>\n",
       "      <td>0.178077</td>\n",
       "      <td>-0.720059</td>\n",
       "      <td>-0.150197</td>\n",
       "      <td>-0.268715</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-0.763691</td>\n",
       "      <td>-1.031834</td>\n",
       "      <td>-0.217194</td>\n",
       "      <td>-0.695701</td>\n",
       "      <td>0.184173</td>\n",
       "      <td>-0.474816</td>\n",
       "      <td>-0.051789</td>\n",
       "      <td>0.344842</td>\n",
       "      <td>-0.642746</td>\n",
       "      <td>0.068863</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.092502</td>\n",
       "      <td>0.010639</td>\n",
       "      <td>-0.465079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.500083</td>\n",
       "      <td>0.112651</td>\n",
       "      <td>1.004660</td>\n",
       "      <td>-0.230304</td>\n",
       "      <td>-0.102416</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>0.230843</td>\n",
       "      <td>0.426624</td>\n",
       "      <td>0.420155</td>\n",
       "      <td>0.303960</td>\n",
       "      <td>0.078341</td>\n",
       "      <td>0.833128</td>\n",
       "      <td>0.205727</td>\n",
       "      <td>-0.434020</td>\n",
       "      <td>0.151202</td>\n",
       "      <td>0.208972</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.021466</td>\n",
       "      <td>-0.049569</td>\n",
       "      <td>0.128461</td>\n",
       "      <td>-0.437797</td>\n",
       "      <td>-0.144343</td>\n",
       "      <td>-0.024981</td>\n",
       "      <td>-0.275897</td>\n",
       "      <td>-0.116367</td>\n",
       "      <td>0.351479</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>-0.635316</td>\n",
       "      <td>-0.478657</td>\n",
       "      <td>0.647004</td>\n",
       "      <td>-0.050322</td>\n",
       "      <td>-0.440517</td>\n",
       "      <td>0.092256</td>\n",
       "      <td>-0.006924</td>\n",
       "      <td>-0.511331</td>\n",
       "      <td>0.101877</td>\n",
       "      <td>0.128163</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.034616</td>\n",
       "      <td>0.158269</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.906358</td>\n",
       "      <td>0.287317</td>\n",
       "      <td>0.745880</td>\n",
       "      <td>-0.320613</td>\n",
       "      <td>-0.180511</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>0.616640</td>\n",
       "      <td>0.767029</td>\n",
       "      <td>-0.231897</td>\n",
       "      <td>0.050841</td>\n",
       "      <td>0.304359</td>\n",
       "      <td>-0.384608</td>\n",
       "      <td>-0.205284</td>\n",
       "      <td>0.513575</td>\n",
       "      <td>0.151769</td>\n",
       "      <td>-0.307037</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.110846</td>\n",
       "      <td>-0.069605</td>\n",
       "      <td>-0.183921</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.681170</td>\n",
       "      <td>0.095601</td>\n",
       "      <td>-0.096716</td>\n",
       "      <td>0.049811</td>\n",
       "      <td>-0.409252</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>-0.059547</td>\n",
       "      <td>-0.526563</td>\n",
       "      <td>-0.269242</td>\n",
       "      <td>-0.176220</td>\n",
       "      <td>0.142132</td>\n",
       "      <td>0.214497</td>\n",
       "      <td>-0.496471</td>\n",
       "      <td>0.450404</td>\n",
       "      <td>-0.533063</td>\n",
       "      <td>-0.398838</td>\n",
       "      <td>...</td>\n",
       "      <td>0.094553</td>\n",
       "      <td>-0.222705</td>\n",
       "      <td>-0.283108</td>\n",
       "      <td>-0.326181</td>\n",
       "      <td>-0.212798</td>\n",
       "      <td>0.535665</td>\n",
       "      <td>0.289848</td>\n",
       "      <td>-0.278484</td>\n",
       "      <td>-0.314278</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>-0.899255</td>\n",
       "      <td>-0.333312</td>\n",
       "      <td>0.404318</td>\n",
       "      <td>-0.570473</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.590039</td>\n",
       "      <td>0.423253</td>\n",
       "      <td>0.114616</td>\n",
       "      <td>0.366197</td>\n",
       "      <td>0.193688</td>\n",
       "      <td>...</td>\n",
       "      <td>0.241629</td>\n",
       "      <td>0.070969</td>\n",
       "      <td>0.768267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.060488</td>\n",
       "      <td>-0.340455</td>\n",
       "      <td>-0.124841</td>\n",
       "      <td>-0.270490</td>\n",
       "      <td>0.145806</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97 rows × 11142 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        A1BG_proteomics  A2M_proteomics  AAAS_proteomics  AACS_proteomics  \\\n",
       "Patient_ID                                                                      \n",
       "C3L-00104          0.077630        0.487228        -0.254208        -0.144373   \n",
       "C3L-00365         -0.145975        0.798796         0.184242        -0.470603   \n",
       "C3L-00674          0.821991        1.096470        -0.094421        -0.106304   \n",
       "C3L-00677         -0.064567        0.129385         0.047751        -0.118187   \n",
       "C3L-01040         -0.763691       -1.031834        -0.217194        -0.695701   \n",
       "...                     ...             ...              ...              ...   \n",
       "C3N-03183          0.230843        0.426624         0.420155         0.303960   \n",
       "C3N-03184         -0.635316       -0.478657         0.647004        -0.050322   \n",
       "C3N-03186          0.616640        0.767029        -0.231897         0.050841   \n",
       "C3N-03188         -0.059547       -0.526563        -0.269242        -0.176220   \n",
       "C3N-03473         -0.899255       -0.333312         0.404318        -0.570473   \n",
       "\n",
       "Name        AADAT_proteomics  AAED1_proteomics  AAGAB_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00104           0.551881         -0.025276         -0.467451   \n",
       "C3L-00365                NaN          0.390211          0.245466   \n",
       "C3L-00674           0.084578          0.176402         -0.248151   \n",
       "C3L-00677           0.237434               NaN          0.303847   \n",
       "C3L-01040           0.184173         -0.474816         -0.051789   \n",
       "...                      ...               ...               ...   \n",
       "C3N-03183           0.078341          0.833128          0.205727   \n",
       "C3N-03184          -0.440517          0.092256         -0.006924   \n",
       "C3N-03186           0.304359         -0.384608         -0.205284   \n",
       "C3N-03188           0.142132          0.214497         -0.496471   \n",
       "C3N-03473                NaN         -0.590039          0.423253   \n",
       "\n",
       "Name        AAK1_proteomics  AAMDC_proteomics  AAMP_proteomics  ...  \\\n",
       "Patient_ID                                                      ...   \n",
       "C3L-00104         -0.089511         -0.078806         0.329603  ...   \n",
       "C3L-00365         -0.609998          0.118625        -0.086927  ...   \n",
       "C3L-00674          0.014061         -0.699773        -0.638462  ...   \n",
       "C3L-00677          0.322163         -0.555479        -0.363414  ...   \n",
       "C3L-01040          0.344842         -0.642746         0.068863  ...   \n",
       "...                     ...               ...              ...  ...   \n",
       "C3N-03183         -0.434020          0.151202         0.208972  ...   \n",
       "C3N-03184         -0.511331          0.101877         0.128163  ...   \n",
       "C3N-03186          0.513575          0.151769        -0.307037  ...   \n",
       "C3N-03188          0.450404         -0.533063        -0.398838  ...   \n",
       "C3N-03473          0.114616          0.366197         0.193688  ...   \n",
       "\n",
       "Name        ZSWIM8_proteomics  ZW10_proteomics  ZWILCH_proteomics  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104           -0.047437        -0.105908          -0.347076   \n",
       "C3L-00365            0.161975        -0.213093           0.235571   \n",
       "C3L-00674           -0.065534        -0.306717           0.879991   \n",
       "C3L-00677           -0.254535         0.463653           0.580230   \n",
       "C3L-01040           -0.092502         0.010639          -0.465079   \n",
       "...                       ...              ...                ...   \n",
       "C3N-03183           -0.021466        -0.049569           0.128461   \n",
       "C3N-03184           -0.034616         0.158269                NaN   \n",
       "C3N-03186           -0.110846        -0.069605          -0.183921   \n",
       "C3N-03188            0.094553        -0.222705          -0.283108   \n",
       "C3N-03473            0.241629         0.070969           0.768267   \n",
       "\n",
       "Name        ZWINT_proteomics  ZXDC_proteomics  ZYG11B_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00104                NaN         0.459635           0.079452   \n",
       "C3L-00365                NaN         0.107421           0.048724   \n",
       "C3L-00674                NaN         0.883564          -0.172222   \n",
       "C3L-00677           0.503044        -0.604986           0.178077   \n",
       "C3L-01040                NaN        -0.500083           0.112651   \n",
       "...                      ...              ...                ...   \n",
       "C3N-03183          -0.437797        -0.144343          -0.024981   \n",
       "C3N-03184                NaN         0.906358           0.287317   \n",
       "C3N-03186                NaN        -0.681170           0.095601   \n",
       "C3N-03188          -0.326181        -0.212798           0.535665   \n",
       "C3N-03473                NaN        -0.060488          -0.340455   \n",
       "\n",
       "Name        ZYX_proteomics  ZZEF1_proteomics  ZZZ3_proteomics        Mutation  \n",
       "Patient_ID                                                                     \n",
       "C3L-00104        -0.784983         -0.488441         0.167990        Deletion  \n",
       "C3L-00365         0.138403         -0.290141         0.405037        Deletion  \n",
       "C3L-00674         0.011876         -0.131889        -0.503581        Deletion  \n",
       "C3L-00677        -0.720059         -0.150197        -0.268715        Deletion  \n",
       "C3L-01040         1.004660         -0.230304        -0.102416        Deletion  \n",
       "...                    ...               ...              ...             ...  \n",
       "C3N-03183        -0.275897         -0.116367         0.351479        Deletion  \n",
       "C3N-03184         0.745880         -0.320613        -0.180511        Deletion  \n",
       "C3N-03186        -0.096716          0.049811        -0.409252        Deletion  \n",
       "C3N-03188         0.289848         -0.278484        -0.314278  Wildtype_Tumor  \n",
       "C3N-03473        -0.124841         -0.270490         0.145806        Deletion  \n",
       "\n",
       "[97 rows x 11142 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = gbm.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = gbm.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "del_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "               Comparison       P_Value\n",
      "0       ARMH3_proteomics  5.396032e-11\n",
      "1        CUTC_proteomics  1.593480e-10\n",
      "2     PIP4K2A_proteomics  1.009419e-09\n",
      "3        CUL2_proteomics  1.122076e-09\n",
      "4        GDI2_proteomics  1.302273e-09\n",
      "...                  ...           ...\n",
      "1882  GOLGA7B_proteomics  8.446816e-03\n",
      "1883     TEX2_proteomics  8.452685e-03\n",
      "1884     MAFF_proteomics  8.463737e-03\n",
      "1885   LRPAP1_proteomics  8.466030e-03\n",
      "1886  PPP2R2B_proteomics  8.466960e-03\n",
      "\n",
      "[1887 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3_proteomics</td>\n",
       "      <td>5.396032e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC_proteomics</td>\n",
       "      <td>1.593480e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A_proteomics</td>\n",
       "      <td>1.009419e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2_proteomics</td>\n",
       "      <td>1.122076e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2_proteomics</td>\n",
       "      <td>1.302273e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11136</th>\n",
       "      <td>ZNF805_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11137</th>\n",
       "      <td>ZNF813_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11138</th>\n",
       "      <td>ZNF814_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11139</th>\n",
       "      <td>ZNF888_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11140</th>\n",
       "      <td>ZSCAN12_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11141 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics   Gbm_P_Value\n",
       "0        ARMH3_proteomics  5.396032e-11\n",
       "1         CUTC_proteomics  1.593480e-10\n",
       "2      PIP4K2A_proteomics  1.009419e-09\n",
       "3         CUL2_proteomics  1.122076e-09\n",
       "4         GDI2_proteomics  1.302273e-09\n",
       "...                   ...           ...\n",
       "11136   ZNF805_proteomics           NaN\n",
       "11137   ZNF813_proteomics           NaN\n",
       "11138   ZNF814_proteomics           NaN\n",
       "11139   ZNF888_proteomics           NaN\n",
       "11140  ZSCAN12_proteomics           NaN\n",
       "\n",
       "[11141 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "g_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if g_sig is not None:\n",
    "    g_sig_list = list(g_sig.Comparison)\n",
    "else: \n",
    "    g_sig_list = None\n",
    "print('significant pvals: \\n',g_sig)\n",
    "\n",
    "# Get all pvals\n",
    "g = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "g_pval = g.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Gbm_P_Value'})\n",
    "#g_pval = g_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "g_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "gbm_d = {}\n",
    "\n",
    "\n",
    "for prot in p:\n",
    "    dif_gbm = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    gbm_d[prot+'_proteomics'] = dif_gbm\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif_gbm)\n",
    "\n",
    "gbm_df = pd.DataFrame.from_dict(gbm_d, orient='index', columns=['Gbm_Median'])\n",
    "gbm_df = gbm_df.reset_index().rename(columns={'index':'Proteomics'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.005661958017430049 - 0.20855503331380604\n"
     ]
    }
   ],
   "source": [
    "#TESTS\n",
    "gbm_df.loc[gbm_df['Proteomics'] == 'MAFF_proteomics']\n",
    "\n",
    "print(wt_med.MAFF_proteomics, '-', del_med.MAFF_proteomics)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3_proteomics</td>\n",
       "      <td>5.396032e-11</td>\n",
       "      <td>0.405134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC_proteomics</td>\n",
       "      <td>1.593480e-10</td>\n",
       "      <td>0.553255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A_proteomics</td>\n",
       "      <td>1.009419e-09</td>\n",
       "      <td>0.838882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2_proteomics</td>\n",
       "      <td>1.122076e-09</td>\n",
       "      <td>0.586396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2_proteomics</td>\n",
       "      <td>1.302273e-09</td>\n",
       "      <td>0.610188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11136</th>\n",
       "      <td>ZNF805_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11137</th>\n",
       "      <td>ZNF813_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11138</th>\n",
       "      <td>ZNF814_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.265936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11139</th>\n",
       "      <td>ZNF888_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11140</th>\n",
       "      <td>ZSCAN12_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11141 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics   Gbm_P_Value  Gbm_Median\n",
       "0        ARMH3_proteomics  5.396032e-11    0.405134\n",
       "1         CUTC_proteomics  1.593480e-10    0.553255\n",
       "2      PIP4K2A_proteomics  1.009419e-09    0.838882\n",
       "3         CUL2_proteomics  1.122076e-09    0.586396\n",
       "4         GDI2_proteomics  1.302273e-09    0.610188\n",
       "...                   ...           ...         ...\n",
       "11136   ZNF805_proteomics           NaN         NaN\n",
       "11137   ZNF813_proteomics           NaN         NaN\n",
       "11138   ZNF814_proteomics           NaN    0.265936\n",
       "11139   ZNF888_proteomics           NaN         NaN\n",
       "11140  ZSCAN12_proteomics           NaN         NaN\n",
       "\n",
       "[11141 rows x 3 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_merged = g_pval.merge(gbm_df, on='Proteomics',how='inner')\n",
    "g_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# all proteins: 10999\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 69 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG_proteomics</th>\n",
       "      <th>A2M_proteomics</th>\n",
       "      <th>A2ML1_proteomics</th>\n",
       "      <th>A4GALT_proteomics</th>\n",
       "      <th>AAAS_proteomics</th>\n",
       "      <th>AACS_proteomics</th>\n",
       "      <th>AADAT_proteomics</th>\n",
       "      <th>AAED1_proteomics</th>\n",
       "      <th>AAGAB_proteomics</th>\n",
       "      <th>AAK1_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM9_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.4120</td>\n",
       "      <td>-0.6640</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02290</td>\n",
       "      <td>0.109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.4330</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.08590</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.4190</td>\n",
       "      <td>-0.5000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.14500</td>\n",
       "      <td>0.01050</td>\n",
       "      <td>-0.116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.0740</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.32000</td>\n",
       "      <td>-0.41900</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>-0.374</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.5370</td>\n",
       "      <td>0.3110</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5650</td>\n",
       "      <td>-0.1010</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.22000</td>\n",
       "      <td>-0.562</td>\n",
       "      <td>0.9370</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.2070</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>0.05130</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00137</th>\n",
       "      <td>-1.320</td>\n",
       "      <td>-0.7080</td>\n",
       "      <td>-0.8080</td>\n",
       "      <td>-0.0709</td>\n",
       "      <td>0.1380</td>\n",
       "      <td>0.6560</td>\n",
       "      <td>-0.2800</td>\n",
       "      <td>-0.1280</td>\n",
       "      <td>0.2170</td>\n",
       "      <td>-0.1330</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00491</td>\n",
       "      <td>0.18000</td>\n",
       "      <td>0.513</td>\n",
       "      <td>0.5490</td>\n",
       "      <td>-0.681</td>\n",
       "      <td>-0.2850</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.08760</td>\n",
       "      <td>0.00698</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00139</th>\n",
       "      <td>-0.467</td>\n",
       "      <td>0.3700</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4340</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>-0.1750</td>\n",
       "      <td>0.1810</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>-0.0512</td>\n",
       "      <td>...</td>\n",
       "      <td>0.23900</td>\n",
       "      <td>0.14000</td>\n",
       "      <td>1.070</td>\n",
       "      <td>0.6070</td>\n",
       "      <td>0.486</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>-0.632</td>\n",
       "      <td>-0.20300</td>\n",
       "      <td>-0.06850</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>-1.070</td>\n",
       "      <td>-0.7120</td>\n",
       "      <td>0.0462</td>\n",
       "      <td>-0.0471</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>1.3500</td>\n",
       "      <td>1.2100</td>\n",
       "      <td>0.0048</td>\n",
       "      <td>0.6820</td>\n",
       "      <td>-0.2240</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.40100</td>\n",
       "      <td>0.55700</td>\n",
       "      <td>0.827</td>\n",
       "      <td>0.3480</td>\n",
       "      <td>-0.187</td>\n",
       "      <td>-0.1070</td>\n",
       "      <td>-0.830</td>\n",
       "      <td>0.06200</td>\n",
       "      <td>-0.52800</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>-1.280</td>\n",
       "      <td>-0.7360</td>\n",
       "      <td>-0.5520</td>\n",
       "      <td>0.1140</td>\n",
       "      <td>0.2730</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.8500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5360</td>\n",
       "      <td>-0.2660</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00364</td>\n",
       "      <td>0.763</td>\n",
       "      <td>0.0784</td>\n",
       "      <td>-0.203</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>-1.200</td>\n",
       "      <td>-0.35700</td>\n",
       "      <td>0.02990</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>-0.290</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>3.1700</td>\n",
       "      <td>-0.9070</td>\n",
       "      <td>0.0317</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2730</td>\n",
       "      <td>-0.0747</td>\n",
       "      <td>0.1240</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.52100</td>\n",
       "      <td>0.05230</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.5850</td>\n",
       "      <td>0.138</td>\n",
       "      <td>-0.1900</td>\n",
       "      <td>-0.966</td>\n",
       "      <td>-0.00627</td>\n",
       "      <td>-0.24900</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>0.266</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.4700</td>\n",
       "      <td>0.3980</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>0.0940</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.01220</td>\n",
       "      <td>0.250</td>\n",
       "      <td>0.5530</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.0642</td>\n",
       "      <td>-0.437</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.49800</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>0.692</td>\n",
       "      <td>0.5890</td>\n",
       "      <td>0.5870</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0835</td>\n",
       "      <td>0.1720</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>0.0140</td>\n",
       "      <td>-0.1460</td>\n",
       "      <td>0.8570</td>\n",
       "      <td>...</td>\n",
       "      <td>0.29800</td>\n",
       "      <td>-0.12400</td>\n",
       "      <td>0.178</td>\n",
       "      <td>0.1470</td>\n",
       "      <td>-0.114</td>\n",
       "      <td>0.1370</td>\n",
       "      <td>1.240</td>\n",
       "      <td>0.00682</td>\n",
       "      <td>-0.02330</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        A1BG_proteomics  A2M_proteomics  A2ML1_proteomics  \\\n",
       "Patient_ID                                                      \n",
       "C3L-00006            -1.180         -0.8630           -0.8020   \n",
       "C3L-00032            -0.528         -1.3200            0.4350   \n",
       "C3L-00098            -0.374         -0.0206           -0.5370   \n",
       "C3L-00137            -1.320         -0.7080           -0.8080   \n",
       "C3L-00139            -0.467          0.3700           -0.3390   \n",
       "...                     ...             ...               ...   \n",
       "C3N-01520            -1.070         -0.7120            0.0462   \n",
       "C3N-01521            -1.280         -0.7360           -0.5520   \n",
       "C3N-01537            -0.290         -0.3200            3.1700   \n",
       "C3N-01802             0.266          1.3900           -0.0655   \n",
       "C3N-01825             0.692          0.5890            0.5870   \n",
       "\n",
       "Name        A4GALT_proteomics  AAAS_proteomics  AACS_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00006              0.2220           0.2560           0.6650   \n",
       "C3L-00032                 NaN          -0.2400           1.0400   \n",
       "C3L-00098              0.3110           0.3750           0.0131   \n",
       "C3L-00137             -0.0709           0.1380           0.6560   \n",
       "C3L-00139                 NaN           0.4340           0.0358   \n",
       "...                       ...              ...              ...   \n",
       "C3N-01520             -0.0471           0.2270           1.3500   \n",
       "C3N-01521              0.1140           0.2730           1.0700   \n",
       "C3N-01537             -0.9070           0.0317          -0.0425   \n",
       "C3N-01802              0.4700           0.3980          -0.1340   \n",
       "C3N-01825                 NaN          -0.0835           0.1720   \n",
       "\n",
       "Name        AADAT_proteomics  AAED1_proteomics  AAGAB_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00006             1.2800           -0.3390            0.4120   \n",
       "C3L-00032            -0.0213           -0.0479            0.4190   \n",
       "C3L-00098            -1.1000               NaN            0.5650   \n",
       "C3L-00137            -0.2800           -0.1280            0.2170   \n",
       "C3L-00139            -0.1750            0.1810            0.1160   \n",
       "...                      ...               ...               ...   \n",
       "C3N-01520             1.2100            0.0048            0.6820   \n",
       "C3N-01521             0.8500               NaN           -0.5360   \n",
       "C3N-01537                NaN           -0.2730           -0.0747   \n",
       "C3N-01802             0.4610            1.0400            0.3630   \n",
       "C3N-01825            -0.2150            0.0140           -0.1460   \n",
       "\n",
       "Name        AAK1_proteomics  ...  ZSWIM9_proteomics  ZW10_proteomics  \\\n",
       "Patient_ID                   ...                                       \n",
       "C3L-00006           -0.6640  ...                NaN          0.02290   \n",
       "C3L-00032           -0.5000  ...           -0.14500          0.01050   \n",
       "C3L-00098           -0.1010  ...                NaN         -1.22000   \n",
       "C3L-00137           -0.1330  ...           -0.00491          0.18000   \n",
       "C3L-00139           -0.0512  ...            0.23900          0.14000   \n",
       "...                     ...  ...                ...              ...   \n",
       "C3N-01520           -0.2240  ...           -0.40100          0.55700   \n",
       "C3N-01521           -0.2660  ...                NaN         -0.00364   \n",
       "C3N-01537            0.1240  ...           -0.52100          0.05230   \n",
       "C3N-01802            0.0940  ...                NaN         -0.01220   \n",
       "C3N-01825            0.8570  ...            0.29800         -0.12400   \n",
       "\n",
       "Name        ZWILCH_proteomics  ZWINT_proteomics  ZXDC_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00006               0.109               NaN           -0.332   \n",
       "C3L-00032              -0.116               NaN            0.151   \n",
       "C3L-00098              -0.562            0.9370           -0.646   \n",
       "C3L-00137               0.513            0.5490           -0.681   \n",
       "C3L-00139               1.070            0.6070            0.486   \n",
       "...                       ...               ...              ...   \n",
       "C3N-01520               0.827            0.3480           -0.187   \n",
       "C3N-01521               0.763            0.0784           -0.203   \n",
       "C3N-01537               0.515            0.5850            0.138   \n",
       "C3N-01802               0.250            0.5530            0.387   \n",
       "C3N-01825               0.178            0.1470           -0.114   \n",
       "\n",
       "Name        ZYG11B_proteomics  ZYX_proteomics  ZZEF1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00006             -0.4330          -1.020          -0.12300   \n",
       "C3L-00032             -0.0740          -0.540           0.32000   \n",
       "C3L-00098              0.2070          -1.850          -0.17600   \n",
       "C3L-00137             -0.2850          -0.564          -0.08760   \n",
       "C3L-00139              0.1690          -0.632          -0.20300   \n",
       "...                       ...             ...               ...   \n",
       "C3N-01520             -0.1070          -0.830           0.06200   \n",
       "C3N-01521              0.0132          -1.200          -0.35700   \n",
       "C3N-01537             -0.1900          -0.966          -0.00627   \n",
       "C3N-01802              0.0642          -0.437           0.10400   \n",
       "C3N-01825              0.1370           1.240           0.00682   \n",
       "\n",
       "Name        ZZZ3_proteomics        Mutation  \n",
       "Patient_ID                                   \n",
       "C3L-00006          -0.08590      Truncation  \n",
       "C3L-00032          -0.41900      Truncation  \n",
       "C3L-00098           0.05130  Wildtype_Tumor  \n",
       "C3L-00137           0.00698      Truncation  \n",
       "C3L-00139          -0.06850  Wildtype_Tumor  \n",
       "...                     ...             ...  \n",
       "C3N-01520          -0.52800      Truncation  \n",
       "C3N-01521           0.02990  Wildtype_Tumor  \n",
       "C3N-01537          -0.24900  Wildtype_Tumor  \n",
       "C3N-01802          -0.49800  Wildtype_Tumor  \n",
       "C3N-01825          -0.02330  Wildtype_Tumor  \n",
       "\n",
       "[68 rows x 11000 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Endo ttest\n",
    "\n",
    "prot = endo.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "print('# all proteins:',len(p))\n",
    "\n",
    "mut_type = endo.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = endo.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "trunc_wt = merged[get]\n",
    "trunc_wt['Mutation'] = np.where(\n",
    "                trunc_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "trunc_wt['Mutation'].value_counts()\n",
    "trunc_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "             Comparison       P_Value\n",
      "0     NOL10_proteomics  8.999599e-08\n",
      "1      ABT1_proteomics  3.141336e-07\n",
      "2     UTP25_proteomics  4.308134e-07\n",
      "3    TOPBP1_proteomics  4.885533e-07\n",
      "4      RIF1_proteomics  9.004982e-07\n",
      "..                 ...           ...\n",
      "397   ACTR5_proteomics  1.788167e-03\n",
      "398   RSPH3_proteomics  1.789386e-03\n",
      "399   COPS5_proteomics  1.793000e-03\n",
      "400   DDAH1_proteomics  1.822621e-03\n",
      "401   UCHL1_proteomics  1.825674e-03\n",
      "\n",
      "[402 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NOL10_proteomics</td>\n",
       "      <td>8.999599e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABT1_proteomics</td>\n",
       "      <td>3.141336e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UTP25_proteomics</td>\n",
       "      <td>4.308134e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TOPBP1_proteomics</td>\n",
       "      <td>4.885533e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RIF1_proteomics</td>\n",
       "      <td>9.004982e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10994</th>\n",
       "      <td>ZNF586_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10995</th>\n",
       "      <td>ZNF630_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10996</th>\n",
       "      <td>ZNF772_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10997</th>\n",
       "      <td>ZNF79_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10998</th>\n",
       "      <td>ZNF805_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10999 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Proteomics    En_P_Value\n",
       "0       NOL10_proteomics  8.999599e-08\n",
       "1        ABT1_proteomics  3.141336e-07\n",
       "2       UTP25_proteomics  4.308134e-07\n",
       "3      TOPBP1_proteomics  4.885533e-07\n",
       "4        RIF1_proteomics  9.004982e-07\n",
       "...                  ...           ...\n",
       "10994  ZNF586_proteomics           NaN\n",
       "10995  ZNF630_proteomics           NaN\n",
       "10996  ZNF772_proteomics           NaN\n",
       "10997   ZNF79_proteomics           NaN\n",
       "10998  ZNF805_proteomics           NaN\n",
       "\n",
       "[10999 rows x 2 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(trunc_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "e_sig = u.wrap_ttest(trunc_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if e_sig is not None:\n",
    "    e_sig_list = list(e_sig.Comparison)\n",
    "else: \n",
    "    e_sig_list = None\n",
    "print('significant pvals: \\n',e_sig)\n",
    "\n",
    "# Get all pvals\n",
    "e_pval = u.wrap_ttest(trunc_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "e_pval = e_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'En_P_Value'})\n",
    "e_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG_proteomics</td>\n",
       "      <td>0.06100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M_proteomics</td>\n",
       "      <td>-0.07400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2ML1_proteomics</td>\n",
       "      <td>-0.22300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A4GALT_proteomics</td>\n",
       "      <td>-0.03995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS_proteomics</td>\n",
       "      <td>0.19200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10994</th>\n",
       "      <td>ZXDC_proteomics</td>\n",
       "      <td>0.42800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10995</th>\n",
       "      <td>ZYG11B_proteomics</td>\n",
       "      <td>0.19420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10996</th>\n",
       "      <td>ZYX_proteomics</td>\n",
       "      <td>-0.21600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10997</th>\n",
       "      <td>ZZEF1_proteomics</td>\n",
       "      <td>-0.05320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10998</th>\n",
       "      <td>ZZZ3_proteomics</td>\n",
       "      <td>0.02120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10999 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Proteomics  En_Median\n",
       "0        A1BG_proteomics    0.06100\n",
       "1         A2M_proteomics   -0.07400\n",
       "2       A2ML1_proteomics   -0.22300\n",
       "3      A4GALT_proteomics   -0.03995\n",
       "4        AAAS_proteomics    0.19200\n",
       "...                  ...        ...\n",
       "10994    ZXDC_proteomics    0.42800\n",
       "10995  ZYG11B_proteomics    0.19420\n",
       "10996     ZYX_proteomics   -0.21600\n",
       "10997   ZZEF1_proteomics   -0.05320\n",
       "10998    ZZZ3_proteomics    0.02120\n",
       "\n",
       "[10999 rows x 2 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = trunc_wt[trunc_wt.Mutation == \"Truncation\"]\n",
    "wt = trunc_wt[trunc_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "trunc_med = t.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "en_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - trunc_med[prot+'_proteomics']\n",
    "    en_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', trunc_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "en_df = pd.DataFrame.from_dict(en_d, orient='index', columns=['En_Median'])\n",
    "en_df = en_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "en_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.16699999999999998 - -0.284\n",
      "            Proteomics  En_P_Value\n",
      "4777  DOCK1_proteomics    0.253937\n"
     ]
    }
   ],
   "source": [
    "#TESTS\n",
    "I = 'DOCK1'\n",
    "en_df.loc[en_df['Proteomics'] == I+'_proteomics']\n",
    "\n",
    "print(wt_med[I+'_proteomics'], '-', trunc_med[I+'_proteomics'])\n",
    "\n",
    "print(e_pval.loc[e_pval['Proteomics'] == I+'_proteomics'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NOL10_proteomics</td>\n",
       "      <td>8.999599e-08</td>\n",
       "      <td>0.44200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABT1_proteomics</td>\n",
       "      <td>3.141336e-07</td>\n",
       "      <td>0.48510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UTP25_proteomics</td>\n",
       "      <td>4.308134e-07</td>\n",
       "      <td>0.73450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TOPBP1_proteomics</td>\n",
       "      <td>4.885533e-07</td>\n",
       "      <td>0.61795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RIF1_proteomics</td>\n",
       "      <td>9.004982e-07</td>\n",
       "      <td>0.54640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10994</th>\n",
       "      <td>ZNF586_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10995</th>\n",
       "      <td>ZNF630_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.99250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10996</th>\n",
       "      <td>ZNF772_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10997</th>\n",
       "      <td>ZNF79_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10998</th>\n",
       "      <td>ZNF805_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10999 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Proteomics    En_P_Value  En_Median\n",
       "0       NOL10_proteomics  8.999599e-08    0.44200\n",
       "1        ABT1_proteomics  3.141336e-07    0.48510\n",
       "2       UTP25_proteomics  4.308134e-07    0.73450\n",
       "3      TOPBP1_proteomics  4.885533e-07    0.61795\n",
       "4        RIF1_proteomics  9.004982e-07    0.54640\n",
       "...                  ...           ...        ...\n",
       "10994  ZNF586_proteomics           NaN        NaN\n",
       "10995  ZNF630_proteomics           NaN   -0.99250\n",
       "10996  ZNF772_proteomics           NaN        NaN\n",
       "10997   ZNF79_proteomics           NaN        NaN\n",
       "10998  ZNF805_proteomics           NaN        NaN\n",
       "\n",
       "[10999 rows x 3 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_merged = e_pval.merge(en_df, on='Proteomics',how='inner')\n",
    "e_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                            \r"
     ]
    }
   ],
   "source": [
    "o = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "10686"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "gene = 'PTEN'\n",
    "\n",
    "prot = o.get_proteomics()\n",
    "prot = o.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "p = list(prot.columns)\n",
    "len(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG_proteomics</th>\n",
       "      <th>A2M_proteomics</th>\n",
       "      <th>A2ML1_proteomics</th>\n",
       "      <th>AAAS_proteomics</th>\n",
       "      <th>AACS_proteomics</th>\n",
       "      <th>AAGAB_proteomics</th>\n",
       "      <th>AAK1_proteomics</th>\n",
       "      <th>AAMDC_proteomics</th>\n",
       "      <th>AAMP_proteomics</th>\n",
       "      <th>AAR2_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSCAN32_proteomics</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01OV007</th>\n",
       "      <td>0.133634</td>\n",
       "      <td>0.765120</td>\n",
       "      <td>-0.778849</td>\n",
       "      <td>0.118236</td>\n",
       "      <td>0.112740</td>\n",
       "      <td>0.054942</td>\n",
       "      <td>0.313055</td>\n",
       "      <td>-0.786270</td>\n",
       "      <td>-0.384680</td>\n",
       "      <td>0.064017</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.323516</td>\n",
       "      <td>0.074209</td>\n",
       "      <td>0.232999</td>\n",
       "      <td>-0.195793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.145053</td>\n",
       "      <td>0.590217</td>\n",
       "      <td>0.019414</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV017</th>\n",
       "      <td>-0.432786</td>\n",
       "      <td>-0.461906</td>\n",
       "      <td>-0.533497</td>\n",
       "      <td>0.041800</td>\n",
       "      <td>0.604114</td>\n",
       "      <td>0.351294</td>\n",
       "      <td>0.546605</td>\n",
       "      <td>0.603935</td>\n",
       "      <td>-0.235374</td>\n",
       "      <td>0.059084</td>\n",
       "      <td>...</td>\n",
       "      <td>0.032004</td>\n",
       "      <td>0.016276</td>\n",
       "      <td>0.587911</td>\n",
       "      <td>0.574339</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.075401</td>\n",
       "      <td>-0.437564</td>\n",
       "      <td>-0.208831</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV018</th>\n",
       "      <td>-0.013943</td>\n",
       "      <td>0.065842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.010514</td>\n",
       "      <td>0.541935</td>\n",
       "      <td>0.169826</td>\n",
       "      <td>-0.023032</td>\n",
       "      <td>-0.057549</td>\n",
       "      <td>-0.397627</td>\n",
       "      <td>0.199581</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002237</td>\n",
       "      <td>0.206993</td>\n",
       "      <td>0.111037</td>\n",
       "      <td>1.020321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.245864</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>0.130504</td>\n",
       "      <td>-0.367433</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV023</th>\n",
       "      <td>1.234203</td>\n",
       "      <td>1.512082</td>\n",
       "      <td>-0.703050</td>\n",
       "      <td>0.181152</td>\n",
       "      <td>0.087719</td>\n",
       "      <td>0.226861</td>\n",
       "      <td>-0.100766</td>\n",
       "      <td>-0.516022</td>\n",
       "      <td>-0.456512</td>\n",
       "      <td>-0.257629</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.529948</td>\n",
       "      <td>0.034713</td>\n",
       "      <td>0.149523</td>\n",
       "      <td>-0.313370</td>\n",
       "      <td>-0.507473</td>\n",
       "      <td>0.221818</td>\n",
       "      <td>0.278313</td>\n",
       "      <td>-1.008998</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV026</th>\n",
       "      <td>-0.482872</td>\n",
       "      <td>0.343872</td>\n",
       "      <td>-1.150936</td>\n",
       "      <td>0.315155</td>\n",
       "      <td>0.097235</td>\n",
       "      <td>-0.160208</td>\n",
       "      <td>-0.125165</td>\n",
       "      <td>-0.768776</td>\n",
       "      <td>-0.372368</td>\n",
       "      <td>0.450086</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.752819</td>\n",
       "      <td>-0.181027</td>\n",
       "      <td>0.179612</td>\n",
       "      <td>0.007854</td>\n",
       "      <td>-0.106049</td>\n",
       "      <td>0.377063</td>\n",
       "      <td>-0.295592</td>\n",
       "      <td>0.011997</td>\n",
       "      <td>-0.236003</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26OV002</th>\n",
       "      <td>1.942785</td>\n",
       "      <td>2.028395</td>\n",
       "      <td>-0.506808</td>\n",
       "      <td>-0.237500</td>\n",
       "      <td>-0.767353</td>\n",
       "      <td>-0.094263</td>\n",
       "      <td>0.164654</td>\n",
       "      <td>-0.120120</td>\n",
       "      <td>-0.560682</td>\n",
       "      <td>0.116122</td>\n",
       "      <td>...</td>\n",
       "      <td>0.291665</td>\n",
       "      <td>-0.205511</td>\n",
       "      <td>0.280838</td>\n",
       "      <td>1.143053</td>\n",
       "      <td>0.545983</td>\n",
       "      <td>0.045314</td>\n",
       "      <td>-0.463719</td>\n",
       "      <td>-0.321074</td>\n",
       "      <td>0.103304</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26OV008</th>\n",
       "      <td>-0.576189</td>\n",
       "      <td>-1.723434</td>\n",
       "      <td>2.023556</td>\n",
       "      <td>0.066470</td>\n",
       "      <td>0.576987</td>\n",
       "      <td>0.725690</td>\n",
       "      <td>0.610382</td>\n",
       "      <td>-0.704039</td>\n",
       "      <td>0.121085</td>\n",
       "      <td>0.534117</td>\n",
       "      <td>...</td>\n",
       "      <td>0.021077</td>\n",
       "      <td>0.054952</td>\n",
       "      <td>0.247094</td>\n",
       "      <td>0.365626</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.101908</td>\n",
       "      <td>0.338000</td>\n",
       "      <td>-0.429887</td>\n",
       "      <td>-0.578476</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26OV009</th>\n",
       "      <td>0.072513</td>\n",
       "      <td>0.594745</td>\n",
       "      <td>0.609137</td>\n",
       "      <td>0.584056</td>\n",
       "      <td>0.041583</td>\n",
       "      <td>-0.269243</td>\n",
       "      <td>-0.461148</td>\n",
       "      <td>-0.297156</td>\n",
       "      <td>-0.027213</td>\n",
       "      <td>0.596118</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.578555</td>\n",
       "      <td>0.533873</td>\n",
       "      <td>0.440283</td>\n",
       "      <td>0.738610</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.482615</td>\n",
       "      <td>-0.685922</td>\n",
       "      <td>-0.286718</td>\n",
       "      <td>-0.542423</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26OV011</th>\n",
       "      <td>-0.835139</td>\n",
       "      <td>-0.861639</td>\n",
       "      <td>-0.834907</td>\n",
       "      <td>-0.019997</td>\n",
       "      <td>-0.381831</td>\n",
       "      <td>-0.186141</td>\n",
       "      <td>0.373510</td>\n",
       "      <td>0.337066</td>\n",
       "      <td>-0.046594</td>\n",
       "      <td>0.236359</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.112791</td>\n",
       "      <td>-0.659868</td>\n",
       "      <td>0.076714</td>\n",
       "      <td>-0.229787</td>\n",
       "      <td>-0.320269</td>\n",
       "      <td>0.549351</td>\n",
       "      <td>0.014579</td>\n",
       "      <td>0.092864</td>\n",
       "      <td>-0.189991</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26OV013</th>\n",
       "      <td>-0.947106</td>\n",
       "      <td>-1.266903</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.248533</td>\n",
       "      <td>-0.186298</td>\n",
       "      <td>0.079681</td>\n",
       "      <td>0.682749</td>\n",
       "      <td>-0.144450</td>\n",
       "      <td>-0.205767</td>\n",
       "      <td>0.091320</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.536167</td>\n",
       "      <td>-0.111325</td>\n",
       "      <td>0.393229</td>\n",
       "      <td>-0.690196</td>\n",
       "      <td>0.166115</td>\n",
       "      <td>-0.079282</td>\n",
       "      <td>-0.216309</td>\n",
       "      <td>-0.262391</td>\n",
       "      <td>-0.021559</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66 rows × 10078 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        A1BG_proteomics  A2M_proteomics  A2ML1_proteomics  \\\n",
       "Patient_ID                                                      \n",
       "01OV007            0.133634        0.765120         -0.778849   \n",
       "01OV017           -0.432786       -0.461906         -0.533497   \n",
       "01OV018           -0.013943        0.065842               NaN   \n",
       "01OV023            1.234203        1.512082         -0.703050   \n",
       "01OV026           -0.482872        0.343872         -1.150936   \n",
       "...                     ...             ...               ...   \n",
       "26OV002            1.942785        2.028395         -0.506808   \n",
       "26OV008           -0.576189       -1.723434          2.023556   \n",
       "26OV009            0.072513        0.594745          0.609137   \n",
       "26OV011           -0.835139       -0.861639         -0.834907   \n",
       "26OV013           -0.947106       -1.266903               NaN   \n",
       "\n",
       "Name        AAAS_proteomics  AACS_proteomics  AAGAB_proteomics  \\\n",
       "Patient_ID                                                       \n",
       "01OV007            0.118236         0.112740          0.054942   \n",
       "01OV017            0.041800         0.604114          0.351294   \n",
       "01OV018           -0.010514         0.541935          0.169826   \n",
       "01OV023            0.181152         0.087719          0.226861   \n",
       "01OV026            0.315155         0.097235         -0.160208   \n",
       "...                     ...              ...               ...   \n",
       "26OV002           -0.237500        -0.767353         -0.094263   \n",
       "26OV008            0.066470         0.576987          0.725690   \n",
       "26OV009            0.584056         0.041583         -0.269243   \n",
       "26OV011           -0.019997        -0.381831         -0.186141   \n",
       "26OV013            0.248533        -0.186298          0.079681   \n",
       "\n",
       "Name        AAK1_proteomics  AAMDC_proteomics  AAMP_proteomics  \\\n",
       "Patient_ID                                                       \n",
       "01OV007            0.313055         -0.786270        -0.384680   \n",
       "01OV017            0.546605          0.603935        -0.235374   \n",
       "01OV018           -0.023032         -0.057549        -0.397627   \n",
       "01OV023           -0.100766         -0.516022        -0.456512   \n",
       "01OV026           -0.125165         -0.768776        -0.372368   \n",
       "...                     ...               ...              ...   \n",
       "26OV002            0.164654         -0.120120        -0.560682   \n",
       "26OV008            0.610382         -0.704039         0.121085   \n",
       "26OV009           -0.461148         -0.297156        -0.027213   \n",
       "26OV011            0.373510          0.337066        -0.046594   \n",
       "26OV013            0.682749         -0.144450        -0.205767   \n",
       "\n",
       "Name        AAR2_proteomics  ...  ZSCAN32_proteomics  ZSWIM8_proteomics  \\\n",
       "Patient_ID                   ...                                          \n",
       "01OV007            0.064017  ...           -0.323516           0.074209   \n",
       "01OV017            0.059084  ...            0.032004           0.016276   \n",
       "01OV018            0.199581  ...            0.002237           0.206993   \n",
       "01OV023           -0.257629  ...                 NaN          -0.529948   \n",
       "01OV026            0.450086  ...           -0.752819          -0.181027   \n",
       "...                     ...  ...                 ...                ...   \n",
       "26OV002            0.116122  ...            0.291665          -0.205511   \n",
       "26OV008            0.534117  ...            0.021077           0.054952   \n",
       "26OV009            0.596118  ...           -0.578555           0.533873   \n",
       "26OV011            0.236359  ...           -0.112791          -0.659868   \n",
       "26OV013            0.091320  ...           -0.536167          -0.111325   \n",
       "\n",
       "Name        ZW10_proteomics  ZWILCH_proteomics  ZWINT_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "01OV007            0.232999          -0.195793               NaN   \n",
       "01OV017            0.587911           0.574339               NaN   \n",
       "01OV018            0.111037           1.020321               NaN   \n",
       "01OV023            0.034713           0.149523         -0.313370   \n",
       "01OV026            0.179612           0.007854         -0.106049   \n",
       "...                     ...                ...               ...   \n",
       "26OV002            0.280838           1.143053          0.545983   \n",
       "26OV008            0.247094           0.365626               NaN   \n",
       "26OV009            0.440283           0.738610               NaN   \n",
       "26OV011            0.076714          -0.229787         -0.320269   \n",
       "26OV013            0.393229          -0.690196          0.166115   \n",
       "\n",
       "Name        ZYG11B_proteomics  ZYX_proteomics  ZZEF1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "01OV007             -0.145053        0.590217          0.019414   \n",
       "01OV017              0.075401       -0.437564         -0.208831   \n",
       "01OV018             -0.245864        0.008411          0.130504   \n",
       "01OV023             -0.507473        0.221818          0.278313   \n",
       "01OV026              0.377063       -0.295592          0.011997   \n",
       "...                       ...             ...               ...   \n",
       "26OV002              0.045314       -0.463719         -0.321074   \n",
       "26OV008              0.101908        0.338000         -0.429887   \n",
       "26OV009              0.482615       -0.685922         -0.286718   \n",
       "26OV011              0.549351        0.014579          0.092864   \n",
       "26OV013             -0.079282       -0.216309         -0.262391   \n",
       "\n",
       "Name        ZZZ3_proteomics        Mutation  \n",
       "Patient_ID                                   \n",
       "01OV007                 NaN  Wildtype_Tumor  \n",
       "01OV017                 NaN        Deletion  \n",
       "01OV018           -0.367433  Wildtype_Tumor  \n",
       "01OV023           -1.008998        Deletion  \n",
       "01OV026           -0.236003  Wildtype_Tumor  \n",
       "...                     ...             ...  \n",
       "26OV002            0.103304        Deletion  \n",
       "26OV008           -0.578476  Wildtype_Tumor  \n",
       "26OV009           -0.542423  Wildtype_Tumor  \n",
       "26OV011           -0.189991  Wildtype_Tumor  \n",
       "26OV013           -0.021559  Wildtype_Tumor  \n",
       "\n",
       "[66 rows x 10078 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = o.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = o.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = o.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "del_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "          Comparison       P_Value\n",
      "0  MMS19_proteomics  1.058914e-08\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MMS19_proteomics</td>\n",
       "      <td>1.058914e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RACK1_proteomics</td>\n",
       "      <td>8.573522e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PI4K2A_proteomics</td>\n",
       "      <td>9.310383e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WAPL_proteomics</td>\n",
       "      <td>1.724680e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IDE_proteomics</td>\n",
       "      <td>2.011475e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10072</th>\n",
       "      <td>SAXO2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10073</th>\n",
       "      <td>SH3GL3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10074</th>\n",
       "      <td>SLC22A3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10075</th>\n",
       "      <td>SSX2B_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10076</th>\n",
       "      <td>TFF2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10077 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics    Ov_P_Value\n",
       "0        MMS19_proteomics  1.058914e-08\n",
       "1        RACK1_proteomics  8.573522e-05\n",
       "2       PI4K2A_proteomics  9.310383e-05\n",
       "3         WAPL_proteomics  1.724680e-04\n",
       "4          IDE_proteomics  2.011475e-04\n",
       "...                   ...           ...\n",
       "10072    SAXO2_proteomics           NaN\n",
       "10073   SH3GL3_proteomics           NaN\n",
       "10074  SLC22A3_proteomics           NaN\n",
       "10075    SSX2B_proteomics           NaN\n",
       "10076     TFF2_proteomics           NaN\n",
       "\n",
       "[10077 rows x 2 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "o_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if o_sig is not None:\n",
    "    o_sig_list = list(o_sig.Comparison)\n",
    "else: \n",
    "    o_sig_list = None\n",
    "print('significant pvals: \\n',o_sig)\n",
    "\n",
    "# Get all pvals\n",
    "o = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "o_pval = o.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Ov_P_Value'})\n",
    "#o_pval = o_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "o_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "ov_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    ov_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "o_df = pd.DataFrame.from_dict(ov_d, orient='index', columns=['Ov_Median'])\n",
    "o_df = o_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#o_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.6560350245 - -0.534134515\n",
      "           Proteomics  Ov_P_Value\n",
      "9772  MAFF_proteomics    0.968044\n"
     ]
    }
   ],
   "source": [
    "#TESTS\n",
    "I = 'MAFF'\n",
    "o_df.loc[o_df['Proteomics'] == I+'_proteomics']\n",
    "\n",
    "print(wt_med[I+'_proteomics'], '-', del_med[I+'_proteomics'])\n",
    "\n",
    "print(o_pval.loc[o_pval['Proteomics'] == I+'_proteomics'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MMS19_proteomics</td>\n",
       "      <td>1.058914e-08</td>\n",
       "      <td>0.324897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RACK1_proteomics</td>\n",
       "      <td>8.573522e-05</td>\n",
       "      <td>0.191250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PI4K2A_proteomics</td>\n",
       "      <td>9.310383e-05</td>\n",
       "      <td>0.298399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WAPL_proteomics</td>\n",
       "      <td>1.724680e-04</td>\n",
       "      <td>0.218775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IDE_proteomics</td>\n",
       "      <td>2.011475e-04</td>\n",
       "      <td>0.208025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10072</th>\n",
       "      <td>SAXO2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.099993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10073</th>\n",
       "      <td>SH3GL3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.843461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10074</th>\n",
       "      <td>SLC22A3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.783325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10075</th>\n",
       "      <td>SSX2B_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10076</th>\n",
       "      <td>TFF2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10077 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics    Ov_P_Value  Ov_Median\n",
       "0        MMS19_proteomics  1.058914e-08   0.324897\n",
       "1        RACK1_proteomics  8.573522e-05   0.191250\n",
       "2       PI4K2A_proteomics  9.310383e-05   0.298399\n",
       "3         WAPL_proteomics  1.724680e-04   0.218775\n",
       "4          IDE_proteomics  2.011475e-04   0.208025\n",
       "...                   ...           ...        ...\n",
       "10072    SAXO2_proteomics           NaN   0.099993\n",
       "10073   SH3GL3_proteomics           NaN  -0.843461\n",
       "10074  SLC22A3_proteomics           NaN  -0.783325\n",
       "10075    SSX2B_proteomics           NaN        NaN\n",
       "10076     TFF2_proteomics           NaN        NaN\n",
       "\n",
       "[10077 rows x 3 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "o_merged = o_pval.merge(o_df, on='Proteomics',how='inner')\n",
    "o_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Breast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "#b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 10)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    80\n",
       "Deletion          24\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = b.get_proteomics()\n",
    "prot = b.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "p = list(prot.columns)\n",
    "len(p)\n",
    "\n",
    "mut_type = b.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = b.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = b.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "         Comparison   P_Value\n",
      "0  PTEN_proteomics  0.000004\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "b_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if b_sig is not None:\n",
    "    b_sig_list = list(b_sig.Comparison)\n",
    "else: \n",
    "    b_sig_list = None\n",
    "print('significant pvals: \\n',b_sig)\n",
    "\n",
    "# Get all pvals\n",
    "b_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "b_pval = b_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Brca_P_Value'})\n",
    "#b_pval = b_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#b_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Brca_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG_proteomics</td>\n",
       "      <td>1.00440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M_proteomics</td>\n",
       "      <td>0.80760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2ML1_proteomics</td>\n",
       "      <td>-1.31745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS_proteomics</td>\n",
       "      <td>-0.34660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AACS_proteomics</td>\n",
       "      <td>-0.73635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9766</th>\n",
       "      <td>ZXDC_proteomics</td>\n",
       "      <td>0.24385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9767</th>\n",
       "      <td>ZYG11B_proteomics</td>\n",
       "      <td>-0.11285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9768</th>\n",
       "      <td>ZYX_proteomics</td>\n",
       "      <td>-0.28070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9769</th>\n",
       "      <td>ZZEF1_proteomics</td>\n",
       "      <td>0.18565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9770</th>\n",
       "      <td>ZZZ3_proteomics</td>\n",
       "      <td>-0.03645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9771 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Brca_Median\n",
       "0       A1BG_proteomics      1.00440\n",
       "1        A2M_proteomics      0.80760\n",
       "2      A2ML1_proteomics     -1.31745\n",
       "3       AAAS_proteomics     -0.34660\n",
       "4       AACS_proteomics     -0.73635\n",
       "...                 ...          ...\n",
       "9766    ZXDC_proteomics      0.24385\n",
       "9767  ZYG11B_proteomics     -0.11285\n",
       "9768     ZYX_proteomics     -0.28070\n",
       "9769   ZZEF1_proteomics      0.18565\n",
       "9770    ZZZ3_proteomics     -0.03645\n",
       "\n",
       "[9771 rows x 2 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "b_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    b_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "b_df = pd.DataFrame.from_dict(b_d, orient='index', columns=['Brca_Median'])\n",
    "b_df = b_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "b_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.32399999999999995 - -0.016050000000000002\n",
      "            Proteomics  Brca_P_Value\n",
      "1119  DOCK1_proteomics      0.056169\n"
     ]
    }
   ],
   "source": [
    "#TESTS\n",
    "I = 'DOCK1'\n",
    "b_df.loc[b_df['Proteomics'] == I+'_proteomics']\n",
    "\n",
    "print(wt_med[I+'_proteomics'], '-', del_med[I+'_proteomics'])\n",
    "\n",
    "b_sig.iloc[350:]\n",
    "print(b_pval.loc[b_pval['Proteomics'] == I+'_proteomics'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>1.00800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EIF4H_proteomics</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>-0.41295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIEF1_proteomics</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>-0.84135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMSB10_proteomics</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>-0.87630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DRG1_proteomics</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>-0.75135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9766</th>\n",
       "      <td>POLR3E_proteomics</td>\n",
       "      <td>0.999356</td>\n",
       "      <td>0.09685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9767</th>\n",
       "      <td>IGFBP7_proteomics</td>\n",
       "      <td>0.999595</td>\n",
       "      <td>-0.01140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9768</th>\n",
       "      <td>AAMP_proteomics</td>\n",
       "      <td>0.999613</td>\n",
       "      <td>-0.15430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9769</th>\n",
       "      <td>TRA2B_proteomics</td>\n",
       "      <td>0.999771</td>\n",
       "      <td>0.17180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9770</th>\n",
       "      <td>KRIT1_proteomics</td>\n",
       "      <td>0.999848</td>\n",
       "      <td>-0.14610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9771 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Brca_P_Value  Brca_Median\n",
       "0       PTEN_proteomics      0.000004      1.00800\n",
       "1      EIF4H_proteomics      0.000025     -0.41295\n",
       "2      MIEF1_proteomics      0.000031     -0.84135\n",
       "3     TMSB10_proteomics      0.000043     -0.87630\n",
       "4       DRG1_proteomics      0.000065     -0.75135\n",
       "...                 ...           ...          ...\n",
       "9766  POLR3E_proteomics      0.999356      0.09685\n",
       "9767  IGFBP7_proteomics      0.999595     -0.01140\n",
       "9768    AAMP_proteomics      0.999613     -0.15430\n",
       "9769   TRA2B_proteomics      0.999771      0.17180\n",
       "9770   KRIT1_proteomics      0.999848     -0.14610\n",
       "\n",
       "[9771 rows x 3 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_merged = b_pval.merge(b_df, on='Proteomics',how='inner')\n",
    "b_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "#col = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    76\n",
       "Deletion          25\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = col.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "len(p)\n",
    "\n",
    "mut_type = col.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = col.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      " None\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "c_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if c_sig is not None:\n",
    "    c_sig_list = list(c_sig.Comparison)\n",
    "else: \n",
    "    c_sig_list = None\n",
    "print('significant pvals: \\n',c_sig)\n",
    "\n",
    "# Get all pvals\n",
    "c_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "c_pval = c_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Colon_P_Value'})\n",
    "#c_pval = c_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#c_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "c_df = pd.DataFrame.from_dict(d, orient='index', columns=['Colon_Median'])\n",
    "c_df = c_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#c_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.19399999999999998 - -0.1295\n",
      "           Proteomics  Colon_P_Value\n",
      "1757  MAFF_proteomics       0.200121\n"
     ]
    }
   ],
   "source": [
    "#TESTS\n",
    "I = 'MAFF'\n",
    "c_df.loc[c_df['Proteomics'] == I+'_proteomics']\n",
    "\n",
    "print(wt_med[I+'_proteomics'], '-', del_med[I+'_proteomics'])\n",
    "\n",
    "print(c_pval.loc[c_pval['Proteomics'] == I+'_proteomics'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DFFA_proteomics</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.19930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WAPL_proteomics</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.19605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SEC14L2_proteomics</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>0.66800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GBF1_proteomics</td>\n",
       "      <td>0.000164</td>\n",
       "      <td>0.12630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>STK11IP_proteomics</td>\n",
       "      <td>0.000239</td>\n",
       "      <td>0.31250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8062</th>\n",
       "      <td>ZNF841_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.78250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8063</th>\n",
       "      <td>ZNF888_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8064</th>\n",
       "      <td>ZNHIT6_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8065</th>\n",
       "      <td>ZNRD1_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.06800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8066</th>\n",
       "      <td>ZWINT_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.18535</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8067 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Proteomics  Colon_P_Value  Colon_Median\n",
       "0        DFFA_proteomics       0.000037       0.19930\n",
       "1        WAPL_proteomics       0.000097       0.19605\n",
       "2     SEC14L2_proteomics       0.000136       0.66800\n",
       "3        GBF1_proteomics       0.000164       0.12630\n",
       "4     STK11IP_proteomics       0.000239       0.31250\n",
       "...                  ...            ...           ...\n",
       "8062   ZNF841_proteomics            NaN      -0.78250\n",
       "8063   ZNF888_proteomics            NaN           NaN\n",
       "8064   ZNHIT6_proteomics            NaN           NaN\n",
       "8065    ZNRD1_proteomics            NaN       0.06800\n",
       "8066    ZWINT_proteomics            NaN      -0.18535\n",
       "\n",
       "[8067 rows x 3 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c_merged = c_pval.merge(c_df, on='Proteomics',how='inner')\n",
    "c_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Head and Neck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "#h = cptac.Hnscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 176 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    87\n",
       "Deletion          18\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = h.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "len(p)\n",
    "\n",
    "mut_type = h.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = h.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "#prot_and_mutations = h.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "              Comparison       P_Value\n",
      "0        NLN_proteomics  1.113430e-09\n",
      "1     AKR1C3_proteomics  1.032977e-08\n",
      "2     AKR1C1_proteomics  6.931627e-08\n",
      "3       ADI1_proteomics  1.601880e-07\n",
      "4      EPHX1_proteomics  3.899394e-07\n",
      "..                  ...           ...\n",
      "523    FLAD1_proteomics  2.232897e-03\n",
      "524    GSTM4_proteomics  2.234528e-03\n",
      "525   ELOVL5_proteomics  2.239241e-03\n",
      "526  SUPT16H_proteomics  2.241138e-03\n",
      "527  ANKRD28_proteomics  2.243254e-03\n",
      "\n",
      "[528 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "h_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if h_sig is not None:\n",
    "    h_sig_list = list(h_sig.Comparison)\n",
    "else: \n",
    "    h_sig_list = None\n",
    "print('significant pvals: \\n',h_sig)\n",
    "\n",
    "# Get all pvals\n",
    "\n",
    "h_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "h_pval = h_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Hnscc_P_Value'})\n",
    "#h_pval = h_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#h_pval\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "h_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    h_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "h_df = pd.DataFrame.from_dict(h_d, orient='index', columns=['Hnscc_Median'])\n",
    "h_df = h_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#h_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25.186277574373303 - 25.0734306518224\n",
      "            Proteomics  Hnscc_P_Value\n",
      "5122  DOCK1_proteomics       0.259882\n"
     ]
    }
   ],
   "source": [
    "#TESTS\n",
    "I = 'DOCK1'\n",
    "h_df.loc[h_df['Proteomics'] == I+'_proteomics']\n",
    "\n",
    "print(wt_med[I+'_proteomics'], '-', del_med[I+'_proteomics'])\n",
    "\n",
    "print(h_pval.loc[h_pval['Proteomics'] == I+'_proteomics'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NLN_proteomics</td>\n",
       "      <td>1.113430e-09</td>\n",
       "      <td>-0.431666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKR1C3_proteomics</td>\n",
       "      <td>1.032977e-08</td>\n",
       "      <td>-0.951905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKR1C1_proteomics</td>\n",
       "      <td>6.931627e-08</td>\n",
       "      <td>-1.961262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ADI1_proteomics</td>\n",
       "      <td>1.601880e-07</td>\n",
       "      <td>-0.513198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EPHX1_proteomics</td>\n",
       "      <td>3.899394e-07</td>\n",
       "      <td>-0.871688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11739</th>\n",
       "      <td>ZNF664_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11740</th>\n",
       "      <td>ZNF836_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11741</th>\n",
       "      <td>ZNF862_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11742</th>\n",
       "      <td>ZSCAN31_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11743</th>\n",
       "      <td>ZXDA_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11744 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics  Hnscc_P_Value  Hnscc_Median\n",
       "0          NLN_proteomics   1.113430e-09     -0.431666\n",
       "1       AKR1C3_proteomics   1.032977e-08     -0.951905\n",
       "2       AKR1C1_proteomics   6.931627e-08     -1.961262\n",
       "3         ADI1_proteomics   1.601880e-07     -0.513198\n",
       "4        EPHX1_proteomics   3.899394e-07     -0.871688\n",
       "...                   ...            ...           ...\n",
       "11739   ZNF664_proteomics            NaN           NaN\n",
       "11740   ZNF836_proteomics            NaN           NaN\n",
       "11741   ZNF862_proteomics            NaN           NaN\n",
       "11742  ZSCAN31_proteomics            NaN           NaN\n",
       "11743     ZXDA_proteomics            NaN           NaN\n",
       "\n",
       "[11744 rows x 3 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h_merged = h_pval.merge(h_df, on='Proteomics',how='inner')\n",
    "#del_wt[['PREX2_proteomics','Mutation']].dropna() # in median, not in pval (only 1 value for deletion)\n",
    "h_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Lung"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "#l = cptac.Luad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n",
      "cptac warning: You tried to flatten an index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-00001.N, C3L-00009.N, C3L-00080.N, C3L-00083.N, C3L-00093.N, C3L-00094.N, C3L-00095.N, C3L-00140.N, C3L-00144.N, C3L-00263.N, C3L-00279.N, C3L-00368.N, C3L-00412.N, C3L-00422.N, C3L-00510.N, C3L-00604.N, C3L-00893.N, C3L-00913.N, C3L-00973.N, C3L-01330.N, C3L-01632.N, C3L-01682.N, C3L-01683.N, C3L-01889.N, C3L-01890, C3L-01890.N, C3L-01924.N, C3L-02219.N, C3L-02345.N, C3L-02348.N, C3L-02350.N, C3L-02365.N, C3L-02508.N, C3L-02549.N, C3N-00167.N, C3N-00169.N, C3N-00175.N, C3N-00180.N, C3N-00199.N, C3N-00203.N, C3N-00217.N, C3N-00223.N, C3N-00293.N, C3N-00433.N, C3N-00546.N, C3N-00547.N, C3N-00549.N, C3N-00550.N, C3N-00551.N, C3N-00552.N, C3N-00556.N, C3N-00559.N, C3N-00560.N, C3N-00572.N, C3N-00574.N, C3N-00578.N, C3N-00579.N, C3N-00580.N, C3N-00704.N, C3N-00737.N, C3N-00738.N, C3N-00959.N, C3N-01016.N, C3N-01021.N, C3N-01023.N, C3N-01024.N, C3N-01030.N, C3N-01071.N, C3N-01072.N, C3N-01405.N, C3N-01410.N, C3N-01413.N, C3N-01414.N, C3N-01415.N, C3N-01416.N, C3N-01488.N, C3N-01489.N, C3N-01799.N, C3N-01823.N, C3N-02000.N, C3N-02002.N, C3N-02003.N, C3N-02067.N, C3N-02087.N, C3N-02089.N, C3N-02145.N, C3N-02149.N, C3N-02155.N, C3N-02158.N, C3N-02379.N, C3N-02380.N, C3N-02421.N, C3N-02423.N, C3N-02424.N, C3N-02433.N, C3N-02529.N, C3N-02572.N, C3N-02582.N, C3N-02586.N, C3N-02587.N, C3N-02588.N, C3N-02729.N (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No_Mutation    88\n",
      "Deletion       17\n",
      "Name: Mutation, dtype: int64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 12)\n",
      "cptac warning: You tried to flatten an index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 12)\n"
     ]
    }
   ],
   "source": [
    "prot = l.get_proteomics()\n",
    "prot = l.reduce_multiindex(prot, levels_to_drop = 1, flatten = True)\n",
    "p = list(prot.columns)\n",
    "len(p)\n",
    "\n",
    "mut_type = l.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.rename(columns={'PTEN':'cnv'})\n",
    "\n",
    "# different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "omics = l.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics',genes1=\"PTEN\", \n",
    "    genes2=p)\n",
    "omics = l.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "omics = omics.drop(columns='PTEN_CNV')\n",
    "\n",
    "# Get only tumor samples\n",
    "p = l.get_proteomics(tissue_type='tumor')\n",
    "tumor_ids = list(p.index)\n",
    "get = omics.index.isin(tumor_ids)\n",
    "omics = omics[get]\n",
    "\n",
    "merged = omics.join(mut_type) # checked and there is 110 tumor samples for lung\n",
    "\n",
    "compare = ['No_Mutation','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "print(del_wt['Mutation'].value_counts())\n",
    "del_wt = del_wt.dropna(axis='columns', how='all')\n",
    "del_wt = del_wt.loc[:,~del_wt.columns.duplicated()] # SH1, PRK3R1 both have double columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "               Comparison       P_Value\n",
      "0        ATE1_proteomics  2.851800e-07\n",
      "1       SHOC2_proteomics  6.424816e-07\n",
      "2        EGFR_proteomics  9.121167e-07\n",
      "3      LANCL2_proteomics  1.742749e-06\n",
      "4        GGCT_proteomics  2.854181e-06\n",
      "..                   ...           ...\n",
      "103    FERMT3_proteomics  4.804371e-04\n",
      "104  ARHGAP15_proteomics  4.875693e-04\n",
      "105    CRYBG2_proteomics  5.063478e-04\n",
      "106     GNAI2_proteomics  5.076782e-04\n",
      "107     CIAO1_proteomics  5.181887e-04\n",
      "\n",
      "[108 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATE1_proteomics</td>\n",
       "      <td>2.851800e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHOC2_proteomics</td>\n",
       "      <td>6.424816e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>9.121167e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LANCL2_proteomics</td>\n",
       "      <td>1.742749e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GGCT_proteomics</td>\n",
       "      <td>2.854181e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10311</th>\n",
       "      <td>COG6_proteomics</td>\n",
       "      <td>9.997105e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10312</th>\n",
       "      <td>PLD1_proteomics</td>\n",
       "      <td>9.997256e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10313</th>\n",
       "      <td>CHMP6_proteomics</td>\n",
       "      <td>9.997659e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10314</th>\n",
       "      <td>FHL3_proteomics</td>\n",
       "      <td>9.999702e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10315</th>\n",
       "      <td>UST_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10316 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Proteomics  Luad_P_Value\n",
       "0        ATE1_proteomics  2.851800e-07\n",
       "1       SHOC2_proteomics  6.424816e-07\n",
       "2        EGFR_proteomics  9.121167e-07\n",
       "3      LANCL2_proteomics  1.742749e-06\n",
       "4        GGCT_proteomics  2.854181e-06\n",
       "...                  ...           ...\n",
       "10311    COG6_proteomics  9.997105e-01\n",
       "10312    PLD1_proteomics  9.997256e-01\n",
       "10313   CHMP6_proteomics  9.997659e-01\n",
       "10314    FHL3_proteomics  9.999702e-01\n",
       "10315     UST_proteomics           NaN\n",
       "\n",
       "[10316 rows x 2 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-2])\n",
    "\n",
    "# Get only sig sites\n",
    "l_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if l_sig is not None:\n",
    "    l_sig_list = list(l_sig.Comparison)\n",
    "else: \n",
    "    l_sig_list = None\n",
    "print('significant pvals: \\n',l_sig)\n",
    "\n",
    "# Get all pvals\n",
    "l_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "l_pval = l_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Luad_P_Value'})\n",
    "#l_pval = l_pval.replace(to_replace = r'_NP_.*$', value = '', regex = True)\n",
    "l_pval # isoforms for some proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Isoform code - fix later\n",
    "#del_wt.replace(to_replace = r'_NP_.*$', value = '', regex = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG_proteomics</td>\n",
       "      <td>0.11030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M_proteomics</td>\n",
       "      <td>0.39840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS_proteomics</td>\n",
       "      <td>0.02265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AACS_proteomics</td>\n",
       "      <td>0.03800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC_proteomics</td>\n",
       "      <td>0.94550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10311</th>\n",
       "      <td>ZXDC_proteomics</td>\n",
       "      <td>-0.20540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10312</th>\n",
       "      <td>ZYG11B_proteomics</td>\n",
       "      <td>-0.45530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10313</th>\n",
       "      <td>ZYX_proteomics</td>\n",
       "      <td>0.00720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10314</th>\n",
       "      <td>ZZEF1_proteomics</td>\n",
       "      <td>0.13655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10315</th>\n",
       "      <td>ZZZ3_proteomics</td>\n",
       "      <td>0.24710</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10316 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Proteomics  Luad_Median\n",
       "0        A1BG_proteomics      0.11030\n",
       "1         A2M_proteomics      0.39840\n",
       "2        AAAS_proteomics      0.02265\n",
       "3        AACS_proteomics      0.03800\n",
       "4       AADAC_proteomics      0.94550\n",
       "...                  ...          ...\n",
       "10311    ZXDC_proteomics     -0.20540\n",
       "10312  ZYG11B_proteomics     -0.45530\n",
       "10313     ZYX_proteomics      0.00720\n",
       "10314   ZZEF1_proteomics      0.13655\n",
       "10315    ZZZ3_proteomics      0.24710\n",
       "\n",
       "[10316 rows x 2 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"No_Mutation\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "l_d = {}\n",
    "\n",
    "for prot in cols:\n",
    "    dif = wt_med[prot] - del_med[prot]\n",
    "    l_d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "l_df = pd.DataFrame.from_dict(l_d, orient='index', columns=['Luad_Median'])\n",
    "l_df = l_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "l_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATE1_proteomics</td>\n",
       "      <td>2.851800e-07</td>\n",
       "      <td>0.88365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHOC2_proteomics</td>\n",
       "      <td>6.424816e-07</td>\n",
       "      <td>0.60600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>9.121167e-07</td>\n",
       "      <td>-2.32970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LANCL2_proteomics</td>\n",
       "      <td>1.742749e-06</td>\n",
       "      <td>-1.21475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GGCT_proteomics</td>\n",
       "      <td>2.854181e-06</td>\n",
       "      <td>-2.12715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10311</th>\n",
       "      <td>COG6_proteomics</td>\n",
       "      <td>9.997105e-01</td>\n",
       "      <td>0.13015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10312</th>\n",
       "      <td>PLD1_proteomics</td>\n",
       "      <td>9.997256e-01</td>\n",
       "      <td>-0.43875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10313</th>\n",
       "      <td>CHMP6_proteomics</td>\n",
       "      <td>9.997659e-01</td>\n",
       "      <td>0.05220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10314</th>\n",
       "      <td>FHL3_proteomics</td>\n",
       "      <td>9.999702e-01</td>\n",
       "      <td>0.06610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10315</th>\n",
       "      <td>UST_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.49145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10316 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Proteomics  Luad_P_Value  Luad_Median\n",
       "0        ATE1_proteomics  2.851800e-07      0.88365\n",
       "1       SHOC2_proteomics  6.424816e-07      0.60600\n",
       "2        EGFR_proteomics  9.121167e-07     -2.32970\n",
       "3      LANCL2_proteomics  1.742749e-06     -1.21475\n",
       "4        GGCT_proteomics  2.854181e-06     -2.12715\n",
       "...                  ...           ...          ...\n",
       "10311    COG6_proteomics  9.997105e-01      0.13015\n",
       "10312    PLD1_proteomics  9.997256e-01     -0.43875\n",
       "10313   CHMP6_proteomics  9.997659e-01      0.05220\n",
       "10314    FHL3_proteomics  9.999702e-01      0.06610\n",
       "10315     UST_proteomics           NaN     -2.49145\n",
       "\n",
       "[10316 rows x 3 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l_merged = l_pval.merge(l_df, on='Proteomics',how='inner')\n",
    "#l_merged = l_merged.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "l_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.07719999999999999 - 0.3594\n",
      "           Proteomics  Luad_P_Value\n",
      "3257  MAFF_proteomics      0.187607\n"
     ]
    }
   ],
   "source": [
    "#TESTS\n",
    "I = 'MAFF'\n",
    "l_df.loc[l_df['Proteomics'] == I+'_proteomics']\n",
    "\n",
    "print(wt_med[I+'_proteomics'], '-', del_med[I+'_proteomics'])\n",
    "\n",
    "print(l_pval.loc[l_pval['Proteomics'] == I+'_proteomics'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3_proteomics</td>\n",
       "      <td>5.396032e-11</td>\n",
       "      <td>0.405134</td>\n",
       "      <td>0.000558</td>\n",
       "      <td>0.135941</td>\n",
       "      <td>0.017908</td>\n",
       "      <td>0.37845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001113</td>\n",
       "      <td>-0.1503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC_proteomics</td>\n",
       "      <td>1.593480e-10</td>\n",
       "      <td>0.553255</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.314307</td>\n",
       "      <td>0.000744</td>\n",
       "      <td>1.01995</td>\n",
       "      <td>0.008295</td>\n",
       "      <td>0.48080</td>\n",
       "      <td>0.060164</td>\n",
       "      <td>0.159424</td>\n",
       "      <td>0.082868</td>\n",
       "      <td>0.2533</td>\n",
       "      <td>0.411506</td>\n",
       "      <td>0.08210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A_proteomics</td>\n",
       "      <td>1.009419e-09</td>\n",
       "      <td>0.838882</td>\n",
       "      <td>0.001544</td>\n",
       "      <td>0.224713</td>\n",
       "      <td>0.560996</td>\n",
       "      <td>-0.04535</td>\n",
       "      <td>0.203952</td>\n",
       "      <td>0.07435</td>\n",
       "      <td>0.389187</td>\n",
       "      <td>-0.052955</td>\n",
       "      <td>0.201771</td>\n",
       "      <td>-0.1690</td>\n",
       "      <td>0.199847</td>\n",
       "      <td>-0.10135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2_proteomics</td>\n",
       "      <td>1.122076e-09</td>\n",
       "      <td>0.586396</td>\n",
       "      <td>0.400554</td>\n",
       "      <td>-0.017886</td>\n",
       "      <td>0.612583</td>\n",
       "      <td>-0.05695</td>\n",
       "      <td>0.198043</td>\n",
       "      <td>-0.17195</td>\n",
       "      <td>0.786184</td>\n",
       "      <td>-0.033091</td>\n",
       "      <td>0.864381</td>\n",
       "      <td>-0.0232</td>\n",
       "      <td>0.349065</td>\n",
       "      <td>0.07690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2_proteomics</td>\n",
       "      <td>1.302273e-09</td>\n",
       "      <td>0.610188</td>\n",
       "      <td>0.630985</td>\n",
       "      <td>-0.001559</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.354712</td>\n",
       "      <td>-0.14995</td>\n",
       "      <td>0.569417</td>\n",
       "      <td>0.067389</td>\n",
       "      <td>0.061660</td>\n",
       "      <td>-0.1286</td>\n",
       "      <td>0.990512</td>\n",
       "      <td>0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14360</th>\n",
       "      <td>WTH3DI_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.40100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14361</th>\n",
       "      <td>ZIC4_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14362</th>\n",
       "      <td>ZNF419_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.38150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14363</th>\n",
       "      <td>ZNF705E_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.12900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14364</th>\n",
       "      <td>ZNF841_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.78250</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14365 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics   Gbm_P_Value  Gbm_Median  Hnscc_P_Value  \\\n",
       "0        ARMH3_proteomics  5.396032e-11    0.405134       0.000558   \n",
       "1         CUTC_proteomics  1.593480e-10    0.553255       0.000189   \n",
       "2      PIP4K2A_proteomics  1.009419e-09    0.838882       0.001544   \n",
       "3         CUL2_proteomics  1.122076e-09    0.586396       0.400554   \n",
       "4         GDI2_proteomics  1.302273e-09    0.610188       0.630985   \n",
       "...                   ...           ...         ...            ...   \n",
       "14360   WTH3DI_proteomics           NaN         NaN            NaN   \n",
       "14361     ZIC4_proteomics           NaN         NaN            NaN   \n",
       "14362   ZNF419_proteomics           NaN         NaN            NaN   \n",
       "14363  ZNF705E_proteomics           NaN         NaN            NaN   \n",
       "14364   ZNF841_proteomics           NaN         NaN            NaN   \n",
       "\n",
       "       Hnscc_Median  Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median  \\\n",
       "0          0.135941      0.017908      0.37845           NaN          NaN   \n",
       "1          0.314307      0.000744      1.01995      0.008295      0.48080   \n",
       "2          0.224713      0.560996     -0.04535      0.203952      0.07435   \n",
       "3         -0.017886      0.612583     -0.05695      0.198043     -0.17195   \n",
       "4         -0.001559           NaN          NaN      0.354712     -0.14995   \n",
       "...             ...           ...          ...           ...          ...   \n",
       "14360           NaN           NaN          NaN           NaN          NaN   \n",
       "14361           NaN           NaN          NaN           NaN          NaN   \n",
       "14362           NaN           NaN          NaN           NaN          NaN   \n",
       "14363           NaN           NaN          NaN           NaN          NaN   \n",
       "14364           NaN           NaN          NaN           NaN          NaN   \n",
       "\n",
       "       Ov_P_Value  Ov_Median  En_P_Value  En_Median  Colon_P_Value  \\\n",
       "0             NaN        NaN    0.001113    -0.1503            NaN   \n",
       "1        0.060164   0.159424    0.082868     0.2533       0.411506   \n",
       "2        0.389187  -0.052955    0.201771    -0.1690       0.199847   \n",
       "3        0.786184  -0.033091    0.864381    -0.0232       0.349065   \n",
       "4        0.569417   0.067389    0.061660    -0.1286       0.990512   \n",
       "...           ...        ...         ...        ...            ...   \n",
       "14360         NaN        NaN         NaN        NaN            NaN   \n",
       "14361         NaN        NaN         NaN        NaN            NaN   \n",
       "14362         NaN        NaN         NaN        NaN            NaN   \n",
       "14363         NaN        NaN         NaN        NaN            NaN   \n",
       "14364         NaN        NaN         NaN        NaN            NaN   \n",
       "\n",
       "       Colon_Median  \n",
       "0               NaN  \n",
       "1           0.08210  \n",
       "2          -0.10135  \n",
       "3           0.07690  \n",
       "4           0.04450  \n",
       "...             ...  \n",
       "14360       0.40100  \n",
       "14361      -0.00200  \n",
       "14362      -0.38150  \n",
       "14363       0.12900  \n",
       "14364      -0.78250  \n",
       "\n",
       "[14365 rows x 15 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n = g_merged.merge(h_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(l_merged, on='Proteomics',how='outer')\n",
    "n= n.merge(b_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(o_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(e_merged, on='Proteomics',how='outer')\n",
    "all_df = n.merge(c_merged, on='Proteomics',how='outer')\n",
    "all_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2584\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3_proteomics</td>\n",
       "      <td>5.396032e-11</td>\n",
       "      <td>0.405134</td>\n",
       "      <td>0.000558</td>\n",
       "      <td>0.135941</td>\n",
       "      <td>0.017908</td>\n",
       "      <td>0.37845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001113</td>\n",
       "      <td>-0.1503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC_proteomics</td>\n",
       "      <td>1.593480e-10</td>\n",
       "      <td>0.553255</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.314307</td>\n",
       "      <td>0.000744</td>\n",
       "      <td>1.01995</td>\n",
       "      <td>0.008295</td>\n",
       "      <td>0.48080</td>\n",
       "      <td>0.060164</td>\n",
       "      <td>0.159424</td>\n",
       "      <td>0.082868</td>\n",
       "      <td>0.2533</td>\n",
       "      <td>0.411506</td>\n",
       "      <td>0.08210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A_proteomics</td>\n",
       "      <td>1.009419e-09</td>\n",
       "      <td>0.838882</td>\n",
       "      <td>0.001544</td>\n",
       "      <td>0.224713</td>\n",
       "      <td>0.560996</td>\n",
       "      <td>-0.04535</td>\n",
       "      <td>0.203952</td>\n",
       "      <td>0.07435</td>\n",
       "      <td>0.389187</td>\n",
       "      <td>-0.052955</td>\n",
       "      <td>0.201771</td>\n",
       "      <td>-0.1690</td>\n",
       "      <td>0.199847</td>\n",
       "      <td>-0.10135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2_proteomics</td>\n",
       "      <td>1.122076e-09</td>\n",
       "      <td>0.586396</td>\n",
       "      <td>0.400554</td>\n",
       "      <td>-0.017886</td>\n",
       "      <td>0.612583</td>\n",
       "      <td>-0.05695</td>\n",
       "      <td>0.198043</td>\n",
       "      <td>-0.17195</td>\n",
       "      <td>0.786184</td>\n",
       "      <td>-0.033091</td>\n",
       "      <td>0.864381</td>\n",
       "      <td>-0.0232</td>\n",
       "      <td>0.349065</td>\n",
       "      <td>0.07690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2_proteomics</td>\n",
       "      <td>1.302273e-09</td>\n",
       "      <td>0.610188</td>\n",
       "      <td>0.630985</td>\n",
       "      <td>-0.001559</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.354712</td>\n",
       "      <td>-0.14995</td>\n",
       "      <td>0.569417</td>\n",
       "      <td>0.067389</td>\n",
       "      <td>0.061660</td>\n",
       "      <td>-0.1286</td>\n",
       "      <td>0.990512</td>\n",
       "      <td>0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12981</th>\n",
       "      <td>SUV39H1_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.169685</td>\n",
       "      <td>-0.37210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000411</td>\n",
       "      <td>0.7950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13236</th>\n",
       "      <td>RSPH3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.336275</td>\n",
       "      <td>0.81260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001789</td>\n",
       "      <td>-1.5200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13251</th>\n",
       "      <td>ARMC3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.367448</td>\n",
       "      <td>0.57430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000933</td>\n",
       "      <td>-1.0990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13275</th>\n",
       "      <td>CCDC57_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.420048</td>\n",
       "      <td>-0.11070</td>\n",
       "      <td>0.621414</td>\n",
       "      <td>0.10775</td>\n",
       "      <td>0.819617</td>\n",
       "      <td>0.087283</td>\n",
       "      <td>0.001480</td>\n",
       "      <td>-0.8445</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13497</th>\n",
       "      <td>ARMC4_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.833684</td>\n",
       "      <td>0.16440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.890458</td>\n",
       "      <td>0.106026</td>\n",
       "      <td>0.000891</td>\n",
       "      <td>-0.8960</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2584 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics   Gbm_P_Value  Gbm_Median  Hnscc_P_Value  \\\n",
       "0        ARMH3_proteomics  5.396032e-11    0.405134       0.000558   \n",
       "1         CUTC_proteomics  1.593480e-10    0.553255       0.000189   \n",
       "2      PIP4K2A_proteomics  1.009419e-09    0.838882       0.001544   \n",
       "3         CUL2_proteomics  1.122076e-09    0.586396       0.400554   \n",
       "4         GDI2_proteomics  1.302273e-09    0.610188       0.630985   \n",
       "...                   ...           ...         ...            ...   \n",
       "12981  SUV39H1_proteomics           NaN         NaN            NaN   \n",
       "13236    RSPH3_proteomics           NaN         NaN            NaN   \n",
       "13251    ARMC3_proteomics           NaN         NaN            NaN   \n",
       "13275   CCDC57_proteomics           NaN         NaN            NaN   \n",
       "13497    ARMC4_proteomics           NaN         NaN            NaN   \n",
       "\n",
       "       Hnscc_Median  Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median  \\\n",
       "0          0.135941      0.017908      0.37845           NaN          NaN   \n",
       "1          0.314307      0.000744      1.01995      0.008295      0.48080   \n",
       "2          0.224713      0.560996     -0.04535      0.203952      0.07435   \n",
       "3         -0.017886      0.612583     -0.05695      0.198043     -0.17195   \n",
       "4         -0.001559           NaN          NaN      0.354712     -0.14995   \n",
       "...             ...           ...          ...           ...          ...   \n",
       "12981           NaN           NaN          NaN      0.169685     -0.37210   \n",
       "13236           NaN      0.336275      0.81260           NaN          NaN   \n",
       "13251           NaN      0.367448      0.57430           NaN          NaN   \n",
       "13275           NaN      0.420048     -0.11070      0.621414      0.10775   \n",
       "13497           NaN      0.833684      0.16440           NaN          NaN   \n",
       "\n",
       "       Ov_P_Value  Ov_Median  En_P_Value  En_Median  Colon_P_Value  \\\n",
       "0             NaN        NaN    0.001113    -0.1503            NaN   \n",
       "1        0.060164   0.159424    0.082868     0.2533       0.411506   \n",
       "2        0.389187  -0.052955    0.201771    -0.1690       0.199847   \n",
       "3        0.786184  -0.033091    0.864381    -0.0232       0.349065   \n",
       "4        0.569417   0.067389    0.061660    -0.1286       0.990512   \n",
       "...           ...        ...         ...        ...            ...   \n",
       "12981         NaN        NaN    0.000411     0.7950            NaN   \n",
       "13236         NaN        NaN    0.001789    -1.5200            NaN   \n",
       "13251         NaN        NaN    0.000933    -1.0990            NaN   \n",
       "13275    0.819617   0.087283    0.001480    -0.8445            NaN   \n",
       "13497    0.890458   0.106026    0.000891    -0.8960            NaN   \n",
       "\n",
       "       Colon_Median  \n",
       "0               NaN  \n",
       "1           0.08210  \n",
       "2          -0.10135  \n",
       "3           0.07690  \n",
       "4           0.04450  \n",
       "...             ...  \n",
       "12981           NaN  \n",
       "13236           NaN  \n",
       "13251           NaN  \n",
       "13275           NaN  \n",
       "13497           NaN  \n",
       "\n",
       "[2584 rows x 15 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Keep only genes significant in > 1 cancer\n",
    "all_sig = []\n",
    "\n",
    "sig_lists = [g_sig_list, e_sig_list, b_sig_list, o_sig_list, c_sig_list, h_sig_list, l_sig_list]\n",
    "\n",
    "for next_list in sig_lists:\n",
    "    if next_list is not None:\n",
    "        all_sig.append(next_list)\n",
    "        \n",
    "\n",
    "flat_list = [item for sublist in all_sig for item in sublist]\n",
    "sig = set(flat_list) # remove duplicates\n",
    "\n",
    "bool_df = all_df['Proteomics'].isin(sig)\n",
    "sig_df = all_df[bool_df]\n",
    "print(len(sig_df))\n",
    "sig_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_df = sig_df.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "sig_df = sig_df.set_index('Proteomics')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteomics</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MAFF</th>\n",
       "      <td>0.008464</td>\n",
       "      <td>-0.202893</td>\n",
       "      <td>0.673107</td>\n",
       "      <td>0.060547</td>\n",
       "      <td>0.187607</td>\n",
       "      <td>-0.2822</td>\n",
       "      <td>0.078518</td>\n",
       "      <td>-0.4992</td>\n",
       "      <td>0.968044</td>\n",
       "      <td>-0.121901</td>\n",
       "      <td>0.741379</td>\n",
       "      <td>0.0704</td>\n",
       "      <td>0.200121</td>\n",
       "      <td>0.3235</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gbm_P_Value  Gbm_Median  Hnscc_P_Value  Hnscc_Median  \\\n",
       "Proteomics                                                         \n",
       "MAFF           0.008464   -0.202893       0.673107      0.060547   \n",
       "\n",
       "            Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median  Ov_P_Value  \\\n",
       "Proteomics                                                                     \n",
       "MAFF            0.187607      -0.2822      0.078518      -0.4992    0.968044   \n",
       "\n",
       "            Ov_Median  En_P_Value  En_Median  Colon_P_Value  Colon_Median  \n",
       "Proteomics                                                                 \n",
       "MAFF        -0.121901    0.741379     0.0704       0.200121        0.3235  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check sig_df\n",
    "\n",
    "sig_df.loc[sig_df.index == 'MAFF']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_df.to_csv('sig_pval_all_proteins.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gbm :\n",
      "1887 \n",
      "\n",
      "En :\n",
      "402 \n",
      "\n",
      "Brca :\n",
      "1 \n",
      "\n",
      "Ov :\n",
      "1 \n",
      "\n",
      "Colon :\n",
      "Hnscc :\n",
      "528 \n",
      "\n",
      "Luad :\n",
      "108 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# see sig in cancer\n",
    "cancer = ['Gbm','En','Brca','Ov','Colon','Hnscc','Luad']\n",
    "i = 0\n",
    "for next_list in sig_lists:\n",
    "\n",
    "    print(cancer[i], ':')\n",
    "    if next_list is not None: \n",
    "        print(len(next_list),'\\n')\n",
    "    #print(next_list, '\\n')\n",
    "    if (i < 6):\n",
    "        i += 1\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HeatMap df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteomics</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARMH3</th>\n",
       "      <td>5.396032e-11</td>\n",
       "      <td>0.405134</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>2.364277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUTC</th>\n",
       "      <td>1.593480e-10</td>\n",
       "      <td>0.553255</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>2.255993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIP4K2A</th>\n",
       "      <td>1.009419e-09</td>\n",
       "      <td>0.838882</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>2.071389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUL2</th>\n",
       "      <td>1.122076e-09</td>\n",
       "      <td>0.586396</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>2.060808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GDI2</th>\n",
       "      <td>1.302273e-09</td>\n",
       "      <td>0.610188</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>2.045916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WWP1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.120300</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZCCHC9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.406000</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZDHHC2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF260</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.052700</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNHIT6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16152 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 P_Value   Medians Cancer      size\n",
       "Proteomics                                         \n",
       "ARMH3       5.396032e-11  0.405134    Gbm  2.364277\n",
       "CUTC        1.593480e-10  0.553255    Gbm  2.255993\n",
       "PIP4K2A     1.009419e-09  0.838882    Gbm  2.071389\n",
       "CUL2        1.122076e-09  0.586396    Gbm  2.060808\n",
       "GDI2        1.302273e-09  0.610188    Gbm  2.045916\n",
       "...                  ...       ...    ...       ...\n",
       "WWP1                 NaN  0.120300  Colon       NaN\n",
       "ZCCHC9               NaN -0.406000  Colon       NaN\n",
       "ZDHHC2               NaN       NaN  Colon       NaN\n",
       "ZNF260               NaN -0.052700  Colon       NaN\n",
       "ZNHIT6               NaN       NaN  Colon       NaN\n",
       "\n",
       "[16152 rows x 4 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create long df for heat map\n",
    "\n",
    "cancer = ['Gbm','Hnscc','Luad','Brca','Ov','En','Colon']\n",
    "merged_dfs = [g_merged,h_merged,l_merged,b_merged,o_merged,e_merged,c_merged]\n",
    "\n",
    "merged = pd.DataFrame()\n",
    "i = 0\n",
    "for c in cancer:\n",
    "    m = merged_dfs[i]\n",
    "    m2 = m.assign(Cancer = c)\n",
    "    m2 = m2.rename(columns={c+'_P_Value': 'P_Value'})\n",
    "    m2 = m2.rename(columns={c+'_Median': 'Medians'})\n",
    "    merged = merged.append(m2) \n",
    "    if i < 6:\n",
    "        i += 1\n",
    "\n",
    "# Keep genes with at least one sig ttest\n",
    "bool_df2 = merged['Proteomics'].isin(sig)\n",
    "plot_df = merged[bool_df2]\n",
    "plot_df = plot_df.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "# log p-vals for right scale in plot (bigger circle, smaller pval)\n",
    "plot_df['size'] = plot_df['P_Value'].apply(lambda x: -1*(np.log(x)/10))\n",
    "plot_df = plot_df.set_index('Proteomics')\n",
    "plot_df#when df of certain cancer doesnt have gene, not added for that cancer since appending(blank in graph, like nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteomics</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DOCK1</th>\n",
       "      <td>5.308552e-07</td>\n",
       "      <td>0.444503</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.444878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOCK1</th>\n",
       "      <td>2.598816e-01</td>\n",
       "      <td>0.112847</td>\n",
       "      <td>Hnscc</td>\n",
       "      <td>0.134753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOCK1</th>\n",
       "      <td>1.314539e-02</td>\n",
       "      <td>0.569100</td>\n",
       "      <td>Luad</td>\n",
       "      <td>0.433168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOCK1</th>\n",
       "      <td>5.616896e-02</td>\n",
       "      <td>0.340050</td>\n",
       "      <td>Brca</td>\n",
       "      <td>0.287939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOCK1</th>\n",
       "      <td>3.248402e-03</td>\n",
       "      <td>0.296576</td>\n",
       "      <td>Ov</td>\n",
       "      <td>0.572959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOCK1</th>\n",
       "      <td>2.539366e-01</td>\n",
       "      <td>0.117000</td>\n",
       "      <td>En</td>\n",
       "      <td>0.137067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOCK1</th>\n",
       "      <td>7.863180e-02</td>\n",
       "      <td>0.044700</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.254298</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 P_Value   Medians Cancer      size\n",
       "Proteomics                                         \n",
       "DOCK1       5.308552e-07  0.444503    Gbm  1.444878\n",
       "DOCK1       2.598816e-01  0.112847  Hnscc  0.134753\n",
       "DOCK1       1.314539e-02  0.569100   Luad  0.433168\n",
       "DOCK1       5.616896e-02  0.340050   Brca  0.287939\n",
       "DOCK1       3.248402e-03  0.296576     Ov  0.572959\n",
       "DOCK1       2.539366e-01  0.117000     En  0.137067\n",
       "DOCK1       7.863180e-02  0.044700  Colon  0.254298"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# CHECK\n",
    "plot_df.loc[plot_df.index == 'DOCK1']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_df.to_csv('heat_map_df.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Keep PosNeg Genes df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "def HasPosNeg(row):\n",
    "    hasPos = False\n",
    "    hasNeg= False\n",
    "\n",
    "    for item in row:\n",
    "        if pd.isnull(item):\n",
    "            continue\n",
    "        if item < -0.3:\n",
    "            hasNeg = True\n",
    "        if item > 0.3:\n",
    "            hasPos = True\n",
    "           \n",
    "    if hasPos & hasNeg:\n",
    "        return True\n",
    "    return False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteomics</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARMH3</th>\n",
       "      <td>0.405134</td>\n",
       "      <td>0.135941</td>\n",
       "      <td>0.37845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1503</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUTC</th>\n",
       "      <td>0.553255</td>\n",
       "      <td>0.314307</td>\n",
       "      <td>1.01995</td>\n",
       "      <td>0.48080</td>\n",
       "      <td>0.159424</td>\n",
       "      <td>0.2533</td>\n",
       "      <td>0.08210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIP4K2A</th>\n",
       "      <td>0.838882</td>\n",
       "      <td>0.224713</td>\n",
       "      <td>-0.04535</td>\n",
       "      <td>0.07435</td>\n",
       "      <td>-0.052955</td>\n",
       "      <td>-0.1690</td>\n",
       "      <td>-0.10135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUL2</th>\n",
       "      <td>0.586396</td>\n",
       "      <td>-0.017886</td>\n",
       "      <td>-0.05695</td>\n",
       "      <td>-0.17195</td>\n",
       "      <td>-0.033091</td>\n",
       "      <td>-0.0232</td>\n",
       "      <td>0.07690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GDI2</th>\n",
       "      <td>0.610188</td>\n",
       "      <td>-0.001559</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.14995</td>\n",
       "      <td>0.067389</td>\n",
       "      <td>-0.1286</td>\n",
       "      <td>0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SUV39H1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.37210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7950</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RSPH3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.81260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.5200</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARMC3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.57430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0990</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC57</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.11070</td>\n",
       "      <td>0.10775</td>\n",
       "      <td>0.087283</td>\n",
       "      <td>-0.8445</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARMC4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.16440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.106026</td>\n",
       "      <td>-0.8960</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2584 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gbm_Median  Hnscc_Median  Luad_Median  Brca_Median  Ov_Median  \\\n",
       "Proteomics                                                                  \n",
       "ARMH3         0.405134      0.135941      0.37845          NaN        NaN   \n",
       "CUTC          0.553255      0.314307      1.01995      0.48080   0.159424   \n",
       "PIP4K2A       0.838882      0.224713     -0.04535      0.07435  -0.052955   \n",
       "CUL2          0.586396     -0.017886     -0.05695     -0.17195  -0.033091   \n",
       "GDI2          0.610188     -0.001559          NaN     -0.14995   0.067389   \n",
       "...                ...           ...          ...          ...        ...   \n",
       "SUV39H1            NaN           NaN          NaN     -0.37210        NaN   \n",
       "RSPH3              NaN           NaN      0.81260          NaN        NaN   \n",
       "ARMC3              NaN           NaN      0.57430          NaN        NaN   \n",
       "CCDC57             NaN           NaN     -0.11070      0.10775   0.087283   \n",
       "ARMC4              NaN           NaN      0.16440          NaN   0.106026   \n",
       "\n",
       "            En_Median  Colon_Median  \n",
       "Proteomics                           \n",
       "ARMH3         -0.1503           NaN  \n",
       "CUTC           0.2533       0.08210  \n",
       "PIP4K2A       -0.1690      -0.10135  \n",
       "CUL2          -0.0232       0.07690  \n",
       "GDI2          -0.1286       0.04450  \n",
       "...               ...           ...  \n",
       "SUV39H1        0.7950           NaN  \n",
       "RSPH3         -1.5200           NaN  \n",
       "ARMC3         -1.0990           NaN  \n",
       "CCDC57        -0.8445           NaN  \n",
       "ARMC4         -0.8960           NaN  \n",
       "\n",
       "[2584 rows x 7 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_df = sig_df.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "\n",
    "only_med = sig_df.drop(columns= ['Gbm_P_Value','Hnscc_P_Value','Luad_P_Value',\n",
    "                     'Brca_P_Value','Ov_P_Value','En_P_Value','Colon_P_Value'])\n",
    "only_med"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  # Remove the CWD from sys.path while we load stuff.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteomics</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PACS2</th>\n",
       "      <td>1.123526e-08</td>\n",
       "      <td>0.386279</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.830421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOCK5</th>\n",
       "      <td>2.754160e-08</td>\n",
       "      <td>0.862819</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.740757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PDZD8</th>\n",
       "      <td>2.315414e-07</td>\n",
       "      <td>0.444260</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.527851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HAGH</th>\n",
       "      <td>3.232029e-07</td>\n",
       "      <td>0.601619</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.494499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DCTN4</th>\n",
       "      <td>5.256343e-07</td>\n",
       "      <td>0.362140</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.445866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPDPF</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.391000</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PYHIN1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.118000</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RANGRF</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.440000</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SIKE1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.607300</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZDHHC2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4242 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 P_Value   Medians Cancer      size\n",
       "Proteomics                                         \n",
       "PACS2       1.123526e-08  0.386279    Gbm  1.830421\n",
       "DOCK5       2.754160e-08  0.862819    Gbm  1.740757\n",
       "PDZD8       2.315414e-07  0.444260    Gbm  1.527851\n",
       "HAGH        3.232029e-07  0.601619    Gbm  1.494499\n",
       "DCTN4       5.256343e-07  0.362140    Gbm  1.445866\n",
       "...                  ...       ...    ...       ...\n",
       "PPDPF                NaN -0.391000  Colon       NaN\n",
       "PYHIN1               NaN  1.118000  Colon       NaN\n",
       "RANGRF               NaN -0.440000  Colon       NaN\n",
       "SIKE1                NaN -0.607300  Colon       NaN\n",
       "ZDHHC2               NaN       NaN  Colon       NaN\n",
       "\n",
       "[4242 rows x 4 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_med[\"Pos_Neg\"] = only_med.apply(HasPosNeg, axis = 1)\n",
    "\n",
    "pn = only_med.loc[only_med['Pos_Neg'] == True]\n",
    "pn_genes = list(pn.index) # list of genes that have posNeg\n",
    "\n",
    "get = plot_df.index.isin(pn_genes)\n",
    "plot_df2 = plot_df[get]\n",
    "\n",
    "plot_df2['size'] = plot_df2['P_Value'].apply(lambda x: -1*(np.log(x)/10))\n",
    "plot_df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteomics</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MAFF</th>\n",
       "      <td>0.008464</td>\n",
       "      <td>-0.202893</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.477196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAFF</th>\n",
       "      <td>0.673107</td>\n",
       "      <td>0.060547</td>\n",
       "      <td>Hnscc</td>\n",
       "      <td>0.039585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAFF</th>\n",
       "      <td>0.187607</td>\n",
       "      <td>-0.282200</td>\n",
       "      <td>Luad</td>\n",
       "      <td>0.167341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAFF</th>\n",
       "      <td>0.078518</td>\n",
       "      <td>-0.499200</td>\n",
       "      <td>Brca</td>\n",
       "      <td>0.254443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAFF</th>\n",
       "      <td>0.968044</td>\n",
       "      <td>-0.121901</td>\n",
       "      <td>Ov</td>\n",
       "      <td>0.003248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAFF</th>\n",
       "      <td>0.741379</td>\n",
       "      <td>0.070400</td>\n",
       "      <td>En</td>\n",
       "      <td>0.029924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAFF</th>\n",
       "      <td>0.200121</td>\n",
       "      <td>0.323500</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.160883</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             P_Value   Medians Cancer      size\n",
       "Proteomics                                     \n",
       "MAFF        0.008464 -0.202893    Gbm  0.477196\n",
       "MAFF        0.673107  0.060547  Hnscc  0.039585\n",
       "MAFF        0.187607 -0.282200   Luad  0.167341\n",
       "MAFF        0.078518 -0.499200   Brca  0.254443\n",
       "MAFF        0.968044 -0.121901     Ov  0.003248\n",
       "MAFF        0.741379  0.070400     En  0.029924\n",
       "MAFF        0.200121  0.323500  Colon  0.160883"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# CHECK\n",
    "plot_df2.loc[plot_df2.index == 'MAFF']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_df2.to_csv('pos_neg_df.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
