{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create tables with p-value and change in medians for multiple cancers "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a dataframe with p-value results from t-tests for all proteins (trans gene proteomics when PTEN has cnv deletions compared to PTEN wildtype). The dataframe also includes the change in medians between deletions and wildtype. Prepare these tables for further analysis by creating csv files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "\n",
    "import plot_utils as pu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Returns a dataframe with proteomics and mutation type\n",
    "\n",
    "# all_prot: list of trans genes\n",
    "\n",
    "def all_prot_format_df(cancer_object, all_prot, gene_in = 'PTEN', utils = u):\n",
    "    mut_type = cancer_object.get_genotype_all_vars(gene_in)\n",
    "    \n",
    "    if cancer_object.get_cancer_type() not in ('luad'):\n",
    "        # Keep only tumor samples from proteomics\n",
    "        prot_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "            mutations_genes = [gene_in], omics_df_name = 'proteomics', omics_genes = all_prot)\n",
    "        # Reduce a multiindex \n",
    "        if isinstance(prot_and_mutations.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "            prot_and_mutations = utils.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "        prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "\n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        mut_type = mut_type[['Mutation']] \n",
    "        prot_df = prot_and_mutations.iloc[:,:-4] # Keep only proteomics\n",
    "        merged = prot_df.join(mut_type)\n",
    "        \n",
    "        # Keep only Wildtype and deletion\n",
    "        compare = ['Wildtype_Tumor','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        del_wt = merged[get]\n",
    "    \n",
    "    \n",
    "    # Luad has no somatic mutations for PTEN which changes some things\n",
    "    else: \n",
    "        # get_genotype_all_vars add cnv data under the column PTEN\n",
    "        mut_type = mut_type.drop(columns= gene_in)\n",
    "        # different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "        omics = cancer_object.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics',genes1= gene_in, \n",
    "            genes2= all_prot)\n",
    "        omics = utils.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "        omics = omics.drop(columns='PTEN_CNV')\n",
    "        # Get only tumor samples\n",
    "        p = cancer_object.get_proteomics(tissue_type='tumor')\n",
    "        tumor_ids = list(p.index)\n",
    "        get = omics.index.isin(tumor_ids)\n",
    "        omics = omics[get]\n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        merged = omics.join(mut_type) \n",
    "        # Keep only No_Mutation (wildtype) and deletion\n",
    "        compare = ['No_Mutation','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        del_wt = merged[get]\n",
    "        del_wt['Mutation'] = np.where(\n",
    "            del_wt['Mutation'] == 'No_Mutation', 'Wildtype_Tumor', 'Deletion')\n",
    "\n",
    "    return del_wt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Params\n",
    "cancer_object: Object. Variable for the loaded cancer data set.\n",
    "cancer_name: Str. name to add to the created dataframe.\n",
    "del_wt_df: DataFrame. Only samples with deletions and wildtype for PTEN. \n",
    "all_prot_list: List. All proteins in proteomics data frame. \n",
    "\n",
    "Returns a dataframe with the difference in medians between proteomics with PTEN wt and del (wt - del). \n",
    "'''\n",
    "\n",
    "def get_change_in_medians_df(cancer_object, cancer_name, del_wt_df, all_prot_list):\n",
    "    d = del_wt_df[del_wt_df.Mutation == \"Deletion\"]\n",
    "    wt = del_wt_df[del_wt_df.Mutation == \"Wildtype_Tumor\"]\n",
    "    del_med = d.median()\n",
    "    wt_med = wt.median()\n",
    "\n",
    "    med_dict = {}\n",
    "    \n",
    "    # Correlation: + is mutant up compared to wt, - is mutant down\n",
    "    for prot in all_prot_list:\n",
    "        dif = del_med[prot+'_proteomics'] - wt_med[prot+'_proteomics']\n",
    "        med_dict[prot+'_proteomics'] = dif\n",
    "        \n",
    "    df = pd.DataFrame.from_dict(med_dict, orient='index', columns=[cancer_name+'_Median'])\n",
    "    df = df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "    \n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Step 1: Create Data frames with p-values and differences in median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each cancer needs a data frame containing only samples that have PTEN cnv deletions and PTEN wildtype with trans proteomics. Use wrap_ttest to run many T-tests for all genes in the proteomics data frame. Use get_change_in_medians_df to create the data frame with change in median values. Merge both data frames. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load in cancer data sets from cptac. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that endometrial index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that luad index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The HNSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LUAD dataset is under publication embargo until July 01, 2020. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 4)\n"
     ]
    }
   ],
   "source": [
    "g = cptac.Gbm()\n",
    "en = cptac.Endometrial()\n",
    "h = cptac.Hnscc()\n",
    "l = cptac.Luad()\n",
    "ls = cptac.Lscc()\n",
    "o = cptac.Ovarian()\n",
    "col = cptac.Colon()\n",
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gbm"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 1: Format data frame using all_prot_format_df. This returns a dataframe with proteomics and a Mutation column containing only PTEN cnv deletions and wildtype tumors. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG_proteomics</th>\n",
       "      <th>A2M_proteomics</th>\n",
       "      <th>AAAS_proteomics</th>\n",
       "      <th>AACS_proteomics</th>\n",
       "      <th>AADAT_proteomics</th>\n",
       "      <th>AAED1_proteomics</th>\n",
       "      <th>AAGAB_proteomics</th>\n",
       "      <th>AAK1_proteomics</th>\n",
       "      <th>AAMDC_proteomics</th>\n",
       "      <th>AAMP_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>0.077630</td>\n",
       "      <td>0.487228</td>\n",
       "      <td>-0.254208</td>\n",
       "      <td>-0.144373</td>\n",
       "      <td>0.551881</td>\n",
       "      <td>-0.025276</td>\n",
       "      <td>-0.467451</td>\n",
       "      <td>-0.089511</td>\n",
       "      <td>-0.078806</td>\n",
       "      <td>0.329603</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047437</td>\n",
       "      <td>-0.105908</td>\n",
       "      <td>-0.347076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.459635</td>\n",
       "      <td>0.079452</td>\n",
       "      <td>-0.784983</td>\n",
       "      <td>-0.488441</td>\n",
       "      <td>0.167990</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>-0.145975</td>\n",
       "      <td>0.798796</td>\n",
       "      <td>0.184242</td>\n",
       "      <td>-0.470603</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.390211</td>\n",
       "      <td>0.245466</td>\n",
       "      <td>-0.609998</td>\n",
       "      <td>0.118625</td>\n",
       "      <td>-0.086927</td>\n",
       "      <td>...</td>\n",
       "      <td>0.161975</td>\n",
       "      <td>-0.213093</td>\n",
       "      <td>0.235571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.107421</td>\n",
       "      <td>0.048724</td>\n",
       "      <td>0.138403</td>\n",
       "      <td>-0.290141</td>\n",
       "      <td>0.405037</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>0.821991</td>\n",
       "      <td>1.096470</td>\n",
       "      <td>-0.094421</td>\n",
       "      <td>-0.106304</td>\n",
       "      <td>0.084578</td>\n",
       "      <td>0.176402</td>\n",
       "      <td>-0.248151</td>\n",
       "      <td>0.014061</td>\n",
       "      <td>-0.699773</td>\n",
       "      <td>-0.638462</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.065534</td>\n",
       "      <td>-0.306717</td>\n",
       "      <td>0.879991</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.883564</td>\n",
       "      <td>-0.172222</td>\n",
       "      <td>0.011876</td>\n",
       "      <td>-0.131889</td>\n",
       "      <td>-0.503581</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>-0.064567</td>\n",
       "      <td>0.129385</td>\n",
       "      <td>0.047751</td>\n",
       "      <td>-0.118187</td>\n",
       "      <td>0.237434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.303847</td>\n",
       "      <td>0.322163</td>\n",
       "      <td>-0.555479</td>\n",
       "      <td>-0.363414</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.254535</td>\n",
       "      <td>0.463653</td>\n",
       "      <td>0.580230</td>\n",
       "      <td>0.503044</td>\n",
       "      <td>-0.604986</td>\n",
       "      <td>0.178077</td>\n",
       "      <td>-0.720059</td>\n",
       "      <td>-0.150197</td>\n",
       "      <td>-0.268715</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-0.763691</td>\n",
       "      <td>-1.031834</td>\n",
       "      <td>-0.217194</td>\n",
       "      <td>-0.695701</td>\n",
       "      <td>0.184173</td>\n",
       "      <td>-0.474816</td>\n",
       "      <td>-0.051789</td>\n",
       "      <td>0.344842</td>\n",
       "      <td>-0.642746</td>\n",
       "      <td>0.068863</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.092502</td>\n",
       "      <td>0.010639</td>\n",
       "      <td>-0.465079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.500083</td>\n",
       "      <td>0.112651</td>\n",
       "      <td>1.004660</td>\n",
       "      <td>-0.230304</td>\n",
       "      <td>-0.102416</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11142 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        A1BG_proteomics  A2M_proteomics  AAAS_proteomics  AACS_proteomics  \\\n",
       "Patient_ID                                                                      \n",
       "C3L-00104          0.077630        0.487228        -0.254208        -0.144373   \n",
       "C3L-00365         -0.145975        0.798796         0.184242        -0.470603   \n",
       "C3L-00674          0.821991        1.096470        -0.094421        -0.106304   \n",
       "C3L-00677         -0.064567        0.129385         0.047751        -0.118187   \n",
       "C3L-01040         -0.763691       -1.031834        -0.217194        -0.695701   \n",
       "\n",
       "Name        AADAT_proteomics  AAED1_proteomics  AAGAB_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00104           0.551881         -0.025276         -0.467451   \n",
       "C3L-00365                NaN          0.390211          0.245466   \n",
       "C3L-00674           0.084578          0.176402         -0.248151   \n",
       "C3L-00677           0.237434               NaN          0.303847   \n",
       "C3L-01040           0.184173         -0.474816         -0.051789   \n",
       "\n",
       "Name        AAK1_proteomics  AAMDC_proteomics  AAMP_proteomics  ...  \\\n",
       "Patient_ID                                                      ...   \n",
       "C3L-00104         -0.089511         -0.078806         0.329603  ...   \n",
       "C3L-00365         -0.609998          0.118625        -0.086927  ...   \n",
       "C3L-00674          0.014061         -0.699773        -0.638462  ...   \n",
       "C3L-00677          0.322163         -0.555479        -0.363414  ...   \n",
       "C3L-01040          0.344842         -0.642746         0.068863  ...   \n",
       "\n",
       "Name        ZSWIM8_proteomics  ZW10_proteomics  ZWILCH_proteomics  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104           -0.047437        -0.105908          -0.347076   \n",
       "C3L-00365            0.161975        -0.213093           0.235571   \n",
       "C3L-00674           -0.065534        -0.306717           0.879991   \n",
       "C3L-00677           -0.254535         0.463653           0.580230   \n",
       "C3L-01040           -0.092502         0.010639          -0.465079   \n",
       "\n",
       "Name        ZWINT_proteomics  ZXDC_proteomics  ZYG11B_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00104                NaN         0.459635           0.079452   \n",
       "C3L-00365                NaN         0.107421           0.048724   \n",
       "C3L-00674                NaN         0.883564          -0.172222   \n",
       "C3L-00677           0.503044        -0.604986           0.178077   \n",
       "C3L-01040                NaN        -0.500083           0.112651   \n",
       "\n",
       "Name        ZYX_proteomics  ZZEF1_proteomics  ZZZ3_proteomics  Mutation  \n",
       "Patient_ID                                                               \n",
       "C3L-00104        -0.784983         -0.488441         0.167990  Deletion  \n",
       "C3L-00365         0.138403         -0.290141         0.405037  Deletion  \n",
       "C3L-00674         0.011876         -0.131889        -0.503581  Deletion  \n",
       "C3L-00677        -0.720059         -0.150197        -0.268715  Deletion  \n",
       "C3L-01040         1.004660         -0.230304        -0.102416  Deletion  \n",
       "\n",
       "[5 rows x 11142 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "g_prot = g.get_proteomics()\n",
    "g_prot_list = list(g_prot.columns)\n",
    "\n",
    "g_del_wt = all_prot_format_df(g, g_prot_list)\n",
    "g_del_wt = g_del_wt.dropna(axis='columns', how='all')\n",
    "g_del_wt.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 2: Run T-tests. Create a data frame of just significant comparisons and another data frame for all comparisons. A moderately stringent correction is used (FDR_BH)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:325: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:333: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:256: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "               Comparison       P_Value\n",
      "0       ARMH3_proteomics  5.396032e-11\n",
      "1        CUTC_proteomics  1.593480e-10\n",
      "2     PIP4K2A_proteomics  1.009419e-09\n",
      "3        CUL2_proteomics  1.122076e-09\n",
      "4        GDI2_proteomics  1.302273e-09\n",
      "...                  ...           ...\n",
      "1882  GOLGA7B_proteomics  8.446816e-03\n",
      "1883     TEX2_proteomics  8.452685e-03\n",
      "1884     MAFF_proteomics  8.463737e-03\n",
      "1885   LRPAP1_proteomics  8.466030e-03\n",
      "1886  PPP2R2B_proteomics  8.466960e-03\n",
      "\n",
      "[1887 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:325: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:333: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:256: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3_proteomics</td>\n",
       "      <td>5.396032e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC_proteomics</td>\n",
       "      <td>1.593480e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A_proteomics</td>\n",
       "      <td>1.009419e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2_proteomics</td>\n",
       "      <td>1.122076e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2_proteomics</td>\n",
       "      <td>1.302273e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1882</th>\n",
       "      <td>GOLGA7B_proteomics</td>\n",
       "      <td>8.446816e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1883</th>\n",
       "      <td>TEX2_proteomics</td>\n",
       "      <td>8.452685e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1884</th>\n",
       "      <td>MAFF_proteomics</td>\n",
       "      <td>8.463737e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1885</th>\n",
       "      <td>LRPAP1_proteomics</td>\n",
       "      <td>8.466030e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1886</th>\n",
       "      <td>PPP2R2B_proteomics</td>\n",
       "      <td>8.466960e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1887 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Comparison       P_Value\n",
       "0       ARMH3_proteomics  5.396032e-11\n",
       "1        CUTC_proteomics  1.593480e-10\n",
       "2     PIP4K2A_proteomics  1.009419e-09\n",
       "3        CUL2_proteomics  1.122076e-09\n",
       "4        GDI2_proteomics  1.302273e-09\n",
       "...                  ...           ...\n",
       "1882  GOLGA7B_proteomics  8.446816e-03\n",
       "1883     TEX2_proteomics  8.452685e-03\n",
       "1884     MAFF_proteomics  8.463737e-03\n",
       "1885   LRPAP1_proteomics  8.466030e-03\n",
       "1886  PPP2R2B_proteomics  8.466960e-03\n",
       "\n",
       "[1887 rows x 2 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(g_del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig genes\n",
    "g_sig = u.wrap_ttest(g_del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "# Create list of sig genes\n",
    "if g_sig is not None:\n",
    "    g_sig_list = list(g_sig.Comparison)\n",
    "else: \n",
    "    g_sig_list = None\n",
    "print('significant pvals: \\n',g_sig)\n",
    "\n",
    "# Get all pvals\n",
    "g = u.wrap_ttest(g_del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "g_pval = g.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Gbm_P_Value'})\n",
    "g_sig"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 3: Create the differences in median df. (median of wildtype tumors - median of cnv deletions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "g_med = get_change_in_medians_df(g, \"Gbm\", g_del_wt, g_prot_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 4: Merge the p-values and the differences in median dfs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3_proteomics</td>\n",
       "      <td>5.396032e-11</td>\n",
       "      <td>-0.405134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC_proteomics</td>\n",
       "      <td>1.593480e-10</td>\n",
       "      <td>-0.553255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A_proteomics</td>\n",
       "      <td>1.009419e-09</td>\n",
       "      <td>-0.838882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2_proteomics</td>\n",
       "      <td>1.122076e-09</td>\n",
       "      <td>-0.586396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2_proteomics</td>\n",
       "      <td>1.302273e-09</td>\n",
       "      <td>-0.610188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11136</th>\n",
       "      <td>ZNF805_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11137</th>\n",
       "      <td>ZNF813_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11138</th>\n",
       "      <td>ZNF814_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.265936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11139</th>\n",
       "      <td>ZNF888_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11140</th>\n",
       "      <td>ZSCAN12_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11141 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics   Gbm_P_Value  Gbm_Median\n",
       "0        ARMH3_proteomics  5.396032e-11   -0.405134\n",
       "1         CUTC_proteomics  1.593480e-10   -0.553255\n",
       "2      PIP4K2A_proteomics  1.009419e-09   -0.838882\n",
       "3         CUL2_proteomics  1.122076e-09   -0.586396\n",
       "4         GDI2_proteomics  1.302273e-09   -0.610188\n",
       "...                   ...           ...         ...\n",
       "11136   ZNF805_proteomics           NaN         NaN\n",
       "11137   ZNF813_proteomics           NaN         NaN\n",
       "11138   ZNF814_proteomics           NaN   -0.265936\n",
       "11139   ZNF888_proteomics           NaN         NaN\n",
       "11140  ZSCAN12_proteomics           NaN         NaN\n",
       "\n",
       "[11141 rows x 3 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_merged = g_pval.merge(g_med, on='Proteomics',how='outer')\n",
    "\n",
    "# Create csv\n",
    "root = R'~\\Github\\WhenMutationsDontMatter\\PTEN\\Step_3_trans_effect\\csv'\n",
    "g_merged.to_csv(root+R'\\gbm_pval_medians.csv',index=False)\n",
    "g_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Repeat for other cancers."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n"
     ]
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "o_prot = o.get_proteomics()\n",
    "o_prot = u.reduce_multiindex(o_prot, levels_to_drop = 1)\n",
    "o_prot_list = list(o_prot.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 14)\n"
     ]
    }
   ],
   "source": [
    "o_del_wt = all_prot_format_df(o, o_prot_list)\n",
    "o_del_wt = o_del_wt.dropna(axis='columns', how='all')\n",
    "# Drop duplicate columns - FIX\n",
    "o_del_wt = o_del_wt.loc[:,~o_del_wt.columns.duplicated()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:325: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:333: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:256: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "          Comparison       P_Value\n",
      "0  MMS19_proteomics  1.058914e-08\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:325: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:333: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:256: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(o_del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "o_sig = u.wrap_ttest(o_del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if o_sig is not None:\n",
    "    o_sig_list = list(o_sig.Comparison)\n",
    "else: \n",
    "    o_sig_list = None\n",
    "print('significant pvals: \\n',o_sig)\n",
    "\n",
    "# Get all pvals\n",
    "o = u.wrap_ttest(o_del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "o_pval = o.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Ov_P_Value'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "o_med = get_change_in_medians_df(o, \"Ov\", o_del_wt, o_prot_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MMS19_proteomics</td>\n",
       "      <td>1.058914e-08</td>\n",
       "      <td>-0.324897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RACK1_proteomics</td>\n",
       "      <td>8.573522e-05</td>\n",
       "      <td>-0.191250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PI4K2A_proteomics</td>\n",
       "      <td>9.310383e-05</td>\n",
       "      <td>-0.298399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WAPL_proteomics</td>\n",
       "      <td>1.724680e-04</td>\n",
       "      <td>-0.218775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IDE_proteomics</td>\n",
       "      <td>2.011475e-04</td>\n",
       "      <td>-0.208025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10072</th>\n",
       "      <td>SAXO2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.099993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10073</th>\n",
       "      <td>SH3GL3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.843461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10074</th>\n",
       "      <td>SLC22A3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.783325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10075</th>\n",
       "      <td>SSX2B_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10076</th>\n",
       "      <td>TFF2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10077 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Proteomics    Ov_P_Value  Ov_Median\n",
       "0        MMS19_proteomics  1.058914e-08  -0.324897\n",
       "1        RACK1_proteomics  8.573522e-05  -0.191250\n",
       "2       PI4K2A_proteomics  9.310383e-05  -0.298399\n",
       "3         WAPL_proteomics  1.724680e-04  -0.218775\n",
       "4          IDE_proteomics  2.011475e-04  -0.208025\n",
       "...                   ...           ...        ...\n",
       "10072    SAXO2_proteomics           NaN  -0.099993\n",
       "10073   SH3GL3_proteomics           NaN   0.843461\n",
       "10074  SLC22A3_proteomics           NaN   0.783325\n",
       "10075    SSX2B_proteomics           NaN        NaN\n",
       "10076     TFF2_proteomics           NaN        NaN\n",
       "\n",
       "[10077 rows x 3 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "o_merged = o_pval.merge(o_med, on='Proteomics',how='outer')\n",
    "\n",
    "# Create csv\n",
    "o_merged.to_csv(root+R'\\ov_pval_medians.csv',index=False)\n",
    "o_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Breast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 14)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG_proteomics</th>\n",
       "      <th>A2M_proteomics</th>\n",
       "      <th>A2ML1_proteomics</th>\n",
       "      <th>AAAS_proteomics</th>\n",
       "      <th>AACS_proteomics</th>\n",
       "      <th>AADAT_proteomics</th>\n",
       "      <th>AAED1_proteomics</th>\n",
       "      <th>AAGAB_proteomics</th>\n",
       "      <th>AAK1_proteomics</th>\n",
       "      <th>AAMDC_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CPT000814</th>\n",
       "      <td>-0.6712</td>\n",
       "      <td>-0.2075</td>\n",
       "      <td>2.7959</td>\n",
       "      <td>1.3969</td>\n",
       "      <td>-1.0899</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.6708</td>\n",
       "      <td>-0.3484</td>\n",
       "      <td>-0.4756</td>\n",
       "      <td>-0.7299</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.6536</td>\n",
       "      <td>0.3384</td>\n",
       "      <td>2.1169</td>\n",
       "      <td>1.3910</td>\n",
       "      <td>-2.1230</td>\n",
       "      <td>0.9136</td>\n",
       "      <td>-0.8082</td>\n",
       "      <td>-1.4793</td>\n",
       "      <td>0.9136</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CPT001846</th>\n",
       "      <td>1.3964</td>\n",
       "      <td>1.3302</td>\n",
       "      <td>-5.0948</td>\n",
       "      <td>0.7674</td>\n",
       "      <td>-1.6845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1022</td>\n",
       "      <td>-0.5814</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>-2.2857</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4711</td>\n",
       "      <td>0.6018</td>\n",
       "      <td>0.2062</td>\n",
       "      <td>-0.2137</td>\n",
       "      <td>-2.1219</td>\n",
       "      <td>0.0860</td>\n",
       "      <td>2.5814</td>\n",
       "      <td>-0.2852</td>\n",
       "      <td>-0.1074</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR001</th>\n",
       "      <td>2.0219</td>\n",
       "      <td>1.6269</td>\n",
       "      <td>-3.2943</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>-1.0739</td>\n",
       "      <td>1.2255</td>\n",
       "      <td>0.2754</td>\n",
       "      <td>-1.1187</td>\n",
       "      <td>-0.0534</td>\n",
       "      <td>-0.2519</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2306</td>\n",
       "      <td>-0.3010</td>\n",
       "      <td>0.3395</td>\n",
       "      <td>-0.5316</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4996</td>\n",
       "      <td>0.7622</td>\n",
       "      <td>-1.5607</td>\n",
       "      <td>0.0256</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR009</th>\n",
       "      <td>1.2556</td>\n",
       "      <td>3.4489</td>\n",
       "      <td>2.8043</td>\n",
       "      <td>-0.2956</td>\n",
       "      <td>-1.7261</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.0471</td>\n",
       "      <td>-0.3547</td>\n",
       "      <td>-0.8298</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2596</td>\n",
       "      <td>0.1898</td>\n",
       "      <td>-0.5010</td>\n",
       "      <td>-0.4189</td>\n",
       "      <td>0.3080</td>\n",
       "      <td>0.5057</td>\n",
       "      <td>0.2181</td>\n",
       "      <td>-0.2288</td>\n",
       "      <td>-0.2750</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR010</th>\n",
       "      <td>-0.3843</td>\n",
       "      <td>-1.0239</td>\n",
       "      <td>-5.3604</td>\n",
       "      <td>0.3739</td>\n",
       "      <td>3.1422</td>\n",
       "      <td>7.6788</td>\n",
       "      <td>0.1961</td>\n",
       "      <td>-1.2437</td>\n",
       "      <td>-0.0015</td>\n",
       "      <td>-0.0953</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0874</td>\n",
       "      <td>1.6506</td>\n",
       "      <td>-2.0636</td>\n",
       "      <td>-1.2733</td>\n",
       "      <td>-1.2338</td>\n",
       "      <td>-0.7794</td>\n",
       "      <td>1.5518</td>\n",
       "      <td>-0.0410</td>\n",
       "      <td>-0.4756</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 9772 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        A1BG_proteomics  A2M_proteomics  A2ML1_proteomics  \\\n",
       "Patient_ID                                                      \n",
       "CPT000814           -0.6712         -0.2075            2.7959   \n",
       "CPT001846            1.3964          1.3302           -5.0948   \n",
       "X01BR001             2.0219          1.6269           -3.2943   \n",
       "X01BR009             1.2556          3.4489            2.8043   \n",
       "X01BR010            -0.3843         -1.0239           -5.3604   \n",
       "\n",
       "Name        AAAS_proteomics  AACS_proteomics  AADAT_proteomics  \\\n",
       "Patient_ID                                                       \n",
       "CPT000814            1.3969          -1.0899               NaN   \n",
       "CPT001846            0.7674          -1.6845               NaN   \n",
       "X01BR001             0.3352          -1.0739            1.2255   \n",
       "X01BR009            -0.2956          -1.7261               NaN   \n",
       "X01BR010             0.3739           3.1422            7.6788   \n",
       "\n",
       "Name        AAED1_proteomics  AAGAB_proteomics  AAK1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "CPT000814             1.6708           -0.3484          -0.4756   \n",
       "CPT001846             2.1022           -0.5814           0.2916   \n",
       "X01BR001              0.2754           -1.1187          -0.0534   \n",
       "X01BR009                 NaN           -2.0471          -0.3547   \n",
       "X01BR010              0.1961           -1.2437          -0.0015   \n",
       "\n",
       "Name        AAMDC_proteomics  ...  ZSWIM8_proteomics  ZW10_proteomics  \\\n",
       "Patient_ID                    ...                                       \n",
       "CPT000814            -0.7299  ...            -0.6536           0.3384   \n",
       "CPT001846            -2.2857  ...             0.4711           0.6018   \n",
       "X01BR001             -0.2519  ...             0.2306          -0.3010   \n",
       "X01BR009             -0.8298  ...            -0.2596           0.1898   \n",
       "X01BR010             -0.0953  ...             0.0874           1.6506   \n",
       "\n",
       "Name        ZWILCH_proteomics  ZWINT_proteomics  ZXDC_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "CPT000814              2.1169            1.3910          -2.1230   \n",
       "CPT001846              0.2062           -0.2137          -2.1219   \n",
       "X01BR001               0.3395           -0.5316              NaN   \n",
       "X01BR009              -0.5010           -0.4189           0.3080   \n",
       "X01BR010              -2.0636           -1.2733          -1.2338   \n",
       "\n",
       "Name        ZYG11B_proteomics  ZYX_proteomics  ZZEF1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "CPT000814              0.9136         -0.8082           -1.4793   \n",
       "CPT001846              0.0860          2.5814           -0.2852   \n",
       "X01BR001               0.4996          0.7622           -1.5607   \n",
       "X01BR009               0.5057          0.2181           -0.2288   \n",
       "X01BR010              -0.7794          1.5518           -0.0410   \n",
       "\n",
       "Name        ZZZ3_proteomics        Mutation  \n",
       "Patient_ID                                   \n",
       "CPT000814            0.9136  Wildtype_Tumor  \n",
       "CPT001846           -0.1074  Wildtype_Tumor  \n",
       "X01BR001             0.0256  Wildtype_Tumor  \n",
       "X01BR009            -0.2750  Wildtype_Tumor  \n",
       "X01BR010            -0.4756        Deletion  \n",
       "\n",
       "[5 rows x 9772 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "b_prot = b.get_proteomics()\n",
    "b_prot = u.reduce_multiindex(b_prot, levels_to_drop = 1)\n",
    "b_prot_list = list(b_prot.columns)\n",
    "\n",
    "b_del_wt = all_prot_format_df(b, b_prot_list)\n",
    "b_del_wt = b_del_wt.dropna(axis='columns', how='all')\n",
    "b_del_wt = b_del_wt.loc[:,~b_del_wt.columns.duplicated()]\n",
    "b_del_wt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "         Comparison   P_Value\n",
      "0  PTEN_proteomics  0.000004\n"
     ]
    }
   ],
   "source": [
    "cols = list(b_del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "b_sig = u.wrap_ttest(b_del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if b_sig is not None:\n",
    "    b_sig_list = list(b_sig.Comparison)\n",
    "else: \n",
    "    b_sig_list = None\n",
    "print('significant pvals: \\n',b_sig)\n",
    "\n",
    "# Get all pvals\n",
    "b_pval = u.wrap_ttest(b_del_wt, 'Mutation', cols, return_all = True)\n",
    "b_pval = b_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Brca_P_Value'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "b_med = get_change_in_medians_df(b, \"Brca\", b_del_wt, b_prot_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>-1.00800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EIF4H_proteomics</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.41295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIEF1_proteomics</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.84135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMSB10_proteomics</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>0.87630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DRG1_proteomics</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.75135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9766</th>\n",
       "      <td>POLR3E_proteomics</td>\n",
       "      <td>0.999356</td>\n",
       "      <td>-0.09685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9767</th>\n",
       "      <td>IGFBP7_proteomics</td>\n",
       "      <td>0.999595</td>\n",
       "      <td>0.01140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9768</th>\n",
       "      <td>AAMP_proteomics</td>\n",
       "      <td>0.999613</td>\n",
       "      <td>0.15430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9769</th>\n",
       "      <td>TRA2B_proteomics</td>\n",
       "      <td>0.999771</td>\n",
       "      <td>-0.17180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9770</th>\n",
       "      <td>KRIT1_proteomics</td>\n",
       "      <td>0.999848</td>\n",
       "      <td>0.14610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9771 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Brca_P_Value  Brca_Median\n",
       "0       PTEN_proteomics      0.000004     -1.00800\n",
       "1      EIF4H_proteomics      0.000025      0.41295\n",
       "2      MIEF1_proteomics      0.000031      0.84135\n",
       "3     TMSB10_proteomics      0.000043      0.87630\n",
       "4       DRG1_proteomics      0.000065      0.75135\n",
       "...                 ...           ...          ...\n",
       "9766  POLR3E_proteomics      0.999356     -0.09685\n",
       "9767  IGFBP7_proteomics      0.999595      0.01140\n",
       "9768    AAMP_proteomics      0.999613      0.15430\n",
       "9769   TRA2B_proteomics      0.999771     -0.17180\n",
       "9770   KRIT1_proteomics      0.999848      0.14610\n",
       "\n",
       "[9771 rows x 3 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_merged = b_pval.merge(b_med, on='Proteomics',how='outer')\n",
    "\n",
    "# Create csv\n",
    "b_merged.to_csv(root+R'\\brca_pval_medians.csv',index=False)\n",
    "b_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "c_prot = col.get_proteomics()\n",
    "c_prot_list = list(c_prot.columns)\n",
    "\n",
    "c_del_wt = all_prot_format_df(col, c_prot_list)\n",
    "c_del_wt = c_del_wt.dropna(axis='columns', how='all')\n",
    "c_del_wt = c_del_wt.loc[:,~c_del_wt.columns.duplicated()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:325: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:333: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:256: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      " None\n"
     ]
    }
   ],
   "source": [
    "cols = list(c_del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "c_sig = u.wrap_ttest(c_del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if c_sig is not None:\n",
    "    c_sig_list = list(c_sig.Comparison)\n",
    "else: \n",
    "    c_sig_list = None\n",
    "print('significant pvals: \\n',c_sig)\n",
    "\n",
    "# Get all pvals\n",
    "c_pval = u.wrap_ttest(c_del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "c_pval = c_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Colon_P_Value'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "c_med = get_change_in_medians_df(col, \"Colon\", c_del_wt, c_prot_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DFFA_proteomics</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>-0.19930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WAPL_proteomics</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>-0.19605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SEC14L2_proteomics</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>-0.66800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GBF1_proteomics</td>\n",
       "      <td>0.000164</td>\n",
       "      <td>-0.12630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>STK11IP_proteomics</td>\n",
       "      <td>0.000239</td>\n",
       "      <td>-0.31250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8062</th>\n",
       "      <td>ZNF841_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.78250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8063</th>\n",
       "      <td>ZNF888_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8064</th>\n",
       "      <td>ZNHIT6_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8065</th>\n",
       "      <td>ZNRD1_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.06800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8066</th>\n",
       "      <td>ZWINT_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.18535</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8067 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Proteomics  Colon_P_Value  Colon_Median\n",
       "0        DFFA_proteomics       0.000037      -0.19930\n",
       "1        WAPL_proteomics       0.000097      -0.19605\n",
       "2     SEC14L2_proteomics       0.000136      -0.66800\n",
       "3        GBF1_proteomics       0.000164      -0.12630\n",
       "4     STK11IP_proteomics       0.000239      -0.31250\n",
       "...                  ...            ...           ...\n",
       "8062   ZNF841_proteomics            NaN       0.78250\n",
       "8063   ZNF888_proteomics            NaN           NaN\n",
       "8064   ZNHIT6_proteomics            NaN           NaN\n",
       "8065    ZNRD1_proteomics            NaN      -0.06800\n",
       "8066    ZWINT_proteomics            NaN       0.18535\n",
       "\n",
       "[8067 rows x 3 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c_merged = c_pval.merge(c_med, on='Proteomics',how='outer')\n",
    "\n",
    "# Create csv\n",
    "c_merged.to_csv(root+R'\\colon_pval_medians.csv',index=False)\n",
    "c_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Hnscc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 313)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 176 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['A1BG',\n",
       " 'A1CF',\n",
       " 'A2M',\n",
       " 'A2ML1',\n",
       " 'A4GALT',\n",
       " 'AAAS',\n",
       " 'AACS',\n",
       " 'AADAC',\n",
       " 'AADAT',\n",
       " 'AAED1',\n",
       " 'AAGAB',\n",
       " 'AAK1',\n",
       " 'AAMDC',\n",
       " 'AAMP',\n",
       " 'AAR2',\n",
       " 'AARS',\n",
       " 'AARS2',\n",
       " 'AARSD1',\n",
       " 'AASDH',\n",
       " 'AASDHPPT',\n",
       " 'AASS',\n",
       " 'AATF',\n",
       " 'ABAT',\n",
       " 'ABCA1',\n",
       " 'ABCA12',\n",
       " 'ABCA2',\n",
       " 'ABCA3',\n",
       " 'ABCA5',\n",
       " 'ABCA6',\n",
       " 'ABCA7',\n",
       " 'ABCA8',\n",
       " 'ABCB1',\n",
       " 'ABCB10',\n",
       " 'ABCB6',\n",
       " 'ABCB7',\n",
       " 'ABCB8',\n",
       " 'ABCC10',\n",
       " 'ABCC2',\n",
       " 'ABCC3',\n",
       " 'ABCC4',\n",
       " 'ABCC5',\n",
       " 'ABCC9',\n",
       " 'ABCD1',\n",
       " 'ABCD3',\n",
       " 'ABCD4',\n",
       " 'ABCE1',\n",
       " 'ABCF1',\n",
       " 'ABCF2',\n",
       " 'ABCF3',\n",
       " 'ABCG1',\n",
       " 'ABCG2',\n",
       " 'ABCG4',\n",
       " 'ABHD10',\n",
       " 'ABHD11',\n",
       " 'ABHD12',\n",
       " 'ABHD13',\n",
       " 'ABHD14A',\n",
       " 'ABHD14B',\n",
       " 'ABHD16A',\n",
       " 'ABHD17A',\n",
       " 'ABHD17B',\n",
       " 'ABHD17C',\n",
       " 'ABHD2',\n",
       " 'ABHD3',\n",
       " 'ABHD4',\n",
       " 'ABHD5',\n",
       " 'ABHD6',\n",
       " 'ABI1',\n",
       " 'ABI2',\n",
       " 'ABI3',\n",
       " 'ABI3BP',\n",
       " 'ABL1',\n",
       " 'ABL2',\n",
       " 'ABLIM1',\n",
       " 'ABLIM2',\n",
       " 'ABLIM3',\n",
       " 'ABO',\n",
       " 'ABR',\n",
       " 'ABRACL',\n",
       " 'ABRAXAS1',\n",
       " 'ABRAXAS2',\n",
       " 'ABT1',\n",
       " 'ABTB1',\n",
       " 'ABTB2',\n",
       " 'ACAA1',\n",
       " 'ACAA2',\n",
       " 'ACACA',\n",
       " 'ACACB',\n",
       " 'ACAD10',\n",
       " 'ACAD11',\n",
       " 'ACAD8',\n",
       " 'ACAD9',\n",
       " 'ACADL',\n",
       " 'ACADM',\n",
       " 'ACADS',\n",
       " 'ACADSB',\n",
       " 'ACADVL',\n",
       " 'ACAN',\n",
       " 'ACAP1',\n",
       " 'ACAP2',\n",
       " 'ACAP3',\n",
       " 'ACAT1',\n",
       " 'ACAT2',\n",
       " 'ACBD3',\n",
       " 'ACBD5',\n",
       " 'ACBD6',\n",
       " 'ACBD7',\n",
       " 'ACD',\n",
       " 'ACE',\n",
       " 'ACHE',\n",
       " 'ACIN1',\n",
       " 'ACKR1',\n",
       " 'ACKR3',\n",
       " 'ACLY',\n",
       " 'ACO1',\n",
       " 'ACO2',\n",
       " 'ACOT11',\n",
       " 'ACOT13',\n",
       " 'ACOT2',\n",
       " 'ACOT7',\n",
       " 'ACOT8',\n",
       " 'ACOT9',\n",
       " 'ACOX1',\n",
       " 'ACOX2',\n",
       " 'ACOX3',\n",
       " 'ACP1',\n",
       " 'ACP2',\n",
       " 'ACP5',\n",
       " 'ACP6',\n",
       " 'ACP7',\n",
       " 'ACPP',\n",
       " 'ACSF2',\n",
       " 'ACSF3',\n",
       " 'ACSL1',\n",
       " 'ACSL3',\n",
       " 'ACSL4',\n",
       " 'ACSL5',\n",
       " 'ACSL6',\n",
       " 'ACSM3',\n",
       " 'ACSM5',\n",
       " 'ACSS1',\n",
       " 'ACSS2',\n",
       " 'ACSS3',\n",
       " 'ACTA1',\n",
       " 'ACTA2',\n",
       " 'ACTB',\n",
       " 'ACTBL2',\n",
       " 'ACTC1',\n",
       " 'ACTG1',\n",
       " 'ACTG2',\n",
       " 'ACTL6A',\n",
       " 'ACTL8',\n",
       " 'ACTN1',\n",
       " 'ACTN2',\n",
       " 'ACTN3',\n",
       " 'ACTN4',\n",
       " 'ACTR10',\n",
       " 'ACTR1A',\n",
       " 'ACTR1B',\n",
       " 'ACTR2',\n",
       " 'ACTR3',\n",
       " 'ACTR3B',\n",
       " 'ACTR5',\n",
       " 'ACTR6',\n",
       " 'ACTR8',\n",
       " 'ACTRT1',\n",
       " 'ACVR1',\n",
       " 'ACVR1B',\n",
       " 'ACVR2A',\n",
       " 'ACVRL1',\n",
       " 'ACY1',\n",
       " 'ACY3',\n",
       " 'ACYP1',\n",
       " 'ACYP2',\n",
       " 'ADA',\n",
       " 'ADA2',\n",
       " 'ADAL',\n",
       " 'ADAM10',\n",
       " 'ADAM12',\n",
       " 'ADAM15',\n",
       " 'ADAM17',\n",
       " 'ADAM22',\n",
       " 'ADAM28',\n",
       " 'ADAM8',\n",
       " 'ADAM9',\n",
       " 'ADAMDEC1',\n",
       " 'ADAMTS1',\n",
       " 'ADAMTS12',\n",
       " 'ADAMTS13',\n",
       " 'ADAMTS14',\n",
       " 'ADAMTS15',\n",
       " 'ADAMTS16',\n",
       " 'ADAMTS2',\n",
       " 'ADAMTS4',\n",
       " 'ADAMTS7',\n",
       " 'ADAMTSL1',\n",
       " 'ADAMTSL2',\n",
       " 'ADAMTSL3',\n",
       " 'ADAMTSL4',\n",
       " 'ADAMTSL5',\n",
       " 'ADAP1',\n",
       " 'ADAP2',\n",
       " 'ADAR',\n",
       " 'ADARB1',\n",
       " 'ADAT1',\n",
       " 'ADAT2',\n",
       " 'ADAT3',\n",
       " 'ADCK1',\n",
       " 'ADCK5',\n",
       " 'ADCY3',\n",
       " 'ADCY4',\n",
       " 'ADCY6',\n",
       " 'ADCY7',\n",
       " 'ADCY9',\n",
       " 'ADD1',\n",
       " 'ADD2',\n",
       " 'ADD3',\n",
       " 'ADGB',\n",
       " 'ADGRA2',\n",
       " 'ADGRA3',\n",
       " 'ADGRE2',\n",
       " 'ADGRE3',\n",
       " 'ADGRE5',\n",
       " 'ADGRF1',\n",
       " 'ADGRF4',\n",
       " 'ADGRF5',\n",
       " 'ADGRG1',\n",
       " 'ADGRG2',\n",
       " 'ADGRG3',\n",
       " 'ADGRG5',\n",
       " 'ADGRG6',\n",
       " 'ADGRL1',\n",
       " 'ADGRL2',\n",
       " 'ADGRL4',\n",
       " 'ADGRV1',\n",
       " 'ADH1A',\n",
       " 'ADH1B',\n",
       " 'ADH1C',\n",
       " 'ADH4',\n",
       " 'ADH5',\n",
       " 'ADH6',\n",
       " 'ADH7',\n",
       " 'ADHFE1',\n",
       " 'ADI1',\n",
       " 'ADIPOQ',\n",
       " 'ADIPOR1',\n",
       " 'ADIRF',\n",
       " 'ADK',\n",
       " 'ADM',\n",
       " 'ADNP',\n",
       " 'ADNP2',\n",
       " 'ADO',\n",
       " 'ADPGK',\n",
       " 'ADPRH',\n",
       " 'ADPRHL1',\n",
       " 'ADPRHL2',\n",
       " 'ADPRM',\n",
       " 'ADRA2A',\n",
       " 'ADRM1',\n",
       " 'ADSL',\n",
       " 'ADSS',\n",
       " 'ADSSL1',\n",
       " 'AEBP1',\n",
       " 'AEBP2',\n",
       " 'AES',\n",
       " 'AFAP1',\n",
       " 'AFAP1L1',\n",
       " 'AFAP1L2',\n",
       " 'AFDN',\n",
       " 'AFF1',\n",
       " 'AFF4',\n",
       " 'AFG1L',\n",
       " 'AFG3L2',\n",
       " 'AFM',\n",
       " 'AFMID',\n",
       " 'AFTPH',\n",
       " 'AGA',\n",
       " 'AGAP1',\n",
       " 'AGAP2',\n",
       " 'AGAP3',\n",
       " 'AGBL1',\n",
       " 'AGER',\n",
       " 'AGFG1',\n",
       " 'AGFG2',\n",
       " 'AGGF1',\n",
       " 'AGK',\n",
       " 'AGL',\n",
       " 'AGMAT',\n",
       " 'AGO1',\n",
       " 'AGO2',\n",
       " 'AGO3',\n",
       " 'AGO4',\n",
       " 'AGPAT1',\n",
       " 'AGPAT2',\n",
       " 'AGPAT3',\n",
       " 'AGPAT4',\n",
       " 'AGPAT5',\n",
       " 'AGPS',\n",
       " 'AGR2',\n",
       " 'AGR3',\n",
       " 'AGRN',\n",
       " 'AGT',\n",
       " 'AGTPBP1',\n",
       " 'AGTRAP',\n",
       " 'AGXT',\n",
       " 'AHCTF1',\n",
       " 'AHCY',\n",
       " 'AHCYL1',\n",
       " 'AHCYL2',\n",
       " 'AHDC1',\n",
       " 'AHNAK',\n",
       " 'AHNAK2',\n",
       " 'AHR',\n",
       " 'AHSA1',\n",
       " 'AHSG',\n",
       " 'AHSP',\n",
       " 'AIDA',\n",
       " 'AIF1',\n",
       " 'AIF1L',\n",
       " 'AIFM1',\n",
       " 'AIFM2',\n",
       " 'AIG1',\n",
       " 'AIM2',\n",
       " 'AIMP1',\n",
       " 'AIMP2',\n",
       " 'AIP',\n",
       " 'AJAP1',\n",
       " 'AJUBA',\n",
       " 'AK1',\n",
       " 'AK2',\n",
       " 'AK3',\n",
       " 'AK4',\n",
       " 'AK5',\n",
       " 'AK6',\n",
       " 'AKAP1',\n",
       " 'AKAP10',\n",
       " 'AKAP11',\n",
       " 'AKAP12',\n",
       " 'AKAP13',\n",
       " 'AKAP17A',\n",
       " 'AKAP5',\n",
       " 'AKAP6',\n",
       " 'AKAP7',\n",
       " 'AKAP8',\n",
       " 'AKAP8L',\n",
       " 'AKAP9',\n",
       " 'AKNA',\n",
       " 'AKR1A1',\n",
       " 'AKR1B1',\n",
       " 'AKR1B10',\n",
       " 'AKR1B15',\n",
       " 'AKR1C1',\n",
       " 'AKR1C2',\n",
       " 'AKR1C3',\n",
       " 'AKR1C4',\n",
       " 'AKR7A2',\n",
       " 'AKR7A3',\n",
       " 'AKT1',\n",
       " 'AKT1S1',\n",
       " 'AKT2',\n",
       " 'AKT3',\n",
       " 'AKTIP',\n",
       " 'ALAD',\n",
       " 'ALAS1',\n",
       " 'ALB',\n",
       " 'ALCAM',\n",
       " 'ALDH16A1',\n",
       " 'ALDH18A1',\n",
       " 'ALDH1A1',\n",
       " 'ALDH1A2',\n",
       " 'ALDH1A3',\n",
       " 'ALDH1B1',\n",
       " 'ALDH1L1',\n",
       " 'ALDH1L2',\n",
       " 'ALDH2',\n",
       " 'ALDH3A1',\n",
       " 'ALDH3A2',\n",
       " 'ALDH3B1',\n",
       " 'ALDH4A1',\n",
       " 'ALDH5A1',\n",
       " 'ALDH6A1',\n",
       " 'ALDH7A1',\n",
       " 'ALDH8A1',\n",
       " 'ALDH9A1',\n",
       " 'ALDOA',\n",
       " 'ALDOB',\n",
       " 'ALDOC',\n",
       " 'ALG1',\n",
       " 'ALG10B',\n",
       " 'ALG11',\n",
       " 'ALG12',\n",
       " 'ALG13',\n",
       " 'ALG14',\n",
       " 'ALG2',\n",
       " 'ALG3',\n",
       " 'ALG5',\n",
       " 'ALG6',\n",
       " 'ALG8',\n",
       " 'ALG9',\n",
       " 'ALKAL2',\n",
       " 'ALKBH1',\n",
       " 'ALKBH2',\n",
       " 'ALKBH3',\n",
       " 'ALKBH4',\n",
       " 'ALKBH5',\n",
       " 'ALKBH6',\n",
       " 'ALKBH7',\n",
       " 'ALKBH8',\n",
       " 'ALMS1',\n",
       " 'ALOX12',\n",
       " 'ALOX12B',\n",
       " 'ALOX15',\n",
       " 'ALOX15B',\n",
       " 'ALOX5',\n",
       " 'ALOX5AP',\n",
       " 'ALOXE3',\n",
       " 'ALPK1',\n",
       " 'ALPK3',\n",
       " 'ALPL',\n",
       " 'ALS2',\n",
       " 'ALS2CL',\n",
       " 'ALYREF',\n",
       " 'AMACR',\n",
       " 'AMBP',\n",
       " 'AMBRA1',\n",
       " 'AMD1',\n",
       " 'AMDHD2',\n",
       " 'AMER1',\n",
       " 'AMFR',\n",
       " 'AMIGO2',\n",
       " 'AMMECR1',\n",
       " 'AMN1',\n",
       " 'AMOT',\n",
       " 'AMOTL1',\n",
       " 'AMOTL2',\n",
       " 'AMPD1',\n",
       " 'AMPD2',\n",
       " 'AMPD3',\n",
       " 'AMPH',\n",
       " 'AMT',\n",
       " 'AMY1B',\n",
       " 'AMZ2',\n",
       " 'ANAPC1',\n",
       " 'ANAPC10',\n",
       " 'ANAPC11',\n",
       " 'ANAPC13',\n",
       " 'ANAPC15',\n",
       " 'ANAPC16',\n",
       " 'ANAPC2',\n",
       " 'ANAPC4',\n",
       " 'ANAPC5',\n",
       " 'ANAPC7',\n",
       " 'ANG',\n",
       " 'ANGEL1',\n",
       " 'ANGEL2',\n",
       " 'ANGPT2',\n",
       " 'ANGPTL1',\n",
       " 'ANGPTL2',\n",
       " 'ANGPTL4',\n",
       " 'ANGPTL7',\n",
       " 'ANK1',\n",
       " 'ANK2',\n",
       " 'ANK3',\n",
       " 'ANKFY1',\n",
       " 'ANKHD1',\n",
       " 'ANKHD1-EIF4EBP3',\n",
       " 'ANKIB1',\n",
       " 'ANKLE2',\n",
       " 'ANKMY2',\n",
       " 'ANKRA2',\n",
       " 'ANKRD1',\n",
       " 'ANKRD10',\n",
       " 'ANKRD11',\n",
       " 'ANKRD12',\n",
       " 'ANKRD13A',\n",
       " 'ANKRD13D',\n",
       " 'ANKRD16',\n",
       " 'ANKRD17',\n",
       " 'ANKRD2',\n",
       " 'ANKRD22',\n",
       " 'ANKRD23',\n",
       " 'ANKRD26',\n",
       " 'ANKRD27',\n",
       " 'ANKRD28',\n",
       " 'ANKRD29',\n",
       " 'ANKRD35',\n",
       " 'ANKRD36C',\n",
       " 'ANKRD39',\n",
       " 'ANKRD40',\n",
       " 'ANKRD44',\n",
       " 'ANKRD46',\n",
       " 'ANKRD49',\n",
       " 'ANKRD50',\n",
       " 'ANKRD52',\n",
       " 'ANKRD54',\n",
       " 'ANKRD62',\n",
       " 'ANKRD9',\n",
       " 'ANKS1A',\n",
       " 'ANKS1B',\n",
       " 'ANKS3',\n",
       " 'ANKS6',\n",
       " 'ANKZF1',\n",
       " 'ANLN',\n",
       " 'ANO1',\n",
       " 'ANO10',\n",
       " 'ANO5',\n",
       " 'ANO6',\n",
       " 'ANO8',\n",
       " 'ANO9',\n",
       " 'ANOS1',\n",
       " 'ANP32A',\n",
       " 'ANP32B',\n",
       " 'ANP32E',\n",
       " 'ANPEP',\n",
       " 'ANTXR1',\n",
       " 'ANTXR2',\n",
       " 'ANXA1',\n",
       " 'ANXA11',\n",
       " 'ANXA13',\n",
       " 'ANXA2',\n",
       " 'ANXA3',\n",
       " 'ANXA4',\n",
       " 'ANXA5',\n",
       " 'ANXA6',\n",
       " 'ANXA7',\n",
       " 'ANXA8',\n",
       " 'ANXA8L1',\n",
       " 'ANXA9',\n",
       " 'AOAH',\n",
       " 'AOC1',\n",
       " 'AOC3',\n",
       " 'AOX1',\n",
       " 'AP1AR',\n",
       " 'AP1B1',\n",
       " 'AP1G1',\n",
       " 'AP1G2',\n",
       " 'AP1M1',\n",
       " 'AP1M2',\n",
       " 'AP1S1',\n",
       " 'AP1S2',\n",
       " 'AP1S3',\n",
       " 'AP2A1',\n",
       " 'AP2A2',\n",
       " 'AP2B1',\n",
       " 'AP2M1',\n",
       " 'AP2S1',\n",
       " 'AP3B1',\n",
       " 'AP3B2',\n",
       " 'AP3D1',\n",
       " 'AP3M1',\n",
       " 'AP3M2',\n",
       " 'AP3S1',\n",
       " 'AP3S2',\n",
       " 'AP4B1',\n",
       " 'AP4E1',\n",
       " 'AP4M1',\n",
       " 'AP4S1',\n",
       " 'AP5B1',\n",
       " 'AP5M1',\n",
       " 'AP5S1',\n",
       " 'AP5Z1',\n",
       " 'APAF1',\n",
       " 'APBA1',\n",
       " 'APBA2',\n",
       " 'APBA3',\n",
       " 'APBB1',\n",
       " 'APBB1IP',\n",
       " 'APBB2',\n",
       " 'APC',\n",
       " 'APCDD1',\n",
       " 'APCS',\n",
       " 'APEH',\n",
       " 'APEX1',\n",
       " 'APEX2',\n",
       " 'APH1A',\n",
       " 'API5',\n",
       " 'APIP',\n",
       " 'APLF',\n",
       " 'APLP2',\n",
       " 'APMAP',\n",
       " 'APOA1',\n",
       " 'APOA2',\n",
       " 'APOA4',\n",
       " 'APOA5',\n",
       " 'APOB',\n",
       " 'APOBEC2',\n",
       " 'APOBEC3A_B',\n",
       " 'APOBEC3B',\n",
       " 'APOBEC3C',\n",
       " 'APOBEC3D',\n",
       " 'APOBEC3F',\n",
       " 'APOBEC3G',\n",
       " 'APOBR',\n",
       " 'APOC1',\n",
       " 'APOC2',\n",
       " 'APOC3',\n",
       " 'APOC4',\n",
       " 'APOD',\n",
       " 'APOE',\n",
       " 'APOF',\n",
       " 'APOH',\n",
       " 'APOL2',\n",
       " 'APOL3',\n",
       " 'APOL4',\n",
       " 'APOL6',\n",
       " 'APOM',\n",
       " 'APOO',\n",
       " 'APOOL',\n",
       " 'APOPT1',\n",
       " 'APP',\n",
       " 'APPBP2',\n",
       " 'APPL1',\n",
       " 'APPL2',\n",
       " 'APRT',\n",
       " 'APTX',\n",
       " 'AQP1',\n",
       " 'AQP10',\n",
       " 'AQP3',\n",
       " 'AQP4',\n",
       " 'AQP5',\n",
       " 'AQR',\n",
       " 'ARAF',\n",
       " 'ARAP1',\n",
       " 'ARAP2',\n",
       " 'ARAP3',\n",
       " 'ARCN1',\n",
       " 'AREG',\n",
       " 'ARF1',\n",
       " 'ARF3',\n",
       " 'ARF4',\n",
       " 'ARF5',\n",
       " 'ARF6',\n",
       " 'ARFGAP1',\n",
       " 'ARFGAP2',\n",
       " 'ARFGAP3',\n",
       " 'ARFGEF1',\n",
       " 'ARFGEF2',\n",
       " 'ARFGEF3',\n",
       " 'ARFIP1',\n",
       " 'ARFIP2',\n",
       " 'ARFRP1',\n",
       " 'ARG1',\n",
       " 'ARG2',\n",
       " 'ARGLU1',\n",
       " 'ARHGAP1',\n",
       " 'ARHGAP10',\n",
       " 'ARHGAP11A',\n",
       " 'ARHGAP12',\n",
       " 'ARHGAP15',\n",
       " 'ARHGAP17',\n",
       " 'ARHGAP18',\n",
       " 'ARHGAP19',\n",
       " 'ARHGAP21',\n",
       " 'ARHGAP23',\n",
       " 'ARHGAP24',\n",
       " 'ARHGAP25',\n",
       " 'ARHGAP26',\n",
       " 'ARHGAP27',\n",
       " 'ARHGAP28',\n",
       " 'ARHGAP29',\n",
       " 'ARHGAP30',\n",
       " 'ARHGAP31',\n",
       " 'ARHGAP32',\n",
       " 'ARHGAP35',\n",
       " 'ARHGAP39',\n",
       " 'ARHGAP4',\n",
       " 'ARHGAP42',\n",
       " 'ARHGAP45',\n",
       " 'ARHGAP5',\n",
       " 'ARHGAP6',\n",
       " 'ARHGAP9',\n",
       " 'ARHGDIA',\n",
       " 'ARHGDIB',\n",
       " 'ARHGEF1',\n",
       " 'ARHGEF10',\n",
       " 'ARHGEF10L',\n",
       " 'ARHGEF11',\n",
       " 'ARHGEF12',\n",
       " 'ARHGEF15',\n",
       " 'ARHGEF16',\n",
       " 'ARHGEF17',\n",
       " 'ARHGEF18',\n",
       " 'ARHGEF19',\n",
       " 'ARHGEF2',\n",
       " 'ARHGEF25',\n",
       " 'ARHGEF26',\n",
       " 'ARHGEF28',\n",
       " 'ARHGEF3',\n",
       " 'ARHGEF35',\n",
       " 'ARHGEF37',\n",
       " 'ARHGEF39',\n",
       " 'ARHGEF4',\n",
       " 'ARHGEF40',\n",
       " 'ARHGEF5',\n",
       " 'ARHGEF6',\n",
       " 'ARHGEF7',\n",
       " 'ARID1A',\n",
       " 'ARID1B',\n",
       " 'ARID2',\n",
       " 'ARID3A',\n",
       " 'ARID3B',\n",
       " 'ARID4A',\n",
       " 'ARID4B',\n",
       " 'ARID5B',\n",
       " 'ARIH1',\n",
       " 'ARIH2',\n",
       " 'ARL1',\n",
       " 'ARL11',\n",
       " 'ARL13A',\n",
       " 'ARL13B',\n",
       " 'ARL14EP',\n",
       " 'ARL15',\n",
       " 'ARL16',\n",
       " 'ARL2',\n",
       " 'ARL2BP',\n",
       " 'ARL3',\n",
       " 'ARL4C',\n",
       " 'ARL5A',\n",
       " 'ARL5B',\n",
       " 'ARL6',\n",
       " 'ARL6IP1',\n",
       " 'ARL6IP4',\n",
       " 'ARL6IP5',\n",
       " 'ARL6IP6',\n",
       " 'ARL8A',\n",
       " 'ARL8B',\n",
       " 'ARMC1',\n",
       " 'ARMC10',\n",
       " 'ARMC5',\n",
       " 'ARMC6',\n",
       " 'ARMC7',\n",
       " 'ARMC8',\n",
       " 'ARMC9',\n",
       " 'ARMCX1',\n",
       " 'ARMCX2',\n",
       " 'ARMCX3',\n",
       " 'ARMCX5',\n",
       " 'ARMCX6',\n",
       " 'ARMH3',\n",
       " 'ARMT1',\n",
       " 'ARNT',\n",
       " 'ARNTL',\n",
       " 'ARNTL2',\n",
       " 'ARPC1A',\n",
       " 'ARPC1B',\n",
       " 'ARPC2',\n",
       " 'ARPC3',\n",
       " 'ARPC4',\n",
       " 'ARPC5',\n",
       " 'ARPC5L',\n",
       " 'ARPIN',\n",
       " 'ARPP19',\n",
       " 'ARPP21',\n",
       " 'ARRB1',\n",
       " 'ARRB2',\n",
       " 'ARRDC1',\n",
       " 'ARSA',\n",
       " 'ARSB',\n",
       " 'ARSD',\n",
       " 'ARSE',\n",
       " 'ARSF',\n",
       " 'ARSK',\n",
       " 'ART3',\n",
       " 'ARV1',\n",
       " 'ARVCF',\n",
       " 'AS3MT',\n",
       " 'ASAH1',\n",
       " 'ASAP1',\n",
       " 'ASAP2',\n",
       " 'ASAP3',\n",
       " 'ASB1',\n",
       " 'ASB10',\n",
       " 'ASB13',\n",
       " 'ASB2',\n",
       " 'ASB6',\n",
       " 'ASB7',\n",
       " 'ASB9',\n",
       " 'ASCC1',\n",
       " 'ASCC2',\n",
       " 'ASCC3',\n",
       " 'ASCL2',\n",
       " 'ASDURF',\n",
       " 'ASF1A',\n",
       " 'ASF1B',\n",
       " 'ASGR2',\n",
       " 'ASH1L',\n",
       " 'ASH2L',\n",
       " 'ASL',\n",
       " 'ASMTL',\n",
       " 'ASNA1',\n",
       " 'ASNS',\n",
       " 'ASNSD1',\n",
       " 'ASPA',\n",
       " 'ASPG',\n",
       " 'ASPH',\n",
       " 'ASPHD2',\n",
       " 'ASPM',\n",
       " 'ASPN',\n",
       " 'ASPRV1',\n",
       " 'ASPSCR1',\n",
       " 'ASRGL1',\n",
       " 'ASS1',\n",
       " 'ASTE1',\n",
       " 'ASXL1',\n",
       " 'ASXL2',\n",
       " 'ATAD1',\n",
       " 'ATAD2',\n",
       " 'ATAD2B',\n",
       " 'ATAD3A',\n",
       " 'ATAD3B',\n",
       " 'ATAD5',\n",
       " 'ATAT1',\n",
       " 'ATE1',\n",
       " 'ATF1',\n",
       " 'ATF2',\n",
       " 'ATF3',\n",
       " 'ATF6',\n",
       " 'ATF6B',\n",
       " 'ATF7',\n",
       " 'ATF7IP',\n",
       " 'ATG10',\n",
       " 'ATG101',\n",
       " 'ATG12',\n",
       " 'ATG13',\n",
       " 'ATG14',\n",
       " 'ATG16L1',\n",
       " 'ATG2A',\n",
       " 'ATG2B',\n",
       " 'ATG3',\n",
       " 'ATG4A',\n",
       " 'ATG4B',\n",
       " 'ATG4C',\n",
       " 'ATG4D',\n",
       " 'ATG5',\n",
       " 'ATG7',\n",
       " 'ATG9A',\n",
       " 'ATIC',\n",
       " 'ATL1',\n",
       " 'ATL2',\n",
       " 'ATL3',\n",
       " 'ATM',\n",
       " 'ATN1',\n",
       " 'ATOX1',\n",
       " 'ATP10A',\n",
       " 'ATP10D',\n",
       " 'ATP11A',\n",
       " 'ATP11B',\n",
       " 'ATP11C',\n",
       " 'ATP12A',\n",
       " 'ATP13A1',\n",
       " 'ATP13A2',\n",
       " 'ATP13A3',\n",
       " 'ATP13A4',\n",
       " 'ATP1A1',\n",
       " 'ATP1A2',\n",
       " 'ATP1A3',\n",
       " 'ATP1B1',\n",
       " 'ATP1B2',\n",
       " 'ATP1B3',\n",
       " 'ATP1B4',\n",
       " 'ATP23',\n",
       " 'ATP2A1',\n",
       " 'ATP2A2',\n",
       " 'ATP2A3',\n",
       " 'ATP2B1',\n",
       " 'ATP2B3',\n",
       " 'ATP2B4',\n",
       " 'ATP2C1',\n",
       " 'ATP2C2',\n",
       " 'ATP4A',\n",
       " 'ATP5F1A',\n",
       " 'ATP5F1B',\n",
       " 'ATP5F1C',\n",
       " 'ATP5F1D',\n",
       " 'ATP5F1E',\n",
       " 'ATP5IF1',\n",
       " 'ATP5MC3',\n",
       " 'ATP5MD',\n",
       " 'ATP5ME',\n",
       " 'ATP5MF',\n",
       " 'ATP5MF-PTCD1',\n",
       " 'ATP5MG',\n",
       " 'ATP5MPL',\n",
       " 'ATP5PB',\n",
       " 'ATP5PD',\n",
       " 'ATP5PF',\n",
       " 'ATP5PO',\n",
       " 'ATP5S',\n",
       " 'ATP6AP1',\n",
       " 'ATP6AP2',\n",
       " 'ATP6V0A1',\n",
       " 'ATP6V0A2',\n",
       " 'ATP6V0A4',\n",
       " 'ATP6V0C',\n",
       " 'ATP6V0D1',\n",
       " 'ATP6V0D2',\n",
       " 'ATP6V1A',\n",
       " 'ATP6V1B1',\n",
       " 'ATP6V1B2',\n",
       " 'ATP6V1C1',\n",
       " 'ATP6V1C2',\n",
       " 'ATP6V1D',\n",
       " 'ATP6V1E1',\n",
       " 'ATP6V1F',\n",
       " 'ATP6V1G1',\n",
       " 'ATP6V1G2',\n",
       " 'ATP6V1H',\n",
       " 'ATP7A',\n",
       " 'ATP8A1',\n",
       " 'ATP8B1',\n",
       " 'ATP8B2',\n",
       " 'ATP8B4',\n",
       " 'ATP9A',\n",
       " 'ATP9B',\n",
       " 'ATPAF1',\n",
       " 'ATPAF2',\n",
       " 'ATR',\n",
       " 'ATRAID',\n",
       " 'ATRIP',\n",
       " 'ATRN',\n",
       " 'ATRX',\n",
       " 'ATXN1',\n",
       " 'ATXN10',\n",
       " 'ATXN1L',\n",
       " 'ATXN2',\n",
       " 'ATXN2L',\n",
       " 'ATXN3',\n",
       " 'ATXN7',\n",
       " 'ATXN7L2',\n",
       " 'ATXN7L3',\n",
       " 'ATXN7L3B',\n",
       " 'AUH',\n",
       " 'AUP1',\n",
       " 'AURKA',\n",
       " 'AURKAIP1',\n",
       " 'AURKB',\n",
       " 'AUTS2',\n",
       " 'AVEN',\n",
       " 'AVIL',\n",
       " 'AVL9',\n",
       " 'AXIN1',\n",
       " 'AXL',\n",
       " 'AZGP1',\n",
       " 'AZI2',\n",
       " 'AZU1',\n",
       " 'B2M',\n",
       " 'B3GALNT1',\n",
       " 'B3GALNT2',\n",
       " 'B3GALT5',\n",
       " 'B3GALT6',\n",
       " 'B3GAT3',\n",
       " 'B3GLCT',\n",
       " 'B3GNT2',\n",
       " 'B3GNT3',\n",
       " 'B3GNT5',\n",
       " 'B3GNT6',\n",
       " 'B3GNT7',\n",
       " 'B3GNT8',\n",
       " 'B3GNT9',\n",
       " 'B3GNTL1',\n",
       " 'B4GALNT3',\n",
       " 'B4GALT1',\n",
       " 'B4GALT2',\n",
       " 'B4GALT3',\n",
       " 'B4GALT4',\n",
       " 'B4GALT5',\n",
       " 'B4GALT7',\n",
       " 'B4GAT1',\n",
       " 'B9D1',\n",
       " 'B9D2',\n",
       " 'BABAM1',\n",
       " 'BABAM2',\n",
       " 'BACE2',\n",
       " 'BACH1',\n",
       " 'BAD',\n",
       " 'BAG1',\n",
       " 'BAG2',\n",
       " 'BAG3',\n",
       " 'BAG4',\n",
       " 'BAG5',\n",
       " 'BAG6',\n",
       " 'BAHD1',\n",
       " 'BAIAP2',\n",
       " 'BAIAP2L1',\n",
       " 'BAIAP2L2',\n",
       " 'BAK1',\n",
       " 'BANF1',\n",
       " 'BANK1',\n",
       " 'BANP',\n",
       " 'BAP1',\n",
       " 'BARD1',\n",
       " 'BARX2',\n",
       " 'BASP1',\n",
       " 'BAX',\n",
       " 'BAZ1A',\n",
       " 'BAZ1B',\n",
       " 'BAZ2A',\n",
       " 'BAZ2B',\n",
       " 'BBIP1',\n",
       " 'BBOX1',\n",
       " ...]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "h_prot = h.get_proteomics()\n",
    "h_prot_list = list(h_prot.columns)\n",
    "\n",
    "h_del_wt = all_prot_format_df(h, h_prot_list)\n",
    "h_del_wt = h_del_wt.dropna(axis='columns', how='all')\n",
    "h_del_wt.head()\n",
    "h_prot_list.remove('ADIPOR2')\n",
    "h_prot_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:325: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:333: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:256: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "              Comparison       P_Value\n",
      "0        NLN_proteomics  1.113430e-09\n",
      "1     AKR1C3_proteomics  1.032977e-08\n",
      "2     AKR1C1_proteomics  6.931627e-08\n",
      "3       ADI1_proteomics  1.601880e-07\n",
      "4      EPHX1_proteomics  3.899394e-07\n",
      "..                  ...           ...\n",
      "523    FLAD1_proteomics  2.232897e-03\n",
      "524    GSTM4_proteomics  2.234528e-03\n",
      "525   ELOVL5_proteomics  2.239241e-03\n",
      "526  SUPT16H_proteomics  2.241138e-03\n",
      "527  ANKRD28_proteomics  2.243254e-03\n",
      "\n",
      "[528 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:325: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:333: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:256: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(h_del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "h_sig = u.wrap_ttest(h_del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if h_sig is not None:\n",
    "    h_sig_list = list(h_sig.Comparison)\n",
    "else: \n",
    "    h_sig_list = None\n",
    "print('significant pvals: \\n',h_sig)\n",
    "\n",
    "# Get all pvals\n",
    "h_pval = u.wrap_ttest(h_del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "h_pval = h_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Hnscc_P_Value'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'CPNE6_proteomics'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\pandas\\core\\indexes\\base.py\u001b[0m in \u001b[0;36mget_value\u001b[1;34m(self, series, key)\u001b[0m\n\u001b[0;32m   4410\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 4411\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mlibindex\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_value_at\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   4412\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mIndexError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.get_value_at\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.get_value_at\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\util.pxd\u001b[0m in \u001b[0;36mpandas._libs.util.get_value_at\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\util.pxd\u001b[0m in \u001b[0;36mpandas._libs.util.validate_indexer\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m: 'str' object cannot be interpreted as an integer",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-58-0f5fd3c69b30>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;31m#h_prot_list = h_prot_list.remove('ADIPOR2')\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[0mh_med\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mget_change_in_medians_df\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mh\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"Hnscc\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mh_del_wt\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mh_prot_list\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      3\u001b[0m \u001b[0mh_med\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-57-1c933d84fcc4>\u001b[0m in \u001b[0;36mget_change_in_medians_df\u001b[1;34m(cancer_object, cancer_name, del_wt_df, all_prot_list)\u001b[0m\n\u001b[0;32m     19\u001b[0m     \u001b[1;31m# Correlation: + is mutant up compared to wt, - is mutant down\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     20\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mprot\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mall_prot_list\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 21\u001b[1;33m         \u001b[0mdif\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdel_med\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mprot\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m'_proteomics'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m-\u001b[0m \u001b[0mwt_med\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mprot\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m'_proteomics'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     22\u001b[0m         \u001b[0mmed_dict\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mprot\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m'_proteomics'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdif\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     23\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\pandas\\core\\series.py\u001b[0m in \u001b[0;36m__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m    869\u001b[0m         \u001b[0mkey\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    870\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 871\u001b[1;33m             \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    872\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    873\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\pandas\\core\\indexes\\base.py\u001b[0m in \u001b[0;36mget_value\u001b[1;34m(self, series, key)\u001b[0m\n\u001b[0;32m   4417\u001b[0m                     \u001b[1;32mraise\u001b[0m \u001b[0mInvalidIndexError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4418\u001b[0m                 \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 4419\u001b[1;33m                     \u001b[1;32mraise\u001b[0m \u001b[0me1\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   4420\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4421\u001b[0m                 \u001b[1;32mraise\u001b[0m \u001b[0me1\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\pandas\\core\\indexes\\base.py\u001b[0m in \u001b[0;36mget_value\u001b[1;34m(self, series, key)\u001b[0m\n\u001b[0;32m   4403\u001b[0m         \u001b[0mk\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_convert_scalar_indexer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mk\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkind\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"getitem\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4404\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 4405\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mk\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtz\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mgetattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mseries\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"tz\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   4406\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me1\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4407\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m>\u001b[0m \u001b[1;36m0\u001b[0m \u001b[1;32mand\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mholds_integer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_boolean\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_value\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_value\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'CPNE6_proteomics'"
     ]
    }
   ],
   "source": [
    "h_med = get_change_in_medians_df(h, \"Hnscc\", h_del_wt, h_prot_list)\n",
    "h_med"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Luad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 176 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "12.757333    1\n",
       "12.543358    1\n",
       "12.558793    1\n",
       "12.593851    1\n",
       "12.817394    1\n",
       "12.437753    1\n",
       "Name: ADIPOR2_proteomics, dtype: int64"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_and_mutations = h.join_omics_to_mutations(\n",
    "            mutations_genes = ['PTEN'], omics_df_name = 'proteomics', omics_genes = 'ADIPOR2')\n",
    "#h_del_wt[['ADIPOR2_proteomics']]\n",
    "prot_and_mutations['ADIPOR2_proteomics'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'\n",
    "l_prot = l.get_proteomics()\n",
    "l_prot = u.reduce_multiindex(l_prot, levels_to_drop = 1)\n",
    "l_prot_list = list(l_prot.columns)\n",
    "\n",
    "l_del_wt = all_prot_format_df(l, l_prot_list)\n",
    "l_del_wt = l_del_wt.dropna(axis='columns', how='all')\n",
    "l_del_wt = l_del_wt.loc[:,~l_del_wt.columns.duplicated()]\n",
    "l_del_wt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Name\n",
       "A1BG_proteomics      27.720639\n",
       "A1CF_proteomics      18.301779\n",
       "A2M_proteomics       28.896864\n",
       "A2ML1_proteomics     26.751857\n",
       "A4GALT_proteomics    17.887410\n",
       "                       ...    \n",
       "ZXDC_proteomics      19.610571\n",
       "ZYG11B_proteomics    22.596113\n",
       "ZYX_proteomics       27.872817\n",
       "ZZEF1_proteomics     24.592125\n",
       "ZZZ3_proteomics      20.055864\n",
       "Length: 11721, dtype: float64"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "d = h_del_wt[h_del_wt.Mutation == \"Deletion\"]\n",
    "wt = h_del_wt[h_del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "wt_med"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = list(l_del_wt.columns[:-2])\n",
    "\n",
    "# Get only sig sites\n",
    "l_sig = u.wrap_ttest(l_del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if l_sig is not None:\n",
    "    l_sig_list = list(l_sig.Comparison)\n",
    "else: \n",
    "    l_sig_list = None\n",
    "print('significant pvals: \\n',l_sig)\n",
    "\n",
    "# Get all pvals\n",
    "l_pval = u.wrap_ttest(l_del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "l_pval = l_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Luad_P_Value'})\n",
    "# isoforms for some proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Isoform code - fix later\n",
    "#del_wt.replace(to_replace = r'_NP_.*$', value = '', regex = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "l_med = get_change_in_medians_df(l, \"Luad\", l_del_wt, l_prot_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "l_merged = l_pval.merge(l_med, on='Proteomics',how='outer')\n",
    "\n",
    "# Create csv\n",
    "l_merged.to_csv(root+R'\\luad_pval_medians.csv',index=False)\n",
    "l_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lscc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'\n",
    "ls_prot = ls.get_proteomics()\n",
    "ls_prot = u.reduce_multiindex(ls_prot, levels_to_drop = 1)\n",
    "ls_prot_list = list(ls_prot.columns)\n",
    "\n",
    "ls_del_wt = all_prot_format_df(ls, ls_prot_list)\n",
    "ls_del_wt = ls_del_wt.dropna(axis='columns', how='all')\n",
    "ls_del_wt = ls_del_wt.loc[:,~ls_del_wt.columns.duplicated()]\n",
    "ls_del_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = list(ls_del_wt.columns[:-2])\n",
    "\n",
    "# Get only sig sites\n",
    "ls_sig = u.wrap_ttest(ls_del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if ls_sig is not None:\n",
    "    ls_sig_list = list(ls_sig.Comparison)\n",
    "else: \n",
    "    ls_sig_list = None\n",
    "print('significant pvals: \\n',ls_sig)\n",
    "\n",
    "# Get all pvals\n",
    "ls_pval = u.wrap_ttest(ls_del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "ls_pval = ls_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Lscc_P_Value'})\n",
    "#ls_pval # isoforms for some proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ls_med = get_change_in_medians_df(ls, \"Lscc\", ls_del_wt, ls_prot_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ls_merged = ls_pval.merge(ls_med, on='Proteomics',how='outer')\n",
    "\n",
    "# Create csv\n",
    "ls_merged.to_csv(root+R'\\lscc_pval_medians.csv',index=False)\n",
    "ls_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Endometrial data set does not have enough cnv deletions to perform a t-test, however the data set does have enough truncation type mutations (nonsense and frame shifts). Different code is needed to create the data frame for Endometrial."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "en = cptac.Endometrial()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'\n",
    "prot = en.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "\n",
    "mut_type = en.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = en.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "trunc_wt = merged[get]\n",
    "trunc_wt['Mutation'] = np.where(\n",
    "                trunc_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "print(len(trunc_wt))\n",
    "trunc_wt = trunc_wt.dropna(axis = 'columns',thresh = 10)\n",
    "len(trunc_wt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = list(trunc_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "e_sig = u.wrap_ttest(trunc_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if e_sig is not None:\n",
    "    e_sig_list = list(e_sig.Comparison)\n",
    "else: \n",
    "    e_sig_list = None\n",
    "print('significant pvals: \\n',e_sig)\n",
    "\n",
    "# Get all pvals\n",
    "e_pval = u.wrap_ttest(trunc_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "e_pval = e_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'En_P_Value'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Differences in median with adaption to trunctation mutations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "t = trunc_wt[trunc_wt.Mutation == \"Truncation\"]\n",
    "wt = trunc_wt[trunc_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "trunc_med = t.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "en_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = trunc_med[prot+'_proteomics'] - wt_med[prot+'_proteomics']\n",
    "    en_d[prot+'_proteomics'] = dif\n",
    "\n",
    "en_med = pd.DataFrame.from_dict(en_d, orient='index', columns=['En_Median'])\n",
    "en_med = en_med.reset_index().rename(columns={'index':'Proteomics'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "e_merged = e_pval.merge(en_med, on='Proteomics',how='outer')\n",
    "\n",
    "# Create csv\n",
    "root = R'~\\Github\\WhenMutationsDontMatter\\PTEN\\Step_3_trans_effect\\csv'\n",
    "e_merged.to_csv(root+R'\\endo_pval_medians.csv',index=False)\n",
    "e_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get a list of significant genes in at least one cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Keep only genes significant in > 1 cancer\n",
    "all_sig = []\n",
    "\n",
    "sig_lists = [g_sig_list, e_sig_list, b_sig_list, o_sig_list, c_sig_list, h_sig_list, l_sig_list, ls_sig_list]\n",
    "\n",
    "for next_list in sig_lists:\n",
    "    if next_list is not None:\n",
    "        all_sig.append(next_list)\n",
    "        \n",
    "\n",
    "flat_list = [item for sublist in all_sig for item in sublist] #change list of lists, to just one list\n",
    "sig = list(set(flat_list)) # remove duplicates\n",
    "len(sig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "s = pd.Series(sig)\n",
    "s = s.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "s.to_csv(root+R'\\list_sig_in_at_least_one_cancer.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get a list of significant genes in multiple cancers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sig in multiple (more than 1) cancers\n",
    "mult = list(set(i for i in flat_list if flat_list.count(i) > 1)) # Keep genes sig in more than 1 cancer\n",
    "len(mult)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = pd.Series(mult)\n",
    "m = m.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "m.to_csv(root+R'\\list_sig_in_multiple_cancers.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# see sig in cancer\n",
    "cancer = ['Gbm','En','Brca','Ov','Colon','Hnscc','Luad']\n",
    "i = 0\n",
    "for next_list in sig_lists:\n",
    "\n",
    "    print(cancer[i], ':')\n",
    "    if next_list is not None: \n",
    "        print(len(next_list),'\\n')\n",
    "    #print(next_list, '\\n')\n",
    "    if (i < 6):\n",
    "        i += 1\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# see sig in cancer\n",
    "cancer = ['Gbm','En','Brca','Ov','Colon','Hnscc','Luad']\n",
    "i = 0\n",
    "for next_list in sig_lists:\n",
    "\n",
    "    print(cancer[i], ':')\n",
    "    if next_list is not None: \n",
    "        print(len(next_list),'\\n')\n",
    "        gl = ['NOL10_proteomics']\n",
    "        if gl in next_list:\n",
    "            print('true')\n",
    "    #print(next_list, '\\n')\n",
    "    if (i < 6):\n",
    "        i += 1\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check if protein in sig_df\n",
    "mcm = ['MCM2_proteomics','MCM3_proteomics','MCM4_proteomics','MCM5_proteomics',\n",
    "            'MCM6_proteomics','MCM7_proteomics']\n",
    "rfc = ['RFC2_proteomics','RFC4_proteomics','RFC5_proteomics']\n",
    "mmr = ['MSH2_proteomics','MSH6_proteomics']\n",
    "\n",
    "sig = h_sig.Comparison\n",
    "test = sig.isin(mmr)\n",
    "sig[test]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
