{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make tables: calculate p-values and differential expressions for each cancer "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a dataframe with p-value results from t-tests for all proteins. The dataframe also includes the change in medians between missense mutation and wildtype. Prepare these tables for further analysis by creating csv files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/Lindsey/anaconda3/lib/python3.7/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Params\n",
    "all_prot: list of trans genes\n",
    "utils: from cptac\n",
    "\n",
    "Returns a dataframe with trans proteomics and mutation status (KRAS missense or wt)\n",
    "'''\n",
    "\n",
    "def all_prot_format_df(cancer_object, all_prot, gene_in = 'KRAS', utils = u):\n",
    "    mut_type = cancer_object.get_genotype_all_vars(gene_in)\n",
    "    \n",
    "    # Keep only tumor samples from proteomics\n",
    "    prot_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "        mutations_genes = [gene_in], omics_df_name = 'proteomics', omics_genes = all_prot,\n",
    "        tissue_type = 'tumor') # drop Normal samples\n",
    "        \n",
    "    # Reduce a multiindex \n",
    "    if isinstance(prot_and_mutations.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "        prot_and_mutations = utils.reduce_multiindex(prot_and_mutations, levels_to_drop = 1) \n",
    "\n",
    "    # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "   \n",
    "    prot_and_mutations= prot_and_mutations.replace(r'[\"nonsynonymous SNV\"]', \"Missense_Mutation\", regex = True)\n",
    "    \n",
    "    prot_and_mutations= prot_and_mutations.replace(r'[\"Missense_Mutation\"]', \"Missense_Mutation\", regex = True)\n",
    "    \n",
    "    hotspot = mut_type[(mut_type.Location.str.contains('G12C')| \n",
    "                    mut_type.Location.str.contains('G12V'))]\n",
    "    wt = mut_type[(mut_type.Mutation.str.contains('Wildtype'))]\n",
    "    hotspot_wt = pd.concat([hotspot, wt]) \n",
    "    hotspot_wt = hotspot_wt[['Mutation']] \n",
    "    prot_df = prot_and_mutations.iloc[:,:-4] # Keep only proteomics\n",
    "    merged = prot_df.join(hotspot_wt)\n",
    "        \n",
    "    # Keep only Wildtype and Missense\n",
    "    compare = ['Wildtype_Tumor','Missense_Mutation']\n",
    "    get = merged['Mutation'].isin(compare)\n",
    "    missense_wt = merged[get]\n",
    "    return missense_wt\n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Params\n",
    "cancer_object: Object. Variable for the loaded cancer data set.\n",
    "cancer_name: Str. name to add to the created dataframe.\n",
    "del_wt_df: DataFrame. Only samples with deletions and wildtype for PIK3CA. \n",
    "all_prot_list: List. All proteins in proteomics data frame. \n",
    "\n",
    "Returns a dataframe with the difference in medians between proteomics with PIK3CA wt and del (wt - Missense). \n",
    "'''\n",
    "\n",
    "def get_change_in_medians_df(cancer_object, cancer_name, miss_wt_df, all_prot_list):\n",
    "    m = miss_wt_df[miss_wt_df.Mutation == \"Missense_Mutation\"]\n",
    "    wt = miss_wt_df[miss_wt_df.Mutation == \"Wildtype_Tumor\"]\n",
    "    miss_med = m.median()\n",
    "    wt_med = wt.median()\n",
    "\n",
    "    med_dict = {}\n",
    "    \n",
    "    # Correlation: + is mutant up compared to wt, - is mutant down\n",
    "    for prot in all_prot_list:\n",
    "        if prot in miss_med.index and prot in wt_med.index:\n",
    "            dif = miss_med[prot] - wt_med[prot]\n",
    "            med_dict[prot] = dif\n",
    "        \n",
    "    df = pd.DataFrame.from_dict(med_dict, orient='index', columns=[cancer_name+'_Median'])\n",
    "    df = df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "    \n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Step 1: Create data frames with p-values and differential expressions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each cancer needs a data frame containing only samples that have PTEN cnv deletions and PTEN wildtype with trans proteomics. Use wrap_ttest to run t-tests for all genes in the proteomic data frame. Use get_change_in_medians_df to create the data frame with differential expression values. Merge both data frames. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load in cancer data sets from cptac. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "\n",
    "en = cptac.Endometrial()\n",
    "col = cptac.Colon()\n",
    "luad = cptac.Luad()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 1: Format data frame using all_prot_format_df. This returns a dataframe with proteomics and a Mutation column containing only PTEN cnv deletions and wildtype tumors. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG_proteomics</th>\n",
       "      <th>A1CF_proteomics</th>\n",
       "      <th>A2M_proteomics</th>\n",
       "      <th>AAAS_proteomics</th>\n",
       "      <th>AACS_proteomics</th>\n",
       "      <th>AAGAB_proteomics</th>\n",
       "      <th>AAK1_proteomics</th>\n",
       "      <th>AAMDC_proteomics</th>\n",
       "      <th>AAMP_proteomics</th>\n",
       "      <th>AAR2_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNRD1_proteomics</th>\n",
       "      <th>ZNRF2_proteomics</th>\n",
       "      <th>ZPR1_proteomics</th>\n",
       "      <th>ZRANB2_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01CO001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO006</th>\n",
       "      <td>-1.120</td>\n",
       "      <td>-0.441</td>\n",
       "      <td>-0.347</td>\n",
       "      <td>-0.0029</td>\n",
       "      <td>0.0957</td>\n",
       "      <td>0.3960</td>\n",
       "      <td>-0.0363</td>\n",
       "      <td>-0.549</td>\n",
       "      <td>0.220</td>\n",
       "      <td>0.248</td>\n",
       "      <td>...</td>\n",
       "      <td>0.324</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3940</td>\n",
       "      <td>0.0846</td>\n",
       "      <td>0.001</td>\n",
       "      <td>-0.345</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.658</td>\n",
       "      <td>0.0052</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO008</th>\n",
       "      <td>-1.200</td>\n",
       "      <td>0.160</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>0.1190</td>\n",
       "      <td>-0.0924</td>\n",
       "      <td>0.0187</td>\n",
       "      <td>-0.2140</td>\n",
       "      <td>0.328</td>\n",
       "      <td>-0.282</td>\n",
       "      <td>-0.348</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3840</td>\n",
       "      <td>-0.1680</td>\n",
       "      <td>0.3570</td>\n",
       "      <td>-0.325</td>\n",
       "      <td>0.349</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.821</td>\n",
       "      <td>0.2000</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO013</th>\n",
       "      <td>-1.890</td>\n",
       "      <td>0.112</td>\n",
       "      <td>-0.329</td>\n",
       "      <td>0.6700</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>0.3130</td>\n",
       "      <td>-0.2380</td>\n",
       "      <td>-0.274</td>\n",
       "      <td>-0.554</td>\n",
       "      <td>0.270</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0656</td>\n",
       "      <td>-0.2950</td>\n",
       "      <td>0.463</td>\n",
       "      <td>-0.448</td>\n",
       "      <td>0.0493</td>\n",
       "      <td>-0.904</td>\n",
       "      <td>-0.1580</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO014</th>\n",
       "      <td>-0.523</td>\n",
       "      <td>-0.248</td>\n",
       "      <td>-0.638</td>\n",
       "      <td>0.2890</td>\n",
       "      <td>0.3780</td>\n",
       "      <td>0.8220</td>\n",
       "      <td>0.0018</td>\n",
       "      <td>-1.080</td>\n",
       "      <td>0.498</td>\n",
       "      <td>-0.120</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0467</td>\n",
       "      <td>0.2170</td>\n",
       "      <td>0.2360</td>\n",
       "      <td>0.287</td>\n",
       "      <td>0.346</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.812</td>\n",
       "      <td>-0.0273</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 8067 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        A1BG_proteomics  A1CF_proteomics  A2M_proteomics  AAAS_proteomics  \\\n",
       "Patient_ID                                                                      \n",
       "01CO001                 NaN              NaN             NaN              NaN   \n",
       "01CO006              -1.120           -0.441          -0.347          -0.0029   \n",
       "01CO008              -1.200            0.160          -1.850           0.1190   \n",
       "01CO013              -1.890            0.112          -0.329           0.6700   \n",
       "01CO014              -0.523           -0.248          -0.638           0.2890   \n",
       "\n",
       "Name        AACS_proteomics  AAGAB_proteomics  AAK1_proteomics  \\\n",
       "Patient_ID                                                       \n",
       "01CO001                 NaN               NaN              NaN   \n",
       "01CO006              0.0957            0.3960          -0.0363   \n",
       "01CO008             -0.0924            0.0187          -0.2140   \n",
       "01CO013              0.1160            0.3130          -0.2380   \n",
       "01CO014              0.3780            0.8220           0.0018   \n",
       "\n",
       "Name        AAMDC_proteomics  AAMP_proteomics  AAR2_proteomics  ...  \\\n",
       "Patient_ID                                                      ...   \n",
       "01CO001                  NaN              NaN              NaN  ...   \n",
       "01CO006               -0.549            0.220            0.248  ...   \n",
       "01CO008                0.328           -0.282           -0.348  ...   \n",
       "01CO013               -0.274           -0.554            0.270  ...   \n",
       "01CO014               -1.080            0.498           -0.120  ...   \n",
       "\n",
       "Name        ZNRD1_proteomics  ZNRF2_proteomics  ZPR1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "01CO001                  NaN               NaN              NaN   \n",
       "01CO006                0.324               NaN          -0.3940   \n",
       "01CO008                  NaN           -0.3840          -0.1680   \n",
       "01CO013                  NaN               NaN          -0.0656   \n",
       "01CO014                  NaN            0.0467           0.2170   \n",
       "\n",
       "Name        ZRANB2_proteomics  ZW10_proteomics  ZWILCH_proteomics  \\\n",
       "Patient_ID                                                          \n",
       "01CO001                   NaN              NaN                NaN   \n",
       "01CO006                0.0846            0.001             -0.345   \n",
       "01CO008                0.3570           -0.325              0.349   \n",
       "01CO013               -0.2950            0.463             -0.448   \n",
       "01CO014                0.2360            0.287              0.346   \n",
       "\n",
       "Name        ZWINT_proteomics  ZYX_proteomics  ZZEF1_proteomics  \\\n",
       "Patient_ID                                                       \n",
       "01CO001                  NaN             NaN               NaN   \n",
       "01CO006                  NaN          -0.658            0.0052   \n",
       "01CO008                  NaN          -0.821            0.2000   \n",
       "01CO013               0.0493          -0.904           -0.1580   \n",
       "01CO014                  NaN          -0.812           -0.0273   \n",
       "\n",
       "Name                 Mutation  \n",
       "Patient_ID                     \n",
       "01CO001        Wildtype_Tumor  \n",
       "01CO006        Wildtype_Tumor  \n",
       "01CO008     Missense_Mutation  \n",
       "01CO013        Wildtype_Tumor  \n",
       "01CO014        Wildtype_Tumor  \n",
       "\n",
       "[5 rows x 8067 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'KRAS'\n",
    "col_prot = col.get_proteomics()\n",
    "col_prot = col_prot.drop(columns = 'KRAS') # cis effect\n",
    "col_prot_list = list(col_prot.columns)\n",
    "\n",
    "col_miss_wt = all_prot_format_df(col, col_prot_list)\n",
    "col_miss_wt.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 2: Run T-tests. Create a data frame of just significant comparisons and another data frame for all comparisons. A moderately stringent correction is used (FDR_BH)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_cols = list(col_miss_wt.columns[:-1])\n",
    "\n",
    "# Get all pvals\n",
    "col_pval = u.wrap_ttest(col_miss_wt, 'Mutation', col_cols, return_all = True, correction_method = 'fdr_bh', mincount = 5, pval_return_corrected=True)\n",
    "col_pval = col_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Colon_P_Value'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 3: Create the differential expression df. (median of tumors with KRAS wildtype - median of tumors with KRAS missense mutation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_med = get_change_in_medians_df(col, \"Colon\", col_miss_wt, col_cols)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "Part 4: Merge the p-value and differential expression dfs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ETFA</td>\n",
       "      <td>0.001645</td>\n",
       "      <td>0.62150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ETFB</td>\n",
       "      <td>0.017909</td>\n",
       "      <td>0.57900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BCAT2</td>\n",
       "      <td>0.046376</td>\n",
       "      <td>0.54165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DHX38</td>\n",
       "      <td>0.056443</td>\n",
       "      <td>-0.33950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CYP2S1</td>\n",
       "      <td>0.166472</td>\n",
       "      <td>0.74945</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Proteomics  Colon_P_Value  Colon_Median\n",
       "0       ETFA       0.001645       0.62150\n",
       "1       ETFB       0.017909       0.57900\n",
       "2      BCAT2       0.046376       0.54165\n",
       "3      DHX38       0.056443      -0.33950\n",
       "4     CYP2S1       0.166472       0.74945"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_merged = col_pval.merge(col_med, on='Proteomics', how='outer')\n",
    "col_merged = col_merged.replace(to_replace = r'_proteomics', value = '', regex = True)\n",
    "\n",
    "# Create csv\n",
    "#en_merged.to_csv('csv/Single_Cancer/Gbm_pval_medians.csv',index=False)\n",
    "col_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Repeat for other cancers."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_objects = { 'Luad':luad, 'Endo':en}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Proteomics  Luad_P_Value  Luad_Median\n",
      "0       ERI3      0.113056     -0.80435\n",
      "1      DOCK5      0.113056      1.09795\n",
      "2      TUFT1      0.113056      1.44105\n",
      "3    ATP13A4      0.131113     -1.62065\n",
      "4     CYP4X1      0.131113     -2.68135 \n",
      "\n",
      "  Proteomics  Endo_P_Value  Endo_Median\n",
      "0     GPRC5A      0.021196       1.6317\n",
      "1      SPRY1      0.062723       0.7132\n",
      "2     SLC7A2      0.092467       1.7340\n",
      "3    ATG16L1      0.133518       0.4410\n",
      "4       SSH3      0.133518       0.4400 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "gene = 'PIK3CA'\n",
    "merged_dfs = {}\n",
    "for cancer in cancer_objects:\n",
    "    prot = cancer_objects[cancer].get_proteomics()\n",
    "    prot = prot.drop(columns = 'PIK3CA') # cis effect\n",
    "    \n",
    "    # Reduce a multiindex \n",
    "    if isinstance(prot.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "        prot = u.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "    prot_list = list(prot.columns)\n",
    "    \n",
    "    # Format df for t-test\n",
    "    miss_wt = all_prot_format_df(cancer_objects[cancer], prot_list)\n",
    "    \n",
    "    # Rename duplicate columns (isoforms)\n",
    "    col_names = pd.Series(miss_wt.columns[:])\n",
    "    for dup in col_names[col_names.duplicated()].unique(): \n",
    "        col_names[col_names[col_names == dup].index.values.tolist()] = [dup + '_isoform_' + str(i) if i != 0 else dup for i in range(sum(col_names == dup))]\n",
    "    miss_wt.columns = col_names # rename the columns with the cols list\n",
    "    \n",
    "    # T-test\n",
    "    cols_in = list(miss_wt.columns[:-1])\n",
    "\n",
    "    # Get all pvals\n",
    "    min_num = 5\n",
    "    all_pval = u.wrap_ttest(miss_wt, 'Mutation', cols_in, return_all = True, correction_method = 'fdr_bh', mincount = min_num, pval_return_corrected=True)\n",
    "    all_pval = all_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': cancer+'_P_Value'})\n",
    "    \n",
    "    # Get difference in medians\n",
    "    delta_median_df = get_change_in_medians_df(cancer_objects[cancer], cancer, miss_wt, cols_in)\n",
    "\n",
    "    # Merge pval_df and delta_median_df\n",
    "    pval_medians_df = all_pval.merge(delta_median_df, on='Proteomics', how='outer')\n",
    "    pval_medians_df = pval_medians_df.replace(to_replace = r'_proteomics', value = '', regex = True)\n",
    "    merged_dfs[cancer] = pval_medians_df # testing purposes\n",
    "    \n",
    "    # Create csv\n",
    "    #pval_medians_df.to_csv('csv/Single_Cancer/'+cancer+'_pval_medians.csv', index=False)\n",
    "    print(pval_medians_df.head(), '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get number of significant genes in each cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add colon to merged_dfs dictionary\n",
    "merged_dfs['Colon'] = col_merged\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Luad sig comparisons: 0\n",
      "Endo sig comparisons: 1\n",
      "Colon sig comparisons: 3\n",
      "\n",
      "Number of significant comparisons in at least 1 cancer: 4\n"
     ]
    }
   ],
   "source": [
    "# Using corrected pvals from wrap_ttest, each cancer's cutoff is 0.05\n",
    "all_sig = []\n",
    "\n",
    "for cancer in merged_dfs:\n",
    "    df = merged_dfs[cancer]\n",
    "    sig_df = df.loc[df[cancer+'_P_Value'] < 0.05]\n",
    "    print(cancer, 'sig comparisons:', len(sig_df))\n",
    "    sig_list = list(sig_df['Proteomics'])\n",
    "    all_sig.append(sig_list)\n",
    "        \n",
    "\n",
    "flat_list = [item for sublist in all_sig for item in sublist] #change list of lists, to just one list\n",
    "sig = list(set(flat_list)) # keep only one if gene in multiple cancer sig lists\n",
    "print('\\nNumber of significant comparisons in at least 1 cancer:', len(sig))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get a list of significant genes in at least one cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant genes in at least one cancer: 4\n"
     ]
    }
   ],
   "source": [
    "s = pd.Series(sig)\n",
    "print('Number of significant genes in at least one cancer:', len(s))\n",
    "#s.to_csv('csv/list_sig_one_cancer.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERI3</td>\n",
       "      <td>0.113056</td>\n",
       "      <td>-0.80435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOCK5</td>\n",
       "      <td>0.113056</td>\n",
       "      <td>1.09795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TUFT1</td>\n",
       "      <td>0.113056</td>\n",
       "      <td>1.44105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATP13A4</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>-1.62065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CYP4X1</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>-2.68135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MPP6</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>-1.27495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BTBD1</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>0.74470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PTGS2</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>5.40095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>F3</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>2.57140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>IFT20</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>-0.74310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>RAP1GAP2</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>1.79715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MTMR12</td>\n",
       "      <td>0.136234</td>\n",
       "      <td>-0.91320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>FAM83A</td>\n",
       "      <td>0.149297</td>\n",
       "      <td>2.60310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NMNAT1</td>\n",
       "      <td>0.149297</td>\n",
       "      <td>-0.80030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CD74_isoform_1</td>\n",
       "      <td>0.150703</td>\n",
       "      <td>-2.36095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>FUK</td>\n",
       "      <td>0.150703</td>\n",
       "      <td>-0.84715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TUBB2B</td>\n",
       "      <td>0.150703</td>\n",
       "      <td>6.88790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>POM121C</td>\n",
       "      <td>0.150703</td>\n",
       "      <td>-0.78550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITPKA</td>\n",
       "      <td>0.165018</td>\n",
       "      <td>1.82610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GAB1</td>\n",
       "      <td>0.165018</td>\n",
       "      <td>-0.76400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>RCC1</td>\n",
       "      <td>0.166957</td>\n",
       "      <td>-1.43555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MKNK2</td>\n",
       "      <td>0.166957</td>\n",
       "      <td>-1.50960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>MMP1</td>\n",
       "      <td>0.166957</td>\n",
       "      <td>3.60950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PUS10</td>\n",
       "      <td>0.183120</td>\n",
       "      <td>-0.64150</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Proteomics  Luad_P_Value  Luad_Median\n",
       "0             ERI3      0.113056     -0.80435\n",
       "1            DOCK5      0.113056      1.09795\n",
       "2            TUFT1      0.113056      1.44105\n",
       "3          ATP13A4      0.131113     -1.62065\n",
       "4           CYP4X1      0.131113     -2.68135\n",
       "5             MPP6      0.131113     -1.27495\n",
       "6            BTBD1      0.131113      0.74470\n",
       "7            PTGS2      0.131113      5.40095\n",
       "8               F3      0.131113      2.57140\n",
       "9            IFT20      0.131113     -0.74310\n",
       "10        RAP1GAP2      0.131113      1.79715\n",
       "11          MTMR12      0.136234     -0.91320\n",
       "12          FAM83A      0.149297      2.60310\n",
       "13          NMNAT1      0.149297     -0.80030\n",
       "14  CD74_isoform_1      0.150703     -2.36095\n",
       "15             FUK      0.150703     -0.84715\n",
       "16          TUBB2B      0.150703      6.88790\n",
       "17         POM121C      0.150703     -0.78550\n",
       "18           ITPKA      0.165018      1.82610\n",
       "19            GAB1      0.165018     -0.76400\n",
       "20            RCC1      0.166957     -1.43555\n",
       "21           MKNK2      0.166957     -1.50960\n",
       "22            MMP1      0.166957      3.60950\n",
       "23           PUS10      0.183120     -0.64150"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Luad_results = merged_dfs[\"Luad\"]\n",
    "\n",
    "Luad_sig_df = Luad_results.loc[Luad_results['Luad_P_Value'] < 0.2]\n",
    "Luad_sig_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Endo_P_Value</th>\n",
       "      <th>Endo_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GPRC5A</td>\n",
       "      <td>0.021196</td>\n",
       "      <td>1.6317</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Proteomics  Endo_P_Value  Endo_Median\n",
       "0     GPRC5A      0.021196       1.6317"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Endo_results = merged_dfs[\"Endo\"]\n",
    "\n",
    "Endo_sig_df = Endo_results.loc[Endo_results['Endo_P_Value'] < 0.05]\n",
    "Endo_sig_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>AKR7A3</td>\n",
       "      <td>0.277713</td>\n",
       "      <td>5.8567</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Proteomics  Luad_P_Value  Luad_Median\n",
       "237     AKR7A3      0.277713       5.8567"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Luad_results.loc[Luad_results['Proteomics'] == \"AKR7A3\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_list = \"ARK7A3\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>ARK7A3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [ARK7A3_proteomics, Mutation]\n",
       "Index: []"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    " # cis effect\n",
    "miss_wt = all_prot_format_df(luad, prot_list)\n",
    "miss_wt.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "Luad_results = merged_dfs[\"Luad\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERI3</td>\n",
       "      <td>0.113056</td>\n",
       "      <td>-0.80435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOCK5</td>\n",
       "      <td>0.113056</td>\n",
       "      <td>1.09795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TUFT1</td>\n",
       "      <td>0.113056</td>\n",
       "      <td>1.44105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATP13A4</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>-1.62065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CYP4X1</td>\n",
       "      <td>0.131113</td>\n",
       "      <td>-2.68135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10693</th>\n",
       "      <td>ZNF112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.11660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10694</th>\n",
       "      <td>ZNF319</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.69230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10695</th>\n",
       "      <td>ZNF438</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.17760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10696</th>\n",
       "      <td>ZNF48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.24970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10697</th>\n",
       "      <td>ZNF799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.60665</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10698 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics  Luad_P_Value  Luad_Median\n",
       "0           ERI3      0.113056     -0.80435\n",
       "1          DOCK5      0.113056      1.09795\n",
       "2          TUFT1      0.113056      1.44105\n",
       "3        ATP13A4      0.131113     -1.62065\n",
       "4         CYP4X1      0.131113     -2.68135\n",
       "...          ...           ...          ...\n",
       "10693     ZNF112           NaN     -0.11660\n",
       "10694     ZNF319           NaN     -1.69230\n",
       "10695     ZNF438           NaN      0.17760\n",
       "10696      ZNF48           NaN      1.24970\n",
       "10697     ZNF799           NaN     -1.60665\n",
       "\n",
       "[10698 rows x 3 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Luad_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Tumor.Purity.byESTIMATE.RNAseq</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00001</th>\n",
       "      <td>0.582012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00009</th>\n",
       "      <td>0.537561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00080</th>\n",
       "      <td>0.762792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00083</th>\n",
       "      <td>0.351477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00093</th>\n",
       "      <td>0.491378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02582.N</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02586.N</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02587.N</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02588.N</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02729.N</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>211 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name         Tumor.Purity.byESTIMATE.RNAseq\n",
       "Patient_ID                                 \n",
       "C3L-00001                          0.582012\n",
       "C3L-00009                          0.537561\n",
       "C3L-00080                          0.762792\n",
       "C3L-00083                          0.351477\n",
       "C3L-00093                          0.491378\n",
       "...                                     ...\n",
       "C3N-02582.N                             NaN\n",
       "C3N-02586.N                             NaN\n",
       "C3N-02587.N                             NaN\n",
       "C3N-02588.N                             NaN\n",
       "C3N-02729.N                             NaN\n",
       "\n",
       "[211 rows x 1 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "purity = luad.get_derived_molecular()\n",
    "purity = purity[[\"Tumor.Purity.byESTIMATE.RNAseq\"]]\n",
    "purity\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>CIN</th>\n",
       "      <th>ESTIMATEScore</th>\n",
       "      <th>ImmuneScore</th>\n",
       "      <th>Integrated.Phenotype</th>\n",
       "      <th>Mutation_Phenotype</th>\n",
       "      <th>Proteomic_subtype</th>\n",
       "      <th>StromalScore</th>\n",
       "      <th>Transcriptomic_subtype</th>\n",
       "      <th>TumorPurity</th>\n",
       "      <th>immuneSubtype</th>\n",
       "      <th>mutation_rate</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01CO001</th>\n",
       "      <td>CIN.low</td>\n",
       "      <td>204.4353</td>\n",
       "      <td>573.3159</td>\n",
       "      <td>Epithelial</td>\n",
       "      <td>MSS</td>\n",
       "      <td>C</td>\n",
       "      <td>-368.8806</td>\n",
       "      <td>CMS3</td>\n",
       "      <td>0.8051</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.8000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO005</th>\n",
       "      <td>CIN.high</td>\n",
       "      <td>-1529.3246</td>\n",
       "      <td>-427.0268</td>\n",
       "      <td>Epithelial</td>\n",
       "      <td>MSS</td>\n",
       "      <td>E</td>\n",
       "      <td>-1102.2979</td>\n",
       "      <td>CMS2</td>\n",
       "      <td>0.9285</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.4000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO006</th>\n",
       "      <td>CIN.low</td>\n",
       "      <td>2008.1980</td>\n",
       "      <td>1542.0233</td>\n",
       "      <td>EMT</td>\n",
       "      <td>MSS</td>\n",
       "      <td>C</td>\n",
       "      <td>466.1746</td>\n",
       "      <td>CMS3</td>\n",
       "      <td>0.6218</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.2667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO008</th>\n",
       "      <td>CIN.high</td>\n",
       "      <td>-1177.5590</td>\n",
       "      <td>-54.7177</td>\n",
       "      <td>Epithelial</td>\n",
       "      <td>MSS</td>\n",
       "      <td>E</td>\n",
       "      <td>-1122.8413</td>\n",
       "      <td>CMS3</td>\n",
       "      <td>0.9081</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.8333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO013</th>\n",
       "      <td>CIN.high</td>\n",
       "      <td>-42.0787</td>\n",
       "      <td>509.0033</td>\n",
       "      <td>Epithelial</td>\n",
       "      <td>MSS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-551.0820</td>\n",
       "      <td>CMS3</td>\n",
       "      <td>0.8260</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.5000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21CO007</th>\n",
       "      <td>CIN.high</td>\n",
       "      <td>-144.8318</td>\n",
       "      <td>130.3882</td>\n",
       "      <td>Epithelial</td>\n",
       "      <td>MSS</td>\n",
       "      <td>E</td>\n",
       "      <td>-275.2200</td>\n",
       "      <td>CMS2</td>\n",
       "      <td>0.8344</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.8000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO004</th>\n",
       "      <td>CIN.high</td>\n",
       "      <td>-305.4899</td>\n",
       "      <td>531.4258</td>\n",
       "      <td>Epithelial</td>\n",
       "      <td>MSS</td>\n",
       "      <td>E</td>\n",
       "      <td>-836.9157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8472</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.2333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO006</th>\n",
       "      <td>CIN.low</td>\n",
       "      <td>3460.3425</td>\n",
       "      <td>2113.5138</td>\n",
       "      <td>EMT</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>C</td>\n",
       "      <td>1346.8287</td>\n",
       "      <td>CMS4</td>\n",
       "      <td>0.4420</td>\n",
       "      <td>2.0</td>\n",
       "      <td>44.8333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24CO005</th>\n",
       "      <td>CIN.high</td>\n",
       "      <td>3099.7718</td>\n",
       "      <td>1679.5852</td>\n",
       "      <td>EMT</td>\n",
       "      <td>MSS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1420.1866</td>\n",
       "      <td>CMS4</td>\n",
       "      <td>0.4889</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.1667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27CO004</th>\n",
       "      <td>CIN.high</td>\n",
       "      <td>1046.1686</td>\n",
       "      <td>653.1256</td>\n",
       "      <td>EMT</td>\n",
       "      <td>MSS</td>\n",
       "      <td>C</td>\n",
       "      <td>393.0430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7258</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.2333</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name             CIN  ESTIMATEScore  ImmuneScore Integrated.Phenotype  \\\n",
       "Patient_ID                                                              \n",
       "01CO001      CIN.low       204.4353     573.3159           Epithelial   \n",
       "01CO005     CIN.high     -1529.3246    -427.0268           Epithelial   \n",
       "01CO006      CIN.low      2008.1980    1542.0233                  EMT   \n",
       "01CO008     CIN.high     -1177.5590     -54.7177           Epithelial   \n",
       "01CO013     CIN.high       -42.0787     509.0033           Epithelial   \n",
       "...              ...            ...          ...                  ...   \n",
       "21CO007     CIN.high      -144.8318     130.3882           Epithelial   \n",
       "22CO004     CIN.high      -305.4899     531.4258           Epithelial   \n",
       "22CO006      CIN.low      3460.3425    2113.5138                  EMT   \n",
       "24CO005     CIN.high      3099.7718    1679.5852                  EMT   \n",
       "27CO004     CIN.high      1046.1686     653.1256                  EMT   \n",
       "\n",
       "Name       Mutation_Phenotype Proteomic_subtype  StromalScore  \\\n",
       "Patient_ID                                                      \n",
       "01CO001                   MSS                 C     -368.8806   \n",
       "01CO005                   MSS                 E    -1102.2979   \n",
       "01CO006                   MSS                 C      466.1746   \n",
       "01CO008                   MSS                 E    -1122.8413   \n",
       "01CO013                   MSS               NaN     -551.0820   \n",
       "...                       ...               ...           ...   \n",
       "21CO007                   MSS                 E     -275.2200   \n",
       "22CO004                   MSS                 E     -836.9157   \n",
       "22CO006                 MSI-H                 C     1346.8287   \n",
       "24CO005                   MSS               NaN     1420.1866   \n",
       "27CO004                   MSS                 C      393.0430   \n",
       "\n",
       "Name       Transcriptomic_subtype  TumorPurity  immuneSubtype  mutation_rate  \n",
       "Patient_ID                                                                    \n",
       "01CO001                      CMS3       0.8051            1.0         4.8000  \n",
       "01CO005                      CMS2       0.9285            1.0        10.4000  \n",
       "01CO006                      CMS3       0.6218            2.0         1.2667  \n",
       "01CO008                      CMS3       0.9081            1.0         1.8333  \n",
       "01CO013                      CMS3       0.8260            1.0        29.5000  \n",
       "...                           ...          ...            ...            ...  \n",
       "21CO007                      CMS2       0.8344            1.0         7.8000  \n",
       "22CO004                       NaN       0.8472            1.0         7.2333  \n",
       "22CO006                      CMS4       0.4420            2.0        44.8333  \n",
       "24CO005                      CMS4       0.4889            2.0         4.1667  \n",
       "27CO004                       NaN       0.7258            2.0         5.2333  \n",
       "\n",
       "[110 rows x 11 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
