{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make tables: calculate p-values and differential expressions for each cancer "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a dataframe with p-value results from t-tests for all proteins (trans proteomics when PTEN has cnv deletions compared to PTEN wildtype). The dataframe also includes the change in medians between deletions and wildtype. Prepare these tables for further analysis by creating csv files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "\n",
    "import cptac\n",
    "import cptac.pancan as pc\n",
    "import cptac.utils as u\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cptac version: 0.8.6\n"
     ]
    }
   ],
   "source": [
    "print('cptac version:', cptac.version())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Params\n",
    "cancer_object: Object. The loaded cancer data set.\n",
    "all_prot: List. trans proteins used to get proteomics data. \n",
    "gene_in: String. Gene used to get mutation status for the Mutation col.\n",
    "utils: utils package from cptac.  \n",
    "\n",
    "Returns a dataframe with trans proteomics and mutation status of a specific gene.\n",
    "'''\n",
    "\n",
    "def all_prot_format_df(cancer_object, all_prot, gene_in = 'PTEN', utils = u):\n",
    "    mut_type = cancer_object.get_genotype_all_vars(gene_in, omics_source = 'washu') # washu has cnv\n",
    "    if cancer_object.get_cancer_type() not in ('pancanluad'):\n",
    "        # Keep only tumor samples from proteomics\n",
    "        prot_and_mutations = cancer_object.join_omics_to_mutations(mutations_genes = [gene_in], \n",
    "                                            omics_df_name = 'proteomics', omics_source = 'umich', \n",
    "                                            omics_genes = all_prot, tissue_type = 'tumor')\n",
    "        # Reduce a multiindex \n",
    "        if isinstance(prot_and_mutations.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "            prot_and_mutations = utils.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "            prot_and_mutations = rename_duplicate_cols(prot_and_mutations) # add '_i' to distinguish\n",
    "            \n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        mut_type = mut_type[['Mutation']] # get mutation column (created in get_genotype_all_vars)\n",
    "        prot_df = prot_and_mutations.iloc[:,:-4] # Keep only proteomics\n",
    "        merged = prot_df.join(mut_type)\n",
    "        \n",
    "        # Keep only Wildtype and deletion\n",
    "        compare = ['Wildtype_Tumor','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "    \n",
    "    \n",
    "    # Luad has no somatic mutations for PTEN which changes some things\n",
    "    elif cancer_object.get_cancer_type() in ('pancanluad'): \n",
    "        # get_genotype_all_vars add cnv data under the column PTEN\n",
    "        mut_type = mut_type.drop(columns= gene_in)\n",
    "        # different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "        omics = cancer_object.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics', df1_source = 'washu', \n",
    "                                                  df2_source = 'umich', genes1= gene_in, genes2= all_prot, \n",
    "                                                  tissue_type = 'tumor')\n",
    "        omics = utils.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "        omics = rename_duplicate_cols(omics)\n",
    "        omics = omics.drop(columns='PTEN_washu_CNV')\n",
    "       \n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        merged = omics.join(mut_type) \n",
    "        # Keep only No_Mutation (wildtype) and deletion\n",
    "        compare = ['No_Mutation','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "        mut_wt['Mutation'] = np.where(\n",
    "            mut_wt['Mutation'] == 'No_Mutation', 'Wildtype_Tumor', 'Deletion')\n",
    "    \n",
    "    mut_wt = mut_wt.rename(columns={col: re.sub(r'_umich_proteomics', '', col) for col in mut_wt.columns.tolist()})\n",
    "    return mut_wt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Params\n",
    "cancer_object: Object of the loaded cancer data set.\n",
    "cancer_name: String. Name to add to the created df.\n",
    "del_wt_df: DataFrame. df of only samples with PTEN deletion and wildtype. \n",
    "all_prot_list: List. All proteins in proteomics df. \n",
    "\n",
    "Returns a df with the differential expression between proteomics with PTEN del and wt (del - wt). \n",
    "'''\n",
    "\n",
    "def get_change_in_medians_df(cancer_object, cancer_name, del_wt_df, all_prot_list):\n",
    "    if  cancer_object.get_cancer_type() in ('pancanucec'):\n",
    "        d = del_wt_df[del_wt_df.Mutation == \"Truncation\"]\n",
    "        wt = del_wt_df[del_wt_df.Mutation == \"Wildtype_Tumor\"]\n",
    "    else:\n",
    "        d = del_wt_df[del_wt_df.Mutation == \"Deletion\"]\n",
    "        wt = del_wt_df[del_wt_df.Mutation == \"Wildtype_Tumor\"]\n",
    "    del_med = d.median()\n",
    "    wt_med = wt.median()\n",
    "\n",
    "    med_dict = {}\n",
    "    \n",
    "    # Correlation: + is mutant up compared to wt, - is mutant down\n",
    "    for prot in all_prot_list:\n",
    "        if prot in del_med.index and prot in wt_med.index:\n",
    "            dif = del_med[prot] - wt_med[prot]\n",
    "            med_dict[prot] = dif\n",
    "        \n",
    "    df = pd.DataFrame.from_dict(med_dict, orient='index', columns=[cancer_name+'_Median'])\n",
    "    df = df.reset_index().rename(columns={'index':'Umich_Proteomics'})\n",
    "    \n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "''' Adds '_i' to duplicate col names where the int i increases by one with each duplicate \n",
    "Returns a df with unique column names. '''\n",
    "\n",
    "def rename_duplicate_cols(df):\n",
    "    cols = pd.Series(df.columns[:])\n",
    "\n",
    "    for dup in cols[cols.duplicated()].unique(): \n",
    "        cols[cols[cols == dup].index.values.tolist()] = [dup + '_' + str(i) if i != 0 else dup for i in range(sum(cols == dup))]\n",
    "\n",
    "    # rename the columns with the cols list.\n",
    "    df.columns=cols\n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Step 1: Create data frames with p-values and differential expressions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each cancer needs a data frame containing only samples that have PTEN cnv deletions and PTEN wildtype with trans proteomics. Use wrap_ttest to run t-tests for all genes in the proteomic data frame. Use get_change_in_medians_df to create the data frame with differential expression values. Merge both data frames. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load in cancer data sets from cptac. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading washuluad v1.0.                          \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brittanyhenderson/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3437: DtypeWarning: Columns (88) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                 \r"
     ]
    }
   ],
   "source": [
    "en = pc.PancanUcec()\n",
    "h = pc.PancanHnscc()\n",
    "l = pc.PancanLuad()\n",
    "ls = pc.PancanLscc()\n",
    "o = pc.PancanOv()\n",
    "r = pc.PancanCcrcc()\n",
    "co = pc.PancanCoad()\n",
    "g = pc.PancanGbm()\n",
    "b = pc.PancanBrca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Please login to Box on the webpage that was just opened and grant access for cptac to download files through your account. If you accidentally closed the browser window, press Ctrl+C and call the download function again.\n",
      "                                                 \r"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pc.download('all', redownload = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gbm"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 1: Format data frame using all_prot_format_df. This returns a dataframe with proteomics and a Mutation column containing only PTEN cnv deletions and wildtype tumors. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brittanyhenderson/anaconda3/lib/python3.8/site-packages/pandas/core/generic.py:4153: PerformanceWarning: dropping on a non-lexsorted multi-index without a level parameter may impact performance.\n",
      "  obj = obj._drop_axis(labels, axis, level=level, errors=errors)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 68 samples for the PTEN gene (/Users/brittanyhenderson/anaconda3/lib/python3.8/site-packages/cptac/pancan/pancandataset.py, line 174)\n",
      "/Users/brittanyhenderson/anaconda3/lib/python3.8/site-packages/numpy/core/_asarray.py:83: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray\n",
      "  return array(a, dtype, copy=False, order=order)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 69 samples for the PTEN gene (/Users/brittanyhenderson/anaconda3/lib/python3.8/site-packages/cptac/pancan/pancandataset.py, line 174)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 1633 duplicated column headers. (<ipython-input-57-7ad4b70bf808>, line 20)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>ARF5</th>\n",
       "      <th>M6PR</th>\n",
       "      <th>ESRRA</th>\n",
       "      <th>FKBP4</th>\n",
       "      <th>NDUFAF7</th>\n",
       "      <th>FUCA2</th>\n",
       "      <th>DBNDD1</th>\n",
       "      <th>HS3ST1</th>\n",
       "      <th>CYP51A1</th>\n",
       "      <th>USP28</th>\n",
       "      <th>...</th>\n",
       "      <th>AP1S2_3</th>\n",
       "      <th>EED</th>\n",
       "      <th>DDHD1_1</th>\n",
       "      <th>WIZ</th>\n",
       "      <th>ZBTB3</th>\n",
       "      <th>CTNND1_2</th>\n",
       "      <th>WIZ_1</th>\n",
       "      <th>WIZ_2</th>\n",
       "      <th>MSANTD2</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>-0.254733</td>\n",
       "      <td>-0.138938</td>\n",
       "      <td>0.662227</td>\n",
       "      <td>-0.074868</td>\n",
       "      <td>0.213529</td>\n",
       "      <td>0.123398</td>\n",
       "      <td>2.105458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.660264</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.446007</td>\n",
       "      <td>0.070101</td>\n",
       "      <td>0.160090</td>\n",
       "      <td>0.129282</td>\n",
       "      <td>-0.166678</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.073448</td>\n",
       "      <td>-0.029020</td>\n",
       "      <td>-0.009463</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>-0.138512</td>\n",
       "      <td>-0.824520</td>\n",
       "      <td>0.494419</td>\n",
       "      <td>0.043783</td>\n",
       "      <td>-0.001394</td>\n",
       "      <td>0.101477</td>\n",
       "      <td>-0.287232</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.402679</td>\n",
       "      <td>0.059975</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.844985</td>\n",
       "      <td>-0.234406</td>\n",
       "      <td>-0.448783</td>\n",
       "      <td>0.078407</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.174780</td>\n",
       "      <td>1.806950</td>\n",
       "      <td>0.033808</td>\n",
       "      <td>0.180578</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>-0.351464</td>\n",
       "      <td>-0.120197</td>\n",
       "      <td>-0.084371</td>\n",
       "      <td>-0.260275</td>\n",
       "      <td>0.096193</td>\n",
       "      <td>-0.360274</td>\n",
       "      <td>0.227499</td>\n",
       "      <td>1.217058</td>\n",
       "      <td>-0.165751</td>\n",
       "      <td>0.010124</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.096328</td>\n",
       "      <td>-0.118506</td>\n",
       "      <td>-0.107690</td>\n",
       "      <td>0.177530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.513869</td>\n",
       "      <td>0.065973</td>\n",
       "      <td>-0.126149</td>\n",
       "      <td>0.465241</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>-0.062869</td>\n",
       "      <td>0.094198</td>\n",
       "      <td>0.391070</td>\n",
       "      <td>-0.030638</td>\n",
       "      <td>0.742258</td>\n",
       "      <td>-0.417291</td>\n",
       "      <td>-0.013377</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.176649</td>\n",
       "      <td>0.535304</td>\n",
       "      <td>...</td>\n",
       "      <td>0.320945</td>\n",
       "      <td>-0.027522</td>\n",
       "      <td>0.104278</td>\n",
       "      <td>0.049948</td>\n",
       "      <td>-0.590267</td>\n",
       "      <td>0.162686</td>\n",
       "      <td>1.964570</td>\n",
       "      <td>0.161229</td>\n",
       "      <td>0.283810</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-0.365351</td>\n",
       "      <td>0.070523</td>\n",
       "      <td>-0.472543</td>\n",
       "      <td>-0.255288</td>\n",
       "      <td>0.096844</td>\n",
       "      <td>0.356271</td>\n",
       "      <td>1.182940</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.307430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.298907</td>\n",
       "      <td>-0.201144</td>\n",
       "      <td>0.440215</td>\n",
       "      <td>0.110757</td>\n",
       "      <td>0.119013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.030719</td>\n",
       "      <td>0.066426</td>\n",
       "      <td>0.189187</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 13002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name            ARF5      M6PR     ESRRA     FKBP4   NDUFAF7     FUCA2  \\\n",
       "Patient_ID                                                               \n",
       "C3L-00104  -0.254733 -0.138938  0.662227 -0.074868  0.213529  0.123398   \n",
       "C3L-00365  -0.138512 -0.824520  0.494419  0.043783 -0.001394  0.101477   \n",
       "C3L-00674  -0.351464 -0.120197 -0.084371 -0.260275  0.096193 -0.360274   \n",
       "C3L-00677  -0.062869  0.094198  0.391070 -0.030638  0.742258 -0.417291   \n",
       "C3L-01040  -0.365351  0.070523 -0.472543 -0.255288  0.096844  0.356271   \n",
       "\n",
       "Name          DBNDD1    HS3ST1   CYP51A1     USP28  ...   AP1S2_3       EED  \\\n",
       "Patient_ID                                          ...                       \n",
       "C3L-00104   2.105458       NaN -0.660264       NaN  ... -0.446007  0.070101   \n",
       "C3L-00365  -0.287232       NaN -0.402679  0.059975  ... -0.844985 -0.234406   \n",
       "C3L-00674   0.227499  1.217058 -0.165751  0.010124  ... -0.096328 -0.118506   \n",
       "C3L-00677  -0.013377       NaN -0.176649  0.535304  ...  0.320945 -0.027522   \n",
       "C3L-01040   1.182940       NaN -0.307430       NaN  ... -0.298907 -0.201144   \n",
       "\n",
       "Name         DDHD1_1       WIZ     ZBTB3  CTNND1_2     WIZ_1     WIZ_2  \\\n",
       "Patient_ID                                                               \n",
       "C3L-00104   0.160090  0.129282 -0.166678       NaN -0.073448 -0.029020   \n",
       "C3L-00365  -0.448783  0.078407       NaN -0.174780  1.806950  0.033808   \n",
       "C3L-00674  -0.107690  0.177530       NaN -1.513869  0.065973 -0.126149   \n",
       "C3L-00677   0.104278  0.049948 -0.590267  0.162686  1.964570  0.161229   \n",
       "C3L-01040   0.440215  0.110757  0.119013       NaN  0.030719  0.066426   \n",
       "\n",
       "Name         MSANTD2  Mutation  \n",
       "Patient_ID                      \n",
       "C3L-00104  -0.009463  Deletion  \n",
       "C3L-00365   0.180578  Deletion  \n",
       "C3L-00674   0.465241  Deletion  \n",
       "C3L-00677   0.283810  Deletion  \n",
       "C3L-01040   0.189187  Deletion  \n",
       "\n",
       "[5 rows x 13002 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "g_prot = g.get_proteomics(source = 'umich')\n",
    "g_prot = g_prot.drop(columns = 'PTEN') # cis effect\n",
    "g_prot_list = list(g_prot.columns.get_level_values(0))\n",
    "\n",
    "g_del_wt = all_prot_format_df(g, g_prot_list)\n",
    "g_del_wt.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 2: Run T-tests. Create a data frame of just significant comparisons and another data frame for all comparisons. A moderately stringent correction is used (FDR_BH)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "g_cols = list(g_del_wt.columns[:-1])\n",
    "\n",
    "# Get all pvals\n",
    "g_pval = u.wrap_ttest(g_del_wt, 'Mutation', g_cols, return_all = True, correction_method = 'fdr_bh', mincount = 5)\n",
    "g_pval = g_pval.rename(columns = {'Comparison': 'Umich_Proteomics','P_Value': 'GBM_P_Value'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Umich_Proteomics</th>\n",
       "      <th>GBM_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CUL2</td>\n",
       "      <td>4.572167e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOCK1_1</td>\n",
       "      <td>4.572167e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATE1_1</td>\n",
       "      <td>2.291619e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GDI2</td>\n",
       "      <td>2.291619e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CUTC</td>\n",
       "      <td>2.291619e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11568</th>\n",
       "      <td>THAP10</td>\n",
       "      <td>9.996790e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11569</th>\n",
       "      <td>ZNF639</td>\n",
       "      <td>9.998768e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11570</th>\n",
       "      <td>HSDL2</td>\n",
       "      <td>9.999004e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11571</th>\n",
       "      <td>ALDOB</td>\n",
       "      <td>9.999004e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11572</th>\n",
       "      <td>GPR37L1</td>\n",
       "      <td>9.999004e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11573 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Umich_Proteomics   GBM_P_Value\n",
       "0                 CUL2  4.572167e-07\n",
       "1              DOCK1_1  4.572167e-07\n",
       "2               ATE1_1  2.291619e-06\n",
       "3                 GDI2  2.291619e-06\n",
       "4                 CUTC  2.291619e-06\n",
       "...                ...           ...\n",
       "11568           THAP10  9.996790e-01\n",
       "11569           ZNF639  9.998768e-01\n",
       "11570            HSDL2  9.999004e-01\n",
       "11571            ALDOB  9.999004e-01\n",
       "11572          GPR37L1  9.999004e-01\n",
       "\n",
       "[11573 rows x 2 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 3: Create the differential expression df. (median of tumors with PTEN wildtype - median of tumors with PTEN cnv deletions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Umich_Proteomics</th>\n",
       "      <th>GBM_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARF5</td>\n",
       "      <td>0.046849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M6PR</td>\n",
       "      <td>-0.104081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ESRRA</td>\n",
       "      <td>-0.038006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FKBP4</td>\n",
       "      <td>0.104346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAF7</td>\n",
       "      <td>-0.038037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12996</th>\n",
       "      <td>ZBTB3</td>\n",
       "      <td>0.231032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12997</th>\n",
       "      <td>CTNND1_2</td>\n",
       "      <td>0.271572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12998</th>\n",
       "      <td>WIZ_1</td>\n",
       "      <td>0.220366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12999</th>\n",
       "      <td>WIZ_2</td>\n",
       "      <td>0.072386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13000</th>\n",
       "      <td>MSANTD2</td>\n",
       "      <td>0.177258</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13001 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Umich_Proteomics  GBM_Median\n",
       "0                 ARF5    0.046849\n",
       "1                 M6PR   -0.104081\n",
       "2                ESRRA   -0.038006\n",
       "3                FKBP4    0.104346\n",
       "4              NDUFAF7   -0.038037\n",
       "...                ...         ...\n",
       "12996            ZBTB3    0.231032\n",
       "12997         CTNND1_2    0.271572\n",
       "12998            WIZ_1    0.220366\n",
       "12999            WIZ_2    0.072386\n",
       "13000          MSANTD2    0.177258\n",
       "\n",
       "[13001 rows x 2 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_cols = list(g_del_wt[:-1])\n",
    "g_med = get_change_in_medians_df(g, \"GBM\", g_del_wt, g_cols)\n",
    "g_med"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Part 4: Merge the p-value and differential expression dfs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Umich_Proteomics</th>\n",
       "      <th>GBM_P_Value</th>\n",
       "      <th>GBM_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CUL2</td>\n",
       "      <td>4.572167e-07</td>\n",
       "      <td>-0.458427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOCK1_1</td>\n",
       "      <td>4.572167e-07</td>\n",
       "      <td>-0.358361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATE1_1</td>\n",
       "      <td>2.291619e-06</td>\n",
       "      <td>-0.481942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GDI2</td>\n",
       "      <td>2.291619e-06</td>\n",
       "      <td>-0.486754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CUTC</td>\n",
       "      <td>2.291619e-06</td>\n",
       "      <td>-0.490992</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Umich_Proteomics   GBM_P_Value  GBM_Median\n",
       "0             CUL2  4.572167e-07   -0.458427\n",
       "1          DOCK1_1  4.572167e-07   -0.358361\n",
       "2           ATE1_1  2.291619e-06   -0.481942\n",
       "3             GDI2  2.291619e-06   -0.486754\n",
       "4             CUTC  2.291619e-06   -0.490992"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_merged = g_pval.merge(g_med, on='Umich_Proteomics', how='outer')\n",
    "g_merged = g_merged.replace(to_replace = r'_proteomics', value = '', regex = True)\n",
    "\n",
    "# Create csv\n",
    "g_merged.to_csv('csv/Single_Cancer/GBM_pval_medians_pancan.csv',index=False)\n",
    "g_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Repeat for other cancers."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_objects = {'OV':o} #{'HNSCC':h, 'LUAD':l, 'LSCC':ls, 'BR':b, 'CO':co, 'OV':o}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'join_omics_to_mutations'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-98-fdf18456d905>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mall_prot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprot_and_iso\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m \u001b[0mmut_type\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mo\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_genotype_all_vars\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgene_in\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_source\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'washu'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# washu has cnv\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m \u001b[0mmut_type\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmut_type\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdrop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m \u001b[0mgene_in\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[0;31m# different code because no somatic mutation data for pten (can't join to somatic mutations)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.8/site-packages/cptac/pancan/pancandataset.py\u001b[0m in \u001b[0;36mget_genotype_all_vars\u001b[0;34m(self, mutations_genes, omics_source, mutations_filter, show_location, mutation_hotspot)\u001b[0m\n\u001b[1;32m    297\u001b[0m         \u001b[0mdrop\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'Database_ID'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    298\u001b[0m         \u001b[0mcombined\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mut\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreduce_multiindex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcombined\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlevels_to_drop\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdrop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 299\u001b[0;31m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    300\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    301\u001b[0m         \u001b[0;31m#If there are hotspot mutations, append 'hotspot' to the mutation type so that it's prioritized correctly\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.8/site-packages/cptac/pancan/pancandataset.py\u001b[0m in \u001b[0;36mjoin_omics_to_mutations\u001b[0;34m(self, omics_df_name, omics_source, mutations_genes, omics_genes, mutations_filter, show_location, how, quiet, tissue_type, mutation_cols)\u001b[0m\n\u001b[1;32m    172\u001b[0m             \u001b[0mmutations_filter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmutations_filter\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    173\u001b[0m             \u001b[0mshow_location\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mshow_location\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 174\u001b[0;31m             \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mhow\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    175\u001b[0m             \u001b[0mquiet\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mquiet\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    176\u001b[0m             \u001b[0mtissue_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtissue_type\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'join_omics_to_mutations'"
     ]
    }
   ],
   "source": [
    "# Luad has no somatic mutations for PTEN which changes some things\n",
    "#elif cancer_object.get_cancer_type() in ('pancanluad'): \n",
    "# get_genotype_all_vars add cnv data under the column PTEN\n",
    "\n",
    "gene_in = 'PTEN'\n",
    "all_prot = prot_and_iso\n",
    "\n",
    "mut_type = o.get_genotype_all_vars(gene_in, omics_source = 'washu') # washu has cnv\n",
    "mut_type = mut_type.drop(columns= gene_in)\n",
    "# different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "omics = o.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics', df1_source = 'washu', \n",
    "                                          df2_source = 'umich', genes1= gene_in, genes2= all_prot, \n",
    "                                          tissue_type = 'tumor')\n",
    "omics = utils.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "omics = rename_duplicate_cols(omics)\n",
    "omics = omics.drop(columns='PTEN_washu_CNV')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "ename": "DataFrameNotIncludedError",
     "evalue": "CNV dataframe not included in the umichov dataset.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mDataFrameNotIncludedError\u001b[0m                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-99-fd9f007a976c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mo\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_CNV\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'umich'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/lib/python3.8/site-packages/cptac/pancan/pancandataset.py\u001b[0m in \u001b[0;36mget_CNV\u001b[0;34m(self, source, tissue_type, imputed)\u001b[0m\n\u001b[1;32m     67\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget_CNV\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0msource\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"washu\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtissue_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"both\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mimputed\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     68\u001b[0m         \u001b[0;34m\"\"\"Get a CNV dataframe from the specified data source.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 69\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_dataframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"CNV\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msource\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtissue_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mimputed\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mimputed\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     70\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     71\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget_deconvolution\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdeconv_algorithm\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msource\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'washu'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtissue_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"both\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mimputed\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.8/site-packages/cptac/pancan/pancandataset.py\u001b[0m in \u001b[0;36m_get_dataframe\u001b[0;34m(self, name, source, tissue_type, imputed)\u001b[0m\n\u001b[1;32m    230\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_get_version\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msource\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    231\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_version\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"latest\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 232\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_version\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    233\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_version\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_version\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0msource\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.8/site-packages/cptac/dataset.py\u001b[0m in \u001b[0;36m_get_dataframe\u001b[0;34m(self, name, tissue_type)\u001b[0m\n\u001b[1;32m    742\u001b[0m         \u001b[0mpandas\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mA\u001b[0m \u001b[0mcopy\u001b[0m \u001b[0mof\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mdesired\u001b[0m \u001b[0mdataframe\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mit\u001b[0m \u001b[0mexists\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mthis\u001b[0m \u001b[0mdataset\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    743\u001b[0m         \"\"\"\n\u001b[0;32m--> 744\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0mname\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    745\u001b[0m             \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    746\u001b[0m             \u001b[0mreturn_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdeep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# We copy it, with deep=True, so edits on their copy don't affect the master for this instance\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mDataFrameNotIncludedError\u001b[0m: CNV dataframe not included in the umichov dataset."
     ]
    }
   ],
   "source": [
    "o.get_CNV('umich')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG_umich_proteomics</th>\n",
       "      <th>A1CF_umich_proteomics</th>\n",
       "      <th>A2M_umich_proteomics</th>\n",
       "      <th>A2ML1_umich_proteomics</th>\n",
       "      <th>A4GALT_umich_proteomics</th>\n",
       "      <th>AAAS_umich_proteomics</th>\n",
       "      <th>AACS_umich_proteomics</th>\n",
       "      <th>AADAC_umich_proteomics</th>\n",
       "      <th>AADAT_umich_proteomics</th>\n",
       "      <th>AAGAB_umich_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_umich_proteomics</th>\n",
       "      <th>ZSWIM9_umich_proteomics</th>\n",
       "      <th>ZW10_umich_proteomics</th>\n",
       "      <th>ZWILCH_umich_proteomics</th>\n",
       "      <th>ZWINT_umich_proteomics</th>\n",
       "      <th>ZXDC_umich_proteomics</th>\n",
       "      <th>ZYG11B_umich_proteomics</th>\n",
       "      <th>ZYX_umich_proteomics</th>\n",
       "      <th>ZZEF1_umich_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11LU013</th>\n",
       "      <td>-0.522000</td>\n",
       "      <td>0.156352</td>\n",
       "      <td>-0.927449</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.072449</td>\n",
       "      <td>-0.423627</td>\n",
       "      <td>-1.564835</td>\n",
       "      <td>0.307324</td>\n",
       "      <td>0.392956</td>\n",
       "      <td>...</td>\n",
       "      <td>0.230056</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.469225</td>\n",
       "      <td>-0.110041</td>\n",
       "      <td>0.424662</td>\n",
       "      <td>0.085748</td>\n",
       "      <td>0.052731</td>\n",
       "      <td>-0.070704</td>\n",
       "      <td>-0.398572</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11LU016</th>\n",
       "      <td>-0.416894</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.034393</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.042920</td>\n",
       "      <td>0.854008</td>\n",
       "      <td>-1.745078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.141026</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.217082</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.504543</td>\n",
       "      <td>0.690377</td>\n",
       "      <td>0.130886</td>\n",
       "      <td>0.239890</td>\n",
       "      <td>0.073790</td>\n",
       "      <td>-0.128560</td>\n",
       "      <td>-0.063684</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11LU022</th>\n",
       "      <td>-0.114286</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.474074</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.184832</td>\n",
       "      <td>0.198384</td>\n",
       "      <td>-3.002410</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.645675</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.023673</td>\n",
       "      <td>0.370696</td>\n",
       "      <td>0.715778</td>\n",
       "      <td>0.281387</td>\n",
       "      <td>0.055611</td>\n",
       "      <td>-0.136202</td>\n",
       "      <td>-0.148158</td>\n",
       "      <td>No_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11LU035</th>\n",
       "      <td>-0.138902</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.525977</td>\n",
       "      <td>-0.080078</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.164643</td>\n",
       "      <td>-0.122591</td>\n",
       "      <td>-0.976153</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.662650</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.092290</td>\n",
       "      <td>-0.125483</td>\n",
       "      <td>0.224966</td>\n",
       "      <td>0.778338</td>\n",
       "      <td>0.704170</td>\n",
       "      <td>-0.320516</td>\n",
       "      <td>-0.085086</td>\n",
       "      <td>-0.098488</td>\n",
       "      <td>-0.209858</td>\n",
       "      <td>No_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00001</th>\n",
       "      <td>-0.779792</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.054531</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.120052</td>\n",
       "      <td>-0.249726</td>\n",
       "      <td>-0.660498</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.097150</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.232436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.286752</td>\n",
       "      <td>-0.284326</td>\n",
       "      <td>-0.143142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.253291</td>\n",
       "      <td>-0.184048</td>\n",
       "      <td>-0.297789</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02582</th>\n",
       "      <td>-0.892348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.200360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.161562</td>\n",
       "      <td>0.884322</td>\n",
       "      <td>-1.422920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.457689</td>\n",
       "      <td>...</td>\n",
       "      <td>0.089350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.262830</td>\n",
       "      <td>-0.258906</td>\n",
       "      <td>-0.248340</td>\n",
       "      <td>-0.480880</td>\n",
       "      <td>0.330258</td>\n",
       "      <td>-0.518121</td>\n",
       "      <td>-0.292263</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02586</th>\n",
       "      <td>-0.417899</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.109325</td>\n",
       "      <td>-0.758060</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.154245</td>\n",
       "      <td>-0.012534</td>\n",
       "      <td>1.446736</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.144611</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.101379</td>\n",
       "      <td>0.079364</td>\n",
       "      <td>-0.090472</td>\n",
       "      <td>-0.220882</td>\n",
       "      <td>-0.684668</td>\n",
       "      <td>-0.197353</td>\n",
       "      <td>0.301477</td>\n",
       "      <td>-0.281588</td>\n",
       "      <td>-0.164234</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02587</th>\n",
       "      <td>-0.309479</td>\n",
       "      <td>-0.252554</td>\n",
       "      <td>-0.525969</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.095018</td>\n",
       "      <td>0.515820</td>\n",
       "      <td>-1.650673</td>\n",
       "      <td>-0.620950</td>\n",
       "      <td>0.104104</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005175</td>\n",
       "      <td>0.131856</td>\n",
       "      <td>0.377512</td>\n",
       "      <td>-0.993616</td>\n",
       "      <td>-0.483909</td>\n",
       "      <td>-0.543046</td>\n",
       "      <td>0.112943</td>\n",
       "      <td>-0.405470</td>\n",
       "      <td>0.088355</td>\n",
       "      <td>No_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02588</th>\n",
       "      <td>-0.611550</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.982055</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.268940</td>\n",
       "      <td>0.272542</td>\n",
       "      <td>-1.333260</td>\n",
       "      <td>0.570812</td>\n",
       "      <td>0.172053</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.257331</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.104694</td>\n",
       "      <td>0.136598</td>\n",
       "      <td>0.569153</td>\n",
       "      <td>0.953944</td>\n",
       "      <td>0.288042</td>\n",
       "      <td>-0.265337</td>\n",
       "      <td>-0.224152</td>\n",
       "      <td>No_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02729</th>\n",
       "      <td>-0.292943</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.267095</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.137302</td>\n",
       "      <td>0.938221</td>\n",
       "      <td>2.009266</td>\n",
       "      <td>1.716120</td>\n",
       "      <td>0.044110</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.093176</td>\n",
       "      <td>-0.717849</td>\n",
       "      <td>0.184317</td>\n",
       "      <td>-0.590719</td>\n",
       "      <td>-0.076732</td>\n",
       "      <td>-0.701981</td>\n",
       "      <td>0.213371</td>\n",
       "      <td>-0.072977</td>\n",
       "      <td>-0.255202</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>111 rows × 13080 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        A1BG_umich_proteomics  A1CF_umich_proteomics  \\\n",
       "Patient_ID                                                 \n",
       "11LU013                 -0.522000               0.156352   \n",
       "11LU016                 -0.416894                    NaN   \n",
       "11LU022                 -0.114286                    NaN   \n",
       "11LU035                 -0.138902                    NaN   \n",
       "C3L-00001               -0.779792                    NaN   \n",
       "...                           ...                    ...   \n",
       "C3N-02582               -0.892348                    NaN   \n",
       "C3N-02586               -0.417899                    NaN   \n",
       "C3N-02587               -0.309479              -0.252554   \n",
       "C3N-02588               -0.611550                    NaN   \n",
       "C3N-02729               -0.292943                    NaN   \n",
       "\n",
       "Name        A2M_umich_proteomics  A2ML1_umich_proteomics  \\\n",
       "Patient_ID                                                 \n",
       "11LU013                -0.927449                     NaN   \n",
       "11LU016                -1.034393                     NaN   \n",
       "11LU022                -0.474074                     NaN   \n",
       "11LU035                -0.525977               -0.080078   \n",
       "C3L-00001              -1.054531                     NaN   \n",
       "...                          ...                     ...   \n",
       "C3N-02582              -1.200360                     NaN   \n",
       "C3N-02586              -0.109325               -0.758060   \n",
       "C3N-02587              -0.525969                     NaN   \n",
       "C3N-02588              -0.982055                     NaN   \n",
       "C3N-02729              -1.267095                     NaN   \n",
       "\n",
       "Name        A4GALT_umich_proteomics  AAAS_umich_proteomics  \\\n",
       "Patient_ID                                                   \n",
       "11LU013                         NaN              -0.072449   \n",
       "11LU016                         NaN               0.042920   \n",
       "11LU022                         NaN               0.184832   \n",
       "11LU035                         NaN               0.164643   \n",
       "C3L-00001                       NaN               0.120052   \n",
       "...                             ...                    ...   \n",
       "C3N-02582                       NaN               0.161562   \n",
       "C3N-02586                       NaN               0.154245   \n",
       "C3N-02587                       NaN              -0.095018   \n",
       "C3N-02588                       NaN              -0.268940   \n",
       "C3N-02729                       NaN               0.137302   \n",
       "\n",
       "Name        AACS_umich_proteomics  AADAC_umich_proteomics  \\\n",
       "Patient_ID                                                  \n",
       "11LU013                 -0.423627               -1.564835   \n",
       "11LU016                  0.854008               -1.745078   \n",
       "11LU022                  0.198384               -3.002410   \n",
       "11LU035                 -0.122591               -0.976153   \n",
       "C3L-00001               -0.249726               -0.660498   \n",
       "...                           ...                     ...   \n",
       "C3N-02582                0.884322               -1.422920   \n",
       "C3N-02586               -0.012534                1.446736   \n",
       "C3N-02587                0.515820               -1.650673   \n",
       "C3N-02588                0.272542               -1.333260   \n",
       "C3N-02729                0.938221                2.009266   \n",
       "\n",
       "Name        AADAT_umich_proteomics  AAGAB_umich_proteomics  ...  \\\n",
       "Patient_ID                                                  ...   \n",
       "11LU013                   0.307324                0.392956  ...   \n",
       "11LU016                        NaN                0.141026  ...   \n",
       "11LU022                        NaN                0.645675  ...   \n",
       "11LU035                        NaN                0.662650  ...   \n",
       "C3L-00001                      NaN               -0.097150  ...   \n",
       "...                            ...                     ...  ...   \n",
       "C3N-02582                      NaN                0.457689  ...   \n",
       "C3N-02586                      NaN                0.144611  ...   \n",
       "C3N-02587                -0.620950                0.104104  ...   \n",
       "C3N-02588                 0.570812                0.172053  ...   \n",
       "C3N-02729                 1.716120                0.044110  ...   \n",
       "\n",
       "Name        ZSWIM8_umich_proteomics  ZSWIM9_umich_proteomics  \\\n",
       "Patient_ID                                                     \n",
       "11LU013                    0.230056                      NaN   \n",
       "11LU016                   -0.217082                      NaN   \n",
       "11LU022                    0.018950                      NaN   \n",
       "11LU035                   -0.092290                -0.125483   \n",
       "C3L-00001                 -0.232436                      NaN   \n",
       "...                             ...                      ...   \n",
       "C3N-02582                  0.089350                      NaN   \n",
       "C3N-02586                 -0.101379                 0.079364   \n",
       "C3N-02587                  0.005175                 0.131856   \n",
       "C3N-02588                 -0.257331                      NaN   \n",
       "C3N-02729                 -0.093176                -0.717849   \n",
       "\n",
       "Name        ZW10_umich_proteomics  ZWILCH_umich_proteomics  \\\n",
       "Patient_ID                                                   \n",
       "11LU013                  0.469225                -0.110041   \n",
       "11LU016                  0.504543                 0.690377   \n",
       "11LU022                 -0.023673                 0.370696   \n",
       "11LU035                  0.224966                 0.778338   \n",
       "C3L-00001                0.286752                -0.284326   \n",
       "...                           ...                      ...   \n",
       "C3N-02582                0.262830                -0.258906   \n",
       "C3N-02586               -0.090472                -0.220882   \n",
       "C3N-02587                0.377512                -0.993616   \n",
       "C3N-02588                0.104694                 0.136598   \n",
       "C3N-02729                0.184317                -0.590719   \n",
       "\n",
       "Name        ZWINT_umich_proteomics  ZXDC_umich_proteomics  \\\n",
       "Patient_ID                                                  \n",
       "11LU013                   0.424662               0.085748   \n",
       "11LU016                   0.130886               0.239890   \n",
       "11LU022                   0.715778               0.281387   \n",
       "11LU035                   0.704170              -0.320516   \n",
       "C3L-00001                -0.143142                    NaN   \n",
       "...                            ...                    ...   \n",
       "C3N-02582                -0.248340              -0.480880   \n",
       "C3N-02586                -0.684668              -0.197353   \n",
       "C3N-02587                -0.483909              -0.543046   \n",
       "C3N-02588                 0.569153               0.953944   \n",
       "C3N-02729                -0.076732              -0.701981   \n",
       "\n",
       "Name        ZYG11B_umich_proteomics  ZYX_umich_proteomics  \\\n",
       "Patient_ID                                                  \n",
       "11LU013                    0.052731             -0.070704   \n",
       "11LU016                    0.073790             -0.128560   \n",
       "11LU022                    0.055611             -0.136202   \n",
       "11LU035                   -0.085086             -0.098488   \n",
       "C3L-00001                  0.253291             -0.184048   \n",
       "...                             ...                   ...   \n",
       "C3N-02582                  0.330258             -0.518121   \n",
       "C3N-02586                  0.301477             -0.281588   \n",
       "C3N-02587                  0.112943             -0.405470   \n",
       "C3N-02588                  0.288042             -0.265337   \n",
       "C3N-02729                  0.213371             -0.072977   \n",
       "\n",
       "Name        ZZEF1_umich_proteomics     Mutation  \n",
       "Patient_ID                                       \n",
       "11LU013                  -0.398572     Deletion  \n",
       "11LU016                  -0.063684     Deletion  \n",
       "11LU022                  -0.148158  No_Mutation  \n",
       "11LU035                  -0.209858  No_Mutation  \n",
       "C3L-00001                -0.297789     Deletion  \n",
       "...                            ...          ...  \n",
       "C3N-02582                -0.292263     Deletion  \n",
       "C3N-02586                -0.164234     Deletion  \n",
       "C3N-02587                 0.088355  No_Mutation  \n",
       "C3N-02588                -0.224152  No_Mutation  \n",
       "C3N-02729                -0.255202     Deletion  \n",
       "\n",
       "[111 rows x 13080 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "merged = omics.join(mut_type)\n",
    "merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD\n",
      "  Umich_Proteomics  LUAD_P_Value  LUAD_Median\n",
      "0           LANCL2      0.000933     0.386394\n",
      "1             EGFR      0.002249     0.598962\n",
      "2             WAPL      0.007815    -0.183739\n",
      "3           ATE1_1      0.011157    -0.315236\n",
      "4             CDK4      0.015576     0.172922 \n",
      "\n",
      "LSCC\n",
      "  Umich_Proteomics  LSCC_P_Value  LSCC_Median\n",
      "0         ABRAXAS2      0.000051    -0.195523\n",
      "1          CCDC186      0.000107    -0.303387\n",
      "2          DENND10      0.000159    -0.190829\n",
      "3            SLK_1      0.000159    -0.288777\n",
      "4             GBF1      0.000159    -0.219999 \n",
      "\n",
      "BR\n",
      "  Umich_Proteomics  BR_P_Value  BR_Median\n",
      "0           RHBDD3    0.759532   0.676329\n",
      "1         IGHV4-34    0.759532   0.817872\n",
      "2           CEMIP2    0.759532   0.475073\n",
      "3           SEMA3F    0.759532  -0.779007\n",
      "4             ITCH    0.759532   0.272043 \n",
      "\n",
      "CO\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'join_omics_to_mutations'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-89-4df7ce286ff0>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     16\u001b[0m     \u001b[0;31m# Format df for t-test\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 17\u001b[0;31m     \u001b[0mdel_wt\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mall_prot_format_df\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcancer_objects\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcancer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprot_list\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     18\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     19\u001b[0m     \u001b[0;31m# Get list of unique cols for t-test and get_change_in_medians\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-88-82f839242db1>\u001b[0m in \u001b[0;36mall_prot_format_df\u001b[0;34m(cancer_object, all_prot, gene_in, utils)\u001b[0m\n\u001b[1;32m     13\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mcancer_object\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_cancer_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;34m'pancanluad'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0;31m# Keep only tumor samples from proteomics\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m         prot_and_mutations = cancer_object.join_omics_to_mutations(mutations_genes = [gene_in], \n\u001b[0m\u001b[1;32m     16\u001b[0m                                             \u001b[0momics_df_name\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'proteomics'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_source\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'umich'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m                                             omics_genes = all_prot, tissue_type = 'tumor')\n",
      "\u001b[0;32m~/anaconda3/lib/python3.8/site-packages/cptac/pancan/pancandataset.py\u001b[0m in \u001b[0;36mjoin_omics_to_mutations\u001b[0;34m(self, omics_df_name, omics_source, mutations_genes, omics_genes, mutations_filter, show_location, how, quiet, tissue_type, mutation_cols)\u001b[0m\n\u001b[1;32m    172\u001b[0m             \u001b[0mmutations_filter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmutations_filter\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    173\u001b[0m             \u001b[0mshow_location\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mshow_location\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 174\u001b[0;31m             \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mhow\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    175\u001b[0m             \u001b[0mquiet\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mquiet\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    176\u001b[0m             \u001b[0mtissue_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtissue_type\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'join_omics_to_mutations'"
     ]
    }
   ],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "gene = 'PTEN'\n",
    "merged_dfs = {}\n",
    "for cancer in cancer_objects:\n",
    "    print(cancer)\n",
    "    prot = cancer_objects[cancer].get_proteomics(source = 'umich')\n",
    "    prot = prot.drop(columns = 'PTEN') # cis effect\n",
    "    \n",
    "    # Reduce a multiindex \n",
    "    if isinstance(prot.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "        prot = u.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "    prot_list = list(prot.columns)\n",
    "    \n",
    "    # Format df for t-test\n",
    "    del_wt = all_prot_format_df(cancer_objects[cancer], prot_list)\n",
    "    \n",
    "    # Get list of unique cols for t-test and get_change_in_medians\n",
    "    prot_and_iso = list(del_wt.columns[:-1])\n",
    "\n",
    "    # Get all pvals\n",
    "    min_num = 5\n",
    "    all_pval = u.wrap_ttest(del_wt, 'Mutation', prot_and_iso, return_all = True, \n",
    "                            correction_method = 'fdr_bh', mincount = min_num)\n",
    "    all_pval = all_pval.rename(columns = {'Comparison': 'Umich_Proteomics','P_Value': cancer+'_P_Value'})\n",
    "    \n",
    "    # Get change in medians\n",
    "    delta_median_df = get_change_in_medians_df(cancer_objects[cancer], cancer, del_wt, prot_and_iso)\n",
    "\n",
    "    # Merge pval_df and delta_median_df\n",
    "    pval_medians_df = all_pval.merge(delta_median_df, on='Umich_Proteomics', how='outer')\n",
    "    merged_dfs[cancer] = pval_medians_df # testing purposes\n",
    "    \n",
    "    # Create csv\n",
    "    pval_medians_df.to_csv('csv/Single_Cancer/'+cancer+'_pval_medians_pancan.csv', index=False)\n",
    "    print(pval_medians_df.head(), '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Endometrial data set does not have enough cnv deletions to perform a t-test, however the data set does have enough truncation type mutations (nonsense and frameshift). Different code is needed to create the data frame for Endometrial."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'\n",
    "prot = en.get_proteomics(source = 'umich')\n",
    "prot = u.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "prot = prot.drop(columns = 'PTEN')\n",
    "e_prot_list = list(prot.columns)\n",
    "\n",
    "# Get mutations based on priority filter when a sample has multiple mutations\n",
    "mut_type = en.get_genotype_all_vars(gene, omics_source = 'washu')\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = en.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_source='umich', omics_genes = e_prot_list, tissue_type = 'tumor') # drop Normal samples\n",
    "prot_and_mutations = u.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "\n",
    "prot_df = prot_and_mutations.iloc[:,:-4] #drop Mutation and Location cols\n",
    "mut_type = mut_type[['Mutation']] # Get Mutation col that includes CNV\n",
    "merged = prot_df.join(mut_type) # merge \n",
    "\n",
    "# Create Truncation category and keep truncation and wt\n",
    "compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "trunc_wt = merged[get]\n",
    "trunc_wt['Mutation'] = np.where(\n",
    "                trunc_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "trunc_wt = rename_duplicate_cols(trunc_wt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "e_cols = list(trunc_wt.columns[:-1])\n",
    "\n",
    "# Get all pvals\n",
    "e_pval = u.wrap_ttest(trunc_wt, 'Mutation', e_cols, return_all = True, correction_method = 'fdr_bh', mincount = 5)\n",
    "e_pval = e_pval.rename(columns = {'Comparison': 'Umich_Proteomics','P_Value': 'EC_P_Value'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Differential expression with adaption for trunctation mutations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "t = trunc_wt[trunc_wt.Mutation == \"Truncation\"]\n",
    "wt = trunc_wt[trunc_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "trunc_med = t.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "en_d = {}\n",
    "\n",
    "for prot in e_cols:\n",
    "    if prot in trunc_med.index and prot in wt_med.index:\n",
    "        dif = trunc_med[prot] - wt_med[prot]\n",
    "        en_d[prot] = dif\n",
    "\n",
    "en_med = pd.DataFrame.from_dict(en_d, orient='index', columns=['EC_Median'])\n",
    "en_med = en_med.reset_index().rename(columns={'index':'Umich_Proteomics'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Umich_Proteomics</th>\n",
       "      <th>EC_P_Value</th>\n",
       "      <th>EC_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TJP3</td>\n",
       "      <td>0.002105</td>\n",
       "      <td>0.562653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABT1</td>\n",
       "      <td>0.002105</td>\n",
       "      <td>-0.479251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WDR75</td>\n",
       "      <td>0.002105</td>\n",
       "      <td>-0.384194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NOL10</td>\n",
       "      <td>0.002105</td>\n",
       "      <td>-0.366391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REXO4</td>\n",
       "      <td>0.002105</td>\n",
       "      <td>-0.444003</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Umich_Proteomics  EC_P_Value  EC_Median\n",
       "0             TJP3    0.002105   0.562653\n",
       "1             ABT1    0.002105  -0.479251\n",
       "2            WDR75    0.002105  -0.384194\n",
       "3            NOL10    0.002105  -0.366391\n",
       "4            REXO4    0.002105  -0.444003"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_merged = e_pval.merge(en_med, on='Umich_Proteomics',how='outer')\n",
    "e_merged = e_merged.replace(to_replace = r'_umich_proteomics', value = '', regex = True)\n",
    "\n",
    "# Create csv\n",
    "e_merged.to_csv('csv/Single_Cancer/EC_pval_medians_pancan.csv',index=False)\n",
    "e_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get number of significant genes in each cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add Gbm and En to merged_dfs dictionary\n",
    "merged_dfs['GBM'] = g_merged\n",
    "merged_dfs['EC'] = e_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD sig comparisons: 17\n",
      "LSCC sig comparisons: 1368\n",
      "BR sig comparisons: 0\n",
      "GBM sig comparisons: 603\n",
      "EC sig comparisons: 490\n",
      "\n",
      "Number of significant comparisons in at least 1 cancer: 2246\n"
     ]
    }
   ],
   "source": [
    "# Using corrected pvals from wrap_ttest, each cancer's cutoff is 0.05\n",
    "all_sig = []\n",
    "\n",
    "for cancer in merged_dfs:\n",
    "    df = merged_dfs[cancer]\n",
    "    sig_df = df.loc[df[cancer+'_P_Value'] < 0.05]\n",
    "    print(cancer, 'sig comparisons:', len(sig_df))\n",
    "    sig_list = list(sig_df['Umich_Proteomics'])\n",
    "    all_sig.append(sig_list)\n",
    "        \n",
    "\n",
    "flat_list = [item for sublist in all_sig for item in sublist] #change list of lists, to just one list\n",
    "sig = list(set(flat_list)) # keep only one if gene in multiple cancer sig lists\n",
    "print('\\nNumber of significant comparisons in at least 1 cancer:', len(sig))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get a list of significant genes in at least one cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant genes in at least one cancer: 2246\n"
     ]
    }
   ],
   "source": [
    "s = pd.Series(sig)\n",
    "print('Number of significant genes in at least one cancer:', len(s))\n",
    "s.to_csv('csv/list_sig_one_cancer_pancan.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get a list of significant genes in multiple cancers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant genes in mult cancers: 223\n"
     ]
    }
   ],
   "source": [
    "# sig in multiple cancers\n",
    "mult = list(set(i for i in flat_list if flat_list.count(i) > 1)) # Keep genes sig in more than 1 cancer\n",
    "print('Number of significant genes in mult cancers:', len(mult))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = pd.Series(mult)\n",
    "m = m.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "m.to_csv('csv/list_sig_multiple_cancers_pancan.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
