{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import operator\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "def wrap_ttest(df, label_column, comparison_columns=None, alpha=.05, return_all=False, correction_method='bonferroni', mincount=3, pval_return_corrected=True):\n",
    "    try:\n",
    "        '''Verify precondition that label column exists and has exactly 2 unique values'''\n",
    "        label_values = df[label_column].unique()\n",
    "        if len(label_values) != 2:\n",
    "            print(\"Incorrectly Formatted Dataframe! Label column must have exactly 2 unique values.\")\n",
    "            return None\n",
    "        \n",
    "        '''Partition dataframe into two sets, one for each of the two unique values from the label column'''\n",
    "        partition1 = df.loc[df[label_column] == label_values[0]]\n",
    "        partition2 = df.loc[df[label_column] == label_values[1]]\n",
    "\n",
    "        '''If no comparison columns specified, use all columns except the specified labed column'''\n",
    "        if not comparison_columns:\n",
    "            comparison_columns = list(df.columns)\n",
    "            comparison_columns.remove(label_column)\n",
    "\n",
    "        '''Determine the number of real valued columns on which we will do t-tests'''\n",
    "        number_of_comparisons = len(comparison_columns)\n",
    "\n",
    "        '''Store comparisons and p-values in two arrays'''\n",
    "        comparisons = []\n",
    "        pvals = []\n",
    "        \n",
    "        '''Loop through each comparison column, perform the t-test, and record the p-val'''\n",
    "        \n",
    "        for column in comparison_columns:\n",
    "            if len(partition1[column].dropna(axis=0)) <= mincount:\n",
    "                continue\n",
    "            elif len(partition2[column].dropna(axis=0)) <= mincount:\n",
    "                continue\n",
    "            else:\n",
    "                stat, pval = scipy.stats.ttest_ind(partition1[column].dropna(axis=0), partition2[column].dropna(axis=0))\n",
    "                comparisons.append(column)\n",
    "                pvals.append(pval)\n",
    "                \n",
    "        '''Correct for multiple testing to determine if each comparison meets the new cutoff'''\n",
    "        results = statsmodels.stats.multitest.multipletests(pvals=pvals, alpha=alpha, method=correction_method)\n",
    "        reject = results[0]\n",
    "\n",
    "        '''Format results in a pandas dataframe'''\n",
    "        results_df = pd.DataFrame(columns=['Comparison','P_Value'])\n",
    "\n",
    "        '''If return all, add all comparisons and p-values to dataframe'''\n",
    "        if return_all:\n",
    "            if pval_return_corrected:\n",
    "                results_df['Comparison'] = comparisons\n",
    "                results_df['P_Value'] = results[1]\n",
    "\n",
    "            else:\n",
    "                results_df['Comparison'] = comparisons\n",
    "                results_df['P_Value'] = pvals\n",
    "\n",
    "            '''Else only add significant comparisons'''\n",
    "        else:\n",
    "            for i in range(0, len(reject)):\n",
    "                if reject[i]:\n",
    "                    if pval_return_corrected:\n",
    "                        results_df = results_df.append({'Comparison':comparisons[i],'P_Value':results[1][i]}, ignore_index=True)\n",
    "                    else:\n",
    "                        results_df = results_df.append({'Comparison':comparisons[i],'P_Value':pvals[i]}, ignore_index=True)\n",
    "\n",
    "\n",
    "        '''Sort dataframe by ascending p-value'''\n",
    "        results_df = results_df.sort_values(by='P_Value', ascending=True)\n",
    "        results_df = results_df.reset_index(drop=True)\n",
    "\n",
    "        '''If results df is not empty, return it, else return None'''\n",
    "        if len(results_df) > 0:\n",
    "            return results_df\n",
    "        else:\n",
    "            return None\n",
    "\n",
    "\n",
    "    except:\n",
    "        print(\"Incorrectly Formatted Dataframe!\")\n",
    "        return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Your version of cptac (0.8.0) is out-of-date. Latest is 0.8.1. Please run 'pip install --upgrade cptac' to update it. (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 10)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load cancer data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "en = cptac.Endometrial()\n",
    "br = cptac.Brca()\n",
    "cl = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PIK3CA'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter out hotspot mutations\n",
    "Hotspots are:\n",
    "E542K\n",
    "E545K\n",
    "H1047R\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the mutation type, and proteomics for PIK3CA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 48 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    }
   ],
   "source": [
    "proteomics = en.get_proteomics(tissue_type=\"tumor\")\n",
    "mut_type = en.get_genotype_all_vars(\"PIK3CA\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join mutation type and proteomics together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "joined = mut_type.join(proteomics)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Select samples containing hotspot mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "en_hotspot_df = joined[joined.Location.str.contains('E542') | \n",
    "                    joined.Location.str.contains('E545') |\n",
    "                    joined.Location.str.contains('H1047')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join the proteomics for the wildtype mutations to the hotspot mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "\n",
    "endo_hotspot = en_hotspot_df.append(wildtype)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Drop unncecesary columns\n",
    "The resulting dataframe is what we will be working with from this point forward. It contains the mutations type and the proteomics for each gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "endo_prot_and_mutations = endo_hotspot.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "# endo_prot_and_mutations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run T-test and difference of Median for each cancer type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the median proteomic expression among all samples, for each gene.\n",
    "This is done for missense mutations and wildtype separately\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "missense = endo_prot_and_mutations[endo_prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = endo_prot_and_mutations[endo_prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #get all wildtype mutations\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the medain difference between missense and wildtype for each gene.\n",
    "This is done by (for each gene) subtracting the median of the missense mutations from the median of the wildtype.\n",
    "This means that if the difference is positive the proteomics of the wildtype have higher expression than the missense mutation. If the difference is negative then the missense mutation has a higher expression than the wildtype."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.01850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.06350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.08085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>0.19145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>0.15610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median\n",
       "A1BG                 0.01850\n",
       "A2M                 -0.06350\n",
       "A2ML1                1.08085\n",
       "A4GALT               0.19145\n",
       "AAAS                 0.15610"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_d = {}\n",
    "\n",
    "for gene in endo_prot_and_mutations:\n",
    "    if gene == \"Mutation\": continue\n",
    "    dif_endo =  missense_med[gene] - wt_med[gene]\n",
    "    endo_d[gene] = dif_endo\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(endo_d, orient='index', columns=['Difference_In_Median'])\n",
    "median_diff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.1800</td>\n",
       "      <td>-0.86300</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>0.22200</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.66500</td>\n",
       "      <td>1.28000</td>\n",
       "      <td>-0.33900</td>\n",
       "      <td>0.41200</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.087700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02290</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3320</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.02000</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.08590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.5280</td>\n",
       "      <td>-1.32000</td>\n",
       "      <td>0.4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.04000</td>\n",
       "      <td>-0.02130</td>\n",
       "      <td>-0.04790</td>\n",
       "      <td>0.41900</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001120</td>\n",
       "      <td>-0.14500</td>\n",
       "      <td>0.01050</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1510</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.54000</td>\n",
       "      <td>0.32000</td>\n",
       "      <td>-0.41900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00362</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.9240</td>\n",
       "      <td>-0.44500</td>\n",
       "      <td>1.5700</td>\n",
       "      <td>-0.67800</td>\n",
       "      <td>0.1730</td>\n",
       "      <td>0.43600</td>\n",
       "      <td>0.22700</td>\n",
       "      <td>-0.00248</td>\n",
       "      <td>0.47900</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.049600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.10800</td>\n",
       "      <td>-0.1530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>-0.33600</td>\n",
       "      <td>-0.82200</td>\n",
       "      <td>-0.03380</td>\n",
       "      <td>0.12100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00601</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.4540</td>\n",
       "      <td>-0.24200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2580</td>\n",
       "      <td>0.21900</td>\n",
       "      <td>-0.24900</td>\n",
       "      <td>-1.33000</td>\n",
       "      <td>0.20400</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.158000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.66700</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>0.4410</td>\n",
       "      <td>0.1300</td>\n",
       "      <td>-0.06590</td>\n",
       "      <td>-0.92300</td>\n",
       "      <td>-0.16300</td>\n",
       "      <td>0.11200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00605</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>0.59400</td>\n",
       "      <td>3.4000</td>\n",
       "      <td>0.15400</td>\n",
       "      <td>0.0932</td>\n",
       "      <td>0.28300</td>\n",
       "      <td>-0.07890</td>\n",
       "      <td>-0.61100</td>\n",
       "      <td>0.37000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.051100</td>\n",
       "      <td>-0.01360</td>\n",
       "      <td>0.21400</td>\n",
       "      <td>0.6830</td>\n",
       "      <td>0.4330</td>\n",
       "      <td>-0.2080</td>\n",
       "      <td>-0.34700</td>\n",
       "      <td>-0.91100</td>\n",
       "      <td>0.06920</td>\n",
       "      <td>-0.23200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00918</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.6980</td>\n",
       "      <td>-0.53700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0854</td>\n",
       "      <td>0.33500</td>\n",
       "      <td>0.06850</td>\n",
       "      <td>0.12300</td>\n",
       "      <td>0.27300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.209000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.12400</td>\n",
       "      <td>0.5880</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5680</td>\n",
       "      <td>-0.22200</td>\n",
       "      <td>-0.14800</td>\n",
       "      <td>0.06220</td>\n",
       "      <td>-0.23100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00921</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-1.19000</td>\n",
       "      <td>3.6600</td>\n",
       "      <td>1.27000</td>\n",
       "      <td>-0.1050</td>\n",
       "      <td>-0.00210</td>\n",
       "      <td>0.08130</td>\n",
       "      <td>-0.69100</td>\n",
       "      <td>-0.29500</td>\n",
       "      <td>...</td>\n",
       "      <td>0.097000</td>\n",
       "      <td>0.25400</td>\n",
       "      <td>0.15100</td>\n",
       "      <td>0.2850</td>\n",
       "      <td>-0.2390</td>\n",
       "      <td>-0.0681</td>\n",
       "      <td>-0.21800</td>\n",
       "      <td>-0.08730</td>\n",
       "      <td>-0.12700</td>\n",
       "      <td>0.37400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00947</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.4850</td>\n",
       "      <td>0.91700</td>\n",
       "      <td>0.3040</td>\n",
       "      <td>-0.00675</td>\n",
       "      <td>0.1780</td>\n",
       "      <td>-0.42300</td>\n",
       "      <td>0.22400</td>\n",
       "      <td>-0.24700</td>\n",
       "      <td>0.07870</td>\n",
       "      <td>...</td>\n",
       "      <td>0.042900</td>\n",
       "      <td>-0.21200</td>\n",
       "      <td>0.16900</td>\n",
       "      <td>0.5210</td>\n",
       "      <td>-0.1470</td>\n",
       "      <td>0.3540</td>\n",
       "      <td>0.17000</td>\n",
       "      <td>-1.01000</td>\n",
       "      <td>-0.11000</td>\n",
       "      <td>0.13100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00323</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.8170</td>\n",
       "      <td>-1.13000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>0.37100</td>\n",
       "      <td>0.64400</td>\n",
       "      <td>0.28100</td>\n",
       "      <td>0.11400</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.022200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.27900</td>\n",
       "      <td>0.4230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2940</td>\n",
       "      <td>-0.13000</td>\n",
       "      <td>-0.10600</td>\n",
       "      <td>-0.29900</td>\n",
       "      <td>0.38500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00324</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.4490</td>\n",
       "      <td>-0.44000</td>\n",
       "      <td>1.1500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6450</td>\n",
       "      <td>0.31600</td>\n",
       "      <td>0.08090</td>\n",
       "      <td>-0.47000</td>\n",
       "      <td>0.80400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.044900</td>\n",
       "      <td>0.04520</td>\n",
       "      <td>-0.15200</td>\n",
       "      <td>-0.1970</td>\n",
       "      <td>0.2950</td>\n",
       "      <td>-0.1880</td>\n",
       "      <td>-0.26500</td>\n",
       "      <td>0.25000</td>\n",
       "      <td>-0.11300</td>\n",
       "      <td>-0.30200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00383</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.0729</td>\n",
       "      <td>0.56700</td>\n",
       "      <td>1.7100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2780</td>\n",
       "      <td>0.64600</td>\n",
       "      <td>0.37800</td>\n",
       "      <td>-0.01460</td>\n",
       "      <td>1.17000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.279000</td>\n",
       "      <td>-0.37800</td>\n",
       "      <td>0.01250</td>\n",
       "      <td>-0.3050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1490</td>\n",
       "      <td>-0.05260</td>\n",
       "      <td>-0.03110</td>\n",
       "      <td>0.43200</td>\n",
       "      <td>0.00801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00743</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.0900</td>\n",
       "      <td>-0.95800</td>\n",
       "      <td>-0.6830</td>\n",
       "      <td>0.48600</td>\n",
       "      <td>0.4540</td>\n",
       "      <td>0.26500</td>\n",
       "      <td>0.46700</td>\n",
       "      <td>-0.67200</td>\n",
       "      <td>0.11900</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.204000</td>\n",
       "      <td>-0.10700</td>\n",
       "      <td>0.23800</td>\n",
       "      <td>0.6160</td>\n",
       "      <td>0.4400</td>\n",
       "      <td>-0.6070</td>\n",
       "      <td>0.01930</td>\n",
       "      <td>0.30500</td>\n",
       "      <td>-0.43400</td>\n",
       "      <td>-0.22900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01212</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>-0.14200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2110</td>\n",
       "      <td>0.92700</td>\n",
       "      <td>0.49200</td>\n",
       "      <td>-0.41600</td>\n",
       "      <td>0.40000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.040600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.09240</td>\n",
       "      <td>0.1210</td>\n",
       "      <td>0.4330</td>\n",
       "      <td>-1.0500</td>\n",
       "      <td>-0.31600</td>\n",
       "      <td>-0.40000</td>\n",
       "      <td>0.12300</td>\n",
       "      <td>-0.39100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01510</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.2690</td>\n",
       "      <td>0.94400</td>\n",
       "      <td>1.2000</td>\n",
       "      <td>-0.15500</td>\n",
       "      <td>0.2660</td>\n",
       "      <td>-0.54800</td>\n",
       "      <td>-0.10500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.40100</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.204000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.58300</td>\n",
       "      <td>0.9130</td>\n",
       "      <td>0.2490</td>\n",
       "      <td>-0.5150</td>\n",
       "      <td>-0.15100</td>\n",
       "      <td>-1.39000</td>\n",
       "      <td>-0.12100</td>\n",
       "      <td>-0.42600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.6850</td>\n",
       "      <td>-1.07000</td>\n",
       "      <td>-0.6840</td>\n",
       "      <td>0.98400</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.33400</td>\n",
       "      <td>1.30000</td>\n",
       "      <td>0.13900</td>\n",
       "      <td>1.33000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.035600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.36300</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.7370</td>\n",
       "      <td>-0.5640</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.13000</td>\n",
       "      <td>-0.07570</td>\n",
       "      <td>-0.47300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.6700</td>\n",
       "      <td>-1.19000</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>0.24300</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.75700</td>\n",
       "      <td>0.74000</td>\n",
       "      <td>-0.92900</td>\n",
       "      <td>0.22900</td>\n",
       "      <td>...</td>\n",
       "      <td>0.072500</td>\n",
       "      <td>-0.05520</td>\n",
       "      <td>-0.07140</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.1560</td>\n",
       "      <td>-0.3980</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.79700</td>\n",
       "      <td>-0.03010</td>\n",
       "      <td>-0.46700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00136</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.0800</td>\n",
       "      <td>-0.70800</td>\n",
       "      <td>-0.1260</td>\n",
       "      <td>-0.42600</td>\n",
       "      <td>-0.1140</td>\n",
       "      <td>-0.11100</td>\n",
       "      <td>0.89500</td>\n",
       "      <td>1.26000</td>\n",
       "      <td>0.15700</td>\n",
       "      <td>...</td>\n",
       "      <td>0.455000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.39700</td>\n",
       "      <td>-0.9990</td>\n",
       "      <td>-0.7300</td>\n",
       "      <td>-0.0229</td>\n",
       "      <td>-0.33100</td>\n",
       "      <td>-1.16000</td>\n",
       "      <td>-0.11600</td>\n",
       "      <td>0.00250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00137</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>-0.70800</td>\n",
       "      <td>-0.8080</td>\n",
       "      <td>-0.07090</td>\n",
       "      <td>0.1380</td>\n",
       "      <td>0.65600</td>\n",
       "      <td>-0.28000</td>\n",
       "      <td>-0.12800</td>\n",
       "      <td>0.21700</td>\n",
       "      <td>...</td>\n",
       "      <td>0.015800</td>\n",
       "      <td>-0.00491</td>\n",
       "      <td>0.18000</td>\n",
       "      <td>0.5130</td>\n",
       "      <td>0.5490</td>\n",
       "      <td>-0.6810</td>\n",
       "      <td>-0.28500</td>\n",
       "      <td>-0.56400</td>\n",
       "      <td>-0.08760</td>\n",
       "      <td>0.00698</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00145</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.7160</td>\n",
       "      <td>-0.88500</td>\n",
       "      <td>2.8200</td>\n",
       "      <td>-0.34300</td>\n",
       "      <td>0.1470</td>\n",
       "      <td>0.44500</td>\n",
       "      <td>-0.05650</td>\n",
       "      <td>-0.83800</td>\n",
       "      <td>0.04900</td>\n",
       "      <td>...</td>\n",
       "      <td>0.084500</td>\n",
       "      <td>-0.64700</td>\n",
       "      <td>0.17500</td>\n",
       "      <td>0.2120</td>\n",
       "      <td>-0.3240</td>\n",
       "      <td>-0.3500</td>\n",
       "      <td>-0.37700</td>\n",
       "      <td>0.38800</td>\n",
       "      <td>0.01100</td>\n",
       "      <td>0.13400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00361</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.3400</td>\n",
       "      <td>-1.87000</td>\n",
       "      <td>-0.2580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1290</td>\n",
       "      <td>2.56000</td>\n",
       "      <td>1.03000</td>\n",
       "      <td>-0.39600</td>\n",
       "      <td>-0.07630</td>\n",
       "      <td>...</td>\n",
       "      <td>0.033000</td>\n",
       "      <td>-0.41900</td>\n",
       "      <td>0.32700</td>\n",
       "      <td>0.1030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6380</td>\n",
       "      <td>-0.51400</td>\n",
       "      <td>-1.42000</td>\n",
       "      <td>0.21800</td>\n",
       "      <td>-0.06110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00413</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.1500</td>\n",
       "      <td>0.98700</td>\n",
       "      <td>5.0700</td>\n",
       "      <td>0.30200</td>\n",
       "      <td>-0.1690</td>\n",
       "      <td>0.39800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.27400</td>\n",
       "      <td>1.09000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.348000</td>\n",
       "      <td>-0.48400</td>\n",
       "      <td>0.14200</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>-0.1420</td>\n",
       "      <td>0.0770</td>\n",
       "      <td>-0.15700</td>\n",
       "      <td>-0.87200</td>\n",
       "      <td>-0.00663</td>\n",
       "      <td>-0.01750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00449</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.1810</td>\n",
       "      <td>-0.28300</td>\n",
       "      <td>0.6890</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.1300</td>\n",
       "      <td>-0.06830</td>\n",
       "      <td>-0.37600</td>\n",
       "      <td>-0.08380</td>\n",
       "      <td>0.16000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.228000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02060</td>\n",
       "      <td>-0.9700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.4520</td>\n",
       "      <td>0.17200</td>\n",
       "      <td>0.62600</td>\n",
       "      <td>0.34200</td>\n",
       "      <td>-0.20300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00767</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.8310</td>\n",
       "      <td>-0.81100</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>0.21200</td>\n",
       "      <td>-0.0578</td>\n",
       "      <td>0.16800</td>\n",
       "      <td>0.42700</td>\n",
       "      <td>-0.75000</td>\n",
       "      <td>0.17600</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>0.18400</td>\n",
       "      <td>0.08680</td>\n",
       "      <td>0.1440</td>\n",
       "      <td>-0.1370</td>\n",
       "      <td>-0.4180</td>\n",
       "      <td>-0.18300</td>\n",
       "      <td>-0.76900</td>\n",
       "      <td>-0.25600</td>\n",
       "      <td>-0.37200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00780</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.5510</td>\n",
       "      <td>0.20300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.4930</td>\n",
       "      <td>0.21100</td>\n",
       "      <td>1.06000</td>\n",
       "      <td>-0.28000</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>...</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.14200</td>\n",
       "      <td>-0.5490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3900</td>\n",
       "      <td>-0.09720</td>\n",
       "      <td>0.00108</td>\n",
       "      <td>0.20000</td>\n",
       "      <td>-0.19400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00932</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.2030</td>\n",
       "      <td>-0.51000</td>\n",
       "      <td>0.1510</td>\n",
       "      <td>1.71000</td>\n",
       "      <td>0.0340</td>\n",
       "      <td>0.08550</td>\n",
       "      <td>-0.04640</td>\n",
       "      <td>0.00430</td>\n",
       "      <td>-0.33700</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.123000</td>\n",
       "      <td>0.14200</td>\n",
       "      <td>-0.17400</td>\n",
       "      <td>-0.5110</td>\n",
       "      <td>-0.1320</td>\n",
       "      <td>0.1530</td>\n",
       "      <td>-0.04300</td>\n",
       "      <td>0.76400</td>\n",
       "      <td>-0.21600</td>\n",
       "      <td>0.07370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00946</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.4920</td>\n",
       "      <td>-0.37100</td>\n",
       "      <td>0.0263</td>\n",
       "      <td>-0.37000</td>\n",
       "      <td>0.1720</td>\n",
       "      <td>0.06760</td>\n",
       "      <td>0.57200</td>\n",
       "      <td>-0.02420</td>\n",
       "      <td>0.64300</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.035100</td>\n",
       "      <td>-0.55900</td>\n",
       "      <td>-0.20600</td>\n",
       "      <td>0.1780</td>\n",
       "      <td>-0.0634</td>\n",
       "      <td>-0.3140</td>\n",
       "      <td>-0.07460</td>\n",
       "      <td>0.08190</td>\n",
       "      <td>0.20500</td>\n",
       "      <td>-0.30500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00949</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.2430</td>\n",
       "      <td>-0.87100</td>\n",
       "      <td>-0.0925</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0807</td>\n",
       "      <td>0.83600</td>\n",
       "      <td>0.51000</td>\n",
       "      <td>-0.24800</td>\n",
       "      <td>-0.08340</td>\n",
       "      <td>...</td>\n",
       "      <td>0.160000</td>\n",
       "      <td>-0.33200</td>\n",
       "      <td>0.02140</td>\n",
       "      <td>-0.1550</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0893</td>\n",
       "      <td>0.01620</td>\n",
       "      <td>0.37700</td>\n",
       "      <td>0.03220</td>\n",
       "      <td>-0.13000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01248</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.2860</td>\n",
       "      <td>-0.51100</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>-0.45900</td>\n",
       "      <td>-0.1520</td>\n",
       "      <td>0.15400</td>\n",
       "      <td>-0.00265</td>\n",
       "      <td>0.20500</td>\n",
       "      <td>-0.13700</td>\n",
       "      <td>...</td>\n",
       "      <td>0.294000</td>\n",
       "      <td>-0.51800</td>\n",
       "      <td>-0.13200</td>\n",
       "      <td>-0.7550</td>\n",
       "      <td>-0.0709</td>\n",
       "      <td>-0.3550</td>\n",
       "      <td>0.14000</td>\n",
       "      <td>1.72000</td>\n",
       "      <td>0.17800</td>\n",
       "      <td>-0.08030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01249</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.2280</td>\n",
       "      <td>-0.24900</td>\n",
       "      <td>1.8600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.9340</td>\n",
       "      <td>-0.28600</td>\n",
       "      <td>0.50700</td>\n",
       "      <td>-0.65900</td>\n",
       "      <td>0.11900</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.145000</td>\n",
       "      <td>-1.17000</td>\n",
       "      <td>-0.59300</td>\n",
       "      <td>-0.7900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4170</td>\n",
       "      <td>0.94400</td>\n",
       "      <td>-0.12000</td>\n",
       "      <td>0.08110</td>\n",
       "      <td>0.48500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01252</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>1.2700</td>\n",
       "      <td>1.60000</td>\n",
       "      <td>0.0955</td>\n",
       "      <td>0.47800</td>\n",
       "      <td>-0.0483</td>\n",
       "      <td>0.08740</td>\n",
       "      <td>-0.06390</td>\n",
       "      <td>-0.04940</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.158000</td>\n",
       "      <td>-0.09790</td>\n",
       "      <td>-0.30400</td>\n",
       "      <td>-0.5420</td>\n",
       "      <td>-0.3020</td>\n",
       "      <td>-0.3070</td>\n",
       "      <td>-0.05490</td>\n",
       "      <td>-0.02140</td>\n",
       "      <td>-0.03970</td>\n",
       "      <td>-0.17100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01256</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.2680</td>\n",
       "      <td>-0.45100</td>\n",
       "      <td>-0.8000</td>\n",
       "      <td>0.04250</td>\n",
       "      <td>-0.0502</td>\n",
       "      <td>0.32200</td>\n",
       "      <td>0.07180</td>\n",
       "      <td>-0.08760</td>\n",
       "      <td>-0.17300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.216000</td>\n",
       "      <td>0.11200</td>\n",
       "      <td>-0.15100</td>\n",
       "      <td>-0.0164</td>\n",
       "      <td>-0.2430</td>\n",
       "      <td>-0.3080</td>\n",
       "      <td>-0.06810</td>\n",
       "      <td>1.18000</td>\n",
       "      <td>-0.05350</td>\n",
       "      <td>-0.02080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01257</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.6390</td>\n",
       "      <td>-0.63000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0175</td>\n",
       "      <td>0.89000</td>\n",
       "      <td>0.32500</td>\n",
       "      <td>-0.07900</td>\n",
       "      <td>-0.14600</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.067000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.13700</td>\n",
       "      <td>0.9570</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2670</td>\n",
       "      <td>-0.26700</td>\n",
       "      <td>-0.45200</td>\n",
       "      <td>0.01940</td>\n",
       "      <td>-0.05460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01275</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.1260</td>\n",
       "      <td>0.61900</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>-0.37300</td>\n",
       "      <td>0.2040</td>\n",
       "      <td>0.05550</td>\n",
       "      <td>0.16300</td>\n",
       "      <td>-0.90700</td>\n",
       "      <td>-0.10400</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.004450</td>\n",
       "      <td>-0.39900</td>\n",
       "      <td>0.09010</td>\n",
       "      <td>0.2780</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>-0.2750</td>\n",
       "      <td>-0.28300</td>\n",
       "      <td>0.41800</td>\n",
       "      <td>-0.02290</td>\n",
       "      <td>0.12700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01282</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.5430</td>\n",
       "      <td>-0.47100</td>\n",
       "      <td>0.0928</td>\n",
       "      <td>-0.16900</td>\n",
       "      <td>-0.1090</td>\n",
       "      <td>-0.27800</td>\n",
       "      <td>-0.23600</td>\n",
       "      <td>-0.31200</td>\n",
       "      <td>-0.13100</td>\n",
       "      <td>...</td>\n",
       "      <td>0.264000</td>\n",
       "      <td>-0.45900</td>\n",
       "      <td>0.01500</td>\n",
       "      <td>-0.2710</td>\n",
       "      <td>-0.0568</td>\n",
       "      <td>-0.3570</td>\n",
       "      <td>0.19200</td>\n",
       "      <td>-0.09430</td>\n",
       "      <td>0.01090</td>\n",
       "      <td>-0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01307</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.6230</td>\n",
       "      <td>0.00807</td>\n",
       "      <td>0.7030</td>\n",
       "      <td>0.26100</td>\n",
       "      <td>0.1060</td>\n",
       "      <td>1.33000</td>\n",
       "      <td>1.15000</td>\n",
       "      <td>0.23400</td>\n",
       "      <td>-0.14200</td>\n",
       "      <td>...</td>\n",
       "      <td>0.126000</td>\n",
       "      <td>-0.54600</td>\n",
       "      <td>0.48800</td>\n",
       "      <td>0.4850</td>\n",
       "      <td>-0.0923</td>\n",
       "      <td>0.1470</td>\n",
       "      <td>0.08140</td>\n",
       "      <td>-0.34800</td>\n",
       "      <td>0.39300</td>\n",
       "      <td>-0.19000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01312</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.5660</td>\n",
       "      <td>-0.32600</td>\n",
       "      <td>-0.0968</td>\n",
       "      <td>-0.15000</td>\n",
       "      <td>-0.1850</td>\n",
       "      <td>-0.00438</td>\n",
       "      <td>0.39300</td>\n",
       "      <td>-0.29100</td>\n",
       "      <td>0.12300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.134000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.14300</td>\n",
       "      <td>-0.7900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.3300</td>\n",
       "      <td>0.12900</td>\n",
       "      <td>-0.40400</td>\n",
       "      <td>0.17700</td>\n",
       "      <td>-0.12000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01744</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.5060</td>\n",
       "      <td>-0.21700</td>\n",
       "      <td>-1.1800</td>\n",
       "      <td>-1.00000</td>\n",
       "      <td>0.1390</td>\n",
       "      <td>0.65400</td>\n",
       "      <td>0.78900</td>\n",
       "      <td>-0.54700</td>\n",
       "      <td>0.49700</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054200</td>\n",
       "      <td>-0.67800</td>\n",
       "      <td>0.34500</td>\n",
       "      <td>0.8860</td>\n",
       "      <td>1.5100</td>\n",
       "      <td>-0.6210</td>\n",
       "      <td>-0.24500</td>\n",
       "      <td>-1.02000</td>\n",
       "      <td>0.26800</td>\n",
       "      <td>-0.02040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01925</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.2720</td>\n",
       "      <td>-1.37000</td>\n",
       "      <td>-0.6660</td>\n",
       "      <td>-0.54400</td>\n",
       "      <td>0.3860</td>\n",
       "      <td>-0.48000</td>\n",
       "      <td>-1.49000</td>\n",
       "      <td>-0.21600</td>\n",
       "      <td>0.05410</td>\n",
       "      <td>...</td>\n",
       "      <td>0.117000</td>\n",
       "      <td>0.00480</td>\n",
       "      <td>0.32200</td>\n",
       "      <td>0.1850</td>\n",
       "      <td>1.5500</td>\n",
       "      <td>0.2040</td>\n",
       "      <td>0.01180</td>\n",
       "      <td>0.34800</td>\n",
       "      <td>0.01220</td>\n",
       "      <td>0.10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00151</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.5560</td>\n",
       "      <td>-0.53600</td>\n",
       "      <td>-1.1600</td>\n",
       "      <td>0.02390</td>\n",
       "      <td>0.0446</td>\n",
       "      <td>0.91900</td>\n",
       "      <td>0.15600</td>\n",
       "      <td>-0.13800</td>\n",
       "      <td>0.81200</td>\n",
       "      <td>...</td>\n",
       "      <td>0.105000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.09630</td>\n",
       "      <td>0.2430</td>\n",
       "      <td>0.6430</td>\n",
       "      <td>-0.5800</td>\n",
       "      <td>0.04090</td>\n",
       "      <td>-0.71000</td>\n",
       "      <td>0.42400</td>\n",
       "      <td>-0.02360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00322</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.5250</td>\n",
       "      <td>-0.79100</td>\n",
       "      <td>0.2120</td>\n",
       "      <td>0.31300</td>\n",
       "      <td>-0.3730</td>\n",
       "      <td>1.02000</td>\n",
       "      <td>-0.44000</td>\n",
       "      <td>0.10100</td>\n",
       "      <td>0.25500</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.051400</td>\n",
       "      <td>-0.11500</td>\n",
       "      <td>-0.19400</td>\n",
       "      <td>0.3600</td>\n",
       "      <td>0.2040</td>\n",
       "      <td>0.2980</td>\n",
       "      <td>0.01610</td>\n",
       "      <td>0.24100</td>\n",
       "      <td>-0.60400</td>\n",
       "      <td>0.14500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00335</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.1260</td>\n",
       "      <td>0.00181</td>\n",
       "      <td>-0.0442</td>\n",
       "      <td>1.68000</td>\n",
       "      <td>0.0711</td>\n",
       "      <td>-0.32100</td>\n",
       "      <td>-0.34600</td>\n",
       "      <td>0.46700</td>\n",
       "      <td>0.75700</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.199000</td>\n",
       "      <td>-0.37100</td>\n",
       "      <td>-0.27400</td>\n",
       "      <td>0.8760</td>\n",
       "      <td>1.4000</td>\n",
       "      <td>0.7160</td>\n",
       "      <td>0.18300</td>\n",
       "      <td>0.45800</td>\n",
       "      <td>-0.41500</td>\n",
       "      <td>0.27200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00339</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.5600</td>\n",
       "      <td>-0.24600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1680</td>\n",
       "      <td>1.17000</td>\n",
       "      <td>1.41000</td>\n",
       "      <td>0.24000</td>\n",
       "      <td>0.53800</td>\n",
       "      <td>...</td>\n",
       "      <td>0.233000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.05930</td>\n",
       "      <td>0.6920</td>\n",
       "      <td>0.1860</td>\n",
       "      <td>0.3390</td>\n",
       "      <td>0.22100</td>\n",
       "      <td>-0.01610</td>\n",
       "      <td>-0.06920</td>\n",
       "      <td>0.26600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00377</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.8660</td>\n",
       "      <td>1.55000</td>\n",
       "      <td>1.1700</td>\n",
       "      <td>0.77600</td>\n",
       "      <td>0.0823</td>\n",
       "      <td>0.50700</td>\n",
       "      <td>0.45300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00740</td>\n",
       "      <td>...</td>\n",
       "      <td>0.134000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.29900</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>1.4600</td>\n",
       "      <td>-0.5700</td>\n",
       "      <td>0.03700</td>\n",
       "      <td>-0.50100</td>\n",
       "      <td>-0.04150</td>\n",
       "      <td>-0.18900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00386</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.9790</td>\n",
       "      <td>-1.11000</td>\n",
       "      <td>0.7740</td>\n",
       "      <td>-1.16000</td>\n",
       "      <td>-0.0186</td>\n",
       "      <td>0.44700</td>\n",
       "      <td>0.84900</td>\n",
       "      <td>-0.23900</td>\n",
       "      <td>0.25400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.519000</td>\n",
       "      <td>-0.65000</td>\n",
       "      <td>0.29800</td>\n",
       "      <td>0.4890</td>\n",
       "      <td>0.0383</td>\n",
       "      <td>0.2830</td>\n",
       "      <td>-0.45100</td>\n",
       "      <td>-0.65900</td>\n",
       "      <td>0.05340</td>\n",
       "      <td>0.30100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00389</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.8070</td>\n",
       "      <td>-0.37900</td>\n",
       "      <td>-1.0600</td>\n",
       "      <td>-0.63300</td>\n",
       "      <td>-0.1960</td>\n",
       "      <td>0.06540</td>\n",
       "      <td>0.39100</td>\n",
       "      <td>-0.13000</td>\n",
       "      <td>0.15200</td>\n",
       "      <td>...</td>\n",
       "      <td>0.048500</td>\n",
       "      <td>-0.34200</td>\n",
       "      <td>-0.06940</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>0.2820</td>\n",
       "      <td>-0.7190</td>\n",
       "      <td>-0.23400</td>\n",
       "      <td>-0.16100</td>\n",
       "      <td>0.10700</td>\n",
       "      <td>0.12700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00734</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.2650</td>\n",
       "      <td>-0.19800</td>\n",
       "      <td>0.2920</td>\n",
       "      <td>-0.02110</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>0.82700</td>\n",
       "      <td>-0.10400</td>\n",
       "      <td>0.16700</td>\n",
       "      <td>0.37900</td>\n",
       "      <td>...</td>\n",
       "      <td>0.108000</td>\n",
       "      <td>-0.16700</td>\n",
       "      <td>0.00762</td>\n",
       "      <td>-0.0456</td>\n",
       "      <td>-0.4750</td>\n",
       "      <td>-0.0226</td>\n",
       "      <td>-0.14800</td>\n",
       "      <td>-0.51400</td>\n",
       "      <td>0.05670</td>\n",
       "      <td>-0.00663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00836</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.8550</td>\n",
       "      <td>0.87600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1270</td>\n",
       "      <td>-0.06820</td>\n",
       "      <td>0.16400</td>\n",
       "      <td>-0.02610</td>\n",
       "      <td>0.00301</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.215000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.05010</td>\n",
       "      <td>0.1800</td>\n",
       "      <td>0.2330</td>\n",
       "      <td>-0.5550</td>\n",
       "      <td>0.19900</td>\n",
       "      <td>0.59000</td>\n",
       "      <td>0.23700</td>\n",
       "      <td>-0.29800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00847</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.3000</td>\n",
       "      <td>-0.96500</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>-0.05380</td>\n",
       "      <td>0.1430</td>\n",
       "      <td>1.15000</td>\n",
       "      <td>0.72600</td>\n",
       "      <td>-0.34400</td>\n",
       "      <td>0.49300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.015000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.31800</td>\n",
       "      <td>0.6660</td>\n",
       "      <td>0.2170</td>\n",
       "      <td>-0.4360</td>\n",
       "      <td>-0.10100</td>\n",
       "      <td>-0.77400</td>\n",
       "      <td>0.20900</td>\n",
       "      <td>0.25300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00850</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.6860</td>\n",
       "      <td>-0.55500</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>-0.30600</td>\n",
       "      <td>-0.1080</td>\n",
       "      <td>0.56600</td>\n",
       "      <td>1.20000</td>\n",
       "      <td>-0.15700</td>\n",
       "      <td>0.11300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.117000</td>\n",
       "      <td>-0.47700</td>\n",
       "      <td>0.32500</td>\n",
       "      <td>0.3400</td>\n",
       "      <td>-0.2590</td>\n",
       "      <td>0.2090</td>\n",
       "      <td>-0.17700</td>\n",
       "      <td>-0.26200</td>\n",
       "      <td>0.04620</td>\n",
       "      <td>-0.02090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-00866</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.2400</td>\n",
       "      <td>-0.08430</td>\n",
       "      <td>0.0142</td>\n",
       "      <td>1.72000</td>\n",
       "      <td>-0.0720</td>\n",
       "      <td>0.53000</td>\n",
       "      <td>-0.19500</td>\n",
       "      <td>-0.29300</td>\n",
       "      <td>0.18200</td>\n",
       "      <td>...</td>\n",
       "      <td>0.077300</td>\n",
       "      <td>0.03270</td>\n",
       "      <td>0.30800</td>\n",
       "      <td>-0.1500</td>\n",
       "      <td>-0.0672</td>\n",
       "      <td>0.0697</td>\n",
       "      <td>-0.22400</td>\n",
       "      <td>-0.66900</td>\n",
       "      <td>0.03750</td>\n",
       "      <td>-0.25500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01211</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.6360</td>\n",
       "      <td>-0.37100</td>\n",
       "      <td>0.9730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6370</td>\n",
       "      <td>-0.16600</td>\n",
       "      <td>-0.12200</td>\n",
       "      <td>0.23200</td>\n",
       "      <td>0.37900</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.416000</td>\n",
       "      <td>-0.75100</td>\n",
       "      <td>0.17900</td>\n",
       "      <td>0.6580</td>\n",
       "      <td>0.4700</td>\n",
       "      <td>0.2700</td>\n",
       "      <td>-0.14800</td>\n",
       "      <td>-0.13900</td>\n",
       "      <td>-0.02990</td>\n",
       "      <td>0.13100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01217</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.7430</td>\n",
       "      <td>-1.37000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1930</td>\n",
       "      <td>0.69400</td>\n",
       "      <td>0.33800</td>\n",
       "      <td>0.09890</td>\n",
       "      <td>-0.01370</td>\n",
       "      <td>...</td>\n",
       "      <td>0.181000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.19600</td>\n",
       "      <td>0.2740</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5010</td>\n",
       "      <td>-0.04680</td>\n",
       "      <td>0.65300</td>\n",
       "      <td>-0.04640</td>\n",
       "      <td>-0.16000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01219</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.2950</td>\n",
       "      <td>-0.05890</td>\n",
       "      <td>0.4900</td>\n",
       "      <td>-0.10900</td>\n",
       "      <td>-0.0495</td>\n",
       "      <td>0.03740</td>\n",
       "      <td>0.10200</td>\n",
       "      <td>-0.32100</td>\n",
       "      <td>0.18300</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.290000</td>\n",
       "      <td>0.04570</td>\n",
       "      <td>0.24600</td>\n",
       "      <td>0.5890</td>\n",
       "      <td>0.3930</td>\n",
       "      <td>-0.1990</td>\n",
       "      <td>-0.15900</td>\n",
       "      <td>-1.09000</td>\n",
       "      <td>0.06270</td>\n",
       "      <td>-0.37900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.2900</td>\n",
       "      <td>-0.32000</td>\n",
       "      <td>3.1700</td>\n",
       "      <td>-0.90700</td>\n",
       "      <td>0.0317</td>\n",
       "      <td>-0.04250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.27300</td>\n",
       "      <td>-0.07470</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.179000</td>\n",
       "      <td>-0.52100</td>\n",
       "      <td>0.05230</td>\n",
       "      <td>0.5150</td>\n",
       "      <td>0.5850</td>\n",
       "      <td>0.1380</td>\n",
       "      <td>-0.19000</td>\n",
       "      <td>-0.96600</td>\n",
       "      <td>-0.00627</td>\n",
       "      <td>-0.24900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.2660</td>\n",
       "      <td>1.39000</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.47000</td>\n",
       "      <td>0.3980</td>\n",
       "      <td>-0.13400</td>\n",
       "      <td>0.46100</td>\n",
       "      <td>1.04000</td>\n",
       "      <td>0.36300</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.140000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.01220</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>0.5530</td>\n",
       "      <td>0.3870</td>\n",
       "      <td>0.06420</td>\n",
       "      <td>-0.43700</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.49800</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation    A1BG      A2M   A2ML1   A4GALT    AAAS  \\\n",
       "Patient_ID                                                                \n",
       "C3L-00006   Missense_Mutation -1.1800 -0.86300 -0.8020  0.22200  0.2560   \n",
       "C3L-00032   Missense_Mutation -0.5280 -1.32000  0.4350      NaN -0.2400   \n",
       "C3L-00362   Missense_Mutation -0.9240 -0.44500  1.5700 -0.67800  0.1730   \n",
       "C3L-00601   Missense_Mutation -0.4540 -0.24200     NaN      NaN  0.2580   \n",
       "C3L-00605   Missense_Mutation -0.2400  0.59400  3.4000  0.15400  0.0932   \n",
       "C3L-00918   Missense_Mutation -0.6980 -0.53700     NaN      NaN -0.0854   \n",
       "C3L-00921   Missense_Mutation -1.1900 -1.19000  3.6600  1.27000 -0.1050   \n",
       "C3L-00947   Missense_Mutation -0.4850  0.91700  0.3040 -0.00675  0.1780   \n",
       "C3N-00323   Missense_Mutation -0.8170 -1.13000     NaN      NaN  0.1020   \n",
       "C3N-00324   Missense_Mutation -0.4490 -0.44000  1.1500      NaN -0.6450   \n",
       "C3N-00383   Missense_Mutation  0.0729  0.56700  1.7100      NaN -0.2780   \n",
       "C3N-00743   Missense_Mutation -1.0900 -0.95800 -0.6830  0.48600  0.4540   \n",
       "C3N-01212   Missense_Mutation  0.0308 -0.14200     NaN      NaN  0.2110   \n",
       "C3N-01510   Missense_Mutation -0.2690  0.94400  1.2000 -0.15500  0.2660   \n",
       "C3L-00008      Wildtype_Tumor -0.6850 -1.07000 -0.6840  0.98400  0.1350   \n",
       "C3L-00090      Wildtype_Tumor -1.6700 -1.19000 -0.4430  0.24300 -0.0993   \n",
       "C3L-00136      Wildtype_Tumor -1.0800 -0.70800 -0.1260 -0.42600 -0.1140   \n",
       "C3L-00137      Wildtype_Tumor -1.3200 -0.70800 -0.8080 -0.07090  0.1380   \n",
       "C3L-00145      Wildtype_Tumor -0.7160 -0.88500  2.8200 -0.34300  0.1470   \n",
       "C3L-00361      Wildtype_Tumor -1.3400 -1.87000 -0.2580      NaN  0.1290   \n",
       "C3L-00413      Wildtype_Tumor  0.1500  0.98700  5.0700  0.30200 -0.1690   \n",
       "C3L-00449      Wildtype_Tumor  0.1810 -0.28300  0.6890  0.10400 -0.1300   \n",
       "C3L-00767      Wildtype_Tumor -0.8310 -0.81100 -0.3200  0.21200 -0.0578   \n",
       "C3L-00780      Wildtype_Tumor  0.5510  0.20300     NaN      NaN -0.4930   \n",
       "C3L-00932      Wildtype_Tumor  0.2030 -0.51000  0.1510  1.71000  0.0340   \n",
       "C3L-00946      Wildtype_Tumor -0.4920 -0.37100  0.0263 -0.37000  0.1720   \n",
       "C3L-00949      Wildtype_Tumor -0.2430 -0.87100 -0.0925      NaN -0.0807   \n",
       "C3L-01248      Wildtype_Tumor -0.2860 -0.51100  1.0700 -0.45900 -0.1520   \n",
       "C3L-01249      Wildtype_Tumor  0.2280 -0.24900  1.8600      NaN -0.9340   \n",
       "C3L-01252      Wildtype_Tumor  1.2700  1.60000  0.0955  0.47800 -0.0483   \n",
       "C3L-01256      Wildtype_Tumor -0.2680 -0.45100 -0.8000  0.04250 -0.0502   \n",
       "C3L-01257      Wildtype_Tumor -0.6390 -0.63000     NaN      NaN -0.0175   \n",
       "C3L-01275      Wildtype_Tumor  0.1260  0.61900  0.3740 -0.37300  0.2040   \n",
       "C3L-01282      Wildtype_Tumor -0.5430 -0.47100  0.0928 -0.16900 -0.1090   \n",
       "C3L-01307      Wildtype_Tumor -0.6230  0.00807  0.7030  0.26100  0.1060   \n",
       "C3L-01312      Wildtype_Tumor -0.5660 -0.32600 -0.0968 -0.15000 -0.1850   \n",
       "C3L-01744      Wildtype_Tumor -0.5060 -0.21700 -1.1800 -1.00000  0.1390   \n",
       "C3L-01925      Wildtype_Tumor -0.2720 -1.37000 -0.6660 -0.54400  0.3860   \n",
       "C3N-00151      Wildtype_Tumor -0.5560 -0.53600 -1.1600  0.02390  0.0446   \n",
       "C3N-00322      Wildtype_Tumor -0.5250 -0.79100  0.2120  0.31300 -0.3730   \n",
       "C3N-00335      Wildtype_Tumor  0.1260  0.00181 -0.0442  1.68000  0.0711   \n",
       "C3N-00339      Wildtype_Tumor -0.5600 -0.24600     NaN      NaN  0.1680   \n",
       "C3N-00377      Wildtype_Tumor  0.8660  1.55000  1.1700  0.77600  0.0823   \n",
       "C3N-00386      Wildtype_Tumor -0.9790 -1.11000  0.7740 -1.16000 -0.0186   \n",
       "C3N-00389      Wildtype_Tumor -0.8070 -0.37900 -1.0600 -0.63300 -0.1960   \n",
       "C3N-00734      Wildtype_Tumor  0.2650 -0.19800  0.2920 -0.02110 -0.2400   \n",
       "C3N-00836      Wildtype_Tumor  0.8550  0.87600     NaN      NaN  0.1270   \n",
       "C3N-00847      Wildtype_Tumor -1.3000 -0.96500  0.7570 -0.05380  0.1430   \n",
       "C3N-00850      Wildtype_Tumor -0.6860 -0.55500  0.3330 -0.30600 -0.1080   \n",
       "C3N-00866      Wildtype_Tumor -1.2400 -0.08430  0.0142  1.72000 -0.0720   \n",
       "C3N-01211      Wildtype_Tumor -0.6360 -0.37100  0.9730      NaN  0.6370   \n",
       "C3N-01217      Wildtype_Tumor -0.7430 -1.37000     NaN      NaN -0.1930   \n",
       "C3N-01219      Wildtype_Tumor -0.2950 -0.05890  0.4900 -0.10900 -0.0495   \n",
       "C3N-01537      Wildtype_Tumor -0.2900 -0.32000  3.1700 -0.90700  0.0317   \n",
       "C3N-01802      Wildtype_Tumor  0.2660  1.39000 -0.0655  0.47000  0.3980   \n",
       "\n",
       "Name           AACS    AADAT    AAED1    AAGAB  ...    ZSWIM8   ZSWIM9  \\\n",
       "Patient_ID                                      ...                      \n",
       "C3L-00006   0.66500  1.28000 -0.33900  0.41200  ... -0.087700      NaN   \n",
       "C3L-00032   1.04000 -0.02130 -0.04790  0.41900  ...  0.001120 -0.14500   \n",
       "C3L-00362   0.43600  0.22700 -0.00248  0.47900  ... -0.049600      NaN   \n",
       "C3L-00601   0.21900 -0.24900 -1.33000  0.20400  ... -0.158000      NaN   \n",
       "C3L-00605   0.28300 -0.07890 -0.61100  0.37000  ... -0.051100 -0.01360   \n",
       "C3L-00918   0.33500  0.06850  0.12300  0.27300  ...  0.209000      NaN   \n",
       "C3L-00921  -0.00210  0.08130 -0.69100 -0.29500  ...  0.097000  0.25400   \n",
       "C3L-00947  -0.42300  0.22400 -0.24700  0.07870  ...  0.042900 -0.21200   \n",
       "C3N-00323   0.37100  0.64400  0.28100  0.11400  ... -0.022200      NaN   \n",
       "C3N-00324   0.31600  0.08090 -0.47000  0.80400  ...  0.044900  0.04520   \n",
       "C3N-00383   0.64600  0.37800 -0.01460  1.17000  ...  0.279000 -0.37800   \n",
       "C3N-00743   0.26500  0.46700 -0.67200  0.11900  ... -0.204000 -0.10700   \n",
       "C3N-01212   0.92700  0.49200 -0.41600  0.40000  ...  0.040600      NaN   \n",
       "C3N-01510  -0.54800 -0.10500      NaN  0.40100  ... -0.204000      NaN   \n",
       "C3L-00008   0.33400  1.30000  0.13900  1.33000  ... -0.035600      NaN   \n",
       "C3L-00090   0.75700  0.74000 -0.92900  0.22900  ...  0.072500 -0.05520   \n",
       "C3L-00136  -0.11100  0.89500  1.26000  0.15700  ...  0.455000      NaN   \n",
       "C3L-00137   0.65600 -0.28000 -0.12800  0.21700  ...  0.015800 -0.00491   \n",
       "C3L-00145   0.44500 -0.05650 -0.83800  0.04900  ...  0.084500 -0.64700   \n",
       "C3L-00361   2.56000  1.03000 -0.39600 -0.07630  ...  0.033000 -0.41900   \n",
       "C3L-00413   0.39800      NaN -0.27400  1.09000  ...  0.348000 -0.48400   \n",
       "C3L-00449  -0.06830 -0.37600 -0.08380  0.16000  ...  0.228000      NaN   \n",
       "C3L-00767   0.16800  0.42700 -0.75000  0.17600  ...  0.000388  0.18400   \n",
       "C3L-00780   0.21100  1.06000 -0.28000  0.20700  ...  0.300000      NaN   \n",
       "C3L-00932   0.08550 -0.04640  0.00430 -0.33700  ... -0.123000  0.14200   \n",
       "C3L-00946   0.06760  0.57200 -0.02420  0.64300  ... -0.035100 -0.55900   \n",
       "C3L-00949   0.83600  0.51000 -0.24800 -0.08340  ...  0.160000 -0.33200   \n",
       "C3L-01248   0.15400 -0.00265  0.20500 -0.13700  ...  0.294000 -0.51800   \n",
       "C3L-01249  -0.28600  0.50700 -0.65900  0.11900  ... -0.145000 -1.17000   \n",
       "C3L-01252   0.08740 -0.06390 -0.04940  0.00000  ... -0.158000 -0.09790   \n",
       "C3L-01256   0.32200  0.07180 -0.08760 -0.17300  ...  0.216000  0.11200   \n",
       "C3L-01257   0.89000  0.32500 -0.07900 -0.14600  ... -0.067000      NaN   \n",
       "C3L-01275   0.05550  0.16300 -0.90700 -0.10400  ... -0.004450 -0.39900   \n",
       "C3L-01282  -0.27800 -0.23600 -0.31200 -0.13100  ...  0.264000 -0.45900   \n",
       "C3L-01307   1.33000  1.15000  0.23400 -0.14200  ...  0.126000 -0.54600   \n",
       "C3L-01312  -0.00438  0.39300 -0.29100  0.12300  ...  0.134000      NaN   \n",
       "C3L-01744   0.65400  0.78900 -0.54700  0.49700  ...  0.054200 -0.67800   \n",
       "C3L-01925  -0.48000 -1.49000 -0.21600  0.05410  ...  0.117000  0.00480   \n",
       "C3N-00151   0.91900  0.15600 -0.13800  0.81200  ...  0.105000      NaN   \n",
       "C3N-00322   1.02000 -0.44000  0.10100  0.25500  ... -0.051400 -0.11500   \n",
       "C3N-00335  -0.32100 -0.34600  0.46700  0.75700  ... -0.199000 -0.37100   \n",
       "C3N-00339   1.17000  1.41000  0.24000  0.53800  ...  0.233000      NaN   \n",
       "C3N-00377   0.50700  0.45300      NaN  0.00740  ...  0.134000      NaN   \n",
       "C3N-00386   0.44700  0.84900 -0.23900  0.25400  ...  0.519000 -0.65000   \n",
       "C3N-00389   0.06540  0.39100 -0.13000  0.15200  ...  0.048500 -0.34200   \n",
       "C3N-00734   0.82700 -0.10400  0.16700  0.37900  ...  0.108000 -0.16700   \n",
       "C3N-00836  -0.06820  0.16400 -0.02610  0.00301  ... -0.215000      NaN   \n",
       "C3N-00847   1.15000  0.72600 -0.34400  0.49300  ...  0.015000      NaN   \n",
       "C3N-00850   0.56600  1.20000 -0.15700  0.11300  ...  0.117000 -0.47700   \n",
       "C3N-00866   0.53000 -0.19500 -0.29300  0.18200  ...  0.077300  0.03270   \n",
       "C3N-01211  -0.16600 -0.12200  0.23200  0.37900  ... -0.416000 -0.75100   \n",
       "C3N-01217   0.69400  0.33800  0.09890 -0.01370  ...  0.181000      NaN   \n",
       "C3N-01219   0.03740  0.10200 -0.32100  0.18300  ... -0.290000  0.04570   \n",
       "C3N-01537  -0.04250      NaN -0.27300 -0.07470  ... -0.179000 -0.52100   \n",
       "C3N-01802  -0.13400  0.46100  1.04000  0.36300  ... -0.140000      NaN   \n",
       "\n",
       "Name           ZW10  ZWILCH   ZWINT    ZXDC   ZYG11B      ZYX    ZZEF1  \\\n",
       "Patient_ID                                                               \n",
       "C3L-00006   0.02290  0.1090     NaN -0.3320 -0.43300 -1.02000 -0.12300   \n",
       "C3L-00032   0.01050 -0.1160     NaN  0.1510 -0.07400 -0.54000  0.32000   \n",
       "C3L-00362   0.10800 -0.1530     NaN  0.1090 -0.33600 -0.82200 -0.03380   \n",
       "C3L-00601   0.66700  1.3000  0.4410  0.1300 -0.06590 -0.92300 -0.16300   \n",
       "C3L-00605   0.21400  0.6830  0.4330 -0.2080 -0.34700 -0.91100  0.06920   \n",
       "C3L-00918   0.12400  0.5880     NaN -0.5680 -0.22200 -0.14800  0.06220   \n",
       "C3L-00921   0.15100  0.2850 -0.2390 -0.0681 -0.21800 -0.08730 -0.12700   \n",
       "C3L-00947   0.16900  0.5210 -0.1470  0.3540  0.17000 -1.01000 -0.11000   \n",
       "C3N-00323  -0.27900  0.4230     NaN  0.2940 -0.13000 -0.10600 -0.29900   \n",
       "C3N-00324  -0.15200 -0.1970  0.2950 -0.1880 -0.26500  0.25000 -0.11300   \n",
       "C3N-00383   0.01250 -0.3050     NaN  0.1490 -0.05260 -0.03110  0.43200   \n",
       "C3N-00743   0.23800  0.6160  0.4400 -0.6070  0.01930  0.30500 -0.43400   \n",
       "C3N-01212   0.09240  0.1210  0.4330 -1.0500 -0.31600 -0.40000  0.12300   \n",
       "C3N-01510  -0.58300  0.9130  0.2490 -0.5150 -0.15100 -1.39000 -0.12100   \n",
       "C3L-00008   0.36300  1.0700  0.7370 -0.5640 -0.00461 -1.13000 -0.07570   \n",
       "C3L-00090  -0.07140  0.0933  0.1560 -0.3980 -0.07520 -0.79700 -0.03010   \n",
       "C3L-00136   0.39700 -0.9990 -0.7300 -0.0229 -0.33100 -1.16000 -0.11600   \n",
       "C3L-00137   0.18000  0.5130  0.5490 -0.6810 -0.28500 -0.56400 -0.08760   \n",
       "C3L-00145   0.17500  0.2120 -0.3240 -0.3500 -0.37700  0.38800  0.01100   \n",
       "C3L-00361   0.32700  0.1030     NaN  0.6380 -0.51400 -1.42000  0.21800   \n",
       "C3L-00413   0.14200  0.2500 -0.1420  0.0770 -0.15700 -0.87200 -0.00663   \n",
       "C3L-00449   0.02060 -0.9700     NaN -0.4520  0.17200  0.62600  0.34200   \n",
       "C3L-00767   0.08680  0.1440 -0.1370 -0.4180 -0.18300 -0.76900 -0.25600   \n",
       "C3L-00780   0.14200 -0.5490     NaN -0.3900 -0.09720  0.00108  0.20000   \n",
       "C3L-00932  -0.17400 -0.5110 -0.1320  0.1530 -0.04300  0.76400 -0.21600   \n",
       "C3L-00946  -0.20600  0.1780 -0.0634 -0.3140 -0.07460  0.08190  0.20500   \n",
       "C3L-00949   0.02140 -0.1550     NaN  0.0893  0.01620  0.37700  0.03220   \n",
       "C3L-01248  -0.13200 -0.7550 -0.0709 -0.3550  0.14000  1.72000  0.17800   \n",
       "C3L-01249  -0.59300 -0.7900     NaN  0.4170  0.94400 -0.12000  0.08110   \n",
       "C3L-01252  -0.30400 -0.5420 -0.3020 -0.3070 -0.05490 -0.02140 -0.03970   \n",
       "C3L-01256  -0.15100 -0.0164 -0.2430 -0.3080 -0.06810  1.18000 -0.05350   \n",
       "C3L-01257   0.13700  0.9570     NaN -0.2670 -0.26700 -0.45200  0.01940   \n",
       "C3L-01275   0.09010  0.2780  0.1020 -0.2750 -0.28300  0.41800 -0.02290   \n",
       "C3L-01282   0.01500 -0.2710 -0.0568 -0.3570  0.19200 -0.09430  0.01090   \n",
       "C3L-01307   0.48800  0.4850 -0.0923  0.1470  0.08140 -0.34800  0.39300   \n",
       "C3L-01312   0.14300 -0.7900     NaN -1.3300  0.12900 -0.40400  0.17700   \n",
       "C3L-01744   0.34500  0.8860  1.5100 -0.6210 -0.24500 -1.02000  0.26800   \n",
       "C3L-01925   0.32200  0.1850  1.5500  0.2040  0.01180  0.34800  0.01220   \n",
       "C3N-00151   0.09630  0.2430  0.6430 -0.5800  0.04090 -0.71000  0.42400   \n",
       "C3N-00322  -0.19400  0.3600  0.2040  0.2980  0.01610  0.24100 -0.60400   \n",
       "C3N-00335  -0.27400  0.8760  1.4000  0.7160  0.18300  0.45800 -0.41500   \n",
       "C3N-00339   0.05930  0.6920  0.1860  0.3390  0.22100 -0.01610 -0.06920   \n",
       "C3N-00377  -0.29900  0.7400  1.4600 -0.5700  0.03700 -0.50100 -0.04150   \n",
       "C3N-00386   0.29800  0.4890  0.0383  0.2830 -0.45100 -0.65900  0.05340   \n",
       "C3N-00389  -0.06940  0.0692  0.2820 -0.7190 -0.23400 -0.16100  0.10700   \n",
       "C3N-00734   0.00762 -0.0456 -0.4750 -0.0226 -0.14800 -0.51400  0.05670   \n",
       "C3N-00836  -0.05010  0.1800  0.2330 -0.5550  0.19900  0.59000  0.23700   \n",
       "C3N-00847   0.31800  0.6660  0.2170 -0.4360 -0.10100 -0.77400  0.20900   \n",
       "C3N-00850   0.32500  0.3400 -0.2590  0.2090 -0.17700 -0.26200  0.04620   \n",
       "C3N-00866   0.30800 -0.1500 -0.0672  0.0697 -0.22400 -0.66900  0.03750   \n",
       "C3N-01211   0.17900  0.6580  0.4700  0.2700 -0.14800 -0.13900 -0.02990   \n",
       "C3N-01217   0.19600  0.2740     NaN -0.5010 -0.04680  0.65300 -0.04640   \n",
       "C3N-01219   0.24600  0.5890  0.3930 -0.1990 -0.15900 -1.09000  0.06270   \n",
       "C3N-01537   0.05230  0.5150  0.5850  0.1380 -0.19000 -0.96600 -0.00627   \n",
       "C3N-01802  -0.01220  0.2500  0.5530  0.3870  0.06420 -0.43700  0.10400   \n",
       "\n",
       "Name           ZZZ3  \n",
       "Patient_ID           \n",
       "C3L-00006  -0.08590  \n",
       "C3L-00032  -0.41900  \n",
       "C3L-00362   0.12100  \n",
       "C3L-00601   0.11200  \n",
       "C3L-00605  -0.23200  \n",
       "C3L-00918  -0.23100  \n",
       "C3L-00921   0.37400  \n",
       "C3L-00947   0.13100  \n",
       "C3N-00323   0.38500  \n",
       "C3N-00324  -0.30200  \n",
       "C3N-00383   0.00801  \n",
       "C3N-00743  -0.22900  \n",
       "C3N-01212  -0.39100  \n",
       "C3N-01510  -0.42600  \n",
       "C3L-00008  -0.47300  \n",
       "C3L-00090  -0.46700  \n",
       "C3L-00136   0.00250  \n",
       "C3L-00137   0.00698  \n",
       "C3L-00145   0.13400  \n",
       "C3L-00361  -0.06110  \n",
       "C3L-00413  -0.01750  \n",
       "C3L-00449  -0.20300  \n",
       "C3L-00767  -0.37200  \n",
       "C3L-00780  -0.19400  \n",
       "C3L-00932   0.07370  \n",
       "C3L-00946  -0.30500  \n",
       "C3L-00949  -0.13000  \n",
       "C3L-01248  -0.08030  \n",
       "C3L-01249   0.48500  \n",
       "C3L-01252  -0.17100  \n",
       "C3L-01256  -0.02080  \n",
       "C3L-01257  -0.05460  \n",
       "C3L-01275   0.12700  \n",
       "C3L-01282  -0.04450  \n",
       "C3L-01307  -0.19000  \n",
       "C3L-01312  -0.12000  \n",
       "C3L-01744  -0.02040  \n",
       "C3L-01925   0.10000  \n",
       "C3N-00151  -0.02360  \n",
       "C3N-00322   0.14500  \n",
       "C3N-00335   0.27200  \n",
       "C3N-00339   0.26600  \n",
       "C3N-00377  -0.18900  \n",
       "C3N-00386   0.30100  \n",
       "C3N-00389   0.12700  \n",
       "C3N-00734  -0.00663  \n",
       "C3N-00836  -0.29800  \n",
       "C3N-00847   0.25300  \n",
       "C3N-00850  -0.02090  \n",
       "C3N-00866  -0.25500  \n",
       "C3N-01211   0.13100  \n",
       "C3N-01217  -0.16000  \n",
       "C3N-01219  -0.37900  \n",
       "C3N-01537  -0.24900  \n",
       "C3N-01802  -0.49800  \n",
       "\n",
       "[55 rows x 11000 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_prot_and_mutations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Do a t-test for every gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = list(endo_prot_and_mutations.columns[1:])\n",
    "n = wrap_ttest(endo_prot_and_mutations, 'Mutation', correction_method=\"fdr_bh\")\n",
    "n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.995678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PPIA</td>\n",
       "      <td>0.995678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PPIAL4G</td>\n",
       "      <td>0.995678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PPIB</td>\n",
       "      <td>0.995678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PPIC</td>\n",
       "      <td>0.995678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10380</th>\n",
       "      <td>DIAPH3</td>\n",
       "      <td>0.999795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10381</th>\n",
       "      <td>POLD4</td>\n",
       "      <td>0.999852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10382</th>\n",
       "      <td>MASTL</td>\n",
       "      <td>0.999852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10383</th>\n",
       "      <td>BAZ1B</td>\n",
       "      <td>0.999876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10384</th>\n",
       "      <td>POGZ</td>\n",
       "      <td>0.999953</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10385 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Comparison   P_Value\n",
       "0           A1BG  0.995678\n",
       "1           PPIA  0.995678\n",
       "2        PPIAL4G  0.995678\n",
       "3           PPIB  0.995678\n",
       "4           PPIC  0.995678\n",
       "...          ...       ...\n",
       "10380     DIAPH3  0.999795\n",
       "10381      POLD4  0.999852\n",
       "10382      MASTL  0.999852\n",
       "10383      BAZ1B  0.999876\n",
       "10384       POGZ  0.999953\n",
       "\n",
       "[10385 rows x 2 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = list(endo_prot_and_mutations.columns[1:])\n",
    "n = wrap_ttest(endo_prot_and_mutations, 'Mutation', correction_method=\"fdr_bh\", return_all=True, pval_return_corrected=True)\n",
    "n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join difference in median and t-test p value into the same dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Step 4 join median and p value together\n",
    "# endo_df = median_diff.join(t_test)\n",
    "# endo_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BRCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "no_duplicates = proteomics[proteomics.columns[~fil]]\n",
    "# duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 83 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    }
   ],
   "source": [
    "#DEALING WITH DUPLICATES\n",
    "#flatten the multiindex\n",
    "duplicates = cptac.utils.reduce_multiindex(duplicates, flatten=True)\n",
    "\n",
    "# We need to figure which of databaseID we want to use, for each protein\n",
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "duplicate_gene_names = list(set(duplicates.columns.get_level_values('Name'))) #get a list of genes names that are duplicated\n",
    "duplicates = cptac.utils.reduce_multiindex(duplicates, flatten=True)\n",
    "duplicates\n",
    "\n",
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "\n",
    "mut_type = br.get_genotype_all_vars(\"PIK3CA\")\n",
    "joined = mut_type.join(duplicates)\n",
    "\n",
    "#select samples containing hotspot mutations\n",
    "br_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "                    joined.Location.str.contains('E545K') |\n",
    "                    joined.Location.str.contains('H1047R')]\n",
    "\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "brca_hotspot = br_hotspot_df.append(wildtype)\n",
    "\n",
    "#drop unncessary columns\n",
    "filter_prot_and_mutations = brca_hotspot.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "\n",
    "#step 3 do a t test for every gene and report the p-value\n",
    "cols = list(filter_prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(filter_prot_and_mutations, 'Mutation', cols, correction_method= \"fdr_bh\", return_all=True)\n",
    "\n",
    "# t_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for each duplicated gene name select the one with the min p val and append[ to no_duplicat df\n",
    "\n",
    "selected_dup = []\n",
    "for name in duplicate_gene_names:\n",
    "    gene_duplicate = t_test[t_test.Comparison.str.contains(name)]\n",
    "    #select the one with the min p value for t test\n",
    "#     import pdb; pdb.set_trace()\n",
    "    min_row = gene_duplicate.loc[gene_duplicate.P_Value == gene_duplicate.P_Value.min()]\n",
    "    name = min_row['Comparison'].values\n",
    "    selected_dup.append(name)\n",
    "\n",
    "selected= []\n",
    "no_dup = cptac.utils.reduce_multiindex(no_duplicates, flatten=True)\n",
    "no_dup.columns\n",
    "for name in no_dup.columns:\n",
    "    selected.append(name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 83 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSCAN31</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CPT001846</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.3964</td>\n",
       "      <td>1.3302</td>\n",
       "      <td>-5.0948</td>\n",
       "      <td>0.7674</td>\n",
       "      <td>-1.6845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1022</td>\n",
       "      <td>-0.5814</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.7592</td>\n",
       "      <td>0.4711</td>\n",
       "      <td>0.6018</td>\n",
       "      <td>0.2062</td>\n",
       "      <td>-0.2137</td>\n",
       "      <td>-2.1219</td>\n",
       "      <td>0.0860</td>\n",
       "      <td>2.5814</td>\n",
       "      <td>-0.2852</td>\n",
       "      <td>-0.1074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X11BR004</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.3627</td>\n",
       "      <td>1.8899</td>\n",
       "      <td>-0.5023</td>\n",
       "      <td>-0.6201</td>\n",
       "      <td>0.8102</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1417</td>\n",
       "      <td>-0.1994</td>\n",
       "      <td>0.3278</td>\n",
       "      <td>...</td>\n",
       "      <td>1.8955</td>\n",
       "      <td>0.3530</td>\n",
       "      <td>0.8214</td>\n",
       "      <td>1.2056</td>\n",
       "      <td>-0.6622</td>\n",
       "      <td>-0.5864</td>\n",
       "      <td>-0.0059</td>\n",
       "      <td>1.3178</td>\n",
       "      <td>0.4372</td>\n",
       "      <td>-1.0408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X11BR013</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.7803</td>\n",
       "      <td>-0.6980</td>\n",
       "      <td>-3.5729</td>\n",
       "      <td>0.7011</td>\n",
       "      <td>-1.8474</td>\n",
       "      <td>-2.7737</td>\n",
       "      <td>-0.0788</td>\n",
       "      <td>0.2499</td>\n",
       "      <td>-1.4275</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0010</td>\n",
       "      <td>-0.6068</td>\n",
       "      <td>0.8283</td>\n",
       "      <td>-1.0147</td>\n",
       "      <td>1.4330</td>\n",
       "      <td>0.2379</td>\n",
       "      <td>-1.0891</td>\n",
       "      <td>-0.2156</td>\n",
       "      <td>-0.3116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X11BR014</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.9126</td>\n",
       "      <td>-0.2126</td>\n",
       "      <td>-8.9169</td>\n",
       "      <td>-0.6406</td>\n",
       "      <td>-1.9189</td>\n",
       "      <td>-2.4744</td>\n",
       "      <td>-2.0662</td>\n",
       "      <td>-0.2834</td>\n",
       "      <td>0.4592</td>\n",
       "      <td>...</td>\n",
       "      <td>1.2953</td>\n",
       "      <td>-0.1190</td>\n",
       "      <td>0.1474</td>\n",
       "      <td>-1.5504</td>\n",
       "      <td>-3.3162</td>\n",
       "      <td>-0.5357</td>\n",
       "      <td>1.4172</td>\n",
       "      <td>1.8508</td>\n",
       "      <td>0.2182</td>\n",
       "      <td>-0.5697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X11BR022</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.9095</td>\n",
       "      <td>1.4993</td>\n",
       "      <td>-4.9660</td>\n",
       "      <td>-0.5247</td>\n",
       "      <td>-0.2128</td>\n",
       "      <td>-3.1250</td>\n",
       "      <td>-0.2501</td>\n",
       "      <td>1.9603</td>\n",
       "      <td>-0.1348</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3823</td>\n",
       "      <td>-0.0501</td>\n",
       "      <td>0.1364</td>\n",
       "      <td>-0.6264</td>\n",
       "      <td>-0.8129</td>\n",
       "      <td>-1.1180</td>\n",
       "      <td>-0.5281</td>\n",
       "      <td>-0.6841</td>\n",
       "      <td>0.0143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X15BR003</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>2.4710</td>\n",
       "      <td>1.0447</td>\n",
       "      <td>1.6000</td>\n",
       "      <td>-0.2389</td>\n",
       "      <td>-0.2619</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.2958</td>\n",
       "      <td>0.1298</td>\n",
       "      <td>0.3695</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3656</td>\n",
       "      <td>-1.0569</td>\n",
       "      <td>0.8718</td>\n",
       "      <td>-0.4324</td>\n",
       "      <td>-0.7942</td>\n",
       "      <td>-2.4072</td>\n",
       "      <td>1.2313</td>\n",
       "      <td>0.6114</td>\n",
       "      <td>-0.6675</td>\n",
       "      <td>-0.8818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X18BR004</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.4652</td>\n",
       "      <td>-0.2157</td>\n",
       "      <td>-3.6000</td>\n",
       "      <td>-0.3496</td>\n",
       "      <td>-0.9725</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2205</td>\n",
       "      <td>-0.7924</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.0780</td>\n",
       "      <td>0.9906</td>\n",
       "      <td>0.3147</td>\n",
       "      <td>0.0305</td>\n",
       "      <td>-0.2289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2189</td>\n",
       "      <td>-0.3281</td>\n",
       "      <td>0.4254</td>\n",
       "      <td>-0.1496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X18BR006</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.0999</td>\n",
       "      <td>0.7852</td>\n",
       "      <td>-8.1266</td>\n",
       "      <td>-0.5701</td>\n",
       "      <td>-2.1021</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5732</td>\n",
       "      <td>0.4928</td>\n",
       "      <td>-0.2838</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3722</td>\n",
       "      <td>0.0816</td>\n",
       "      <td>0.0086</td>\n",
       "      <td>0.3131</td>\n",
       "      <td>-0.2260</td>\n",
       "      <td>-0.6128</td>\n",
       "      <td>0.7913</td>\n",
       "      <td>0.8553</td>\n",
       "      <td>-0.0371</td>\n",
       "      <td>-0.3874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X18BR017</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.8934</td>\n",
       "      <td>1.6886</td>\n",
       "      <td>-6.0217</td>\n",
       "      <td>-0.0481</td>\n",
       "      <td>0.1299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.5536</td>\n",
       "      <td>0.5748</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2404</td>\n",
       "      <td>0.6150</td>\n",
       "      <td>0.2906</td>\n",
       "      <td>0.2303</td>\n",
       "      <td>-0.3553</td>\n",
       "      <td>0.4313</td>\n",
       "      <td>1.0197</td>\n",
       "      <td>1.0025</td>\n",
       "      <td>1.8005</td>\n",
       "      <td>-0.5533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X20BR001</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>1.0545</td>\n",
       "      <td>1.0394</td>\n",
       "      <td>-3.5713</td>\n",
       "      <td>-0.7841</td>\n",
       "      <td>-1.1205</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2457</td>\n",
       "      <td>-0.5970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4385</td>\n",
       "      <td>0.1229</td>\n",
       "      <td>0.1229</td>\n",
       "      <td>-1.2017</td>\n",
       "      <td>-1.4247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7539</td>\n",
       "      <td>-0.2229</td>\n",
       "      <td>0.1569</td>\n",
       "      <td>0.2495</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71 rows × 9451 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation    A1BG     A2M   A2ML1    AAAS    AACS   AADAT  \\\n",
       "Patient_ID                                                                      \n",
       "CPT001846   Missense_Mutation  1.3964  1.3302 -5.0948  0.7674 -1.6845     NaN   \n",
       "X11BR004    Missense_Mutation  1.3627  1.8899 -0.5023 -0.6201  0.8102     NaN   \n",
       "X11BR013    Missense_Mutation  0.7803 -0.6980 -3.5729  0.7011 -1.8474 -2.7737   \n",
       "X11BR014    Missense_Mutation  0.9126 -0.2126 -8.9169 -0.6406 -1.9189 -2.4744   \n",
       "X11BR022    Missense_Mutation  1.9095  1.4993 -4.9660 -0.5247 -0.2128 -3.1250   \n",
       "...                       ...     ...     ...     ...     ...     ...     ...   \n",
       "X15BR003       Wildtype_Tumor  2.4710  1.0447  1.6000 -0.2389 -0.2619     NaN   \n",
       "X18BR004       Wildtype_Tumor -0.4652 -0.2157 -3.6000 -0.3496 -0.9725     NaN   \n",
       "X18BR006       Wildtype_Tumor  0.0999  0.7852 -8.1266 -0.5701 -2.1021     NaN   \n",
       "X18BR017       Wildtype_Tumor  0.8934  1.6886 -6.0217 -0.0481  0.1299     NaN   \n",
       "X20BR001       Wildtype_Tumor  1.0545  1.0394 -3.5713 -0.7841 -1.1205     NaN   \n",
       "\n",
       "Name         AAED1   AAGAB    AAK1  ...  ZSCAN31  ZSWIM8    ZW10  ZWILCH  \\\n",
       "Patient_ID                          ...                                    \n",
       "CPT001846   2.1022 -0.5814  0.2916  ...  -0.7592  0.4711  0.6018  0.2062   \n",
       "X11BR004   -1.1417 -0.1994  0.3278  ...   1.8955  0.3530  0.8214  1.2056   \n",
       "X11BR013   -0.0788  0.2499 -1.4275  ...      NaN  1.0010 -0.6068  0.8283   \n",
       "X11BR014   -2.0662 -0.2834  0.4592  ...   1.2953 -0.1190  0.1474 -1.5504   \n",
       "X11BR022   -0.2501  1.9603 -0.1348  ...      NaN -0.3823 -0.0501  0.1364   \n",
       "...            ...     ...     ...  ...      ...     ...     ...     ...   \n",
       "X15BR003    1.2958  0.1298  0.3695  ...  -0.3656 -1.0569  0.8718 -0.4324   \n",
       "X18BR004       NaN  0.2205 -0.7924  ...  -2.0780  0.9906  0.3147  0.0305   \n",
       "X18BR006   -0.5732  0.4928 -0.2838  ...  -0.3722  0.0816  0.0086  0.3131   \n",
       "X18BR017       NaN  1.5536  0.5748  ...  -0.2404  0.6150  0.2906  0.2303   \n",
       "X20BR001       NaN  0.2457 -0.5970  ...   0.4385  0.1229  0.1229 -1.2017   \n",
       "\n",
       "Name         ZWINT    ZXDC  ZYG11B     ZYX   ZZEF1    ZZZ3  \n",
       "Patient_ID                                                  \n",
       "CPT001846  -0.2137 -2.1219  0.0860  2.5814 -0.2852 -0.1074  \n",
       "X11BR004   -0.6622 -0.5864 -0.0059  1.3178  0.4372 -1.0408  \n",
       "X11BR013   -1.0147  1.4330  0.2379 -1.0891 -0.2156 -0.3116  \n",
       "X11BR014   -3.3162 -0.5357  1.4172  1.8508  0.2182 -0.5697  \n",
       "X11BR022   -0.6264 -0.8129 -1.1180 -0.5281 -0.6841  0.0143  \n",
       "...            ...     ...     ...     ...     ...     ...  \n",
       "X15BR003   -0.7942 -2.4072  1.2313  0.6114 -0.6675 -0.8818  \n",
       "X18BR004   -0.2289     NaN  0.2189 -0.3281  0.4254 -0.1496  \n",
       "X18BR006   -0.2260 -0.6128  0.7913  0.8553 -0.0371 -0.3874  \n",
       "X18BR017   -0.3553  0.4313  1.0197  1.0025  1.8005 -0.5533  \n",
       "X20BR001   -1.4247     NaN -0.7539 -0.2229  0.1569  0.2495  \n",
       "\n",
       "[71 rows x 9451 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#use the list of selected genes to filter dataframe\n",
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "proteomics = cptac.utils.reduce_multiindex(proteomics, flatten=True)\n",
    "selected_prot = proteomics[proteomics.columns.intersection(selected)]\n",
    "\n",
    "#get mutation_tyoe\n",
    "mut_type = br.get_genotype_all_vars('PIK3CA')\n",
    "\n",
    "#join proteomics and mutation type\n",
    "joined = mut_type.join(selected_prot)\n",
    "\n",
    "#select samples containing hotspot mutations\n",
    "br_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "                    joined.Location.str.contains('E545K') |\n",
    "                    joined.Location.str.contains('H1047R')]\n",
    "\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "brca_hotspot = br_hotspot_df.append(wildtype)\n",
    "\n",
    "#drop unncessary columns\n",
    "brca_prot_and_mutations = brca_hotspot.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "#drop databaseID identifier\n",
    "split = brca_prot_and_mutations.columns.str.split('_', 1).str[0]\n",
    "brca_prot_and_mutations.columns=split\n",
    "\n",
    "#remove duplicate columns\n",
    "brca_prot_and_mutations = brca_prot_and_mutations.loc[:,~brca_prot_and_mutations.columns.duplicated()]\n",
    "brca_prot_and_mutations \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(brca_prot_and_mutations.loc[brca_prot_and_mutations[\"Mutation\"]==\"Missense_Mutation\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HSPE1-MOB4</td>\n",
       "      <td>0.011107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UBE4A</td>\n",
       "      <td>0.086181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IDH3A</td>\n",
       "      <td>0.091678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HSPE1</td>\n",
       "      <td>0.091678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACAP2</td>\n",
       "      <td>0.091678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>OXSM</td>\n",
       "      <td>0.112433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PUS1</td>\n",
       "      <td>0.115105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HSPD1</td>\n",
       "      <td>0.169127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CD320</td>\n",
       "      <td>0.169127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MRPS6</td>\n",
       "      <td>0.169127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>0.169127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>LONP1</td>\n",
       "      <td>0.216125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CHDH</td>\n",
       "      <td>0.224365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TUFM</td>\n",
       "      <td>0.227058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>0.227058</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Comparison   P_Value\n",
       "0   HSPE1-MOB4  0.011107\n",
       "1        UBE4A  0.086181\n",
       "2        IDH3A  0.091678\n",
       "3        HSPE1  0.091678\n",
       "4        ACAP2  0.091678\n",
       "5         OXSM  0.112433\n",
       "6         PUS1  0.115105\n",
       "7        HSPD1  0.169127\n",
       "8        CD320  0.169127\n",
       "9        MRPS6  0.169127\n",
       "10     DNAJC13  0.169127\n",
       "11       LONP1  0.216125\n",
       "12        CHDH  0.224365\n",
       "13        TUFM  0.227058\n",
       "14        ABL1  0.227058"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n = wrap_ttest(brca_prot_and_mutations, 'Mutation', correction_method= \"fdr_bh\", pval_return_corrected=True, return_all=True)\n",
    "n.head(15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HSPE1-MOB4</td>\n",
       "      <td>0.011107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UBE4A</td>\n",
       "      <td>0.086181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IDH3A</td>\n",
       "      <td>0.091678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HSPE1</td>\n",
       "      <td>0.091678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACAP2</td>\n",
       "      <td>0.091678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9431</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.999473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9432</th>\n",
       "      <td>MAX</td>\n",
       "      <td>0.999599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9433</th>\n",
       "      <td>RASIP1</td>\n",
       "      <td>0.999642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9434</th>\n",
       "      <td>ARHGEF17</td>\n",
       "      <td>0.999816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9435</th>\n",
       "      <td>SPARCL1</td>\n",
       "      <td>0.999816</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9436 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Comparison   P_Value\n",
       "0     HSPE1-MOB4  0.011107\n",
       "1          UBE4A  0.086181\n",
       "2          IDH3A  0.091678\n",
       "3          HSPE1  0.091678\n",
       "4          ACAP2  0.091678\n",
       "...          ...       ...\n",
       "9431         A2M  0.999473\n",
       "9432         MAX  0.999599\n",
       "9433      RASIP1  0.999642\n",
       "9434    ARHGEF17  0.999816\n",
       "9435     SPARCL1  0.999816\n",
       "\n",
       "[9436 rows x 2 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n = wrap_ttest(brca_prot_and_mutations, 'Mutation', correction_method= \"fdr_bh\", pval_return_corrected=True, return_all=True)\n",
    "n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "53"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# b_mut = brca_prot_and_mutations.loc[brca_prot_and_mutations[\"Mutation\"]==\"Missense_Mutation\"]\n",
    "b_mut=brca_prot_and_mutations\n",
    "full = b_mut.dropna(axis=1)\n",
    "full\n",
    "len(brca_prot_and_mutations.loc[brca_prot_and_mutations[\"Mutation\"]==\"Wildtype_Tumor\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HSPE1-MOB4</td>\n",
       "      <td>0.011107</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Comparison   P_Value\n",
       "0  HSPE1-MOB4  0.011107"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now run ttest\n",
    "n = wrap_ttest(brca_prot_and_mutations, 'Mutation', correction_method= \"fdr_bh\", pval_return_corrected=True)\n",
    "n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#let make a box plot. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>AAMDC</th>\n",
       "      <th>AAMP</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNHIT6</th>\n",
       "      <th>ZNRD1</th>\n",
       "      <th>ZNRF2</th>\n",
       "      <th>ZPR1</th>\n",
       "      <th>ZRANB2</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>05CO028</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.340</td>\n",
       "      <td>0.343</td>\n",
       "      <td>-1.6600</td>\n",
       "      <td>0.5040</td>\n",
       "      <td>0.1620</td>\n",
       "      <td>0.3190</td>\n",
       "      <td>-0.2980</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>-0.1050</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0595</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>0.1450</td>\n",
       "      <td>0.1970</td>\n",
       "      <td>-0.3990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7540</td>\n",
       "      <td>-0.4230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.220</td>\n",
       "      <td>-0.575</td>\n",
       "      <td>-0.5960</td>\n",
       "      <td>0.2990</td>\n",
       "      <td>-0.0254</td>\n",
       "      <td>0.7640</td>\n",
       "      <td>0.1040</td>\n",
       "      <td>-0.6940</td>\n",
       "      <td>0.3560</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2850</td>\n",
       "      <td>-0.1480</td>\n",
       "      <td>0.0751</td>\n",
       "      <td>0.0503</td>\n",
       "      <td>0.2300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5900</td>\n",
       "      <td>-0.0644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>09CO005</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.490</td>\n",
       "      <td>-0.200</td>\n",
       "      <td>-1.2800</td>\n",
       "      <td>-0.1220</td>\n",
       "      <td>0.2730</td>\n",
       "      <td>0.2230</td>\n",
       "      <td>0.3590</td>\n",
       "      <td>-1.1200</td>\n",
       "      <td>-0.0215</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2660</td>\n",
       "      <td>0.0078</td>\n",
       "      <td>0.5310</td>\n",
       "      <td>-0.3550</td>\n",
       "      <td>-0.6400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.4300</td>\n",
       "      <td>-0.3340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO027</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.090</td>\n",
       "      <td>0.236</td>\n",
       "      <td>-1.0400</td>\n",
       "      <td>0.3070</td>\n",
       "      <td>0.0530</td>\n",
       "      <td>0.4660</td>\n",
       "      <td>-0.0733</td>\n",
       "      <td>-0.0244</td>\n",
       "      <td>0.8510</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3930</td>\n",
       "      <td>0.7610</td>\n",
       "      <td>0.3730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0261</td>\n",
       "      <td>-0.1030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO052</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.280</td>\n",
       "      <td>-0.623</td>\n",
       "      <td>-0.9320</td>\n",
       "      <td>-0.0443</td>\n",
       "      <td>0.0090</td>\n",
       "      <td>0.0958</td>\n",
       "      <td>0.1710</td>\n",
       "      <td>-0.3130</td>\n",
       "      <td>0.0981</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1510</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2730</td>\n",
       "      <td>0.2830</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>0.1180</td>\n",
       "      <td>0.2930</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>-0.0604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20CO001</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.629</td>\n",
       "      <td>-0.068</td>\n",
       "      <td>0.0734</td>\n",
       "      <td>-0.6540</td>\n",
       "      <td>0.2130</td>\n",
       "      <td>0.1550</td>\n",
       "      <td>0.5650</td>\n",
       "      <td>-0.0406</td>\n",
       "      <td>-0.1540</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4150</td>\n",
       "      <td>-0.0765</td>\n",
       "      <td>-0.0385</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4260</td>\n",
       "      <td>0.5190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20CO003</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.123</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-0.4520</td>\n",
       "      <td>0.2660</td>\n",
       "      <td>-0.2280</td>\n",
       "      <td>0.0484</td>\n",
       "      <td>0.1710</td>\n",
       "      <td>0.7850</td>\n",
       "      <td>0.3000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0415</td>\n",
       "      <td>0.0830</td>\n",
       "      <td>-0.0374</td>\n",
       "      <td>-0.1320</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6810</td>\n",
       "      <td>-0.2970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO004</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.280</td>\n",
       "      <td>0.518</td>\n",
       "      <td>-0.8720</td>\n",
       "      <td>0.2580</td>\n",
       "      <td>0.4240</td>\n",
       "      <td>0.2980</td>\n",
       "      <td>-0.1500</td>\n",
       "      <td>-0.6580</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4490</td>\n",
       "      <td>-0.0616</td>\n",
       "      <td>0.3850</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>-0.0437</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.8840</td>\n",
       "      <td>0.0088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO006</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.515</td>\n",
       "      <td>-1.210</td>\n",
       "      <td>-0.0283</td>\n",
       "      <td>0.1780</td>\n",
       "      <td>-0.3030</td>\n",
       "      <td>0.1500</td>\n",
       "      <td>0.3990</td>\n",
       "      <td>-1.0000</td>\n",
       "      <td>0.2060</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5610</td>\n",
       "      <td>0.4740</td>\n",
       "      <td>0.0836</td>\n",
       "      <td>-0.2510</td>\n",
       "      <td>0.1760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0584</td>\n",
       "      <td>-0.3010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24CO005</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74 rows × 8068 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation   A1BG   A1CF     A2M    AAAS    AACS   AAGAB  \\\n",
       "Patient_ID                                                                    \n",
       "05CO028     Missense_Mutation -1.340  0.343 -1.6600  0.5040  0.1620  0.3190   \n",
       "05CO032     Missense_Mutation -1.220 -0.575 -0.5960  0.2990 -0.0254  0.7640   \n",
       "09CO005     Missense_Mutation -1.490 -0.200 -1.2800 -0.1220  0.2730  0.2230   \n",
       "11CO027     Missense_Mutation -1.090  0.236 -1.0400  0.3070  0.0530  0.4660   \n",
       "11CO052     Missense_Mutation -0.280 -0.623 -0.9320 -0.0443  0.0090  0.0958   \n",
       "...                       ...    ...    ...     ...     ...     ...     ...   \n",
       "20CO001        Wildtype_Tumor  0.629 -0.068  0.0734 -0.6540  0.2130  0.1550   \n",
       "20CO003        Wildtype_Tumor -0.123 -1.070 -0.4520  0.2660 -0.2280  0.0484   \n",
       "22CO004        Wildtype_Tumor -1.280  0.518 -0.8720  0.2580  0.4240  0.2980   \n",
       "22CO006        Wildtype_Tumor  0.515 -1.210 -0.0283  0.1780 -0.3030  0.1500   \n",
       "24CO005        Wildtype_Tumor    NaN    NaN     NaN     NaN     NaN     NaN   \n",
       "\n",
       "Name          AAK1   AAMDC    AAMP  ...  ZNHIT6  ZNRD1   ZNRF2    ZPR1  \\\n",
       "Patient_ID                          ...                                  \n",
       "05CO028    -0.2980 -0.2150 -0.1050  ...     NaN    NaN -0.0595  0.1690   \n",
       "05CO032     0.1040 -0.6940  0.3560  ...  0.0105    NaN -0.2850 -0.1480   \n",
       "09CO005     0.3590 -1.1200 -0.0215  ...     NaN    NaN -0.2660  0.0078   \n",
       "11CO027    -0.0733 -0.0244  0.8510  ...     NaN    NaN     NaN  0.3930   \n",
       "11CO052     0.1710 -0.3130  0.0981  ...  0.1510    NaN  0.2730  0.2830   \n",
       "...            ...     ...     ...  ...     ...    ...     ...     ...   \n",
       "20CO001     0.5650 -0.0406 -0.1540  ...     NaN    NaN -0.1340  0.4150   \n",
       "20CO003     0.1710  0.7850  0.3000  ...     NaN    NaN     NaN -0.0415   \n",
       "22CO004    -0.1500 -0.6580  0.0066  ...     NaN    NaN  0.4490 -0.0616   \n",
       "22CO006     0.3990 -1.0000  0.2060  ...     NaN    NaN -0.5610  0.4740   \n",
       "24CO005        NaN     NaN     NaN  ...     NaN    NaN     NaN     NaN   \n",
       "\n",
       "Name        ZRANB2    ZW10  ZWILCH  ZWINT     ZYX   ZZEF1  \n",
       "Patient_ID                                                 \n",
       "05CO028     0.1450  0.1970 -0.3990    NaN -0.7540 -0.4230  \n",
       "05CO032     0.0751  0.0503  0.2300    NaN -0.5900 -0.0644  \n",
       "09CO005     0.5310 -0.3550 -0.6400    NaN -0.4300 -0.3340  \n",
       "11CO027     0.7610  0.3730     NaN    NaN  0.0261 -0.1030  \n",
       "11CO052     0.1690  0.1180  0.2930    NaN -0.1450 -0.0604  \n",
       "...            ...     ...     ...    ...     ...     ...  \n",
       "20CO001    -0.0765 -0.0385     NaN    NaN  0.4260  0.5190  \n",
       "20CO003     0.0830 -0.0374 -0.1320    NaN  0.6810 -0.2970  \n",
       "22CO004     0.3850  0.1160 -0.0437    NaN -0.8840  0.0088  \n",
       "22CO006     0.0836 -0.2510  0.1760    NaN  0.0584 -0.3010  \n",
       "24CO005        NaN     NaN     NaN    NaN     NaN     NaN  \n",
       "\n",
       "[74 rows x 8068 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "proteomics = cl.get_proteomics(tissue_type=\"tumor\")\n",
    "\n",
    "prot_list = list(proteomics.columns)\n",
    "\n",
    "mut_type = cl.get_genotype_all_vars(\"PIK3CA\")\n",
    "\n",
    "\n",
    "joined = mut_type.join(proteomics)\n",
    "\n",
    "#select samples containing hotspot mutations\n",
    "cl_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "                    joined.Location.str.contains('E545K') |\n",
    "                    joined.Location.str.contains('H1047R')]\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "colon_hotspot = cl_hotspot_df.append(wildtype)\n",
    "\n",
    "#drop unncessary columns\n",
    "col_prot_and_mutations = colon_hotspot.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "col_prot_and_mutations[\"Mutation\"] = col_prot_and_mutations['Mutation'].replace(['nonsynonymous SNV'], 'Missense_Mutation')\n",
    "col_prot_and_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>AAMDC</th>\n",
       "      <th>AAMP</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNHIT6</th>\n",
       "      <th>ZNRD1</th>\n",
       "      <th>ZNRF2</th>\n",
       "      <th>ZPR1</th>\n",
       "      <th>ZRANB2</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>05CO028</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.3400</td>\n",
       "      <td>0.3430</td>\n",
       "      <td>-1.660</td>\n",
       "      <td>0.5040</td>\n",
       "      <td>0.1620</td>\n",
       "      <td>0.3190</td>\n",
       "      <td>-0.2980</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>-0.1050</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0595</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>0.1450</td>\n",
       "      <td>0.1970</td>\n",
       "      <td>-0.399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7540</td>\n",
       "      <td>-0.4230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.2200</td>\n",
       "      <td>-0.5750</td>\n",
       "      <td>-0.596</td>\n",
       "      <td>0.2990</td>\n",
       "      <td>-0.0254</td>\n",
       "      <td>0.7640</td>\n",
       "      <td>0.1040</td>\n",
       "      <td>-0.6940</td>\n",
       "      <td>0.3560</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2850</td>\n",
       "      <td>-0.1480</td>\n",
       "      <td>0.0751</td>\n",
       "      <td>0.0503</td>\n",
       "      <td>0.230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5900</td>\n",
       "      <td>-0.0644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>09CO005</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.4900</td>\n",
       "      <td>-0.2000</td>\n",
       "      <td>-1.280</td>\n",
       "      <td>-0.1220</td>\n",
       "      <td>0.2730</td>\n",
       "      <td>0.2230</td>\n",
       "      <td>0.3590</td>\n",
       "      <td>-1.1200</td>\n",
       "      <td>-0.0215</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2660</td>\n",
       "      <td>0.0078</td>\n",
       "      <td>0.5310</td>\n",
       "      <td>-0.3550</td>\n",
       "      <td>-0.640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.4300</td>\n",
       "      <td>-0.3340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO027</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.0900</td>\n",
       "      <td>0.2360</td>\n",
       "      <td>-1.040</td>\n",
       "      <td>0.3070</td>\n",
       "      <td>0.0530</td>\n",
       "      <td>0.4660</td>\n",
       "      <td>-0.0733</td>\n",
       "      <td>-0.0244</td>\n",
       "      <td>0.8510</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3930</td>\n",
       "      <td>0.7610</td>\n",
       "      <td>0.3730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0261</td>\n",
       "      <td>-0.1030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO052</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.2800</td>\n",
       "      <td>-0.6230</td>\n",
       "      <td>-0.932</td>\n",
       "      <td>-0.0443</td>\n",
       "      <td>0.0090</td>\n",
       "      <td>0.0958</td>\n",
       "      <td>0.1710</td>\n",
       "      <td>-0.3130</td>\n",
       "      <td>0.0981</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1510</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2730</td>\n",
       "      <td>0.2830</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>0.1180</td>\n",
       "      <td>0.293</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>-0.0604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO062</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.0901</td>\n",
       "      <td>-0.0789</td>\n",
       "      <td>0.340</td>\n",
       "      <td>-1.2300</td>\n",
       "      <td>0.5190</td>\n",
       "      <td>0.7130</td>\n",
       "      <td>-0.0840</td>\n",
       "      <td>-0.9340</td>\n",
       "      <td>-0.0101</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>0.2040</td>\n",
       "      <td>-0.1130</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0222</td>\n",
       "      <td>-0.0726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16CO006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.1780</td>\n",
       "      <td>0.0866</td>\n",
       "      <td>-0.525</td>\n",
       "      <td>-0.1290</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.2190</td>\n",
       "      <td>-0.1290</td>\n",
       "      <td>-0.1980</td>\n",
       "      <td>0.0203</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5150</td>\n",
       "      <td>0.2840</td>\n",
       "      <td>0.0238</td>\n",
       "      <td>0.0339</td>\n",
       "      <td>0.157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2590</td>\n",
       "      <td>-0.0582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16CO012</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20CO004</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.8700</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.1570</td>\n",
       "      <td>1.2900</td>\n",
       "      <td>0.1330</td>\n",
       "      <td>-0.2230</td>\n",
       "      <td>-0.2460</td>\n",
       "      <td>0.2350</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1970</td>\n",
       "      <td>-0.1240</td>\n",
       "      <td>-0.2830</td>\n",
       "      <td>0.0343</td>\n",
       "      <td>-0.376</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5030</td>\n",
       "      <td>-0.5220</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9 rows × 8068 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation    A1BG    A1CF    A2M    AAAS    AACS   AAGAB  \\\n",
       "Patient_ID                                                                     \n",
       "05CO028     Missense_Mutation -1.3400  0.3430 -1.660  0.5040  0.1620  0.3190   \n",
       "05CO032     Missense_Mutation -1.2200 -0.5750 -0.596  0.2990 -0.0254  0.7640   \n",
       "09CO005     Missense_Mutation -1.4900 -0.2000 -1.280 -0.1220  0.2730  0.2230   \n",
       "11CO027     Missense_Mutation -1.0900  0.2360 -1.040  0.3070  0.0530  0.4660   \n",
       "11CO052     Missense_Mutation -0.2800 -0.6230 -0.932 -0.0443  0.0090  0.0958   \n",
       "11CO062     Missense_Mutation -0.0901 -0.0789  0.340 -1.2300  0.5190  0.7130   \n",
       "16CO006     Missense_Mutation  0.1780  0.0866 -0.525 -0.1290 -0.1450  0.2190   \n",
       "16CO012     Missense_Mutation     NaN     NaN    NaN     NaN     NaN     NaN   \n",
       "20CO004     Missense_Mutation -1.8700  0.4610 -1.850 -0.1570  1.2900  0.1330   \n",
       "\n",
       "Name          AAK1   AAMDC    AAMP  ...  ZNHIT6  ZNRD1   ZNRF2    ZPR1  \\\n",
       "Patient_ID                          ...                                  \n",
       "05CO028    -0.2980 -0.2150 -0.1050  ...     NaN    NaN -0.0595  0.1690   \n",
       "05CO032     0.1040 -0.6940  0.3560  ...  0.0105    NaN -0.2850 -0.1480   \n",
       "09CO005     0.3590 -1.1200 -0.0215  ...     NaN    NaN -0.2660  0.0078   \n",
       "11CO027    -0.0733 -0.0244  0.8510  ...     NaN    NaN     NaN  0.3930   \n",
       "11CO052     0.1710 -0.3130  0.0981  ...  0.1510    NaN  0.2730  0.2830   \n",
       "11CO062    -0.0840 -0.9340 -0.0101  ...     NaN    NaN     NaN  0.2270   \n",
       "16CO006    -0.1290 -0.1980  0.0203  ...     NaN    NaN -0.5150  0.2840   \n",
       "16CO012        NaN     NaN     NaN  ...     NaN    NaN     NaN     NaN   \n",
       "20CO004    -0.2230 -0.2460  0.2350  ...  0.5430    NaN -0.1970 -0.1240   \n",
       "\n",
       "Name        ZRANB2    ZW10  ZWILCH  ZWINT     ZYX   ZZEF1  \n",
       "Patient_ID                                                 \n",
       "05CO028     0.1450  0.1970  -0.399    NaN -0.7540 -0.4230  \n",
       "05CO032     0.0751  0.0503   0.230    NaN -0.5900 -0.0644  \n",
       "09CO005     0.5310 -0.3550  -0.640    NaN -0.4300 -0.3340  \n",
       "11CO027     0.7610  0.3730     NaN    NaN  0.0261 -0.1030  \n",
       "11CO052     0.1690  0.1180   0.293    NaN -0.1450 -0.0604  \n",
       "11CO062     0.2040 -0.1130     NaN    NaN  0.0222 -0.0726  \n",
       "16CO006     0.0238  0.0339   0.157    NaN  0.2590 -0.0582  \n",
       "16CO012        NaN     NaN     NaN    NaN     NaN     NaN  \n",
       "20CO004    -0.2830  0.0343  -0.376    NaN -0.5030 -0.5220  \n",
       "\n",
       "[9 rows x 8068 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_prot_and_mutations.loc[col_prot_and_mutations[\"Mutation\"]==\"Missense_Mutation\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get the difference in medians\n",
    "missense = col_prot_and_mutations[col_prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = col_prot_and_mutations[col_prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "colon_d = {}\n",
    "\n",
    "for prot in proteomics:\n",
    "    dif_colon = missense_med[prot] - wt_med[prot]\n",
    "    colon_d[prot] = dif_colon\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(colon_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#do a t test for every gene and report the p-value\n",
    "cols = list(col_prot_and_mutations.columns[1:])\n",
    "t_test = wrap_ttest(col_prot_and_mutations, 'Mutation', cols, correction_method= \"fdr_bh\")\n",
    "# t_test = t_test.set_index('Comparison')\n",
    "t_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# t_test.head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'NoneType' object is not iterable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-30-ce1fb3f92ead>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m#join median and p value together\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mcolon_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmedian_diff\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mt_test\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mcolon_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mjoin\u001b[0;34m(self, other, on, how, lsuffix, rsuffix, sort)\u001b[0m\n\u001b[1;32m   7207\u001b[0m         \"\"\"\n\u001b[1;32m   7208\u001b[0m         return self._join_compat(\n\u001b[0;32m-> 7209\u001b[0;31m             \u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mon\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mhow\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlsuffix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlsuffix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrsuffix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrsuffix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msort\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msort\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   7210\u001b[0m         )\n\u001b[1;32m   7211\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_join_compat\u001b[0;34m(self, other, on, how, lsuffix, rsuffix, sort)\u001b[0m\n\u001b[1;32m   7238\u001b[0m                 )\n\u001b[1;32m   7239\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 7240\u001b[0;31m             \u001b[0mframes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   7241\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   7242\u001b[0m             \u001b[0mcan_concat\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_unique\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mdf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mframes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: 'NoneType' object is not iterable"
     ]
    }
   ],
   "source": [
    "#join median and p value together\n",
    "colon_df = median_diff.join(t_test)\n",
    "colon_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine Endo, Brca, and Colon into one table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine brca and endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "brca_df=brca_df.rename_axis(None)\n",
    "brca_endo = brca_df.join(endo_df, lsuffix='_Brca', rsuffix='_Endo')\n",
    "brca_endo.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_table = brca_endo.join(colon_df, rsuffix='_colon')\n",
    "final_table.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Only rows that contain a significant pvalue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "significant = final_table.loc[(final_table['P_Value_Brca'] <= .05) |\n",
    "                (final_table['P_Value_Endo'] <= .05) |\n",
    "                (final_table['P_Value'] <= .05)]\n",
    "significant.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "significant.to_csv(\"~/WhenMutationsDontMatter/PIK3CA/csv_files/proteomics.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
