{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load cancer data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "en = cptac.Endometrial()\n",
    "br = cptac.Brca()\n",
    "cl = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 48 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>Amplification</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Single_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E726K</td>\n",
       "      <td>Single_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>Amplification</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Single_mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.K111E</td>\n",
       "      <td>Single_mutation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation       Location  Mutation_Status\n",
       "Patient_ID                                                   \n",
       "C3L-00006   Missense_Mutation        p.E545K  Single_mutation\n",
       "C3L-00008      Wildtype_Tumor    No_mutation   Wildtype_Tumor\n",
       "C3L-00032   Missense_Mutation        p.E545K  Single_mutation\n",
       "C3L-00090      Wildtype_Tumor    No_mutation   Wildtype_Tumor\n",
       "C3L-00098       Amplification  Amplification  Single_mutation\n",
       "...                       ...            ...              ...\n",
       "C3N-01520   Missense_Mutation        p.E726K  Single_mutation\n",
       "C3N-01521       Amplification  Amplification  Single_mutation\n",
       "C3N-01537      Wildtype_Tumor    No_mutation   Wildtype_Tumor\n",
       "C3N-01802      Wildtype_Tumor    No_mutation   Wildtype_Tumor\n",
       "C3N-01825   Missense_Mutation        p.K111E  Single_mutation\n",
       "\n",
       "[95 rows x 3 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut = en.get_genotype_all_vars(\"PIK3CA\")\n",
    "mut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "95"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mut.loc[mut[\"Mutation\"]==\"Missense_Mutation\"])\n",
    "len(mut.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.1230</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.684</td>\n",
       "      <td>0.984</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.737</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.0757</td>\n",
       "      <td>-0.4730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.3200</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.443</td>\n",
       "      <td>0.243</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.0714</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.156</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.0301</td>\n",
       "      <td>-0.4670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>Amplification</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.374</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.537</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.2200</td>\n",
       "      <td>-0.5620</td>\n",
       "      <td>0.937</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.1760</td>\n",
       "      <td>0.0513</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation       Location  Mutation_Status   A1BG     A2M  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00006   Missense_Mutation        p.E545K  Single_mutation -1.180 -0.8630   \n",
       "C3L-00008      Wildtype_Tumor    No_mutation   Wildtype_Tumor -0.685 -1.0700   \n",
       "C3L-00032   Missense_Mutation        p.E545K  Single_mutation -0.528 -1.3200   \n",
       "C3L-00090      Wildtype_Tumor    No_mutation   Wildtype_Tumor -1.670 -1.1900   \n",
       "C3L-00098       Amplification  Amplification  Single_mutation -0.374 -0.0206   \n",
       "\n",
       "Name        A2ML1  A4GALT    AAAS    AACS   AADAT  ...   ZSWIM8  ZSWIM9  \\\n",
       "Patient_ID                                         ...                    \n",
       "C3L-00006  -0.802   0.222  0.2560  0.6650  1.2800  ... -0.08770     NaN   \n",
       "C3L-00008  -0.684   0.984  0.1350  0.3340  1.3000  ... -0.03560     NaN   \n",
       "C3L-00032   0.435     NaN -0.2400  1.0400 -0.0213  ...  0.00112 -0.1450   \n",
       "C3L-00090  -0.443   0.243 -0.0993  0.7570  0.7400  ...  0.07250 -0.0552   \n",
       "C3L-00098  -0.537   0.311  0.3750  0.0131 -1.1000  ... -0.17600     NaN   \n",
       "\n",
       "Name          ZW10  ZWILCH  ZWINT   ZXDC   ZYG11B    ZYX   ZZEF1    ZZZ3  \n",
       "Patient_ID                                                                \n",
       "C3L-00006   0.0229  0.1090    NaN -0.332 -0.43300 -1.020 -0.1230 -0.0859  \n",
       "C3L-00008   0.3630  1.0700  0.737 -0.564 -0.00461 -1.130 -0.0757 -0.4730  \n",
       "C3L-00032   0.0105 -0.1160    NaN  0.151 -0.07400 -0.540  0.3200 -0.4190  \n",
       "C3L-00090  -0.0714  0.0933  0.156 -0.398 -0.07520 -0.797 -0.0301 -0.4670  \n",
       "C3L-00098  -1.2200 -0.5620  0.937 -0.646  0.20700 -1.850 -0.1760  0.0513  \n",
       "\n",
       "[5 rows x 11002 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = en.get_proteomics(tissue_type=\"tumor\")\n",
    "joined = mut.join(prot)\n",
    "joined.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.4330</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.0740</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.32000</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00139</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.Q546P</td>\n",
       "      <td>Multiple_mutation</td>\n",
       "      <td>-0.467</td>\n",
       "      <td>0.3700</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4340</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>-0.1750</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.67500</td>\n",
       "      <td>0.2390</td>\n",
       "      <td>0.1400</td>\n",
       "      <td>1.070</td>\n",
       "      <td>0.6070</td>\n",
       "      <td>0.486</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>-0.632</td>\n",
       "      <td>-0.20300</td>\n",
       "      <td>-0.0685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00143</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.R115L</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-1.120</td>\n",
       "      <td>-1.3100</td>\n",
       "      <td>0.9120</td>\n",
       "      <td>0.418</td>\n",
       "      <td>-0.0768</td>\n",
       "      <td>0.8460</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00212</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-1.270</td>\n",
       "      <td>-1.2700</td>\n",
       "      <td>-0.222</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>-0.620</td>\n",
       "      <td>0.36300</td>\n",
       "      <td>-0.4630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00156</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.L628R</td>\n",
       "      <td>Multiple_mutation</td>\n",
       "      <td>-0.232</td>\n",
       "      <td>-0.0910</td>\n",
       "      <td>0.6180</td>\n",
       "      <td>1.630</td>\n",
       "      <td>-0.2500</td>\n",
       "      <td>1.3300</td>\n",
       "      <td>-1.7200</td>\n",
       "      <td>...</td>\n",
       "      <td>0.06520</td>\n",
       "      <td>-0.2360</td>\n",
       "      <td>0.5010</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>-0.428</td>\n",
       "      <td>-0.2120</td>\n",
       "      <td>-1.340</td>\n",
       "      <td>0.31700</td>\n",
       "      <td>-0.9430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01211</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.636</td>\n",
       "      <td>-0.3710</td>\n",
       "      <td>0.9730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6370</td>\n",
       "      <td>-0.1660</td>\n",
       "      <td>-0.1220</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.41600</td>\n",
       "      <td>-0.7510</td>\n",
       "      <td>0.1790</td>\n",
       "      <td>0.658</td>\n",
       "      <td>0.4700</td>\n",
       "      <td>0.270</td>\n",
       "      <td>-0.1480</td>\n",
       "      <td>-0.139</td>\n",
       "      <td>-0.02990</td>\n",
       "      <td>0.1310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01217</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.743</td>\n",
       "      <td>-1.3700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1930</td>\n",
       "      <td>0.6940</td>\n",
       "      <td>0.3380</td>\n",
       "      <td>...</td>\n",
       "      <td>0.18100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1960</td>\n",
       "      <td>0.274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.501</td>\n",
       "      <td>-0.0468</td>\n",
       "      <td>0.653</td>\n",
       "      <td>-0.04640</td>\n",
       "      <td>-0.1600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01219</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.295</td>\n",
       "      <td>-0.0589</td>\n",
       "      <td>0.4900</td>\n",
       "      <td>-0.109</td>\n",
       "      <td>-0.0495</td>\n",
       "      <td>0.0374</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.29000</td>\n",
       "      <td>0.0457</td>\n",
       "      <td>0.2460</td>\n",
       "      <td>0.589</td>\n",
       "      <td>0.3930</td>\n",
       "      <td>-0.199</td>\n",
       "      <td>-0.1590</td>\n",
       "      <td>-1.090</td>\n",
       "      <td>0.06270</td>\n",
       "      <td>-0.3790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.290</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>3.1700</td>\n",
       "      <td>-0.907</td>\n",
       "      <td>0.0317</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17900</td>\n",
       "      <td>-0.5210</td>\n",
       "      <td>0.0523</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.5850</td>\n",
       "      <td>0.138</td>\n",
       "      <td>-0.1900</td>\n",
       "      <td>-0.966</td>\n",
       "      <td>-0.00627</td>\n",
       "      <td>-0.2490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.266</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.470</td>\n",
       "      <td>0.3980</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.14000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0122</td>\n",
       "      <td>0.250</td>\n",
       "      <td>0.5530</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.0642</td>\n",
       "      <td>-0.437</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.4980</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84 rows × 11002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation     Location    Mutation_Status   A1BG     A2M  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00006   Missense_Mutation      p.E545K    Single_mutation -1.180 -0.8630   \n",
       "C3L-00032   Missense_Mutation      p.E545K    Single_mutation -0.528 -1.3200   \n",
       "C3L-00139   Missense_Mutation      p.Q546P  Multiple_mutation -0.467  0.3700   \n",
       "C3L-00143   Missense_Mutation      p.R115L    Single_mutation -1.120 -1.3100   \n",
       "C3L-00156   Missense_Mutation      p.L628R  Multiple_mutation -0.232 -0.0910   \n",
       "...                       ...          ...                ...    ...     ...   \n",
       "C3N-01211      Wildtype_Tumor  No_mutation     Wildtype_Tumor -0.636 -0.3710   \n",
       "C3N-01217      Wildtype_Tumor  No_mutation     Wildtype_Tumor -0.743 -1.3700   \n",
       "C3N-01219      Wildtype_Tumor  No_mutation     Wildtype_Tumor -0.295 -0.0589   \n",
       "C3N-01537      Wildtype_Tumor  No_mutation     Wildtype_Tumor -0.290 -0.3200   \n",
       "C3N-01802      Wildtype_Tumor  No_mutation     Wildtype_Tumor  0.266  1.3900   \n",
       "\n",
       "Name         A2ML1  A4GALT    AAAS    AACS   AADAT  ...   ZSWIM8  ZSWIM9  \\\n",
       "Patient_ID                                          ...                    \n",
       "C3L-00006  -0.8020   0.222  0.2560  0.6650  1.2800  ... -0.08770     NaN   \n",
       "C3L-00032   0.4350     NaN -0.2400  1.0400 -0.0213  ...  0.00112 -0.1450   \n",
       "C3L-00139  -0.3390     NaN  0.4340  0.0358 -0.1750  ... -0.67500  0.2390   \n",
       "C3L-00143   0.9120   0.418 -0.0768  0.8460 -0.1210  ... -0.00212     NaN   \n",
       "C3L-00156   0.6180   1.630 -0.2500  1.3300 -1.7200  ...  0.06520 -0.2360   \n",
       "...            ...     ...     ...     ...     ...  ...      ...     ...   \n",
       "C3N-01211   0.9730     NaN  0.6370 -0.1660 -0.1220  ... -0.41600 -0.7510   \n",
       "C3N-01217      NaN     NaN -0.1930  0.6940  0.3380  ...  0.18100     NaN   \n",
       "C3N-01219   0.4900  -0.109 -0.0495  0.0374  0.1020  ... -0.29000  0.0457   \n",
       "C3N-01537   3.1700  -0.907  0.0317 -0.0425     NaN  ... -0.17900 -0.5210   \n",
       "C3N-01802  -0.0655   0.470  0.3980 -0.1340  0.4610  ... -0.14000     NaN   \n",
       "\n",
       "Name          ZW10  ZWILCH   ZWINT   ZXDC  ZYG11B    ZYX    ZZEF1    ZZZ3  \n",
       "Patient_ID                                                                 \n",
       "C3L-00006   0.0229   0.109     NaN -0.332 -0.4330 -1.020 -0.12300 -0.0859  \n",
       "C3L-00032   0.0105  -0.116     NaN  0.151 -0.0740 -0.540  0.32000 -0.4190  \n",
       "C3L-00139   0.1400   1.070  0.6070  0.486  0.1690 -0.632 -0.20300 -0.0685  \n",
       "C3L-00143  -1.1900  -1.270 -1.2700 -0.222 -0.3200 -0.620  0.36300 -0.4630  \n",
       "C3L-00156   0.5010   0.352  0.0839 -0.428 -0.2120 -1.340  0.31700 -0.9430  \n",
       "...            ...     ...     ...    ...     ...    ...      ...     ...  \n",
       "C3N-01211   0.1790   0.658  0.4700  0.270 -0.1480 -0.139 -0.02990  0.1310  \n",
       "C3N-01217   0.1960   0.274     NaN -0.501 -0.0468  0.653 -0.04640 -0.1600  \n",
       "C3N-01219   0.2460   0.589  0.3930 -0.199 -0.1590 -1.090  0.06270 -0.3790  \n",
       "C3N-01537   0.0523   0.515  0.5850  0.138 -0.1900 -0.966 -0.00627 -0.2490  \n",
       "C3N-01802  -0.0122   0.250  0.5530  0.387  0.0642 -0.437  0.10400 -0.4980  \n",
       "\n",
       "[84 rows x 11002 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missense_mutations = joined.loc[joined[\"Mutation\"] == \"Missense_Mutation\"]\n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "endo_missense = missense_mutations.append(wildtype)\n",
    "endo_missense"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.863</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.412</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.433</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.123</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.320</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.419</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.145</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.074</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.320</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00139</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.467</td>\n",
       "      <td>0.370</td>\n",
       "      <td>-0.339</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4340</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>-0.1750</td>\n",
       "      <td>0.1810</td>\n",
       "      <td>0.116</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.67500</td>\n",
       "      <td>0.239</td>\n",
       "      <td>0.1400</td>\n",
       "      <td>1.070</td>\n",
       "      <td>0.6070</td>\n",
       "      <td>0.486</td>\n",
       "      <td>0.169</td>\n",
       "      <td>-0.632</td>\n",
       "      <td>-0.203</td>\n",
       "      <td>-0.0685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00143</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.120</td>\n",
       "      <td>-1.310</td>\n",
       "      <td>0.912</td>\n",
       "      <td>0.418</td>\n",
       "      <td>-0.0768</td>\n",
       "      <td>0.8460</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.311</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00212</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-1.270</td>\n",
       "      <td>-1.2700</td>\n",
       "      <td>-0.222</td>\n",
       "      <td>-0.320</td>\n",
       "      <td>-0.620</td>\n",
       "      <td>0.363</td>\n",
       "      <td>-0.4630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00156</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.232</td>\n",
       "      <td>-0.091</td>\n",
       "      <td>0.618</td>\n",
       "      <td>1.630</td>\n",
       "      <td>-0.2500</td>\n",
       "      <td>1.3300</td>\n",
       "      <td>-1.7200</td>\n",
       "      <td>-0.8890</td>\n",
       "      <td>0.633</td>\n",
       "      <td>...</td>\n",
       "      <td>0.06520</td>\n",
       "      <td>-0.236</td>\n",
       "      <td>0.5010</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>-0.428</td>\n",
       "      <td>-0.212</td>\n",
       "      <td>-1.340</td>\n",
       "      <td>0.317</td>\n",
       "      <td>-0.9430</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation   A1BG    A2M  A2ML1  A4GALT    AAAS    AACS  \\\n",
       "Patient_ID                                                                   \n",
       "C3L-00006   Missense_Mutation -1.180 -0.863 -0.802   0.222  0.2560  0.6650   \n",
       "C3L-00032   Missense_Mutation -0.528 -1.320  0.435     NaN -0.2400  1.0400   \n",
       "C3L-00139   Missense_Mutation -0.467  0.370 -0.339     NaN  0.4340  0.0358   \n",
       "C3L-00143   Missense_Mutation -1.120 -1.310  0.912   0.418 -0.0768  0.8460   \n",
       "C3L-00156   Missense_Mutation -0.232 -0.091  0.618   1.630 -0.2500  1.3300   \n",
       "\n",
       "Name         AADAT   AAED1  AAGAB  ...   ZSWIM8  ZSWIM9    ZW10  ZWILCH  \\\n",
       "Patient_ID                         ...                                    \n",
       "C3L-00006   1.2800 -0.3390  0.412  ... -0.08770     NaN  0.0229   0.109   \n",
       "C3L-00032  -0.0213 -0.0479  0.419  ...  0.00112  -0.145  0.0105  -0.116   \n",
       "C3L-00139  -0.1750  0.1810  0.116  ... -0.67500   0.239  0.1400   1.070   \n",
       "C3L-00143  -0.1210     NaN -0.311  ... -0.00212     NaN -1.1900  -1.270   \n",
       "C3L-00156  -1.7200 -0.8890  0.633  ...  0.06520  -0.236  0.5010   0.352   \n",
       "\n",
       "Name         ZWINT   ZXDC  ZYG11B    ZYX  ZZEF1    ZZZ3  \n",
       "Patient_ID                                               \n",
       "C3L-00006      NaN -0.332  -0.433 -1.020 -0.123 -0.0859  \n",
       "C3L-00032      NaN  0.151  -0.074 -0.540  0.320 -0.4190  \n",
       "C3L-00139   0.6070  0.486   0.169 -0.632 -0.203 -0.0685  \n",
       "C3L-00143  -1.2700 -0.222  -0.320 -0.620  0.363 -0.4630  \n",
       "C3L-00156   0.0839 -0.428  -0.212 -1.340  0.317 -0.9430  \n",
       "\n",
       "[5 rows x 11000 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_and_mutations = endo_missense.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "prot_and_mutations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter out hotspot mutations\n",
    "Hotspots are:\n",
    "E542K\n",
    "E545K\n",
    "H1047R\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 94 samples for the ZZZ3 gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#### Get the mutation type, and ptorteomics for PIK3CA\n",
    "\n",
    "en.get_genotype_all_vars(gene)\n",
    "\n",
    "len(mut.loc[mut[\"Mutation\"]==\"Missense_Mutation\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 94 samples for the ZZZ3 gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 311)\n"
     ]
    }
   ],
   "source": [
    "prot = en.get_proteomics(tissue_type=\"tumor\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join mutation type and proteomics together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.1230</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.684</td>\n",
       "      <td>0.984</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.737</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.0757</td>\n",
       "      <td>-0.4730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.3200</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.443</td>\n",
       "      <td>0.243</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.0714</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.156</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.0301</td>\n",
       "      <td>-0.4670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.374</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.537</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.2200</td>\n",
       "      <td>-0.5620</td>\n",
       "      <td>0.937</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.1760</td>\n",
       "      <td>0.0513</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name              Mutation     Location Mutation_Status   A1BG     A2M  A2ML1  \\\n",
       "Patient_ID                                                                      \n",
       "C3L-00006   Wildtype_Tumor  No_mutation  Wildtype_Tumor -1.180 -0.8630 -0.802   \n",
       "C3L-00008   Wildtype_Tumor  No_mutation  Wildtype_Tumor -0.685 -1.0700 -0.684   \n",
       "C3L-00032   Wildtype_Tumor  No_mutation  Wildtype_Tumor -0.528 -1.3200  0.435   \n",
       "C3L-00090   Wildtype_Tumor  No_mutation  Wildtype_Tumor -1.670 -1.1900 -0.443   \n",
       "C3L-00098   Wildtype_Tumor  No_mutation  Wildtype_Tumor -0.374 -0.0206 -0.537   \n",
       "\n",
       "Name        A4GALT    AAAS    AACS   AADAT  ...   ZSWIM8  ZSWIM9    ZW10  \\\n",
       "Patient_ID                                  ...                            \n",
       "C3L-00006    0.222  0.2560  0.6650  1.2800  ... -0.08770     NaN  0.0229   \n",
       "C3L-00008    0.984  0.1350  0.3340  1.3000  ... -0.03560     NaN  0.3630   \n",
       "C3L-00032      NaN -0.2400  1.0400 -0.0213  ...  0.00112 -0.1450  0.0105   \n",
       "C3L-00090    0.243 -0.0993  0.7570  0.7400  ...  0.07250 -0.0552 -0.0714   \n",
       "C3L-00098    0.311  0.3750  0.0131 -1.1000  ... -0.17600     NaN -1.2200   \n",
       "\n",
       "Name        ZWILCH  ZWINT   ZXDC   ZYG11B    ZYX   ZZEF1    ZZZ3  \n",
       "Patient_ID                                                        \n",
       "C3L-00006   0.1090    NaN -0.332 -0.43300 -1.020 -0.1230 -0.0859  \n",
       "C3L-00008   1.0700  0.737 -0.564 -0.00461 -1.130 -0.0757 -0.4730  \n",
       "C3L-00032  -0.1160    NaN  0.151 -0.07400 -0.540  0.3200 -0.4190  \n",
       "C3L-00090   0.0933  0.156 -0.398 -0.07520 -0.797 -0.0301 -0.4670  \n",
       "C3L-00098  -0.5620  0.937 -0.646  0.20700 -1.850 -0.1760  0.0513  \n",
       "\n",
       "[5 rows x 11002 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "joined = mut.join(prot)\n",
    "joined.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Select samples containing missense mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# en_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "#                     joined.Location.str.contains('E545K') |\n",
    "#                     joined.Location.str.contains('H1047R')]\n",
    "missense_mutations = joined.loc[joined[\"Mutation\"] == \"Missense_Mutation\"]\n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join the proteomics for the wildtype mutations to the hotspot mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-01311</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.Y891C</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>-0.261</td>\n",
       "      <td>0.0677</td>\n",
       "      <td>-1.2400</td>\n",
       "      <td>-0.5110</td>\n",
       "      <td>0.1310</td>\n",
       "      <td>0.1210</td>\n",
       "      <td>1.3100</td>\n",
       "      <td>...</td>\n",
       "      <td>0.08220</td>\n",
       "      <td>-0.6450</td>\n",
       "      <td>0.1940</td>\n",
       "      <td>0.0633</td>\n",
       "      <td>0.440</td>\n",
       "      <td>-0.397</td>\n",
       "      <td>-0.30800</td>\n",
       "      <td>-0.484</td>\n",
       "      <td>-0.14400</td>\n",
       "      <td>-0.1340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.6840</td>\n",
       "      <td>0.9840</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.737</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.07570</td>\n",
       "      <td>-0.4730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.32000</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.4430</td>\n",
       "      <td>0.2430</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.0714</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.156</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.03010</td>\n",
       "      <td>-0.4670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01510</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.269</td>\n",
       "      <td>0.9440</td>\n",
       "      <td>1.2000</td>\n",
       "      <td>-0.1550</td>\n",
       "      <td>0.2660</td>\n",
       "      <td>-0.5480</td>\n",
       "      <td>-0.1050</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.20400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5830</td>\n",
       "      <td>0.9130</td>\n",
       "      <td>0.249</td>\n",
       "      <td>-0.515</td>\n",
       "      <td>-0.15100</td>\n",
       "      <td>-1.390</td>\n",
       "      <td>-0.12100</td>\n",
       "      <td>-0.4260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-0.7120</td>\n",
       "      <td>0.0462</td>\n",
       "      <td>-0.0471</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>1.3500</td>\n",
       "      <td>1.2100</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.06990</td>\n",
       "      <td>-0.4010</td>\n",
       "      <td>0.5570</td>\n",
       "      <td>0.8270</td>\n",
       "      <td>0.348</td>\n",
       "      <td>-0.187</td>\n",
       "      <td>-0.10700</td>\n",
       "      <td>-0.830</td>\n",
       "      <td>0.06200</td>\n",
       "      <td>-0.5280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.290</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>3.1700</td>\n",
       "      <td>-0.9070</td>\n",
       "      <td>0.0317</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17900</td>\n",
       "      <td>-0.5210</td>\n",
       "      <td>0.0523</td>\n",
       "      <td>0.5150</td>\n",
       "      <td>0.585</td>\n",
       "      <td>0.138</td>\n",
       "      <td>-0.19000</td>\n",
       "      <td>-0.966</td>\n",
       "      <td>-0.00627</td>\n",
       "      <td>-0.2490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.266</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.4700</td>\n",
       "      <td>0.3980</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.14000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0122</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>0.553</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.06420</td>\n",
       "      <td>-0.437</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.4980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>No_mutation</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.692</td>\n",
       "      <td>0.5890</td>\n",
       "      <td>0.5870</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0835</td>\n",
       "      <td>0.1720</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.02250</td>\n",
       "      <td>0.2980</td>\n",
       "      <td>-0.1240</td>\n",
       "      <td>0.1780</td>\n",
       "      <td>0.147</td>\n",
       "      <td>-0.114</td>\n",
       "      <td>0.13700</td>\n",
       "      <td>1.240</td>\n",
       "      <td>0.00682</td>\n",
       "      <td>-0.0233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89 rows × 11002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation     Location  Mutation_Status   A1BG     A2M  \\\n",
       "Patient_ID                                                                   \n",
       "C3L-01311   Missense_Mutation      p.Y891C  Single_mutation -0.261  0.0677   \n",
       "C3L-00006      Wildtype_Tumor  No_mutation   Wildtype_Tumor -1.180 -0.8630   \n",
       "C3L-00008      Wildtype_Tumor  No_mutation   Wildtype_Tumor -0.685 -1.0700   \n",
       "C3L-00032      Wildtype_Tumor  No_mutation   Wildtype_Tumor -0.528 -1.3200   \n",
       "C3L-00090      Wildtype_Tumor  No_mutation   Wildtype_Tumor -1.670 -1.1900   \n",
       "...                       ...          ...              ...    ...     ...   \n",
       "C3N-01510      Wildtype_Tumor  No_mutation   Wildtype_Tumor -0.269  0.9440   \n",
       "C3N-01520      Wildtype_Tumor  No_mutation   Wildtype_Tumor -1.070 -0.7120   \n",
       "C3N-01537      Wildtype_Tumor  No_mutation   Wildtype_Tumor -0.290 -0.3200   \n",
       "C3N-01802      Wildtype_Tumor  No_mutation   Wildtype_Tumor  0.266  1.3900   \n",
       "C3N-01825      Wildtype_Tumor  No_mutation   Wildtype_Tumor  0.692  0.5890   \n",
       "\n",
       "Name         A2ML1  A4GALT    AAAS    AACS   AADAT  ...   ZSWIM8  ZSWIM9  \\\n",
       "Patient_ID                                          ...                    \n",
       "C3L-01311  -1.2400 -0.5110  0.1310  0.1210  1.3100  ...  0.08220 -0.6450   \n",
       "C3L-00006  -0.8020  0.2220  0.2560  0.6650  1.2800  ... -0.08770     NaN   \n",
       "C3L-00008  -0.6840  0.9840  0.1350  0.3340  1.3000  ... -0.03560     NaN   \n",
       "C3L-00032   0.4350     NaN -0.2400  1.0400 -0.0213  ...  0.00112 -0.1450   \n",
       "C3L-00090  -0.4430  0.2430 -0.0993  0.7570  0.7400  ...  0.07250 -0.0552   \n",
       "...            ...     ...     ...     ...     ...  ...      ...     ...   \n",
       "C3N-01510   1.2000 -0.1550  0.2660 -0.5480 -0.1050  ... -0.20400     NaN   \n",
       "C3N-01520   0.0462 -0.0471  0.2270  1.3500  1.2100  ... -0.06990 -0.4010   \n",
       "C3N-01537   3.1700 -0.9070  0.0317 -0.0425     NaN  ... -0.17900 -0.5210   \n",
       "C3N-01802  -0.0655  0.4700  0.3980 -0.1340  0.4610  ... -0.14000     NaN   \n",
       "C3N-01825   0.5870     NaN -0.0835  0.1720 -0.2150  ... -0.02250  0.2980   \n",
       "\n",
       "Name          ZW10  ZWILCH  ZWINT   ZXDC   ZYG11B    ZYX    ZZEF1    ZZZ3  \n",
       "Patient_ID                                                                 \n",
       "C3L-01311   0.1940  0.0633  0.440 -0.397 -0.30800 -0.484 -0.14400 -0.1340  \n",
       "C3L-00006   0.0229  0.1090    NaN -0.332 -0.43300 -1.020 -0.12300 -0.0859  \n",
       "C3L-00008   0.3630  1.0700  0.737 -0.564 -0.00461 -1.130 -0.07570 -0.4730  \n",
       "C3L-00032   0.0105 -0.1160    NaN  0.151 -0.07400 -0.540  0.32000 -0.4190  \n",
       "C3L-00090  -0.0714  0.0933  0.156 -0.398 -0.07520 -0.797 -0.03010 -0.4670  \n",
       "...            ...     ...    ...    ...      ...    ...      ...     ...  \n",
       "C3N-01510  -0.5830  0.9130  0.249 -0.515 -0.15100 -1.390 -0.12100 -0.4260  \n",
       "C3N-01520   0.5570  0.8270  0.348 -0.187 -0.10700 -0.830  0.06200 -0.5280  \n",
       "C3N-01537   0.0523  0.5150  0.585  0.138 -0.19000 -0.966 -0.00627 -0.2490  \n",
       "C3N-01802  -0.0122  0.2500  0.553  0.387  0.06420 -0.437  0.10400 -0.4980  \n",
       "C3N-01825  -0.1240  0.1780  0.147 -0.114  0.13700  1.240  0.00682 -0.0233  \n",
       "\n",
       "[89 rows x 11002 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c# wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "\n",
    "# endo_hotspot = en_hotspot_df.append(wildtype)\n",
    "endo_missense"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Drop unncecesary columns\n",
    "The resulting dataframe is what we will be working with from this point forward. It contains the mutations type and the proteomics for each gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-01311</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.261</td>\n",
       "      <td>0.0677</td>\n",
       "      <td>-1.240</td>\n",
       "      <td>-0.511</td>\n",
       "      <td>0.1310</td>\n",
       "      <td>0.121</td>\n",
       "      <td>1.3100</td>\n",
       "      <td>-0.0531</td>\n",
       "      <td>-0.255</td>\n",
       "      <td>...</td>\n",
       "      <td>0.08220</td>\n",
       "      <td>-0.6450</td>\n",
       "      <td>0.1940</td>\n",
       "      <td>0.0633</td>\n",
       "      <td>0.440</td>\n",
       "      <td>-0.397</td>\n",
       "      <td>-0.30800</td>\n",
       "      <td>-0.484</td>\n",
       "      <td>-0.1440</td>\n",
       "      <td>-0.1340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.665</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.412</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.1230</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.684</td>\n",
       "      <td>0.984</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.334</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>0.1390</td>\n",
       "      <td>1.330</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.737</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.0757</td>\n",
       "      <td>-0.4730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.040</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.419</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.3200</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.443</td>\n",
       "      <td>0.243</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.757</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>-0.9290</td>\n",
       "      <td>0.229</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.0714</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.156</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.0301</td>\n",
       "      <td>-0.4670</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation   A1BG     A2M  A2ML1  A4GALT    AAAS   AACS  \\\n",
       "Patient_ID                                                                   \n",
       "C3L-01311   Missense_Mutation -0.261  0.0677 -1.240  -0.511  0.1310  0.121   \n",
       "C3L-00006      Wildtype_Tumor -1.180 -0.8630 -0.802   0.222  0.2560  0.665   \n",
       "C3L-00008      Wildtype_Tumor -0.685 -1.0700 -0.684   0.984  0.1350  0.334   \n",
       "C3L-00032      Wildtype_Tumor -0.528 -1.3200  0.435     NaN -0.2400  1.040   \n",
       "C3L-00090      Wildtype_Tumor -1.670 -1.1900 -0.443   0.243 -0.0993  0.757   \n",
       "\n",
       "Name         AADAT   AAED1  AAGAB  ...   ZSWIM8  ZSWIM9    ZW10  ZWILCH  \\\n",
       "Patient_ID                         ...                                    \n",
       "C3L-01311   1.3100 -0.0531 -0.255  ...  0.08220 -0.6450  0.1940  0.0633   \n",
       "C3L-00006   1.2800 -0.3390  0.412  ... -0.08770     NaN  0.0229  0.1090   \n",
       "C3L-00008   1.3000  0.1390  1.330  ... -0.03560     NaN  0.3630  1.0700   \n",
       "C3L-00032  -0.0213 -0.0479  0.419  ...  0.00112 -0.1450  0.0105 -0.1160   \n",
       "C3L-00090   0.7400 -0.9290  0.229  ...  0.07250 -0.0552 -0.0714  0.0933   \n",
       "\n",
       "Name        ZWINT   ZXDC   ZYG11B    ZYX   ZZEF1    ZZZ3  \n",
       "Patient_ID                                                \n",
       "C3L-01311   0.440 -0.397 -0.30800 -0.484 -0.1440 -0.1340  \n",
       "C3L-00006     NaN -0.332 -0.43300 -1.020 -0.1230 -0.0859  \n",
       "C3L-00008   0.737 -0.564 -0.00461 -1.130 -0.0757 -0.4730  \n",
       "C3L-00032     NaN  0.151 -0.07400 -0.540  0.3200 -0.4190  \n",
       "C3L-00090   0.156 -0.398 -0.07520 -0.797 -0.0301 -0.4670  \n",
       "\n",
       "[5 rows x 11000 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_and_mutations = endo_missense.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "prot_and_mutations.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run T-test and difference of Median for each cancer type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the median proteomic expression among all samples, for each gene.\n",
    "This is done for missense mutations and wildtype separately\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #get all wildtype mutations\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the medain difference between missense and wildtype for each gene.\n",
    "This is done by (for each gene) subtracting the median of the missense mutations from the median of the wildtype.\n",
    "This means that if the difference is positive the proteomics of the wildtype have higher expression than the missense mutation. If the difference is negative then the missense mutation has a higher expression than the wildtype."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.07100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.10300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.47135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>0.22545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>0.07410</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median\n",
       "A1BG                 0.07100\n",
       "A2M                  0.10300\n",
       "A2ML1                0.47135\n",
       "A4GALT               0.22545\n",
       "AAAS                 0.07410"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_d = {}\n",
    "\n",
    "for gene in prot_and_mutations:\n",
    "    if gene == \"Mutation\": continue\n",
    "    dif_endo =  missense_med[gene] - wt_med[gene]\n",
    "    endo_d[gene] = dif_endo\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(endo_d, orient='index', columns=['Difference_In_Median'])\n",
    "median_diff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(prot_and_mutations.loc[prot_and_mutations['Mutation']==\"Missense_Mutation\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.4120</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.4330</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.4190</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.0740</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.32000</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00139</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.467</td>\n",
       "      <td>0.3700</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4340</td>\n",
       "      <td>0.0358</td>\n",
       "      <td>-0.1750</td>\n",
       "      <td>0.1810</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.67500</td>\n",
       "      <td>0.2390</td>\n",
       "      <td>0.1400</td>\n",
       "      <td>1.070</td>\n",
       "      <td>0.6070</td>\n",
       "      <td>0.486</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>-0.632</td>\n",
       "      <td>-0.20300</td>\n",
       "      <td>-0.0685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00143</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.120</td>\n",
       "      <td>-1.3100</td>\n",
       "      <td>0.9120</td>\n",
       "      <td>0.418</td>\n",
       "      <td>-0.0768</td>\n",
       "      <td>0.8460</td>\n",
       "      <td>-0.1210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3110</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00212</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-1.270</td>\n",
       "      <td>-1.2700</td>\n",
       "      <td>-0.222</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>-0.620</td>\n",
       "      <td>0.36300</td>\n",
       "      <td>-0.4630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00156</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.232</td>\n",
       "      <td>-0.0910</td>\n",
       "      <td>0.6180</td>\n",
       "      <td>1.630</td>\n",
       "      <td>-0.2500</td>\n",
       "      <td>1.3300</td>\n",
       "      <td>-1.7200</td>\n",
       "      <td>-0.8890</td>\n",
       "      <td>0.6330</td>\n",
       "      <td>...</td>\n",
       "      <td>0.06520</td>\n",
       "      <td>-0.2360</td>\n",
       "      <td>0.5010</td>\n",
       "      <td>0.352</td>\n",
       "      <td>0.0839</td>\n",
       "      <td>-0.428</td>\n",
       "      <td>-0.2120</td>\n",
       "      <td>-1.340</td>\n",
       "      <td>0.31700</td>\n",
       "      <td>-0.9430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01211</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.636</td>\n",
       "      <td>-0.3710</td>\n",
       "      <td>0.9730</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6370</td>\n",
       "      <td>-0.1660</td>\n",
       "      <td>-0.1220</td>\n",
       "      <td>0.2320</td>\n",
       "      <td>0.3790</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.41600</td>\n",
       "      <td>-0.7510</td>\n",
       "      <td>0.1790</td>\n",
       "      <td>0.658</td>\n",
       "      <td>0.4700</td>\n",
       "      <td>0.270</td>\n",
       "      <td>-0.1480</td>\n",
       "      <td>-0.139</td>\n",
       "      <td>-0.02990</td>\n",
       "      <td>0.1310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01217</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.743</td>\n",
       "      <td>-1.3700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1930</td>\n",
       "      <td>0.6940</td>\n",
       "      <td>0.3380</td>\n",
       "      <td>0.0989</td>\n",
       "      <td>-0.0137</td>\n",
       "      <td>...</td>\n",
       "      <td>0.18100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1960</td>\n",
       "      <td>0.274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.501</td>\n",
       "      <td>-0.0468</td>\n",
       "      <td>0.653</td>\n",
       "      <td>-0.04640</td>\n",
       "      <td>-0.1600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01219</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.295</td>\n",
       "      <td>-0.0589</td>\n",
       "      <td>0.4900</td>\n",
       "      <td>-0.109</td>\n",
       "      <td>-0.0495</td>\n",
       "      <td>0.0374</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>-0.3210</td>\n",
       "      <td>0.1830</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.29000</td>\n",
       "      <td>0.0457</td>\n",
       "      <td>0.2460</td>\n",
       "      <td>0.589</td>\n",
       "      <td>0.3930</td>\n",
       "      <td>-0.199</td>\n",
       "      <td>-0.1590</td>\n",
       "      <td>-1.090</td>\n",
       "      <td>0.06270</td>\n",
       "      <td>-0.3790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>-0.290</td>\n",
       "      <td>-0.3200</td>\n",
       "      <td>3.1700</td>\n",
       "      <td>-0.907</td>\n",
       "      <td>0.0317</td>\n",
       "      <td>-0.0425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2730</td>\n",
       "      <td>-0.0747</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17900</td>\n",
       "      <td>-0.5210</td>\n",
       "      <td>0.0523</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.5850</td>\n",
       "      <td>0.138</td>\n",
       "      <td>-0.1900</td>\n",
       "      <td>-0.966</td>\n",
       "      <td>-0.00627</td>\n",
       "      <td>-0.2490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>0.266</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.470</td>\n",
       "      <td>0.3980</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.4610</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.14000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0122</td>\n",
       "      <td>0.250</td>\n",
       "      <td>0.5530</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.0642</td>\n",
       "      <td>-0.437</td>\n",
       "      <td>0.10400</td>\n",
       "      <td>-0.4980</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation   A1BG     A2M   A2ML1  A4GALT    AAAS    AACS  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00006   Missense_Mutation -1.180 -0.8630 -0.8020   0.222  0.2560  0.6650   \n",
       "C3L-00032   Missense_Mutation -0.528 -1.3200  0.4350     NaN -0.2400  1.0400   \n",
       "C3L-00139   Missense_Mutation -0.467  0.3700 -0.3390     NaN  0.4340  0.0358   \n",
       "C3L-00143   Missense_Mutation -1.120 -1.3100  0.9120   0.418 -0.0768  0.8460   \n",
       "C3L-00156   Missense_Mutation -0.232 -0.0910  0.6180   1.630 -0.2500  1.3300   \n",
       "...                       ...    ...     ...     ...     ...     ...     ...   \n",
       "C3N-01211      Wildtype_Tumor -0.636 -0.3710  0.9730     NaN  0.6370 -0.1660   \n",
       "C3N-01217      Wildtype_Tumor -0.743 -1.3700     NaN     NaN -0.1930  0.6940   \n",
       "C3N-01219      Wildtype_Tumor -0.295 -0.0589  0.4900  -0.109 -0.0495  0.0374   \n",
       "C3N-01537      Wildtype_Tumor -0.290 -0.3200  3.1700  -0.907  0.0317 -0.0425   \n",
       "C3N-01802      Wildtype_Tumor  0.266  1.3900 -0.0655   0.470  0.3980 -0.1340   \n",
       "\n",
       "Name         AADAT   AAED1   AAGAB  ...   ZSWIM8  ZSWIM9    ZW10  ZWILCH  \\\n",
       "Patient_ID                          ...                                    \n",
       "C3L-00006   1.2800 -0.3390  0.4120  ... -0.08770     NaN  0.0229   0.109   \n",
       "C3L-00032  -0.0213 -0.0479  0.4190  ...  0.00112 -0.1450  0.0105  -0.116   \n",
       "C3L-00139  -0.1750  0.1810  0.1160  ... -0.67500  0.2390  0.1400   1.070   \n",
       "C3L-00143  -0.1210     NaN -0.3110  ... -0.00212     NaN -1.1900  -1.270   \n",
       "C3L-00156  -1.7200 -0.8890  0.6330  ...  0.06520 -0.2360  0.5010   0.352   \n",
       "...            ...     ...     ...  ...      ...     ...     ...     ...   \n",
       "C3N-01211  -0.1220  0.2320  0.3790  ... -0.41600 -0.7510  0.1790   0.658   \n",
       "C3N-01217   0.3380  0.0989 -0.0137  ...  0.18100     NaN  0.1960   0.274   \n",
       "C3N-01219   0.1020 -0.3210  0.1830  ... -0.29000  0.0457  0.2460   0.589   \n",
       "C3N-01537      NaN -0.2730 -0.0747  ... -0.17900 -0.5210  0.0523   0.515   \n",
       "C3N-01802   0.4610  1.0400  0.3630  ... -0.14000     NaN -0.0122   0.250   \n",
       "\n",
       "Name         ZWINT   ZXDC  ZYG11B    ZYX    ZZEF1    ZZZ3  \n",
       "Patient_ID                                                 \n",
       "C3L-00006      NaN -0.332 -0.4330 -1.020 -0.12300 -0.0859  \n",
       "C3L-00032      NaN  0.151 -0.0740 -0.540  0.32000 -0.4190  \n",
       "C3L-00139   0.6070  0.486  0.1690 -0.632 -0.20300 -0.0685  \n",
       "C3L-00143  -1.2700 -0.222 -0.3200 -0.620  0.36300 -0.4630  \n",
       "C3L-00156   0.0839 -0.428 -0.2120 -1.340  0.31700 -0.9430  \n",
       "...            ...    ...     ...    ...      ...     ...  \n",
       "C3N-01211   0.4700  0.270 -0.1480 -0.139 -0.02990  0.1310  \n",
       "C3N-01217      NaN -0.501 -0.0468  0.653 -0.04640 -0.1600  \n",
       "C3N-01219   0.3930 -0.199 -0.1590 -1.090  0.06270 -0.3790  \n",
       "C3N-01537   0.5850  0.138 -0.1900 -0.966 -0.00627 -0.2490  \n",
       "C3N-01802   0.5530  0.387  0.0642 -0.437  0.10400 -0.4980  \n",
       "\n",
       "[84 rows x 11000 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_and_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:153: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:257: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "u.wrap_ttest(prot_and_mutations, 'Mutation')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import operator\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Missense_Mutation', 'Wildtype_Tumor'], dtype=object)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    " prot_and_mutations[\"Mutation\"].unique()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "label_values = prot_and_mutations[\"Mutation\"].unique()\n",
    "\n",
    "'''Partition dataframe into two sets, one for each of the two unique values from the label column'''\n",
    "partition1 = prot_and_mutations.loc[prot_and_mutations[\"Mutation\"] == label_values[0]]\n",
    "partition2 = prot_and_mutations.loc[prot_and_mutations[\"Mutation\"] == label_values[1]]\n",
    "\n",
    "'''If no comparison columns specified, use all columns except the specified labed column'''\n",
    "comparison_columns = list(prot_and_mutations.columns)\n",
    "comparison_columns.remove(\"Mutation\")\n",
    "\n",
    "number_of_comparisons = len(comparison_columns)\n",
    "\n",
    "'''Store comparisons and p-values in two arrays'''\n",
    "comparisons = []\n",
    "pvals = []\n",
    "\n",
    "'''Loop through each comparison column, perform the t-test, and record the p-val'''\n",
    "times_through = 0\n",
    "for column in comparison_columns:  \n",
    "    times_through += 1\n",
    "    if len(partition1[column].dropna(axis=0)) <= 1:\n",
    "#         comparison_columns.remove(column)\n",
    "        continue\n",
    "    elif len(partition2[column].dropna(axis=0)) <= 1:\n",
    "#         comparison_columns.remove(column)\n",
    "        continue\n",
    "    else:\n",
    "        stat, pval = scipy.stats.ttest_ind(partition1[column].dropna(axis=0), partition2[column].dropna(axis=0))\n",
    "        comparisons.append(column)\n",
    "        pvals.append(pval)\n",
    "    \n",
    "'''Correct for multiple testing to determine if each comparison meets the new cutoff'''\n",
    "results = statsmodels.stats.multitest.multipletests(pvals=pvals, alpha=.05, method='fdr_bh')\n",
    "reject = results[0]\n",
    "\n",
    "\n",
    "'''Format results in a pandas dataframe'''\n",
    "results_df = pd.DataFrame(columns=['Comparison','P_Value'])\n",
    "\n",
    "for i in range(0, len(reject)):\n",
    "    if reject[i]:\n",
    "        results_df = results_df.append({'Comparison':comparisons[i],'P_Value':pvals[i]}, ignore_index=True)\n",
    "\n",
    "\n",
    "'''Sort dataframe by ascending p-value'''\n",
    "results_df = results_df.sort_values(by='P_Value', ascending=True)\n",
    "results_df = results_df.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0006916028868550783"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "min(pvals)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<div id=\"altair-viz-12608220942c49b8b52f8a10a9987b42\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-12608220942c49b8b52f8a10a9987b42\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-12608220942c49b8b52f8a10a9987b42\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm//vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm//vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm//vega-lite@4.8.1?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm//vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function loadScript(lib) {\n",
       "      return new Promise(function(resolve, reject) {\n",
       "        var s = document.createElement('script');\n",
       "        s.src = paths[lib];\n",
       "        s.async = true;\n",
       "        s.onload = () => resolve(paths[lib]);\n",
       "        s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "        document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "      });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else if (typeof vegaEmbed === \"function\") {\n",
       "      displayChart(vegaEmbed);\n",
       "    } else {\n",
       "      loadScript(\"vega\")\n",
       "        .then(() => loadScript(\"vega-lite\"))\n",
       "        .then(() => loadScript(\"vega-embed\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 400, \"continuousHeight\": 300}}, \"data\": {\"url\": \"altair-data-c39da52ecf51ef7687f6d8e00f09b897.json\", \"format\": {\"type\": \"json\"}}, \"mark\": \"bar\", \"encoding\": {\"x\": {\"type\": \"quantitative\", \"bin\": {\"step\": 0.05}, \"field\": \"p_val\"}, \"y\": {\"type\": \"quantitative\", \"aggregate\": \"count\"}}, \"$schema\": \"https://vega.github.io/schema/vega-lite/v4.8.1.json\"}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import altair as alt\n",
    "alt.data_transformers.disable_max_rows()\n",
    "alt.data_transformers.enable('json')\n",
    "alt.Chart(pd.DataFrame({\"p_val\":pvals})).mark_bar().encode(\n",
    "    x=alt.X(\"p_val:Q\",bin=alt.Bin(step=0.05)),\n",
    "    y=\"count()\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Do a t-test for every gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = list(prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(prot_and_mutations, 'Mutation')\n",
    "# t_test = t_test.set_index('Comparison')\n",
    "t_test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join difference in median and t-test p value into the same dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 4 join median and p value together\n",
    "endo_df = median_diff.join(t_test)\n",
    "endo_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Brca\n",
    "The Brca data uses different database_IDs, giving us multiple proteomic values for each gene. To deal with this, we will first separate genes that have multiple proteomics values recoreded (multiple database_IDs) from ones that don't."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Dealing with multiple database_IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Separate genes with multiple database_IDs from genes with multiple database_IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "no_duplicates = proteomics[proteomics.columns[~fil]]\n",
    "no_duplicates"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Dealing with genes that don't have multiple database_IDs\n",
    "These will be processed the same way the data for endometrial was."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#FOR NO DUPLICATES\n",
    "#Flatten the multiindex (since there are no duplicate gene names, this won't be a problem)\n",
    "no_duplicates = cptac.utils.reduce_multiindex(no_duplicates, flatten=True)\n",
    "\n",
    "#get mutation_tyoe\n",
    "mut_type = br.get_genotype_all_vars('PIK3CA')\n",
    "\n",
    "#join proteomics and mutation type\n",
    "joined = mut_type.join(no_duplicates)\n",
    "\n",
    "missense_mutations = joined.loc[joined[\"Mutation\"] == \"Missense_Mutation\"]\n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "brca_missense = missense_mutations.append(wildtype)\n",
    "\n",
    "\n",
    "#drop unncessary columns\n",
    "prot_and_mutations = brca_missense.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "# Get the difference in medians\n",
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "no_dup_d = {}\n",
    "\n",
    "for prot in no_duplicates:\n",
    "    dif_brca = missense_med[prot] - wt_med[prot]\n",
    "    no_dup_d[prot] = dif_brca\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(no_dup_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#Do a t test for every gene and report the p-value\n",
    "genes = list(prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(prot_and_mutations, 'Mutation', correction_method=\"fdr_bh\")\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "#join median and p value together\n",
    "no_dup_df = median_diff.join(t_test)\n",
    "no_dup_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Dealing with genes that do have multiple database_IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get t-test results and median difference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#DEALING WITH DUPLICATES\n",
    "#flatten the multiindex\n",
    "duplicates = cptac.utils.reduce_multiindex(duplicates, flatten=True)\n",
    "\n",
    "# We need to figure which of databaseID we want to use, for each protein\n",
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "duplicate_gene_names = list(set(duplicates.columns.get_level_values('Name'))) #get a list of genes names that are duplicated\n",
    "duplicates = cptac.utils.reduce_multiindex(duplicates, flatten=True)\n",
    "duplicates\n",
    "\n",
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "\n",
    "mut_type = br.get_genotype_all_vars(\"PIK3CA\")\n",
    "\n",
    "\n",
    "joined = mut_type.join(duplicates)\n",
    "\n",
    "missense_mutations = joined.loc[joined[\"Mutation\"] == \"Missense_Mutation\"]\n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "brca_missense = missense_mutations.append(wildtype)\n",
    "\n",
    "\n",
    "#drop unncessary columns\n",
    "prot_and_mutations = brca_missense.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "\n",
    "# get the difference in medians\n",
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "duplicates_d = {}\n",
    "\n",
    "for prot in duplicates:\n",
    "    dif_brca =  missense_med[prot] - wt_med[prot]\n",
    "    duplicates_d[prot] = dif_brca\n",
    "    \n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(duplicates_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "\n",
    "#step 3 do a t test for every gene and report the p-value\n",
    "cols = list(prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(prot_and_mutations, 'Mutation', correction_method=\"fdr_bh\")\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "# Step 4 join median and p value together\n",
    "dup_df = median_diff.join(t_test)\n",
    "dup_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Select the database_ID that gives a lower p value from the t-test, and append selected database_ID to the no_duplicate dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for each duplicated gene name select the one with the min p val and append to no_duplicat df\n",
    "for name in duplicate_gene_names:\n",
    "    gene_duplicate = dup_df[dup_df.index.str.contains(name)]\n",
    "    #select the one with the min p value for t test\n",
    "    min_row = gene_duplicate[gene_duplicate.P_Value == gene_duplicate.P_Value.min()]\n",
    "    #append the selected version onto no_duplicate dataframe\n",
    "    no_dup_df=no_dup_df.append(min_row)\n",
    "no_dup_df=no_dup_df.sort_index()\n",
    "no_dup_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Reformat the table by getting rid of databaseIDs so we can join it to Endo and Brca"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "split = no_dup_df.index.str.split('_', 1).str[0]\n",
    "no_dup_df['new_index']=split\n",
    "brca_df = no_dup_df.set_index(\"new_index\")\n",
    "brca_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "proteomics = cl.get_proteomics(tissue_type=\"tumor\")\n",
    "\n",
    "prot_list = list(proteomics.columns)\n",
    "\n",
    "mut_type = cl.get_genotype_all_vars(\"PIK3CA\")\n",
    "\n",
    "\n",
    "joined = mut_type.join(proteomics)\n",
    "joined[\"Mutation\"] = joined['Mutation'].replace(['nonsynonymous SNV'], 'Missense_Mutation')\n",
    "\n",
    "\n",
    "#select samples containing hotspot mutations\n",
    "missense_mutations = joined.loc[joined[\"Mutation\"] == \"Missense_Mutation\"]\n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "colon_missense = missense_mutations.append(wildtype)\n",
    "\n",
    "\n",
    "#drop unncessary columns\n",
    "prot_and_mutations = colon_missense.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "\n",
    "#get the difference in medians\n",
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"nonsynonymous SNV\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "colon_d = {}\n",
    "\n",
    "for prot in proteomics:\n",
    "    dif_colon = missense_med[prot] - wt_med[prot]\n",
    "    colon_d[prot] = dif_colon\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(colon_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#do a t test for every gene and report the p-value\n",
    "cols = list(prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(prot_and_mutations, 'Mutation', correction_method=\"fdr_bh\")\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "#join median and p value together\n",
    "colon_df = median_diff.join(t_test)\n",
    "colon_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine Endo, Brca, and Colon into one table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine brca and endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "brca_df=brca_df.rename_axis(None)\n",
    "brca_endo = brca_df.join(endo_df, lsuffix='_Brca', rsuffix='_Endo')\n",
    "brca_endo.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_table = brca_endo.join(colon_df, rsuffix='_colon')\n",
    "final_table.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Only rows that contain a significant pvalue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "significant = final_table.loc[(final_table['P_Value_Brca'] <= .05) |\n",
    "                (final_table['P_Value_Endo'] <= .05) |\n",
    "                (final_table['P_Value'] <= .05)]\n",
    "significant.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "significant.to_csv(\"~/WhenMutationsDontMatter/PIK3CA/csv_files/proteomics_no_hotspots.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
