{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make tables: calculate p-values and differential expressions for each cancer "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a dataframe with p-value results from t-tests for all proteins (trans proteomics when PTEN has cnv deletions compared to PTEN wildtype). The dataframe also includes the change in medians between deletions and wildtype. Prepare these tables for further analysis by creating csv files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import re\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Returns a dataframe with proteomics and mutation type\n",
    "\n",
    "# all_prot: list of trans genes\n",
    "\n",
    "def all_prot_format_df(cancer_object, all_prot, gene_in = 'PTEN', utils = u):\n",
    "    mut_type = cancer_object.get_genotype_all_vars(gene_in)\n",
    "    \n",
    "    if cancer_object.get_cancer_type() not in ('luad'):\n",
    "        # Keep only tumor samples from proteomics\n",
    "        prot_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "            mutations_genes = [gene_in], omics_df_name = 'proteomics', omics_genes = all_prot, tissue_type = 'tumor')\n",
    "        # Reduce a multiindex \n",
    "        if isinstance(prot_and_mutations.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "            prot_and_mutations = utils.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "            prot_and_mutations = rename_duplicate_cols(prot_and_mutations)\n",
    "        \n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        mut_type = mut_type[['Mutation']] \n",
    "        prot_df = prot_and_mutations.iloc[:,:-4] # Keep only proteomics\n",
    "        merged = prot_df.join(mut_type)\n",
    "        \n",
    "        # Keep only Wildtype and deletion\n",
    "        compare = ['Wildtype_Tumor','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "    \n",
    "    \n",
    "    # Luad has no somatic mutations for PTEN which changes some things\n",
    "    elif cancer_object.get_cancer_type() in ('luad'): \n",
    "        # get_genotype_all_vars add cnv data under the column PTEN\n",
    "        mut_type = mut_type.drop(columns= gene_in)\n",
    "        # different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "        omics = cancer_object.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics',genes1= gene_in, \n",
    "            genes2= all_prot, tissue_type = 'tumor')\n",
    "        omics = utils.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "        omics = rename_duplicate_cols(omics)\n",
    "        omics = omics.drop(columns='PTEN_CNV')\n",
    "       \n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        merged = omics.join(mut_type) \n",
    "        # Keep only No_Mutation (wildtype) and deletion\n",
    "        compare = ['No_Mutation','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        mut_wt = merged[get]\n",
    "        mut_wt['Mutation'] = np.where(\n",
    "            mut_wt['Mutation'] == 'No_Mutation', 'Wildtype_Tumor', 'Deletion')\n",
    "    \n",
    "    mut_wt = mut_wt.rename(columns={col: re.sub(r'_proteomics', '', col) for col in mut_wt.columns.tolist()})\n",
    "    return mut_wt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Params\n",
    "cancer_object: Object. Variable for the loaded cancer data set.\n",
    "cancer_name: Str. name to add to the created dataframe.\n",
    "del_wt_df: DataFrame. Only samples with deletions and wildtype for PTEN. \n",
    "all_prot_list: List. All proteins in proteomics data frame. \n",
    "\n",
    "Returns a dataframe with the difference in medians between proteomics with PTEN wt and del (wt - del). \n",
    "'''\n",
    "\n",
    "def get_change_in_medians_df(cancer_object, cancer_name, del_wt_df, all_prot_list):\n",
    "    if  cancer_object.get_cancer_type() in ('endometrial'):\n",
    "        d = del_wt_df[del_wt_df.Mutation == \"Truncation\"]\n",
    "        wt = del_wt_df[del_wt_df.Mutation == \"Wildtype_Tumor\"]\n",
    "    else:\n",
    "        d = del_wt_df[del_wt_df.Mutation == \"Deletion\"]\n",
    "        wt = del_wt_df[del_wt_df.Mutation == \"Wildtype_Tumor\"]\n",
    "    del_med = d.median()\n",
    "    wt_med = wt.median()\n",
    "\n",
    "    med_dict = {}\n",
    "    \n",
    "    # Correlation: + is mutant up compared to wt, - is mutant down\n",
    "    for prot in all_prot_list:\n",
    "        if prot in del_med.index and prot in wt_med.index:\n",
    "            dif = del_med[prot] - wt_med[prot]\n",
    "            med_dict[prot] = dif\n",
    "        \n",
    "    df = pd.DataFrame.from_dict(med_dict, orient='index', columns=[cancer_name+'_Median'])\n",
    "    df = df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "    \n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Adds '_i' to duplicate col names where i is a number (increases with each duplicate)\n",
    "\n",
    "def rename_duplicate_cols(df):\n",
    "    cols = pd.Series(df.columns[:])\n",
    "\n",
    "    for dup in cols[cols.duplicated()].unique(): \n",
    "        cols[cols[cols == dup].index.values.tolist()] = [dup + '_' + str(i) if i != 0 else dup for i in range(sum(cols == dup))]\n",
    "\n",
    "    # rename the columns with the cols list.\n",
    "    df.columns=cols\n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Step 1: Create data frames with p-values and differential expressions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each cancer needs a data frame containing only samples that have PTEN cnv deletions and PTEN wildtype with trans proteomics. Use wrap_ttest to run t-tests for all genes in the proteomic data frame. Use get_change_in_medians_df to create the data frame with differential expression values. Merge both data frames. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load in cancer data sets from cptac. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "#g = cptac.Gbm()\n",
    "#en = cptac.Endometrial()\n",
    "#h = cptac.Hnscc()\n",
    "#l = cptac.Luad()\n",
    "#ls = cptac.Lscc()\n",
    "#o = cptac.Ovarian()\n",
    "col = cptac.Colon()\n",
    "#b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>AAMDC</th>\n",
       "      <th>AAMP</th>\n",
       "      <th>AAR2</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNRD1</th>\n",
       "      <th>ZNRF2</th>\n",
       "      <th>ZPR1</th>\n",
       "      <th>ZRANB2</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01CO001</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO005</th>\n",
       "      <td>-1.10</td>\n",
       "      <td>0.318</td>\n",
       "      <td>-0.487</td>\n",
       "      <td>0.0995</td>\n",
       "      <td>0.1550</td>\n",
       "      <td>0.1690</td>\n",
       "      <td>0.0653</td>\n",
       "      <td>-0.147</td>\n",
       "      <td>0.114</td>\n",
       "      <td>0.341</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0384</td>\n",
       "      <td>0.0221</td>\n",
       "      <td>0.2500</td>\n",
       "      <td>0.0869</td>\n",
       "      <td>0.0331</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.733</td>\n",
       "      <td>-0.2650</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO006</th>\n",
       "      <td>-1.12</td>\n",
       "      <td>-0.441</td>\n",
       "      <td>-0.347</td>\n",
       "      <td>-0.0029</td>\n",
       "      <td>0.0957</td>\n",
       "      <td>0.3960</td>\n",
       "      <td>-0.0363</td>\n",
       "      <td>-0.549</td>\n",
       "      <td>0.220</td>\n",
       "      <td>0.248</td>\n",
       "      <td>...</td>\n",
       "      <td>0.324</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3940</td>\n",
       "      <td>0.0846</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>-0.3450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.658</td>\n",
       "      <td>0.0052</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO008</th>\n",
       "      <td>-1.20</td>\n",
       "      <td>0.160</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>0.1190</td>\n",
       "      <td>-0.0924</td>\n",
       "      <td>0.0187</td>\n",
       "      <td>-0.2140</td>\n",
       "      <td>0.328</td>\n",
       "      <td>-0.282</td>\n",
       "      <td>-0.348</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3840</td>\n",
       "      <td>-0.1680</td>\n",
       "      <td>0.3570</td>\n",
       "      <td>-0.3250</td>\n",
       "      <td>0.3490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.821</td>\n",
       "      <td>0.2000</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO013</th>\n",
       "      <td>-1.89</td>\n",
       "      <td>0.112</td>\n",
       "      <td>-0.329</td>\n",
       "      <td>0.6700</td>\n",
       "      <td>0.1160</td>\n",
       "      <td>0.3130</td>\n",
       "      <td>-0.2380</td>\n",
       "      <td>-0.274</td>\n",
       "      <td>-0.554</td>\n",
       "      <td>0.270</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0656</td>\n",
       "      <td>-0.2950</td>\n",
       "      <td>0.4630</td>\n",
       "      <td>-0.4480</td>\n",
       "      <td>0.0493</td>\n",
       "      <td>-0.904</td>\n",
       "      <td>-0.1580</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 8067 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        A1BG   A1CF    A2M    AAAS    AACS   AAGAB    AAK1  AAMDC   AAMP  \\\n",
       "Patient_ID                                                                     \n",
       "01CO001      NaN    NaN    NaN     NaN     NaN     NaN     NaN    NaN    NaN   \n",
       "01CO005    -1.10  0.318 -0.487  0.0995  0.1550  0.1690  0.0653 -0.147  0.114   \n",
       "01CO006    -1.12 -0.441 -0.347 -0.0029  0.0957  0.3960 -0.0363 -0.549  0.220   \n",
       "01CO008    -1.20  0.160 -1.850  0.1190 -0.0924  0.0187 -0.2140  0.328 -0.282   \n",
       "01CO013    -1.89  0.112 -0.329  0.6700  0.1160  0.3130 -0.2380 -0.274 -0.554   \n",
       "\n",
       "Name         AAR2  ...  ZNRD1   ZNRF2    ZPR1  ZRANB2    ZW10  ZWILCH   ZWINT  \\\n",
       "Patient_ID         ...                                                          \n",
       "01CO001       NaN  ...    NaN     NaN     NaN     NaN     NaN     NaN     NaN   \n",
       "01CO005     0.341  ...    NaN  0.0384  0.0221  0.2500  0.0869  0.0331     NaN   \n",
       "01CO006     0.248  ...  0.324     NaN -0.3940  0.0846  0.0010 -0.3450     NaN   \n",
       "01CO008    -0.348  ...    NaN -0.3840 -0.1680  0.3570 -0.3250  0.3490     NaN   \n",
       "01CO013     0.270  ...    NaN     NaN -0.0656 -0.2950  0.4630 -0.4480  0.0493   \n",
       "\n",
       "Name          ZYX   ZZEF1        Mutation  \n",
       "Patient_ID                                 \n",
       "01CO001       NaN     NaN  Wildtype_Tumor  \n",
       "01CO005    -0.733 -0.2650        Deletion  \n",
       "01CO006    -0.658  0.0052  Wildtype_Tumor  \n",
       "01CO008    -0.821  0.2000  Wildtype_Tumor  \n",
       "01CO013    -0.904 -0.1580  Wildtype_Tumor  \n",
       "\n",
       "[5 rows x 8067 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get df with PTEN deletions and wt\n",
    "cancer = col\n",
    "gene = 'PTEN'\n",
    "prot = cancer.get_proteomics()\n",
    "#prot= u.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "prot = prot.drop(columns = 'PTEN') # cis effect\n",
    "prot_list = list(prot.columns)\n",
    "\n",
    "del_wt = all_prot_format_df(cancer, prot_list)\n",
    "del_wt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total samples with PTEN deletion: 25\n"
     ]
    }
   ],
   "source": [
    "# get ids of samples with PTEN del\n",
    "df = del_wt.loc[del_wt['Mutation'] == 'Deletion']\n",
    "del_ids = list(df.index)\n",
    "print('total samples with PTEN deletion:', len(del_ids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total samples: \n",
      "\n",
      "Stage III    48\n",
      "Stage II     42\n",
      "Stage I      12\n",
      "Stage IV      8\n",
      "Name: Stage, dtype: int64 \n",
      "\n",
      "samples with PTEN loss:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Stage II     12\n",
       "Stage III     8\n",
       "Stage I       4\n",
       "Stage IV      1\n",
       "Name: Stage, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check how many samples are in stage 1 and have PTEN loss\n",
    "clinical = cancer.get_clinical()\n",
    "#print(clinical.columns)\n",
    "print('total samples:', '\\n')\n",
    "print(clinical.Stage.value_counts(), '\\n')\n",
    "\n",
    "only_del = clinical[clinical.index.isin(del_ids)]\n",
    "print('samples with PTEN loss:')\n",
    "only_del.Stage.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "54"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get a list of all samples with initial stages\n",
    "id_list = []\n",
    "\n",
    "stage_list = ['Stage I', 'Stage II'] # colon\n",
    "#stage_list = ['I', 'II', 'IA', 'IB', 'IIA', 'IIB'] # LSCC\n",
    "\n",
    "for stage in stage_list:\n",
    "    stage_df = clinical.loc[clinical['Stage'] == stage]\n",
    "    s1 = list(stage_df.index)\n",
    "    id_list += s1\n",
    "len(id_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    36\n",
       "Deletion          16\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Keep samples in the initial stages\n",
    "initial = del_wt[del_wt.index.isin(id_list)]\n",
    "initial.Mutation.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NAXE</td>\n",
       "      <td>0.090697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FAM45A</td>\n",
       "      <td>0.166665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FHL2</td>\n",
       "      <td>0.321489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NDUFAF3</td>\n",
       "      <td>0.321489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LTBP2</td>\n",
       "      <td>0.362640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6784</th>\n",
       "      <td>FAU</td>\n",
       "      <td>0.999204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6785</th>\n",
       "      <td>TRIM22</td>\n",
       "      <td>0.999204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6786</th>\n",
       "      <td>RRM2</td>\n",
       "      <td>0.999204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6787</th>\n",
       "      <td>PGLYRP2</td>\n",
       "      <td>0.999378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6788</th>\n",
       "      <td>TEX10</td>\n",
       "      <td>0.999901</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6789 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Comparison   P_Value\n",
       "0          NAXE  0.090697\n",
       "1        FAM45A  0.166665\n",
       "2          FHL2  0.321489\n",
       "3       NDUFAF3  0.321489\n",
       "4         LTBP2  0.362640\n",
       "...         ...       ...\n",
       "6784        FAU  0.999204\n",
       "6785     TRIM22  0.999204\n",
       "6786       RRM2  0.999204\n",
       "6787    PGLYRP2  0.999378\n",
       "6788      TEX10  0.999901\n",
       "\n",
       "[6789 rows x 2 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_cols = list(initial.columns[:-1])\n",
    "\n",
    "# Get all pvals\n",
    "pval = u.wrap_ttest(initial, 'Mutation', prot_cols, return_all = True, correction_method = 'fdr_bh', mincount = 5)\n",
    "pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>NAXE</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01CO005</th>\n",
       "      <td>0.1470</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO008</th>\n",
       "      <td>-0.2520</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO013</th>\n",
       "      <td>0.0459</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO015</th>\n",
       "      <td>0.0692</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01CO022</th>\n",
       "      <td>-0.4700</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO003</th>\n",
       "      <td>-0.1480</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO005</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO011</th>\n",
       "      <td>0.4020</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO015</th>\n",
       "      <td>-0.3390</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO020</th>\n",
       "      <td>-0.3630</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO028</th>\n",
       "      <td>-0.2800</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO029</th>\n",
       "      <td>-0.0225</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO035</th>\n",
       "      <td>-0.0985</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO037</th>\n",
       "      <td>-0.1620</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO039</th>\n",
       "      <td>-0.3300</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO041</th>\n",
       "      <td>-0.5950</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO044</th>\n",
       "      <td>-0.1480</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO047</th>\n",
       "      <td>-0.0318</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO049</th>\n",
       "      <td>-0.0728</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO053</th>\n",
       "      <td>-0.2090</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05CO054</th>\n",
       "      <td>-0.1080</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>09CO006</th>\n",
       "      <td>-0.1260</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>09CO011</th>\n",
       "      <td>-0.1450</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>09CO014</th>\n",
       "      <td>0.5600</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>09CO019</th>\n",
       "      <td>0.2250</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>09CO022</th>\n",
       "      <td>-0.0262</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO010</th>\n",
       "      <td>-0.2340</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO018</th>\n",
       "      <td>-0.0504</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO027</th>\n",
       "      <td>-0.3680</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO030</th>\n",
       "      <td>0.0422</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO031</th>\n",
       "      <td>0.5180</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO039</th>\n",
       "      <td>-0.1860</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO042</th>\n",
       "      <td>-0.0186</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO043</th>\n",
       "      <td>-0.1340</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO047</th>\n",
       "      <td>-0.1860</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO048</th>\n",
       "      <td>-0.4470</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO051</th>\n",
       "      <td>-0.1290</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO053</th>\n",
       "      <td>0.2120</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO054</th>\n",
       "      <td>-0.1740</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO058</th>\n",
       "      <td>0.1300</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO059</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO060</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO061</th>\n",
       "      <td>-0.3050</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11CO077</th>\n",
       "      <td>-0.0149</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14CO002</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14CO005</th>\n",
       "      <td>0.2120</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15CO002</th>\n",
       "      <td>-0.2750</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16CO003</th>\n",
       "      <td>0.2210</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16CO012</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20CO007</th>\n",
       "      <td>0.2440</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21CO006</th>\n",
       "      <td>-0.1070</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22CO004</th>\n",
       "      <td>-0.0905</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name          NAXE        Mutation\n",
       "Patient_ID                        \n",
       "01CO005     0.1470        Deletion\n",
       "01CO008    -0.2520  Wildtype_Tumor\n",
       "01CO013     0.0459  Wildtype_Tumor\n",
       "01CO015     0.0692        Deletion\n",
       "01CO022    -0.4700  Wildtype_Tumor\n",
       "05CO003    -0.1480  Wildtype_Tumor\n",
       "05CO005        NaN  Wildtype_Tumor\n",
       "05CO011     0.4020        Deletion\n",
       "05CO015    -0.3390  Wildtype_Tumor\n",
       "05CO020    -0.3630  Wildtype_Tumor\n",
       "05CO028    -0.2800  Wildtype_Tumor\n",
       "05CO029    -0.0225  Wildtype_Tumor\n",
       "05CO035    -0.0985  Wildtype_Tumor\n",
       "05CO037    -0.1620  Wildtype_Tumor\n",
       "05CO039    -0.3300  Wildtype_Tumor\n",
       "05CO041    -0.5950  Wildtype_Tumor\n",
       "05CO044    -0.1480  Wildtype_Tumor\n",
       "05CO047    -0.0318        Deletion\n",
       "05CO049    -0.0728  Wildtype_Tumor\n",
       "05CO053    -0.2090        Deletion\n",
       "05CO054    -0.1080  Wildtype_Tumor\n",
       "09CO006    -0.1260  Wildtype_Tumor\n",
       "09CO011    -0.1450        Deletion\n",
       "09CO014     0.5600        Deletion\n",
       "09CO019     0.2250        Deletion\n",
       "09CO022    -0.0262  Wildtype_Tumor\n",
       "11CO010    -0.2340  Wildtype_Tumor\n",
       "11CO018    -0.0504  Wildtype_Tumor\n",
       "11CO027    -0.3680  Wildtype_Tumor\n",
       "11CO030     0.0422  Wildtype_Tumor\n",
       "11CO031     0.5180        Deletion\n",
       "11CO039    -0.1860  Wildtype_Tumor\n",
       "11CO042    -0.0186        Deletion\n",
       "11CO043    -0.1340        Deletion\n",
       "11CO047    -0.1860  Wildtype_Tumor\n",
       "11CO048    -0.4470  Wildtype_Tumor\n",
       "11CO051    -0.1290  Wildtype_Tumor\n",
       "11CO053     0.2120  Wildtype_Tumor\n",
       "11CO054    -0.1740  Wildtype_Tumor\n",
       "11CO058     0.1300        Deletion\n",
       "11CO059        NaN  Wildtype_Tumor\n",
       "11CO060        NaN        Deletion\n",
       "11CO061    -0.3050  Wildtype_Tumor\n",
       "11CO077    -0.0149        Deletion\n",
       "14CO002        NaN  Wildtype_Tumor\n",
       "14CO005     0.2120        Deletion\n",
       "15CO002    -0.2750  Wildtype_Tumor\n",
       "16CO003     0.2210  Wildtype_Tumor\n",
       "16CO012        NaN  Wildtype_Tumor\n",
       "20CO007     0.2440        Deletion\n",
       "21CO006    -0.1070  Wildtype_Tumor\n",
       "22CO004    -0.0905  Wildtype_Tumor"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "initial[['NAXE', 'Mutation']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
