{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load cancer data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "en = cptac.Endometrial()\n",
    "br = cptac.Brca()\n",
    "cl = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PIK3CA'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter out hotspot mutations\n",
    "Hotspots are:\n",
    "E542K\n",
    "E545K\n",
    "H1047R\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the mutation type, and ptorteomics for PIK3CA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 48 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 383)\n"
     ]
    }
   ],
   "source": [
    "mut = en.get_genotype_all_vars(gene, mutation_hotspot=['E542K', 'E545K', 'H1047R'])\n",
    "prot = en.get_proteomics(tissue_type=\"tumor\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join mutation type and proteomics together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "joined = mut.join(prot)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Select samples containing hotspot mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "en_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "                    joined.Location.str.contains('E545K') |\n",
    "                    joined.Location.str.contains('H1047R')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join the proteomics for the wildtype mutations to the hotspot mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "\n",
    "endo_hotspot = en_hotspot_df.append(wildtype)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Drop unncecesary columns\n",
    "The resulting dataframe is what we will be working with from this point forward. It contains the mutations type and the proteomics for each gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.863</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.665</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.33900</td>\n",
       "      <td>0.412</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.4330</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.1230</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.320</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.040</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.04790</td>\n",
       "      <td>0.419</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.0740</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.3200</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00362</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.924</td>\n",
       "      <td>-0.445</td>\n",
       "      <td>1.570</td>\n",
       "      <td>-0.678</td>\n",
       "      <td>0.1730</td>\n",
       "      <td>0.436</td>\n",
       "      <td>0.2270</td>\n",
       "      <td>-0.00248</td>\n",
       "      <td>0.479</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.04960</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1080</td>\n",
       "      <td>-0.153</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.109</td>\n",
       "      <td>-0.3360</td>\n",
       "      <td>-0.822</td>\n",
       "      <td>-0.0338</td>\n",
       "      <td>0.1210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00601</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.454</td>\n",
       "      <td>-0.242</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2580</td>\n",
       "      <td>0.219</td>\n",
       "      <td>-0.2490</td>\n",
       "      <td>-1.33000</td>\n",
       "      <td>0.204</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.15800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6670</td>\n",
       "      <td>1.300</td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.130</td>\n",
       "      <td>-0.0659</td>\n",
       "      <td>-0.923</td>\n",
       "      <td>-0.1630</td>\n",
       "      <td>0.1120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00605</th>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>-0.240</td>\n",
       "      <td>0.594</td>\n",
       "      <td>3.400</td>\n",
       "      <td>0.154</td>\n",
       "      <td>0.0932</td>\n",
       "      <td>0.283</td>\n",
       "      <td>-0.0789</td>\n",
       "      <td>-0.61100</td>\n",
       "      <td>0.370</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.05110</td>\n",
       "      <td>-0.0136</td>\n",
       "      <td>0.2140</td>\n",
       "      <td>0.683</td>\n",
       "      <td>0.433</td>\n",
       "      <td>-0.208</td>\n",
       "      <td>-0.3470</td>\n",
       "      <td>-0.911</td>\n",
       "      <td>0.0692</td>\n",
       "      <td>-0.2320</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                 Mutation   A1BG    A2M  A2ML1  A4GALT    AAAS   AACS  \\\n",
       "Patient_ID                                                                  \n",
       "C3L-00006   Missense_Mutation -1.180 -0.863 -0.802   0.222  0.2560  0.665   \n",
       "C3L-00032   Missense_Mutation -0.528 -1.320  0.435     NaN -0.2400  1.040   \n",
       "C3L-00362   Missense_Mutation -0.924 -0.445  1.570  -0.678  0.1730  0.436   \n",
       "C3L-00601   Missense_Mutation -0.454 -0.242    NaN     NaN  0.2580  0.219   \n",
       "C3L-00605   Missense_Mutation -0.240  0.594  3.400   0.154  0.0932  0.283   \n",
       "\n",
       "Name         AADAT    AAED1  AAGAB  ...   ZSWIM8  ZSWIM9    ZW10  ZWILCH  \\\n",
       "Patient_ID                          ...                                    \n",
       "C3L-00006   1.2800 -0.33900  0.412  ... -0.08770     NaN  0.0229   0.109   \n",
       "C3L-00032  -0.0213 -0.04790  0.419  ...  0.00112 -0.1450  0.0105  -0.116   \n",
       "C3L-00362   0.2270 -0.00248  0.479  ... -0.04960     NaN  0.1080  -0.153   \n",
       "C3L-00601  -0.2490 -1.33000  0.204  ... -0.15800     NaN  0.6670   1.300   \n",
       "C3L-00605  -0.0789 -0.61100  0.370  ... -0.05110 -0.0136  0.2140   0.683   \n",
       "\n",
       "Name        ZWINT   ZXDC  ZYG11B    ZYX   ZZEF1    ZZZ3  \n",
       "Patient_ID                                               \n",
       "C3L-00006     NaN -0.332 -0.4330 -1.020 -0.1230 -0.0859  \n",
       "C3L-00032     NaN  0.151 -0.0740 -0.540  0.3200 -0.4190  \n",
       "C3L-00362     NaN  0.109 -0.3360 -0.822 -0.0338  0.1210  \n",
       "C3L-00601   0.441  0.130 -0.0659 -0.923 -0.1630  0.1120  \n",
       "C3L-00605   0.433 -0.208 -0.3470 -0.911  0.0692 -0.2320  \n",
       "\n",
       "[5 rows x 11000 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_and_mutations = endo_hotspot.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "prot_and_mutations.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run T-test and difference of Median for each cancer type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the median proteomic expression among all samples, for each gene.\n",
    "This is done for missense mutations and wildtype separately\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #get all wildtype mutations\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the medain difference between missense and wildtype for each gene.\n",
    "This is done by (for each gene) subtracting the median of the missense mutations from the median of the wildtype.\n",
    "This means that if the difference is positive the proteomics of the wildtype have higher expression than the missense mutation. If the difference is negative then the missense mutation has a higher expression than the wildtype."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.055500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.038000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>-1.105850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>-0.111075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.156100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median\n",
       "A1BG               -0.055500\n",
       "A2M                -0.038000\n",
       "A2ML1              -1.105850\n",
       "A4GALT             -0.111075\n",
       "AAAS               -0.156100"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_d = {}\n",
    "\n",
    "for gene in prot_and_mutations:\n",
    "    if gene == \"Mutation\": continue\n",
    "    dif_endo = wt_med[gene] - missense_med[gene]\n",
    "    endo_d[gene] = dif_endo\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(endo_d, orient='index', columns=['Difference_In_Median'])\n",
    "median_diff.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Do a t-test for every gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:153: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:257: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Comparison</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SLC25A14</th>\n",
       "      <td>0.000645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORO7-PAM16</th>\n",
       "      <td>0.000877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RIPK1</th>\n",
       "      <td>0.001035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LYRM9</th>\n",
       "      <td>0.001181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAS6</th>\n",
       "      <td>0.001208</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              P_Value\n",
       "Comparison           \n",
       "SLC25A14     0.000645\n",
       "CORO7-PAM16  0.000877\n",
       "RIPK1        0.001035\n",
       "LYRM9        0.001181\n",
       "GAS6         0.001208"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = list(prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(prot_and_mutations, 'Mutation', genes, return_all = True)\n",
    "t_test = t_test.set_index('Comparison')\n",
    "t_test.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Join difference in median and t-test p value into the same dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.055500</td>\n",
       "      <td>0.434646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.038000</td>\n",
       "      <td>0.714215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>-1.105850</td>\n",
       "      <td>0.037044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>-0.111075</td>\n",
       "      <td>0.837021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.156100</td>\n",
       "      <td>0.664545</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Difference_In_Median   P_Value\n",
       "A1BG               -0.055500  0.434646\n",
       "A2M                -0.038000  0.714215\n",
       "A2ML1              -1.105850  0.037044\n",
       "A4GALT             -0.111075  0.837021\n",
       "AAAS               -0.156100  0.664545"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 4 join median and p value together\n",
    "endo_df = median_diff.join(t_test)\n",
    "endo_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Brca\n",
    "The Brca data uses different database_IDs, giving us multiple proteomic values for each gene. To deal with this, we will first separate genes that have multiple proteomics values recoreded (multiple database_IDs) from ones that don't."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Dealing with multiple database_IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Separate genes with multiple database_IDs from genes with multiple database_IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>AAMDC</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSCAN31</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Database_ID</th>\n",
       "      <th>NP_570602.2</th>\n",
       "      <th>NP_000005.2</th>\n",
       "      <th>NP_653271.2|NP_001269353.1</th>\n",
       "      <th>NP_056480.1|NP_001166937.1</th>\n",
       "      <th>NP_076417.2|NP_001306769.1|NP_001306768.1</th>\n",
       "      <th>NP_057312.1|NP_001273611.1</th>\n",
       "      <th>NP_714542.1</th>\n",
       "      <th>NP_078942.3|NP_001258815.1</th>\n",
       "      <th>NP_055726.3</th>\n",
       "      <th>NP_078960.1|NP_001303886.1|NP_001303887.1</th>\n",
       "      <th>...</th>\n",
       "      <th>NP_665916.1|NP_001230172.1|NP_116078.4|NP_001166148.1|NP_001229731.1|NP_001012458.1</th>\n",
       "      <th>NP_055852.2|NP_001229416.1|NP_001229417.1</th>\n",
       "      <th>NP_004715.1</th>\n",
       "      <th>NP_060445.3|NP_001274750.1</th>\n",
       "      <th>NP_008988.2|NP_001005413.1</th>\n",
       "      <th>NP_079388.3|NP_001035743.1</th>\n",
       "      <th>NP_078922.1</th>\n",
       "      <th>NP_001010972.1</th>\n",
       "      <th>NP_055928.3</th>\n",
       "      <th>NP_056349.1|NP_001295166.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CPT000814</th>\n",
       "      <td>-0.6712</td>\n",
       "      <td>-0.2075</td>\n",
       "      <td>2.7959</td>\n",
       "      <td>1.3969</td>\n",
       "      <td>-1.0899</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.6708</td>\n",
       "      <td>-0.3484</td>\n",
       "      <td>-0.4756</td>\n",
       "      <td>-0.7299</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.2868</td>\n",
       "      <td>-0.6536</td>\n",
       "      <td>0.3384</td>\n",
       "      <td>2.1169</td>\n",
       "      <td>1.3910</td>\n",
       "      <td>-2.1230</td>\n",
       "      <td>0.9136</td>\n",
       "      <td>-0.8082</td>\n",
       "      <td>-1.4793</td>\n",
       "      <td>0.9136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CPT001846</th>\n",
       "      <td>1.3964</td>\n",
       "      <td>1.3302</td>\n",
       "      <td>-5.0948</td>\n",
       "      <td>0.7674</td>\n",
       "      <td>-1.6845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1022</td>\n",
       "      <td>-0.5814</td>\n",
       "      <td>0.2916</td>\n",
       "      <td>-2.2857</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.7592</td>\n",
       "      <td>0.4711</td>\n",
       "      <td>0.6018</td>\n",
       "      <td>0.2062</td>\n",
       "      <td>-0.2137</td>\n",
       "      <td>-2.1219</td>\n",
       "      <td>0.0860</td>\n",
       "      <td>2.5814</td>\n",
       "      <td>-0.2852</td>\n",
       "      <td>-0.1074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR001</th>\n",
       "      <td>2.0219</td>\n",
       "      <td>1.6269</td>\n",
       "      <td>-3.2943</td>\n",
       "      <td>0.3352</td>\n",
       "      <td>-1.0739</td>\n",
       "      <td>1.2255</td>\n",
       "      <td>0.2754</td>\n",
       "      <td>-1.1187</td>\n",
       "      <td>-0.0534</td>\n",
       "      <td>-0.2519</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2306</td>\n",
       "      <td>-0.3010</td>\n",
       "      <td>0.3395</td>\n",
       "      <td>-0.5316</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4996</td>\n",
       "      <td>0.7622</td>\n",
       "      <td>-1.5607</td>\n",
       "      <td>0.0256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR008</th>\n",
       "      <td>-0.5290</td>\n",
       "      <td>0.3267</td>\n",
       "      <td>1.4342</td>\n",
       "      <td>0.4938</td>\n",
       "      <td>-2.8676</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0691</td>\n",
       "      <td>-0.3643</td>\n",
       "      <td>-1.8173</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.1789</td>\n",
       "      <td>0.2695</td>\n",
       "      <td>0.1506</td>\n",
       "      <td>1.0498</td>\n",
       "      <td>0.7546</td>\n",
       "      <td>1.7889</td>\n",
       "      <td>-0.2499</td>\n",
       "      <td>-0.2590</td>\n",
       "      <td>-0.1263</td>\n",
       "      <td>0.3725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR009</th>\n",
       "      <td>1.2556</td>\n",
       "      <td>3.4489</td>\n",
       "      <td>2.8043</td>\n",
       "      <td>-0.2956</td>\n",
       "      <td>-1.7261</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.0471</td>\n",
       "      <td>-0.3547</td>\n",
       "      <td>-0.8298</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.3990</td>\n",
       "      <td>-0.2596</td>\n",
       "      <td>0.1898</td>\n",
       "      <td>-0.5010</td>\n",
       "      <td>-0.4189</td>\n",
       "      <td>0.3080</td>\n",
       "      <td>0.5057</td>\n",
       "      <td>0.2181</td>\n",
       "      <td>-0.2288</td>\n",
       "      <td>-0.2750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X21BR001</th>\n",
       "      <td>-0.6610</td>\n",
       "      <td>-0.6402</td>\n",
       "      <td>-4.8578</td>\n",
       "      <td>1.2319</td>\n",
       "      <td>-1.6491</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3074</td>\n",
       "      <td>-0.3074</td>\n",
       "      <td>-0.0266</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2528</td>\n",
       "      <td>0.5090</td>\n",
       "      <td>0.0306</td>\n",
       "      <td>0.4908</td>\n",
       "      <td>-0.5570</td>\n",
       "      <td>2.3864</td>\n",
       "      <td>0.3764</td>\n",
       "      <td>-0.6974</td>\n",
       "      <td>1.3541</td>\n",
       "      <td>1.1123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X21BR002</th>\n",
       "      <td>-1.3735</td>\n",
       "      <td>0.4227</td>\n",
       "      <td>-4.9553</td>\n",
       "      <td>0.6327</td>\n",
       "      <td>-3.1434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3071</td>\n",
       "      <td>0.7562</td>\n",
       "      <td>-1.6912</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.3351</td>\n",
       "      <td>0.1548</td>\n",
       "      <td>1.0792</td>\n",
       "      <td>-0.6619</td>\n",
       "      <td>-1.4444</td>\n",
       "      <td>-0.3704</td>\n",
       "      <td>0.4909</td>\n",
       "      <td>0.3938</td>\n",
       "      <td>0.2992</td>\n",
       "      <td>-0.3494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X21BR010</th>\n",
       "      <td>1.1583</td>\n",
       "      <td>0.3329</td>\n",
       "      <td>-5.7358</td>\n",
       "      <td>-0.1658</td>\n",
       "      <td>-2.0413</td>\n",
       "      <td>-1.2433</td>\n",
       "      <td>0.9090</td>\n",
       "      <td>-0.2410</td>\n",
       "      <td>0.6717</td>\n",
       "      <td>-0.1651</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.7054</td>\n",
       "      <td>0.2752</td>\n",
       "      <td>0.8850</td>\n",
       "      <td>-2.6704</td>\n",
       "      <td>-0.9444</td>\n",
       "      <td>-1.9717</td>\n",
       "      <td>0.0650</td>\n",
       "      <td>0.6300</td>\n",
       "      <td>-0.0686</td>\n",
       "      <td>0.1798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X22BR005</th>\n",
       "      <td>0.4948</td>\n",
       "      <td>-1.0986</td>\n",
       "      <td>-8.8314</td>\n",
       "      <td>0.2826</td>\n",
       "      <td>-1.0123</td>\n",
       "      <td>-2.5732</td>\n",
       "      <td>5.7567</td>\n",
       "      <td>1.7644</td>\n",
       "      <td>0.5415</td>\n",
       "      <td>0.1531</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3936</td>\n",
       "      <td>-0.0340</td>\n",
       "      <td>-0.9367</td>\n",
       "      <td>-0.1922</td>\n",
       "      <td>1.2572</td>\n",
       "      <td>1.3220</td>\n",
       "      <td>-1.0698</td>\n",
       "      <td>0.4012</td>\n",
       "      <td>-0.3792</td>\n",
       "      <td>1.2752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X22BR006</th>\n",
       "      <td>0.5049</td>\n",
       "      <td>-0.6582</td>\n",
       "      <td>-7.4699</td>\n",
       "      <td>0.6570</td>\n",
       "      <td>-0.7239</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6420</td>\n",
       "      <td>2.2376</td>\n",
       "      <td>-0.7597</td>\n",
       "      <td>0.1589</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5526</td>\n",
       "      <td>0.3468</td>\n",
       "      <td>1.0655</td>\n",
       "      <td>-1.2607</td>\n",
       "      <td>-2.2866</td>\n",
       "      <td>1.0775</td>\n",
       "      <td>-0.1334</td>\n",
       "      <td>2.5269</td>\n",
       "      <td>1.4294</td>\n",
       "      <td>0.4542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122 rows × 9452 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name               A1BG         A2M                      A2ML1  \\\n",
       "Database_ID NP_570602.2 NP_000005.2 NP_653271.2|NP_001269353.1   \n",
       "CPT000814       -0.6712     -0.2075                     2.7959   \n",
       "CPT001846        1.3964      1.3302                    -5.0948   \n",
       "X01BR001         2.0219      1.6269                    -3.2943   \n",
       "X01BR008        -0.5290      0.3267                     1.4342   \n",
       "X01BR009         1.2556      3.4489                     2.8043   \n",
       "...                 ...         ...                        ...   \n",
       "X21BR001        -0.6610     -0.6402                    -4.8578   \n",
       "X21BR002        -1.3735      0.4227                    -4.9553   \n",
       "X21BR010         1.1583      0.3329                    -5.7358   \n",
       "X22BR005         0.4948     -1.0986                    -8.8314   \n",
       "X22BR006         0.5049     -0.6582                    -7.4699   \n",
       "\n",
       "Name                              AAAS  \\\n",
       "Database_ID NP_056480.1|NP_001166937.1   \n",
       "CPT000814                       1.3969   \n",
       "CPT001846                       0.7674   \n",
       "X01BR001                        0.3352   \n",
       "X01BR008                        0.4938   \n",
       "X01BR009                       -0.2956   \n",
       "...                                ...   \n",
       "X21BR001                        1.2319   \n",
       "X21BR002                        0.6327   \n",
       "X21BR010                       -0.1658   \n",
       "X22BR005                        0.2826   \n",
       "X22BR006                        0.6570   \n",
       "\n",
       "Name                                             AACS  \\\n",
       "Database_ID NP_076417.2|NP_001306769.1|NP_001306768.1   \n",
       "CPT000814                                     -1.0899   \n",
       "CPT001846                                     -1.6845   \n",
       "X01BR001                                      -1.0739   \n",
       "X01BR008                                      -2.8676   \n",
       "X01BR009                                      -1.7261   \n",
       "...                                               ...   \n",
       "X21BR001                                      -1.6491   \n",
       "X21BR002                                      -3.1434   \n",
       "X21BR010                                      -2.0413   \n",
       "X22BR005                                      -1.0123   \n",
       "X22BR006                                      -0.7239   \n",
       "\n",
       "Name                             AADAT       AAED1                      AAGAB  \\\n",
       "Database_ID NP_057312.1|NP_001273611.1 NP_714542.1 NP_078942.3|NP_001258815.1   \n",
       "CPT000814                          NaN      1.6708                    -0.3484   \n",
       "CPT001846                          NaN      2.1022                    -0.5814   \n",
       "X01BR001                        1.2255      0.2754                    -1.1187   \n",
       "X01BR008                           NaN         NaN                    -1.0691   \n",
       "X01BR009                           NaN         NaN                    -2.0471   \n",
       "...                                ...         ...                        ...   \n",
       "X21BR001                           NaN         NaN                    -0.3074   \n",
       "X21BR002                           NaN         NaN                     0.3071   \n",
       "X21BR010                       -1.2433      0.9090                    -0.2410   \n",
       "X22BR005                       -2.5732      5.7567                     1.7644   \n",
       "X22BR006                           NaN      0.6420                     2.2376   \n",
       "\n",
       "Name               AAK1                                     AAMDC  ...  \\\n",
       "Database_ID NP_055726.3 NP_078960.1|NP_001303886.1|NP_001303887.1  ...   \n",
       "CPT000814       -0.4756                                   -0.7299  ...   \n",
       "CPT001846        0.2916                                   -2.2857  ...   \n",
       "X01BR001        -0.0534                                   -0.2519  ...   \n",
       "X01BR008        -0.3643                                   -1.8173  ...   \n",
       "X01BR009        -0.3547                                   -0.8298  ...   \n",
       "...                 ...                                       ...  ...   \n",
       "X21BR001        -0.3074                                   -0.0266  ...   \n",
       "X21BR002         0.7562                                   -1.6912  ...   \n",
       "X21BR010         0.6717                                   -0.1651  ...   \n",
       "X22BR005         0.5415                                    0.1531  ...   \n",
       "X22BR006        -0.7597                                    0.1589  ...   \n",
       "\n",
       "Name                                                                                    ZSCAN31  \\\n",
       "Database_ID NP_665916.1|NP_001230172.1|NP_116078.4|NP_001166148.1|NP_001229731.1|NP_001012458.1   \n",
       "CPT000814                                              -5.2868                                    \n",
       "CPT001846                                              -0.7592                                    \n",
       "X01BR001                                                   NaN                                    \n",
       "X01BR008                                               -2.1789                                    \n",
       "X01BR009                                               -2.3990                                    \n",
       "...                                                        ...                                    \n",
       "X21BR001                                               -0.2528                                    \n",
       "X21BR002                                               -3.3351                                    \n",
       "X21BR010                                               -0.7054                                    \n",
       "X22BR005                                               -0.3936                                    \n",
       "X22BR006                                                0.5526                                    \n",
       "\n",
       "Name                                           ZSWIM8        ZW10  \\\n",
       "Database_ID NP_055852.2|NP_001229416.1|NP_001229417.1 NP_004715.1   \n",
       "CPT000814                                     -0.6536      0.3384   \n",
       "CPT001846                                      0.4711      0.6018   \n",
       "X01BR001                                       0.2306     -0.3010   \n",
       "X01BR008                                       0.2695      0.1506   \n",
       "X01BR009                                      -0.2596      0.1898   \n",
       "...                                               ...         ...   \n",
       "X21BR001                                       0.5090      0.0306   \n",
       "X21BR002                                       0.1548      1.0792   \n",
       "X21BR010                                       0.2752      0.8850   \n",
       "X22BR005                                      -0.0340     -0.9367   \n",
       "X22BR006                                       0.3468      1.0655   \n",
       "\n",
       "Name                            ZWILCH                      ZWINT  \\\n",
       "Database_ID NP_060445.3|NP_001274750.1 NP_008988.2|NP_001005413.1   \n",
       "CPT000814                       2.1169                     1.3910   \n",
       "CPT001846                       0.2062                    -0.2137   \n",
       "X01BR001                        0.3395                    -0.5316   \n",
       "X01BR008                        1.0498                     0.7546   \n",
       "X01BR009                       -0.5010                    -0.4189   \n",
       "...                                ...                        ...   \n",
       "X21BR001                        0.4908                    -0.5570   \n",
       "X21BR002                       -0.6619                    -1.4444   \n",
       "X21BR010                       -2.6704                    -0.9444   \n",
       "X22BR005                       -0.1922                     1.2572   \n",
       "X22BR006                       -1.2607                    -2.2866   \n",
       "\n",
       "Name                              ZXDC      ZYG11B            ZYX       ZZEF1  \\\n",
       "Database_ID NP_079388.3|NP_001035743.1 NP_078922.1 NP_001010972.1 NP_055928.3   \n",
       "CPT000814                      -2.1230      0.9136        -0.8082     -1.4793   \n",
       "CPT001846                      -2.1219      0.0860         2.5814     -0.2852   \n",
       "X01BR001                           NaN      0.4996         0.7622     -1.5607   \n",
       "X01BR008                        1.7889     -0.2499        -0.2590     -0.1263   \n",
       "X01BR009                        0.3080      0.5057         0.2181     -0.2288   \n",
       "...                                ...         ...            ...         ...   \n",
       "X21BR001                        2.3864      0.3764        -0.6974      1.3541   \n",
       "X21BR002                       -0.3704      0.4909         0.3938      0.2992   \n",
       "X21BR010                       -1.9717      0.0650         0.6300     -0.0686   \n",
       "X22BR005                        1.3220     -1.0698         0.4012     -0.3792   \n",
       "X22BR006                        1.0775     -0.1334         2.5269      1.4294   \n",
       "\n",
       "Name                              ZZZ3  \n",
       "Database_ID NP_056349.1|NP_001295166.1  \n",
       "CPT000814                       0.9136  \n",
       "CPT001846                      -0.1074  \n",
       "X01BR001                        0.0256  \n",
       "X01BR008                        0.3725  \n",
       "X01BR009                       -0.2750  \n",
       "...                                ...  \n",
       "X21BR001                        1.1123  \n",
       "X21BR002                       -0.3494  \n",
       "X21BR010                        0.1798  \n",
       "X22BR005                        1.2752  \n",
       "X22BR006                        0.4542  \n",
       "\n",
       "[122 rows x 9452 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "no_duplicates = proteomics[proteomics.columns[~fil]]\n",
    "no_duplicates"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Dealing with genes that don't have multiple database_IDs\n",
    "These will be processed the same way the data for endometrial was."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 83 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 383)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:153: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:257: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG_NP_570602.2</th>\n",
       "      <td>-0.63510</td>\n",
       "      <td>0.830859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M_NP_000005.2</th>\n",
       "      <td>-0.05110</td>\n",
       "      <td>0.999049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1_NP_653271.2|NP_001269353.1</th>\n",
       "      <td>1.26120</td>\n",
       "      <td>0.165194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS_NP_056480.1|NP_001166937.1</th>\n",
       "      <td>-0.13550</td>\n",
       "      <td>0.201707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS_NP_076417.2|NP_001306769.1|NP_001306768.1</th>\n",
       "      <td>0.01775</td>\n",
       "      <td>0.652935</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Difference_In_Median   P_Value\n",
       "A1BG_NP_570602.2                                            -0.63510  0.830859\n",
       "A2M_NP_000005.2                                             -0.05110  0.999049\n",
       "A2ML1_NP_653271.2|NP_001269353.1                             1.26120  0.165194\n",
       "AAAS_NP_056480.1|NP_001166937.1                             -0.13550  0.201707\n",
       "AACS_NP_076417.2|NP_001306769.1|NP_001306768.1               0.01775  0.652935"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#FOR NO DUPLICATES\n",
    "#Flatten the multiindex (since there are no duplicate gene names, this won't be a problem)\n",
    "no_duplicates = br.reduce_multiindex(no_duplicates, flatten=True)\n",
    "\n",
    "#get mutation_tyoe\n",
    "mut_type = br.get_genotype_all_vars('PIK3CA')\n",
    "\n",
    "#join proteomics and mutation type\n",
    "joined = mut_type.join(no_duplicates)\n",
    "\n",
    "#select samples containing hotspot mutations\n",
    "br_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "                    joined.Location.str.contains('E545K') |\n",
    "                    joined.Location.str.contains('H1047R')]\n",
    "\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "brca_hotspot = br_hotspot_df.append(wildtype)\n",
    "\n",
    "#drop unncessary columns\n",
    "prot_and_mutations = brca_hotspot.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "# Get the difference in medians\n",
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "no_dup_d = {}\n",
    "\n",
    "for prot in no_duplicates:\n",
    "    dif_brca = wt_med[prot] - missense_med[prot]\n",
    "    no_dup_d[prot] = dif_brca\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(no_dup_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#Do a t test for every gene and report the p-value\n",
    "genes = list(prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(prot_and_mutations, 'Mutation', genes, return_all = True)\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "#join median and p value together\n",
    "no_dup_df = median_diff.join(t_test)\n",
    "no_dup_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Dealing with genes that do have multiple database_IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get t-test results and median difference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 83 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 383)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABCB7_NP_001258628.1|NP_001258626.1</th>\n",
       "      <td>0.2932</td>\n",
       "      <td>0.238399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCB7_NP_004290.2|NP_001258625.1|NP_001258627.1</th>\n",
       "      <td>0.2573</td>\n",
       "      <td>0.537629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABLIM3_NP_001287947.1|NP_001287944.1</th>\n",
       "      <td>-0.1830</td>\n",
       "      <td>0.781017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABLIM3_NP_001287956.1</th>\n",
       "      <td>0.4862</td>\n",
       "      <td>0.166418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACIN1_NP_001158286.1</th>\n",
       "      <td>-0.1520</td>\n",
       "      <td>0.783389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Difference_In_Median  \\\n",
       "ABCB7_NP_001258628.1|NP_001258626.1                            0.2932   \n",
       "ABCB7_NP_004290.2|NP_001258625.1|NP_001258627.1                0.2573   \n",
       "ABLIM3_NP_001287947.1|NP_001287944.1                          -0.1830   \n",
       "ABLIM3_NP_001287956.1                                          0.4862   \n",
       "ACIN1_NP_001158286.1                                          -0.1520   \n",
       "\n",
       "                                                  P_Value  \n",
       "ABCB7_NP_001258628.1|NP_001258626.1              0.238399  \n",
       "ABCB7_NP_004290.2|NP_001258625.1|NP_001258627.1  0.537629  \n",
       "ABLIM3_NP_001287947.1|NP_001287944.1             0.781017  \n",
       "ABLIM3_NP_001287956.1                            0.166418  \n",
       "ACIN1_NP_001158286.1                             0.783389  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#DEALING WITH DUPLICATES\n",
    "#flatten the multiindex\n",
    "duplicates = br.reduce_multiindex(duplicates, flatten=True)\n",
    "\n",
    "# We need to figure which of databaseID we want to use, for each protein\n",
    "proteomics = br.get_proteomics(tissue_type=\"tumor\")\n",
    "fil = proteomics.columns.get_level_values('Name').duplicated(False)\n",
    "duplicates = proteomics[proteomics.columns[fil]]\n",
    "duplicate_gene_names = list(set(duplicates.columns.get_level_values('Name'))) #get a list of genes names that are duplicated\n",
    "duplicates = br.reduce_multiindex(duplicates, flatten=True)\n",
    "duplicates\n",
    "\n",
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "\n",
    "mut_type = br.get_genotype_all_vars(\"PIK3CA\")\n",
    "\n",
    "\n",
    "joined = mut_type.join(duplicates)\n",
    "\n",
    "#select samples containing hotspot mutations\n",
    "br_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "                    joined.Location.str.contains('E545K') |\n",
    "                    joined.Location.str.contains('H1047R')]\n",
    "\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "brca_hotspot = br_hotspot_df.append(wildtype)\n",
    "\n",
    "#drop unncessary columns\n",
    "prot_and_mutations = brca_hotspot.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "\n",
    "# get the difference in medians\n",
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"Missense_Mutation\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "duplicates_d = {}\n",
    "\n",
    "for prot in duplicates:\n",
    "    dif_brca = wt_med[prot] - missense_med[prot]\n",
    "    duplicates_d[prot] = dif_brca\n",
    "    \n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(duplicates_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "\n",
    "#step 3 do a t test for every gene and report the p-value\n",
    "cols = list(prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(prot_and_mutations, 'Mutation', cols, return_all = True)\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "# Step 4 join median and p value together\n",
    "dup_df = median_diff.join(t_test)\n",
    "dup_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Select the database_ID that gives a lower p value from the t-test, and append selected database_ID to the no_duplicate dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG_NP_570602.2</th>\n",
       "      <td>-0.63510</td>\n",
       "      <td>0.830859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1_NP_653271.2|NP_001269353.1</th>\n",
       "      <td>1.26120</td>\n",
       "      <td>0.165194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M_NP_000005.2</th>\n",
       "      <td>-0.05110</td>\n",
       "      <td>0.999049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS_NP_056480.1|NP_001166937.1</th>\n",
       "      <td>-0.13550</td>\n",
       "      <td>0.201707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS_NP_076417.2|NP_001306769.1|NP_001306768.1</th>\n",
       "      <td>0.01775</td>\n",
       "      <td>0.652935</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Difference_In_Median   P_Value\n",
       "A1BG_NP_570602.2                                            -0.63510  0.830859\n",
       "A2ML1_NP_653271.2|NP_001269353.1                             1.26120  0.165194\n",
       "A2M_NP_000005.2                                             -0.05110  0.999049\n",
       "AAAS_NP_056480.1|NP_001166937.1                             -0.13550  0.201707\n",
       "AACS_NP_076417.2|NP_001306769.1|NP_001306768.1               0.01775  0.652935"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#for each duplicated gene name select the one with the min p val and append to no_duplicat df\n",
    "for name in duplicate_gene_names:\n",
    "    gene_duplicate = dup_df[dup_df.index.str.contains(name)]\n",
    "    #select the one with the min p value for t test\n",
    "    min_row = gene_duplicate[gene_duplicate.P_Value == gene_duplicate.P_Value.min()]\n",
    "    #append the selected version onto no_duplicate dataframe\n",
    "    no_dup_df=no_dup_df.append(min_row)\n",
    "no_dup_df=no_dup_df.sort_index()\n",
    "no_dup_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Reformat the table by getting rid of databaseIDs so we can join it to Endo and Brca"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>new_index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.63510</td>\n",
       "      <td>0.830859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.26120</td>\n",
       "      <td>0.165194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.05110</td>\n",
       "      <td>0.999049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.13550</td>\n",
       "      <td>0.201707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0.01775</td>\n",
       "      <td>0.652935</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Difference_In_Median   P_Value\n",
       "new_index                                \n",
       "A1BG                   -0.63510  0.830859\n",
       "A2ML1                   1.26120  0.165194\n",
       "A2M                    -0.05110  0.999049\n",
       "AAAS                   -0.13550  0.201707\n",
       "AACS                    0.01775  0.652935"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split = no_dup_df.index.str.split('_', 1).str[0]\n",
    "no_dup_df['new_index']=split\n",
    "brca_df = no_dup_df.set_index(\"new_index\")\n",
    "brca_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PIK3CA gene (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/cptac/dataset.py, line 383)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:153: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "/Users/hannahboekweg/anaconda3/lib/python3.7/site-packages/statsmodels/stats/multitest.py:257: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.61600</td>\n",
       "      <td>0.263957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.03225</td>\n",
       "      <td>0.913020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.26800</td>\n",
       "      <td>0.265891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>0.26315</td>\n",
       "      <td>0.070194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0.00850</td>\n",
       "      <td>0.113686</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Difference_In_Median   P_Value\n",
       "A1BG               0.61600  0.263957\n",
       "A1CF               0.03225  0.913020\n",
       "A2M                0.26800  0.265891\n",
       "AAAS               0.26315  0.070194\n",
       "AACS               0.00850  0.113686"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#  Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "proteomics = cl.get_proteomics(tissue_type=\"tumor\")\n",
    "\n",
    "prot_list = list(proteomics.columns)\n",
    "\n",
    "mut_type = cl.get_genotype_all_vars(\"PIK3CA\")\n",
    "\n",
    "\n",
    "joined = mut_type.join(proteomics)\n",
    "\n",
    "#select samples containing hotspot mutations\n",
    "cl_hotspot_df = joined[joined.Location.str.contains('E542K') | \n",
    "                    joined.Location.str.contains('E545K') |\n",
    "                    joined.Location.str.contains('H1047R')]\n",
    "\n",
    "#join wildtype proteomics to missense mutation proteomics \n",
    "wildtype  = joined.loc[joined.Mutation == \"Wildtype_Tumor\"]\n",
    "colon_hotspot = cl_hotspot_df.append(wildtype)\n",
    "\n",
    "#drop unncessary columns\n",
    "prot_and_mutations = colon_hotspot.drop(columns = [\"Mutation_Status\", \"Location\"])\n",
    "\n",
    "\n",
    "#get the difference in medians\n",
    "missense = prot_and_mutations[prot_and_mutations.Mutation == \"nonsynonymous SNV\"] #get all missense_mutations\n",
    "wt = prot_and_mutations[prot_and_mutations.Mutation == \"Wildtype_Tumor\"] #getn all wildtype\n",
    "missense_med = missense.median() #get the medain of the missense. This will give the median expression among all samples, for each gene\n",
    "wt_med = wt.median() #get the median of the wildtype\n",
    "\n",
    "\n",
    "colon_d = {}\n",
    "\n",
    "for prot in proteomics:\n",
    "    dif_colon = wt_med[prot] - missense_med[prot]\n",
    "    colon_d[prot] = dif_colon\n",
    "\n",
    "median_diff = pd.DataFrame.from_dict(colon_d, orient='index', columns=['Difference_In_Median'])\n",
    "\n",
    "#do a t test for every gene and report the p-value\n",
    "cols = list(prot_and_mutations.columns[1:])\n",
    "t_test = u.wrap_ttest(prot_and_mutations, 'Mutation', cols, return_all = True)\n",
    "t_test = t_test.set_index('Comparison')\n",
    "\n",
    "#join median and p value together\n",
    "colon_df = median_diff.join(t_test)\n",
    "colon_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine Endo, Brca, and Colon into one table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine brca and endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median_Brca</th>\n",
       "      <th>P_Value_Brca</th>\n",
       "      <th>Difference_In_Median_Endo</th>\n",
       "      <th>P_Value_Endo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.63510</td>\n",
       "      <td>0.830859</td>\n",
       "      <td>-0.05550</td>\n",
       "      <td>0.434646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.05110</td>\n",
       "      <td>0.999049</td>\n",
       "      <td>-0.03800</td>\n",
       "      <td>0.714215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.26120</td>\n",
       "      <td>0.165194</td>\n",
       "      <td>-1.10585</td>\n",
       "      <td>0.037044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.13550</td>\n",
       "      <td>0.201707</td>\n",
       "      <td>-0.15610</td>\n",
       "      <td>0.664545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0.01775</td>\n",
       "      <td>0.652935</td>\n",
       "      <td>-0.02150</td>\n",
       "      <td>0.732714</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Difference_In_Median_Brca  P_Value_Brca  Difference_In_Median_Endo  \\\n",
       "A1BG                    -0.63510      0.830859                   -0.05550   \n",
       "A2M                     -0.05110      0.999049                   -0.03800   \n",
       "A2ML1                    1.26120      0.165194                   -1.10585   \n",
       "AAAS                    -0.13550      0.201707                   -0.15610   \n",
       "AACS                     0.01775      0.652935                   -0.02150   \n",
       "\n",
       "       P_Value_Endo  \n",
       "A1BG       0.434646  \n",
       "A2M        0.714215  \n",
       "A2ML1      0.037044  \n",
       "AAAS       0.664545  \n",
       "AACS       0.732714  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_df=brca_df.rename_axis(None)\n",
    "brca_endo = brca_df.join(endo_df, lsuffix='_Brca', rsuffix='_Endo')\n",
    "brca_endo.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median_Brca</th>\n",
       "      <th>P_Value_Brca</th>\n",
       "      <th>Difference_In_Median_Endo</th>\n",
       "      <th>P_Value_Endo</th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.63510</td>\n",
       "      <td>0.830859</td>\n",
       "      <td>-0.05550</td>\n",
       "      <td>0.434646</td>\n",
       "      <td>0.61600</td>\n",
       "      <td>0.263957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.05110</td>\n",
       "      <td>0.999049</td>\n",
       "      <td>-0.03800</td>\n",
       "      <td>0.714215</td>\n",
       "      <td>0.26800</td>\n",
       "      <td>0.265891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.26120</td>\n",
       "      <td>0.165194</td>\n",
       "      <td>-1.10585</td>\n",
       "      <td>0.037044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>-0.13550</td>\n",
       "      <td>0.201707</td>\n",
       "      <td>-0.15610</td>\n",
       "      <td>0.664545</td>\n",
       "      <td>0.26315</td>\n",
       "      <td>0.070194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>0.01775</td>\n",
       "      <td>0.652935</td>\n",
       "      <td>-0.02150</td>\n",
       "      <td>0.732714</td>\n",
       "      <td>0.00850</td>\n",
       "      <td>0.113686</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Difference_In_Median_Brca  P_Value_Brca  Difference_In_Median_Endo  \\\n",
       "A1BG                    -0.63510      0.830859                   -0.05550   \n",
       "A2M                     -0.05110      0.999049                   -0.03800   \n",
       "A2ML1                    1.26120      0.165194                   -1.10585   \n",
       "AAAS                    -0.13550      0.201707                   -0.15610   \n",
       "AACS                     0.01775      0.652935                   -0.02150   \n",
       "\n",
       "       P_Value_Endo  Difference_In_Median   P_Value  \n",
       "A1BG       0.434646               0.61600  0.263957  \n",
       "A2M        0.714215               0.26800  0.265891  \n",
       "A2ML1      0.037044                   NaN       NaN  \n",
       "AAAS       0.664545               0.26315  0.070194  \n",
       "AACS       0.732714               0.00850  0.113686  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_table = brca_endo.join(colon_df, rsuffix='_colon')\n",
    "final_table.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Only rows that contain a significant pvalue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Difference_In_Median_Brca</th>\n",
       "      <th>P_Value_Brca</th>\n",
       "      <th>Difference_In_Median_Endo</th>\n",
       "      <th>P_Value_Endo</th>\n",
       "      <th>Difference_In_Median</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.26120</td>\n",
       "      <td>0.165194</td>\n",
       "      <td>-1.10585</td>\n",
       "      <td>0.037044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADAT</th>\n",
       "      <td>1.33200</td>\n",
       "      <td>0.027477</td>\n",
       "      <td>0.18535</td>\n",
       "      <td>0.690613</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAGAB</th>\n",
       "      <td>0.09455</td>\n",
       "      <td>0.909622</td>\n",
       "      <td>-0.24350</td>\n",
       "      <td>0.139632</td>\n",
       "      <td>-0.1390</td>\n",
       "      <td>0.039189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AASDHPPT</th>\n",
       "      <td>-0.51445</td>\n",
       "      <td>0.003620</td>\n",
       "      <td>0.10590</td>\n",
       "      <td>0.428884</td>\n",
       "      <td>0.0628</td>\n",
       "      <td>0.973515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AATF</th>\n",
       "      <td>0.16905</td>\n",
       "      <td>0.042271</td>\n",
       "      <td>0.18170</td>\n",
       "      <td>0.915482</td>\n",
       "      <td>-0.0250</td>\n",
       "      <td>0.568821</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Difference_In_Median_Brca  P_Value_Brca  Difference_In_Median_Endo  \\\n",
       "A2ML1                       1.26120      0.165194                   -1.10585   \n",
       "AADAT                       1.33200      0.027477                    0.18535   \n",
       "AAGAB                       0.09455      0.909622                   -0.24350   \n",
       "AASDHPPT                   -0.51445      0.003620                    0.10590   \n",
       "AATF                        0.16905      0.042271                    0.18170   \n",
       "\n",
       "          P_Value_Endo  Difference_In_Median   P_Value  \n",
       "A2ML1         0.037044                   NaN       NaN  \n",
       "AADAT         0.690613                   NaN       NaN  \n",
       "AAGAB         0.139632               -0.1390  0.039189  \n",
       "AASDHPPT      0.428884                0.0628  0.973515  \n",
       "AATF          0.915482               -0.0250  0.568821  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "significant = final_table.loc[(final_table['P_Value_Brca'] <= .05) |\n",
    "                (final_table['P_Value_Endo'] <= .05) |\n",
    "                (final_table['P_Value'] <= .05)]\n",
    "significant.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# significant.to_csv(\"~/WhenMutationsDontMatter/PIK3CA/csv_files/proteomics.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
