{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make EGFR Figure 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebooks takes all the trans genes that have positive and negative results and runs a GSEA using Reactome. It then takes a subset of genes from the top hit(Hemostasis) pathway and maps them on a large circle heat map. This heatmap focuses on coagulation and urokinase related genes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:5: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  \"\"\"\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas.util.testing as tm\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "import statsmodels.stats.multitest\n",
    "\n",
    "import gseapy as gp\n",
    "from gseapy.plot import barplot, dotplot\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 1: Find Trans proteins with opposite effects in different cancers "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load df with all of the genes that are FDR significant."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation_Gbm</th>\n",
       "      <th>P_value_Gbm</th>\n",
       "      <th>Correlation_kidney</th>\n",
       "      <th>P_value_kidney</th>\n",
       "      <th>Correlation_Ovar</th>\n",
       "      <th>P_value_Ovar</th>\n",
       "      <th>Correlation_Brca</th>\n",
       "      <th>P_value_Brca</th>\n",
       "      <th>Correlation_Luad</th>\n",
       "      <th>P_value_Luad</th>\n",
       "      <th>Correlation_hnscc</th>\n",
       "      <th>P_value_hnscc</th>\n",
       "      <th>Correlation_Lscc</th>\n",
       "      <th>P_value_Lscc</th>\n",
       "      <th>Correlation_Colon</th>\n",
       "      <th>P_value_Colon</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Comparison</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PHLDA1</th>\n",
       "      <td>0.816848</td>\n",
       "      <td>3.507071e-21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.364797</td>\n",
       "      <td>0.002164</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.664271</td>\n",
       "      <td>8.888640e-12</td>\n",
       "      <td>0.713420</td>\n",
       "      <td>2.644826e-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2</th>\n",
       "      <td>-0.610889</td>\n",
       "      <td>6.729990e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.302439</td>\n",
       "      <td>0.020631</td>\n",
       "      <td>-0.532341</td>\n",
       "      <td>3.320092e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDH4</th>\n",
       "      <td>0.559180</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PLA2G15</th>\n",
       "      <td>-0.556624</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>-0.298029</td>\n",
       "      <td>0.02216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.274185</td>\n",
       "      <td>0.016061</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOCS2</th>\n",
       "      <td>0.562720</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.472624</td>\n",
       "      <td>1.417921e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.287735</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SSR2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.301814</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSMB1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.287839</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL39</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.288110</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAP3K11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.498761</td>\n",
       "      <td>0.049785</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6230 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Correlation_Gbm   P_value_Gbm  Correlation_kidney  P_value_kidney  \\\n",
       "Comparison                                                                      \n",
       "PHLDA1             0.816848  3.507071e-21                 NaN             NaN   \n",
       "GRB2              -0.610889  6.729990e-08                 NaN             NaN   \n",
       "CDH4               0.559180  3.420388e-06                 NaN             NaN   \n",
       "PLA2G15           -0.556624  3.420388e-06           -0.298029         0.02216   \n",
       "SOCS2              0.562720  3.420388e-06                 NaN             NaN   \n",
       "...                     ...           ...                 ...             ...   \n",
       "CCDC6                   NaN           NaN                 NaN             NaN   \n",
       "SSR2                    NaN           NaN                 NaN             NaN   \n",
       "PSMB1                   NaN           NaN                 NaN             NaN   \n",
       "MRPL39                  NaN           NaN                 NaN             NaN   \n",
       "MAP3K11                 NaN           NaN                 NaN             NaN   \n",
       "\n",
       "            Correlation_Ovar  P_value_Ovar  Correlation_Brca  P_value_Brca  \\\n",
       "Comparison                                                                   \n",
       "PHLDA1                   NaN           NaN          0.364797      0.002164   \n",
       "GRB2                     NaN           NaN               NaN           NaN   \n",
       "CDH4                     NaN           NaN               NaN           NaN   \n",
       "PLA2G15                  NaN           NaN          0.274185      0.016061   \n",
       "SOCS2                    NaN           NaN               NaN           NaN   \n",
       "...                      ...           ...               ...           ...   \n",
       "CCDC6                    NaN           NaN               NaN           NaN   \n",
       "SSR2                     NaN           NaN               NaN           NaN   \n",
       "PSMB1                    NaN           NaN               NaN           NaN   \n",
       "MRPL39                   NaN           NaN               NaN           NaN   \n",
       "MAP3K11                  NaN           NaN               NaN           NaN   \n",
       "\n",
       "            Correlation_Luad  P_value_Luad  Correlation_hnscc  P_value_hnscc  \\\n",
       "Comparison                                                                     \n",
       "PHLDA1                   NaN           NaN           0.664271   8.888640e-12   \n",
       "GRB2               -0.302439      0.020631          -0.532341   3.320092e-06   \n",
       "CDH4                     NaN           NaN                NaN            NaN   \n",
       "PLA2G15                  NaN           NaN                NaN            NaN   \n",
       "SOCS2                    NaN           NaN                NaN            NaN   \n",
       "...                      ...           ...                ...            ...   \n",
       "CCDC6                    NaN           NaN                NaN            NaN   \n",
       "SSR2                     NaN           NaN                NaN            NaN   \n",
       "PSMB1                    NaN           NaN                NaN            NaN   \n",
       "MRPL39                   NaN           NaN                NaN            NaN   \n",
       "MAP3K11                  NaN           NaN                NaN            NaN   \n",
       "\n",
       "            Correlation_Lscc  P_value_Lscc  Correlation_Colon  P_value_Colon  \n",
       "Comparison                                                                    \n",
       "PHLDA1              0.713420  2.644826e-14                NaN            NaN  \n",
       "GRB2                     NaN           NaN                NaN            NaN  \n",
       "CDH4                     NaN           NaN                NaN            NaN  \n",
       "PLA2G15                  NaN           NaN                NaN            NaN  \n",
       "SOCS2               0.472624  1.417921e-02                NaN            NaN  \n",
       "...                      ...           ...                ...            ...  \n",
       "CCDC6                    NaN           NaN           0.287735       0.049589  \n",
       "SSR2                     NaN           NaN           0.301814       0.049589  \n",
       "PSMB1                    NaN           NaN           0.287839       0.049589  \n",
       "MRPL39                   NaN           NaN          -0.288110       0.049589  \n",
       "MAP3K11                  NaN           NaN           0.498761       0.049785  \n",
       "\n",
       "[6230 rows x 16 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FDR_sig = pd.read_csv(\"Pval_corr_table_Fig_2_return_sig.csv\")\n",
    "FDR_sig = FDR_sig.replace(to_replace ='_proteomics', value = '', regex = True) #Remove proteomics labeling\n",
    "FDR_sig= FDR_sig.drop(['Unnamed: 0'], axis=1)\n",
    "FDR_sig = FDR_sig.set_index(\"Comparison\")\n",
    "FDR_sig = FDR_sig[1:] #Drop EGFR \n",
    "FDR_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def HasPosNeg(row):\n",
    "    hasPos = False\n",
    "    hasNeg= False\n",
    "\n",
    "    for item in row:\n",
    "        if pd.isnull(item):\n",
    "            continue\n",
    "        if item < 0:\n",
    "            hasNeg = True\n",
    "        if item > 0:\n",
    "            hasPos = True\n",
    "            \n",
    "    if hasPos & hasNeg:\n",
    "        return True\n",
    "    return False\n",
    "\n",
    "def CountPosNeg(row):\n",
    "    hasPos = False\n",
    "    hasNeg= False\n",
    "    counter = 0\n",
    "    for item in row:\n",
    "        if pd.isnull(item):\n",
    "            continue\n",
    "        if item < -0:\n",
    "            hasNeg = True\n",
    "            counter += 1\n",
    "        if item > 0:\n",
    "            hasPos = True\n",
    "            counter += 1\n",
    "    return counter\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Subset data frame to include only trans genes that have opposite effects in different cancers by using apply function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "col = [\"Correlation_Gbm\",\"Correlation_kidney\",\"Correlation_Ovar\",\"Correlation_Brca\",\"Correlation_Luad\",\"Correlation_hnscc\",\"Correlation_Lscc\",\"Correlation_Colon\"]\n",
    "FDR_corr = FDR_sig[col]\n",
    "FDR_corr[\"Pos_Neg\"] = FDR_corr.apply(HasPosNeg, axis = 1)\n",
    "FDR_corr_False = FDR_corr[FDR_corr['Pos_Neg']==False]\n",
    "FDR_corr_False = FDR_corr_False.drop([\"Pos_Neg\"], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation_Gbm</th>\n",
       "      <th>Correlation_kidney</th>\n",
       "      <th>Correlation_Ovar</th>\n",
       "      <th>Correlation_Brca</th>\n",
       "      <th>Correlation_Luad</th>\n",
       "      <th>Correlation_hnscc</th>\n",
       "      <th>Correlation_Lscc</th>\n",
       "      <th>Correlation_Colon</th>\n",
       "      <th>Num_corr</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Comparison</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PHLDA1</th>\n",
       "      <td>0.816848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.364797</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.664271</td>\n",
       "      <td>0.713420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2</th>\n",
       "      <td>-0.610889</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.302439</td>\n",
       "      <td>-0.532341</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOCS2</th>\n",
       "      <td>0.562720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.472624</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLA</th>\n",
       "      <td>-0.550491</td>\n",
       "      <td>-0.298348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHLDA3</th>\n",
       "      <td>0.525883</td>\n",
       "      <td>0.432530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.763784</td>\n",
       "      <td>0.682034</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S100A14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370227</td>\n",
       "      <td>0.351373</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MOV10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.302787</td>\n",
       "      <td>0.299218</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP2B1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.298873</td>\n",
       "      <td>0.346409</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MFSD6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.295201</td>\n",
       "      <td>0.380169</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SART1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.292189</td>\n",
       "      <td>-0.306539</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1136 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Correlation_Gbm  Correlation_kidney  Correlation_Ovar  \\\n",
       "Comparison                                                          \n",
       "PHLDA1             0.816848                 NaN               NaN   \n",
       "GRB2              -0.610889                 NaN               NaN   \n",
       "SOCS2              0.562720                 NaN               NaN   \n",
       "GLA               -0.550491           -0.298348               NaN   \n",
       "PHLDA3             0.525883            0.432530               NaN   \n",
       "...                     ...                 ...               ...   \n",
       "S100A14                 NaN                 NaN               NaN   \n",
       "MOV10                   NaN                 NaN               NaN   \n",
       "ATP2B1                  NaN                 NaN               NaN   \n",
       "MFSD6                   NaN                 NaN               NaN   \n",
       "SART1                   NaN                 NaN               NaN   \n",
       "\n",
       "            Correlation_Brca  Correlation_Luad  Correlation_hnscc  \\\n",
       "Comparison                                                          \n",
       "PHLDA1              0.364797               NaN           0.664271   \n",
       "GRB2                     NaN         -0.302439          -0.532341   \n",
       "SOCS2                    NaN               NaN                NaN   \n",
       "GLA                      NaN               NaN                NaN   \n",
       "PHLDA3                   NaN               NaN           0.763784   \n",
       "...                      ...               ...                ...   \n",
       "S100A14                  NaN               NaN                NaN   \n",
       "MOV10                    NaN               NaN                NaN   \n",
       "ATP2B1                   NaN               NaN                NaN   \n",
       "MFSD6                    NaN               NaN                NaN   \n",
       "SART1                    NaN               NaN                NaN   \n",
       "\n",
       "            Correlation_Lscc  Correlation_Colon  Num_corr  \n",
       "Comparison                                                 \n",
       "PHLDA1              0.713420                NaN         4  \n",
       "GRB2                     NaN                NaN         3  \n",
       "SOCS2               0.472624                NaN         2  \n",
       "GLA                      NaN                NaN         2  \n",
       "PHLDA3              0.682034                NaN         4  \n",
       "...                      ...                ...       ...  \n",
       "S100A14             0.370227           0.351373         2  \n",
       "MOV10               0.302787           0.299218         2  \n",
       "ATP2B1              0.298873           0.346409         2  \n",
       "MFSD6               0.295201           0.380169         2  \n",
       "SART1              -0.292189          -0.306539         2  \n",
       "\n",
       "[1136 rows x 9 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FDR_corr_False[\"Num_corr\"] = FDR_corr_False.apply(CountPosNeg, axis = 1)\n",
    "FDR_corr_False.sort_values(by=['Num_corr'],ascending=False)\n",
    "FDR_corr_False = FDR_corr_False[FDR_corr_False['Num_corr'].isin([2,3,4,5,6])]\n",
    "FDR_corr_False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The manuscript mentions the total number of trans proteins that opposite effects in different cancers. Here is the derivation of that number "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of trans proteins with same effects in different cancers is 1136\n"
     ]
    }
   ],
   "source": [
    "same_sign_prot = FDR_corr_False.index.tolist()\n",
    "print(\"Total number of trans proteins with same effects in different cancers is \" + str(len(same_sign_prot)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run GSEA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Term</th>\n",
       "      <th>Overlap</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Adjusted P-value</th>\n",
       "      <th>Old P-value</th>\n",
       "      <th>Old Adjusted P-value</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>Combined Score</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Gene_set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Arrhythmogenic right ventricular cardiomyopath...</td>\n",
       "      <td>18/74</td>\n",
       "      <td>1.159284e-07</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.282451</td>\n",
       "      <td>68.392004</td>\n",
       "      <td>DSP;ITGA4;JUP;ITGB4;ACTN2;ITGA2;GJA1;CDH2;PKP2...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Metabolic pathways Homo sapiens hsa01100</td>\n",
       "      <td>114/1239</td>\n",
       "      <td>1.813751e-07</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.619889</td>\n",
       "      <td>25.145046</td>\n",
       "      <td>MSMO1;ENO1;NSDHL;NAMPT;ENPP1;ENPP3;ATP6V1E1;PT...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Regulation of actin cytoskeleton Homo sapiens ...</td>\n",
       "      <td>32/214</td>\n",
       "      <td>5.044867e-07</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.632618</td>\n",
       "      <td>38.172238</td>\n",
       "      <td>CYFIP2;NCKAP1;ITGB4;WAS;PIK3CD;IQGAP2;ITGAL;FG...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Biosynthesis of amino acids Homo sapiens hsa01230</td>\n",
       "      <td>17/74</td>\n",
       "      <td>6.249417e-07</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.044537</td>\n",
       "      <td>57.778675</td>\n",
       "      <td>PRPS1;ARG1;PGAM1;GPT;ENO1;PFKL;PKM;CBS;PSAT1;P...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Glycolysis / Gluconeogenesis Homo sapiens hsa0...</td>\n",
       "      <td>16/67</td>\n",
       "      <td>7.613663e-07</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.204330</td>\n",
       "      <td>59.231243</td>\n",
       "      <td>GPI;ACSS2;ADH1B;PGAM1;ENO1;HK2;LDHA;PFKL;PKM;P...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Bacterial invasion of epithelial cells Homo sa...</td>\n",
       "      <td>17/78</td>\n",
       "      <td>1.377343e-06</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.837125</td>\n",
       "      <td>51.783367</td>\n",
       "      <td>WAS;FN1;PIK3CD;SEPT1;PIK3CG;SEPT2;CD2AP;SEPT9;...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>B cell receptor signaling pathway Homo sapiens...</td>\n",
       "      <td>16/73</td>\n",
       "      <td>2.588556e-06</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.858769</td>\n",
       "      <td>49.640789</td>\n",
       "      <td>CR2;JUN;SYK;INPPL1;PIK3CD;NFATC2;PIK3CG;VAV1;P...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Carbon metabolism Homo sapiens hsa01200</td>\n",
       "      <td>20/113</td>\n",
       "      <td>5.248171e-06</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.116041</td>\n",
       "      <td>37.883681</td>\n",
       "      <td>GPI;PRPS1;H6PD;ACSS2;ECHS1;PGAM1;GPT;ENO1;HK2;...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>T cell receptor signaling pathway Homo sapiens...</td>\n",
       "      <td>19/104</td>\n",
       "      <td>5.547202e-06</td>\n",
       "      <td>0.000181</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.216414</td>\n",
       "      <td>38.925738</td>\n",
       "      <td>JUN;NFATC2;PIK3CD;CD3G;PIK3CG;VAV1;ZAP70;CD4;P...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Axon guidance Homo sapiens hsa04360</td>\n",
       "      <td>21/127</td>\n",
       "      <td>9.304350e-06</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.911168</td>\n",
       "      <td>33.725962</td>\n",
       "      <td>EPHA4;SEMA4A;SEMA4D;SEMA4B;LIMK1;GNAI3;NFATC2;...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Term   Overlap       P-value  \\\n",
       "0  Arrhythmogenic right ventricular cardiomyopath...     18/74  1.159284e-07   \n",
       "1           Metabolic pathways Homo sapiens hsa01100  114/1239  1.813751e-07   \n",
       "2  Regulation of actin cytoskeleton Homo sapiens ...    32/214  5.044867e-07   \n",
       "3  Biosynthesis of amino acids Homo sapiens hsa01230     17/74  6.249417e-07   \n",
       "4  Glycolysis / Gluconeogenesis Homo sapiens hsa0...     16/67  7.613663e-07   \n",
       "5  Bacterial invasion of epithelial cells Homo sa...     17/78  1.377343e-06   \n",
       "6  B cell receptor signaling pathway Homo sapiens...     16/73  2.588556e-06   \n",
       "7            Carbon metabolism Homo sapiens hsa01200    20/113  5.248171e-06   \n",
       "8  T cell receptor signaling pathway Homo sapiens...    19/104  5.547202e-06   \n",
       "9                Axon guidance Homo sapiens hsa04360    21/127  9.304350e-06   \n",
       "\n",
       "   Adjusted P-value  Old P-value  Old Adjusted P-value  Odds Ratio  \\\n",
       "0          0.000034            0                     0    4.282451   \n",
       "1          0.000027            0                     0    1.619889   \n",
       "2          0.000049            0                     0    2.632618   \n",
       "3          0.000046            0                     0    4.044537   \n",
       "4          0.000045            0                     0    4.204330   \n",
       "5          0.000067            0                     0    3.837125   \n",
       "6          0.000108            0                     0    3.858769   \n",
       "7          0.000192            0                     0    3.116041   \n",
       "8          0.000181            0                     0    3.216414   \n",
       "9          0.000273            0                     0    2.911168   \n",
       "\n",
       "   Combined Score                                              Genes  \\\n",
       "0       68.392004  DSP;ITGA4;JUP;ITGB4;ACTN2;ITGA2;GJA1;CDH2;PKP2...   \n",
       "1       25.145046  MSMO1;ENO1;NSDHL;NAMPT;ENPP1;ENPP3;ATP6V1E1;PT...   \n",
       "2       38.172238  CYFIP2;NCKAP1;ITGB4;WAS;PIK3CD;IQGAP2;ITGAL;FG...   \n",
       "3       57.778675  PRPS1;ARG1;PGAM1;GPT;ENO1;PFKL;PKM;CBS;PSAT1;P...   \n",
       "4       59.231243  GPI;ACSS2;ADH1B;PGAM1;ENO1;HK2;LDHA;PFKL;PKM;P...   \n",
       "5       51.783367  WAS;FN1;PIK3CD;SEPT1;PIK3CG;SEPT2;CD2AP;SEPT9;...   \n",
       "6       49.640789  CR2;JUN;SYK;INPPL1;PIK3CD;NFATC2;PIK3CG;VAV1;P...   \n",
       "7       37.883681  GPI;PRPS1;H6PD;ACSS2;ECHS1;PGAM1;GPT;ENO1;HK2;...   \n",
       "8       38.925738  JUN;NFATC2;PIK3CD;CD3G;PIK3CG;VAV1;ZAP70;CD4;P...   \n",
       "9       33.725962  EPHA4;SEMA4A;SEMA4D;SEMA4B;LIMK1;GNAI3;NFATC2;...   \n",
       "\n",
       "    Gene_set  \n",
       "0  KEGG_2016  \n",
       "1  KEGG_2016  \n",
       "2  KEGG_2016  \n",
       "3  KEGG_2016  \n",
       "4  KEGG_2016  \n",
       "5  KEGG_2016  \n",
       "6  KEGG_2016  \n",
       "7  KEGG_2016  \n",
       "8  KEGG_2016  \n",
       "9  KEGG_2016  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "same_sign_enr = gp.enrichr(gene_list = same_sign_prot, description='Tumor_partition', gene_sets='KEGG_2016', \n",
    "                       outdir='test/enrichr_KEGG')\n",
    "same_sign_enr.res2d.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "list.remove(x): x not in list",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-8-695d213258e9>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     13\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m \u001b[1;31m#remove = [\"CR2\",\"NCK1\",\"CARD11\",\"INPPL1\",\"JUN\",\"KRAS\"]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 15\u001b[1;33m \u001b[0mimmune\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mremove\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"JUN\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     16\u001b[0m \u001b[0mimmune\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mremove\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"KRAS\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m \u001b[0mimmune\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mremove\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"CR2\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: list.remove(x): x not in list"
     ]
    }
   ],
   "source": [
    "#get just the clotting cascade genes and add urokinase genes \n",
    "same_sign_df = same_sign_enr.res2d\n",
    "bcell = same_sign_df.iloc[6,9]\n",
    "bcell = bcell.split(';')\n",
    "tcell = same_sign_df.iloc[8,9]\n",
    "tcell = tcell.split(\";\")\n",
    "immune_both =  bcell + tcell\n",
    "immune = set(immune_both)\n",
    "immune = list(immune)\n",
    "btcell_apm = ([\"CD5\",\"CD8A\",\"BLK\",\"BLNK\",\"CD79A\",\"LYN\",\"CD3E\",\"HLA-DMA\",\"HLA-DMB\"])\n",
    "\n",
    "immune= immune + btcell_apm\n",
    "\n",
    "#remove = [\"CR2\",\"NCK1\",\"CARD11\",\"INPPL1\",\"JUN\",\"KRAS\"]\n",
    "immune.remove(\"JUN\")\n",
    "immune.remove(\"KRAS\")\n",
    "immune.remove(\"CR2\")\n",
    "immune.remove(\"NCK1\")\n",
    "immune.remove(\"INPPL1\")\n",
    "immune.remove(\"CARD11\")\n",
    "immune"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 3 Make Data frame for figure 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sig PTEN # CD2? CHUK\n",
    "t_only = ['CD3E', 'CD3G', 'CD4', 'CD8A', 'GRAP2', 'FYB1', 'PRKCQ'] \n",
    "b_only = ['PPP3CC', 'CD79A', 'SYK', 'BTK', 'LYN', 'BLNK', 'BLK', 'HLA-DMA', 'HLA-DMB']\n",
    "both = ['CD2', 'CD5', 'GRB2', 'LCK', 'ZAP70', 'VAV1', 'PTPN6', 'PTPRC', 'PIK3CD', 'PIK3CG', 'INPP5D',\n",
    "        'PLCG1', 'PLCG2', 'NFATC2', 'PRKCB', 'CHUK', 'NFKB2', 'DOCK2', 'RAC2', 'ELMO1', 'WAS']\n",
    "immune = t_only + b_only + both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Correlation</th>\n",
       "      <th>P_value</th>\n",
       "      <th>Cancer Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>-0.610889</td>\n",
       "      <td>6.729990e-08</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>WAS</td>\n",
       "      <td>-0.501918</td>\n",
       "      <td>4.961122e-05</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>PLCG2</td>\n",
       "      <td>-0.472453</td>\n",
       "      <td>1.145235e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>SYK</td>\n",
       "      <td>-0.470849</td>\n",
       "      <td>1.202510e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>CD4</td>\n",
       "      <td>-0.467479</td>\n",
       "      <td>1.362923e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7781</th>\n",
       "      <td>PRKCQ</td>\n",
       "      <td>-0.284462</td>\n",
       "      <td>3.293176e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7876</th>\n",
       "      <td>RAC2</td>\n",
       "      <td>-0.277100</td>\n",
       "      <td>3.855935e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8461</th>\n",
       "      <td>NFATC2</td>\n",
       "      <td>-0.297428</td>\n",
       "      <td>4.438700e-02</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8573</th>\n",
       "      <td>PRKCB</td>\n",
       "      <td>0.395927</td>\n",
       "      <td>6.516094e-03</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8693</th>\n",
       "      <td>NFKB2</td>\n",
       "      <td>0.357365</td>\n",
       "      <td>1.247990e-02</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Comparison  Correlation       P_value    Cancer Type\n",
       "1          GRB2    -0.610889  6.729990e-08            GBM\n",
       "24          WAS    -0.501918  4.961122e-05            GBM\n",
       "72        PLCG2    -0.472453  1.145235e-04            GBM\n",
       "75          SYK    -0.470849  1.202510e-04            GBM\n",
       "82          CD4    -0.467479  1.362923e-04            GBM\n",
       "...         ...          ...           ...            ...\n",
       "7781      PRKCQ    -0.284462  3.293176e-02  Head and Neck\n",
       "7876       RAC2    -0.277100  3.855935e-02  Head and Neck\n",
       "8461     NFATC2    -0.297428  4.438700e-02           Lscc\n",
       "8573      PRKCB     0.395927  6.516094e-03          Colon\n",
       "8693      NFKB2     0.357365  1.247990e-02          Colon\n",
       "\n",
       "[87 rows x 4 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Get append version of the df with all cancer type, fdr sig trans results\n",
    "df_FDR_append = pd.read_csv(\"Pval_corr_table_Fig_2_return_sig_append.csv\")\n",
    "df_FDR_append = df_FDR_append.drop(['Unnamed: 0'], axis=1)\n",
    "\n",
    "#Remove Proteomics labeling \n",
    "df_FDR_append = df_FDR_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_FDR_append= df_FDR_append[df_FDR_append.Comparison.isin(immune)]\n",
    "df_FDR_append['Cancer Type'].unique()\n",
    "df_FDR_append"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set add new column to be unique index and order the new index. This way genes will be grouped by coagulation factors, regulators, and urokinase genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CD3E_Breast', 'CD3E_Colon', 'CD3E_GBM', 'CD3E_Head and Neck', 'CD3E_Kidney', 'CD3E_Lscc', 'CD3E_Luad', 'CD3G_Breast', 'CD3G_Colon', 'CD3G_GBM', 'CD3G_Head and Neck', 'CD3G_Kidney', 'CD3G_Lscc', 'CD3G_Luad', 'CD4_Breast', 'CD4_Colon', 'CD4_GBM', 'CD4_Head and Neck', 'CD4_Kidney', 'CD4_Lscc', 'CD4_Luad', 'CD8A_Breast', 'CD8A_Colon', 'CD8A_GBM', 'CD8A_Head and Neck', 'CD8A_Kidney', 'CD8A_Lscc', 'CD8A_Luad', 'GRAP2_Breast', 'GRAP2_Colon', 'GRAP2_GBM', 'GRAP2_Head and Neck', 'GRAP2_Kidney', 'GRAP2_Lscc', 'GRAP2_Luad', 'FYB1_Breast', 'FYB1_Colon', 'FYB1_GBM', 'FYB1_Head and Neck', 'FYB1_Kidney', 'FYB1_Lscc', 'FYB1_Luad', 'PRKCQ_Breast', 'PRKCQ_Colon', 'PRKCQ_GBM', 'PRKCQ_Head and Neck', 'PRKCQ_Kidney', 'PRKCQ_Lscc', 'PRKCQ_Luad', 'PPP3CC_Breast', 'PPP3CC_Colon', 'PPP3CC_GBM', 'PPP3CC_Head and Neck', 'PPP3CC_Kidney', 'PPP3CC_Lscc', 'PPP3CC_Luad', 'CD79A_Breast', 'CD79A_Colon', 'CD79A_GBM', 'CD79A_Head and Neck', 'CD79A_Kidney', 'CD79A_Lscc', 'CD79A_Luad', 'SYK_Breast', 'SYK_Colon', 'SYK_GBM', 'SYK_Head and Neck', 'SYK_Kidney', 'SYK_Lscc', 'SYK_Luad', 'BTK_Breast', 'BTK_Colon', 'BTK_GBM', 'BTK_Head and Neck', 'BTK_Kidney', 'BTK_Lscc', 'BTK_Luad', 'LYN_Breast', 'LYN_Colon', 'LYN_GBM', 'LYN_Head and Neck', 'LYN_Kidney', 'LYN_Lscc', 'LYN_Luad', 'BLNK_Breast', 'BLNK_Colon', 'BLNK_GBM', 'BLNK_Head and Neck', 'BLNK_Kidney', 'BLNK_Lscc', 'BLNK_Luad', 'BLK_Breast', 'BLK_Colon', 'BLK_GBM', 'BLK_Head and Neck', 'BLK_Kidney', 'BLK_Lscc', 'BLK_Luad', 'HLA-DMA_Breast', 'HLA-DMA_Colon', 'HLA-DMA_GBM', 'HLA-DMA_Head and Neck', 'HLA-DMA_Kidney', 'HLA-DMA_Lscc', 'HLA-DMA_Luad', 'HLA-DMB_Breast', 'HLA-DMB_Colon', 'HLA-DMB_GBM', 'HLA-DMB_Head and Neck', 'HLA-DMB_Kidney', 'HLA-DMB_Lscc', 'HLA-DMB_Luad', 'CD2_Breast', 'CD2_Colon', 'CD2_GBM', 'CD2_Head and Neck', 'CD2_Kidney', 'CD2_Lscc', 'CD2_Luad', 'CD5_Breast', 'CD5_Colon', 'CD5_GBM', 'CD5_Head and Neck', 'CD5_Kidney', 'CD5_Lscc', 'CD5_Luad', 'GRB2_Breast', 'GRB2_Colon', 'GRB2_GBM', 'GRB2_Head and Neck', 'GRB2_Kidney', 'GRB2_Lscc', 'GRB2_Luad', 'LCK_Breast', 'LCK_Colon', 'LCK_GBM', 'LCK_Head and Neck', 'LCK_Kidney', 'LCK_Lscc', 'LCK_Luad', 'ZAP70_Breast', 'ZAP70_Colon', 'ZAP70_GBM', 'ZAP70_Head and Neck', 'ZAP70_Kidney', 'ZAP70_Lscc', 'ZAP70_Luad', 'VAV1_Breast', 'VAV1_Colon', 'VAV1_GBM', 'VAV1_Head and Neck', 'VAV1_Kidney', 'VAV1_Lscc', 'VAV1_Luad', 'PTPN6_Breast', 'PTPN6_Colon', 'PTPN6_GBM', 'PTPN6_Head and Neck', 'PTPN6_Kidney', 'PTPN6_Lscc', 'PTPN6_Luad', 'PTPRC_Breast', 'PTPRC_Colon', 'PTPRC_GBM', 'PTPRC_Head and Neck', 'PTPRC_Kidney', 'PTPRC_Lscc', 'PTPRC_Luad', 'PIK3CD_Breast', 'PIK3CD_Colon', 'PIK3CD_GBM', 'PIK3CD_Head and Neck', 'PIK3CD_Kidney', 'PIK3CD_Lscc', 'PIK3CD_Luad', 'PIK3CG_Breast', 'PIK3CG_Colon', 'PIK3CG_GBM', 'PIK3CG_Head and Neck', 'PIK3CG_Kidney', 'PIK3CG_Lscc', 'PIK3CG_Luad', 'INPP5D_Breast', 'INPP5D_Colon', 'INPP5D_GBM', 'INPP5D_Head and Neck', 'INPP5D_Kidney', 'INPP5D_Lscc', 'INPP5D_Luad', 'PLCG1_Breast', 'PLCG1_Colon', 'PLCG1_GBM', 'PLCG1_Head and Neck', 'PLCG1_Kidney', 'PLCG1_Lscc', 'PLCG1_Luad', 'PLCG2_Breast', 'PLCG2_Colon', 'PLCG2_GBM', 'PLCG2_Head and Neck', 'PLCG2_Kidney', 'PLCG2_Lscc', 'PLCG2_Luad', 'NFATC2_Breast', 'NFATC2_Colon', 'NFATC2_GBM', 'NFATC2_Head and Neck', 'NFATC2_Kidney', 'NFATC2_Lscc', 'NFATC2_Luad', 'PRKCB_Breast', 'PRKCB_Colon', 'PRKCB_GBM', 'PRKCB_Head and Neck', 'PRKCB_Kidney', 'PRKCB_Lscc', 'PRKCB_Luad', 'CHUK_Breast', 'CHUK_Colon', 'CHUK_GBM', 'CHUK_Head and Neck', 'CHUK_Kidney', 'CHUK_Lscc', 'CHUK_Luad', 'NFKB2_Breast', 'NFKB2_Colon', 'NFKB2_GBM', 'NFKB2_Head and Neck', 'NFKB2_Kidney', 'NFKB2_Lscc', 'NFKB2_Luad', 'DOCK2_Breast', 'DOCK2_Colon', 'DOCK2_GBM', 'DOCK2_Head and Neck', 'DOCK2_Kidney', 'DOCK2_Lscc', 'DOCK2_Luad', 'RAC2_Breast', 'RAC2_Colon', 'RAC2_GBM', 'RAC2_Head and Neck', 'RAC2_Kidney', 'RAC2_Lscc', 'RAC2_Luad', 'ELMO1_Breast', 'ELMO1_Colon', 'ELMO1_GBM', 'ELMO1_Head and Neck', 'ELMO1_Kidney', 'ELMO1_Lscc', 'ELMO1_Luad', 'WAS_Breast', 'WAS_Colon', 'WAS_GBM', 'WAS_Head and Neck', 'WAS_Kidney', 'WAS_Lscc', 'WAS_Luad']\n",
      "Empty DataFrame\n",
      "Columns: [Comparison, Correlation, P_value, Cancer Type]\n",
      "Index: []\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'P_Value'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\pandas\\core\\indexes\\base.py\u001b[0m in \u001b[0;36mget_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   2888\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2889\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2890\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'P_Value'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-23-41307bd1b0c1>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     10\u001b[0m \u001b[0mordered_df\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdf_FDR_append\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreindex\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mordered_list\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     11\u001b[0m \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mordered_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdropna\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 12\u001b[1;33m \u001b[0mordered_df\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mordered_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mordered_df\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'P_Value'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m<\u001b[0m \u001b[1;36m0.05\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;31m# Keep sig\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     13\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m \u001b[0mdf_ordered\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m40\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\pandas\\core\\frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m   2897\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[1;33m>\u001b[0m \u001b[1;36m1\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2898\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2899\u001b[1;33m             \u001b[0mindexer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2900\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2901\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\pandas\\core\\indexes\\base.py\u001b[0m in \u001b[0;36mget_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   2889\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2890\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2891\u001b[1;33m                 \u001b[1;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2892\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2893\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mtolerance\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'P_Value'"
     ]
    }
   ],
   "source": [
    "\n",
    "df_FDR_append[\"Index\"] = df_FDR_append[\"Comparison\"] + \" \" + df_FDR_append[\"Cancer Type\"]\n",
    "df_FDR_append = df_FDR_append.set_index(\"Index\")\n",
    "\n",
    "ordered_list = []\n",
    "cancer_list = ['Breast', 'Colon', 'GBM', 'Head and Neck', 'Kidney', 'Lscc', 'Luad']\n",
    "for gene in immune:\n",
    "    for cancer in cancer_list:\n",
    "        ordered_list.append(gene+'_'+cancer)\n",
    "\n",
    "ordered_df = df_FDR_append.reindex(ordered_list)\n",
    "ordered_df = ordered_df.loc[ordered_df['P_value'] < 0.05] # Keep sig \n",
    "                                    \n",
    "df_ordered.head(40)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 4: Plot Figure 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "expected an element of either Seq(String), Seq(Tuple(String, String)) or Seq(Tuple(String, String, String)), got ['CD3E', nan, 'CD8A', 'CD79A', 'CD5', 'LYN', 'BLNK', 'BLK', 'HLA-DMA', 'HLA-DMB']",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-12-88b3ec90500b>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[0mlegend_min\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdf_ordered\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"P_value\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[1;31m#Make plot using plot utils\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[0mp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mplotCircleHeatMap\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf_ordered\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcircle_var\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"P_value\"\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mcolor_var\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"Correlation\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mx_axis\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"Comparison\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my_axis\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"Cancer Type\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mplot_width\u001b[0m\u001b[1;33m=\u001b[0m \u001b[1;36m1000\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mplot_height\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;36m500\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlegend_min\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mlegend_min\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlegend_max\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;36m0.05\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfont_size\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;36m10\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mshow_legend\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m \u001b[1;33m,\u001b[0m \u001b[0msave_png\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"Figure2.png\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\plot_utils\\__init__.py\u001b[0m in \u001b[0;36mplotCircleHeatMap\u001b[1;34m(df, circle_var, color_var, x_axis, y_axis, plot_width, plot_height, font_size, x_axis_lab, y_axis_lab, show_plot, save_png, legend_min, legend_med, legend_max, show_legend)\u001b[0m\n\u001b[0;32m     58\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     59\u001b[0m     \u001b[1;31m#p.x_range.factors = sorted(df[x_axis].unique().tolist())\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 60\u001b[1;33m     \u001b[0mp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mx_range\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfactors\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mx_axis\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munique\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     61\u001b[0m     \u001b[0mp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0my_range\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfactors\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msorted\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0my_axis\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munique\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mreverse\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     62\u001b[0m     \u001b[0mp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mxaxis\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmajor_label_orientation\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmath\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpi\u001b[0m\u001b[1;33m/\u001b[0m\u001b[1;36m2\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\bokeh\\core\\has_props.py\u001b[0m in \u001b[0;36m__setattr__\u001b[1;34m(self, name, value)\u001b[0m\n\u001b[0;32m    272\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    273\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mname\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mprops\u001b[0m \u001b[1;32mor\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mdescriptor\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mdescriptor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfset\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 274\u001b[1;33m             \u001b[0msuper\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    275\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    276\u001b[0m             \u001b[0mmatches\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtext\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdifflib\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_close_matches\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mprops\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"similar\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\bokeh\\core\\property\\descriptors.py\u001b[0m in \u001b[0;36m__set__\u001b[1;34m(self, obj, value, setter)\u001b[0m\n\u001b[0;32m    537\u001b[0m             \u001b[1;32mraise\u001b[0m \u001b[0mRuntimeError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"%s.%s is a readonly property\"\u001b[0m \u001b[1;33m%\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__class__\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__name__\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    538\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 539\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_internal_set\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msetter\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0msetter\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    540\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    541\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__delete__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mobj\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\bokeh\\core\\property\\descriptors.py\u001b[0m in \u001b[0;36m_internal_set\u001b[1;34m(self, obj, value, hint, setter)\u001b[0m\n\u001b[0;32m    758\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    759\u001b[0m         '''\n\u001b[1;32m--> 760\u001b[1;33m         \u001b[0mvalue\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mproperty\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mprepare_value\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    761\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    762\u001b[0m         \u001b[0mold\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__get__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mobj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__class__\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\bokeh\\core\\property\\bases.py\u001b[0m in \u001b[0;36mprepare_value\u001b[1;34m(self, obj_or_cls, name, value)\u001b[0m\n\u001b[0;32m    329\u001b[0m                     \u001b[1;32mbreak\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    330\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 331\u001b[1;33m                 \u001b[1;32mraise\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    332\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    333\u001b[0m             \u001b[0mvalue\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtransform\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\bokeh\\core\\property\\bases.py\u001b[0m in \u001b[0;36mprepare_value\u001b[1;34m(self, obj_or_cls, name, value)\u001b[0m\n\u001b[0;32m    322\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    323\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mvalidation_on\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 324\u001b[1;33m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvalidate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    325\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mValueError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    326\u001b[0m             \u001b[1;32mfor\u001b[0m \u001b[0mtp\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mconverter\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0malternatives\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\bokeh\\core\\property\\either.py\u001b[0m in \u001b[0;36mvalidate\u001b[1;34m(self, value, detail)\u001b[0m\n\u001b[0;32m    107\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mvalue\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0many\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mparam\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_valid\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mparam\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtype_params\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    108\u001b[0m             \u001b[0mmsg\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"\"\u001b[0m \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mdetail\u001b[0m \u001b[1;32melse\u001b[0m \u001b[1;34m\"expected an element of either %s, got %r\"\u001b[0m \u001b[1;33m%\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mnice_join\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtype_params\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 109\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    110\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    111\u001b[0m     \u001b[1;31m# TODO (bev) implement this\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: expected an element of either Seq(String), Seq(Tuple(String, String)) or Seq(Tuple(String, String, String)), got ['CD3E', nan, 'CD8A', 'CD79A', 'CD5', 'LYN', 'BLNK', 'BLK', 'HLA-DMA', 'HLA-DMB']"
     ]
    }
   ],
   "source": [
    "legend_min = df_ordered[\"P_value\"].min()\n",
    "#Make plot using plot utils\n",
    "p.plotCircleHeatMap(df_ordered, circle_var = \"P_value\",color_var = \"Correlation\", x_axis = \"Comparison\", y_axis = \"Cancer Type\", plot_width= 1000, plot_height = 500, legend_min = legend_min, legend_max = 0.05, font_size = 10, show_legend = True , save_png = \"Figure2.png\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check if blanks are due to no data "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The follow code chunks show that the following cancers/genes don't have data: colon THBD, Kidney MMP12, and Ovarian MMP12. (As mentioned in EGFR Figure 2 legend)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get append version of the df with all proteins \n",
    "df_all_prot_append = pd.read_csv(\"Pval_corr_table_Fig_2_return_all_append.csv\")\n",
    "df_all_prot_append = df_all_prot_append.drop(['Unnamed: 0'], axis=1)\n",
    "#Remove Proteomics labeling \n",
    "df_all_prot_append = df_all_prot_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#MHC class2 (CD4) molecules\n",
    "mhc2 = [\"HLA-DPA1\",\"HLA-DPB1\",\"HLA-DMA\",\"HLA-DMB\",\"HLA-DOA\",\"HLA-DOB\",\"HLA-DQ\",\"HLA-DRA\",\"HLA-DRB1\"]\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_all_mhc2 = df_all_prot_append[df_all_prot_append.Comparison.isin(mhc2)]\n",
    "print(\"Number of missing columns\" )\n",
    "print(56 - len(df_all_mhc2))\n",
    "#HLA-DP and HLA-DM missing for all. HLA-DRA and HLA-DRB1 consistently not sig \n",
    "df_all_mhc2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CD8 and MHC class 1 molecules\n",
    "mhc1 = [\"CD8A\",\"HLA-A\",\"HLA-B\",\"HLA-C\"]\n",
    "\n",
    "df_all_mhc1 = df_all_prot_append[df_all_prot_append.Comparison.isin(mhc1)]\n",
    "df_all_mhc1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Remove Proteomics labeling \n",
    "df_all_prot_append = df_all_prot_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_all_comp_coag = df_all_prot_append[df_all_prot_append.Comparison.isin(coag_upa)]\n",
    "print(\"Number of rows in data frame \" + str(len(df_all_comp_coag)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Our Figure includes 26 genes for 8 cancers. If all data was present there would be 208 rows. However, the data frame only has 205 rows. 3 genes are missing. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_missing_genes(test_list, full_list):\n",
    "    for gene in full_list:\n",
    "        if (gene not in test_list):\n",
    "            print(gene)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get list of genes for colon, kidney, and ovarian\n",
    "colon = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Colon\"]\n",
    "colon_list = colon.Comparison.to_list()\n",
    "\n",
    "Kidney = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Kidney\"]\n",
    "Kidney_list = Kidney.Comparison.to_list()\n",
    "\n",
    "Ovarian = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Ovarian\"]\n",
    "Ovarian_list = Ovarian.Comparison.to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Show the 3 missing genes \n",
    "print(\"Ovarian missing genes: \")\n",
    "find_missing_genes(Ovarian_list, coag_upa)\n",
    "print(\"Kidney missing genes: \")\n",
    "find_missing_genes(Kidney_list, coag_upa)\n",
    "print(\"Colon missing genes: \")\n",
    "find_missing_genes(colon_list, coag_upa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
