{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make EGFR Figure 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebooks takes all the trans genes that have positive and negative results and runs a GSEA using Reactome. It then takes a subset of genes from the top hit(Hemostasis) pathway and maps them on a large circle heat map. This heatmap focuses on coagulation and urokinase related genes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:5: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  \"\"\"\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas.util.testing as tm\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "import statsmodels.stats.multitest\n",
    "\n",
    "import gseapy as gp\n",
    "from gseapy.plot import barplot, dotplot\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 1: Find Trans proteins with opposite effects in different cancers "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load df with all of the genes that are FDR significant."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation_Gbm</th>\n",
       "      <th>P_value_Gbm</th>\n",
       "      <th>Correlation_kidney</th>\n",
       "      <th>P_value_kidney</th>\n",
       "      <th>Correlation_Ovar</th>\n",
       "      <th>P_value_Ovar</th>\n",
       "      <th>Correlation_Brca</th>\n",
       "      <th>P_value_Brca</th>\n",
       "      <th>Correlation_Luad</th>\n",
       "      <th>P_value_Luad</th>\n",
       "      <th>Correlation_hnscc</th>\n",
       "      <th>P_value_hnscc</th>\n",
       "      <th>Correlation_Lscc</th>\n",
       "      <th>P_value_Lscc</th>\n",
       "      <th>Correlation_Colon</th>\n",
       "      <th>P_value_Colon</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Comparison</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PHLDA1</th>\n",
       "      <td>0.816848</td>\n",
       "      <td>3.507071e-21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.364797</td>\n",
       "      <td>0.002164</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.664271</td>\n",
       "      <td>8.888640e-12</td>\n",
       "      <td>0.713420</td>\n",
       "      <td>2.644826e-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2</th>\n",
       "      <td>-0.610889</td>\n",
       "      <td>6.729990e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.302439</td>\n",
       "      <td>0.020631</td>\n",
       "      <td>-0.532341</td>\n",
       "      <td>3.320092e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDH4</th>\n",
       "      <td>0.559180</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PLA2G15</th>\n",
       "      <td>-0.556624</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>-0.298029</td>\n",
       "      <td>0.02216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.274185</td>\n",
       "      <td>0.016061</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOCS2</th>\n",
       "      <td>0.562720</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.472624</td>\n",
       "      <td>1.417921e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.287735</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SSR2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.301814</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSMB1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.287839</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL39</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.288110</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAP3K11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.498761</td>\n",
       "      <td>0.049785</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6230 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Correlation_Gbm   P_value_Gbm  Correlation_kidney  P_value_kidney  \\\n",
       "Comparison                                                                      \n",
       "PHLDA1             0.816848  3.507071e-21                 NaN             NaN   \n",
       "GRB2              -0.610889  6.729990e-08                 NaN             NaN   \n",
       "CDH4               0.559180  3.420388e-06                 NaN             NaN   \n",
       "PLA2G15           -0.556624  3.420388e-06           -0.298029         0.02216   \n",
       "SOCS2              0.562720  3.420388e-06                 NaN             NaN   \n",
       "...                     ...           ...                 ...             ...   \n",
       "CCDC6                   NaN           NaN                 NaN             NaN   \n",
       "SSR2                    NaN           NaN                 NaN             NaN   \n",
       "PSMB1                   NaN           NaN                 NaN             NaN   \n",
       "MRPL39                  NaN           NaN                 NaN             NaN   \n",
       "MAP3K11                 NaN           NaN                 NaN             NaN   \n",
       "\n",
       "            Correlation_Ovar  P_value_Ovar  Correlation_Brca  P_value_Brca  \\\n",
       "Comparison                                                                   \n",
       "PHLDA1                   NaN           NaN          0.364797      0.002164   \n",
       "GRB2                     NaN           NaN               NaN           NaN   \n",
       "CDH4                     NaN           NaN               NaN           NaN   \n",
       "PLA2G15                  NaN           NaN          0.274185      0.016061   \n",
       "SOCS2                    NaN           NaN               NaN           NaN   \n",
       "...                      ...           ...               ...           ...   \n",
       "CCDC6                    NaN           NaN               NaN           NaN   \n",
       "SSR2                     NaN           NaN               NaN           NaN   \n",
       "PSMB1                    NaN           NaN               NaN           NaN   \n",
       "MRPL39                   NaN           NaN               NaN           NaN   \n",
       "MAP3K11                  NaN           NaN               NaN           NaN   \n",
       "\n",
       "            Correlation_Luad  P_value_Luad  Correlation_hnscc  P_value_hnscc  \\\n",
       "Comparison                                                                     \n",
       "PHLDA1                   NaN           NaN           0.664271   8.888640e-12   \n",
       "GRB2               -0.302439      0.020631          -0.532341   3.320092e-06   \n",
       "CDH4                     NaN           NaN                NaN            NaN   \n",
       "PLA2G15                  NaN           NaN                NaN            NaN   \n",
       "SOCS2                    NaN           NaN                NaN            NaN   \n",
       "...                      ...           ...                ...            ...   \n",
       "CCDC6                    NaN           NaN                NaN            NaN   \n",
       "SSR2                     NaN           NaN                NaN            NaN   \n",
       "PSMB1                    NaN           NaN                NaN            NaN   \n",
       "MRPL39                   NaN           NaN                NaN            NaN   \n",
       "MAP3K11                  NaN           NaN                NaN            NaN   \n",
       "\n",
       "            Correlation_Lscc  P_value_Lscc  Correlation_Colon  P_value_Colon  \n",
       "Comparison                                                                    \n",
       "PHLDA1              0.713420  2.644826e-14                NaN            NaN  \n",
       "GRB2                     NaN           NaN                NaN            NaN  \n",
       "CDH4                     NaN           NaN                NaN            NaN  \n",
       "PLA2G15                  NaN           NaN                NaN            NaN  \n",
       "SOCS2               0.472624  1.417921e-02                NaN            NaN  \n",
       "...                      ...           ...                ...            ...  \n",
       "CCDC6                    NaN           NaN           0.287735       0.049589  \n",
       "SSR2                     NaN           NaN           0.301814       0.049589  \n",
       "PSMB1                    NaN           NaN           0.287839       0.049589  \n",
       "MRPL39                   NaN           NaN          -0.288110       0.049589  \n",
       "MAP3K11                  NaN           NaN           0.498761       0.049785  \n",
       "\n",
       "[6230 rows x 16 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FDR_sig = pd.read_csv(\"Pval_corr_table_Fig_2_return_sig.csv\")\n",
    "FDR_sig = FDR_sig.replace(to_replace ='_proteomics', value = '', regex = True) #Remove proteomics labeling\n",
    "FDR_sig= FDR_sig.drop(['Unnamed: 0'], axis=1)\n",
    "FDR_sig = FDR_sig.set_index(\"Comparison\")\n",
    "FDR_sig = FDR_sig[1:] #Drop EGFR \n",
    "FDR_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def HasPosNeg(row):\n",
    "    hasPos = False\n",
    "    hasNeg= False\n",
    "\n",
    "    for item in row:\n",
    "        if pd.isnull(item):\n",
    "            continue\n",
    "        if item < 0:\n",
    "            hasNeg = True\n",
    "        if item > 0:\n",
    "            hasPos = True\n",
    "            \n",
    "    if hasPos & hasNeg:\n",
    "        return True\n",
    "    return False\n",
    "\n",
    "def CountPosNeg(row):\n",
    "    hasPos = False\n",
    "    hasNeg= False\n",
    "    counter = 0\n",
    "    for item in row:\n",
    "        if pd.isnull(item):\n",
    "            continue\n",
    "        if item < -0:\n",
    "            hasNeg = True\n",
    "            counter += 1\n",
    "        if item > 0:\n",
    "            hasPos = True\n",
    "            counter += 1\n",
    "    return counter\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Subset data frame to include only trans genes that have opposite effects in different cancers by using apply function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "col = [\"Correlation_Gbm\",\"Correlation_kidney\",\"Correlation_Ovar\",\"Correlation_Brca\",\"Correlation_Luad\",\"Correlation_hnscc\",\"Correlation_Lscc\",\"Correlation_Colon\"]\n",
    "FDR_corr = FDR_sig[col]\n",
    "FDR_corr[\"Pos_Neg\"] = FDR_corr.apply(HasPosNeg, axis = 1)\n",
    "FDR_corr_False = FDR_corr[FDR_corr['Pos_Neg']==False]\n",
    "FDR_corr_False = FDR_corr_False.drop([\"Pos_Neg\"], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation_Gbm</th>\n",
       "      <th>Correlation_kidney</th>\n",
       "      <th>Correlation_Ovar</th>\n",
       "      <th>Correlation_Brca</th>\n",
       "      <th>Correlation_Luad</th>\n",
       "      <th>Correlation_hnscc</th>\n",
       "      <th>Correlation_Lscc</th>\n",
       "      <th>Correlation_Colon</th>\n",
       "      <th>Num_corr</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Comparison</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PHLDA1</th>\n",
       "      <td>0.816848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.364797</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.664271</td>\n",
       "      <td>0.713420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2</th>\n",
       "      <td>-0.610889</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.302439</td>\n",
       "      <td>-0.532341</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOCS2</th>\n",
       "      <td>0.562720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.472624</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLA</th>\n",
       "      <td>-0.550491</td>\n",
       "      <td>-0.298348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHLDA3</th>\n",
       "      <td>0.525883</td>\n",
       "      <td>0.432530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.763784</td>\n",
       "      <td>0.682034</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S100A14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370227</td>\n",
       "      <td>0.351373</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MOV10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.302787</td>\n",
       "      <td>0.299218</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP2B1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.298873</td>\n",
       "      <td>0.346409</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MFSD6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.295201</td>\n",
       "      <td>0.380169</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SART1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.292189</td>\n",
       "      <td>-0.306539</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1136 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Correlation_Gbm  Correlation_kidney  Correlation_Ovar  \\\n",
       "Comparison                                                          \n",
       "PHLDA1             0.816848                 NaN               NaN   \n",
       "GRB2              -0.610889                 NaN               NaN   \n",
       "SOCS2              0.562720                 NaN               NaN   \n",
       "GLA               -0.550491           -0.298348               NaN   \n",
       "PHLDA3             0.525883            0.432530               NaN   \n",
       "...                     ...                 ...               ...   \n",
       "S100A14                 NaN                 NaN               NaN   \n",
       "MOV10                   NaN                 NaN               NaN   \n",
       "ATP2B1                  NaN                 NaN               NaN   \n",
       "MFSD6                   NaN                 NaN               NaN   \n",
       "SART1                   NaN                 NaN               NaN   \n",
       "\n",
       "            Correlation_Brca  Correlation_Luad  Correlation_hnscc  \\\n",
       "Comparison                                                          \n",
       "PHLDA1              0.364797               NaN           0.664271   \n",
       "GRB2                     NaN         -0.302439          -0.532341   \n",
       "SOCS2                    NaN               NaN                NaN   \n",
       "GLA                      NaN               NaN                NaN   \n",
       "PHLDA3                   NaN               NaN           0.763784   \n",
       "...                      ...               ...                ...   \n",
       "S100A14                  NaN               NaN                NaN   \n",
       "MOV10                    NaN               NaN                NaN   \n",
       "ATP2B1                   NaN               NaN                NaN   \n",
       "MFSD6                    NaN               NaN                NaN   \n",
       "SART1                    NaN               NaN                NaN   \n",
       "\n",
       "            Correlation_Lscc  Correlation_Colon  Num_corr  \n",
       "Comparison                                                 \n",
       "PHLDA1              0.713420                NaN         4  \n",
       "GRB2                     NaN                NaN         3  \n",
       "SOCS2               0.472624                NaN         2  \n",
       "GLA                      NaN                NaN         2  \n",
       "PHLDA3              0.682034                NaN         4  \n",
       "...                      ...                ...       ...  \n",
       "S100A14             0.370227           0.351373         2  \n",
       "MOV10               0.302787           0.299218         2  \n",
       "ATP2B1              0.298873           0.346409         2  \n",
       "MFSD6               0.295201           0.380169         2  \n",
       "SART1              -0.292189          -0.306539         2  \n",
       "\n",
       "[1136 rows x 9 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FDR_corr_False[\"Num_corr\"] = FDR_corr_False.apply(CountPosNeg, axis = 1)\n",
    "FDR_corr_False.sort_values(by=['Num_corr'],ascending=False)\n",
    "FDR_corr_False = FDR_corr_False[FDR_corr_False['Num_corr'].isin([2,3,4,5,6])]\n",
    "FDR_corr_False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The manuscript mentions the total number of trans proteins that opposite effects in different cancers. Here is the derivation of that number "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of trans proteins with same effects in different cancers is 1136\n"
     ]
    }
   ],
   "source": [
    "same_sign_prot = FDR_corr_False.index.tolist()\n",
    "print(\"Total number of trans proteins with same effects in different cancers is \" + str(len(same_sign_prot)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run GSEA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Term</th>\n",
       "      <th>Overlap</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Adjusted P-value</th>\n",
       "      <th>Old P-value</th>\n",
       "      <th>Old Adjusted P-value</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>Combined Score</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Gene_set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Arrhythmogenic right ventricular cardiomyopath...</td>\n",
       "      <td>18/74</td>\n",
       "      <td>1.159284e-07</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.282451</td>\n",
       "      <td>68.392004</td>\n",
       "      <td>DSP;ITGA4;JUP;ITGB4;ACTN2;ITGA2;GJA1;CDH2;PKP2...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Metabolic pathways Homo sapiens hsa01100</td>\n",
       "      <td>114/1239</td>\n",
       "      <td>1.813751e-07</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.619889</td>\n",
       "      <td>25.145046</td>\n",
       "      <td>MSMO1;ENO1;NSDHL;NAMPT;ENPP1;ENPP3;ATP6V1E1;PT...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Regulation of actin cytoskeleton Homo sapiens ...</td>\n",
       "      <td>32/214</td>\n",
       "      <td>5.044867e-07</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.632618</td>\n",
       "      <td>38.172238</td>\n",
       "      <td>CYFIP2;NCKAP1;ITGB4;WAS;PIK3CD;IQGAP2;ITGAL;FG...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Biosynthesis of amino acids Homo sapiens hsa01230</td>\n",
       "      <td>17/74</td>\n",
       "      <td>6.249417e-07</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.044537</td>\n",
       "      <td>57.778675</td>\n",
       "      <td>PRPS1;ARG1;PGAM1;GPT;ENO1;PFKL;PKM;CBS;PSAT1;P...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Glycolysis / Gluconeogenesis Homo sapiens hsa0...</td>\n",
       "      <td>16/67</td>\n",
       "      <td>7.613663e-07</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.204330</td>\n",
       "      <td>59.231243</td>\n",
       "      <td>GPI;ACSS2;ADH1B;PGAM1;ENO1;HK2;LDHA;PFKL;PKM;P...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Bacterial invasion of epithelial cells Homo sa...</td>\n",
       "      <td>17/78</td>\n",
       "      <td>1.377343e-06</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.837125</td>\n",
       "      <td>51.783367</td>\n",
       "      <td>WAS;FN1;PIK3CD;SEPT1;PIK3CG;SEPT2;CD2AP;SEPT9;...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>B cell receptor signaling pathway Homo sapiens...</td>\n",
       "      <td>16/73</td>\n",
       "      <td>2.588556e-06</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.858769</td>\n",
       "      <td>49.640789</td>\n",
       "      <td>CR2;JUN;SYK;INPPL1;PIK3CD;NFATC2;PIK3CG;VAV1;P...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Carbon metabolism Homo sapiens hsa01200</td>\n",
       "      <td>20/113</td>\n",
       "      <td>5.248171e-06</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.116041</td>\n",
       "      <td>37.883681</td>\n",
       "      <td>GPI;PRPS1;H6PD;ACSS2;ECHS1;PGAM1;GPT;ENO1;HK2;...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>T cell receptor signaling pathway Homo sapiens...</td>\n",
       "      <td>19/104</td>\n",
       "      <td>5.547202e-06</td>\n",
       "      <td>0.000181</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.216414</td>\n",
       "      <td>38.925738</td>\n",
       "      <td>JUN;NFATC2;PIK3CD;CD3G;PIK3CG;VAV1;ZAP70;CD4;P...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Axon guidance Homo sapiens hsa04360</td>\n",
       "      <td>21/127</td>\n",
       "      <td>9.304350e-06</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.911168</td>\n",
       "      <td>33.725962</td>\n",
       "      <td>EPHA4;SEMA4A;SEMA4D;SEMA4B;LIMK1;GNAI3;NFATC2;...</td>\n",
       "      <td>KEGG_2016</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Term   Overlap       P-value  \\\n",
       "0  Arrhythmogenic right ventricular cardiomyopath...     18/74  1.159284e-07   \n",
       "1           Metabolic pathways Homo sapiens hsa01100  114/1239  1.813751e-07   \n",
       "2  Regulation of actin cytoskeleton Homo sapiens ...    32/214  5.044867e-07   \n",
       "3  Biosynthesis of amino acids Homo sapiens hsa01230     17/74  6.249417e-07   \n",
       "4  Glycolysis / Gluconeogenesis Homo sapiens hsa0...     16/67  7.613663e-07   \n",
       "5  Bacterial invasion of epithelial cells Homo sa...     17/78  1.377343e-06   \n",
       "6  B cell receptor signaling pathway Homo sapiens...     16/73  2.588556e-06   \n",
       "7            Carbon metabolism Homo sapiens hsa01200    20/113  5.248171e-06   \n",
       "8  T cell receptor signaling pathway Homo sapiens...    19/104  5.547202e-06   \n",
       "9                Axon guidance Homo sapiens hsa04360    21/127  9.304350e-06   \n",
       "\n",
       "   Adjusted P-value  Old P-value  Old Adjusted P-value  Odds Ratio  \\\n",
       "0          0.000034            0                     0    4.282451   \n",
       "1          0.000027            0                     0    1.619889   \n",
       "2          0.000049            0                     0    2.632618   \n",
       "3          0.000046            0                     0    4.044537   \n",
       "4          0.000045            0                     0    4.204330   \n",
       "5          0.000067            0                     0    3.837125   \n",
       "6          0.000108            0                     0    3.858769   \n",
       "7          0.000192            0                     0    3.116041   \n",
       "8          0.000181            0                     0    3.216414   \n",
       "9          0.000273            0                     0    2.911168   \n",
       "\n",
       "   Combined Score                                              Genes  \\\n",
       "0       68.392004  DSP;ITGA4;JUP;ITGB4;ACTN2;ITGA2;GJA1;CDH2;PKP2...   \n",
       "1       25.145046  MSMO1;ENO1;NSDHL;NAMPT;ENPP1;ENPP3;ATP6V1E1;PT...   \n",
       "2       38.172238  CYFIP2;NCKAP1;ITGB4;WAS;PIK3CD;IQGAP2;ITGAL;FG...   \n",
       "3       57.778675  PRPS1;ARG1;PGAM1;GPT;ENO1;PFKL;PKM;CBS;PSAT1;P...   \n",
       "4       59.231243  GPI;ACSS2;ADH1B;PGAM1;ENO1;HK2;LDHA;PFKL;PKM;P...   \n",
       "5       51.783367  WAS;FN1;PIK3CD;SEPT1;PIK3CG;SEPT2;CD2AP;SEPT9;...   \n",
       "6       49.640789  CR2;JUN;SYK;INPPL1;PIK3CD;NFATC2;PIK3CG;VAV1;P...   \n",
       "7       37.883681  GPI;PRPS1;H6PD;ACSS2;ECHS1;PGAM1;GPT;ENO1;HK2;...   \n",
       "8       38.925738  JUN;NFATC2;PIK3CD;CD3G;PIK3CG;VAV1;ZAP70;CD4;P...   \n",
       "9       33.725962  EPHA4;SEMA4A;SEMA4D;SEMA4B;LIMK1;GNAI3;NFATC2;...   \n",
       "\n",
       "    Gene_set  \n",
       "0  KEGG_2016  \n",
       "1  KEGG_2016  \n",
       "2  KEGG_2016  \n",
       "3  KEGG_2016  \n",
       "4  KEGG_2016  \n",
       "5  KEGG_2016  \n",
       "6  KEGG_2016  \n",
       "7  KEGG_2016  \n",
       "8  KEGG_2016  \n",
       "9  KEGG_2016  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "same_sign_enr = gp.enrichr(gene_list = same_sign_prot, description='Tumor_partition', gene_sets='KEGG_2016', \n",
    "                       outdir='test/enrichr_KEGG')\n",
    "same_sign_enr.res2d.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "list.remove(x): x not in list",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-8-695d213258e9>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     13\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m \u001b[1;31m#remove = [\"CR2\",\"NCK1\",\"CARD11\",\"INPPL1\",\"JUN\",\"KRAS\"]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 15\u001b[1;33m \u001b[0mimmune\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mremove\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"JUN\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     16\u001b[0m \u001b[0mimmune\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mremove\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"KRAS\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m \u001b[0mimmune\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mremove\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"CR2\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: list.remove(x): x not in list"
     ]
    }
   ],
   "source": [
    "#get just the clotting cascade genes and add urokinase genes \n",
    "same_sign_df = same_sign_enr.res2d\n",
    "bcell = same_sign_df.iloc[6,9]\n",
    "bcell = bcell.split(';')\n",
    "tcell = same_sign_df.iloc[8,9]\n",
    "tcell = tcell.split(\";\")\n",
    "immune_both =  bcell + tcell\n",
    "immune = set(immune_both)\n",
    "immune = list(immune)\n",
    "btcell_apm = ([\"CD5\",\"CD8A\",\"BLK\",\"BLNK\",\"CD79A\",\"LYN\",\"CD3E\",\"HLA-DMA\",\"HLA-DMB\"])\n",
    "\n",
    "immune= immune + btcell_apm\n",
    "\n",
    "#remove = [\"CR2\",\"NCK1\",\"CARD11\",\"INPPL1\",\"JUN\",\"KRAS\"]\n",
    "immune.remove(\"JUN\")\n",
    "immune.remove(\"KRAS\")\n",
    "immune.remove(\"CR2\")\n",
    "immune.remove(\"NCK1\")\n",
    "immune.remove(\"INPPL1\")\n",
    "immune.remove(\"CARD11\")\n",
    "immune"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 3 Make Data frame for figure 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sig PTEN # CD2? CHUK\n",
    "t_only = ['CD3E', 'CD3G', 'CD4', 'CD8A', 'GRAP2', 'FYB1', 'PRKCQ'] \n",
    "b_only = ['PPP3CC', 'CD79A', 'SYK', 'BTK', 'LYN', 'BLNK', 'BLK', 'HLA-DMA', 'HLA-DMB']\n",
    "both = ['CD2', 'CD5', 'GRB2', 'LCK', 'ZAP70', 'VAV1', 'PTPN6', 'PTPRC', 'PIK3CD', 'PIK3CG', 'INPP5D',\n",
    "        'PLCG1', 'PLCG2', 'NFATC2', 'PRKCB', 'CHUK', 'NFKB2', 'DOCK2', 'RAC2', 'ELMO1', 'WAS']\n",
    "immune = t_only + b_only + both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Correlation</th>\n",
       "      <th>P_value</th>\n",
       "      <th>Cancer Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>-0.610889</td>\n",
       "      <td>6.729990e-08</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>WAS</td>\n",
       "      <td>-0.501918</td>\n",
       "      <td>4.961122e-05</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>PLCG2</td>\n",
       "      <td>-0.472453</td>\n",
       "      <td>1.145235e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>SYK</td>\n",
       "      <td>-0.470849</td>\n",
       "      <td>1.202510e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>CD4</td>\n",
       "      <td>-0.467479</td>\n",
       "      <td>1.362923e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7781</th>\n",
       "      <td>PRKCQ</td>\n",
       "      <td>-0.284462</td>\n",
       "      <td>3.293176e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7876</th>\n",
       "      <td>RAC2</td>\n",
       "      <td>-0.277100</td>\n",
       "      <td>3.855935e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8461</th>\n",
       "      <td>NFATC2</td>\n",
       "      <td>-0.297428</td>\n",
       "      <td>4.438700e-02</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8573</th>\n",
       "      <td>PRKCB</td>\n",
       "      <td>0.395927</td>\n",
       "      <td>6.516094e-03</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8693</th>\n",
       "      <td>NFKB2</td>\n",
       "      <td>0.357365</td>\n",
       "      <td>1.247990e-02</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Comparison  Correlation       P_value    Cancer Type\n",
       "1          GRB2    -0.610889  6.729990e-08            GBM\n",
       "24          WAS    -0.501918  4.961122e-05            GBM\n",
       "72        PLCG2    -0.472453  1.145235e-04            GBM\n",
       "75          SYK    -0.470849  1.202510e-04            GBM\n",
       "82          CD4    -0.467479  1.362923e-04            GBM\n",
       "...         ...          ...           ...            ...\n",
       "7781      PRKCQ    -0.284462  3.293176e-02  Head and Neck\n",
       "7876       RAC2    -0.277100  3.855935e-02  Head and Neck\n",
       "8461     NFATC2    -0.297428  4.438700e-02           Lscc\n",
       "8573      PRKCB     0.395927  6.516094e-03          Colon\n",
       "8693      NFKB2     0.357365  1.247990e-02          Colon\n",
       "\n",
       "[87 rows x 4 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Get append version of the df with all cancer type, fdr sig trans results\n",
    "df_FDR_append = pd.read_csv(\"Pval_corr_table_Fig_2_return_sig_append.csv\")\n",
    "df_FDR_append = df_FDR_append.drop(['Unnamed: 0'], axis=1)\n",
    "\n",
    "#Remove Proteomics labeling \n",
    "df_FDR_append = df_FDR_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_FDR_append= df_FDR_append[df_FDR_append.Comparison.isin(immune)]\n",
    "df_FDR_append['Cancer Type'].unique()\n",
    "df_FDR_append"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set add new column to be unique index and order the new index. This way genes will be grouped by coagulation factors, regulators, and urokinase genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Correlation</th>\n",
       "      <th>P_value</th>\n",
       "      <th>Cancer Type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CD3E Head and Neck</th>\n",
       "      <td>CD3E</td>\n",
       "      <td>-0.339994</td>\n",
       "      <td>0.020794</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD3G Head and Neck</th>\n",
       "      <td>CD3G</td>\n",
       "      <td>-0.381032</td>\n",
       "      <td>0.003211</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD3G Luad</th>\n",
       "      <td>CD3G</td>\n",
       "      <td>-0.333706</td>\n",
       "      <td>0.040933</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD4 GBM</th>\n",
       "      <td>CD4</td>\n",
       "      <td>-0.467479</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD4 Head and Neck</th>\n",
       "      <td>CD4</td>\n",
       "      <td>-0.417232</td>\n",
       "      <td>0.000566</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ELMO1 Luad</th>\n",
       "      <td>ELMO1</td>\n",
       "      <td>-0.410098</td>\n",
       "      <td>0.000810</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WAS GBM</th>\n",
       "      <td>WAS</td>\n",
       "      <td>-0.501918</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WAS Head and Neck</th>\n",
       "      <td>WAS</td>\n",
       "      <td>-0.503427</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WAS Kidney</th>\n",
       "      <td>WAS</td>\n",
       "      <td>-0.279814</td>\n",
       "      <td>0.035337</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WAS Luad</th>\n",
       "      <td>WAS</td>\n",
       "      <td>-0.337909</td>\n",
       "      <td>0.008234</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Comparison  Correlation   P_value    Cancer Type\n",
       "Index                                                              \n",
       "CD3E Head and Neck       CD3E    -0.339994  0.020794  Head and Neck\n",
       "CD3G Head and Neck       CD3G    -0.381032  0.003211  Head and Neck\n",
       "CD3G Luad                CD3G    -0.333706  0.040933           Luad\n",
       "CD4 GBM                   CD4    -0.467479  0.000136            GBM\n",
       "CD4 Head and Neck         CD4    -0.417232  0.000566  Head and Neck\n",
       "...                       ...          ...       ...            ...\n",
       "ELMO1 Luad              ELMO1    -0.410098  0.000810           Luad\n",
       "WAS GBM                   WAS    -0.501918  0.000050            GBM\n",
       "WAS Head and Neck         WAS    -0.503427  0.000016  Head and Neck\n",
       "WAS Kidney                WAS    -0.279814  0.035337         Kidney\n",
       "WAS Luad                  WAS    -0.337909  0.008234           Luad\n",
       "\n",
       "[87 rows x 4 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "df_FDR_append[\"Index\"] = df_FDR_append[\"Comparison\"] + \" \" + df_FDR_append[\"Cancer Type\"]\n",
    "df_FDR_append = df_FDR_append.set_index(\"Index\")\n",
    "\n",
    "ordered_list = []\n",
    "cancer_list = ['Breast', 'Colon', 'GBM', 'Head and Neck', 'Kidney', 'Lscc', 'Luad']\n",
    "for gene in immune:\n",
    "    for cancer in cancer_list:\n",
    "        ordered_list.append(gene+' '+cancer)\n",
    "\n",
    "ordered_df = df_FDR_append.reindex(ordered_list)\n",
    "ordered_df = ordered_df.loc[ordered_df['P_value'] < 0.05] # Keep sig \n",
    "                                    \n",
    "ordered_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 4: Plot Figure 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"3307\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"3307\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"3307\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '3307' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"3307\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"3307\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"3307\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '3307' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"3307\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"411956d0-ef38-4e23-a610-46f33e8f2523\" data-root-id=\"3308\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"82e3c20c-52b8-442b-be01-fbc6ebf85e00\":{\"roots\":{\"references\":[{\"attributes\":{\"children\":[{\"id\":\"3259\"},{\"id\":\"3285\"}]},\"id\":\"3308\",\"type\":\"Row\"},{\"attributes\":{},\"id\":\"3267\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"axis\":{\"id\":\"3295\"},\"dimension\":1,\"ticker\":null},\"id\":\"3297\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"3409\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"factors\":[\"5.0e-02\",\"1.0e-04\",\"1.0e-08\"]},\"id\":\"3284\",\"type\":\"FactorRange\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"3272\"}]},\"id\":\"3273\",\"type\":\"Toolbar\"},{\"attributes\":{\"callback\":null},\"id\":\"3298\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"3408\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"high\":1.5,\"low\":1.5,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"3282\",\"type\":\"LinearColorMapper\"},{\"attributes\":{},\"id\":\"3414\",\"type\":\"Selection\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"3298\"}]},\"id\":\"3299\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"3405\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"factors\":[\"CD3E\",\"CD3G\",\"CD4\",\"CD8A\",\"GRAP2\",\"FYB1\",\"PRKCQ\",\"PPP3CC\",\"CD79A\",\"SYK\",\"BTK\",\"LYN\",\"BLNK\",\"BLK\",\"HLA-DMA\",\"HLA-DMB\",\"CD2\",\"CD5\",\"GRB2\",\"LCK\",\"ZAP70\",\"VAV1\",\"PTPN6\",\"PTPRC\",\"PIK3CD\",\"PIK3CG\",\"INPP5D\",\"PLCG1\",\"PLCG2\",\"NFATC2\",\"PRKCB\",\"NFKB2\",\"DOCK2\",\"RAC2\",\"ELMO1\",\"WAS\"]},\"id\":\"3257\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"3403\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"source\":{\"id\":\"3275\"}},\"id\":\"3280\",\"type\":\"CDSView\"},{\"attributes\":{\"data_source\":{\"id\":\"3301\"},\"glyph\":{\"id\":\"3303\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"3304\"},\"selection_glyph\":null,\"view\":{\"id\":\"3306\"}},\"id\":\"3305\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"formatter\":{\"id\":\"3403\"},\"major_label_orientation\":1.5707963267948966,\"major_label_text_font_size\":\"12pt\",\"ticker\":{\"id\":\"3267\"}},\"id\":\"3266\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"factors\":[\"Luad\",\"Lscc\",\"Kidney\",\"Head and Neck\",\"GBM\",\"Colon\",\"Breast\"]},\"id\":\"3258\",\"type\":\"FactorRange\"},{\"attributes\":{\"fill_color\":{\"field\":\"Correlation\",\"transform\":{\"id\":\"3256\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Comparison\"},\"y\":{\"field\":\"Cancer Type\"}},\"id\":\"3277\",\"type\":\"Scatter\"},{\"attributes\":{\"data_source\":{\"id\":\"3275\"},\"glyph\":{\"id\":\"3277\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"3278\"},\"selection_glyph\":null,\"view\":{\"id\":\"3280\"}},\"id\":\"3279\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"text\":\"\"},\"id\":\"3400\",\"type\":\"Title\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Correlation\",\"transform\":{\"id\":\"3282\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"3304\",\"type\":\"Scatter\"},{\"attributes\":{\"below\":[{\"id\":\"3266\"}],\"center\":[{\"id\":\"3268\"},{\"id\":\"3271\"}],\"left\":[{\"id\":\"3269\"}],\"plot_height\":500,\"plot_width\":1000,\"renderers\":[{\"id\":\"3279\"}],\"right\":[{\"id\":\"3281\"}],\"title\":{\"id\":\"3400\"},\"toolbar\":{\"id\":\"3273\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"3257\"},\"x_scale\":{\"id\":\"3262\"},\"y_range\":{\"id\":\"3258\"},\"y_scale\":{\"id\":\"3264\"}},\"id\":\"3259\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Correlation\",\"transform\":{\"id\":\"3256\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Comparison\"},\"y\":{\"field\":\"Cancer Type\"}},\"id\":\"3278\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"3264\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"source\":{\"id\":\"3301\"}},\"id\":\"3306\",\"type\":\"CDSView\"},{\"attributes\":{},\"id\":\"3262\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"color_mapper\":{\"id\":\"3256\"},\"formatter\":{\"id\":\"3408\"},\"location\":[0,0],\"ticker\":{\"id\":\"3407\"}},\"id\":\"3281\",\"type\":\"ColorBar\"},{\"attributes\":{\"formatter\":{\"id\":\"3411\"},\"major_label_text_font_size\":\"11pt\",\"ticker\":{\"id\":\"3296\"}},\"id\":\"3295\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"3407\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"3293\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"data\":{\"Cancer Type\":[\"Head and Neck\",\"Head and Neck\",\"Luad\",\"GBM\",\"Head and Neck\",\"Luad\",\"Head and Neck\",\"Kidney\",\"GBM\",\"Head and Neck\",\"Luad\",\"GBM\",\"Head and Neck\",\"Luad\",\"Head and Neck\",\"Luad\",\"Head and Neck\",\"GBM\",\"Head and Neck\",\"GBM\",\"Head and Neck\",\"Luad\",\"Head and Neck\",\"GBM\",\"Luad\",\"GBM\",\"GBM\",\"Head and Neck\",\"Head and Neck\",\"Luad\",\"Head and Neck\",\"GBM\",\"Head and Neck\",\"Luad\",\"Head and Neck\",\"Luad\",\"Head and Neck\",\"Kidney\",\"Luad\",\"GBM\",\"Head and Neck\",\"Kidney\",\"GBM\",\"Head and Neck\",\"Luad\",\"GBM\",\"Head and Neck\",\"GBM\",\"Luad\",\"GBM\",\"Head and Neck\",\"GBM\",\"Head and Neck\",\"Kidney\",\"Luad\",\"Breast\",\"GBM\",\"Kidney\",\"Breast\",\"GBM\",\"Head and Neck\",\"Kidney\",\"Luad\",\"GBM\",\"Head and Neck\",\"Lscc\",\"Colon\",\"Head and Neck\",\"Kidney\",\"Luad\",\"Colon\",\"GBM\",\"Luad\",\"GBM\",\"Head and Neck\",\"Kidney\",\"Luad\",\"GBM\",\"Head and Neck\",\"Luad\",\"GBM\",\"Head and Neck\",\"Luad\",\"GBM\",\"Head and Neck\",\"Kidney\",\"Luad\"],\"Comparison\":[\"CD3E\",\"CD3G\",\"CD3G\",\"CD4\",\"CD4\",\"CD8A\",\"GRAP2\",\"GRAP2\",\"FYB1\",\"FYB1\",\"FYB1\",\"PRKCQ\",\"PRKCQ\",\"PRKCQ\",\"PPP3CC\",\"PPP3CC\",\"CD79A\",\"SYK\",\"SYK\",\"BTK\",\"BTK\",\"BTK\",\"LYN\",\"BLNK\",\"BLK\",\"HLA-DMA\",\"HLA-DMB\",\"HLA-DMB\",\"CD2\",\"CD2\",\"CD5\",\"GRB2\",\"GRB2\",\"GRB2\",\"LCK\",\"LCK\",\"ZAP70\",\"ZAP70\",\"ZAP70\",\"VAV1\",\"VAV1\",\"VAV1\",\"PTPN6\",\"PTPN6\",\"PTPN6\",\"PTPRC\",\"PTPRC\",\"PIK3CD\",\"PIK3CD\",\"PIK3CG\",\"PIK3CG\",\"INPP5D\",\"INPP5D\",\"INPP5D\",\"INPP5D\",\"PLCG1\",\"PLCG1\",\"PLCG1\",\"PLCG2\",\"PLCG2\",\"PLCG2\",\"PLCG2\",\"PLCG2\",\"NFATC2\",\"NFATC2\",\"NFATC2\",\"PRKCB\",\"PRKCB\",\"PRKCB\",\"PRKCB\",\"NFKB2\",\"NFKB2\",\"NFKB2\",\"DOCK2\",\"DOCK2\",\"DOCK2\",\"DOCK2\",\"RAC2\",\"RAC2\",\"RAC2\",\"ELMO1\",\"ELMO1\",\"ELMO1\",\"WAS\",\"WAS\",\"WAS\",\"WAS\"],\"Correlation\":{\"__ndarray__\":\"DB6XHHfC1b81uFzO1GLYv45JW2puW9W/YVKSLS3r3b8jnPCv77Pav72yODj1t9K/eLBLqt9l279X2H7K3YnTv6BXau0hw92/VrnKiXW7179Kqb0TG8XRv23VvHPiGtW/bsoYAaI00r9WSSmhYsPav0WGL0yq/Na/e3ebBlaq278G/hlc5XfXv/YZa+1iIt6/+Qs4t3IZ078uIV2PqBjbvysmIG2iHN2/RU5V3cvK0b/E6xrDOfvXv0t5O9yM19a/k/MekYpk17/ofRLRWOrXv2OV9f/QSdy/Apw7ZFGY0r/x1htzQVPZv5isPzq98tK/BsrI2qJn2b+WT/NUZ4zjvwBLwnHwCOG/KpFDSSdb0787glW7iTHVv/YNZ8a4vNe/0g2eui5A2r+WPjcsaavRv0NKxBJ+ndi/cFKT5vyd3L/MNEU9LSDcvzhPEpHLSNK/SQrkIqW62b9Ztm0dFlPav1rrNXewzte/l+Ckj4Iy2794z4Bz3cDZv1zBpLZI192/4GyOdY471b9M9C4FFjjZvwWFlkVMwtm/KctljaEf2r92dMul6incv20nfcWoZtG/DktVFFtn0b8wGFf+NT3WPzFaZqExZtE/+JNw/VFn0r8CKuuE8zDRP2BRGRWrPN6/jJ+ECLZA2L9x8O9w4EnUv8YGH0cvH9q/ZLySgGUg2b/0ilSIRkjdvxfDE/cPCdO/t+XY3t9W2T9wRTkWjhLdv6JD5HLnqNS/kbqIgKub3L9lg8ggE9/WPwwVX3nDg9q/L5I9zdMa2r9ijoM7IHLbvxX5kwdZKty/ULAxw+TM0r8qYGbBPqvSv91QrLsvjtS/Row7xwG80b+s7x4IXWXWv6x15spiU9S/Ouy6aAXb17/G1lJ/DD/av+62E2q2D+C/51cBqhIc4L/F4iZzeOjRv9m7MMNLoNW/\",\"dtype\":\"float64\",\"shape\":[87]},\"Index\":[\"CD3E Head and Neck\",\"CD3G Head and Neck\",\"CD3G Luad\",\"CD4 GBM\",\"CD4 Head and Neck\",\"CD8A Luad\",\"GRAP2 Head and Neck\",\"GRAP2 Kidney\",\"FYB1 GBM\",\"FYB1 Head and Neck\",\"FYB1 Luad\",\"PRKCQ GBM\",\"PRKCQ Head and Neck\",\"PRKCQ Luad\",\"PPP3CC Head and Neck\",\"PPP3CC Luad\",\"CD79A Head and Neck\",\"SYK GBM\",\"SYK Head and Neck\",\"BTK GBM\",\"BTK Head and Neck\",\"BTK Luad\",\"LYN Head and Neck\",\"BLNK GBM\",\"BLK Luad\",\"HLA-DMA GBM\",\"HLA-DMB GBM\",\"HLA-DMB Head and Neck\",\"CD2 Head and Neck\",\"CD2 Luad\",\"CD5 Head and Neck\",\"GRB2 GBM\",\"GRB2 Head and Neck\",\"GRB2 Luad\",\"LCK Head and Neck\",\"LCK Luad\",\"ZAP70 Head and Neck\",\"ZAP70 Kidney\",\"ZAP70 Luad\",\"VAV1 GBM\",\"VAV1 Head and Neck\",\"VAV1 Kidney\",\"PTPN6 GBM\",\"PTPN6 Head and Neck\",\"PTPN6 Luad\",\"PTPRC GBM\",\"PTPRC Head and Neck\",\"PIK3CD GBM\",\"PIK3CD Luad\",\"PIK3CG GBM\",\"PIK3CG Head and Neck\",\"INPP5D GBM\",\"INPP5D Head and Neck\",\"INPP5D Kidney\",\"INPP5D Luad\",\"PLCG1 Breast\",\"PLCG1 GBM\",\"PLCG1 Kidney\",\"PLCG2 Breast\",\"PLCG2 GBM\",\"PLCG2 Head and Neck\",\"PLCG2 Kidney\",\"PLCG2 Luad\",\"NFATC2 GBM\",\"NFATC2 Head and Neck\",\"NFATC2 Lscc\",\"PRKCB Colon\",\"PRKCB Head and Neck\",\"PRKCB Kidney\",\"PRKCB Luad\",\"NFKB2 Colon\",\"NFKB2 GBM\",\"NFKB2 Luad\",\"DOCK2 GBM\",\"DOCK2 Head and Neck\",\"DOCK2 Kidney\",\"DOCK2 Luad\",\"RAC2 GBM\",\"RAC2 Head and Neck\",\"RAC2 Luad\",\"ELMO1 GBM\",\"ELMO1 Head and Neck\",\"ELMO1 Luad\",\"WAS GBM\",\"WAS Head and Neck\",\"WAS Kidney\",\"WAS Luad\"],\"P_value\":{\"__ndarray__\":\"Tawh6gdLlT9Pb+lHfk1qP4DqXnYm9aQ/YWmEIzbdIT9ZwATwlo5CPwdCWEGFb5o/RdziP8eINz9mvZaYmcmVPwyf0XtLrCI/ZNbIDUpIZj8YCKcKckeiP9aIbnaMmZA/rPChqW7coD/y8MWXhZdEP21/ak+uC3A/vxw4dsPiNz/Qzzx26mlpP7xWTa3nhR8/yd7DLMbXlz8Am51dQXhFP2n1yarnoyA/NsrYhcovoj9n2L0mHlljP9pyAE0Yv3Q/fETI/VP0gT+rpYm5F7Z0P56rH2kjyDY/K7EYskoSnT/wy9mFeRNUP97Nd1cIVZg/xvToiwHxWT+8gzFP0BByPsPS2sPX2cs+5iYePUgglT8RQz45B3iDP41I7r/FzWo/HXnl7hSfRz8bDDwNKKGjP2/saukEJWE/cfffZlL4Mj+llNVbXa4tPxQFzFPx+p4/GAYEtBlZVD9B68DBaH9GP/cBnnFL4Wk/5+J8XmFORD8Z0GFchAJQPyWY7u2IViI/Ub0t0HvRgz86RMcPRUBaPxnQYVyEAlA/V3Y+0KCIUD+1JOqBDP8sP1xvRoVOaqU/LhOOqBWEpD/noGm/0FhcP6Js9nRspqc/bvBySC8+oz+YWvDymiyTP9SUHV2LBR4/hzVESkUOYT9zC/ng1wmMP+0hFmvReEw/HwNkpICMWz8yee8jKCYdPzeTnL3kuaY/KpBIj56wej9RCOQOFP4gPxQxNJrHPYg/gV21yQMaKT8xIb9ND4+JP6xjoWD+/0o/SX0OlkCmTD8NV5BerxJCP7Uk6oEM/yw/DgfuEQwsmT9V0UOMDa6aPwTkhF82pYs/aalsIw2+oz/l+0Flf094P+lWgPWlxo4/l+xJz5rJZD8kzucmVYtKP9/bWp20Ago/3AjCN6hl8D6xAG1NqheiP/Uz7e413YA/\",\"dtype\":\"float64\",\"shape\":[87]},\"size\":{\"__ndarray__\":\"nje5phA9J0AJd1n/RTkxQLC4Q4HILCNAZix/gb6zOkCe2RRp2G02QNgIRTnI8CVAVekISLLLN0CsHCnXzhknQNJT4Gq7kTpABG5se6q4MUDwaBC80v4jQEQolkaEuyhAa5APF9t6JEBrvv9G7x02QEaSnlyCoDBA2/mifE/AN0BqI9jOrVMxQIF84PzpEztAAf5r7FaPJkAyyZlN3/01QINmOYw/6jpAHt+ZdpsGJEDgwKvCGyUyQLm/BQpcti9AmABJBrNrLEC+mAQa97gvQNnDOnKn5DdAcxngYMteJUBITnWXFB00QNJOeDsjcCZAfGZ3fDtYM0A22I7JasVIQAug3ORb7EJA7P5eiChJJ0DSHi0eUO8rQEp55KrHKjFAA3q5goW0NUClcft8V5EjQImYOuv3gTJAKO2H1kZxOEBwovM10S05QFBpDJEd/SRAkkvO+b4SNEDEgqxc9Nk1QF3lmuS2RTFA2VtTSKooNkCoETgH5so0QEJYPqainzpAY/bSGf3TK0CCf8/DHk8zQKgROAfmyjRAlwliqSuyNEDu17Wqvj85QHInWmyaCyNApdSEvH9NI0Cp86fBIBQzQNamTYcicyJAoimSQuWvI0BlXceMD94nQDCzR45kOTtAMNxDmfWFMkC44GxyCr8pQJEPjOYVJTVARLRVmQ8qM0CMgg7ZC1A7QIbpUI1YsCJAn0+B9WwzLkAJb7syKdo6QMwIB2mOnipAsm5le4SuOUDimlDoQ00qQBKGtX/aTTVASDQiKlAgNUCMeeViJII2QO7Xtaq+PzlAeqyRJwM8JkC3g+aPp+IlQABmNDW61ClAlM1h9IiII0Cif3fp2cIuQIjTSXjzLylAbiEdof/tMUCqUeUu7Vo1QKxIdpffuz1AvCi7QkiZQEDtA01elg4kQA4PQ0/tyyxA\",\"dtype\":\"float64\",\"shape\":[87]},\"size2\":{\"__ndarray__\":\"fp+hMxb8DkAM9CH/B/cWQED2BFcLkQlAmR2qqynNIUB9d8Y2y+cdQHW2BvcKQQ1AcYy2CkO6H0A6JozJE80OQIzi6vF8tiFAWj07T+OgF0Dqi8D6GKkKQC1wudkCfRBAj8AUdHlOC0Dk/VQJP30dQF4YfntYKxZAeaIuphSrH0Dj2coTPRoXQFaolahGDSJArFLl5XMUDkCYYSISf1IdQALv0LJ/8SFAKNR383mzCkAqAeVYejEYQCbVA1w9JBVAZQCGWXfyEkDUZVgR+iUVQMyvo+2J2x9AmcwqgWR+DEC2vfHJcNEaQBhpoKTZ6g1A+91J+/nKGUDPOl+GnIMwQLkqJjElOylA5lMpCzYMD0A3v8i+ip8SQGJM2+O04xZABPihA7LwHEAxQk9RHxcKQGHL+I5KrRhAxUgFj4RLIECgwfcjNskgQGuMEGwn/AtAbQ8TTanDGkAFWTsmmyIdQHzcI4aeBxdAd3rEtY2LHUA2wkq0MrkbQCyQKcTBvyFAQqSMZlONEkCuVL+v074ZQDbCSrQyuRtAH2ItNzqYG0BJ5SPHKdUgQO2JzeXNZAlAhxuxUKq8CUCM79+sK3AZQMgzEl+DmQhALuLCWNw/CkAxJ19mv9IPQCAihQlDJiJA6npazJyyGEB7QPP2BioRQGy/uohyMRxABZvHIWqNGUCzAV87XTUiQAjia7wg6whAaoqro0giFEAGStIhxuYhQN0FWpsJvxFAzEnu/AIfIUCXvDXw14gRQMOynP/NZxxAYPCCjRUrHEC7TIcu2wIeQEnlI8cp1SBAopAXilmlDUD0BN6/NC4NQABEeCN8OBFAxbzXRWELCkDB/6Sb5oEUQLA3MVCiyhBAkyx8gf/nF0COF4c+PHkcQHMwpA+V0iNA+uBOrrUhJkCRWhHTHb4KQF9f1zSeMhNA\",\"dtype\":\"float64\",\"shape\":[87]}},\"selected\":{\"id\":\"3414\"},\"selection_policy\":{\"id\":\"3413\"}},\"id\":\"3275\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"axis\":{\"id\":\"3292\"},\"ticker\":null},\"id\":\"3294\",\"type\":\"Grid\"},{\"attributes\":{\"high\":0.6108891162098071,\"low\":-0.6108891162098071,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"3256\",\"type\":\"LinearColorMapper\"},{\"attributes\":{},\"id\":\"3416\",\"type\":\"Selection\"},{\"attributes\":{},\"id\":\"3288\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"text\":\"\"},\"id\":\"3402\",\"type\":\"Title\"},{\"attributes\":{\"factors\":[\"\"]},\"id\":\"3283\",\"type\":\"FactorRange\"},{\"attributes\":{\"fill_color\":{\"field\":\"Correlation\",\"transform\":{\"id\":\"3282\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"3303\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"3411\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{},\"id\":\"3413\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"3290\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"callback\":null},\"id\":\"3272\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"3415\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"3296\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"axis\":{\"id\":\"3269\"},\"dimension\":1,\"ticker\":null},\"id\":\"3271\",\"type\":\"Grid\"},{\"attributes\":{\"below\":[{\"id\":\"3292\"}],\"center\":[{\"id\":\"3294\"},{\"id\":\"3297\"}],\"left\":[{\"id\":\"3295\"}],\"plot_height\":200,\"plot_width\":120,\"renderers\":[{\"id\":\"3305\"}],\"title\":{\"id\":\"3402\"},\"toolbar\":{\"id\":\"3299\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"3283\"},\"x_scale\":{\"id\":\"3288\"},\"y_range\":{\"id\":\"3284\"},\"y_scale\":{\"id\":\"3290\"}},\"id\":\"3285\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{},\"id\":\"3270\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"formatter\":{\"id\":\"3405\"},\"major_label_text_font_size\":\"12pt\",\"ticker\":{\"id\":\"3270\"}},\"id\":\"3269\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"axis_label\":\"FDR_P-Values\",\"formatter\":{\"id\":\"3409\"},\"major_label_orientation\":1.5707963267948966,\"major_label_text_font_size\":\"11pt\",\"ticker\":{\"id\":\"3293\"}},\"id\":\"3292\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"axis\":{\"id\":\"3266\"},\"ticker\":null},\"id\":\"3268\",\"type\":\"Grid\"},{\"attributes\":{\"data\":{\"Correlation\":{\"__ndarray__\":\"AAAAAAAA+D8AAAAAAAD4PwAAAAAAAPg/\",\"dtype\":\"float64\",\"shape\":[3]},\"P_Value\":{\"__ndarray__\":\"mpmZmZmZqT8tQxzr4jYaPzqMMOKOeUU+\",\"dtype\":\"float64\",\"shape\":[3]},\"index\":[0,1,2],\"size\":{\"__ndarray__\":\"reqW3HH5IUCg/4+ZiqE7QKH/j5mKoUtA\",\"dtype\":\"float64\",\"shape\":[3]},\"size2\":{\"__ndarray__\":\"keNze0L3B0AVVbW7sWsiQBZVtbuxazJA\",\"dtype\":\"float64\",\"shape\":[3]},\"x_axis\":[\"\",\"\",\"\"],\"y_axis\":[\"5.0e-02\",\"1.0e-04\",\"1.0e-08\"]},\"selected\":{\"id\":\"3416\"},\"selection_policy\":{\"id\":\"3415\"}},\"id\":\"3301\",\"type\":\"ColumnDataSource\"}],\"root_ids\":[\"3308\"]},\"title\":\"Bokeh Application\",\"version\":\"2.0.1\"}};\n",
       "  var render_items = [{\"docid\":\"82e3c20c-52b8-442b-be01-fbc6ebf85e00\",\"root_ids\":[\"3308\"],\"roots\":{\"3308\":\"411956d0-ef38-4e23-a610-46f33e8f2523\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "3308"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "legend_min = ordered_df[\"P_value\"].min()\n",
    "#Make plot using plot utils\n",
    "p.plotCircleHeatMap(ordered_df, circle_var = \"P_value\",color_var = \"Correlation\", x_axis = \"Comparison\", \n",
    "                    y_axis = \"Cancer Type\", plot_width= 1000, plot_height = 500, legend_min = 1e-8, \n",
    "                    legend_max = 0.05, font_size = 12, show_legend = True , save_png = \"Figure2.png\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sig PTEN # CD2? CHUK\n",
    "t_only = ['CD3E', 'CD3G', 'CD4', 'CD8A', 'GRAP2', 'FYB1', 'PRKCQ'] \n",
    "b_only = ['PPP3CC', 'CD79A', 'SYK', 'BTK', 'LYN', 'BLNK', 'BLK', 'HLA-DMA', 'HLA-DMB']\n",
    "both = ['CD2', 'CD5', 'GRB2', 'LCK', 'ZAP70', 'VAV1', 'PTPN6', 'PTPRC', 'PIK3CD', 'PIK3CG', 'INPP5D',\n",
    "        'PLCG1', 'PLCG2', 'NFATC2', 'PRKCB', 'CHUK', 'NFKB2', 'DOCK2', 'RAC2', 'ELMO1', 'WAS']\n",
    "immune = t_only + b_only + both"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check if blanks are due to no data "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The follow code chunks show that the following cancers/genes don't have data: colon THBD, Kidney MMP12, and Ovarian MMP12. (As mentioned in EGFR Figure 2 legend)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get append version of the df with all proteins \n",
    "df_all_prot_append = pd.read_csv(\"Pval_corr_table_Fig_2_return_all_append.csv\")\n",
    "df_all_prot_append = df_all_prot_append.drop(['Unnamed: 0'], axis=1)\n",
    "#Remove Proteomics labeling \n",
    "df_all_prot_append = df_all_prot_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#MHC class2 (CD4) molecules\n",
    "mhc2 = [\"HLA-DPA1\",\"HLA-DPB1\",\"HLA-DMA\",\"HLA-DMB\",\"HLA-DOA\",\"HLA-DOB\",\"HLA-DQ\",\"HLA-DRA\",\"HLA-DRB1\"]\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_all_mhc2 = df_all_prot_append[df_all_prot_append.Comparison.isin(mhc2)]\n",
    "print(\"Number of missing columns\" )\n",
    "print(56 - len(df_all_mhc2))\n",
    "#HLA-DP and HLA-DM missing for all. HLA-DRA and HLA-DRB1 consistently not sig \n",
    "df_all_mhc2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CD8 and MHC class 1 molecules\n",
    "mhc1 = [\"CD8A\",\"HLA-A\",\"HLA-B\",\"HLA-C\"]\n",
    "\n",
    "df_all_mhc1 = df_all_prot_append[df_all_prot_append.Comparison.isin(mhc1)]\n",
    "df_all_mhc1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Remove Proteomics labeling \n",
    "df_all_prot_append = df_all_prot_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_all_comp_coag = df_all_prot_append[df_all_prot_append.Comparison.isin(coag_upa)]\n",
    "print(\"Number of rows in data frame \" + str(len(df_all_comp_coag)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Our Figure includes 26 genes for 8 cancers. If all data was present there would be 208 rows. However, the data frame only has 205 rows. 3 genes are missing. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_missing_genes(test_list, full_list):\n",
    "    for gene in full_list:\n",
    "        if (gene not in test_list):\n",
    "            print(gene)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get list of genes for colon, kidney, and ovarian\n",
    "colon = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Colon\"]\n",
    "colon_list = colon.Comparison.to_list()\n",
    "\n",
    "Kidney = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Kidney\"]\n",
    "Kidney_list = Kidney.Comparison.to_list()\n",
    "\n",
    "Ovarian = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Ovarian\"]\n",
    "Ovarian_list = Ovarian.Comparison.to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Show the 3 missing genes \n",
    "print(\"Ovarian missing genes: \")\n",
    "find_missing_genes(Ovarian_list, coag_upa)\n",
    "print(\"Kidney missing genes: \")\n",
    "find_missing_genes(Kidney_list, coag_upa)\n",
    "print(\"Colon missing genes: \")\n",
    "find_missing_genes(colon_list, coag_upa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
