{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Interacting Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "\n",
    "import sys\n",
    "sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\WhenMutationsDontMatter\\\\')\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "gbm = cptac.Gbm()\n",
    "endo = cptac.Endometrial()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ip = u.get_interacting_proteins('PTEN')\n",
    "len(ip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>MDM2_proteomics</th>\n",
       "      <th>CSNK2A1_proteomics</th>\n",
       "      <th>PDGFRB_proteomics</th>\n",
       "      <th>PIK3R2_proteomics</th>\n",
       "      <th>PIK3C3_proteomics</th>\n",
       "      <th>CSNK2A2_proteomics</th>\n",
       "      <th>SLC9A3R1_proteomics</th>\n",
       "      <th>PIK3R3_proteomics</th>\n",
       "      <th>USP13_proteomics</th>\n",
       "      <th>PIK3CA_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>PIK3CD_proteomics</th>\n",
       "      <th>ROCK1_proteomics</th>\n",
       "      <th>SHC1_proteomics</th>\n",
       "      <th>NEDD4_proteomics</th>\n",
       "      <th>INPP4B_proteomics</th>\n",
       "      <th>PIK3R1_proteomics</th>\n",
       "      <th>AKT1_proteomics</th>\n",
       "      <th>MVP_proteomics</th>\n",
       "      <th>MAGI3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>-0.886531</td>\n",
       "      <td>0.221578</td>\n",
       "      <td>-0.108555</td>\n",
       "      <td>-0.068085</td>\n",
       "      <td>0.081642</td>\n",
       "      <td>-0.273672</td>\n",
       "      <td>0.616424</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.423661</td>\n",
       "      <td>0.124542</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.315003</td>\n",
       "      <td>-0.319701</td>\n",
       "      <td>-0.095985</td>\n",
       "      <td>-0.437514</td>\n",
       "      <td>0.060124</td>\n",
       "      <td>-0.379646</td>\n",
       "      <td>-0.065930</td>\n",
       "      <td>0.147936</td>\n",
       "      <td>0.311427</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.169391</td>\n",
       "      <td>-0.109383</td>\n",
       "      <td>0.079710</td>\n",
       "      <td>-0.027524</td>\n",
       "      <td>-0.035616</td>\n",
       "      <td>0.450206</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155746</td>\n",
       "      <td>-0.211025</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.629943</td>\n",
       "      <td>0.013908</td>\n",
       "      <td>0.003049</td>\n",
       "      <td>0.179955</td>\n",
       "      <td>-0.639545</td>\n",
       "      <td>-0.283144</td>\n",
       "      <td>-0.285929</td>\n",
       "      <td>0.371665</td>\n",
       "      <td>0.259059</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.208625</td>\n",
       "      <td>0.414121</td>\n",
       "      <td>0.078025</td>\n",
       "      <td>0.062739</td>\n",
       "      <td>-0.105403</td>\n",
       "      <td>0.307356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.021276</td>\n",
       "      <td>-0.573919</td>\n",
       "      <td>...</td>\n",
       "      <td>0.085276</td>\n",
       "      <td>-0.048311</td>\n",
       "      <td>0.051279</td>\n",
       "      <td>0.265735</td>\n",
       "      <td>0.669283</td>\n",
       "      <td>0.061201</td>\n",
       "      <td>-0.222607</td>\n",
       "      <td>0.329753</td>\n",
       "      <td>0.128442</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>-0.564682</td>\n",
       "      <td>-0.292558</td>\n",
       "      <td>-0.141799</td>\n",
       "      <td>0.024117</td>\n",
       "      <td>0.087752</td>\n",
       "      <td>-0.189861</td>\n",
       "      <td>-0.040643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.322976</td>\n",
       "      <td>-0.360962</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.382652</td>\n",
       "      <td>-0.046639</td>\n",
       "      <td>0.288725</td>\n",
       "      <td>0.706601</td>\n",
       "      <td>0.451016</td>\n",
       "      <td>-0.496340</td>\n",
       "      <td>-0.370584</td>\n",
       "      <td>-0.518407</td>\n",
       "      <td>-0.099551</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-0.243835</td>\n",
       "      <td>0.132716</td>\n",
       "      <td>0.812053</td>\n",
       "      <td>-0.029297</td>\n",
       "      <td>-0.057555</td>\n",
       "      <td>0.239919</td>\n",
       "      <td>-0.155468</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.049040</td>\n",
       "      <td>0.137414</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.537787</td>\n",
       "      <td>0.237764</td>\n",
       "      <td>-0.337079</td>\n",
       "      <td>0.133808</td>\n",
       "      <td>-0.024038</td>\n",
       "      <td>-0.146416</td>\n",
       "      <td>0.004573</td>\n",
       "      <td>-0.741119</td>\n",
       "      <td>-0.017146</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.015279</td>\n",
       "      <td>0.121787</td>\n",
       "      <td>-0.116620</td>\n",
       "      <td>-0.092783</td>\n",
       "      <td>-0.022308</td>\n",
       "      <td>0.128944</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.096330</td>\n",
       "      <td>-0.218186</td>\n",
       "      <td>...</td>\n",
       "      <td>0.395350</td>\n",
       "      <td>0.141372</td>\n",
       "      <td>0.557741</td>\n",
       "      <td>0.418850</td>\n",
       "      <td>0.010608</td>\n",
       "      <td>-0.093624</td>\n",
       "      <td>-0.002002</td>\n",
       "      <td>-0.090315</td>\n",
       "      <td>-0.079546</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312023</td>\n",
       "      <td>0.361701</td>\n",
       "      <td>-0.023912</td>\n",
       "      <td>-0.160911</td>\n",
       "      <td>0.266436</td>\n",
       "      <td>-1.291855</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.276721</td>\n",
       "      <td>-0.049803</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.556546</td>\n",
       "      <td>-0.158482</td>\n",
       "      <td>-0.282123</td>\n",
       "      <td>0.028673</td>\n",
       "      <td>-0.219639</td>\n",
       "      <td>-0.164426</td>\n",
       "      <td>0.501695</td>\n",
       "      <td>-0.994267</td>\n",
       "      <td>-0.364258</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>0.137052</td>\n",
       "      <td>-0.069012</td>\n",
       "      <td>0.606401</td>\n",
       "      <td>0.626821</td>\n",
       "      <td>-0.029083</td>\n",
       "      <td>-0.161898</td>\n",
       "      <td>0.075817</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.091852</td>\n",
       "      <td>0.815177</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.009546</td>\n",
       "      <td>-0.222781</td>\n",
       "      <td>0.061741</td>\n",
       "      <td>0.013211</td>\n",
       "      <td>0.348604</td>\n",
       "      <td>0.443719</td>\n",
       "      <td>0.196700</td>\n",
       "      <td>0.086260</td>\n",
       "      <td>0.541898</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.276126</td>\n",
       "      <td>-0.108478</td>\n",
       "      <td>-0.135303</td>\n",
       "      <td>0.173834</td>\n",
       "      <td>0.246281</td>\n",
       "      <td>0.090879</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.405654</td>\n",
       "      <td>0.143824</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.255032</td>\n",
       "      <td>-0.256044</td>\n",
       "      <td>-0.240227</td>\n",
       "      <td>-0.209677</td>\n",
       "      <td>0.055190</td>\n",
       "      <td>0.084610</td>\n",
       "      <td>-0.198548</td>\n",
       "      <td>-0.012143</td>\n",
       "      <td>0.260033</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.154969</td>\n",
       "      <td>-0.228369</td>\n",
       "      <td>0.186960</td>\n",
       "      <td>-0.073499</td>\n",
       "      <td>0.064866</td>\n",
       "      <td>-0.275365</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.252806</td>\n",
       "      <td>0.163076</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002891</td>\n",
       "      <td>-0.191965</td>\n",
       "      <td>0.123735</td>\n",
       "      <td>0.013306</td>\n",
       "      <td>-0.077057</td>\n",
       "      <td>-0.166501</td>\n",
       "      <td>-0.055991</td>\n",
       "      <td>-0.740250</td>\n",
       "      <td>-0.134775</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        MDM2_proteomics  CSNK2A1_proteomics  PDGFRB_proteomics  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00104         -0.886531            0.221578          -0.108555   \n",
       "C3L-00365               NaN            0.169391          -0.109383   \n",
       "C3L-00674               NaN           -0.208625           0.414121   \n",
       "C3L-00677         -0.564682           -0.292558          -0.141799   \n",
       "C3L-01040         -0.243835            0.132716           0.812053   \n",
       "...                     ...                 ...                ...   \n",
       "C3N-03183               NaN           -0.015279           0.121787   \n",
       "C3N-03184               NaN            0.312023           0.361701   \n",
       "C3N-03186          0.137052           -0.069012           0.606401   \n",
       "C3N-03188               NaN            0.276126          -0.108478   \n",
       "C3N-03473               NaN            0.154969          -0.228369   \n",
       "\n",
       "Name        PIK3R2_proteomics  PIK3C3_proteomics  CSNK2A2_proteomics  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00104           -0.068085           0.081642           -0.273672   \n",
       "C3L-00365            0.079710          -0.027524           -0.035616   \n",
       "C3L-00674            0.078025           0.062739           -0.105403   \n",
       "C3L-00677            0.024117           0.087752           -0.189861   \n",
       "C3L-01040           -0.029297          -0.057555            0.239919   \n",
       "...                       ...                ...                 ...   \n",
       "C3N-03183           -0.116620          -0.092783           -0.022308   \n",
       "C3N-03184           -0.023912          -0.160911            0.266436   \n",
       "C3N-03186            0.626821          -0.029083           -0.161898   \n",
       "C3N-03188           -0.135303           0.173834            0.246281   \n",
       "C3N-03473            0.186960          -0.073499            0.064866   \n",
       "\n",
       "Name        SLC9A3R1_proteomics  PIK3R3_proteomics  USP13_proteomics  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00104              0.616424                NaN         -0.423661   \n",
       "C3L-00365              0.450206                NaN          0.155746   \n",
       "C3L-00674              0.307356                NaN         -0.021276   \n",
       "C3L-00677             -0.040643                NaN         -0.322976   \n",
       "C3L-01040             -0.155468                NaN         -0.049040   \n",
       "...                         ...                ...               ...   \n",
       "C3N-03183              0.128944                NaN         -0.096330   \n",
       "C3N-03184             -1.291855                NaN          0.276721   \n",
       "C3N-03186              0.075817                NaN          0.091852   \n",
       "C3N-03188              0.090879                NaN         -0.405654   \n",
       "C3N-03473             -0.275365                NaN          0.252806   \n",
       "\n",
       "Name        PIK3CA_proteomics  ...  PIK3CD_proteomics  ROCK1_proteomics  \\\n",
       "Patient_ID                     ...                                        \n",
       "C3L-00104            0.124542  ...          -0.315003         -0.319701   \n",
       "C3L-00365           -0.211025  ...          -0.629943          0.013908   \n",
       "C3L-00674           -0.573919  ...           0.085276         -0.048311   \n",
       "C3L-00677           -0.360962  ...          -0.382652         -0.046639   \n",
       "C3L-01040            0.137414  ...          -0.537787          0.237764   \n",
       "...                       ...  ...                ...               ...   \n",
       "C3N-03183           -0.218186  ...           0.395350          0.141372   \n",
       "C3N-03184           -0.049803  ...          -0.556546         -0.158482   \n",
       "C3N-03186            0.815177  ...          -0.009546         -0.222781   \n",
       "C3N-03188            0.143824  ...          -0.255032         -0.256044   \n",
       "C3N-03473            0.163076  ...           0.002891         -0.191965   \n",
       "\n",
       "Name        SHC1_proteomics  NEDD4_proteomics  INPP4B_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00104         -0.095985         -0.437514           0.060124   \n",
       "C3L-00365          0.003049          0.179955          -0.639545   \n",
       "C3L-00674          0.051279          0.265735           0.669283   \n",
       "C3L-00677          0.288725          0.706601           0.451016   \n",
       "C3L-01040         -0.337079          0.133808          -0.024038   \n",
       "...                     ...               ...                ...   \n",
       "C3N-03183          0.557741          0.418850           0.010608   \n",
       "C3N-03184         -0.282123          0.028673          -0.219639   \n",
       "C3N-03186          0.061741          0.013211           0.348604   \n",
       "C3N-03188         -0.240227         -0.209677           0.055190   \n",
       "C3N-03473          0.123735          0.013306          -0.077057   \n",
       "\n",
       "Name        PIK3R1_proteomics  AKT1_proteomics  MVP_proteomics  \\\n",
       "Patient_ID                                                       \n",
       "C3L-00104           -0.379646        -0.065930        0.147936   \n",
       "C3L-00365           -0.283144        -0.285929        0.371665   \n",
       "C3L-00674            0.061201        -0.222607        0.329753   \n",
       "C3L-00677           -0.496340        -0.370584       -0.518407   \n",
       "C3L-01040           -0.146416         0.004573       -0.741119   \n",
       "...                       ...              ...             ...   \n",
       "C3N-03183           -0.093624        -0.002002       -0.090315   \n",
       "C3N-03184           -0.164426         0.501695       -0.994267   \n",
       "C3N-03186            0.443719         0.196700        0.086260   \n",
       "C3N-03188            0.084610        -0.198548       -0.012143   \n",
       "C3N-03473           -0.166501        -0.055991       -0.740250   \n",
       "\n",
       "Name        MAGI3_proteomics        Mutation  \n",
       "Patient_ID                                    \n",
       "C3L-00104           0.311427        Deletion  \n",
       "C3L-00365           0.259059        Deletion  \n",
       "C3L-00674           0.128442        Deletion  \n",
       "C3L-00677          -0.099551        Deletion  \n",
       "C3L-01040          -0.017146        Deletion  \n",
       "...                      ...             ...  \n",
       "C3N-03183          -0.079546        Deletion  \n",
       "C3N-03184          -0.364258        Deletion  \n",
       "C3N-03186           0.541898        Deletion  \n",
       "C3N-03188           0.260033  Wildtype_Tumor  \n",
       "C3N-03473          -0.134775        Deletion  \n",
       "\n",
       "[99 rows x 29 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = gbm.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = gbm.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.219523e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>1.030815e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>1.225949e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Comparison       P_Value\n",
       "0    PTEN_proteomics  1.219523e-07\n",
       "1  PIK3CD_proteomics  1.030815e-03\n",
       "2  PIK3C3_proteomics  1.225949e-03"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "u.wrap_ttest(del_wt, 'Mutation', cols)#, return_all = True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gbm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MDM2</th>\n",
       "      <td>-0.127685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A1</th>\n",
       "      <td>-0.123874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PDGFRB</th>\n",
       "      <td>-0.028576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R2</th>\n",
       "      <td>0.015429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3C3</th>\n",
       "      <td>0.102518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A2</th>\n",
       "      <td>-0.102838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC9A3R1</th>\n",
       "      <td>0.289788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R3</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP13</th>\n",
       "      <td>-0.111700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>0.105746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>-0.256630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR</th>\n",
       "      <td>-1.001035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CB</th>\n",
       "      <td>0.042720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTK2</th>\n",
       "      <td>-0.093309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP7</th>\n",
       "      <td>-0.154035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XIAP</th>\n",
       "      <td>-0.088967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>0.511726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREX2</th>\n",
       "      <td>-0.151642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAST2</th>\n",
       "      <td>0.139061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CD</th>\n",
       "      <td>0.492099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ROCK1</th>\n",
       "      <td>0.080280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SHC1</th>\n",
       "      <td>0.153123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEDD4</th>\n",
       "      <td>-0.079598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INPP4B</th>\n",
       "      <td>0.341829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R1</th>\n",
       "      <td>0.188390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT1</th>\n",
       "      <td>-0.169565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MVP</th>\n",
       "      <td>0.474151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAGI3</th>\n",
       "      <td>0.050953</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Gbm\n",
       "MDM2     -0.127685\n",
       "CSNK2A1  -0.123874\n",
       "PDGFRB   -0.028576\n",
       "PIK3R2    0.015429\n",
       "PIK3C3    0.102518\n",
       "CSNK2A2  -0.102838\n",
       "SLC9A3R1  0.289788\n",
       "PIK3R3         NaN\n",
       "USP13    -0.111700\n",
       "PIK3CA    0.105746\n",
       "TP53     -0.256630\n",
       "EGFR     -1.001035\n",
       "PIK3CB    0.042720\n",
       "PTK2     -0.093309\n",
       "USP7     -0.154035\n",
       "XIAP     -0.088967\n",
       "PTEN      0.511726\n",
       "PREX2    -0.151642\n",
       "MAST2     0.139061\n",
       "PIK3CD    0.492099\n",
       "ROCK1     0.080280\n",
       "SHC1      0.153123\n",
       "NEDD4    -0.079598\n",
       "INPP4B    0.341829\n",
       "PIK3R1    0.188390\n",
       "AKT1     -0.169565\n",
       "MVP       0.474151\n",
       "MAGI3     0.050953"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "gbm_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif_gbm = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    gbm_d[prot] = dif_gbm\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif_gbm)\n",
    "\n",
    "gbm_df = pd.DataFrame.from_dict(gbm_d, orient='index', columns=['Gbm'])\n",
    "gbm_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: MDM2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 69 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    15\n",
       "Deletion           3\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Endo ttest\n",
    "\n",
    "mut_type_gbm = endo.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = endo.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "o = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: MDM2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 8)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 8)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    46\n",
       "Deletion          20\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "gene = 'PTEN'\n",
    "\n",
    "mut_type_gbm = o.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = o.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = o.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000909</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Comparison   P_Value\n",
       "0  PTEN_proteomics  0.000909"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "o_pval = u.wrap_ttest(del_wt, 'Mutation', cols)#, return_all = True)\n",
    "o_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ovarian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MDM2</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A1</th>\n",
       "      <td>-0.015518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PDGFRB</th>\n",
       "      <td>-0.163327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R2</th>\n",
       "      <td>-0.240986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3C3</th>\n",
       "      <td>-0.104687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A2</th>\n",
       "      <td>-0.011741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC9A3R1</th>\n",
       "      <td>0.146323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R3</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP13</th>\n",
       "      <td>0.154918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>-0.199296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>-0.196586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR</th>\n",
       "      <td>0.063388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CB</th>\n",
       "      <td>-0.276354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTK2</th>\n",
       "      <td>-0.229494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP7</th>\n",
       "      <td>0.017808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XIAP</th>\n",
       "      <td>0.010591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>0.169972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREX2</th>\n",
       "      <td>-0.505300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAST2</th>\n",
       "      <td>0.085809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CD</th>\n",
       "      <td>0.035308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ROCK1</th>\n",
       "      <td>0.026832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SHC1</th>\n",
       "      <td>-0.091044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEDD4</th>\n",
       "      <td>-0.001637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INPP4B</th>\n",
       "      <td>-0.370647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R1</th>\n",
       "      <td>-0.179425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT1</th>\n",
       "      <td>-0.145812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MVP</th>\n",
       "      <td>-0.193693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAGI3</th>\n",
       "      <td>-0.133108</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Ovarian\n",
       "MDM2           NaN\n",
       "CSNK2A1  -0.015518\n",
       "PDGFRB   -0.163327\n",
       "PIK3R2   -0.240986\n",
       "PIK3C3   -0.104687\n",
       "CSNK2A2  -0.011741\n",
       "SLC9A3R1  0.146323\n",
       "PIK3R3         NaN\n",
       "USP13     0.154918\n",
       "PIK3CA   -0.199296\n",
       "TP53     -0.196586\n",
       "EGFR      0.063388\n",
       "PIK3CB   -0.276354\n",
       "PTK2     -0.229494\n",
       "USP7      0.017808\n",
       "XIAP      0.010591\n",
       "PTEN      0.169972\n",
       "PREX2    -0.505300\n",
       "MAST2     0.085809\n",
       "PIK3CD    0.035308\n",
       "ROCK1     0.026832\n",
       "SHC1     -0.091044\n",
       "NEDD4    -0.001637\n",
       "INPP4B   -0.370647\n",
       "PIK3R1   -0.179425\n",
       "AKT1     -0.145812\n",
       "MVP      -0.193693\n",
       "MAGI3    -0.133108"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "o_df = pd.DataFrame.from_dict(d, orient='index', columns=['Ovarian'])\n",
    "o_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Breast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CSNK2A1, MDM2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 6)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    80\n",
       "Deletion          24\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = b.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = b.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = b.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Comparison   P_Value\n",
       "0  PTEN_proteomics  0.000004"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "b_pval = u.wrap_ttest(del_wt, 'Mutation', cols)#, return_all = True)\n",
    "b_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Brca</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MDM2</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A1</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PDGFRB</th>\n",
       "      <td>-0.04105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R2</th>\n",
       "      <td>0.00110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3C3</th>\n",
       "      <td>-0.11490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A2</th>\n",
       "      <td>-0.28100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC9A3R1</th>\n",
       "      <td>0.37085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R3</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP13</th>\n",
       "      <td>-0.28270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>0.27750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>-0.11100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR</th>\n",
       "      <td>-0.58645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CB</th>\n",
       "      <td>-0.01540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTK2</th>\n",
       "      <td>0.04900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP7</th>\n",
       "      <td>0.14305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XIAP</th>\n",
       "      <td>-0.15040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>1.00800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREX2</th>\n",
       "      <td>0.02575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAST2</th>\n",
       "      <td>0.29705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CD</th>\n",
       "      <td>-0.15015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ROCK1</th>\n",
       "      <td>-0.00695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SHC1</th>\n",
       "      <td>-0.42620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEDD4</th>\n",
       "      <td>0.48570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INPP4B</th>\n",
       "      <td>0.37695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R1</th>\n",
       "      <td>1.03615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT1</th>\n",
       "      <td>0.13185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MVP</th>\n",
       "      <td>-0.01090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAGI3</th>\n",
       "      <td>-0.15150</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Brca\n",
       "MDM2          NaN\n",
       "CSNK2A1       NaN\n",
       "PDGFRB   -0.04105\n",
       "PIK3R2    0.00110\n",
       "PIK3C3   -0.11490\n",
       "CSNK2A2  -0.28100\n",
       "SLC9A3R1  0.37085\n",
       "PIK3R3        NaN\n",
       "USP13    -0.28270\n",
       "PIK3CA    0.27750\n",
       "TP53     -0.11100\n",
       "EGFR     -0.58645\n",
       "PIK3CB   -0.01540\n",
       "PTK2      0.04900\n",
       "USP7      0.14305\n",
       "XIAP     -0.15040\n",
       "PTEN      1.00800\n",
       "PREX2     0.02575\n",
       "MAST2     0.29705\n",
       "PIK3CD   -0.15015\n",
       "ROCK1    -0.00695\n",
       "SHC1     -0.42620\n",
       "NEDD4     0.48570\n",
       "INPP4B    0.37695\n",
       "PIK3R1    1.03615\n",
       "AKT1      0.13185\n",
       "MVP      -0.01090\n",
       "MAGI3    -0.15150"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "b_df = pd.DataFrame.from_dict(d, orient='index', columns=['Brca'])\n",
    "b_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "col = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: MAGI3, MAST2, MDM2, PIK3R3, PREX2 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    76\n",
       "Deletion          25\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = col.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = col.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.012872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.030107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.116419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.118135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.131200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.144425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.182782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.237441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.250276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.326372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.328626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.332902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.354186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.366886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.375198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.491790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.516876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.522867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.561268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.665493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.968870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.972777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.974887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Comparison   P_Value\n",
       "0       PTEN_proteomics  0.012872\n",
       "1       SHC1_proteomics  0.030107\n",
       "2     PIK3C3_proteomics  0.116419\n",
       "3     INPP4B_proteomics  0.118135\n",
       "4     PIK3CD_proteomics  0.131200\n",
       "5    CSNK2A1_proteomics  0.144425\n",
       "6        MVP_proteomics  0.182782\n",
       "7       USP7_proteomics  0.237441\n",
       "8     PDGFRB_proteomics  0.250276\n",
       "9    CSNK2A2_proteomics  0.326372\n",
       "10  SLC9A3R1_proteomics  0.328626\n",
       "11    PIK3CA_proteomics  0.332902\n",
       "12      AKT1_proteomics  0.354186\n",
       "13      EGFR_proteomics  0.366886\n",
       "14    PIK3R2_proteomics  0.375198\n",
       "15     NEDD4_proteomics  0.491790\n",
       "16      XIAP_proteomics  0.516876\n",
       "17      PTK2_proteomics  0.522867\n",
       "18     ROCK1_proteomics  0.561268\n",
       "19    PIK3CB_proteomics  0.665493\n",
       "20      TP53_proteomics  0.968870\n",
       "21    PIK3R1_proteomics  0.972777\n",
       "22     USP13_proteomics  0.974887\n",
       "23      MDM2_proteomics       NaN\n",
       "24    PIK3R3_proteomics       NaN\n",
       "25     PREX2_proteomics       NaN\n",
       "26     MAST2_proteomics       NaN\n",
       "27     MAGI3_proteomics       NaN"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "c_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "c_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Brca</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MDM2</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A1</th>\n",
       "      <td>0.04845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PDGFRB</th>\n",
       "      <td>-0.02450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R2</th>\n",
       "      <td>-0.05730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3C3</th>\n",
       "      <td>0.07995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A2</th>\n",
       "      <td>-0.04825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC9A3R1</th>\n",
       "      <td>0.22100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R3</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP13</th>\n",
       "      <td>-0.18050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>-0.03385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>-0.04905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR</th>\n",
       "      <td>0.02350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CB</th>\n",
       "      <td>0.08550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTK2</th>\n",
       "      <td>0.00595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP7</th>\n",
       "      <td>-0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XIAP</th>\n",
       "      <td>0.06900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>0.20400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREX2</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAST2</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CD</th>\n",
       "      <td>0.07600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ROCK1</th>\n",
       "      <td>-0.03900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SHC1</th>\n",
       "      <td>-0.11610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEDD4</th>\n",
       "      <td>-0.14490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INPP4B</th>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R1</th>\n",
       "      <td>0.01245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT1</th>\n",
       "      <td>0.00670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MVP</th>\n",
       "      <td>-0.23690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAGI3</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Brca\n",
       "MDM2          NaN\n",
       "CSNK2A1   0.04845\n",
       "PDGFRB   -0.02450\n",
       "PIK3R2   -0.05730\n",
       "PIK3C3    0.07995\n",
       "CSNK2A2  -0.04825\n",
       "SLC9A3R1  0.22100\n",
       "PIK3R3        NaN\n",
       "USP13    -0.18050\n",
       "PIK3CA   -0.03385\n",
       "TP53     -0.04905\n",
       "EGFR      0.02350\n",
       "PIK3CB    0.08550\n",
       "PTK2      0.00595\n",
       "USP7     -0.04450\n",
       "XIAP      0.06900\n",
       "PTEN      0.20400\n",
       "PREX2         NaN\n",
       "MAST2         NaN\n",
       "PIK3CD    0.07600\n",
       "ROCK1    -0.03900\n",
       "SHC1     -0.11610\n",
       "NEDD4    -0.14490\n",
       "INPP4B   -0.08220\n",
       "PIK3R1    0.01245\n",
       "AKT1      0.00670\n",
       "MVP      -0.23690\n",
       "MAGI3         NaN"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "c_df = pd.DataFrame.from_dict(d, orient='index', columns=['Brca'])\n",
    "c_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Kidney"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "r = cptac.Ccrcc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 105 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 382)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 189 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\pandas\\core\\reshape\\merge.py:618: UserWarning: merging between different levels can give an unintended result (2 levels on the left, 1 on the right)\n",
      "  warnings.warn(msg, UserWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    95\n",
       "Deletion          11\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = r.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = r.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ccrcc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AKT1_CNV</td>\n",
       "      <td>0.120693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CSNK2A1_CNV</td>\n",
       "      <td>-0.035823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CSNK2A2_CNV</td>\n",
       "      <td>-0.014239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>EGFR_CNV</td>\n",
       "      <td>-0.006142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>INPP4B_CNV</td>\n",
       "      <td>-0.026517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>MAGI3_CNV</td>\n",
       "      <td>-0.032643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>MAST2_CNV</td>\n",
       "      <td>-0.031930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>MDM2_CNV</td>\n",
       "      <td>-0.035195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>MVP_CNV</td>\n",
       "      <td>-0.015314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NEDD4_CNV</td>\n",
       "      <td>-0.034182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PDGFRB_CNV</td>\n",
       "      <td>0.132965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3C3_CNV</td>\n",
       "      <td>-0.032898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3CA_CNV</td>\n",
       "      <td>-0.045139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3CB_CNV</td>\n",
       "      <td>-0.050940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3CD_CNV</td>\n",
       "      <td>-0.031275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3R1_CNV</td>\n",
       "      <td>-0.027801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3R2_CNV</td>\n",
       "      <td>-0.017630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3R3_CNV</td>\n",
       "      <td>-0.031930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PREX2_CNV</td>\n",
       "      <td>-0.033309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PTEN_CNV</td>\n",
       "      <td>-0.057497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PTK2_CNV</td>\n",
       "      <td>-0.031336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ROCK1_CNV</td>\n",
       "      <td>-0.032898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>SHC1_CNV</td>\n",
       "      <td>-0.023641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>SLC9A3R1_CNV</td>\n",
       "      <td>-0.033200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TP53_CNV</td>\n",
       "      <td>-0.016247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>USP13_CNV</td>\n",
       "      <td>-0.045139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>USP7_CNV</td>\n",
       "      <td>-0.019387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XIAP_CNV</td>\n",
       "      <td>-0.030506</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Ccrcc\n",
       "AKT1_CNV      0.120693\n",
       "CSNK2A1_CNV  -0.035823\n",
       "CSNK2A2_CNV  -0.014239\n",
       "EGFR_CNV     -0.006142\n",
       "INPP4B_CNV   -0.026517\n",
       "MAGI3_CNV    -0.032643\n",
       "MAST2_CNV    -0.031930\n",
       "MDM2_CNV     -0.035195\n",
       "MVP_CNV      -0.015314\n",
       "NEDD4_CNV    -0.034182\n",
       "PDGFRB_CNV    0.132965\n",
       "PIK3C3_CNV   -0.032898\n",
       "PIK3CA_CNV   -0.045139\n",
       "PIK3CB_CNV   -0.050940\n",
       "PIK3CD_CNV   -0.031275\n",
       "PIK3R1_CNV   -0.027801\n",
       "PIK3R2_CNV   -0.017630\n",
       "PIK3R3_CNV   -0.031930\n",
       "PREX2_CNV    -0.033309\n",
       "PTEN_CNV     -0.057497\n",
       "PTK2_CNV     -0.031336\n",
       "ROCK1_CNV    -0.032898\n",
       "SHC1_CNV     -0.023641\n",
       "SLC9A3R1_CNV -0.033200\n",
       "TP53_CNV     -0.016247\n",
       "USP13_CNV    -0.045139\n",
       "USP7_CNV     -0.019387\n",
       "XIAP_CNV     -0.030506"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for prot in prot_list:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "c_df = pd.DataFrame.from_dict(d, orient='index', columns=['Colon'])\n",
    "c_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Lung - no pten mutations data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Head and Neck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "h = cptac.Hnscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>PTEN_CNV</th>\n",
       "      <th>binary_mutations</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>C3L-00977</td>\n",
       "      <td>0.061917</td>\n",
       "      <td>Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00987</td>\n",
       "      <td>-0.284854</td>\n",
       "      <td>Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00994</td>\n",
       "      <td>-0.013106</td>\n",
       "      <td>Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00995</td>\n",
       "      <td>-0.029976</td>\n",
       "      <td>Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00997</td>\n",
       "      <td>-0.140809</td>\n",
       "      <td>Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-04277</td>\n",
       "      <td>-0.017048</td>\n",
       "      <td>Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-04278</td>\n",
       "      <td>-0.002343</td>\n",
       "      <td>Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-04279</td>\n",
       "      <td>0.105120</td>\n",
       "      <td>Amp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-04280</td>\n",
       "      <td>-0.042670</td>\n",
       "      <td>Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-04611</td>\n",
       "      <td>-0.113783</td>\n",
       "      <td>Del</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>111 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        PTEN_CNV binary_mutations\n",
       "Patient_ID                           \n",
       "C3L-00977   0.061917              Amp\n",
       "C3L-00987  -0.284854              Del\n",
       "C3L-00994  -0.013106              Del\n",
       "C3L-00995  -0.029976              Del\n",
       "C3L-00997  -0.140809              Del\n",
       "...              ...              ...\n",
       "C3N-04277  -0.017048              Del\n",
       "C3N-04278  -0.002343              Del\n",
       "C3N-04279   0.105120              Amp\n",
       "C3N-04280  -0.042670              Del\n",
       "C3N-04611  -0.113783              Del\n",
       "\n",
       "[111 rows x 2 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1 - Create dataframe \n",
    "prot_and_mutations = h.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'CNV', omics_genes = ip)  \n",
    "#prot_and_mutations = h.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "# Step 2 - Create binary column \n",
    "prot_and_mutations['binary_mutations'] = np.where(\n",
    "            prot_and_mutations[gene+'_CNV'] >= 0.02, 'Amp', 'Del')\n",
    "\n",
    "# Step 3 - Format the dataframe correctly \n",
    "tumors = prot_and_mutations[prot_and_mutations.Sample_Status == 'Tumor'] #drop Normal samples\n",
    "columns_to_drop = [gene+\"_Mutation\", gene+\"_Location\", gene+\"_Mutation_Status\", \"Sample_Status\"]\n",
    "mut_status = tumors.drop(columns_to_drop, axis = 1)\n",
    "mut_status = mut_status.dropna(axis=1,how='all')\n",
    "h_prot = list(mut_status.columns[:-1]) # new prot list without na in all tumor (sometimes values in normal)\n",
    "mut_status[['PTEN_CNV','binary_mutations']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hnscc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>MDM2_CNV</td>\n",
       "      <td>-0.029787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CSNK2A1_CNV</td>\n",
       "      <td>-0.048607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PDGFRB_CNV</td>\n",
       "      <td>-0.019600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3R2_CNV</td>\n",
       "      <td>-0.024264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3C3_CNV</td>\n",
       "      <td>-0.042857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CSNK2A2_CNV</td>\n",
       "      <td>-0.023380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>SLC9A3R1_CNV</td>\n",
       "      <td>-0.045898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3R3_CNV</td>\n",
       "      <td>-0.043826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>USP13_CNV</td>\n",
       "      <td>-0.157708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3CA_CNV</td>\n",
       "      <td>-0.157708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TP53_CNV</td>\n",
       "      <td>-0.035777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>EGFR_CNV</td>\n",
       "      <td>-0.061896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3CB_CNV</td>\n",
       "      <td>0.002074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PTK2_CNV</td>\n",
       "      <td>0.059077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>USP7_CNV</td>\n",
       "      <td>-0.023623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XIAP_CNV</td>\n",
       "      <td>-0.052603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PTEN_CNV</td>\n",
       "      <td>-0.085870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PREX2_CNV</td>\n",
       "      <td>0.030508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>MAST2_CNV</td>\n",
       "      <td>-0.048937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3CD_CNV</td>\n",
       "      <td>-0.029514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ROCK1_CNV</td>\n",
       "      <td>-0.033082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>SHC1_CNV</td>\n",
       "      <td>-0.034168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NEDD4_CNV</td>\n",
       "      <td>-0.057806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>INPP4B_CNV</td>\n",
       "      <td>-0.004886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PIK3R1_CNV</td>\n",
       "      <td>0.002269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AKT1_CNV</td>\n",
       "      <td>-0.040452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>MVP_CNV</td>\n",
       "      <td>-0.024298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>MAGI3_CNV</td>\n",
       "      <td>-0.012427</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Hnscc\n",
       "MDM2_CNV     -0.029787\n",
       "CSNK2A1_CNV  -0.048607\n",
       "PDGFRB_CNV   -0.019600\n",
       "PIK3R2_CNV   -0.024264\n",
       "PIK3C3_CNV   -0.042857\n",
       "CSNK2A2_CNV  -0.023380\n",
       "SLC9A3R1_CNV -0.045898\n",
       "PIK3R3_CNV   -0.043826\n",
       "USP13_CNV    -0.157708\n",
       "PIK3CA_CNV   -0.157708\n",
       "TP53_CNV     -0.035777\n",
       "EGFR_CNV     -0.061896\n",
       "PIK3CB_CNV    0.002074\n",
       "PTK2_CNV      0.059077\n",
       "USP7_CNV     -0.023623\n",
       "XIAP_CNV     -0.052603\n",
       "PTEN_CNV     -0.085870\n",
       "PREX2_CNV     0.030508\n",
       "MAST2_CNV    -0.048937\n",
       "PIK3CD_CNV   -0.029514\n",
       "ROCK1_CNV    -0.033082\n",
       "SHC1_CNV     -0.034168\n",
       "NEDD4_CNV    -0.057806\n",
       "INPP4B_CNV   -0.004886\n",
       "PIK3R1_CNV    0.002269\n",
       "AKT1_CNV     -0.040452\n",
       "MVP_CNV      -0.024298\n",
       "MAGI3_CNV    -0.012427"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amp_df = mut_status.loc[mut_status['binary_mutations'] == 'Amp']\n",
    "del_df = mut_status.loc[mut_status['binary_mutations'] == 'Del']\n",
    "\n",
    "amp_med = amp_df.median()\n",
    "del_med = del_df.median()\n",
    "\n",
    "h_dict = {}\n",
    "\n",
    "for prot in h_prot:\n",
    "    dif = del_med[prot] - amp_med[prot]\n",
    "    h_dict[prot] = dif\n",
    "    #print(del_med[prot], '-', amp_med[prot], '=',dif)\n",
    "\n",
    "h_df = pd.DataFrame.from_dict(h_dict, orient='index', columns=['Hnscc'])\n",
    "h_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MDM2_CNV</th>\n",
       "      <th>CSNK2A1_CNV</th>\n",
       "      <th>PDGFRB_CNV</th>\n",
       "      <th>PIK3R2_CNV</th>\n",
       "      <th>PIK3C3_CNV</th>\n",
       "      <th>CSNK2A2_CNV</th>\n",
       "      <th>SLC9A3R1_CNV</th>\n",
       "      <th>PIK3R3_CNV</th>\n",
       "      <th>USP13_CNV</th>\n",
       "      <th>PIK3CA_CNV</th>\n",
       "      <th>...</th>\n",
       "      <th>MAST2_CNV</th>\n",
       "      <th>PIK3CD_CNV</th>\n",
       "      <th>ROCK1_CNV</th>\n",
       "      <th>SHC1_CNV</th>\n",
       "      <th>NEDD4_CNV</th>\n",
       "      <th>INPP4B_CNV</th>\n",
       "      <th>PIK3R1_CNV</th>\n",
       "      <th>AKT1_CNV</th>\n",
       "      <th>MVP_CNV</th>\n",
       "      <th>MAGI3_CNV</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cancer</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Gbm</td>\n",
       "      <td>-0.112350</td>\n",
       "      <td>-0.062947</td>\n",
       "      <td>-0.039516</td>\n",
       "      <td>0.027473</td>\n",
       "      <td>-0.016396</td>\n",
       "      <td>-0.118437</td>\n",
       "      <td>-0.034208</td>\n",
       "      <td>-0.071091</td>\n",
       "      <td>-0.008994</td>\n",
       "      <td>-0.005148</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.071091</td>\n",
       "      <td>-0.046664</td>\n",
       "      <td>-0.010449</td>\n",
       "      <td>-0.110476</td>\n",
       "      <td>-0.032116</td>\n",
       "      <td>0.119056</td>\n",
       "      <td>-0.026384</td>\n",
       "      <td>-0.044454</td>\n",
       "      <td>-0.109663</td>\n",
       "      <td>-0.070727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Endometrial</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.015000</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.015000</td>\n",
       "      <td>0.020000</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.020000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.020000</td>\n",
       "      <td>0.020000</td>\n",
       "      <td>0.020000</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.020000</td>\n",
       "      <td>0.010000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ovarian</td>\n",
       "      <td>-0.072600</td>\n",
       "      <td>0.029040</td>\n",
       "      <td>-0.081950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.069850</td>\n",
       "      <td>0.034150</td>\n",
       "      <td>0.038350</td>\n",
       "      <td>0.015700</td>\n",
       "      <td>0.042050</td>\n",
       "      <td>0.062450</td>\n",
       "      <td>...</td>\n",
       "      <td>0.017500</td>\n",
       "      <td>0.160550</td>\n",
       "      <td>0.075700</td>\n",
       "      <td>-0.064400</td>\n",
       "      <td>-0.153428</td>\n",
       "      <td>-0.079850</td>\n",
       "      <td>-0.003788</td>\n",
       "      <td>0.001850</td>\n",
       "      <td>-0.062550</td>\n",
       "      <td>0.005600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Colon</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>-0.336000</td>\n",
       "      <td>-0.004000</td>\n",
       "      <td>0.050500</td>\n",
       "      <td>-0.315500</td>\n",
       "      <td>-0.004000</td>\n",
       "      <td>0.022500</td>\n",
       "      <td>0.015000</td>\n",
       "      <td>0.007000</td>\n",
       "      <td>0.002000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.028500</td>\n",
       "      <td>0.041500</td>\n",
       "      <td>-0.003000</td>\n",
       "      <td>-0.002000</td>\n",
       "      <td>-0.124500</td>\n",
       "      <td>-0.014000</td>\n",
       "      <td>-0.060000</td>\n",
       "      <td>0.002500</td>\n",
       "      <td>0.064500</td>\n",
       "      <td>-0.001500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Ccrcc</td>\n",
       "      <td>-0.035195</td>\n",
       "      <td>-0.035823</td>\n",
       "      <td>0.132965</td>\n",
       "      <td>-0.017630</td>\n",
       "      <td>-0.032898</td>\n",
       "      <td>-0.014239</td>\n",
       "      <td>-0.033200</td>\n",
       "      <td>-0.031930</td>\n",
       "      <td>-0.045139</td>\n",
       "      <td>-0.045139</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031930</td>\n",
       "      <td>-0.031275</td>\n",
       "      <td>-0.032898</td>\n",
       "      <td>-0.023641</td>\n",
       "      <td>-0.034182</td>\n",
       "      <td>-0.026517</td>\n",
       "      <td>-0.027801</td>\n",
       "      <td>0.120693</td>\n",
       "      <td>-0.015314</td>\n",
       "      <td>-0.032643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Hnscc</td>\n",
       "      <td>-0.029787</td>\n",
       "      <td>-0.048607</td>\n",
       "      <td>-0.019600</td>\n",
       "      <td>-0.024264</td>\n",
       "      <td>-0.042857</td>\n",
       "      <td>-0.023380</td>\n",
       "      <td>-0.045898</td>\n",
       "      <td>-0.043826</td>\n",
       "      <td>-0.157708</td>\n",
       "      <td>-0.157708</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.048937</td>\n",
       "      <td>-0.029514</td>\n",
       "      <td>-0.033082</td>\n",
       "      <td>-0.034168</td>\n",
       "      <td>-0.057806</td>\n",
       "      <td>-0.004886</td>\n",
       "      <td>0.002269</td>\n",
       "      <td>-0.040452</td>\n",
       "      <td>-0.024298</td>\n",
       "      <td>-0.012427</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             MDM2_CNV  CSNK2A1_CNV  PDGFRB_CNV  PIK3R2_CNV  PIK3C3_CNV  \\\n",
       "Cancer                                                                   \n",
       "Gbm         -0.112350    -0.062947   -0.039516    0.027473   -0.016396   \n",
       "Endometrial  0.010000     0.010000    0.015000    0.010000    0.000000   \n",
       "Ovarian     -0.072600     0.029040   -0.081950         NaN   -0.069850   \n",
       "Colon        0.002500    -0.336000   -0.004000    0.050500   -0.315500   \n",
       "Ccrcc       -0.035195    -0.035823    0.132965   -0.017630   -0.032898   \n",
       "Hnscc       -0.029787    -0.048607   -0.019600   -0.024264   -0.042857   \n",
       "\n",
       "             CSNK2A2_CNV  SLC9A3R1_CNV  PIK3R3_CNV  USP13_CNV  PIK3CA_CNV  \\\n",
       "Cancer                                                                      \n",
       "Gbm            -0.118437     -0.034208   -0.071091  -0.008994   -0.005148   \n",
       "Endometrial     0.015000      0.020000    0.010000   0.010000    0.010000   \n",
       "Ovarian         0.034150      0.038350    0.015700   0.042050    0.062450   \n",
       "Colon          -0.004000      0.022500    0.015000   0.007000    0.002000   \n",
       "Ccrcc          -0.014239     -0.033200   -0.031930  -0.045139   -0.045139   \n",
       "Hnscc          -0.023380     -0.045898   -0.043826  -0.157708   -0.157708   \n",
       "\n",
       "             ...  MAST2_CNV  PIK3CD_CNV  ROCK1_CNV  SHC1_CNV  NEDD4_CNV  \\\n",
       "Cancer       ...                                                          \n",
       "Gbm          ...  -0.071091   -0.046664  -0.010449 -0.110476  -0.032116   \n",
       "Endometrial  ...   0.010000    0.020000   0.000000 -0.020000   0.020000   \n",
       "Ovarian      ...   0.017500    0.160550   0.075700 -0.064400  -0.153428   \n",
       "Colon        ...   0.028500    0.041500  -0.003000 -0.002000  -0.124500   \n",
       "Ccrcc        ...  -0.031930   -0.031275  -0.032898 -0.023641  -0.034182   \n",
       "Hnscc        ...  -0.048937   -0.029514  -0.033082 -0.034168  -0.057806   \n",
       "\n",
       "             INPP4B_CNV  PIK3R1_CNV  AKT1_CNV   MVP_CNV  MAGI3_CNV  \n",
       "Cancer                                                              \n",
       "Gbm            0.119056   -0.026384 -0.044454 -0.109663  -0.070727  \n",
       "Endometrial    0.020000    0.010000  0.010000  0.020000   0.010000  \n",
       "Ovarian       -0.079850   -0.003788  0.001850 -0.062550   0.005600  \n",
       "Colon         -0.014000   -0.060000  0.002500  0.064500  -0.001500  \n",
       "Ccrcc         -0.026517   -0.027801  0.120693 -0.015314  -0.032643  \n",
       "Hnscc         -0.004886    0.002269 -0.040452 -0.024298  -0.012427  \n",
       "\n",
       "[6 rows x 28 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Lung - PTEN not found in mutation dat\n",
    "\n",
    "n = gbm_df.join(e_df)\n",
    "m = n.join(ov_df)\n",
    "o = m.join(col_df)\n",
    "l = o.join(k_df)\n",
    "l = l.join(h_df)\n",
    "\n",
    "p = l.transpose()\n",
    "ip_df = p.dropna(axis = 1,thresh=4)\n",
    "#ip_df = ip_df.drop('NEDD4_proteomics', axis = 1) # fix -2 values for ov\n",
    "#ip_df = ip_df.reset_index()\n",
    "#ip_df = ip_df.rename(columns = {'index':'Cancer'})\n",
    "ip_df.index.name = 'Cancer'\n",
    "ip_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "ip_df.to_csv('cnv_del_vs_amp_interacting.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = ip_df.std()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = stdev.to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>EGFR_CNV</td>\n",
       "      <td>0.212839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PTEN_CNV</td>\n",
       "      <td>0.216789</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 0\n",
       "EGFR_CNV  0.212839\n",
       "PTEN_CNV  0.216789"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stdev.loc[stdev[0] > .2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    0.047669\n",
       "dtype: float64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d_stdev = stdev.std()\n",
    "d_stdev"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0, 0.5, 'Count')"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmYAAAElCAYAAABOCC7VAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3XmYZFV5+PHvC8M+wyY4kXVUFDWiKC1GjXFGiaK4JZqIERVjMkHj9hONGLO4BrKYaBbjkhhUhHGJGIWoIZHWmEhwRlE2F4RhX2R1GgEF398f5xRzp6aqu7qqu+vO9PfzPPV01z13ee+555773qWqIjORJEnS+G0z7gAkSZJUmJhJkiS1hImZJElSS5iYSZIktYSJmSRJUkuYmEmSJLVEqxOziFgZEVfN4fxWRERGxJK5mucsln1ARExFxLZDTv+HEfFPcx3XsCJieUR8NSI2RMS7xx2PphcRx0bE1wYc9+SIeOeQyxmpnY/bXPc5I8SxSV8VEV+IiJc2yt8ZETdGxHXzWed1vg+Y6/kOKyJeERHX17juM+54NL2IeH9E/PG445hr871eMyZmEfHLEfG/EXFbRNwcEf8TEY+pZQN39luCiFgfEXfUZOPWut7HRcTICWxmXpGZSzPzngHi2OzgkJl/lpm/M2occ2g1cCOwa2Ye311YD+4/rR3ozRFxVkQ8ZOHDHL8oXhMRF0TE7RFxVUR8KiIOqeUn14Pw4Y1pDoqIrP9/ICI+2mO+j4iIuyJiz4Vbm81iWB8RR3Tez6ada3CZ+fTM/AhAROwPHA88LDN/Ya7qPCImI2KTPqbO99JR5jtXImI74K+Bp9a4buoq7ySzU/W1PiJOaCSunVfW/bDz/old/VXn9e2u+Z7ZtbxTIuKtC7DeA58oRcRbI+KU+Y6pz7I3ywcy87jMfMc8LKvnth5hfpu1/enM13p1TJtwRMSuwBnA3wF7AvsCbwPumq+A5koMf1XsWZm5DDgQOAl4E/DPcxbY1uNA4KKc/huK/yIzlwL7ATcAJy9EYOMyTZt7L/Ba4DWU/ejBwGeBoxrj3Az063xPBn49InbpGv4S4IzMvHnYmLVFOhC4KTNvGHcgC2w5sCNw4Qzj7V77nRcCf0JJYJd2XnWcRzaG/Xcd9hfN8TLzkV3z/aWIeMKcrU0LjXDcHJdNtnVEHNk9wha4TpCZfV/ABHBrn7KHAncC9wBTnfEoB5tvAT8GrgTe2phmBZDAS4ErKFdc3tIo34lyELoFuAh4I3BVo/wE4IfAhlr+a42yY4H/Af6GjQe5bYG/qsu5FPj9uvwlfdZpPXBE17DDgZ8DD6/vd6jzvAK4Hng/sFMtuxh4ZmPaJXXZj26s+5Ja9rI6/oYa2+/V4bsAd9RlTtXXPsBbgVMa8342pYO6FZgEHtq1Hm8AvgPcBnwC2LGW7UVJtm+t9fTfwDZ96uPxwDfqPL4BPL4OPxn4GfDTGt8RPaY9GXhn4/1RwFSjTr9eY7gW+Htg+8a4CRwH/KC2hX8AolH+u426uwh49IDt42t1290CXAY8vVG+J/AvwDW1/LONsmcC59V4/xd4RFddv6nW9V10tS3gQZR95PBp9rOTKVcCrgOeVIcdBGRjnO8BL2m837bG+uw+87wP8DnKfngu8A7ga43yhwBn1TbwPeA3p9l2Pdcf+Bilnd5R28EfsHk736fGcTNwCfC7jfm+Ffgk8NG6zS4EJhrlbwKurmXfA57SZ13nrc/psaz31mX8GFgHPLFrfT4FnFJjPp+ShL+ZcmJyJeVqT2f8SeDEun1uA/4N2LMr7iWNcX8HOIJN+4eTe4zbsy0De1D2/R/V4WcA+9Wyd1Ha6Z11vn/f2BcPqv/vVrfVj4DLgT+i9h3MvH8dS+nnNtSyF/Wp3x2A99TYr6n/71Dr8fYazxTw5R7TblIPddg3gDd0jXfvOvVr833m+ybg7MbwU2i0ta5ptqn1c3nd9h8FdhukTU7Xl043LXAkpU/+Wa2jbze22z9T+tqrqcfGxnbpPm4+EPgycFOd/8cpCVAnnv2Bz9R2cBOl/+6XDzRjXwlcRbnae0ON52VdfdbnKfvWN2osX5thm/Tc1rXs9ynHkMvqsH7Hs35tf6A+coD1egalb9lQ6/8NvdZpk/WbthB2rRX/EeDpwB5d5cd2V1wN8hBKw3wEJXl5bldlfojSIT6ScjB7aC0/iZIo7Fk3/gVsmpj9BqWj3wZ4AWVHvV8jlruBV1MSop0oB/fv1nntCZzdvTG7Yl9P7yTjCuAV9f/3UA40ewLLakM6sZb9CfDxrgPGd/t0tEdRdoAAngT8hI0Jxkq6Dg40EjM2dlK/CmxHOSBeQk1u6nqcW+tqT0oSc1wtO5GSTG5XX0+kkfQ0lrcnpYN9ca3PF9b395mpI+vRcJcCpwL/Xd8fBvxSne+KGt/rujrOM4DdgQMoHcCRjTZwNfCYWncHAQcO2D5+RknqtgVeQen4o5afSUlg96j18qQ6/NGUne2xdbqX1vrdoVHX51Ha2E496uE44PIZ9rOTKZ3Qa6j7E5snZm8B/rPx/mm1XrbrM881lKRnF+Dhtc46896FkiS8rG6DR1M64F/sse0GWf8jGstdwabt/CvA+yhXOg6tMT+l0abvpHRc21La5jm17OAa4z6N+T6wz7quZJ76nB7LOoZyAFlC6YivY+NJT2d9nlbLP0pJQt5CaVO/Sz1I1PEn63Z5eN0m/8rGfby7HieB3+nVP/QYt19bvg/wPGBnSt/1KTY9Abl3GV37Yicx+ygleVxWl/l94OUz7V913X4MHFzHvR+1rfWo37cD5wD3BfamnAi8o9d69pj23vK63CdQ+tWn9FunXv3VNPNdWrfXEXX4dInZb1P65AfU6T4DfGyQNjlDXzrttHSdwNdhnwU+ULfDfSnHhs6FgGPZ/Lh5EOXYskPdBl8F3lPH3xb4NiWR24WyX/9yY17d+UAz9pV1WW+ntMtn1O2zR6PPWkNpnw+j7P8zJma9tnUtO4uyX+/EzMezSRptn9n1kTOt17XUEzjKPvno6Y4HmTMkZnVGD61BXFUX/jlgeb8N0WP69wB/01WZ+zXKzwWOrv9fSj0A1/ermb6TPA94TiOWK7rKv0xNSOr7pzJcYnYOpXMNysH+gY2yx7ExIz+IkhXvXN9/HPiTATuVzwKvbWzo6RKzPwY+2SjbhtJhrGysxzGN8r8A3t/o+P6Nro6pRzwvBs7tGvZ14NiZOrJG+Z2UqyzX1XbT78D6OuD0xvuk7uz1/SeBE+r/X+rU0wBtt7t9XNIo27ku5xcoB4qf03XiUcf7R+qBoTHse2w82K0HfnuaGN5CTTZmqKt3UjrCKygnQd2J2QGUA1/nCsfHgff2md+2ddyHNIb9GRsTsxdQk+RG+QeAP+3etgOuf8/EjJLo3AMsa5SfCJzcaNPNZPNhwB2NfekGyhWinsnnNPU5b31Oj2XdQrkt1lmfsxplz6KcgXeuTiyrsexe308CJ3Wt/0/r9ru3HhvjzpiYMU1b7hH7ocAtjff3LqNrXzyoxnQX5bZgp+z3gMkB9q9dKP3A8+hx8tK1vB8Cz2i8fxqwvns9+0zbKb+1bpeLgdf0GK9fYtbprzqvj/So31ey8eRhusTsv4BXNt4fTNknlzBDm+zXPwzYnt/KpndWltfttlNj2AupV/7ocdzssfznAt+q/z+OcnK12TZgsMTsDja9ynUD5SS902cd3Cgb5IpZz21dy57ceD/T8WySTROz2fSRfder/n8FZV/ZdaZ9svOa8aH2zLw4M4/NzP0oZ3b7UDq+niLisRFxdkT8KCJuo1wx2KtrtOsa//+EckZBnfeVjbLLu+b9kog4rz6Yf2uNpznv5rQzzm8W9qVcztyb0uGsa8TwxTqczLyE0kCeFRE7U243ntprhhHx9Ig4pz4Yfysly+6up372aa5LZv6csp77NsbpV8d/STmT+4+IuHSaByY3WUZ1edcyZvJXmbl7lgeUn52ZPwSIiAdHxBlRPlH2Y0rSMGgb2Z/SeW9mgPZx7zwz8yf136V1njdn5i09ZnsgcHxnnnW++1Pqp6O73TXdRDlYzigz76LccnwH5SSgWXYF5cz1mIhYSuksP9JnVntTDgD92v6BwGO71ulFlINot0HWv599KPW6oSuO6drpjhGxpO5Lr6McaG6IiDUR0XOZ89nn9FjW8RFxcZQPQ91KuU3UXNb1jf/vAG7MjQ/l31H/Lm2M073s7XrEPht923JE7Fw/SHJ53e++Cuweg32acy9gezatn77bsrl/ZebtlAPdccC1EXHmNB8E6u53LmewtrZJrJm5R2Y+NDP/dhbTdfqrzuulPcb5ELA8Ip41w7x6rccSSqLU0a9NDmLQaQ+ktKlrG/vvByhXzjo26b8i4r51f7u6tpNT2Ngm96fcAbh7FrE23dQ1bSf2Xn3WdP1qx3Tbujn9bI9ns+kjof96QTkheQZweUR8JSIeN8M6ze7rMjLzu5RM8eGdQT1GO5VydWT/zNyNctsseozXy7WUDd9xQOefiDiQslO8inL5cXfKbYfmvLvj6Tu/QUX5BOq+lOcnbqR0rr/Y2Hl3y40PlAKcRjkjeQ7l4fhLesxzB8pti7+iXH3cHfj3xrr0qtemaygNpzO/oKzn1TOtT2ZuyMzjM/MBlDP610fEU2ZaRnXAIMsYwD9SbjE/KDN3Bf6QwdvIlZRbwJsYsH1MN889I2L3PmXv6uqwd87M0xrjTLe9/gvYLyImBogDyrNBuwG/1qPsI5QH/p9HuUr7zT7z+BHl6na/tn8l8JWudVqama/oMa+Z1n+6db+GUq/LuuIYqA1l5qmZ+cuUdpjAn/cZdV76nG4R8UTKc0a/SbkitTvleZVBl9VL97J/RulnhjVdWz6ecuXmsXW/+5U6fJB+58YaW7NPmM22/FJm/irlJOW7lH21l+5+54A6rBUy82eUD8BtdvLUpdd63M2mift86N6GV1KumO3V2H93zcxfnGaaE+uwR9R2cgwb1/VK4IA+D9TPdNyaTqfP2q8xbP8+4w6qGc9Mx7Ne9TZoHzl9EJnfyMznUJLhz1LuAE1rpk9lPqSeIe5X3+9PSTrOqaNcTznobN+YbBnljO3OKB///61ZrMMngTdHxB51ma9ulO1Cqbwf1VhexsYEcbr5vSYi9ouIPSgPhw8kInaNiGdS7nmfkpnn1ytTHwL+JiLuW8fbNyKe1ph0DeWW6Svoc7WMcua5Q12XuyPi6XWajuuB+0TEbtOs11ER8ZQoHyE/nrLz/e8A6/XMKF/FEJTnPu6pr27/Djw4In4rIpZExAsot1rOmGkZA1hWlz1Vz5xn09j/CXhDRBwWxUE1KRumfQCQmdcCXwDeV9vedhHROWh9CDiuXpWJiNglIo7qSjamm/cPKM9YnRbla1C2j4gdI+LoXlcr61nXWykJQLd/pXRWb6P/1TLqFZrPAG+tV0keRnk2rOMMyrZ9cV3X7SLiMRHx0B6zm2n9r6c8R9MrjispbfLEus6PAF5OuQ07rYg4OCKeXE9i7qScEPX7Ooj56nN6Ledu6q2ciPgTynO4ozgmIh4W5Qr724FP5whfezFDW15Gqcdbo3zFyp92TT7dtryHUlfviohldZ97PeVqyrSifOfhs6N8qvguyu3dfut4GvBHEbF3ROxFeW53LF//MI2PUfrvzT4B2HAa8P8i4v5RrnD/GfCJEa40Dep6YEXUr3iq7eE/gHfXY9o2EfHAiHjSNPNYRn2APyL2pXwgpuNcysnMSbUv2DE2flK1Vz4wkB591kMoJ6FzZabjWXfbn00f2Vft718UEbvVpL5zzJ3WTFfMNlAe+v2/iLidkpBdQEkEoDzDdSFwXUR0zvJeCbw9IjZQdqoZs8OGt1EuL15GaUwf6xRk5kXAuyn3ha+nPOz7PzPM70OUZ5K+DXyTsuFn8vka+5WU54P+mvIAYMebKLcCz4lymfc/KWehnTivrTE+nvIA7mbqrZ3XUOrmFsqB5HON8u9SduxLo1xG3adr+u9RzmL+jnIm+yzK13z8dID1e1CNearG+b7MnOwR402UT+MdT7kd9weUT5yOcjbf8QbKOm+gbKOe9dRLZn6K8imaU+v0n6V8km2Y9tH0YsoVge9Sng94XV3eWsoDzX9P2VaXUJ6lmI3X1On/gfJMxA8pV8Q+32f80yid3ybqLaFOcjZTcvMqyqX06yhXuf+lMZ8NlBOBoylnktdRrkbt0GOZM63/iZQD6a0R8YYecbyQ8jzINcDplGc0zpohdmosJ1Ha93WUs80/7DPuvPQ5PXyJkvR8v05zJ4PdbpnOxyjb5zrKg9SvGXF+0KctUx5B2YlSp+dQHsNoei/w/Ii4JSJ63QJ8NeUZ20spdxBOBT48QDzbUPqRayiPhDyJss16eSewlvIp5/Mp/fZQX3Y8hD+ITb/HrGdfV5OIP6U8UN7Phynb9quUtnUn0yf9c+VT9e9NEdG5ov4SysWAiyj78KeZ/vGKt1Eedr+N8kGSe4+bdd2fRXnu8ArKs+cvqMW98oHZeBXlbsF1lLo7jTn6aq4BjmebtP3Z9JEDeDGwvuYLx1GO3dPqfCJNkrSAImKScjW+Nb/oIbVFRPw58At9nvXbqrX6J5kkSdLWL8qjU4+oj0wcTnns4fRxxzUOW9434kqSpK3NMsrty30ot+DfTflqp0XHW5mSJEkt4a1MSZKkljAxkyRJagkTM0mSpJYwMZMkSWoJEzNJkqSWMDGTJElqCRMzSZKkljAxkyRJagkTM0mSpJYwMZMkSWoJEzNJkqSWMDGTJElqCRMzSZKkljAxkyRJagkTM0mSpJYwMZMkSWqJJeMOQFuWvfbaK1esWDHr6W6//XZ22WWXuQ9oDrQ5NjC+URnfaNocX5tjg43xrVu37sbM3Hvc8WjLYGKmWVmxYgVr166d9XSTk5OsXLly7gOaA22ODYxvVMY3mjbH1+bYYGN8EXH5uGPRlsNbmZIkSS1hYiZJktQSJmaSJEktYWImSZLUEiZmkiRJLWFiJkmS1BImZotcROweEZ+OiO9GxMUR8bhxxyRJ0mLl95jpvcAXM/P5EbE9sPO4A5IkabEyMVvEImJX4FeAYwEy86fAT8cZkyRJi1lk5rhj0JhExKHAB4GLgEcC64DXZubtXeOtBlYDLF++/LA1a9bMellTU1MsXbp06FjPv/q2oac9ZN/dpi0fNbb5ZnyjMb7RtDm+NscGG+NbtWrVusycGHc82jKYmC1iETEBnAM8ITP/LyLeC/w4M/+43zQTExM5jp9kWnHCmUNPu/6ko6Yt31J+1qWtjG80xje8NscGm/wkk4mZBubD/4vbVcBVmfl/9f2ngUePMR5JkhY1E7NFLDOvA66MiIProKdQbmtKkqQx8OF/vRr4eP1E5qXAy8YcjyRJi5aJ2SKXmecBPvsgSVILeCtTkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwkSZJawsRMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwkSZJawsRMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwkSZJawsRMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwkSZJawsRMkiSpJZaMOwCNV0SsBzYA9wB3Z+bEeCOSJGnxMjETwKrMvHHcQUiStNh5K1OSJKklIjPHHYPGKCIuA24BEvhAZn6wxzirgdUAy5cvP2zNmjWzXs7U1BRLly4dOs7zr75t6GkP2Xe3actHjW2+Gd9ojG80bY6vzbHBxvhWrVq1zsdENCgTs0UuIvbJzGsi4r7AWcCrM/Or/cafmJjItWvXzno5k5OTrFy5cug4V5xw5tDTrj/pqGnLR41tvhnfaIxvNG2Or82xwcb4IsLETAPzVuYil5nX1L83AKcDh483IkmSFi8Ts0UsInaJiGWd/4GnAheMNypJkhYvP5W5uC0HTo8IKG3h1Mz84nhDkiRp8TIxW8Qy81LgkeOOQ5IkFd7KlCRJagkTM0mSpJYwMZMkSWoJEzNJkqSWMDGTJElqCRMzSZKkljAxkyRJagkTM0mSpJYwMZMkSWoJEzNJkqSWMDGTJElqCRMzSZKkljAxkyRJagkTM0mSpJYwMZMkSWoJEzNJkqSWMDGTJElqCRMzSZKkljAxkyRJagkTM0mSpJYwMZMkSWoJEzNJkqSWMDGTJElqCRMzSZKkljAxkyRJagkTM0mSpJYwMRMRsW1EfCsizhh3LJIkLWYmZgJ4LXDxuIOQJGmxMzFb5CJiP+Ao4J/GHYskSYtdZOa4Y9AYRcSngROBZcAbMvOZPcZZDawGWL58+WFr1qyZ9XKmpqZYunTp0HGef/VtQ097yL67TVs+amzTmYu45zO+uWB8ozG+4bU5NtgY36pVq9Zl5sS449GWYcm4A9D4RMQzgRsyc11ErOw3XmZ+EPggwMTERK5c2XfUviYnJxlmuo5jTzhz6GnXv2j65Y4a23TmIu75jG8uGN9ojG94bY4N2h+f2slbmYvbE4BnR8R6YA3w5Ig4ZbwhSZK0eJmYLWKZ+ebM3C8zVwBHA1/OzGPGHJYkSYuWiZkkSVJL+IyZAMjMSWByzGFIkrSoecVMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwkSZJawsRMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMy2EhHxhEGGSZKk9jIx23r83YDDJElSSy0ZdwAaTUQ8Dng8sHdEvL5RtCuw7XiikiRJwzAx2/JtDyylbMtljeE/Bp4/logkSdJQTMy2cJn5FeArEXFyZl4+7ngkSdLwTMy2HjtExAeBFTS2a2Y+eWwRSZKkWTEx23p8Cng/8E/APWOORZIkDcHEbOtxd2b+47iDkCRJw/PrMrYen4+IV0bE/SJiz85r3EFJkqTBecVs6/HS+veNjWEJPGAMsUiSpCGYmG0lMvP+445BkiSNxsRsKxERL+k1PDM/utCxSJKk4ZiYbT0e0/h/R+ApwDcBEzNJkrYQJmZbicx8dfN9ROwGfGxM4UiSpCH4qcyt10+AB003QkTsGBHnRsS3I+LCiHjbAsUmSZJ68IrZViIiPk/5FCaUHy9/KPDJGSa7C3hyZk5FxHbA1yLiC5l5zjyGKkmS+jAx23r8VeP/u4HLM/Oq6SbIzASm6tvt6iv7TyFJkuZTlGOztgYRsZyNHwI4NzNvGGCabYF1wEHAP2Tmm3qMsxpYDbB8+fLD1qxZM+vYpqamuOy2dv5S1PKd4Po7xh1Ff8PEd8i+u81PMD1MTU2xdOnSBVvebBnfaNocX5tjg43xrVq1al1mTow7Hm0ZTMy2EhHxm8BfApNAAE8E3piZnx5w+t2B04FXZ+YF/cabmJjItWvXzjq+yclJjv3i7bOebiEcf8jdvPv89l48Hia+9ScdNU/RbG5ycpKVK1cu2PJmy/hG0+b42hwbbIwvIkzMNLD2Ho00W28BHtO5ShYRewP/CQyUmGXmrRExCRwJ9E3MJEnS/PFTmVuPbbpuXd7EDNs3IvauV8qIiJ2AI4Dvzl+IkiRpOl4x23p8MSK+BJxW378A+PcZprkf8JH6nNk2wCcz84x5jFGSJE3DxGwLFxEHAcsz840R8evAL1OeMfs68PHpps3M7wCPmv8oJUnSILyVueV7D7ABIDM/k5mvz8z/R7la9p6xRiZJkmbFxGzLt6Je+dpEZq4FVix8OJIkaVgmZlu+Hacp22nBopAkSSMzMdvyfSMifrd7YES8nPLFsZIkaQvhw/9bvtcBp0fEi9iYiE0A2wO/NraoJEnSrJmYbeEy83rg8RGxCnh4HXxmZn55jGFJkqQhmJhtJTLzbODsccchSZKG5zNmkiRJLWFiJkmS1BImZpIkSS1hYiZJktQSJmaSJEktYWImSZLUEiZmkiRJLWFiJkmS1BImZpIkSS1hYiZJktQSJmaSJEktYWImSZLUEiZmkiRJLWFiJkmS1BImZpIkSS1hYiZJktQSJmaSJEktYWImSZLUEiZmkiRJLWFitohFxP4RcXZEXBwRF0bEa8cdkyRJi9mScQegsbobOD4zvxkRy4B1EXFWZl407sAkSVqMvGK2iGXmtZn5zfr/BuBiYN/xRiVJ0uIVmTnuGNQCEbEC+Crw8Mz8cVfZamA1wPLlyw9bs2bNrOc/NTXFZbfdM3qg82D5TnD9HeOOor9h4jtk393mJ5gepqamWLp06YItb7aMbzRtjm+2sZ1/9W1DL2uYfaoT36pVq9Zl5sTQC9eiYmImImIp8BXgXZn5menGnZiYyLVr1856GZOTkxz7xduHjHB+HX/I3bz7/Pbe1R8mvvUnHTVP0WxucnKSlStXLtjyZsv4RtPm+GYb24oTzhx6WcPsU534IsLETAPzVuYiFxHbAf8KfHympEySJM0vE7NFLCIC+Gfg4sz863HHI0nSYmditrg9AXgx8OSIOK++njHuoCRJWqza+2CN5l1mfg2IccchSZIKr5hJkiS1hImZJElSS5iYSZIktYSJmSRJUkuYmEmSJLWEiZkkSVJLmJhJkiS1hImZJElSS5iYSZIktYSJmSRJUkuYmEmSJLWEiZkkSVJLmJhJkiS1hImZJElSS5iYSZIktYSJmSRJUkuYmEmSJLWEiZkkSVJLmJhJkiS1hImZJElSS5iYSZIktYSJmSRJUkuYmEmSJLWEiZkkSVJLmJhJkiS1hInZIhYRH46IGyLignHHIkmSTMwWu5OBI8cdhCRJKkzMFrHM/Cpw87jjkCRJRWTmuGPQGEXECuCMzHz4NOOsBlYDLF++/LA1a9bMejlTU1Ncdts9Q0Y5v5bvBNffMe4o+tvS4jtk392Gntf5V9829LT9ljs1NcXSpUsXfLmDGiS+cWpzfLONbaG3cye+VatWrcvMiaEXrkXFxGyRGyQxa5qYmMi1a9fOejmTk5Mc+8XbZz3dQjj+kLt59/lLxh1GX1tafOtPOmroea044cyhp+233MnJSVauXLngyx3UIPGNU5vjm21sC72dO/FFhImZBuatTEmSpJYwMZMkSWoJE7NFLCJOA74OHBwRV0XEy8cdkyRJi1n6oANxAAAJS0lEQVR7H1zRvMvMF447BkmStJFXzCRJklrCxEySJKklTMwkSZJawsRMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwkSZJawsRMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwkSZJawsRMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwkSZJawsRMkiSpJUzMJEmSWsLETJIkqSVMzCRJklrCxEySJKklTMwWuYg4MiK+FxGXRMQJ445HkqTFzMRsEYuIbYF/AJ4OPAx4YUQ8bLxRSZK0eJmYLW6HA5dk5qWZ+VNgDfCcMcckSdKiFZk57hg0JhHxfODIzPyd+v7FwGMz81Vd460GVte3BwPfG2JxewE3jhDufGpzbGB8ozK+0bQ5vjbHBhvjOzAz9x53MNoyLBl3ABqr6DFss0w9Mz8IfHCkBUWszcyJUeYxX9ocGxjfqIxvNG2Or82xQfvjUzt5K3NxuwrYv/F+P+CaMcUiSdKiZ2K2uH0DeFBE3D8itgeOBj435pgkSVq0vJW5iGXm3RHxKuBLwLbAhzPzwnla3Ei3QudZm2MD4xuV8Y2mzfG1OTZof3xqIR/+lyRJaglvZUqSJLWEiZkkSVJLmJhp1mb6GaeI2CEiPlHL/y8iVtThvxoR6yLi/Pr3yY1pJus8z6uv+44hvhURcUcjhvc3pjmsxn1JRPxtRPT6qpH5ju9FjdjOi4ifR8ShtWwh6+9XIuKbEXF3/S68ZtlLI+IH9fXSxvA5qb9hY4uIQyPi6xFxYUR8JyJe0Cg7OSIua9TdocPENkp8teyeRgyfawy/f20HP6jtYvuFji8iVnW1vTsj4rm1bCHr7/URcVHdhv8VEQc2yua17Y0S30K1P20lMtOXr4FflA8J/BB4ALA98G3gYV3jvBJ4f/3/aOAT9f9HAfvU/x8OXN2YZhKYGHN8K4AL+sz3XOBxlO9++wLw9IWOr2ucQ4BLx1R/K4BHAB8Fnt8Yvidwaf27R/1/j7mqvxFjezDwoPr/PsC1wO71/cnNccdRd7Vsqs98PwkcXf9/P/CKccTXtZ1vBnYeQ/2taiz3FWzcd+e17c1BfPPe/nxtPS+vmGm2BvkZp+cAH6n/fxp4SkREZn4rMzvfk3YhsGNE7NCW+PrNMCLuB+yamV/PzKQctJ475vheCJw2ZAwjxZeZ6zPzO8DPu6Z9GnBWZt6cmbcAZwFHzmH9DR1bZn4/M39Q/78GuAGY629iH6Xueqrb/cmUdgClXcxb2xswvucDX8jMnwwZxyjxnd1Y7jmU716E+W97I8W3QO1PWwkTM83WvsCVjfdX1WE9x8nMu4HbgPt0jfM84FuZeVdj2L/US/l/PMLthlHju39EfCsivhIRT2yMf9UM81yo+DpewOaJ2ULV32ynnav6GyW2e0XE4ZQrHj9sDH5XvcX0NyOcLIwa344RsTYizuncJqRs91trOxhmnnMZX8fRbN72xlF/L6dcAZtu2oXed/vFd695bH/aSpiYabYG+RmnaceJiF8E/hz4vUb5izLzEOCJ9fXiMcR3LXBAZj4KeD1wakTsOuA8FyK+UhjxWOAnmXlBo3wh62+2085V/Y08n3oF5WPAyzKzc1XozcBDgMdQboW9aYjY5iK+A7L8fM9vAe+JiAfOwTyb5qr+DqF892HHgtdfRBwDTAB/OcO0Y6m/HvF1hs9n+9NWwsRMszXIzzjdO05ELAF2ozyTQkTsB5wOvCQz7z1jzMyr698NwKmU2wYLGl9m3pWZN9U41lHOaB9cx9+vMf0oP101Uv1Vm12xWOD6m+20c1V/I/2EWE2yzwT+KDPP6QzPzGuzuAv4F8ZTd51bXGTmpZRnBh9F+QHs3Ws7mPU85zK+6jeB0zPzZ50BC11/EXEE8Bbg2Y0r7vPd9kaNbyHan7YSJmaarUF+xulzQOdTUc8HvpyZGRG7UzqmN2fm/3RGjoglEbFX/X874JnABQxnlPj2johtaxwPAB5EecD+WmBDRPxSvUX4EuDfFjq+Gtc2wG9Qnm+hDlvo+uvnS8BTI2KPiNgDeCrwpTmsv6Fjq+OfDnw0Mz/VVXa/+jcozx8teN3VOtuh/r8X8ATgorrdz6a0AyjtYj7b3kw2e7ZxIesvIh4FfICS9NzQKJrvtjdSfAvU/rS1GOYTA74W9wt4BvB9yhWlt9Rhb6d0RgA7Ap8CLqF8IuoBdfgfAbcD5zVe9wV2AdYB36F8KOC9wLZjiO95dfnfBr4JPKsxzwlKh/lD4O+pv5qxkPHVspXAOV3zW+j6ewzl6sHtwE3AhY1pf7vGfQnlds2c1t+wsQHHAD/ranuH1rIvA+fX+E4Bli503QGPrzF8u/59eWOeD6jt4JLaLnYY07ZdAVwNbNM1z4Wsv/8Erm9sw88tVNsbJb6Fan++to6XP8kkSZLUEt7KlCRJagkTM0mSpJYwMZMkSWoJEzNJkqSWMDGTJElqCRMzSa0SEW+JiAvrT9ScFxGPjYjXRcTOjXHWR8T59XVRRLxzpp+yiYjJiHha17DXRcT75mtdJGm2TMwktUZEPI7yBbmPzsxHAEdQfp/wdcDOXaOvyvIzVIdTvuvrgzPM/jTKl4I29frdR0kaGxMzSW1yP+DGrD9lk5k3Ur71fh/g7Ig4u3uCzJwCjgOeGxF7AkTEGyPiG/Wq29vqqJ8Gntn4hv0Vdb5fm99VkqTBmZhJapP/APaPiO9HxPsi4kmZ+beU3yRclZmrek2UmT8GLqP8ZM5TKT+ndThwKHBYRPxKlt9BPRc4sk52NPCJ9Fu2JbWIiZmk1qhXvw4DVgM/Aj4REccOOHnUv0+tr29RflrrIZREDTa9neltTEmts2TcAUhSU2beA0wCkxFxPht/0L2viFhG+S3H71MStBMz8wM9Rv0s8NcR8Whgp8z85lzFLUlzwStmklojIg6OiAc1Bh0KXA5sAJb1mWYp8D7gs5l5C/Al4LfrcCJi34i4L9x7RW4S+DBeLZPUQl4xk9QmS4G/i4jdgbuBSyi3NV8IfCEirm08Z3Z2RATlBPN04B0AmfkfEfFQ4OulmCngGOCGOt1pwGfY/BOakjR24XOvkiRJ7eCtTEmSpJYwMZMkSWoJEzNJkqSWMDGTJElqCRMzSZKkljAxkyRJagkTM0mSpJb4/6+mbtj7DKZMAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "a = stdev.hist(bins = 20)\n",
    "plt.title(\"Standard Deviations of Pancancer CNV deletions and amplifications of PTEN on Interacting Proteins\\n\")\n",
    "plt.xlabel('StDeV')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(10,8)\n",
    "sns.set(font_scale = 1.3)\n",
    "a = sns.distplot(stdev, bins = 20)\n",
    "\n",
    "a.set_title(\"Standard Deviation Distribution for Pancancer PTEN Mutation Effect on Proteomics\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "a.set(xlabel = 'StDeV', ylabel = 'Frequency')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_df = stdev.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = mean_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Mean between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Mean')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m_df = stdev.median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = m_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Medians between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Median')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
