{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Phospho for Interacting Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that endometrial index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "gbm = cptac.Gbm()\n",
    "endo = cptac.Endometrial()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ip = u.get_interacting_proteins('PTEN')\n",
    "len(ip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gbm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the phosphoproteomics dataframe, so they were inserted into joined table, but filled with NaN: PIK3CB, PIK3R3, USP13 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>AKT1_phosphoproteomics_S122S124S126_S*GS*PS*DNSGAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_S122S129_S*GSPSDNS*GAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_S124_SGS*PSDNSGAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_S124S126_SGS*PS*DNSGAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_S124S126S129_SGS*PS*DNS*GAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_S124S129_SGS*PSDNS*GAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_S126_SGSPS*DNSGAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_S126S129_SGSPS*DNS*GAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_S129_SGSPSDNS*GAEEMEVSLAK</th>\n",
       "      <th>AKT1_phosphoproteomics_T450_YFDEEFTAQMITIT*PPDQDDSMECVDSER</th>\n",
       "      <th>...</th>\n",
       "      <th>USP7_phosphoproteomics_S49T54_ITQNPVINGNVALS*DGHNT*AEEDMEDDTSWR</th>\n",
       "      <th>USP7_phosphoproteomics_S963_IIGVHQEDELLECLS*PATSR</th>\n",
       "      <th>USP7_phosphoproteomics_T54_ITQNPVINGNVALSDGHNT*AEEDMEDDTSWR</th>\n",
       "      <th>USP7_phosphoproteomics_Y143S149S151_IINY*RDDEKS*FS*RR</th>\n",
       "      <th>XIAP_phosphoproteomics_S361T363_TPS*LT*RR</th>\n",
       "      <th>XIAP_phosphoproteomics_S406_S*LEVLVADLVNAQK</th>\n",
       "      <th>PTEN_Mutation__</th>\n",
       "      <th>PTEN_Location__</th>\n",
       "      <th>PTEN_Mutation_Status__</th>\n",
       "      <th>Sample_Status__</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.266735</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.003699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.278996</td>\n",
       "      <td>-0.258196</td>\n",
       "      <td>-0.023265</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.232297</td>\n",
       "      <td>-0.669862</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.066926</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.G251D]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.126213</td>\n",
       "      <td>0.167336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.444445</td>\n",
       "      <td>-0.306907</td>\n",
       "      <td>0.249322</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.628643</td>\n",
       "      <td>-0.133168</td>\n",
       "      <td>0.074903</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.099868</td>\n",
       "      <td>0.415818</td>\n",
       "      <td>-0.025736</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.442646</td>\n",
       "      <td>0.283930</td>\n",
       "      <td>0.080429</td>\n",
       "      <td>0.595095</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.251479</td>\n",
       "      <td>0.878238</td>\n",
       "      <td>-0.126044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.346832</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.C136Y]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.474102</td>\n",
       "      <td>0.157080</td>\n",
       "      <td>0.081324</td>\n",
       "      <td>-0.256534</td>\n",
       "      <td>-0.261877</td>\n",
       "      <td>-0.069383</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.476500</td>\n",
       "      <td>0.726599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.443770</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.R173H]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.213932</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.081431</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.187680</td>\n",
       "      <td>0.093658</td>\n",
       "      <td>0.149175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.228913</td>\n",
       "      <td>0.294451</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.155544</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>0.135839</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.048229</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.333319</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.073602</td>\n",
       "      <td>0.435285</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.104482</td>\n",
       "      <td>-0.721475</td>\n",
       "      <td>-0.144328</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.D252G]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.029779</td>\n",
       "      <td>0.562808</td>\n",
       "      <td>0.375545</td>\n",
       "      <td>-0.122466</td>\n",
       "      <td>-0.212794</td>\n",
       "      <td>0.082460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.311864</td>\n",
       "      <td>0.048687</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.190067</td>\n",
       "      <td>0.204214</td>\n",
       "      <td>-0.020094</td>\n",
       "      <td>0.208449</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.129185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.293418</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.024489</td>\n",
       "      <td>0.244612</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.455009</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>-0.212662</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.288704</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.050538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.330554</td>\n",
       "      <td>0.085500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.083619</td>\n",
       "      <td>0.086609</td>\n",
       "      <td>-0.803688</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>0.384405</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.485617</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.520556</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.275765</td>\n",
       "      <td>-0.124927</td>\n",
       "      <td>0.163806</td>\n",
       "      <td>0.013293</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.307149</td>\n",
       "      <td>-0.364647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.180454</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 347 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        AKT1_phosphoproteomics_S122S124S126_S*GS*PS*DNSGAEEMEVSLAK  \\\n",
       "Patient_ID                                                               \n",
       "C3L-00104                                                 NaN            \n",
       "C3L-00365                                                 NaN            \n",
       "C3L-00674                                                 NaN            \n",
       "C3L-00677                                                 NaN            \n",
       "C3L-01040                                                 NaN            \n",
       "...                                                       ...            \n",
       "C3N-03183                                            0.135839            \n",
       "C3N-03184                                                 NaN            \n",
       "C3N-03186                                                 NaN            \n",
       "C3N-03188                                           -0.212662            \n",
       "C3N-03473                                            0.384405            \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S122S129_S*GSPSDNS*GAEEMEVSLAK  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104                                                 NaN       \n",
       "C3L-00365                                                 NaN       \n",
       "C3L-00674                                                 NaN       \n",
       "C3L-00677                                                 NaN       \n",
       "C3L-01040                                                 NaN       \n",
       "...                                                       ...       \n",
       "C3N-03183                                                 NaN       \n",
       "C3N-03184                                                 NaN       \n",
       "C3N-03186                                                 NaN       \n",
       "C3N-03188                                                 NaN       \n",
       "C3N-03473                                                 NaN       \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S124_SGS*PSDNSGAEEMEVSLAK  \\\n",
       "Patient_ID                                                     \n",
       "C3L-00104                                          -0.266735   \n",
       "C3L-00365                                                NaN   \n",
       "C3L-00674                                          -0.099868   \n",
       "C3L-00677                                          -0.474102   \n",
       "C3L-01040                                           0.213932   \n",
       "...                                                      ...   \n",
       "C3N-03183                                          -0.048229   \n",
       "C3N-03184                                                NaN   \n",
       "C3N-03186                                           0.190067   \n",
       "C3N-03188                                          -0.288704   \n",
       "C3N-03473                                          -0.485617   \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S124S126_SGS*PS*DNSGAEEMEVSLAK  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104                                                 NaN       \n",
       "C3L-00365                                           -0.126213       \n",
       "C3L-00674                                            0.415818       \n",
       "C3L-00677                                            0.157080       \n",
       "C3L-01040                                                 NaN       \n",
       "...                                                       ...       \n",
       "C3N-03183                                                 NaN       \n",
       "C3N-03184                                            0.029779       \n",
       "C3N-03186                                            0.204214       \n",
       "C3N-03188                                                 NaN       \n",
       "C3N-03473                                                 NaN       \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S124S126S129_SGS*PS*DNS*GAEEMEVSLAK  \\\n",
       "Patient_ID                                                               \n",
       "C3L-00104                                           -0.003699            \n",
       "C3L-00365                                            0.167336            \n",
       "C3L-00674                                           -0.025736            \n",
       "C3L-00677                                            0.081324            \n",
       "C3L-01040                                            0.081431            \n",
       "...                                                       ...            \n",
       "C3N-03183                                            0.333319            \n",
       "C3N-03184                                            0.562808            \n",
       "C3N-03186                                           -0.020094            \n",
       "C3N-03188                                            0.050538            \n",
       "C3N-03473                                            0.520556            \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S124S129_SGS*PSDNS*GAEEMEVSLAK  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104                                                 NaN       \n",
       "C3L-00365                                                 NaN       \n",
       "C3L-00674                                                 NaN       \n",
       "C3L-00677                                           -0.256534       \n",
       "C3L-01040                                                 NaN       \n",
       "...                                                       ...       \n",
       "C3N-03183                                                 NaN       \n",
       "C3N-03184                                            0.375545       \n",
       "C3N-03186                                            0.208449       \n",
       "C3N-03188                                                 NaN       \n",
       "C3N-03473                                                 NaN       \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S126_SGSPS*DNSGAEEMEVSLAK  \\\n",
       "Patient_ID                                                     \n",
       "C3L-00104                                          -0.278996   \n",
       "C3L-00365                                          -0.444445   \n",
       "C3L-00674                                          -0.442646   \n",
       "C3L-00677                                          -0.261877   \n",
       "C3L-01040                                          -0.187680   \n",
       "...                                                      ...   \n",
       "C3N-03183                                                NaN   \n",
       "C3N-03184                                          -0.122466   \n",
       "C3N-03186                                                NaN   \n",
       "C3N-03188                                                NaN   \n",
       "C3N-03473                                          -0.275765   \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S126S129_SGSPS*DNS*GAEEMEVSLAK  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104                                           -0.258196       \n",
       "C3L-00365                                           -0.306907       \n",
       "C3L-00674                                            0.283930       \n",
       "C3L-00677                                           -0.069383       \n",
       "C3L-01040                                            0.093658       \n",
       "...                                                       ...       \n",
       "C3N-03183                                            0.073602       \n",
       "C3N-03184                                           -0.212794       \n",
       "C3N-03186                                           -0.129185       \n",
       "C3N-03188                                           -0.330554       \n",
       "C3N-03473                                           -0.124927       \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S129_SGSPSDNS*GAEEMEVSLAK  \\\n",
       "Patient_ID                                                     \n",
       "C3L-00104                                          -0.023265   \n",
       "C3L-00365                                           0.249322   \n",
       "C3L-00674                                           0.080429   \n",
       "C3L-00677                                                NaN   \n",
       "C3L-01040                                           0.149175   \n",
       "...                                                      ...   \n",
       "C3N-03183                                           0.435285   \n",
       "C3N-03184                                           0.082460   \n",
       "C3N-03186                                                NaN   \n",
       "C3N-03188                                           0.085500   \n",
       "C3N-03473                                           0.163806   \n",
       "\n",
       "Name        AKT1_phosphoproteomics_T450_YFDEEFTAQMITIT*PPDQDDSMECVDSER  ...  \\\n",
       "Patient_ID                                                              ...   \n",
       "C3L-00104                                                 NaN           ...   \n",
       "C3L-00365                                                 NaN           ...   \n",
       "C3L-00674                                            0.595095           ...   \n",
       "C3L-00677                                                 NaN           ...   \n",
       "C3L-01040                                                 NaN           ...   \n",
       "...                                                       ...           ...   \n",
       "C3N-03183                                                 NaN           ...   \n",
       "C3N-03184                                                 NaN           ...   \n",
       "C3N-03186                                           -0.293418           ...   \n",
       "C3N-03188                                                 NaN           ...   \n",
       "C3N-03473                                            0.013293           ...   \n",
       "\n",
       "Name        USP7_phosphoproteomics_S49T54_ITQNPVINGNVALS*DGHNT*AEEDMEDDTSWR  \\\n",
       "Patient_ID                                                                    \n",
       "C3L-00104                                                 NaN                 \n",
       "C3L-00365                                           -0.628643                 \n",
       "C3L-00674                                                 NaN                 \n",
       "C3L-00677                                                 NaN                 \n",
       "C3L-01040                                                 NaN                 \n",
       "...                                                       ...                 \n",
       "C3N-03183                                           -0.104482                 \n",
       "C3N-03184                                           -0.311864                 \n",
       "C3N-03186                                                 NaN                 \n",
       "C3N-03188                                           -0.083619                 \n",
       "C3N-03473                                                 NaN                 \n",
       "\n",
       "Name        USP7_phosphoproteomics_S963_IIGVHQEDELLECLS*PATSR  \\\n",
       "Patient_ID                                                      \n",
       "C3L-00104                                           -0.232297   \n",
       "C3L-00365                                           -0.133168   \n",
       "C3L-00674                                            0.251479   \n",
       "C3L-00677                                            0.476500   \n",
       "C3L-01040                                            0.228913   \n",
       "...                                                       ...   \n",
       "C3N-03183                                           -0.721475   \n",
       "C3N-03184                                            0.048687   \n",
       "C3N-03186                                           -0.024489   \n",
       "C3N-03188                                            0.086609   \n",
       "C3N-03473                                           -0.307149   \n",
       "\n",
       "Name        USP7_phosphoproteomics_T54_ITQNPVINGNVALSDGHNT*AEEDMEDDTSWR  \\\n",
       "Patient_ID                                                                \n",
       "C3L-00104                                           -0.669862             \n",
       "C3L-00365                                            0.074903             \n",
       "C3L-00674                                            0.878238             \n",
       "C3L-00677                                            0.726599             \n",
       "C3L-01040                                            0.294451             \n",
       "...                                                       ...             \n",
       "C3N-03183                                           -0.144328             \n",
       "C3N-03184                                                 NaN             \n",
       "C3N-03186                                            0.244612             \n",
       "C3N-03188                                           -0.803688             \n",
       "C3N-03473                                           -0.364647             \n",
       "\n",
       "Name        USP7_phosphoproteomics_Y143S149S151_IINY*RDDEKS*FS*RR  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104                                                 NaN       \n",
       "C3L-00365                                                 NaN       \n",
       "C3L-00674                                           -0.126044       \n",
       "C3L-00677                                                 NaN       \n",
       "C3L-01040                                                 NaN       \n",
       "...                                                       ...       \n",
       "C3N-03183                                                 NaN       \n",
       "C3N-03184                                                 NaN       \n",
       "C3N-03186                                                 NaN       \n",
       "C3N-03188                                                 NaN       \n",
       "C3N-03473                                                 NaN       \n",
       "\n",
       "Name        XIAP_phosphoproteomics_S361T363_TPS*LT*RR  \\\n",
       "Patient_ID                                              \n",
       "C3L-00104                                    0.066926   \n",
       "C3L-00365                                         NaN   \n",
       "C3L-00674                                         NaN   \n",
       "C3L-00677                                         NaN   \n",
       "C3L-01040                                   -0.155544   \n",
       "...                                               ...   \n",
       "C3N-03183                                         NaN   \n",
       "C3N-03184                                         NaN   \n",
       "C3N-03186                                         NaN   \n",
       "C3N-03188                                         NaN   \n",
       "C3N-03473                                         NaN   \n",
       "\n",
       "Name        XIAP_phosphoproteomics_S406_S*LEVLVADLVNAQK      PTEN_Mutation__  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00104                                           NaN  [Missense_Mutation]   \n",
       "C3L-00365                                           NaN     [Wildtype_Tumor]   \n",
       "C3L-00674                                     -0.346832  [Missense_Mutation]   \n",
       "C3L-00677                                     -0.443770  [Missense_Mutation]   \n",
       "C3L-01040                                           NaN     [Wildtype_Tumor]   \n",
       "...                                                 ...                  ...   \n",
       "C3N-03183                                           NaN  [Missense_Mutation]   \n",
       "C3N-03184                                           NaN     [Wildtype_Tumor]   \n",
       "C3N-03186                                      0.455009     [Wildtype_Tumor]   \n",
       "C3N-03188                                           NaN     [Wildtype_Tumor]   \n",
       "C3N-03473                                      0.180454     [Wildtype_Tumor]   \n",
       "\n",
       "Name        PTEN_Location__  PTEN_Mutation_Status__  Sample_Status__  \n",
       "Patient_ID                                                            \n",
       "C3L-00104         [p.G251D]         Single_mutation            Tumor  \n",
       "C3L-00365     [No_mutation]          Wildtype_Tumor            Tumor  \n",
       "C3L-00674         [p.C136Y]         Single_mutation            Tumor  \n",
       "C3L-00677         [p.R173H]         Single_mutation            Tumor  \n",
       "C3L-01040     [No_mutation]          Wildtype_Tumor            Tumor  \n",
       "...                     ...                     ...              ...  \n",
       "C3N-03183         [p.D252G]         Single_mutation            Tumor  \n",
       "C3N-03184     [No_mutation]          Wildtype_Tumor            Tumor  \n",
       "C3N-03186     [No_mutation]          Wildtype_Tumor            Tumor  \n",
       "C3N-03188     [No_mutation]          Wildtype_Tumor            Tumor  \n",
       "C3N-03473     [No_mutation]          Wildtype_Tumor            Tumor  \n",
       "\n",
       "[99 rows x 347 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = gbm.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = gbm.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'phosphoproteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "prot_and_mutations = gbm.reduce_multiindex(prot_and_mutations, levels_to_drop = 3, flatten = True)\n",
    "\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "del_wt\n",
    "prot_and_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTK2_phosphoproteomics_T660Y661_YMEDST*Y*YK</td>\n",
       "      <td>0.000220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S280_EALAEAALES*PRP...</td>\n",
       "      <td>0.001065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3R1_phosphoproteomics_S83_KIS*PPTPK</td>\n",
       "      <td>0.001357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EGFR_phosphoproteomics_S1042S1045_TPLLSSLSATS*...</td>\n",
       "      <td>0.001429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1_phosphoproteomics_S124S126S129_SGS*PS*DNS...</td>\n",
       "      <td>0.001594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S288T293_S*ASSDT*SE...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S291S294S299_SASS*D...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>USP13_phosphoproteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>USP7_phosphoproteomics_Y143S149S151_IINY*RDDEK...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>XIAP_phosphoproteomics_S361T363_TPS*LT*RR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>343 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Phosphoproteomics  Gbm_P_Value\n",
       "0          PTK2_phosphoproteomics_T660Y661_YMEDST*Y*YK     0.000220\n",
       "1    SLC9A3R1_phosphoproteomics_S280_EALAEAALES*PRP...     0.001065\n",
       "2               PIK3R1_phosphoproteomics_S83_KIS*PPTPK     0.001357\n",
       "3    EGFR_phosphoproteomics_S1042S1045_TPLLSSLSATS*...     0.001429\n",
       "4    AKT1_phosphoproteomics_S124S126S129_SGS*PS*DNS...     0.001594\n",
       "..                                                 ...          ...\n",
       "338  SLC9A3R1_phosphoproteomics_S288T293_S*ASSDT*SE...          NaN\n",
       "339  SLC9A3R1_phosphoproteomics_S291S294S299_SASS*D...          NaN\n",
       "340                            USP13_phosphoproteomics          NaN\n",
       "341  USP7_phosphoproteomics_Y143S149S151_IINY*RDDEK...          NaN\n",
       "342          XIAP_phosphoproteomics_S361T363_TPS*LT*RR          NaN\n",
       "\n",
       "[343 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "g = u.wrap_ttest(del_wt, 'Mutation', cols,return_all = True, correction_method = 'fdr_bh')\n",
    "g_pval = g.rename(columns = {'Comparison': 'Phosphoproteomics','P_Value': 'Gbm_P_Value'})\n",
    "g_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1_phosphoproteomics_S122S124S126_S*GS*PS*DN...</td>\n",
       "      <td>-0.278248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1_phosphoproteomics_S122S129_S*GSPSDNS*GAEE...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1_phosphoproteomics_S124_SGS*PSDNSGAEEMEVSLAK</td>\n",
       "      <td>0.098392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1_phosphoproteomics_S124S126_SGS*PS*DNSGAEE...</td>\n",
       "      <td>-0.264459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1_phosphoproteomics_S124S126S129_SGS*PS*DNS...</td>\n",
       "      <td>-0.299351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>USP7_phosphoproteomics_S963_IIGVHQEDELLECLS*PATSR</td>\n",
       "      <td>0.197240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>USP7_phosphoproteomics_T54_ITQNPVINGNVALSDGHNT...</td>\n",
       "      <td>-0.073719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>USP7_phosphoproteomics_Y143S149S151_IINY*RDDEK...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>XIAP_phosphoproteomics_S361T363_TPS*LT*RR</td>\n",
       "      <td>-0.041787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>XIAP_phosphoproteomics_S406_S*LEVLVADLVNAQK</td>\n",
       "      <td>-0.290781</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>343 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Phosphoproteomics  Gbm_Median\n",
       "0    AKT1_phosphoproteomics_S122S124S126_S*GS*PS*DN...   -0.278248\n",
       "1    AKT1_phosphoproteomics_S122S129_S*GSPSDNS*GAEE...         NaN\n",
       "2     AKT1_phosphoproteomics_S124_SGS*PSDNSGAEEMEVSLAK    0.098392\n",
       "3    AKT1_phosphoproteomics_S124S126_SGS*PS*DNSGAEE...   -0.264459\n",
       "4    AKT1_phosphoproteomics_S124S126S129_SGS*PS*DNS...   -0.299351\n",
       "..                                                 ...         ...\n",
       "338  USP7_phosphoproteomics_S963_IIGVHQEDELLECLS*PATSR    0.197240\n",
       "339  USP7_phosphoproteomics_T54_ITQNPVINGNVALSDGHNT...   -0.073719\n",
       "340  USP7_phosphoproteomics_Y143S149S151_IINY*RDDEK...         NaN\n",
       "341          XIAP_phosphoproteomics_S361T363_TPS*LT*RR   -0.041787\n",
       "342        XIAP_phosphoproteomics_S406_S*LEVLVADLVNAQK   -0.290781\n",
       "\n",
       "[343 rows x 2 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "gbm_d = {}\n",
    "\n",
    "for phos in cols:\n",
    "    dif_gbm = wt_med[phos] - del_med[phos]\n",
    "    gbm_d[phos] = dif_gbm\n",
    "\n",
    "gbm_df = pd.DataFrame.from_dict(gbm_d, orient='index', columns=['Gbm_Median'])\n",
    "gbm_df = gbm_df.reset_index().rename(columns={'index':'Phosphoproteomics'})\n",
    "gbm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTK2_T660Y661</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>-0.764033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SLC9A3R1_S280</td>\n",
       "      <td>0.001065</td>\n",
       "      <td>0.728675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3R1_S83</td>\n",
       "      <td>0.001357</td>\n",
       "      <td>0.580407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EGFR_S1042S1045</td>\n",
       "      <td>0.001429</td>\n",
       "      <td>-0.349234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1_S124S126S129</td>\n",
       "      <td>0.001594</td>\n",
       "      <td>-0.299351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>SLC9A3R1_S288T293</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>SLC9A3R1_S291S294S299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.559956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>USP13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>USP7_Y143S149S151</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>XIAP_S361T363</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.041787</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>343 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Phosphoproteomics  Gbm_P_Value  Gbm_Median\n",
       "0            PTK2_T660Y661     0.000220   -0.764033\n",
       "1            SLC9A3R1_S280     0.001065    0.728675\n",
       "2               PIK3R1_S83     0.001357    0.580407\n",
       "3          EGFR_S1042S1045     0.001429   -0.349234\n",
       "4        AKT1_S124S126S129     0.001594   -0.299351\n",
       "..                     ...          ...         ...\n",
       "338      SLC9A3R1_S288T293          NaN         NaN\n",
       "339  SLC9A3R1_S291S294S299          NaN   -0.559956\n",
       "340                  USP13          NaN         NaN\n",
       "341      USP7_Y143S149S151          NaN         NaN\n",
       "342          XIAP_S361T363          NaN   -0.041787\n",
       "\n",
       "[343 rows x 3 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_merged = g_pval.merge(gbm_df, on='Phosphoproteomics',how='outer')\n",
    "g_merged = g_merged.replace(to_replace = r'_phosphoproteomics', value = '', regex = True)\n",
    "g_merged = g_merged.replace(to_replace = r'_[A-Z*]*$', value = '', regex = True)\n",
    "g_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the phosphoproteomics dataframe, so they were inserted into joined table, but filled with NaN: CSNK2A1, PIK3CB, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 69 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>AKT1_phosphoproteomics_S122</th>\n",
       "      <th>AKT1_phosphoproteomics_S124</th>\n",
       "      <th>AKT1_phosphoproteomics_S126</th>\n",
       "      <th>AKT1_phosphoproteomics_S129</th>\n",
       "      <th>AKT1_phosphoproteomics_T450</th>\n",
       "      <th>CSNK2A1_phosphoproteomics</th>\n",
       "      <th>CSNK2A2_phosphoproteomics_S18</th>\n",
       "      <th>CSNK2A2_phosphoproteomics_S21</th>\n",
       "      <th>EGFR_phosphoproteomics_S1026</th>\n",
       "      <th>EGFR_phosphoproteomics_S1037</th>\n",
       "      <th>...</th>\n",
       "      <th>XIAP_phosphoproteomics_S261</th>\n",
       "      <th>XIAP_phosphoproteomics_S40</th>\n",
       "      <th>XIAP_phosphoproteomics_S402</th>\n",
       "      <th>XIAP_phosphoproteomics_S406</th>\n",
       "      <th>XIAP_phosphoproteomics_S427</th>\n",
       "      <th>XIAP_phosphoproteomics_S430</th>\n",
       "      <th>PTEN_Mutation_</th>\n",
       "      <th>PTEN_Location_</th>\n",
       "      <th>PTEN_Mutation_Status_</th>\n",
       "      <th>Sample_Status_</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.04605</td>\n",
       "      <td>0.14810</td>\n",
       "      <td>-0.04605</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.227</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.437</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Missense_Mutation, Nonsense_Mutation]</td>\n",
       "      <td>[p.R130Q, p.R233*]</td>\n",
       "      <td>Multiple_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00008</th>\n",
       "      <td>-0.2500</td>\n",
       "      <td>-0.26300</td>\n",
       "      <td>-0.12100</td>\n",
       "      <td>-0.18550</td>\n",
       "      <td>-0.00876</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0107</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.90</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.G127R]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.21400</td>\n",
       "      <td>0.05050</td>\n",
       "      <td>0.03195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Nonsense_Mutation]</td>\n",
       "      <td>[p.W111*]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00090</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00300</td>\n",
       "      <td>-0.02820</td>\n",
       "      <td>0.04540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.07190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.203</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.R130G]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.18965</td>\n",
       "      <td>-0.03930</td>\n",
       "      <td>-0.07715</td>\n",
       "      <td>-0.30100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.710</td>\n",
       "      <td>0.424</td>\n",
       "      <td>0.614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.35400</td>\n",
       "      <td>-0.30800</td>\n",
       "      <td>-0.17250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.519</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.163</td>\n",
       "      <td>-0.333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Frame_Shift_Ins, Frame_Shift_Del]</td>\n",
       "      <td>[p.D268Gfs*30, p.N323Mfs*21]</td>\n",
       "      <td>Multiple_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.16350</td>\n",
       "      <td>-0.09890</td>\n",
       "      <td>-0.12995</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.277</td>\n",
       "      <td>-0.511</td>\n",
       "      <td>-0.267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>0.0241</td>\n",
       "      <td>0.02410</td>\n",
       "      <td>-0.01185</td>\n",
       "      <td>-0.04780</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>-0.321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>0.3130</td>\n",
       "      <td>-0.71750</td>\n",
       "      <td>-0.12300</td>\n",
       "      <td>-0.38450</td>\n",
       "      <td>-0.00828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.630</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.15</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>0.1410</td>\n",
       "      <td>0.14100</td>\n",
       "      <td>0.19050</td>\n",
       "      <td>0.15855</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.211</td>\n",
       "      <td>0.699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 232 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        AKT1_phosphoproteomics_S122  AKT1_phosphoproteomics_S124  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00006                           NaN                     -0.04605   \n",
       "C3L-00008                       -0.2500                     -0.26300   \n",
       "C3L-00032                           NaN                      0.21400   \n",
       "C3L-00090                           NaN                     -0.00300   \n",
       "C3L-00098                           NaN                     -0.18965   \n",
       "...                                 ...                          ...   \n",
       "C3N-01520                           NaN                     -0.35400   \n",
       "C3N-01521                           NaN                     -0.16350   \n",
       "C3N-01537                        0.0241                      0.02410   \n",
       "C3N-01802                        0.3130                     -0.71750   \n",
       "C3N-01825                        0.1410                      0.14100   \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S126  AKT1_phosphoproteomics_S129  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00006                       0.14810                     -0.04605   \n",
       "C3L-00008                      -0.12100                     -0.18550   \n",
       "C3L-00032                       0.05050                      0.03195   \n",
       "C3L-00090                      -0.02820                      0.04540   \n",
       "C3L-00098                      -0.03930                     -0.07715   \n",
       "...                                 ...                          ...   \n",
       "C3N-01520                      -0.30800                     -0.17250   \n",
       "C3N-01521                      -0.09890                     -0.12995   \n",
       "C3N-01537                      -0.01185                     -0.04780   \n",
       "C3N-01802                      -0.12300                     -0.38450   \n",
       "C3N-01825                       0.19050                      0.15855   \n",
       "\n",
       "Name        AKT1_phosphoproteomics_T450  CSNK2A1_phosphoproteomics  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00006                           NaN                        NaN   \n",
       "C3L-00008                      -0.00876                        NaN   \n",
       "C3L-00032                           NaN                        NaN   \n",
       "C3L-00090                           NaN                        NaN   \n",
       "C3L-00098                      -0.30100                        NaN   \n",
       "...                                 ...                        ...   \n",
       "C3N-01520                           NaN                        NaN   \n",
       "C3N-01521                       0.00000                        NaN   \n",
       "C3N-01537                           NaN                        NaN   \n",
       "C3N-01802                      -0.00828                        NaN   \n",
       "C3N-01825                           NaN                        NaN   \n",
       "\n",
       "Name        CSNK2A2_phosphoproteomics_S18  CSNK2A2_phosphoproteomics_S21  \\\n",
       "Patient_ID                                                                 \n",
       "C3L-00006                             NaN                            NaN   \n",
       "C3L-00008                          0.0107                            NaN   \n",
       "C3L-00032                             NaN                            NaN   \n",
       "C3L-00090                             NaN                       -0.07190   \n",
       "C3L-00098                             NaN                            NaN   \n",
       "...                                   ...                            ...   \n",
       "C3N-01520                             NaN                       -0.00115   \n",
       "C3N-01521                             NaN                            NaN   \n",
       "C3N-01537                             NaN                        1.00000   \n",
       "C3N-01802                          0.3740                            NaN   \n",
       "C3N-01825                          0.6210                            NaN   \n",
       "\n",
       "Name        EGFR_phosphoproteomics_S1026  EGFR_phosphoproteomics_S1037  ...  \\\n",
       "Patient_ID                                                              ...   \n",
       "C3L-00006                         -0.227                           NaN  ...   \n",
       "C3L-00008                            NaN                           NaN  ...   \n",
       "C3L-00032                            NaN                           NaN  ...   \n",
       "C3L-00090                            NaN                           NaN  ...   \n",
       "C3L-00098                            NaN                           NaN  ...   \n",
       "...                                  ...                           ...  ...   \n",
       "C3N-01520                            NaN                           NaN  ...   \n",
       "C3N-01521                            NaN                           NaN  ...   \n",
       "C3N-01537                         -0.321                           NaN  ...   \n",
       "C3N-01802                            NaN                           NaN  ...   \n",
       "C3N-01825                            NaN                           NaN  ...   \n",
       "\n",
       "Name        XIAP_phosphoproteomics_S261  XIAP_phosphoproteomics_S40  \\\n",
       "Patient_ID                                                            \n",
       "C3L-00006                           NaN                         NaN   \n",
       "C3L-00008                           NaN                         NaN   \n",
       "C3L-00032                           NaN                         NaN   \n",
       "C3L-00090                        -0.311                         NaN   \n",
       "C3L-00098                           NaN                         NaN   \n",
       "...                                 ...                         ...   \n",
       "C3N-01520                        -0.519                         NaN   \n",
       "C3N-01521                           NaN                         NaN   \n",
       "C3N-01537                           NaN                         NaN   \n",
       "C3N-01802                           NaN                         NaN   \n",
       "C3N-01825                           NaN                         NaN   \n",
       "\n",
       "Name        XIAP_phosphoproteomics_S402  XIAP_phosphoproteomics_S406  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00006                        -0.437                          NaN   \n",
       "C3L-00008                           NaN                        0.116   \n",
       "C3L-00032                           NaN                          NaN   \n",
       "C3L-00090                         0.000                       -0.203   \n",
       "C3L-00098                         1.710                        0.424   \n",
       "...                                 ...                          ...   \n",
       "C3N-01520                         0.163                       -0.333   \n",
       "C3N-01521                        -0.277                       -0.511   \n",
       "C3N-01537                           NaN                          NaN   \n",
       "C3N-01802                           NaN                       -1.630   \n",
       "C3N-01825                         0.211                        0.699   \n",
       "\n",
       "Name        XIAP_phosphoproteomics_S427  XIAP_phosphoproteomics_S430  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00006                           NaN                          NaN   \n",
       "C3L-00008                           NaN                        -1.90   \n",
       "C3L-00032                           NaN                          NaN   \n",
       "C3L-00090                           NaN                          NaN   \n",
       "C3L-00098                         0.614                          NaN   \n",
       "...                                 ...                          ...   \n",
       "C3N-01520                           NaN                          NaN   \n",
       "C3N-01521                        -0.267                          NaN   \n",
       "C3N-01537                           NaN                          NaN   \n",
       "C3N-01802                           NaN                         1.15   \n",
       "C3N-01825                           NaN                          NaN   \n",
       "\n",
       "Name                                PTEN_Mutation_  \\\n",
       "Patient_ID                                           \n",
       "C3L-00006   [Missense_Mutation, Nonsense_Mutation]   \n",
       "C3L-00008                      [Missense_Mutation]   \n",
       "C3L-00032                      [Nonsense_Mutation]   \n",
       "C3L-00090                      [Missense_Mutation]   \n",
       "C3L-00098                         [Wildtype_Tumor]   \n",
       "...                                            ...   \n",
       "C3N-01520       [Frame_Shift_Ins, Frame_Shift_Del]   \n",
       "C3N-01521                         [Wildtype_Tumor]   \n",
       "C3N-01537                         [Wildtype_Tumor]   \n",
       "C3N-01802                         [Wildtype_Tumor]   \n",
       "C3N-01825                         [Wildtype_Tumor]   \n",
       "\n",
       "Name                      PTEN_Location_  PTEN_Mutation_Status_  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00006             [p.R130Q, p.R233*]      Multiple_mutation   \n",
       "C3L-00008                      [p.G127R]        Single_mutation   \n",
       "C3L-00032                      [p.W111*]        Single_mutation   \n",
       "C3L-00090                      [p.R130G]        Single_mutation   \n",
       "C3L-00098                  [No_mutation]         Wildtype_Tumor   \n",
       "...                                  ...                    ...   \n",
       "C3N-01520   [p.D268Gfs*30, p.N323Mfs*21]      Multiple_mutation   \n",
       "C3N-01521                  [No_mutation]         Wildtype_Tumor   \n",
       "C3N-01537                  [No_mutation]         Wildtype_Tumor   \n",
       "C3N-01802                  [No_mutation]         Wildtype_Tumor   \n",
       "C3N-01825                  [No_mutation]         Wildtype_Tumor   \n",
       "\n",
       "Name        Sample_Status_  \n",
       "Patient_ID                  \n",
       "C3L-00006            Tumor  \n",
       "C3L-00008            Tumor  \n",
       "C3L-00032            Tumor  \n",
       "C3L-00090            Tumor  \n",
       "C3L-00098            Tumor  \n",
       "...                    ...  \n",
       "C3N-01520            Tumor  \n",
       "C3N-01521            Tumor  \n",
       "C3N-01537            Tumor  \n",
       "C3N-01802            Tumor  \n",
       "C3N-01825            Tumor  \n",
       "\n",
       "[95 rows x 232 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Endo ttest\n",
    "\n",
    "mut_type_gbm = endo.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = endo.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'phosphoproteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "prot_and_mutations = endo.reduce_multiindex(prot_and_mutations, flatten = True)\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "trunc_wt = merged[get]\n",
    "trunc_wt['Mutation'] = np.where(\n",
    "                trunc_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "trunc_wt['Mutation'].value_counts()\n",
    "#trunc_wt\n",
    "prot_and_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CSNK2A2_phosphoproteomics_S21</td>\n",
       "      <td>0.001495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PTEN_phosphoproteomics_S475</td>\n",
       "      <td>0.009049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53_phosphoproteomics_S315</td>\n",
       "      <td>0.009797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PTK2_phosphoproteomics_S807</td>\n",
       "      <td>0.010277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ROCK1_phosphoproteomics_S704</td>\n",
       "      <td>0.024254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>SHC1_phosphoproteomics_T380</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S162</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S46</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>XIAP_phosphoproteomics_S40</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>XIAP_phosphoproteomics_S430</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>228 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Phosphoproteomics  En_P_Value\n",
       "0      CSNK2A2_phosphoproteomics_S21    0.001495\n",
       "1        PTEN_phosphoproteomics_S475    0.009049\n",
       "2        TP53_phosphoproteomics_S315    0.009797\n",
       "3        PTK2_phosphoproteomics_S807    0.010277\n",
       "4       ROCK1_phosphoproteomics_S704    0.024254\n",
       "..                               ...         ...\n",
       "223      SHC1_phosphoproteomics_T380         NaN\n",
       "224  SLC9A3R1_phosphoproteomics_S162         NaN\n",
       "225   SLC9A3R1_phosphoproteomics_S46         NaN\n",
       "226       XIAP_phosphoproteomics_S40         NaN\n",
       "227      XIAP_phosphoproteomics_S430         NaN\n",
       "\n",
       "[228 rows x 2 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(trunc_wt.columns[:-1])\n",
    "\n",
    "e_pval = u.wrap_ttest(trunc_wt, 'Mutation', cols, return_all = True,correction_method = 'fdr_bh')\n",
    "e_pval = e_pval.rename(columns = {'Comparison': 'Phosphoproteomics','P_Value': 'En_P_Value'})\n",
    "e_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1_phosphoproteomics_S122</td>\n",
       "      <td>0.07530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1_phosphoproteomics_S124</td>\n",
       "      <td>0.11230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1_phosphoproteomics_S126</td>\n",
       "      <td>0.11370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1_phosphoproteomics_S129</td>\n",
       "      <td>0.04810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1_phosphoproteomics_T450</td>\n",
       "      <td>-0.02238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>XIAP_phosphoproteomics_S40</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>XIAP_phosphoproteomics_S402</td>\n",
       "      <td>0.20150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>XIAP_phosphoproteomics_S406</td>\n",
       "      <td>-0.37250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>XIAP_phosphoproteomics_S427</td>\n",
       "      <td>0.28950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>XIAP_phosphoproteomics_S430</td>\n",
       "      <td>0.30550</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>228 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Phosphoproteomics  En_Median\n",
       "0    AKT1_phosphoproteomics_S122    0.07530\n",
       "1    AKT1_phosphoproteomics_S124    0.11230\n",
       "2    AKT1_phosphoproteomics_S126    0.11370\n",
       "3    AKT1_phosphoproteomics_S129    0.04810\n",
       "4    AKT1_phosphoproteomics_T450   -0.02238\n",
       "..                           ...        ...\n",
       "223   XIAP_phosphoproteomics_S40        NaN\n",
       "224  XIAP_phosphoproteomics_S402    0.20150\n",
       "225  XIAP_phosphoproteomics_S406   -0.37250\n",
       "226  XIAP_phosphoproteomics_S427    0.28950\n",
       "227  XIAP_phosphoproteomics_S430    0.30550\n",
       "\n",
       "[228 rows x 2 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = trunc_wt[trunc_wt.Mutation == \"Truncation\"]\n",
    "wt = trunc_wt[trunc_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "trunc_med = t.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "en_d = {}\n",
    "\n",
    "for phos in cols:\n",
    "    dif = wt_med[phos] - trunc_med[phos]\n",
    "    en_d[phos] = dif\n",
    "\n",
    "en_df = pd.DataFrame.from_dict(en_d, orient='index', columns=['En_Median'])\n",
    "en_df = en_df.reset_index().rename(columns={'index':'Phosphoproteomics'})\n",
    "en_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CSNK2A2_phosphoproteomics_S21</td>\n",
       "      <td>0.001495</td>\n",
       "      <td>0.839075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PTEN_phosphoproteomics_S475</td>\n",
       "      <td>0.009049</td>\n",
       "      <td>0.393800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53_phosphoproteomics_S315</td>\n",
       "      <td>0.009797</td>\n",
       "      <td>0.157500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PTK2_phosphoproteomics_S807</td>\n",
       "      <td>0.010277</td>\n",
       "      <td>0.263150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ROCK1_phosphoproteomics_S704</td>\n",
       "      <td>0.024254</td>\n",
       "      <td>0.299950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>SHC1_phosphoproteomics_T380</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S46</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.178760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>XIAP_phosphoproteomics_S40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>XIAP_phosphoproteomics_S430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.305500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>228 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Phosphoproteomics  En_P_Value  En_Median\n",
       "0      CSNK2A2_phosphoproteomics_S21    0.001495   0.839075\n",
       "1        PTEN_phosphoproteomics_S475    0.009049   0.393800\n",
       "2        TP53_phosphoproteomics_S315    0.009797   0.157500\n",
       "3        PTK2_phosphoproteomics_S807    0.010277   0.263150\n",
       "4       ROCK1_phosphoproteomics_S704    0.024254   0.299950\n",
       "..                               ...         ...        ...\n",
       "223      SHC1_phosphoproteomics_T380         NaN        NaN\n",
       "224  SLC9A3R1_phosphoproteomics_S162         NaN        NaN\n",
       "225   SLC9A3R1_phosphoproteomics_S46         NaN  -0.178760\n",
       "226       XIAP_phosphoproteomics_S40         NaN        NaN\n",
       "227      XIAP_phosphoproteomics_S430         NaN   0.305500\n",
       "\n",
       "[228 rows x 3 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_merged = e_pval.merge(en_df, on='Phosphoproteomics',how='outer')\n",
    "e_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                            \r"
     ]
    }
   ],
   "source": [
    "o = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the phosphoproteomics dataframe, so they were inserted into joined table, but filled with NaN: CSNK2A1, INPP4B, PIK3CB (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "157\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 328 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>AKT1_phosphoproteomics_S124_R.SGS*PSDNSGAEEMEVSLAKPK.H</th>\n",
       "      <th>AKT1_phosphoproteomics_S126_R.SGSPS*DNSGAEEMEVSLAK.P</th>\n",
       "      <th>AKT1_phosphoproteomics_S126_R.SGSPS*DNSGAEEMEVSLAKPK.H</th>\n",
       "      <th>AKT1_phosphoproteomics_S129_R.SGSPSDNS*GAEEMEVSLAKPK.H</th>\n",
       "      <th>AKT1_phosphoproteomics_T450_I.TIT*PPDQDDSMECVDSER.R</th>\n",
       "      <th>AKT1_phosphoproteomics_T450_Q.MITIT*PPDQDDSMECVDSER.R</th>\n",
       "      <th>CSNK2A2_phosphoproteomics_S18_R.VYAEVNS*LR.S</th>\n",
       "      <th>EGFR_phosphoproteomics_S1057_R.NGLQS*CPIKEDSFLQR.Y</th>\n",
       "      <th>EGFR_phosphoproteomics_S1064_R.NGLQSCPIKEDS*FLQR.Y</th>\n",
       "      <th>EGFR_phosphoproteomics_S1104_P.AGS*VQNPVYHNQPLNPAPSR.D</th>\n",
       "      <th>...</th>\n",
       "      <th>SLC9A3R1_phosphoproteomics_T293_R.SASSDT*SEELNSQDSPPK.Q</th>\n",
       "      <th>TP53_phosphoproteomics_S315_K.RALPNNTSSS*PQPK.K</th>\n",
       "      <th>TP53_phosphoproteomics_S315_R.ALPNNTSSS*PQPK.K</th>\n",
       "      <th>USP13_phosphoproteomics_S630_R.GLQPGEEELPDIS*PPIVIPDDSKDR.L</th>\n",
       "      <th>USP7_phosphoproteomics_S18_K.AGEQQLS*EPEDMEMEAGDTDDPPR.I</th>\n",
       "      <th>XIAP_phosphoproteomics_S402_K.IQISGS*NYK.S</th>\n",
       "      <th>XIAP_phosphoproteomics_S87_R.KVS*PNCR.F</th>\n",
       "      <th>XIAP_phosphoproteomics_T356_R.TT*EKTPSLTR.R</th>\n",
       "      <th>XIAP_phosphoproteomics_T359_R.TTEKT*PSLTR.R</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01OV007</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.336</td>\n",
       "      <td>-1.332</td>\n",
       "      <td>-1.191</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.419</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV017</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.022</td>\n",
       "      <td>-0.989</td>\n",
       "      <td>-1.169</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.061</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.693</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV018</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV023</th>\n",
       "      <td>-0.739</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.371</td>\n",
       "      <td>-3.195</td>\n",
       "      <td>-2.372</td>\n",
       "      <td>-1.771</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.967</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.914</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV026</th>\n",
       "      <td>-1.454</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.094</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.278</td>\n",
       "      <td>-2.915</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.686</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.869</td>\n",
       "      <td>-0.348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 154 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        AKT1_phosphoproteomics_S124_R.SGS*PSDNSGAEEMEVSLAKPK.H  \\\n",
       "Patient_ID                                                           \n",
       "01OV007                                                   NaN        \n",
       "01OV017                                                   NaN        \n",
       "01OV018                                                   NaN        \n",
       "01OV023                                                -0.739        \n",
       "01OV026                                                -1.454        \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S126_R.SGSPS*DNSGAEEMEVSLAK.P  \\\n",
       "Patient_ID                                                         \n",
       "01OV007                                                -2.336      \n",
       "01OV017                                                -2.022      \n",
       "01OV018                                                   NaN      \n",
       "01OV023                                                   NaN      \n",
       "01OV026                                                   NaN      \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S126_R.SGSPS*DNSGAEEMEVSLAKPK.H  \\\n",
       "Patient_ID                                                           \n",
       "01OV007                                                -1.332        \n",
       "01OV017                                                -0.989        \n",
       "01OV018                                                -1.503        \n",
       "01OV023                                                   NaN        \n",
       "01OV026                                                   NaN        \n",
       "\n",
       "Name        AKT1_phosphoproteomics_S129_R.SGSPSDNS*GAEEMEVSLAKPK.H  \\\n",
       "Patient_ID                                                           \n",
       "01OV007                                                -1.191        \n",
       "01OV017                                                -1.169        \n",
       "01OV018                                                   NaN        \n",
       "01OV023                                                -1.371        \n",
       "01OV026                                                -2.094        \n",
       "\n",
       "Name        AKT1_phosphoproteomics_T450_I.TIT*PPDQDDSMECVDSER.R  \\\n",
       "Patient_ID                                                        \n",
       "01OV007                                                   NaN     \n",
       "01OV017                                                   NaN     \n",
       "01OV018                                                -2.250     \n",
       "01OV023                                                -3.195     \n",
       "01OV026                                                   NaN     \n",
       "\n",
       "Name        AKT1_phosphoproteomics_T450_Q.MITIT*PPDQDDSMECVDSER.R  \\\n",
       "Patient_ID                                                          \n",
       "01OV007                                                   NaN       \n",
       "01OV017                                                   NaN       \n",
       "01OV018                                                   NaN       \n",
       "01OV023                                                -2.372       \n",
       "01OV026                                                   NaN       \n",
       "\n",
       "Name        CSNK2A2_phosphoproteomics_S18_R.VYAEVNS*LR.S  \\\n",
       "Patient_ID                                                 \n",
       "01OV007                                              NaN   \n",
       "01OV017                                              NaN   \n",
       "01OV018                                              NaN   \n",
       "01OV023                                           -1.771   \n",
       "01OV026                                              NaN   \n",
       "\n",
       "Name        EGFR_phosphoproteomics_S1057_R.NGLQS*CPIKEDSFLQR.Y  \\\n",
       "Patient_ID                                                       \n",
       "01OV007                                                   NaN    \n",
       "01OV017                                                   NaN    \n",
       "01OV018                                                   NaN    \n",
       "01OV023                                                   NaN    \n",
       "01OV026                                                   NaN    \n",
       "\n",
       "Name        EGFR_phosphoproteomics_S1064_R.NGLQSCPIKEDS*FLQR.Y  \\\n",
       "Patient_ID                                                       \n",
       "01OV007                                                   NaN    \n",
       "01OV017                                                   NaN    \n",
       "01OV018                                                   NaN    \n",
       "01OV023                                                   NaN    \n",
       "01OV026                                                -2.278    \n",
       "\n",
       "Name        EGFR_phosphoproteomics_S1104_P.AGS*VQNPVYHNQPLNPAPSR.D  ...  \\\n",
       "Patient_ID                                                          ...   \n",
       "01OV007                                                -2.419       ...   \n",
       "01OV017                                                -2.061       ...   \n",
       "01OV018                                                   NaN       ...   \n",
       "01OV023                                                -1.967       ...   \n",
       "01OV026                                                -2.915       ...   \n",
       "\n",
       "Name        SLC9A3R1_phosphoproteomics_T293_R.SASSDT*SEELNSQDSPPK.Q  \\\n",
       "Patient_ID                                                            \n",
       "01OV007                                                -4.115         \n",
       "01OV017                                                -3.693         \n",
       "01OV018                                                   NaN         \n",
       "01OV023                                                   NaN         \n",
       "01OV026                                                -4.686         \n",
       "\n",
       "Name        TP53_phosphoproteomics_S315_K.RALPNNTSSS*PQPK.K  \\\n",
       "Patient_ID                                                    \n",
       "01OV007                                                 NaN   \n",
       "01OV017                                                 NaN   \n",
       "01OV018                                                 NaN   \n",
       "01OV023                                                 NaN   \n",
       "01OV026                                                 NaN   \n",
       "\n",
       "Name        TP53_phosphoproteomics_S315_R.ALPNNTSSS*PQPK.K  \\\n",
       "Patient_ID                                                   \n",
       "01OV007                                                NaN   \n",
       "01OV017                                                NaN   \n",
       "01OV018                                                NaN   \n",
       "01OV023                                             -3.017   \n",
       "01OV026                                             -1.976   \n",
       "\n",
       "Name        USP13_phosphoproteomics_S630_R.GLQPGEEELPDIS*PPIVIPDDSKDR.L  \\\n",
       "Patient_ID                                                                \n",
       "01OV007                                                   NaN             \n",
       "01OV017                                                   NaN             \n",
       "01OV018                                                   NaN             \n",
       "01OV023                                                   NaN             \n",
       "01OV026                                                   NaN             \n",
       "\n",
       "Name        USP7_phosphoproteomics_S18_K.AGEQQLS*EPEDMEMEAGDTDDPPR.I  \\\n",
       "Patient_ID                                                             \n",
       "01OV007                                                   NaN          \n",
       "01OV017                                                   NaN          \n",
       "01OV018                                                   NaN          \n",
       "01OV023                                                   NaN          \n",
       "01OV026                                                   NaN          \n",
       "\n",
       "Name        XIAP_phosphoproteomics_S402_K.IQISGS*NYK.S  \\\n",
       "Patient_ID                                               \n",
       "01OV007                                            NaN   \n",
       "01OV017                                            NaN   \n",
       "01OV018                                            NaN   \n",
       "01OV023                                            NaN   \n",
       "01OV026                                            NaN   \n",
       "\n",
       "Name        XIAP_phosphoproteomics_S87_R.KVS*PNCR.F  \\\n",
       "Patient_ID                                            \n",
       "01OV007                                         NaN   \n",
       "01OV017                                         NaN   \n",
       "01OV018                                         NaN   \n",
       "01OV023                                         NaN   \n",
       "01OV026                                      -0.869   \n",
       "\n",
       "Name        XIAP_phosphoproteomics_T356_R.TT*EKTPSLTR.R  \\\n",
       "Patient_ID                                                \n",
       "01OV007                                             NaN   \n",
       "01OV017                                             NaN   \n",
       "01OV018                                             NaN   \n",
       "01OV023                                             NaN   \n",
       "01OV026                                          -0.348   \n",
       "\n",
       "Name        XIAP_phosphoproteomics_T359_R.TTEKT*PSLTR.R        Mutation  \n",
       "Patient_ID                                                               \n",
       "01OV007                                             NaN  Wildtype_Tumor  \n",
       "01OV017                                             NaN        Deletion  \n",
       "01OV018                                             NaN  Wildtype_Tumor  \n",
       "01OV023                                          -2.914        Deletion  \n",
       "01OV026                                             NaN  Wildtype_Tumor  \n",
       "\n",
       "[5 rows x 154 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "\n",
    "mut_type = o.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = o.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'phosphoproteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "prot_and_mutations = o.reduce_multiindex(prot_and_mutations, levels_to_drop = 3, flatten=True)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns #FIX, \n",
    "# dropped 328 cols!  \n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "print(len(del_wt.columns))\n",
    "del_wt = del_wt.dropna(axis = 'columns', how='all')\n",
    "del_wt.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIK3R1_phosphoproteomics_Y470_R.LYEEY*TR.T</td>\n",
       "      <td>0.002950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3R3_phosphoproteomics_Y219_R.LYEEY*TR.T</td>\n",
       "      <td>0.002950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NEDD4_phosphoproteomics_S884_R.GKTS*LDTSNDLGPL...</td>\n",
       "      <td>0.004646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EGFR_phosphoproteomics_S1104_P.AGS*VQNPVYHNQPL...</td>\n",
       "      <td>0.007877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EGFR_phosphoproteomics_S991_R.MHLPS*PTDSNFYR.A</td>\n",
       "      <td>0.029326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S280_R.EALAEAALES*P...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S288_R.S*ASSDTSEELN...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S291S302_R.SASS*DTS...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>XIAP_phosphoproteomics_S87_R.KVS*PNCR.F</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>XIAP_phosphoproteomics_T356_R.TT*EKTPSLTR.R</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Phosphoproteomics  Ov_P_Value\n",
       "0           PIK3R1_phosphoproteomics_Y470_R.LYEEY*TR.T    0.002950\n",
       "1           PIK3R3_phosphoproteomics_Y219_R.LYEEY*TR.T    0.002950\n",
       "2    NEDD4_phosphoproteomics_S884_R.GKTS*LDTSNDLGPL...    0.004646\n",
       "3    EGFR_phosphoproteomics_S1104_P.AGS*VQNPVYHNQPL...    0.007877\n",
       "4       EGFR_phosphoproteomics_S991_R.MHLPS*PTDSNFYR.A    0.029326\n",
       "..                                                 ...         ...\n",
       "148  SLC9A3R1_phosphoproteomics_S280_R.EALAEAALES*P...         NaN\n",
       "149  SLC9A3R1_phosphoproteomics_S288_R.S*ASSDTSEELN...         NaN\n",
       "150  SLC9A3R1_phosphoproteomics_S291S302_R.SASS*DTS...         NaN\n",
       "151            XIAP_phosphoproteomics_S87_R.KVS*PNCR.F         NaN\n",
       "152        XIAP_phosphoproteomics_T356_R.TT*EKTPSLTR.R         NaN\n",
       "\n",
       "[153 rows x 2 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "o_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "o_pval = o_pval.rename(columns = {'Comparison': 'Phosphoproteomics','P_Value': 'Ov_P_Value'})\n",
    "o_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Ov_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1_phosphoproteomics_S124_R.SGS*PSDNSGAEEMEV...</td>\n",
       "      <td>-0.2670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1_phosphoproteomics_S126_R.SGSPS*DNSGAEEMEV...</td>\n",
       "      <td>0.0810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1_phosphoproteomics_S126_R.SGSPS*DNSGAEEMEV...</td>\n",
       "      <td>-0.3165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1_phosphoproteomics_S129_R.SGSPSDNS*GAEEMEV...</td>\n",
       "      <td>-0.1040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1_phosphoproteomics_T450_I.TIT*PPDQDDSMECVD...</td>\n",
       "      <td>0.2370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>USP7_phosphoproteomics_S18_K.AGEQQLS*EPEDMEMEA...</td>\n",
       "      <td>0.3790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>XIAP_phosphoproteomics_S402_K.IQISGS*NYK.S</td>\n",
       "      <td>-1.1640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>XIAP_phosphoproteomics_S87_R.KVS*PNCR.F</td>\n",
       "      <td>-0.4010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>XIAP_phosphoproteomics_T356_R.TT*EKTPSLTR.R</td>\n",
       "      <td>0.8270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>XIAP_phosphoproteomics_T359_R.TTEKT*PSLTR.R</td>\n",
       "      <td>-1.6880</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Phosphoproteomics  Ov_Median\n",
       "0    AKT1_phosphoproteomics_S124_R.SGS*PSDNSGAEEMEV...    -0.2670\n",
       "1    AKT1_phosphoproteomics_S126_R.SGSPS*DNSGAEEMEV...     0.0810\n",
       "2    AKT1_phosphoproteomics_S126_R.SGSPS*DNSGAEEMEV...    -0.3165\n",
       "3    AKT1_phosphoproteomics_S129_R.SGSPSDNS*GAEEMEV...    -0.1040\n",
       "4    AKT1_phosphoproteomics_T450_I.TIT*PPDQDDSMECVD...     0.2370\n",
       "..                                                 ...        ...\n",
       "148  USP7_phosphoproteomics_S18_K.AGEQQLS*EPEDMEMEA...     0.3790\n",
       "149         XIAP_phosphoproteomics_S402_K.IQISGS*NYK.S    -1.1640\n",
       "150            XIAP_phosphoproteomics_S87_R.KVS*PNCR.F    -0.4010\n",
       "151        XIAP_phosphoproteomics_T356_R.TT*EKTPSLTR.R     0.8270\n",
       "152        XIAP_phosphoproteomics_T359_R.TTEKT*PSLTR.R    -1.6880\n",
       "\n",
       "[153 rows x 2 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "ov_d = {}\n",
    "\n",
    "for phos in cols:\n",
    "    dif = wt_med[phos] - del_med[phos]\n",
    "    ov_d[phos] = dif\n",
    "\n",
    "o_df = pd.DataFrame.from_dict(ov_d, orient='index', columns=['Ov_Median'])\n",
    "o_df = o_df.reset_index().rename(columns={'index':'Phosphoproteomics'})\n",
    "o_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIK3R1_Y470</td>\n",
       "      <td>0.002950</td>\n",
       "      <td>-0.6640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3R3_Y219</td>\n",
       "      <td>0.002950</td>\n",
       "      <td>-0.6640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NEDD4_S884</td>\n",
       "      <td>0.004646</td>\n",
       "      <td>-0.4495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EGFR_S1104</td>\n",
       "      <td>0.007877</td>\n",
       "      <td>-0.6180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EGFR_S991</td>\n",
       "      <td>0.029326</td>\n",
       "      <td>-1.0500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>SLC9A3R1_S280</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>SLC9A3R1_S288</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>SLC9A3R1_S291S302</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>XIAP_S87</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.4010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>XIAP_T356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8270</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Phosphoproteomics  Ov_P_Value  Ov_Median\n",
       "0          PIK3R1_Y470    0.002950    -0.6640\n",
       "1          PIK3R3_Y219    0.002950    -0.6640\n",
       "2           NEDD4_S884    0.004646    -0.4495\n",
       "3           EGFR_S1104    0.007877    -0.6180\n",
       "4            EGFR_S991    0.029326    -1.0500\n",
       "..                 ...         ...        ...\n",
       "148      SLC9A3R1_S280         NaN    -0.3780\n",
       "149      SLC9A3R1_S288         NaN    -0.5650\n",
       "150  SLC9A3R1_S291S302         NaN    -0.3350\n",
       "151           XIAP_S87         NaN    -0.4010\n",
       "152          XIAP_T356         NaN     0.8270\n",
       "\n",
       "[153 rows x 3 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "o_merged = o_pval.merge(o_df, on='Phosphoproteomics',how='outer')\n",
    "o_merged = o_merged.replace(to_replace = r'_phosphoproteomics', value = '', regex = True)\n",
    "o_merged = o_merged.replace(to_replace = r'_[A-Z.*]*$', value = '', regex = True)\n",
    "o_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Breast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    }
   ],
   "source": [
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the phosphoproteomics dataframe, so they were inserted into joined table, but filled with NaN: INPP4B, PIK3CB, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    80\n",
       "Deletion          24\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = b.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = b.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'phosphoproteomics', omics_genes = ip)\n",
    "#prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "prot_and_mutations = b.reduce_multiindex(prot_and_mutations, levels_to_drop = 3)\n",
    "prot_and_mutations = b.reduce_multiindex(prot_and_mutations, levels_to_drop = 2, flatten = True)\n",
    "prot_and_mutations\n",
    "\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAST2_phosphoproteomics_S74S81</td>\n",
       "      <td>0.003703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAGI3_phosphoproteomics_S1407</td>\n",
       "      <td>0.005574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SLC9A3R1_phosphoproteomics_S170</td>\n",
       "      <td>0.007694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ROCK1_phosphoproteomics_S2</td>\n",
       "      <td>0.008402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CSNK2A2_phosphoproteomics_S18</td>\n",
       "      <td>0.011835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>USP7_phosphoproteomics_S963</td>\n",
       "      <td>0.973910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>MAST2_phosphoproteomics_S92</td>\n",
       "      <td>0.983045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>INPP4B_phosphoproteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>PIK3CB_phosphoproteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>PIK3R3_phosphoproteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>141 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Phosphoproteomics  Brca_P_Value\n",
       "0     MAST2_phosphoproteomics_S74S81      0.003703\n",
       "1      MAGI3_phosphoproteomics_S1407      0.005574\n",
       "2    SLC9A3R1_phosphoproteomics_S170      0.007694\n",
       "3         ROCK1_phosphoproteomics_S2      0.008402\n",
       "4      CSNK2A2_phosphoproteomics_S18      0.011835\n",
       "..                               ...           ...\n",
       "136      USP7_phosphoproteomics_S963      0.973910\n",
       "137      MAST2_phosphoproteomics_S92      0.983045\n",
       "138         INPP4B_phosphoproteomics           NaN\n",
       "139         PIK3CB_phosphoproteomics           NaN\n",
       "140         PIK3R3_phosphoproteomics           NaN\n",
       "\n",
       "[141 rows x 2 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "b_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "b_pval = b_pval.rename(columns = {'Comparison': 'Phosphoproteomics','P_Value': 'Brca_P_Value'})\n",
    "b_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Brca_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1_phosphoproteomics_M446T450</td>\n",
       "      <td>0.51865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT1_phosphoproteomics_S124S126S129</td>\n",
       "      <td>-0.37585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1_phosphoproteomics_S126</td>\n",
       "      <td>0.13445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT1_phosphoproteomics_S126S129</td>\n",
       "      <td>-0.16085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1_phosphoproteomics_S129</td>\n",
       "      <td>0.02820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>USP7_phosphoproteomics_S49T54</td>\n",
       "      <td>-0.43095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>USP7_phosphoproteomics_S752</td>\n",
       "      <td>0.09245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>USP7_phosphoproteomics_S963</td>\n",
       "      <td>0.03725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>XIAP_phosphoproteomics_S402</td>\n",
       "      <td>-0.26415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>XIAP_phosphoproteomics_S5</td>\n",
       "      <td>-0.90855</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>141 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Phosphoproteomics  Brca_Median\n",
       "0        AKT1_phosphoproteomics_M446T450      0.51865\n",
       "1    AKT1_phosphoproteomics_S124S126S129     -0.37585\n",
       "2            AKT1_phosphoproteomics_S126      0.13445\n",
       "3        AKT1_phosphoproteomics_S126S129     -0.16085\n",
       "4            AKT1_phosphoproteomics_S129      0.02820\n",
       "..                                   ...          ...\n",
       "136        USP7_phosphoproteomics_S49T54     -0.43095\n",
       "137          USP7_phosphoproteomics_S752      0.09245\n",
       "138          USP7_phosphoproteomics_S963      0.03725\n",
       "139          XIAP_phosphoproteomics_S402     -0.26415\n",
       "140            XIAP_phosphoproteomics_S5     -0.90855\n",
       "\n",
       "[141 rows x 2 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "b_d = {}\n",
    "\n",
    "for phos in cols:\n",
    "    dif = wt_med[phos] - del_med[phos]\n",
    "    b_d[phos] = dif\n",
    "\n",
    "b_df = pd.DataFrame.from_dict(b_d, orient='index', columns=['Brca_Median'])\n",
    "b_df = b_df.reset_index().rename(columns={'index':'Phosphoproteomics'})\n",
    "b_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Phosphoproteomics</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAST2_S74S81</td>\n",
       "      <td>0.003703</td>\n",
       "      <td>-0.96550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAGI3_S1407</td>\n",
       "      <td>0.005574</td>\n",
       "      <td>-1.31335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SLC9A3R1_S170</td>\n",
       "      <td>0.007694</td>\n",
       "      <td>1.33170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ROCK1_S2</td>\n",
       "      <td>0.008402</td>\n",
       "      <td>0.91895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CSNK2A2_S18</td>\n",
       "      <td>0.011835</td>\n",
       "      <td>-0.66225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>USP7_S963</td>\n",
       "      <td>0.973910</td>\n",
       "      <td>0.03725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>MAST2_S92</td>\n",
       "      <td>0.983045</td>\n",
       "      <td>0.23720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>PIK3CB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>PIK3R3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>141 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Phosphoproteomics  Brca_P_Value  Brca_Median\n",
       "0        MAST2_S74S81      0.003703     -0.96550\n",
       "1         MAGI3_S1407      0.005574     -1.31335\n",
       "2       SLC9A3R1_S170      0.007694      1.33170\n",
       "3            ROCK1_S2      0.008402      0.91895\n",
       "4         CSNK2A2_S18      0.011835     -0.66225\n",
       "..                ...           ...          ...\n",
       "136         USP7_S963      0.973910      0.03725\n",
       "137         MAST2_S92      0.983045      0.23720\n",
       "138            INPP4B           NaN          NaN\n",
       "139            PIK3CB           NaN          NaN\n",
       "140            PIK3R3           NaN          NaN\n",
       "\n",
       "[141 rows x 3 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_merged = b_pval.merge(b_df, on='Phosphoproteomics',how='outer')\n",
    "b_merged = b_merged.replace(to_replace = r'_phosphoproteomics', value = '', regex = True)\n",
    "b_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Formatting dataframes...                  \r"
     ]
    }
   ],
   "source": [
    "col = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mut_type = col.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = col.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'phosphoproteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "#prot_and_mutations = col.reduce_multiindex(prot_and_mutations, levels_to_drop = 3)\n",
    "prot_and_mutations = col.reduce_multiindex(prot_and_mutations, levels_to_drop = 2, flatten = True)\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "prot_and_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "c_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "c_pval = c_pval.rename(columns = {'Comparison': 'Phosphoproteomics','P_Value': 'Colon_P_Value'})\n",
    "c_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for phos in cols:\n",
    "    dif = wt_med[phos] - del_med[phos]\n",
    "    d[phos] = dif\n",
    "\n",
    "c_df = pd.DataFrame.from_dict(d, orient='index', columns=['Colon_Median'])\n",
    "c_df = c_df.reset_index().rename(columns={'index':'Phosphoproteomics'})\n",
    "c_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "c_merged = c_pval.merge(c_df, on='Phosphoproteomics',how='outer')\n",
    "c_merged = c_merged.replace(to_replace = r'_phosphoproteomics', value = '', regex = True)\n",
    "c_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Head and Neck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "h = cptac.Hnscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mut_type = h.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = h.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'phosphoproteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "prot_and_mutations = h.reduce_multiindex(prot_and_mutations, levels_to_drop = 3)\n",
    "prot_and_mutations = h.reduce_multiindex(prot_and_mutations, levels_to_drop = 2, flatten=True)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "# FIX - 17 duplicates\n",
    "\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "del_wt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "h_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "h_pval = h_pval.rename(columns = {'Comparison': 'Phosphoproteomics','P_Value': 'Hnscc_P_Value'})\n",
    "h_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "h_d = {}\n",
    "\n",
    "for phos in cols:\n",
    "    dif = wt_med[phos] - del_med[phos]\n",
    "    h_d[phos] = dif\n",
    "\n",
    "h_df = pd.DataFrame.from_dict(h_d, orient='index', columns=['Hnscc_Median'])\n",
    "h_df = h_df.reset_index().rename(columns={'index':'Phosphoproteomics'})\n",
    "h_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "h_merged = h_pval.merge(h_df, on='Phosphoproteomics',how='outer')\n",
    "h_merged = h_merged.replace(to_replace = r'_phosphoproteomics', value = '', regex = True)\n",
    "h_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lung"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "l = cptac.Luad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mut_type = l.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.rename(columns={'PTEN':'cnv'})\n",
    "\n",
    "# different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "phos = l.get_phosphoproteomics(tissue_type='tumor')\n",
    "ip_phos = phos[['PDGFRB', 'PIK3R2', 'PIK3C3','SLC9A3R1',\n",
    "     'USP13','PIK3CA','TP53','EGFR','PTK2','USP7','XIAP','PTEN','PREX2','MAST2','PIK3CD',\n",
    "    'ROCK1','NEDD4','AKT1', 'MVP', 'MAGI3','CSNK2A2', 'MDM2', 'CSNK2A1']] \n",
    "# 'PIK3CB' 'INPP4B' 'SH1' 'PRK3R1' 'PIK3R3' not in index\n",
    "ip_phos = l.reduce_multiindex(ip_phos, levels_to_drop =3)\n",
    "ip_phos = l.reduce_multiindex(ip_phos, levels_to_drop = 2, flatten = True)\n",
    "\n",
    "merged = ip_phos.join(mut_type) # checked and there is 110 tumor samples for lung\n",
    "\n",
    "compare = ['No_Mutation','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "print(del_wt['Mutation'].value_counts())\n",
    "del_wt.columns[:-2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = list(del_wt.columns[:-2])\n",
    "\n",
    "l_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "l_pval = l_pval.rename(columns = {'Comparison': 'Phosphoproteomics','P_Value': 'Luad_P_Value'})\n",
    "l_pval "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"No_Mutation\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "l_d = {}\n",
    "\n",
    "for phos in cols:\n",
    "    dif = wt_med[phos] - del_med[phos]\n",
    "    l_d[phos] = dif\n",
    "\n",
    "l_df = pd.DataFrame.from_dict(l_d, orient='index', columns=['Luad_Median'])\n",
    "l_df = l_df.reset_index().rename(columns={'index':'Phosphoproteomics'})\n",
    "l_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "l_merged = l_pval.merge(l_df, on='Phosphoproteomics',how='outer')\n",
    "l_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "n = g_merged.merge(h_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(l_merged, on='Proteomics',how='outer')\n",
    "n= n.merge(b_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(o_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(e_merged, on='Proteomics',how='outer')\n",
    "all_df = n.merge(c_merged, on='Proteomics',how='outer')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Keep only genes significant in > 1 cancer\n",
    "cols = ['Gbm_P_Value','Hnscc_P_Value','Luad_P_Value','Brca_P_Value','Ov_P_Value','En_P_Value','Colon_P_Value']\n",
    "\n",
    "bc = .05/28\n",
    "\n",
    "t = []\n",
    "\n",
    "for c in cols:\n",
    "    df = all_df[all_df[c] <= bc]\n",
    "    t.append(list(df.Proteomics))\n",
    "    \n",
    "flat_list = [item for sublist in t for item in sublist]\n",
    "sig = set(flat_list) # remove duplicates\n",
    "\n",
    "\n",
    "\n",
    "bool_df = all_df['Proteomics'].isin(sig)\n",
    "sig_df = all_df[bool_df]\n",
    "sig_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_df.to_csv('at_least_one_significant_interacting.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = ip_df.std()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = stdev.to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev.loc[stdev[0] > .2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d_stdev = stdev.std()\n",
    "d_stdev"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = stdev.hist(bins = 20)\n",
    "plt.title(\"Standard Deviations of Pancancer CNV deletions and amplifications of PTEN on Interacting Proteins\\n\")\n",
    "plt.xlabel('StDeV')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(10,8)\n",
    "sns.set(font_scale = 1.3)\n",
    "a = sns.distplot(stdev, bins = 20)\n",
    "\n",
    "a.set_title(\"Standard Deviation Distribution for Pancancer PTEN Mutation Effect on Proteomics\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "a.set(xlabel = 'StDeV', ylabel = 'Frequency')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_df = stdev.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = mean_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Mean between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Mean')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m_df = stdev.median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = m_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Medians between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Median')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
