{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Interacting Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\nimport sys\\nsys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\WhenMutationsDontMatter\\\\')\\nimport plot_utils as p\""
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "'''\n",
    "import sys\n",
    "sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\WhenMutationsDontMatter\\\\')\n",
    "import plot_utils as p'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that endometrial index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "gbm = cptac.Gbm()\n",
    "endo = cptac.Endometrial()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ip = u.get_interacting_proteins('PTEN')\n",
    "len(ip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>MDM2_proteomics</th>\n",
       "      <th>CSNK2A1_proteomics</th>\n",
       "      <th>PDGFRB_proteomics</th>\n",
       "      <th>PIK3R2_proteomics</th>\n",
       "      <th>PIK3C3_proteomics</th>\n",
       "      <th>CSNK2A2_proteomics</th>\n",
       "      <th>SLC9A3R1_proteomics</th>\n",
       "      <th>PIK3R3_proteomics</th>\n",
       "      <th>USP13_proteomics</th>\n",
       "      <th>PIK3CA_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>PIK3CD_proteomics</th>\n",
       "      <th>ROCK1_proteomics</th>\n",
       "      <th>SHC1_proteomics</th>\n",
       "      <th>NEDD4_proteomics</th>\n",
       "      <th>INPP4B_proteomics</th>\n",
       "      <th>PIK3R1_proteomics</th>\n",
       "      <th>AKT1_proteomics</th>\n",
       "      <th>MVP_proteomics</th>\n",
       "      <th>MAGI3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>-0.886531</td>\n",
       "      <td>0.221578</td>\n",
       "      <td>-0.108555</td>\n",
       "      <td>-0.068085</td>\n",
       "      <td>0.081642</td>\n",
       "      <td>-0.273672</td>\n",
       "      <td>0.616424</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.423661</td>\n",
       "      <td>0.124542</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.315003</td>\n",
       "      <td>-0.319701</td>\n",
       "      <td>-0.095985</td>\n",
       "      <td>-0.437514</td>\n",
       "      <td>0.060124</td>\n",
       "      <td>-0.379646</td>\n",
       "      <td>-0.065930</td>\n",
       "      <td>0.147936</td>\n",
       "      <td>0.311427</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.169391</td>\n",
       "      <td>-0.109383</td>\n",
       "      <td>0.079710</td>\n",
       "      <td>-0.027524</td>\n",
       "      <td>-0.035616</td>\n",
       "      <td>0.450206</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155746</td>\n",
       "      <td>-0.211025</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.629943</td>\n",
       "      <td>0.013908</td>\n",
       "      <td>0.003049</td>\n",
       "      <td>0.179955</td>\n",
       "      <td>-0.639545</td>\n",
       "      <td>-0.283144</td>\n",
       "      <td>-0.285929</td>\n",
       "      <td>0.371665</td>\n",
       "      <td>0.259059</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.208625</td>\n",
       "      <td>0.414121</td>\n",
       "      <td>0.078025</td>\n",
       "      <td>0.062739</td>\n",
       "      <td>-0.105403</td>\n",
       "      <td>0.307356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.021276</td>\n",
       "      <td>-0.573919</td>\n",
       "      <td>...</td>\n",
       "      <td>0.085276</td>\n",
       "      <td>-0.048311</td>\n",
       "      <td>0.051279</td>\n",
       "      <td>0.265735</td>\n",
       "      <td>0.669283</td>\n",
       "      <td>0.061201</td>\n",
       "      <td>-0.222607</td>\n",
       "      <td>0.329753</td>\n",
       "      <td>0.128442</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>-0.564682</td>\n",
       "      <td>-0.292558</td>\n",
       "      <td>-0.141799</td>\n",
       "      <td>0.024117</td>\n",
       "      <td>0.087752</td>\n",
       "      <td>-0.189861</td>\n",
       "      <td>-0.040643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.322976</td>\n",
       "      <td>-0.360962</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.382652</td>\n",
       "      <td>-0.046639</td>\n",
       "      <td>0.288725</td>\n",
       "      <td>0.706601</td>\n",
       "      <td>0.451016</td>\n",
       "      <td>-0.496340</td>\n",
       "      <td>-0.370584</td>\n",
       "      <td>-0.518407</td>\n",
       "      <td>-0.099551</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-0.243835</td>\n",
       "      <td>0.132716</td>\n",
       "      <td>0.812053</td>\n",
       "      <td>-0.029297</td>\n",
       "      <td>-0.057555</td>\n",
       "      <td>0.239919</td>\n",
       "      <td>-0.155468</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.049040</td>\n",
       "      <td>0.137414</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.537787</td>\n",
       "      <td>0.237764</td>\n",
       "      <td>-0.337079</td>\n",
       "      <td>0.133808</td>\n",
       "      <td>-0.024038</td>\n",
       "      <td>-0.146416</td>\n",
       "      <td>0.004573</td>\n",
       "      <td>-0.741119</td>\n",
       "      <td>-0.017146</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.015279</td>\n",
       "      <td>0.121787</td>\n",
       "      <td>-0.116620</td>\n",
       "      <td>-0.092783</td>\n",
       "      <td>-0.022308</td>\n",
       "      <td>0.128944</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.096330</td>\n",
       "      <td>-0.218186</td>\n",
       "      <td>...</td>\n",
       "      <td>0.395350</td>\n",
       "      <td>0.141372</td>\n",
       "      <td>0.557741</td>\n",
       "      <td>0.418850</td>\n",
       "      <td>0.010608</td>\n",
       "      <td>-0.093624</td>\n",
       "      <td>-0.002002</td>\n",
       "      <td>-0.090315</td>\n",
       "      <td>-0.079546</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312023</td>\n",
       "      <td>0.361701</td>\n",
       "      <td>-0.023912</td>\n",
       "      <td>-0.160911</td>\n",
       "      <td>0.266436</td>\n",
       "      <td>-1.291855</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.276721</td>\n",
       "      <td>-0.049803</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.556546</td>\n",
       "      <td>-0.158482</td>\n",
       "      <td>-0.282123</td>\n",
       "      <td>0.028673</td>\n",
       "      <td>-0.219639</td>\n",
       "      <td>-0.164426</td>\n",
       "      <td>0.501695</td>\n",
       "      <td>-0.994267</td>\n",
       "      <td>-0.364258</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>0.137052</td>\n",
       "      <td>-0.069012</td>\n",
       "      <td>0.606401</td>\n",
       "      <td>0.626821</td>\n",
       "      <td>-0.029083</td>\n",
       "      <td>-0.161898</td>\n",
       "      <td>0.075817</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.091852</td>\n",
       "      <td>0.815177</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.009546</td>\n",
       "      <td>-0.222781</td>\n",
       "      <td>0.061741</td>\n",
       "      <td>0.013211</td>\n",
       "      <td>0.348604</td>\n",
       "      <td>0.443719</td>\n",
       "      <td>0.196700</td>\n",
       "      <td>0.086260</td>\n",
       "      <td>0.541898</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.276126</td>\n",
       "      <td>-0.108478</td>\n",
       "      <td>-0.135303</td>\n",
       "      <td>0.173834</td>\n",
       "      <td>0.246281</td>\n",
       "      <td>0.090879</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.405654</td>\n",
       "      <td>0.143824</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.255032</td>\n",
       "      <td>-0.256044</td>\n",
       "      <td>-0.240227</td>\n",
       "      <td>-0.209677</td>\n",
       "      <td>0.055190</td>\n",
       "      <td>0.084610</td>\n",
       "      <td>-0.198548</td>\n",
       "      <td>-0.012143</td>\n",
       "      <td>0.260033</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.154969</td>\n",
       "      <td>-0.228369</td>\n",
       "      <td>0.186960</td>\n",
       "      <td>-0.073499</td>\n",
       "      <td>0.064866</td>\n",
       "      <td>-0.275365</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.252806</td>\n",
       "      <td>0.163076</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002891</td>\n",
       "      <td>-0.191965</td>\n",
       "      <td>0.123735</td>\n",
       "      <td>0.013306</td>\n",
       "      <td>-0.077057</td>\n",
       "      <td>-0.166501</td>\n",
       "      <td>-0.055991</td>\n",
       "      <td>-0.740250</td>\n",
       "      <td>-0.134775</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        MDM2_proteomics  CSNK2A1_proteomics  PDGFRB_proteomics  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00104         -0.886531            0.221578          -0.108555   \n",
       "C3L-00365               NaN            0.169391          -0.109383   \n",
       "C3L-00674               NaN           -0.208625           0.414121   \n",
       "C3L-00677         -0.564682           -0.292558          -0.141799   \n",
       "C3L-01040         -0.243835            0.132716           0.812053   \n",
       "...                     ...                 ...                ...   \n",
       "C3N-03183               NaN           -0.015279           0.121787   \n",
       "C3N-03184               NaN            0.312023           0.361701   \n",
       "C3N-03186          0.137052           -0.069012           0.606401   \n",
       "C3N-03188               NaN            0.276126          -0.108478   \n",
       "C3N-03473               NaN            0.154969          -0.228369   \n",
       "\n",
       "Name        PIK3R2_proteomics  PIK3C3_proteomics  CSNK2A2_proteomics  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00104           -0.068085           0.081642           -0.273672   \n",
       "C3L-00365            0.079710          -0.027524           -0.035616   \n",
       "C3L-00674            0.078025           0.062739           -0.105403   \n",
       "C3L-00677            0.024117           0.087752           -0.189861   \n",
       "C3L-01040           -0.029297          -0.057555            0.239919   \n",
       "...                       ...                ...                 ...   \n",
       "C3N-03183           -0.116620          -0.092783           -0.022308   \n",
       "C3N-03184           -0.023912          -0.160911            0.266436   \n",
       "C3N-03186            0.626821          -0.029083           -0.161898   \n",
       "C3N-03188           -0.135303           0.173834            0.246281   \n",
       "C3N-03473            0.186960          -0.073499            0.064866   \n",
       "\n",
       "Name        SLC9A3R1_proteomics  PIK3R3_proteomics  USP13_proteomics  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00104              0.616424                NaN         -0.423661   \n",
       "C3L-00365              0.450206                NaN          0.155746   \n",
       "C3L-00674              0.307356                NaN         -0.021276   \n",
       "C3L-00677             -0.040643                NaN         -0.322976   \n",
       "C3L-01040             -0.155468                NaN         -0.049040   \n",
       "...                         ...                ...               ...   \n",
       "C3N-03183              0.128944                NaN         -0.096330   \n",
       "C3N-03184             -1.291855                NaN          0.276721   \n",
       "C3N-03186              0.075817                NaN          0.091852   \n",
       "C3N-03188              0.090879                NaN         -0.405654   \n",
       "C3N-03473             -0.275365                NaN          0.252806   \n",
       "\n",
       "Name        PIK3CA_proteomics  ...  PIK3CD_proteomics  ROCK1_proteomics  \\\n",
       "Patient_ID                     ...                                        \n",
       "C3L-00104            0.124542  ...          -0.315003         -0.319701   \n",
       "C3L-00365           -0.211025  ...          -0.629943          0.013908   \n",
       "C3L-00674           -0.573919  ...           0.085276         -0.048311   \n",
       "C3L-00677           -0.360962  ...          -0.382652         -0.046639   \n",
       "C3L-01040            0.137414  ...          -0.537787          0.237764   \n",
       "...                       ...  ...                ...               ...   \n",
       "C3N-03183           -0.218186  ...           0.395350          0.141372   \n",
       "C3N-03184           -0.049803  ...          -0.556546         -0.158482   \n",
       "C3N-03186            0.815177  ...          -0.009546         -0.222781   \n",
       "C3N-03188            0.143824  ...          -0.255032         -0.256044   \n",
       "C3N-03473            0.163076  ...           0.002891         -0.191965   \n",
       "\n",
       "Name        SHC1_proteomics  NEDD4_proteomics  INPP4B_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00104         -0.095985         -0.437514           0.060124   \n",
       "C3L-00365          0.003049          0.179955          -0.639545   \n",
       "C3L-00674          0.051279          0.265735           0.669283   \n",
       "C3L-00677          0.288725          0.706601           0.451016   \n",
       "C3L-01040         -0.337079          0.133808          -0.024038   \n",
       "...                     ...               ...                ...   \n",
       "C3N-03183          0.557741          0.418850           0.010608   \n",
       "C3N-03184         -0.282123          0.028673          -0.219639   \n",
       "C3N-03186          0.061741          0.013211           0.348604   \n",
       "C3N-03188         -0.240227         -0.209677           0.055190   \n",
       "C3N-03473          0.123735          0.013306          -0.077057   \n",
       "\n",
       "Name        PIK3R1_proteomics  AKT1_proteomics  MVP_proteomics  \\\n",
       "Patient_ID                                                       \n",
       "C3L-00104           -0.379646        -0.065930        0.147936   \n",
       "C3L-00365           -0.283144        -0.285929        0.371665   \n",
       "C3L-00674            0.061201        -0.222607        0.329753   \n",
       "C3L-00677           -0.496340        -0.370584       -0.518407   \n",
       "C3L-01040           -0.146416         0.004573       -0.741119   \n",
       "...                       ...              ...             ...   \n",
       "C3N-03183           -0.093624        -0.002002       -0.090315   \n",
       "C3N-03184           -0.164426         0.501695       -0.994267   \n",
       "C3N-03186            0.443719         0.196700        0.086260   \n",
       "C3N-03188            0.084610        -0.198548       -0.012143   \n",
       "C3N-03473           -0.166501        -0.055991       -0.740250   \n",
       "\n",
       "Name        MAGI3_proteomics        Mutation  \n",
       "Patient_ID                                    \n",
       "C3L-00104           0.311427        Deletion  \n",
       "C3L-00365           0.259059        Deletion  \n",
       "C3L-00674           0.128442        Deletion  \n",
       "C3L-00677          -0.099551        Deletion  \n",
       "C3L-01040          -0.017146        Deletion  \n",
       "...                      ...             ...  \n",
       "C3N-03183          -0.079546        Deletion  \n",
       "C3N-03184          -0.364258        Deletion  \n",
       "C3N-03186           0.541898        Deletion  \n",
       "C3N-03188           0.260033  Wildtype_Tumor  \n",
       "C3N-03473          -0.134775        Deletion  \n",
       "\n",
       "[97 rows x 29 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = gbm.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = gbm.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "del_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "           Comparison       P_Value\n",
      "0    PTEN_proteomics  1.219523e-07\n",
      "1  PIK3CD_proteomics  1.030815e-03\n",
      "2  PIK3C3_proteomics  1.225949e-03\n",
      "3    USP7_proteomics  4.017792e-03\n",
      "4  PIK3R1_proteomics  4.282296e-03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.219523e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>1.030815e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>1.225949e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>4.017792e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>4.282296e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>1.260293e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>1.455578e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>2.487273e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>4.010871e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>5.478559e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>5.604258e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>1.040103e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.167020e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>1.184655e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>1.655635e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.242562e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>2.562153e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>2.876721e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>4.531770e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>5.493659e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>5.791747e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>6.343051e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>6.673038e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>6.682500e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>8.049218e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>8.663143e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.128906e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics   Gbm_P_Value\n",
       "0       PTEN_proteomics  1.219523e-07\n",
       "1     PIK3CD_proteomics  1.030815e-03\n",
       "2     PIK3C3_proteomics  1.225949e-03\n",
       "3       USP7_proteomics  4.017792e-03\n",
       "4     PIK3R1_proteomics  4.282296e-03\n",
       "5      MAST2_proteomics  1.260293e-02\n",
       "6     INPP4B_proteomics  1.455578e-02\n",
       "7   SLC9A3R1_proteomics  2.487273e-02\n",
       "8       EGFR_proteomics  4.010871e-02\n",
       "9      USP13_proteomics  5.478559e-02\n",
       "10   CSNK2A1_proteomics  5.604258e-02\n",
       "11    PIK3CB_proteomics  1.040103e-01\n",
       "12       MVP_proteomics  1.167020e-01\n",
       "13      XIAP_proteomics  1.184655e-01\n",
       "14      AKT1_proteomics  1.655635e-01\n",
       "15     ROCK1_proteomics  2.242562e-01\n",
       "16     MAGI3_proteomics  2.562153e-01\n",
       "17    PIK3CA_proteomics  2.876721e-01\n",
       "18   CSNK2A2_proteomics  4.531770e-01\n",
       "19      SHC1_proteomics  5.493659e-01\n",
       "20     PREX2_proteomics  5.791747e-01\n",
       "21    PIK3R2_proteomics  6.343051e-01\n",
       "22      PTK2_proteomics  6.673038e-01\n",
       "23      TP53_proteomics  6.682500e-01\n",
       "24     NEDD4_proteomics  8.049218e-01\n",
       "25      MDM2_proteomics  8.663143e-01\n",
       "26    PDGFRB_proteomics  9.128906e-01\n",
       "27    PIK3R3_proteomics           NaN"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "g_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if g_sig is not None:\n",
    "    g_sig_list = list(g_sig.Comparison)\n",
    "else: \n",
    "    l_sig_list = None\n",
    "print('significant pvals: \\n',g_sig)\n",
    "\n",
    "# Get all pvals\n",
    "g = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "g_pval = g.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Gbm_P_Value'})\n",
    "#g_pval = g_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "g_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>-0.127685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>-0.123874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>-0.028576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.015429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.102518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>-0.102838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.289788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>-0.111700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.105746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>-0.256630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>-1.001035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.042720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>-0.093309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>-0.154035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>-0.088967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.511726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>-0.151642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.139061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.492099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.080280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.153123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>-0.079598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.341829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.188390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>-0.169565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.474151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.050953</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Gbm_Median\n",
       "0       MDM2_proteomics   -0.127685\n",
       "1    CSNK2A1_proteomics   -0.123874\n",
       "2     PDGFRB_proteomics   -0.028576\n",
       "3     PIK3R2_proteomics    0.015429\n",
       "4     PIK3C3_proteomics    0.102518\n",
       "5    CSNK2A2_proteomics   -0.102838\n",
       "6   SLC9A3R1_proteomics    0.289788\n",
       "7     PIK3R3_proteomics         NaN\n",
       "8      USP13_proteomics   -0.111700\n",
       "9     PIK3CA_proteomics    0.105746\n",
       "10      TP53_proteomics   -0.256630\n",
       "11      EGFR_proteomics   -1.001035\n",
       "12    PIK3CB_proteomics    0.042720\n",
       "13      PTK2_proteomics   -0.093309\n",
       "14      USP7_proteomics   -0.154035\n",
       "15      XIAP_proteomics   -0.088967\n",
       "16      PTEN_proteomics    0.511726\n",
       "17     PREX2_proteomics   -0.151642\n",
       "18     MAST2_proteomics    0.139061\n",
       "19    PIK3CD_proteomics    0.492099\n",
       "20     ROCK1_proteomics    0.080280\n",
       "21      SHC1_proteomics    0.153123\n",
       "22     NEDD4_proteomics   -0.079598\n",
       "23    INPP4B_proteomics    0.341829\n",
       "24    PIK3R1_proteomics    0.188390\n",
       "25      AKT1_proteomics   -0.169565\n",
       "26       MVP_proteomics    0.474151\n",
       "27     MAGI3_proteomics    0.050953"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "gbm_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif_gbm = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    gbm_d[prot+'_proteomics'] = dif_gbm\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif_gbm)\n",
    "\n",
    "gbm_df = pd.DataFrame.from_dict(gbm_d, orient='index', columns=['Gbm_Median'])\n",
    "gbm_df = gbm_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "gbm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.219523e-07</td>\n",
       "      <td>0.511726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>1.030815e-03</td>\n",
       "      <td>0.492099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>1.225949e-03</td>\n",
       "      <td>0.102518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>4.017792e-03</td>\n",
       "      <td>-0.154035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>4.282296e-03</td>\n",
       "      <td>0.188390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>1.260293e-02</td>\n",
       "      <td>0.139061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>1.455578e-02</td>\n",
       "      <td>0.341829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>2.487273e-02</td>\n",
       "      <td>0.289788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>4.010871e-02</td>\n",
       "      <td>-1.001035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>5.478559e-02</td>\n",
       "      <td>-0.111700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>5.604258e-02</td>\n",
       "      <td>-0.123874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>1.040103e-01</td>\n",
       "      <td>0.042720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.167020e-01</td>\n",
       "      <td>0.474151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>1.184655e-01</td>\n",
       "      <td>-0.088967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>1.655635e-01</td>\n",
       "      <td>-0.169565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.242562e-01</td>\n",
       "      <td>0.080280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>2.562153e-01</td>\n",
       "      <td>0.050953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>2.876721e-01</td>\n",
       "      <td>0.105746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>4.531770e-01</td>\n",
       "      <td>-0.102838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>5.493659e-01</td>\n",
       "      <td>0.153123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>5.791747e-01</td>\n",
       "      <td>-0.151642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>6.343051e-01</td>\n",
       "      <td>0.015429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>6.673038e-01</td>\n",
       "      <td>-0.093309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>6.682500e-01</td>\n",
       "      <td>-0.256630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>8.049218e-01</td>\n",
       "      <td>-0.079598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>8.663143e-01</td>\n",
       "      <td>-0.127685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.128906e-01</td>\n",
       "      <td>-0.028576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics   Gbm_P_Value  Gbm_Median\n",
       "0       PTEN_proteomics  1.219523e-07    0.511726\n",
       "1     PIK3CD_proteomics  1.030815e-03    0.492099\n",
       "2     PIK3C3_proteomics  1.225949e-03    0.102518\n",
       "3       USP7_proteomics  4.017792e-03   -0.154035\n",
       "4     PIK3R1_proteomics  4.282296e-03    0.188390\n",
       "5      MAST2_proteomics  1.260293e-02    0.139061\n",
       "6     INPP4B_proteomics  1.455578e-02    0.341829\n",
       "7   SLC9A3R1_proteomics  2.487273e-02    0.289788\n",
       "8       EGFR_proteomics  4.010871e-02   -1.001035\n",
       "9      USP13_proteomics  5.478559e-02   -0.111700\n",
       "10   CSNK2A1_proteomics  5.604258e-02   -0.123874\n",
       "11    PIK3CB_proteomics  1.040103e-01    0.042720\n",
       "12       MVP_proteomics  1.167020e-01    0.474151\n",
       "13      XIAP_proteomics  1.184655e-01   -0.088967\n",
       "14      AKT1_proteomics  1.655635e-01   -0.169565\n",
       "15     ROCK1_proteomics  2.242562e-01    0.080280\n",
       "16     MAGI3_proteomics  2.562153e-01    0.050953\n",
       "17    PIK3CA_proteomics  2.876721e-01    0.105746\n",
       "18   CSNK2A2_proteomics  4.531770e-01   -0.102838\n",
       "19      SHC1_proteomics  5.493659e-01    0.153123\n",
       "20     PREX2_proteomics  5.791747e-01   -0.151642\n",
       "21    PIK3R2_proteomics  6.343051e-01    0.015429\n",
       "22      PTK2_proteomics  6.673038e-01   -0.093309\n",
       "23      TP53_proteomics  6.682500e-01   -0.256630\n",
       "24     NEDD4_proteomics  8.049218e-01   -0.079598\n",
       "25      MDM2_proteomics  8.663143e-01   -0.127685\n",
       "26    PDGFRB_proteomics  9.128906e-01   -0.028576\n",
       "27    PIK3R3_proteomics           NaN         NaN"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_merged = g_pval.merge(gbm_df, on='Proteomics',how='inner')\n",
    "g_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: MDM2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 69 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>MDM2_proteomics</th>\n",
       "      <th>CSNK2A1_proteomics</th>\n",
       "      <th>PDGFRB_proteomics</th>\n",
       "      <th>PIK3R2_proteomics</th>\n",
       "      <th>PIK3C3_proteomics</th>\n",
       "      <th>CSNK2A2_proteomics</th>\n",
       "      <th>SLC9A3R1_proteomics</th>\n",
       "      <th>PIK3R3_proteomics</th>\n",
       "      <th>USP13_proteomics</th>\n",
       "      <th>PIK3CA_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>PIK3CD_proteomics</th>\n",
       "      <th>ROCK1_proteomics</th>\n",
       "      <th>SHC1_proteomics</th>\n",
       "      <th>NEDD4_proteomics</th>\n",
       "      <th>INPP4B_proteomics</th>\n",
       "      <th>PIK3R1_proteomics</th>\n",
       "      <th>AKT1_proteomics</th>\n",
       "      <th>MVP_proteomics</th>\n",
       "      <th>MAGI3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01510</td>\n",
       "      <td>-0.4940</td>\n",
       "      <td>0.0540</td>\n",
       "      <td>-0.41600</td>\n",
       "      <td>0.2670</td>\n",
       "      <td>0.404</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0543</td>\n",
       "      <td>-0.4500</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.25200</td>\n",
       "      <td>-0.367</td>\n",
       "      <td>-0.1250</td>\n",
       "      <td>-0.8590</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.481</td>\n",
       "      <td>-0.2550</td>\n",
       "      <td>-0.167</td>\n",
       "      <td>-0.3070</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.28000</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>0.0502</td>\n",
       "      <td>-0.00815</td>\n",
       "      <td>-0.2410</td>\n",
       "      <td>0.659</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2780</td>\n",
       "      <td>-0.1780</td>\n",
       "      <td>...</td>\n",
       "      <td>0.45700</td>\n",
       "      <td>0.180</td>\n",
       "      <td>-0.0194</td>\n",
       "      <td>-0.7620</td>\n",
       "      <td>1.320</td>\n",
       "      <td>0.266</td>\n",
       "      <td>-0.0722</td>\n",
       "      <td>0.273</td>\n",
       "      <td>0.4150</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.35000</td>\n",
       "      <td>-1.3400</td>\n",
       "      <td>0.0690</td>\n",
       "      <td>0.04510</td>\n",
       "      <td>-0.0431</td>\n",
       "      <td>-1.150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0664</td>\n",
       "      <td>0.2870</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00543</td>\n",
       "      <td>-0.164</td>\n",
       "      <td>0.1030</td>\n",
       "      <td>-0.0684</td>\n",
       "      <td>0.201</td>\n",
       "      <td>-0.612</td>\n",
       "      <td>-0.6410</td>\n",
       "      <td>1.200</td>\n",
       "      <td>-0.2160</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00137</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.14300</td>\n",
       "      <td>-0.6530</td>\n",
       "      <td>0.1550</td>\n",
       "      <td>-0.59600</td>\n",
       "      <td>0.1050</td>\n",
       "      <td>-0.626</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>-0.3170</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.05000</td>\n",
       "      <td>-0.178</td>\n",
       "      <td>-0.1610</td>\n",
       "      <td>-1.2500</td>\n",
       "      <td>-0.859</td>\n",
       "      <td>-1.910</td>\n",
       "      <td>0.1430</td>\n",
       "      <td>-0.639</td>\n",
       "      <td>-0.0705</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00139</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.61300</td>\n",
       "      <td>-0.4280</td>\n",
       "      <td>0.6930</td>\n",
       "      <td>-0.07510</td>\n",
       "      <td>0.1470</td>\n",
       "      <td>-0.108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3000</td>\n",
       "      <td>0.2820</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.20400</td>\n",
       "      <td>-0.222</td>\n",
       "      <td>-0.4690</td>\n",
       "      <td>-0.9890</td>\n",
       "      <td>-0.329</td>\n",
       "      <td>-0.496</td>\n",
       "      <td>-0.7250</td>\n",
       "      <td>-0.565</td>\n",
       "      <td>0.0858</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.21200</td>\n",
       "      <td>-0.7660</td>\n",
       "      <td>0.1080</td>\n",
       "      <td>0.36700</td>\n",
       "      <td>-0.2410</td>\n",
       "      <td>1.150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5140</td>\n",
       "      <td>-0.4860</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.12900</td>\n",
       "      <td>-0.176</td>\n",
       "      <td>-0.4970</td>\n",
       "      <td>-0.9740</td>\n",
       "      <td>0.152</td>\n",
       "      <td>0.218</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>-0.707</td>\n",
       "      <td>-0.0286</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00398</td>\n",
       "      <td>-1.1200</td>\n",
       "      <td>-0.0944</td>\n",
       "      <td>-0.28300</td>\n",
       "      <td>0.1530</td>\n",
       "      <td>0.027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0900</td>\n",
       "      <td>0.0820</td>\n",
       "      <td>...</td>\n",
       "      <td>0.20500</td>\n",
       "      <td>-0.327</td>\n",
       "      <td>0.8070</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>0.820</td>\n",
       "      <td>0.117</td>\n",
       "      <td>0.1720</td>\n",
       "      <td>0.396</td>\n",
       "      <td>0.4750</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.07130</td>\n",
       "      <td>-0.7420</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>0.18900</td>\n",
       "      <td>0.1240</td>\n",
       "      <td>0.758</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2650</td>\n",
       "      <td>-0.0312</td>\n",
       "      <td>...</td>\n",
       "      <td>0.74600</td>\n",
       "      <td>-0.174</td>\n",
       "      <td>-0.1530</td>\n",
       "      <td>-0.8110</td>\n",
       "      <td>-0.224</td>\n",
       "      <td>-0.333</td>\n",
       "      <td>-0.2300</td>\n",
       "      <td>0.310</td>\n",
       "      <td>0.2760</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.28900</td>\n",
       "      <td>-0.8020</td>\n",
       "      <td>-0.4200</td>\n",
       "      <td>0.28500</td>\n",
       "      <td>0.0649</td>\n",
       "      <td>0.207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5340</td>\n",
       "      <td>0.1040</td>\n",
       "      <td>...</td>\n",
       "      <td>0.69100</td>\n",
       "      <td>0.280</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>0.1600</td>\n",
       "      <td>0.296</td>\n",
       "      <td>-0.446</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>1.820</td>\n",
       "      <td>0.0052</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.08870</td>\n",
       "      <td>0.2880</td>\n",
       "      <td>0.1840</td>\n",
       "      <td>0.26800</td>\n",
       "      <td>-0.0830</td>\n",
       "      <td>0.595</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2470</td>\n",
       "      <td>0.6810</td>\n",
       "      <td>...</td>\n",
       "      <td>0.58300</td>\n",
       "      <td>0.474</td>\n",
       "      <td>-0.0868</td>\n",
       "      <td>0.4280</td>\n",
       "      <td>-0.129</td>\n",
       "      <td>0.244</td>\n",
       "      <td>0.2330</td>\n",
       "      <td>0.186</td>\n",
       "      <td>-0.0173</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        MDM2_proteomics  CSNK2A1_proteomics  PDGFRB_proteomics  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00006               NaN             0.01510            -0.4940   \n",
       "C3L-00032               NaN            -0.28000             0.0143   \n",
       "C3L-00098               NaN             0.35000            -1.3400   \n",
       "C3L-00137               NaN             0.14300            -0.6530   \n",
       "C3L-00139               NaN             0.61300            -0.4280   \n",
       "...                     ...                 ...                ...   \n",
       "C3N-01520               NaN            -0.21200            -0.7660   \n",
       "C3N-01521               NaN            -0.00398            -1.1200   \n",
       "C3N-01537               NaN            -0.07130            -0.7420   \n",
       "C3N-01802               NaN             0.28900            -0.8020   \n",
       "C3N-01825               NaN             0.08870             0.2880   \n",
       "\n",
       "Name        PIK3R2_proteomics  PIK3C3_proteomics  CSNK2A2_proteomics  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00006              0.0540           -0.41600              0.2670   \n",
       "C3L-00032              0.0502           -0.00815             -0.2410   \n",
       "C3L-00098              0.0690            0.04510             -0.0431   \n",
       "C3L-00137              0.1550           -0.59600              0.1050   \n",
       "C3L-00139              0.6930           -0.07510              0.1470   \n",
       "...                       ...                ...                 ...   \n",
       "C3N-01520              0.1080            0.36700             -0.2410   \n",
       "C3N-01521             -0.0944           -0.28300              0.1530   \n",
       "C3N-01537             -0.1340            0.18900              0.1240   \n",
       "C3N-01802             -0.4200            0.28500              0.0649   \n",
       "C3N-01825              0.1840            0.26800             -0.0830   \n",
       "\n",
       "Name        SLC9A3R1_proteomics  PIK3R3_proteomics  USP13_proteomics  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00006                 0.404                NaN           -0.0543   \n",
       "C3L-00032                 0.659                NaN           -0.2780   \n",
       "C3L-00098                -1.150                NaN           -0.0664   \n",
       "C3L-00137                -0.626                NaN            0.1250   \n",
       "C3L-00139                -0.108                NaN           -0.3000   \n",
       "...                         ...                ...               ...   \n",
       "C3N-01520                 1.150                NaN            0.5140   \n",
       "C3N-01521                 0.027                NaN            1.0900   \n",
       "C3N-01537                 0.758                NaN           -0.2650   \n",
       "C3N-01802                 0.207                NaN           -0.5340   \n",
       "C3N-01825                 0.595                NaN            0.2470   \n",
       "\n",
       "Name        PIK3CA_proteomics  ...  PIK3CD_proteomics  ROCK1_proteomics  \\\n",
       "Patient_ID                     ...                                        \n",
       "C3L-00006             -0.4500  ...           -0.25200            -0.367   \n",
       "C3L-00032             -0.1780  ...            0.45700             0.180   \n",
       "C3L-00098              0.2870  ...            0.00543            -0.164   \n",
       "C3L-00137             -0.3170  ...           -1.05000            -0.178   \n",
       "C3L-00139              0.2820  ...           -0.20400            -0.222   \n",
       "...                       ...  ...                ...               ...   \n",
       "C3N-01520             -0.4860  ...           -0.12900            -0.176   \n",
       "C3N-01521              0.0820  ...            0.20500            -0.327   \n",
       "C3N-01537             -0.0312  ...            0.74600            -0.174   \n",
       "C3N-01802              0.1040  ...            0.69100             0.280   \n",
       "C3N-01825              0.6810  ...            0.58300             0.474   \n",
       "\n",
       "Name        SHC1_proteomics  NEDD4_proteomics  INPP4B_proteomics  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00006           -0.1250           -0.8590             -0.540   \n",
       "C3L-00032           -0.0194           -0.7620              1.320   \n",
       "C3L-00098            0.1030           -0.0684              0.201   \n",
       "C3L-00137           -0.1610           -1.2500             -0.859   \n",
       "C3L-00139           -0.4690           -0.9890             -0.329   \n",
       "...                     ...               ...                ...   \n",
       "C3N-01520           -0.4970           -0.9740              0.152   \n",
       "C3N-01521            0.8070           -0.4740              0.820   \n",
       "C3N-01537           -0.1530           -0.8110             -0.224   \n",
       "C3N-01802            0.2510            0.1600              0.296   \n",
       "C3N-01825           -0.0868            0.4280             -0.129   \n",
       "\n",
       "Name        PIK3R1_proteomics  AKT1_proteomics  MVP_proteomics  \\\n",
       "Patient_ID                                                       \n",
       "C3L-00006               0.481          -0.2550          -0.167   \n",
       "C3L-00032               0.266          -0.0722           0.273   \n",
       "C3L-00098              -0.612          -0.6410           1.200   \n",
       "C3L-00137              -1.910           0.1430          -0.639   \n",
       "C3L-00139              -0.496          -0.7250          -0.565   \n",
       "...                       ...              ...             ...   \n",
       "C3N-01520               0.218           0.0217          -0.707   \n",
       "C3N-01521               0.117           0.1720           0.396   \n",
       "C3N-01537              -0.333          -0.2300           0.310   \n",
       "C3N-01802              -0.446          -0.2150           1.820   \n",
       "C3N-01825               0.244           0.2330           0.186   \n",
       "\n",
       "Name        MAGI3_proteomics        Mutation  \n",
       "Patient_ID                                    \n",
       "C3L-00006            -0.3070      Truncation  \n",
       "C3L-00032             0.4150      Truncation  \n",
       "C3L-00098            -0.2160  Wildtype_Tumor  \n",
       "C3L-00137            -0.0705      Truncation  \n",
       "C3L-00139             0.0858  Wildtype_Tumor  \n",
       "...                      ...             ...  \n",
       "C3N-01520            -0.0286      Truncation  \n",
       "C3N-01521             0.4750  Wildtype_Tumor  \n",
       "C3N-01537             0.2760  Wildtype_Tumor  \n",
       "C3N-01802             0.0052  Wildtype_Tumor  \n",
       "C3N-01825            -0.0173  Wildtype_Tumor  \n",
       "\n",
       "[68 rows x 29 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Endo ttest\n",
    "\n",
    "mut_type_gbm = endo.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = endo.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "trunc_wt = merged[get]\n",
    "trunc_wt['Mutation'] = np.where(\n",
    "                trunc_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "trunc_wt['Mutation'].value_counts()\n",
    "trunc_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "            Comparison   P_Value\n",
      "0     PTEN_proteomics  0.000640\n",
      "1     TP53_proteomics  0.001246\n",
      "2   PIK3CA_proteomics  0.001761\n",
      "3  CSNK2A1_proteomics  0.002364\n",
      "4     EGFR_proteomics  0.006128\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.001246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.001761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.002364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.006128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.018340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.035238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.036592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.045489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.053078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.135943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.182172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.205099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.228439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.391469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.454039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.489186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.572500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.622263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.681581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.690012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.724325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.732837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.793385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.913864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.933397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  En_P_Value\n",
       "0       PTEN_proteomics    0.000640\n",
       "1       TP53_proteomics    0.001246\n",
       "2     PIK3CA_proteomics    0.001761\n",
       "3    CSNK2A1_proteomics    0.002364\n",
       "4       EGFR_proteomics    0.006128\n",
       "5       SHC1_proteomics    0.018340\n",
       "6    CSNK2A2_proteomics    0.035238\n",
       "7     PIK3C3_proteomics    0.036592\n",
       "8     PIK3CD_proteomics    0.045489\n",
       "9      PREX2_proteomics    0.053078\n",
       "10    PIK3R1_proteomics    0.135943\n",
       "11  SLC9A3R1_proteomics    0.182172\n",
       "12     NEDD4_proteomics    0.205099\n",
       "13     MAST2_proteomics    0.228439\n",
       "14    PDGFRB_proteomics    0.391469\n",
       "15      PTK2_proteomics    0.454039\n",
       "16    INPP4B_proteomics    0.489186\n",
       "17      USP7_proteomics    0.572500\n",
       "18       MVP_proteomics    0.622263\n",
       "19     MAGI3_proteomics    0.681581\n",
       "20     ROCK1_proteomics    0.690012\n",
       "21    PIK3CB_proteomics    0.724325\n",
       "22      AKT1_proteomics    0.732837\n",
       "23     USP13_proteomics    0.793385\n",
       "24      XIAP_proteomics    0.913864\n",
       "25    PIK3R2_proteomics    0.933397\n",
       "26      MDM2_proteomics         NaN\n",
       "27    PIK3R3_proteomics         NaN"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(trunc_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "e_sig = u.wrap_ttest(trunc_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if e_sig is not None:\n",
    "    e_sig_list = list(e_sig.Comparison)\n",
    "else: \n",
    "    e_sig_list = None\n",
    "print('significant pvals: \\n',e_sig)\n",
    "\n",
    "# Get all pvals\n",
    "e_pval = u.wrap_ttest(trunc_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "e_pval = e_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'En_P_Value'})\n",
    "e_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.10412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>-0.17100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.01500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.13160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.15970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>-0.33300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>-0.03000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.35300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.80260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>-0.52600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>-0.03370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.02340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>-0.03834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>-0.09772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.74660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>-0.45800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>-0.07090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.18243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.24980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.04500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>-0.16320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>-0.33990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.05160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.11890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.00430</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  En_Median\n",
       "0       MDM2_proteomics        NaN\n",
       "1    CSNK2A1_proteomics    0.10412\n",
       "2     PDGFRB_proteomics   -0.17100\n",
       "3     PIK3R2_proteomics    0.01500\n",
       "4     PIK3C3_proteomics    0.13160\n",
       "5    CSNK2A2_proteomics    0.15970\n",
       "6   SLC9A3R1_proteomics   -0.33300\n",
       "7     PIK3R3_proteomics        NaN\n",
       "8      USP13_proteomics   -0.03000\n",
       "9     PIK3CA_proteomics    0.35300\n",
       "10      TP53_proteomics    0.80260\n",
       "11      EGFR_proteomics   -0.52600\n",
       "12    PIK3CB_proteomics   -0.03370\n",
       "13      PTK2_proteomics    0.02340\n",
       "14      USP7_proteomics   -0.03834\n",
       "15      XIAP_proteomics   -0.09772\n",
       "16      PTEN_proteomics    0.74660\n",
       "17     PREX2_proteomics   -0.45800\n",
       "18     MAST2_proteomics   -0.07090\n",
       "19    PIK3CD_proteomics    0.18243\n",
       "20     ROCK1_proteomics   -0.08220\n",
       "21      SHC1_proteomics    0.24980\n",
       "22     NEDD4_proteomics    0.04500\n",
       "23    INPP4B_proteomics   -0.16320\n",
       "24    PIK3R1_proteomics   -0.33990\n",
       "25      AKT1_proteomics    0.05160\n",
       "26       MVP_proteomics    0.11890\n",
       "27     MAGI3_proteomics    0.00430"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = trunc_wt[trunc_wt.Mutation == \"Truncation\"]\n",
    "wt = trunc_wt[trunc_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "trunc_med = t.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "en_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - trunc_med[prot+'_proteomics']\n",
    "    en_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', trunc_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "en_df = pd.DataFrame.from_dict(en_d, orient='index', columns=['En_Median'])\n",
    "en_df = en_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "en_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>0.74660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.001246</td>\n",
       "      <td>0.80260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.001761</td>\n",
       "      <td>0.35300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.002364</td>\n",
       "      <td>0.10412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.006128</td>\n",
       "      <td>-0.52600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.018340</td>\n",
       "      <td>0.24980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.035238</td>\n",
       "      <td>0.15970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.036592</td>\n",
       "      <td>0.13160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.045489</td>\n",
       "      <td>0.18243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.053078</td>\n",
       "      <td>-0.45800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.135943</td>\n",
       "      <td>-0.33990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.182172</td>\n",
       "      <td>-0.33300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.205099</td>\n",
       "      <td>0.04500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.228439</td>\n",
       "      <td>-0.07090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.391469</td>\n",
       "      <td>-0.17100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.454039</td>\n",
       "      <td>0.02340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.489186</td>\n",
       "      <td>-0.16320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.572500</td>\n",
       "      <td>-0.03834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.622263</td>\n",
       "      <td>0.11890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.681581</td>\n",
       "      <td>0.00430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.690012</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.724325</td>\n",
       "      <td>-0.03370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.732837</td>\n",
       "      <td>0.05160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.793385</td>\n",
       "      <td>-0.03000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.913864</td>\n",
       "      <td>-0.09772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.933397</td>\n",
       "      <td>0.01500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  En_P_Value  En_Median\n",
       "0       PTEN_proteomics    0.000640    0.74660\n",
       "1       TP53_proteomics    0.001246    0.80260\n",
       "2     PIK3CA_proteomics    0.001761    0.35300\n",
       "3    CSNK2A1_proteomics    0.002364    0.10412\n",
       "4       EGFR_proteomics    0.006128   -0.52600\n",
       "5       SHC1_proteomics    0.018340    0.24980\n",
       "6    CSNK2A2_proteomics    0.035238    0.15970\n",
       "7     PIK3C3_proteomics    0.036592    0.13160\n",
       "8     PIK3CD_proteomics    0.045489    0.18243\n",
       "9      PREX2_proteomics    0.053078   -0.45800\n",
       "10    PIK3R1_proteomics    0.135943   -0.33990\n",
       "11  SLC9A3R1_proteomics    0.182172   -0.33300\n",
       "12     NEDD4_proteomics    0.205099    0.04500\n",
       "13     MAST2_proteomics    0.228439   -0.07090\n",
       "14    PDGFRB_proteomics    0.391469   -0.17100\n",
       "15      PTK2_proteomics    0.454039    0.02340\n",
       "16    INPP4B_proteomics    0.489186   -0.16320\n",
       "17      USP7_proteomics    0.572500   -0.03834\n",
       "18       MVP_proteomics    0.622263    0.11890\n",
       "19     MAGI3_proteomics    0.681581    0.00430\n",
       "20     ROCK1_proteomics    0.690012   -0.08220\n",
       "21    PIK3CB_proteomics    0.724325   -0.03370\n",
       "22      AKT1_proteomics    0.732837    0.05160\n",
       "23     USP13_proteomics    0.793385   -0.03000\n",
       "24      XIAP_proteomics    0.913864   -0.09772\n",
       "25    PIK3R2_proteomics    0.933397    0.01500\n",
       "26      MDM2_proteomics         NaN        NaN\n",
       "27    PIK3R3_proteomics         NaN        NaN"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_merged = e_pval.merge(en_df, on='Proteomics',how='inner')\n",
    "e_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                            \r"
     ]
    }
   ],
   "source": [
    "o = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: MDM2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 8)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 8)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    46\n",
       "Deletion          20\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "gene = 'PTEN'\n",
    "\n",
    "mut_type_gbm = o.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = o.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = o.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "         Comparison   P_Value\n",
      "0  PTEN_proteomics  0.000909\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "o_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if o_sig is not None:\n",
    "    o_sig_list = list(o_sig.Comparison)\n",
    "else: \n",
    "    o_sig_list = None\n",
    "print('significant pvals: \\n',o_sig)\n",
    "\n",
    "# Get all pvals\n",
    "o = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "o_pval = o.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Ov_P_Value'})\n",
    "#o_pval = o_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "#o_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "ov_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    ov_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "o_df = pd.DataFrame.from_dict(ov_d, orient='index', columns=['Ov_Median'])\n",
    "o_df = o_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#o_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000909</td>\n",
       "      <td>0.169972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.005865</td>\n",
       "      <td>-0.240986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.008335</td>\n",
       "      <td>-0.199296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.050313</td>\n",
       "      <td>-0.276354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.060296</td>\n",
       "      <td>-0.505300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.080941</td>\n",
       "      <td>-0.229494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.150601</td>\n",
       "      <td>-0.133108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.164860</td>\n",
       "      <td>0.085809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.219977</td>\n",
       "      <td>-0.179425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.231014</td>\n",
       "      <td>-0.163327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.337367</td>\n",
       "      <td>-0.001637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.353620</td>\n",
       "      <td>-0.370647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.429951</td>\n",
       "      <td>-0.145812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.444583</td>\n",
       "      <td>-0.104687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.486184</td>\n",
       "      <td>0.026832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.525069</td>\n",
       "      <td>0.017808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.530323</td>\n",
       "      <td>0.146323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.580920</td>\n",
       "      <td>-0.015518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.587071</td>\n",
       "      <td>-0.193693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.685027</td>\n",
       "      <td>0.010591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.838960</td>\n",
       "      <td>0.063388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.926598</td>\n",
       "      <td>0.154918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.937984</td>\n",
       "      <td>0.035308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.951727</td>\n",
       "      <td>-0.011741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.980145</td>\n",
       "      <td>-0.196586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.997559</td>\n",
       "      <td>-0.091044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Ov_P_Value  Ov_Median\n",
       "0       PTEN_proteomics    0.000909   0.169972\n",
       "1     PIK3R2_proteomics    0.005865  -0.240986\n",
       "2     PIK3CA_proteomics    0.008335  -0.199296\n",
       "3     PIK3CB_proteomics    0.050313  -0.276354\n",
       "4      PREX2_proteomics    0.060296  -0.505300\n",
       "5       PTK2_proteomics    0.080941  -0.229494\n",
       "6      MAGI3_proteomics    0.150601  -0.133108\n",
       "7      MAST2_proteomics    0.164860   0.085809\n",
       "8     PIK3R1_proteomics    0.219977  -0.179425\n",
       "9     PDGFRB_proteomics    0.231014  -0.163327\n",
       "10     NEDD4_proteomics    0.337367  -0.001637\n",
       "11    INPP4B_proteomics    0.353620  -0.370647\n",
       "12      AKT1_proteomics    0.429951  -0.145812\n",
       "13    PIK3C3_proteomics    0.444583  -0.104687\n",
       "14     ROCK1_proteomics    0.486184   0.026832\n",
       "15      USP7_proteomics    0.525069   0.017808\n",
       "16  SLC9A3R1_proteomics    0.530323   0.146323\n",
       "17   CSNK2A1_proteomics    0.580920  -0.015518\n",
       "18       MVP_proteomics    0.587071  -0.193693\n",
       "19      XIAP_proteomics    0.685027   0.010591\n",
       "20      EGFR_proteomics    0.838960   0.063388\n",
       "21     USP13_proteomics    0.926598   0.154918\n",
       "22    PIK3CD_proteomics    0.937984   0.035308\n",
       "23   CSNK2A2_proteomics    0.951727  -0.011741\n",
       "24      TP53_proteomics    0.980145  -0.196586\n",
       "25      SHC1_proteomics    0.997559  -0.091044\n",
       "26      MDM2_proteomics         NaN        NaN\n",
       "27    PIK3R3_proteomics         NaN        NaN"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "o_merged = o_pval.merge(o_df, on='Proteomics',how='inner')\n",
    "o_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Breast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    }
   ],
   "source": [
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CSNK2A1, MDM2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 6)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    80\n",
       "Deletion          24\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = b.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = b.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = b.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "         Comparison   P_Value\n",
      "0  PTEN_proteomics  0.000004\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "b_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if b_sig is not None:\n",
    "    b_sig_list = list(b_sig.Comparison)\n",
    "else: \n",
    "    b_sig_list = None\n",
    "print('significant pvals: \\n',b_sig)\n",
    "\n",
    "# Get all pvals\n",
    "b_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "b_pval = b_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Brca_P_Value'})\n",
    "#b_pval = b_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#b_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "b_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    b_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "b_df = pd.DataFrame.from_dict(b_d, orient='index', columns=['Brca_Median'])\n",
    "b_df = b_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#b_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>1.00800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.014172</td>\n",
       "      <td>-0.42620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.016614</td>\n",
       "      <td>-0.58645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.068427</td>\n",
       "      <td>-0.11100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.164475</td>\n",
       "      <td>1.03615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.213636</td>\n",
       "      <td>0.29705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.262209</td>\n",
       "      <td>-0.28100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.295970</td>\n",
       "      <td>0.37695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.325600</td>\n",
       "      <td>-0.28270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.353546</td>\n",
       "      <td>0.04900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.382956</td>\n",
       "      <td>-0.15150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.383804</td>\n",
       "      <td>-0.15040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.392146</td>\n",
       "      <td>-0.01540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.445080</td>\n",
       "      <td>-0.11490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.460310</td>\n",
       "      <td>0.27750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.522034</td>\n",
       "      <td>0.37085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.529497</td>\n",
       "      <td>0.48570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.684989</td>\n",
       "      <td>0.00110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.714108</td>\n",
       "      <td>0.13185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.831541</td>\n",
       "      <td>-0.00695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.836701</td>\n",
       "      <td>-0.15015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.885732</td>\n",
       "      <td>0.02575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.921867</td>\n",
       "      <td>-0.01090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.929331</td>\n",
       "      <td>0.14305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.980902</td>\n",
       "      <td>-0.04105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Brca_P_Value  Brca_Median\n",
       "0       PTEN_proteomics      0.000004      1.00800\n",
       "1       SHC1_proteomics      0.014172     -0.42620\n",
       "2       EGFR_proteomics      0.016614     -0.58645\n",
       "3       TP53_proteomics      0.068427     -0.11100\n",
       "4     PIK3R1_proteomics      0.164475      1.03615\n",
       "5      MAST2_proteomics      0.213636      0.29705\n",
       "6    CSNK2A2_proteomics      0.262209     -0.28100\n",
       "7     INPP4B_proteomics      0.295970      0.37695\n",
       "8      USP13_proteomics      0.325600     -0.28270\n",
       "9       PTK2_proteomics      0.353546      0.04900\n",
       "10     MAGI3_proteomics      0.382956     -0.15150\n",
       "11      XIAP_proteomics      0.383804     -0.15040\n",
       "12    PIK3CB_proteomics      0.392146     -0.01540\n",
       "13    PIK3C3_proteomics      0.445080     -0.11490\n",
       "14    PIK3CA_proteomics      0.460310      0.27750\n",
       "15  SLC9A3R1_proteomics      0.522034      0.37085\n",
       "16     NEDD4_proteomics      0.529497      0.48570\n",
       "17    PIK3R2_proteomics      0.684989      0.00110\n",
       "18      AKT1_proteomics      0.714108      0.13185\n",
       "19     ROCK1_proteomics      0.831541     -0.00695\n",
       "20    PIK3CD_proteomics      0.836701     -0.15015\n",
       "21     PREX2_proteomics      0.885732      0.02575\n",
       "22       MVP_proteomics      0.921867     -0.01090\n",
       "23      USP7_proteomics      0.929331      0.14305\n",
       "24    PDGFRB_proteomics      0.980902     -0.04105\n",
       "25   CSNK2A1_proteomics           NaN          NaN\n",
       "26      MDM2_proteomics           NaN          NaN\n",
       "27    PIK3R3_proteomics           NaN          NaN"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_merged = b_pval.merge(b_df, on='Proteomics',how='inner')\n",
    "b_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "col = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: MAGI3, MAST2, MDM2, PIK3R3, PREX2 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    76\n",
       "Deletion          25\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = col.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = col.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      " None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "c_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if c_sig is not None:\n",
    "    c_sig_list = list(c_sig.Comparison)\n",
    "else: \n",
    "    c_sig_list = None\n",
    "print('significant pvals: \\n',c_sig)\n",
    "\n",
    "# Get all pvals\n",
    "c_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "c_pval = c_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Colon_P_Value'})\n",
    "#c_pval = c_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#c_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "c_df = pd.DataFrame.from_dict(d, orient='index', columns=['Colon_Median'])\n",
    "c_df = c_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#c_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.20400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.030107</td>\n",
       "      <td>-0.11610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.116419</td>\n",
       "      <td>0.07995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.118135</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.131200</td>\n",
       "      <td>0.07600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.144425</td>\n",
       "      <td>0.04845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.182782</td>\n",
       "      <td>-0.23690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.237441</td>\n",
       "      <td>-0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.250276</td>\n",
       "      <td>-0.02450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.326372</td>\n",
       "      <td>-0.04825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.328626</td>\n",
       "      <td>0.22100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.332902</td>\n",
       "      <td>-0.03385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.354186</td>\n",
       "      <td>0.00670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.366886</td>\n",
       "      <td>0.02350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.375198</td>\n",
       "      <td>-0.05730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.491790</td>\n",
       "      <td>-0.14490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.516876</td>\n",
       "      <td>0.06900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.522867</td>\n",
       "      <td>0.00595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.561268</td>\n",
       "      <td>-0.03900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.665493</td>\n",
       "      <td>0.08550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.968870</td>\n",
       "      <td>-0.04905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.972777</td>\n",
       "      <td>0.01245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.974887</td>\n",
       "      <td>-0.18050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Colon_P_Value  Colon_Median\n",
       "0       PTEN_proteomics       0.012872       0.20400\n",
       "1       SHC1_proteomics       0.030107      -0.11610\n",
       "2     PIK3C3_proteomics       0.116419       0.07995\n",
       "3     INPP4B_proteomics       0.118135      -0.08220\n",
       "4     PIK3CD_proteomics       0.131200       0.07600\n",
       "5    CSNK2A1_proteomics       0.144425       0.04845\n",
       "6        MVP_proteomics       0.182782      -0.23690\n",
       "7       USP7_proteomics       0.237441      -0.04450\n",
       "8     PDGFRB_proteomics       0.250276      -0.02450\n",
       "9    CSNK2A2_proteomics       0.326372      -0.04825\n",
       "10  SLC9A3R1_proteomics       0.328626       0.22100\n",
       "11    PIK3CA_proteomics       0.332902      -0.03385\n",
       "12      AKT1_proteomics       0.354186       0.00670\n",
       "13      EGFR_proteomics       0.366886       0.02350\n",
       "14    PIK3R2_proteomics       0.375198      -0.05730\n",
       "15     NEDD4_proteomics       0.491790      -0.14490\n",
       "16      XIAP_proteomics       0.516876       0.06900\n",
       "17      PTK2_proteomics       0.522867       0.00595\n",
       "18     ROCK1_proteomics       0.561268      -0.03900\n",
       "19    PIK3CB_proteomics       0.665493       0.08550\n",
       "20      TP53_proteomics       0.968870      -0.04905\n",
       "21    PIK3R1_proteomics       0.972777       0.01245\n",
       "22     USP13_proteomics       0.974887      -0.18050\n",
       "23      MDM2_proteomics            NaN           NaN\n",
       "24    PIK3R3_proteomics            NaN           NaN\n",
       "25     PREX2_proteomics            NaN           NaN\n",
       "26     MAST2_proteomics            NaN           NaN\n",
       "27     MAGI3_proteomics            NaN           NaN"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c_merged = c_pval.merge(c_df, on='Proteomics',how='inner')\n",
    "c_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Head and Neck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "h = cptac.Hnscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: MDM2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 176 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    87\n",
       "Deletion          18\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = h.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = h.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "#prot_and_mutations = h.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "           Comparison   P_Value\n",
      "0    PTEN_proteomics  0.000023\n",
      "1  INPP4B_proteomics  0.000676\n",
      "2  PIK3CA_proteomics  0.001677\n",
      "3   USP13_proteomics  0.005645\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "h_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if h_sig is not None:\n",
    "    h_sig_list = list(h_sig.Comparison)\n",
    "else: \n",
    "    h_sig_list = None\n",
    "print('significant pvals: \\n',h_sig)\n",
    "\n",
    "# Get all pvals\n",
    "\n",
    "h_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "h_pval = h_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Hnscc_P_Value'})\n",
    "#h_pval = h_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#h_pval\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "h_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    h_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "h_df = pd.DataFrame.from_dict(h_d, orient='index', columns=['Hnscc_Median'])\n",
    "h_df = h_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#h_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>0.318812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.000676</td>\n",
       "      <td>0.591596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.001677</td>\n",
       "      <td>-0.163663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.005645</td>\n",
       "      <td>-0.217338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.014016</td>\n",
       "      <td>0.238171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.017719</td>\n",
       "      <td>-0.467227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.031385</td>\n",
       "      <td>-0.155484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.031801</td>\n",
       "      <td>-0.705560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.040586</td>\n",
       "      <td>-0.079700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.064384</td>\n",
       "      <td>-0.094081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.090920</td>\n",
       "      <td>-0.153286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.104981</td>\n",
       "      <td>-0.049397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.105902</td>\n",
       "      <td>-0.138133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.120966</td>\n",
       "      <td>0.045723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.145523</td>\n",
       "      <td>0.099953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.272988</td>\n",
       "      <td>-0.042043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.312904</td>\n",
       "      <td>0.041287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.328437</td>\n",
       "      <td>0.101923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.434622</td>\n",
       "      <td>-0.072064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.500462</td>\n",
       "      <td>-0.059333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.684058</td>\n",
       "      <td>0.000479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.699824</td>\n",
       "      <td>0.036617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.713094</td>\n",
       "      <td>0.081410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.783206</td>\n",
       "      <td>0.090114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.891248</td>\n",
       "      <td>0.093924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.995386</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Hnscc_P_Value  Hnscc_Median\n",
       "0       PTEN_proteomics       0.000023      0.318812\n",
       "1     INPP4B_proteomics       0.000676      0.591596\n",
       "2     PIK3CA_proteomics       0.001677     -0.163663\n",
       "3      USP13_proteomics       0.005645     -0.217338\n",
       "4     PIK3CD_proteomics       0.014016      0.238171\n",
       "5       EGFR_proteomics       0.017719     -0.467227\n",
       "6   SLC9A3R1_proteomics       0.031385     -0.155484\n",
       "7       TP53_proteomics       0.031801     -0.705560\n",
       "8    CSNK2A1_proteomics       0.040586     -0.079700\n",
       "9       USP7_proteomics       0.064384     -0.094081\n",
       "10    PIK3CB_proteomics       0.090920     -0.153286\n",
       "11      XIAP_proteomics       0.104981     -0.049397\n",
       "12      PTK2_proteomics       0.105902     -0.138133\n",
       "13      SHC1_proteomics       0.120966      0.045723\n",
       "14    PDGFRB_proteomics       0.145523      0.099953\n",
       "15    PIK3R2_proteomics       0.272988     -0.042043\n",
       "16    PIK3C3_proteomics       0.312904      0.041287\n",
       "17     ROCK1_proteomics       0.328437      0.101923\n",
       "18    PIK3R1_proteomics       0.434622     -0.072064\n",
       "19     MAGI3_proteomics       0.500462     -0.059333\n",
       "20      AKT1_proteomics       0.684058      0.000479\n",
       "21       MVP_proteomics       0.699824      0.036617\n",
       "22     MAST2_proteomics       0.713094      0.081410\n",
       "23     NEDD4_proteomics       0.783206      0.090114\n",
       "24   CSNK2A2_proteomics       0.891248      0.093924\n",
       "25      MDM2_proteomics            NaN           NaN\n",
       "26    PIK3R3_proteomics            NaN           NaN\n",
       "27     PREX2_proteomics            NaN     -0.995386"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h_merged = h_pval.merge(h_df, on='Proteomics',how='inner')\n",
    "#del_wt[['PREX2_proteomics','Mutation']].dropna() # in median, not in pval (only 1 value for deletion)\n",
    "h_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Lung"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LUAD dataset is under publication embargo until July 01, 2020. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    }
   ],
   "source": [
    "l = cptac.Luad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No_Mutation    88\n",
      "Deletion       17\n",
      "Name: Mutation, dtype: int64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-00001.N, C3L-00009.N, C3L-00080.N, C3L-00083.N, C3L-00093.N, C3L-00094.N, C3L-00095.N, C3L-00140.N, C3L-00144.N, C3L-00263.N, C3L-00279.N, C3L-00368.N, C3L-00412.N, C3L-00422.N, C3L-00510.N, C3L-00604.N, C3L-00893.N, C3L-00913.N, C3L-00973.N, C3L-01330.N, C3L-01632.N, C3L-01682.N, C3L-01683.N, C3L-01889.N, C3L-01890, C3L-01890.N, C3L-01924.N, C3L-02219.N, C3L-02345.N, C3L-02348.N, C3L-02350.N, C3L-02365.N, C3L-02508.N, C3L-02549.N, C3N-00167.N, C3N-00169.N, C3N-00175.N, C3N-00180.N, C3N-00199.N, C3N-00203.N, C3N-00217.N, C3N-00223.N, C3N-00293.N, C3N-00433.N, C3N-00546.N, C3N-00547.N, C3N-00549.N, C3N-00550.N, C3N-00551.N, C3N-00552.N, C3N-00556.N, C3N-00559.N, C3N-00560.N, C3N-00572.N, C3N-00574.N, C3N-00578.N, C3N-00579.N, C3N-00580.N, C3N-00704.N, C3N-00737.N, C3N-00738.N, C3N-00959.N, C3N-01016.N, C3N-01021.N, C3N-01023.N, C3N-01024.N, C3N-01030.N, C3N-01071.N, C3N-01072.N, C3N-01405.N, C3N-01410.N, C3N-01413.N, C3N-01414.N, C3N-01415.N, C3N-01416.N, C3N-01488.N, C3N-01489.N, C3N-01799.N, C3N-01823.N, C3N-02000.N, C3N-02002.N, C3N-02003.N, C3N-02067.N, C3N-02087.N, C3N-02089.N, C3N-02145.N, C3N-02149.N, C3N-02155.N, C3N-02158.N, C3N-02379.N, C3N-02380.N, C3N-02421.N, C3N-02423.N, C3N-02424.N, C3N-02433.N, C3N-02529.N, C3N-02572.N, C3N-02582.N, C3N-02586.N, C3N-02587.N, C3N-02588.N, C3N-02729.N (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: You tried to flatten an index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 12)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Index(['AKT1_proteomics', 'EGFR_proteomics', 'INPP4B_proteomics',\n",
       "       'MAGI3_proteomics', 'MAST2_proteomics', 'MVP_proteomics',\n",
       "       'NEDD4_proteomics', 'PDGFRB_proteomics', 'PIK3C3_proteomics',\n",
       "       'PIK3CA_proteomics', 'PIK3CB_proteomics', 'PIK3CD_proteomics',\n",
       "       'PIK3R2_proteomics', 'PREX2_proteomics', 'PTEN_proteomics',\n",
       "       'PTK2_proteomics', 'ROCK1_proteomics', 'SLC9A3R1_proteomics',\n",
       "       'TP53_proteomics', 'USP13_proteomics', 'USP7_proteomics',\n",
       "       'XIAP_proteomics'],\n",
       "      dtype='object', name='Name')"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = l.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.rename(columns={'PTEN':'cnv'})\n",
    "\n",
    "ip_prot = ['PDGFRB', 'PIK3R2', 'PIK3C3','SLC9A3R1',\n",
    "     'USP13','PIK3CA','TP53','EGFR','PIK3CB','PTK2','USP7','XIAP','PTEN','PREX2','MAST2','PIK3CD',\n",
    "    'ROCK1','NEDD4','INPP4B','AKT1', 'MVP', 'MAGI3'] # 'CSNK2A2', 'MDM2', 'CSNK2A1', 'PIK3R3' not in index\n",
    "# SH1, PRK3R1 both have double columns\n",
    "\n",
    "# different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "omics = l.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics',genes1=\"PTEN\", \n",
    "    genes2=ip_prot)\n",
    "omics = l.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "omics = omics.drop(columns='PTEN_CNV')\n",
    "\n",
    "# Get only tumor samples\n",
    "phos = l.get_proteomics(tissue_type='tumor')\n",
    "tumor_ids = list(phos.index)\n",
    "get = omics.index.isin(tumor_ids)\n",
    "omics = omics[get]\n",
    "\n",
    "merged = omics.join(mut_type) # checked and there is 110 tumor samples for lung\n",
    "\n",
    "compare = ['No_Mutation','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "print(del_wt['Mutation'].value_counts())\n",
    "del_wt.columns[:-2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot = l.get_proteomics(tissue_type='tumor')\n",
    "ip_prot = prot[['PDGFRB', 'PIK3R2', 'PIK3C3','SLC9A3R1',\n",
    "     'USP13','PIK3CA','TP53','EGFR','PIK3CB','PTK2','USP7','XIAP','PTEN','PREX2','MAST2','PIK3CD',\n",
    "    'ROCK1','NEDD4','INPP4B','AKT1', 'MVP', 'MAGI3']] # 'CSNK2A2', 'MDM2', 'CSNK2A1', 'PIK3R3' not in index\n",
    "# SH1, PRK3R1 both have double columns\n",
    "ip_prot = l.reduce_multiindex(ip_prot, levels_to_drop =1)\n",
    "#ip_prot = ip_prot.drop('SHC1_NP_001123512.1|NP_003020.2', axis = 1)\n",
    "ip_prot.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "           Comparison       P_Value\n",
      "0    EGFR_proteomics  9.121167e-07\n",
      "1  PIK3CD_proteomics  4.312883e-04\n",
      "2    XIAP_proteomics  4.264738e-03\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>9.121167e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>4.312883e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>4.264738e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.435202e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>2.191472e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>4.551992e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>7.637239e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>9.086574e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>9.235432e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.718800e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.303937e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>2.111085e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>2.392126e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.598046e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>2.766716e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>3.694158e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>3.715141e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>3.935556e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>4.395072e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>4.795317e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>5.491624e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>9.919357e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Luad_P_Value\n",
       "0       EGFR_proteomics  9.121167e-07\n",
       "1     PIK3CD_proteomics  4.312883e-04\n",
       "2       XIAP_proteomics  4.264738e-03\n",
       "3       PTEN_proteomics  1.435202e-02\n",
       "4       AKT1_proteomics  2.191472e-02\n",
       "5     PIK3CB_proteomics  4.551992e-02\n",
       "6     PIK3CA_proteomics  7.637239e-02\n",
       "7       PTK2_proteomics  9.086574e-02\n",
       "8      NEDD4_proteomics  9.235432e-02\n",
       "9     PDGFRB_proteomics  9.718800e-02\n",
       "10       MVP_proteomics  1.303937e-01\n",
       "11     PREX2_proteomics  2.111085e-01\n",
       "12      TP53_proteomics  2.392126e-01\n",
       "13     ROCK1_proteomics  2.598046e-01\n",
       "14    INPP4B_proteomics  2.766716e-01\n",
       "15     USP13_proteomics  3.694158e-01\n",
       "16      USP7_proteomics  3.715141e-01\n",
       "17     MAGI3_proteomics  3.935556e-01\n",
       "18    PIK3R2_proteomics  4.395072e-01\n",
       "19  SLC9A3R1_proteomics  4.795317e-01\n",
       "20    PIK3C3_proteomics  5.491624e-01\n",
       "21     MAST2_proteomics  9.919357e-01"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-2])\n",
    "\n",
    "# Get only sig sites\n",
    "l_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if l_sig is not None:\n",
    "    l_sig_list = list(l_sig.Comparison)\n",
    "else: \n",
    "    l_sig_list = None\n",
    "print('significant pvals: \\n',l_sig)\n",
    "\n",
    "# Get all pvals\n",
    "l_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "l_pval = l_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Luad_P_Value'})\n",
    "#l_pval = l_pval.replace(to_replace = r'_NP_.*$', value = '', regex = True)\n",
    "l_pval # isoforms for some proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Isoform code - fix later\n",
    "#del_wt.replace(to_replace = r'_NP_.*$', value = '', regex = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>-0.28010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>-2.32970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>-1.87510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>-0.45165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.01015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>-0.03475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.13545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.48860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.10100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.13565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.28425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.84795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>-0.27005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.79590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.36490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>-0.53510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>-0.27425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>-0.38490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>-2.76780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>-0.06650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>-0.13260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>-0.52480</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Luad_Median\n",
       "0       AKT1_proteomics     -0.28010\n",
       "1       EGFR_proteomics     -2.32970\n",
       "2     INPP4B_proteomics     -1.87510\n",
       "3      MAGI3_proteomics     -0.45165\n",
       "4      MAST2_proteomics      0.01015\n",
       "5        MVP_proteomics     -0.03475\n",
       "6      NEDD4_proteomics      0.13545\n",
       "7     PDGFRB_proteomics      0.48860\n",
       "8     PIK3C3_proteomics      0.10100\n",
       "9     PIK3CA_proteomics      0.13565\n",
       "10    PIK3CB_proteomics      0.28425\n",
       "11    PIK3CD_proteomics      0.84795\n",
       "12    PIK3R2_proteomics     -0.27005\n",
       "13     PREX2_proteomics      0.79590\n",
       "14      PTEN_proteomics      0.36490\n",
       "15      PTK2_proteomics     -0.53510\n",
       "16     ROCK1_proteomics     -0.27425\n",
       "17  SLC9A3R1_proteomics     -0.38490\n",
       "18      TP53_proteomics     -2.76780\n",
       "19     USP13_proteomics     -0.06650\n",
       "20      USP7_proteomics     -0.13260\n",
       "21      XIAP_proteomics     -0.52480"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"No_Mutation\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "l_d = {}\n",
    "\n",
    "for prot in cols:\n",
    "    dif = wt_med[prot] - del_med[prot]\n",
    "    l_d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "l_df = pd.DataFrame.from_dict(l_d, orient='index', columns=['Luad_Median'])\n",
    "l_df = l_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "l_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>9.121167e-07</td>\n",
       "      <td>-2.32970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>4.312883e-04</td>\n",
       "      <td>0.84795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>4.264738e-03</td>\n",
       "      <td>-0.52480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.435202e-02</td>\n",
       "      <td>0.36490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>2.191472e-02</td>\n",
       "      <td>-0.28010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>4.551992e-02</td>\n",
       "      <td>0.28425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>7.637239e-02</td>\n",
       "      <td>0.13565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>9.086574e-02</td>\n",
       "      <td>-0.53510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>9.235432e-02</td>\n",
       "      <td>0.13545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.718800e-02</td>\n",
       "      <td>0.48860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.303937e-01</td>\n",
       "      <td>-0.03475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>2.111085e-01</td>\n",
       "      <td>0.79590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>2.392126e-01</td>\n",
       "      <td>-2.76780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.598046e-01</td>\n",
       "      <td>-0.27425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>2.766716e-01</td>\n",
       "      <td>-1.87510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>3.694158e-01</td>\n",
       "      <td>-0.06650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>3.715141e-01</td>\n",
       "      <td>-0.13260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>3.935556e-01</td>\n",
       "      <td>-0.45165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>4.395072e-01</td>\n",
       "      <td>-0.27005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>4.795317e-01</td>\n",
       "      <td>-0.38490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>5.491624e-01</td>\n",
       "      <td>0.10100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>9.919357e-01</td>\n",
       "      <td>0.01015</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Luad_P_Value  Luad_Median\n",
       "0       EGFR_proteomics  9.121167e-07     -2.32970\n",
       "1     PIK3CD_proteomics  4.312883e-04      0.84795\n",
       "2       XIAP_proteomics  4.264738e-03     -0.52480\n",
       "3       PTEN_proteomics  1.435202e-02      0.36490\n",
       "4       AKT1_proteomics  2.191472e-02     -0.28010\n",
       "5     PIK3CB_proteomics  4.551992e-02      0.28425\n",
       "6     PIK3CA_proteomics  7.637239e-02      0.13565\n",
       "7       PTK2_proteomics  9.086574e-02     -0.53510\n",
       "8      NEDD4_proteomics  9.235432e-02      0.13545\n",
       "9     PDGFRB_proteomics  9.718800e-02      0.48860\n",
       "10       MVP_proteomics  1.303937e-01     -0.03475\n",
       "11     PREX2_proteomics  2.111085e-01      0.79590\n",
       "12      TP53_proteomics  2.392126e-01     -2.76780\n",
       "13     ROCK1_proteomics  2.598046e-01     -0.27425\n",
       "14    INPP4B_proteomics  2.766716e-01     -1.87510\n",
       "15     USP13_proteomics  3.694158e-01     -0.06650\n",
       "16      USP7_proteomics  3.715141e-01     -0.13260\n",
       "17     MAGI3_proteomics  3.935556e-01     -0.45165\n",
       "18    PIK3R2_proteomics  4.395072e-01     -0.27005\n",
       "19  SLC9A3R1_proteomics  4.795317e-01     -0.38490\n",
       "20    PIK3C3_proteomics  5.491624e-01      0.10100\n",
       "21     MAST2_proteomics  9.919357e-01      0.01015"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l_merged = l_pval.merge(l_df, on='Proteomics',how='inner')\n",
    "#l_merged = l_merged.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "l_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.219523e-07</td>\n",
       "      <td>0.511726</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>0.318812</td>\n",
       "      <td>1.435202e-02</td>\n",
       "      <td>0.36490</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>1.00800</td>\n",
       "      <td>0.000909</td>\n",
       "      <td>0.169972</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>0.74660</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.20400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>1.030815e-03</td>\n",
       "      <td>0.492099</td>\n",
       "      <td>0.014016</td>\n",
       "      <td>0.238171</td>\n",
       "      <td>4.312883e-04</td>\n",
       "      <td>0.84795</td>\n",
       "      <td>0.836701</td>\n",
       "      <td>-0.15015</td>\n",
       "      <td>0.937984</td>\n",
       "      <td>0.035308</td>\n",
       "      <td>0.045489</td>\n",
       "      <td>0.18243</td>\n",
       "      <td>0.131200</td>\n",
       "      <td>0.07600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>1.225949e-03</td>\n",
       "      <td>0.102518</td>\n",
       "      <td>0.312904</td>\n",
       "      <td>0.041287</td>\n",
       "      <td>5.491624e-01</td>\n",
       "      <td>0.10100</td>\n",
       "      <td>0.445080</td>\n",
       "      <td>-0.11490</td>\n",
       "      <td>0.444583</td>\n",
       "      <td>-0.104687</td>\n",
       "      <td>0.036592</td>\n",
       "      <td>0.13160</td>\n",
       "      <td>0.116419</td>\n",
       "      <td>0.07995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>4.017792e-03</td>\n",
       "      <td>-0.154035</td>\n",
       "      <td>0.064384</td>\n",
       "      <td>-0.094081</td>\n",
       "      <td>3.715141e-01</td>\n",
       "      <td>-0.13260</td>\n",
       "      <td>0.929331</td>\n",
       "      <td>0.14305</td>\n",
       "      <td>0.525069</td>\n",
       "      <td>0.017808</td>\n",
       "      <td>0.572500</td>\n",
       "      <td>-0.03834</td>\n",
       "      <td>0.237441</td>\n",
       "      <td>-0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>4.282296e-03</td>\n",
       "      <td>0.188390</td>\n",
       "      <td>0.434622</td>\n",
       "      <td>-0.072064</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.164475</td>\n",
       "      <td>1.03615</td>\n",
       "      <td>0.219977</td>\n",
       "      <td>-0.179425</td>\n",
       "      <td>0.135943</td>\n",
       "      <td>-0.33990</td>\n",
       "      <td>0.972777</td>\n",
       "      <td>0.01245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>1.260293e-02</td>\n",
       "      <td>0.139061</td>\n",
       "      <td>0.713094</td>\n",
       "      <td>0.081410</td>\n",
       "      <td>9.919357e-01</td>\n",
       "      <td>0.01015</td>\n",
       "      <td>0.213636</td>\n",
       "      <td>0.29705</td>\n",
       "      <td>0.164860</td>\n",
       "      <td>0.085809</td>\n",
       "      <td>0.228439</td>\n",
       "      <td>-0.07090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>1.455578e-02</td>\n",
       "      <td>0.341829</td>\n",
       "      <td>0.000676</td>\n",
       "      <td>0.591596</td>\n",
       "      <td>2.766716e-01</td>\n",
       "      <td>-1.87510</td>\n",
       "      <td>0.295970</td>\n",
       "      <td>0.37695</td>\n",
       "      <td>0.353620</td>\n",
       "      <td>-0.370647</td>\n",
       "      <td>0.489186</td>\n",
       "      <td>-0.16320</td>\n",
       "      <td>0.118135</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>2.487273e-02</td>\n",
       "      <td>0.289788</td>\n",
       "      <td>0.031385</td>\n",
       "      <td>-0.155484</td>\n",
       "      <td>4.795317e-01</td>\n",
       "      <td>-0.38490</td>\n",
       "      <td>0.522034</td>\n",
       "      <td>0.37085</td>\n",
       "      <td>0.530323</td>\n",
       "      <td>0.146323</td>\n",
       "      <td>0.182172</td>\n",
       "      <td>-0.33300</td>\n",
       "      <td>0.328626</td>\n",
       "      <td>0.22100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>4.010871e-02</td>\n",
       "      <td>-1.001035</td>\n",
       "      <td>0.017719</td>\n",
       "      <td>-0.467227</td>\n",
       "      <td>9.121167e-07</td>\n",
       "      <td>-2.32970</td>\n",
       "      <td>0.016614</td>\n",
       "      <td>-0.58645</td>\n",
       "      <td>0.838960</td>\n",
       "      <td>0.063388</td>\n",
       "      <td>0.006128</td>\n",
       "      <td>-0.52600</td>\n",
       "      <td>0.366886</td>\n",
       "      <td>0.02350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>5.478559e-02</td>\n",
       "      <td>-0.111700</td>\n",
       "      <td>0.005645</td>\n",
       "      <td>-0.217338</td>\n",
       "      <td>3.694158e-01</td>\n",
       "      <td>-0.06650</td>\n",
       "      <td>0.325600</td>\n",
       "      <td>-0.28270</td>\n",
       "      <td>0.926598</td>\n",
       "      <td>0.154918</td>\n",
       "      <td>0.793385</td>\n",
       "      <td>-0.03000</td>\n",
       "      <td>0.974887</td>\n",
       "      <td>-0.18050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>5.604258e-02</td>\n",
       "      <td>-0.123874</td>\n",
       "      <td>0.040586</td>\n",
       "      <td>-0.079700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.580920</td>\n",
       "      <td>-0.015518</td>\n",
       "      <td>0.002364</td>\n",
       "      <td>0.10412</td>\n",
       "      <td>0.144425</td>\n",
       "      <td>0.04845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>1.040103e-01</td>\n",
       "      <td>0.042720</td>\n",
       "      <td>0.090920</td>\n",
       "      <td>-0.153286</td>\n",
       "      <td>4.551992e-02</td>\n",
       "      <td>0.28425</td>\n",
       "      <td>0.392146</td>\n",
       "      <td>-0.01540</td>\n",
       "      <td>0.050313</td>\n",
       "      <td>-0.276354</td>\n",
       "      <td>0.724325</td>\n",
       "      <td>-0.03370</td>\n",
       "      <td>0.665493</td>\n",
       "      <td>0.08550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.167020e-01</td>\n",
       "      <td>0.474151</td>\n",
       "      <td>0.699824</td>\n",
       "      <td>0.036617</td>\n",
       "      <td>1.303937e-01</td>\n",
       "      <td>-0.03475</td>\n",
       "      <td>0.921867</td>\n",
       "      <td>-0.01090</td>\n",
       "      <td>0.587071</td>\n",
       "      <td>-0.193693</td>\n",
       "      <td>0.622263</td>\n",
       "      <td>0.11890</td>\n",
       "      <td>0.182782</td>\n",
       "      <td>-0.23690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>1.184655e-01</td>\n",
       "      <td>-0.088967</td>\n",
       "      <td>0.104981</td>\n",
       "      <td>-0.049397</td>\n",
       "      <td>4.264738e-03</td>\n",
       "      <td>-0.52480</td>\n",
       "      <td>0.383804</td>\n",
       "      <td>-0.15040</td>\n",
       "      <td>0.685027</td>\n",
       "      <td>0.010591</td>\n",
       "      <td>0.913864</td>\n",
       "      <td>-0.09772</td>\n",
       "      <td>0.516876</td>\n",
       "      <td>0.06900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>1.655635e-01</td>\n",
       "      <td>-0.169565</td>\n",
       "      <td>0.684058</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>2.191472e-02</td>\n",
       "      <td>-0.28010</td>\n",
       "      <td>0.714108</td>\n",
       "      <td>0.13185</td>\n",
       "      <td>0.429951</td>\n",
       "      <td>-0.145812</td>\n",
       "      <td>0.732837</td>\n",
       "      <td>0.05160</td>\n",
       "      <td>0.354186</td>\n",
       "      <td>0.00670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.242562e-01</td>\n",
       "      <td>0.080280</td>\n",
       "      <td>0.328437</td>\n",
       "      <td>0.101923</td>\n",
       "      <td>2.598046e-01</td>\n",
       "      <td>-0.27425</td>\n",
       "      <td>0.831541</td>\n",
       "      <td>-0.00695</td>\n",
       "      <td>0.486184</td>\n",
       "      <td>0.026832</td>\n",
       "      <td>0.690012</td>\n",
       "      <td>-0.08220</td>\n",
       "      <td>0.561268</td>\n",
       "      <td>-0.03900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>2.562153e-01</td>\n",
       "      <td>0.050953</td>\n",
       "      <td>0.500462</td>\n",
       "      <td>-0.059333</td>\n",
       "      <td>3.935556e-01</td>\n",
       "      <td>-0.45165</td>\n",
       "      <td>0.382956</td>\n",
       "      <td>-0.15150</td>\n",
       "      <td>0.150601</td>\n",
       "      <td>-0.133108</td>\n",
       "      <td>0.681581</td>\n",
       "      <td>0.00430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>2.876721e-01</td>\n",
       "      <td>0.105746</td>\n",
       "      <td>0.001677</td>\n",
       "      <td>-0.163663</td>\n",
       "      <td>7.637239e-02</td>\n",
       "      <td>0.13565</td>\n",
       "      <td>0.460310</td>\n",
       "      <td>0.27750</td>\n",
       "      <td>0.008335</td>\n",
       "      <td>-0.199296</td>\n",
       "      <td>0.001761</td>\n",
       "      <td>0.35300</td>\n",
       "      <td>0.332902</td>\n",
       "      <td>-0.03385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>4.531770e-01</td>\n",
       "      <td>-0.102838</td>\n",
       "      <td>0.891248</td>\n",
       "      <td>0.093924</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.262209</td>\n",
       "      <td>-0.28100</td>\n",
       "      <td>0.951727</td>\n",
       "      <td>-0.011741</td>\n",
       "      <td>0.035238</td>\n",
       "      <td>0.15970</td>\n",
       "      <td>0.326372</td>\n",
       "      <td>-0.04825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>5.493659e-01</td>\n",
       "      <td>0.153123</td>\n",
       "      <td>0.120966</td>\n",
       "      <td>0.045723</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.014172</td>\n",
       "      <td>-0.42620</td>\n",
       "      <td>0.997559</td>\n",
       "      <td>-0.091044</td>\n",
       "      <td>0.018340</td>\n",
       "      <td>0.24980</td>\n",
       "      <td>0.030107</td>\n",
       "      <td>-0.11610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>5.791747e-01</td>\n",
       "      <td>-0.151642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.995386</td>\n",
       "      <td>2.111085e-01</td>\n",
       "      <td>0.79590</td>\n",
       "      <td>0.885732</td>\n",
       "      <td>0.02575</td>\n",
       "      <td>0.060296</td>\n",
       "      <td>-0.505300</td>\n",
       "      <td>0.053078</td>\n",
       "      <td>-0.45800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>6.343051e-01</td>\n",
       "      <td>0.015429</td>\n",
       "      <td>0.272988</td>\n",
       "      <td>-0.042043</td>\n",
       "      <td>4.395072e-01</td>\n",
       "      <td>-0.27005</td>\n",
       "      <td>0.684989</td>\n",
       "      <td>0.00110</td>\n",
       "      <td>0.005865</td>\n",
       "      <td>-0.240986</td>\n",
       "      <td>0.933397</td>\n",
       "      <td>0.01500</td>\n",
       "      <td>0.375198</td>\n",
       "      <td>-0.05730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>6.673038e-01</td>\n",
       "      <td>-0.093309</td>\n",
       "      <td>0.105902</td>\n",
       "      <td>-0.138133</td>\n",
       "      <td>9.086574e-02</td>\n",
       "      <td>-0.53510</td>\n",
       "      <td>0.353546</td>\n",
       "      <td>0.04900</td>\n",
       "      <td>0.080941</td>\n",
       "      <td>-0.229494</td>\n",
       "      <td>0.454039</td>\n",
       "      <td>0.02340</td>\n",
       "      <td>0.522867</td>\n",
       "      <td>0.00595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>6.682500e-01</td>\n",
       "      <td>-0.256630</td>\n",
       "      <td>0.031801</td>\n",
       "      <td>-0.705560</td>\n",
       "      <td>2.392126e-01</td>\n",
       "      <td>-2.76780</td>\n",
       "      <td>0.068427</td>\n",
       "      <td>-0.11100</td>\n",
       "      <td>0.980145</td>\n",
       "      <td>-0.196586</td>\n",
       "      <td>0.001246</td>\n",
       "      <td>0.80260</td>\n",
       "      <td>0.968870</td>\n",
       "      <td>-0.04905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>8.049218e-01</td>\n",
       "      <td>-0.079598</td>\n",
       "      <td>0.783206</td>\n",
       "      <td>0.090114</td>\n",
       "      <td>9.235432e-02</td>\n",
       "      <td>0.13545</td>\n",
       "      <td>0.529497</td>\n",
       "      <td>0.48570</td>\n",
       "      <td>0.337367</td>\n",
       "      <td>-0.001637</td>\n",
       "      <td>0.205099</td>\n",
       "      <td>0.04500</td>\n",
       "      <td>0.491790</td>\n",
       "      <td>-0.14490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>8.663143e-01</td>\n",
       "      <td>-0.127685</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.128906e-01</td>\n",
       "      <td>-0.028576</td>\n",
       "      <td>0.145523</td>\n",
       "      <td>0.099953</td>\n",
       "      <td>9.718800e-02</td>\n",
       "      <td>0.48860</td>\n",
       "      <td>0.980902</td>\n",
       "      <td>-0.04105</td>\n",
       "      <td>0.231014</td>\n",
       "      <td>-0.163327</td>\n",
       "      <td>0.391469</td>\n",
       "      <td>-0.17100</td>\n",
       "      <td>0.250276</td>\n",
       "      <td>-0.02450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics   Gbm_P_Value  Gbm_Median  Hnscc_P_Value  \\\n",
       "0       PTEN_proteomics  1.219523e-07    0.511726       0.000023   \n",
       "1     PIK3CD_proteomics  1.030815e-03    0.492099       0.014016   \n",
       "2     PIK3C3_proteomics  1.225949e-03    0.102518       0.312904   \n",
       "3       USP7_proteomics  4.017792e-03   -0.154035       0.064384   \n",
       "4     PIK3R1_proteomics  4.282296e-03    0.188390       0.434622   \n",
       "5      MAST2_proteomics  1.260293e-02    0.139061       0.713094   \n",
       "6     INPP4B_proteomics  1.455578e-02    0.341829       0.000676   \n",
       "7   SLC9A3R1_proteomics  2.487273e-02    0.289788       0.031385   \n",
       "8       EGFR_proteomics  4.010871e-02   -1.001035       0.017719   \n",
       "9      USP13_proteomics  5.478559e-02   -0.111700       0.005645   \n",
       "10   CSNK2A1_proteomics  5.604258e-02   -0.123874       0.040586   \n",
       "11    PIK3CB_proteomics  1.040103e-01    0.042720       0.090920   \n",
       "12       MVP_proteomics  1.167020e-01    0.474151       0.699824   \n",
       "13      XIAP_proteomics  1.184655e-01   -0.088967       0.104981   \n",
       "14      AKT1_proteomics  1.655635e-01   -0.169565       0.684058   \n",
       "15     ROCK1_proteomics  2.242562e-01    0.080280       0.328437   \n",
       "16     MAGI3_proteomics  2.562153e-01    0.050953       0.500462   \n",
       "17    PIK3CA_proteomics  2.876721e-01    0.105746       0.001677   \n",
       "18   CSNK2A2_proteomics  4.531770e-01   -0.102838       0.891248   \n",
       "19      SHC1_proteomics  5.493659e-01    0.153123       0.120966   \n",
       "20     PREX2_proteomics  5.791747e-01   -0.151642            NaN   \n",
       "21    PIK3R2_proteomics  6.343051e-01    0.015429       0.272988   \n",
       "22      PTK2_proteomics  6.673038e-01   -0.093309       0.105902   \n",
       "23      TP53_proteomics  6.682500e-01   -0.256630       0.031801   \n",
       "24     NEDD4_proteomics  8.049218e-01   -0.079598       0.783206   \n",
       "25      MDM2_proteomics  8.663143e-01   -0.127685            NaN   \n",
       "26    PDGFRB_proteomics  9.128906e-01   -0.028576       0.145523   \n",
       "27    PIK3R3_proteomics           NaN         NaN            NaN   \n",
       "\n",
       "    Hnscc_Median  Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median  \\\n",
       "0       0.318812  1.435202e-02      0.36490      0.000004      1.00800   \n",
       "1       0.238171  4.312883e-04      0.84795      0.836701     -0.15015   \n",
       "2       0.041287  5.491624e-01      0.10100      0.445080     -0.11490   \n",
       "3      -0.094081  3.715141e-01     -0.13260      0.929331      0.14305   \n",
       "4      -0.072064           NaN          NaN      0.164475      1.03615   \n",
       "5       0.081410  9.919357e-01      0.01015      0.213636      0.29705   \n",
       "6       0.591596  2.766716e-01     -1.87510      0.295970      0.37695   \n",
       "7      -0.155484  4.795317e-01     -0.38490      0.522034      0.37085   \n",
       "8      -0.467227  9.121167e-07     -2.32970      0.016614     -0.58645   \n",
       "9      -0.217338  3.694158e-01     -0.06650      0.325600     -0.28270   \n",
       "10     -0.079700           NaN          NaN           NaN          NaN   \n",
       "11     -0.153286  4.551992e-02      0.28425      0.392146     -0.01540   \n",
       "12      0.036617  1.303937e-01     -0.03475      0.921867     -0.01090   \n",
       "13     -0.049397  4.264738e-03     -0.52480      0.383804     -0.15040   \n",
       "14      0.000479  2.191472e-02     -0.28010      0.714108      0.13185   \n",
       "15      0.101923  2.598046e-01     -0.27425      0.831541     -0.00695   \n",
       "16     -0.059333  3.935556e-01     -0.45165      0.382956     -0.15150   \n",
       "17     -0.163663  7.637239e-02      0.13565      0.460310      0.27750   \n",
       "18      0.093924           NaN          NaN      0.262209     -0.28100   \n",
       "19      0.045723           NaN          NaN      0.014172     -0.42620   \n",
       "20     -0.995386  2.111085e-01      0.79590      0.885732      0.02575   \n",
       "21     -0.042043  4.395072e-01     -0.27005      0.684989      0.00110   \n",
       "22     -0.138133  9.086574e-02     -0.53510      0.353546      0.04900   \n",
       "23     -0.705560  2.392126e-01     -2.76780      0.068427     -0.11100   \n",
       "24      0.090114  9.235432e-02      0.13545      0.529497      0.48570   \n",
       "25           NaN           NaN          NaN           NaN          NaN   \n",
       "26      0.099953  9.718800e-02      0.48860      0.980902     -0.04105   \n",
       "27           NaN           NaN          NaN           NaN          NaN   \n",
       "\n",
       "    Ov_P_Value  Ov_Median  En_P_Value  En_Median  Colon_P_Value  Colon_Median  \n",
       "0     0.000909   0.169972    0.000640    0.74660       0.012872       0.20400  \n",
       "1     0.937984   0.035308    0.045489    0.18243       0.131200       0.07600  \n",
       "2     0.444583  -0.104687    0.036592    0.13160       0.116419       0.07995  \n",
       "3     0.525069   0.017808    0.572500   -0.03834       0.237441      -0.04450  \n",
       "4     0.219977  -0.179425    0.135943   -0.33990       0.972777       0.01245  \n",
       "5     0.164860   0.085809    0.228439   -0.07090            NaN           NaN  \n",
       "6     0.353620  -0.370647    0.489186   -0.16320       0.118135      -0.08220  \n",
       "7     0.530323   0.146323    0.182172   -0.33300       0.328626       0.22100  \n",
       "8     0.838960   0.063388    0.006128   -0.52600       0.366886       0.02350  \n",
       "9     0.926598   0.154918    0.793385   -0.03000       0.974887      -0.18050  \n",
       "10    0.580920  -0.015518    0.002364    0.10412       0.144425       0.04845  \n",
       "11    0.050313  -0.276354    0.724325   -0.03370       0.665493       0.08550  \n",
       "12    0.587071  -0.193693    0.622263    0.11890       0.182782      -0.23690  \n",
       "13    0.685027   0.010591    0.913864   -0.09772       0.516876       0.06900  \n",
       "14    0.429951  -0.145812    0.732837    0.05160       0.354186       0.00670  \n",
       "15    0.486184   0.026832    0.690012   -0.08220       0.561268      -0.03900  \n",
       "16    0.150601  -0.133108    0.681581    0.00430            NaN           NaN  \n",
       "17    0.008335  -0.199296    0.001761    0.35300       0.332902      -0.03385  \n",
       "18    0.951727  -0.011741    0.035238    0.15970       0.326372      -0.04825  \n",
       "19    0.997559  -0.091044    0.018340    0.24980       0.030107      -0.11610  \n",
       "20    0.060296  -0.505300    0.053078   -0.45800            NaN           NaN  \n",
       "21    0.005865  -0.240986    0.933397    0.01500       0.375198      -0.05730  \n",
       "22    0.080941  -0.229494    0.454039    0.02340       0.522867       0.00595  \n",
       "23    0.980145  -0.196586    0.001246    0.80260       0.968870      -0.04905  \n",
       "24    0.337367  -0.001637    0.205099    0.04500       0.491790      -0.14490  \n",
       "25         NaN        NaN         NaN        NaN            NaN           NaN  \n",
       "26    0.231014  -0.163327    0.391469   -0.17100       0.250276      -0.02450  \n",
       "27         NaN        NaN         NaN        NaN            NaN           NaN  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n = g_merged.merge(h_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(l_merged, on='Proteomics',how='outer')\n",
    "n= n.merge(b_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(o_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(e_merged, on='Proteomics',how='outer')\n",
    "all_df = n.merge(c_merged, on='Proteomics',how='outer')\n",
    "all_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.219523e-07</td>\n",
       "      <td>0.511726</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>0.318812</td>\n",
       "      <td>1.435202e-02</td>\n",
       "      <td>0.36490</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>1.00800</td>\n",
       "      <td>0.000909</td>\n",
       "      <td>0.169972</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>0.74660</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.20400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>1.030815e-03</td>\n",
       "      <td>0.492099</td>\n",
       "      <td>0.014016</td>\n",
       "      <td>0.238171</td>\n",
       "      <td>4.312883e-04</td>\n",
       "      <td>0.84795</td>\n",
       "      <td>0.836701</td>\n",
       "      <td>-0.15015</td>\n",
       "      <td>0.937984</td>\n",
       "      <td>0.035308</td>\n",
       "      <td>0.045489</td>\n",
       "      <td>0.18243</td>\n",
       "      <td>0.131200</td>\n",
       "      <td>0.07600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>1.225949e-03</td>\n",
       "      <td>0.102518</td>\n",
       "      <td>0.312904</td>\n",
       "      <td>0.041287</td>\n",
       "      <td>5.491624e-01</td>\n",
       "      <td>0.10100</td>\n",
       "      <td>0.445080</td>\n",
       "      <td>-0.11490</td>\n",
       "      <td>0.444583</td>\n",
       "      <td>-0.104687</td>\n",
       "      <td>0.036592</td>\n",
       "      <td>0.13160</td>\n",
       "      <td>0.116419</td>\n",
       "      <td>0.07995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>1.455578e-02</td>\n",
       "      <td>0.341829</td>\n",
       "      <td>0.000676</td>\n",
       "      <td>0.591596</td>\n",
       "      <td>2.766716e-01</td>\n",
       "      <td>-1.87510</td>\n",
       "      <td>0.295970</td>\n",
       "      <td>0.37695</td>\n",
       "      <td>0.353620</td>\n",
       "      <td>-0.370647</td>\n",
       "      <td>0.489186</td>\n",
       "      <td>-0.16320</td>\n",
       "      <td>0.118135</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>4.010871e-02</td>\n",
       "      <td>-1.001035</td>\n",
       "      <td>0.017719</td>\n",
       "      <td>-0.467227</td>\n",
       "      <td>9.121167e-07</td>\n",
       "      <td>-2.32970</td>\n",
       "      <td>0.016614</td>\n",
       "      <td>-0.58645</td>\n",
       "      <td>0.838960</td>\n",
       "      <td>0.063388</td>\n",
       "      <td>0.006128</td>\n",
       "      <td>-0.52600</td>\n",
       "      <td>0.366886</td>\n",
       "      <td>0.02350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>2.876721e-01</td>\n",
       "      <td>0.105746</td>\n",
       "      <td>0.001677</td>\n",
       "      <td>-0.163663</td>\n",
       "      <td>7.637239e-02</td>\n",
       "      <td>0.13565</td>\n",
       "      <td>0.460310</td>\n",
       "      <td>0.27750</td>\n",
       "      <td>0.008335</td>\n",
       "      <td>-0.199296</td>\n",
       "      <td>0.001761</td>\n",
       "      <td>0.35300</td>\n",
       "      <td>0.332902</td>\n",
       "      <td>-0.03385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>6.682500e-01</td>\n",
       "      <td>-0.256630</td>\n",
       "      <td>0.031801</td>\n",
       "      <td>-0.705560</td>\n",
       "      <td>2.392126e-01</td>\n",
       "      <td>-2.76780</td>\n",
       "      <td>0.068427</td>\n",
       "      <td>-0.11100</td>\n",
       "      <td>0.980145</td>\n",
       "      <td>-0.196586</td>\n",
       "      <td>0.001246</td>\n",
       "      <td>0.80260</td>\n",
       "      <td>0.968870</td>\n",
       "      <td>-0.04905</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Proteomics   Gbm_P_Value  Gbm_Median  Hnscc_P_Value  Hnscc_Median  \\\n",
       "0     PTEN_proteomics  1.219523e-07    0.511726       0.000023      0.318812   \n",
       "1   PIK3CD_proteomics  1.030815e-03    0.492099       0.014016      0.238171   \n",
       "2   PIK3C3_proteomics  1.225949e-03    0.102518       0.312904      0.041287   \n",
       "6   INPP4B_proteomics  1.455578e-02    0.341829       0.000676      0.591596   \n",
       "8     EGFR_proteomics  4.010871e-02   -1.001035       0.017719     -0.467227   \n",
       "17  PIK3CA_proteomics  2.876721e-01    0.105746       0.001677     -0.163663   \n",
       "23    TP53_proteomics  6.682500e-01   -0.256630       0.031801     -0.705560   \n",
       "\n",
       "    Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median  Ov_P_Value  \\\n",
       "0   1.435202e-02      0.36490      0.000004      1.00800    0.000909   \n",
       "1   4.312883e-04      0.84795      0.836701     -0.15015    0.937984   \n",
       "2   5.491624e-01      0.10100      0.445080     -0.11490    0.444583   \n",
       "6   2.766716e-01     -1.87510      0.295970      0.37695    0.353620   \n",
       "8   9.121167e-07     -2.32970      0.016614     -0.58645    0.838960   \n",
       "17  7.637239e-02      0.13565      0.460310      0.27750    0.008335   \n",
       "23  2.392126e-01     -2.76780      0.068427     -0.11100    0.980145   \n",
       "\n",
       "    Ov_Median  En_P_Value  En_Median  Colon_P_Value  Colon_Median  \n",
       "0    0.169972    0.000640    0.74660       0.012872       0.20400  \n",
       "1    0.035308    0.045489    0.18243       0.131200       0.07600  \n",
       "2   -0.104687    0.036592    0.13160       0.116419       0.07995  \n",
       "6   -0.370647    0.489186   -0.16320       0.118135      -0.08220  \n",
       "8    0.063388    0.006128   -0.52600       0.366886       0.02350  \n",
       "17  -0.199296    0.001761    0.35300       0.332902      -0.03385  \n",
       "23  -0.196586    0.001246    0.80260       0.968870      -0.04905  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = ['Gbm_P_Value','Hnscc_P_Value','Luad_P_Value','Brca_P_Value','Ov_P_Value','En_P_Value','Colon_P_Value']\n",
    "\n",
    "bc = .05/28\n",
    "\n",
    "t = []\n",
    "\n",
    "for c in cols:\n",
    "    df = all_df[all_df[c] <= bc]\n",
    "    t.append(list(df.Proteomics))\n",
    "    \n",
    "flat_list = [item for sublist in t for item in sublist]\n",
    "sig = set(flat_list) # remove duplicates\n",
    "\n",
    "\n",
    "\n",
    "bool_df = all_df['Proteomics'].isin(sig)\n",
    "sig_df = all_df[bool_df]\n",
    "sig_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Keep only genes significant in > 1 cancer\n",
    "all_sig = []\n",
    "\n",
    "sig_lists = [g_sig_list, e_sig_list, b_sig_list, o_sig_list, c_sig_list, h_sig_list, h_sig_list, l_sig_list]\n",
    "\n",
    "for next_list in sig_lists:\n",
    "    if next_list is not None:\n",
    "        all_sig.append(next_list)\n",
    "        \n",
    "flat_list = [item for sublist in all_sig for item in sublist]\n",
    "sig = set(flat_list) # remove duplicates\n",
    "\n",
    "#sig = [re.sub(r'_phosphoproteomics', '', i) for i in sig] \n",
    "\n",
    "bool_df = all_df['Proteomics'].isin(sig)\n",
    "sig_df = all_df[bool_df]\n",
    "print(len(sig_df))\n",
    "sig"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_df.to_csv('at_least_one_significant_interacting.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = ip_df.std()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = stdev.to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev.loc[stdev[0] > .2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d_stdev = stdev.std()\n",
    "d_stdev"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = stdev.hist(bins = 20)\n",
    "plt.title(\"Standard Deviations of Pancancer CNV deletions and amplifications of PTEN on Interacting Proteins\\n\")\n",
    "plt.xlabel('StDeV')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(10,8)\n",
    "sns.set(font_scale = 1.3)\n",
    "a = sns.distplot(stdev, bins = 20)\n",
    "\n",
    "a.set_title(\"Standard Deviation Distribution for Pancancer PTEN Mutation Effect on Proteomics\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "a.set(xlabel = 'StDeV', ylabel = 'Frequency')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_df = stdev.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = mean_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Mean between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Mean')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m_df = stdev.median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = m_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Medians between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Median')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
