{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Interacting Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "\n",
    "import sys\n",
    "sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\WhenMutationsDontMatter\\\\')\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that endometrial index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that ovarian index is up-to-date...    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LUAD dataset is under publication embargo until July 01, 2020. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 4)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                            \r"
     ]
    }
   ],
   "source": [
    "gbm = cptac.Gbm()\n",
    "endo = cptac.Endometrial()\n",
    "h = cptac.Hnscc()\n",
    "l = cptac.Luad()\n",
    "o = cptac.Ovarian()\n",
    "col = cptac.Colon()\n",
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "46\n"
     ]
    }
   ],
   "source": [
    "ip = u.get_interacting_proteins('PTEN')\n",
    "\n",
    "# RAS-RAF-MEK-ERK MAPK Pathway\n",
    "ip.append('GRB2')\n",
    "ip.append('SHC1') # (SHC in paper)\n",
    "ip.append('SOS1') # (SOS in the paper) *GEF for RAS \"recruitment to the plasma membrane requires GRB2, PIP2, PA\" \n",
    "ip.append('KRAS') # (RAS in paper) possibly add \"RAS regulators such as PHLPP, SHP-2, and NF-2\"\n",
    "ip.append('RAF1') # (RAF-1 in paper)\n",
    "ip.append('MAP2K1') #(MEK1 protein name in paper)\n",
    "ip.append('MAP2K2 ') # (MEK2 in paper)\n",
    "ip.append('MAPK3') # (ERK1 in paper) https://www.uniprot.org/uniprot/P27361\n",
    "ip.append('MAPK1') # (ERK2 in paper)\n",
    "\n",
    "#mTOR Pathway\n",
    "ip.append('GAB1')\n",
    "#ip.append('PI3K')\n",
    "ip.append('AKT1') # (AKT in paper) *binds PIP3 (see info for isoform AKT1 at: https://www.uniprot.org/uniprot/P31749)\n",
    "ip.append('PDK1') #activate AKT\n",
    "ip.append('MTORC2') #activate AKT\n",
    "ip.append('GSK3B') # (GSK-3B in paper)inhibit cyclin D (akt inhibits GSK3B)\n",
    "ip.append('FOXO1') #(FOXO in paper)\n",
    "ip.append('TSC2') # repress mTOR, AKT inhibits\n",
    "ip.append('MTOR') # promote cyclin D\n",
    "ip.append('BAD')\n",
    "ip.append('CAS9')\n",
    "\n",
    "ip.append('CCND1') # Cyclin D1 -> G1/S \n",
    "\n",
    "\n",
    "singles = set(ip)\n",
    "ip = list(singles)\n",
    "print(len(ip))\n",
    "#ip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CAS9, MAP2K2 , MTORC2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>MAGI3_proteomics</th>\n",
       "      <th>PREX2_proteomics</th>\n",
       "      <th>PIK3R1_proteomics</th>\n",
       "      <th>NEDD4_proteomics</th>\n",
       "      <th>FOXO1_proteomics</th>\n",
       "      <th>SHC1_proteomics</th>\n",
       "      <th>PIK3R2_proteomics</th>\n",
       "      <th>MTOR_proteomics</th>\n",
       "      <th>AKT1_proteomics</th>\n",
       "      <th>MTORC2_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>SOS1_proteomics</th>\n",
       "      <th>PIK3CD_proteomics</th>\n",
       "      <th>MAPK1_proteomics</th>\n",
       "      <th>CAS9_proteomics</th>\n",
       "      <th>CSNK2A2_proteomics</th>\n",
       "      <th>PIK3CB_proteomics</th>\n",
       "      <th>PDK1_proteomics</th>\n",
       "      <th>XIAP_proteomics</th>\n",
       "      <th>PIK3C3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>0.311427</td>\n",
       "      <td>-0.285658</td>\n",
       "      <td>-0.379646</td>\n",
       "      <td>-0.437514</td>\n",
       "      <td>-0.085520</td>\n",
       "      <td>-0.095985</td>\n",
       "      <td>-0.068085</td>\n",
       "      <td>0.136882</td>\n",
       "      <td>-0.065930</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.086085</td>\n",
       "      <td>-0.315003</td>\n",
       "      <td>0.364629</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.273672</td>\n",
       "      <td>-0.550558</td>\n",
       "      <td>0.857020</td>\n",
       "      <td>-0.069611</td>\n",
       "      <td>0.081642</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>0.259059</td>\n",
       "      <td>-0.077928</td>\n",
       "      <td>-0.283144</td>\n",
       "      <td>0.179955</td>\n",
       "      <td>0.066860</td>\n",
       "      <td>0.003049</td>\n",
       "      <td>0.079710</td>\n",
       "      <td>0.184699</td>\n",
       "      <td>-0.285929</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.198497</td>\n",
       "      <td>-0.629943</td>\n",
       "      <td>0.419605</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.035616</td>\n",
       "      <td>0.191167</td>\n",
       "      <td>0.180948</td>\n",
       "      <td>0.028903</td>\n",
       "      <td>-0.027524</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>0.128442</td>\n",
       "      <td>-0.189948</td>\n",
       "      <td>0.061201</td>\n",
       "      <td>0.265735</td>\n",
       "      <td>0.185222</td>\n",
       "      <td>0.051279</td>\n",
       "      <td>0.078025</td>\n",
       "      <td>-0.285577</td>\n",
       "      <td>-0.222607</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.166658</td>\n",
       "      <td>0.085276</td>\n",
       "      <td>-0.369258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.105403</td>\n",
       "      <td>-0.548042</td>\n",
       "      <td>0.344142</td>\n",
       "      <td>-0.479773</td>\n",
       "      <td>0.062739</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>-0.099551</td>\n",
       "      <td>-0.455972</td>\n",
       "      <td>-0.496340</td>\n",
       "      <td>0.706601</td>\n",
       "      <td>0.251746</td>\n",
       "      <td>0.288725</td>\n",
       "      <td>0.024117</td>\n",
       "      <td>0.093500</td>\n",
       "      <td>-0.370584</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.014030</td>\n",
       "      <td>-0.382652</td>\n",
       "      <td>-0.090202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.189861</td>\n",
       "      <td>-0.397225</td>\n",
       "      <td>-0.014611</td>\n",
       "      <td>-0.205255</td>\n",
       "      <td>0.087752</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-0.017146</td>\n",
       "      <td>1.097753</td>\n",
       "      <td>-0.146416</td>\n",
       "      <td>0.133808</td>\n",
       "      <td>-0.190030</td>\n",
       "      <td>-0.337079</td>\n",
       "      <td>-0.029297</td>\n",
       "      <td>0.023296</td>\n",
       "      <td>0.004573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.269299</td>\n",
       "      <td>-0.537787</td>\n",
       "      <td>-0.919902</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.239919</td>\n",
       "      <td>-0.395284</td>\n",
       "      <td>-0.674527</td>\n",
       "      <td>-0.123213</td>\n",
       "      <td>-0.057555</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>-0.079546</td>\n",
       "      <td>-0.356248</td>\n",
       "      <td>-0.093624</td>\n",
       "      <td>0.418850</td>\n",
       "      <td>0.207731</td>\n",
       "      <td>0.557741</td>\n",
       "      <td>-0.116620</td>\n",
       "      <td>0.023302</td>\n",
       "      <td>-0.002002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.014041</td>\n",
       "      <td>0.395350</td>\n",
       "      <td>0.382110</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.022308</td>\n",
       "      <td>-0.223580</td>\n",
       "      <td>-0.262954</td>\n",
       "      <td>-0.042216</td>\n",
       "      <td>-0.092783</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>-0.364258</td>\n",
       "      <td>0.279289</td>\n",
       "      <td>-0.164426</td>\n",
       "      <td>0.028673</td>\n",
       "      <td>0.036764</td>\n",
       "      <td>-0.282123</td>\n",
       "      <td>-0.023912</td>\n",
       "      <td>-0.018453</td>\n",
       "      <td>0.501695</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.132077</td>\n",
       "      <td>-0.556546</td>\n",
       "      <td>-0.337446</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.266436</td>\n",
       "      <td>-0.036881</td>\n",
       "      <td>-1.511121</td>\n",
       "      <td>-0.012862</td>\n",
       "      <td>-0.160911</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>0.541898</td>\n",
       "      <td>0.221264</td>\n",
       "      <td>0.443719</td>\n",
       "      <td>0.013211</td>\n",
       "      <td>0.135086</td>\n",
       "      <td>0.061741</td>\n",
       "      <td>0.626821</td>\n",
       "      <td>-0.024103</td>\n",
       "      <td>0.196700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.052877</td>\n",
       "      <td>-0.009546</td>\n",
       "      <td>0.088424</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.161898</td>\n",
       "      <td>0.236419</td>\n",
       "      <td>0.468991</td>\n",
       "      <td>-0.206451</td>\n",
       "      <td>-0.029083</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>0.260033</td>\n",
       "      <td>0.265185</td>\n",
       "      <td>0.084610</td>\n",
       "      <td>-0.209677</td>\n",
       "      <td>-0.159436</td>\n",
       "      <td>-0.240227</td>\n",
       "      <td>-0.135303</td>\n",
       "      <td>0.159754</td>\n",
       "      <td>-0.198548</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.097524</td>\n",
       "      <td>-0.255032</td>\n",
       "      <td>0.070822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.246281</td>\n",
       "      <td>0.107452</td>\n",
       "      <td>-0.371473</td>\n",
       "      <td>0.336283</td>\n",
       "      <td>0.173834</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>-0.134775</td>\n",
       "      <td>-0.572131</td>\n",
       "      <td>-0.166501</td>\n",
       "      <td>0.013306</td>\n",
       "      <td>0.103327</td>\n",
       "      <td>0.123735</td>\n",
       "      <td>0.186960</td>\n",
       "      <td>0.032411</td>\n",
       "      <td>-0.055991</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.030563</td>\n",
       "      <td>0.002891</td>\n",
       "      <td>-0.166356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.064866</td>\n",
       "      <td>-0.054008</td>\n",
       "      <td>-0.658736</td>\n",
       "      <td>0.096203</td>\n",
       "      <td>-0.073499</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97 rows Ã— 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        MAGI3_proteomics  PREX2_proteomics  PIK3R1_proteomics  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104           0.311427         -0.285658          -0.379646   \n",
       "C3L-00365           0.259059         -0.077928          -0.283144   \n",
       "C3L-00674           0.128442         -0.189948           0.061201   \n",
       "C3L-00677          -0.099551         -0.455972          -0.496340   \n",
       "C3L-01040          -0.017146          1.097753          -0.146416   \n",
       "...                      ...               ...                ...   \n",
       "C3N-03183          -0.079546         -0.356248          -0.093624   \n",
       "C3N-03184          -0.364258          0.279289          -0.164426   \n",
       "C3N-03186           0.541898          0.221264           0.443719   \n",
       "C3N-03188           0.260033          0.265185           0.084610   \n",
       "C3N-03473          -0.134775         -0.572131          -0.166501   \n",
       "\n",
       "Name        NEDD4_proteomics  FOXO1_proteomics  SHC1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00104          -0.437514         -0.085520        -0.095985   \n",
       "C3L-00365           0.179955          0.066860         0.003049   \n",
       "C3L-00674           0.265735          0.185222         0.051279   \n",
       "C3L-00677           0.706601          0.251746         0.288725   \n",
       "C3L-01040           0.133808         -0.190030        -0.337079   \n",
       "...                      ...               ...              ...   \n",
       "C3N-03183           0.418850          0.207731         0.557741   \n",
       "C3N-03184           0.028673          0.036764        -0.282123   \n",
       "C3N-03186           0.013211          0.135086         0.061741   \n",
       "C3N-03188          -0.209677         -0.159436        -0.240227   \n",
       "C3N-03473           0.013306          0.103327         0.123735   \n",
       "\n",
       "Name        PIK3R2_proteomics  MTOR_proteomics  AKT1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00104           -0.068085         0.136882        -0.065930   \n",
       "C3L-00365            0.079710         0.184699        -0.285929   \n",
       "C3L-00674            0.078025        -0.285577        -0.222607   \n",
       "C3L-00677            0.024117         0.093500        -0.370584   \n",
       "C3L-01040           -0.029297         0.023296         0.004573   \n",
       "...                       ...              ...              ...   \n",
       "C3N-03183           -0.116620         0.023302        -0.002002   \n",
       "C3N-03184           -0.023912        -0.018453         0.501695   \n",
       "C3N-03186            0.626821        -0.024103         0.196700   \n",
       "C3N-03188           -0.135303         0.159754        -0.198548   \n",
       "C3N-03473            0.186960         0.032411        -0.055991   \n",
       "\n",
       "Name        MTORC2_proteomics  ...  SOS1_proteomics  PIK3CD_proteomics  \\\n",
       "Patient_ID                     ...                                       \n",
       "C3L-00104                 NaN  ...         0.086085          -0.315003   \n",
       "C3L-00365                 NaN  ...         0.198497          -0.629943   \n",
       "C3L-00674                 NaN  ...        -0.166658           0.085276   \n",
       "C3L-00677                 NaN  ...        -0.014030          -0.382652   \n",
       "C3L-01040                 NaN  ...        -0.269299          -0.537787   \n",
       "...                       ...  ...              ...                ...   \n",
       "C3N-03183                 NaN  ...        -0.014041           0.395350   \n",
       "C3N-03184                 NaN  ...        -0.132077          -0.556546   \n",
       "C3N-03186                 NaN  ...        -0.052877          -0.009546   \n",
       "C3N-03188                 NaN  ...        -0.097524          -0.255032   \n",
       "C3N-03473                 NaN  ...         0.030563           0.002891   \n",
       "\n",
       "Name        MAPK1_proteomics  CAS9_proteomics  CSNK2A2_proteomics  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104           0.364629              NaN           -0.273672   \n",
       "C3L-00365           0.419605              NaN           -0.035616   \n",
       "C3L-00674          -0.369258              NaN           -0.105403   \n",
       "C3L-00677          -0.090202              NaN           -0.189861   \n",
       "C3L-01040          -0.919902              NaN            0.239919   \n",
       "...                      ...              ...                 ...   \n",
       "C3N-03183           0.382110              NaN           -0.022308   \n",
       "C3N-03184          -0.337446              NaN            0.266436   \n",
       "C3N-03186           0.088424              NaN           -0.161898   \n",
       "C3N-03188           0.070822              NaN            0.246281   \n",
       "C3N-03473          -0.166356              NaN            0.064866   \n",
       "\n",
       "Name        PIK3CB_proteomics  PDK1_proteomics  XIAP_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00104           -0.550558         0.857020        -0.069611   \n",
       "C3L-00365            0.191167         0.180948         0.028903   \n",
       "C3L-00674           -0.548042         0.344142        -0.479773   \n",
       "C3L-00677           -0.397225        -0.014611        -0.205255   \n",
       "C3L-01040           -0.395284        -0.674527        -0.123213   \n",
       "...                       ...              ...              ...   \n",
       "C3N-03183           -0.223580        -0.262954        -0.042216   \n",
       "C3N-03184           -0.036881        -1.511121        -0.012862   \n",
       "C3N-03186            0.236419         0.468991        -0.206451   \n",
       "C3N-03188            0.107452        -0.371473         0.336283   \n",
       "C3N-03473           -0.054008        -0.658736         0.096203   \n",
       "\n",
       "Name        PIK3C3_proteomics        Mutation  \n",
       "Patient_ID                                     \n",
       "C3L-00104            0.081642        Deletion  \n",
       "C3L-00365           -0.027524        Deletion  \n",
       "C3L-00674            0.062739        Deletion  \n",
       "C3L-00677            0.087752        Deletion  \n",
       "C3L-01040           -0.057555        Deletion  \n",
       "...                       ...             ...  \n",
       "C3N-03183           -0.092783        Deletion  \n",
       "C3N-03184           -0.160911        Deletion  \n",
       "C3N-03186           -0.029083        Deletion  \n",
       "C3N-03188            0.173834  Wildtype_Tumor  \n",
       "C3N-03473           -0.073499        Deletion  \n",
       "\n",
       "[97 rows x 47 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = gbm.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = gbm.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "#del_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "           Comparison       P_Value\n",
      "0    PTEN_proteomics  1.219523e-07\n",
      "1   MAPK3_proteomics  1.020660e-06\n",
      "2  PIK3CD_proteomics  1.030815e-03\n",
      "3    GRB2_proteomics  1.089951e-03\n",
      "4  PIK3C3_proteomics  1.225949e-03\n",
      "5    USP7_proteomics  4.017792e-03\n",
      "6  MAP2K1_proteomics  4.138456e-03\n",
      "7  PIK3R1_proteomics  4.282296e-03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.219523e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>1.020660e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>1.030815e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>1.089951e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>1.225949e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>4.017792e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>4.138456e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>4.282296e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>1.260293e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>1.455578e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>2.487273e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>3.212345e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>4.010871e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>5.478559e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>5.604258e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>7.197879e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>7.511173e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>9.521641e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>1.040103e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.167020e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>1.184655e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>1.439290e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>1.655635e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>2.195053e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.242562e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>2.562153e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>2.876721e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>3.018625e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>3.746728e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>4.531770e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>4.749988e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>5.493659e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>5.791747e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>6.343051e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>6.673038e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>6.682500e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>7.879877e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>7.974795e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>8.049218e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>8.663143e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.128906e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>9.387589e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics   Gbm_P_Value\n",
       "0       PTEN_proteomics  1.219523e-07\n",
       "1      MAPK3_proteomics  1.020660e-06\n",
       "2     PIK3CD_proteomics  1.030815e-03\n",
       "3       GRB2_proteomics  1.089951e-03\n",
       "4     PIK3C3_proteomics  1.225949e-03\n",
       "5       USP7_proteomics  4.017792e-03\n",
       "6     MAP2K1_proteomics  4.138456e-03\n",
       "7     PIK3R1_proteomics  4.282296e-03\n",
       "8      MAST2_proteomics  1.260293e-02\n",
       "9     INPP4B_proteomics  1.455578e-02\n",
       "10  SLC9A3R1_proteomics  2.487273e-02\n",
       "11      TSC2_proteomics  3.212345e-02\n",
       "12      EGFR_proteomics  4.010871e-02\n",
       "13     USP13_proteomics  5.478559e-02\n",
       "14   CSNK2A1_proteomics  5.604258e-02\n",
       "15     MAPK1_proteomics  7.197879e-02\n",
       "16      KRAS_proteomics  7.511173e-02\n",
       "17      PDK1_proteomics  9.521641e-02\n",
       "18    PIK3CB_proteomics  1.040103e-01\n",
       "19       MVP_proteomics  1.167020e-01\n",
       "20      XIAP_proteomics  1.184655e-01\n",
       "21      RAF1_proteomics  1.439290e-01\n",
       "22      AKT1_proteomics  1.655635e-01\n",
       "23      SOS1_proteomics  2.195053e-01\n",
       "24     ROCK1_proteomics  2.242562e-01\n",
       "25     MAGI3_proteomics  2.562153e-01\n",
       "26    PIK3CA_proteomics  2.876721e-01\n",
       "27     GSK3B_proteomics  3.018625e-01\n",
       "28      GAB1_proteomics  3.746728e-01\n",
       "29   CSNK2A2_proteomics  4.531770e-01\n",
       "30       BAD_proteomics  4.749988e-01\n",
       "31      SHC1_proteomics  5.493659e-01\n",
       "32     PREX2_proteomics  5.791747e-01\n",
       "33    PIK3R2_proteomics  6.343051e-01\n",
       "34      PTK2_proteomics  6.673038e-01\n",
       "35      TP53_proteomics  6.682500e-01\n",
       "36      MTOR_proteomics  7.879877e-01\n",
       "37     FOXO1_proteomics  7.974795e-01\n",
       "38     NEDD4_proteomics  8.049218e-01\n",
       "39      MDM2_proteomics  8.663143e-01\n",
       "40    PDGFRB_proteomics  9.128906e-01\n",
       "41     CCND1_proteomics  9.387589e-01\n",
       "42    MTORC2_proteomics           NaN\n",
       "43   MAP2K2 _proteomics           NaN\n",
       "44    PIK3R3_proteomics           NaN\n",
       "45      CAS9_proteomics           NaN"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "g_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if g_sig is not None:\n",
    "    g_sig_list = list(g_sig.Comparison)\n",
    "else: \n",
    "    l_sig_list = None\n",
    "print('significant pvals: \\n',g_sig)\n",
    "\n",
    "# Get all pvals\n",
    "g = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "g_pval = g.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Gbm_P_Value'})\n",
    "#g_pval = g_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "g_pval.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.050953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>-0.151642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.188390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>-0.079598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>-0.083073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.153123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.015429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>-0.080901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>-0.169565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.139061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.087843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>-0.127685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>-0.123874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>-0.256630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>-1.001035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>-0.028576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>-0.154035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.289788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.080280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>-0.091412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.592389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>0.665105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>-0.136083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.341829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>0.186256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.105746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.474151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>0.126064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.312146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>-0.093309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>-0.111700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.511726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>-0.035604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>0.185622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>0.034435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.492099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>0.206357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>-0.102838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.042720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>-0.199948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>-0.088967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.102518</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Gbm_Median\n",
       "0      MAGI3_proteomics    0.050953\n",
       "1      PREX2_proteomics   -0.151642\n",
       "2     PIK3R1_proteomics    0.188390\n",
       "3      NEDD4_proteomics   -0.079598\n",
       "4      FOXO1_proteomics   -0.083073\n",
       "5       SHC1_proteomics    0.153123\n",
       "6     PIK3R2_proteomics    0.015429\n",
       "7       MTOR_proteomics   -0.080901\n",
       "8       AKT1_proteomics   -0.169565\n",
       "9     MTORC2_proteomics         NaN\n",
       "10   MAP2K2 _proteomics         NaN\n",
       "11     MAST2_proteomics    0.139061\n",
       "12       BAD_proteomics    0.087843\n",
       "13      MDM2_proteomics   -0.127685\n",
       "14   CSNK2A1_proteomics   -0.123874\n",
       "15      TP53_proteomics   -0.256630\n",
       "16      EGFR_proteomics   -1.001035\n",
       "17    PDGFRB_proteomics   -0.028576\n",
       "18      USP7_proteomics   -0.154035\n",
       "19  SLC9A3R1_proteomics    0.289788\n",
       "20     ROCK1_proteomics    0.080280\n",
       "21      RAF1_proteomics   -0.091412\n",
       "22    MAP2K1_proteomics    0.592389\n",
       "23     MAPK3_proteomics    0.665105\n",
       "24    PIK3R3_proteomics         NaN\n",
       "25      GAB1_proteomics   -0.136083\n",
       "26    INPP4B_proteomics    0.341829\n",
       "27      KRAS_proteomics    0.186256\n",
       "28    PIK3CA_proteomics    0.105746\n",
       "29       MVP_proteomics    0.474151\n",
       "30     CCND1_proteomics    0.126064\n",
       "31      GRB2_proteomics    0.312146\n",
       "32      PTK2_proteomics   -0.093309\n",
       "33     USP13_proteomics   -0.111700\n",
       "34      PTEN_proteomics    0.511726\n",
       "35     GSK3B_proteomics   -0.035604\n",
       "36      TSC2_proteomics    0.185622\n",
       "37      SOS1_proteomics    0.034435\n",
       "38    PIK3CD_proteomics    0.492099\n",
       "39     MAPK1_proteomics    0.206357\n",
       "40      CAS9_proteomics         NaN\n",
       "41   CSNK2A2_proteomics   -0.102838\n",
       "42    PIK3CB_proteomics    0.042720\n",
       "43      PDK1_proteomics   -0.199948\n",
       "44      XIAP_proteomics   -0.088967\n",
       "45    PIK3C3_proteomics    0.102518"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "gbm_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif_gbm = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    gbm_d[prot+'_proteomics'] = dif_gbm\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif_gbm)\n",
    "\n",
    "gbm_df = pd.DataFrame.from_dict(gbm_d, orient='index', columns=['Gbm_Median'])\n",
    "gbm_df = gbm_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "gbm_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.219523e-07</td>\n",
       "      <td>0.511726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>1.020660e-06</td>\n",
       "      <td>0.665105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>1.030815e-03</td>\n",
       "      <td>0.492099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>1.089951e-03</td>\n",
       "      <td>0.312146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>1.225949e-03</td>\n",
       "      <td>0.102518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>4.017792e-03</td>\n",
       "      <td>-0.154035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>4.138456e-03</td>\n",
       "      <td>0.592389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>4.282296e-03</td>\n",
       "      <td>0.188390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>1.260293e-02</td>\n",
       "      <td>0.139061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>1.455578e-02</td>\n",
       "      <td>0.341829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>2.487273e-02</td>\n",
       "      <td>0.289788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>3.212345e-02</td>\n",
       "      <td>0.185622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>4.010871e-02</td>\n",
       "      <td>-1.001035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>5.478559e-02</td>\n",
       "      <td>-0.111700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>5.604258e-02</td>\n",
       "      <td>-0.123874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>7.197879e-02</td>\n",
       "      <td>0.206357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>7.511173e-02</td>\n",
       "      <td>0.186256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>9.521641e-02</td>\n",
       "      <td>-0.199948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>1.040103e-01</td>\n",
       "      <td>0.042720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.167020e-01</td>\n",
       "      <td>0.474151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>1.184655e-01</td>\n",
       "      <td>-0.088967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>1.439290e-01</td>\n",
       "      <td>-0.091412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>1.655635e-01</td>\n",
       "      <td>-0.169565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>2.195053e-01</td>\n",
       "      <td>0.034435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.242562e-01</td>\n",
       "      <td>0.080280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>2.562153e-01</td>\n",
       "      <td>0.050953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>2.876721e-01</td>\n",
       "      <td>0.105746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>3.018625e-01</td>\n",
       "      <td>-0.035604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>3.746728e-01</td>\n",
       "      <td>-0.136083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>4.531770e-01</td>\n",
       "      <td>-0.102838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>4.749988e-01</td>\n",
       "      <td>0.087843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>5.493659e-01</td>\n",
       "      <td>0.153123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>5.791747e-01</td>\n",
       "      <td>-0.151642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>6.343051e-01</td>\n",
       "      <td>0.015429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>6.673038e-01</td>\n",
       "      <td>-0.093309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>6.682500e-01</td>\n",
       "      <td>-0.256630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>7.879877e-01</td>\n",
       "      <td>-0.080901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>7.974795e-01</td>\n",
       "      <td>-0.083073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>8.049218e-01</td>\n",
       "      <td>-0.079598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>8.663143e-01</td>\n",
       "      <td>-0.127685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.128906e-01</td>\n",
       "      <td>-0.028576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>9.387589e-01</td>\n",
       "      <td>0.126064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics   Gbm_P_Value  Gbm_Median\n",
       "0       PTEN_proteomics  1.219523e-07    0.511726\n",
       "1      MAPK3_proteomics  1.020660e-06    0.665105\n",
       "2     PIK3CD_proteomics  1.030815e-03    0.492099\n",
       "3       GRB2_proteomics  1.089951e-03    0.312146\n",
       "4     PIK3C3_proteomics  1.225949e-03    0.102518\n",
       "5       USP7_proteomics  4.017792e-03   -0.154035\n",
       "6     MAP2K1_proteomics  4.138456e-03    0.592389\n",
       "7     PIK3R1_proteomics  4.282296e-03    0.188390\n",
       "8      MAST2_proteomics  1.260293e-02    0.139061\n",
       "9     INPP4B_proteomics  1.455578e-02    0.341829\n",
       "10  SLC9A3R1_proteomics  2.487273e-02    0.289788\n",
       "11      TSC2_proteomics  3.212345e-02    0.185622\n",
       "12      EGFR_proteomics  4.010871e-02   -1.001035\n",
       "13     USP13_proteomics  5.478559e-02   -0.111700\n",
       "14   CSNK2A1_proteomics  5.604258e-02   -0.123874\n",
       "15     MAPK1_proteomics  7.197879e-02    0.206357\n",
       "16      KRAS_proteomics  7.511173e-02    0.186256\n",
       "17      PDK1_proteomics  9.521641e-02   -0.199948\n",
       "18    PIK3CB_proteomics  1.040103e-01    0.042720\n",
       "19       MVP_proteomics  1.167020e-01    0.474151\n",
       "20      XIAP_proteomics  1.184655e-01   -0.088967\n",
       "21      RAF1_proteomics  1.439290e-01   -0.091412\n",
       "22      AKT1_proteomics  1.655635e-01   -0.169565\n",
       "23      SOS1_proteomics  2.195053e-01    0.034435\n",
       "24     ROCK1_proteomics  2.242562e-01    0.080280\n",
       "25     MAGI3_proteomics  2.562153e-01    0.050953\n",
       "26    PIK3CA_proteomics  2.876721e-01    0.105746\n",
       "27     GSK3B_proteomics  3.018625e-01   -0.035604\n",
       "28      GAB1_proteomics  3.746728e-01   -0.136083\n",
       "29   CSNK2A2_proteomics  4.531770e-01   -0.102838\n",
       "30       BAD_proteomics  4.749988e-01    0.087843\n",
       "31      SHC1_proteomics  5.493659e-01    0.153123\n",
       "32     PREX2_proteomics  5.791747e-01   -0.151642\n",
       "33    PIK3R2_proteomics  6.343051e-01    0.015429\n",
       "34      PTK2_proteomics  6.673038e-01   -0.093309\n",
       "35      TP53_proteomics  6.682500e-01   -0.256630\n",
       "36      MTOR_proteomics  7.879877e-01   -0.080901\n",
       "37     FOXO1_proteomics  7.974795e-01   -0.083073\n",
       "38     NEDD4_proteomics  8.049218e-01   -0.079598\n",
       "39      MDM2_proteomics  8.663143e-01   -0.127685\n",
       "40    PDGFRB_proteomics  9.128906e-01   -0.028576\n",
       "41     CCND1_proteomics  9.387589e-01    0.126064\n",
       "42    MTORC2_proteomics           NaN         NaN\n",
       "43   MAP2K2 _proteomics           NaN         NaN\n",
       "44    PIK3R3_proteomics           NaN         NaN\n",
       "45      CAS9_proteomics           NaN         NaN"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_merged = g_pval.merge(gbm_df, on='Proteomics',how='inner')\n",
    "g_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CAS9, MAP2K2 , MDM2, MTORC2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 69 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>MAGI3_proteomics</th>\n",
       "      <th>PREX2_proteomics</th>\n",
       "      <th>PIK3R1_proteomics</th>\n",
       "      <th>NEDD4_proteomics</th>\n",
       "      <th>FOXO1_proteomics</th>\n",
       "      <th>SHC1_proteomics</th>\n",
       "      <th>PIK3R2_proteomics</th>\n",
       "      <th>MTOR_proteomics</th>\n",
       "      <th>AKT1_proteomics</th>\n",
       "      <th>MTORC2_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>SOS1_proteomics</th>\n",
       "      <th>PIK3CD_proteomics</th>\n",
       "      <th>MAPK1_proteomics</th>\n",
       "      <th>CAS9_proteomics</th>\n",
       "      <th>CSNK2A2_proteomics</th>\n",
       "      <th>PIK3CB_proteomics</th>\n",
       "      <th>PDK1_proteomics</th>\n",
       "      <th>XIAP_proteomics</th>\n",
       "      <th>PIK3C3_proteomics</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00006</th>\n",
       "      <td>-0.3070</td>\n",
       "      <td>0.5570</td>\n",
       "      <td>0.481</td>\n",
       "      <td>-0.8590</td>\n",
       "      <td>-0.566</td>\n",
       "      <td>-0.1250</td>\n",
       "      <td>0.0540</td>\n",
       "      <td>0.0659</td>\n",
       "      <td>-0.2550</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3670</td>\n",
       "      <td>-0.25200</td>\n",
       "      <td>-0.2840</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2670</td>\n",
       "      <td>-0.0761</td>\n",
       "      <td>1.090</td>\n",
       "      <td>-0.3060</td>\n",
       "      <td>-0.41600</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00032</th>\n",
       "      <td>0.4150</td>\n",
       "      <td>-0.0581</td>\n",
       "      <td>0.266</td>\n",
       "      <td>-0.7620</td>\n",
       "      <td>-0.457</td>\n",
       "      <td>-0.0194</td>\n",
       "      <td>0.0502</td>\n",
       "      <td>-0.0120</td>\n",
       "      <td>-0.0722</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.1120</td>\n",
       "      <td>0.45700</td>\n",
       "      <td>-0.1380</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2410</td>\n",
       "      <td>0.3620</td>\n",
       "      <td>0.607</td>\n",
       "      <td>0.2220</td>\n",
       "      <td>-0.00815</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00098</th>\n",
       "      <td>-0.2160</td>\n",
       "      <td>0.9430</td>\n",
       "      <td>-0.612</td>\n",
       "      <td>-0.0684</td>\n",
       "      <td>-2.980</td>\n",
       "      <td>0.1030</td>\n",
       "      <td>0.0690</td>\n",
       "      <td>-0.2120</td>\n",
       "      <td>-0.6410</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2140</td>\n",
       "      <td>0.00543</td>\n",
       "      <td>-0.1150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0431</td>\n",
       "      <td>0.3730</td>\n",
       "      <td>0.742</td>\n",
       "      <td>0.6050</td>\n",
       "      <td>0.04510</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00137</th>\n",
       "      <td>-0.0705</td>\n",
       "      <td>1.2400</td>\n",
       "      <td>-1.910</td>\n",
       "      <td>-1.2500</td>\n",
       "      <td>-0.621</td>\n",
       "      <td>-0.1610</td>\n",
       "      <td>0.1550</td>\n",
       "      <td>0.0597</td>\n",
       "      <td>0.1430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2220</td>\n",
       "      <td>-1.05000</td>\n",
       "      <td>-0.2800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1050</td>\n",
       "      <td>-0.7650</td>\n",
       "      <td>0.273</td>\n",
       "      <td>-0.0210</td>\n",
       "      <td>-0.59600</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00139</th>\n",
       "      <td>0.0858</td>\n",
       "      <td>0.2140</td>\n",
       "      <td>-0.496</td>\n",
       "      <td>-0.9890</td>\n",
       "      <td>0.230</td>\n",
       "      <td>-0.4690</td>\n",
       "      <td>0.6930</td>\n",
       "      <td>0.2910</td>\n",
       "      <td>-0.7250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0330</td>\n",
       "      <td>-0.20400</td>\n",
       "      <td>-0.4110</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1470</td>\n",
       "      <td>-0.1750</td>\n",
       "      <td>-0.282</td>\n",
       "      <td>-0.3040</td>\n",
       "      <td>-0.07510</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>-0.0286</td>\n",
       "      <td>0.0739</td>\n",
       "      <td>0.218</td>\n",
       "      <td>-0.9740</td>\n",
       "      <td>-0.822</td>\n",
       "      <td>-0.4970</td>\n",
       "      <td>0.1080</td>\n",
       "      <td>-0.2090</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2790</td>\n",
       "      <td>-0.12900</td>\n",
       "      <td>-0.0859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2410</td>\n",
       "      <td>0.0470</td>\n",
       "      <td>0.725</td>\n",
       "      <td>-0.4500</td>\n",
       "      <td>0.36700</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>0.4750</td>\n",
       "      <td>-0.7160</td>\n",
       "      <td>0.117</td>\n",
       "      <td>-0.4740</td>\n",
       "      <td>-1.610</td>\n",
       "      <td>0.8070</td>\n",
       "      <td>-0.0944</td>\n",
       "      <td>0.1450</td>\n",
       "      <td>0.1720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2130</td>\n",
       "      <td>0.20500</td>\n",
       "      <td>-0.5260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1530</td>\n",
       "      <td>1.1700</td>\n",
       "      <td>1.190</td>\n",
       "      <td>-0.5480</td>\n",
       "      <td>-0.28300</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>0.2760</td>\n",
       "      <td>0.4530</td>\n",
       "      <td>-0.333</td>\n",
       "      <td>-0.8110</td>\n",
       "      <td>-0.567</td>\n",
       "      <td>-0.1530</td>\n",
       "      <td>-0.1340</td>\n",
       "      <td>-0.1180</td>\n",
       "      <td>-0.2300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.1140</td>\n",
       "      <td>0.74600</td>\n",
       "      <td>-0.7120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1240</td>\n",
       "      <td>-0.0841</td>\n",
       "      <td>1.110</td>\n",
       "      <td>0.0726</td>\n",
       "      <td>0.18900</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>0.0052</td>\n",
       "      <td>0.6040</td>\n",
       "      <td>-0.446</td>\n",
       "      <td>0.1600</td>\n",
       "      <td>-0.122</td>\n",
       "      <td>0.2510</td>\n",
       "      <td>-0.4200</td>\n",
       "      <td>0.2800</td>\n",
       "      <td>-0.2150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0937</td>\n",
       "      <td>0.69100</td>\n",
       "      <td>0.5820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0649</td>\n",
       "      <td>0.2820</td>\n",
       "      <td>0.503</td>\n",
       "      <td>-0.3440</td>\n",
       "      <td>0.28500</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>-0.0173</td>\n",
       "      <td>-0.5080</td>\n",
       "      <td>0.244</td>\n",
       "      <td>0.4280</td>\n",
       "      <td>0.273</td>\n",
       "      <td>-0.0868</td>\n",
       "      <td>0.1840</td>\n",
       "      <td>0.2350</td>\n",
       "      <td>0.2330</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2000</td>\n",
       "      <td>0.58300</td>\n",
       "      <td>0.2680</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0830</td>\n",
       "      <td>0.1850</td>\n",
       "      <td>-0.050</td>\n",
       "      <td>-0.1370</td>\n",
       "      <td>0.26800</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows Ã— 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        MAGI3_proteomics  PREX2_proteomics  PIK3R1_proteomics  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00006            -0.3070            0.5570              0.481   \n",
       "C3L-00032             0.4150           -0.0581              0.266   \n",
       "C3L-00098            -0.2160            0.9430             -0.612   \n",
       "C3L-00137            -0.0705            1.2400             -1.910   \n",
       "C3L-00139             0.0858            0.2140             -0.496   \n",
       "...                      ...               ...                ...   \n",
       "C3N-01520            -0.0286            0.0739              0.218   \n",
       "C3N-01521             0.4750           -0.7160              0.117   \n",
       "C3N-01537             0.2760            0.4530             -0.333   \n",
       "C3N-01802             0.0052            0.6040             -0.446   \n",
       "C3N-01825            -0.0173           -0.5080              0.244   \n",
       "\n",
       "Name        NEDD4_proteomics  FOXO1_proteomics  SHC1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00006            -0.8590            -0.566          -0.1250   \n",
       "C3L-00032            -0.7620            -0.457          -0.0194   \n",
       "C3L-00098            -0.0684            -2.980           0.1030   \n",
       "C3L-00137            -1.2500            -0.621          -0.1610   \n",
       "C3L-00139            -0.9890             0.230          -0.4690   \n",
       "...                      ...               ...              ...   \n",
       "C3N-01520            -0.9740            -0.822          -0.4970   \n",
       "C3N-01521            -0.4740            -1.610           0.8070   \n",
       "C3N-01537            -0.8110            -0.567          -0.1530   \n",
       "C3N-01802             0.1600            -0.122           0.2510   \n",
       "C3N-01825             0.4280             0.273          -0.0868   \n",
       "\n",
       "Name        PIK3R2_proteomics  MTOR_proteomics  AKT1_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00006              0.0540           0.0659          -0.2550   \n",
       "C3L-00032              0.0502          -0.0120          -0.0722   \n",
       "C3L-00098              0.0690          -0.2120          -0.6410   \n",
       "C3L-00137              0.1550           0.0597           0.1430   \n",
       "C3L-00139              0.6930           0.2910          -0.7250   \n",
       "...                       ...              ...              ...   \n",
       "C3N-01520              0.1080          -0.2090           0.0217   \n",
       "C3N-01521             -0.0944           0.1450           0.1720   \n",
       "C3N-01537             -0.1340          -0.1180          -0.2300   \n",
       "C3N-01802             -0.4200           0.2800          -0.2150   \n",
       "C3N-01825              0.1840           0.2350           0.2330   \n",
       "\n",
       "Name        MTORC2_proteomics  ...  SOS1_proteomics  PIK3CD_proteomics  \\\n",
       "Patient_ID                     ...                                       \n",
       "C3L-00006                 NaN  ...          -0.3670           -0.25200   \n",
       "C3L-00032                 NaN  ...          -0.1120            0.45700   \n",
       "C3L-00098                 NaN  ...           0.2140            0.00543   \n",
       "C3L-00137                 NaN  ...          -0.2220           -1.05000   \n",
       "C3L-00139                 NaN  ...          -0.0330           -0.20400   \n",
       "...                       ...  ...              ...                ...   \n",
       "C3N-01520                 NaN  ...          -0.2790           -0.12900   \n",
       "C3N-01521                 NaN  ...          -0.2130            0.20500   \n",
       "C3N-01537                 NaN  ...          -0.1140            0.74600   \n",
       "C3N-01802                 NaN  ...           0.0937            0.69100   \n",
       "C3N-01825                 NaN  ...           0.2000            0.58300   \n",
       "\n",
       "Name        MAPK1_proteomics  CAS9_proteomics  CSNK2A2_proteomics  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00006            -0.2840              NaN              0.2670   \n",
       "C3L-00032            -0.1380              NaN             -0.2410   \n",
       "C3L-00098            -0.1150              NaN             -0.0431   \n",
       "C3L-00137            -0.2800              NaN              0.1050   \n",
       "C3L-00139            -0.4110              NaN              0.1470   \n",
       "...                      ...              ...                 ...   \n",
       "C3N-01520            -0.0859              NaN             -0.2410   \n",
       "C3N-01521            -0.5260              NaN              0.1530   \n",
       "C3N-01537            -0.7120              NaN              0.1240   \n",
       "C3N-01802             0.5820              NaN              0.0649   \n",
       "C3N-01825             0.2680              NaN             -0.0830   \n",
       "\n",
       "Name        PIK3CB_proteomics  PDK1_proteomics  XIAP_proteomics  \\\n",
       "Patient_ID                                                        \n",
       "C3L-00006             -0.0761            1.090          -0.3060   \n",
       "C3L-00032              0.3620            0.607           0.2220   \n",
       "C3L-00098              0.3730            0.742           0.6050   \n",
       "C3L-00137             -0.7650            0.273          -0.0210   \n",
       "C3L-00139             -0.1750           -0.282          -0.3040   \n",
       "...                       ...              ...              ...   \n",
       "C3N-01520              0.0470            0.725          -0.4500   \n",
       "C3N-01521              1.1700            1.190          -0.5480   \n",
       "C3N-01537             -0.0841            1.110           0.0726   \n",
       "C3N-01802              0.2820            0.503          -0.3440   \n",
       "C3N-01825              0.1850           -0.050          -0.1370   \n",
       "\n",
       "Name        PIK3C3_proteomics        Mutation  \n",
       "Patient_ID                                     \n",
       "C3L-00006            -0.41600      Truncation  \n",
       "C3L-00032            -0.00815      Truncation  \n",
       "C3L-00098             0.04510  Wildtype_Tumor  \n",
       "C3L-00137            -0.59600      Truncation  \n",
       "C3L-00139            -0.07510  Wildtype_Tumor  \n",
       "...                       ...             ...  \n",
       "C3N-01520             0.36700      Truncation  \n",
       "C3N-01521            -0.28300  Wildtype_Tumor  \n",
       "C3N-01537             0.18900  Wildtype_Tumor  \n",
       "C3N-01802             0.28500  Wildtype_Tumor  \n",
       "C3N-01825             0.26800  Wildtype_Tumor  \n",
       "\n",
       "[68 rows x 47 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Endo ttest\n",
    "\n",
    "mut_type_gbm = endo.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = endo.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "trunc_wt = merged[get]\n",
    "trunc_wt['Mutation'] = np.where(\n",
    "                trunc_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "trunc_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "            Comparison   P_Value\n",
      "0     PTEN_proteomics  0.000640\n",
      "1     TP53_proteomics  0.001246\n",
      "2   PIK3CA_proteomics  0.001761\n",
      "3  CSNK2A1_proteomics  0.002364\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.001246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.001761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.002364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.006128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.018340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.035238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.036592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.045489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.053078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>0.058960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>0.081623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>0.091748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.135943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>0.138674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.160116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.182172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.205099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.228439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>0.254287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>0.266253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>0.272111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.391469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>0.430651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.454039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>0.463516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.489186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>0.497471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>0.543094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.572500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.622263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.675793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.681581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.690012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.724325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.732837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.793385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.802934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.913864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.933397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>0.955021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  En_P_Value\n",
       "0       PTEN_proteomics    0.000640\n",
       "1       TP53_proteomics    0.001246\n",
       "2     PIK3CA_proteomics    0.001761\n",
       "3    CSNK2A1_proteomics    0.002364\n",
       "4       EGFR_proteomics    0.006128\n",
       "5       SHC1_proteomics    0.018340\n",
       "6    CSNK2A2_proteomics    0.035238\n",
       "7     PIK3C3_proteomics    0.036592\n",
       "8     PIK3CD_proteomics    0.045489\n",
       "9      PREX2_proteomics    0.053078\n",
       "10     MAPK1_proteomics    0.058960\n",
       "11      GAB1_proteomics    0.081623\n",
       "12      KRAS_proteomics    0.091748\n",
       "13    PIK3R1_proteomics    0.135943\n",
       "14     FOXO1_proteomics    0.138674\n",
       "15      GRB2_proteomics    0.160116\n",
       "16  SLC9A3R1_proteomics    0.182172\n",
       "17     NEDD4_proteomics    0.205099\n",
       "18     MAST2_proteomics    0.228439\n",
       "19     CCND1_proteomics    0.254287\n",
       "20      MTOR_proteomics    0.266253\n",
       "21     MAPK3_proteomics    0.272111\n",
       "22    PDGFRB_proteomics    0.391469\n",
       "23      SOS1_proteomics    0.430651\n",
       "24      PTK2_proteomics    0.454039\n",
       "25      RAF1_proteomics    0.463516\n",
       "26    INPP4B_proteomics    0.489186\n",
       "27     GSK3B_proteomics    0.497471\n",
       "28      PDK1_proteomics    0.543094\n",
       "29      USP7_proteomics    0.572500\n",
       "30       MVP_proteomics    0.622263\n",
       "31    MAP2K1_proteomics    0.675793\n",
       "32     MAGI3_proteomics    0.681581\n",
       "33     ROCK1_proteomics    0.690012\n",
       "34    PIK3CB_proteomics    0.724325\n",
       "35      AKT1_proteomics    0.732837\n",
       "36     USP13_proteomics    0.793385\n",
       "37       BAD_proteomics    0.802934\n",
       "38      XIAP_proteomics    0.913864\n",
       "39    PIK3R2_proteomics    0.933397\n",
       "40      TSC2_proteomics    0.955021\n",
       "41    MTORC2_proteomics         NaN\n",
       "42   MAP2K2 _proteomics         NaN\n",
       "43      MDM2_proteomics         NaN\n",
       "44    PIK3R3_proteomics         NaN\n",
       "45      CAS9_proteomics         NaN"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(trunc_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "e_sig = u.wrap_ttest(trunc_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if e_sig is not None:\n",
    "    e_sig_list = list(e_sig.Comparison)\n",
    "else: \n",
    "    e_sig_list = None\n",
    "print('significant pvals: \\n',e_sig)\n",
    "\n",
    "# Get all pvals\n",
    "e_pval = u.wrap_ttest(trunc_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "e_pval = e_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'En_P_Value'})\n",
    "#e_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.00430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>-0.45800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>-0.33990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.04500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>-0.09000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.24980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.01500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>-0.03380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.05160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>-0.07090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.11250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.10412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.80260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>-0.52600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>-0.17100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>-0.03834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>-0.33300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>0.00010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.09190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>-0.08200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>-0.28500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>-0.16320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>-0.19180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.35300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.11890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>-0.05260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.26400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.02340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>-0.03000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.74660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>0.02178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>-0.02550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>0.14100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.18243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>-0.13670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.15970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>-0.03370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>0.07800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>-0.09772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.13160</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  En_Median\n",
       "0      MAGI3_proteomics    0.00430\n",
       "1      PREX2_proteomics   -0.45800\n",
       "2     PIK3R1_proteomics   -0.33990\n",
       "3      NEDD4_proteomics    0.04500\n",
       "4      FOXO1_proteomics   -0.09000\n",
       "5       SHC1_proteomics    0.24980\n",
       "6     PIK3R2_proteomics    0.01500\n",
       "7       MTOR_proteomics   -0.03380\n",
       "8       AKT1_proteomics    0.05160\n",
       "9     MTORC2_proteomics        NaN\n",
       "10   MAP2K2 _proteomics        NaN\n",
       "11     MAST2_proteomics   -0.07090\n",
       "12       BAD_proteomics    0.11250\n",
       "13      MDM2_proteomics        NaN\n",
       "14   CSNK2A1_proteomics    0.10412\n",
       "15      TP53_proteomics    0.80260\n",
       "16      EGFR_proteomics   -0.52600\n",
       "17    PDGFRB_proteomics   -0.17100\n",
       "18      USP7_proteomics   -0.03834\n",
       "19  SLC9A3R1_proteomics   -0.33300\n",
       "20     ROCK1_proteomics   -0.08220\n",
       "21      RAF1_proteomics    0.00010\n",
       "22    MAP2K1_proteomics    0.09190\n",
       "23     MAPK3_proteomics   -0.08200\n",
       "24    PIK3R3_proteomics        NaN\n",
       "25      GAB1_proteomics   -0.28500\n",
       "26    INPP4B_proteomics   -0.16320\n",
       "27      KRAS_proteomics   -0.19180\n",
       "28    PIK3CA_proteomics    0.35300\n",
       "29       MVP_proteomics    0.11890\n",
       "30     CCND1_proteomics   -0.05260\n",
       "31      GRB2_proteomics    0.26400\n",
       "32      PTK2_proteomics    0.02340\n",
       "33     USP13_proteomics   -0.03000\n",
       "34      PTEN_proteomics    0.74660\n",
       "35     GSK3B_proteomics    0.02178\n",
       "36      TSC2_proteomics   -0.02550\n",
       "37      SOS1_proteomics    0.14100\n",
       "38    PIK3CD_proteomics    0.18243\n",
       "39     MAPK1_proteomics   -0.13670\n",
       "40      CAS9_proteomics        NaN\n",
       "41   CSNK2A2_proteomics    0.15970\n",
       "42    PIK3CB_proteomics   -0.03370\n",
       "43      PDK1_proteomics    0.07800\n",
       "44      XIAP_proteomics   -0.09772\n",
       "45    PIK3C3_proteomics    0.13160"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = trunc_wt[trunc_wt.Mutation == \"Truncation\"]\n",
    "wt = trunc_wt[trunc_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "trunc_med = t.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "en_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - trunc_med[prot+'_proteomics']\n",
    "    en_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', trunc_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "en_df = pd.DataFrame.from_dict(en_d, orient='index', columns=['En_Median'])\n",
    "en_df = en_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#en_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>0.74660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.001246</td>\n",
       "      <td>0.80260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.001761</td>\n",
       "      <td>0.35300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.002364</td>\n",
       "      <td>0.10412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.006128</td>\n",
       "      <td>-0.52600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.018340</td>\n",
       "      <td>0.24980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.035238</td>\n",
       "      <td>0.15970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.036592</td>\n",
       "      <td>0.13160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.045489</td>\n",
       "      <td>0.18243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.053078</td>\n",
       "      <td>-0.45800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>0.058960</td>\n",
       "      <td>-0.13670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>0.081623</td>\n",
       "      <td>-0.28500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>0.091748</td>\n",
       "      <td>-0.19180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.135943</td>\n",
       "      <td>-0.33990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>0.138674</td>\n",
       "      <td>-0.09000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.160116</td>\n",
       "      <td>0.26400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.182172</td>\n",
       "      <td>-0.33300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.205099</td>\n",
       "      <td>0.04500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.228439</td>\n",
       "      <td>-0.07090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>0.254287</td>\n",
       "      <td>-0.05260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>0.266253</td>\n",
       "      <td>-0.03380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>0.272111</td>\n",
       "      <td>-0.08200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.391469</td>\n",
       "      <td>-0.17100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>0.430651</td>\n",
       "      <td>0.14100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.454039</td>\n",
       "      <td>0.02340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>0.463516</td>\n",
       "      <td>0.00010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.489186</td>\n",
       "      <td>-0.16320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>0.497471</td>\n",
       "      <td>0.02178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>0.543094</td>\n",
       "      <td>0.07800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.572500</td>\n",
       "      <td>-0.03834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.622263</td>\n",
       "      <td>0.11890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.675793</td>\n",
       "      <td>0.09190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.681581</td>\n",
       "      <td>0.00430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.690012</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.724325</td>\n",
       "      <td>-0.03370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.732837</td>\n",
       "      <td>0.05160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.793385</td>\n",
       "      <td>-0.03000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.802934</td>\n",
       "      <td>0.11250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.913864</td>\n",
       "      <td>-0.09772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.933397</td>\n",
       "      <td>0.01500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>0.955021</td>\n",
       "      <td>-0.02550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  En_P_Value  En_Median\n",
       "0       PTEN_proteomics    0.000640    0.74660\n",
       "1       TP53_proteomics    0.001246    0.80260\n",
       "2     PIK3CA_proteomics    0.001761    0.35300\n",
       "3    CSNK2A1_proteomics    0.002364    0.10412\n",
       "4       EGFR_proteomics    0.006128   -0.52600\n",
       "5       SHC1_proteomics    0.018340    0.24980\n",
       "6    CSNK2A2_proteomics    0.035238    0.15970\n",
       "7     PIK3C3_proteomics    0.036592    0.13160\n",
       "8     PIK3CD_proteomics    0.045489    0.18243\n",
       "9      PREX2_proteomics    0.053078   -0.45800\n",
       "10     MAPK1_proteomics    0.058960   -0.13670\n",
       "11      GAB1_proteomics    0.081623   -0.28500\n",
       "12      KRAS_proteomics    0.091748   -0.19180\n",
       "13    PIK3R1_proteomics    0.135943   -0.33990\n",
       "14     FOXO1_proteomics    0.138674   -0.09000\n",
       "15      GRB2_proteomics    0.160116    0.26400\n",
       "16  SLC9A3R1_proteomics    0.182172   -0.33300\n",
       "17     NEDD4_proteomics    0.205099    0.04500\n",
       "18     MAST2_proteomics    0.228439   -0.07090\n",
       "19     CCND1_proteomics    0.254287   -0.05260\n",
       "20      MTOR_proteomics    0.266253   -0.03380\n",
       "21     MAPK3_proteomics    0.272111   -0.08200\n",
       "22    PDGFRB_proteomics    0.391469   -0.17100\n",
       "23      SOS1_proteomics    0.430651    0.14100\n",
       "24      PTK2_proteomics    0.454039    0.02340\n",
       "25      RAF1_proteomics    0.463516    0.00010\n",
       "26    INPP4B_proteomics    0.489186   -0.16320\n",
       "27     GSK3B_proteomics    0.497471    0.02178\n",
       "28      PDK1_proteomics    0.543094    0.07800\n",
       "29      USP7_proteomics    0.572500   -0.03834\n",
       "30       MVP_proteomics    0.622263    0.11890\n",
       "31    MAP2K1_proteomics    0.675793    0.09190\n",
       "32     MAGI3_proteomics    0.681581    0.00430\n",
       "33     ROCK1_proteomics    0.690012   -0.08220\n",
       "34    PIK3CB_proteomics    0.724325   -0.03370\n",
       "35      AKT1_proteomics    0.732837    0.05160\n",
       "36     USP13_proteomics    0.793385   -0.03000\n",
       "37       BAD_proteomics    0.802934    0.11250\n",
       "38      XIAP_proteomics    0.913864   -0.09772\n",
       "39    PIK3R2_proteomics    0.933397    0.01500\n",
       "40      TSC2_proteomics    0.955021   -0.02550\n",
       "41    MTORC2_proteomics         NaN        NaN\n",
       "42   MAP2K2 _proteomics         NaN        NaN\n",
       "43      MDM2_proteomics         NaN        NaN\n",
       "44    PIK3R3_proteomics         NaN        NaN\n",
       "45      CAS9_proteomics         NaN        NaN"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_merged = e_pval.merge(en_df, on='Proteomics',how='inner')\n",
    "e_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                            \r"
     ]
    }
   ],
   "source": [
    "#o = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CAS9, CCND1, MAP2K2 , MDM2, MTORC2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 8)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 8)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 4 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    46\n",
       "Deletion          20\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "gene = 'PTEN'\n",
    "\n",
    "mut_type_gbm = o.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = o.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = o.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "         Comparison   P_Value\n",
      "0  PTEN_proteomics  0.000909\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "o_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if o_sig is not None:\n",
    "    o_sig_list = list(o_sig.Comparison)\n",
    "else: \n",
    "    o_sig_list = None\n",
    "print('significant pvals: \\n',o_sig)\n",
    "\n",
    "# Get all pvals\n",
    "o = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "o_pval = o.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Ov_P_Value'})\n",
    "#o_pval = o_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "#o_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "ov_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    ov_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "o_df = pd.DataFrame.from_dict(ov_d, orient='index', columns=['Ov_Median'])\n",
    "o_df = o_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#o_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000909</td>\n",
       "      <td>0.169972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.005865</td>\n",
       "      <td>-0.240986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.008335</td>\n",
       "      <td>-0.199296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.050313</td>\n",
       "      <td>-0.276354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.060296</td>\n",
       "      <td>-0.505300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.080941</td>\n",
       "      <td>-0.229494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.086428</td>\n",
       "      <td>-0.184243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.150601</td>\n",
       "      <td>-0.133108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.164860</td>\n",
       "      <td>0.085809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>0.213423</td>\n",
       "      <td>-0.064073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.219977</td>\n",
       "      <td>-0.179425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.231014</td>\n",
       "      <td>-0.163327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>0.273504</td>\n",
       "      <td>0.035997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.337367</td>\n",
       "      <td>-0.001637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.353620</td>\n",
       "      <td>-0.370647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.429951</td>\n",
       "      <td>-0.145812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.444583</td>\n",
       "      <td>-0.104687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.472124</td>\n",
       "      <td>0.025625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.486184</td>\n",
       "      <td>0.026832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>0.506805</td>\n",
       "      <td>0.076732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.525069</td>\n",
       "      <td>0.017808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.530323</td>\n",
       "      <td>0.146323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>0.551643</td>\n",
       "      <td>0.019180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>0.569027</td>\n",
       "      <td>-0.040657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.580920</td>\n",
       "      <td>-0.015518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.587071</td>\n",
       "      <td>-0.193693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.682356</td>\n",
       "      <td>0.167967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.685027</td>\n",
       "      <td>0.010591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>0.740772</td>\n",
       "      <td>-0.003763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.838960</td>\n",
       "      <td>0.063388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>0.842428</td>\n",
       "      <td>-0.059507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>0.859802</td>\n",
       "      <td>0.027015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>0.876446</td>\n",
       "      <td>0.039263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>0.909874</td>\n",
       "      <td>-0.043522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.926598</td>\n",
       "      <td>0.154918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.937984</td>\n",
       "      <td>0.035308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.951727</td>\n",
       "      <td>-0.011741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>0.972748</td>\n",
       "      <td>-0.004340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.980145</td>\n",
       "      <td>-0.196586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.997559</td>\n",
       "      <td>-0.091044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Ov_P_Value  Ov_Median\n",
       "0       PTEN_proteomics    0.000909   0.169972\n",
       "1     PIK3R2_proteomics    0.005865  -0.240986\n",
       "2     PIK3CA_proteomics    0.008335  -0.199296\n",
       "3     PIK3CB_proteomics    0.050313  -0.276354\n",
       "4      PREX2_proteomics    0.060296  -0.505300\n",
       "5       PTK2_proteomics    0.080941  -0.229494\n",
       "6       GRB2_proteomics    0.086428  -0.184243\n",
       "7      MAGI3_proteomics    0.150601  -0.133108\n",
       "8      MAST2_proteomics    0.164860   0.085809\n",
       "9       MTOR_proteomics    0.213423  -0.064073\n",
       "10    PIK3R1_proteomics    0.219977  -0.179425\n",
       "11    PDGFRB_proteomics    0.231014  -0.163327\n",
       "12      KRAS_proteomics    0.273504   0.035997\n",
       "13     NEDD4_proteomics    0.337367  -0.001637\n",
       "14    INPP4B_proteomics    0.353620  -0.370647\n",
       "15      AKT1_proteomics    0.429951  -0.145812\n",
       "16    PIK3C3_proteomics    0.444583  -0.104687\n",
       "17    MAP2K1_proteomics    0.472124   0.025625\n",
       "18     ROCK1_proteomics    0.486184   0.026832\n",
       "19     GSK3B_proteomics    0.506805   0.076732\n",
       "20      USP7_proteomics    0.525069   0.017808\n",
       "21  SLC9A3R1_proteomics    0.530323   0.146323\n",
       "22      PDK1_proteomics    0.551643   0.019180\n",
       "23      SOS1_proteomics    0.569027  -0.040657\n",
       "24   CSNK2A1_proteomics    0.580920  -0.015518\n",
       "25       MVP_proteomics    0.587071  -0.193693\n",
       "26       BAD_proteomics    0.682356   0.167967\n",
       "27      XIAP_proteomics    0.685027   0.010591\n",
       "28     MAPK1_proteomics    0.740772  -0.003763\n",
       "29      EGFR_proteomics    0.838960   0.063388\n",
       "30     MAPK3_proteomics    0.842428  -0.059507\n",
       "31      TSC2_proteomics    0.859802   0.027015\n",
       "32     FOXO1_proteomics    0.876446   0.039263\n",
       "33      RAF1_proteomics    0.909874  -0.043522\n",
       "34     USP13_proteomics    0.926598   0.154918\n",
       "35    PIK3CD_proteomics    0.937984   0.035308\n",
       "36   CSNK2A2_proteomics    0.951727  -0.011741\n",
       "37      GAB1_proteomics    0.972748  -0.004340\n",
       "38      TP53_proteomics    0.980145  -0.196586\n",
       "39      SHC1_proteomics    0.997559  -0.091044\n",
       "40      CAS9_proteomics         NaN        NaN\n",
       "41     CCND1_proteomics         NaN        NaN\n",
       "42   MAP2K2 _proteomics         NaN        NaN\n",
       "43      MDM2_proteomics         NaN        NaN\n",
       "44    MTORC2_proteomics         NaN        NaN\n",
       "45    PIK3R3_proteomics         NaN        NaN"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "o_merged = o_pval.merge(o_df, on='Proteomics',how='inner')\n",
    "o_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Breast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CAS9, CSNK2A1, MAP2K2 , MDM2, MTORC2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 6)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    80\n",
       "Deletion          24\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = b.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = b.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = b.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "         Comparison   P_Value\n",
      "0  PTEN_proteomics  0.000004\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "b_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if b_sig is not None:\n",
    "    b_sig_list = list(b_sig.Comparison)\n",
    "else: \n",
    "    b_sig_list = None\n",
    "print('significant pvals: \\n',b_sig)\n",
    "\n",
    "# Get all pvals\n",
    "b_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "b_pval = b_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Brca_P_Value'})\n",
    "#b_pval = b_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#b_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "b_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    b_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "b_df = pd.DataFrame.from_dict(b_d, orient='index', columns=['Brca_Median'])\n",
    "b_df = b_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#b_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>1.00800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.014172</td>\n",
       "      <td>-0.42620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.016614</td>\n",
       "      <td>-0.58645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.061517</td>\n",
       "      <td>0.31470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.068427</td>\n",
       "      <td>-0.11100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>0.088200</td>\n",
       "      <td>-0.18225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>0.131515</td>\n",
       "      <td>0.21150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>0.151628</td>\n",
       "      <td>0.46115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.164475</td>\n",
       "      <td>1.03615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.213636</td>\n",
       "      <td>0.29705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.262209</td>\n",
       "      <td>-0.28100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>0.271869</td>\n",
       "      <td>-0.20785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.279670</td>\n",
       "      <td>0.85950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>0.285412</td>\n",
       "      <td>0.22335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.295970</td>\n",
       "      <td>0.37695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.325600</td>\n",
       "      <td>-0.28270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.353546</td>\n",
       "      <td>0.04900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.382956</td>\n",
       "      <td>-0.15150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.383804</td>\n",
       "      <td>-0.15040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.392146</td>\n",
       "      <td>-0.01540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.445080</td>\n",
       "      <td>-0.11490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.460310</td>\n",
       "      <td>0.27750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>0.464174</td>\n",
       "      <td>0.05670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.522034</td>\n",
       "      <td>0.37085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.529497</td>\n",
       "      <td>0.48570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.684989</td>\n",
       "      <td>0.00110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.714108</td>\n",
       "      <td>0.13185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>0.735044</td>\n",
       "      <td>-0.13505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>0.735666</td>\n",
       "      <td>0.23985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.757927</td>\n",
       "      <td>0.03045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>0.815146</td>\n",
       "      <td>-0.13505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.831541</td>\n",
       "      <td>-0.00695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.836701</td>\n",
       "      <td>-0.15015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.885732</td>\n",
       "      <td>0.02575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>0.896082</td>\n",
       "      <td>-0.30440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>0.912203</td>\n",
       "      <td>-0.14820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.921867</td>\n",
       "      <td>-0.01090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.929331</td>\n",
       "      <td>0.14305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.980902</td>\n",
       "      <td>-0.04105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>0.986037</td>\n",
       "      <td>-0.04200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Brca_P_Value  Brca_Median\n",
       "0       PTEN_proteomics      0.000004      1.00800\n",
       "1       SHC1_proteomics      0.014172     -0.42620\n",
       "2       EGFR_proteomics      0.016614     -0.58645\n",
       "3       GRB2_proteomics      0.061517      0.31470\n",
       "4       TP53_proteomics      0.068427     -0.11100\n",
       "5      GSK3B_proteomics      0.088200     -0.18225\n",
       "6       RAF1_proteomics      0.131515      0.21150\n",
       "7      CCND1_proteomics      0.151628      0.46115\n",
       "8     PIK3R1_proteomics      0.164475      1.03615\n",
       "9      MAST2_proteomics      0.213636      0.29705\n",
       "10   CSNK2A2_proteomics      0.262209     -0.28100\n",
       "11      TSC2_proteomics      0.271869     -0.20785\n",
       "12       BAD_proteomics      0.279670      0.85950\n",
       "13     MAPK3_proteomics      0.285412      0.22335\n",
       "14    INPP4B_proteomics      0.295970      0.37695\n",
       "15     USP13_proteomics      0.325600     -0.28270\n",
       "16      PTK2_proteomics      0.353546      0.04900\n",
       "17     MAGI3_proteomics      0.382956     -0.15150\n",
       "18      XIAP_proteomics      0.383804     -0.15040\n",
       "19    PIK3CB_proteomics      0.392146     -0.01540\n",
       "20    PIK3C3_proteomics      0.445080     -0.11490\n",
       "21    PIK3CA_proteomics      0.460310      0.27750\n",
       "22      KRAS_proteomics      0.464174      0.05670\n",
       "23  SLC9A3R1_proteomics      0.522034      0.37085\n",
       "24     NEDD4_proteomics      0.529497      0.48570\n",
       "25    PIK3R2_proteomics      0.684989      0.00110\n",
       "26      AKT1_proteomics      0.714108      0.13185\n",
       "27      MTOR_proteomics      0.735044     -0.13505\n",
       "28      GAB1_proteomics      0.735666      0.23985\n",
       "29    MAP2K1_proteomics      0.757927      0.03045\n",
       "30      SOS1_proteomics      0.815146     -0.13505\n",
       "31     ROCK1_proteomics      0.831541     -0.00695\n",
       "32    PIK3CD_proteomics      0.836701     -0.15015\n",
       "33     PREX2_proteomics      0.885732      0.02575\n",
       "34      PDK1_proteomics      0.896082     -0.30440\n",
       "35     FOXO1_proteomics      0.912203     -0.14820\n",
       "36       MVP_proteomics      0.921867     -0.01090\n",
       "37      USP7_proteomics      0.929331      0.14305\n",
       "38    PDGFRB_proteomics      0.980902     -0.04105\n",
       "39     MAPK1_proteomics      0.986037     -0.04200\n",
       "40      CAS9_proteomics           NaN          NaN\n",
       "41   CSNK2A1_proteomics           NaN          NaN\n",
       "42   MAP2K2 _proteomics           NaN          NaN\n",
       "43      MDM2_proteomics           NaN          NaN\n",
       "44    MTORC2_proteomics           NaN          NaN\n",
       "45    PIK3R3_proteomics           NaN          NaN"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_merged = b_pval.merge(b_df, on='Proteomics',how='inner')\n",
    "b_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#col = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CAS9, CCND1, FOXO1, MAGI3, MAP2K2 , MAST2, MDM2, MTORC2, PIK3R3, PREX2 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    76\n",
       "Deletion          25\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = col.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = col.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      " None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "c_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if c_sig is not None:\n",
    "    c_sig_list = list(c_sig.Comparison)\n",
    "else: \n",
    "    c_sig_list = None\n",
    "print('significant pvals: \\n',c_sig)\n",
    "\n",
    "# Get all pvals\n",
    "c_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "c_pval = c_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Colon_P_Value'})\n",
    "#c_pval = c_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#c_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "c_df = pd.DataFrame.from_dict(d, orient='index', columns=['Colon_Median'])\n",
    "c_df = c_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#c_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>0.003390</td>\n",
       "      <td>-0.19490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.20400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.030107</td>\n",
       "      <td>-0.11610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>0.050997</td>\n",
       "      <td>0.07900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>0.062750</td>\n",
       "      <td>0.10560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.116419</td>\n",
       "      <td>0.07995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.118135</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.131200</td>\n",
       "      <td>0.07600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.144425</td>\n",
       "      <td>0.04845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.182782</td>\n",
       "      <td>-0.23690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.237441</td>\n",
       "      <td>-0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.250276</td>\n",
       "      <td>-0.02450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>0.295137</td>\n",
       "      <td>0.10140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.326372</td>\n",
       "      <td>-0.04825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.328626</td>\n",
       "      <td>0.22100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.332902</td>\n",
       "      <td>-0.03385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.354186</td>\n",
       "      <td>0.00670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.366886</td>\n",
       "      <td>0.02350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.375198</td>\n",
       "      <td>-0.05730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.421338</td>\n",
       "      <td>0.11550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.491790</td>\n",
       "      <td>-0.14490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>0.514866</td>\n",
       "      <td>-0.02370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.516876</td>\n",
       "      <td>0.06900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.522867</td>\n",
       "      <td>0.00595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>0.531454</td>\n",
       "      <td>0.10670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>0.558901</td>\n",
       "      <td>-0.02300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>0.559999</td>\n",
       "      <td>0.01375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.561268</td>\n",
       "      <td>-0.03900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.665493</td>\n",
       "      <td>0.08550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.717025</td>\n",
       "      <td>0.02675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>0.757922</td>\n",
       "      <td>-0.05530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.827435</td>\n",
       "      <td>-0.01680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>0.883689</td>\n",
       "      <td>0.04335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.968870</td>\n",
       "      <td>-0.04905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.972777</td>\n",
       "      <td>0.01245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.974887</td>\n",
       "      <td>-0.18050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Colon_P_Value  Colon_Median\n",
       "0      MAPK3_proteomics       0.003390      -0.19490\n",
       "1       PTEN_proteomics       0.012872       0.20400\n",
       "2       SHC1_proteomics       0.030107      -0.11610\n",
       "3       PDK1_proteomics       0.050997       0.07900\n",
       "4       KRAS_proteomics       0.062750       0.10560\n",
       "5     PIK3C3_proteomics       0.116419       0.07995\n",
       "6     INPP4B_proteomics       0.118135      -0.08220\n",
       "7     PIK3CD_proteomics       0.131200       0.07600\n",
       "8    CSNK2A1_proteomics       0.144425       0.04845\n",
       "9        MVP_proteomics       0.182782      -0.23690\n",
       "10      USP7_proteomics       0.237441      -0.04450\n",
       "11    PDGFRB_proteomics       0.250276      -0.02450\n",
       "12      RAF1_proteomics       0.295137       0.10140\n",
       "13   CSNK2A2_proteomics       0.326372      -0.04825\n",
       "14  SLC9A3R1_proteomics       0.328626       0.22100\n",
       "15    PIK3CA_proteomics       0.332902      -0.03385\n",
       "16      AKT1_proteomics       0.354186       0.00670\n",
       "17      EGFR_proteomics       0.366886       0.02350\n",
       "18    PIK3R2_proteomics       0.375198      -0.05730\n",
       "19       BAD_proteomics       0.421338       0.11550\n",
       "20     NEDD4_proteomics       0.491790      -0.14490\n",
       "21     MAPK1_proteomics       0.514866      -0.02370\n",
       "22      XIAP_proteomics       0.516876       0.06900\n",
       "23      PTK2_proteomics       0.522867       0.00595\n",
       "24      TSC2_proteomics       0.531454       0.10670\n",
       "25      GAB1_proteomics       0.558901      -0.02300\n",
       "26      SOS1_proteomics       0.559999       0.01375\n",
       "27     ROCK1_proteomics       0.561268      -0.03900\n",
       "28    PIK3CB_proteomics       0.665493       0.08550\n",
       "29    MAP2K1_proteomics       0.717025       0.02675\n",
       "30      MTOR_proteomics       0.757922      -0.05530\n",
       "31      GRB2_proteomics       0.827435      -0.01680\n",
       "32     GSK3B_proteomics       0.883689       0.04335\n",
       "33      TP53_proteomics       0.968870      -0.04905\n",
       "34    PIK3R1_proteomics       0.972777       0.01245\n",
       "35     USP13_proteomics       0.974887      -0.18050\n",
       "36     MAGI3_proteomics            NaN           NaN\n",
       "37     PREX2_proteomics            NaN           NaN\n",
       "38     FOXO1_proteomics            NaN           NaN\n",
       "39    MTORC2_proteomics            NaN           NaN\n",
       "40   MAP2K2 _proteomics            NaN           NaN\n",
       "41     MAST2_proteomics            NaN           NaN\n",
       "42      MDM2_proteomics            NaN           NaN\n",
       "43    PIK3R3_proteomics            NaN           NaN\n",
       "44     CCND1_proteomics            NaN           NaN\n",
       "45      CAS9_proteomics            NaN           NaN"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c_merged = c_pval.merge(c_df, on='Proteomics',how='inner')\n",
    "c_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Head and Neck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "#h = cptac.Hnscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CAS9, MAP2K2 , MDM2, MTORC2, PIK3R3 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 176 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    87\n",
       "Deletion          18\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = h.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = h.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = ip)\n",
    "#prot_and_mutations = h.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "           Comparison   P_Value\n",
      "0    PTEN_proteomics  0.000023\n",
      "1  INPP4B_proteomics  0.000676\n",
      "2  PIK3CA_proteomics  0.001677\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:320: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals_sorted <= ecdffactor*alpha\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:328: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "# Get only sig sites\n",
    "h_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if h_sig is not None:\n",
    "    h_sig_list = list(h_sig.Comparison)\n",
    "else: \n",
    "    h_sig_list = None\n",
    "print('significant pvals: \\n',h_sig)\n",
    "\n",
    "# Get all pvals\n",
    "\n",
    "h_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "h_pval = h_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Hnscc_P_Value'})\n",
    "#h_pval = h_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#h_pval\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "h_d = {}\n",
    "\n",
    "for prot in ip:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    h_d[prot+'_proteomics'] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "h_df = pd.DataFrame.from_dict(h_d, orient='index', columns=['Hnscc_Median'])\n",
    "h_df = h_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#h_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>0.318812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>0.000676</td>\n",
       "      <td>0.591596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.001677</td>\n",
       "      <td>-0.163663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>0.005645</td>\n",
       "      <td>-0.217338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TSC2_proteomics</td>\n",
       "      <td>0.012038</td>\n",
       "      <td>0.081507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.014016</td>\n",
       "      <td>0.238171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>0.017719</td>\n",
       "      <td>-0.467227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>0.025498</td>\n",
       "      <td>-0.101291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>0.031385</td>\n",
       "      <td>-0.155484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>0.031801</td>\n",
       "      <td>-0.705560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>0.040586</td>\n",
       "      <td>-0.079700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>0.064384</td>\n",
       "      <td>-0.094081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MTOR_proteomics</td>\n",
       "      <td>0.066571</td>\n",
       "      <td>0.056843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.090920</td>\n",
       "      <td>-0.153286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.104723</td>\n",
       "      <td>0.190047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>0.104981</td>\n",
       "      <td>-0.049397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>0.105902</td>\n",
       "      <td>-0.138133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.120966</td>\n",
       "      <td>0.045723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>0.129494</td>\n",
       "      <td>0.061417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.137960</td>\n",
       "      <td>0.156746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.145523</td>\n",
       "      <td>0.099953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RAF1_proteomics</td>\n",
       "      <td>0.196196</td>\n",
       "      <td>0.028147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>0.256525</td>\n",
       "      <td>-0.079953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>0.272988</td>\n",
       "      <td>-0.042043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>0.300865</td>\n",
       "      <td>-0.070350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.312904</td>\n",
       "      <td>0.041287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>0.328437</td>\n",
       "      <td>0.101923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SOS1_proteomics</td>\n",
       "      <td>0.418398</td>\n",
       "      <td>0.057115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>0.434622</td>\n",
       "      <td>-0.072064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>0.455638</td>\n",
       "      <td>-0.088882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>0.500462</td>\n",
       "      <td>-0.059333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.550116</td>\n",
       "      <td>-0.008519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>0.570060</td>\n",
       "      <td>-0.064930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>0.597115</td>\n",
       "      <td>0.000771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>0.684058</td>\n",
       "      <td>0.000479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>0.698163</td>\n",
       "      <td>0.067759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>0.699824</td>\n",
       "      <td>0.036617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.713094</td>\n",
       "      <td>0.081410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.783206</td>\n",
       "      <td>0.090114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>CSNK2A2_proteomics</td>\n",
       "      <td>0.891248</td>\n",
       "      <td>0.093924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.995386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>MTORC2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MAP2K2 _proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MDM2_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PIK3R3_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CAS9_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Hnscc_P_Value  Hnscc_Median\n",
       "0       PTEN_proteomics       0.000023      0.318812\n",
       "1     INPP4B_proteomics       0.000676      0.591596\n",
       "2     PIK3CA_proteomics       0.001677     -0.163663\n",
       "3      USP13_proteomics       0.005645     -0.217338\n",
       "4       TSC2_proteomics       0.012038      0.081507\n",
       "5     PIK3CD_proteomics       0.014016      0.238171\n",
       "6       EGFR_proteomics       0.017719     -0.467227\n",
       "7       KRAS_proteomics       0.025498     -0.101291\n",
       "8   SLC9A3R1_proteomics       0.031385     -0.155484\n",
       "9       TP53_proteomics       0.031801     -0.705560\n",
       "10   CSNK2A1_proteomics       0.040586     -0.079700\n",
       "11      USP7_proteomics       0.064384     -0.094081\n",
       "12      MTOR_proteomics       0.066571      0.056843\n",
       "13    PIK3CB_proteomics       0.090920     -0.153286\n",
       "14      GRB2_proteomics       0.104723      0.190047\n",
       "15      XIAP_proteomics       0.104981     -0.049397\n",
       "16      PTK2_proteomics       0.105902     -0.138133\n",
       "17      SHC1_proteomics       0.120966      0.045723\n",
       "18     MAPK3_proteomics       0.129494      0.061417\n",
       "19       BAD_proteomics       0.137960      0.156746\n",
       "20    PDGFRB_proteomics       0.145523      0.099953\n",
       "21      RAF1_proteomics       0.196196      0.028147\n",
       "22      PDK1_proteomics       0.256525     -0.079953\n",
       "23    PIK3R2_proteomics       0.272988     -0.042043\n",
       "24      GAB1_proteomics       0.300865     -0.070350\n",
       "25    PIK3C3_proteomics       0.312904      0.041287\n",
       "26     ROCK1_proteomics       0.328437      0.101923\n",
       "27      SOS1_proteomics       0.418398      0.057115\n",
       "28    PIK3R1_proteomics       0.434622     -0.072064\n",
       "29     GSK3B_proteomics       0.455638     -0.088882\n",
       "30     MAGI3_proteomics       0.500462     -0.059333\n",
       "31    MAP2K1_proteomics       0.550116     -0.008519\n",
       "32     FOXO1_proteomics       0.570060     -0.064930\n",
       "33     MAPK1_proteomics       0.597115      0.000771\n",
       "34      AKT1_proteomics       0.684058      0.000479\n",
       "35     CCND1_proteomics       0.698163      0.067759\n",
       "36       MVP_proteomics       0.699824      0.036617\n",
       "37     MAST2_proteomics       0.713094      0.081410\n",
       "38     NEDD4_proteomics       0.783206      0.090114\n",
       "39   CSNK2A2_proteomics       0.891248      0.093924\n",
       "40     PREX2_proteomics            NaN     -0.995386\n",
       "41    MTORC2_proteomics            NaN           NaN\n",
       "42   MAP2K2 _proteomics            NaN           NaN\n",
       "43      MDM2_proteomics            NaN           NaN\n",
       "44    PIK3R3_proteomics            NaN           NaN\n",
       "45      CAS9_proteomics            NaN           NaN"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h_merged = h_pval.merge(h_df, on='Proteomics',how='inner')\n",
    "#del_wt[['PREX2_proteomics','Mutation']].dropna() # in median, not in pval (only 1 value for deletion)\n",
    "h_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Lung"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "#l = cptac.Luad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No_Mutation    88\n",
      "Deletion       17\n",
      "Name: Mutation, dtype: int64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: CAS9, CSNK2A1, CSNK2A2, MAP2K2 , MDM2, MTOR, MTORC2, PIK3R3, RAF1, SOS1, TSC2 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 6)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-00001.N, C3L-00009.N, C3L-00080.N, C3L-00083.N, C3L-00093.N, C3L-00094.N, C3L-00095.N, C3L-00140.N, C3L-00144.N, C3L-00263.N, C3L-00279.N, C3L-00368.N, C3L-00412.N, C3L-00422.N, C3L-00510.N, C3L-00604.N, C3L-00893.N, C3L-00913.N, C3L-00973.N, C3L-01330.N, C3L-01632.N, C3L-01682.N, C3L-01683.N, C3L-01889.N, C3L-01890, C3L-01890.N, C3L-01924.N, C3L-02219.N, C3L-02345.N, C3L-02348.N, C3L-02350.N, C3L-02365.N, C3L-02508.N, C3L-02549.N, C3N-00167.N, C3N-00169.N, C3N-00175.N, C3N-00180.N, C3N-00199.N, C3N-00203.N, C3N-00217.N, C3N-00223.N, C3N-00293.N, C3N-00433.N, C3N-00546.N, C3N-00547.N, C3N-00549.N, C3N-00550.N, C3N-00551.N, C3N-00552.N, C3N-00556.N, C3N-00559.N, C3N-00560.N, C3N-00572.N, C3N-00574.N, C3N-00578.N, C3N-00579.N, C3N-00580.N, C3N-00704.N, C3N-00737.N, C3N-00738.N, C3N-00959.N, C3N-01016.N, C3N-01021.N, C3N-01023.N, C3N-01024.N, C3N-01030.N, C3N-01071.N, C3N-01072.N, C3N-01405.N, C3N-01410.N, C3N-01413.N, C3N-01414.N, C3N-01415.N, C3N-01416.N, C3N-01488.N, C3N-01489.N, C3N-01799.N, C3N-01823.N, C3N-02000.N, C3N-02002.N, C3N-02003.N, C3N-02067.N, C3N-02087.N, C3N-02089.N, C3N-02145.N, C3N-02149.N, C3N-02155.N, C3N-02158.N, C3N-02379.N, C3N-02380.N, C3N-02421.N, C3N-02423.N, C3N-02424.N, C3N-02433.N, C3N-02529.N, C3N-02572.N, C3N-02582.N, C3N-02586.N, C3N-02587.N, C3N-02588.N, C3N-02729.N (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 6)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 4 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: You tried to flatten an index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n"
     ]
    }
   ],
   "source": [
    "mut_type = l.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.rename(columns={'PTEN':'cnv'})\n",
    "\n",
    "# different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "omics = l.join_omics_to_omics(df1_name = 'CNV', df2_name='proteomics',genes1=\"PTEN\", \n",
    "    genes2=ip)\n",
    "omics = l.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "omics = omics.drop(columns='PTEN_CNV')\n",
    "\n",
    "# Get only tumor samples\n",
    "p = l.get_proteomics(tissue_type='tumor')\n",
    "tumor_ids = list(p.index)\n",
    "get = omics.index.isin(tumor_ids)\n",
    "omics = omics[get]\n",
    "\n",
    "merged = omics.join(mut_type) # checked and there is 110 tumor samples for lung\n",
    "\n",
    "compare = ['No_Mutation','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "print(del_wt['Mutation'].value_counts())\n",
    "del_wt = del_wt.dropna(axis='columns', how='all')\n",
    "del_wt = del_wt.loc[:,~del_wt.columns.duplicated()] # SH1, PRK3R1 both have double columns\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant pvals: \n",
      "           Comparison       P_Value\n",
      "0    EGFR_proteomics  9.121167e-07\n",
      "1  PIK3CD_proteomics  4.312883e-04\n",
      "2    XIAP_proteomics  4.264738e-03\n",
      "3   FOXO1_proteomics  5.642364e-03\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>9.121167e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>4.312883e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>4.264738e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>5.642364e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>7.160789e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.435202e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>2.191472e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>2.581832e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>4.551992e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>7.637239e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>8.855596e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>9.086574e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>9.235432e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.718800e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.303937e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>1.698543e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>2.111085e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>2.392126e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>2.426636e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.598046e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>2.766716e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>3.694158e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>3.715141e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>3.935556e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>4.395072e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>4.795317e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>4.930512e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>5.256187e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>5.491624e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>6.486765e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>6.811128e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>8.904743e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>9.199836e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>9.265762e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>9.919357e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Luad_P_Value\n",
       "0       EGFR_proteomics  9.121167e-07\n",
       "1     PIK3CD_proteomics  4.312883e-04\n",
       "2       XIAP_proteomics  4.264738e-03\n",
       "3      FOXO1_proteomics  5.642364e-03\n",
       "4       GRB2_proteomics  7.160789e-03\n",
       "5       PTEN_proteomics  1.435202e-02\n",
       "6       AKT1_proteomics  2.191472e-02\n",
       "7      CCND1_proteomics  2.581832e-02\n",
       "8     PIK3CB_proteomics  4.551992e-02\n",
       "9     PIK3CA_proteomics  7.637239e-02\n",
       "10      PDK1_proteomics  8.855596e-02\n",
       "11      PTK2_proteomics  9.086574e-02\n",
       "12     NEDD4_proteomics  9.235432e-02\n",
       "13    PDGFRB_proteomics  9.718800e-02\n",
       "14       MVP_proteomics  1.303937e-01\n",
       "15      SHC1_proteomics  1.698543e-01\n",
       "16     PREX2_proteomics  2.111085e-01\n",
       "17      TP53_proteomics  2.392126e-01\n",
       "18    MAP2K1_proteomics  2.426636e-01\n",
       "19     ROCK1_proteomics  2.598046e-01\n",
       "20    INPP4B_proteomics  2.766716e-01\n",
       "21     USP13_proteomics  3.694158e-01\n",
       "22      USP7_proteomics  3.715141e-01\n",
       "23     MAGI3_proteomics  3.935556e-01\n",
       "24    PIK3R2_proteomics  4.395072e-01\n",
       "25  SLC9A3R1_proteomics  4.795317e-01\n",
       "26       BAD_proteomics  4.930512e-01\n",
       "27     MAPK3_proteomics  5.256187e-01\n",
       "28    PIK3C3_proteomics  5.491624e-01\n",
       "29    PIK3R1_proteomics  6.486765e-01\n",
       "30     MAPK1_proteomics  6.811128e-01\n",
       "31      GAB1_proteomics  8.904743e-01\n",
       "32     GSK3B_proteomics  9.199836e-01\n",
       "33      KRAS_proteomics  9.265762e-01\n",
       "34     MAST2_proteomics  9.919357e-01"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-2])\n",
    "\n",
    "# Get only sig sites\n",
    "l_sig = u.wrap_ttest(del_wt, 'Mutation', cols, correction_method = 'fdr_bh')\n",
    "if l_sig is not None:\n",
    "    l_sig_list = list(l_sig.Comparison)\n",
    "else: \n",
    "    l_sig_list = None\n",
    "print('significant pvals: \\n',l_sig)\n",
    "\n",
    "# Get all pvals\n",
    "l_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True, correction_method = 'fdr_bh')\n",
    "l_pval = l_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Luad_P_Value'})\n",
    "#l_pval = l_pval.replace(to_replace = r'_NP_.*$', value = '', regex = True)\n",
    "l_pval.head() # isoforms for some proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Isoform code - fix later\n",
    "#del_wt.replace(to_replace = r'_NP_.*$', value = '', regex = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>-0.28010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>0.05135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>1.18450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>-2.32970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>0.57005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>-0.02360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>0.55885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>0.04540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>-1.87510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>0.27130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>-0.45165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>0.10475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>0.06395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>-0.33845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>0.01015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>-0.03475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>0.13545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>0.48860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>0.62525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>0.10100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>0.13565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>0.28425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>0.84795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>-0.47740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>-0.27005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>0.79590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>0.36490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>-0.53510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>-0.27425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>0.24225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>-0.38490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>-2.76780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>-0.06650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>-0.13260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>-0.52480</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Luad_Median\n",
       "0       AKT1_proteomics     -0.28010\n",
       "1        BAD_proteomics      0.05135\n",
       "2      CCND1_proteomics      1.18450\n",
       "3       EGFR_proteomics     -2.32970\n",
       "4      FOXO1_proteomics      0.57005\n",
       "5       GAB1_proteomics     -0.02360\n",
       "6       GRB2_proteomics      0.55885\n",
       "7      GSK3B_proteomics      0.04540\n",
       "8     INPP4B_proteomics     -1.87510\n",
       "9       KRAS_proteomics      0.27130\n",
       "10     MAGI3_proteomics     -0.45165\n",
       "11    MAP2K1_proteomics      0.10475\n",
       "12     MAPK1_proteomics      0.06395\n",
       "13     MAPK3_proteomics     -0.33845\n",
       "14     MAST2_proteomics      0.01015\n",
       "15       MVP_proteomics     -0.03475\n",
       "16     NEDD4_proteomics      0.13545\n",
       "17    PDGFRB_proteomics      0.48860\n",
       "18      PDK1_proteomics      0.62525\n",
       "19    PIK3C3_proteomics      0.10100\n",
       "20    PIK3CA_proteomics      0.13565\n",
       "21    PIK3CB_proteomics      0.28425\n",
       "22    PIK3CD_proteomics      0.84795\n",
       "23    PIK3R1_proteomics     -0.47740\n",
       "24    PIK3R2_proteomics     -0.27005\n",
       "25     PREX2_proteomics      0.79590\n",
       "26      PTEN_proteomics      0.36490\n",
       "27      PTK2_proteomics     -0.53510\n",
       "28     ROCK1_proteomics     -0.27425\n",
       "29      SHC1_proteomics      0.24225\n",
       "30  SLC9A3R1_proteomics     -0.38490\n",
       "31      TP53_proteomics     -2.76780\n",
       "32     USP13_proteomics     -0.06650\n",
       "33      USP7_proteomics     -0.13260\n",
       "34      XIAP_proteomics     -0.52480"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"No_Mutation\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "l_d = {}\n",
    "\n",
    "for prot in cols:\n",
    "    dif = wt_med[prot] - del_med[prot]\n",
    "    l_d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "l_df = pd.DataFrame.from_dict(l_d, orient='index', columns=['Luad_Median'])\n",
    "l_df = l_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "l_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>9.121167e-07</td>\n",
       "      <td>-2.32970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>4.312883e-04</td>\n",
       "      <td>0.84795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>4.264738e-03</td>\n",
       "      <td>-0.52480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>5.642364e-03</td>\n",
       "      <td>0.57005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>7.160789e-03</td>\n",
       "      <td>0.55885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.435202e-02</td>\n",
       "      <td>0.36490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AKT1_proteomics</td>\n",
       "      <td>2.191472e-02</td>\n",
       "      <td>-0.28010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CCND1_proteomics</td>\n",
       "      <td>2.581832e-02</td>\n",
       "      <td>1.18450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3CB_proteomics</td>\n",
       "      <td>4.551992e-02</td>\n",
       "      <td>0.28425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>7.637239e-02</td>\n",
       "      <td>0.13565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PDK1_proteomics</td>\n",
       "      <td>8.855596e-02</td>\n",
       "      <td>0.62525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>9.086574e-02</td>\n",
       "      <td>-0.53510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NEDD4_proteomics</td>\n",
       "      <td>9.235432e-02</td>\n",
       "      <td>0.13545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PDGFRB_proteomics</td>\n",
       "      <td>9.718800e-02</td>\n",
       "      <td>0.48860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MVP_proteomics</td>\n",
       "      <td>1.303937e-01</td>\n",
       "      <td>-0.03475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SHC1_proteomics</td>\n",
       "      <td>1.698543e-01</td>\n",
       "      <td>0.24225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PREX2_proteomics</td>\n",
       "      <td>2.111085e-01</td>\n",
       "      <td>0.79590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>2.392126e-01</td>\n",
       "      <td>-2.76780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>2.426636e-01</td>\n",
       "      <td>0.10475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ROCK1_proteomics</td>\n",
       "      <td>2.598046e-01</td>\n",
       "      <td>-0.27425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>2.766716e-01</td>\n",
       "      <td>-1.87510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>USP13_proteomics</td>\n",
       "      <td>3.694158e-01</td>\n",
       "      <td>-0.06650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>3.715141e-01</td>\n",
       "      <td>-0.13260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MAGI3_proteomics</td>\n",
       "      <td>3.935556e-01</td>\n",
       "      <td>-0.45165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PIK3R2_proteomics</td>\n",
       "      <td>4.395072e-01</td>\n",
       "      <td>-0.27005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SLC9A3R1_proteomics</td>\n",
       "      <td>4.795317e-01</td>\n",
       "      <td>-0.38490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>BAD_proteomics</td>\n",
       "      <td>4.930512e-01</td>\n",
       "      <td>0.05135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>5.256187e-01</td>\n",
       "      <td>-0.33845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>5.491624e-01</td>\n",
       "      <td>0.10100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>6.486765e-01</td>\n",
       "      <td>-0.47740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MAPK1_proteomics</td>\n",
       "      <td>6.811128e-01</td>\n",
       "      <td>0.06395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GAB1_proteomics</td>\n",
       "      <td>8.904743e-01</td>\n",
       "      <td>-0.02360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GSK3B_proteomics</td>\n",
       "      <td>9.199836e-01</td>\n",
       "      <td>0.04540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>KRAS_proteomics</td>\n",
       "      <td>9.265762e-01</td>\n",
       "      <td>0.27130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>MAST2_proteomics</td>\n",
       "      <td>9.919357e-01</td>\n",
       "      <td>0.01015</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Proteomics  Luad_P_Value  Luad_Median\n",
       "0       EGFR_proteomics  9.121167e-07     -2.32970\n",
       "1     PIK3CD_proteomics  4.312883e-04      0.84795\n",
       "2       XIAP_proteomics  4.264738e-03     -0.52480\n",
       "3      FOXO1_proteomics  5.642364e-03      0.57005\n",
       "4       GRB2_proteomics  7.160789e-03      0.55885\n",
       "5       PTEN_proteomics  1.435202e-02      0.36490\n",
       "6       AKT1_proteomics  2.191472e-02     -0.28010\n",
       "7      CCND1_proteomics  2.581832e-02      1.18450\n",
       "8     PIK3CB_proteomics  4.551992e-02      0.28425\n",
       "9     PIK3CA_proteomics  7.637239e-02      0.13565\n",
       "10      PDK1_proteomics  8.855596e-02      0.62525\n",
       "11      PTK2_proteomics  9.086574e-02     -0.53510\n",
       "12     NEDD4_proteomics  9.235432e-02      0.13545\n",
       "13    PDGFRB_proteomics  9.718800e-02      0.48860\n",
       "14       MVP_proteomics  1.303937e-01     -0.03475\n",
       "15      SHC1_proteomics  1.698543e-01      0.24225\n",
       "16     PREX2_proteomics  2.111085e-01      0.79590\n",
       "17      TP53_proteomics  2.392126e-01     -2.76780\n",
       "18    MAP2K1_proteomics  2.426636e-01      0.10475\n",
       "19     ROCK1_proteomics  2.598046e-01     -0.27425\n",
       "20    INPP4B_proteomics  2.766716e-01     -1.87510\n",
       "21     USP13_proteomics  3.694158e-01     -0.06650\n",
       "22      USP7_proteomics  3.715141e-01     -0.13260\n",
       "23     MAGI3_proteomics  3.935556e-01     -0.45165\n",
       "24    PIK3R2_proteomics  4.395072e-01     -0.27005\n",
       "25  SLC9A3R1_proteomics  4.795317e-01     -0.38490\n",
       "26       BAD_proteomics  4.930512e-01      0.05135\n",
       "27     MAPK3_proteomics  5.256187e-01     -0.33845\n",
       "28    PIK3C3_proteomics  5.491624e-01      0.10100\n",
       "29    PIK3R1_proteomics  6.486765e-01     -0.47740\n",
       "30     MAPK1_proteomics  6.811128e-01      0.06395\n",
       "31      GAB1_proteomics  8.904743e-01     -0.02360\n",
       "32     GSK3B_proteomics  9.199836e-01      0.04540\n",
       "33      KRAS_proteomics  9.265762e-01      0.27130\n",
       "34     MAST2_proteomics  9.919357e-01      0.01015"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l_merged = l_pval.merge(l_df, on='Proteomics',how='inner')\n",
    "#l_merged = l_merged.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "l_merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "n = g_merged.merge(h_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(l_merged, on='Proteomics',how='outer')\n",
    "n= n.merge(b_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(o_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(e_merged, on='Proteomics',how='outer')\n",
    "all_df = n.merge(c_merged, on='Proteomics',how='outer')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN_proteomics</td>\n",
       "      <td>1.219523e-07</td>\n",
       "      <td>0.511726</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>0.318812</td>\n",
       "      <td>1.435202e-02</td>\n",
       "      <td>0.36490</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>1.00800</td>\n",
       "      <td>0.000909</td>\n",
       "      <td>0.169972</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>0.74660</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.20400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAPK3_proteomics</td>\n",
       "      <td>1.020660e-06</td>\n",
       "      <td>0.665105</td>\n",
       "      <td>0.129494</td>\n",
       "      <td>0.061417</td>\n",
       "      <td>5.256187e-01</td>\n",
       "      <td>-0.33845</td>\n",
       "      <td>0.285412</td>\n",
       "      <td>0.22335</td>\n",
       "      <td>0.842428</td>\n",
       "      <td>-0.059507</td>\n",
       "      <td>0.272111</td>\n",
       "      <td>-0.08200</td>\n",
       "      <td>0.003390</td>\n",
       "      <td>-0.19490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CD_proteomics</td>\n",
       "      <td>1.030815e-03</td>\n",
       "      <td>0.492099</td>\n",
       "      <td>0.014016</td>\n",
       "      <td>0.238171</td>\n",
       "      <td>4.312883e-04</td>\n",
       "      <td>0.84795</td>\n",
       "      <td>0.836701</td>\n",
       "      <td>-0.15015</td>\n",
       "      <td>0.937984</td>\n",
       "      <td>0.035308</td>\n",
       "      <td>0.045489</td>\n",
       "      <td>0.18243</td>\n",
       "      <td>0.131200</td>\n",
       "      <td>0.07600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRB2_proteomics</td>\n",
       "      <td>1.089951e-03</td>\n",
       "      <td>0.312146</td>\n",
       "      <td>0.104723</td>\n",
       "      <td>0.190047</td>\n",
       "      <td>7.160789e-03</td>\n",
       "      <td>0.55885</td>\n",
       "      <td>0.061517</td>\n",
       "      <td>0.31470</td>\n",
       "      <td>0.086428</td>\n",
       "      <td>-0.184243</td>\n",
       "      <td>0.160116</td>\n",
       "      <td>0.26400</td>\n",
       "      <td>0.827435</td>\n",
       "      <td>-0.01680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3C3_proteomics</td>\n",
       "      <td>1.225949e-03</td>\n",
       "      <td>0.102518</td>\n",
       "      <td>0.312904</td>\n",
       "      <td>0.041287</td>\n",
       "      <td>5.491624e-01</td>\n",
       "      <td>0.10100</td>\n",
       "      <td>0.445080</td>\n",
       "      <td>-0.11490</td>\n",
       "      <td>0.444583</td>\n",
       "      <td>-0.104687</td>\n",
       "      <td>0.036592</td>\n",
       "      <td>0.13160</td>\n",
       "      <td>0.116419</td>\n",
       "      <td>0.07995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>USP7_proteomics</td>\n",
       "      <td>4.017792e-03</td>\n",
       "      <td>-0.154035</td>\n",
       "      <td>0.064384</td>\n",
       "      <td>-0.094081</td>\n",
       "      <td>3.715141e-01</td>\n",
       "      <td>-0.13260</td>\n",
       "      <td>0.929331</td>\n",
       "      <td>0.14305</td>\n",
       "      <td>0.525069</td>\n",
       "      <td>0.017808</td>\n",
       "      <td>0.572500</td>\n",
       "      <td>-0.03834</td>\n",
       "      <td>0.237441</td>\n",
       "      <td>-0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MAP2K1_proteomics</td>\n",
       "      <td>4.138456e-03</td>\n",
       "      <td>0.592389</td>\n",
       "      <td>0.550116</td>\n",
       "      <td>-0.008519</td>\n",
       "      <td>2.426636e-01</td>\n",
       "      <td>0.10475</td>\n",
       "      <td>0.757927</td>\n",
       "      <td>0.03045</td>\n",
       "      <td>0.472124</td>\n",
       "      <td>0.025625</td>\n",
       "      <td>0.675793</td>\n",
       "      <td>0.09190</td>\n",
       "      <td>0.717025</td>\n",
       "      <td>0.02675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3R1_proteomics</td>\n",
       "      <td>4.282296e-03</td>\n",
       "      <td>0.188390</td>\n",
       "      <td>0.434622</td>\n",
       "      <td>-0.072064</td>\n",
       "      <td>6.486765e-01</td>\n",
       "      <td>-0.47740</td>\n",
       "      <td>0.164475</td>\n",
       "      <td>1.03615</td>\n",
       "      <td>0.219977</td>\n",
       "      <td>-0.179425</td>\n",
       "      <td>0.135943</td>\n",
       "      <td>-0.33990</td>\n",
       "      <td>0.972777</td>\n",
       "      <td>0.01245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>INPP4B_proteomics</td>\n",
       "      <td>1.455578e-02</td>\n",
       "      <td>0.341829</td>\n",
       "      <td>0.000676</td>\n",
       "      <td>0.591596</td>\n",
       "      <td>2.766716e-01</td>\n",
       "      <td>-1.87510</td>\n",
       "      <td>0.295970</td>\n",
       "      <td>0.37695</td>\n",
       "      <td>0.353620</td>\n",
       "      <td>-0.370647</td>\n",
       "      <td>0.489186</td>\n",
       "      <td>-0.16320</td>\n",
       "      <td>0.118135</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>EGFR_proteomics</td>\n",
       "      <td>4.010871e-02</td>\n",
       "      <td>-1.001035</td>\n",
       "      <td>0.017719</td>\n",
       "      <td>-0.467227</td>\n",
       "      <td>9.121167e-07</td>\n",
       "      <td>-2.32970</td>\n",
       "      <td>0.016614</td>\n",
       "      <td>-0.58645</td>\n",
       "      <td>0.838960</td>\n",
       "      <td>0.063388</td>\n",
       "      <td>0.006128</td>\n",
       "      <td>-0.52600</td>\n",
       "      <td>0.366886</td>\n",
       "      <td>0.02350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CSNK2A1_proteomics</td>\n",
       "      <td>5.604258e-02</td>\n",
       "      <td>-0.123874</td>\n",
       "      <td>0.040586</td>\n",
       "      <td>-0.079700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.580920</td>\n",
       "      <td>-0.015518</td>\n",
       "      <td>0.002364</td>\n",
       "      <td>0.10412</td>\n",
       "      <td>0.144425</td>\n",
       "      <td>0.04845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>XIAP_proteomics</td>\n",
       "      <td>1.184655e-01</td>\n",
       "      <td>-0.088967</td>\n",
       "      <td>0.104981</td>\n",
       "      <td>-0.049397</td>\n",
       "      <td>4.264738e-03</td>\n",
       "      <td>-0.52480</td>\n",
       "      <td>0.383804</td>\n",
       "      <td>-0.15040</td>\n",
       "      <td>0.685027</td>\n",
       "      <td>0.010591</td>\n",
       "      <td>0.913864</td>\n",
       "      <td>-0.09772</td>\n",
       "      <td>0.516876</td>\n",
       "      <td>0.06900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PIK3CA_proteomics</td>\n",
       "      <td>2.876721e-01</td>\n",
       "      <td>0.105746</td>\n",
       "      <td>0.001677</td>\n",
       "      <td>-0.163663</td>\n",
       "      <td>7.637239e-02</td>\n",
       "      <td>0.13565</td>\n",
       "      <td>0.460310</td>\n",
       "      <td>0.27750</td>\n",
       "      <td>0.008335</td>\n",
       "      <td>-0.199296</td>\n",
       "      <td>0.001761</td>\n",
       "      <td>0.35300</td>\n",
       "      <td>0.332902</td>\n",
       "      <td>-0.03385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>6.682500e-01</td>\n",
       "      <td>-0.256630</td>\n",
       "      <td>0.031801</td>\n",
       "      <td>-0.705560</td>\n",
       "      <td>2.392126e-01</td>\n",
       "      <td>-2.76780</td>\n",
       "      <td>0.068427</td>\n",
       "      <td>-0.11100</td>\n",
       "      <td>0.980145</td>\n",
       "      <td>-0.196586</td>\n",
       "      <td>0.001246</td>\n",
       "      <td>0.80260</td>\n",
       "      <td>0.968870</td>\n",
       "      <td>-0.04905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>FOXO1_proteomics</td>\n",
       "      <td>7.974795e-01</td>\n",
       "      <td>-0.083073</td>\n",
       "      <td>0.570060</td>\n",
       "      <td>-0.064930</td>\n",
       "      <td>5.642364e-03</td>\n",
       "      <td>0.57005</td>\n",
       "      <td>0.912203</td>\n",
       "      <td>-0.14820</td>\n",
       "      <td>0.876446</td>\n",
       "      <td>0.039263</td>\n",
       "      <td>0.138674</td>\n",
       "      <td>-0.09000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Proteomics   Gbm_P_Value  Gbm_Median  Hnscc_P_Value  Hnscc_Median  \\\n",
       "0      PTEN_proteomics  1.219523e-07    0.511726       0.000023      0.318812   \n",
       "1     MAPK3_proteomics  1.020660e-06    0.665105       0.129494      0.061417   \n",
       "2    PIK3CD_proteomics  1.030815e-03    0.492099       0.014016      0.238171   \n",
       "3      GRB2_proteomics  1.089951e-03    0.312146       0.104723      0.190047   \n",
       "4    PIK3C3_proteomics  1.225949e-03    0.102518       0.312904      0.041287   \n",
       "5      USP7_proteomics  4.017792e-03   -0.154035       0.064384     -0.094081   \n",
       "6    MAP2K1_proteomics  4.138456e-03    0.592389       0.550116     -0.008519   \n",
       "7    PIK3R1_proteomics  4.282296e-03    0.188390       0.434622     -0.072064   \n",
       "9    INPP4B_proteomics  1.455578e-02    0.341829       0.000676      0.591596   \n",
       "12     EGFR_proteomics  4.010871e-02   -1.001035       0.017719     -0.467227   \n",
       "14  CSNK2A1_proteomics  5.604258e-02   -0.123874       0.040586     -0.079700   \n",
       "20     XIAP_proteomics  1.184655e-01   -0.088967       0.104981     -0.049397   \n",
       "26   PIK3CA_proteomics  2.876721e-01    0.105746       0.001677     -0.163663   \n",
       "35     TP53_proteomics  6.682500e-01   -0.256630       0.031801     -0.705560   \n",
       "37    FOXO1_proteomics  7.974795e-01   -0.083073       0.570060     -0.064930   \n",
       "\n",
       "    Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median  Ov_P_Value  \\\n",
       "0   1.435202e-02      0.36490      0.000004      1.00800    0.000909   \n",
       "1   5.256187e-01     -0.33845      0.285412      0.22335    0.842428   \n",
       "2   4.312883e-04      0.84795      0.836701     -0.15015    0.937984   \n",
       "3   7.160789e-03      0.55885      0.061517      0.31470    0.086428   \n",
       "4   5.491624e-01      0.10100      0.445080     -0.11490    0.444583   \n",
       "5   3.715141e-01     -0.13260      0.929331      0.14305    0.525069   \n",
       "6   2.426636e-01      0.10475      0.757927      0.03045    0.472124   \n",
       "7   6.486765e-01     -0.47740      0.164475      1.03615    0.219977   \n",
       "9   2.766716e-01     -1.87510      0.295970      0.37695    0.353620   \n",
       "12  9.121167e-07     -2.32970      0.016614     -0.58645    0.838960   \n",
       "14           NaN          NaN           NaN          NaN    0.580920   \n",
       "20  4.264738e-03     -0.52480      0.383804     -0.15040    0.685027   \n",
       "26  7.637239e-02      0.13565      0.460310      0.27750    0.008335   \n",
       "35  2.392126e-01     -2.76780      0.068427     -0.11100    0.980145   \n",
       "37  5.642364e-03      0.57005      0.912203     -0.14820    0.876446   \n",
       "\n",
       "    Ov_Median  En_P_Value  En_Median  Colon_P_Value  Colon_Median  \n",
       "0    0.169972    0.000640    0.74660       0.012872       0.20400  \n",
       "1   -0.059507    0.272111   -0.08200       0.003390      -0.19490  \n",
       "2    0.035308    0.045489    0.18243       0.131200       0.07600  \n",
       "3   -0.184243    0.160116    0.26400       0.827435      -0.01680  \n",
       "4   -0.104687    0.036592    0.13160       0.116419       0.07995  \n",
       "5    0.017808    0.572500   -0.03834       0.237441      -0.04450  \n",
       "6    0.025625    0.675793    0.09190       0.717025       0.02675  \n",
       "7   -0.179425    0.135943   -0.33990       0.972777       0.01245  \n",
       "9   -0.370647    0.489186   -0.16320       0.118135      -0.08220  \n",
       "12   0.063388    0.006128   -0.52600       0.366886       0.02350  \n",
       "14  -0.015518    0.002364    0.10412       0.144425       0.04845  \n",
       "20   0.010591    0.913864   -0.09772       0.516876       0.06900  \n",
       "26  -0.199296    0.001761    0.35300       0.332902      -0.03385  \n",
       "35  -0.196586    0.001246    0.80260       0.968870      -0.04905  \n",
       "37   0.039263    0.138674   -0.09000            NaN           NaN  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Keep only genes significant in > 1 cancer\n",
    "all_sig = []\n",
    "\n",
    "sig_lists = [g_sig_list, e_sig_list, b_sig_list, o_sig_list, c_sig_list, h_sig_list, l_sig_list]\n",
    "\n",
    "for next_list in sig_lists:\n",
    "    if next_list is not None:\n",
    "        all_sig.append(next_list)\n",
    "        \n",
    "flat_list = [item for sublist in all_sig for item in sublist]\n",
    "sig = set(flat_list) # remove duplicates\n",
    "\n",
    "\n",
    "bool_df = all_df['Proteomics'].isin(sig)\n",
    "sig_df = all_df[bool_df]\n",
    "print(len(sig_df))\n",
    "sig_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_df.to_csv('sig_pval_15_interacting.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Lists sig in cancer\n",
    "cancer = ['Gbm','En','Brca','Ov','Colon','Hnscc','Luad']\n",
    "i = 0\n",
    "for next_list in sig_lists:\n",
    "\n",
    "    print(cancer[i])\n",
    "    print(next_list, '\\n')\n",
    "    if (i < 6):\n",
    "        i += 1\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HeatMaps\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>1.219523e-07</td>\n",
       "      <td>0.511726</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.591964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>1.020660e-06</td>\n",
       "      <td>0.665105</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.379506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CD</td>\n",
       "      <td>1.030815e-03</td>\n",
       "      <td>0.492099</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.687741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>1.089951e-03</td>\n",
       "      <td>0.312146</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.682162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3C3</td>\n",
       "      <td>1.225949e-03</td>\n",
       "      <td>0.102518</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.670404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>7.170253e-01</td>\n",
       "      <td>0.026750</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.033264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>8.274353e-01</td>\n",
       "      <td>-0.016800</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.018942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TP53</td>\n",
       "      <td>9.688700e-01</td>\n",
       "      <td>-0.049050</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.003162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>9.727770e-01</td>\n",
       "      <td>0.012450</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.002760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>FOXO1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Colon</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>104 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Proteomics       P_Value   Medians Cancer      size\n",
       "0        PTEN  1.219523e-07  0.511726    Gbm  1.591964\n",
       "1       MAPK3  1.020660e-06  0.665105    Gbm  1.379506\n",
       "2      PIK3CD  1.030815e-03  0.492099    Gbm  0.687741\n",
       "3        GRB2  1.089951e-03  0.312146    Gbm  0.682162\n",
       "4      PIK3C3  1.225949e-03  0.102518    Gbm  0.670404\n",
       "..        ...           ...       ...    ...       ...\n",
       "29     MAP2K1  7.170253e-01  0.026750  Colon  0.033264\n",
       "31       GRB2  8.274353e-01 -0.016800  Colon  0.018942\n",
       "33       TP53  9.688700e-01 -0.049050  Colon  0.003162\n",
       "34     PIK3R1  9.727770e-01  0.012450  Colon  0.002760\n",
       "38      FOXO1           NaN       NaN  Colon       NaN\n",
       "\n",
       "[104 rows x 5 columns]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create df for heat map\n",
    "\n",
    "cancer = ['Gbm','Hnscc','Luad','Brca','Ov','En','Colon']\n",
    "merged_dfs = [g_merged,h_merged,l_merged,b_merged,o_merged,e_merged,c_merged]\n",
    "\n",
    "merged = pd.DataFrame()\n",
    "i = 0\n",
    "for c in cancer:\n",
    "    m = merged_dfs[i]\n",
    "    m2 = m.assign(Cancer = c)\n",
    "    m2 = m2.rename(columns={c+'_P_Value': 'P_Value'})\n",
    "    m2 = m2.rename(columns={c+'_Median': 'Medians'})\n",
    "    merged = merged.append(m2)\n",
    "    if i < 6:\n",
    "        i += 1\n",
    "        \n",
    "bool_df2 = merged['Proteomics'].isin(sig)\n",
    "plot_df = merged[bool_df2]\n",
    "plot_df = plot_df.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "# log p-vals for right scale in plot (bigger circle, smaller pval)\n",
    "plot_df['size'] = plot_df['P_Value'].apply(lambda x: -1*(np.log(x)/10))\n",
    "\n",
    "plot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"2083\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"2083\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"2083\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '2083' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"2083\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"2083\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"2083\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '2083' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"2083\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"86709e16-d9e2-4f77-b479-9399661054cf\" data-root-id=\"2060\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"9f2e5281-9a70-4b3c-bcaa-e73a352ccf53\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"2067\"}],\"center\":[{\"id\":\"2069\"},{\"id\":\"2072\"}],\"left\":[{\"id\":\"2070\"}],\"plot_height\":450,\"plot_width\":700,\"renderers\":[{\"id\":\"2080\"}],\"right\":[{\"id\":\"2082\"}],\"title\":{\"id\":\"2165\"},\"toolbar\":{\"id\":\"2074\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"2058\"},\"x_scale\":{\"id\":\"2063\"},\"y_range\":{\"id\":\"2059\"},\"y_scale\":{\"id\":\"2065\"}},\"id\":\"2060\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{},\"id\":\"2171\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"formatter\":{\"id\":\"2169\"},\"ticker\":{\"id\":\"2071\"}},\"id\":\"2070\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"2065\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"data\":{\"Cancer\":[\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\"],\"Medians\":{\"__ndarray__\":\"jGwIKw5g4D+/ll42ikjlPygHUt6Oft8/HckqfTH60z9ksWq9nT66P4iMpUNrt8O/5kywOtr04j/IXfSVLB3IP1zsGQeG4NU/IvwGUT0E8L8pj79tN7a/v2qEMUmGxra/z58OTSQSuz+aGp9poGzQv6AnYJVMRLW/gGyKmWtn1D8gbd9YWu7iP4Cm1rLr8sS/AImKGGZ8zj8AA6CbDefdv6D1nV3zk+a/AK88/DpntL8AO8i1rxW4v4BOdDF3U8g/AFT15ZtKqb8A/rGtAnKvPwBS7sSDI6U/AP2Jrc5ysr8AmB3NkHKBvwDPteo8n7C/+u3rwDmjAsAYldQJaCLrPwkbnl4py+C/rrZif9k94j/E0ytlGeLhP5jdk4eFWtc/QOjZrPpcwT9N845TdCQGwDeJQWDl0Lo/cayL22gA/r9+jLlrCfnAv2IyVTAqqdW/QmDl0CLbuT/MEMe6uI3ev1TjpZvEIPA/h6dXyjLE4r/cRgN4CyTUP+Ai2/l+ary/pAG8BRKU8D+E4seYu5bMP3KKjuTyH9g/VMGopE5Aw79gdk8eFmq9v8L1KFyPwtE/8B2n6Egunz+ASL99HTjDv5wzorQ3+MK/rK3YX3ZPwj8AAAAAAAD4/0tthnSowcU/MgA2G4mCyb8YeL4ERpXHv0cTitVo98a/FxfS16+417/uQD2AvMy6v5BX2u6NPZo/AHB4g2A8kj8AfJRqz8ePv1DVde97sIU/rCmbAjE6sD+gUV87vXeuvzAif6BEGqQ/vn554ugToj+acEXouynJv/kx5q4l5Oc/kKD4Meau6T9kO99PjZfWP4EJ3Lqbp7o/okW28/3U4L8qqRPQRNjAP3yW58HdWcc//kP67evA1b8M16NwPQq3v0w3iUFg5dA/yqFFtvP9tL+HhVrTvOPEvxrdQexMoaO/SZ2AJsKGtz9ORSqMLQS5v27F/rJ78si/6iYxCKwcyj/zsFBrmne0P+wvuycPC7W/2/l+arx0sz+Qfvs6cM6oP2QQWDm0yKa/aW/whclUob+g8dJNYhCYP0SLbOf7qbE/MQisHFpkmz8UjErqBDSRv1zTvOMUHam/YH2utmJ/iT8AAAAAAAD4/w==\",\"dtype\":\"float64\",\"shape\":[104]},\"P_Value\":{\"__ndarray__\":\"oqcy7D9egD7pRM2esx+xPiRRp7iM41A/mPRJMpbbUT90yliQABZUP+mDmML1dHA/iJMgUnzzcD+MLLUnUIpxP4SR9rxrz40/LSuaISGJpD+eMmgInbGsP5Dy+NLBU74/19G/XThp0j9ONHPQTWLlP/HHi6vzhOk/A0h5FIH29z5j/9ZB9CJGP3ZhO3BFeFs/WutnzoK0jD8dH5YB2iSSPx9r7+UsSKA/qI1QaKjHpD+ZG4OhfnuwPy2EoCgdz7o/Hrl8bAvguj+5iAUbQ5PAPxQyeaCfBtQ/IgSrINrQ2z9Nu/rii5rhP/zps7PuPeI/xy2F6QWbrj7sxBA50UM8P4YF0f7md3E/Ad3biXIcdz8+dm27oVR9PwfzGyeYZI0/toutBCSNsz8nPS+phJ7OP4EEkzWaD88/GvhR3Py00T/++I8+48bXP3ItMDfe0eA/7sQL9byS4T/T7PAp9cHkPyog947hItI+KH+yGDEDkT+//qlHF3+vP3xj1edzhLE/uikfnoINxT9XoRhTL0TSP2jHJncq8dI/xOu7MECQ2D/kQqi2MnzcP1RJRaO5dd0/us5ucu9A6D+UtsmIQcbqP+Du27PDMO0/VkVWnBS97T8AAAAAAAD4f66Dwe6Yy00/2q3gLNwRgT9CAeLBJCC2Px/EU4w2KMw/OFkWuLSh1j/S7OIhDHTcP7bN4iBGN94/PspbSV3N4D8Yzdd55ZbiP1m707K96+U/AajQZcPY6j9YqI8KLPXqP7vXqELYC+w/jxB0H/YD7j/du5lxWV3vP0hDNks7+UQ/hFfkXDhpVD8xf9M4Q9xcP4qdAevGXGM/vnq0L8wZeT+9Qz/YLryiP3+DkjBTSqc/6lndjZdmwT96nxO8DsDBP7wGihavfsQ/llfVXEVq0T/4go/n0k7fP7TlZnLsUeI/Hl85CRig5T8Usxk0Xz7tPy90KZB2xGs/fgFuFIxcij91bvHCm829P4ff800WPr4/XSiA7CzLwD/l9vw8hnzCP4JGoUZ2ZM4/Wif/0EVO1T9e33NXDnvXP3rrI35AiuA/XsWbGd/x5j98oSaVWXrqPzyIx5j7AO8/Cdu4Jv0g7z8AAAAAAAD4fw==\",\"dtype\":\"float64\",\"shape\":[104]},\"Proteomics\":[\"PTEN\",\"MAPK3\",\"PIK3CD\",\"GRB2\",\"PIK3C3\",\"USP7\",\"MAP2K1\",\"PIK3R1\",\"INPP4B\",\"EGFR\",\"CSNK2A1\",\"XIAP\",\"PIK3CA\",\"TP53\",\"FOXO1\",\"PTEN\",\"INPP4B\",\"PIK3CA\",\"PIK3CD\",\"EGFR\",\"TP53\",\"CSNK2A1\",\"USP7\",\"GRB2\",\"XIAP\",\"MAPK3\",\"PIK3C3\",\"PIK3R1\",\"MAP2K1\",\"FOXO1\",\"EGFR\",\"PIK3CD\",\"XIAP\",\"FOXO1\",\"GRB2\",\"PTEN\",\"PIK3CA\",\"TP53\",\"MAP2K1\",\"INPP4B\",\"USP7\",\"MAPK3\",\"PIK3C3\",\"PIK3R1\",\"PTEN\",\"EGFR\",\"GRB2\",\"TP53\",\"PIK3R1\",\"MAPK3\",\"INPP4B\",\"XIAP\",\"PIK3C3\",\"PIK3CA\",\"MAP2K1\",\"PIK3CD\",\"FOXO1\",\"USP7\",\"CSNK2A1\",\"PTEN\",\"PIK3CA\",\"GRB2\",\"PIK3R1\",\"INPP4B\",\"PIK3C3\",\"MAP2K1\",\"USP7\",\"CSNK2A1\",\"XIAP\",\"EGFR\",\"MAPK3\",\"FOXO1\",\"PIK3CD\",\"TP53\",\"PTEN\",\"TP53\",\"PIK3CA\",\"CSNK2A1\",\"EGFR\",\"PIK3C3\",\"PIK3CD\",\"PIK3R1\",\"FOXO1\",\"GRB2\",\"MAPK3\",\"INPP4B\",\"USP7\",\"MAP2K1\",\"XIAP\",\"MAPK3\",\"PTEN\",\"PIK3C3\",\"INPP4B\",\"PIK3CD\",\"CSNK2A1\",\"USP7\",\"PIK3CA\",\"EGFR\",\"XIAP\",\"MAP2K1\",\"GRB2\",\"TP53\",\"PIK3R1\",\"FOXO1\"],\"index\":[0,1,2,3,4,5,6,7,9,12,14,20,26,35,37,0,1,2,5,6,9,10,11,14,15,18,25,28,31,32,0,1,2,3,4,5,9,17,18,20,22,27,28,29,0,2,3,4,8,13,14,18,20,21,29,32,35,37,41,0,2,6,10,14,16,17,20,24,27,29,30,32,35,38,0,1,2,3,4,7,8,13,14,15,21,26,29,31,38,0,1,5,6,7,8,10,15,17,22,29,31,33,34,38],\"size\":{\"__ndarray__\":\"oZbIjkjmU0ArQdNlaz5RQDYkdDCKMUFABzsHm9YNQUDGTG7wlcJAQDPSnQvKlTtAhvaG4ulvO0BvmXMrLkQ7QPOsthUaJjVAn7uD2a8UMED7l2XdAdEsQHTLHDHSVCVA+935dS7rGEAhGc3uqx8AQGwCXXibGvI/wGInwlG3SkDiSRJp/D9CQPvaEwVy9D9A7v8to3RWNUBVk46HaSo0QBdKkZfIPTFAWwRvaY0FMEB7NoudyG0rQLBU2W97kCZAfgbAtSyKJkBSJAmK8nAkQItCmH+2PBdAbQCb9GGqEEDUkM87s+cHQL+KZ5QEewZAQFtwOWZiUUBk8SuJMF9DQF4psHRwSTtAHuxDhiPjOUApx7jHF7I4QP6K+W8lODVAO+B9y6m4KUDObIJRqZscQOIg/ARTUhxAUiLdj8+yGUDWG7V2qc0TQPsVsf0nuglA5VCOPXX5B0DLd2vFFlABQPK6NhjZ4E5ArzsoGdV8NEBihMZ4buIrQMWRNHjg0SpA4YEmb8sMIkB609VzkhMZQFYYZ7CXWRhAwyeCuQYnE0AU8MUipDAQQI1Nunq/CA9AUzfK3Wks9j/3M3pIsIbsPynFNTHhZ90/sHwnfe5z1z8AAAAAAAD4f8fz6BXUgUFAY86Ry8HvN0AY/l0yBHwoQJqnB1DdSB5Avmid1GjKFEAJwW/8XTYQQOcWYVZEBQ5AxOaVQ9/ECUDMenXkxrkFQGhBRBKGQ/4/BqRprj0Y7D+GZ3rTR2/rP0uN1TvRGeU///F4XcB81D8sMj3vequ5Pz8Z+FuGYkJAR0g2lk+4QEACgpE9PLU/QCr9+PDmPD5ALRxzNG55OUAwZJ9SJYowQHFAKAYi5y5A48yR/YX0I0CvBqo9nsEjQFFUrkSNUSJA2ms+EOgHGkA7IK42upkMQOzj8JZGTwZAsAcFkXlZ/z8zw7Uf2NLcPzkq6VxsbzxA7aWPSnnDNUApT9MFcYElQHM2Lvn5WyVA9+JjR29PJEAyzKV0lVkjQA/mdFS5wRxAmD0SvoT/FUCQdO892Q0UQLOMcP3mZQpAKbjBLI2c+j+5dDB40k7uPwbxEf1qPcQ/D+1Gyg6qwT8AAAAAAAD4fw==\",\"dtype\":\"float64\",\"shape\":[104]}},\"selected\":{\"id\":\"2173\"},\"selection_policy\":{\"id\":\"2172\"}},\"id\":\"2076\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"2057\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Cancer\"},\"y\":{\"field\":\"Proteomics\"}},\"id\":\"2078\",\"type\":\"Scatter\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"2073\"}]},\"id\":\"2074\",\"type\":\"Toolbar\"},{\"attributes\":{\"data_source\":{\"id\":\"2076\"},\"glyph\":{\"id\":\"2078\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"2079\"},\"selection_glyph\":null,\"view\":{\"id\":\"2081\"}},\"id\":\"2080\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"factors\":[\"Brca\",\"Colon\",\"En\",\"Gbm\",\"Hnscc\",\"Luad\",\"Ov\"]},\"id\":\"2058\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"2169\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{},\"id\":\"2173\",\"type\":\"Selection\"},{\"attributes\":{\"callback\":null},\"id\":\"2073\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"2071\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"axis\":{\"id\":\"2070\"},\"dimension\":1,\"ticker\":null},\"id\":\"2072\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"2068\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"factors\":[\"XIAP\",\"USP7\",\"TP53\",\"PTEN\",\"PIK3R1\",\"PIK3CD\",\"PIK3CA\",\"PIK3C3\",\"MAPK3\",\"MAP2K1\",\"INPP4B\",\"GRB2\",\"FOXO1\",\"EGFR\",\"CSNK2A1\"]},\"id\":\"2059\",\"type\":\"FactorRange\"},{\"attributes\":{\"text\":\"\"},\"id\":\"2165\",\"type\":\"Title\"},{\"attributes\":{\"high\":1,\"low\":-1,\"palette\":[\"#b2182b\",\"#d6604d\",\"#f4a582\",\"#fddbc7\",\"#f7f7f7\",\"#d1e5f0\",\"#92c5de\",\"#4393c3\",\"#2166ac\"]},\"id\":\"2057\",\"type\":\"LinearColorMapper\"},{\"attributes\":{},\"id\":\"2166\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"2172\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"2167\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"axis\":{\"id\":\"2067\"},\"ticker\":null},\"id\":\"2069\",\"type\":\"Grid\"},{\"attributes\":{\"color_mapper\":{\"id\":\"2057\"},\"formatter\":{\"id\":\"2167\"},\"location\":[0,0],\"ticker\":{\"id\":\"2166\"}},\"id\":\"2082\",\"type\":\"ColorBar\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"2057\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Cancer\"},\"y\":{\"field\":\"Proteomics\"}},\"id\":\"2079\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"2063\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"source\":{\"id\":\"2076\"}},\"id\":\"2081\",\"type\":\"CDSView\"},{\"attributes\":{\"formatter\":{\"id\":\"2171\"},\"ticker\":{\"id\":\"2068\"}},\"id\":\"2067\",\"type\":\"CategoricalAxis\"}],\"root_ids\":[\"2060\"]},\"title\":\"Bokeh Application\",\"version\":\"2.0.1\"}};\n",
       "  var render_items = [{\"docid\":\"9f2e5281-9a70-4b3c-bcaa-e73a352ccf53\",\"root_ids\":[\"2060\"],\"roots\":{\"2060\":\"86709e16-d9e2-4f77-b479-9399661054cf\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "2060"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "p.plotCircleHeatMap(plot_df, circle_var = 'size', color_var='Medians', x_axis= 'Cancer', y_axis = 'Proteomics')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Keep PosNeg Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "def HasPosNeg(row):\n",
    "    hasPos = False\n",
    "    hasNeg= False\n",
    "\n",
    "    for item in row:\n",
    "        if pd.isnull(item):\n",
    "            continue\n",
    "        if item < -0.3:\n",
    "            hasNeg = True\n",
    "        if item > 0.3:\n",
    "            hasPos = True\n",
    "           \n",
    "    if hasPos & hasNeg:\n",
    "        return True\n",
    "    return False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Proteomics</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>0.511726</td>\n",
       "      <td>0.318812</td>\n",
       "      <td>0.36490</td>\n",
       "      <td>1.00800</td>\n",
       "      <td>0.169972</td>\n",
       "      <td>0.74660</td>\n",
       "      <td>0.20400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAPK3</th>\n",
       "      <td>0.665105</td>\n",
       "      <td>0.061417</td>\n",
       "      <td>-0.33845</td>\n",
       "      <td>0.22335</td>\n",
       "      <td>-0.059507</td>\n",
       "      <td>-0.08200</td>\n",
       "      <td>-0.19490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CD</th>\n",
       "      <td>0.492099</td>\n",
       "      <td>0.238171</td>\n",
       "      <td>0.84795</td>\n",
       "      <td>-0.15015</td>\n",
       "      <td>0.035308</td>\n",
       "      <td>0.18243</td>\n",
       "      <td>0.07600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2</th>\n",
       "      <td>0.312146</td>\n",
       "      <td>0.190047</td>\n",
       "      <td>0.55885</td>\n",
       "      <td>0.31470</td>\n",
       "      <td>-0.184243</td>\n",
       "      <td>0.26400</td>\n",
       "      <td>-0.01680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3C3</th>\n",
       "      <td>0.102518</td>\n",
       "      <td>0.041287</td>\n",
       "      <td>0.10100</td>\n",
       "      <td>-0.11490</td>\n",
       "      <td>-0.104687</td>\n",
       "      <td>0.13160</td>\n",
       "      <td>0.07995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP7</th>\n",
       "      <td>-0.154035</td>\n",
       "      <td>-0.094081</td>\n",
       "      <td>-0.13260</td>\n",
       "      <td>0.14305</td>\n",
       "      <td>0.017808</td>\n",
       "      <td>-0.03834</td>\n",
       "      <td>-0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAP2K1</th>\n",
       "      <td>0.592389</td>\n",
       "      <td>-0.008519</td>\n",
       "      <td>0.10475</td>\n",
       "      <td>0.03045</td>\n",
       "      <td>0.025625</td>\n",
       "      <td>0.09190</td>\n",
       "      <td>0.02675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R1</th>\n",
       "      <td>0.188390</td>\n",
       "      <td>-0.072064</td>\n",
       "      <td>-0.47740</td>\n",
       "      <td>1.03615</td>\n",
       "      <td>-0.179425</td>\n",
       "      <td>-0.33990</td>\n",
       "      <td>0.01245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INPP4B</th>\n",
       "      <td>0.341829</td>\n",
       "      <td>0.591596</td>\n",
       "      <td>-1.87510</td>\n",
       "      <td>0.37695</td>\n",
       "      <td>-0.370647</td>\n",
       "      <td>-0.16320</td>\n",
       "      <td>-0.08220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR</th>\n",
       "      <td>-1.001035</td>\n",
       "      <td>-0.467227</td>\n",
       "      <td>-2.32970</td>\n",
       "      <td>-0.58645</td>\n",
       "      <td>0.063388</td>\n",
       "      <td>-0.52600</td>\n",
       "      <td>0.02350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSNK2A1</th>\n",
       "      <td>-0.123874</td>\n",
       "      <td>-0.079700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.015518</td>\n",
       "      <td>0.10412</td>\n",
       "      <td>0.04845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XIAP</th>\n",
       "      <td>-0.088967</td>\n",
       "      <td>-0.049397</td>\n",
       "      <td>-0.52480</td>\n",
       "      <td>-0.15040</td>\n",
       "      <td>0.010591</td>\n",
       "      <td>-0.09772</td>\n",
       "      <td>0.06900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>0.105746</td>\n",
       "      <td>-0.163663</td>\n",
       "      <td>0.13565</td>\n",
       "      <td>0.27750</td>\n",
       "      <td>-0.199296</td>\n",
       "      <td>0.35300</td>\n",
       "      <td>-0.03385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>-0.256630</td>\n",
       "      <td>-0.705560</td>\n",
       "      <td>-2.76780</td>\n",
       "      <td>-0.11100</td>\n",
       "      <td>-0.196586</td>\n",
       "      <td>0.80260</td>\n",
       "      <td>-0.04905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FOXO1</th>\n",
       "      <td>-0.083073</td>\n",
       "      <td>-0.064930</td>\n",
       "      <td>0.57005</td>\n",
       "      <td>-0.14820</td>\n",
       "      <td>0.039263</td>\n",
       "      <td>-0.09000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Gbm_Median  Hnscc_Median  Luad_Median  Brca_Median  Ov_Median  \\\n",
       "Proteomics                                                                  \n",
       "PTEN          0.511726      0.318812      0.36490      1.00800   0.169972   \n",
       "MAPK3         0.665105      0.061417     -0.33845      0.22335  -0.059507   \n",
       "PIK3CD        0.492099      0.238171      0.84795     -0.15015   0.035308   \n",
       "GRB2          0.312146      0.190047      0.55885      0.31470  -0.184243   \n",
       "PIK3C3        0.102518      0.041287      0.10100     -0.11490  -0.104687   \n",
       "USP7         -0.154035     -0.094081     -0.13260      0.14305   0.017808   \n",
       "MAP2K1        0.592389     -0.008519      0.10475      0.03045   0.025625   \n",
       "PIK3R1        0.188390     -0.072064     -0.47740      1.03615  -0.179425   \n",
       "INPP4B        0.341829      0.591596     -1.87510      0.37695  -0.370647   \n",
       "EGFR         -1.001035     -0.467227     -2.32970     -0.58645   0.063388   \n",
       "CSNK2A1      -0.123874     -0.079700          NaN          NaN  -0.015518   \n",
       "XIAP         -0.088967     -0.049397     -0.52480     -0.15040   0.010591   \n",
       "PIK3CA        0.105746     -0.163663      0.13565      0.27750  -0.199296   \n",
       "TP53         -0.256630     -0.705560     -2.76780     -0.11100  -0.196586   \n",
       "FOXO1        -0.083073     -0.064930      0.57005     -0.14820   0.039263   \n",
       "\n",
       "            En_Median  Colon_Median  \n",
       "Proteomics                           \n",
       "PTEN          0.74660       0.20400  \n",
       "MAPK3        -0.08200      -0.19490  \n",
       "PIK3CD        0.18243       0.07600  \n",
       "GRB2          0.26400      -0.01680  \n",
       "PIK3C3        0.13160       0.07995  \n",
       "USP7         -0.03834      -0.04450  \n",
       "MAP2K1        0.09190       0.02675  \n",
       "PIK3R1       -0.33990       0.01245  \n",
       "INPP4B       -0.16320      -0.08220  \n",
       "EGFR         -0.52600       0.02350  \n",
       "CSNK2A1       0.10412       0.04845  \n",
       "XIAP         -0.09772       0.06900  \n",
       "PIK3CA        0.35300      -0.03385  \n",
       "TP53          0.80260      -0.04905  \n",
       "FOXO1        -0.09000           NaN  "
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_df = sig_df.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "\n",
    "p = sig_df.set_index('Proteomics')\n",
    "p = p.drop(columns= ['Gbm_P_Value','Hnscc_P_Value','Luad_P_Value',\n",
    "                     'Brca_P_Value','Ov_P_Value','En_P_Value','Colon_P_Value'])\n",
    "p.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  if __name__ == '__main__':\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.665105</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>1.379506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.004282</td>\n",
       "      <td>0.188390</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.545327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.014556</td>\n",
       "      <td>0.341829</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.422977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.668250</td>\n",
       "      <td>-0.256630</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.040309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.000676</td>\n",
       "      <td>0.591596</td>\n",
       "      <td>Hnscc</td>\n",
       "      <td>0.729998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.031801</td>\n",
       "      <td>-0.705560</td>\n",
       "      <td>Hnscc</td>\n",
       "      <td>0.344827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>0.129494</td>\n",
       "      <td>0.061417</td>\n",
       "      <td>Hnscc</td>\n",
       "      <td>0.204412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.434622</td>\n",
       "      <td>-0.072064</td>\n",
       "      <td>Hnscc</td>\n",
       "      <td>0.083328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.239213</td>\n",
       "      <td>-2.767800</td>\n",
       "      <td>Luad</td>\n",
       "      <td>0.143040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.276672</td>\n",
       "      <td>-1.875100</td>\n",
       "      <td>Luad</td>\n",
       "      <td>0.128492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>0.525619</td>\n",
       "      <td>-0.338450</td>\n",
       "      <td>Luad</td>\n",
       "      <td>0.064318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.648676</td>\n",
       "      <td>-0.477400</td>\n",
       "      <td>Luad</td>\n",
       "      <td>0.043282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.068427</td>\n",
       "      <td>-0.111000</td>\n",
       "      <td>Brca</td>\n",
       "      <td>0.268198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.164475</td>\n",
       "      <td>1.036150</td>\n",
       "      <td>Brca</td>\n",
       "      <td>0.180500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>0.285412</td>\n",
       "      <td>0.223350</td>\n",
       "      <td>Brca</td>\n",
       "      <td>0.125382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.295970</td>\n",
       "      <td>0.376950</td>\n",
       "      <td>Brca</td>\n",
       "      <td>0.121750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.219977</td>\n",
       "      <td>-0.179425</td>\n",
       "      <td>Ov</td>\n",
       "      <td>0.151423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.353620</td>\n",
       "      <td>-0.370647</td>\n",
       "      <td>Ov</td>\n",
       "      <td>0.103953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>0.842428</td>\n",
       "      <td>-0.059507</td>\n",
       "      <td>Ov</td>\n",
       "      <td>0.017147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.980145</td>\n",
       "      <td>-0.196586</td>\n",
       "      <td>Ov</td>\n",
       "      <td>0.002005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.001246</td>\n",
       "      <td>0.802600</td>\n",
       "      <td>En</td>\n",
       "      <td>0.668799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.135943</td>\n",
       "      <td>-0.339900</td>\n",
       "      <td>En</td>\n",
       "      <td>0.199552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>0.272111</td>\n",
       "      <td>-0.082000</td>\n",
       "      <td>En</td>\n",
       "      <td>0.130154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.489186</td>\n",
       "      <td>-0.163200</td>\n",
       "      <td>En</td>\n",
       "      <td>0.071501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>0.003390</td>\n",
       "      <td>-0.194900</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.568705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>0.118135</td>\n",
       "      <td>-0.082200</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.213593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.968870</td>\n",
       "      <td>-0.049050</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.003162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.972777</td>\n",
       "      <td>0.012450</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.002760</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Proteomics   P_Value   Medians Cancer      size\n",
       "1       MAPK3  0.000001  0.665105    Gbm  1.379506\n",
       "7      PIK3R1  0.004282  0.188390    Gbm  0.545327\n",
       "9      INPP4B  0.014556  0.341829    Gbm  0.422977\n",
       "35       TP53  0.668250 -0.256630    Gbm  0.040309\n",
       "1      INPP4B  0.000676  0.591596  Hnscc  0.729998\n",
       "9        TP53  0.031801 -0.705560  Hnscc  0.344827\n",
       "18      MAPK3  0.129494  0.061417  Hnscc  0.204412\n",
       "28     PIK3R1  0.434622 -0.072064  Hnscc  0.083328\n",
       "17       TP53  0.239213 -2.767800   Luad  0.143040\n",
       "20     INPP4B  0.276672 -1.875100   Luad  0.128492\n",
       "27      MAPK3  0.525619 -0.338450   Luad  0.064318\n",
       "29     PIK3R1  0.648676 -0.477400   Luad  0.043282\n",
       "4        TP53  0.068427 -0.111000   Brca  0.268198\n",
       "8      PIK3R1  0.164475  1.036150   Brca  0.180500\n",
       "13      MAPK3  0.285412  0.223350   Brca  0.125382\n",
       "14     INPP4B  0.295970  0.376950   Brca  0.121750\n",
       "10     PIK3R1  0.219977 -0.179425     Ov  0.151423\n",
       "14     INPP4B  0.353620 -0.370647     Ov  0.103953\n",
       "30      MAPK3  0.842428 -0.059507     Ov  0.017147\n",
       "38       TP53  0.980145 -0.196586     Ov  0.002005\n",
       "1        TP53  0.001246  0.802600     En  0.668799\n",
       "13     PIK3R1  0.135943 -0.339900     En  0.199552\n",
       "21      MAPK3  0.272111 -0.082000     En  0.130154\n",
       "26     INPP4B  0.489186 -0.163200     En  0.071501\n",
       "0       MAPK3  0.003390 -0.194900  Colon  0.568705\n",
       "6      INPP4B  0.118135 -0.082200  Colon  0.213593\n",
       "33       TP53  0.968870 -0.049050  Colon  0.003162\n",
       "34     PIK3R1  0.972777  0.012450  Colon  0.002760"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p[\"Pos_Neg\"] = p.apply(HasPosNeg, axis = 1)\n",
    "\n",
    "pn = p.loc[p['Pos_Neg'] == True]\n",
    "pn_genes = list(pn.index)\n",
    "\n",
    "get = plot_df['Proteomics'].isin(pn_genes)\n",
    "plot_df2 = plot_df[get]\n",
    "\n",
    "plot_df2['size'] = plot_df2['P_Value'].apply(lambda x: -1*(np.log(x)/10))\n",
    "plot_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:12: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  if sys.path[0] == '':\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"2260\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"2260\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"2260\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '2260' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"2260\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"2260\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"2260\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '2260' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"2260\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"58c4351f-9c30-402d-a12f-615101d468c7\" data-root-id=\"2237\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"a0243aed-e3d7-4a4e-9e1a-cb10a481a6a7\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"2244\"}],\"center\":[{\"id\":\"2246\"},{\"id\":\"2249\"}],\"left\":[{\"id\":\"2247\"}],\"plot_height\":450,\"plot_width\":700,\"renderers\":[{\"id\":\"2257\"}],\"right\":[{\"id\":\"2259\"}],\"title\":{\"id\":\"2352\"},\"toolbar\":{\"id\":\"2251\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"2235\"},\"x_scale\":{\"id\":\"2240\"},\"y_range\":{\"id\":\"2236\"},\"y_scale\":{\"id\":\"2242\"}},\"id\":\"2237\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{},\"id\":\"2242\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"formatter\":{\"id\":\"2356\"},\"ticker\":{\"id\":\"2248\"}},\"id\":\"2247\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"2360\",\"type\":\"Selection\"},{\"attributes\":{\"axis\":{\"id\":\"2247\"},\"dimension\":1,\"ticker\":null},\"id\":\"2249\",\"type\":\"Grid\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"2234\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Cancer\"},\"y\":{\"field\":\"Proteomics\"}},\"id\":\"2255\",\"type\":\"Scatter\"},{\"attributes\":{\"high\":1,\"low\":-1,\"palette\":[\"#b2182b\",\"#d6604d\",\"#f4a582\",\"#fddbc7\",\"#f7f7f7\",\"#d1e5f0\",\"#92c5de\",\"#4393c3\",\"#2166ac\"]},\"id\":\"2234\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"formatter\":{\"id\":\"2358\"},\"ticker\":{\"id\":\"2245\"}},\"id\":\"2244\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"source\":{\"id\":\"2253\"}},\"id\":\"2258\",\"type\":\"CDSView\"},{\"attributes\":{\"text\":\"\"},\"id\":\"2352\",\"type\":\"Title\"},{\"attributes\":{\"callback\":null},\"id\":\"2250\",\"type\":\"HoverTool\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"2250\"}]},\"id\":\"2251\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"2359\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"axis\":{\"id\":\"2244\"},\"ticker\":null},\"id\":\"2246\",\"type\":\"Grid\"},{\"attributes\":{\"color_mapper\":{\"id\":\"2234\"},\"formatter\":{\"id\":\"2354\"},\"location\":[0,0],\"ticker\":{\"id\":\"2353\"}},\"id\":\"2259\",\"type\":\"ColorBar\"},{\"attributes\":{},\"id\":\"2353\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"2356\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"data_source\":{\"id\":\"2253\"},\"glyph\":{\"id\":\"2255\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"2256\"},\"selection_glyph\":null,\"view\":{\"id\":\"2258\"}},\"id\":\"2257\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"2234\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Cancer\"},\"y\":{\"field\":\"Proteomics\"}},\"id\":\"2256\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"2358\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"data\":{\"Cancer\":[\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"En\",\"En\",\"En\",\"En\",\"Colon\",\"Colon\",\"Colon\",\"Colon\"],\"Medians\":{\"__ndarray__\":\"v5ZeNopI5T/IXfSVLB3IP1zsGQeG4NU/mhqfaaBs0L8gbd9YWu7iP6D1nV3zk+a/AP6xrQJyrz8A/YmtznKyv03zjlN0JAbAcayL22gA/r9iMlUwKqnVv8wQx7q4jd6/4CLb+X5qvL+kAbwFEpTwP4Tix5i7lsw/coqO5PIf2D9HE4rVaPfGvxcX0tevuNe/oFFfO713rr+acEXouynJv5Cg+DHmruk//kP67evA1b/KoUW28/20v4eFWtO848S/bsX+snvyyL/sL7snDwu1v1zTvOMUHam/YH2utmJ/iT8=\",\"dtype\":\"float64\",\"shape\":[28]},\"P_Value\":{\"__ndarray__\":\"6UTNnrMfsT6MLLUnUIpxP4SR9rxrz40/TjRz0E1i5T9j/9ZB9CJGPx9r7+UsSKA/uYgFG0OTwD8iBKsg2tDbPyc9L6mEns4/GvhR3Py00T9yLTA33tHgP9Ps8Cn1weQ/fGPV53OEsT+6KR+egg3FP1ehGFMvRNI/aMcmdyrx0j8fxFOMNijMPzhZFri0odY/WKiPCiz16j/du5lxWV3vP4RX5Fw4aVQ/6lndjZdmwT+WV9VcRWrRP/iCj+fSTt8/L3QpkHbEaz+H3/NNFj6+PzyIx5j7AO8/Cdu4Jv0g7z8=\",\"dtype\":\"float64\",\"shape\":[28]},\"Proteomics\":[\"MAPK3\",\"PIK3R1\",\"INPP4B\",\"TP53\",\"INPP4B\",\"TP53\",\"MAPK3\",\"PIK3R1\",\"TP53\",\"INPP4B\",\"MAPK3\",\"PIK3R1\",\"TP53\",\"PIK3R1\",\"MAPK3\",\"INPP4B\",\"PIK3R1\",\"INPP4B\",\"MAPK3\",\"TP53\",\"TP53\",\"PIK3R1\",\"MAPK3\",\"INPP4B\",\"MAPK3\",\"INPP4B\",\"TP53\",\"PIK3R1\"],\"index\":[1,7,9,35,1,9,18,28,17,20,27,29,4,8,13,14,10,14,30,38,1,13,21,26,0,6,33,34],\"size\":{\"__ndarray__\":\"K0HTZWs+UUBvmXMrLkQ7QPOsthUaJjVAIRnN7qsfAEDiSRJp/D9CQBdKkZfIPTFAUiQJivJwJEBtAJv0YaoQQM5sglGpmxxAUiLdj8+yGUD7FbH9J7oJQMt3a8UWUAFAxZE0eODRKkDhgSZvywwiQHrT1XOSExlAVhhnsJdZGECapwdQ3UgeQL5ondRoyhRAhmd600dv6z8sMj3vequ5P0dINpZPuEBA48yR/YX0I0Daaz4Q6AcaQDsgrja6mQxAOSrpXGxvPEBzNi75+VslQAbxEf1qPcQ/D+1Gyg6qwT8=\",\"dtype\":\"float64\",\"shape\":[28]}},\"selected\":{\"id\":\"2360\"},\"selection_policy\":{\"id\":\"2359\"}},\"id\":\"2253\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"2245\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"factors\":[\"TP53\",\"PIK3R1\",\"MAPK3\",\"INPP4B\"]},\"id\":\"2236\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"2354\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"2248\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"2240\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"factors\":[\"Brca\",\"Colon\",\"En\",\"Gbm\",\"Hnscc\",\"Luad\",\"Ov\"]},\"id\":\"2235\",\"type\":\"FactorRange\"}],\"root_ids\":[\"2237\"]},\"title\":\"Bokeh Application\",\"version\":\"2.0.1\"}};\n",
       "  var render_items = [{\"docid\":\"a0243aed-e3d7-4a4e-9e1a-cb10a481a6a7\",\"root_ids\":[\"2237\"],\"roots\":{\"2237\":\"58c4351f-9c30-402d-a12f-615101d468c7\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "2237"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "p.plotCircleHeatMap(plot_df2, circle_var = 'size', color_var='Medians', x_axis= 'Cancer', y_axis = 'Proteomics')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Extras"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"1916\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"1916\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"1916\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '1916' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"1916\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"1916\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"1916\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '1916' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"1916\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"34a461a6-7205-4620-9241-9b48de509605\" data-root-id=\"1893\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"4682f99c-b32c-4655-8281-4f4f48d603a3\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"1900\"}],\"center\":[{\"id\":\"1902\"},{\"id\":\"1905\"}],\"left\":[{\"id\":\"1903\"}],\"plot_height\":450,\"plot_width\":700,\"renderers\":[{\"id\":\"1913\"}],\"right\":[{\"id\":\"1915\"}],\"title\":{\"id\":\"1988\"},\"toolbar\":{\"id\":\"1907\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"1891\"},\"x_scale\":{\"id\":\"1896\"},\"y_range\":{\"id\":\"1892\"},\"y_scale\":{\"id\":\"1898\"}},\"id\":\"1893\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"data_source\":{\"id\":\"1909\"},\"glyph\":{\"id\":\"1911\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1912\"},\"selection_glyph\":null,\"view\":{\"id\":\"1914\"}},\"id\":\"1913\",\"type\":\"GlyphRenderer\"},{\"attributes\":{},\"id\":\"1989\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"1904\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"1996\",\"type\":\"Selection\"},{\"attributes\":{\"high\":1,\"low\":-1,\"palette\":[\"#b2182b\",\"#d6604d\",\"#f4a582\",\"#fddbc7\",\"#f7f7f7\",\"#d1e5f0\",\"#92c5de\",\"#4393c3\",\"#2166ac\"]},\"id\":\"1890\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"data\":{\"Cancer\":[\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\"],\"Medians\":{\"__ndarray__\":\"jGwIKw5g4D+/ll42ikjlPygHUt6Oft8/HckqfTH60z9ksWq9nT66P4iMpUNrt8O/5kywOtr04j/IXfSVLB3IP1zsGQeG4NU/IvwGUT0E8L8pj79tN7a/v2qEMUmGxra/z58OTSQSuz+aGp9poGzQv6AnYJVMRLW/gGyKmWtn1D8gbd9YWu7iP4Cm1rLr8sS/AImKGGZ8zj8AA6CbDefdv6D1nV3zk+a/AK88/DpntL8AO8i1rxW4v4BOdDF3U8g/AFT15ZtKqb8A/rGtAnKvPwBS7sSDI6U/AP2Jrc5ysr8AmB3NkHKBvwDPteo8n7C/+u3rwDmjAsAYldQJaCLrPwkbnl4py+C/rrZif9k94j/E0ytlGeLhP5jdk4eFWtc/QOjZrPpcwT9N845TdCQGwDeJQWDl0Lo/cayL22gA/r9+jLlrCfnAv2IyVTAqqdW/QmDl0CLbuT/MEMe6uI3ev1TjpZvEIPA/h6dXyjLE4r/cRgN4CyTUP+Ai2/l+ary/pAG8BRKU8D+E4seYu5bMP3KKjuTyH9g/VMGopE5Aw79gdk8eFmq9v8L1KFyPwtE/8B2n6Egunz+ASL99HTjDv5wzorQ3+MK/rK3YX3ZPwj8AAAAAAAD4/0tthnSowcU/MgA2G4mCyb8YeL4ERpXHv0cTitVo98a/FxfS16+417/uQD2AvMy6v5BX2u6NPZo/AHB4g2A8kj8AfJRqz8ePv1DVde97sIU/rCmbAjE6sD+gUV87vXeuvzAif6BEGqQ/vn554ugToj+acEXouynJv/kx5q4l5Oc/kKD4Meau6T9kO99PjZfWP4EJ3Lqbp7o/okW28/3U4L8qqRPQRNjAP3yW58HdWcc//kP67evA1b8M16NwPQq3v0w3iUFg5dA/yqFFtvP9tL+HhVrTvOPEvxrdQexMoaO/SZ2AJsKGtz9ORSqMLQS5v27F/rJ78si/6iYxCKwcyj/zsFBrmne0P+wvuycPC7W/2/l+arx0sz+Qfvs6cM6oP2QQWDm0yKa/aW/whclUob+g8dJNYhCYP0SLbOf7qbE/MQisHFpkmz8UjErqBDSRv1zTvOMUHam/YH2utmJ/iT8AAAAAAAD4/w==\",\"dtype\":\"float64\",\"shape\":[104]},\"P_Value\":{\"__ndarray__\":\"oqcy7D9egD7pRM2esx+xPiRRp7iM41A/mPRJMpbbUT90yliQABZUP+mDmML1dHA/iJMgUnzzcD+MLLUnUIpxP4SR9rxrz40/LSuaISGJpD+eMmgInbGsP5Dy+NLBU74/19G/XThp0j9ONHPQTWLlP/HHi6vzhOk/A0h5FIH29z5j/9ZB9CJGP3ZhO3BFeFs/WutnzoK0jD8dH5YB2iSSPx9r7+UsSKA/qI1QaKjHpD+ZG4OhfnuwPy2EoCgdz7o/Hrl8bAvguj+5iAUbQ5PAPxQyeaCfBtQ/IgSrINrQ2z9Nu/rii5rhP/zps7PuPeI/xy2F6QWbrj7sxBA50UM8P4YF0f7md3E/Ad3biXIcdz8+dm27oVR9PwfzGyeYZI0/toutBCSNsz8nPS+phJ7OP4EEkzWaD88/GvhR3Py00T/++I8+48bXP3ItMDfe0eA/7sQL9byS4T/T7PAp9cHkPyog947hItI+KH+yGDEDkT+//qlHF3+vP3xj1edzhLE/uikfnoINxT9XoRhTL0TSP2jHJncq8dI/xOu7MECQ2D/kQqi2MnzcP1RJRaO5dd0/us5ucu9A6D+UtsmIQcbqP+Du27PDMO0/VkVWnBS97T8AAAAAAAD4f66Dwe6Yy00/2q3gLNwRgT9CAeLBJCC2Px/EU4w2KMw/OFkWuLSh1j/S7OIhDHTcP7bN4iBGN94/PspbSV3N4D8Yzdd55ZbiP1m707K96+U/AajQZcPY6j9YqI8KLPXqP7vXqELYC+w/jxB0H/YD7j/du5lxWV3vP0hDNks7+UQ/hFfkXDhpVD8xf9M4Q9xcP4qdAevGXGM/vnq0L8wZeT+9Qz/YLryiP3+DkjBTSqc/6lndjZdmwT96nxO8DsDBP7wGihavfsQ/llfVXEVq0T/4go/n0k7fP7TlZnLsUeI/Hl85CRig5T8Usxk0Xz7tPy90KZB2xGs/fgFuFIxcij91bvHCm829P4ff800WPr4/XSiA7CzLwD/l9vw8hnzCP4JGoUZ2ZM4/Wif/0EVO1T9e33NXDnvXP3rrI35AiuA/XsWbGd/x5j98oSaVWXrqPzyIx5j7AO8/Cdu4Jv0g7z8AAAAAAAD4fw==\",\"dtype\":\"float64\",\"shape\":[104]},\"Proteomics\":[\"PTEN\",\"MAPK3\",\"PIK3CD\",\"GRB2\",\"PIK3C3\",\"USP7\",\"MAP2K1\",\"PIK3R1\",\"INPP4B\",\"EGFR\",\"CSNK2A1\",\"XIAP\",\"PIK3CA\",\"TP53\",\"FOXO1\",\"PTEN\",\"INPP4B\",\"PIK3CA\",\"PIK3CD\",\"EGFR\",\"TP53\",\"CSNK2A1\",\"USP7\",\"GRB2\",\"XIAP\",\"MAPK3\",\"PIK3C3\",\"PIK3R1\",\"MAP2K1\",\"FOXO1\",\"EGFR\",\"PIK3CD\",\"XIAP\",\"FOXO1\",\"GRB2\",\"PTEN\",\"PIK3CA\",\"TP53\",\"MAP2K1\",\"INPP4B\",\"USP7\",\"MAPK3\",\"PIK3C3\",\"PIK3R1\",\"PTEN\",\"EGFR\",\"GRB2\",\"TP53\",\"PIK3R1\",\"MAPK3\",\"INPP4B\",\"XIAP\",\"PIK3C3\",\"PIK3CA\",\"MAP2K1\",\"PIK3CD\",\"FOXO1\",\"USP7\",\"CSNK2A1\",\"PTEN\",\"PIK3CA\",\"GRB2\",\"PIK3R1\",\"INPP4B\",\"PIK3C3\",\"MAP2K1\",\"USP7\",\"CSNK2A1\",\"XIAP\",\"EGFR\",\"MAPK3\",\"FOXO1\",\"PIK3CD\",\"TP53\",\"PTEN\",\"TP53\",\"PIK3CA\",\"CSNK2A1\",\"EGFR\",\"PIK3C3\",\"PIK3CD\",\"PIK3R1\",\"FOXO1\",\"GRB2\",\"MAPK3\",\"INPP4B\",\"USP7\",\"MAP2K1\",\"XIAP\",\"MAPK3\",\"PTEN\",\"PIK3C3\",\"INPP4B\",\"PIK3CD\",\"CSNK2A1\",\"USP7\",\"PIK3CA\",\"EGFR\",\"XIAP\",\"MAP2K1\",\"GRB2\",\"TP53\",\"PIK3R1\",\"FOXO1\"],\"index\":[0,1,2,3,4,5,6,7,9,12,14,20,26,35,37,0,1,2,5,6,9,10,11,14,15,18,25,28,31,32,0,1,2,3,4,5,9,17,18,20,22,27,28,29,0,2,3,4,8,13,14,18,20,21,29,32,35,37,41,0,2,6,10,14,16,17,20,24,27,29,30,32,35,38,0,1,2,3,4,7,8,13,14,15,21,26,29,31,38,0,1,5,6,7,8,10,15,17,22,29,31,33,34,38],\"size\":{\"__ndarray__\":\"myk9IxaWOUDF5Xn6q6BAQJcVsJ3fmjhAPdqSU+02L0CWWvs764AUQJSrspmXzh5AJ3iz+5SePUBE6SbVytYiQKhAjLVoFzFA9emazp8GSUDYp71NW8YYQHOvLukYyxFA2mwzXCwmFUCRmQiluqkpQPUes9RbnRBAiEnoH5jhL0CCGt0qbZQ9QBSytyPIXSBACDsswy/RJ0BYApWhglw3QOVnIyGWo0FAcNEeKkzhD0AYbgRG8dASQFTdoh41ASNAoKmny0nCA0BwDrMXEpEIQBAw2vG+gwBAUJsn72LTDECAPX5AAkPbP3ATvC7P+AlA16NwPQofXUB7FK5H4TJFQD4K16NwPTpAcD0K16OAPEDiehSuR/E7QB+F61G4PjJA5HoUrkchG0AUrkfhekxhQDMzMzMz8xRAuB6F61FwV0CF61G4HoUaQF2PwvUo7DBANDMzMzMzFEAfhetRuN43QDMzMzMzM0lAw/UoXI9SPUC4HoXrUXgvQD8zMzMzMxZAkML1KFznSUD3KFyPwlUmQClcj8L12DJAE65H4XoUHkB7FK5H4foWQP//////vytAZI/C9Shc+D9I4XoUrgceQKRwPQrXox1AXY/C9SicHEAAAAAAAAD4/2MFCZtL/yBAJzBCHfvtI0DTzbSznmwiQA/f0+ZJ8SFACiKcYEmIMkC62i9E8+8UQGiUmuIWgPQ/AC9szVZ+7D/gAEQLGtToP6YOFNPg8eA/HXESlOxaCUDFd2LWi80HQGulxjoLaf8/Cdbd4Rs//D/4P3bNmqgjQAvXo3A9qkJAcD0K16MQREBmZmZmZqYxQG3n+6nx0hRAzczMzMxMOkBSuB6F61EaQJHtfD81PiJAHoXrUbj+MEABAAAAAAASQGdmZmZmZipAZmZmZmZmEEBRuB6F61EgQHnpJjEIrP4/4XoUrkdhEkAlBoGVQ4sTQD4K16NwfSNAZ2ZmZmZmJEB8FK5H4foPQHA9CtejcBBAZmZmZmZmDkDgehSuR2EDQM7MzMzMzAFAFK5H4XoU+z/FzMzMzMzyP5qZmZmZmQtAZmZmZmZm9T/fehSuR+HqPyCF61G4ngNA81G4HoXr4z8AAAAAAAD4/w==\",\"dtype\":\"float64\",\"shape\":[104]}},\"selected\":{\"id\":\"1996\"},\"selection_policy\":{\"id\":\"1995\"}},\"id\":\"1909\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1890\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Cancer\"},\"y\":{\"field\":\"Proteomics\"}},\"id\":\"1911\",\"type\":\"Scatter\"},{\"attributes\":{\"factors\":[\"Brca\",\"Colon\",\"En\",\"Gbm\",\"Hnscc\",\"Luad\",\"Ov\"]},\"id\":\"1891\",\"type\":\"FactorRange\"},{\"attributes\":{\"axis\":{\"id\":\"1900\"},\"ticker\":null},\"id\":\"1902\",\"type\":\"Grid\"},{\"attributes\":{\"formatter\":{\"id\":\"1992\"},\"ticker\":{\"id\":\"1904\"}},\"id\":\"1903\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"1990\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"1898\",\"type\":\"CategoricalScale\"},{\"attributes\":{},\"id\":\"1992\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"callback\":null},\"id\":\"1906\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"1995\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"1896\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"axis\":{\"id\":\"1903\"},\"dimension\":1,\"ticker\":null},\"id\":\"1905\",\"type\":\"Grid\"},{\"attributes\":{\"formatter\":{\"id\":\"1994\"},\"ticker\":{\"id\":\"1901\"}},\"id\":\"1900\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"1901\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"1994\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"1906\"}]},\"id\":\"1907\",\"type\":\"Toolbar\"},{\"attributes\":{\"source\":{\"id\":\"1909\"}},\"id\":\"1914\",\"type\":\"CDSView\"},{\"attributes\":{\"text\":\"\"},\"id\":\"1988\",\"type\":\"Title\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1890\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Cancer\"},\"y\":{\"field\":\"Proteomics\"}},\"id\":\"1912\",\"type\":\"Scatter\"},{\"attributes\":{\"factors\":[\"XIAP\",\"USP7\",\"TP53\",\"PTEN\",\"PIK3R1\",\"PIK3CD\",\"PIK3CA\",\"PIK3C3\",\"MAPK3\",\"MAP2K1\",\"INPP4B\",\"GRB2\",\"FOXO1\",\"EGFR\",\"CSNK2A1\"]},\"id\":\"1892\",\"type\":\"FactorRange\"},{\"attributes\":{\"color_mapper\":{\"id\":\"1890\"},\"formatter\":{\"id\":\"1990\"},\"location\":[0,0],\"ticker\":{\"id\":\"1989\"}},\"id\":\"1915\",\"type\":\"ColorBar\"}],\"root_ids\":[\"1893\"]},\"title\":\"Bokeh Application\",\"version\":\"2.0.1\"}};\n",
       "  var render_items = [{\"docid\":\"4682f99c-b32c-4655-8281-4f4f48d603a3\",\"root_ids\":[\"1893\"],\"roots\":{\"1893\":\"34a461a6-7205-4620-9241-9b48de509605\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "1893"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plotCircleHeatMap(plot_df, circle_var = 'Medians', color_var='Medians', x_axis= 'Cancer', y_axis = 'Proteomics')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Medians</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>1.219523e-07</td>\n",
       "      <td>0.511726</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAPK3</td>\n",
       "      <td>1.020660e-06</td>\n",
       "      <td>0.665105</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CD</td>\n",
       "      <td>1.030815e-03</td>\n",
       "      <td>0.492099</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>1.089951e-03</td>\n",
       "      <td>0.312146</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3C3</td>\n",
       "      <td>1.225949e-03</td>\n",
       "      <td>0.102518</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>0.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>7.170253e-01</td>\n",
       "      <td>0.026750</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>8.274353e-01</td>\n",
       "      <td>-0.016800</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TP53</td>\n",
       "      <td>9.688700e-01</td>\n",
       "      <td>-0.049050</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>9.727770e-01</td>\n",
       "      <td>0.012450</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>FOXO1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Colon</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>104 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Proteomics       P_Value   Medians Cancer  size\n",
       "0        PTEN  1.219523e-07  0.511726    Gbm  0.75\n",
       "1       MAPK3  1.020660e-06  0.665105    Gbm  0.75\n",
       "2      PIK3CD  1.030815e-03  0.492099    Gbm  0.75\n",
       "3        GRB2  1.089951e-03  0.312146    Gbm  0.50\n",
       "4      PIK3C3  1.225949e-03  0.102518    Gbm  0.50\n",
       "..        ...           ...       ...    ...   ...\n",
       "29     MAP2K1  7.170253e-01  0.026750  Colon  0.00\n",
       "31       GRB2  8.274353e-01 -0.016800  Colon  0.00\n",
       "33       TP53  9.688700e-01 -0.049050  Colon  0.00\n",
       "34     PIK3R1  9.727770e-01  0.012450  Colon  0.00\n",
       "38      FOXO1           NaN       NaN  Colon  0.00\n",
       "\n",
       "[104 rows x 5 columns]"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Code for only keeping data for sig pvals\n",
    "# 0 for not sig (no circle)\n",
    "bc = .05/46\n",
    "c = .05/36\n",
    "r = .01\n",
    "\n",
    "plot_df['size'] = plot_df['P_Value'].apply(lambda x: .75 if x <= bc else (.5 if x <= c else (.25 if x <= r else 0)))\n",
    "\n",
    "plot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"1231\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"1231\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"1231\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '1231' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"1231\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"1231\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"1231\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '1231' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"1231\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"ae3d58e6-e6a0-418d-818d-87959f9e9ccc\" data-root-id=\"1208\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"2071ddc2-956d-45f5-9035-ca78be7be4b4\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"1215\"}],\"center\":[{\"id\":\"1217\"},{\"id\":\"1220\"}],\"left\":[{\"id\":\"1218\"}],\"plot_height\":450,\"plot_width\":700,\"renderers\":[{\"id\":\"1228\"}],\"right\":[{\"id\":\"1230\"}],\"title\":{\"id\":\"1253\"},\"toolbar\":{\"id\":\"1222\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"1206\"},\"x_scale\":{\"id\":\"1211\"},\"y_range\":{\"id\":\"1207\"},\"y_scale\":{\"id\":\"1213\"}},\"id\":\"1208\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"data_source\":{\"id\":\"1224\"},\"glyph\":{\"id\":\"1226\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1227\"},\"selection_glyph\":null,\"view\":{\"id\":\"1229\"}},\"id\":\"1228\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"axis\":{\"id\":\"1215\"},\"ticker\":null},\"id\":\"1217\",\"type\":\"Grid\"},{\"attributes\":{\"factors\":[\"Brca\",\"Colon\",\"En\",\"Gbm\",\"Hnscc\",\"Luad\",\"Ov\"]},\"id\":\"1206\",\"type\":\"FactorRange\"},{\"attributes\":{\"formatter\":{\"id\":\"1257\"},\"ticker\":{\"id\":\"1219\"}},\"id\":\"1218\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"high\":1,\"low\":-1,\"palette\":[\"#b2182b\",\"#d6604d\",\"#f4a582\",\"#fddbc7\",\"#f7f7f7\",\"#d1e5f0\",\"#92c5de\",\"#4393c3\",\"#2166ac\"]},\"id\":\"1205\",\"type\":\"LinearColorMapper\"},{\"attributes\":{},\"id\":\"1254\",\"type\":\"BasicTicker\"},{\"attributes\":{\"formatter\":{\"id\":\"1259\"},\"ticker\":{\"id\":\"1216\"}},\"id\":\"1215\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"1216\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"1257\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"1221\"}]},\"id\":\"1222\",\"type\":\"Toolbar\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1205\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Cancer\"},\"y\":{\"field\":\"Proteomics\"}},\"id\":\"1227\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"1255\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"1219\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"data\":{\"Cancer\":[\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Brca\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"Ov\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"En\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\",\"Colon\"],\"Medians\":{\"__ndarray__\":\"jGwIKw5g4D+/ll42ikjlPygHUt6Oft8/HckqfTH60z9ksWq9nT66P4iMpUNrt8O/5kywOtr04j/IXfSVLB3IP1zsGQeG4NU/IvwGUT0E8L8pj79tN7a/v2qEMUmGxra/z58OTSQSuz+aGp9poGzQv6AnYJVMRLW/gGyKmWtn1D8gbd9YWu7iP4Cm1rLr8sS/AImKGGZ8zj8AA6CbDefdv6D1nV3zk+a/AK88/DpntL8AO8i1rxW4v4BOdDF3U8g/AFT15ZtKqb8A/rGtAnKvPwBS7sSDI6U/AP2Jrc5ysr8AmB3NkHKBvwDPteo8n7C/+u3rwDmjAsAYldQJaCLrPwkbnl4py+C/rrZif9k94j/E0ytlGeLhP5jdk4eFWtc/QOjZrPpcwT9N845TdCQGwDeJQWDl0Lo/cayL22gA/r9+jLlrCfnAv2IyVTAqqdW/QmDl0CLbuT/MEMe6uI3ev1TjpZvEIPA/h6dXyjLE4r/cRgN4CyTUP+Ai2/l+ary/pAG8BRKU8D+E4seYu5bMP3KKjuTyH9g/VMGopE5Aw79gdk8eFmq9v8L1KFyPwtE/8B2n6Egunz+ASL99HTjDv5wzorQ3+MK/rK3YX3ZPwj8AAAAAAAD4/0tthnSowcU/MgA2G4mCyb8YeL4ERpXHv0cTitVo98a/FxfS16+417/uQD2AvMy6v5BX2u6NPZo/AHB4g2A8kj8AfJRqz8ePv1DVde97sIU/rCmbAjE6sD+gUV87vXeuvzAif6BEGqQ/vn554ugToj+acEXouynJv/kx5q4l5Oc/kKD4Meau6T9kO99PjZfWP4EJ3Lqbp7o/okW28/3U4L8qqRPQRNjAP3yW58HdWcc//kP67evA1b8M16NwPQq3v0w3iUFg5dA/yqFFtvP9tL+HhVrTvOPEvxrdQexMoaO/SZ2AJsKGtz9ORSqMLQS5v27F/rJ78si/6iYxCKwcyj/zsFBrmne0P+wvuycPC7W/2/l+arx0sz+Qfvs6cM6oP2QQWDm0yKa/aW/whclUob+g8dJNYhCYP0SLbOf7qbE/MQisHFpkmz8UjErqBDSRv1zTvOMUHam/YH2utmJ/iT8AAAAAAAD4/w==\",\"dtype\":\"float64\",\"shape\":[104]},\"P_Value\":{\"__ndarray__\":\"oqcy7D9egD7pRM2esx+xPiRRp7iM41A/mPRJMpbbUT90yliQABZUP+mDmML1dHA/iJMgUnzzcD+MLLUnUIpxP4SR9rxrz40/LSuaISGJpD+eMmgInbGsP5Dy+NLBU74/19G/XThp0j9ONHPQTWLlP/HHi6vzhOk/A0h5FIH29z5j/9ZB9CJGP3ZhO3BFeFs/WutnzoK0jD8dH5YB2iSSPx9r7+UsSKA/qI1QaKjHpD+ZG4OhfnuwPy2EoCgdz7o/Hrl8bAvguj+5iAUbQ5PAPxQyeaCfBtQ/IgSrINrQ2z9Nu/rii5rhP/zps7PuPeI/xy2F6QWbrj7sxBA50UM8P4YF0f7md3E/Ad3biXIcdz8+dm27oVR9PwfzGyeYZI0/toutBCSNsz8nPS+phJ7OP4EEkzWaD88/GvhR3Py00T/++I8+48bXP3ItMDfe0eA/7sQL9byS4T/T7PAp9cHkPyog947hItI+KH+yGDEDkT+//qlHF3+vP3xj1edzhLE/uikfnoINxT9XoRhTL0TSP2jHJncq8dI/xOu7MECQ2D/kQqi2MnzcP1RJRaO5dd0/us5ucu9A6D+UtsmIQcbqP+Du27PDMO0/VkVWnBS97T8AAAAAAAD4f66Dwe6Yy00/2q3gLNwRgT9CAeLBJCC2Px/EU4w2KMw/OFkWuLSh1j/S7OIhDHTcP7bN4iBGN94/PspbSV3N4D8Yzdd55ZbiP1m707K96+U/AajQZcPY6j9YqI8KLPXqP7vXqELYC+w/jxB0H/YD7j/du5lxWV3vP0hDNks7+UQ/hFfkXDhpVD8xf9M4Q9xcP4qdAevGXGM/vnq0L8wZeT+9Qz/YLryiP3+DkjBTSqc/6lndjZdmwT96nxO8DsDBP7wGihavfsQ/llfVXEVq0T/4go/n0k7fP7TlZnLsUeI/Hl85CRig5T8Usxk0Xz7tPy90KZB2xGs/fgFuFIxcij91bvHCm829P4ff800WPr4/XSiA7CzLwD/l9vw8hnzCP4JGoUZ2ZM4/Wif/0EVO1T9e33NXDnvXP3rrI35AiuA/XsWbGd/x5j98oSaVWXrqPzyIx5j7AO8/Cdu4Jv0g7z8AAAAAAAD4fw==\",\"dtype\":\"float64\",\"shape\":[104]},\"Proteomics\":[\"PTEN\",\"MAPK3\",\"PIK3CD\",\"GRB2\",\"PIK3C3\",\"USP7\",\"MAP2K1\",\"PIK3R1\",\"INPP4B\",\"EGFR\",\"CSNK2A1\",\"XIAP\",\"PIK3CA\",\"TP53\",\"FOXO1\",\"PTEN\",\"INPP4B\",\"PIK3CA\",\"PIK3CD\",\"EGFR\",\"TP53\",\"CSNK2A1\",\"USP7\",\"GRB2\",\"XIAP\",\"MAPK3\",\"PIK3C3\",\"PIK3R1\",\"MAP2K1\",\"FOXO1\",\"EGFR\",\"PIK3CD\",\"XIAP\",\"FOXO1\",\"GRB2\",\"PTEN\",\"PIK3CA\",\"TP53\",\"MAP2K1\",\"INPP4B\",\"USP7\",\"MAPK3\",\"PIK3C3\",\"PIK3R1\",\"PTEN\",\"EGFR\",\"GRB2\",\"TP53\",\"PIK3R1\",\"MAPK3\",\"INPP4B\",\"XIAP\",\"PIK3C3\",\"PIK3CA\",\"MAP2K1\",\"PIK3CD\",\"FOXO1\",\"USP7\",\"CSNK2A1\",\"PTEN\",\"PIK3CA\",\"GRB2\",\"PIK3R1\",\"INPP4B\",\"PIK3C3\",\"MAP2K1\",\"USP7\",\"CSNK2A1\",\"XIAP\",\"EGFR\",\"MAPK3\",\"FOXO1\",\"PIK3CD\",\"TP53\",\"PTEN\",\"TP53\",\"PIK3CA\",\"CSNK2A1\",\"EGFR\",\"PIK3C3\",\"PIK3CD\",\"PIK3R1\",\"FOXO1\",\"GRB2\",\"MAPK3\",\"INPP4B\",\"USP7\",\"MAP2K1\",\"XIAP\",\"MAPK3\",\"PTEN\",\"PIK3C3\",\"INPP4B\",\"PIK3CD\",\"CSNK2A1\",\"USP7\",\"PIK3CA\",\"EGFR\",\"XIAP\",\"MAP2K1\",\"GRB2\",\"TP53\",\"PIK3R1\",\"FOXO1\"],\"index\":[0,1,2,3,4,5,6,7,9,12,14,20,26,35,37,0,1,2,5,6,9,10,11,14,15,18,25,28,31,32,0,1,2,3,4,5,9,17,18,20,22,27,28,29,0,2,3,4,8,13,14,18,20,21,29,32,35,37,41,0,2,6,10,14,16,17,20,24,27,29,30,32,35,38,0,1,2,3,4,7,8,13,14,15,21,26,29,31,38,0,1,5,6,7,8,10,15,17,22,29,31,33,34,38],\"size\":{\"__ndarray__\":\"AAAAAADAQkAAAAAAAMBCQAAAAAAAwEJAAAAAAAAAOUAAAAAAAAA5QAAAAAAAAClAAAAAAAAAKUAAAAAAAAApQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAQkAAAAAAAMBCQAAAAAAAAClAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAQkAAAAAAAMBCQAAAAAAAAClAAAAAAAAAKUAAAAAAAAApQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwEJAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwEJAAAAAAAAAKUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwEJAAAAAAAAAOUAAAAAAAAApQAAAAAAAAClAAAAAAAAAKUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAClAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==\",\"dtype\":\"float64\",\"shape\":[104]}},\"selected\":{\"id\":\"1261\"},\"selection_policy\":{\"id\":\"1260\"}},\"id\":\"1224\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"color_mapper\":{\"id\":\"1205\"},\"formatter\":{\"id\":\"1255\"},\"location\":[0,0],\"ticker\":{\"id\":\"1254\"}},\"id\":\"1230\",\"type\":\"ColorBar\"},{\"attributes\":{\"text\":\"\"},\"id\":\"1253\",\"type\":\"Title\"},{\"attributes\":{},\"id\":\"1261\",\"type\":\"Selection\"},{\"attributes\":{},\"id\":\"1213\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"callback\":null},\"id\":\"1221\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"1211\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"source\":{\"id\":\"1224\"}},\"id\":\"1229\",\"type\":\"CDSView\"},{\"attributes\":{\"axis\":{\"id\":\"1218\"},\"dimension\":1,\"ticker\":null},\"id\":\"1220\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"1259\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{},\"id\":\"1260\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"factors\":[\"XIAP\",\"USP7\",\"TP53\",\"PTEN\",\"PIK3R1\",\"PIK3CD\",\"PIK3CA\",\"PIK3C3\",\"MAPK3\",\"MAP2K1\",\"INPP4B\",\"GRB2\",\"FOXO1\",\"EGFR\",\"CSNK2A1\"]},\"id\":\"1207\",\"type\":\"FactorRange\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"1205\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Cancer\"},\"y\":{\"field\":\"Proteomics\"}},\"id\":\"1226\",\"type\":\"Scatter\"}],\"root_ids\":[\"1208\"]},\"title\":\"Bokeh Application\",\"version\":\"2.0.1\"}};\n",
       "  var render_items = [{\"docid\":\"2071ddc2-956d-45f5-9035-ca78be7be4b4\",\"root_ids\":[\"1208\"],\"roots\":{\"1208\":\"ae3d58e6-e6a0-418d-818d-87959f9e9ccc\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "1208"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "p.plotCircleHeatMap(plot_df, circle_var = 'size', color_var='Medians', x_axis= 'Cancer', y_axis = 'Proteomics')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_df.to_csv('at_least_one_significant_interacting.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = ip_df.std()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = stdev.to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev.loc[stdev[0] > .2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d_stdev = stdev.std()\n",
    "d_stdev"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = stdev.hist(bins = 20)\n",
    "plt.title(\"Standard Deviations of Pancancer CNV deletions and amplifications of PTEN on Interacting Proteins\\n\")\n",
    "plt.xlabel('StDeV')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(10,8)\n",
    "sns.set(font_scale = 1.3)\n",
    "a = sns.distplot(stdev, bins = 20)\n",
    "\n",
    "a.set_title(\"Standard Deviation Distribution for Pancancer PTEN Mutation Effect on Proteomics\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "a.set(xlabel = 'StDeV', ylabel = 'Frequency')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_df = stdev.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = mean_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Mean between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Mean')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m_df = stdev.median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = m_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Medians between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Median')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
