{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Interacting Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "\n",
    "import sys\n",
    "sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\WhenMutationsDontMatter\\\\')\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that endometrial index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "gbm = cptac.Gbm()\n",
    "endo = cptac.Endometrial()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11141"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = gbm.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "len(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'PTEN'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Deletion          81\n",
       "Wildtype_Tumor    16\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type_gbm = gbm.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = gbm.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type_gbm = mut_type_gbm[['Mutation']] \n",
    "merged = ip_df.join(mut_type_gbm) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3</td>\n",
       "      <td>5.396032e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC</td>\n",
       "      <td>1.593480e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A</td>\n",
       "      <td>1.009419e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2</td>\n",
       "      <td>1.122076e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2</td>\n",
       "      <td>1.302273e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11136</th>\n",
       "      <td>ZNF805</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11137</th>\n",
       "      <td>ZNF813</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11138</th>\n",
       "      <td>ZNF814</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11139</th>\n",
       "      <td>ZNF888</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11140</th>\n",
       "      <td>ZSCAN12</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11141 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics   Gbm_P_Value\n",
       "0          ARMH3  5.396032e-11\n",
       "1           CUTC  1.593480e-10\n",
       "2        PIP4K2A  1.009419e-09\n",
       "3           CUL2  1.122076e-09\n",
       "4           GDI2  1.302273e-09\n",
       "...          ...           ...\n",
       "11136     ZNF805           NaN\n",
       "11137     ZNF813           NaN\n",
       "11138     ZNF814           NaN\n",
       "11139     ZNF888           NaN\n",
       "11140    ZSCAN12           NaN\n",
       "\n",
       "[11141 rows x 2 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "g = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "g_pval = g.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Gbm_P_Value'})\n",
    "g_pval = g_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "g_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.245016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.680437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>-0.128834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AACS</td>\n",
       "      <td>0.005207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>0.200270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11136</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.010868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11137</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>-0.185994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11138</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>-0.314468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11139</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.131491</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11140</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.153449</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11141 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics  Gbm_Median\n",
       "0           A1BG    0.245016\n",
       "1            A2M    0.680437\n",
       "2           AAAS   -0.128834\n",
       "3           AACS    0.005207\n",
       "4          AADAT    0.200270\n",
       "...          ...         ...\n",
       "11136       ZXDC    0.010868\n",
       "11137     ZYG11B   -0.185994\n",
       "11138        ZYX   -0.314468\n",
       "11139      ZZEF1    0.131491\n",
       "11140       ZZZ3    0.153449\n",
       "\n",
       "[11141 rows x 2 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "gbm_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif_gbm = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    gbm_d[prot] = dif_gbm\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif_gbm)\n",
    "\n",
    "gbm_df = pd.DataFrame.from_dict(gbm_d, orient='index', columns=['Gbm_Median'])\n",
    "gbm_df = gbm_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "gbm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3</td>\n",
       "      <td>5.396032e-11</td>\n",
       "      <td>0.405134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC</td>\n",
       "      <td>1.593480e-10</td>\n",
       "      <td>0.553255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A</td>\n",
       "      <td>1.009419e-09</td>\n",
       "      <td>0.838882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2</td>\n",
       "      <td>1.122076e-09</td>\n",
       "      <td>0.586396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2</td>\n",
       "      <td>1.302273e-09</td>\n",
       "      <td>0.610188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11136</th>\n",
       "      <td>ZNF805</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11137</th>\n",
       "      <td>ZNF813</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11138</th>\n",
       "      <td>ZNF814</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.265936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11139</th>\n",
       "      <td>ZNF888</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11140</th>\n",
       "      <td>ZSCAN12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11141 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics   Gbm_P_Value  Gbm_Median\n",
       "0          ARMH3  5.396032e-11    0.405134\n",
       "1           CUTC  1.593480e-10    0.553255\n",
       "2        PIP4K2A  1.009419e-09    0.838882\n",
       "3           CUL2  1.122076e-09    0.586396\n",
       "4           GDI2  1.302273e-09    0.610188\n",
       "...          ...           ...         ...\n",
       "11136     ZNF805           NaN         NaN\n",
       "11137     ZNF813           NaN         NaN\n",
       "11138     ZNF814           NaN    0.265936\n",
       "11139     ZNF888           NaN         NaN\n",
       "11140    ZSCAN12           NaN         NaN\n",
       "\n",
       "[11141 rows x 3 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_merged = g_pval.merge(gbm_df, on='Proteomics',how='inner')\n",
    "g_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Endo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10999"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = endo.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "len(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 20 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 69 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Truncation        53\n",
       "Wildtype_Tumor    15\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Endo ttest\n",
    "\n",
    "mut_type = endo.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = endo.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Nonsense_Mutation','Frame_Shift_Ins','Frame_Shift_Del']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "trunc_wt = merged[get]\n",
    "trunc_wt['Mutation'] = np.where(\n",
    "                trunc_wt['Mutation'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "trunc_wt['Mutation'].value_counts()\n",
    "#trunc_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NOL10</td>\n",
       "      <td>8.999599e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABT1</td>\n",
       "      <td>3.141336e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UTP25</td>\n",
       "      <td>4.308134e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TOPBP1</td>\n",
       "      <td>4.885533e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RIF1</td>\n",
       "      <td>9.004982e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10994</th>\n",
       "      <td>ZNF586</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10995</th>\n",
       "      <td>ZNF630</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10996</th>\n",
       "      <td>ZNF772</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10997</th>\n",
       "      <td>ZNF79</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10998</th>\n",
       "      <td>ZNF805</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10999 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics    En_P_Value\n",
       "0          NOL10  8.999599e-08\n",
       "1           ABT1  3.141336e-07\n",
       "2          UTP25  4.308134e-07\n",
       "3         TOPBP1  4.885533e-07\n",
       "4           RIF1  9.004982e-07\n",
       "...          ...           ...\n",
       "10994     ZNF586           NaN\n",
       "10995     ZNF630           NaN\n",
       "10996     ZNF772           NaN\n",
       "10997      ZNF79           NaN\n",
       "10998     ZNF805           NaN\n",
       "\n",
       "[10999 rows x 2 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(trunc_wt.columns[:-1])\n",
    "\n",
    "e_pval = u.wrap_ttest(trunc_wt, 'Mutation', cols, return_all = True)\n",
    "e_pval = e_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'En_P_Value'})\n",
    "e_pval = e_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "e_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.06100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.07400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>-0.22300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>-0.03995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.19200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10994</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.42800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10995</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.19420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10996</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>-0.21600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10997</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-0.05320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10998</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.02120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10999 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics  En_Median\n",
       "0           A1BG    0.06100\n",
       "1            A2M   -0.07400\n",
       "2          A2ML1   -0.22300\n",
       "3         A4GALT   -0.03995\n",
       "4           AAAS    0.19200\n",
       "...          ...        ...\n",
       "10994       ZXDC    0.42800\n",
       "10995     ZYG11B    0.19420\n",
       "10996        ZYX   -0.21600\n",
       "10997      ZZEF1   -0.05320\n",
       "10998       ZZZ3    0.02120\n",
       "\n",
       "[10999 rows x 2 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = trunc_wt[trunc_wt.Mutation == \"Truncation\"]\n",
    "wt = trunc_wt[trunc_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "trunc_med = t.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "en_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - trunc_med[prot+'_proteomics']\n",
    "    en_d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', trunc_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "en_df = pd.DataFrame.from_dict(en_d, orient='index', columns=['En_Median'])\n",
    "en_df = en_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "en_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NOL10</td>\n",
       "      <td>8.999599e-08</td>\n",
       "      <td>0.44200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABT1</td>\n",
       "      <td>3.141336e-07</td>\n",
       "      <td>0.48510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UTP25</td>\n",
       "      <td>4.308134e-07</td>\n",
       "      <td>0.73450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TOPBP1</td>\n",
       "      <td>4.885533e-07</td>\n",
       "      <td>0.61795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RIF1</td>\n",
       "      <td>9.004982e-07</td>\n",
       "      <td>0.54640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10994</th>\n",
       "      <td>ZNF586</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10995</th>\n",
       "      <td>ZNF630</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.99250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10996</th>\n",
       "      <td>ZNF772</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10997</th>\n",
       "      <td>ZNF79</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10998</th>\n",
       "      <td>ZNF805</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10999 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics    En_P_Value  En_Median\n",
       "0          NOL10  8.999599e-08    0.44200\n",
       "1           ABT1  3.141336e-07    0.48510\n",
       "2          UTP25  4.308134e-07    0.73450\n",
       "3         TOPBP1  4.885533e-07    0.61795\n",
       "4           RIF1  9.004982e-07    0.54640\n",
       "...          ...           ...        ...\n",
       "10994     ZNF586           NaN        NaN\n",
       "10995     ZNF630           NaN   -0.99250\n",
       "10996     ZNF772           NaN        NaN\n",
       "10997      ZNF79           NaN        NaN\n",
       "10998     ZNF805           NaN        NaN\n",
       "\n",
       "[10999 rows x 3 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e_merged = e_pval.merge(en_df, on='Proteomics',how='inner')\n",
    "e_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                            \r"
     ]
    }
   ],
   "source": [
    "o = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>AAMDC</th>\n",
       "      <th>AAMP</th>\n",
       "      <th>AAR2</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSCAN31</th>\n",
       "      <th>ZSCAN32</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01OV007</th>\n",
       "      <td>0.133634</td>\n",
       "      <td>0.765120</td>\n",
       "      <td>-0.778849</td>\n",
       "      <td>0.118236</td>\n",
       "      <td>0.112740</td>\n",
       "      <td>0.054942</td>\n",
       "      <td>0.313055</td>\n",
       "      <td>-0.786270</td>\n",
       "      <td>-0.384680</td>\n",
       "      <td>0.064017</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.382350</td>\n",
       "      <td>-0.323516</td>\n",
       "      <td>0.074209</td>\n",
       "      <td>0.232999</td>\n",
       "      <td>-0.195793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.145053</td>\n",
       "      <td>0.590217</td>\n",
       "      <td>0.019414</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV017</th>\n",
       "      <td>-0.432786</td>\n",
       "      <td>-0.461906</td>\n",
       "      <td>-0.533497</td>\n",
       "      <td>0.041800</td>\n",
       "      <td>0.604114</td>\n",
       "      <td>0.351294</td>\n",
       "      <td>0.546605</td>\n",
       "      <td>0.603935</td>\n",
       "      <td>-0.235374</td>\n",
       "      <td>0.059084</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.008922</td>\n",
       "      <td>0.032004</td>\n",
       "      <td>0.016276</td>\n",
       "      <td>0.587911</td>\n",
       "      <td>0.574339</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.075401</td>\n",
       "      <td>-0.437564</td>\n",
       "      <td>-0.208831</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV018</th>\n",
       "      <td>-0.013943</td>\n",
       "      <td>0.065842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.010514</td>\n",
       "      <td>0.541935</td>\n",
       "      <td>0.169826</td>\n",
       "      <td>-0.023032</td>\n",
       "      <td>-0.057549</td>\n",
       "      <td>-0.397627</td>\n",
       "      <td>0.199581</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003784</td>\n",
       "      <td>0.002237</td>\n",
       "      <td>0.206993</td>\n",
       "      <td>0.111037</td>\n",
       "      <td>1.020321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.245864</td>\n",
       "      <td>0.008411</td>\n",
       "      <td>0.130504</td>\n",
       "      <td>-0.367433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV023</th>\n",
       "      <td>1.234203</td>\n",
       "      <td>1.512082</td>\n",
       "      <td>-0.703050</td>\n",
       "      <td>0.181152</td>\n",
       "      <td>0.087719</td>\n",
       "      <td>0.226861</td>\n",
       "      <td>-0.100766</td>\n",
       "      <td>-0.516022</td>\n",
       "      <td>-0.456512</td>\n",
       "      <td>-0.257629</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.277046</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.529948</td>\n",
       "      <td>0.034713</td>\n",
       "      <td>0.149523</td>\n",
       "      <td>-0.313370</td>\n",
       "      <td>-0.507473</td>\n",
       "      <td>0.221818</td>\n",
       "      <td>0.278313</td>\n",
       "      <td>-1.008998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01OV026</th>\n",
       "      <td>-0.482872</td>\n",
       "      <td>0.343872</td>\n",
       "      <td>-1.150936</td>\n",
       "      <td>0.315155</td>\n",
       "      <td>0.097235</td>\n",
       "      <td>-0.160208</td>\n",
       "      <td>-0.125165</td>\n",
       "      <td>-0.768776</td>\n",
       "      <td>-0.372368</td>\n",
       "      <td>0.450086</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.096284</td>\n",
       "      <td>-0.752819</td>\n",
       "      <td>-0.181027</td>\n",
       "      <td>0.179612</td>\n",
       "      <td>0.007854</td>\n",
       "      <td>-0.106049</td>\n",
       "      <td>0.377063</td>\n",
       "      <td>-0.295592</td>\n",
       "      <td>0.011997</td>\n",
       "      <td>-0.236003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17OV001.N</th>\n",
       "      <td>-0.012139</td>\n",
       "      <td>0.437837</td>\n",
       "      <td>-0.704253</td>\n",
       "      <td>0.411204</td>\n",
       "      <td>-0.418927</td>\n",
       "      <td>0.016434</td>\n",
       "      <td>0.504605</td>\n",
       "      <td>0.022115</td>\n",
       "      <td>-0.163317</td>\n",
       "      <td>0.290281</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.351444</td>\n",
       "      <td>0.812955</td>\n",
       "      <td>0.010818</td>\n",
       "      <td>0.321095</td>\n",
       "      <td>-0.612827</td>\n",
       "      <td>-0.149425</td>\n",
       "      <td>0.017523</td>\n",
       "      <td>0.442398</td>\n",
       "      <td>0.088551</td>\n",
       "      <td>-0.614771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17OV002.N</th>\n",
       "      <td>1.949941</td>\n",
       "      <td>1.803647</td>\n",
       "      <td>0.924203</td>\n",
       "      <td>-0.025210</td>\n",
       "      <td>0.593528</td>\n",
       "      <td>-0.427610</td>\n",
       "      <td>-0.094656</td>\n",
       "      <td>0.082444</td>\n",
       "      <td>0.065310</td>\n",
       "      <td>-0.322335</td>\n",
       "      <td>...</td>\n",
       "      <td>0.483503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.530407</td>\n",
       "      <td>-0.157267</td>\n",
       "      <td>0.716779</td>\n",
       "      <td>-0.676827</td>\n",
       "      <td>-0.569823</td>\n",
       "      <td>0.640503</td>\n",
       "      <td>0.763645</td>\n",
       "      <td>-1.370872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17OV003.N</th>\n",
       "      <td>1.523833</td>\n",
       "      <td>3.418092</td>\n",
       "      <td>0.281521</td>\n",
       "      <td>-0.148852</td>\n",
       "      <td>-0.051749</td>\n",
       "      <td>0.297993</td>\n",
       "      <td>0.160205</td>\n",
       "      <td>-0.318083</td>\n",
       "      <td>-0.382827</td>\n",
       "      <td>0.420826</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.070859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.778621</td>\n",
       "      <td>0.425679</td>\n",
       "      <td>-0.048390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.675466</td>\n",
       "      <td>0.267934</td>\n",
       "      <td>0.583772</td>\n",
       "      <td>0.501873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17OV004.N</th>\n",
       "      <td>1.091843</td>\n",
       "      <td>1.942523</td>\n",
       "      <td>3.192955</td>\n",
       "      <td>-0.439487</td>\n",
       "      <td>0.901140</td>\n",
       "      <td>-0.146974</td>\n",
       "      <td>-0.106039</td>\n",
       "      <td>0.361625</td>\n",
       "      <td>-0.475584</td>\n",
       "      <td>-0.291893</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.359314</td>\n",
       "      <td>0.537220</td>\n",
       "      <td>0.090890</td>\n",
       "      <td>-0.085341</td>\n",
       "      <td>0.952951</td>\n",
       "      <td>0.874700</td>\n",
       "      <td>-0.185089</td>\n",
       "      <td>-0.426710</td>\n",
       "      <td>0.008868</td>\n",
       "      <td>-0.278545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17OV005.N</th>\n",
       "      <td>1.511505</td>\n",
       "      <td>1.498532</td>\n",
       "      <td>2.861443</td>\n",
       "      <td>-0.002351</td>\n",
       "      <td>0.351521</td>\n",
       "      <td>-0.391236</td>\n",
       "      <td>-0.038395</td>\n",
       "      <td>0.286218</td>\n",
       "      <td>-0.159277</td>\n",
       "      <td>0.428637</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.557919</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.749876</td>\n",
       "      <td>-0.364281</td>\n",
       "      <td>0.467271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.228770</td>\n",
       "      <td>0.182717</td>\n",
       "      <td>0.298465</td>\n",
       "      <td>-0.591317</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>103 rows Ã— 10686 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name            A1BG       A2M     A2ML1      AAAS      AACS     AAGAB  \\\n",
       "Patient_ID                                                               \n",
       "01OV007     0.133634  0.765120 -0.778849  0.118236  0.112740  0.054942   \n",
       "01OV017    -0.432786 -0.461906 -0.533497  0.041800  0.604114  0.351294   \n",
       "01OV018    -0.013943  0.065842       NaN -0.010514  0.541935  0.169826   \n",
       "01OV023     1.234203  1.512082 -0.703050  0.181152  0.087719  0.226861   \n",
       "01OV026    -0.482872  0.343872 -1.150936  0.315155  0.097235 -0.160208   \n",
       "...              ...       ...       ...       ...       ...       ...   \n",
       "17OV001.N  -0.012139  0.437837 -0.704253  0.411204 -0.418927  0.016434   \n",
       "17OV002.N   1.949941  1.803647  0.924203 -0.025210  0.593528 -0.427610   \n",
       "17OV003.N   1.523833  3.418092  0.281521 -0.148852 -0.051749  0.297993   \n",
       "17OV004.N   1.091843  1.942523  3.192955 -0.439487  0.901140 -0.146974   \n",
       "17OV005.N   1.511505  1.498532  2.861443 -0.002351  0.351521 -0.391236   \n",
       "\n",
       "Name            AAK1     AAMDC      AAMP      AAR2  ...   ZSCAN31   ZSCAN32  \\\n",
       "Patient_ID                                          ...                       \n",
       "01OV007     0.313055 -0.786270 -0.384680  0.064017  ... -0.382350 -0.323516   \n",
       "01OV017     0.546605  0.603935 -0.235374  0.059084  ... -0.008922  0.032004   \n",
       "01OV018    -0.023032 -0.057549 -0.397627  0.199581  ...  0.003784  0.002237   \n",
       "01OV023    -0.100766 -0.516022 -0.456512 -0.257629  ... -0.277046       NaN   \n",
       "01OV026    -0.125165 -0.768776 -0.372368  0.450086  ... -0.096284 -0.752819   \n",
       "...              ...       ...       ...       ...  ...       ...       ...   \n",
       "17OV001.N   0.504605  0.022115 -0.163317  0.290281  ... -0.351444  0.812955   \n",
       "17OV002.N  -0.094656  0.082444  0.065310 -0.322335  ...  0.483503       NaN   \n",
       "17OV003.N   0.160205 -0.318083 -0.382827  0.420826  ... -0.070859       NaN   \n",
       "17OV004.N  -0.106039  0.361625 -0.475584 -0.291893  ... -0.359314  0.537220   \n",
       "17OV005.N  -0.038395  0.286218 -0.159277  0.428637  ... -0.557919       NaN   \n",
       "\n",
       "Name          ZSWIM8      ZW10    ZWILCH     ZWINT    ZYG11B       ZYX  \\\n",
       "Patient_ID                                                               \n",
       "01OV007     0.074209  0.232999 -0.195793       NaN -0.145053  0.590217   \n",
       "01OV017     0.016276  0.587911  0.574339       NaN  0.075401 -0.437564   \n",
       "01OV018     0.206993  0.111037  1.020321       NaN -0.245864  0.008411   \n",
       "01OV023    -0.529948  0.034713  0.149523 -0.313370 -0.507473  0.221818   \n",
       "01OV026    -0.181027  0.179612  0.007854 -0.106049  0.377063 -0.295592   \n",
       "...              ...       ...       ...       ...       ...       ...   \n",
       "17OV001.N   0.010818  0.321095 -0.612827 -0.149425  0.017523  0.442398   \n",
       "17OV002.N  -0.530407 -0.157267  0.716779 -0.676827 -0.569823  0.640503   \n",
       "17OV003.N   0.778621  0.425679 -0.048390       NaN  0.675466  0.267934   \n",
       "17OV004.N   0.090890 -0.085341  0.952951  0.874700 -0.185089 -0.426710   \n",
       "17OV005.N  -0.749876 -0.364281  0.467271       NaN -0.228770  0.182717   \n",
       "\n",
       "Name           ZZEF1      ZZZ3  \n",
       "Patient_ID                      \n",
       "01OV007     0.019414       NaN  \n",
       "01OV017    -0.208831       NaN  \n",
       "01OV018     0.130504 -0.367433  \n",
       "01OV023     0.278313 -1.008998  \n",
       "01OV026     0.011997 -0.236003  \n",
       "...              ...       ...  \n",
       "17OV001.N   0.088551 -0.614771  \n",
       "17OV002.N   0.763645 -1.370872  \n",
       "17OV003.N   0.583772  0.501873  \n",
       "17OV004.N   0.008868 -0.278545  \n",
       "17OV005.N   0.298465 -0.591317  \n",
       "\n",
       "[103 rows x 10686 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "\n",
    "prot = o.get_proteomics()\n",
    "prot = o.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "p = list(prot.columns)\n",
    "len(p)\n",
    "prot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 6)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "10081"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = o.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = o.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = o.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()\n",
    "len(prot_and_mutations.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MMS19</td>\n",
       "      <td>1.058914e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RACK1</td>\n",
       "      <td>8.573522e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PI4K2A</td>\n",
       "      <td>9.310383e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WAPL</td>\n",
       "      <td>1.724680e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IDE</td>\n",
       "      <td>2.011475e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10072</th>\n",
       "      <td>SAXO2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10073</th>\n",
       "      <td>SH3GL3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10074</th>\n",
       "      <td>SLC22A3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10075</th>\n",
       "      <td>SSX2B</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10076</th>\n",
       "      <td>TFF2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10077 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics    Ov_P_Value\n",
       "0          MMS19  1.058914e-08\n",
       "1          RACK1  8.573522e-05\n",
       "2         PI4K2A  9.310383e-05\n",
       "3           WAPL  1.724680e-04\n",
       "4            IDE  2.011475e-04\n",
       "...          ...           ...\n",
       "10072      SAXO2           NaN\n",
       "10073     SH3GL3           NaN\n",
       "10074    SLC22A3           NaN\n",
       "10075      SSX2B           NaN\n",
       "10076       TFF2           NaN\n",
       "\n",
       "[10077 rows x 2 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "o = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "o = o.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Ov_P_Value'})\n",
    "o_pval = o.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "o_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "ov_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    ov_d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "o_df = pd.DataFrame.from_dict(ov_d, orient='index', columns=['Ov_Median'])\n",
    "o_df = o_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#o_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MMS19</td>\n",
       "      <td>1.058914e-08</td>\n",
       "      <td>0.324897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RACK1</td>\n",
       "      <td>8.573522e-05</td>\n",
       "      <td>0.191250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PI4K2A</td>\n",
       "      <td>9.310383e-05</td>\n",
       "      <td>0.298399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WAPL</td>\n",
       "      <td>1.724680e-04</td>\n",
       "      <td>0.218775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IDE</td>\n",
       "      <td>2.011475e-04</td>\n",
       "      <td>0.208025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10072</th>\n",
       "      <td>SAXO2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.099993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10073</th>\n",
       "      <td>SH3GL3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.843461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10074</th>\n",
       "      <td>SLC22A3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.783325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10075</th>\n",
       "      <td>SSX2B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10076</th>\n",
       "      <td>TFF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10077 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics    Ov_P_Value  Ov_Median\n",
       "0          MMS19  1.058914e-08   0.324897\n",
       "1          RACK1  8.573522e-05   0.191250\n",
       "2         PI4K2A  9.310383e-05   0.298399\n",
       "3           WAPL  1.724680e-04   0.218775\n",
       "4            IDE  2.011475e-04   0.208025\n",
       "...          ...           ...        ...\n",
       "10072      SAXO2           NaN   0.099993\n",
       "10073     SH3GL3           NaN  -0.843461\n",
       "10074    SLC22A3           NaN  -0.783325\n",
       "10075      SSX2B           NaN        NaN\n",
       "10076       TFF2           NaN        NaN\n",
       "\n",
       "[10077 rows x 3 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "o_merged = o_pval.merge(o_df, on='Proteomics',how='inner')\n",
    "o_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Breast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    }
   ],
   "source": [
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "10107"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = b.get_proteomics()\n",
    "prot = b.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "p = list(prot.columns)\n",
    "len(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 6)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    80\n",
       "Deletion          24\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = b.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = b.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = b.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations.loc[:,~prot_and_mutations.columns.duplicated()] # drop duplicated columns\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "b_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "b_pval = b_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Brca_P_Value'})\n",
    "b_pval = b_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#b_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "b_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    b_d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "b_df = pd.DataFrame.from_dict(b_d, orient='index', columns=['Brca_Median'])\n",
    "b_df = b_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#b_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>1.00800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EIF4H</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>-0.41295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIEF1</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>-0.84135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMSB10</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>-0.87630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DRG1</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>-0.75135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9766</th>\n",
       "      <td>POLR3E</td>\n",
       "      <td>0.999356</td>\n",
       "      <td>0.09685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9767</th>\n",
       "      <td>IGFBP7</td>\n",
       "      <td>0.999595</td>\n",
       "      <td>-0.01140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9768</th>\n",
       "      <td>AAMP</td>\n",
       "      <td>0.999613</td>\n",
       "      <td>-0.15430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9769</th>\n",
       "      <td>TRA2B</td>\n",
       "      <td>0.999771</td>\n",
       "      <td>0.17180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9770</th>\n",
       "      <td>KRIT1</td>\n",
       "      <td>0.999848</td>\n",
       "      <td>-0.14610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9771 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Proteomics  Brca_P_Value  Brca_Median\n",
       "0          PTEN      0.000004      1.00800\n",
       "1         EIF4H      0.000025     -0.41295\n",
       "2         MIEF1      0.000031     -0.84135\n",
       "3        TMSB10      0.000043     -0.87630\n",
       "4          DRG1      0.000065     -0.75135\n",
       "...         ...           ...          ...\n",
       "9766     POLR3E      0.999356      0.09685\n",
       "9767     IGFBP7      0.999595     -0.01140\n",
       "9768       AAMP      0.999613     -0.15430\n",
       "9769      TRA2B      0.999771      0.17180\n",
       "9770      KRIT1      0.999848     -0.14610\n",
       "\n",
       "[9771 rows x 3 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_merged = b_pval.merge(b_df, on='Proteomics',how='inner')\n",
    "b_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "col = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8067"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = col.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "len(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n",
      "cptac warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    76\n",
       "Deletion          25\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = col.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = col.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "c_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "c_pval = c_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Colon_P_Value'})\n",
    "c_pval = c_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#c_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "c_df = pd.DataFrame.from_dict(d, orient='index', columns=['Colon_Median'])\n",
    "c_df = c_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#c_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DFFA</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.19930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WAPL</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.19605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SEC14L2</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>0.66800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GBF1</td>\n",
       "      <td>0.000164</td>\n",
       "      <td>0.12630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>STK11IP</td>\n",
       "      <td>0.000239</td>\n",
       "      <td>0.31250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8062</th>\n",
       "      <td>ZNF841</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.78250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8063</th>\n",
       "      <td>ZNF888</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8064</th>\n",
       "      <td>ZNHIT6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8065</th>\n",
       "      <td>ZNRD1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.06800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8066</th>\n",
       "      <td>ZWINT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.18535</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8067 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Proteomics  Colon_P_Value  Colon_Median\n",
       "0          DFFA       0.000037       0.19930\n",
       "1          WAPL       0.000097       0.19605\n",
       "2       SEC14L2       0.000136       0.66800\n",
       "3          GBF1       0.000164       0.12630\n",
       "4       STK11IP       0.000239       0.31250\n",
       "...         ...            ...           ...\n",
       "8062     ZNF841            NaN      -0.78250\n",
       "8063     ZNF888            NaN           NaN\n",
       "8064     ZNHIT6            NaN           NaN\n",
       "8065      ZNRD1            NaN       0.06800\n",
       "8066      ZWINT            NaN      -0.18535\n",
       "\n",
       "[8067 rows x 3 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c_merged = c_pval.merge(c_df, on='Proteomics',how='inner')\n",
    "c_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Head and Neck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "h = cptac.Hnscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11744"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = h.get_proteomics()\n",
    "p = list(prot.columns)\n",
    "len(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 383)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 176 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype_Tumor    87\n",
       "Deletion          18\n",
       "Name: Mutation, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = h.get_genotype_all_vars(gene)\n",
    "\n",
    "# merge cnv with genotype all mut type\n",
    "prot_and_mutations = h.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'proteomics', omics_genes = p)\n",
    "#prot_and_mutations = h.reduce_multiindex(prot_and_mutations, levels_to_drop = 1)\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "ip_df = prot_and_mutations.iloc[:,:-4] #drop mutation, location cols\n",
    "mut_type = mut_type[['Mutation']] \n",
    "merged = ip_df.join(mut_type) # merge mutation col from function (includes cnv)\n",
    "\n",
    "# Keep two values to compare\n",
    "compare = ['Wildtype_Tumor','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "del_wt['Mutation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-1])\n",
    "\n",
    "h_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "h_pval = h_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Hnscc_P_Value'})\n",
    "h_pval = h_pval.replace(to_replace = r'_proteomics$', value = '', regex = True)\n",
    "\n",
    "#h_pval\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\lib\\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice\n",
      "  return np.nanmean(a, axis, out=out, keepdims=keepdims)\n"
     ]
    }
   ],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"Wildtype_Tumor\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "h_d = {}\n",
    "\n",
    "for prot in p:\n",
    "    dif = wt_med[prot+'_proteomics'] - del_med[prot+'_proteomics']\n",
    "    h_d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "h_df = pd.DataFrame.from_dict(h_d, orient='index', columns=['Hnscc_Median'])\n",
    "h_df = h_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#h_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NLN</td>\n",
       "      <td>1.113430e-09</td>\n",
       "      <td>-0.431666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKR1C3</td>\n",
       "      <td>1.032977e-08</td>\n",
       "      <td>-0.951905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKR1C1</td>\n",
       "      <td>6.931627e-08</td>\n",
       "      <td>-1.961262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ADI1</td>\n",
       "      <td>1.601880e-07</td>\n",
       "      <td>-0.513198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EPHX1</td>\n",
       "      <td>3.899394e-07</td>\n",
       "      <td>-0.871688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11739</th>\n",
       "      <td>ZNF664</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11740</th>\n",
       "      <td>ZNF836</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11741</th>\n",
       "      <td>ZNF862</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11742</th>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11743</th>\n",
       "      <td>ZXDA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11744 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics  Hnscc_P_Value  Hnscc_Median\n",
       "0            NLN   1.113430e-09     -0.431666\n",
       "1         AKR1C3   1.032977e-08     -0.951905\n",
       "2         AKR1C1   6.931627e-08     -1.961262\n",
       "3           ADI1   1.601880e-07     -0.513198\n",
       "4          EPHX1   3.899394e-07     -0.871688\n",
       "...          ...            ...           ...\n",
       "11739     ZNF664            NaN           NaN\n",
       "11740     ZNF836            NaN           NaN\n",
       "11741     ZNF862            NaN           NaN\n",
       "11742    ZSCAN31            NaN           NaN\n",
       "11743       ZXDA            NaN           NaN\n",
       "\n",
       "[11744 rows x 3 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h_merged = h_pval.merge(h_df, on='Proteomics',how='inner')\n",
    "#del_wt[['PREX2_proteomics','Mutation']].dropna() # in median, not in pval (only 1 value for deletion)\n",
    "h_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Lung -dropped isoforms so some genes not tested. FIX"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LUAD dataset is under publication embargo until July 01, 2020. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    }
   ],
   "source": [
    "l = cptac.Luad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "10699"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot = l.get_proteomics(tissue_type='tumor')\n",
    "prot = l.reduce_multiindex(prot, levels_to_drop = 1)\n",
    "p = list(prot.columns)\n",
    "len(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No_Mutation    88\n",
      "Deletion       17\n",
      "Name: Mutation, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Index(['A1BG', 'A2M', 'AAAS', 'AACS', 'AADAC', 'AADAT', 'AAED1', 'AAGAB',\n",
       "       'AAMDC', 'AAMP',\n",
       "       ...\n",
       "       'ZSCAN31', 'ZSWIM9', 'ZW10', 'ZWILCH', 'ZWINT', 'ZXDC', 'ZYG11B', 'ZYX',\n",
       "       'ZZEF1', 'ZZZ3'],\n",
       "      dtype='object', name='Name', length=10316)"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_type = l.get_genotype_all_vars(gene)\n",
    "mut_type = mut_type.rename(columns={'PTEN':'cnv'})\n",
    "\n",
    "merged = prot.join(mut_type) # checked and there is 110 tumor samples for lung\n",
    "merged = merged.loc[:,~merged.columns.duplicated()] # drop duplicate columns (isoforms)\n",
    "\n",
    "compare = ['No_Mutation','Deletion']\n",
    "get = merged['Mutation'].isin(compare)\n",
    "del_wt = merged[get]\n",
    "print(del_wt['Mutation'].value_counts())\n",
    "#del_wt.columns[:-2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATE1</td>\n",
       "      <td>2.851800e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHOC2</td>\n",
       "      <td>6.424816e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>9.121167e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LANCL2</td>\n",
       "      <td>1.742749e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GGCT</td>\n",
       "      <td>2.854181e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>3.173508e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PPP3CB</td>\n",
       "      <td>3.708740e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Proteomics  Luad_P_Value\n",
       "0       ATE1  2.851800e-07\n",
       "1      SHOC2  6.424816e-07\n",
       "2       EGFR  9.121167e-07\n",
       "3     LANCL2  1.742749e-06\n",
       "4       GGCT  2.854181e-06\n",
       "5    CWF19L1  3.173508e-06\n",
       "6     PPP3CB  3.708740e-06"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(del_wt.columns[:-2])\n",
    "\n",
    "l_pval = u.wrap_ttest(del_wt, 'Mutation', cols, return_all = True)\n",
    "l_pval = l_pval.rename(columns = {'Comparison': 'Proteomics','P_Value': 'Luad_P_Value'})\n",
    "\n",
    "l_pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Isoform code - fix later\n",
    "#del_wt.replace(to_replace = r'_NP_.*$', value = '', regex = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = del_wt[del_wt.Mutation == \"Deletion\"]\n",
    "wt = del_wt[del_wt.Mutation == \"No_Mutation\"]\n",
    "del_med = d.median()\n",
    "wt_med = wt.median()\n",
    "\n",
    "l_d = {}\n",
    "\n",
    "for prot in cols:\n",
    "    dif = wt_med[prot] - del_med[prot]\n",
    "    l_d[prot] = dif\n",
    "    #print(wt_med[prot+'_proteomics'], '-', del_med[prot+'_proteomics'], '=',dif)\n",
    "\n",
    "l_df = pd.DataFrame.from_dict(l_d, orient='index', columns=['Luad_Median'])\n",
    "l_df = l_df.reset_index().rename(columns={'index':'Proteomics'})\n",
    "#l_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATE1</td>\n",
       "      <td>2.851800e-07</td>\n",
       "      <td>0.88365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHOC2</td>\n",
       "      <td>6.424816e-07</td>\n",
       "      <td>0.60600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>9.121167e-07</td>\n",
       "      <td>-2.32970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LANCL2</td>\n",
       "      <td>1.742749e-06</td>\n",
       "      <td>-1.21475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GGCT</td>\n",
       "      <td>2.854181e-06</td>\n",
       "      <td>-2.12715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10311</th>\n",
       "      <td>COG6</td>\n",
       "      <td>9.997105e-01</td>\n",
       "      <td>0.13015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10312</th>\n",
       "      <td>PLD1</td>\n",
       "      <td>9.997256e-01</td>\n",
       "      <td>-0.43875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10313</th>\n",
       "      <td>CHMP6</td>\n",
       "      <td>9.997659e-01</td>\n",
       "      <td>0.05220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10314</th>\n",
       "      <td>FHL3</td>\n",
       "      <td>9.999702e-01</td>\n",
       "      <td>0.06610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10315</th>\n",
       "      <td>UST</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.49145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10316 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics  Luad_P_Value  Luad_Median\n",
       "0           ATE1  2.851800e-07      0.88365\n",
       "1          SHOC2  6.424816e-07      0.60600\n",
       "2           EGFR  9.121167e-07     -2.32970\n",
       "3         LANCL2  1.742749e-06     -1.21475\n",
       "4           GGCT  2.854181e-06     -2.12715\n",
       "...          ...           ...          ...\n",
       "10311       COG6  9.997105e-01      0.13015\n",
       "10312       PLD1  9.997256e-01     -0.43875\n",
       "10313      CHMP6  9.997659e-01      0.05220\n",
       "10314       FHL3  9.999702e-01      0.06610\n",
       "10315        UST           NaN     -2.49145\n",
       "\n",
       "[10316 rows x 3 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l_merged = l_pval.merge(l_df, on='Proteomics',how='inner')\n",
    "l_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "n = g_merged.merge(h_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(l_merged, on='Proteomics',how='outer')\n",
    "n= n.merge(b_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(o_merged, on='Proteomics',how='outer')\n",
    "n = n.merge(e_merged, on='Proteomics',how='outer')\n",
    "all_df = n.merge(c_merged, on='Proteomics',how='outer')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3</td>\n",
       "      <td>5.396032e-11</td>\n",
       "      <td>0.405134</td>\n",
       "      <td>0.000558</td>\n",
       "      <td>0.135941</td>\n",
       "      <td>0.017908</td>\n",
       "      <td>0.37845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001113</td>\n",
       "      <td>-0.1503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC</td>\n",
       "      <td>1.593480e-10</td>\n",
       "      <td>0.553255</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.314307</td>\n",
       "      <td>0.000744</td>\n",
       "      <td>1.01995</td>\n",
       "      <td>0.008295</td>\n",
       "      <td>0.48080</td>\n",
       "      <td>0.060164</td>\n",
       "      <td>0.159424</td>\n",
       "      <td>0.082868</td>\n",
       "      <td>0.2533</td>\n",
       "      <td>0.411506</td>\n",
       "      <td>0.08210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A</td>\n",
       "      <td>1.009419e-09</td>\n",
       "      <td>0.838882</td>\n",
       "      <td>0.001544</td>\n",
       "      <td>0.224713</td>\n",
       "      <td>0.560996</td>\n",
       "      <td>-0.04535</td>\n",
       "      <td>0.203952</td>\n",
       "      <td>0.07435</td>\n",
       "      <td>0.389187</td>\n",
       "      <td>-0.052955</td>\n",
       "      <td>0.201771</td>\n",
       "      <td>-0.1690</td>\n",
       "      <td>0.199847</td>\n",
       "      <td>-0.10135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2</td>\n",
       "      <td>1.122076e-09</td>\n",
       "      <td>0.586396</td>\n",
       "      <td>0.400554</td>\n",
       "      <td>-0.017886</td>\n",
       "      <td>0.612583</td>\n",
       "      <td>-0.05695</td>\n",
       "      <td>0.198043</td>\n",
       "      <td>-0.17195</td>\n",
       "      <td>0.786184</td>\n",
       "      <td>-0.033091</td>\n",
       "      <td>0.864381</td>\n",
       "      <td>-0.0232</td>\n",
       "      <td>0.349065</td>\n",
       "      <td>0.07690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2</td>\n",
       "      <td>1.302273e-09</td>\n",
       "      <td>0.610188</td>\n",
       "      <td>0.630985</td>\n",
       "      <td>-0.001559</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.354712</td>\n",
       "      <td>-0.14995</td>\n",
       "      <td>0.569417</td>\n",
       "      <td>0.067389</td>\n",
       "      <td>0.061660</td>\n",
       "      <td>-0.1286</td>\n",
       "      <td>0.990512</td>\n",
       "      <td>0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14360</th>\n",
       "      <td>WTH3DI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.40100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14361</th>\n",
       "      <td>ZIC4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14362</th>\n",
       "      <td>ZNF419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.38150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14363</th>\n",
       "      <td>ZNF705E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.12900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14364</th>\n",
       "      <td>ZNF841</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.78250</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14365 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics   Gbm_P_Value  Gbm_Median  Hnscc_P_Value  Hnscc_Median  \\\n",
       "0          ARMH3  5.396032e-11    0.405134       0.000558      0.135941   \n",
       "1           CUTC  1.593480e-10    0.553255       0.000189      0.314307   \n",
       "2        PIP4K2A  1.009419e-09    0.838882       0.001544      0.224713   \n",
       "3           CUL2  1.122076e-09    0.586396       0.400554     -0.017886   \n",
       "4           GDI2  1.302273e-09    0.610188       0.630985     -0.001559   \n",
       "...          ...           ...         ...            ...           ...   \n",
       "14360     WTH3DI           NaN         NaN            NaN           NaN   \n",
       "14361       ZIC4           NaN         NaN            NaN           NaN   \n",
       "14362     ZNF419           NaN         NaN            NaN           NaN   \n",
       "14363    ZNF705E           NaN         NaN            NaN           NaN   \n",
       "14364     ZNF841           NaN         NaN            NaN           NaN   \n",
       "\n",
       "       Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median  Ov_P_Value  \\\n",
       "0          0.017908      0.37845           NaN          NaN         NaN   \n",
       "1          0.000744      1.01995      0.008295      0.48080    0.060164   \n",
       "2          0.560996     -0.04535      0.203952      0.07435    0.389187   \n",
       "3          0.612583     -0.05695      0.198043     -0.17195    0.786184   \n",
       "4               NaN          NaN      0.354712     -0.14995    0.569417   \n",
       "...             ...          ...           ...          ...         ...   \n",
       "14360           NaN          NaN           NaN          NaN         NaN   \n",
       "14361           NaN          NaN           NaN          NaN         NaN   \n",
       "14362           NaN          NaN           NaN          NaN         NaN   \n",
       "14363           NaN          NaN           NaN          NaN         NaN   \n",
       "14364           NaN          NaN           NaN          NaN         NaN   \n",
       "\n",
       "       Ov_Median  En_P_Value  En_Median  Colon_P_Value  Colon_Median  \n",
       "0            NaN    0.001113    -0.1503            NaN           NaN  \n",
       "1       0.159424    0.082868     0.2533       0.411506       0.08210  \n",
       "2      -0.052955    0.201771    -0.1690       0.199847      -0.10135  \n",
       "3      -0.033091    0.864381    -0.0232       0.349065       0.07690  \n",
       "4       0.067389    0.061660    -0.1286       0.990512       0.04450  \n",
       "...          ...         ...        ...            ...           ...  \n",
       "14360        NaN         NaN        NaN            NaN       0.40100  \n",
       "14361        NaN         NaN        NaN            NaN      -0.00200  \n",
       "14362        NaN         NaN        NaN            NaN      -0.38150  \n",
       "14363        NaN         NaN        NaN            NaN       0.12900  \n",
       "14364        NaN         NaN        NaN            NaN      -0.78250  \n",
       "\n",
       "[14365 rows x 15 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARMH3</td>\n",
       "      <td>5.396032e-11</td>\n",
       "      <td>0.405134</td>\n",
       "      <td>5.584141e-04</td>\n",
       "      <td>0.135941</td>\n",
       "      <td>0.017908</td>\n",
       "      <td>0.37845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001113</td>\n",
       "      <td>-0.15030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUTC</td>\n",
       "      <td>1.593480e-10</td>\n",
       "      <td>0.553255</td>\n",
       "      <td>1.894453e-04</td>\n",
       "      <td>0.314307</td>\n",
       "      <td>0.000744</td>\n",
       "      <td>1.01995</td>\n",
       "      <td>0.008295</td>\n",
       "      <td>0.48080</td>\n",
       "      <td>0.060164</td>\n",
       "      <td>0.159424</td>\n",
       "      <td>0.082868</td>\n",
       "      <td>0.25330</td>\n",
       "      <td>0.411506</td>\n",
       "      <td>0.08210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP4K2A</td>\n",
       "      <td>1.009419e-09</td>\n",
       "      <td>0.838882</td>\n",
       "      <td>1.543642e-03</td>\n",
       "      <td>0.224713</td>\n",
       "      <td>0.560996</td>\n",
       "      <td>-0.04535</td>\n",
       "      <td>0.203952</td>\n",
       "      <td>0.07435</td>\n",
       "      <td>0.389187</td>\n",
       "      <td>-0.052955</td>\n",
       "      <td>0.201771</td>\n",
       "      <td>-0.16900</td>\n",
       "      <td>0.199847</td>\n",
       "      <td>-0.10135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUL2</td>\n",
       "      <td>1.122076e-09</td>\n",
       "      <td>0.586396</td>\n",
       "      <td>4.005541e-01</td>\n",
       "      <td>-0.017886</td>\n",
       "      <td>0.612583</td>\n",
       "      <td>-0.05695</td>\n",
       "      <td>0.198043</td>\n",
       "      <td>-0.17195</td>\n",
       "      <td>0.786184</td>\n",
       "      <td>-0.033091</td>\n",
       "      <td>0.864381</td>\n",
       "      <td>-0.02320</td>\n",
       "      <td>0.349065</td>\n",
       "      <td>0.07690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GDI2</td>\n",
       "      <td>1.302273e-09</td>\n",
       "      <td>0.610188</td>\n",
       "      <td>6.309846e-01</td>\n",
       "      <td>-0.001559</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.354712</td>\n",
       "      <td>-0.14995</td>\n",
       "      <td>0.569417</td>\n",
       "      <td>0.067389</td>\n",
       "      <td>0.061660</td>\n",
       "      <td>-0.12860</td>\n",
       "      <td>0.990512</td>\n",
       "      <td>0.04450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10465</th>\n",
       "      <td>SENP2</td>\n",
       "      <td>8.945796e-01</td>\n",
       "      <td>-0.013401</td>\n",
       "      <td>6.561188e-01</td>\n",
       "      <td>-0.086021</td>\n",
       "      <td>0.023553</td>\n",
       "      <td>-0.80130</td>\n",
       "      <td>0.375510</td>\n",
       "      <td>0.38550</td>\n",
       "      <td>0.869603</td>\n",
       "      <td>0.036145</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.48150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10709</th>\n",
       "      <td>STAT1</td>\n",
       "      <td>9.404646e-01</td>\n",
       "      <td>0.180847</td>\n",
       "      <td>1.305089e-01</td>\n",
       "      <td>0.353813</td>\n",
       "      <td>0.061263</td>\n",
       "      <td>0.60540</td>\n",
       "      <td>0.434161</td>\n",
       "      <td>-0.24190</td>\n",
       "      <td>0.097508</td>\n",
       "      <td>-0.142470</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0.93700</td>\n",
       "      <td>0.811050</td>\n",
       "      <td>0.06350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11013</th>\n",
       "      <td>PIR</td>\n",
       "      <td>9.948134e-01</td>\n",
       "      <td>0.087051</td>\n",
       "      <td>2.395809e-06</td>\n",
       "      <td>-1.051236</td>\n",
       "      <td>0.176542</td>\n",
       "      <td>-0.20135</td>\n",
       "      <td>0.056007</td>\n",
       "      <td>-1.09030</td>\n",
       "      <td>0.068513</td>\n",
       "      <td>-0.109151</td>\n",
       "      <td>0.184500</td>\n",
       "      <td>-0.19000</td>\n",
       "      <td>0.292827</td>\n",
       "      <td>-0.07240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11141</th>\n",
       "      <td>ADH7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.321698e-07</td>\n",
       "      <td>-2.155736</td>\n",
       "      <td>0.401404</td>\n",
       "      <td>0.12470</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.079428</td>\n",
       "      <td>-0.52485</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12724</th>\n",
       "      <td>BSPRY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.436117e-01</td>\n",
       "      <td>0.250476</td>\n",
       "      <td>0.099639</td>\n",
       "      <td>-0.91175</td>\n",
       "      <td>0.027229</td>\n",
       "      <td>0.70770</td>\n",
       "      <td>0.462840</td>\n",
       "      <td>-0.040360</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>-0.72800</td>\n",
       "      <td>0.112486</td>\n",
       "      <td>0.17590</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>136 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Proteomics   Gbm_P_Value  Gbm_Median  Hnscc_P_Value  Hnscc_Median  \\\n",
       "0          ARMH3  5.396032e-11    0.405134   5.584141e-04      0.135941   \n",
       "1           CUTC  1.593480e-10    0.553255   1.894453e-04      0.314307   \n",
       "2        PIP4K2A  1.009419e-09    0.838882   1.543642e-03      0.224713   \n",
       "3           CUL2  1.122076e-09    0.586396   4.005541e-01     -0.017886   \n",
       "4           GDI2  1.302273e-09    0.610188   6.309846e-01     -0.001559   \n",
       "...          ...           ...         ...            ...           ...   \n",
       "10465      SENP2  8.945796e-01   -0.013401   6.561188e-01     -0.086021   \n",
       "10709      STAT1  9.404646e-01    0.180847   1.305089e-01      0.353813   \n",
       "11013        PIR  9.948134e-01    0.087051   2.395809e-06     -1.051236   \n",
       "11141       ADH7           NaN         NaN   4.321698e-07     -2.155736   \n",
       "12724      BSPRY           NaN         NaN   9.436117e-01      0.250476   \n",
       "\n",
       "       Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median  Ov_P_Value  \\\n",
       "0          0.017908      0.37845           NaN          NaN         NaN   \n",
       "1          0.000744      1.01995      0.008295      0.48080    0.060164   \n",
       "2          0.560996     -0.04535      0.203952      0.07435    0.389187   \n",
       "3          0.612583     -0.05695      0.198043     -0.17195    0.786184   \n",
       "4               NaN          NaN      0.354712     -0.14995    0.569417   \n",
       "...             ...          ...           ...          ...         ...   \n",
       "10465      0.023553     -0.80130      0.375510      0.38550    0.869603   \n",
       "10709      0.061263      0.60540      0.434161     -0.24190    0.097508   \n",
       "11013      0.176542     -0.20135      0.056007     -1.09030    0.068513   \n",
       "11141      0.401404      0.12470           NaN          NaN         NaN   \n",
       "12724      0.099639     -0.91175      0.027229      0.70770    0.462840   \n",
       "\n",
       "       Ov_Median  En_P_Value  En_Median  Colon_P_Value  Colon_Median  \n",
       "0            NaN    0.001113   -0.15030            NaN           NaN  \n",
       "1       0.159424    0.082868    0.25330       0.411506       0.08210  \n",
       "2      -0.052955    0.201771   -0.16900       0.199847      -0.10135  \n",
       "3      -0.033091    0.864381   -0.02320       0.349065       0.07690  \n",
       "4       0.067389    0.061660   -0.12860       0.990512       0.04450  \n",
       "...          ...         ...        ...            ...           ...  \n",
       "10465   0.036145    0.000002    0.48150            NaN           NaN  \n",
       "10709  -0.142470    0.000005    0.93700       0.811050       0.06350  \n",
       "11013  -0.109151    0.184500   -0.19000       0.292827      -0.07240  \n",
       "11141        NaN    0.079428   -0.52485            NaN           NaN  \n",
       "12724  -0.040360    0.000003   -0.72800       0.112486       0.17590  \n",
       "\n",
       "[136 rows x 15 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = ['Gbm_P_Value','Hnscc_P_Value', 'Luad_P_Value', 'Brca_P_Value','Ov_P_Value','En_P_Value','Colon_P_Value']\n",
    "\n",
    "bc = .05/8000\n",
    "\n",
    "t = []\n",
    "\n",
    "for c in cols:\n",
    "    df = all_df[all_df[c] <= bc]\n",
    "    t.append(list(df.Proteomics))\n",
    "    \n",
    "flat_list = [item for sublist in t for item in sublist]\n",
    "sig = set(flat_list) # remove duplicates\n",
    "\n",
    "\n",
    "\n",
    "bool_df = all_df['Proteomics'].isin(sig)\n",
    "sig_df = all_df[bool_df]\n",
    "sig_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Gbm_P_Value</th>\n",
       "      <th>Gbm_Median</th>\n",
       "      <th>Hnscc_P_Value</th>\n",
       "      <th>Hnscc_Median</th>\n",
       "      <th>Luad_P_Value</th>\n",
       "      <th>Luad_Median</th>\n",
       "      <th>Brca_P_Value</th>\n",
       "      <th>Brca_Median</th>\n",
       "      <th>Ov_P_Value</th>\n",
       "      <th>Ov_Median</th>\n",
       "      <th>En_P_Value</th>\n",
       "      <th>En_Median</th>\n",
       "      <th>Colon_P_Value</th>\n",
       "      <th>Colon_Median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>MMS19</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.324287</td>\n",
       "      <td>0.020464</td>\n",
       "      <td>0.123084</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.019642</td>\n",
       "      <td>0.6534</td>\n",
       "      <td>1.058914e-08</td>\n",
       "      <td>0.324897</td>\n",
       "      <td>0.215233</td>\n",
       "      <td>0.0847</td>\n",
       "      <td>0.005686</td>\n",
       "      <td>0.1218</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Proteomics  Gbm_P_Value  Gbm_Median  Hnscc_P_Value  Hnscc_Median  \\\n",
       "66      MMS19     0.000004    0.324287       0.020464      0.123084   \n",
       "\n",
       "    Luad_P_Value  Luad_Median  Brca_P_Value  Brca_Median    Ov_P_Value  \\\n",
       "66      0.000088         0.62      0.019642       0.6534  1.058914e-08   \n",
       "\n",
       "    Ov_Median  En_P_Value  En_Median  Colon_P_Value  Colon_Median  \n",
       "66   0.324897    0.215233     0.0847       0.005686        0.1218  "
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_df.loc[sig_df['Proteomics'] == 'MMS19']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "t\n"
     ]
    }
   ],
   "source": [
    "s = list(sig)\n",
    "if 'MMS19' in s:\n",
    "    print('t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_df.to_csv('at_least_one_significant_all.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = ip_df.std()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev = stdev.to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stdev.loc[stdev[0] > .2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d_stdev = stdev.std()\n",
    "d_stdev"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = stdev.hist(bins = 20)\n",
    "plt.title(\"Standard Deviations of Pancancer CNV deletions and amplifications of PTEN on Interacting Proteins\\n\")\n",
    "plt.xlabel('StDeV')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(10,8)\n",
    "sns.set(font_scale = 1.3)\n",
    "a = sns.distplot(stdev, bins = 20)\n",
    "\n",
    "a.set_title(\"Standard Deviation Distribution for Pancancer PTEN Mutation Effect on Proteomics\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "a.set(xlabel = 'StDeV', ylabel = 'Frequency')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_df = stdev.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = mean_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Mean between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Mean')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m_df = stdev.median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = m_df.hist(bins = 20)\n",
    "plt.title(\"Distribution of the Medians between Cancers for PTEN Mutation Effect on all Proteins\\n\\n (Median of proteomics with Wildtype PTEN - Median of proteomics with Mutated PTEN)\")\n",
    "plt.xlabel('Median')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
