{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create Heat Map"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook looks at the significant genes in at least one cancer in the DNA Replication pathway. Pancancer heat maps are created with circle size showing significance and color showing differences in median."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import math as math\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys\n",
    "import statsmodels.stats.multitest\n",
    "from bokeh.palettes import RdBu\n",
    "from bokeh.models import LinearColorMapper, ColorBar\n",
    "from bokeh.models.ranges import FactorRange\n",
    "from bokeh.plotting import figure, show\n",
    "from bokeh.io import output_notebook, export_png, export_svgs\n",
    "from bokeh.layouts import row\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import gseapy as gp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_circle_legend_df(legend_min, legend_max):\n",
    "    \n",
    "    # Find middle pvals\n",
    "    # Find difference between exponents of the min and max\n",
    "    exp_legend_min = np.log10(legend_min)\n",
    "    exp_legend_max = np.log10(legend_max)\n",
    "    delta = exp_legend_min - exp_legend_max \n",
    "    # Split difference into quarters \n",
    "    num = 4\n",
    "    val = delta / num\n",
    "    # Find middle exponents\n",
    "    exp2 = round(exp_legend_min - val)\n",
    "    exp3 = round(exp_legend_max + val)\n",
    "    # Create middle pvals\n",
    "    pval2 = 1*10**exp2\n",
    "    pval3 = 1*10**exp3\n",
    "    \n",
    "    # Foramat scientific notation pvals as strings for y_axis labels  \n",
    "    max_str = \"{:.1e}\".format(legend_max, '.2f')\n",
    "    pval_str_3 = \"{:.1e}\".format(pval3, '.2f')\n",
    "    pval_str_2 = \"{:.1e}\".format(pval2, '.2f')\n",
    "    min_str = \"{:.1e}\".format(legend_min, '.2f')\n",
    "    \n",
    "    # max to min\n",
    "    data = {'P_Value':  [legend_max, pval3, pval2, legend_min],\n",
    "            'y_axis': [max_str, pval_str_3, pval_str_2, min_str],\n",
    "            'x_axis': ['', '', '', ''],\n",
    "            'Medians': [1.5, 1.5, 1.5, 1.5]}\n",
    "\n",
    "    fake_df = pd.DataFrame (data, columns = ['x_axis', 'y_axis', 'P_Value', 'Medians'])\n",
    "    \n",
    "    fake_df[\"size2\"] = fake_df['P_Value'].apply(lambda x: -1*(np.log(x)))\n",
    "    fake_df['size'] = (fake_df[\"size2\"])*3\n",
    "    \n",
    "    return fake_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_circle_legend(df, circle_var, color_var, legend_min, legend_max,\n",
    "                         x_axis = 'x_axis', y_axis = 'y_axis', \n",
    "                         plot_height = 200, plot_width = 120):\n",
    "    # Use the smallest pval\n",
    "    if df[circle_var].min() < legend_min:\n",
    "        legend_min = df[circle_var].min()\n",
    "    circle_df = create_circle_legend_df(legend_min, legend_max)\n",
    "    \n",
    "    maxval = circle_df[color_var].max()\n",
    "    minval = circle_df[color_var].min()\n",
    "    if maxval > abs(minval):\n",
    "        minval = maxval * -1 \n",
    "    if maxval < abs(minval):\n",
    "        maxval = minval * -1\n",
    "    colors = list((RdBu[9]))\n",
    "    exp_cmap = LinearColorMapper(palette=colors, low = minval, high = maxval)\n",
    "    \n",
    "    circle = figure(x_range = FactorRange(), y_range = FactorRange(), plot_width= plot_width, \n",
    "               plot_height=plot_height, toolbar_location=None, tools=\"hover\")\n",
    "\n",
    "    circle.scatter(x_axis, y_axis, source = circle_df, fill_alpha=1,  line_width=0, size=\"size\", \n",
    "              fill_color={\"field\":color_var, \"transform\":exp_cmap})\n",
    "    \n",
    "    circle.x_range.factors = sorted(circle_df[x_axis].unique().tolist())\n",
    "    circle.y_range.factors = circle_df[y_axis].unique().tolist() # plots in reverse order of df (max to min)\n",
    "    circle.xaxis.major_label_orientation = math.pi/2\n",
    "    \n",
    "    circle.xaxis.axis_label = 'P-Values'\n",
    "    \n",
    "    return circle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 1: Run GSEA for significant genes in at least 1 cancer"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First read in sig_pval_all_proteins.csv into a df. This csv file contains only genes with a significant p-value in at least one cancer. Then run GSEA using a list of genes from the df."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "root = R'~\\Github\\WhenMutationsDontMatter\\PTEN\\Step_3_trans_effect\\csv'\n",
    "sig_df = pd.read_csv(root+R'\\sig_pval_heatmap.csv')\n",
    "\n",
    "prot_list = list(sig_df.Proteomics) # list of genes with a sig pval in >= 1 cancer\n",
    "prot_enr = gp.enrichr(gene_list = prot_list, description='Tumor_partition', gene_sets='Reactome_2016', \n",
    "                       outdir='/Enrichr')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Term</th>\n",
       "      <th>Overlap</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Adjusted P-value</th>\n",
       "      <th>Old P-value</th>\n",
       "      <th>Old Adjusted P-value</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>Combined Score</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Gene_set</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gene Expression Homo sapiens R-HSA-74160</td>\n",
       "      <td>440/1631</td>\n",
       "      <td>6.403332e-53</td>\n",
       "      <td>9.797098e-50</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.015489</td>\n",
       "      <td>242.221802</td>\n",
       "      <td>TDRKH;RPL4;ATF2;MDC1;NUP107;HNRNPU;EHMT1;PHAX;...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Processing of Capped Intron-Containing Pre-mRN...</td>\n",
       "      <td>116/193</td>\n",
       "      <td>1.992679e-52</td>\n",
       "      <td>1.524399e-49</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.490372</td>\n",
       "      <td>534.556067</td>\n",
       "      <td>NUP107;NUP188;EIF4A3;HNRNPU;EFTUD2;SNRPD2;SNRP...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mRNA Splicing - Major Pathway Homo sapiens R-H...</td>\n",
       "      <td>85/134</td>\n",
       "      <td>2.432993e-41</td>\n",
       "      <td>1.240826e-38</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.739099</td>\n",
       "      <td>443.185660</td>\n",
       "      <td>EIF4A3;HNRNPU;YBX1;PRPF19;USP39;ELAVL1;EFTUD2;...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mRNA Splicing Homo sapiens R-HSA-72172</td>\n",
       "      <td>85/144</td>\n",
       "      <td>7.488070e-38</td>\n",
       "      <td>2.864187e-35</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.409995</td>\n",
       "      <td>376.988047</td>\n",
       "      <td>EIF4A3;HNRNPU;YBX1;PRPF19;USP39;ELAVL1;EFTUD2;...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Major pathway of rRNA processing in the nucleo...</td>\n",
       "      <td>82/166</td>\n",
       "      <td>4.309535e-29</td>\n",
       "      <td>1.318718e-26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.690519</td>\n",
       "      <td>241.043035</td>\n",
       "      <td>LTV1;RPL4;DDX47;RPP30;WDR3;RPLP0;PWP2;RPL10A;R...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rRNA processing Homo sapiens R-HSA-72312</td>\n",
       "      <td>85/180</td>\n",
       "      <td>2.535428e-28</td>\n",
       "      <td>6.465341e-26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.527996</td>\n",
       "      <td>224.175973</td>\n",
       "      <td>LTV1;RPL4;DDX47;RPP30;WDR3;RPLP0;NAT10;PWP2;RP...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Infectious disease Homo sapiens R-HSA-5663205</td>\n",
       "      <td>127/348</td>\n",
       "      <td>4.238130e-28</td>\n",
       "      <td>9.263341e-26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.726504</td>\n",
       "      <td>171.846789</td>\n",
       "      <td>RPL4;NUP107;NUP188;RPL10A;RPL9;PSMD8;RPS15;PSM...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Cell Cycle, Mitotic Homo sapiens R-HSA-69278</td>\n",
       "      <td>145/462</td>\n",
       "      <td>3.302158e-24</td>\n",
       "      <td>6.315378e-22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.344810</td>\n",
       "      <td>126.777933</td>\n",
       "      <td>ANKLE2;NUP107;NUP188;ZWILCH;BUB1B;PPP2R2A;SMC4...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Cell Cycle Homo sapiens R-HSA-1640170</td>\n",
       "      <td>163/566</td>\n",
       "      <td>1.242046e-22</td>\n",
       "      <td>2.111478e-20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.151557</td>\n",
       "      <td>108.524761</td>\n",
       "      <td>MDC1;ANKLE2;NUP107;NUP188;ZWILCH;BUB1B;PPP2R2A...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Transport of Mature Transcript to Cytoplasm Ho...</td>\n",
       "      <td>46/74</td>\n",
       "      <td>1.962264e-22</td>\n",
       "      <td>3.002265e-20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.644166</td>\n",
       "      <td>232.128295</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;EIF4A3;NUP160;NXF1;...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Influenza Infection Homo sapiens R-HSA-168254</td>\n",
       "      <td>68/147</td>\n",
       "      <td>2.730640e-22</td>\n",
       "      <td>3.798072e-20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.455996</td>\n",
       "      <td>171.598263</td>\n",
       "      <td>RPL4;NUP107;NUP188;RPLP0;RPL10A;RPL9;RPS15;RPS...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Influenza Life Cycle Homo sapiens R-HSA-168255</td>\n",
       "      <td>64/136</td>\n",
       "      <td>1.472133e-21</td>\n",
       "      <td>1.876969e-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.515788</td>\n",
       "      <td>168.643828</td>\n",
       "      <td>RPL4;NUP107;NUP188;RPLP0;RPL10A;RPL9;RPS15;RPS...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Transport of Mature mRNA derived from an Intro...</td>\n",
       "      <td>43/68</td>\n",
       "      <td>1.739344e-21</td>\n",
       "      <td>2.047074e-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.724340</td>\n",
       "      <td>225.827144</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;EIF4A3;NUP160;NXF1;...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Influenza Viral RNA Transcription and Replicat...</td>\n",
       "      <td>61/128</td>\n",
       "      <td>5.756029e-21</td>\n",
       "      <td>6.290518e-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.560422</td>\n",
       "      <td>165.930051</td>\n",
       "      <td>RPL4;NUP107;NUP188;RPLP0;RPL10A;RPL9;RPS15;RPS...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HIV Infection Homo sapiens R-HSA-162906</td>\n",
       "      <td>83/222</td>\n",
       "      <td>1.459921e-19</td>\n",
       "      <td>1.489119e-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.793230</td>\n",
       "      <td>121.144449</td>\n",
       "      <td>FEN1;NUP107;ELL;NUP188;PSMD8;PSMD9;XPO1;PSMD5;...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Transport of Mature mRNA Derived from an Intro...</td>\n",
       "      <td>28/37</td>\n",
       "      <td>1.125061e-17</td>\n",
       "      <td>1.075840e-15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.653767</td>\n",
       "      <td>220.644541</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;NUP160;NXF1;NUP85;T...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Metabolism of non-coding RNA Homo sapiens R-HS...</td>\n",
       "      <td>33/50</td>\n",
       "      <td>1.207686e-17</td>\n",
       "      <td>1.086917e-15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.930893</td>\n",
       "      <td>192.084116</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;DDX20;PHAX;NUP160;S...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>snRNP Assembly Homo sapiens R-HSA-191859</td>\n",
       "      <td>33/50</td>\n",
       "      <td>1.207686e-17</td>\n",
       "      <td>1.026533e-15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.930893</td>\n",
       "      <td>192.084116</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;DDX20;PHAX;NUP160;S...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Transport of Mature mRNAs Derived from Intronl...</td>\n",
       "      <td>28/38</td>\n",
       "      <td>3.728557e-17</td>\n",
       "      <td>3.002470e-15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.504984</td>\n",
       "      <td>208.242122</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;NUP160;NXF1;NUP85;T...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>DNA strand elongation Homo sapiens R-HSA-69190</td>\n",
       "      <td>25/32</td>\n",
       "      <td>1.717948e-16</td>\n",
       "      <td>1.314230e-14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.836758</td>\n",
       "      <td>211.875647</td>\n",
       "      <td>FEN1;PCNA;MCM7;PRIM1;POLD3;POLD1;POLD2;GINS1;R...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Synthesis of DNA Homo sapiens R-HSA-69239</td>\n",
       "      <td>46/97</td>\n",
       "      <td>4.694039e-16</td>\n",
       "      <td>3.419943e-14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.542972</td>\n",
       "      <td>125.049438</td>\n",
       "      <td>PSMD10;FEN1;PSMD11;CDKN1B;PCNA;MCM7;PRIM1;PSMB...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Transport of the SLBP independent Mature mRNA ...</td>\n",
       "      <td>25/33</td>\n",
       "      <td>6.183251e-16</td>\n",
       "      <td>4.300170e-14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.659886</td>\n",
       "      <td>198.206480</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;NUP160;NXF1;NUP85;T...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HIV Life Cycle Homo sapiens R-HSA-162587</td>\n",
       "      <td>57/141</td>\n",
       "      <td>1.254669e-15</td>\n",
       "      <td>8.346280e-14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.020212</td>\n",
       "      <td>103.629210</td>\n",
       "      <td>FEN1;NUP107;NUP188;ELL;XPO1;NELFB;NELFE;NUP214...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Transport of the SLBP Dependant Mature mRNA Ho...</td>\n",
       "      <td>25/34</td>\n",
       "      <td>2.038196e-15</td>\n",
       "      <td>1.299350e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.493419</td>\n",
       "      <td>185.824294</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;NUP160;NXF1;NUP85;T...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Late Phase of HIV Life Cycle Homo sapiens R-HS...</td>\n",
       "      <td>53/128</td>\n",
       "      <td>3.757830e-15</td>\n",
       "      <td>2.299792e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.093482</td>\n",
       "      <td>102.749786</td>\n",
       "      <td>NUP107;NUP188;ELL;XPO1;NELFB;NELFE;NUP214;NDC1...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Interactions of Rev with host cellular protein...</td>\n",
       "      <td>25/35</td>\n",
       "      <td>6.224682e-15</td>\n",
       "      <td>3.662986e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.336464</td>\n",
       "      <td>174.557095</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;NUP160;XPO1;NUP85;T...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Rev-mediated nuclear export of HIV RNA Homo sa...</td>\n",
       "      <td>24/33</td>\n",
       "      <td>1.130501e-14</td>\n",
       "      <td>6.406172e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.433491</td>\n",
       "      <td>174.488568</td>\n",
       "      <td>RANBP2;NUP214;NDC1;RANBP1;NUP205;NUP107;SEH1L;...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Disease Homo sapiens R-HSA-1643685</td>\n",
       "      <td>172/725</td>\n",
       "      <td>1.142882e-14</td>\n",
       "      <td>6.245032e-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.772442</td>\n",
       "      <td>56.900069</td>\n",
       "      <td>RPL4;SEMA5B;NUP107;NUP188;HDAC11;RPL10A;RPL9;P...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>DNA Replication Homo sapiens R-HSA-69306</td>\n",
       "      <td>46/105</td>\n",
       "      <td>1.958357e-14</td>\n",
       "      <td>1.033202e-12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.273031</td>\n",
       "      <td>103.310239</td>\n",
       "      <td>PSMD10;FEN1;PSMD11;CDKN1B;PCNA;MCM7;PRIM1;PSMB...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NS1 Mediated Effects on Host Pathways Homo sap...</td>\n",
       "      <td>26/39</td>\n",
       "      <td>2.423739e-14</td>\n",
       "      <td>1.236107e-12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.980700</td>\n",
       "      <td>156.149321</td>\n",
       "      <td>NUP205;NUP107;SEH1L;NUP188;NUP160;PABPN1;NUP85...</td>\n",
       "      <td>Reactome_2016</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Term   Overlap       P-value  \\\n",
       "0            Gene Expression Homo sapiens R-HSA-74160  440/1631  6.403332e-53   \n",
       "1   Processing of Capped Intron-Containing Pre-mRN...   116/193  1.992679e-52   \n",
       "2   mRNA Splicing - Major Pathway Homo sapiens R-H...    85/134  2.432993e-41   \n",
       "3              mRNA Splicing Homo sapiens R-HSA-72172    85/144  7.488070e-38   \n",
       "4   Major pathway of rRNA processing in the nucleo...    82/166  4.309535e-29   \n",
       "5            rRNA processing Homo sapiens R-HSA-72312    85/180  2.535428e-28   \n",
       "6       Infectious disease Homo sapiens R-HSA-5663205   127/348  4.238130e-28   \n",
       "7        Cell Cycle, Mitotic Homo sapiens R-HSA-69278   145/462  3.302158e-24   \n",
       "8               Cell Cycle Homo sapiens R-HSA-1640170   163/566  1.242046e-22   \n",
       "9   Transport of Mature Transcript to Cytoplasm Ho...     46/74  1.962264e-22   \n",
       "10      Influenza Infection Homo sapiens R-HSA-168254    68/147  2.730640e-22   \n",
       "11     Influenza Life Cycle Homo sapiens R-HSA-168255    64/136  1.472133e-21   \n",
       "12  Transport of Mature mRNA derived from an Intro...     43/68  1.739344e-21   \n",
       "13  Influenza Viral RNA Transcription and Replicat...    61/128  5.756029e-21   \n",
       "14            HIV Infection Homo sapiens R-HSA-162906    83/222  1.459921e-19   \n",
       "15  Transport of Mature mRNA Derived from an Intro...     28/37  1.125061e-17   \n",
       "16  Metabolism of non-coding RNA Homo sapiens R-HS...     33/50  1.207686e-17   \n",
       "17           snRNP Assembly Homo sapiens R-HSA-191859     33/50  1.207686e-17   \n",
       "18  Transport of Mature mRNAs Derived from Intronl...     28/38  3.728557e-17   \n",
       "19     DNA strand elongation Homo sapiens R-HSA-69190     25/32  1.717948e-16   \n",
       "20          Synthesis of DNA Homo sapiens R-HSA-69239     46/97  4.694039e-16   \n",
       "21  Transport of the SLBP independent Mature mRNA ...     25/33  6.183251e-16   \n",
       "22           HIV Life Cycle Homo sapiens R-HSA-162587    57/141  1.254669e-15   \n",
       "23  Transport of the SLBP Dependant Mature mRNA Ho...     25/34  2.038196e-15   \n",
       "24  Late Phase of HIV Life Cycle Homo sapiens R-HS...    53/128  3.757830e-15   \n",
       "25  Interactions of Rev with host cellular protein...     25/35  6.224682e-15   \n",
       "26  Rev-mediated nuclear export of HIV RNA Homo sa...     24/33  1.130501e-14   \n",
       "27                 Disease Homo sapiens R-HSA-1643685   172/725  1.142882e-14   \n",
       "28           DNA Replication Homo sapiens R-HSA-69306    46/105  1.958357e-14   \n",
       "29  NS1 Mediated Effects on Host Pathways Homo sap...     26/39  2.423739e-14   \n",
       "\n",
       "    Adjusted P-value  Old P-value  Old Adjusted P-value  Odds Ratio  \\\n",
       "0       9.797098e-50            0                     0    2.015489   \n",
       "1       1.524399e-49            0                     0    4.490372   \n",
       "2       1.240826e-38            0                     0    4.739099   \n",
       "3       2.864187e-35            0                     0    4.409995   \n",
       "4       1.318718e-26            0                     0    3.690519   \n",
       "5       6.465341e-26            0                     0    3.527996   \n",
       "6       9.263341e-26            0                     0    2.726504   \n",
       "7       6.315378e-22            0                     0    2.344810   \n",
       "8       2.111478e-20            0                     0    2.151557   \n",
       "9       3.002265e-20            0                     0    4.644166   \n",
       "10      3.798072e-20            0                     0    3.455996   \n",
       "11      1.876969e-19            0                     0    3.515788   \n",
       "12      2.047074e-19            0                     0    4.724340   \n",
       "13      6.290518e-19            0                     0    3.560422   \n",
       "14      1.489119e-17            0                     0    2.793230   \n",
       "15      1.075840e-15            0                     0    5.653767   \n",
       "16      1.086917e-15            0                     0    4.930893   \n",
       "17      1.026533e-15            0                     0    4.930893   \n",
       "18      3.002470e-15            0                     0    5.504984   \n",
       "19      1.314230e-14            0                     0    5.836758   \n",
       "20      3.419943e-14            0                     0    3.542972   \n",
       "21      4.300170e-14            0                     0    5.659886   \n",
       "22      8.346280e-14            0                     0    3.020212   \n",
       "23      1.299350e-13            0                     0    5.493419   \n",
       "24      2.299792e-13            0                     0    3.093482   \n",
       "25      3.662986e-13            0                     0    5.336464   \n",
       "26      6.406172e-13            0                     0    5.433491   \n",
       "27      6.245032e-13            0                     0    1.772442   \n",
       "28      1.033202e-12            0                     0    3.273031   \n",
       "29      1.236107e-12            0                     0    4.980700   \n",
       "\n",
       "    Combined Score                                              Genes  \\\n",
       "0       242.221802  TDRKH;RPL4;ATF2;MDC1;NUP107;HNRNPU;EHMT1;PHAX;...   \n",
       "1       534.556067  NUP107;NUP188;EIF4A3;HNRNPU;EFTUD2;SNRPD2;SNRP...   \n",
       "2       443.185660  EIF4A3;HNRNPU;YBX1;PRPF19;USP39;ELAVL1;EFTUD2;...   \n",
       "3       376.988047  EIF4A3;HNRNPU;YBX1;PRPF19;USP39;ELAVL1;EFTUD2;...   \n",
       "4       241.043035  LTV1;RPL4;DDX47;RPP30;WDR3;RPLP0;PWP2;RPL10A;R...   \n",
       "5       224.175973  LTV1;RPL4;DDX47;RPP30;WDR3;RPLP0;NAT10;PWP2;RP...   \n",
       "6       171.846789  RPL4;NUP107;NUP188;RPL10A;RPL9;PSMD8;RPS15;PSM...   \n",
       "7       126.777933  ANKLE2;NUP107;NUP188;ZWILCH;BUB1B;PPP2R2A;SMC4...   \n",
       "8       108.524761  MDC1;ANKLE2;NUP107;NUP188;ZWILCH;BUB1B;PPP2R2A...   \n",
       "9       232.128295  NUP205;NUP107;SEH1L;NUP188;EIF4A3;NUP160;NXF1;...   \n",
       "10      171.598263  RPL4;NUP107;NUP188;RPLP0;RPL10A;RPL9;RPS15;RPS...   \n",
       "11      168.643828  RPL4;NUP107;NUP188;RPLP0;RPL10A;RPL9;RPS15;RPS...   \n",
       "12      225.827144  NUP205;NUP107;SEH1L;NUP188;EIF4A3;NUP160;NXF1;...   \n",
       "13      165.930051  RPL4;NUP107;NUP188;RPLP0;RPL10A;RPL9;RPS15;RPS...   \n",
       "14      121.144449  FEN1;NUP107;ELL;NUP188;PSMD8;PSMD9;XPO1;PSMD5;...   \n",
       "15      220.644541  NUP205;NUP107;SEH1L;NUP188;NUP160;NXF1;NUP85;T...   \n",
       "16      192.084116  NUP205;NUP107;SEH1L;NUP188;DDX20;PHAX;NUP160;S...   \n",
       "17      192.084116  NUP205;NUP107;SEH1L;NUP188;DDX20;PHAX;NUP160;S...   \n",
       "18      208.242122  NUP205;NUP107;SEH1L;NUP188;NUP160;NXF1;NUP85;T...   \n",
       "19      211.875647  FEN1;PCNA;MCM7;PRIM1;POLD3;POLD1;POLD2;GINS1;R...   \n",
       "20      125.049438  PSMD10;FEN1;PSMD11;CDKN1B;PCNA;MCM7;PRIM1;PSMB...   \n",
       "21      198.206480  NUP205;NUP107;SEH1L;NUP188;NUP160;NXF1;NUP85;T...   \n",
       "22      103.629210  FEN1;NUP107;NUP188;ELL;XPO1;NELFB;NELFE;NUP214...   \n",
       "23      185.824294  NUP205;NUP107;SEH1L;NUP188;NUP160;NXF1;NUP85;T...   \n",
       "24      102.749786  NUP107;NUP188;ELL;XPO1;NELFB;NELFE;NUP214;NDC1...   \n",
       "25      174.557095  NUP205;NUP107;SEH1L;NUP188;NUP160;XPO1;NUP85;T...   \n",
       "26      174.488568  RANBP2;NUP214;NDC1;RANBP1;NUP205;NUP107;SEH1L;...   \n",
       "27       56.900069  RPL4;SEMA5B;NUP107;NUP188;HDAC11;RPL10A;RPL9;P...   \n",
       "28      103.310239  PSMD10;FEN1;PSMD11;CDKN1B;PCNA;MCM7;PRIM1;PSMB...   \n",
       "29      156.149321  NUP205;NUP107;SEH1L;NUP188;NUP160;PABPN1;NUP85...   \n",
       "\n",
       "         Gene_set  \n",
       "0   Reactome_2016  \n",
       "1   Reactome_2016  \n",
       "2   Reactome_2016  \n",
       "3   Reactome_2016  \n",
       "4   Reactome_2016  \n",
       "5   Reactome_2016  \n",
       "6   Reactome_2016  \n",
       "7   Reactome_2016  \n",
       "8   Reactome_2016  \n",
       "9   Reactome_2016  \n",
       "10  Reactome_2016  \n",
       "11  Reactome_2016  \n",
       "12  Reactome_2016  \n",
       "13  Reactome_2016  \n",
       "14  Reactome_2016  \n",
       "15  Reactome_2016  \n",
       "16  Reactome_2016  \n",
       "17  Reactome_2016  \n",
       "18  Reactome_2016  \n",
       "19  Reactome_2016  \n",
       "20  Reactome_2016  \n",
       "21  Reactome_2016  \n",
       "22  Reactome_2016  \n",
       "23  Reactome_2016  \n",
       "24  Reactome_2016  \n",
       "25  Reactome_2016  \n",
       "26  Reactome_2016  \n",
       "27  Reactome_2016  \n",
       "28  Reactome_2016  \n",
       "29  Reactome_2016  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_enr.res2d.head(30)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 2: Get the list of significant genes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cell Cycle, Mitotic Homo sapiens R-HSA-69278\n",
      "total genes: 20\n"
     ]
    }
   ],
   "source": [
    "index = 7\n",
    "trans = prot_enr.res2d.Genes[index]\n",
    "genes = trans.split(';')\n",
    "print(prot_enr.res2d.Term[index])\n",
    "genes = genes[:20]\n",
    "print('total genes:',len(genes))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 3: Create HeatMap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Slice out genes from the DNA Replication pathway from the df with genes sig in > 1 cancer."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sig > 1 cancer\n",
    "bool_df = sig_df.Proteomics.isin(genes)\n",
    "plot_df = sig_df[bool_df]\n",
    "len(plot_df.Proteomics.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Only include p-values < a certain cutoff\n",
    "a = 0.05\n",
    "plot_df = plot_df.loc[plot_df['P_Value'] <= a]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plotCircleHeatMap (df, circle_var, color_var, x_axis, y_axis,plot_width= 1000, font_size=12,\n",
    "                       plot_height = 650, legend_min = 1e-6, legend_max = 0.01,\n",
    "                       x_axis_lab = \"no_label\", y_axis_lab = \"no_label\", \n",
    "                       show_plot = True, save_png = \"plot.png\"):\n",
    "  \n",
    "    # circle_var designed for pvalues. Normalized by taking log 10 of values and multiplying by 5 \n",
    "    #added a new column to make the plot size\n",
    "    \n",
    "    df[\"size2\"] = df[circle_var].apply(lambda x: -1*(np.log(x)))\n",
    "    df['size'] = (df[\"size2\"])*3\n",
    "    #find values to set color bar min/ max as \n",
    "    maxval = df[color_var].max()\n",
    "    minval = df[color_var].min()\n",
    "    if maxval > abs(minval):\n",
    "        minval = maxval * -1 \n",
    "    if maxval < abs(minval):\n",
    "        maxval = minval * -1\n",
    "    colors = list((RdBu[9]))\n",
    "    exp_cmap = LinearColorMapper(palette=colors, low = minval, high = maxval)\n",
    "    p = figure(x_range = FactorRange(), y_range = FactorRange(), plot_width= plot_width, \n",
    "               plot_height=plot_height, \n",
    "               toolbar_location=None, tools=\"hover\")\n",
    "\n",
    "    p.scatter(x_axis,y_axis,source=df, fill_alpha=1,  line_width=0, size=\"size\", \n",
    "              fill_color={\"field\":color_var, \"transform\":exp_cmap})\n",
    "    \n",
    "    p.x_range.factors = sorted(df[x_axis].unique().tolist())\n",
    "    p.y_range.factors = sorted(df[y_axis].unique().tolist(), reverse = True)\n",
    "    p.xaxis.major_label_orientation = math.pi/2\n",
    "    \n",
    "    # font size\n",
    "    p.axis.major_label_text_font_size = str(font_size)+\"pt\"\n",
    "    \n",
    "    if (x_axis_lab != \"no_label\" ):\n",
    "        p.xaxis.axis_label = x_axis_lab\n",
    "        p.xaxis.axis_label_text_font_size = '12pt'\n",
    "    if (y_axis_lab != \"no_label\" ):   ### change to y\n",
    "        p.yaxis.axis_label = y_axis_lab\n",
    "        p.yaxis.axis_label_text_font_size = '12pt'\n",
    "\n",
    "    bar = ColorBar(color_mapper=exp_cmap, location=(0,0))\n",
    "    p.add_layout(bar, \"right\")\n",
    "    \n",
    "    # Create Circle Legend\n",
    "    circle_legend = create_circle_legend(df, circle_var, color_var, \n",
    "                                         legend_min = legend_min, legend_max = legend_max)\n",
    "    \n",
    "    if show_plot:\n",
    "        output_notebook()\n",
    "        show(row(p, circle_legend))\n",
    "      \n",
    "    if save_png != \"plot.png\":\n",
    "        export_png(row(p, circle_legend), filename= save_png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"2351\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"2351\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      if (url in hashes) {\n",
       "        element.crossOrigin = \"anonymous\";\n",
       "        element.integrity = \"sha384-\" + hashes[url];\n",
       "      }\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"2351\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '2351' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  \n",
       "  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n",
       "  var css_urls = [];\n",
       "  \n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    function(Bokeh) {\n",
       "    \n",
       "    \n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      \n",
       "    for (var i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "    if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"2351\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"2351\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    const hashes = {\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\": \"JpP8FXbgAZLkfur7LiK3j9AGBhHNIvF742meBJrjO2ShJDhCG2I1uVvW+0DUtrmc\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\": \"xZlADit0Q04ISQEdKg2k3L4W9AwQBAuDs9nJL9fM/WwzL1tEU9VPNezOFX0nLEAz\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\": \"4BuPRZkdMKSnj3zoxiNrQ86XgNw0rYmBOxe7nshquXwwcauupgBF2DHLVG1WuZlV\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\": \"Dv1SQ87hmDqK6S5OhBf0bCuwAEvL5QYL0PuR/F1SPVhCS/r/abjkbpKDYL2zeM19\"};\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      if (url in hashes) {\n        element.crossOrigin = \"anonymous\";\n        element.integrity = \"sha384-\" + hashes[url];\n      }\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"2351\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '2351' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  \n  var js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.0.1.min.js\"];\n  var css_urls = [];\n  \n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    function(Bokeh) {\n    \n    \n    }\n  ];\n\n  function run_inline_js() {\n    \n    if (root.Bokeh !== undefined || force === true) {\n      \n    for (var i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n    if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"2351\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"1ba91f0f-e95e-4816-a46d-574254a03f11\" data-root-id=\"2352\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"290475f7-5e9b-4885-8dd9-2bf075d268c1\":{\"roots\":{\"references\":[{\"attributes\":{\"children\":[{\"id\":\"2303\"},{\"id\":\"2329\"}]},\"id\":\"2352\",\"type\":\"Row\"},{\"attributes\":{\"text\":\"\"},\"id\":\"2407\",\"type\":\"Title\"},{\"attributes\":{},\"id\":\"2334\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"factors\":[\"ANKLE2\",\"BUB1B\",\"CDC26\",\"GOLGA2\",\"GTSE1\",\"NUP107\",\"NUP188\",\"NUP210\",\"NUP214\",\"PPP2R2A\",\"PRKCB\",\"PSMD1\",\"PSMD2\",\"PSMD5\",\"PSMD8\",\"PSMD9\",\"SKP1\",\"SMC2\",\"SMC4\",\"ZWILCH\"]},\"id\":\"2301\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"2421\",\"type\":\"Selection\"},{\"attributes\":{\"data_source\":{\"id\":\"2319\"},\"glyph\":{\"id\":\"2321\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"2322\"},\"selection_glyph\":null,\"view\":{\"id\":\"2324\"}},\"id\":\"2323\",\"type\":\"GlyphRenderer\"},{\"attributes\":{},\"id\":\"2332\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"axis_label\":\"P-Values\",\"formatter\":{\"id\":\"2419\"},\"major_label_orientation\":1.5707963267948966,\"ticker\":{\"id\":\"2337\"}},\"id\":\"2336\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"2300\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Proteomics\"},\"y\":{\"field\":\"Cancer\"}},\"id\":\"2322\",\"type\":\"Scatter\"},{\"attributes\":{\"below\":[{\"id\":\"2310\"}],\"center\":[{\"id\":\"2312\"},{\"id\":\"2315\"}],\"left\":[{\"id\":\"2313\"}],\"plot_height\":300,\"plot_width\":500,\"renderers\":[{\"id\":\"2323\"}],\"right\":[{\"id\":\"2325\"}],\"title\":{\"id\":\"2407\"},\"toolbar\":{\"id\":\"2317\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"2301\"},\"x_scale\":{\"id\":\"2306\"},\"y_range\":{\"id\":\"2302\"},\"y_scale\":{\"id\":\"2308\"}},\"id\":\"2303\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"text\":\"\"},\"id\":\"2409\",\"type\":\"Title\"},{\"attributes\":{\"source\":{\"id\":\"2319\"}},\"id\":\"2324\",\"type\":\"CDSView\"},{\"attributes\":{},\"id\":\"2337\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"2415\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"axis\":{\"id\":\"2336\"},\"ticker\":null},\"id\":\"2338\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"2419\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"source\":{\"id\":\"2345\"}},\"id\":\"2350\",\"type\":\"CDSView\"},{\"attributes\":{\"formatter\":{\"id\":\"2417\"},\"ticker\":{\"id\":\"2340\"}},\"id\":\"2339\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"2424\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"2340\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"2306\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"formatter\":{\"id\":\"2411\"},\"major_label_text_font_size\":\"10pt\",\"ticker\":{\"id\":\"2314\"}},\"id\":\"2313\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"axis\":{\"id\":\"2339\"},\"dimension\":1,\"ticker\":null},\"id\":\"2341\",\"type\":\"Grid\"},{\"attributes\":{\"color_mapper\":{\"id\":\"2300\"},\"formatter\":{\"id\":\"2413\"},\"location\":[0,0],\"ticker\":{\"id\":\"2414\"}},\"id\":\"2325\",\"type\":\"ColorBar\"},{\"attributes\":{\"high\":1.5,\"low\":1.5,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"2326\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"data_source\":{\"id\":\"2345\"},\"glyph\":{\"id\":\"2347\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"2348\"},\"selection_glyph\":null,\"view\":{\"id\":\"2350\"}},\"id\":\"2349\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"axis\":{\"id\":\"2310\"},\"ticker\":null},\"id\":\"2312\",\"type\":\"Grid\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"2300\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Proteomics\"},\"y\":{\"field\":\"Cancer\"}},\"id\":\"2321\",\"type\":\"Scatter\"},{\"attributes\":{\"callback\":null},\"id\":\"2342\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"2422\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"2342\"}]},\"id\":\"2343\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"2423\",\"type\":\"Selection\"},{\"attributes\":{\"factors\":[\"5.0e-02\",\"1.0e-02\",\"5.0e-03\"]},\"id\":\"2328\",\"type\":\"FactorRange\"},{\"attributes\":{\"axis\":{\"id\":\"2313\"},\"dimension\":1,\"ticker\":null},\"id\":\"2315\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"2414\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"2314\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"2326\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"2347\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"2411\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"data\":{\"Cancer\":[\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Gbm\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Hnscc\",\"Luad\",\"Luad\",\"En\",\"En\",\"En\"],\"Medians\":{\"__ndarray__\":\"zajGWyS75T9qB8zqAF3Xv5BWVbgAH+k/sFAde3VPzD/isO/GiZnwv6dH0Cuve80/WcVcWZ6E6T/87WnE0CHQPwD4POziysc/td6aBLx4zr+pJevAbljIPx0wEqiIjeY/K3lCk8PZyD+x12p1fvPkP+ApP4K4N+K/ANUwl85ZxL+gV54YAx/hPwDSX4OaQ7k/AB5D/unnvT8BQX2IQ9W3P9k9eVioNfu/MuauJeQD9r8W2c73U+Phv8strYbEPc4/m1Wfq63Y1z8=\",\"dtype\":\"float64\",\"shape\":[25]},\"P_Value\":{\"__ndarray__\":\"T2d2j/PrdT/1MaDusp13P9Bl1DdPBng/RezMK4UTfj/PKLjZsBF/P6S/zYzNroQ/7SUpVS33hj+ya846hAWQP7QanyFCw5Y/3q8/7Zu5mD8wmseZCPWhPwyppWwjfKQ/QoLnFrcopj/PQMHn5UqmP3igLJS7wJI/m0D2GgTnlz9R/gRzdM6bP6J67TfVr5w/eIJljVUEnT/p8c/u9iugP58lUdcOzXg/QM5E1oO7oz8rgRzKnKSaP/cYGnQ8W6M/oKt+qLaGqD8=\",\"dtype\":\"float64\",\"shape\":[25]},\"Proteomics\":[\"SMC4\",\"SKP1\",\"SMC2\",\"CDC26\",\"PRKCB\",\"NUP214\",\"GTSE1\",\"GOLGA2\",\"ANKLE2\",\"PPP2R2A\",\"NUP188\",\"BUB1B\",\"NUP107\",\"ZWILCH\",\"PRKCB\",\"PSMD9\",\"NUP210\",\"PSMD2\",\"PSMD8\",\"PSMD1\",\"PRKCB\",\"NUP210\",\"BUB1B\",\"PSMD5\",\"PPP2R2A\"],\"index\":[398,426,438,524,532,671,738,932,1171,1219,1535,1668,1756,1768,2804,2891,2924,2934,2948,2995,5011,5081,14471,14537,14603],\"size\":{\"__ndarray__\":\"KI4vpblhL0Cq1tZDX+8uQCAoW4cF1S5AmG5bWPN7LUDsSsdzEEotQNkid4lnkitAUq1ZWJnxKkBe7AIk+vEoQNAmHa6M1iZAW10C64hXJkCtzbeMHRokQF7IAxzTTyNAHHyFRSnXIkBSod/s7s0iQNSBd1AxAChAFxJxPIKLJkCal0HUIKMlQK8oejRAcyVAdjRdhq1hJUCxUvqy+LokQLpXbaorpC5A7TsZVk6JI0BUBmpwx+QlQAz8QSPdpiNA3Cu370I7IkA=\",\"dtype\":\"float64\",\"shape\":[25]},\"size2\":{\"__ndarray__\":\"xV7KGNHrFEAcj+TXlJ8UQMAakq8DjhRAZfQ8kPenE0Cd3ISitYYTQObBpFuaYRJAjHPmOmb2EUA/SFdtUaEQQMAzfD1mcw5AJCcD5AvKDUDnvJ+7fM0KQH1gWiXEvwlAJlAHsuEeCUDCgX/mkxIJQOOrT+AgABBAycKWUFgPDkB4HwIbgdkMQOngovCqmQxAnvAmszyCDEBBbvhD9qMLQHw6nnFybRRA50/McmgMCkBwCDjrCTENQLr6V4TRMwpA0I9J6gNPCEA=\",\"dtype\":\"float64\",\"shape\":[25]}},\"selected\":{\"id\":\"2421\"},\"selection_policy\":{\"id\":\"2422\"}},\"id\":\"2319\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"2308\",\"type\":\"CategoricalScale\"},{\"attributes\":{},\"id\":\"2413\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"2316\"}]},\"id\":\"2317\",\"type\":\"Toolbar\"},{\"attributes\":{\"factors\":[\"Luad\",\"Hnscc\",\"Gbm\",\"En\"]},\"id\":\"2302\",\"type\":\"FactorRange\"},{\"attributes\":{\"factors\":[\"\"]},\"id\":\"2327\",\"type\":\"FactorRange\"},{\"attributes\":{},\"id\":\"2417\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"high\":1.7006,\"low\":-1.7006,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"2300\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"callback\":null},\"id\":\"2316\",\"type\":\"HoverTool\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"field\":\"Medians\",\"transform\":{\"id\":\"2326\"}},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"2348\",\"type\":\"Scatter\"},{\"attributes\":{\"below\":[{\"id\":\"2336\"}],\"center\":[{\"id\":\"2338\"},{\"id\":\"2341\"}],\"left\":[{\"id\":\"2339\"}],\"plot_height\":200,\"plot_width\":120,\"renderers\":[{\"id\":\"2349\"}],\"title\":{\"id\":\"2409\"},\"toolbar\":{\"id\":\"2343\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"2327\"},\"x_scale\":{\"id\":\"2332\"},\"y_range\":{\"id\":\"2328\"},\"y_scale\":{\"id\":\"2334\"}},\"id\":\"2329\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"axis_label\":\"Proteomics\",\"formatter\":{\"id\":\"2415\"},\"major_label_orientation\":1.5707963267948966,\"major_label_text_font_size\":\"10pt\",\"ticker\":{\"id\":\"2311\"}},\"id\":\"2310\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"2311\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"data\":{\"Medians\":{\"__ndarray__\":\"AAAAAAAA+D8AAAAAAAD4PwAAAAAAAPg/AAAAAAAA+D8=\",\"dtype\":\"float64\",\"shape\":[4]},\"P_Value\":{\"__ndarray__\":\"mpmZmZmZqT97FK5H4XqEP3sUrkfheoQ/exSuR+F6dD8=\",\"dtype\":\"float64\",\"shape\":[4]},\"index\":[0,1,2,3],\"size\":{\"__ndarray__\":\"reqW3HH5IUCg/4+ZiqErQKD/j5mKoStAfOpeKTfKL0A=\",\"dtype\":\"float64\",\"shape\":[4]},\"size2\":{\"__ndarray__\":\"keNze0L3B0AVVbW7sWsSQBVVtbuxaxJAU5yUG3oxFUA=\",\"dtype\":\"float64\",\"shape\":[4]},\"x_axis\":[\"\",\"\",\"\",\"\"],\"y_axis\":[\"5.0e-02\",\"1.0e-02\",\"1.0e-02\",\"5.0e-03\"]},\"selected\":{\"id\":\"2423\"},\"selection_policy\":{\"id\":\"2424\"}},\"id\":\"2345\",\"type\":\"ColumnDataSource\"}],\"root_ids\":[\"2352\"]},\"title\":\"Bokeh Application\",\"version\":\"2.0.1\"}};\n",
       "  var render_items = [{\"docid\":\"290475f7-5e9b-4885-8dd9-2bf075d268c1\",\"root_ids\":[\"2352\"],\"roots\":{\"2352\":\"1ba91f0f-e95e-4816-a46d-574254a03f11\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else {\n",
       "        attempts++;\n",
       "        if (attempts > 100) {\n",
       "          clearInterval(timer);\n",
       "          console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        }\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "2352"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "p.plotCircleHeatMap(plot_df, circle_var = 'P_Value', color_var='Medians', x_axis= 'Proteomics', y_axis = 'Cancer',\n",
    "                    plot_height=300, plot_width= 500, font_size= 10, legend_min = 5e-3, legend_max = 0.05, \n",
    "                    x_axis_lab= 'Proteomics', save_png = 'test4.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
