{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that lscc index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LUAD dataset is under publication embargo until July 01, 2020. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that brca index is up-to-date...   \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 8)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    }
   ],
   "source": [
    "e = cptac.Endometrial()\n",
    "h = cptac.Hnscc()\n",
    "l = cptac.Luad()\n",
    "ls = cptac.Lscc()\n",
    "o = cptac.Ovarian()\n",
    "c = cptac.Ccrcc()\n",
    "col = cptac.Colon()\n",
    "g = cptac.Gbm()\n",
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check mutation names\n",
    "cancers = [g,h,l,ls,o,e,c,col,b]\n",
    "a = []\n",
    "\n",
    "for c in cancers:\n",
    "    s = c.get_somatic_mutation()\n",
    "    n = list(s.Mutation.unique()) \n",
    "    a.append(n)\n",
    "\n",
    "flat_list = [item for sublist in a for item in sublist] #change list of lists, to just one list\n",
    "m = set(flat_list) # remove duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_frequently_mutated(cancer_object, cutoff = 0.1):  \n",
    "    # Get total tumors/patients\n",
    "    omics_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "        mutations_genes = 'TP53', omics_df_name = 'proteomics', omics_genes = 'TP53')\n",
    "    tumors = omics_and_mutations.Sample_Status\n",
    "\n",
    "    if isinstance(tumors, pd.DataFrame): # This would happen if our proteomics dataframe has a column multiindex, which leads to a joined df with a column multiindex, and causes our selection to be a dataframe instead of a series.\n",
    "        tumors = tumors.iloc[:, 0]\n",
    "        tumors.name = \"Sample_Status\"\n",
    "\n",
    "    v = tumors.value_counts()\n",
    "    total_tumors = v['Tumor']\n",
    "    total_tumor_count = float(total_tumors)\n",
    "    print(total_tumor_count)\n",
    "    \n",
    "    # Get mutations data frame\n",
    "    somatic_mutations = cancer_object.get_somatic_mutation() \n",
    "    #test#somatic_mutations = somatic_mutations.loc[somatic_mutations['Gene'] == 'APOB']\n",
    "\n",
    "    # Drop silent mutations for Hnscc, Ovarian, and Ccrcc dataset, and synonymous SNV (i.e. silent) mutations in HNSCC\n",
    "    if 'Silent' in somatic_mutations['Mutation'].unique():\n",
    "        origin_df = somatic_mutations.loc[somatic_mutations['Mutation'] != 'Silent'].reset_index()\n",
    "        (print('sil'))\n",
    "    elif 'synonymous SNV' in somatic_mutations['Mutation'].unique():\n",
    "        origin_df = somatic_mutations.loc[somatic_mutations['Mutation'] != 'synonymous SNV'].reset_index()\n",
    "    else:\n",
    "        origin_df = somatic_mutations.reset_index() #prepare to count unique samples\n",
    "        \n",
    "    # Create two categories in Mutation column - 'M': Missense, 'T': Truncation\n",
    "    if cancer_object.get_cancer_type() in ('hnscc') and cancer_object.version() == '0.1':\n",
    "        dif_mut_names = True\n",
    "    elif cancer_object.get_cancer_type() in ('colon'):\n",
    "        dif_mut_names = True\n",
    "    else: \n",
    "        dif_mut_names = False\n",
    "        \n",
    "    if dif_mut_names == True:\n",
    "        missense_truncation_groups = {'frameshift substitution': 'T', \n",
    "            'frameshift deletion': 'T', 'frameshift insertion': 'T', \n",
    "            'stopgain': 'T', 'stoploss': 'T', 'nonsynonymous SNV': 'M',\n",
    "            'nonframeshift insertion': 'M','nonframeshift deletion': 'M', \n",
    "            'nonframeshift substitution': 'M'}\n",
    "    else: \n",
    "        missense_truncation_groups = {'In_Frame_Del': 'M', 'In_Frame_Ins': 'M',\n",
    "            'Missense_Mutation': 'M', 'Frame_Shift_Del': 'T','Nonsense_Mutation': 'T', \n",
    "            'Splice_Site': 'T', 'Frame_Shift_Ins': 'T','Nonstop_Mutation':'T'}\n",
    "    \n",
    "    mutations_replaced_M_T = origin_df.replace(missense_truncation_groups)\n",
    "    \n",
    "    # replace non_coding mutations for Gbm\n",
    "    unique_mutations = len(mutations_replaced_M_T['Mutation'].unique())\n",
    "    gbm = False\n",
    "    if cancer_object.get_cancer_type() == 'gbm':\n",
    "        gbm = True\n",
    "        non_coding = {'Intron': 'NC', 'RNA': 'NC', \"5'Flank\": 'NC', \"3'Flank\": 'NC', \n",
    "            \"5'UTR\": 'NC', \"3'UTR\": 'NC', 'Splice_Region' : 'NC'}\n",
    "        mutations_replaced_M_T = mutations_replaced_M_T.replace(non_coding)\n",
    "        unique_mutations_2 = len(mutations_replaced_M_T['Mutation'].unique())\n",
    "        \n",
    "    elif unique_mutations != 2: # Check that all mutation names are catagorized\n",
    "        print('Warning: New mutation name not classified. Counts will be affected.')\n",
    "    \n",
    "    # Find frequently mutated genes (total fraction > cutoff)\n",
    "    # Same steps will be repeated for finding the missense and truncation mutation frequencies\n",
    "    # Step 1 - group by gene and count unique samples\n",
    "    # Step 2 - format\n",
    "    # Step 3 - filter using the cutoff and create fraction \n",
    "    count_mutations = origin_df.groupby(['Gene']).nunique()\n",
    "    count_mutations = count_mutations.rename(columns={\"Patient_ID\": \"Unique_Samples_Mut\"}) # Step 2 \n",
    "    count_mutations = count_mutations.drop(['Gene', 'Mutation', 'Location'], axis = 1)\n",
    "    fraction_mutated = count_mutations.apply(lambda x: x / total_tumor_count) # Step 3 \n",
    "    fraction_greater_than_cutoff = fraction_mutated.where(lambda x: x > cutoff) #na used when not > cutoff\n",
    "    filtered_gene_df = fraction_greater_than_cutoff.dropna() # drop genes below cutoff\n",
    "    \n",
    "    # Create and join Missense column (following similar steps as seen above) *Counts missense once in sample\n",
    "    miss = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'M']\n",
    "    count_miss = miss.groupby(['Gene']).nunique()\n",
    "    missense_df = count_miss.rename(columns={\"Patient_ID\": \"Missense_Mut\"})\n",
    "    #test#print(missense_df)\n",
    "    missense_df = missense_df.drop(['Gene', 'Mutation', 'Location'], axis = 1)\n",
    "    fraction_missense = missense_df.apply(lambda x: x / total_tumor_count)\n",
    "    freq_mutated_df = filtered_gene_df.join(fraction_missense, how='left').fillna(0)\n",
    "    \n",
    "    # Create and join Truncation column (following similar steps as seen above)\n",
    "    trunc = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'T']\n",
    "    count_trunc = trunc.groupby(['Gene']).nunique()\n",
    "    truncation_df = count_trunc.rename(columns={\"Patient_ID\": \"Truncation_Mut\"})\n",
    "    truncation_df = truncation_df.drop(['Gene', 'Mutation', 'Location'], axis = 1)\n",
    "    fraction_truncation = truncation_df.apply(lambda x: x / total_tumor_count)\n",
    "    freq_mutated_df = freq_mutated_df.join(fraction_truncation, how='left').fillna(0)\n",
    "    \n",
    "    if gbm == True:\n",
    "        # Create and join non-coding column (following similar steps as seen above)\n",
    "        nc = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'NC']\n",
    "        count_nc = nc.groupby(['Gene']).nunique()\n",
    "        nc_df = count_nc.rename(columns={\"Patient_ID\": \"Non-Coding\"})\n",
    "        nc_df = nc_df.drop(['Gene', 'Mutation', 'Location'], axis = 1)\n",
    "        fraction_nc = nc_df.apply(lambda x: x / total_tumor_count)\n",
    "        freq_mutated_df = freq_mutated_df.join(fraction_nc, how='left').fillna(0)\n",
    "        \n",
    "    freq_mutated_df = freq_mutated_df.reset_index() #move genes to their own column\n",
    "    \n",
    "    return freq_mutated_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the TP53 gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\utils\\__init__.py, line 372)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Unique_Samples_Mut</th>\n",
       "      <th>Missense_Mut</th>\n",
       "      <th>Truncation_Mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA13</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ADGRV1</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.009174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AHNAK</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.073394</td>\n",
       "      <td>0.027523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AJUBA</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.027523</td>\n",
       "      <td>0.091743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>APOB</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASPM</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.091743</td>\n",
       "      <td>0.009174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ATP10A</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CCDC168</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.082569</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>0.266055</td>\n",
       "      <td>0.045872</td>\n",
       "      <td>0.229358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>COL22A1</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.073394</td>\n",
       "      <td>0.027523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CSMD1</td>\n",
       "      <td>0.146789</td>\n",
       "      <td>0.137615</td>\n",
       "      <td>0.027523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CSMD3</td>\n",
       "      <td>0.238532</td>\n",
       "      <td>0.201835</td>\n",
       "      <td>0.045872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CTNNA2</td>\n",
       "      <td>0.174312</td>\n",
       "      <td>0.146789</td>\n",
       "      <td>0.027523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>DNAH11</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.091743</td>\n",
       "      <td>0.009174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>DNAH5</td>\n",
       "      <td>0.192661</td>\n",
       "      <td>0.155963</td>\n",
       "      <td>0.045872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>EP300</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.091743</td>\n",
       "      <td>0.027523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>EYS</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.082569</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>0.238532</td>\n",
       "      <td>0.082569</td>\n",
       "      <td>0.192661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>FAT2</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.064220</td>\n",
       "      <td>0.036697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>FMN2</td>\n",
       "      <td>0.137615</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.009174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>FSIP2</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.027523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>0.174312</td>\n",
       "      <td>0.082569</td>\n",
       "      <td>0.128440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>LRP1B</td>\n",
       "      <td>0.155963</td>\n",
       "      <td>0.137615</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>MAP2</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.009174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MDGA2</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.036697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.174312</td>\n",
       "      <td>0.146789</td>\n",
       "      <td>0.036697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.027523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MUC5B</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.027523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NAV3</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>0.238532</td>\n",
       "      <td>0.155963</td>\n",
       "      <td>0.119266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>PCDH15</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PCLO</td>\n",
       "      <td>0.137615</td>\n",
       "      <td>0.137615</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.091743</td>\n",
       "      <td>0.009174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PKHD1L1</td>\n",
       "      <td>0.110092</td>\n",
       "      <td>0.073394</td>\n",
       "      <td>0.036697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>RELN</td>\n",
       "      <td>0.119266</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>RYR1</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.128440</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>0.174312</td>\n",
       "      <td>0.137615</td>\n",
       "      <td>0.036697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>SPATA31D1</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>SYNE2</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.082569</td>\n",
       "      <td>0.018349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.880734</td>\n",
       "      <td>0.541284</td>\n",
       "      <td>0.422018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.403670</td>\n",
       "      <td>0.366972</td>\n",
       "      <td>0.110092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>XIRP2</td>\n",
       "      <td>0.100917</td>\n",
       "      <td>0.073394</td>\n",
       "      <td>0.036697</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       Gene  Unique_Samples_Mut  Missense_Mut  Truncation_Mut\n",
       "0        ABCA13            0.128440      0.119266        0.018349\n",
       "1        ADGRV1            0.110092      0.110092        0.009174\n",
       "2         AHNAK            0.100917      0.073394        0.027523\n",
       "3         AJUBA            0.110092      0.027523        0.091743\n",
       "4          APOB            0.110092      0.110092        0.000000\n",
       "5          ASPM            0.100917      0.091743        0.009174\n",
       "6        ATP10A            0.119266      0.110092        0.018349\n",
       "7       CCDC168            0.100917      0.082569        0.018349\n",
       "8        CDKN2A            0.266055      0.045872        0.229358\n",
       "9       COL22A1            0.100917      0.073394        0.027523\n",
       "10        CSMD1            0.146789      0.137615        0.027523\n",
       "11        CSMD3            0.238532      0.201835        0.045872\n",
       "12       CTNNA2            0.174312      0.146789        0.027523\n",
       "13       DNAH11            0.100917      0.091743        0.009174\n",
       "14        DNAH5            0.192661      0.155963        0.045872\n",
       "15        EP300            0.119266      0.091743        0.027523\n",
       "16          EYS            0.100917      0.082569        0.018349\n",
       "17         FAT1            0.238532      0.082569        0.192661\n",
       "18         FAT2            0.100917      0.064220        0.036697\n",
       "19         FAT3            0.119266      0.110092        0.018349\n",
       "20         FMN2            0.137615      0.128440        0.009174\n",
       "21        FSIP2            0.128440      0.100917        0.027523\n",
       "22        KMT2D            0.174312      0.082569        0.128440\n",
       "23        LRP1B            0.155963      0.137615        0.018349\n",
       "24         MAP2            0.119266      0.110092        0.009174\n",
       "25        MDGA2            0.128440      0.100917        0.036697\n",
       "26        MUC16            0.174312      0.146789        0.036697\n",
       "27         MUC4            0.128440      0.100917        0.027523\n",
       "28        MUC5B            0.119266      0.100917        0.027523\n",
       "29         NAV3            0.119266      0.100917        0.018349\n",
       "30       NOTCH1            0.238532      0.155963        0.119266\n",
       "31       PCDH15            0.119266      0.100917        0.018349\n",
       "32         PCLO            0.137615      0.137615        0.000000\n",
       "33       PIK3CA            0.100917      0.091743        0.009174\n",
       "34      PKHD1L1            0.110092      0.073394        0.036697\n",
       "35         PLEC            0.119266      0.119266        0.000000\n",
       "36         RELN            0.119266      0.100917        0.018349\n",
       "37         RYR1            0.128440      0.128440        0.000000\n",
       "38         RYR2            0.174312      0.137615        0.036697\n",
       "39    SPATA31D1            0.100917      0.100917        0.000000\n",
       "40        SYNE2            0.100917      0.082569        0.018349\n",
       "41         TP53            0.880734      0.541284        0.422018\n",
       "42          TTN            0.403670      0.366972        0.110092\n",
       "43        XIRP2            0.100917      0.073394        0.036697"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "u.get_frequently_mutated(h)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "t\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'0.1'"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if h.version() == '0.1':\n",
    "    print('t')\n",
    "h.version()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading hnscc v0.1.                       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Old hnscc data version. Latest is 2.0. This is 0.1. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                         \r"
     ]
    }
   ],
   "source": [
    "h = cptac.Hnscc(version='0.1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00999</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.R80X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-04849</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.E88X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3N-00297</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.E120X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3N-00299</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.R80X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3N-00825</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.W110X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C3N-01339</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.E61X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>C3N-02700</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.W110X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>C3N-02730</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.R80X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>C3N-03226</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.Y129X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>C3N-03226</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.E88X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>C3N-03620</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.E88X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>C3N-03837</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.R80X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>C3N-03888</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.R80X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>C3N-03933</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.R80X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>C3N-04276</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>p.W110X</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name Patient_ID    Gene  Mutation Location\n",
       "0     C3L-00999  CDKN2A  stopgain   p.R80X\n",
       "1     C3L-04849  CDKN2A  stopgain   p.E88X\n",
       "2     C3N-00297  CDKN2A  stopgain  p.E120X\n",
       "3     C3N-00299  CDKN2A  stopgain   p.R80X\n",
       "4     C3N-00825  CDKN2A  stopgain  p.W110X\n",
       "5     C3N-01339  CDKN2A  stopgain   p.E61X\n",
       "6     C3N-02700  CDKN2A  stopgain  p.W110X\n",
       "7     C3N-02730  CDKN2A  stopgain   p.R80X\n",
       "8     C3N-03226  CDKN2A  stopgain  p.Y129X\n",
       "9     C3N-03226  CDKN2A  stopgain   p.E88X\n",
       "10    C3N-03620  CDKN2A  stopgain   p.E88X\n",
       "11    C3N-03837  CDKN2A  stopgain   p.R80X\n",
       "12    C3N-03888  CDKN2A  stopgain   p.R80X\n",
       "13    C3N-03933  CDKN2A  stopgain   p.R80X\n",
       "14    C3N-04276  CDKN2A  stopgain  p.W110X"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s = h.get_somatic_mutation()\n",
    "s = s.loc[s['Mutation'] != 'Silent']\n",
    "g = s.loc[s['Gene'] == 'CDKN2A']\n",
    "r = g.reset_index()\n",
    "r.Patient_ID.value_counts()\n",
    "r\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the TP53 gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "109"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omics_and_mutations = h.join_omics_to_mutations(\n",
    "    mutations_genes = 'TP53', omics_df_name = 'proteomics', omics_genes = 'TP53')\n",
    "tumors = omics_and_mutations.Sample_Status\n",
    "\n",
    "v = tumors.value_counts()\n",
    "total_tumors = v['Tumor']\n",
    "total_tumor_count = int(total_tumors)\n",
    "total_tumor_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
