{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gene vs phospho correlation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook finds the correlation of phosphorylation between an individual site and all phosphylation for the corresponding gene. It thens find the percent of genes that have atleast 1 site with a corelation greater than 0.9 (or 0.8,"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/Lindsey/anaconda3/lib/python3.7/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "import statsmodels.stats.multitest\n",
    "import math\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "@Param df: Data frame. Takes a data frame of proteomic data frame.\n",
    "This function takes the data frame from get_proteomics and labels any duplicate columns due to mutliple isoforms.\n",
    "Then it returns the list of all proteins\n",
    "'''\n",
    "\n",
    "def create_prot_list(df):\n",
    "    #Add _(number) to keep track of duplicates due to multiple isoforms \n",
    "    cols = pd.Series(df.columns[:])\n",
    "    for dup in cols[cols.duplicated()].unique(): \n",
    "        cols[cols[cols == dup].index.values.tolist()] = [dup + '_' + str(i) if i != 0 else dup for i in range(sum(cols == dup))]\n",
    "    df.columns=cols\n",
    "    #get list of proteins\n",
    "    prot_genes_list = df.columns.values.tolist()\n",
    "    return prot_genes_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def wrap_pearson_corr(df,label_column, alpha=.05,comparison_columns=None,correction_method='bonferroni',return_all = True, return_corrected_pvals = False):\n",
    "\n",
    "\n",
    "    #df = df.dropna(axis=1, how=\"all\")\n",
    "\n",
    "    '''If no comparison columns specified, use all columns except the specified labed column'''\n",
    "    if not comparison_columns:\n",
    "        comparison_columns = list(df.columns)\n",
    "        comparison_columns.remove(label_column)\n",
    "    '''Store comparisons,p-values, correlation in their own array'''\n",
    "    comparisons = []\n",
    "    pvals = []\n",
    "    correlation=[]\n",
    "\n",
    "\n",
    "    '''Format results in a pandas dataframe'''\n",
    "    \n",
    "    newdf = pd.DataFrame()\n",
    "    for gene in comparison_columns:\n",
    "        #create subset df with interacting gene/ gene (otherwise drop NaN drops everything)\n",
    "        df_subset = df[[label_column,gene]]\n",
    "        #do a linear regression to see if it's a meaningful association\n",
    "        #dropna will remove rows with nan\n",
    "        df_subset = df_subset.dropna(axis=0, how=\"any\")\n",
    "        count_row = df_subset.shape[0]\n",
    "        if count_row > 30:\n",
    "            x1 = df_subset[[label_column]].values\n",
    "            y1 = df_subset[[gene]].values\n",
    "            x1 = x1[:,0]\n",
    "            y1 = y1[:,0]\n",
    "            corr, pval = scipy.stats.pearsonr(x1,y1)\n",
    "\n",
    "            comparisons.append(gene)\n",
    "            pvals.append(pval)\n",
    "            correlation.append(corr)\n",
    "            \n",
    "    if correction_method != \"none\":\n",
    "\n",
    "        '''Correct for multiple testing to determine if each comparison meets the new cutoff'''\n",
    "        results = statsmodels.stats.multitest.multipletests(pvals=pvals, alpha=alpha, method=correction_method)\n",
    "        reject = results[0]\n",
    "        corrected_pval = results[1]\n",
    "\n",
    "        if return_all:\n",
    "            if return_corrected_pvals:\n",
    "                for i in range(0,len(comparisons)):\n",
    "                    newdf = newdf.append({'Comparison': comparisons[i],\"Correlation\": correlation[i], \"P_value\": corrected_pval[i]}, ignore_index=True)\n",
    "\n",
    "            if return_corrected_pvals == False:\n",
    "                for i in range(0,len(comparisons)):\n",
    "                    newdf = newdf.append({'Comparison': comparisons[i],\"Correlation\": correlation[i],'P_value': pvals[i]}, ignore_index=True)\n",
    "\n",
    "        '''Else only add significant comparisons'''\n",
    "        if (return_all == False):\n",
    "\n",
    "            if return_corrected_pvals:\n",
    "                for i in range(0, len(reject)):\n",
    "                    if reject[i]:\n",
    "                        newdf = newdf.append({'Comparison': comparisons[i],\"Correlation\": correlation[i],\"P_value\": corrected_pval[i]}, ignore_index=True)\n",
    "            if return_corrected_pvals == False:\n",
    "                    for i in range(0, len(reject)):\n",
    "                        if reject[i]:\n",
    "                            newdf = newdf.append({'Comparison': comparisons[i],\"Correlation\": correlation[i],\"P_value\": pvals[i]}, ignore_index=True)\n",
    "\n",
    "    if correction_method == \"none\":\n",
    "        for i in range(0, len(comparisons)):\n",
    "            newdf = newdf.append({'Comparison': comparisons[i],\"Correlation\": correlation[i],\"P_value\": pvals[i]}, ignore_index=True)\n",
    "\n",
    "   # newdf = newdf.sort_values(by= 'P_value', ascending=True)\n",
    "    '''If results df is not empty, return it, else return None'''\n",
    "    return newdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.8.6'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#cptac version\n",
    "cptac.version()   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "#load cptac data \n",
    "brain = cptac.Gbm()\n",
    "kidney = cptac.Ccrcc()\n",
    "ovar = cptac.Ovarian()\n",
    "colon = cptac.Colon()\n",
    "brca = cptac.Brca()\n",
    "luad = cptac.Luad()\n",
    "hnscc = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()\n",
    "endo = cptac.Endometrial()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def phospho_gene_vs_site(cancer):\n",
    "    #get phosphoproteomic data \n",
    "    phospho_site = cancer.get_phosphoproteomics(\"tumor\")\n",
    "    #drop database_id and peptide \n",
    "    if (cancer != colon) & (cancer != endo):\n",
    "        phospho_site = u.reduce_multiindex(phospho_site, levels_to_drop = 2)\n",
    "        phospho_site = u.reduce_multiindex(phospho_site, levels_to_drop = 2)\n",
    "        phospho_gene = u.reduce_multiindex(phospho_site, levels_to_drop = 1)\n",
    "    if (cancer == colon): \n",
    "        phospho_site = u.reduce_multiindex(phospho_site, levels_to_drop = 2)\n",
    "        phospho_gene = u.reduce_multiindex(phospho_site, levels_to_drop = 1)\n",
    "    if cancer == endo:\n",
    "         phospho_gene = u.reduce_multiindex(phospho_site, levels_to_drop = 1)\n",
    "    genes = phospho_gene.columns.to_list() \n",
    "    genes = list(set(genes))#get unique list of all genes\n",
    "\n",
    "    phospho_site.columns =phospho_site.columns.map('_'.join) #join gene and site\n",
    "    phospho_site = phospho_site.loc[:,~phospho_site.T.duplicated(keep='first')] # drop columns where values are idendical\n",
    "\n",
    "    phospho_site = phospho_site.dropna(thresh=30, axis=1) #drop columns with less than 30 values \n",
    "    \n",
    "    df = pd.DataFrame() \n",
    "    for gene in genes:\n",
    "        phospho_gene = phospho_site.filter(regex=gene +\"_\",axis=1)\n",
    "        prot_genes_list = create_prot_list(phospho_gene)\n",
    "\n",
    "        if len(prot_genes_list) > 1: # only find correlation if there is more than one site \n",
    "            phospho_gene[gene] = phospho_gene.mean(numeric_only=True, axis=1) # average phospho across all sites for gene\n",
    "            #find correlations for all sites within that gene \n",
    "            corr_df = wrap_pearson_corr(phospho_gene,gene,comparison_columns= prot_genes_list,correction_method='none', return_all = True, return_corrected_pvals = False)\n",
    "            corr_df[\"Gene\"] = gene \n",
    "            df = df.append(corr_df)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def phospho_gene_vs_site(cancer):\n",
    "    #get phosphoproteomic data \n",
    "    phospho_site = cancer.get_phosphoproteomics(\"tumor\")\n",
    "    #drop database_id and peptide \n",
    "    if (cancer != colon) & (cancer != endo):\n",
    "        phospho_site = u.reduce_multiindex(phospho_site, levels_to_drop = 2)\n",
    "        phospho_site = u.reduce_multiindex(phospho_site, levels_to_drop = 2)\n",
    "        phospho_gene = u.reduce_multiindex(phospho_site, levels_to_drop = 1)\n",
    "    if (cancer == colon): \n",
    "        phospho_site = u.reduce_multiindex(phospho_site, levels_to_drop = 2)\n",
    "        phospho_gene = u.reduce_multiindex(phospho_site, levels_to_drop = 1)\n",
    "    if cancer == endo:\n",
    "         phospho_gene = u.reduce_multiindex(phospho_site, levels_to_drop = 1)\n",
    "    genes = phospho_gene.columns.to_list() \n",
    "    genes = list(set(genes))#get unique list of all genes\n",
    "\n",
    "    phospho_site.columns =phospho_site.columns.map('_'.join) #join gene and site\n",
    "    phospho_site = phospho_site.loc[:,~phospho_site.T.duplicated(keep='first')] # drop columns where values are idendical\n",
    "\n",
    "    df = pd.DataFrame() \n",
    "    for gene in genes:\n",
    "        phospho_gene = phospho_site.filter(regex=gene +\"_\",axis=1)\n",
    "        prot_genes_list = create_prot_list(phospho_gene)\n",
    "        if len(prot_genes_list) > 1: # only find correlation if there is more than one site \n",
    "        \n",
    "            site_corrs = phospho_gene.corr(method = \"pearson\", min_periods = 20) #pairwise correclations\n",
    "            site_corrs = site_corrs.values.tolist()#array to list\n",
    "            from pandas.core.common import flatten\n",
    "            site_corrs = list(flatten(site_corrs))\n",
    "            site_corrs = [x for x in site_corrs if (math.isnan(x) != True)] #remove na \n",
    "       \n",
    "            #site_corrs.remove(1.0)# remove comparison with same gene\n",
    "            site_corrs = list(set(site_corrs))# remove duplicate pairwise correlations\n",
    "      \n",
    "          \n",
    "            abs_val_corrs = [abs(x) for x in site_corrs]\n",
    "            cutoffs = [0.9,0.8,0.7,0.6,0.1]\n",
    "            for corr in cutoffs:\n",
    "\n",
    "                result = all(x >= corr for x in abs_val_corrs)\n",
    "                if not abs_val_corrs:\n",
    "                    result = False\n",
    "                df = df.append({'Gene': gene,\"Correlation\": corr,\"Pass_cutoff\": result}, ignore_index=True)\n",
    "\n",
    "        \n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Pass_cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.9</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.8</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.7</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.6</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.1</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33510</th>\n",
       "      <td>0.9</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33511</th>\n",
       "      <td>0.8</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33512</th>\n",
       "      <td>0.7</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33513</th>\n",
       "      <td>0.6</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33514</th>\n",
       "      <td>0.1</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33515 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Correlation   Gene  Pass_cutoff\n",
       "0              0.9  HECW2          0.0\n",
       "1              0.8  HECW2          0.0\n",
       "2              0.7  HECW2          0.0\n",
       "3              0.6  HECW2          0.0\n",
       "4              0.1  HECW2          0.0\n",
       "...            ...    ...          ...\n",
       "33510          0.9  TMUB1          1.0\n",
       "33511          0.8  TMUB1          1.0\n",
       "33512          0.7  TMUB1          1.0\n",
       "33513          0.6  TMUB1          1.0\n",
       "33514          0.1  TMUB1          1.0\n",
       "\n",
       "[33515 rows x 3 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbm_df = phospho_gene_vs_site(brain)\n",
    "\n",
    "gbm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Pass_cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.9</td>\n",
       "      <td>ZNF276</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.9</td>\n",
       "      <td>WBP1L</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.9</td>\n",
       "      <td>STX4</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.9</td>\n",
       "      <td>PGLS</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0.9</td>\n",
       "      <td>RSPH1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33490</th>\n",
       "      <td>0.9</td>\n",
       "      <td>EBF1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33495</th>\n",
       "      <td>0.9</td>\n",
       "      <td>OPLAH</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33500</th>\n",
       "      <td>0.9</td>\n",
       "      <td>AHDC1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33505</th>\n",
       "      <td>0.9</td>\n",
       "      <td>RALY</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33510</th>\n",
       "      <td>0.9</td>\n",
       "      <td>DST</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6703 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Correlation    Gene  Pass_cutoff\n",
       "0              0.9  ZNF276          0.0\n",
       "5              0.9   WBP1L          1.0\n",
       "10             0.9    STX4          0.0\n",
       "15             0.9    PGLS          1.0\n",
       "20             0.9   RSPH1          1.0\n",
       "...            ...     ...          ...\n",
       "33490          0.9    EBF1          1.0\n",
       "33495          0.9   OPLAH          0.0\n",
       "33500          0.9   AHDC1          0.0\n",
       "33505          0.9    RALY          0.0\n",
       "33510          0.9     DST          0.0\n",
       "\n",
       "[6703 rows x 3 columns]"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbm_df_9 = gbm_df.loc[gbm_df['Correlation'] == 0.9]\n",
    "gbm_df_9\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.21139788154557662"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gbm_df_9.Pass_cutoff.sum()/ len(gbm_df_9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#gbm_df = phospho_gene_vs_site(brain)\n",
    "ovar_df = phospho_gene_vs_site(ovar)\n",
    "brca_df = phospho_gene_vs_site(brca)\n",
    "luad_df = phospho_gene_vs_site(luad)\n",
    "hnscc_df = phospho_gene_vs_site(hnscc)\n",
    "lscc_df = phospho_gene_vs_site(lscc)\n",
    "kidney_df = phospho_gene_vs_site(kidney)\n",
    "colon_df = phospho_gene_vs_site(colon)\n",
    "endo_df = phospho_gene_vs_site(endo)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Pass_cutoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.9</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.8</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.7</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.6</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.1</td>\n",
       "      <td>HECW2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31670</th>\n",
       "      <td>0.9</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31671</th>\n",
       "      <td>0.8</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31672</th>\n",
       "      <td>0.7</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31673</th>\n",
       "      <td>0.6</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31674</th>\n",
       "      <td>0.1</td>\n",
       "      <td>TMUB1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31675 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Correlation   Gene  Pass_cutoff\n",
       "0              0.9  HECW2          0.0\n",
       "1              0.8  HECW2          0.0\n",
       "2              0.7  HECW2          0.0\n",
       "3              0.6  HECW2          0.0\n",
       "4              0.1  HECW2          1.0\n",
       "...            ...    ...          ...\n",
       "31670          0.9  TMUB1          1.0\n",
       "31671          0.8  TMUB1          1.0\n",
       "31672          0.7  TMUB1          1.0\n",
       "31673          0.6  TMUB1          1.0\n",
       "31674          0.1  TMUB1          1.0\n",
       "\n",
       "[31675 rows x 3 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_dfs = {'GBM':gbm_df, 'HNSCC':hnscc_df, 'LUAD':luad_df, 'BR':brca_df, 'CO':colon_df, 'OV':ovar_df, \"ccRCC\":kidney_df,\"EC\": endo_df}\n",
    "\n",
    "newdf = pd.DataFrame()\n",
    "for cancer in cancer_dfs:\n",
    "    df = cancer_dfs[cancer]\n",
    "    total = df.Gene.to_list()\n",
    "    total = len(set(total))\n",
    "    test_nums = [0.9,0.8,0.7,0.6]\n",
    "    all_percents = []\n",
    "    for num in test_nums:\n",
    "        \n",
    "        df_num = df[df.Correlation > num]\n",
    "        list1 = df_num.Gene.to_list()\n",
    "        num_corr = len(set(list1))\n",
    "        percent = num_corr/total\n",
    "        \n",
    "        all_percents.append(percent)\n",
    "        \n",
    "    newdf = newdf.append({'Cancer': cancer ,\"percent_0.9 corr\": all_percents[0], \"percent_0.8 corr\":all_percents[1], \"percent_0.7 corr\":all_percents[2],\"percent_0.6 corr\": all_percents[3] } ,ignore_index=True )\n",
    "\n",
    "    \n",
    "    \n",
    "\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "newdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AAAS_S495</th>\n",
       "      <th>AAAS_S495</th>\n",
       "      <th>AAAS_S525</th>\n",
       "      <th>AAAS_S541</th>\n",
       "      <th>AAED1_S12</th>\n",
       "      <th>AAGAB_S196S199</th>\n",
       "      <th>AAGAB_S215</th>\n",
       "      <th>AAGAB_S311</th>\n",
       "      <th>AAGAB_T216S218</th>\n",
       "      <th>AAK1_S14</th>\n",
       "      <th>...</th>\n",
       "      <th>ZZEF1_T1512S1518</th>\n",
       "      <th>ZZEF1_T1521</th>\n",
       "      <th>ZZEF1_T1521T1523</th>\n",
       "      <th>ZZEF1_T2074</th>\n",
       "      <th>ZZEF1_T66</th>\n",
       "      <th>ZZZ3_S113</th>\n",
       "      <th>ZZZ3_S113</th>\n",
       "      <th>ZZZ3_S314</th>\n",
       "      <th>ZZZ3_S391</th>\n",
       "      <th>ZZZ3_S397</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>0.053418</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.118623</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.749345</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.095324</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.012659</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.162407</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>-0.093105</td>\n",
       "      <td>-0.095450</td>\n",
       "      <td>0.069499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.221096</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.245832</td>\n",
       "      <td>-0.427945</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.661599</td>\n",
       "      <td>...</td>\n",
       "      <td>0.368710</td>\n",
       "      <td>-0.235903</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.883682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.520211</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>-1.084975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.368630</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.497238</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.317659</td>\n",
       "      <td>...</td>\n",
       "      <td>1.001173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.818227</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.114993</td>\n",
       "      <td>0.171042</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.037834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>-0.260149</td>\n",
       "      <td>0.390607</td>\n",
       "      <td>0.336969</td>\n",
       "      <td>0.411596</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.118422</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.442111</td>\n",
       "      <td>0.512767</td>\n",
       "      <td>0.515239</td>\n",
       "      <td>-0.538142</td>\n",
       "      <td>-0.132746</td>\n",
       "      <td>0.013019</td>\n",
       "      <td>0.287170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-0.344549</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.082962</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.017669</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.548392</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.128901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.151054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>-0.046331</td>\n",
       "      <td>0.223123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.129488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.291690</td>\n",
       "      <td>-0.257608</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.096117</td>\n",
       "      <td>-0.163778</td>\n",
       "      <td>-0.458433</td>\n",
       "      <td>-0.442097</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.326221</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.424204</td>\n",
       "      <td>0.485515</td>\n",
       "      <td>0.200915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>0.439788</td>\n",
       "      <td>0.799441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.359969</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.669859</td>\n",
       "      <td>0.211198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.085896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.179171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.147963</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.394168</td>\n",
       "      <td>0.393012</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>-0.284134</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.530625</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.645441</td>\n",
       "      <td>0.292919</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.022576</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.769847</td>\n",
       "      <td>0.391365</td>\n",
       "      <td>0.419967</td>\n",
       "      <td>-0.384171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.054770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>-1.037135</td>\n",
       "      <td>0.371373</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.086821</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.275246</td>\n",
       "      <td>-0.064109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.600983</td>\n",
       "      <td>0.305315</td>\n",
       "      <td>0.601319</td>\n",
       "      <td>0.435356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.873856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.247936</td>\n",
       "      <td>-0.459617</td>\n",
       "      <td>-0.328697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>0.557386</td>\n",
       "      <td>1.762413</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.196274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.512274</td>\n",
       "      <td>0.095313</td>\n",
       "      <td>0.328732</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.568578</td>\n",
       "      <td>0.070018</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.286549</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.798986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.471217</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 56271 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            AAAS_S495  AAAS_S495  AAAS_S525  AAAS_S541  AAED1_S12  \\\n",
       "Patient_ID                                                          \n",
       "C3L-00104    0.053418        NaN   0.118623        NaN        NaN   \n",
       "C3L-00365   -0.093105  -0.095450   0.069499        NaN  -1.221096   \n",
       "C3L-00674   -1.084975        NaN        NaN   1.368630        NaN   \n",
       "C3L-00677   -0.260149   0.390607   0.336969   0.411596        NaN   \n",
       "C3L-01040   -0.344549        NaN  -0.082962        NaN        NaN   \n",
       "...               ...        ...        ...        ...        ...   \n",
       "C3N-03183   -0.046331   0.223123        NaN        NaN   0.129488   \n",
       "C3N-03184    0.439788   0.799441        NaN  -0.359969        NaN   \n",
       "C3N-03186   -0.284134        NaN        NaN   0.530625        NaN   \n",
       "C3N-03188   -1.037135   0.371373        NaN        NaN  -0.086821   \n",
       "C3N-03473    0.557386   1.762413        NaN  -0.196274        NaN   \n",
       "\n",
       "            AAGAB_S196S199  AAGAB_S215  AAGAB_S311  AAGAB_T216S218  AAK1_S14  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00104              NaN   -0.749345         NaN             NaN  0.095324   \n",
       "C3L-00365              NaN    0.245832   -0.427945             NaN -0.661599   \n",
       "C3L-00674              NaN         NaN    0.497238             NaN  0.317659   \n",
       "C3L-00677         0.118422         NaN         NaN             NaN       NaN   \n",
       "C3L-01040              NaN   -0.017669         NaN             NaN  0.548392   \n",
       "...                    ...         ...         ...             ...       ...   \n",
       "C3N-03183              NaN   -0.291690   -0.257608             NaN       NaN   \n",
       "C3N-03184        -0.669859    0.211198         NaN             NaN       NaN   \n",
       "C3N-03186              NaN         NaN    0.645441        0.292919       NaN   \n",
       "C3N-03188              NaN   -0.275246   -0.064109             NaN       NaN   \n",
       "C3N-03473              NaN    0.512274    0.095313        0.328732       NaN   \n",
       "\n",
       "            ...  ZZEF1_T1512S1518  ZZEF1_T1521  ZZEF1_T1521T1523  ZZEF1_T2074  \\\n",
       "Patient_ID  ...                                                                 \n",
       "C3L-00104   ...               NaN          NaN               NaN    -0.012659   \n",
       "C3L-00365   ...          0.368710    -0.235903               NaN    -0.883682   \n",
       "C3L-00674   ...          1.001173          NaN               NaN     0.818227   \n",
       "C3L-00677   ...         -0.442111     0.512767          0.515239    -0.538142   \n",
       "C3L-01040   ...               NaN          NaN               NaN     0.128901   \n",
       "...         ...               ...          ...               ...          ...   \n",
       "C3N-03183   ...          0.096117    -0.163778         -0.458433    -0.442097   \n",
       "C3N-03184   ...         -0.085896          NaN               NaN    -0.179171   \n",
       "C3N-03186   ...          0.022576          NaN          0.769847     0.391365   \n",
       "C3N-03188   ...          0.600983     0.305315          0.601319     0.435356   \n",
       "C3N-03473   ...         -0.568578     0.070018               NaN    -0.286549   \n",
       "\n",
       "            ZZEF1_T66  ZZZ3_S113  ZZZ3_S113  ZZZ3_S314  ZZZ3_S391  ZZZ3_S397  \n",
       "Patient_ID                                                                    \n",
       "C3L-00104         NaN  -0.162407        NaN        NaN        NaN        NaN  \n",
       "C3L-00365         NaN   0.520211        NaN        NaN        NaN        NaN  \n",
       "C3L-00674         NaN   0.114993   0.171042        NaN        NaN  -1.037834  \n",
       "C3L-00677   -0.132746   0.013019   0.287170        NaN        NaN        NaN  \n",
       "C3L-01040         NaN  -0.151054        NaN        NaN        NaN        NaN  \n",
       "...               ...        ...        ...        ...        ...        ...  \n",
       "C3N-03183         NaN   0.326221        NaN   0.424204   0.485515   0.200915  \n",
       "C3N-03184         NaN   0.147963        NaN  -0.394168   0.393012        NaN  \n",
       "C3N-03186    0.419967  -0.384171        NaN   0.054770        NaN        NaN  \n",
       "C3N-03188         NaN  -0.873856        NaN  -0.247936  -0.459617  -0.328697  \n",
       "C3N-03473         NaN        NaN        NaN   0.798986        NaN   0.471217  \n",
       "\n",
       "[99 rows x 56271 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phospho_site = brain.get_phosphoproteomics(\"tumor\")\n",
    "    #drop database_id and peptide \n",
    "\n",
    "phospho_site = u.reduce_multiindex(phospho_site, levels_to_drop = 2)\n",
    "phospho_site = u.reduce_multiindex(phospho_site, levels_to_drop = 2)\n",
    "phospho_gene = u.reduce_multiindex(phospho_site, levels_to_drop = 1)\n",
    "genes = phospho_gene.columns.to_list() \n",
    "\n",
    "genes = list(set(genes))#get unique list of all genes\n",
    "\n",
    "phospho_site.columns =phospho_site.columns.map('_'.join) #join gene and site\n",
    "phospho_site = phospho_site.loc[:,~phospho_site.T.duplicated(keep='first')] # drop columns where values are idendical\n",
    "\n",
    "phospho_site = phospho_site.dropna(thresh=20, axis=1) #drop columns with less than 30 values \n",
    "    \n",
    "phospho_site"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PGLS_S49</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>-0.008787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>-0.186648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>0.391807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>0.291408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>0.250515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>0.516006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>-0.995225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>0.221631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>-0.438092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>-0.325650</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PGLS_S49\n",
       "Patient_ID          \n",
       "C3L-00104  -0.008787\n",
       "C3L-00365  -0.186648\n",
       "C3L-00674   0.391807\n",
       "C3L-00677   0.291408\n",
       "C3L-01040   0.250515\n",
       "...              ...\n",
       "C3N-03183   0.516006\n",
       "C3N-03184  -0.995225\n",
       "C3N-03186   0.221631\n",
       "C3N-03188  -0.438092\n",
       "C3N-03473  -0.325650\n",
       "\n",
       "[99 rows x 1 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = \"PGLS\"\n",
    "phospho_gene_test = phospho_site.filter(regex=gene +\"_\",axis=1)\n",
    "phospho_gene_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_df = phospho_site\n",
    "\n",
    "#result = test_df.corr(method = \"pearson\", min_periods = 20)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "bad operand type for abs(): 'list'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-92-96e6406fd838>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mremove\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1.0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0mcleanedList\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mx\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mvalues\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misnan\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m \u001b[0mabs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcleanedList\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: bad operand type for abs(): 'list'"
     ]
    }
   ],
   "source": [
    "values = result.values.tolist()\n",
    "from pandas.core.common import flatten\n",
    "values = list(flatten(values))\n",
    "values = list(set(values))\n",
    "\n",
    "values.remove(1.0)\n",
    "cleanedList = [x for x in values if (math.isnan(x) != True)]\n",
    "abs(cleanedList)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abs_val_corrs = [1.0,0.95]\n",
    "result = all(x >= 0.9 for x in abs_val_corrs)\n",
    "\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "def phospho_gene_vs_site_test(test_df):\n",
    "    phospho_site = test_df\n",
    "    df = pd.DataFrame() \n",
    "    for gene in genes:\n",
    "        phospho_gene = phospho_site.filter(regex=\"PGLS\" +\"_\",axis=1)\n",
    "        prot_genes_list = create_prot_list(phospho_gene)\n",
    "        test = False\n",
    "        if len(prot_genes_list) > 1: # only find correlation if there is more than one site \n",
    "        \n",
    "            site_corrs = phospho_gene.corr(method = \"pearson\", min_periods = 20) #pairwise correclations\n",
    "            site_corrs = site_corrs.values.tolist()#array to list\n",
    "            from pandas.core.common import flatten\n",
    "            site_corrs = list(flatten(site_corrs))\n",
    "            site_corrs = [x for x in site_corrs if (math.isnan(x) != True)] #remove na \n",
    "       \n",
    "            #site_corrs.remove(1.0)# remove comparison with same gene\n",
    "            site_corrs = list(set(site_corrs))# remove duplicate pairwise correlations\n",
    "      \n",
    "          \n",
    "            abs_val_corrs = [abs(x) for x in site_corrs]\n",
    "            cutoffs = [0.9,0.8,0.7,0.6,0.1]\n",
    "            test = True\n",
    "            #for corr in cutoffs:\n",
    "\n",
    "             #   result = all(x >= corr for x in abs_val_corrs)\n",
    "              #  if not abs_val_corrs:\n",
    "               #     result = False\n",
    "                #df = df.append({'Gene': gene,\"Correlation\": corr,\"Pass_cutoff\": result}, ignore_index=True)\n",
    "\n",
    "        \n",
    "    return test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-22-e2e47cffe611>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mgbm_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mphospho_gene_vs_site_test\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtest_df\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mgbm_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-21-37a40f23017b>\u001b[0m in \u001b[0;36mphospho_gene_vs_site_test\u001b[0;34m(test_df)\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mgene\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m         \u001b[0mphospho_gene\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mphospho_site\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfilter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mregex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"PGLS\"\u001b[0m \u001b[0;34m+\u001b[0m\u001b[0;34m\"_\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m         \u001b[0mprot_genes_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcreate_prot_list\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mphospho_gene\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m         \u001b[0mtest\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36mfilter\u001b[0;34m(self, items, like, regex, axis)\u001b[0m\n\u001b[1;32m   4711\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4712\u001b[0m             \u001b[0mmatcher\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mre\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mregex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4713\u001b[0;31m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4714\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4715\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mmap\u001b[0;34m(self, mapper, na_action)\u001b[0m\n\u001b[1;32m   4563\u001b[0m         \u001b[0;32mfrom\u001b[0m \u001b[0mpandas\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindexes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmulti\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mMultiIndex\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4564\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4565\u001b[0;31m         \u001b[0mnew_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_map_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmapper\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mna_action\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mna_action\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4566\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4567\u001b[0m         \u001b[0mattributes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_attributes_dict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/base.py\u001b[0m in \u001b[0;36m_map_values\u001b[0;34m(self, mapper, na_action)\u001b[0m\n\u001b[1;32m   1152\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1153\u001b[0m         \u001b[0;31m# mapper is a function\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1154\u001b[0;31m         \u001b[0mnew_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmap_f\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmapper\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1155\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1156\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mnew_values\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/lib.pyx\u001b[0m in \u001b[0;36mpandas._libs.lib.map_infer\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36mf\u001b[0;34m(x)\u001b[0m\n\u001b[1;32m   4707\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mregex\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4708\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4709\u001b[0;31m             \u001b[0;32mdef\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4710\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mmatcher\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msearch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mensure_str\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4711\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "gbm_df = phospho_gene_vs_site_test(test_df)\n",
    "\n",
    "gbm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    }
   ],
   "source": [
    "phospho_site = test_df\n",
    "df = pd.DataFrame() \n",
    "\n",
    "phospho_gene = phospho_site.filter(regex=\"PGLS\" +\"_\",axis=1)\n",
    "prot_genes_list = create_prot_list(phospho_gene)\n",
    "test = False\n",
    "if len(prot_genes_list) > 1: # only find correlation if there is more than one site \n",
    "    test = True\n",
    "print(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
