{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pancancer Boxplot for PTEN trans effect on MCM2 phosphoproteomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook creates a boxplot for the trans effect of PTEN cnv deletions in 8 cancers. These 8 cancers were chosen because they have enough samples with PTEN cnv deletions to do a t-test. Only Ccrcc does not have enough samples with deletions. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import gseapy as gp\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "\n",
    "import sys\n",
    "import plot_utils as ut"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Functions:\n",
    "\n",
    "format_df prepares a data frame with PTEN proteomics and PTEN mutation types from get_genotype_all_vars.\n",
    "\n",
    "format_pval_annotation is used to add marks to the boxplot. A star represents significant p-value and \"ns\" represents a nonsignificant p-value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Returns a dataframe with proteomics and mutation type\n",
    "\n",
    "def format_df(cancer_object, trans_gene, gene_in = 'PTEN', drop_level = 3):\n",
    "    mut_type = cancer_object.get_genotype_all_vars(gene_in)\n",
    "    \n",
    "    if cancer_object.get_cancer_type() not in ('luad'):\n",
    "        # Keep only tumor samples from proteomics\n",
    "        prot_and_mutations = cancer_object.join_omics_to_mutations(\n",
    "            mutations_genes = [gene_in], omics_df_name = 'phosphoproteomics', omics_genes = trans_gene)\n",
    "        prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "        \n",
    "        # Reduce a multiindex \n",
    "        if isinstance(prot_and_mutations.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "            prot_and_mutations = cancer_object.reduce_multiindex(prot_and_mutations,\n",
    "                                                                 levels_to_drop=[drop_level], flatten=True)\n",
    "\n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        mut_type = mut_type[['Mutation']] \n",
    "        prot_df = prot_and_mutations.iloc[:,:-4] # Keep only proteomics\n",
    "        merged = prot_df.join(mut_type)\n",
    "        \n",
    "        # Keep only Wildtype and deletion\n",
    "        compare = ['Wildtype_Tumor','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        del_wt = merged[get]\n",
    "    \n",
    "    \n",
    "    # Luad has no somatic mutations for PTEN which changes some things\n",
    "    else: \n",
    "        # get_genotype_all_vars add cnv data under the column PTEN\n",
    "        mut_type = mut_type.drop(columns= gene_in)\n",
    "        # different code because no somatic mutation data for pten (can't join to somatic mutations)\n",
    "        omics = cancer_object.join_omics_to_omics(df1_name = 'CNV', df2_name='phosphoproteomics',genes1= gene_in, \n",
    "            genes2=trans_gene)\n",
    "        omics = cancer_object.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "        omics = omics.drop(columns='PTEN_CNV__')\n",
    "        # Get only tumor samples\n",
    "        p = l.get_proteomics(tissue_type='tumor')\n",
    "        tumor_ids = list(p.index)\n",
    "        get = omics.index.isin(tumor_ids)\n",
    "        omics = omics[get]\n",
    "        # Merge Mutation column from get_genotype_all_vars (includes cnv) with proteomics\n",
    "        merged = omics.join(mut_type) \n",
    "        # Keep only No_Mutation (wildtype) and deletion\n",
    "        compare = ['No_Mutation','Deletion']\n",
    "        get = merged['Mutation'].isin(compare)\n",
    "        del_wt = merged[get]\n",
    "        del_wt['Mutation'] = np.where(\n",
    "            del_wt['Mutation'] == 'No_Mutation', 'Wildtype_Tumor', 'Deletion')\n",
    "\n",
    "    return del_wt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAsQR</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_GLLYDsDEEDEERPAR</th>\n",
       "      <th>MCM2_phosphoproteomics_S170_ATEDGEEDEEMIEsIENLEDLK</th>\n",
       "      <th>MCM2_phosphoproteomics_S26S27_RGNDPLTssPGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S27_GNDPLTSsPGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S381_IQEsPGK</th>\n",
       "      <th>MCM2_phosphoproteomics_S40S41_TDALTssPGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S41M77_RTDALTSsPGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGmER</th>\n",
       "      <th>MCM2_phosphoproteomics_T158_QVERAtEDGEEDEEMIESIENLEDLK</th>\n",
       "      <th>MCM2_phosphoproteomics_T25_RRGNDPLtSSPGR</th>\n",
       "      <th>PTEN_Mutation__</th>\n",
       "      <th>PTEN_Location__</th>\n",
       "      <th>PTEN_Mutation_Status__</th>\n",
       "      <th>Sample_Status__</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00081</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0437</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0374</td>\n",
       "      <td>0.5615</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3830</td>\n",
       "      <td>0.3563</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.9446</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00415</th>\n",
       "      <td>1.4818</td>\n",
       "      <td>0.7056</td>\n",
       "      <td>1.1065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.2280</td>\n",
       "      <td>0.9578</td>\n",
       "      <td>0.7101</td>\n",
       "      <td>-0.3497</td>\n",
       "      <td>0.3243</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00445</th>\n",
       "      <td>0.0843</td>\n",
       "      <td>0.5319</td>\n",
       "      <td>-0.0160</td>\n",
       "      <td>0.5257</td>\n",
       "      <td>1.3807</td>\n",
       "      <td>1.1300</td>\n",
       "      <td>0.6222</td>\n",
       "      <td>1.7907</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3802</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00568</th>\n",
       "      <td>-0.1885</td>\n",
       "      <td>0.6663</td>\n",
       "      <td>0.4402</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1167</td>\n",
       "      <td>-0.1636</td>\n",
       "      <td>1.4081</td>\n",
       "      <td>1.3506</td>\n",
       "      <td>0.8407</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00603</th>\n",
       "      <td>0.0742</td>\n",
       "      <td>1.4027</td>\n",
       "      <td>2.1734</td>\n",
       "      <td>2.3722</td>\n",
       "      <td>1.4256</td>\n",
       "      <td>1.1982</td>\n",
       "      <td>1.1453</td>\n",
       "      <td>0.7978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8536</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03886</th>\n",
       "      <td>1.8236</td>\n",
       "      <td>2.6035</td>\n",
       "      <td>2.4655</td>\n",
       "      <td>1.9592</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.3977</td>\n",
       "      <td>2.6647</td>\n",
       "      <td>2.9886</td>\n",
       "      <td>1.8224</td>\n",
       "      <td>3.0354</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04124</th>\n",
       "      <td>-0.4913</td>\n",
       "      <td>-0.4429</td>\n",
       "      <td>-0.4319</td>\n",
       "      <td>1.7152</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1387</td>\n",
       "      <td>-0.4164</td>\n",
       "      <td>-1.0705</td>\n",
       "      <td>-1.2665</td>\n",
       "      <td>-0.3636</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04127</th>\n",
       "      <td>0.1444</td>\n",
       "      <td>-0.6074</td>\n",
       "      <td>0.4294</td>\n",
       "      <td>-1.2400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3851</td>\n",
       "      <td>-0.2032</td>\n",
       "      <td>-1.5512</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.4215</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04155</th>\n",
       "      <td>0.8447</td>\n",
       "      <td>1.8117</td>\n",
       "      <td>1.3487</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.3865</td>\n",
       "      <td>0.2179</td>\n",
       "      <td>1.3739</td>\n",
       "      <td>2.0212</td>\n",
       "      <td>1.2022</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-04162</th>\n",
       "      <td>0.2213</td>\n",
       "      <td>0.8798</td>\n",
       "      <td>0.5217</td>\n",
       "      <td>0.0050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.6922</td>\n",
       "      <td>0.3414</td>\n",
       "      <td>0.6274</td>\n",
       "      <td>0.0614</td>\n",
       "      <td>0.6960</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAsQR  \\\n",
       "Patient_ID                                                             \n",
       "C3L-00081                                                 NaN          \n",
       "C3L-00415                                              1.4818          \n",
       "C3L-00445                                              0.0843          \n",
       "C3L-00568                                             -0.1885          \n",
       "C3L-00603                                              0.0742          \n",
       "...                                                       ...          \n",
       "C3N-03886                                              1.8236          \n",
       "C3N-04124                                             -0.4913          \n",
       "C3N-04127                                              0.1444          \n",
       "C3N-04155                                              0.8447          \n",
       "C3N-04162                                              0.2213          \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S139_GLLYDsDEEDEERPAR  \\\n",
       "Patient_ID                                                 \n",
       "C3L-00081                                         1.0437   \n",
       "C3L-00415                                         0.7056   \n",
       "C3L-00445                                         0.5319   \n",
       "C3L-00568                                         0.6663   \n",
       "C3L-00603                                         1.4027   \n",
       "...                                                  ...   \n",
       "C3N-03886                                         2.6035   \n",
       "C3N-04124                                        -0.4429   \n",
       "C3N-04127                                        -0.6074   \n",
       "C3N-04155                                         1.8117   \n",
       "C3N-04162                                         0.8798   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S170_ATEDGEEDEEMIEsIENLEDLK  \\\n",
       "Patient_ID                                                       \n",
       "C3L-00081                                                 NaN    \n",
       "C3L-00415                                              1.1065    \n",
       "C3L-00445                                             -0.0160    \n",
       "C3L-00568                                              0.4402    \n",
       "C3L-00603                                              2.1734    \n",
       "...                                                       ...    \n",
       "C3N-03886                                              2.4655    \n",
       "C3N-04124                                             -0.4319    \n",
       "C3N-04127                                              0.4294    \n",
       "C3N-04155                                              1.3487    \n",
       "C3N-04162                                              0.5217    \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S26S27_RGNDPLTssPGR  \\\n",
       "Patient_ID                                               \n",
       "C3L-00081                                      -0.0374   \n",
       "C3L-00415                                          NaN   \n",
       "C3L-00445                                       0.5257   \n",
       "C3L-00568                                          NaN   \n",
       "C3L-00603                                       2.3722   \n",
       "...                                                ...   \n",
       "C3N-03886                                       1.9592   \n",
       "C3N-04124                                       1.7152   \n",
       "C3N-04127                                      -1.2400   \n",
       "C3N-04155                                          NaN   \n",
       "C3N-04162                                       0.0050   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S27_GNDPLTSsPGR  \\\n",
       "Patient_ID                                           \n",
       "C3L-00081                                   0.5615   \n",
       "C3L-00415                                      NaN   \n",
       "C3L-00445                                   1.3807   \n",
       "C3L-00568                                      NaN   \n",
       "C3L-00603                                   1.4256   \n",
       "...                                            ...   \n",
       "C3N-03886                                      NaN   \n",
       "C3N-04124                                      NaN   \n",
       "C3N-04127                                      NaN   \n",
       "C3N-04155                                      NaN   \n",
       "C3N-04162                                      NaN   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S381_IQEsPGK  \\\n",
       "Patient_ID                                        \n",
       "C3L-00081                                   NaN   \n",
       "C3L-00415                               -1.2280   \n",
       "C3L-00445                                1.1300   \n",
       "C3L-00568                                1.1167   \n",
       "C3L-00603                                1.1982   \n",
       "...                                         ...   \n",
       "C3N-03886                                3.3977   \n",
       "C3N-04124                               -1.1387   \n",
       "C3N-04127                               -0.3851   \n",
       "C3N-04155                                1.3865   \n",
       "C3N-04162                                1.6922   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S40S41_TDALTssPGR  \\\n",
       "Patient_ID                                             \n",
       "C3L-00081                                     0.3830   \n",
       "C3L-00415                                     0.9578   \n",
       "C3L-00445                                     0.6222   \n",
       "C3L-00568                                    -0.1636   \n",
       "C3L-00603                                     1.1453   \n",
       "...                                              ...   \n",
       "C3N-03886                                     2.6647   \n",
       "C3N-04124                                    -0.4164   \n",
       "C3N-04127                                    -0.2032   \n",
       "C3N-04155                                     0.2179   \n",
       "C3N-04162                                     0.3414   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S41M77_RTDALTSsPGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGmER  \\\n",
       "Patient_ID                                                                                 \n",
       "C3L-00081                                              0.3563                              \n",
       "C3L-00415                                              0.7101                              \n",
       "C3L-00445                                              1.7907                              \n",
       "C3L-00568                                              1.4081                              \n",
       "C3L-00603                                              0.7978                              \n",
       "...                                                       ...                              \n",
       "C3N-03886                                              2.9886                              \n",
       "C3N-04124                                             -1.0705                              \n",
       "C3N-04127                                             -1.5512                              \n",
       "C3N-04155                                              1.3739                              \n",
       "C3N-04162                                              0.6274                              \n",
       "\n",
       "Name        MCM2_phosphoproteomics_T158_QVERAtEDGEEDEEMIESIENLEDLK  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00081                                                 NaN        \n",
       "C3L-00415                                             -0.3497        \n",
       "C3L-00445                                                 NaN        \n",
       "C3L-00568                                              1.3506        \n",
       "C3L-00603                                                 NaN        \n",
       "...                                                       ...        \n",
       "C3N-03886                                              1.8224        \n",
       "C3N-04124                                             -1.2665        \n",
       "C3N-04127                                                 NaN        \n",
       "C3N-04155                                              2.0212        \n",
       "C3N-04162                                              0.0614        \n",
       "\n",
       "Name        MCM2_phosphoproteomics_T25_RRGNDPLtSSPGR   PTEN_Mutation__  \\\n",
       "Patient_ID                                                               \n",
       "C3L-00081                                     1.9446  [Wildtype_Tumor]   \n",
       "C3L-00415                                     0.3243  [Wildtype_Tumor]   \n",
       "C3L-00445                                     0.3802  [Wildtype_Tumor]   \n",
       "C3L-00568                                     0.8407  [Wildtype_Tumor]   \n",
       "C3L-00603                                     0.8536  [Wildtype_Tumor]   \n",
       "...                                              ...               ...   \n",
       "C3N-03886                                     3.0354  [Wildtype_Tumor]   \n",
       "C3N-04124                                    -0.3636  [Wildtype_Tumor]   \n",
       "C3N-04127                                    -0.4215  [Wildtype_Tumor]   \n",
       "C3N-04155                                     1.2022  [Wildtype_Tumor]   \n",
       "C3N-04162                                     0.6960  [Wildtype_Tumor]   \n",
       "\n",
       "Name       PTEN_Location__ PTEN_Mutation_Status__ Sample_Status__  \n",
       "Patient_ID                                                         \n",
       "C3L-00081    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "C3L-00415    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "C3L-00445    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "C3L-00568    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "C3L-00603    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "...                    ...                    ...             ...  \n",
       "C3N-03886    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "C3N-04124    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "C3N-04127    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "C3N-04155    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "C3N-04162    [No_mutation]         Wildtype_Tumor           Tumor  \n",
       "\n",
       "[110 rows x 14 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_and_mutations = l.join_omics_to_mutations(\n",
    "    mutations_genes = ['PTEN'], omics_df_name = 'phosphoproteomics', omics_genes = 'MCM2')\n",
    "prot_and_mutations = prot_and_mutations[prot_and_mutations.Sample_Status == \"Tumor\"] # drop Normal samples\n",
    "\n",
    "# Reduce a multiindex \n",
    "if isinstance(prot_and_mutations.keys(), pd.core.indexes.multi.MultiIndex):\n",
    "    prot_and_mutations = ls.reduce_multiindex(prot_and_mutations, levels_to_drop=[3], flatten=True)\n",
    "prot_and_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-00001.N, C3L-00009.N, C3L-00080.N, C3L-00083.N, C3L-00093.N, C3L-00094.N, C3L-00095.N, C3L-00140.N, C3L-00144.N, C3L-00263.N, C3L-00279.N, C3L-00368.N, C3L-00412.N, C3L-00422.N, C3L-00510.N, C3L-00604.N, C3L-00893.N, C3L-00913.N, C3L-00973.N, C3L-01330.N, C3L-01632.N, C3L-01682.N, C3L-01683.N, C3L-01889.N, C3L-01890, C3L-01890.N, C3L-01924.N, C3L-02219.N, C3L-02345.N, C3L-02348.N, C3L-02350.N, C3L-02365.N, C3L-02508.N, C3L-02549.N, C3N-00167.N, C3N-00169.N, C3N-00175.N, C3N-00180.N, C3N-00199.N, C3N-00203.N, C3N-00217.N, C3N-00223.N, C3N-00293.N, C3N-00433.N, C3N-00546.N, C3N-00547.N, C3N-00549.N, C3N-00550.N, C3N-00551.N, C3N-00552.N, C3N-00556.N, C3N-00559.N, C3N-00560.N, C3N-00572.N, C3N-00574.N, C3N-00578.N, C3N-00579.N, C3N-00580.N, C3N-00704.N, C3N-00737.N, C3N-00738.N, C3N-00959.N, C3N-01016.N, C3N-01021.N, C3N-01023.N, C3N-01024.N, C3N-01030.N, C3N-01071.N, C3N-01072.N, C3N-01405.N, C3N-01410.N, C3N-01413.N, C3N-01414.N, C3N-01415.N, C3N-01416.N, C3N-01488.N, C3N-01489.N, C3N-01799.N, C3N-01823.N, C3N-02000.N, C3N-02002.N, C3N-02003.N, C3N-02067.N, C3N-02087.N, C3N-02089.N, C3N-02145.N, C3N-02149.N, C3N-02155.N, C3N-02158.N, C3N-02379.N, C3N-02380.N, C3N-02421.N, C3N-02423.N, C3N-02424.N, C3N-02433.N, C3N-02529.N, C3N-02572.N, C3N-02582.N, C3N-02586.N, C3N-02587.N, C3N-02588.N, C3N-02729.N (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>MCM2_phosphoproteomics_AIPELDAYEAEGLALDDEDVEELTAsQR_NP_004517.2</th>\n",
       "      <th>MCM2_phosphoproteomics_GLLYDsDEEDEERPAR_NP_004517.2</th>\n",
       "      <th>MCM2_phosphoproteomics_RGNDPLTSsPGR_NP_004517.2</th>\n",
       "      <th>MCM2_phosphoproteomics_IQEsPGK_NP_004517.2</th>\n",
       "      <th>MCM2_phosphoproteomics_TDALTssPGR_NP_004517.2</th>\n",
       "      <th>MCM2_phosphoproteomics_RTDALtSSPGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGmER_NP_004517.2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00001</th>\n",
       "      <td>-0.3442</td>\n",
       "      <td>-2.1796</td>\n",
       "      <td>-1.0381</td>\n",
       "      <td>-2.8939</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.9930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00009</th>\n",
       "      <td>-1.3712</td>\n",
       "      <td>-0.0050</td>\n",
       "      <td>-0.3403</td>\n",
       "      <td>0.2616</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00080</th>\n",
       "      <td>0.7855</td>\n",
       "      <td>2.4530</td>\n",
       "      <td>2.5073</td>\n",
       "      <td>4.6866</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.4019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00083</th>\n",
       "      <td>-0.2696</td>\n",
       "      <td>-1.4357</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00093</th>\n",
       "      <td>-0.4590</td>\n",
       "      <td>0.8262</td>\n",
       "      <td>-0.2096</td>\n",
       "      <td>-1.7823</td>\n",
       "      <td>-0.1550</td>\n",
       "      <td>-0.8178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02582.N</th>\n",
       "      <td>0.5485</td>\n",
       "      <td>-1.2192</td>\n",
       "      <td>-0.6149</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1586</td>\n",
       "      <td>0.2833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02586.N</th>\n",
       "      <td>-0.6541</td>\n",
       "      <td>-3.0871</td>\n",
       "      <td>-1.9174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.3776</td>\n",
       "      <td>-1.1223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02587.N</th>\n",
       "      <td>-1.4323</td>\n",
       "      <td>-1.7396</td>\n",
       "      <td>-2.3258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02588.N</th>\n",
       "      <td>0.6357</td>\n",
       "      <td>-1.7150</td>\n",
       "      <td>-1.7763</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.4599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-02729.N</th>\n",
       "      <td>-0.7019</td>\n",
       "      <td>-2.9489</td>\n",
       "      <td>-2.2395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.3284</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>211 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name         MCM2_phosphoproteomics_AIPELDAYEAEGLALDDEDVEELTAsQR_NP_004517.2  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00001                                              -0.3442                 \n",
       "C3L-00009                                              -1.3712                 \n",
       "C3L-00080                                               0.7855                 \n",
       "C3L-00083                                              -0.2696                 \n",
       "C3L-00093                                              -0.4590                 \n",
       "...                                                        ...                 \n",
       "C3N-02582.N                                             0.5485                 \n",
       "C3N-02586.N                                            -0.6541                 \n",
       "C3N-02587.N                                            -1.4323                 \n",
       "C3N-02588.N                                             0.6357                 \n",
       "C3N-02729.N                                            -0.7019                 \n",
       "\n",
       "Name         MCM2_phosphoproteomics_GLLYDsDEEDEERPAR_NP_004517.2  \\\n",
       "Patient_ID                                                         \n",
       "C3L-00001                                              -2.1796     \n",
       "C3L-00009                                              -0.0050     \n",
       "C3L-00080                                               2.4530     \n",
       "C3L-00083                                              -1.4357     \n",
       "C3L-00093                                               0.8262     \n",
       "...                                                        ...     \n",
       "C3N-02582.N                                            -1.2192     \n",
       "C3N-02586.N                                            -3.0871     \n",
       "C3N-02587.N                                            -1.7396     \n",
       "C3N-02588.N                                            -1.7150     \n",
       "C3N-02729.N                                            -2.9489     \n",
       "\n",
       "Name         MCM2_phosphoproteomics_RGNDPLTSsPGR_NP_004517.2  \\\n",
       "Patient_ID                                                     \n",
       "C3L-00001                                            -1.0381   \n",
       "C3L-00009                                            -0.3403   \n",
       "C3L-00080                                             2.5073   \n",
       "C3L-00083                                             0.0017   \n",
       "C3L-00093                                            -0.2096   \n",
       "...                                                      ...   \n",
       "C3N-02582.N                                          -0.6149   \n",
       "C3N-02586.N                                          -1.9174   \n",
       "C3N-02587.N                                          -2.3258   \n",
       "C3N-02588.N                                          -1.7763   \n",
       "C3N-02729.N                                          -2.2395   \n",
       "\n",
       "Name         MCM2_phosphoproteomics_IQEsPGK_NP_004517.2  \\\n",
       "Patient_ID                                                \n",
       "C3L-00001                                       -2.8939   \n",
       "C3L-00009                                        0.2616   \n",
       "C3L-00080                                        4.6866   \n",
       "C3L-00083                                           NaN   \n",
       "C3L-00093                                       -1.7823   \n",
       "...                                                 ...   \n",
       "C3N-02582.N                                         NaN   \n",
       "C3N-02586.N                                         NaN   \n",
       "C3N-02587.N                                         NaN   \n",
       "C3N-02588.N                                         NaN   \n",
       "C3N-02729.N                                         NaN   \n",
       "\n",
       "Name         MCM2_phosphoproteomics_TDALTssPGR_NP_004517.2  \\\n",
       "Patient_ID                                                   \n",
       "C3L-00001                                              NaN   \n",
       "C3L-00009                                              NaN   \n",
       "C3L-00080                                              NaN   \n",
       "C3L-00083                                              NaN   \n",
       "C3L-00093                                          -0.1550   \n",
       "...                                                    ...   \n",
       "C3N-02582.N                                        -0.1586   \n",
       "C3N-02586.N                                        -1.3776   \n",
       "C3N-02587.N                                            NaN   \n",
       "C3N-02588.N                                            NaN   \n",
       "C3N-02729.N                                            NaN   \n",
       "\n",
       "Name         MCM2_phosphoproteomics_RTDALtSSPGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGmER_NP_004517.2  \n",
       "Patient_ID                                                                                      \n",
       "C3L-00001                                              -0.9930                                  \n",
       "C3L-00009                                              -0.6235                                  \n",
       "C3L-00080                                               2.4019                                  \n",
       "C3L-00083                                              -0.2083                                  \n",
       "C3L-00093                                              -0.8178                                  \n",
       "...                                                        ...                                  \n",
       "C3N-02582.N                                             0.2833                                  \n",
       "C3N-02586.N                                            -1.1223                                  \n",
       "C3N-02587.N                                             0.2958                                  \n",
       "C3N-02588.N                                            -1.4599                                  \n",
       "C3N-02729.N                                            -1.3284                                  \n",
       "\n",
       "[211 rows x 6 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omics = l.join_omics_to_omics(df1_name = 'CNV', df2_name='phosphoproteomics',genes1= 'PTEN', \n",
    "            genes2='MCM2')\n",
    "omics = l.reduce_multiindex(omics, levels_to_drop = 1, flatten = True)\n",
    "omics = omics.drop(columns='PTEN_CNV__')\n",
    "omics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_pval_annotation(pval_symbol, x1, x2, line_start = .05, line_height=.05):\n",
    "    # for manual adjustment to pval annotations\n",
    "    \n",
    "    y, h = line_start, line_height\n",
    "    plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "             [y, y+h, y+h, y], #vertical line\n",
    "             lw=1.5, color= '.3')\n",
    "    plt.text((x1+x2)*.5, # half between x coord\n",
    "             y+h, pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 1: Create data frames with PTEN cnv deletion and Proteomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each cancer needs a data frame containing only samples that have PTEN cnv deletions or PTEN wildtype tumors and PTEN proteomics."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, load in cancer data sets from cptac. Note: Keep the variable names the same because they are used in format_df to reduce multiindexs found in certain cancer data sets. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that luad index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The HNSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that lscc index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LUAD dataset is under publication embargo until July 01, 2020. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that ovarian index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 4)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that brca index is up-to-date...   \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    }
   ],
   "source": [
    "en = cptac.Endometrial()\n",
    "h = cptac.Hnscc()\n",
    "l = cptac.Luad()\n",
    "ls = cptac.Lscc()\n",
    "o = cptac.Ovarian()\n",
    "col = cptac.Colon()\n",
    "g = cptac.Gbm()\n",
    "b = cptac.Brca()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, use format_df to create the specific data frame for each cancer. Use wrap_ttest to test PTEN deletion vs. wildtype. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gbm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 72 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-01834 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 82 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QR</th>\n",
       "      <th>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QREAAER</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPAR</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPARK</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPAR</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPARK</th>\n",
       "      <th>MCM2_phosphoproteomics_S26S27_GNDPLTS*S*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S27_GNDPLTSS*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S27_RGNDPLTSS*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S27S31_GNDPLTSS*PGRS*SR</th>\n",
       "      <th>MCM2_phosphoproteomics_S27S31_RGNDPLTSS*PGRS*SR</th>\n",
       "      <th>MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER</th>\n",
       "      <th>MCM2_phosphoproteomics_S41_RTDALTSS*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S41_TDALTSS*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S41_TDALTSS*PGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER</th>\n",
       "      <th>MCM2_phosphoproteomics_T39_TDALT*SSPGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER</th>\n",
       "      <th>MCM2_phosphoproteomics_T59_RTDALTSSPGRDLPPFEDESEGLLGT*EGPLEEEEDGEELIGDGMER</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>0.250213</td>\n",
       "      <td>0.469123</td>\n",
       "      <td>0.553357</td>\n",
       "      <td>1.007214</td>\n",
       "      <td>0.116577</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.526739</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.846266</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.743703</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>0.680272</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.592142</td>\n",
       "      <td>1.128073</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.940525</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.48957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.212528</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.843078</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>-1.100628</td>\n",
       "      <td>1.074279</td>\n",
       "      <td>-0.417157</td>\n",
       "      <td>-0.105052</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.412701</td>\n",
       "      <td>-0.455602</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.108264</td>\n",
       "      <td>-0.367349</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.049988</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.539857</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.118192</td>\n",
       "      <td>1.192966</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.194149</td>\n",
       "      <td>-0.680688</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.802540</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-1.149843</td>\n",
       "      <td>-1.269670</td>\n",
       "      <td>-1.141344</td>\n",
       "      <td>-0.707192</td>\n",
       "      <td>-0.493904</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.524123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.339159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.556890</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QR  \\\n",
       "Patient_ID                                                              \n",
       "C3L-00104                                            0.250213           \n",
       "C3L-00365                                            0.680272           \n",
       "C3L-00674                                           -1.100628           \n",
       "C3L-00677                                                 NaN           \n",
       "C3L-01040                                           -1.149843           \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QREAAER  \\\n",
       "Patient_ID                                                                   \n",
       "C3L-00104                                            0.469123                \n",
       "C3L-00365                                                 NaN                \n",
       "C3L-00674                                            1.074279                \n",
       "C3L-00677                                            0.539857                \n",
       "C3L-01040                                           -1.269670                \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPAR  \\\n",
       "Patient_ID                                                  \n",
       "C3L-00104                                        0.553357   \n",
       "C3L-00365                                        0.592142   \n",
       "C3L-00674                                       -0.417157   \n",
       "C3L-00677                                             NaN   \n",
       "C3L-01040                                       -1.141344   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPARK  \\\n",
       "Patient_ID                                                   \n",
       "C3L-00104                                         1.007214   \n",
       "C3L-00365                                         1.128073   \n",
       "C3L-00674                                        -0.105052   \n",
       "C3L-00677                                         0.118192   \n",
       "C3L-01040                                        -0.707192   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPAR  \\\n",
       "Patient_ID                                                   \n",
       "C3L-00104                                         0.116577   \n",
       "C3L-00365                                              NaN   \n",
       "C3L-00674                                              NaN   \n",
       "C3L-00677                                         1.192966   \n",
       "C3L-01040                                        -0.493904   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPARK  \\\n",
       "Patient_ID                                                    \n",
       "C3L-00104                                               NaN   \n",
       "C3L-00365                                               NaN   \n",
       "C3L-00674                                               NaN   \n",
       "C3L-00677                                               NaN   \n",
       "C3L-01040                                               NaN   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S26S27_GNDPLTS*S*PGR  \\\n",
       "Patient_ID                                                \n",
       "C3L-00104                                           NaN   \n",
       "C3L-00365                                           NaN   \n",
       "C3L-00674                                           NaN   \n",
       "C3L-00677                                           NaN   \n",
       "C3L-01040                                           NaN   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S27_GNDPLTSS*PGR  \\\n",
       "Patient_ID                                            \n",
       "C3L-00104                                  0.526739   \n",
       "C3L-00365                                       NaN   \n",
       "C3L-00674                                 -0.412701   \n",
       "C3L-00677                                 -0.194149   \n",
       "C3L-01040                                 -0.524123   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S27_RGNDPLTSS*PGR  \\\n",
       "Patient_ID                                             \n",
       "C3L-00104                                        NaN   \n",
       "C3L-00365                                   0.940525   \n",
       "C3L-00674                                  -0.455602   \n",
       "C3L-00677                                  -0.680688   \n",
       "C3L-01040                                        NaN   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S27S31_GNDPLTSS*PGRS*SR  \\\n",
       "Patient_ID                                                   \n",
       "C3L-00104                                              NaN   \n",
       "C3L-00365                                              NaN   \n",
       "C3L-00674                                              NaN   \n",
       "C3L-00677                                              NaN   \n",
       "C3L-01040                                              NaN   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S27S31_RGNDPLTSS*PGRS*SR  \\\n",
       "Patient_ID                                                    \n",
       "C3L-00104                                               NaN   \n",
       "C3L-00365                                               NaN   \n",
       "C3L-00674                                               NaN   \n",
       "C3L-00677                                               NaN   \n",
       "C3L-01040                                               NaN   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGR  \\\n",
       "Patient_ID                                                \n",
       "C3L-00104                                           NaN   \n",
       "C3L-00365                                       0.48957   \n",
       "C3L-00674                                           NaN   \n",
       "C3L-00677                                           NaN   \n",
       "C3L-01040                                           NaN   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER  \\\n",
       "Patient_ID                                                                                   \n",
       "C3L-00104                                                 NaN                                \n",
       "C3L-00365                                                 NaN                                \n",
       "C3L-00674                                                 NaN                                \n",
       "C3L-00677                                                 NaN                                \n",
       "C3L-01040                                                 NaN                                \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S41_RTDALTSS*PGR  \\\n",
       "Patient_ID                                            \n",
       "C3L-00104                                       NaN   \n",
       "C3L-00365                                  1.212528   \n",
       "C3L-00674                                 -0.108264   \n",
       "C3L-00677                                       NaN   \n",
       "C3L-01040                                       NaN   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S41_TDALTSS*PGR  \\\n",
       "Patient_ID                                           \n",
       "C3L-00104                                 0.846266   \n",
       "C3L-00365                                      NaN   \n",
       "C3L-00674                                -0.367349   \n",
       "C3L-00677                                      NaN   \n",
       "C3L-01040                                -1.339159   \n",
       "\n",
       "Name        MCM2_phosphoproteomics_S41_TDALTSS*PGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER  \\\n",
       "Patient_ID                                                                              \n",
       "C3L-00104                                                 NaN                           \n",
       "C3L-00365                                                 NaN                           \n",
       "C3L-00674                                                 NaN                           \n",
       "C3L-00677                                                 NaN                           \n",
       "C3L-01040                                                 NaN                           \n",
       "\n",
       "Name        MCM2_phosphoproteomics_T39_TDALT*SSPGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER  \\\n",
       "Patient_ID                                                                              \n",
       "C3L-00104                                                 NaN                           \n",
       "C3L-00365                                                 NaN                           \n",
       "C3L-00674                                                 NaN                           \n",
       "C3L-00677                                                 NaN                           \n",
       "C3L-01040                                                 NaN                           \n",
       "\n",
       "Name        MCM2_phosphoproteomics_T59_RTDALTSSPGRDLPPFEDESEGLLGT*EGPLEEEEDGEELIGDGMER  \\\n",
       "Patient_ID                                                                               \n",
       "C3L-00104                                            0.743703                            \n",
       "C3L-00365                                            0.843078                            \n",
       "C3L-00674                                            1.049988                            \n",
       "C3L-00677                                           -0.802540                            \n",
       "C3L-01040                                           -1.556890                            \n",
       "\n",
       "Name        Mutation  \n",
       "Patient_ID            \n",
       "C3L-00104   Deletion  \n",
       "C3L-00365   Deletion  \n",
       "C3L-00674   Deletion  \n",
       "C3L-00677   Deletion  \n",
       "C3L-01040   Deletion  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "t_gene = 'MCM2'\n",
    "g_del_wt = format_df(g, t_gene)\n",
    "g_del_wt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QR',\n",
       " 'MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QREAAER',\n",
       " 'MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPAR',\n",
       " 'MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPARK',\n",
       " 'MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPAR',\n",
       " 'MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPARK',\n",
       " 'MCM2_phosphoproteomics_S26S27_GNDPLTS*S*PGR',\n",
       " 'MCM2_phosphoproteomics_S27_GNDPLTSS*PGR',\n",
       " 'MCM2_phosphoproteomics_S27_RGNDPLTSS*PGR',\n",
       " 'MCM2_phosphoproteomics_S27S31_GNDPLTSS*PGRS*SR',\n",
       " 'MCM2_phosphoproteomics_S27S31_RGNDPLTSS*PGRS*SR',\n",
       " 'MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGR',\n",
       " 'MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER',\n",
       " 'MCM2_phosphoproteomics_S41_RTDALTSS*PGR',\n",
       " 'MCM2_phosphoproteomics_S41_TDALTSS*PGR',\n",
       " 'MCM2_phosphoproteomics_S41_TDALTSS*PGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER',\n",
       " 'MCM2_phosphoproteomics_T39_TDALT*SSPGRDLPPFEDESEGLLGTEGPLEEEEDGEELIGDGMER',\n",
       " 'MCM2_phosphoproteomics_T59_RTDALTSSPGRDLPPFEDESEGLLGT*EGPLEEEEDGEELIGDGMER']"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(g_del_wt.columns[:-1])\n",
    "cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPARK</td>\n",
       "      <td>0.000192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPAR</td>\n",
       "      <td>0.001449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPAR</td>\n",
       "      <td>0.006748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCM2_phosphoproteomics_S27_GNDPLTSS*PGR</td>\n",
       "      <td>0.007865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_RTDALTSS*PGR</td>\n",
       "      <td>0.034238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MCM2_phosphoproteomics_T59_RTDALTSSPGRDLPPFEDE...</td>\n",
       "      <td>0.043905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MCM2_phosphoproteomics_S26S27_GNDPLTS*S*PGR</td>\n",
       "      <td>0.046890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MCM2_phosphoproteomics_S27S31_RGNDPLTSS*PGRS*SR</td>\n",
       "      <td>0.062649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MCM2_phosphoproteomics_T39_TDALT*SSPGRDLPPFEDE...</td>\n",
       "      <td>0.207886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGRDLP...</td>\n",
       "      <td>0.255586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGR</td>\n",
       "      <td>0.302922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_TDALTSS*PGR</td>\n",
       "      <td>0.388402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...</td>\n",
       "      <td>0.465424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MCM2_phosphoproteomics_S27_RGNDPLTSS*PGR</td>\n",
       "      <td>0.492027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_TDALTSS*PGRDLPPFEDE...</td>\n",
       "      <td>0.663548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...</td>\n",
       "      <td>0.918292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPARK</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MCM2_phosphoproteomics_S27S31_GNDPLTSS*PGRS*SR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           Comparison   P_Value\n",
       "0      MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPARK  0.000192\n",
       "1      MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPAR  0.001449\n",
       "2       MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPAR  0.006748\n",
       "3             MCM2_phosphoproteomics_S27_GNDPLTSS*PGR  0.007865\n",
       "4             MCM2_phosphoproteomics_S41_RTDALTSS*PGR  0.034238\n",
       "5   MCM2_phosphoproteomics_T59_RTDALTSSPGRDLPPFEDE...  0.043905\n",
       "6         MCM2_phosphoproteomics_S26S27_GNDPLTS*S*PGR  0.046890\n",
       "7     MCM2_phosphoproteomics_S27S31_RGNDPLTSS*PGRS*SR  0.062649\n",
       "8   MCM2_phosphoproteomics_T39_TDALT*SSPGRDLPPFEDE...  0.207886\n",
       "9   MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGRDLP...  0.255586\n",
       "10        MCM2_phosphoproteomics_S40S41_RTDALTS*S*PGR  0.302922\n",
       "11             MCM2_phosphoproteomics_S41_TDALTSS*PGR  0.388402\n",
       "12  MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...  0.465424\n",
       "13           MCM2_phosphoproteomics_S27_RGNDPLTSS*PGR  0.492027\n",
       "14  MCM2_phosphoproteomics_S41_TDALTSS*PGRDLPPFEDE...  0.663548\n",
       "15  MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...  0.918292\n",
       "16    MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPARK       NaN\n",
       "17     MCM2_phosphoproteomics_S27S31_GNDPLTSS*PGRS*SR       NaN"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_pval = u.wrap_ttest(g_del_wt, 'Mutation', cols, return_all = True)\n",
    "gp = g_pval['P_Value'][0]\n",
    "g_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Luad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-00001.N, C3L-00009.N, C3L-00080.N, C3L-00083.N, C3L-00093.N, C3L-00094.N, C3L-00095.N, C3L-00140.N, C3L-00144.N, C3L-00263.N, C3L-00279.N, C3L-00368.N, C3L-00412.N, C3L-00422.N, C3L-00510.N, C3L-00604.N, C3L-00893.N, C3L-00913.N, C3L-00973.N, C3L-01330.N, C3L-01632.N, C3L-01682.N, C3L-01683.N, C3L-01889.N, C3L-01890, C3L-01890.N, C3L-01924.N, C3L-02219.N, C3L-02345.N, C3L-02348.N, C3L-02350.N, C3L-02365.N, C3L-02508.N, C3L-02549.N, C3N-00167.N, C3N-00169.N, C3N-00175.N, C3N-00180.N, C3N-00199.N, C3N-00203.N, C3N-00217.N, C3N-00223.N, C3N-00293.N, C3N-00433.N, C3N-00546.N, C3N-00547.N, C3N-00549.N, C3N-00550.N, C3N-00551.N, C3N-00552.N, C3N-00556.N, C3N-00559.N, C3N-00560.N, C3N-00572.N, C3N-00574.N, C3N-00578.N, C3N-00579.N, C3N-00580.N, C3N-00704.N, C3N-00737.N, C3N-00738.N, C3N-00959.N, C3N-01016.N, C3N-01021.N, C3N-01023.N, C3N-01024.N, C3N-01030.N, C3N-01071.N, C3N-01072.N, C3N-01405.N, C3N-01410.N, C3N-01413.N, C3N-01414.N, C3N-01415.N, C3N-01416.N, C3N-01488.N, C3N-01489.N, C3N-01799.N, C3N-01823.N, C3N-02000.N, C3N-02002.N, C3N-02003.N, C3N-02067.N, C3N-02087.N, C3N-02089.N, C3N-02145.N, C3N-02149.N, C3N-02155.N, C3N-02158.N, C3N-02379.N, C3N-02380.N, C3N-02421.N, C3N-02423.N, C3N-02424.N, C3N-02433.N, C3N-02529.N, C3N-02572.N, C3N-02582.N, C3N-02586.N, C3N-02587.N, C3N-02588.N, C3N-02729.N (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 33)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:48: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MCM2_phosphoproteomics_RTDALtSSPGRDLPPFEDESEGL...</td>\n",
       "      <td>0.232525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MCM2_phosphoproteomics_TDALTssPGR_NP_004517.2</td>\n",
       "      <td>0.340057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MCM2_phosphoproteomics_GLLYDsDEEDEERPAR_NP_004...</td>\n",
       "      <td>0.348227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCM2_phosphoproteomics_RGNDPLTSsPGR_NP_004517.2</td>\n",
       "      <td>0.364915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MCM2_phosphoproteomics_IQEsPGK_NP_004517.2</td>\n",
       "      <td>0.677160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MCM2_phosphoproteomics_AIPELDAYEAEGLALDDEDVEEL...</td>\n",
       "      <td>0.728188</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Comparison   P_Value\n",
       "0  MCM2_phosphoproteomics_RTDALtSSPGRDLPPFEDESEGL...  0.232525\n",
       "1      MCM2_phosphoproteomics_TDALTssPGR_NP_004517.2  0.340057\n",
       "2  MCM2_phosphoproteomics_GLLYDsDEEDEERPAR_NP_004...  0.348227\n",
       "3    MCM2_phosphoproteomics_RGNDPLTSsPGR_NP_004517.2  0.364915\n",
       "4         MCM2_phosphoproteomics_IQEsPGK_NP_004517.2  0.677160\n",
       "5  MCM2_phosphoproteomics_AIPELDAYEAEGLALDDEDVEEL...  0.728188"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "l_del_wt = format_df(l, t_gene)\n",
    "\n",
    "# t-test\n",
    "cols = list(l_del_wt.columns[:-1])\n",
    "l_pval = u.wrap_ttest(l_del_wt, 'Mutation', cols, return_all = True)\n",
    "lp = l_pval['P_Value'][0]\n",
    "l_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lscc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MCM2_phosphoproteomics_T25_RRGNDPLtSSPGR</td>\n",
       "      <td>0.000268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MCM2_phosphoproteomics_S41M77_RTDALTSsPGRDLPPF...</td>\n",
       "      <td>0.000956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_GLLYDsDEEDEERPAR</td>\n",
       "      <td>0.001413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCM2_phosphoproteomics_S381_IQEsPGK</td>\n",
       "      <td>0.030974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MCM2_phosphoproteomics_S40S41_TDALTssPGR</td>\n",
       "      <td>0.081259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MCM2_phosphoproteomics_S170_ATEDGEEDEEMIEsIENL...</td>\n",
       "      <td>0.210675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MCM2_phosphoproteomics_S27_GNDPLTSsPGR</td>\n",
       "      <td>0.292314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MCM2_phosphoproteomics_T158_QVERAtEDGEEDEEMIES...</td>\n",
       "      <td>0.344540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MCM2_phosphoproteomics_S26S27_RGNDPLTssPGR</td>\n",
       "      <td>0.462780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...</td>\n",
       "      <td>0.906169</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Comparison   P_Value\n",
       "0           MCM2_phosphoproteomics_T25_RRGNDPLtSSPGR  0.000268\n",
       "1  MCM2_phosphoproteomics_S41M77_RTDALTSsPGRDLPPF...  0.000956\n",
       "2       MCM2_phosphoproteomics_S139_GLLYDsDEEDEERPAR  0.001413\n",
       "3                MCM2_phosphoproteomics_S381_IQEsPGK  0.030974\n",
       "4           MCM2_phosphoproteomics_S40S41_TDALTssPGR  0.081259\n",
       "5  MCM2_phosphoproteomics_S170_ATEDGEEDEEMIEsIENL...  0.210675\n",
       "6             MCM2_phosphoproteomics_S27_GNDPLTSsPGR  0.292314\n",
       "7  MCM2_phosphoproteomics_T158_QVERAtEDGEEDEEMIES...  0.344540\n",
       "8         MCM2_phosphoproteomics_S26S27_RGNDPLTssPGR  0.462780\n",
       "9  MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...  0.906169"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ls_del_wt = format_df(ls, t_gene)\n",
    "\n",
    "# t-test\n",
    "cols = list(ls_del_wt.columns[:-1])\n",
    "ls_pval = u.wrap_ttest(ls_del_wt, 'Mutation', cols, return_all = True)\n",
    "lsp = ls_pval['P_Value'][0]\n",
    "ls_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 01OV029, 02OV015 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_G.LLYDS*DEEDEERPAR.K</td>\n",
       "      <td>0.093656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_R.GLLYDS*DEEDEERPAR.K</td>\n",
       "      <td>0.106059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_R.GLLYDS*DEEDEER.P</td>\n",
       "      <td>0.156596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCM2_phosphoproteomics_S27_R.GNDPLTSS*PGR.S</td>\n",
       "      <td>0.373581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_R.RTDALTSS*PGR.D</td>\n",
       "      <td>0.473149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_L.YDS*DEEDEERPAR.K</td>\n",
       "      <td>0.581648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_R.GLLYDS*DEEDEERPA.R</td>\n",
       "      <td>0.616750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_R.TDALTSS*PGR.D</td>\n",
       "      <td>0.637227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MCM2_phosphoproteomics_S27_R.RGNDPLTSS*PGR.S</td>\n",
       "      <td>0.663227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MCM2_phosphoproteomics_S13_F.TMASS*PAQR.R</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Comparison   P_Value\n",
       "0   MCM2_phosphoproteomics_S139_G.LLYDS*DEEDEERPAR.K  0.093656\n",
       "1  MCM2_phosphoproteomics_S139_R.GLLYDS*DEEDEERPAR.K  0.106059\n",
       "2     MCM2_phosphoproteomics_S139_R.GLLYDS*DEEDEER.P  0.156596\n",
       "3        MCM2_phosphoproteomics_S27_R.GNDPLTSS*PGR.S  0.373581\n",
       "4        MCM2_phosphoproteomics_S41_R.RTDALTSS*PGR.D  0.473149\n",
       "5     MCM2_phosphoproteomics_S139_L.YDS*DEEDEERPAR.K  0.581648\n",
       "6   MCM2_phosphoproteomics_S139_R.GLLYDS*DEEDEERPA.R  0.616750\n",
       "7         MCM2_phosphoproteomics_S41_R.TDALTSS*PGR.D  0.637227\n",
       "8       MCM2_phosphoproteomics_S27_R.RGNDPLTSS*PGR.S  0.663227\n",
       "9          MCM2_phosphoproteomics_S13_F.TMASS*PAQR.R       NaN"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "o_del_wt = format_df(o, t_gene)\n",
    "\n",
    "# t-test\n",
    "cols = list(o_del_wt.columns[:-1])\n",
    "o_pval = u.wrap_ttest(o_del_wt, 'Mutation', cols, return_all = True)\n",
    "op = o_pval['P_Value'][0]\n",
    "o_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Brca"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 113 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MCM2_phosphoproteomics_S381_IQEsPGK</td>\n",
       "      <td>0.227976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...</td>\n",
       "      <td>0.376895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MCM2_phosphoproteomics_S26S27_RGNDPLTssPGR</td>\n",
       "      <td>0.382376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCM2_phosphoproteomics_S40S41_RTDALTssPGR</td>\n",
       "      <td>0.581410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_RTDALTSsPGRDLPPFEDE...</td>\n",
       "      <td>0.728327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MCM2_phosphoproteomics_T158_AtEDGEEDEEMIESIENL...</td>\n",
       "      <td>0.761973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_GLLYDsDEEDEERPAR</td>\n",
       "      <td>0.773645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MCM2_phosphoproteomics_S170_ATEDGEEDEEMIEsIENL...</td>\n",
       "      <td>0.808375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MCM2_phosphoproteomics_S27_RGNDPLTSsPGR</td>\n",
       "      <td>0.830809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_TDALTSsPGRDLPPFEDES...</td>\n",
       "      <td>0.849992</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          Comparison   P_Value\n",
       "0                MCM2_phosphoproteomics_S381_IQEsPGK  0.227976\n",
       "1  MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...  0.376895\n",
       "2         MCM2_phosphoproteomics_S26S27_RGNDPLTssPGR  0.382376\n",
       "3          MCM2_phosphoproteomics_S40S41_RTDALTssPGR  0.581410\n",
       "4  MCM2_phosphoproteomics_S41_RTDALTSsPGRDLPPFEDE...  0.728327\n",
       "5  MCM2_phosphoproteomics_T158_AtEDGEEDEEMIESIENL...  0.761973\n",
       "6       MCM2_phosphoproteomics_S139_GLLYDsDEEDEERPAR  0.773645\n",
       "7  MCM2_phosphoproteomics_S170_ATEDGEEDEEMIEsIENL...  0.808375\n",
       "8            MCM2_phosphoproteomics_S27_RGNDPLTSsPGR  0.830809\n",
       "9  MCM2_phosphoproteomics_S41_TDALTSsPGRDLPPFEDES...  0.849992"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b_del_wt = format_df(b, t_gene)\n",
    "\n",
    "# t-test\n",
    "cols = list(b_del_wt.columns[:-1])\n",
    "b_pval = u.wrap_ttest(b_del_wt, 'Mutation', cols, return_all = True)\n",
    "bp = b_pval['P_Value'][0]\n",
    "b_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: 05CO045 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 201 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n",
      "cptac warning: phosphoproteomics data was not found for the following samples, so phosphoproteomics data columns were filled with NaN for these samples: 01CO001, 05CO005, 05CO007, 05CO014, 05CO055, 11CO059, 11CO060, 14CO002, 16CO012, 24CO005 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MCM2_phosphoproteomics_S53</td>\n",
       "      <td>0.345397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MCM2_phosphoproteomics_S27</td>\n",
       "      <td>0.656717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MCM2_phosphoproteomics_S139</td>\n",
       "      <td>0.879570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCM2_phosphoproteomics_S41</td>\n",
       "      <td>0.985340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MCM2_phosphoproteomics_S108</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MCM2_phosphoproteomics_T59</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Comparison   P_Value\n",
       "0   MCM2_phosphoproteomics_S53  0.345397\n",
       "1   MCM2_phosphoproteomics_S27  0.656717\n",
       "2  MCM2_phosphoproteomics_S139  0.879570\n",
       "3   MCM2_phosphoproteomics_S41  0.985340\n",
       "4  MCM2_phosphoproteomics_S108       NaN\n",
       "5   MCM2_phosphoproteomics_T59       NaN"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c_del_wt = format_df(col, t_gene, drop_level = 2)\n",
    "\n",
    "# t-test\n",
    "cols = list(c_del_wt.columns[:-1])\n",
    "c_pval = u.wrap_ttest(c_del_wt, 'Mutation', cols, return_all = True)\n",
    "cp = c_pval['P_Value'][0]\n",
    "c_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Hnscc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 109 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: CNV data was not found for the following samples, so CNV data columns were filled with NaN for these samples: C3L-02617, C3N-02727 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\cptac\\dataset.py, line 394)\n",
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 176 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 9)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_RTDALTSsPGR</td>\n",
       "      <td>0.031083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MCM2_phosphoproteomics_T39_RTDALtSSPGRDLPPFEDE...</td>\n",
       "      <td>0.033700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MCM2_phosphoproteomics_S381_IQEsPGK</td>\n",
       "      <td>0.049361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCM2_phosphoproteomics_S27_RRGNDPLTSsPGR</td>\n",
       "      <td>0.055931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MCM2_phosphoproteomics_S170_ATEDGEEDEEMIEsIENL...</td>\n",
       "      <td>0.119922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MCM2_phosphoproteomics_S139_RGLLYDsDEEDEERPARK</td>\n",
       "      <td>0.189839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...</td>\n",
       "      <td>0.203755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MCM2_phosphoproteomics_S27S31_RGNDPLTSsPGRsSR</td>\n",
       "      <td>0.216745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MCM2_phosphoproteomics_S700_EEEGLANGsAAEPAMPNT...</td>\n",
       "      <td>0.244018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MCM2_phosphoproteomics_S53_DLPPFEDEsEGLLGTEGPL...</td>\n",
       "      <td>0.281392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MCM2_phosphoproteomics_S31_RGNDPLTSSPGRsSR</td>\n",
       "      <td>0.368037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MCM2_phosphoproteomics_S41_RTDALTSsPGRDLPPFEDE...</td>\n",
       "      <td>0.482874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MCM2_phosphoproteomics_S40S41_RTDALTssPGR</td>\n",
       "      <td>0.522822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MCM2_phosphoproteomics_Y137_RGLLyDSDEEDEERPARK</td>\n",
       "      <td>0.528980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MCM2_phosphoproteomics_S26S27_RGNDPLTssPGR</td>\n",
       "      <td>0.554811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MCM2_phosphoproteomics_T59_DLPPFEDESEGLLGtEGPL...</td>\n",
       "      <td>0.663532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MCM2_phosphoproteomics_S40S41_RTDALTssPGRDLPPF...</td>\n",
       "      <td>0.891482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MCM2_phosphoproteomics_S209_THVDsHGHNVFK</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MCM2_phosphoproteomics_S41T59_RTDALTSsPGRDLPPF...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MCM2_phosphoproteomics_T106_AIPELDAYEAEGLALDDE...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MCM2_phosphoproteomics_T25_RGNDPLtSSPGR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MCM2_phosphoproteomics_T25S26_RGNDPLtsSPGR</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           Comparison   P_Value\n",
       "0              MCM2_phosphoproteomics_S41_RTDALTSsPGR  0.031083\n",
       "1   MCM2_phosphoproteomics_T39_RTDALtSSPGRDLPPFEDE...  0.033700\n",
       "2                 MCM2_phosphoproteomics_S381_IQEsPGK  0.049361\n",
       "3            MCM2_phosphoproteomics_S27_RRGNDPLTSsPGR  0.055931\n",
       "4   MCM2_phosphoproteomics_S170_ATEDGEEDEEMIEsIENL...  0.119922\n",
       "5      MCM2_phosphoproteomics_S139_RGLLYDsDEEDEERPARK  0.189839\n",
       "6   MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDE...  0.203755\n",
       "7       MCM2_phosphoproteomics_S27S31_RGNDPLTSsPGRsSR  0.216745\n",
       "8   MCM2_phosphoproteomics_S700_EEEGLANGsAAEPAMPNT...  0.244018\n",
       "9   MCM2_phosphoproteomics_S53_DLPPFEDEsEGLLGTEGPL...  0.281392\n",
       "10         MCM2_phosphoproteomics_S31_RGNDPLTSSPGRsSR  0.368037\n",
       "11  MCM2_phosphoproteomics_S41_RTDALTSsPGRDLPPFEDE...  0.482874\n",
       "12          MCM2_phosphoproteomics_S40S41_RTDALTssPGR  0.522822\n",
       "13     MCM2_phosphoproteomics_Y137_RGLLyDSDEEDEERPARK  0.528980\n",
       "14         MCM2_phosphoproteomics_S26S27_RGNDPLTssPGR  0.554811\n",
       "15  MCM2_phosphoproteomics_T59_DLPPFEDESEGLLGtEGPL...  0.663532\n",
       "16  MCM2_phosphoproteomics_S40S41_RTDALTssPGRDLPPF...  0.891482\n",
       "17           MCM2_phosphoproteomics_S209_THVDsHGHNVFK       NaN\n",
       "18  MCM2_phosphoproteomics_S41T59_RTDALTSsPGRDLPPF...       NaN\n",
       "19  MCM2_phosphoproteomics_T106_AIPELDAYEAEGLALDDE...       NaN\n",
       "20            MCM2_phosphoproteomics_T25_RGNDPLtSSPGR       NaN\n",
       "21         MCM2_phosphoproteomics_T25S26_RGNDPLtsSPGR       NaN"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h_del_wt = format_df(h, t_gene, drop_level=3)\n",
    "\n",
    "# t-test\n",
    "cols = list(h_del_wt.columns[:-1])\n",
    "h_pval = u.wrap_ttest(h_del_wt, 'Mutation', cols, return_all = True)\n",
    "hp = h_pval['P_Value'][0]\n",
    "h_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Endometrial data set does not have enough cnv deletions to perform a t-test, however the data set does have enough truncation type mutations (nonsense and frame shifts). Different code is needed to create the data frame for Endometrial."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nonsense_Mutation    25\n",
      "Frame_Shift_Del      22\n",
      "Wildtype_Tumor       20\n",
      "Frame_Shift_Ins       6\n",
      "Name: PTEN_Mutation_, dtype: int64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 69 samples for the PTEN gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 4)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  del sys.path[0]\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3367: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\numpy\\core\\_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less\n",
      "  return (self.a < x) & (x < self.b)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= self.a)\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:147: RuntimeWarning: invalid value encountered in less_equal\n",
      "  reject = pvals <= alphacBonf\n",
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\statsmodels\\stats\\multitest.py:251: RuntimeWarning: invalid value encountered in greater\n",
      "  pvals_corrected[pvals_corrected>1] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MCM2_phosphoproteomics_S139</td>\n",
       "      <td>0.012954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MCM2_phosphoproteomics_S41</td>\n",
       "      <td>0.028419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MCM2_phosphoproteomics_S27</td>\n",
       "      <td>0.033836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MCM2_phosphoproteomics_T59</td>\n",
       "      <td>0.076825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MCM2_phosphoproteomics_S40</td>\n",
       "      <td>0.149126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MCM2_phosphoproteomics_S170</td>\n",
       "      <td>0.187925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MCM2_phosphoproteomics_T39</td>\n",
       "      <td>0.448562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MCM2_phosphoproteomics_S108</td>\n",
       "      <td>0.476061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MCM2_phosphoproteomics_Y137</td>\n",
       "      <td>0.629660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MCM2_phosphoproteomics_S26</td>\n",
       "      <td>0.667774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MCM2_phosphoproteomics_S31</td>\n",
       "      <td>0.846722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MCM2_phosphoproteomics_S53</td>\n",
       "      <td>0.987694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MCM2_phosphoproteomics_Y90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Comparison   P_Value\n",
       "0   MCM2_phosphoproteomics_S139  0.012954\n",
       "1    MCM2_phosphoproteomics_S41  0.028419\n",
       "2    MCM2_phosphoproteomics_S27  0.033836\n",
       "3    MCM2_phosphoproteomics_T59  0.076825\n",
       "4    MCM2_phosphoproteomics_S40  0.149126\n",
       "5   MCM2_phosphoproteomics_S170  0.187925\n",
       "6    MCM2_phosphoproteomics_T39  0.448562\n",
       "7   MCM2_phosphoproteomics_S108  0.476061\n",
       "8   MCM2_phosphoproteomics_Y137  0.629660\n",
       "9    MCM2_phosphoproteomics_S26  0.667774\n",
       "10   MCM2_phosphoproteomics_S31  0.846722\n",
       "11   MCM2_phosphoproteomics_S53  0.987694\n",
       "12   MCM2_phosphoproteomics_Y90       NaN"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1 - Create proteomics and truncations dataframe\n",
    "prot_and_mutations = en.join_omics_to_mutations(\n",
    "    mutations_genes = [gene], omics_df_name = 'phosphoproteomics', omics_genes = t_gene, \n",
    "    mutations_filter = ['Nonsense_Mutation','Frame_Shift_Del','Frame_Shift_Ins']) \n",
    "prot_and_mutations = en.reduce_multiindex(prot_and_mutations, flatten = True)\n",
    "keep = ['Nonsense_Mutation','Frame_Shift_Del','Frame_Shift_Ins','Wildtype_Tumor']\n",
    "in_keep = prot_and_mutations['PTEN_Mutation_'].isin(keep) \n",
    "trunc_mutations = prot_and_mutations[in_keep]\n",
    "print(trunc_mutations['PTEN_Mutation_'].value_counts()) # may need to change extra char\n",
    "\n",
    "# Step 2 - Create binary column \n",
    "trunc_mutations['Mutation'] = np.where(\n",
    "            trunc_mutations[gene+'_Mutation_Status_'] == 'Wildtype_Tumor', 'Wildtype_Tumor', 'Truncation')\n",
    "\n",
    "# Step 3 - Format the dataframe correctly for the t-test(just omics and binary columns for tumors)\n",
    "tumors = trunc_mutations[trunc_mutations.Sample_Status_ == 'Tumor'] #drop Normal samples\n",
    "columns_to_drop = [gene+\"_Mutation_\", gene+\"_Location_\", gene+\"_Mutation_Status_\", \"Sample_Status_\"]\n",
    "e_trunc_wt = tumors.drop(columns_to_drop, axis = 1)\n",
    "e_trunc_wt = e_trunc_wt.dropna(axis=1,how='all')\n",
    "e_trunc_wt['Mutation'].value_counts()\n",
    "\n",
    "# t-test\n",
    "cols = list(e_trunc_wt.columns[:-1])\n",
    "e_pval = u.wrap_ttest(e_trunc_wt, 'Mutation', cols, return_all=True)\n",
    "ep = e_pval['P_Value'][0]\n",
    "e_pval"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 2: Create a long dataframe for the boxplot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The boxplot will take three columns: Proteomics, Mutation, and Cancer. We need to append all the individual cancer dfs into one long pancancer df. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First create the Cancer column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "g_del_wt = g_del_wt.assign(cancer = 'Gbm')\n",
    "l_del_wt = l_del_wt.assign(cancer = 'Luad')\n",
    "ls_del_wt = ls_del_wt.assign(cancer = 'Lscc')\n",
    "b_del_wt = b_del_wt.assign(cancer = 'Brca')\n",
    "o_del_wt = o_del_wt.assign(cancer = 'Ovarian')\n",
    "c_del_wt = c_del_wt.assign(cancer = 'Colon')\n",
    "h_del_wt = h_del_wt.assign(cancer = 'Hnscc') # higher scale\n",
    "e_trunc_wt = e_trunc_wt.assign(cancer = 'Endometrial')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next append the dfs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QR</th>\n",
       "      <th>MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QREAAER</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPAR</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPARK</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPAR</th>\n",
       "      <th>MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPARK</th>\n",
       "      <th>MCM2_phosphoproteomics_S26S27_GNDPLTS*S*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S27_GNDPLTSS*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S27_RGNDPLTSS*PGR</th>\n",
       "      <th>MCM2_phosphoproteomics_S27S31_GNDPLTSS*PGRS*SR</th>\n",
       "      <th>...</th>\n",
       "      <th>MCM2_phosphoproteomics_S41</th>\n",
       "      <th>MCM2_phosphoproteomics_S53</th>\n",
       "      <th>MCM2_phosphoproteomics_T59</th>\n",
       "      <th>MCM2_phosphoproteomics_S170</th>\n",
       "      <th>MCM2_phosphoproteomics_S26</th>\n",
       "      <th>MCM2_phosphoproteomics_S31</th>\n",
       "      <th>MCM2_phosphoproteomics_S40</th>\n",
       "      <th>MCM2_phosphoproteomics_T39</th>\n",
       "      <th>MCM2_phosphoproteomics_Y137</th>\n",
       "      <th>MCM2_phosphoproteomics_Y90</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>0.250213</td>\n",
       "      <td>0.469123</td>\n",
       "      <td>0.553357</td>\n",
       "      <td>1.007214</td>\n",
       "      <td>0.116577</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.526739</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>0.680272</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.592142</td>\n",
       "      <td>1.128073</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.940525</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>-1.100628</td>\n",
       "      <td>1.074279</td>\n",
       "      <td>-0.417157</td>\n",
       "      <td>-0.105052</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.412701</td>\n",
       "      <td>-0.455602</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.539857</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.118192</td>\n",
       "      <td>1.192966</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.194149</td>\n",
       "      <td>-0.680688</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>-1.149843</td>\n",
       "      <td>-1.269670</td>\n",
       "      <td>-1.141344</td>\n",
       "      <td>-0.707192</td>\n",
       "      <td>-0.493904</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.524123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01520</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.9590</td>\n",
       "      <td>-1.3700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01521</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4945</td>\n",
       "      <td>0.1825</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01537</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4560</td>\n",
       "      <td>0.265</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.199</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01802</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01825</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3560</td>\n",
       "      <td>0.4675</td>\n",
       "      <td>0.0207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.295</td>\n",
       "      <td>-0.0287</td>\n",
       "      <td>0.7905</td>\n",
       "      <td>-0.0289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>647 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QR  \\\n",
       "Patient_ID                                                              \n",
       "C3L-00104                                            0.250213           \n",
       "C3L-00365                                            0.680272           \n",
       "C3L-00674                                           -1.100628           \n",
       "C3L-00677                                                 NaN           \n",
       "C3L-01040                                           -1.149843           \n",
       "...                                                       ...           \n",
       "C3N-01520                                                 NaN           \n",
       "C3N-01521                                                 NaN           \n",
       "C3N-01537                                                 NaN           \n",
       "C3N-01802                                                 NaN           \n",
       "C3N-01825                                                 NaN           \n",
       "\n",
       "            MCM2_phosphoproteomics_S108_AIPELDAYEAEGLALDDEDVEELTAS*QREAAER  \\\n",
       "Patient_ID                                                                   \n",
       "C3L-00104                                            0.469123                \n",
       "C3L-00365                                                 NaN                \n",
       "C3L-00674                                            1.074279                \n",
       "C3L-00677                                            0.539857                \n",
       "C3L-01040                                           -1.269670                \n",
       "...                                                       ...                \n",
       "C3N-01520                                                 NaN                \n",
       "C3N-01521                                                 NaN                \n",
       "C3N-01537                                                 NaN                \n",
       "C3N-01802                                                 NaN                \n",
       "C3N-01825                                                 NaN                \n",
       "\n",
       "            MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPAR  \\\n",
       "Patient_ID                                                  \n",
       "C3L-00104                                        0.553357   \n",
       "C3L-00365                                        0.592142   \n",
       "C3L-00674                                       -0.417157   \n",
       "C3L-00677                                             NaN   \n",
       "C3L-01040                                       -1.141344   \n",
       "...                                                   ...   \n",
       "C3N-01520                                             NaN   \n",
       "C3N-01521                                             NaN   \n",
       "C3N-01537                                             NaN   \n",
       "C3N-01802                                             NaN   \n",
       "C3N-01825                                             NaN   \n",
       "\n",
       "            MCM2_phosphoproteomics_S139_GLLYDS*DEEDEERPARK  \\\n",
       "Patient_ID                                                   \n",
       "C3L-00104                                         1.007214   \n",
       "C3L-00365                                         1.128073   \n",
       "C3L-00674                                        -0.105052   \n",
       "C3L-00677                                         0.118192   \n",
       "C3L-01040                                        -0.707192   \n",
       "...                                                    ...   \n",
       "C3N-01520                                              NaN   \n",
       "C3N-01521                                              NaN   \n",
       "C3N-01537                                              NaN   \n",
       "C3N-01802                                              NaN   \n",
       "C3N-01825                                              NaN   \n",
       "\n",
       "            MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPAR  \\\n",
       "Patient_ID                                                   \n",
       "C3L-00104                                         0.116577   \n",
       "C3L-00365                                              NaN   \n",
       "C3L-00674                                              NaN   \n",
       "C3L-00677                                         1.192966   \n",
       "C3L-01040                                        -0.493904   \n",
       "...                                                    ...   \n",
       "C3N-01520                                              NaN   \n",
       "C3N-01521                                              NaN   \n",
       "C3N-01537                                              NaN   \n",
       "C3N-01802                                              NaN   \n",
       "C3N-01825                                              NaN   \n",
       "\n",
       "            MCM2_phosphoproteomics_S139_RGLLYDS*DEEDEERPARK  \\\n",
       "Patient_ID                                                    \n",
       "C3L-00104                                               NaN   \n",
       "C3L-00365                                               NaN   \n",
       "C3L-00674                                               NaN   \n",
       "C3L-00677                                               NaN   \n",
       "C3L-01040                                               NaN   \n",
       "...                                                     ...   \n",
       "C3N-01520                                               NaN   \n",
       "C3N-01521                                               NaN   \n",
       "C3N-01537                                               NaN   \n",
       "C3N-01802                                               NaN   \n",
       "C3N-01825                                               NaN   \n",
       "\n",
       "            MCM2_phosphoproteomics_S26S27_GNDPLTS*S*PGR  \\\n",
       "Patient_ID                                                \n",
       "C3L-00104                                           NaN   \n",
       "C3L-00365                                           NaN   \n",
       "C3L-00674                                           NaN   \n",
       "C3L-00677                                           NaN   \n",
       "C3L-01040                                           NaN   \n",
       "...                                                 ...   \n",
       "C3N-01520                                           NaN   \n",
       "C3N-01521                                           NaN   \n",
       "C3N-01537                                           NaN   \n",
       "C3N-01802                                           NaN   \n",
       "C3N-01825                                           NaN   \n",
       "\n",
       "            MCM2_phosphoproteomics_S27_GNDPLTSS*PGR  \\\n",
       "Patient_ID                                            \n",
       "C3L-00104                                  0.526739   \n",
       "C3L-00365                                       NaN   \n",
       "C3L-00674                                 -0.412701   \n",
       "C3L-00677                                 -0.194149   \n",
       "C3L-01040                                 -0.524123   \n",
       "...                                             ...   \n",
       "C3N-01520                                       NaN   \n",
       "C3N-01521                                       NaN   \n",
       "C3N-01537                                       NaN   \n",
       "C3N-01802                                       NaN   \n",
       "C3N-01825                                       NaN   \n",
       "\n",
       "            MCM2_phosphoproteomics_S27_RGNDPLTSS*PGR  \\\n",
       "Patient_ID                                             \n",
       "C3L-00104                                        NaN   \n",
       "C3L-00365                                   0.940525   \n",
       "C3L-00674                                  -0.455602   \n",
       "C3L-00677                                  -0.680688   \n",
       "C3L-01040                                        NaN   \n",
       "...                                              ...   \n",
       "C3N-01520                                        NaN   \n",
       "C3N-01521                                        NaN   \n",
       "C3N-01537                                        NaN   \n",
       "C3N-01802                                        NaN   \n",
       "C3N-01825                                        NaN   \n",
       "\n",
       "            MCM2_phosphoproteomics_S27S31_GNDPLTSS*PGRS*SR  ...  \\\n",
       "Patient_ID                                                  ...   \n",
       "C3L-00104                                              NaN  ...   \n",
       "C3L-00365                                              NaN  ...   \n",
       "C3L-00674                                              NaN  ...   \n",
       "C3L-00677                                              NaN  ...   \n",
       "C3L-01040                                              NaN  ...   \n",
       "...                                                    ...  ...   \n",
       "C3N-01520                                              NaN  ...   \n",
       "C3N-01521                                              NaN  ...   \n",
       "C3N-01537                                              NaN  ...   \n",
       "C3N-01802                                              NaN  ...   \n",
       "C3N-01825                                              NaN  ...   \n",
       "\n",
       "            MCM2_phosphoproteomics_S41  MCM2_phosphoproteomics_S53  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00104                          NaN                         NaN   \n",
       "C3L-00365                          NaN                         NaN   \n",
       "C3L-00674                          NaN                         NaN   \n",
       "C3L-00677                          NaN                         NaN   \n",
       "C3L-01040                          NaN                         NaN   \n",
       "...                                ...                         ...   \n",
       "C3N-01520                      -0.9590                     -1.3700   \n",
       "C3N-01521                       0.4945                      0.1825   \n",
       "C3N-01537                          NaN                         NaN   \n",
       "C3N-01802                      -0.0144                         NaN   \n",
       "C3N-01825                       0.3560                      0.4675   \n",
       "\n",
       "            MCM2_phosphoproteomics_T59  MCM2_phosphoproteomics_S170  \\\n",
       "Patient_ID                                                            \n",
       "C3L-00104                          NaN                          NaN   \n",
       "C3L-00365                          NaN                          NaN   \n",
       "C3L-00674                          NaN                          NaN   \n",
       "C3L-00677                          NaN                          NaN   \n",
       "C3L-01040                          NaN                          NaN   \n",
       "...                                ...                          ...   \n",
       "C3N-01520                          NaN                          NaN   \n",
       "C3N-01521                          NaN                          NaN   \n",
       "C3N-01537                       0.4560                        0.265   \n",
       "C3N-01802                          NaN                          NaN   \n",
       "C3N-01825                       0.0207                          NaN   \n",
       "\n",
       "            MCM2_phosphoproteomics_S26  MCM2_phosphoproteomics_S31  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00104                          NaN                         NaN   \n",
       "C3L-00365                          NaN                         NaN   \n",
       "C3L-00674                          NaN                         NaN   \n",
       "C3L-00677                          NaN                         NaN   \n",
       "C3L-01040                          NaN                         NaN   \n",
       "...                                ...                         ...   \n",
       "C3N-01520                          NaN                         NaN   \n",
       "C3N-01521                          NaN                         NaN   \n",
       "C3N-01537                          NaN                         NaN   \n",
       "C3N-01802                          NaN                         NaN   \n",
       "C3N-01825                        0.295                     -0.0287   \n",
       "\n",
       "            MCM2_phosphoproteomics_S40  MCM2_phosphoproteomics_T39  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00104                          NaN                         NaN   \n",
       "C3L-00365                          NaN                         NaN   \n",
       "C3L-00674                          NaN                         NaN   \n",
       "C3L-00677                          NaN                         NaN   \n",
       "C3L-01040                          NaN                         NaN   \n",
       "...                                ...                         ...   \n",
       "C3N-01520                          NaN                         NaN   \n",
       "C3N-01521                          NaN                         NaN   \n",
       "C3N-01537                          NaN                         NaN   \n",
       "C3N-01802                          NaN                         NaN   \n",
       "C3N-01825                       0.7905                     -0.0289   \n",
       "\n",
       "           MCM2_phosphoproteomics_Y137 MCM2_phosphoproteomics_Y90  \n",
       "Patient_ID                                                         \n",
       "C3L-00104                          NaN                        NaN  \n",
       "C3L-00365                          NaN                        NaN  \n",
       "C3L-00674                          NaN                        NaN  \n",
       "C3L-00677                          NaN                        NaN  \n",
       "C3L-01040                          NaN                        NaN  \n",
       "...                                ...                        ...  \n",
       "C3N-01520                          NaN                        NaN  \n",
       "C3N-01521                          NaN                        NaN  \n",
       "C3N-01537                        0.199                        NaN  \n",
       "C3N-01802                          NaN                        NaN  \n",
       "C3N-01825                          NaN                        NaN  \n",
       "\n",
       "[647 rows x 64 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = g_del_wt.append(l_del_wt)\n",
    "df2 = df.append(ls_del_wt)\n",
    "df3 = df2.append(b_del_wt)\n",
    "df4 = df3.append(o_del_wt)\n",
    "df5 = df4.append(c_del_wt)\n",
    "df6 = df5.append(e_trunc_wt)\n",
    "#df7 = df6.append(h_del_wt) # Leave out Hnscc because of high proteomics numbers\n",
    "\n",
    "df6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 3: Create the Pancancer Boxplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Could not interpret input 'MCM2_proteomics'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-89-82e5dd795539>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m boxplot = sns.boxplot(x='cancer', y=t_gene+\"_proteomics\", data = df6, hue = 'Mutation',\n\u001b[1;32m----> 6\u001b[1;33m                       hue_order = [\"Wildtype_Tumor\", \"Deletion\",'Truncation'], showfliers = False)    \n\u001b[0m\u001b[0;32m      7\u001b[0m \u001b[0mboxplot\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mset_title\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'Pancancer trans effect of PTEN CNV Deletions on '\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mt_gene\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      8\u001b[0m boxplot = sns.stripplot(x='cancer', y=t_gene+\"_proteomics\", data = df6, jitter = True, \n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\seaborn\\categorical.py\u001b[0m in \u001b[0;36mboxplot\u001b[1;34m(x, y, hue, data, order, hue_order, orient, color, palette, saturation, width, dodge, fliersize, linewidth, whis, notch, ax, **kwargs)\u001b[0m\n\u001b[0;32m   2229\u001b[0m     plotter = _BoxPlotter(x, y, hue, data, order, hue_order,\n\u001b[0;32m   2230\u001b[0m                           \u001b[0morient\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcolor\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpalette\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msaturation\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2231\u001b[1;33m                           width, dodge, fliersize, linewidth)\n\u001b[0m\u001b[0;32m   2232\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2233\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0max\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\seaborn\\categorical.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, x, y, hue, data, order, hue_order, orient, color, palette, saturation, width, dodge, fliersize, linewidth)\u001b[0m\n\u001b[0;32m    444\u001b[0m                  width, dodge, fliersize, linewidth):\n\u001b[0;32m    445\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 446\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mestablish_variables\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mhue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0morient\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0morder\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mhue_order\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    447\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mestablish_colors\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcolor\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpalette\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msaturation\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    448\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\seaborn\\categorical.py\u001b[0m in \u001b[0;36mestablish_variables\u001b[1;34m(self, x, y, hue, data, orient, order, hue_order, units)\u001b[0m\n\u001b[0;32m    153\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mstring_types\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    154\u001b[0m                     \u001b[0merr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"Could not interpret input '{}'\"\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 155\u001b[1;33m                     \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0merr\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    156\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    157\u001b[0m             \u001b[1;31m# Figure out the plotting orientation\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: Could not interpret input 'MCM2_proteomics'"
     ]
    }
   ],
   "source": [
    "gene = 'PTEN'\n",
    "plt.rcParams['figure.figsize']=(15,10) #size of plot\n",
    "sns.set(font_scale = 1.2)\n",
    "\n",
    "boxplot = sns.boxplot(x='cancer', y=t_gene+\"_proteomics\", data = df6, hue = 'Mutation',\n",
    "                      hue_order = [\"Wildtype_Tumor\", \"Deletion\",'Truncation'], showfliers = False)    \n",
    "boxplot.set_title('Pancancer trans effect of PTEN CNV Deletions on '+t_gene)\n",
    "boxplot = sns.stripplot(x='cancer', y=t_gene+\"_proteomics\", data = df6, jitter = True, \n",
    "                           color = \".3\", hue = 'Mutation', hue_order = [\"Wildtype_Tumor\", \"Deletion\", 'Truncation'],dodge = True)\n",
    "boxplot.set(xlabel = \"\\n\"+gene+\" Wildtype/CNV Deletion\", ylabel = t_gene+' Phosphoproteomics')\n",
    "\n",
    "# format legend\n",
    "handles, labels = boxplot.get_legend_handles_labels()\n",
    "plt.legend(handles[0:3], labels[0:3])\n",
    "\n",
    "\n",
    "# Create significance symbols:\n",
    "# * P ≤ 0.05   ** P ≤ 0.01   *** P ≤ 0.001 \n",
    "pvals = [gp, lp, lsp, bp, op, cp, ep] # same order as appended in long df\n",
    "symbols = {}\n",
    "\n",
    "for p in pvals:\n",
    "    if p <= 0.001:\n",
    "        symbols[p] = '***'\n",
    "    elif p <= 0.01:\n",
    "        symbols[p] = '**'\n",
    "    elif p <= .05:\n",
    "        symbols[p] = '*'\n",
    "    else:\n",
    "        symbols[p] = 'ns'\n",
    "#print(symbols)\n",
    "\n",
    "format_pval_annotation(symbols[gp], -.3, 0, 4) # Gbm\n",
    "format_pval_annotation(symbols[lp], .7, 1, 6) # Luad\n",
    "format_pval_annotation(symbols[lsp], 1.7, 2, 5) # Lscc\n",
    "format_pval_annotation(symbols[bp], 2.7, 3, 4) # Brca\n",
    "format_pval_annotation(symbols[op], 3.7, 4, 2.6) # Ovarian\n",
    "format_pval_annotation(symbols[cp], 4.7, 5, 2.6)  # Colon\n",
    "format_pval_annotation(symbols[ep], 5.7, 6.2, 2.6) # Endometrial\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save figure\n",
    "#fig = boxplot.get_figure()\n",
    "#fig.savefig(\".png\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
