{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8a220f11",
   "metadata": {},
   "source": [
    "# Part 1: Create database set\n",
    "By inputting a target gene, a nested dictionary will be created, with each key being the name of a cancer and each value being a dictionary containing the genomic, proteomic, transcriptomic, and phosphoproteomic data. In other words, the dictionary will be:\n",
    "\n",
    "{\n",
    "    cancer : {   \n",
    "    \n",
    "            \"gen\" : <cancer's genomic data>,\n",
    "            \"prot\" : <cancer's proteomic data>,\n",
    "            \"trans\": <cancer's transcriptomic data>,\n",
    "            \"phos\": <cancer's phosphoproteomic data>\n",
    "        }\n",
    "    ... for each cancer\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "dede19fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import cptac.pancan as pc\n",
    "import numpy as np\n",
    "import cptac.utils as ut\n",
    "import plot_utils as plt\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4159d5ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PancanBrca',\n",
       " 'PancanCcrcc',\n",
       " 'PancanCoad',\n",
       " 'PancanGbm',\n",
       " 'PancanHnscc',\n",
       " 'PancanLscc',\n",
       " 'PancanLuad',\n",
       " 'PancanOv',\n",
       " 'PancanUcec',\n",
       " 'PancanPdac']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset_names = pc.list_datasets(False)\n",
    "dataset_names  = dataset_names.split('\\n')\n",
    "dataset_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c45e7a87",
   "metadata": {},
   "outputs": [],
   "source": [
    "target_gene = \"TP53\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "36f26ae4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading broadbrca v1.0...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/sdsquires/anaconda3/lib/python3.9/site-packages/gtfparse/read_gtf.py:151: FutureWarning: The warn_bad_lines argument has been deprecated and will be removed in a future version.\n",
      "\n",
      "\n",
      "  result = parse_gtf(\n",
      "/home/sdsquires/anaconda3/lib/python3.9/site-packages/gtfparse/read_gtf.py:151: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version.\n",
      "\n",
      "\n",
      "  result = parse_gtf(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 68 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n",
      "cptac warning: washu_CNV data was not found for the following samples, so washu_CNV data columns were filled with NaN for these samples: 01BR044, 604 (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 104 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 44 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                      \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 67 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 14 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                        \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 9 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                        \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 50 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                   \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 5 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                        \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 73 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                        \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 41 samples for the TP53 gene (/home/sdsquires/anaconda3/lib/python3.9/site-packages/cptac/pancan/pancandataset.py, line 174)\n"
     ]
    }
   ],
   "source": [
    "datasets = {}\n",
    "for name in dataset_names:\n",
    "    exec(f\"cancer = pc.{name}(no_internet = True)\")  \n",
    "    genotype_data = cancer.get_genotype_all_vars(mutations_genes = target_gene, omics_source = \"washu\")\n",
    "    \n",
    "    proteomic_data = cancer.get_proteomics(source = 'umich')\n",
    "    proteomic_data = proteomic_data.droplevel(\"Database_ID\", axis = 1)\n",
    "    \n",
    "    phospho_data = cancer.get_phosphoproteomics()\n",
    "    phospho_data.loc[:, phospho_data.columns.get_level_values(0) == \"TP53\"].droplevel([\"Database_ID\", \"Peptide\"], axis = 1)\n",
    "    \n",
    "    transcript_data = cancer.get_transcriptomics(source = \"washu\")\n",
    "    transcript_data = transcript_data.droplevel(\"Database_ID\", axis = 1)\n",
    "    \n",
    "    datasets[name[6:]] = {\n",
    "                        \"gen\" : genotype_data,\n",
    "                        \"prot\": proteomic_data,\n",
    "                        \"trans\": transcript_data,\n",
    "                        \"phos\" : phospho_data\n",
    "                         }    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a54fff79",
   "metadata": {},
   "source": [
    "# Part 2: Select mutations\n",
    "Determine which cancers have mutation types with rates above 20%.\n",
    "\n",
    "To facilitate that, the mutation types from the get_genotype_all_vars must be grouped into functional types.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b6178983",
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_types = [\"Deletion\", \"Indel\", \"Missense\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9bfa5779",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Grouping the mutation types for each cancer\n",
    "for name, dataset in datasets.items():\n",
    "    genotypes = dataset[\"gen\"]\n",
    "    genotypes.replace([\"Nonsense_Mutation\", \"Frame_Shift_Ins\", \"Frame_Shift_Del\"], \"Truncation\", inplace = True)\n",
    "    genotypes.replace([\"In_Frame_Del\", \"In_Frame_Ins\"], \"Indel\", inplace = True)\n",
    "    genotypes.replace([\"Missense_Mutation\", \"Wildtype_Tumor\"], [\"Missense\", \"Wildtype\"], inplace = True)\n",
    "    genotypes = genotypes.loc[genotypes[\"Mutation\"] != \"Silent\"]\n",
    "    datasets[name][\"gen\"] = genotypes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9ddba048",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a bargraph displaying the mutation types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3e8e0509",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Brca': {'Deletion'}, 'Ccrcc': set(), 'Coad': {'Deletion'}, 'Gbm': set(), 'Hnscc': {'Missense'}, 'Lscc': set(), 'Luad': set(), 'Ov': {'Deletion', 'Missense'}, 'Ucec': set(), 'Pdac': set()}\n"
     ]
    }
   ],
   "source": [
    "# Determine which cancers have mutation rates over 20%\n",
    "variants_to_analyze = {}\n",
    "for name, dataset in datasets.items():\n",
    "    mut_type_freq = dataset[\"gen\"][\"Mutation\"].value_counts() / len(dataset[\"prot\"])\n",
    "    mut_type_freq = mut_type_freq.loc[mut_type_freq >= 0.2]\n",
    "    variants_to_analyze[name] = set(mut_type_freq.index.tolist())\n",
    "    variants_to_analyze[name].discard(\"Wildtype\")\n",
    "print(str(variants_to_analyze))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "368f9dd3",
   "metadata": {},
   "source": [
    "# Part 3: Cis effects\n",
    "Determines which genes that, given a mutation, will change the expression in their own proteomics, transcriptomics, or phosphoproteomics.This analysis is repeated for every cancer, and every mutation type.\n",
    "\n",
    "Writes out a file containing the infomation for each mutation and cancer, and creates boxplot summarizing the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "588253ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import ttest_ind\n",
    "omics_key = {\"prot\":\"proteomics\", \"trans\":\"transcriptomics\",\"phos\":\"phosphoproteomics\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "0aa454a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a function that will analyze the protein cis effects for a given mutation type\n",
    "def analyze_cis_mutation(omics_type, mutation_type, write_out = True, pval_only = True):\n",
    "    \"\"\"Analyzes the cis effects for a given mutation type across all cancers.\n",
    "    \n",
    "    Parameters:\n",
    "    mutation_type (str): The mutation type to analyze, as a string. Possible values given in mutation_types.\n",
    "    omics_type (str): The omics to analyze. Possible values are in list [\"prot\", \"trans\", \"phos\"]\n",
    "    write_out (bool, optional): Whether to write out the data to a separate file.\n",
    "    pval_only (bool, optional): If write_out, only writes out the p_value for each analysis. If not write_out, has no effect.\n",
    "    \n",
    "    Returns the output of scipy.ttest_ind() between \"Wildtype\" and mutation_type tumors. If analyzing phosphoproteomics, returns a list of such outputs\n",
    "    \"\"\"\n",
    "    if write_out:\n",
    "        # Clears the target file\n",
    "        file_name = omics_key[omics_type].capitalize() + '_' + mutation_type.capitalize() + 's.txt'\n",
    "        with open(\"Cis_Effect_output/\" + file_name, 'w') as out_file: pass\n",
    "    for name, dataset in datasets.items():\n",
    "        if mutation_type not in variants_to_analyze[name]: continue\n",
    "        \n",
    "        # Separate the wildtype and mutated samples\n",
    "        genotype_and_omic = pd.merge(dataset[\"gen\"], dataset[omics_type], left_index = True, right_index = True)\n",
    "        mutation = genotype_and_omic.loc[genotype_and_omic[\"Mutation\"] == mutation_type][target_gene].dropna()\n",
    "        wildtype = genotype_and_omic.loc[genotype_and_omic[\"Mutation\"] == \"Wildtype\"][target_gene].dropna()\n",
    "        \n",
    "        # Phosphoproteomics must be analyzed at each site in the gene\n",
    "        if omics_type != \"phos\":\n",
    "            result = ttest_ind(mutation, wildtype)[1]\n",
    "        else:\n",
    "            result = [(site, ttest_ind(mutation[site], wildtype[site])[1]) for site in mutation.columns if site in wildtype.columns]\n",
    "                \n",
    "        if write_out:\n",
    "            output_string = \"\"\n",
    "            if not pval_only:\n",
    "                output_string += f\"{name}\\n\"\n",
    "                output_string += f\"Number of {mutation_type}s = {len(mutation)}\\n\"\n",
    "                output_string += f\"Number of Wildtypes = {len(wildtype)}\\n\"\n",
    "            if omics_type != \"phos\":\n",
    "                output_string += f\"{name} {mutation_type} P-value = {result}\\n\\n\"\n",
    "            else:\n",
    "                output_string += f\"{name} {mutation_type} P-values:\\n\"\n",
    "                for site in result:\n",
    "                    output_string += f\"   {site[0]}: {site[1]}\\n\"\n",
    "                ouput_string += '\\n'\n",
    "            with open(\"Cis_Effect_output/\" + file_name, \"a\") as out_file:\n",
    "                print(output_string)\n",
    "                out_file.write(\"output_string\") \n",
    "                \n",
    "    try:\n",
    "        return result\n",
    "    except UnboundLocalError:\n",
    "        return\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8292543a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# analyze_cis_mutation(\"Deletion\", \"prot\", True, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "d7643c46",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Brca\n",
      "Number of Deletions = 32\n",
      "Number of Wildtypes = 46\n",
      "Brca Deletion P-value = 0.2621251374049456\n",
      "\n",
      "\n",
      "Coad\n",
      "Number of Deletions = 34\n",
      "Number of Wildtypes = 16\n",
      "Coad Deletion P-value = 0.01191040809308662\n",
      "\n",
      "\n",
      "Ov\n",
      "Number of Deletions = 40\n",
      "Number of Wildtypes = 4\n",
      "Ov Deletion P-value = 0.36078400842934055\n",
      "\n",
      "\n",
      "Hnscc\n",
      "Number of Missenses = 44\n",
      "Number of Wildtypes = 11\n",
      "Hnscc Missense P-value = 2.176079318065972e-06\n",
      "\n",
      "\n",
      "Ov\n",
      "Number of Missenses = 24\n",
      "Number of Wildtypes = 4\n",
      "Ov Missense P-value = 0.0011348104751968877\n",
      "\n",
      "\n",
      "Brca\n",
      "Number of Deletions = 32\n",
      "Number of Wildtypes = 46\n",
      "Brca Deletion P-value = 0.0006822070180091262\n",
      "\n",
      "\n",
      "Coad\n",
      "Number of Deletions = 57\n",
      "Number of Wildtypes = 35\n",
      "Coad Deletion P-value = 0.009417539095200425\n",
      "\n",
      "\n",
      "Ov\n",
      "Number of Deletions = 40\n",
      "Number of Wildtypes = 4\n",
      "Ov Deletion P-value = 0.1685545651519631\n",
      "\n",
      "\n",
      "Hnscc\n",
      "Number of Missenses = 46\n",
      "Number of Wildtypes = 13\n",
      "Hnscc Missense P-value = 0.12091072132069322\n",
      "\n",
      "\n",
      "Ov\n",
      "Number of Missenses = 24\n",
      "Number of Wildtypes = 4\n",
      "Ov Missense P-value = 0.11273859276247053\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_15401/1793494427.py:21: FutureWarning: merging between different levels is deprecated and will be removed in a future version. (1 levels on the left,4 on the right)\n",
      "  genotype_and_omic = pd.merge(dataset[\"gen\"], dataset[omics_type], left_index = True, right_index = True)\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'TP53'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3360\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3361\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3362\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'TP53'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_15401/2998773360.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0momics_type\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m\"prot\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"trans\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"phos\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mmutation_type\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mmutation_types\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m         \u001b[0manalyze_cis_mutation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutation_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwrite_out\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpval_only\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/tmp/ipykernel_15401/1793494427.py\u001b[0m in \u001b[0;36manalyze_cis_mutation\u001b[0;34m(omics_type, mutation_type, write_out, pval_only)\u001b[0m\n\u001b[1;32m     20\u001b[0m         \u001b[0;31m# Separate the wildtype and mutated samples\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m         \u001b[0mgenotype_and_omic\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdataset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"gen\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdataset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0momics_type\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mleft_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mright_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m         \u001b[0mmutation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgenotype_and_omic\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mgenotype_and_omic\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Mutation\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mmutation_type\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtarget_gene\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdropna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m         \u001b[0mwildtype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgenotype_and_omic\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mgenotype_and_omic\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Mutation\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"Wildtype\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtarget_gene\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdropna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3456\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3457\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3458\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3459\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3460\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3361\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3362\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3363\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3364\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3365\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhasnans\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'TP53'"
     ]
    }
   ],
   "source": [
    "#Now, run the analysis for every mutation_type, for every omics_type\n",
    "for omics_type in [\"prot\", \"trans\", \"phos\"]:\n",
    "    for mutation_type in mutation_types:\n",
    "        analyze_cis_mutation(omics_type, mutation_type, write_out = True, pval_only = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "51e689a6",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'deletion_sets' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_15401/243624243.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"deletion_protein_data.txt\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0moutFile\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdataset\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdatasets\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0mname\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdeletion_sets\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m         \u001b[0moutFile\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m'\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m         \u001b[0mgenotype_and_prot\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdataset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdataset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mleft_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mright_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'deletion_sets' is not defined"
     ]
    }
   ],
   "source": [
    "# Create a boxplot for each type, either in the function or outside of it\n",
    "# Corrections must be performed to reduce the FDR"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "73617b45",
   "metadata": {},
   "source": [
    "# Part 4: Trans Effects\n",
    "Determines which genes that, given a mutation, will change the expression in another protein's proteomics, transcriptomics, or phosphoproteomics.This analysis is repeated for every cancer, and every mutation type.\n",
    "\n",
    "To reduce the computational intensity and the corrections needed, the scope will be restricted to all genes whose corresponding proteins have a known interaction with the proteins corresponding to the target_gene. *(For example, Given the target_gene 'TP53', the set of genes will be all those whose proteins share a pathway with TP53.)*\n",
    "\n",
    "Writes out a file containing the infomation for each mutation and cancer, and creates boxplot summarizing the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "6aac2474",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>pathway</th>\n",
       "      <th>pathway_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Activation of NOXA and translocation to mitoch...</td>\n",
       "      <td>R-HSA-111448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Activation of PUMA and translocation to mitoch...</td>\n",
       "      <td>R-HSA-139915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Pre-NOTCH Transcription and Translation</td>\n",
       "      <td>R-HSA-1912408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Oxidative Stress Induced Senescence</td>\n",
       "      <td>R-HSA-2559580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Formation of Senescence-Associated Heterochrom...</td>\n",
       "      <td>R-HSA-2559584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Oncogene Induced Senescence</td>\n",
       "      <td>R-HSA-2559585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TP53</td>\n",
       "      <td>DNA Damage/Telomere Stress Induced Senescence</td>\n",
       "      <td>R-HSA-2559586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TP53</td>\n",
       "      <td>SUMOylation of transcription factors</td>\n",
       "      <td>R-HSA-3232118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Autodegradation of the E3 ubiquitin ligase COP1</td>\n",
       "      <td>R-HSA-349425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Association of TriC/CCT with target proteins d...</td>\n",
       "      <td>R-HSA-390471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Pyroptosis</td>\n",
       "      <td>R-HSA-5620971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 Regulates Metabolic Genes</td>\n",
       "      <td>R-HSA-5628897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Ub-specific processing proteases</td>\n",
       "      <td>R-HSA-5689880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Ovarian tumor domain proteases</td>\n",
       "      <td>R-HSA-5689896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Recruitment and ATM-mediated phosphorylation o...</td>\n",
       "      <td>R-HSA-5693565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Interleukin-4 and Interleukin-13 signaling</td>\n",
       "      <td>R-HSA-6785807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 Regulates Transcription of DNA Repair Genes</td>\n",
       "      <td>R-HSA-6796648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 Regulates Transcription of Genes Involved...</td>\n",
       "      <td>R-HSA-6803204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 regulates transcription of several additi...</td>\n",
       "      <td>R-HSA-6803205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 Regulates Transcription of Caspase Activa...</td>\n",
       "      <td>R-HSA-6803207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 Regulates Transcription of Death Receptor...</td>\n",
       "      <td>R-HSA-6803211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 Regulates Transcription of Genes Involved...</td>\n",
       "      <td>R-HSA-6804114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 regulates transcription of additional cel...</td>\n",
       "      <td>R-HSA-6804115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 Regulates Transcription of Genes Involved...</td>\n",
       "      <td>R-HSA-6804116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Regulation of TP53 Expression</td>\n",
       "      <td>R-HSA-6804754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Regulation of TP53 Activity through Phosphoryl...</td>\n",
       "      <td>R-HSA-6804756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Regulation of TP53 Degradation</td>\n",
       "      <td>R-HSA-6804757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Regulation of TP53 Activity through Acetylation</td>\n",
       "      <td>R-HSA-6804758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Regulation of TP53 Activity through Associatio...</td>\n",
       "      <td>R-HSA-6804759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Regulation of TP53 Activity through Methylation</td>\n",
       "      <td>R-HSA-6804760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TP53</td>\n",
       "      <td>PI5P Regulates TP53 Acetylation</td>\n",
       "      <td>R-HSA-6811555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TP53</td>\n",
       "      <td>G2/M DNA damage checkpoint</td>\n",
       "      <td>R-HSA-69473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TP53</td>\n",
       "      <td>G2/M Checkpoints</td>\n",
       "      <td>R-HSA-69481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Stabilization of p53</td>\n",
       "      <td>R-HSA-69541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Transcriptional  activation of  cell cycle inh...</td>\n",
       "      <td>R-HSA-69895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>TP53</td>\n",
       "      <td>The role of GTSE1 in G2/M progression after G2...</td>\n",
       "      <td>R-HSA-8852276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Transcriptional Regulation by VENTX</td>\n",
       "      <td>R-HSA-8853884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>TP53</td>\n",
       "      <td>RUNX3 regulates CDKN1A transcription</td>\n",
       "      <td>R-HSA-8941855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Regulation of PTEN gene transcription</td>\n",
       "      <td>R-HSA-8943724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Loss of function of TP53 in cancer due to loss...</td>\n",
       "      <td>R-HSA-9723905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Factors involved in megakaryocyte development ...</td>\n",
       "      <td>R-HSA-983231</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      id                                            pathway     pathway_id\n",
       "0   TP53  Activation of NOXA and translocation to mitoch...   R-HSA-111448\n",
       "1   TP53  Activation of PUMA and translocation to mitoch...   R-HSA-139915\n",
       "2   TP53            Pre-NOTCH Transcription and Translation  R-HSA-1912408\n",
       "3   TP53                Oxidative Stress Induced Senescence  R-HSA-2559580\n",
       "4   TP53  Formation of Senescence-Associated Heterochrom...  R-HSA-2559584\n",
       "5   TP53                        Oncogene Induced Senescence  R-HSA-2559585\n",
       "6   TP53      DNA Damage/Telomere Stress Induced Senescence  R-HSA-2559586\n",
       "7   TP53               SUMOylation of transcription factors  R-HSA-3232118\n",
       "8   TP53    Autodegradation of the E3 ubiquitin ligase COP1   R-HSA-349425\n",
       "9   TP53  Association of TriC/CCT with target proteins d...   R-HSA-390471\n",
       "10  TP53                                         Pyroptosis  R-HSA-5620971\n",
       "11  TP53                     TP53 Regulates Metabolic Genes  R-HSA-5628897\n",
       "12  TP53                   Ub-specific processing proteases  R-HSA-5689880\n",
       "13  TP53                     Ovarian tumor domain proteases  R-HSA-5689896\n",
       "14  TP53  Recruitment and ATM-mediated phosphorylation o...  R-HSA-5693565\n",
       "15  TP53         Interleukin-4 and Interleukin-13 signaling  R-HSA-6785807\n",
       "16  TP53   TP53 Regulates Transcription of DNA Repair Genes  R-HSA-6796648\n",
       "17  TP53  TP53 Regulates Transcription of Genes Involved...  R-HSA-6803204\n",
       "18  TP53  TP53 regulates transcription of several additi...  R-HSA-6803205\n",
       "19  TP53  TP53 Regulates Transcription of Caspase Activa...  R-HSA-6803207\n",
       "20  TP53  TP53 Regulates Transcription of Death Receptor...  R-HSA-6803211\n",
       "21  TP53  TP53 Regulates Transcription of Genes Involved...  R-HSA-6804114\n",
       "22  TP53  TP53 regulates transcription of additional cel...  R-HSA-6804115\n",
       "23  TP53  TP53 Regulates Transcription of Genes Involved...  R-HSA-6804116\n",
       "24  TP53                      Regulation of TP53 Expression  R-HSA-6804754\n",
       "25  TP53  Regulation of TP53 Activity through Phosphoryl...  R-HSA-6804756\n",
       "26  TP53                     Regulation of TP53 Degradation  R-HSA-6804757\n",
       "27  TP53    Regulation of TP53 Activity through Acetylation  R-HSA-6804758\n",
       "28  TP53  Regulation of TP53 Activity through Associatio...  R-HSA-6804759\n",
       "29  TP53    Regulation of TP53 Activity through Methylation  R-HSA-6804760\n",
       "30  TP53                    PI5P Regulates TP53 Acetylation  R-HSA-6811555\n",
       "31  TP53                         G2/M DNA damage checkpoint    R-HSA-69473\n",
       "32  TP53                                   G2/M Checkpoints    R-HSA-69481\n",
       "33  TP53                               Stabilization of p53    R-HSA-69541\n",
       "34  TP53  Transcriptional  activation of  cell cycle inh...    R-HSA-69895\n",
       "35  TP53  The role of GTSE1 in G2/M progression after G2...  R-HSA-8852276\n",
       "36  TP53                Transcriptional Regulation by VENTX  R-HSA-8853884\n",
       "37  TP53               RUNX3 regulates CDKN1A transcription  R-HSA-8941855\n",
       "38  TP53              Regulation of PTEN gene transcription  R-HSA-8943724\n",
       "39  TP53  Loss of function of TP53 in cancer due to loss...  R-HSA-9723905\n",
       "40  TP53  Factors involved in megakaryocyte development ...   R-HSA-983231"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "paths = ut.get_pathways_with_proteins(proteins = target_gene, database = 'reactome')\n",
    "paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "b1daf485",
   "metadata": {},
   "outputs": [],
   "source": [
    "interacting_proteins = ut.get_proteins_in_pathways(paths[\"pathway_id\"], database = 'reactome')\n",
    "interacting_proteins = list(interacting_proteins[\"member\"])\n",
    "\n",
    "genes_measured = list(datasets[\"Brca\"][\"prot\"].columns)\n",
    "new_genes = sorted(list(set([protein for protein in interacting_proteins if protein in genes_measured])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "58036cab",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1077\n",
      "12046\n",
      "884\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['ABL1',\n",
       " 'ABRAXAS1',\n",
       " 'ACD',\n",
       " 'ACTB',\n",
       " 'ADRM1',\n",
       " 'AGO1',\n",
       " 'AGO2',\n",
       " 'AGO3',\n",
       " 'AGO4',\n",
       " 'AIFM2',\n",
       " 'AK3',\n",
       " 'AKAP1',\n",
       " 'AKAP10',\n",
       " 'AKT1',\n",
       " 'AKT2',\n",
       " 'AKT3',\n",
       " 'ALOX15',\n",
       " 'ALOX5',\n",
       " 'ANAPC1',\n",
       " 'ANAPC10',\n",
       " 'ANAPC11',\n",
       " 'ANAPC15',\n",
       " 'ANAPC16',\n",
       " 'ANAPC2',\n",
       " 'ANAPC4',\n",
       " 'ANAPC5',\n",
       " 'ANAPC7',\n",
       " 'ANXA1',\n",
       " 'AP3M1',\n",
       " 'APAF1',\n",
       " 'APBB1',\n",
       " 'APC',\n",
       " 'AR',\n",
       " 'ARFGEF2',\n",
       " 'ARID3A',\n",
       " 'ARRB1',\n",
       " 'ARRB2',\n",
       " 'ASF1A',\n",
       " 'ATF2',\n",
       " 'ATM',\n",
       " 'ATN1',\n",
       " 'ATR',\n",
       " 'ATRIP',\n",
       " 'ATXN7',\n",
       " 'AURKA',\n",
       " 'AURKB',\n",
       " 'AXIN1',\n",
       " 'BABAM1',\n",
       " 'BABAM2',\n",
       " 'BAK1',\n",
       " 'BANP',\n",
       " 'BAP1',\n",
       " 'BARD1',\n",
       " 'BATF',\n",
       " 'BAX',\n",
       " 'BAZ1B',\n",
       " 'BBC3',\n",
       " 'BCL2',\n",
       " 'BCL2L1',\n",
       " 'BCL2L14',\n",
       " 'BCL6',\n",
       " 'BECN1',\n",
       " 'BIRC2',\n",
       " 'BIRC3',\n",
       " 'BIRC5',\n",
       " 'BLM',\n",
       " 'BNIP3L',\n",
       " 'BRCA1',\n",
       " 'BRCC3',\n",
       " 'BRD1',\n",
       " 'BRD7',\n",
       " 'BRIP1',\n",
       " 'BRPF1',\n",
       " 'BRPF3',\n",
       " 'BTG2',\n",
       " 'CABIN1',\n",
       " 'CABLES1',\n",
       " 'CABLES2',\n",
       " 'CAPZA1',\n",
       " 'CAPZA2',\n",
       " 'CAPZB',\n",
       " 'CARM1',\n",
       " 'CARMIL1',\n",
       " 'CASP1',\n",
       " 'CASP10',\n",
       " 'CASP2',\n",
       " 'CASP3',\n",
       " 'CASP4',\n",
       " 'CASP5',\n",
       " 'CASP6',\n",
       " 'CBX2',\n",
       " 'CBX4',\n",
       " 'CBX5',\n",
       " 'CBX6',\n",
       " 'CBX8',\n",
       " 'CCNA2',\n",
       " 'CCNB1',\n",
       " 'CCNB2',\n",
       " 'CCND1',\n",
       " 'CCNE1',\n",
       " 'CCNE2',\n",
       " 'CCNH',\n",
       " 'CCNK',\n",
       " 'CCNT1',\n",
       " 'CCNT2',\n",
       " 'CCP110',\n",
       " 'CCT2',\n",
       " 'CCT3',\n",
       " 'CCT4',\n",
       " 'CCT5',\n",
       " 'CCT6A',\n",
       " 'CCT6B',\n",
       " 'CCT7',\n",
       " 'CCT8',\n",
       " 'CD36',\n",
       " 'CDC16',\n",
       " 'CDC20',\n",
       " 'CDC23',\n",
       " 'CDC25C',\n",
       " 'CDC26',\n",
       " 'CDC27',\n",
       " 'CDC42',\n",
       " 'CDC45',\n",
       " 'CDC6',\n",
       " 'CDC7',\n",
       " 'CDK1',\n",
       " 'CDK12',\n",
       " 'CDK13',\n",
       " 'CDK2',\n",
       " 'CDK4',\n",
       " 'CDK5',\n",
       " 'CDK6',\n",
       " 'CDK7',\n",
       " 'CDK9',\n",
       " 'CDKN1A',\n",
       " 'CDKN1B',\n",
       " 'CDKN2A',\n",
       " 'CDKN2B',\n",
       " 'CDKN2C',\n",
       " 'CDKN2D',\n",
       " 'CEBPB',\n",
       " 'CEBPD',\n",
       " 'CENPE',\n",
       " 'CENPJ',\n",
       " 'CFTR',\n",
       " 'CHD3',\n",
       " 'CHD4',\n",
       " 'CHEK1',\n",
       " 'CHEK2',\n",
       " 'CHM',\n",
       " 'CHMP2A',\n",
       " 'CHMP2B',\n",
       " 'CHMP3',\n",
       " 'CHMP4A',\n",
       " 'CHMP4B',\n",
       " 'CHMP4C',\n",
       " 'CHMP6',\n",
       " 'CHMP7',\n",
       " 'CLSPN',\n",
       " 'CNOT1',\n",
       " 'CNOT10',\n",
       " 'CNOT11',\n",
       " 'CNOT2',\n",
       " 'CNOT3',\n",
       " 'CNOT4',\n",
       " 'CNOT6',\n",
       " 'CNOT6L',\n",
       " 'CNOT7',\n",
       " 'CNOT8',\n",
       " 'CNOT9',\n",
       " 'COL1A2',\n",
       " 'COP1',\n",
       " 'COX11',\n",
       " 'COX14',\n",
       " 'COX16',\n",
       " 'COX18',\n",
       " 'COX19',\n",
       " 'COX20',\n",
       " 'COX4I1',\n",
       " 'COX5A',\n",
       " 'COX5B',\n",
       " 'COX6A1',\n",
       " 'COX6B1',\n",
       " 'COX6C',\n",
       " 'COX7A2L',\n",
       " 'COX7B',\n",
       " 'COX7C',\n",
       " 'CRADD',\n",
       " 'CREBBP',\n",
       " 'CSF1R',\n",
       " 'CSNK2A1',\n",
       " 'CSNK2A2',\n",
       " 'CSNK2B',\n",
       " 'CTDP1',\n",
       " 'CTNNB1',\n",
       " 'CYLD',\n",
       " 'DAXX',\n",
       " 'DBF4',\n",
       " 'DCAF7',\n",
       " 'DDB2',\n",
       " 'DDIT4',\n",
       " 'DDX58',\n",
       " 'DNA2',\n",
       " 'DOCK1',\n",
       " 'DOCK10',\n",
       " 'DOCK11',\n",
       " 'DOCK2',\n",
       " 'DOCK4',\n",
       " 'DOCK5',\n",
       " 'DOCK6',\n",
       " 'DOCK7',\n",
       " 'DOCK8',\n",
       " 'DOCK9',\n",
       " 'DYRK2',\n",
       " 'E2F1',\n",
       " 'E2F3',\n",
       " 'E2F4',\n",
       " 'E2F7',\n",
       " 'E2F8',\n",
       " 'EED',\n",
       " 'EGR1',\n",
       " 'EHD1',\n",
       " 'EHD2',\n",
       " 'EHD3',\n",
       " 'EHMT1',\n",
       " 'EHMT2',\n",
       " 'ELANE',\n",
       " 'ELF3',\n",
       " 'ELL',\n",
       " 'ELOA',\n",
       " 'ELOB',\n",
       " 'ELOC',\n",
       " 'EP300',\n",
       " 'EP400',\n",
       " 'ERCC2',\n",
       " 'ERCC3',\n",
       " 'ERF',\n",
       " 'ESR1',\n",
       " 'ETS1',\n",
       " 'EXO1',\n",
       " 'EYA2',\n",
       " 'EYA3',\n",
       " 'EZH2',\n",
       " 'F13A1',\n",
       " 'FANCD2',\n",
       " 'FANCI',\n",
       " 'FAS',\n",
       " 'FBXL3',\n",
       " 'FBXO4',\n",
       " 'FBXO6',\n",
       " 'FBXW2',\n",
       " 'FBXW4',\n",
       " 'FBXW5',\n",
       " 'FBXW7',\n",
       " 'FBXW9',\n",
       " 'FGF2',\n",
       " 'FKBP8',\n",
       " 'FKBP9',\n",
       " 'FKBPL',\n",
       " 'FN1',\n",
       " 'FOS',\n",
       " 'FOXO1',\n",
       " 'FOXO3',\n",
       " 'FOXO4',\n",
       " 'FSCN1',\n",
       " 'FZR1',\n",
       " 'G6PD',\n",
       " 'GAPDHS',\n",
       " 'GATA2',\n",
       " 'GATA3',\n",
       " 'GATA6',\n",
       " 'GATAD2A',\n",
       " 'GATAD2B',\n",
       " 'GBA',\n",
       " 'GLS',\n",
       " 'GLS2',\n",
       " 'GPI',\n",
       " 'GPX2',\n",
       " 'GSDMD',\n",
       " 'GSDME',\n",
       " 'GSR',\n",
       " 'GTF2F1',\n",
       " 'GTF2F2',\n",
       " 'GTF2H1',\n",
       " 'GTF2H3',\n",
       " 'GTF2H4',\n",
       " 'GTF2H5',\n",
       " 'GTSE1',\n",
       " 'GZMB',\n",
       " 'H1-0',\n",
       " 'H1-1',\n",
       " 'H1-2',\n",
       " 'H1-3',\n",
       " 'H1-4',\n",
       " 'H1-5',\n",
       " 'H2AC21',\n",
       " 'H2AJ',\n",
       " 'H2AX',\n",
       " 'H2AZ2',\n",
       " 'H2BC1',\n",
       " 'H2BC18',\n",
       " 'H2BC5',\n",
       " 'HBD',\n",
       " 'HBG1',\n",
       " 'HBG2',\n",
       " 'HDAC1',\n",
       " 'HDAC2',\n",
       " 'HDAC3',\n",
       " 'HDAC5',\n",
       " 'HDAC7',\n",
       " 'HERC2',\n",
       " 'HGF',\n",
       " 'HGS',\n",
       " 'HIC1',\n",
       " 'HIF1A',\n",
       " 'HIPK1',\n",
       " 'HIPK2',\n",
       " 'HIRA',\n",
       " 'HMG20B',\n",
       " 'HMGA1',\n",
       " 'HMGA2',\n",
       " 'HMGB1',\n",
       " 'HMOX1',\n",
       " 'HSP90AA1',\n",
       " 'HSP90AB1',\n",
       " 'HSP90B1',\n",
       " 'HSPA8',\n",
       " 'HUS1',\n",
       " 'ICAM1',\n",
       " 'ID1',\n",
       " 'IDE',\n",
       " 'IFIH1',\n",
       " 'IGFBP3',\n",
       " 'IGHE',\n",
       " 'IGHG1',\n",
       " 'IGHG4',\n",
       " 'IKBKG',\n",
       " 'IL13RA1',\n",
       " 'IL18',\n",
       " 'IL33',\n",
       " 'IL6R',\n",
       " 'ING2',\n",
       " 'ING5',\n",
       " 'IRF1',\n",
       " 'IRF2',\n",
       " 'IRF4',\n",
       " 'ITGAM',\n",
       " 'ITGAX',\n",
       " 'ITGB1',\n",
       " 'ITGB2',\n",
       " 'ITPK1',\n",
       " 'JAK1',\n",
       " 'JAK2',\n",
       " 'JAK3',\n",
       " 'JMJD1C',\n",
       " 'JMY',\n",
       " 'JUN',\n",
       " 'JUNB',\n",
       " 'KAT2A',\n",
       " 'KAT2B',\n",
       " 'KAT5',\n",
       " 'KAT6A',\n",
       " 'KDM1A',\n",
       " 'KDM4A',\n",
       " 'KDM4B',\n",
       " 'KDM6B',\n",
       " 'KEAP1',\n",
       " 'KIF11',\n",
       " 'KIF12',\n",
       " 'KIF13A',\n",
       " 'KIF13B',\n",
       " 'KIF15',\n",
       " 'KIF16B',\n",
       " 'KIF18A',\n",
       " 'KIF18B',\n",
       " 'KIF1A',\n",
       " 'KIF1B',\n",
       " 'KIF1C',\n",
       " 'KIF20A',\n",
       " 'KIF20B',\n",
       " 'KIF21A',\n",
       " 'KIF21B',\n",
       " 'KIF22',\n",
       " 'KIF23',\n",
       " 'KIF26B',\n",
       " 'KIF27',\n",
       " 'KIF2A',\n",
       " 'KIF2C',\n",
       " 'KIF3A',\n",
       " 'KIF3B',\n",
       " 'KIF3C',\n",
       " 'KIF4A',\n",
       " 'KIF5A',\n",
       " 'KIF5B',\n",
       " 'KIF5C',\n",
       " 'KIFAP3',\n",
       " 'KIFC1',\n",
       " 'KIFC3',\n",
       " 'KLC1',\n",
       " 'KLC2',\n",
       " 'KLC3',\n",
       " 'KLC4',\n",
       " 'KMT5A',\n",
       " 'LAMA5',\n",
       " 'LAMTOR1',\n",
       " 'LAMTOR2',\n",
       " 'LAMTOR3',\n",
       " 'LAMTOR4',\n",
       " 'LAMTOR5',\n",
       " 'LBP',\n",
       " 'LCN2',\n",
       " 'LEF1',\n",
       " 'LIF',\n",
       " 'LMNB1',\n",
       " 'LONP2',\n",
       " 'LRPPRC',\n",
       " 'MAF1',\n",
       " 'MAFF',\n",
       " 'MAFG',\n",
       " 'MAFK',\n",
       " 'MAML1',\n",
       " 'MAML2',\n",
       " 'MAML3',\n",
       " 'MAMLD1',\n",
       " 'MAOA',\n",
       " 'MAP2K3',\n",
       " 'MAP2K4',\n",
       " 'MAP2K6',\n",
       " 'MAP2K7',\n",
       " 'MAP3K5',\n",
       " 'MAP3K7',\n",
       " 'MAP4K4',\n",
       " 'MAPK1',\n",
       " 'MAPK10',\n",
       " 'MAPK11',\n",
       " 'MAPK14',\n",
       " 'MAPK3',\n",
       " 'MAPK8',\n",
       " 'MAPK9',\n",
       " 'MAPKAP1',\n",
       " 'MAPKAPK2',\n",
       " 'MAPKAPK3',\n",
       " 'MAPKAPK5',\n",
       " 'MAPRE1',\n",
       " 'MAT2B',\n",
       " 'MAVS',\n",
       " 'MBD3',\n",
       " 'MCL1',\n",
       " 'MCM10',\n",
       " 'MCM2',\n",
       " 'MCM3',\n",
       " 'MCM4',\n",
       " 'MCM5',\n",
       " 'MCM6',\n",
       " 'MCM7',\n",
       " 'MDC1',\n",
       " 'MDM2',\n",
       " 'MDM4',\n",
       " 'MEAF6',\n",
       " 'MECOM',\n",
       " 'MFN1',\n",
       " 'MFN2',\n",
       " 'MICAL1',\n",
       " 'MINK1',\n",
       " 'MITF',\n",
       " 'MLH1',\n",
       " 'MLST8',\n",
       " 'MMP1',\n",
       " 'MMP2',\n",
       " 'MMP3',\n",
       " 'MMP9',\n",
       " 'MNAT1',\n",
       " 'MOV10',\n",
       " 'MRE11',\n",
       " 'MSH2',\n",
       " 'MT-CO1',\n",
       " 'MT-CO2',\n",
       " 'MT-CO3',\n",
       " 'MTA1',\n",
       " 'MTA2',\n",
       " 'MTA3',\n",
       " 'MTOR',\n",
       " 'MUC1',\n",
       " 'MUL1',\n",
       " 'MYB',\n",
       " 'MYC',\n",
       " 'NBN',\n",
       " 'NDN',\n",
       " 'NDRG1',\n",
       " 'NDUFA4',\n",
       " 'NELFA',\n",
       " 'NELFB',\n",
       " 'NELFCD',\n",
       " 'NELFE',\n",
       " 'NFKB1',\n",
       " 'NFKBIA',\n",
       " 'NLRC4',\n",
       " 'NOC2L',\n",
       " 'NOD1',\n",
       " 'NOD2',\n",
       " 'NOP56',\n",
       " 'NOS2',\n",
       " 'NOTCH1',\n",
       " 'NOTCH2',\n",
       " 'NOTCH3',\n",
       " 'NPM1',\n",
       " 'NSD2',\n",
       " 'NUAK1',\n",
       " 'ORC1',\n",
       " 'ORC2',\n",
       " 'ORC3',\n",
       " 'ORC4',\n",
       " 'ORC5',\n",
       " 'ORC6',\n",
       " 'OTUB1',\n",
       " 'OTUB2',\n",
       " 'OTUD3',\n",
       " 'OTUD5',\n",
       " 'OTUD7B',\n",
       " 'PCBP4',\n",
       " 'PCNA',\n",
       " 'PDPK1',\n",
       " 'PHC1',\n",
       " 'PHC2',\n",
       " 'PHC3',\n",
       " 'PHF20',\n",
       " 'PHF21A',\n",
       " 'PIAS1',\n",
       " 'PIAS2',\n",
       " 'PIAS3',\n",
       " 'PIAS4',\n",
       " 'PIK3R1',\n",
       " 'PIN1',\n",
       " 'PIP4K2A',\n",
       " 'PIP4K2B',\n",
       " 'PIP4K2C',\n",
       " 'PIP4P1',\n",
       " 'PKMYT1',\n",
       " 'PLK1',\n",
       " 'PMAIP1',\n",
       " 'PML',\n",
       " 'PMS2',\n",
       " 'POLB',\n",
       " 'POLR2A',\n",
       " 'POLR2B',\n",
       " 'POLR2C',\n",
       " 'POLR2D',\n",
       " 'POLR2E',\n",
       " 'POLR2F',\n",
       " 'POLR2G',\n",
       " 'POLR2H',\n",
       " 'POLR2I',\n",
       " 'POLR2K',\n",
       " 'POLR2L',\n",
       " 'POT1',\n",
       " 'POU2F1',\n",
       " 'PPARG',\n",
       " 'PPP1R13B',\n",
       " 'PPP1R13L',\n",
       " 'PPP2CB',\n",
       " 'PPP2R1A',\n",
       " 'PPP2R1B',\n",
       " 'PPP2R5C',\n",
       " 'PPP5C',\n",
       " 'PRDM1',\n",
       " 'PRDX2',\n",
       " 'PRDX5',\n",
       " 'PRELID1',\n",
       " 'PRKAA1',\n",
       " 'PRKAA2',\n",
       " 'PRKAB1',\n",
       " 'PRKAB2',\n",
       " 'PRKACA',\n",
       " 'PRKACB',\n",
       " 'PRKAG1',\n",
       " 'PRKAG2',\n",
       " 'PRKAR1A',\n",
       " 'PRKAR1B',\n",
       " 'PRKAR2A',\n",
       " 'PRKAR2B',\n",
       " 'PRKCI',\n",
       " 'PRMT1',\n",
       " 'PRMT5',\n",
       " 'PRR5',\n",
       " 'PSMA1',\n",
       " 'PSMA3',\n",
       " 'PSMA4',\n",
       " 'PSMA5',\n",
       " 'PSMA6',\n",
       " 'PSMA7',\n",
       " 'PSMA8',\n",
       " 'PSMB1',\n",
       " 'PSMB10',\n",
       " 'PSMB2',\n",
       " 'PSMB3',\n",
       " 'PSMB4',\n",
       " 'PSMB5',\n",
       " 'PSMB6',\n",
       " 'PSMB7',\n",
       " 'PSMB8',\n",
       " 'PSMB9',\n",
       " 'PSMC1',\n",
       " 'PSMC2',\n",
       " 'PSMC3',\n",
       " 'PSMC4',\n",
       " 'PSMC5',\n",
       " 'PSMC6',\n",
       " 'PSMD1',\n",
       " 'PSMD10',\n",
       " 'PSMD11',\n",
       " 'PSMD12',\n",
       " 'PSMD13',\n",
       " 'PSMD14',\n",
       " 'PSMD2',\n",
       " 'PSMD3',\n",
       " 'PSMD4',\n",
       " 'PSMD5',\n",
       " 'PSMD6',\n",
       " 'PSMD7',\n",
       " 'PSMD8',\n",
       " 'PSMD9',\n",
       " 'PSME1',\n",
       " 'PSME2',\n",
       " 'PSME3',\n",
       " 'PSME4',\n",
       " 'PSMF1',\n",
       " 'PTEN',\n",
       " 'PTRH2',\n",
       " 'RAB5A',\n",
       " 'RABGGTA',\n",
       " 'RABGGTB',\n",
       " 'RAC1',\n",
       " 'RACGAP1',\n",
       " 'RAD1',\n",
       " 'RAD17',\n",
       " 'RAD50',\n",
       " 'RAD51B',\n",
       " 'RAD51C',\n",
       " 'RAD51D',\n",
       " 'RAD9A',\n",
       " 'RB1',\n",
       " 'RBBP4',\n",
       " 'RBBP7',\n",
       " 'RBBP8',\n",
       " 'RBL1',\n",
       " 'RBL2',\n",
       " 'RBPJ',\n",
       " 'RBSN',\n",
       " 'RCE1',\n",
       " 'RCOR1',\n",
       " 'RELA',\n",
       " 'REST',\n",
       " 'RFC2',\n",
       " 'RFC3',\n",
       " 'RFC4',\n",
       " 'RFC5',\n",
       " 'RFFL',\n",
       " 'RGCC',\n",
       " 'RHEB',\n",
       " 'RHOA',\n",
       " 'RHOT1',\n",
       " 'RICTOR',\n",
       " 'RING1',\n",
       " 'RIPK1',\n",
       " 'RIPK2',\n",
       " 'RMI1',\n",
       " 'RMI2',\n",
       " 'RNF123',\n",
       " 'RNF135',\n",
       " 'RNF146',\n",
       " 'RNF168',\n",
       " 'RNF2',\n",
       " 'RNF34',\n",
       " 'RNF8',\n",
       " 'RORA',\n",
       " 'RORC',\n",
       " 'RPA1',\n",
       " 'RPA2',\n",
       " 'RPA3',\n",
       " 'RPTOR',\n",
       " 'RRAGA',\n",
       " 'RRAGB',\n",
       " 'RRAGC',\n",
       " 'RRAGD',\n",
       " 'RRM2B',\n",
       " 'RUNX1',\n",
       " 'RUNX3',\n",
       " 'RUVBL1',\n",
       " 'SAA1',\n",
       " 'SCMH1',\n",
       " 'SCO1',\n",
       " 'SCO2',\n",
       " 'SESN1',\n",
       " 'SESN2',\n",
       " 'SESN3',\n",
       " 'SETD9',\n",
       " 'SFN',\n",
       " 'SH2B1',\n",
       " 'SH2B2',\n",
       " 'SH2B3',\n",
       " 'SIAH2',\n",
       " 'SIN3A',\n",
       " 'SIRT6',\n",
       " 'SKIV2L',\n",
       " 'SKP2',\n",
       " 'SLC38A9',\n",
       " 'SMAD1',\n",
       " 'SMAD2',\n",
       " 'SMAD3',\n",
       " 'SMAD4',\n",
       " 'SMARCA5',\n",
       " 'SMURF2',\n",
       " 'SMYD2',\n",
       " 'SNAI1',\n",
       " 'SNAI2',\n",
       " 'SNW1',\n",
       " 'SNX3',\n",
       " 'SOX2',\n",
       " 'SP1',\n",
       " 'SP3',\n",
       " 'SPHK1',\n",
       " 'SSRP1',\n",
       " 'STAM2',\n",
       " 'STAT1',\n",
       " 'STAT3',\n",
       " 'STAT6',\n",
       " 'STEAP3',\n",
       " 'STK11',\n",
       " 'SUDS3',\n",
       " 'SUMO2',\n",
       " 'SUMO3',\n",
       " 'SUPT16H',\n",
       " 'SUPT4H1',\n",
       " 'SUPT5H',\n",
       " 'SURF1',\n",
       " 'SUZ12',\n",
       " 'TAB1',\n",
       " 'TACO1',\n",
       " 'TADA2B',\n",
       " 'TADA3',\n",
       " 'TAF1',\n",
       " 'TAF10',\n",
       " 'TAF11',\n",
       " 'TAF12',\n",
       " 'TAF13',\n",
       " 'TAF15',\n",
       " 'TAF2',\n",
       " 'TAF3',\n",
       " 'TAF4',\n",
       " 'TAF5',\n",
       " 'TAF6',\n",
       " 'TAF7',\n",
       " 'TAF9',\n",
       " 'TAF9B',\n",
       " 'TBP',\n",
       " 'TCEA1',\n",
       " 'TCF7L2',\n",
       " 'TCP1',\n",
       " 'TERF1',\n",
       " 'TERF2',\n",
       " 'TERF2IP',\n",
       " 'TFAP2A',\n",
       " 'TFAP2B',\n",
       " 'TFAP2C',\n",
       " 'TFDP1',\n",
       " 'TFDP2',\n",
       " 'TGFB1',\n",
       " 'TGFBR1',\n",
       " 'TIGAR',\n",
       " 'TIMP1',\n",
       " 'TINF2',\n",
       " 'TMEM219',\n",
       " 'TNFAIP3',\n",
       " 'TNFRSF10A',\n",
       " 'TNFRSF10B',\n",
       " 'TNFRSF10C',\n",
       " 'TNFRSF1B',\n",
       " 'TNIK',\n",
       " 'TNIP1',\n",
       " 'TNIP2',\n",
       " 'TNKS',\n",
       " 'TNKS1BP1',\n",
       " 'TNKS2',\n",
       " 'TNRC6A',\n",
       " 'TNRC6B',\n",
       " 'TNRC6C',\n",
       " 'TOMM20',\n",
       " 'TOMM70',\n",
       " 'TOP3A',\n",
       " 'TOPBP1',\n",
       " 'TP53BP1',\n",
       " 'TP53BP2',\n",
       " 'TP53I3',\n",
       " 'TP53RK',\n",
       " 'TP63',\n",
       " 'TPX2',\n",
       " 'TRAF2',\n",
       " 'TRAF3',\n",
       " 'TRAF6',\n",
       " 'TRIAP1',\n",
       " 'TRIM25',\n",
       " 'TRIM4',\n",
       " 'TRRAP',\n",
       " 'TSC1',\n",
       " 'TSC2',\n",
       " 'TTC5',\n",
       " 'TUBA1C',\n",
       " 'TUBA3E',\n",
       " 'TUBA4A',\n",
       " 'TUBA8',\n",
       " 'TUBB1',\n",
       " 'TUBB2A',\n",
       " 'TUBB2B',\n",
       " 'TUBB4A',\n",
       " 'TUBB4B',\n",
       " 'TUBB6',\n",
       " 'TUBB8',\n",
       " 'TUBB8B',\n",
       " 'TXNRD1',\n",
       " 'TYK2',\n",
       " 'UBA52',\n",
       " 'UBE2D1',\n",
       " 'UBE2N',\n",
       " 'UBE2S',\n",
       " 'UBE2V2',\n",
       " 'UBN1',\n",
       " 'UBXN1',\n",
       " 'UFD1',\n",
       " 'UIMC1',\n",
       " 'USP10',\n",
       " 'USP11',\n",
       " 'USP12',\n",
       " 'USP13',\n",
       " 'USP14',\n",
       " 'USP15',\n",
       " 'USP16',\n",
       " 'USP18',\n",
       " 'USP19',\n",
       " 'USP20',\n",
       " 'USP21',\n",
       " 'USP22',\n",
       " 'USP24',\n",
       " 'USP25',\n",
       " 'USP28',\n",
       " 'USP3',\n",
       " 'USP30',\n",
       " 'USP33',\n",
       " 'USP34',\n",
       " 'USP37',\n",
       " 'USP4',\n",
       " 'USP42',\n",
       " 'USP47',\n",
       " 'USP48',\n",
       " 'USP5',\n",
       " 'USP8',\n",
       " 'USP9X',\n",
       " 'VCAM1',\n",
       " 'VCP',\n",
       " 'VCPIP1',\n",
       " 'VDAC1',\n",
       " 'VDAC2',\n",
       " 'VDAC3',\n",
       " 'VEGFA',\n",
       " 'VIM',\n",
       " 'VPS45',\n",
       " 'WDR20',\n",
       " 'WDR48',\n",
       " 'WEE1',\n",
       " 'WRAP53',\n",
       " 'WRN',\n",
       " 'XRN2',\n",
       " 'YOD1',\n",
       " 'YWHAB',\n",
       " 'YWHAE',\n",
       " 'YWHAG',\n",
       " 'YWHAH',\n",
       " 'YWHAQ',\n",
       " 'YWHAZ',\n",
       " 'ZEB1',\n",
       " 'ZFHX3',\n",
       " 'ZFPM1',\n",
       " 'ZNF385A',\n",
       " 'ZNF420',\n",
       " 'ZRANB1']"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interacting_proteins = set(interacting_proteins)\n",
    "print(len(interacting_proteins))\n",
    "genes_measured = set(genes_measured)\n",
    "if target_gene in new_genes: new_genes.remove(target_gene)\n",
    "print(len(genes_measured))\n",
    "print(len(new_genes))\n",
    "new_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "52b4340d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a function that will analyze the protein cis effects for a given mutation type\n",
    "def analyze_trans_mutation(omics_type, mutation_type, gene_list, write_out = True, pval_only = True):\n",
    "    \"\"\"Analyzes the trans effects for a given mutation type across all cancers.\n",
    "    \n",
    "    Parameters:\n",
    "    mutation_type (str): The mutation type to analyze, as a string. Possible values given in mutation_types.\n",
    "    omics_type (str): The omics to analyze. Possible values are in list [\"prot\", \"trans\", \"phos\"]\n",
    "    gene_list (list[str]): A list of genes with proteins in the same pathway as those from target_gene\n",
    "    write_out (bool, optional): Whether to write out the data to a separate file.\n",
    "    pval_only (bool, optional): If write_out, only writes out the p_value for each analysis. If not write_out, has no effect.\n",
    "    \n",
    "    Returns a list of the outputs of scipy.ttest_ind() between \"Wildtype\" and mutation_type tumors, for each of the genes. If analyzing phosphoproteomics, returns a list of lists, in the form [[(site_name, pval) for each site] for each entry]\n",
    "    \"\"\"\n",
    "    if write_out:\n",
    "        # Clears the target file\n",
    "        file_name = omics_key[omics_type].capitalize() + '_' + mutation_type.capitalize() + 's.txt'\n",
    "        with open(\"Trans_Effect_output/\" + file_name, 'w') as out_file: pass\n",
    "    for name, dataset in datasets.items():\n",
    "        if mutation_type not in variants_to_analyze[name]: continue\n",
    "        \n",
    "        # Separate the wildtype and mutated samples\n",
    "        genotype_and_omic = pd.merge(dataset[\"gen\"], dataset[omics_type], left_index = True, right_index = True)\n",
    "        mutation = genotype_and_omic.loc[genotype_and_omic[\"Mutation\"] == mutation_type].dropna()\n",
    "        wildtype = genotype_and_omic.loc[genotype_and_omic[\"Mutation\"] == \"Wildtype\"].dropna()\n",
    "        \n",
    "        # Phosphoproteomics must be analyzed at each site in the gene\n",
    "        if omics_type != \"phos\":\n",
    "            result = [ttest_ind(mutation[trans_gene].dropna(), wildtype[trans_gene].dropna())[1] for trans_gene in gene_list]\n",
    "        else:\n",
    "            result = [(site, ttest_ind(mutation[trans_gene][site].dropna(), wildtype[trans_gene][site])[1]) for site in trans_gene for trans_gene in mutation.columns if site in wildtype[trans_gene].columns]\n",
    "                \n",
    "        if write_out:\n",
    "            output_string = \"\"\n",
    "            if not pval_only:\n",
    "                output_string += f\"{name}\\n\"\n",
    "                output_string += f\"Number of {mutation_type}s = {len(mutation)}\\n\"\n",
    "                output_string += f\"Number of Wildtypes = {len(wildtype)}\\n\"\n",
    "            if omics_type != \"phos\":\n",
    "                output_string += f\"{name} {mutation_type} P-value = {result}\\n\\n\"\n",
    "            else:\n",
    "                output_string += f\"{name} {mutation_type} P-values:\\n\"\n",
    "                for site in result:\n",
    "                    output_string += f\"   {site[0]}: {site[1]}\\n\"\n",
    "                ouput_string += '\\n'\n",
    "            with open(\"Trans_Effect_output/\" + file_name, \"a\") as out_file:\n",
    "                print(output_string)\n",
    "                out_file.write(\"output_string\") \n",
    "                \n",
    "    try:\n",
    "        return result\n",
    "    except UnboundLocalError:\n",
    "        return\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "0fe9ee79",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'Analysis_output/Proteomics_Deletions.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_15401/2998773360.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0momics_type\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m\"prot\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"trans\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"phos\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mmutation_type\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mmutation_types\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m         \u001b[0manalyze_cis_mutation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutation_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwrite_out\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpval_only\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/tmp/ipykernel_15401/1793494427.py\u001b[0m in \u001b[0;36manalyze_cis_mutation\u001b[0;34m(omics_type, mutation_type, write_out, pval_only)\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0;31m# Clears the target file\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m         \u001b[0mfile_name\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0momics_key\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0momics_type\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcapitalize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m'_'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mmutation_type\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcapitalize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m's.txt'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m         \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Analysis_output/\"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mfile_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mout_file\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;32mpass\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     17\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdataset\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdatasets\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmutation_type\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mvariants_to_analyze\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'Analysis_output/Proteomics_Deletions.txt'"
     ]
    }
   ],
   "source": [
    "#Now, run the analysis for every mutation_type, for every omics_type\n",
    "for omics_type in [\"prot\", \"trans\", \"phos\"]:\n",
    "    for mutation_type in mutation_types:\n",
    "        analyze_trans_mutation(omics_type, mutation_type, new_genes, write_out = True, pval_only = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44c4801b",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"deletion_trans_data.txt\", 'w') as outFile:\n",
    "    all_significant = {}\n",
    "    for name, dataset in datasets.items():\n",
    "        if name not in deletion_sets: continue\n",
    "        print(name + '\\n')\n",
    "        outFile.write(name + '\\n')\n",
    "        significant_changes = {}\n",
    "        for trans_gene in interacting_proteins:\n",
    "            genotype_and_prot = dataset[0].merge(dataset[1], left_index = True, right_index = True)\n",
    "            if trans_gene not in genotype_and_prot: continue\n",
    "            deletions = genotype_and_prot.loc[genotype_and_prot[\"Mutation\"] == \"Deletion\"][trans_gene].dropna()\n",
    "            wildtypes = genotype_and_prot.loc[genotype_and_prot[\"Mutation\"] == \"Wildtype_Tumor\"][trans_gene].dropna()\n",
    "            result = ttest_ind(deletions, wildtypes)\n",
    "            print(trans_gene + \" P-value with \" + name + \" deletion: \" + str(result[1]))\n",
    "            outFile.write(name + \" deletion P-value: \" + str(result[1]))\n",
    "        print('\\n')\n",
    "        outFile.write('\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9fd7f56",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
