{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8a220f11",
   "metadata": {},
   "source": [
    "# Part 1: Create database set\n",
    "By inputting a target gene, a nested dictionary will be created, with each key being the name of a cancer and each value being a dictionary containing the genomic, proteomic, transcriptomic, and phosphoproteomic data. In other words, the dictionary will be:\n",
    "\n",
    "{\n",
    "    cancer : {   \n",
    "    \n",
    "            \"gen\" : <cancer's genomic data>,\n",
    "            \"prot\" : <cancer's proteomic data>,\n",
    "            \"trans\": <cancer's transcriptomic data>,\n",
    "            \"phos\": <cancer's phosphoproteomic data>\n",
    "        }\n",
    "    ... for each cancer\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "dede19fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import cptac.pancan as pc\n",
    "import numpy as np\n",
    "import cptac.utils as ut\n",
    "import plot_utils as plt_ut\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "930371b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4159d5ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PancanBrca',\n",
       " 'PancanCcrcc',\n",
       " 'PancanCoad',\n",
       " 'PancanGbm',\n",
       " 'PancanHnscc',\n",
       " 'PancanLscc',\n",
       " 'PancanLuad',\n",
       " 'PancanOv',\n",
       " 'PancanUcec',\n",
       " 'PancanPdac']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset_names = pc.list_datasets(False)\n",
    "dataset_names  = dataset_names.split('\\n')\n",
    "dataset_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c45e7a87",
   "metadata": {},
   "outputs": [],
   "source": [
    "target_gene = \"TP53\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6aac2474",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TP53</td>\n",
       "      <td>MAPK Signaling Pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>DNA Damage Response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Wnt Signaling Pathway and Pluripotency</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Pathways Affected in Adenoid Cystic Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Retinoblastoma Gene in Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Mammary gland development pathway - Involution...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TP53</td>\n",
       "      <td>miRNA regulation of p53 pathway in prostate ca...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TP53</td>\n",
       "      <td>miRNA Regulation of DNA Damage Response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Ultraconserved region 339 modulation of tumor ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Gastric Cancer Network 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TP53 Network</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Chromosomal and microsatellite instability in ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Circadian rhythm related genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Breast cancer pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TP53</td>\n",
       "      <td>RAC1/PAK1/p38/MMP2 Pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Apoptosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Hepatitis C and Hepatocellular Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TP53</td>\n",
       "      <td>The effect of progerin on the involved genes i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TP53</td>\n",
       "      <td>G1 to S cell cycle control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Fluoropyrimidine Activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>TP53</td>\n",
       "      <td>PI3K-Akt Signaling Pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Melanoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Pancreatic adenocarcinoma pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Spinal Cord Injury</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Apoptosis Modulation and Signaling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Metastatic brain tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>TP53</td>\n",
       "      <td>TGF-beta Signaling Pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>TP53</td>\n",
       "      <td>LncRNA-mediated mechanisms of therapeutic resi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Head and Neck Squamous Cell Carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TP53</td>\n",
       "      <td>DNA IR-damage and cellular response via ATR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Non-small cell lung cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TP53</td>\n",
       "      <td>MicroRNA network associated with chronic lymph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Ferroptosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Signaling Pathways in Glioblastoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Alzheimers Disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Bladder Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Photodynamic therapy-induced HIF-1 survival si...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>TP53</td>\n",
       "      <td>IL-18 signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Epithelial to mesenchymal transition in colore...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Endometrial cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>TP53</td>\n",
       "      <td>miRNAs involved in DNA damage response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>TP53</td>\n",
       "      <td>DNA IR-Double Strand Breaks (DSBs) and cellula...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Senescence and Autophagy in Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Copper homeostasis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Amyotrophic lateral sclerosis (ALS)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ATM Signaling Pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Netrin-UNC5B signaling Pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Small cell lung cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>TP53</td>\n",
       "      <td>DNA Damage Response (only ATM dependent)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      id                                            pathway\n",
       "0   TP53                             MAPK Signaling Pathway\n",
       "1   TP53                                DNA Damage Response\n",
       "2   TP53             Wnt Signaling Pathway and Pluripotency\n",
       "3   TP53      Pathways Affected in Adenoid Cystic Carcinoma\n",
       "4   TP53                      Retinoblastoma Gene in Cancer\n",
       "5   TP53  Mammary gland development pathway - Involution...\n",
       "6   TP53  miRNA regulation of p53 pathway in prostate ca...\n",
       "7   TP53            miRNA Regulation of DNA Damage Response\n",
       "8   TP53  Ultraconserved region 339 modulation of tumor ...\n",
       "9   TP53                           Gastric Cancer Network 2\n",
       "10  TP53                                       TP53 Network\n",
       "11  TP53  Chromosomal and microsatellite instability in ...\n",
       "12  TP53                     Circadian rhythm related genes\n",
       "13  TP53                              Breast cancer pathway\n",
       "14  TP53                         RAC1/PAK1/p38/MMP2 Pathway\n",
       "15  TP53                                          Apoptosis\n",
       "16  TP53           Hepatitis C and Hepatocellular Carcinoma\n",
       "17  TP53  The effect of progerin on the involved genes i...\n",
       "18  TP53                         G1 to S cell cycle control\n",
       "19  TP53                          Fluoropyrimidine Activity\n",
       "20  TP53                         PI3K-Akt Signaling Pathway\n",
       "21  TP53                                           Melanoma\n",
       "22  TP53                  Pancreatic adenocarcinoma pathway\n",
       "23  TP53                                 Spinal Cord Injury\n",
       "24  TP53                 Apoptosis Modulation and Signaling\n",
       "25  TP53                             Metastatic brain tumor\n",
       "26  TP53                         TGF-beta Signaling Pathway\n",
       "27  TP53  LncRNA-mediated mechanisms of therapeutic resi...\n",
       "28  TP53              Head and Neck Squamous Cell Carcinoma\n",
       "29  TP53        DNA IR-damage and cellular response via ATR\n",
       "30  TP53                         Non-small cell lung cancer\n",
       "31  TP53  MicroRNA network associated with chronic lymph...\n",
       "32  TP53                                        Ferroptosis\n",
       "33  TP53                 Signaling Pathways in Glioblastoma\n",
       "34  TP53                                 Alzheimers Disease\n",
       "35  TP53                                     Bladder Cancer\n",
       "36  TP53  Photodynamic therapy-induced HIF-1 survival si...\n",
       "37  TP53                            IL-18 signaling pathway\n",
       "38  TP53  Epithelial to mesenchymal transition in colore...\n",
       "39  TP53                                 Endometrial cancer\n",
       "40  TP53             miRNAs involved in DNA damage response\n",
       "41  TP53  DNA IR-Double Strand Breaks (DSBs) and cellula...\n",
       "42  TP53                 Senescence and Autophagy in Cancer\n",
       "43  TP53                                 Copper homeostasis\n",
       "44  TP53                Amyotrophic lateral sclerosis (ALS)\n",
       "45  TP53                              ATM Signaling Pathway\n",
       "46  TP53                     Netrin-UNC5B signaling Pathway\n",
       "47  TP53                             Small cell lung cancer\n",
       "48  TP53           DNA Damage Response (only ATM dependent)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of pathways to select for the trans analysis\n",
    "paths = ut.get_pathways_with_proteins(proteins = target_gene, database = 'wikipathways')\n",
    "paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0d72985d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Indicate the NUMERICAL INDICES of the pathways of interest\n",
    "path_indices=[0,1,2,10,18]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "61771c6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import multiprocessing\n",
    "# pool_size = multiprocessing.cpu_count()\n",
    "# pool = multiprocessing.Pool(pool_size)\n",
    "\n",
    "# pool.map(load_data, dataset_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f73069a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading broadlscc v1.0.          gbm v1.0.Loading bcmlscc v1.0.Loading bcmov v1.0..Loading bcmhnscc v1.0...Loading bcmlscc v1.0...Loading bcmgbm v1.0..\r"
     ]
    }
   ],
   "source": [
    "import concurrent.futures\n",
    "\n",
    "def load_data(name):\n",
    "    exec(f\"cancer = pc.{name}(no_internet = True)\")  \n",
    "    genotype_data = cancer.get_genotype_all_vars(target_gene, omics_source = \"washu\")\n",
    "    \n",
    "    proteomic_data = cancer.get_proteomics(source = 'umich')\n",
    "    proteomic_data = proteomic_data.droplevel(\"Database_ID\", axis = 1)\n",
    "    \n",
    "    phospho_data = cancer.get_phosphoproteomics(source = 'umich')\n",
    "    phospho_data = phospho_data.droplevel(\"Database_ID\", axis = 1).droplevel(\"Peptide\", axis = 1)\n",
    "    \n",
    "    transcript_data = cancer.get_transcriptomics(source = \"washu\")\n",
    "    transcript_data = transcript_data.droplevel(\"Database_ID\", axis = 1)\n",
    "    \n",
    "    return (name[6:], {\n",
    "                        \"gen\" : genotype_data,\n",
    "                        \"prot\": proteomic_data,\n",
    "                        \"trans\": transcript_data,\n",
    "                        \"phos\" : phospho_data\n",
    "                    })\n",
    "\n",
    "datasets = {}\n",
    "if __name__ == '__main__':\n",
    "    with concurrent.futures.ProcessPoolExecutor() as executor:\n",
    "        results = executor.map(load_data, dataset_names)\n",
    "        for result in results:\n",
    "            print(f\"appending {result[0]}\")\n",
    "            datasets[result[0]] = datasets[result[1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36f26ae4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# datasets = {}\n",
    "# for name in dataset_names:\n",
    "#     exec(f\"cancer = pc.{name}(no_internet = True)\")  \n",
    "#     genotype_data = cancer.get_genotype_all_vars(target_gene, omics_source = \"washu\")\n",
    "    \n",
    "#     proteomic_data = cancer.get_proteomics(source = 'umich')\n",
    "#     proteomic_data = proteomic_data.droplevel(\"Database_ID\", axis = 1)\n",
    "    \n",
    "#     phospho_data = cancer.get_phosphoproteomics(source = 'umich')\n",
    "#     phospho_data = phospho_data.droplevel(\"Database_ID\", axis = 1).droplevel(\"Peptide\", axis = 1)\n",
    "    \n",
    "#     transcript_data = cancer.get_transcriptomics(source = \"washu\")\n",
    "#     transcript_data = transcript_data.droplevel(\"Database_ID\", axis = 1)\n",
    "    \n",
    "#     datasets[name[6:]] = {\n",
    "#                         \"gen\" : genotype_data,\n",
    "#                         \"prot\": proteomic_data,\n",
    "#                         \"trans\": transcript_data,\n",
    "#                         \"phos\" : phospho_data\n",
    "#                          }    \n",
    "#     print(f\"Loaded {name}...\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fd85c29",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DELETE THIS CELL\n",
    "#I use this to let me know when the data has finished loading haha\n",
    "import webbrowser\n",
    "webbrowser.open(\"https://www.youtube.com/watch?v=dQw4w9WgXcQ\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a54fff79",
   "metadata": {},
   "source": [
    "# Part 2: Select mutations\n",
    "Determine which cancers have mutation types with rates above 20%.\n",
    "\n",
    "To facilitate that, the mutation types from the get_genotype_all_vars must be grouped into functional types.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b6178983",
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_types = [\"Deletion\", \"Amplification\", \"Truncation\", \"Indel\", \"Missense\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bfa5779",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Grouping the mutation types for each cancer\n",
    "for name, dataset in datasets.items():\n",
    "    genotypes = dataset[\"gen\"]\n",
    "    genotypes.replace([\"Nonsense_Mutation\", \"Frame_Shift_Ins\", \"Frame_Shift_Del\"], \"Truncation\", inplace = True)\n",
    "    genotypes.replace([\"In_Frame_Del\", \"In_Frame_Ins\"], \"Indel\", inplace = True)\n",
    "    genotypes.replace([\"Missense_Mutation\", \"Wildtype_Tumor\"], [\"Missense\", \"Wildtype\"], inplace = True)\n",
    "    genotypes = genotypes.loc[~genotypes[\"Mutation\"].isin([\"Intron\",\"Silent\"])]\n",
    "    datasets[name][\"gen\"] = genotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9ddba048",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a bargraph displaying the mutation types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e8e0509",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Determine which cancers have mutation rates over 20%\n",
    "variants_to_analyze = {}\n",
    "for name, dataset in datasets.items():\n",
    "    mut_type_freq = dataset[\"gen\"][\"Mutation\"].value_counts() / len(dataset[\"gen\"])\n",
    "    \n",
    "    mut_type_freq = mut_type_freq.loc[mut_type_freq >= 0.2]\n",
    "    variants_to_analyze[name] = set(mut_type_freq.index.tolist())\n",
    "    variants_to_analyze[name].discard(\"Wildtype\")\n",
    "print(str(variants_to_analyze))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0a75f2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plt.figure1_plot_mutations(datasets.values(), list(datasets.keys()), save_to_path = \"Test_Figure.png\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "368f9dd3",
   "metadata": {},
   "source": [
    "# Part 3: Cis effects\n",
    "Determines which genes that, given a mutation, will change the expression in their own proteomics, transcriptomics, or phosphoproteomics.This analysis is repeated for every cancer, and every mutation type.\n",
    "\n",
    "Writes out a file containing the infomation for each mutation and cancer, and creates boxplot summarizing the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "588253ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import ttest_ind\n",
    "omics_key = {\"prot\":\"proteomics\", \"trans\":\"transcriptomics\",\"phos\":\"phosphoproteomics\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b4a3f8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Parameters\n",
    "pval_symbol: String. The symbol associated with the significance of the pvalue (NS, *, **, or ***)\n",
    "x1, x2: Float. x coordinates for the start and end of the annotation line.\n",
    "line_start: Float. y coordinate for the horizontal line (Default = .05).\n",
    "line_height: Float. The height of the vertical lines at the start and end of the horizontal line (Default = .05).\n",
    "\n",
    "Returns a formated annotation line showing the significance symbol of each pvalue.\n",
    "'''\n",
    "def format_pval_annotation(pval_symbol, x1, x2, line_start = .05, line_height=.05):\n",
    "    # for manual adjustment to pval annotations\n",
    "    \n",
    "    y, h = line_start, line_height\n",
    "    plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "             [y, y+h, y+h, y], #vertical line\n",
    "             lw=1.5, color= '.3')\n",
    "    plt.text((x1+x2)*.5, # half between x coord\n",
    "             y+h, pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf4312c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_cis_plot(boxplot_data, omics_type, pval_df):\n",
    "    plt.rcParams['figure.figsize'] = (15,10) # size of plot\n",
    "#     if len(gene_list) > 15:\n",
    "#         plt.rcParams['figure.figsize']=(20,10)\n",
    "    sns.set(font_scale = 1.2, style = \"white\")\n",
    "\n",
    "    for site in boxplot_data[\"Site\"].unique():\n",
    "        mutations = sorted(boxplot_data[\"Mutation\"].unique(), reverse = True)\n",
    "        boxplot = sns.boxplot(x = \"cancer_type\", y = \"P_Value\", hue = \"Mutation\", showfliers = False, \n",
    "                              data = boxplot_data[boxplot_data[\"Site\"] == site], hue_order = mutations)\n",
    "        boxplot = sns.stripplot(x = \"cancer_type\", y = \"P_Value\", hue = \"Mutation\", jitter = True, hue_order = mutations,\n",
    "                               color = \".3\", dodge = True, data = boxplot_data[boxplot_data[\"Site\"] == site])\n",
    "#         gene_site = site if site == target_gene else target_gene+'.'+site\n",
    "        gene_site = site\n",
    "        boxplot.set(xlabel = \"\", ylabel = f\"{gene_site}_{omics_key[omics_type]}\")\n",
    "        boxplot.set(title = f\"Pancan Analysis of {gene_site} {omics_key[omics_type].capitalize()} Mutations\")\n",
    "#         if len(gene_list) > 15:\n",
    "#             boxplot.set_xticklabels(boxplot.get_xticklabels(),rotation=50)\n",
    "        \n",
    "        num = len(boxplot_data[\"Mutation\"].unique())\n",
    "        handles, labels = boxplot.get_legend_handles_labels()\n",
    "        legend = plt.legend(handles[0:num], labels[0:num], loc = \"lower right\")\n",
    "        \n",
    "        ## CREATE PVAL ANNOTATIONS\n",
    "        symbols = {}\n",
    "        start = -0.3\n",
    "        cancer_list = boxplot.get_xticklabels() # get a list of cancer types actually plotted \n",
    "        first_mutation = mutations[1]\n",
    "        bottom,top = plt.ylim()\n",
    "        annot_size = top + 0.5\n",
    "        annot_scale = (top - bottom) * 0.05 # sets annotation height between mutations based on y-axis scale\n",
    "        for cancer in cancer_list:\n",
    "            cancer = cancer.get_text()\n",
    "            mutation_list = sorted(boxplot_data.loc[boxplot_data[\"cancer_type\"] == cancer][\"Mutation\"].unique().tolist(),\n",
    "                                   reverse = True)\n",
    "            mutation_list.remove(\"Wildtype\")\n",
    "            offset = 0.4\n",
    "            annot_offset = annot_scale\n",
    "            for mutation in mutation_list:\n",
    "                pval = pval_df.loc[(pval_df[\"Cancer\"] == cancer)&(pval_df[\"Mutation\"] == mutation)&\n",
    "                                   (pval_df[\"Site\"] == site)]['P_Value']\n",
    "                if pval.tolist():\n",
    "                    pval = pval.tolist()[0]\n",
    "                    if pval  <= 0.001:\n",
    "                         symbols[cancer] = '***'\n",
    "                    elif pval  <= 0.01:\n",
    "                        symbols[cancer] = '**'  \n",
    "                    elif pval <= 0.05:\n",
    "                        symbols[cancer] = '*'\n",
    "                    else:\n",
    "                        symbols[cancer] = 'ns'\n",
    "                    if mutation == first_mutation:\n",
    "                        format_pval_annotation(symbols[cancer], start, start + 0.2, annot_size)\n",
    "                    else:\n",
    "                        format_pval_annotation(symbols[cancer], start, start + offset, annot_size + annot_offset)\n",
    "                        offset += 0.2\n",
    "                        annot_offset += annot_scale\n",
    "            start += 1\n",
    "        plt.show()\n",
    "        plt.clf()\n",
    "        plt.close()\n",
    "            \n",
    "#         fig = boxplot.get_figure()\n",
    "#         fig.savefig(f\"Cis_Effect_output/Test_{gene_site}_{omics_key[omics_type]}.png\")\n",
    "#         fig.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8e97355",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a function that will analyze the protein cis effects for a given mutation type\n",
    "def analyze_cis_mutation(omics_type, genes_to_analyze = [target_gene], write_out = True, file_name = None, output_boxplot = True, write_insignificant=False):\n",
    "    \"\"\"Analyzes the cis effects for a given mutation type across all cancers.\n",
    "    \n",
    "    Parameters:\n",
    "    omics_type (str): The omics to analyze. Possible values are in list [\"prot\", \"trans\", \"phos\"]\n",
    "    genes_to_analyze (list): A list of strings representing genes to analyze. For cis effects, this is always [target_gene]\n",
    "    write_out (bool, default=True): Whether to write out the data to a separate file.\n",
    "    output_boxplot (bool, default=True): Outputs the data as a boxplot given by the generate_cis_plot function\n",
    "    write_insignificant (bool, default=True): Returns data only when the p-value <= 0.05. If write_out, it only writes out the data under that condition.\n",
    "    \n",
    "    Returns a dataframe containing output of scipy.ttest_ind() between \"Wildtype\" and mutation_type tumors.\n",
    "    \"\"\"\n",
    "    if write_out:# Sets file name and clears target files\n",
    "        import os\n",
    "        if file_name is None:\n",
    "            effect_type = \"Cis\" if genes_to_analyze == [target_gene] else \"Trans\"\n",
    "            file_name = f\"{effect_type}_Effect_output/{omics_key[omics_type].capitalize()}_Analysis.txt\" \n",
    "            if '/' in file_name:\n",
    "                directory = file_name[:file_name.rfind('/')]\n",
    "                if not os.path.exists(directory):\n",
    "                    os.makedirs(directory)\n",
    "        with open(file_name, 'w') as out_file: pass\n",
    "                \n",
    "                \n",
    "    for cancer_type, dataset in datasets.items():\n",
    "        ## PREPARE DATA ##\n",
    "        genotype_and_omic = dataset[omics_type].copy()\n",
    "        if isinstance(genotype_and_omic.columns, pd.MultiIndex):\n",
    "            genotype_and_omic.columns = ['_'.join(col) for col in genotype_and_omic.columns.values]\n",
    "        if genes_to_analyze is not None:\n",
    "            genotype_and_omic = genotype_and_omic.filter(regex='(' + '|'.join(genes_to_analyze) + ')(_.*)?$', axis=1)\n",
    "        else:\n",
    "            genotype_and_omic.drop(target_gene, axis = 1, inplace=True)\n",
    "        genotype_and_omic[\"Mutation\"] = dataset[\"gen\"][\"Mutation\"].copy()\n",
    "        genotype_and_omic = genotype_and_omic.loc[:,~genotype_and_omic.columns.duplicated()]\n",
    "        \n",
    "        for mutation_type in mutation_types:\n",
    "            if mutation_type not in variants_to_analyze[cancer_type]: \n",
    "                continue\n",
    "            if write_out:\n",
    "                with open(file_name, 'a') as out_file:\n",
    "                    out_file.write(f\"\\n\\n{cancer_type.upper()} {mutation_type.upper()}: \")\n",
    "                    \n",
    "            print(f\"   {cancer_type} {mutation_type}s\") #debug temp\n",
    "            target_mut_data = genotype_and_omic.loc[genotype_and_omic[\"Mutation\"].isin([\"Wildtype\", mutation_type])]\n",
    "            ## ANALYZE P-VALUES ##\n",
    "            # DIFF b/w trans effects: loops through each column to avoid correction\n",
    "            for site in target_mut_data.drop(\"Mutation\", axis=1).columns:\n",
    "                mutation_analysis = ut.wrap_ttest(target_mut_data, label_column=\"Mutation\",\n",
    "                                    correction_method = \"fdr_bh\", return_all=True)\n",
    "                mutation_analysis.rename({'Comparison':'Site'}, axis = 1, inplace = True)\n",
    "                mutation_analysis['Cancer'], mutation_analysis['Mutation'] = cancer_type, mutation_type\n",
    "            \n",
    "                try:\n",
    "                    result = pd.concat([result, mutation_analysis])\n",
    "                except NameError:\n",
    "                    result = mutation_analysis\n",
    "            #END DIFF\n",
    "            \n",
    "                ## GATHER DATA FOR BOXPLOT ##\n",
    "                if output_boxplot:\n",
    "                    new_data = genotype_and_omic[(genotype_and_omic[\"Mutation\"] == mutation_type)|\n",
    "                                                (genotype_and_omic[\"Mutation\"] == \"Wildtype\")]\n",
    "                    new_data[\"Site\"], new_data[\"cancer_type\"] = site, cancer_type\n",
    "                    new_data.rename({site:\"P_Value\"}, axis=1, inplace=True)\n",
    "                    try:\n",
    "                        boxplot_data = pd.concat([boxplot_data, new_data], axis=0)\n",
    "                    except NameError:\n",
    "                        boxplot_data = new_data\n",
    "\n",
    "            ## WRITE OUT RESULT ##\n",
    "            if write_out:\n",
    "                with open(file_name, 'a') as out_file:\n",
    "                    out_file.write(f\"{len(result.loc[result['P_Value'] <= 0.05])} Significant Entries\\n\")\n",
    "                    if not write_insignificant:\n",
    "                        result_display = result.loc[result['P_Value'] <= 0.05]\n",
    "\n",
    "                    for row in result_display.iterrows():\n",
    "                        data = row[1]\n",
    "                        out_file.write(f\"   {data['Site']}:\".ljust(15) + f\"{data['P_Value']}\\n\")\n",
    "\n",
    "    ## CREATE BOXPLOT ##\n",
    "    if output_boxplot:\n",
    "        generate_cis_plot(boxplot_data, omics_type, result) \n",
    "\n",
    "    result.reset_index(drop=True, inplace=True)\n",
    "    result[\"Omics\"] = omics_type\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a28f9e0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "cis_effects = pd.concat([analyze_cis_mutation(omics_type, genes_to_analyze=[target_gene]) for omics_type in [\"prot\", \"trans\", \"phos\"]])\n",
    "cis_effects.reset_index(drop=True, inplace=True)\n",
    "cis_effects\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65dd39af",
   "metadata": {},
   "source": [
    "## TEMP NOTE\n",
    "* Get the plots working\n",
    "* Do a pathway enrichment for both restricted and unrestricted\n",
    "* Find other analyses\n",
    "\n",
    "### Problems\n",
    "* When wrap ttest is passed the phospho data, it performs a correction on the phospho data. Should the data in general get a correction?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "73617b45",
   "metadata": {},
   "source": [
    "# Part 4: Trans Effects\n",
    "Determines which genes that, given a mutation, will change the expression in another protein's proteomics, transcriptomics, or phosphoproteomics.This analysis is repeated for every cancer, and every mutation type.\n",
    "\n",
    "To reduce the computational intensity and the corrections needed, the scope will be restricted to genes that share specific pathways with the target gene. These pathways are hypothesized to be significant in the gene's function, and **must be specificed beforehand**, either at the beginning of this notebook or in the following cells.\n",
    "\n",
    "The analysis will then be repeated for all genes in the dataset. As the number of comparisons is greater, the correction applied will be more substantial.\n",
    "\n",
    "Writes out a file containing the infomation for each mutation and cancer, and creates boxplot summarizing the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "402cf454",
   "metadata": {},
   "outputs": [],
   "source": [
    "if path_indices == []:\n",
    "    print(paths)\n",
    "    path_indices = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1daf485",
   "metadata": {},
   "outputs": [],
   "source": [
    "interacting_proteins = ut.get_proteins_in_pathways(paths[\"pathway\"].loc[path_indices], database = 'wikipathways')\n",
    "interacting_proteins = set(interacting_proteins[\"member\"])\n",
    "\n",
    "# genes_measured = [gene for dataset in datasets.values() for gene in dataset[\"prot\"].columns]\n",
    "genes_measured = set([gene for gene in dataset[\"prot\"].columns for dataset in datasets.values()])\n",
    "genes_to_analyze = sorted([protein for protein in interacting_proteins if protein in genes_measured])\n",
    "no_analysis = sorted([protein for protein in interacting_proteins if protein not in genes_measured])# debug temp\n",
    "\n",
    "if target_gene in genes_to_analyze:\n",
    "    genes_to_analyze.remove(target_gene)\n",
    "print(f\"Interacting genes: {len(interacting_proteins)}\")\n",
    "print(f\"Not measured in cptac: {len(no_analysis)}\") #debug temp\n",
    "print(f\"Genes to analyze: {len(genes_to_analyze)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8dd51f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a function that will analyze the protein cis effects for a given mutation type\n",
    "def analyze_trans_mutation(omics_type, genes_to_analyze = None, write_out = True, file_name = None, output_boxplot = True, write_insignificant=False):\n",
    "    \"\"\"Analyzes the cis effects for a given mutation type across all cancers.\n",
    "    \n",
    "    Parameters:\n",
    "    omics_type (str): The omics to analyze. Possible values are in list [\"prot\", \"trans\", \"phos\"]\n",
    "    genes_to_analyze (list, default=None): The list of genes to consider for trans effects. Defaults to all genes in dataset, excluding target_gene\n",
    "    write_out (bool, default=True): Whether to write out the data to a separate file.\n",
    "    output_boxplot (bool, default=True): Outputs the data as a boxplot given by the generate_trans_plot function\n",
    "    write_insignificant (bool, default=True): Returns data only when the p-value <= 0.05. If write_out, it only writes out the data under that condition.\n",
    "    \n",
    "    Returns the output of scipy.ttest_ind() between \"Wildtype\" and mutation_type tumors. If analyzing phosphoproteomics, returns a list of such outputs\n",
    "    \"\"\"\n",
    "    if write_out:# Sets file name and clears target files\n",
    "        import os\n",
    "        if file_name is None:\n",
    "            effect_type = \"Cis\" if genes_to_analyze == [target_gene] else \"Trans\"\n",
    "            file_name = f\"{effect_type}_Effect_output/{omics_key[omics_type].capitalize()}_Analysis_trans.txt\"\n",
    "        if '/' in file_name:\n",
    "            directory = file_name[:file_name.rfind('/')]\n",
    "            if not os.path.exists(directory):\n",
    "                os.makedirs(directory)\n",
    "        with open(file_name, 'w') as out_file: pass\n",
    "                \n",
    "                \n",
    "    for cancer_type, dataset in datasets.items():\n",
    "        ## PREPARE DATA ##\n",
    "        genotype_and_omic = dataset[omics_type].copy()\n",
    "        if isinstance(genotype_and_omic.columns, pd.MultiIndex):\n",
    "            genotype_and_omic.columns = ['_'.join(col) for col in genotype_and_omic.columns.values]\n",
    "        if genes_to_analyze is not None:\n",
    "            genotype_and_omic = genotype_and_omic.filter(regex='(' + '|'.join(genes_to_analyze) + ')(_.*)?$', axis=1)\n",
    "        else:\n",
    "            genotype_and_omic.drop(target_gene, axis = 1, inplace=True)\n",
    "        genotype_and_omic[\"Mutation\"] = dataset[\"gen\"][\"Mutation\"].copy()\n",
    "        genotype_and_omic = genotype_and_omic.loc[:,~genotype_and_omic.columns.duplicated()]\n",
    "        \n",
    "        for mutation_type in mutation_types:\n",
    "            if mutation_type not in variants_to_analyze[cancer_type]: \n",
    "                continue\n",
    "            if write_out:\n",
    "                with open(file_name, 'a') as out_file:\n",
    "                    out_file.write(f\"\\n\\n{cancer_type.upper()} {mutation_type.upper()}: \")\n",
    "            print(f\"   {cancer_type} {mutation_type}s\") #debug temp\n",
    "\n",
    "            target_mut_data = genotype_and_omic.loc[genotype_and_omic[\"Mutation\"].isin([\"Wildtype\", mutation_type])]\n",
    "   \n",
    "            ## ANALYZE P-VALUES ##\n",
    "            mutation_analysis = ut.wrap_ttest(target_mut_data, label_column=\"Mutation\",\n",
    "                                correction_method = \"fdr_bh\", return_all=True)\n",
    "            mutation_analysis.rename({'Comparison':'Site'}, axis = 1, inplace = True)\n",
    "            mutation_analysis['Cancer'], mutation_analysis['Mutation'] = cancer_type, mutation_type\n",
    "            \n",
    "            try:\n",
    "                result = pd.concat([result, mutation_analysis])\n",
    "            except NameError:\n",
    "                result = mutation_analysis\n",
    "\n",
    "            ## GATHER DATA FOR BOXPLOT ##\n",
    "            if output_boxplot:\n",
    "                new_data = genotype_and_omic[(genotype_and_omic[\"Mutation\"] == mutation_type)|\n",
    "                                            (genotype_and_omic[\"Mutation\"] == \"Wildtype\")]\n",
    "                new_data[\"cancer_type\"] = cancer_type\n",
    "                try:\n",
    "                    boxplot_data = pd.concat([boxplot_data, new_data], axis=0)\n",
    "                except NameError:\n",
    "                    boxplot_data = new_data\n",
    "\n",
    "            ## WRITE OUT RESULT ##\n",
    "            if write_out:\n",
    "                with open(file_name, 'a') as out_file:\n",
    "                    out_file.write(f\"{len(result.loc[result['P_Value'] <= 0.05])} Significant Entries\\n\")\n",
    "                    if not write_insignificant:\n",
    "                        result = result.loc[result['P_Value'] <= 0.05]\n",
    "                    \n",
    "                    for row in result.iterrows():\n",
    "                        data = row[1]\n",
    "                        out_file.write(f\"   {data['Site']}:\".ljust(15) + f\"{data['P_Value']}\\n\")\n",
    "\n",
    "    ## CREATE BOXPLOT ##\n",
    "#     if output_boxplot:\n",
    "#         pval_df = pd.DataFrame(pval_dict)\n",
    "#         generate_cis_plot(boxplot_data, omics_type, pval_df) \n",
    "        \n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78608115",
   "metadata": {},
   "outputs": [],
   "source": [
    "trans_effects = None\n",
    "for omics_type in [\"prot\", \"trans\", \"phos\"]:\n",
    "    print(omics_type.upper() + ':')\n",
    "    data = analyze_trans_mutation(omics_type, genes_to_analyze=genes_to_analyze, write_insignificant=True)\n",
    "    data[\"Omics\"] = omics_type\n",
    "    if trans_effects is not None:\n",
    "        trans_effects = pd.concat([trans_effects, data])\n",
    "    else:\n",
    "        trans_effects = data\n",
    "    print()\n",
    "trans_effects.reset_index(drop=True, inplace=True)\n",
    "trans_effects"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b737522d",
   "metadata": {},
   "source": [
    "#Part 5: Conclusion\n",
    "The following "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4767df0c",
   "metadata": {},
   "source": [
    "# Part 5: Discussion\n",
    "Looks for ptaterns in the analysis that will indicate meaningful conclusions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c64d79a2",
   "metadata": {},
   "source": [
    "### 5.1: Pathway Enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ccd0a51",
   "metadata": {},
   "outputs": [],
   "source": [
    "from gprofiler import GProfiler\n",
    "cancer = \"Brca\"\n",
    "mutation = \"Deletion\"\n",
    "\n",
    "up_in_mutation = trans_effects.loc[(trans_effects[\"Cancer\"]==cancer)&\n",
    "                                   (trans_effects[\"Mutation\"]==mutation)&\n",
    "                                   (trans_effects[\"Omics\"] == \"prot\")][\"Site\"]\n",
    "up_in_mutation = list(up_in_mutation)\n",
    "up_in_mutation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd066950",
   "metadata": {},
   "outputs": [],
   "source": [
    "GProfiler(return_dataframe=True).profile(organism='hsapiens', query=up_in_mutation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57a7d587",
   "metadata": {},
   "outputs": [],
   "source": [
    "# enrichments = {cancer + '_' + mutation : set(GProfiler(return_dataframe=True).profile(organism='hsapiens', query=up_in_mutation)[\"name\"])\n",
    "#                 for cancer in trans_effects[\"Cancer\"]\n",
    "#                 for mutation in trans_effects[\"Mutation\"]}\n",
    "# enrichments\n",
    "#Taking a long time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4436372",
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichments = {}\n",
    "omics_effects = trans_effects.loc[trans_effects[\"Omics\"] == \"prot\"]\n",
    "for cancer in omics_effects[\"Cancer\"].unique():\n",
    "    cancer_effects = omics_effects.loc[omics_effects[\"Cancer\"] == cancer]\n",
    "    for mutation in cancer_effects[\"Mutation\"].unique():\n",
    "        print(cancer+'_'+mutation)\n",
    "        query = list(trans_effects.loc[(trans_effects[\"Mutation\"]==mutation)][\"Site\"])\n",
    "        enrichments[cancer+'_'+mutation] = GProfiler(return_dataframe=True).profile(organism='hsapiens', query=query)\n",
    "        enrichments[cancer+'_'+mutation]\n",
    "        break\n",
    "    break\n",
    "enrichments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ad58218",
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichments[\"Brca_Deletion\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "380829a1",
   "metadata": {},
   "source": [
    "### TEMP\n",
    "This is junk that will be deleted soon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "53f15dff",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "425a5771",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def generate_cis_plot(boxplot_data, omics_type, pval_df):\n",
    "#     plt.rcParams['figure.figsize'] = (15,10) # size of plot\n",
    "# #     if len(gene_list) > 15:\n",
    "# #         plt.rcParams['figure.figsize']=(20,10)\n",
    "#     sns.set(font_scale = 1.2, style = \"white\")\n",
    "\n",
    "#     for site in boxplot_data[\"site\"].unique():\n",
    "#         mutations = sorted(boxplot_data[\"Mutation\"].unique(), reverse = True)\n",
    "#         boxplot = sns.boxplot(x = \"cancer_type\", y = site, hue = \"Mutation\", showfliers = False, \n",
    "#                               data = boxplot_data[boxplot_data[\"site\"] == site], hue_order = mutations)\n",
    "#         boxplot = sns.stripplot(x = \"cancer_type\", y = site, hue = \"Mutation\", jitter = True, hue_order = mutations,\n",
    "#                                color = \".3\", dodge = True, data = boxplot_data[boxplot_data[\"site\"] == site])\n",
    "#         gene_site = site if site == target_gene else target_gene+'.'+site\n",
    "#         boxplot.set(xlabel = \"\", ylabel = f\"{gene_site}_{omics_key[omics_type]}\")\n",
    "#         boxplot.set(title = f\"Pancan Analysis of {gene_site} {omics_key[omics_type].capitalize()} Mutations\")\n",
    "# #         if len(gene_list) > 15:\n",
    "# #             boxplot.set_xticklabels(boxplot.get_xticklabels(),rotation=50)\n",
    "        \n",
    "#         num = len(boxplot_data[\"Mutation\"].unique())\n",
    "#         handles, labels = boxplot.get_legend_handles_labels()\n",
    "#         legend = plt.legend(handles[0:num], labels[0:num], loc = \"lower right\")\n",
    "        \n",
    "#         ## CREATE PVAL ANNOTATIONS\n",
    "#         symbols = {}\n",
    "#         start = -0.3\n",
    "#         cancer_list = boxplot.get_xticklabels() # get a list of cancer types actually plotted \n",
    "#         first_mutation = mutations[1]\n",
    "#         bottom,top = plt.ylim()\n",
    "#         annot_size = top + 0.5\n",
    "#         annot_scale = (top - bottom) * 0.05 # sets annotation height between mutations based on y-axis scale\n",
    "#         for cancer in cancer_list:\n",
    "#             cancer = cancer.get_text()\n",
    "#             mutation_list = sorted(boxplot_data.loc[boxplot_data[\"cancer_type\"] == cancer][\"Mutation\"].unique().tolist(),\n",
    "#                                    reverse = True)\n",
    "#             mutation_list.remove(\"Wildtype\")\n",
    "#             offset = 0.4\n",
    "#             annot_offset = annot_scale\n",
    "#             for mutation in mutation_list:\n",
    "#                 pval = pval_df.loc[(pval_df[\"Cancer\"] == cancer)&(pval_df[\"Mutation\"] == mutation)&\n",
    "#                                    (pval_df[\"Site\"] == site)]['P_Value']\n",
    "#                 if pval.tolist():\n",
    "#                     pval = pval.tolist()[0]\n",
    "#                     if pval  <= 0.001:\n",
    "#                          symbols[cancer] = '***'\n",
    "#                     elif pval  <= 0.01:\n",
    "#                         symbols[cancer] = '**'  \n",
    "#                     elif pval <= 0.05:\n",
    "#                         symbols[cancer] = '*'\n",
    "#                     else:\n",
    "#                         symbols[cancer] = 'ns'\n",
    "#                     if mutation == first_mutation:\n",
    "#                         format_pval_annotation(symbols[cancer], start, start + 0.2, annot_size)\n",
    "#                     else:\n",
    "#                         format_pval_annotation(symbols[cancer], start, start + offset, annot_size + annot_offset)\n",
    "#                         offset += 0.2\n",
    "#                         annot_offset += annot_scale\n",
    "#             start += 1\n",
    "        \n",
    "#         plt.show()\n",
    "#         plt.clf()\n",
    "#         plt.close()\n",
    "            \n",
    "# #         fig = boxplot.get_figure()\n",
    "# #         fig.savefig(f\"Cis_Effect_output/Test_{gene_site}_{omics_key[omics_type]}.png\")\n",
    "# #         fig.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0aa454a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Create a function that will analyze the protein cis effects for a given mutation type\n",
    "# def analyze_cis_mutation(omics_type, write_out = True, file_name = None, output_boxplot = True, write_insignificant=False):\n",
    "#     \"\"\"Analyzes the cis effects for a given mutation type across all cancers.\n",
    "    \n",
    "#     Parameters:\n",
    "#     omics_type (str): The omics to analyze. Possible values are in list [\"prot\", \"trans\", \"phos\"]\n",
    "#     write_out (bool, optional): Whether to write out the data to a separate file.\n",
    "    \n",
    "#     Returns the output of scipy.ttest_ind() between \"Wildtype\" and mutation_type tumors. If analyzing phosphoproteomics, returns a list of such outputs\n",
    "#     \"\"\"\n",
    "#     if write_out:# Sets file name and clears target files\n",
    "#         import os\n",
    "#         if file_name is None:\n",
    "#             file_name = f\"Cis_Effect_output/{omics_key[omics_type].capitalize()}_Analysis.txt\"\n",
    "#         if '/' in file_name:\n",
    "#             directory = file_name[:file_name.rfind('/')]\n",
    "#             if not os.path.exists(directory):\n",
    "#                 os.makedirs(directory)\n",
    "#         with open(file_name, 'w') as out_file: pass\n",
    "                \n",
    "#     result = []\n",
    "#     pval_dict = {'Cancer':[], 'Mutation':[], 'Site':[], 'P_Value':[]}\n",
    "#     boxplot_data = []\n",
    "#     for cancer_type, dataset in datasets.items():\n",
    "#         if write_out:\n",
    "#             with open(file_name, 'a') as out_file:\n",
    "#                 out_file.write(cancer_type.upper()+'\\n')\n",
    "            \n",
    "#         for mutation_type in mutation_types:\n",
    "#             ## PREPARE DATA ##\n",
    "#             if mutation_type not in variants_to_analyze[cancer_type]: \n",
    "#                 continue\n",
    "#             print(f\"{cancer_type} {mutation_type}s\") #debug temp\n",
    "#             genotype_and_omic = pd.merge(dataset[\"gen\"][\"Mutation\"], dataset[omics_type][target_gene], left_index = True,\n",
    "#                                          right_index = True)\n",
    "#             genotype_and_omic = genotype_and_omic.loc[genotype_and_omic[\"Mutation\"].isin([\"Wildtype\", mutation_type])]\n",
    "                            \n",
    "#             for site in genotype_and_omic.drop(\"Mutation\", axis = 1).columns:\n",
    "#                 ## ANALYZE P-VALUES ##\n",
    "#                 try:\n",
    "#                     pval = ut.wrap_ttest(genotype_and_omic, label_column=\"Mutation\", comparison_columns=[site],\n",
    "#                                          correction_method = \"fdr_bh\", return_all=True).iloc[0,1]\n",
    "#                 except:\n",
    "#                     continue\n",
    "#                 pval_dict['Cancer'].append(cancer_type)\n",
    "#                 pval_dict['Mutation'].append(mutation_type)\n",
    "#                 pval_dict['Site'].append(site)\n",
    "#                 pval_dict['P_Value'].append(pval)\n",
    "                \n",
    "#                 result.append ({\n",
    "#                     \"omics_type\":omics_type,\n",
    "#                     \"site\":site,\n",
    "#                     \"mutation_type\":mutation_type,\n",
    "#                     \"cancer_type\":cancer_type,\n",
    "#                     \"pval\":pval\n",
    "#                 })\n",
    "                    \n",
    "#                 ## GATHER DATA FOR BOXPLOT ##\n",
    "#                 if output_boxplot:\n",
    "#                     new_data = genotype_and_omic[(genotype_and_omic[\"Mutation\"] == mutation_type)|\n",
    "#                                                 (genotype_and_omic[\"Mutation\"] == \"Wildtype\")]\n",
    "#                     new_data[\"site\"], new_data[\"cancer_type\"] = site, cancer_type\n",
    "#                     boxplot_data.append(pd.DataFrame(new_data))\n",
    "                \n",
    "#                 ## WRITE OUT RESULT ##\n",
    "#                 if write_out:\n",
    "#                     with open(file_name, 'a') as out_file: \n",
    "#                         out_file.write(f\"   {mutation_type} p-val: {pval}\\n\")\n",
    "    \n",
    "#     ## CREATE BOXPLOT ##\n",
    "#     if output_boxplot:\n",
    "#         pval_df = pd.DataFrame(pval_dict)\n",
    "#         boxplot_data = pd.concat(boxplot_data)\n",
    "#         generate_cis_plot(boxplot_data, omics_type, pval_df) \n",
    "        \n",
    "#     result = pd.DataFrame(result)\n",
    "#     return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4d42f57",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "cis_effects = pd.concat([analyze_cis_mutation(omics_type) for omics_type in [\"prot\", \"trans\", \"phos\"]])\n",
    "cis_effects.reset_index(drop=True, inplace=True)\n",
    "cis_effects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bed7aed3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
