{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make EGFR Figure 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebooks takes all the trans genes that have positive and negative results and runs a GSEA using Reactome. It then takes a subset of genes from the top hit(Hemostasis) pathway and maps them on a large circle heat map. This heatmap focuses on coagulation and urokinase related genes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/Lindsey/anaconda3/lib/python3.7/site-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas.util.testing as tm\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "import statsmodels.stats.multitest\n",
    "\n",
    "import gseapy as gp\n",
    "from gseapy.plot import barplot, dotplot\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n",
    "import plot_utils as p "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 1: Find Trans proteins with opposite effects in different cancers "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load df with all of the genes that are FDR significant."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation_Gbm</th>\n",
       "      <th>P_value_Gbm</th>\n",
       "      <th>Correlation_kidney</th>\n",
       "      <th>P_value_kidney</th>\n",
       "      <th>Correlation_Ovar</th>\n",
       "      <th>P_value_Ovar</th>\n",
       "      <th>Correlation_Brca</th>\n",
       "      <th>P_value_Brca</th>\n",
       "      <th>Correlation_Luad</th>\n",
       "      <th>P_value_Luad</th>\n",
       "      <th>Correlation_hnscc</th>\n",
       "      <th>P_value_hnscc</th>\n",
       "      <th>Correlation_Lscc</th>\n",
       "      <th>P_value_Lscc</th>\n",
       "      <th>Correlation_Colon</th>\n",
       "      <th>P_value_Colon</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Comparison</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PHLDA1</th>\n",
       "      <td>0.816848</td>\n",
       "      <td>3.507071e-21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.364797</td>\n",
       "      <td>0.002164</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.664271</td>\n",
       "      <td>8.888640e-12</td>\n",
       "      <td>0.713420</td>\n",
       "      <td>2.644826e-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2</th>\n",
       "      <td>-0.610889</td>\n",
       "      <td>6.729990e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.302439</td>\n",
       "      <td>0.020631</td>\n",
       "      <td>-0.532341</td>\n",
       "      <td>3.320092e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDH4</th>\n",
       "      <td>0.559180</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PLA2G15</th>\n",
       "      <td>-0.556624</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>-0.298029</td>\n",
       "      <td>0.02216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.274185</td>\n",
       "      <td>0.016061</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOCS2</th>\n",
       "      <td>0.562720</td>\n",
       "      <td>3.420388e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.472624</td>\n",
       "      <td>1.417921e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.287735</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SSR2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.301814</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSMB1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.287839</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL39</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.288110</td>\n",
       "      <td>0.049589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAP3K11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.498761</td>\n",
       "      <td>0.049785</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6230 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Correlation_Gbm   P_value_Gbm  Correlation_kidney  P_value_kidney  \\\n",
       "Comparison                                                                      \n",
       "PHLDA1             0.816848  3.507071e-21                 NaN             NaN   \n",
       "GRB2              -0.610889  6.729990e-08                 NaN             NaN   \n",
       "CDH4               0.559180  3.420388e-06                 NaN             NaN   \n",
       "PLA2G15           -0.556624  3.420388e-06           -0.298029         0.02216   \n",
       "SOCS2              0.562720  3.420388e-06                 NaN             NaN   \n",
       "...                     ...           ...                 ...             ...   \n",
       "CCDC6                   NaN           NaN                 NaN             NaN   \n",
       "SSR2                    NaN           NaN                 NaN             NaN   \n",
       "PSMB1                   NaN           NaN                 NaN             NaN   \n",
       "MRPL39                  NaN           NaN                 NaN             NaN   \n",
       "MAP3K11                 NaN           NaN                 NaN             NaN   \n",
       "\n",
       "            Correlation_Ovar  P_value_Ovar  Correlation_Brca  P_value_Brca  \\\n",
       "Comparison                                                                   \n",
       "PHLDA1                   NaN           NaN          0.364797      0.002164   \n",
       "GRB2                     NaN           NaN               NaN           NaN   \n",
       "CDH4                     NaN           NaN               NaN           NaN   \n",
       "PLA2G15                  NaN           NaN          0.274185      0.016061   \n",
       "SOCS2                    NaN           NaN               NaN           NaN   \n",
       "...                      ...           ...               ...           ...   \n",
       "CCDC6                    NaN           NaN               NaN           NaN   \n",
       "SSR2                     NaN           NaN               NaN           NaN   \n",
       "PSMB1                    NaN           NaN               NaN           NaN   \n",
       "MRPL39                   NaN           NaN               NaN           NaN   \n",
       "MAP3K11                  NaN           NaN               NaN           NaN   \n",
       "\n",
       "            Correlation_Luad  P_value_Luad  Correlation_hnscc  P_value_hnscc  \\\n",
       "Comparison                                                                     \n",
       "PHLDA1                   NaN           NaN           0.664271   8.888640e-12   \n",
       "GRB2               -0.302439      0.020631          -0.532341   3.320092e-06   \n",
       "CDH4                     NaN           NaN                NaN            NaN   \n",
       "PLA2G15                  NaN           NaN                NaN            NaN   \n",
       "SOCS2                    NaN           NaN                NaN            NaN   \n",
       "...                      ...           ...                ...            ...   \n",
       "CCDC6                    NaN           NaN                NaN            NaN   \n",
       "SSR2                     NaN           NaN                NaN            NaN   \n",
       "PSMB1                    NaN           NaN                NaN            NaN   \n",
       "MRPL39                   NaN           NaN                NaN            NaN   \n",
       "MAP3K11                  NaN           NaN                NaN            NaN   \n",
       "\n",
       "            Correlation_Lscc  P_value_Lscc  Correlation_Colon  P_value_Colon  \n",
       "Comparison                                                                    \n",
       "PHLDA1              0.713420  2.644826e-14                NaN            NaN  \n",
       "GRB2                     NaN           NaN                NaN            NaN  \n",
       "CDH4                     NaN           NaN                NaN            NaN  \n",
       "PLA2G15                  NaN           NaN                NaN            NaN  \n",
       "SOCS2               0.472624  1.417921e-02                NaN            NaN  \n",
       "...                      ...           ...                ...            ...  \n",
       "CCDC6                    NaN           NaN           0.287735       0.049589  \n",
       "SSR2                     NaN           NaN           0.301814       0.049589  \n",
       "PSMB1                    NaN           NaN           0.287839       0.049589  \n",
       "MRPL39                   NaN           NaN          -0.288110       0.049589  \n",
       "MAP3K11                  NaN           NaN           0.498761       0.049785  \n",
       "\n",
       "[6230 rows x 16 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FDR_sig = pd.read_csv(\"Pval_corr_table_Fig_2_return_sig.csv\")\n",
    "FDR_sig = FDR_sig.replace(to_replace ='_proteomics', value = '', regex = True) #Remove proteomics labeling\n",
    "FDR_sig= FDR_sig.drop(['Unnamed: 0'], axis=1)\n",
    "FDR_sig = FDR_sig.set_index(\"Comparison\")\n",
    "FDR_sig = FDR_sig[1:] #Drop EGFR \n",
    "FDR_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def HasPosNeg(row):\n",
    "    hasPos = False\n",
    "    hasNeg= False\n",
    "\n",
    "    for item in row:\n",
    "        if pd.isnull(item):\n",
    "            continue\n",
    "        if item < 0:\n",
    "            hasNeg = True\n",
    "        if item > 0:\n",
    "            hasPos = True\n",
    "            \n",
    "    if hasPos & hasNeg:\n",
    "        return True\n",
    "    return False\n",
    "\n",
    "def CountPosNeg(row):\n",
    "    hasPos = False\n",
    "    hasNeg= False\n",
    "    counter = 0\n",
    "    for item in row:\n",
    "        if pd.isnull(item):\n",
    "            continue\n",
    "        if item < -0:\n",
    "            hasNeg = True\n",
    "            counter += 1\n",
    "        if item > 0:\n",
    "            hasPos = True\n",
    "            counter += 1\n",
    "    return counter\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Subset data frame to include only trans genes that have opposite effects in different cancers by using apply function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/Lindsey/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    }
   ],
   "source": [
    "col = [\"Correlation_Gbm\",\"Correlation_kidney\",\"Correlation_Ovar\",\"Correlation_Brca\",\"Correlation_Luad\",\"Correlation_hnscc\",\"Correlation_Lscc\",\"Correlation_Colon\"]\n",
    "FDR_corr = FDR_sig[col]\n",
    "FDR_corr[\"Pos_Neg\"] = FDR_corr.apply(HasPosNeg, axis = 1)\n",
    "FDR_corr_False = FDR_corr[FDR_corr['Pos_Neg']==False]\n",
    "FDR_corr_False = FDR_corr_False.drop([\"Pos_Neg\"], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Correlation_Gbm</th>\n",
       "      <th>Correlation_kidney</th>\n",
       "      <th>Correlation_Ovar</th>\n",
       "      <th>Correlation_Brca</th>\n",
       "      <th>Correlation_Luad</th>\n",
       "      <th>Correlation_hnscc</th>\n",
       "      <th>Correlation_Lscc</th>\n",
       "      <th>Correlation_Colon</th>\n",
       "      <th>Num_corr</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Comparison</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PHLDA1</th>\n",
       "      <td>0.816848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.364797</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.664271</td>\n",
       "      <td>0.713420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2</th>\n",
       "      <td>-0.610889</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.302439</td>\n",
       "      <td>-0.532341</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOCS2</th>\n",
       "      <td>0.562720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.472624</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLA</th>\n",
       "      <td>-0.550491</td>\n",
       "      <td>-0.298348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHLDA3</th>\n",
       "      <td>0.525883</td>\n",
       "      <td>0.432530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.763784</td>\n",
       "      <td>0.682034</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S100A14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.370227</td>\n",
       "      <td>0.351373</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MOV10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.302787</td>\n",
       "      <td>0.299218</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP2B1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.298873</td>\n",
       "      <td>0.346409</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MFSD6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.295201</td>\n",
       "      <td>0.380169</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SART1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.292189</td>\n",
       "      <td>-0.306539</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1136 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Correlation_Gbm  Correlation_kidney  Correlation_Ovar  \\\n",
       "Comparison                                                          \n",
       "PHLDA1             0.816848                 NaN               NaN   \n",
       "GRB2              -0.610889                 NaN               NaN   \n",
       "SOCS2              0.562720                 NaN               NaN   \n",
       "GLA               -0.550491           -0.298348               NaN   \n",
       "PHLDA3             0.525883            0.432530               NaN   \n",
       "...                     ...                 ...               ...   \n",
       "S100A14                 NaN                 NaN               NaN   \n",
       "MOV10                   NaN                 NaN               NaN   \n",
       "ATP2B1                  NaN                 NaN               NaN   \n",
       "MFSD6                   NaN                 NaN               NaN   \n",
       "SART1                   NaN                 NaN               NaN   \n",
       "\n",
       "            Correlation_Brca  Correlation_Luad  Correlation_hnscc  \\\n",
       "Comparison                                                          \n",
       "PHLDA1              0.364797               NaN           0.664271   \n",
       "GRB2                     NaN         -0.302439          -0.532341   \n",
       "SOCS2                    NaN               NaN                NaN   \n",
       "GLA                      NaN               NaN                NaN   \n",
       "PHLDA3                   NaN               NaN           0.763784   \n",
       "...                      ...               ...                ...   \n",
       "S100A14                  NaN               NaN                NaN   \n",
       "MOV10                    NaN               NaN                NaN   \n",
       "ATP2B1                   NaN               NaN                NaN   \n",
       "MFSD6                    NaN               NaN                NaN   \n",
       "SART1                    NaN               NaN                NaN   \n",
       "\n",
       "            Correlation_Lscc  Correlation_Colon  Num_corr  \n",
       "Comparison                                                 \n",
       "PHLDA1              0.713420                NaN         4  \n",
       "GRB2                     NaN                NaN         3  \n",
       "SOCS2               0.472624                NaN         2  \n",
       "GLA                      NaN                NaN         2  \n",
       "PHLDA3              0.682034                NaN         4  \n",
       "...                      ...                ...       ...  \n",
       "S100A14             0.370227           0.351373         2  \n",
       "MOV10               0.302787           0.299218         2  \n",
       "ATP2B1              0.298873           0.346409         2  \n",
       "MFSD6               0.295201           0.380169         2  \n",
       "SART1              -0.292189          -0.306539         2  \n",
       "\n",
       "[1136 rows x 9 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FDR_corr_False[\"Num_corr\"] = FDR_corr_False.apply(CountPosNeg, axis = 1)\n",
    "FDR_corr_False.sort_values(by=['Num_corr'],ascending=False)\n",
    "FDR_corr_False = FDR_corr_False[FDR_corr_False['Num_corr'].isin([2,3,4,5,6])]\n",
    "FDR_corr_False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The manuscript mentions the total number of trans proteins that opposite effects in different cancers. Here is the derivation of that number "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of trans proteins with same effects in different cancers is 1136\n"
     ]
    }
   ],
   "source": [
    "same_sign_prot = FDR_corr_False.index.tolist()\n",
    "print(\"Total number of trans proteins with same effects in different cancers is \" + str(len(same_sign_prot)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run GSEA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_set</th>\n",
       "      <th>Term</th>\n",
       "      <th>Overlap</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Adjusted P-value</th>\n",
       "      <th>Old P-value</th>\n",
       "      <th>Old Adjusted P-value</th>\n",
       "      <th>Odds Ratio</th>\n",
       "      <th>Combined Score</th>\n",
       "      <th>Genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>Arrhythmogenic right ventricular cardiomyopath...</td>\n",
       "      <td>18/74</td>\n",
       "      <td>1.159284e-07</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.282451</td>\n",
       "      <td>68.392004</td>\n",
       "      <td>DSP;ITGA4;JUP;ITGB4;ACTN2;ITGA2;GJA1;CDH2;PKP2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>Metabolic pathways Homo sapiens hsa01100</td>\n",
       "      <td>114/1239</td>\n",
       "      <td>1.813751e-07</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.619889</td>\n",
       "      <td>25.145046</td>\n",
       "      <td>MSMO1;ENO1;NSDHL;NAMPT;ENPP1;ENPP3;ATP6V1E1;PT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>Regulation of actin cytoskeleton Homo sapiens ...</td>\n",
       "      <td>32/214</td>\n",
       "      <td>5.044867e-07</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.632618</td>\n",
       "      <td>38.172238</td>\n",
       "      <td>CYFIP2;NCKAP1;ITGB4;WAS;PIK3CD;IQGAP2;ITGAL;FG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>Biosynthesis of amino acids Homo sapiens hsa01230</td>\n",
       "      <td>17/74</td>\n",
       "      <td>6.249417e-07</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.044537</td>\n",
       "      <td>57.778675</td>\n",
       "      <td>PRPS1;ARG1;PGAM1;GPT;ENO1;PFKL;PKM;CBS;PSAT1;P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>Glycolysis / Gluconeogenesis Homo sapiens hsa0...</td>\n",
       "      <td>16/67</td>\n",
       "      <td>7.613663e-07</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.204330</td>\n",
       "      <td>59.231243</td>\n",
       "      <td>GPI;ACSS2;ADH1B;PGAM1;ENO1;HK2;LDHA;PFKL;PKM;P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>Bacterial invasion of epithelial cells Homo sa...</td>\n",
       "      <td>17/78</td>\n",
       "      <td>1.377343e-06</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.837125</td>\n",
       "      <td>51.783367</td>\n",
       "      <td>WAS;FN1;PIK3CD;SEPT1;PIK3CG;SEPT2;CD2AP;SEPT9;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>B cell receptor signaling pathway Homo sapiens...</td>\n",
       "      <td>16/73</td>\n",
       "      <td>2.588556e-06</td>\n",
       "      <td>0.000108</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.858769</td>\n",
       "      <td>49.640789</td>\n",
       "      <td>CR2;JUN;SYK;INPPL1;PIK3CD;NFATC2;PIK3CG;VAV1;P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>Carbon metabolism Homo sapiens hsa01200</td>\n",
       "      <td>20/113</td>\n",
       "      <td>5.248171e-06</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.116041</td>\n",
       "      <td>37.883681</td>\n",
       "      <td>GPI;PRPS1;H6PD;ACSS2;ECHS1;PGAM1;GPT;ENO1;HK2;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>T cell receptor signaling pathway Homo sapiens...</td>\n",
       "      <td>19/104</td>\n",
       "      <td>5.547202e-06</td>\n",
       "      <td>0.000181</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.216414</td>\n",
       "      <td>38.925738</td>\n",
       "      <td>JUN;NFATC2;PIK3CD;CD3G;PIK3CG;VAV1;ZAP70;CD4;P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KEGG_2016</td>\n",
       "      <td>Axon guidance Homo sapiens hsa04360</td>\n",
       "      <td>21/127</td>\n",
       "      <td>9.304350e-06</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.911168</td>\n",
       "      <td>33.725962</td>\n",
       "      <td>EPHA4;SEMA4A;SEMA4D;SEMA4B;LIMK1;GNAI3;NFATC2;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene_set                                               Term   Overlap  \\\n",
       "0  KEGG_2016  Arrhythmogenic right ventricular cardiomyopath...     18/74   \n",
       "1  KEGG_2016           Metabolic pathways Homo sapiens hsa01100  114/1239   \n",
       "2  KEGG_2016  Regulation of actin cytoskeleton Homo sapiens ...    32/214   \n",
       "3  KEGG_2016  Biosynthesis of amino acids Homo sapiens hsa01230     17/74   \n",
       "4  KEGG_2016  Glycolysis / Gluconeogenesis Homo sapiens hsa0...     16/67   \n",
       "5  KEGG_2016  Bacterial invasion of epithelial cells Homo sa...     17/78   \n",
       "6  KEGG_2016  B cell receptor signaling pathway Homo sapiens...     16/73   \n",
       "7  KEGG_2016            Carbon metabolism Homo sapiens hsa01200    20/113   \n",
       "8  KEGG_2016  T cell receptor signaling pathway Homo sapiens...    19/104   \n",
       "9  KEGG_2016                Axon guidance Homo sapiens hsa04360    21/127   \n",
       "\n",
       "        P-value  Adjusted P-value  Old P-value  Old Adjusted P-value  \\\n",
       "0  1.159284e-07          0.000034            0                     0   \n",
       "1  1.813751e-07          0.000027            0                     0   \n",
       "2  5.044867e-07          0.000049            0                     0   \n",
       "3  6.249417e-07          0.000046            0                     0   \n",
       "4  7.613663e-07          0.000045            0                     0   \n",
       "5  1.377343e-06          0.000067            0                     0   \n",
       "6  2.588556e-06          0.000108            0                     0   \n",
       "7  5.248171e-06          0.000192            0                     0   \n",
       "8  5.547202e-06          0.000181            0                     0   \n",
       "9  9.304350e-06          0.000273            0                     0   \n",
       "\n",
       "   Odds Ratio  Combined Score  \\\n",
       "0    4.282451       68.392004   \n",
       "1    1.619889       25.145046   \n",
       "2    2.632618       38.172238   \n",
       "3    4.044537       57.778675   \n",
       "4    4.204330       59.231243   \n",
       "5    3.837125       51.783367   \n",
       "6    3.858769       49.640789   \n",
       "7    3.116041       37.883681   \n",
       "8    3.216414       38.925738   \n",
       "9    2.911168       33.725962   \n",
       "\n",
       "                                               Genes  \n",
       "0  DSP;ITGA4;JUP;ITGB4;ACTN2;ITGA2;GJA1;CDH2;PKP2...  \n",
       "1  MSMO1;ENO1;NSDHL;NAMPT;ENPP1;ENPP3;ATP6V1E1;PT...  \n",
       "2  CYFIP2;NCKAP1;ITGB4;WAS;PIK3CD;IQGAP2;ITGAL;FG...  \n",
       "3  PRPS1;ARG1;PGAM1;GPT;ENO1;PFKL;PKM;CBS;PSAT1;P...  \n",
       "4  GPI;ACSS2;ADH1B;PGAM1;ENO1;HK2;LDHA;PFKL;PKM;P...  \n",
       "5  WAS;FN1;PIK3CD;SEPT1;PIK3CG;SEPT2;CD2AP;SEPT9;...  \n",
       "6  CR2;JUN;SYK;INPPL1;PIK3CD;NFATC2;PIK3CG;VAV1;P...  \n",
       "7  GPI;PRPS1;H6PD;ACSS2;ECHS1;PGAM1;GPT;ENO1;HK2;...  \n",
       "8  JUN;NFATC2;PIK3CD;CD3G;PIK3CG;VAV1;ZAP70;CD4;P...  \n",
       "9  EPHA4;SEMA4A;SEMA4D;SEMA4B;LIMK1;GNAI3;NFATC2;...  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "same_sign_enr = gp.enrichr(gene_list = same_sign_prot, description='Tumor_partition', gene_sets='KEGG_2016', \n",
    "                       outdir='test/enrichr_KEGG')\n",
    "same_sign_enr.res2d.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PRKCQ',\n",
       " 'PPP3CC',\n",
       " 'LCP2',\n",
       " 'INPP5D',\n",
       " 'PIK3CD',\n",
       " 'GRAP2',\n",
       " 'CD4',\n",
       " 'PTPN6',\n",
       " 'BTK',\n",
       " 'NFATC2',\n",
       " 'PIK3CG',\n",
       " 'SYK',\n",
       " 'PTPRC',\n",
       " 'ZAP70',\n",
       " 'VAV1',\n",
       " 'RAC2',\n",
       " 'LCK',\n",
       " 'GRB2',\n",
       " 'CD3G',\n",
       " 'CD5',\n",
       " 'CD8A',\n",
       " 'BLK',\n",
       " 'BLNK',\n",
       " 'CD79A',\n",
       " 'LYN',\n",
       " 'CD3E',\n",
       " 'HLA-DMA',\n",
       " 'HLA-DMB']"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get just the clotting cascade genes and add urokinase genes \n",
    "same_sign_df = same_sign_enr.res2d\n",
    "bcell = same_sign_df.iloc[6,9]\n",
    "bcell = bcell.split(';')\n",
    "tcell = same_sign_df.iloc[8,9]\n",
    "tcell = tcell.split(\";\")\n",
    "immune_both =  bcell + tcell\n",
    "immune = set(immune_both)\n",
    "immune = list(immune)\n",
    "btcell_apm = ([\"CD5\",\"CD8A\",\"BLK\",\"BLNK\",\"CD79A\",\"LYN\",\"CD3E\",\"HLA-DMA\",\"HLA-DMB\"])\n",
    "\n",
    "immune= immune + btcell_apm\n",
    "\n",
    "#remove = [\"CR2\",\"NCK1\",\"CARD11\",\"INPPL1\",\"JUN\",\"KRAS\"]\n",
    "immune.remove(\"JUN\")\n",
    "immune.remove(\"KRAS\")\n",
    "immune.remove(\"CR2\")\n",
    "immune.remove(\"NCK1\")\n",
    "immune.remove(\"INPPL1\")\n",
    "immune.remove(\"CARD11\")\n",
    "immune"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 3 Make Data frame for figure 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Correlation</th>\n",
       "      <th>P_value</th>\n",
       "      <th>Cancer Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>-0.610889</td>\n",
       "      <td>6.729990e-08</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>SYK</td>\n",
       "      <td>-0.470849</td>\n",
       "      <td>1.202510e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>CD4</td>\n",
       "      <td>-0.467479</td>\n",
       "      <td>1.362923e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>PIK3CD</td>\n",
       "      <td>-0.466265</td>\n",
       "      <td>1.399080e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>LCP2</td>\n",
       "      <td>-0.460881</td>\n",
       "      <td>1.665085e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>VAV1</td>\n",
       "      <td>-0.447143</td>\n",
       "      <td>2.894593e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>-0.442005</td>\n",
       "      <td>3.476225e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>PTPRC</td>\n",
       "      <td>-0.424958</td>\n",
       "      <td>6.196952e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>BTK</td>\n",
       "      <td>-0.423380</td>\n",
       "      <td>6.552047e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>INPP5D</td>\n",
       "      <td>-0.408181</td>\n",
       "      <td>1.009137e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>PTPN6</td>\n",
       "      <td>-0.402017</td>\n",
       "      <td>1.241946e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>-0.394048</td>\n",
       "      <td>1.602237e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>NFATC2</td>\n",
       "      <td>-0.392602</td>\n",
       "      <td>1.681448e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>603</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>-0.373678</td>\n",
       "      <td>5.056470e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>605</th>\n",
       "      <td>BLNK</td>\n",
       "      <td>-0.356906</td>\n",
       "      <td>5.065055e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>941</th>\n",
       "      <td>RAC2</td>\n",
       "      <td>-0.321178</td>\n",
       "      <td>1.349871e-02</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>PRKCQ</td>\n",
       "      <td>-0.329766</td>\n",
       "      <td>1.621074e-02</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2270</th>\n",
       "      <td>GRAP2</td>\n",
       "      <td>-0.305290</td>\n",
       "      <td>2.127686e-02</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2400</th>\n",
       "      <td>VAV1</td>\n",
       "      <td>-0.285693</td>\n",
       "      <td>3.025415e-02</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2492</th>\n",
       "      <td>ZAP70</td>\n",
       "      <td>-0.276087</td>\n",
       "      <td>3.833890e-02</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2550</th>\n",
       "      <td>INPP5D</td>\n",
       "      <td>-0.271891</td>\n",
       "      <td>4.182668e-02</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5750</th>\n",
       "      <td>PPP3CC</td>\n",
       "      <td>-0.432271</td>\n",
       "      <td>3.644683e-04</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5773</th>\n",
       "      <td>PRKCQ</td>\n",
       "      <td>-0.418175</td>\n",
       "      <td>6.284144e-04</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5851</th>\n",
       "      <td>ZAP70</td>\n",
       "      <td>-0.384613</td>\n",
       "      <td>2.092847e-03</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5893</th>\n",
       "      <td>PTPN6</td>\n",
       "      <td>-0.371990</td>\n",
       "      <td>3.159187e-03</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5897</th>\n",
       "      <td>LCK</td>\n",
       "      <td>-0.370894</td>\n",
       "      <td>3.271948e-03</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6002</th>\n",
       "      <td>RAC2</td>\n",
       "      <td>-0.349937</td>\n",
       "      <td>5.935190e-03</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6094</th>\n",
       "      <td>BLK</td>\n",
       "      <td>-0.365512</td>\n",
       "      <td>8.766800e-03</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6121</th>\n",
       "      <td>PIK3CD</td>\n",
       "      <td>-0.331760</td>\n",
       "      <td>9.676902e-03</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6364</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>-0.302439</td>\n",
       "      <td>2.063096e-02</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6480</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>-0.292478</td>\n",
       "      <td>2.581604e-02</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6666</th>\n",
       "      <td>BTK</td>\n",
       "      <td>-0.278003</td>\n",
       "      <td>3.552087e-02</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6762</th>\n",
       "      <td>INPP5D</td>\n",
       "      <td>-0.271933</td>\n",
       "      <td>4.007023e-02</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6777</th>\n",
       "      <td>CD3G</td>\n",
       "      <td>-0.333706</td>\n",
       "      <td>4.093285e-02</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6932</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>-0.532341</td>\n",
       "      <td>3.320092e-06</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6978</th>\n",
       "      <td>NFATC2</td>\n",
       "      <td>-0.457536</td>\n",
       "      <td>1.111948e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6980</th>\n",
       "      <td>BTK</td>\n",
       "      <td>-0.454873</td>\n",
       "      <td>1.269551e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6997</th>\n",
       "      <td>INPP5D</td>\n",
       "      <td>-0.440058</td>\n",
       "      <td>2.212241e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7005</th>\n",
       "      <td>VAV1</td>\n",
       "      <td>-0.439464</td>\n",
       "      <td>2.264489e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7024</th>\n",
       "      <td>GRAP2</td>\n",
       "      <td>-0.428093</td>\n",
       "      <td>3.591048e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7042</th>\n",
       "      <td>CD4</td>\n",
       "      <td>-0.417232</td>\n",
       "      <td>5.663144e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7059</th>\n",
       "      <td>PTPN6</td>\n",
       "      <td>-0.411321</td>\n",
       "      <td>6.865751e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7061</th>\n",
       "      <td>ZAP70</td>\n",
       "      <td>-0.410167</td>\n",
       "      <td>7.208683e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7077</th>\n",
       "      <td>PTPRC</td>\n",
       "      <td>-0.402397</td>\n",
       "      <td>9.771626e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7078</th>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>-0.402484</td>\n",
       "      <td>9.771626e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7121</th>\n",
       "      <td>LCP2</td>\n",
       "      <td>-0.390098</td>\n",
       "      <td>1.462062e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7141</th>\n",
       "      <td>CD5</td>\n",
       "      <td>-0.396950</td>\n",
       "      <td>1.583339e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7187</th>\n",
       "      <td>LYN</td>\n",
       "      <td>-0.374709</td>\n",
       "      <td>2.361831e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7219</th>\n",
       "      <td>CD79A</td>\n",
       "      <td>-0.366693</td>\n",
       "      <td>3.102262e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7222</th>\n",
       "      <td>CD3G</td>\n",
       "      <td>-0.381032</td>\n",
       "      <td>3.210780e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7257</th>\n",
       "      <td>PPP3CC</td>\n",
       "      <td>-0.359171</td>\n",
       "      <td>3.917390e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7401</th>\n",
       "      <td>LCK</td>\n",
       "      <td>-0.331149</td>\n",
       "      <td>9.506279e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7603</th>\n",
       "      <td>CD3E</td>\n",
       "      <td>-0.339994</td>\n",
       "      <td>2.079403e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7646</th>\n",
       "      <td>SYK</td>\n",
       "      <td>-0.298428</td>\n",
       "      <td>2.328405e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7721</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>-0.290547</td>\n",
       "      <td>2.839009e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7781</th>\n",
       "      <td>PRKCQ</td>\n",
       "      <td>-0.284462</td>\n",
       "      <td>3.293176e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7876</th>\n",
       "      <td>RAC2</td>\n",
       "      <td>-0.277100</td>\n",
       "      <td>3.855935e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8461</th>\n",
       "      <td>NFATC2</td>\n",
       "      <td>-0.297428</td>\n",
       "      <td>4.438700e-02</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Comparison  Correlation       P_value    Cancer Type\n",
       "1          GRB2    -0.610889  6.729990e-08            GBM\n",
       "75          SYK    -0.470849  1.202510e-04            GBM\n",
       "82          CD4    -0.467479  1.362923e-04            GBM\n",
       "86       PIK3CD    -0.466265  1.399080e-04            GBM\n",
       "100        LCP2    -0.460881  1.665085e-04            GBM\n",
       "125        VAV1    -0.447143  2.894593e-04            GBM\n",
       "141     HLA-DMB    -0.442005  3.476225e-04            GBM\n",
       "199       PTPRC    -0.424958  6.196952e-04            GBM\n",
       "204         BTK    -0.423380  6.552047e-04            GBM\n",
       "287      INPP5D    -0.408181  1.009137e-03            GBM\n",
       "318       PTPN6    -0.402017  1.241946e-03            GBM\n",
       "361      PIK3CG    -0.394048  1.602237e-03            GBM\n",
       "370      NFATC2    -0.392602  1.681448e-03            GBM\n",
       "603     HLA-DMA    -0.373678  5.056470e-03            GBM\n",
       "605        BLNK    -0.356906  5.065055e-03            GBM\n",
       "941        RAC2    -0.321178  1.349871e-02            GBM\n",
       "990       PRKCQ    -0.329766  1.621074e-02            GBM\n",
       "2270      GRAP2    -0.305290  2.127686e-02         Kidney\n",
       "2400       VAV1    -0.285693  3.025415e-02         Kidney\n",
       "2492      ZAP70    -0.276087  3.833890e-02         Kidney\n",
       "2550     INPP5D    -0.271891  4.182668e-02         Kidney\n",
       "5750     PPP3CC    -0.432271  3.644683e-04           Luad\n",
       "5773      PRKCQ    -0.418175  6.284144e-04           Luad\n",
       "5851      ZAP70    -0.384613  2.092847e-03           Luad\n",
       "5893      PTPN6    -0.371990  3.159187e-03           Luad\n",
       "5897        LCK    -0.370894  3.271948e-03           Luad\n",
       "6002       RAC2    -0.349937  5.935190e-03           Luad\n",
       "6094        BLK    -0.365512  8.766800e-03           Luad\n",
       "6121     PIK3CD    -0.331760  9.676902e-03           Luad\n",
       "6364       GRB2    -0.302439  2.063096e-02           Luad\n",
       "6480       CD8A    -0.292478  2.581604e-02           Luad\n",
       "6666        BTK    -0.278003  3.552087e-02           Luad\n",
       "6762     INPP5D    -0.271933  4.007023e-02           Luad\n",
       "6777       CD3G    -0.333706  4.093285e-02           Luad\n",
       "6932       GRB2    -0.532341  3.320092e-06  Head and Neck\n",
       "6978     NFATC2    -0.457536  1.111948e-04  Head and Neck\n",
       "6980        BTK    -0.454873  1.269551e-04  Head and Neck\n",
       "6997     INPP5D    -0.440058  2.212241e-04  Head and Neck\n",
       "7005       VAV1    -0.439464  2.264489e-04  Head and Neck\n",
       "7024      GRAP2    -0.428093  3.591048e-04  Head and Neck\n",
       "7042        CD4    -0.417232  5.663144e-04  Head and Neck\n",
       "7059      PTPN6    -0.411321  6.865751e-04  Head and Neck\n",
       "7061      ZAP70    -0.410167  7.208683e-04  Head and Neck\n",
       "7077      PTPRC    -0.402397  9.771626e-04  Head and Neck\n",
       "7078     PIK3CG    -0.402484  9.771626e-04  Head and Neck\n",
       "7121       LCP2    -0.390098  1.462062e-03  Head and Neck\n",
       "7141        CD5    -0.396950  1.583339e-03  Head and Neck\n",
       "7187        LYN    -0.374709  2.361831e-03  Head and Neck\n",
       "7219      CD79A    -0.366693  3.102262e-03  Head and Neck\n",
       "7222       CD3G    -0.381032  3.210780e-03  Head and Neck\n",
       "7257     PPP3CC    -0.359171  3.917390e-03  Head and Neck\n",
       "7401        LCK    -0.331149  9.506279e-03  Head and Neck\n",
       "7603       CD3E    -0.339994  2.079403e-02  Head and Neck\n",
       "7646        SYK    -0.298428  2.328405e-02  Head and Neck\n",
       "7721    HLA-DMB    -0.290547  2.839009e-02  Head and Neck\n",
       "7781      PRKCQ    -0.284462  3.293176e-02  Head and Neck\n",
       "7876       RAC2    -0.277100  3.855935e-02  Head and Neck\n",
       "8461     NFATC2    -0.297428  4.438700e-02           Lscc"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Get append version of the df with all cancer type, fdr sig trans results\n",
    "df_FDR_append = pd.read_csv(\"Pval_corr_table_Fig_2_return_sig_append.csv\")\n",
    "df_FDR_append = df_FDR_append.drop(['Unnamed: 0'], axis=1)\n",
    "\n",
    "#Remove Proteomics labeling \n",
    "df_FDR_append = df_FDR_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_FDR_append= df_FDR_append[df_FDR_append.Comparison.isin(immune)]\n",
    "df_FDR_append"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set add new column to be unique index and order the new index. This way genes will be grouped by coagulation factors, regulators, and urokinase genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Correlation</th>\n",
       "      <th>P_value</th>\n",
       "      <th>Cancer Type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CD3E Head and Neck</th>\n",
       "      <td>CD3E</td>\n",
       "      <td>-0.339994</td>\n",
       "      <td>2.079403e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD3G Head and Neck</th>\n",
       "      <td>CD3G</td>\n",
       "      <td>-0.381032</td>\n",
       "      <td>3.210780e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD4 Head and Neck</th>\n",
       "      <td>CD4</td>\n",
       "      <td>-0.417232</td>\n",
       "      <td>5.663144e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD8A Luad</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>-0.292478</td>\n",
       "      <td>2.581604e-02</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LCK Head and Neck</th>\n",
       "      <td>LCK</td>\n",
       "      <td>-0.331149</td>\n",
       "      <td>9.506279e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZAP70 Head and Neck</th>\n",
       "      <td>ZAP70</td>\n",
       "      <td>-0.410167</td>\n",
       "      <td>7.208683e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LCP2 Head and Neck</th>\n",
       "      <td>LCP2</td>\n",
       "      <td>-0.390098</td>\n",
       "      <td>1.462062e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRAP2 Head and Neck</th>\n",
       "      <td>GRAP2</td>\n",
       "      <td>-0.428093</td>\n",
       "      <td>3.591048e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VAV1 Head and Neck</th>\n",
       "      <td>VAV1</td>\n",
       "      <td>-0.439464</td>\n",
       "      <td>2.264489e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2 Head and Neck</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>-0.532341</td>\n",
       "      <td>3.320092e-06</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFATC2 Head and Neck</th>\n",
       "      <td>NFATC2</td>\n",
       "      <td>-0.457536</td>\n",
       "      <td>1.111948e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPP3CC Head and Neck</th>\n",
       "      <td>PPP3CC</td>\n",
       "      <td>-0.359171</td>\n",
       "      <td>3.917390e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAC2 Head and Neck</th>\n",
       "      <td>RAC2</td>\n",
       "      <td>-0.277100</td>\n",
       "      <td>3.855935e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD79A Head and Neck</th>\n",
       "      <td>CD79A</td>\n",
       "      <td>-0.366693</td>\n",
       "      <td>3.102262e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SYK Head and Neck</th>\n",
       "      <td>SYK</td>\n",
       "      <td>-0.298428</td>\n",
       "      <td>2.328405e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BTK Head and Neck</th>\n",
       "      <td>BTK</td>\n",
       "      <td>-0.454873</td>\n",
       "      <td>1.269551e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD5 Head and Neck</th>\n",
       "      <td>CD5</td>\n",
       "      <td>-0.396950</td>\n",
       "      <td>1.583339e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTPN6 Head and Neck</th>\n",
       "      <td>PTPN6</td>\n",
       "      <td>-0.411321</td>\n",
       "      <td>6.865751e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LYN Head and Neck</th>\n",
       "      <td>LYN</td>\n",
       "      <td>-0.374709</td>\n",
       "      <td>2.361831e-03</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTPRC Head and Neck</th>\n",
       "      <td>PTPRC</td>\n",
       "      <td>-0.402397</td>\n",
       "      <td>9.771626e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CG Head and Neck</th>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>-0.402484</td>\n",
       "      <td>9.771626e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INPP5D Head and Neck</th>\n",
       "      <td>INPP5D</td>\n",
       "      <td>-0.440058</td>\n",
       "      <td>2.212241e-04</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRKCQ Head and Neck</th>\n",
       "      <td>PRKCQ</td>\n",
       "      <td>-0.284462</td>\n",
       "      <td>3.293176e-02</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD4 GBM</th>\n",
       "      <td>CD4</td>\n",
       "      <td>-0.467479</td>\n",
       "      <td>1.362923e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LCP2 GBM</th>\n",
       "      <td>LCP2</td>\n",
       "      <td>-0.460881</td>\n",
       "      <td>1.665085e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VAV1 GBM</th>\n",
       "      <td>VAV1</td>\n",
       "      <td>-0.447143</td>\n",
       "      <td>2.894593e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRB2 GBM</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>-0.610889</td>\n",
       "      <td>6.729990e-08</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFATC2 GBM</th>\n",
       "      <td>NFATC2</td>\n",
       "      <td>-0.392602</td>\n",
       "      <td>1.681448e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAC2 GBM</th>\n",
       "      <td>RAC2</td>\n",
       "      <td>-0.321178</td>\n",
       "      <td>1.349871e-02</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SYK GBM</th>\n",
       "      <td>SYK</td>\n",
       "      <td>-0.470849</td>\n",
       "      <td>1.202510e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BTK GBM</th>\n",
       "      <td>BTK</td>\n",
       "      <td>-0.423380</td>\n",
       "      <td>6.552047e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BLNK GBM</th>\n",
       "      <td>BLNK</td>\n",
       "      <td>-0.356906</td>\n",
       "      <td>5.065055e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTPN6 GBM</th>\n",
       "      <td>PTPN6</td>\n",
       "      <td>-0.402017</td>\n",
       "      <td>1.241946e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTPRC GBM</th>\n",
       "      <td>PTPRC</td>\n",
       "      <td>-0.424958</td>\n",
       "      <td>6.196952e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CD GBM</th>\n",
       "      <td>PIK3CD</td>\n",
       "      <td>-0.466265</td>\n",
       "      <td>1.399080e-04</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CG GBM</th>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>-0.394048</td>\n",
       "      <td>1.602237e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INPP5D GBM</th>\n",
       "      <td>INPP5D</td>\n",
       "      <td>-0.408181</td>\n",
       "      <td>1.009137e-03</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRKCQ GBM</th>\n",
       "      <td>PRKCQ</td>\n",
       "      <td>-0.329766</td>\n",
       "      <td>1.621074e-02</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD3G Luad</th>\n",
       "      <td>CD3G</td>\n",
       "      <td>-0.333706</td>\n",
       "      <td>4.093285e-02</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LCK Luad</th>\n",
       "      <td>LCK</td>\n",
       "      <td>-0.370894</td>\n",
       "      <td>3.271948e-03</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Comparison  Correlation       P_value    Cancer Type\n",
       "Index                                                                    \n",
       "CD3E Head and Neck         CD3E    -0.339994  2.079403e-02  Head and Neck\n",
       "CD3G Head and Neck         CD3G    -0.381032  3.210780e-03  Head and Neck\n",
       "CD4 Head and Neck           CD4    -0.417232  5.663144e-04  Head and Neck\n",
       "CD8A Luad                  CD8A    -0.292478  2.581604e-02           Luad\n",
       "LCK Head and Neck           LCK    -0.331149  9.506279e-03  Head and Neck\n",
       "ZAP70 Head and Neck       ZAP70    -0.410167  7.208683e-04  Head and Neck\n",
       "LCP2 Head and Neck         LCP2    -0.390098  1.462062e-03  Head and Neck\n",
       "GRAP2 Head and Neck       GRAP2    -0.428093  3.591048e-04  Head and Neck\n",
       "VAV1 Head and Neck         VAV1    -0.439464  2.264489e-04  Head and Neck\n",
       "GRB2 Head and Neck         GRB2    -0.532341  3.320092e-06  Head and Neck\n",
       "NFATC2 Head and Neck     NFATC2    -0.457536  1.111948e-04  Head and Neck\n",
       "PPP3CC Head and Neck     PPP3CC    -0.359171  3.917390e-03  Head and Neck\n",
       "RAC2 Head and Neck         RAC2    -0.277100  3.855935e-02  Head and Neck\n",
       "CD79A Head and Neck       CD79A    -0.366693  3.102262e-03  Head and Neck\n",
       "SYK Head and Neck           SYK    -0.298428  2.328405e-02  Head and Neck\n",
       "BTK Head and Neck           BTK    -0.454873  1.269551e-04  Head and Neck\n",
       "CD5 Head and Neck           CD5    -0.396950  1.583339e-03  Head and Neck\n",
       "PTPN6 Head and Neck       PTPN6    -0.411321  6.865751e-04  Head and Neck\n",
       "LYN Head and Neck           LYN    -0.374709  2.361831e-03  Head and Neck\n",
       "PTPRC Head and Neck       PTPRC    -0.402397  9.771626e-04  Head and Neck\n",
       "PIK3CG Head and Neck     PIK3CG    -0.402484  9.771626e-04  Head and Neck\n",
       "INPP5D Head and Neck     INPP5D    -0.440058  2.212241e-04  Head and Neck\n",
       "PRKCQ Head and Neck       PRKCQ    -0.284462  3.293176e-02  Head and Neck\n",
       "CD4 GBM                     CD4    -0.467479  1.362923e-04            GBM\n",
       "LCP2 GBM                   LCP2    -0.460881  1.665085e-04            GBM\n",
       "VAV1 GBM                   VAV1    -0.447143  2.894593e-04            GBM\n",
       "GRB2 GBM                   GRB2    -0.610889  6.729990e-08            GBM\n",
       "NFATC2 GBM               NFATC2    -0.392602  1.681448e-03            GBM\n",
       "RAC2 GBM                   RAC2    -0.321178  1.349871e-02            GBM\n",
       "SYK GBM                     SYK    -0.470849  1.202510e-04            GBM\n",
       "BTK GBM                     BTK    -0.423380  6.552047e-04            GBM\n",
       "BLNK GBM                   BLNK    -0.356906  5.065055e-03            GBM\n",
       "PTPN6 GBM                 PTPN6    -0.402017  1.241946e-03            GBM\n",
       "PTPRC GBM                 PTPRC    -0.424958  6.196952e-04            GBM\n",
       "PIK3CD GBM               PIK3CD    -0.466265  1.399080e-04            GBM\n",
       "PIK3CG GBM               PIK3CG    -0.394048  1.602237e-03            GBM\n",
       "INPP5D GBM               INPP5D    -0.408181  1.009137e-03            GBM\n",
       "PRKCQ GBM                 PRKCQ    -0.329766  1.621074e-02            GBM\n",
       "CD3G Luad                  CD3G    -0.333706  4.093285e-02           Luad\n",
       "LCK Luad                    LCK    -0.370894  3.271948e-03           Luad"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "df_FDR_append[\"Index\"] = df_FDR_append[\"Comparison\"] + \" \" + df_FDR_append[\"Cancer Type\"]\n",
    "df_FDR_append = df_FDR_append.set_index(\"Index\")\n",
    "df_ordered = df_FDR_append.reindex(['CD3E Head and Neck', 'CD3G Head and Neck','CD4 Head and Neck','CD8A Luad', 'LCK Head and Neck', 'ZAP70 Head and Neck','LCP2 Head and Neck', 'GRAP2 Head and Neck','VAV1 Head and Neck','GRB2 Head and Neck', 'NFATC2 Head and Neck','PPP3CC Head and Neck', 'RAC2 Head and Neck','CD79A Head and Neck','SYK Head and Neck','BTK Head and Neck','CD5 Head and Neck','PTPN6 Head and Neck','LYN Head and Neck','PTPRC Head and Neck','PIK3CG Head and Neck','INPP5D Head and Neck','PRKCQ Head and Neck',\n",
    "                                    'CD4 GBM', 'LCP2 GBM', 'VAV1 GBM','GRB2 GBM', 'NFATC2 GBM','RAC2 GBM','SYK GBM','BTK GBM','BLNK GBM', 'PTPN6 GBM','PTPRC GBM', 'PIK3CD GBM', 'PIK3CG GBM','INPP5D GBM','PRKCQ GBM', \n",
    "                                    'CD3G Luad','LCK Luad', 'ZAP70 Luad','GRB2 Luad','PPP3CC Luad', 'RAC2 Luad','BLK Luad','BTK Luad','PTPN6 Luad','PIK3CD Luad','INPP5D Luad','PRKCQ Luad',\n",
    "                                    'VAV1 Kidney',\"ZAP70 Kidney\", \"INPP5D Kidney\",\"GRAP2 Kidney\",\"HLA-DMA GBM\",\"HLA-DMB GBM\", \n",
    "                                    \"NFATC2 Lscc\"])\n",
    "                                    \n",
    "df_ordered.head(40)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Step 4: Plot Figure 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.pydata.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"3050\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"3050\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error() {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < css_urls.length; i++) {\n",
       "      var url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error;\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };var element = document.getElementById(\"3050\");\n",
       "  if (element == null) {\n",
       "    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '3050' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.2.0.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.2.0.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.2.0.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.2.0.min.js\"];\n",
       "  var css_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.2.0.min.css\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.2.0.min.css\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.2.0.min.css\"];\n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    \n",
       "    function(Bokeh) {\n",
       "      \n",
       "    },\n",
       "    function(Bokeh) {} // ensure no trailing comma for IE\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if ((root.Bokeh !== undefined) || (force === true)) {\n",
       "      for (var i = 0; i < inline_js.length; i++) {\n",
       "        inline_js[i].call(root, root.Bokeh);\n",
       "      }if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"3050\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"3050\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error() {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (var i = 0; i < css_urls.length; i++) {\n      var url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };var element = document.getElementById(\"3050\");\n  if (element == null) {\n    console.error(\"Bokeh: ERROR: autoload.js configured with elementid '3050' but no matching script tag was found. \")\n    return false;\n  }\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.2.0.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.2.0.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.2.0.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.2.0.min.js\"];\n  var css_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.2.0.min.css\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.2.0.min.css\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.2.0.min.css\"];\n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    \n    function(Bokeh) {\n      \n    },\n    function(Bokeh) {} // ensure no trailing comma for IE\n  ];\n\n  function run_inline_js() {\n    \n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (var i = 0; i < inline_js.length; i++) {\n        inline_js[i].call(root, root.Bokeh);\n      }if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"3050\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"ccbfa427-af04-4315-8740-6fd558ebdfba\" data-root-id=\"3051\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"bd613a43-9134-4e83-a7c9-5c34bff1bf0e\":{\"roots\":{\"references\":[{\"attributes\":{\"children\":[{\"id\":\"2971\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"id\":\"3026\",\"subtype\":\"Figure\",\"type\":\"Plot\"}]},\"id\":\"3051\",\"type\":\"Row\"},{\"attributes\":{\"text\":\"\"},\"id\":\"3143\",\"type\":\"Title\"},{\"attributes\":{\"data_source\":{\"id\":\"3044\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"3046\",\"type\":\"Scatter\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"3047\",\"type\":\"Scatter\"},\"selection_glyph\":null,\"view\":{\"id\":\"3049\",\"type\":\"CDSView\"}},\"id\":\"3048\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"high\":0.6108891162098071,\"low\":-0.6108891162098071,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"2968\",\"type\":\"LinearColorMapper\"},{\"attributes\":{\"formatter\":{\"id\":\"3155\",\"type\":\"CategoricalTickFormatter\"},\"ticker\":{\"id\":\"3038\",\"type\":\"CategoricalTicker\"}},\"id\":\"3037\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"3031\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"fill_color\":{\"field\":\"Correlation\",\"transform\":{\"id\":\"2968\",\"type\":\"LinearColorMapper\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Comparison\"},\"y\":{\"field\":\"Cancer Type\"}},\"id\":\"2991\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"3034\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"2979\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"callback\":null,\"factors\":[\"\"]},\"id\":\"3024\",\"type\":\"FactorRange\"},{\"attributes\":{\"dimension\":1,\"ticker\":{\"id\":\"3038\",\"type\":\"CategoricalTicker\"}},\"id\":\"3040\",\"type\":\"Grid\"},{\"attributes\":{\"callback\":null,\"factors\":[\"CD3E\",\"CD3G\",\"CD4\",\"CD8A\",\"LCK\",\"ZAP70\",\"LCP2\",\"GRAP2\",\"VAV1\",\"GRB2\",\"NFATC2\",\"PPP3CC\",\"RAC2\",\"CD79A\",\"SYK\",\"BTK\",\"CD5\",\"PTPN6\",\"LYN\",\"PTPRC\",\"PIK3CG\",\"INPP5D\",\"PRKCQ\",\"BLNK\",\"PIK3CD\",\"BLK\",\"HLA-DMA\",\"HLA-DMB\"]},\"id\":\"2969\",\"type\":\"FactorRange\"},{\"attributes\":{\"axis_label\":\"FDR_P-Values\",\"formatter\":{\"id\":\"3153\",\"type\":\"CategoricalTickFormatter\"},\"major_label_orientation\":1.5707963267948966,\"ticker\":{\"id\":\"3034\",\"type\":\"CategoricalTicker\"}},\"id\":\"3033\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"2983\",\"type\":\"CategoricalTicker\"},{\"attributes\":{},\"id\":\"3155\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"callback\":null},\"id\":\"3041\",\"type\":\"HoverTool\"},{\"attributes\":{\"formatter\":{\"id\":\"3151\",\"type\":\"CategoricalTickFormatter\"},\"major_label_text_font_size\":{\"value\":\"10pt\"},\"ticker\":{\"id\":\"2983\",\"type\":\"CategoricalTicker\"}},\"id\":\"2982\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"3151\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"formatter\":{\"id\":\"3149\",\"type\":\"CategoricalTickFormatter\"},\"major_label_orientation\":1.5707963267948966,\"major_label_text_font_size\":{\"value\":\"10pt\"},\"ticker\":{\"id\":\"2979\",\"type\":\"CategoricalTicker\"}},\"id\":\"2978\",\"type\":\"CategoricalAxis\"},{\"attributes\":{\"dimension\":1,\"ticker\":{\"id\":\"2983\",\"type\":\"CategoricalTicker\"}},\"id\":\"2985\",\"type\":\"Grid\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"2986\",\"type\":\"HoverTool\"}]},\"id\":\"2987\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"3149\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"below\":[{\"id\":\"2978\",\"type\":\"CategoricalAxis\"}],\"center\":[{\"id\":\"2981\",\"type\":\"Grid\"},{\"id\":\"2985\",\"type\":\"Grid\"}],\"left\":[{\"id\":\"2982\",\"type\":\"CategoricalAxis\"}],\"plot_height\":500,\"plot_width\":1000,\"renderers\":[{\"id\":\"2993\",\"type\":\"GlyphRenderer\"}],\"right\":[{\"id\":\"2995\",\"type\":\"ColorBar\"}],\"title\":{\"id\":\"3143\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"2987\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"2969\",\"type\":\"FactorRange\"},\"x_scale\":{\"id\":\"2974\",\"type\":\"CategoricalScale\"},\"y_range\":{\"id\":\"2970\",\"type\":\"FactorRange\"},\"y_scale\":{\"id\":\"2976\",\"type\":\"CategoricalScale\"}},\"id\":\"2971\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"high\":1.5,\"low\":1.5,\"palette\":[\"#2166ac\",\"#4393c3\",\"#92c5de\",\"#d1e5f0\",\"#f7f7f7\",\"#fddbc7\",\"#f4a582\",\"#d6604d\",\"#b2182b\"]},\"id\":\"3023\",\"type\":\"LinearColorMapper\"},{\"attributes\":{},\"id\":\"3159\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"callback\":null,\"data\":{\"Cancer Type\":[\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Luad\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"Head and Neck\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"GBM\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Luad\",\"Kidney\",\"Kidney\",\"Kidney\",\"Kidney\",\"GBM\",\"GBM\",\"Lscc\"],\"Comparison\":[\"CD3E\",\"CD3G\",\"CD4\",\"CD8A\",\"LCK\",\"ZAP70\",\"LCP2\",\"GRAP2\",\"VAV1\",\"GRB2\",\"NFATC2\",\"PPP3CC\",\"RAC2\",\"CD79A\",\"SYK\",\"BTK\",\"CD5\",\"PTPN6\",\"LYN\",\"PTPRC\",\"PIK3CG\",\"INPP5D\",\"PRKCQ\",\"CD4\",\"LCP2\",\"VAV1\",\"GRB2\",\"NFATC2\",\"RAC2\",\"SYK\",\"BTK\",\"BLNK\",\"PTPN6\",\"PTPRC\",\"PIK3CD\",\"PIK3CG\",\"INPP5D\",\"PRKCQ\",\"CD3G\",\"LCK\",\"ZAP70\",\"GRB2\",\"PPP3CC\",\"RAC2\",\"BLK\",\"BTK\",\"PTPN6\",\"PIK3CD\",\"INPP5D\",\"PRKCQ\",\"VAV1\",\"ZAP70\",\"INPP5D\",\"GRAP2\",\"HLA-DMA\",\"HLA-DMB\",\"NFATC2\"],\"Correlation\":{\"__ndarray__\":\"DB6XHHfC1b82uFzO1GLYvyKc8K/vs9q/vbI4OPW30r87glW7iTHVv9INnrouQNq/A50RvFv32L94sEuq32Xbv8w0RT0tINy/AEvCcfAI4b/0ilSIRkjdv0WGL0yq/Na/Row7xwG80b8G/hlc5XfXv/kLOLdyGdO/KyYgbaIc3b8GysjaomfZv1u2bR0WU9q/xOsawzn71794z4Bz3cDZvwWFlkVMwtm/dnTLpeop3L9uyhgBojTSv2FSki0t692/gTmE5RJ/3b9wUpPm/J3cv5ZP81RnjOO/ZLySgGUg2b/eUKy7L47Uv/YZa+1iIt6/LiFdj6gY279LeTvcjNfWv0oK5CKlutm/luCkj4Iy279cwaS2SNfdv0z0LgUWONm/J8tljaEf2r9t1bxz4hrVv45JW2puW9W/9g1nxri8179DSsQSfp3YvyqRQ0knW9O/e3ebBlaq27+s7x4IXWXWv5PzHpGKZNe/RU5V3cvK0b9a6zV3sM7Xv+BsjnWOO9W/DktVFFtn0b9WSSmhYsPavzhPEpHLSNK/lj43LGmr0b9sJ33FqGbRv1fYfsrdidO/6H0S0Vjq179jlfX/0EncvxfDE/cPCdO/\",\"dtype\":\"float64\",\"shape\":[57]},\"Index\":[\"CD3E Head and Neck\",\"CD3G Head and Neck\",\"CD4 Head and Neck\",\"CD8A Luad\",\"LCK Head and Neck\",\"ZAP70 Head and Neck\",\"LCP2 Head and Neck\",\"GRAP2 Head and Neck\",\"VAV1 Head and Neck\",\"GRB2 Head and Neck\",\"NFATC2 Head and Neck\",\"PPP3CC Head and Neck\",\"RAC2 Head and Neck\",\"CD79A Head and Neck\",\"SYK Head and Neck\",\"BTK Head and Neck\",\"CD5 Head and Neck\",\"PTPN6 Head and Neck\",\"LYN Head and Neck\",\"PTPRC Head and Neck\",\"PIK3CG Head and Neck\",\"INPP5D Head and Neck\",\"PRKCQ Head and Neck\",\"CD4 GBM\",\"LCP2 GBM\",\"VAV1 GBM\",\"GRB2 GBM\",\"NFATC2 GBM\",\"RAC2 GBM\",\"SYK GBM\",\"BTK GBM\",\"BLNK GBM\",\"PTPN6 GBM\",\"PTPRC GBM\",\"PIK3CD GBM\",\"PIK3CG GBM\",\"INPP5D GBM\",\"PRKCQ GBM\",\"CD3G Luad\",\"LCK Luad\",\"ZAP70 Luad\",\"GRB2 Luad\",\"PPP3CC Luad\",\"RAC2 Luad\",\"BLK Luad\",\"BTK Luad\",\"PTPN6 Luad\",\"PIK3CD Luad\",\"INPP5D Luad\",\"PRKCQ Luad\",\"VAV1 Kidney\",\"ZAP70 Kidney\",\"INPP5D Kidney\",\"GRAP2 Kidney\",\"HLA-DMA GBM\",\"HLA-DMB GBM\",\"NFATC2 Lscc\"],\"P_value\":{\"__ndarray__\":\"Tawh6gdLlT9Kb+lHfk1qP1nABPCWjkI/B0JYQYVvmj8RQz45B3iDPx155e4Un0c/zY7rAFX0Vz9F3OI/x4g3P6WU1Vtdri0/w9Law9fZyz4yee8jKCYdP21/ak+uC3A/aalsIw2+oz/Qzzx26mlpP4fewyzG15c/afXJquejID/G9OiLAfFZP/zqwMFof0Y/Z9i9Jh5ZYz8Z0GFchAJQPxnQYVyEAlA/tSTqgQz/LD+s8KGpbtygP2FphCM23SE/0ElVGhnTJT9x999mUvgyP7yDMU/QEHI+HwNkpICMWz8O5IRfNqWLP7xWTa3nhR8/AJudXUF4RT/acgBNGL90P+EFBLQZWVQ/F+N8XmFORD8lmO7tiFYiPzpExw9FQFo/V3Y+0KCIUD/WiG52jJmQP4DqXnYm9aQ/jUjuv8XNaj9v7GrpBCVhP+YmHj1IIJU/vxw4dsPiNz/l+0Flf094P3xEyP1T9IE/MMrYhcovoj/3AZ5xS+FpP1G9LdB70YM/LhOOqBWEpD/y8MWXhZdEPxQFzFPx+p4/Gww8DSihoz9cb0aFTmqlP2a9lpiZyZU/q6WJuRe2dD+eqx9pI8g2PzeTnL3kuaY/\",\"dtype\":\"float64\",\"shape\":[57]},\"size\":{\"__ndarray__\":\"nje5phA9J0AKd1n/RTkxQJ7ZFGnYbTZA2AhFOcjwJUDSHi0eUO8rQAN6uYKFtDVACGG02m6VM0BV6QhIsss3QHCi8zXRLTlAC6Dc5FvsQkCMgg7ZC1A7QEaSnlyCoDBAlM1h9IiII0BqI9jOrVMxQBL+a+xWjyZAg2Y5jD/qOkB8Znd8O1gzQM2CrFz02TVA4MCrwhslMkCoETgH5so0QKgROAfmyjRA7te1qr4/OUBrkA8X23okQGYsf4G+szpA4g083vQZOkAo7YfWRnE4QDbYjslqxUhARLRVmQ8qM0D+ZTQ1utQpQIF84PzpEztAMsmZTd/9NUC5vwUKXLYvQJpLzvm+EjRA0ltTSKooNkBCWD6mop86QIJ/z8MeTzNAlwliqSuyNEBEKJZGhLsoQLC4Q4HILCNASnnkqscqMUCJmDrr94EyQOz+XogoSSdA2/mifE/AN0Cif3fp2cIuQJgASQazayxAIN+ZdpsGJEBd5ZrktkUxQGP20hn90ytApdSEvH9NI0Brvv9G7x02QFBpDJEd/SRApXH7fFeRI0ByJ1psmgsjQKwcKdfOGSdAvpgEGve4L0DZwzpyp+Q3QIbpUI1YsCJA\",\"dtype\":\"float64\",\"shape\":[57]},\"size2\":{\"__ndarray__\":\"fp+hMxb8DkAN9CH/B/cWQH13xjbL5x1AdbYG9wpBDUA3v8i+ip8SQAT4oQOy8BxAttZFzpMcGkBxjLYKQ7ofQKDB9yM2ySBAuSomMSU7KUCzAV87XTUiQF4YfntYKxZAxbzXRWELCkDj2coTPRoXQMJS5eVzFA5AAu/Qsn/xIUD73Un7+coZQBFZOyabIh1AKgHlWHoxGEA2wkq0MrkbQDbCSrQyuRtASeUjxynVIECPwBR0eU4LQJkdqqspzSFA7LPSPqNmIUDFSAWPhEsgQM86X4acgzBABZvHIWqNGUD/Q3gjfDgRQFaolahGDSJAmGEiEn9SHUAm1QNcPSQVQHgPE02pwxpAbnrEtY2LHUAskCnEwb8hQK5Uv6/TvhlAH2ItNzqYG0AtcLnZAn0QQED2BFcLkQlAYkzb47TjFkBhy/iOSq0YQOZTKQs2DA9AeaIuphSrH0DB/6Sb5oEUQGUAhll38hJAKtR383mzCkB83COGngcXQEKkjGZTjRJAhxuxUKq8CUDk/VQJP30dQGuMEGwn/AtAMUJPUR8XCkDtic3lzWQJQDomjMkTzQ5A1GVYEfolFUDMr6PtidsfQAjia7wg6whA\",\"dtype\":\"float64\",\"shape\":[57]}},\"selected\":{\"id\":\"3156\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"3157\",\"type\":\"UnionRenderers\"}},\"id\":\"2989\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"2976\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"callback\":null,\"data\":{\"Correlation\":{\"__ndarray__\":\"AAAAAAAA+D8AAAAAAAD4PwAAAAAAAPg/\",\"dtype\":\"float64\",\"shape\":[3]},\"P_Value\":{\"__ndarray__\":\"mpmZmZmZqT+N7bWg98awPryDMU/QEHI+\",\"dtype\":\"float64\",\"shape\":[3]},\"index\":[0,1,2],\"size\":{\"__ndarray__\":\"reqW3HH5IUC4/yvzJ7lEQDbYjslqxUhA\",\"dtype\":\"float64\",\"shape\":[3]},\"size2\":{\"__ndarray__\":\"keNze0L3B0Cg/4+ZiqErQM86X4acgzBA\",\"dtype\":\"float64\",\"shape\":[3]},\"x_axis\":[\"\",\"\",\"\"],\"y_axis\":[\"5.0e-02\",\"1.0e-06\",\"6.7e-08\"]},\"selected\":{\"id\":\"3158\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"3159\",\"type\":\"UnionRenderers\"}},\"id\":\"3044\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"3146\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"3029\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"fill_color\":{\"field\":\"Correlation\",\"transform\":{\"id\":\"3023\",\"type\":\"LinearColorMapper\"}},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"3046\",\"type\":\"Scatter\"},{\"attributes\":{},\"id\":\"3157\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"ticker\":{\"id\":\"2979\",\"type\":\"CategoricalTicker\"}},\"id\":\"2981\",\"type\":\"Grid\"},{\"attributes\":{\"data_source\":{\"id\":\"2989\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"2991\",\"type\":\"Scatter\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"2992\",\"type\":\"Scatter\"},\"selection_glyph\":null,\"view\":{\"id\":\"2994\",\"type\":\"CDSView\"}},\"id\":\"2993\",\"type\":\"GlyphRenderer\"},{\"attributes\":{},\"id\":\"3147\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"below\":[{\"id\":\"3033\",\"type\":\"CategoricalAxis\"}],\"center\":[{\"id\":\"3036\",\"type\":\"Grid\"},{\"id\":\"3040\",\"type\":\"Grid\"}],\"left\":[{\"id\":\"3037\",\"type\":\"CategoricalAxis\"}],\"plot_height\":200,\"plot_width\":120,\"renderers\":[{\"id\":\"3048\",\"type\":\"GlyphRenderer\"}],\"title\":{\"id\":\"3145\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"3042\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"3024\",\"type\":\"FactorRange\"},\"x_scale\":{\"id\":\"3029\",\"type\":\"CategoricalScale\"},\"y_range\":{\"id\":\"3025\",\"type\":\"FactorRange\"},\"y_scale\":{\"id\":\"3031\",\"type\":\"CategoricalScale\"}},\"id\":\"3026\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"Comparison\"},\"y\":{\"field\":\"Cancer Type\"}},\"id\":\"2992\",\"type\":\"Scatter\"},{\"attributes\":{\"callback\":null,\"factors\":[\"Luad\",\"Lscc\",\"Kidney\",\"Head and Neck\",\"GBM\"]},\"id\":\"2970\",\"type\":\"FactorRange\"},{\"attributes\":{\"ticker\":{\"id\":\"3034\",\"type\":\"CategoricalTicker\"}},\"id\":\"3036\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"3153\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"source\":{\"id\":\"2989\",\"type\":\"ColumnDataSource\"}},\"id\":\"2994\",\"type\":\"CDSView\"},{\"attributes\":{\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"line_width\":{\"value\":0},\"size\":{\"field\":\"size\",\"units\":\"screen\"},\"x\":{\"field\":\"x_axis\"},\"y\":{\"field\":\"y_axis\"}},\"id\":\"3047\",\"type\":\"Scatter\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"3041\",\"type\":\"HoverTool\"}]},\"id\":\"3042\",\"type\":\"Toolbar\"},{\"attributes\":{\"callback\":null},\"id\":\"2986\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"3156\",\"type\":\"Selection\"},{\"attributes\":{},\"id\":\"3158\",\"type\":\"Selection\"},{\"attributes\":{\"source\":{\"id\":\"3044\",\"type\":\"ColumnDataSource\"}},\"id\":\"3049\",\"type\":\"CDSView\"},{\"attributes\":{\"callback\":null,\"factors\":[\"5.0e-02\",\"1.0e-06\",\"6.7e-08\"]},\"id\":\"3025\",\"type\":\"FactorRange\"},{\"attributes\":{\"color_mapper\":{\"id\":\"2968\",\"type\":\"LinearColorMapper\"},\"formatter\":{\"id\":\"3147\",\"type\":\"BasicTickFormatter\"},\"location\":[0,0],\"ticker\":{\"id\":\"3146\",\"type\":\"BasicTicker\"}},\"id\":\"2995\",\"type\":\"ColorBar\"},{\"attributes\":{},\"id\":\"3038\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"text\":\"\"},\"id\":\"3145\",\"type\":\"Title\"},{\"attributes\":{},\"id\":\"2974\",\"type\":\"CategoricalScale\"}],\"root_ids\":[\"3051\"]},\"title\":\"Bokeh Application\",\"version\":\"1.2.0\"}};\n",
       "  var render_items = [{\"docid\":\"bd613a43-9134-4e83-a7c9-5c34bff1bf0e\",\"roots\":{\"3051\":\"ccbfa427-af04-4315-8740-6fd558ebdfba\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        embed_document(root);\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "      attempts++;\n",
       "      if (attempts > 100) {\n",
       "        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "3051"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "legend_min = df_ordered[\"P_value\"].min()\n",
    "#Make plot using plot utils\n",
    "p.plotCircleHeatMap(df_ordered, circle_var = \"P_value\",color_var = \"Correlation\", x_axis = \"Comparison\", y_axis = \"Cancer Type\", plot_width= 1000, plot_height = 500, legend_min = legend_min, legend_max = 0.05, font_size = 10, show_legend = True , save_png = \"Figure2.png\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check if blanks are due to no data "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The follow code chunks show that the following cancers/genes don't have data: colon THBD, Kidney MMP12, and Ovarian MMP12. (As mentioned in EGFR Figure 2 legend)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'coag_upa' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-27-043a7f1758aa>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;31m#subset dataframe to include genes only desired for figure\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m \u001b[0mdf_all_comp_coag\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_all_prot_append\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf_all_prot_append\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mComparison\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoag_upa\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Number of rows in data frame \"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_all_comp_coag\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[0mdf_all_prot_append\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'coag_upa' is not defined"
     ]
    }
   ],
   "source": [
    "#Get append version of the df with all proteins \n",
    "df_all_prot_append = pd.read_csv(\"Pval_corr_table_Fig_2_return_all_append.csv\")\n",
    "df_all_prot_append = df_all_prot_append.drop(['Unnamed: 0'], axis=1)\n",
    "#Remove Proteomics labeling \n",
    "df_all_prot_append = df_all_prot_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of missing columns\n",
      "4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Correlation</th>\n",
       "      <th>P_value</th>\n",
       "      <th>Cancer Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>-0.442005</td>\n",
       "      <td>0.000348</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>603</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>-0.373678</td>\n",
       "      <td>0.005056</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1667</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>-0.262423</td>\n",
       "      <td>0.055717</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1681</th>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>-0.261173</td>\n",
       "      <td>0.057358</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2494</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>-0.218791</td>\n",
       "      <td>0.126822</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2705</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>-0.209905</td>\n",
       "      <td>0.146458</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12448</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>-0.222952</td>\n",
       "      <td>0.109021</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12942</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>-0.199014</td>\n",
       "      <td>0.164162</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13489</th>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>-0.176880</td>\n",
       "      <td>0.229302</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15164</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>0.123195</td>\n",
       "      <td>0.443495</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18445</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>-0.044398</td>\n",
       "      <td>0.827363</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18648</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>-0.039792</td>\n",
       "      <td>0.847542</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24454</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>0.146296</td>\n",
       "      <td>0.495865</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25596</th>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>-0.115924</td>\n",
       "      <td>0.605777</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28370</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>-0.052360</td>\n",
       "      <td>0.837835</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28694</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>0.044980</td>\n",
       "      <td>0.863132</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29864</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>0.019519</td>\n",
       "      <td>0.947539</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30661</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>-0.002345</td>\n",
       "      <td>0.995827</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34370</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>0.196824</td>\n",
       "      <td>0.098855</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35195</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>0.166702</td>\n",
       "      <td>0.172019</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36631</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>0.134573</td>\n",
       "      <td>0.356425</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37685</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>0.088111</td>\n",
       "      <td>0.511556</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38265</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>0.072042</td>\n",
       "      <td>0.600532</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38956</th>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>0.052811</td>\n",
       "      <td>0.711526</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41990</th>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>0.270801</td>\n",
       "      <td>0.040933</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43833</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>0.184269</td>\n",
       "      <td>0.196235</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44037</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>0.177418</td>\n",
       "      <td>0.216594</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44401</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>0.165584</td>\n",
       "      <td>0.255514</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45569</th>\n",
       "      <td>HLA-DOA</td>\n",
       "      <td>0.151911</td>\n",
       "      <td>0.383575</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45873</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>0.124300</td>\n",
       "      <td>0.420163</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48784</th>\n",
       "      <td>HLA-DOB</td>\n",
       "      <td>-0.066582</td>\n",
       "      <td>0.741646</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50193</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>-0.027738</td>\n",
       "      <td>0.889242</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52379</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>-0.290547</td>\n",
       "      <td>0.028390</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52951</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>-0.252281</td>\n",
       "      <td>0.076947</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53389</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>-0.221519</td>\n",
       "      <td>0.118457</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54502</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>-0.175415</td>\n",
       "      <td>0.242017</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55143</th>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>-0.154935</td>\n",
       "      <td>0.313356</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56922</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>-0.106554</td>\n",
       "      <td>0.524974</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63269</th>\n",
       "      <td>HLA-DOB</td>\n",
       "      <td>-0.227253</td>\n",
       "      <td>0.159880</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63784</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>-0.201418</td>\n",
       "      <td>0.232838</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64326</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>-0.179610</td>\n",
       "      <td>0.307518</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65257</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>-0.151224</td>\n",
       "      <td>0.414930</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66021</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>-0.132824</td>\n",
       "      <td>0.486062</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66928</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>-0.110666</td>\n",
       "      <td>0.592151</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67687</th>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>-0.096418</td>\n",
       "      <td>0.648168</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68087</th>\n",
       "      <td>HLA-DOA</td>\n",
       "      <td>-0.147955</td>\n",
       "      <td>0.684254</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74659</th>\n",
       "      <td>HLA-DRB1</td>\n",
       "      <td>0.243673</td>\n",
       "      <td>0.102476</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79534</th>\n",
       "      <td>HLA-DMB</td>\n",
       "      <td>0.037077</td>\n",
       "      <td>0.851579</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79568</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>0.036115</td>\n",
       "      <td>0.855265</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79983</th>\n",
       "      <td>HLA-DRA</td>\n",
       "      <td>-0.022677</td>\n",
       "      <td>0.910492</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80054</th>\n",
       "      <td>HLA-DMA</td>\n",
       "      <td>0.027898</td>\n",
       "      <td>0.917367</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80198</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>-0.015465</td>\n",
       "      <td>0.939902</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Comparison  Correlation   P_value    Cancer Type\n",
       "141      HLA-DMB    -0.442005  0.000348            GBM\n",
       "603      HLA-DMA    -0.373678  0.005056            GBM\n",
       "1667    HLA-DPB1    -0.262423  0.055717            GBM\n",
       "1681     HLA-DRA    -0.261173  0.057358            GBM\n",
       "2494    HLA-DPA1    -0.218791  0.126822            GBM\n",
       "2705    HLA-DRB1    -0.209905  0.146458            GBM\n",
       "12448   HLA-DPB1    -0.222952  0.109021         Kidney\n",
       "12942   HLA-DPA1    -0.199014  0.164162         Kidney\n",
       "13489    HLA-DRA    -0.176880  0.229302         Kidney\n",
       "15164    HLA-DMA     0.123195  0.443495         Kidney\n",
       "18445   HLA-DRB1    -0.044398  0.827363         Kidney\n",
       "18648    HLA-DMB    -0.039792  0.847542         Kidney\n",
       "24454   HLA-DRB1     0.146296  0.495865        Ovarian\n",
       "25596    HLA-DRA    -0.115924  0.605777        Ovarian\n",
       "28370   HLA-DPA1    -0.052360  0.837835        Ovarian\n",
       "28694   HLA-DPB1     0.044980  0.863132        Ovarian\n",
       "29864    HLA-DMB     0.019519  0.947539        Ovarian\n",
       "30661    HLA-DMA    -0.002345  0.995827        Ovarian\n",
       "34370   HLA-DRB1     0.196824  0.098855         Breast\n",
       "35195   HLA-DPA1     0.166702  0.172019         Breast\n",
       "36631    HLA-DMA     0.134573  0.356425         Breast\n",
       "37685    HLA-DMB     0.088111  0.511556         Breast\n",
       "38265   HLA-DPB1     0.072042  0.600532         Breast\n",
       "38956    HLA-DRA     0.052811  0.711526         Breast\n",
       "41990    HLA-DRA     0.270801  0.040933           Luad\n",
       "43833   HLA-DPA1     0.184269  0.196235           Luad\n",
       "44037    HLA-DMB     0.177418  0.216594           Luad\n",
       "44401   HLA-DPB1     0.165584  0.255514           Luad\n",
       "45569    HLA-DOA     0.151911  0.383575           Luad\n",
       "45873   HLA-DRB1     0.124300  0.420163           Luad\n",
       "48784    HLA-DOB    -0.066582  0.741646           Luad\n",
       "50193    HLA-DMA    -0.027738  0.889242           Luad\n",
       "52379    HLA-DMB    -0.290547  0.028390  Head and Neck\n",
       "52951    HLA-DMA    -0.252281  0.076947  Head and Neck\n",
       "53389   HLA-DPB1    -0.221519  0.118457  Head and Neck\n",
       "54502   HLA-DRB1    -0.175415  0.242017  Head and Neck\n",
       "55143    HLA-DRA    -0.154935  0.313356  Head and Neck\n",
       "56922   HLA-DPA1    -0.106554  0.524974  Head and Neck\n",
       "63269    HLA-DOB    -0.227253  0.159880           Lscc\n",
       "63784    HLA-DMB    -0.201418  0.232838           Lscc\n",
       "64326   HLA-DPB1    -0.179610  0.307518           Lscc\n",
       "65257   HLA-DRB1    -0.151224  0.414930           Lscc\n",
       "66021    HLA-DMA    -0.132824  0.486062           Lscc\n",
       "66928   HLA-DPA1    -0.110666  0.592151           Lscc\n",
       "67687    HLA-DRA    -0.096418  0.648168           Lscc\n",
       "68087    HLA-DOA    -0.147955  0.684254           Lscc\n",
       "74659   HLA-DRB1     0.243673  0.102476          Colon\n",
       "79534    HLA-DMB     0.037077  0.851579          Colon\n",
       "79568   HLA-DPA1     0.036115  0.855265          Colon\n",
       "79983    HLA-DRA    -0.022677  0.910492          Colon\n",
       "80054    HLA-DMA     0.027898  0.917367          Colon\n",
       "80198   HLA-DPB1    -0.015465  0.939902          Colon"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#MHC class2 (CD4) molecules\n",
    "mhc2 = [\"HLA-DPA1\",\"HLA-DPB1\",\"HLA-DMA\",\"HLA-DMB\",\"HLA-DOA\",\"HLA-DOB\",\"HLA-DQ\",\"HLA-DRA\",\"HLA-DRB1\"]\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_all_mhc2 = df_all_prot_append[df_all_prot_append.Comparison.isin(mhc2)]\n",
    "print(\"Number of missing columns\" )\n",
    "print(56 - len(df_all_mhc2))\n",
    "#HLA-DP and HLA-DM missing for all. HLA-DRA and HLA-DRB1 consistently not sig \n",
    "df_all_mhc2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Correlation</th>\n",
       "      <th>P_value</th>\n",
       "      <th>Cancer Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4460</th>\n",
       "      <td>HLA-C</td>\n",
       "      <td>0.149001</td>\n",
       "      <td>0.338227</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4767</th>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.140454</td>\n",
       "      <td>0.371550</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6132</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>-0.155200</td>\n",
       "      <td>0.528789</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8068</th>\n",
       "      <td>HLA-A</td>\n",
       "      <td>-0.058388</td>\n",
       "      <td>0.749978</td>\n",
       "      <td>GBM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13260</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>-0.240222</td>\n",
       "      <td>0.201476</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16995</th>\n",
       "      <td>HLA-A</td>\n",
       "      <td>-0.082140</td>\n",
       "      <td>0.673774</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17871</th>\n",
       "      <td>HLA-B</td>\n",
       "      <td>-0.056599</td>\n",
       "      <td>0.769694</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19497</th>\n",
       "      <td>HLA-C</td>\n",
       "      <td>-0.022882</td>\n",
       "      <td>0.917089</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26082</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>0.108046</td>\n",
       "      <td>0.656611</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26858</th>\n",
       "      <td>HLA-C</td>\n",
       "      <td>0.085650</td>\n",
       "      <td>0.719148</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28553</th>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.048173</td>\n",
       "      <td>0.853008</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28760</th>\n",
       "      <td>HLA-A</td>\n",
       "      <td>0.043865</td>\n",
       "      <td>0.866842</td>\n",
       "      <td>Ovarian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35058</th>\n",
       "      <td>HLA-A</td>\n",
       "      <td>0.178391</td>\n",
       "      <td>0.156305</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36799</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>0.117063</td>\n",
       "      <td>0.379066</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36884</th>\n",
       "      <td>HLA-C</td>\n",
       "      <td>0.111106</td>\n",
       "      <td>0.393245</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40220</th>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.018660</td>\n",
       "      <td>0.903281</td>\n",
       "      <td>Breast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41694</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>-0.292478</td>\n",
       "      <td>0.025816</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44338</th>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.167621</td>\n",
       "      <td>0.248299</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47413</th>\n",
       "      <td>HLA-A</td>\n",
       "      <td>-0.097372</td>\n",
       "      <td>0.587799</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51159</th>\n",
       "      <td>HLA-C</td>\n",
       "      <td>-0.010527</td>\n",
       "      <td>0.963644</td>\n",
       "      <td>Luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52718</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>-0.261069</td>\n",
       "      <td>0.055836</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56637</th>\n",
       "      <td>HLA-A</td>\n",
       "      <td>-0.114655</td>\n",
       "      <td>0.482804</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61086</th>\n",
       "      <td>HLA-C</td>\n",
       "      <td>0.016743</td>\n",
       "      <td>0.941696</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61711</th>\n",
       "      <td>HLA-B</td>\n",
       "      <td>-0.003951</td>\n",
       "      <td>0.990387</td>\n",
       "      <td>Head and Neck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62767</th>\n",
       "      <td>HLA-A</td>\n",
       "      <td>0.260466</td>\n",
       "      <td>0.092815</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62923</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>-0.246814</td>\n",
       "      <td>0.120444</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67880</th>\n",
       "      <td>HLA-B</td>\n",
       "      <td>0.092719</td>\n",
       "      <td>0.664397</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73161</th>\n",
       "      <td>HLA-C</td>\n",
       "      <td>-0.005874</td>\n",
       "      <td>0.983589</td>\n",
       "      <td>Lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74959</th>\n",
       "      <td>HLA-A</td>\n",
       "      <td>-0.223933</td>\n",
       "      <td>0.137092</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77570</th>\n",
       "      <td>HLA-B</td>\n",
       "      <td>-0.105428</td>\n",
       "      <td>0.536007</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78243</th>\n",
       "      <td>CD8A</td>\n",
       "      <td>0.116971</td>\n",
       "      <td>0.647257</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79644</th>\n",
       "      <td>HLA-C</td>\n",
       "      <td>0.033580</td>\n",
       "      <td>0.866157</td>\n",
       "      <td>Colon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Comparison  Correlation   P_value    Cancer Type\n",
       "4460       HLA-C     0.149001  0.338227            GBM\n",
       "4767       HLA-B     0.140454  0.371550            GBM\n",
       "6132        CD8A    -0.155200  0.528789            GBM\n",
       "8068       HLA-A    -0.058388  0.749978            GBM\n",
       "13260       CD8A    -0.240222  0.201476         Kidney\n",
       "16995      HLA-A    -0.082140  0.673774         Kidney\n",
       "17871      HLA-B    -0.056599  0.769694         Kidney\n",
       "19497      HLA-C    -0.022882  0.917089         Kidney\n",
       "26082       CD8A     0.108046  0.656611        Ovarian\n",
       "26858      HLA-C     0.085650  0.719148        Ovarian\n",
       "28553      HLA-B     0.048173  0.853008        Ovarian\n",
       "28760      HLA-A     0.043865  0.866842        Ovarian\n",
       "35058      HLA-A     0.178391  0.156305         Breast\n",
       "36799       CD8A     0.117063  0.379066         Breast\n",
       "36884      HLA-C     0.111106  0.393245         Breast\n",
       "40220      HLA-B     0.018660  0.903281         Breast\n",
       "41694       CD8A    -0.292478  0.025816           Luad\n",
       "44338      HLA-B     0.167621  0.248299           Luad\n",
       "47413      HLA-A    -0.097372  0.587799           Luad\n",
       "51159      HLA-C    -0.010527  0.963644           Luad\n",
       "52718       CD8A    -0.261069  0.055836  Head and Neck\n",
       "56637      HLA-A    -0.114655  0.482804  Head and Neck\n",
       "61086      HLA-C     0.016743  0.941696  Head and Neck\n",
       "61711      HLA-B    -0.003951  0.990387  Head and Neck\n",
       "62767      HLA-A     0.260466  0.092815           Lscc\n",
       "62923       CD8A    -0.246814  0.120444           Lscc\n",
       "67880      HLA-B     0.092719  0.664397           Lscc\n",
       "73161      HLA-C    -0.005874  0.983589           Lscc\n",
       "74959      HLA-A    -0.223933  0.137092          Colon\n",
       "77570      HLA-B    -0.105428  0.536007          Colon\n",
       "78243       CD8A     0.116971  0.647257          Colon\n",
       "79644      HLA-C     0.033580  0.866157          Colon"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#CD8 and MHC class 1 molecules\n",
    "mhc1 = [\"CD8A\",\"HLA-A\",\"HLA-B\",\"HLA-C\"]\n",
    "\n",
    "df_all_mhc1 = df_all_prot_append[df_all_prot_append.Comparison.isin(mhc1)]\n",
    "df_all_mhc1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Remove Proteomics labeling \n",
    "df_all_prot_append = df_all_prot_append.replace(to_replace ='_proteomics', value = '', regex = True) \n",
    "\n",
    "#subset dataframe to include genes only desired for figure \n",
    "df_all_comp_coag = df_all_prot_append[df_all_prot_append.Comparison.isin(coag_upa)]\n",
    "print(\"Number of rows in data frame \" + str(len(df_all_comp_coag)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Our Figure includes 26 genes for 8 cancers. If all data was present there would be 208 rows. However, the data frame only has 205 rows. 3 genes are missing. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_missing_genes(test_list, full_list):\n",
    "    for gene in full_list:\n",
    "        if (gene not in test_list):\n",
    "            print(gene)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get list of genes for colon, kidney, and ovarian\n",
    "colon = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Colon\"]\n",
    "colon_list = colon.Comparison.to_list()\n",
    "\n",
    "Kidney = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Kidney\"]\n",
    "Kidney_list = Kidney.Comparison.to_list()\n",
    "\n",
    "Ovarian = df_all_prot_append[df_all_prot_append[\"Cancer Type\"] == \"Ovarian\"]\n",
    "Ovarian_list = Ovarian.Comparison.to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Show the 3 missing genes \n",
    "print(\"Ovarian missing genes: \")\n",
    "find_missing_genes(Ovarian_list, coag_upa)\n",
    "print(\"Kidney missing genes: \")\n",
    "find_missing_genes(Kidney_list, coag_upa)\n",
    "print(\"Colon missing genes: \")\n",
    "find_missing_genes(colon_list, coag_upa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
